<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */
</style>

<style>
  div.csl-bib-body { }
  div.csl-entry {
    clear: both;
    }
  .hanging div.csl-entry {
    margin-left:2em;
    text-indent:-2em;
  }
  div.csl-left-margin {
    min-width:2em;
    float:left;
  }
  div.csl-right-inline {
    margin-left:2em;
    padding-left:1em;
  }
  div.csl-indent {
    margin-left: 2em;
  }
</style>

  <!--radix_placeholder_meta_tags-->
  <title>: 5. Alpha Diversity Estimates</title>

  <meta property="description" itemprop="description" content="Reproducible workflow to assess alpha diversity across temperature treatments using Hill numbers."/>

  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>


  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content=": 5. Alpha Diversity Estimates"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="Reproducible workflow to assess alpha diversity across temperature treatments using Hill numbers."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content=""/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content=": 5. Alpha Diversity Estimates"/>
  <meta property="twitter:description" content="Reproducible workflow to assess alpha diversity across temperature treatments using Hill numbers."/>

  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Hilldiv: An r package for the integral analysis of diversity based on hill numbers;citation_publication_date=2019;citation_author=Antton Alberdi;citation_author=M Thomas P Gilbert"/>
  <meta name="citation_reference" content="citation_title=A guide to the application of hill numbers to DNA-based diversity analyses;citation_publication_date=2019;citation_volume=19;citation_author=Antton Alberdi;citation_author=M Thomas P Gilbert"/>
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","bibliography","output"]}},"value":[{"type":"character","attributes":{},"value":["5. Alpha Diversity Estimates"]},{"type":"character","attributes":{},"value":["Reproducible workflow to assess alpha diversity across temperature treatments using Hill numbers.\n"]},{"type":"character","attributes":{},"value":["assets/cite.bib"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["distill::distill_article"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["css","toc","toc_depth"]}},"value":[{"type":"character","attributes":{},"value":["assets/styles.css"]},{"type":"logical","attributes":{},"value":[true]},{"type":"integer","attributes":{},"value":[3]}]}]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }

  </style>

  <script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <script src="site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
  <script src="site_libs/fuse-6.4.1/fuse.min.js"></script>

  <script type="application/javascript">

  function getMeta(metaName) {
    var metas = document.getElementsByTagName('meta');
    for (let i = 0; i < metas.length; i++) {
      if (metas[i].getAttribute('name') === metaName) {
        return metas[i].getAttribute('content');
      }
    }
    return '';
  }

  function offsetURL(url) {
    var offset = getMeta('distill:offset');
    return offset ? offset + '/' + url : url;
  }

  function createFuseIndex() {

    // create fuse index
    var options = {
      keys: [
        { name: 'title', weight: 20 },
        { name: 'categories', weight: 15 },
        { name: 'description', weight: 10 },
        { name: 'contents', weight: 5 },
      ],
      ignoreLocation: true,
      threshold: 0
    };
    var fuse = new window.Fuse([], options);

    // fetch the main search.json
    return fetch(offsetURL('search.json'))
      .then(function(response) {
        if (response.status == 200) {
          return response.json().then(function(json) {
            // index main articles
            json.articles.forEach(function(article) {
              fuse.add(article);
            });
            // download collections and index their articles
            return Promise.all(json.collections.map(function(collection) {
              return fetch(offsetURL(collection)).then(function(response) {
                if (response.status === 200) {
                  return response.json().then(function(articles) {
                    articles.forEach(function(article) {
                      fuse.add(article);
                    });
                  })
                } else {
                  return Promise.reject(
                    new Error('Unexpected status from search index request: ' +
                              response.status)
                  );
                }
              });
            })).then(function() {
              return fuse;
            });
          });

        } else {
          return Promise.reject(
            new Error('Unexpected status from search index request: ' +
                        response.status)
          );
        }
      });
  }

  window.document.addEventListener("DOMContentLoaded", function (event) {

    // get search element (bail if we don't have one)
    var searchEl = window.document.getElementById('distill-search');
    if (!searchEl)
      return;

    createFuseIndex()
      .then(function(fuse) {

        // make search box visible
        searchEl.classList.remove('hidden');

        // initialize autocomplete
        var options = {
          autoselect: true,
          hint: false,
          minLength: 2,
        };
        window.autocomplete(searchEl, options, [{
          source: function(query, callback) {
            const searchOptions = {
              isCaseSensitive: false,
              shouldSort: true,
              minMatchCharLength: 2,
              limit: 10,
            };
            var results = fuse.search(query, searchOptions);
            callback(results
              .map(function(result) { return result.item; })
            );
          },
          templates: {
            suggestion: function(suggestion) {
              var img = suggestion.preview && Object.keys(suggestion.preview).length > 0
                ? `<img src="${offsetURL(suggestion.preview)}"</img>`
                : '';
              var html = `
                <div class="search-item">
                  <h3>${suggestion.title}</h3>
                  <div class="search-item-description">
                    ${suggestion.description || ''}
                  </div>
                  <div class="search-item-preview">
                    ${img}
                  </div>
                </div>
              `;
              return html;
            }
          }
        }]).on('autocomplete:selected', function(event, suggestion) {
          window.location.href = offsetURL(suggestion.path);
        });
        // remove inline display style on autocompleter (we want to
        // manage responsive display via css)
        $('.algolia-autocomplete').css("display", "");
      })
      .catch(function(error) {
        console.log(error);
      });

  });

  </script>

  <style type="text/css">

  .nav-search {
    font-size: x-small;
  }

  /* Algolioa Autocomplete */

  .algolia-autocomplete {
    display: inline-block;
    margin-left: 10px;
    vertical-align: sub;
    background-color: white;
    color: black;
    padding: 6px;
    padding-top: 8px;
    padding-bottom: 0;
    border-radius: 6px;
    border: 1px #0F2E3D solid;
    width: 180px;
  }


  @media screen and (max-width: 768px) {
    .distill-site-nav .algolia-autocomplete {
      display: none;
      visibility: hidden;
    }
    .distill-site-nav.responsive .algolia-autocomplete {
      display: inline-block;
      visibility: visible;
    }
    .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
      margin-left: 0;
      width: 400px;
      max-height: 400px;
    }
  }

  .algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
    width: 90%;
    outline: none;
    border: none;
  }

  .algolia-autocomplete .aa-hint {
    color: #999;
  }
  .algolia-autocomplete .aa-dropdown-menu {
    width: 550px;
    max-height: 70vh;
    overflow-x: visible;
    overflow-y: scroll;
    padding: 5px;
    margin-top: 3px;
    margin-left: -150px;
    background-color: #fff;
    border-radius: 5px;
    border: 1px solid #999;
    border-top: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
    cursor: pointer;
    padding: 5px 4px;
    border-bottom: 1px solid #eee;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
    border-bottom: none;
    margin-bottom: 2px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
    overflow: hidden;
    font-size: 0.8em;
    line-height: 1.4em;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
    font-size: 1rem;
    margin-block-start: 0;
    margin-block-end: 5px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
    display: inline-block;
    overflow: hidden;
    height: 2.8em;
    width: 80%;
    margin-right: 4%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
    display: inline-block;
    width: 15%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
    height: 3em;
    width: auto;
    display: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
    display: initial;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
    background-color: #eee;
  }
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
    font-weight: bold;
    font-style: normal;
  }

  </style>


  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure img {
    width: 100%;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refs = $(this).attr('data-cites').split(" ");
        var refHtml = refs.map(function(ref) {
          return "<p>" + $('#ref-' + ref).html() + "</p>";
        }).join("\n");
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
  <script src="site_libs/clipboard-2.0.6/clipboard.min.js"></script>
  <link href="site_libs/xaringanExtra-clipboard-0.2.6/xaringanExtra-clipboard.css" rel="stylesheet" />
  <script src="site_libs/xaringanExtra-clipboard-0.2.6/xaringanExtra-clipboard.js"></script>
  <script>window.xaringanExtraClipboard(null, {"button":"<i class=\"fa fa-clone fa-2x\" style=\"color: #004276\"><\/i>","success":"<i class=\"fa fa-check fa-2x\" style=\"color: #90BE6D\"><\/i>","error":"<i class=\"fa fa-times fa-2x\" style=\"color: #F94144\"><\/i>"})</script>
  <script src="site_libs/font-awesome-5.13.0/js/script.js"></script>
  <link href="site_libs/panelset-0.2.6/panelset.css" rel="stylesheet" />
  <script src="site_libs/panelset-0.2.6/panelset.js"></script>
  <script src="site_libs/core-js-2.5.3/shim.min.js"></script>
  <script src="site_libs/react-17.0.0/react.min.js"></script>
  <script src="site_libs/react-17.0.0/react-dom.min.js"></script>
  <script src="site_libs/reactwidget-1.0.0/react-tools.js"></script>
  <script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
  <script src="site_libs/reactable-binding-0.2.3.9000/reactable.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Inconsolata|Source+Sans+Pro|Work+Sans:200,300,400,500|Karla:400,400i,700,700i');
  /********************************/
  /* controls general properties of site*/
  /********************************/

  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    background-color: #FEFEFE;
  }

  d-title h1{
  font-size: 2.1em;
  font-family: 'Work Sans';
  font-weight: 300;
  }

  d-article h1 {
    font-size: 1.8em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 300;
  }

  d-article h2 {
    font-size: 1.65em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 500;
  }

  d-article h3 {
    margin-top: 1em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
  }

  d-byline .authors-affiliations {
    grid-column-end: span 3;
    grid-column-gap: 8px;
  }

  img.orcid-id {
    margin: 0;
  }

  a.name:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  a.affiliation:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  blockquote p {
  	font-size: 1em;
  	font-family: 'Work Sans';
    font-style: normal;
    margin: 0.5em 0 0.5em;
  }

  d-article li {
    margin: 0px 0px 4px;
    font-size: 18px;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
  }

  /********************************/
  /* controls CODE*/
  /********************************/

  /** pre.sourceCode controls code block without lang **/

  /** Controls r code  **/

  pre.sourceCode.r {
    margin-top: 0px;
     color: #333333;
     background-color: #E8E8E8;
     border-radius: 3px;
     font-size: 0.80em;
     font-family: 'Monaco', 'PT Mono', monospace;
     overflow-x: scroll !important;
     border-left: 6px solid #2271B2;
     border-top: 1px solid #B7B7B7;
     border-right: 1px solid #B7B7B7;
     border-bottom: 1px solid #B7B7B7;
     padding: 18px 0px 18px 20px;
   }

   pre.sourceCode.bash {
     margin-top: 0px;
      color: #333333;
      background-color: #E8E8E8;
      border-radius: 3px;
      font-size: 0.80em;
      font-family: 'Monaco', 'PT Mono', monospace;
      overflow-x: scroll !important;
      border-left: 6px solid #B22271;
      border-top: 1px solid #B7B7B7;
      border-right: 1px solid #B7B7B7;
      border-bottom: 1px solid #B7B7B7;
      padding: 18px 0px 18px 20px;
    }
  /** Controls code output **/

  d-article pre {
     margin-top: 0px;
      color: #333333;
      background-color: #FAFAFA;
      font-size: 0.8em;
      border: 1px solid #ccc;
      overflow-x: scroll !important;
      padding:  10px 10px 10px 18px;
      border-radius: 5px;
    }

  /** Controls inline code  **/

  p code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }

  li code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }

  d-article aside code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-family: 'Monaco', 'PT Mono', monospace;
    color: #BF2727;
  }

  /********************************/
  /* controls NAV BAR and HEADERS*/
  /********************************/

  .nav-dropdown-content .nav-dropdown-header {
    text-transform: uppercase;
    color: #686868;
  }

  .nav-dropdown-content a {
    font-family: 'Work Sans'
  }
  .nav-dropdown .nav-dropbtn {
    color: #FEFEFE;
    font-weight: 400;
  }

  .distill-site-header .logo img {
    margin-top: 10px;
    max-height: 30px;
    margin-bottom: auto;
  }

  .distill-site-nav {
      color: #FEFEFE;
      background-color: #113858;
      font-size: 0.95em;
      font-weight: 400;
      min-width: 165px;
      position: absolute;
      font-family: 'Karla';
      letter-spacing: .1em;
      text-transform: uppercase;
  }

  img {
    margin: 5px;
  }

  /** LIGHTBOX MARKUP **/
  .thumbnail {
    max-width: 100%;
    position: relative;
  }

  /** DETAILS fold down**/

  details {
    background-color: #f6f6f6;
    cursor: pointer;
    color: #2d2d2d;
    margin-bottom:1vh;
    padding: 1em;
    border: none;
    text-align: left;
    outline: none;
  }

  .active, details:hover {
    transition: max-height 0.2s ease-out;
    outline: none;

    /* background-color: #acc2c2; */
  }

  details {
    display: flex;
    justify-content: space-between;
    align-content: center;
    border-radius: 4px;
    outline: none;
  }

  details.active {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    outline: none;
  }

  summary::-webkit-details-marker {
    display: none;
    outline: none;
  }

  summary:focus {
    outline: none;
  }

  summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f0fe";
  }

  details[open] summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f146";
  }

  /********************************/
  /* controls downloadthis  */
  /********************************/

  button.btn.btn-default {
    font-size: 0.8em;
    font-family: "PT Sans";
    border-color: #B22271;
    border-width: medium;
    margin-bottom: 15px;
  }

  /********************************/
  /* controls reactable  */
  /********************************/

  div.rt-text-content {
    font-size: 1.2em;
  }

  /********************************/
  /* next and previous buttons at the end of the article */
  /********************************/

  .btnnav {
    border: none;
    background-color: #113858;
    color: #f2f2f2;
    padding: 14px 28px;
    font-size: 16px;
    cursor: pointer;
    display: inline-block;
  }

  .button_round {border-radius: 12px;}

  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"5. Alpha Diversity Estimates","description":"Reproducible workflow to assess alpha diversity across temperature treatments using Hill numbers.","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a class="logo" href="index.html">
<img src="assets/icon.png" alt="Logo"/>
</a>
<a href="index.html" class="title"></a>
</div>
<div class="nav-right">
<a href="index.html">
<i class="fa fa-home" aria-hidden="true"></i>
</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
A. Processing
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">Workflows</span>
<hr/>
<a href="dada2.html">1. DADA2</a>
<a href="data-prep.html">2. Data Set Prep</a>
<a href="filtering.html">3. Filtering</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
B. Diversity
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">Workflows</span>
<hr/>
<a href="taxa.html">4. Taxonomic</a>
<a href="alpha.html">5. Alpha Diversity</a>
<a href="beta.html">6. Beta Diversity</a>
<a href="da.html">7. DA ASVs &amp; Taxa</a>
<a href="metadata.html">8. Metadata</a>
</div>
</div>
<a href="data-availability.html">Data</a>
<a href="som.html">SOM</a>
<a href="https://github.com/sweltr/high-temp/">
<i class="fa fa-github fa-lg" aria-hidden="true"></i>
</a>
<a href="about.html">
<i class="fa fa-question" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>5. Alpha Diversity Estimates</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->
<p><p>Reproducible workflow to assess alpha diversity across temperature
treatments using Hill numbers.</p></p>
</div>


<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#synopsis">Synopsis</a>
<ul>
<li><a href="#workflow-input">Workflow Input</a></li>
</ul></li>
<li><a href="#s-rrna">16s rRNA</a>
<ul>
<li><a href="#calculate-hill-numbers">Calculate Hill Numbers</a></li>
<li><a href="#hill-numbers-summary">Hill Numbers Summary</a>
<ul>
<li><a href="#hill-summary">Hill Summary</a></li>
</ul></li>
<li><a href="#normality-tests">Normality Tests</a></li>
<li><a href="#significance-tests">Significance Tests</a></li>
<li><a href="#summary">Summary</a></li>
<li><a href="#posthoc-analyses">PostHoc Analyses</a></li>
<li><a href="#alpha-diversity-plots">Alpha Diversity Plots</a></li>
</ul></li>
<li><a href="#its">ITS</a>
<ul>
<li><a href="#calculate-hill-numbers-1">Calculate Hill Numbers</a></li>
<li><a href="#hill-numbers-summary-1">Hill Numbers Summary</a>
<ul>
<li><a href="#hill-summary-1">Hill Summary</a></li>
</ul></li>
<li><a href="#normality-tests-1">Normality Tests</a></li>
<li><a href="#significance-tests-1">Significance Tests</a></li>
<li><a href="#summary-1">Summary</a></li>
<li><a href="#posthoc-analyses-1">PostHoc Analyses</a></li>
<li><a href="#alpha-diversity-plots-1">Alpha Diversity Plots</a></li>
</ul></li>
<li><a href="#workflow-output">Workflow Output</a>
<ul>
<li><a href="#source-code">Source Code</a></li>
<li><a href="#data-availability">Data Availability</a></li>
<li><a href="#last-updated-on">Last updated on</a></li>
</ul></li>
</ul>
</nav>
</div>
<details>
<summary>
<strong>Click here</strong> for setup information.
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">echo =</span> <span class="cn">TRUE</span>, <span class="at">eval =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">119</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#library(conflicted)</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq); <span class="fu">packageVersion</span>(<span class="st">&quot;phyloseq&quot;</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings); <span class="fu">packageVersion</span>(<span class="st">&quot;Biostrings&quot;</span>)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(tidyverse, DT, ampvis2, hilldiv, </span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>               microbiome, phytools, phangorn, </span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>               pairwiseAdonis, naniar, </span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>               labdsv, patchwork, agricolae, ggpubr,</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>               reactable, downloadthis, captioner,</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">install =</span> <span class="cn">FALSE</span>, <span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;hack_code/div_test_plot_jjs.R&quot;</span>)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;hack_code/div_test_jjs.R&quot;</span>)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">scipen=</span><span class="dv">999</span>)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_current<span class="sc">$</span><span class="fu">get</span>(<span class="fu">c</span>(</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache&quot;</span>,</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.path&quot;</span>,</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.rebuild&quot;</span>,</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;dependson&quot;</span>,</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;autodep&quot;</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>))</span></code></pre></div>
</div>
</details>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="synopsis">Synopsis</h1>
<p>This workflow contains diversity assessments for the 2018 high
temperature data sets. In order to run the workflow, you either need to
first run the <a href="dada2.html">DADA2 Workflow</a>
<strong>and</strong> the <a href="data-prep.html">Data Preparation
workflow</a> <strong>or</strong> download the files linked below. See
the <a href="data-availability.html">Data Availability</a> page for
complete details.</p>
<h2 id="workflow-input">Workflow Input</h2>
<p>Files needed to run this workflow can be downloaded from
figshare.</p>
<iframe src="https://widgets.figshare.com/articles/14690739/embed?show_title=1" width="568" height="251" allowfullscreen frameborder="0">
</iframe>
<iframe src="https://widgets.figshare.com/articles/14701440/embed?show_title=1" width="568" height="351" allowfullscreen frameborder="0">
</iframe>
<h1 id="s-rrna">16s rRNA</h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>To account for presence of rare sequence variants caused by
sequencing errors or other technical artifacts, we use Hill numbers
<span class="citation" data-cites="alberdi2019guide">(<a
href="#ref-alberdi2019guide" role="doc-biblioref">Alberdi and Gilbert
2019a</a>)</span>. Hill numbers allow the weight put on rare versus
abundant sequence variants to be scaled while providing intuitive
comparisons of diversity levels using effective number of ASVs as a
measuring unit. This approach allows for balancing the over
representation of rare ASVs that might be inflated due to sequencing
errors.</p>
<p>We will then use Shapiro-Wilk tests to test for normalcy and then,
depending on the results, either use parametric ANOVA or non-parametric
Kruskal-Wallis to compare alpha diversity among treatments.</p>
<h2 id="calculate-hill-numbers">Calculate Hill Numbers</h2>
<p>To calculate Hill numbers, we use the R package <code>hilldiv</code>
<span class="citation" data-cites="alberdi2019hilldiv">(<a
href="#ref-alberdi2019hilldiv" role="doc-biblioref">Alberdi and Gilbert
2019b</a>)</span>. We calculate three metrics that put more or less
weight on common species:</p>
<ol type="1">
<li>Observed richness, where <code>q-value = 0</code>.</li>
<li>Shannon exponential, which weighs ASVs by their frequency, where
<code>q-value = 1</code>.</li>
<li>Simpson multiplicative inverse, which over weighs abundant ASVs,
where <code>q-value = 2</code>.</li>
</ol>
<p>We perform each analysis against the Full (unfiltered) data set as
well as the Arbitrary, PERfect and PIME, filtered data sets using the
function <code>hill_div</code>.</p>
<p>The command is as follows:</p>
<pre><code>hill_div(count = x, qvalue = i, tree = ultrametric_tree)</code></pre>
<p>where <code>x</code> is the sample by ASV table, <code>i</code> is
the q-value corresponding to the metric of interest and
<code>tree</code> is an ultrametric formatted phylogenetic tree if you
choose to look at lineage, rather than ASV, diversity.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>We first transform all the data to relative abundance values, and
compute new trees.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>ssu18_alpha_ds <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ssu18_ps_work&quot;</span>, <span class="st">&quot;ssu18_ps_filt&quot;</span>, <span class="st">&quot;ssu18_ps_perfect&quot;</span>, <span class="st">&quot;ssu18_ps_pime&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_alpha_ds) {</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  tmp_ps <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(<span class="fu">get</span>(i), <span class="cf">function</span>(otu) otu<span class="sc">/</span><span class="fu">sum</span>(otu))</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  tmp_ps<span class="sc">@</span>phy_tree <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  tmp_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(tmp_ps), <span class="at">rooted =</span> <span class="cn">TRUE</span>,</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(tmp_ps))</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  tmp_ps_norm <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(tmp_ps, sample_data, tmp_tree)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  tmp_asv <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(tmp_ps_norm)))</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>  tmp_ps_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_norm&quot;</span>))</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_ps_name, tmp_ps_norm)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  tmp_asv_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_tu&quot;</span>))</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_asv_name, tmp_asv)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<p>Next, we run the analysis for all three metrics on the data sets
(without a tree).</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>qvalue <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> qvalue) {</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (j <span class="cf">in</span> ssu18_alpha_ds) {</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>     tmp_asv <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(j, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_tu&quot;</span>)))</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>     tmp_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">hill_div</span>(tmp_asv, <span class="at">qvalue =</span> i))</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>     tmp_df <span class="ot">&lt;-</span> tmp_df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;tmp_name&quot;</span> <span class="ot">=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>                              tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SamName&quot;</span>)</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(j, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_h&quot;</span>, i))</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_name)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_df)</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;_h&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>And make summary tables to add back into each <code>ps</code>
object.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_alpha_ds) {</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>     tmp_obs <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_h0&quot;</span>)))</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>     tmp_sha <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_h1&quot;</span>)))</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>     tmp_sim <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_h2&quot;</span>)))</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>     tmp_hill <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_obs, tmp_sha, <span class="at">by =</span> <span class="st">&quot;SamName&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>       dplyr<span class="sc">::</span><span class="fu">left_join</span>(., tmp_sim, <span class="at">by =</span> <span class="st">&quot;SamName&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>       dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;Observed&quot;</span> <span class="ot">=</span> <span class="dv">2</span>, <span class="st">&quot;Shannon_exp&quot;</span> <span class="ot">=</span> <span class="dv">3</span>, <span class="st">&quot;InvSimpson&quot;</span> <span class="ot">=</span> <span class="dv">4</span>)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_hill&quot;</span>))</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_hill)</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;_hill&quot;</span>)</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span></code></pre></div>
</details>
</div>
<p>And then create the new objects with the diversity data.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_alpha_ds) {</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>     tmp_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(tmp_ps), <span class="at">rooted =</span> <span class="cn">TRUE</span>,</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(tmp_ps))</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(tmp_ps, sample_data, tmp_tree)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>     tmp_hill <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_hill&quot;</span>)))</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>     tmp_hill_samp <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(<span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_ps)),</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>                                         tmp_hill, <span class="at">by =</span> <span class="st">&quot;SamName&quot;</span>)</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>     tmp_hill_samp<span class="sc">$</span>ID <span class="ot">&lt;-</span> tmp_hill_samp<span class="sc">$</span>SamName</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>     tmp_hill_samp <span class="ot">&lt;-</span> tmp_hill_samp <span class="sc">%&gt;%</span>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;ID&quot;</span>)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>     tmp_ps2 <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(<span class="fu">otu_table</span>(tmp_ps),</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">sample_data</span>(tmp_hill_samp),</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">tax_table</span>(tmp_ps),</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">phy_tree</span>(tmp_ps))</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(i, tmp_ps2)</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>     tmp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&quot;files/alpha/rdata/&quot;</span>)</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>     <span class="fu">saveRDS</span>(tmp_ps2, <span class="fu">paste</span>(tmp_path, i, <span class="st">&quot;.rds&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="hill-numbers-summary">Hill Numbers Summary</h2>
<p>Now we summarize the data for each sample against all three metrics.
The table contains the results of ASV diversity estimates from the full
data set and the three filtered data sets.</p>
<p>The suffix <code>_fi</code> indicates metrics for the Arbitrary data
set. The suffix <code>_pe</code> indicates metrics for the PERfect data
set and the suffix <code>_pet</code> indicates the lineage diversity for
the PERfect data set.</p>
<p>The suffix <code>_pi</code> indicates metrics for the PIME data set
and the suffix <code>_pit</code> indicates the lineage diversity for the
PIME data set.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="hill-summary">Hill Summary</h3>
<p><small><strong>(16S rRNA) Table 1 | Hill numbers for samples in the
unfiltered &amp; filtered data sets.</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a onclick="fetch(&#39;data:text/csv;base64,U2FtcGxlX0lELE9ic2VydmVkLE9ic2VydmVkX2ZpLE9ic2VydmVkX3BlLE9ic2VydmVkX3BpLFNoYW5ub25fZXhwLFNoYW5ub25fZXhwX2ZpLFNoYW5ub25fZXhwX3BlLFNoYW5ub25fZXhwX3BpLEludlNpbXBzb24sSW52U2ltcHNvbl9maSxJbnZTaW1wc29uX3BlLEludlNpbXBzb25fcGkKUDAxX0QwMF8wMTBfVzNBLDIzNzEsODMxLDcwNyw1MDIsNzYwLjcsMzg1LjEsMzUzLjQsMjI2LjMsMjAwLjMsMTI4LDEyMy42LDg0LjIKUDAxX0QwMF8wMTBfVzhBLDE2MTAsNjIwLDU2MywzMjIsMjM2LjEsMTI3LjgsMTI2LjUsNTkuMiwyNS41LDE4LjksMTkuMSwxMgpQMDJfRDAwXzAxMF9DMEEsMjI3Miw4ODQsNzM3LDU5MCw3ODEuMSw0MzcuMSwzOTcuNCwyOTEuMiwyNjEuNSwxNzguNSwxNzEuNiwxMjkuMwpQMDNfRDAwXzAxMF9XM0IsMjM1Niw5MjIsNzY0LDU5Niw2ODYuOCwzNzMuNywzMzcuMywyMzUsMTU5LjQsMTA4LjcsMTAzLjgsNzUuOApQMDNfRDAwXzAxMF9XOEIsMTU3NSw0MjIsNDExLDIyNCw1MDEsMTgxLDIxMS45LDkwLjEsMTUzLjMsNjkuMiw4Mi42LDM4LjYKUDA0X0QwMF8wMTBfQzBCLDIwNDAsODAyLDY2Myw1NzAsNzk3LjgsNDM5LjQsMzg4LjQsMzA5LjIsMjcyLDE3OC44LDE2Ni44LDEzMC43ClAwNV9EMDBfMDEwX1czQywxNjYzLDY4MCw1NzQsNDk5LDU3OC44LDMyOS40LDI5OS4yLDI0Ni4xLDIwMS4yLDEzOS41LDEzMi42LDExMQpQMDVfRDAwXzAxMF9XOEMsMjEyNCw1NzIsNjYxLDI4OSwzMDcuMywxMDIuNCwxNDYuOSw0Ny45LDI3LjMsMTQuOSwxOC41LDkuNgpQMDZfRDAwXzAxMF9DMEMsMzA2NSwxMDExLDg1Nyw2NjQsMTA5OC40LDUxOC43LDQ4MS4zLDMyNC43LDMwOS40LDE4MiwxNzkuNywxMjAuNgpQMDdfRDAwXzAxMF9XM0QsMjAxMCw4NDMsNzI1LDUxNiw1OTguOCwzNTMuNCwzMjcuNSwyMDgsMTQ2LjgsMTAzLjcsOTkuNyw2Ny41ClAwN19EMDBfMDEwX1c4RCwyMTU3LDc3MSw2NzEsMjc2LDU2MC44LDI4NC4yLDI3MC4zLDg4LjEsOTEuNSw2MC4yLDYwLjIsMjQuNApQMDhfRDAwXzAxMF9DMEQsMjQ0MSw4MTMsNjgxLDU5NCw5NzMuMyw0NjQuOCw0MTkuOCwzMzEuMywzOTQuMiwyMzAsMjIwLjUsMTY4LjcKUDA5X0QwMF8wMTBfVzNFLDIzODYsODc0LDc1MSw1NjMsNjgwLjEsMzUyLjcsMzM0LjUsMjI0LjQsMTY4LjUsMTEwLjMsMTA5LjQsNzYuNQpQMDlfRDAwXzAxMF9XOEUsODEzLDM3NiwzMzIsMTcwLDM3NS4zLDE2NS42LDE0OS45LDY3LjksMTIwLjIsNTUuNyw1Ni44LDI2LjQKUDEwX0QwMF8wMTBfQzBFLDIwNjYsNjcwLDU3MSw0OTEsODg2LjgsNDE2LjUsMzc5LjQsMjk4LjMsNDI3LjYsMjQyLjgsMjMxLjQsMTc0LjcK&#39;).then(res =&gt; res.blob()).then(blob =&gt; {&#10;      const downloadURL = window.URL.createObjectURL(blob);&#10;      const a = document.createElement(&#39;a&#39;);&#10;      document.body.appendChild(a);&#10;      a.href = downloadURL;&#10;      a.download = &#39;ssu_tab_alpha_div.csv&#39;; a.click();&#10;      window.URL.revokeObjectURL(downloadURL);&#10;      document.body.removeChild(a);&#10;    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-page">
<div id="htmlwidget-a193bdecd0d80589300a" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-a193bdecd0d80589300a">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Sample_ID":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"Observed":[2371,1610,2272,2356,1575,2040,1663,2124,3065,2010,2157,2441,2386,813,2066],"Observed_fi":[831,620,884,922,422,802,680,572,1011,843,771,813,874,376,670],"Observed_pe":[707,563,737,764,411,663,574,661,857,725,671,681,751,332,571],"Observed_pi":[502,322,590,596,224,570,499,289,664,516,276,594,563,170,491],"Shannon_exp":[760.7,236.1,781.1,686.8,501,797.8,578.8,307.3,1098.4,598.8,560.8,973.3,680.1,375.3,886.8],"Shannon_exp_fi":[385.1,127.8,437.1,373.7,181,439.4,329.4,102.4,518.7,353.4,284.2,464.8,352.7,165.6,416.5],"Shannon_exp_pe":[353.4,126.5,397.4,337.3,211.9,388.4,299.2,146.9,481.3,327.5,270.3,419.8,334.5,149.9,379.4],"Shannon_exp_pi":[226.3,59.2,291.2,235,90.1,309.2,246.1,47.9,324.7,208,88.1,331.3,224.4,67.9,298.3],"InvSimpson":[200.3,25.5,261.5,159.4,153.3,272,201.2,27.3,309.4,146.8,91.5,394.2,168.5,120.2,427.6],"InvSimpson_fi":[128,18.9,178.5,108.7,69.2,178.8,139.5,14.9,182,103.7,60.2,230,110.3,55.7,242.8],"InvSimpson_pe":[123.6,19.1,171.6,103.8,82.6,166.8,132.6,18.5,179.7,99.7,60.2,220.5,109.4,56.8,231.4],"InvSimpson_pi":[84.2,12,129.3,75.8,38.6,130.7,111,9.6,120.6,67.5,24.4,168.7,76.5,26.4,174.7]},"columns":[{"accessor":"Sample_ID","name":"Sample_ID","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"header":"Sample ID","minWidth":150,"align":"left","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"5px solid #eee"},"headerStyle":{"borderRight":"5px solid #eee"}},{"accessor":"Observed","name":"FULL","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["2371","1610","2272","2356","1575","2040","1663","2124","3065","2010","2157","2441","2386","813","2066"],"header":"FULL","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Observed_fi","name":"Arbitrary","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["831","620","884","922","422","802","680","572","1011","843","771","813","874","376","670"],"header":"Arbitrary","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Observed_pe","name":"PERfect","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["707","563","737","764","411","663","574","661","857","725","671","681","751","332","571"],"header":"PERfect","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Observed_pi","name":"PIME","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["502","322","590","596","224","570","499","289","664","516","276","594","563","170","491"],"header":"PIME","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"},"style":{"borderRight":"5px solid #eee"},"headerStyle":{"borderRight":"5px solid #eee"}},{"accessor":"Shannon_exp","name":"FULL","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["760.7","236.1","781.1","686.8","501","797.8","578.8","307.3","1098.4","598.8","560.8","973.3","680.1","375.3","886.8"],"header":"FULL","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Shannon_exp_fi","name":"Arbitrary","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["385.1","127.8","437.1","373.7","181","439.4","329.4","102.4","518.7","353.4","284.2","464.8","352.7","165.6","416.5"],"header":"Arbitrary","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Shannon_exp_pe","name":"PERfect","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["353.4","126.5","397.4","337.3","211.9","388.4","299.2","146.9","481.3","327.5","270.3","419.8","334.5","149.9","379.4"],"header":"PERfect","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Shannon_exp_pi","name":"PIME","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["226.3","59.2","291.2","235","90.1","309.2","246.1","47.9","324.7","208","88.1","331.3","224.4","67.9","298.3"],"header":"PIME","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"},"style":{"borderRight":"5px solid #eee"},"headerStyle":{"borderRight":"5px solid #eee"}},{"accessor":"InvSimpson","name":"FULL","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["200.3","25.5","261.5","159.4","153.3","272","201.2","27.3","309.4","146.8","91.5","394.2","168.5","120.2","427.6"],"header":"FULL","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"InvSimpson_fi","name":"Arbitrary","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["128","18.9","178.5","108.7","69.2","178.8","139.5","14.9","182","103.7","60.2","230","110.3","55.7","242.8"],"header":"Arbitrary","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"InvSimpson_pe","name":"PERfect","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["123.6","19.1","171.6","103.8","82.6","166.8","132.6","18.5","179.7","99.7","60.2","220.5","109.4","56.8","231.4"],"header":"PERfect","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"InvSimpson_pi","name":"PIME","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["84.2","12","129.3","75.8","38.6","130.7","111","9.6","120.6","67.5","24.4","168.7","76.5","26.4","174.7"],"header":"PIME","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}}],"columnGroups":[{"name":"Observed richness","headerStyle":{"borderRight":"5px solid #eee","fontSize":"1.1em"},"columns":["Observed","Observed_fi","Observed_pe","Observed_pi"]},{"name":"Shannon exponential","headerStyle":{"fontSize":"1.1em"},"columns":["Shannon_exp","Shannon_exp_fi","Shannon_exp_pe","Shannon_exp_pi"]},{"name":"Simpson multiplicative inverse","headerStyle":{"fontSize":"1.1em"},"columns":["InvSimpson","InvSimpson_fi","InvSimpson_pe","InvSimpson_pi"]}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPagination":true,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"9e0e1d6b4f7db79bfc534bba13d02de4"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
<h2 id="normality-tests">Normality Tests</h2>
<p>Before running significance tests, we need to know if data is
normally distributed, which will tell use whether to use a parametric or
non-parametric test. To test if the data are normally distributed, we
use the Shapiro-Wilk Normality test and the Bartlett Test of Homogeneity
of Variances.</p>
<p>If the p-values are both <strong>not significant</strong> (p &gt;
0.05) from the tests, we accept the null hypothesis (that the results
are normally distributed) and test for significance between samples
using an ANOVA. If the p-values are both <strong>significant</strong> (p
&lt; 0.05), we reject the null hypothesis (that the results are normally
distributed) and test for significance between samples using
Kruskal-Wallis (non-parametric equivalent of ANOVA).</p>
<p>The commands are as follows:</p>
<p><code>shapiro.test(x)</code>, where <code>x</code> is a numeric
vector of alpha diversity values from the sample data table.</p>
<p><code>bartlett.test(Value ~ Group, data = df)</code> Where
<code>Value</code> is the metric of interest, <code>Group</code> in the
treatment to compare, and <code>df</code> is the data frame.</p>
<p>First the Shapiro-Wilk Normality test.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>ssu_tab_alpha_div</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>ssu18_div_tab <span class="ot">&lt;-</span> ssu_tab_alpha_div</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>ssu18_shap_tests_asv <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">colnames</span>(ssu18_div_tab[,<span class="dv">7</span><span class="sc">:</span><span class="fu">ncol</span>(ssu18_div_tab)])) {</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>   tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">&quot;ssu18_shap_asv_&quot;</span>, .))</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>   ssu18_shap_tests_asv <span class="ot">&lt;-</span> <span class="fu">append</span>(ssu18_shap_tests_asv, tmp_name)</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>   tmp_test <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">shapiro.test</span>(ssu18_div_tab[[i]]))</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>   tmp_test<span class="sc">$</span>data.name <span class="ot">&lt;-</span> tmp_name</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>   <span class="fu">assign</span>(tmp_name, tmp_test)</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>   <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;shap_&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>And then the Bartlett Test of Homogeneity of Variances.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>ssu18_div_tab_asv <span class="ot">&lt;-</span> ssu_tab_alpha_div</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>ssu18_bart_tests_asv <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">colnames</span>(ssu18_div_tab_asv[,<span class="dv">7</span><span class="sc">:</span><span class="fu">ncol</span>(ssu18_div_tab_asv)])) {</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>   tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">&quot;ssu18_bart_asv_&quot;</span>, .))</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>   ssu18_bart_tests_asv <span class="ot">&lt;-</span> <span class="fu">append</span>(ssu18_bart_tests_asv, tmp_name)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>   tmp_test <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">bartlett.test</span>(ssu18_div_tab_asv[[i]] <span class="sc">~</span> TEMP, <span class="at">data =</span> ssu18_div_tab_asv))</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>   tmp_test<span class="sc">$</span>data.name <span class="ot">&lt;-</span> tmp_name</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>   <span class="fu">assign</span>(tmp_name, tmp_test)</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>   <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;_bart_&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>Here we see which Shapiro-Wilk Normality <strong>and</strong>
Bartlett tests were significant and which were not. So wherever the
value of both p-values in <code>&gt; 0.05</code> we can use an ANOVA,
otherwise we use Kruskal-Wallis.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><small><strong>(16S rRNA) Table 2 | Summary of Normality Tests for
unfiltered &amp; filtered data sets. P-values in red indicate
significance (p-value &lt; 0.05).</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a onclick="fetch(&#39;data:text/csv;base64,bWV0cmljLGRhdGFzZXQscC12YWx1ZSAobm9ybWFsaXR5KSxwLXZhbHVlIChob21vZ2VuZWl0eSksbWV0aG9kCk9ic2VydmVkLEZVTEwsMC4yNjgyLDAuNTk5MixBTk9WQQpPYnNlcnZlZCxGSUxULDAuMzY3MSwwLjU4NTEsQU5PVkEKT2JzZXJ2ZWQsUEVSZmVjdCwwLjE5MDUsMC40MzcsQU5PVkEKT2JzZXJ2ZWQsUElNRSwwLjA1NTIsMC43NTUxLEFOT1ZBClNoYW5ub24sRlVMTCwwLjk5NCwwLjQ5MDUsQU5PVkEKU2hhbm5vbixGSUxULDAuMjI5MywwLjEwNjQsQU5PVkEKU2hhbm5vbixQRVJmZWN0LDAuMzMxMSwwLjE1OTIsQU5PVkEKU2hhbm5vbixQSU1FLDAuMDM2NSwwLjg4MTIsS3J1c2thbC1XYWxsaXMKSW52U2ltcHNvbixGVUxMLDAuNTg0NCwwLjE1NDQsQU5PVkEKSW52U2ltcHNvbixGSUxULDAuNjczMSwwLjQxMTYsQU5PVkEKSW52U2ltcHNvbixQRVJmZWN0LDAuNzQ2NiwwLjM0OTcsQU5PVkEKSW52U2ltcHNvbixQSU1FLDAuMzcwOCwwLjM3MjcsQU5PVkEK&#39;).then(res =&gt; res.blob()).then(blob =&gt; {&#10;      const downloadURL = window.URL.createObjectURL(blob);&#10;      const a = document.createElement(&#39;a&#39;);&#10;      document.body.appendChild(a);&#10;      a.href = downloadURL;&#10;      a.download = &#39;ssu_norm_tests.csv&#39;; a.click();&#10;      window.URL.revokeObjectURL(downloadURL);&#10;      document.body.removeChild(a);&#10;    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-9ecb46afe5020008df84" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-9ecb46afe5020008df84">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"metric":["Observed","Observed","Observed","Observed","Shannon","Shannon","Shannon","Shannon","InvSimpson","InvSimpson","InvSimpson","InvSimpson"],"dataset":["FULL","FILT","PERfect","PIME","FULL","FILT","PERfect","PIME","FULL","FILT","PERfect","PIME"],"p-value (normality)":[0.2682,0.3671,0.1905,0.0552,0.994,0.2293,0.3311,0.0365,0.5844,0.6731,0.7466,0.3708],"p-value (homogeneity)":[0.5992,0.5851,0.437,0.7551,0.4905,0.1064,0.1592,0.8812,0.1544,0.4116,0.3497,0.3727],"method":["ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","Kruskal-Wallis","ANOVA","ANOVA","ANOVA","ANOVA"]},"columns":[{"accessor":"metric","name":"Metric","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["Observed","Observed","Observed","Observed","Shannon","Shannon","Shannon","Shannon","InvSimpson","InvSimpson","InvSimpson","InvSimpson"],"header":"Metric","minWidth":100,"align":"left","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"dataset","name":"dataset","type":"character","sortable":true,"resizable":false,"filterable":false,"cell":["FULL","FILT","PERfect","PIME","FULL","FILT","PERfect","PIME","FULL","FILT","PERfect","PIME"],"header":"dataset","minWidth":80,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"p-value (normality)","name":"p-value (normality)","type":"numeric","sortable":true,"resizable":false,"filterable":false,"cell":["0.2682","0.3671","0.1905","0.0552","0.994","0.2293","0.3311","0.0365","0.5844","0.6731","0.7466","0.3708"],"header":"p-value (normality)","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"p-value (homogeneity)","name":"p-value (homogeneity)","type":"numeric","sortable":true,"resizable":false,"filterable":false,"cell":["0.5992","0.5851","0.437","0.7551","0.4905","0.1064","0.1592","0.8812","0.1544","0.4116","0.3497","0.3727"],"header":"p-value (homogeneity)","minWidth":130,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"method","name":"method","type":"character","sortable":true,"resizable":false,"filterable":false,"cell":["ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","Kruskal-Wallis","ANOVA","ANOVA","ANOVA","ANOVA"],"header":"method","minWidth":120,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"}}],"defaultPageSize":12,"showPageSizeOptions":true,"pageSizeOptions":[5,10,12],"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"88811c9ba2d03ce860741156afc68a50"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
<h2 id="significance-tests">Significance Tests</h2>
<p>To begin, we need to create a hierarchy variable; a two-column matrix
specifying the relationship between samples (first column) and groups
(second column).</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>ssu18_hill_hier <span class="ot">&lt;-</span> ssu_tab_alpha_div</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>ssu18_hill_hier <span class="ot">&lt;-</span> ssu18_hill_hier <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="st">&quot;Sample_ID&quot;</span>, <span class="st">&quot;TEMP&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">remove_rownames</span>()</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>ssu18_hill_hier <span class="ot">&lt;-</span> ssu18_hill_hier[<span class="fu">order</span>(ssu18_hill_hier<span class="sc">$</span>TEMP), ]</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>ssu18_hill_hier<span class="sc">$</span>TEMP <span class="ot">=</span> <span class="fu">paste</span>(ssu18_hill_hier<span class="sc">$</span>TEMP, <span class="st">&#39;C&#39;</span>, <span class="at">sep=</span><span class="st">&#39;&#39;</span>)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>ssu18_hill_hier <span class="ot">&lt;-</span> ssu18_hill_hier <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">remove_rownames</span>()</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu18_hill_hier, <span class="st">&quot;files/alpha/rdata/ssu18_hill_hier.rds&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>Again, we start by testing significance of ASV diversity for the data
sets against each of the three metrics using the <code>div_test</code>
function.</p>
<p>The command is as follows:</p>
<pre><code>div_test(countable = x, qvalue = i, hierarchy = hier, 
         tree = ultrametric_tree, posthoc = TRUE)</code></pre>
<p>where <code>x</code> is ASV by sample table, <code>i</code> is the
q-value corresponding to the metric of interest, <code>hier</code> is
the hierarchy matrix, <code>tree</code> is an ultrametric formatted
phylogenetic tree (if you choose to explore lineage diversity), and
<code>posthoc</code> indicates whether to run post hoc pairwise
analyses.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>qvalue <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_alpha_ds) {</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>     <span class="cf">for</span> (j <span class="cf">in</span> qvalue) {</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>         tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_tu&quot;</span>)))</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>         tmp_test <span class="ot">&lt;-</span> <span class="fu">div_test</span>(tmp_get, <span class="at">qvalue =</span> j,</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">hierarchy =</span> ssu18_hill_hier,</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">posthoc =</span> <span class="cn">TRUE</span>)</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>         tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_q&quot;</span>, j, <span class="st">&quot;_adt&quot;</span>))</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>         <span class="fu">print</span>(tmp_name)</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>         <span class="fu">assign</span>(tmp_name, tmp_test)</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>         <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a> }</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="summary">Summary</h2>
<p><small><strong>(16S rRNA) Table 3 | Summary of significance tests for
unfiltered &amp; filtered data sets. P-values in red for post hoc
analysis indicate significance (p-value &lt; 0.05).</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a onclick="fetch(&#39;data:text/csv;base64,ZGF0YXNldCxtZXRyaWMscS12YWx1ZSxub3JtYWxpdHkgcC12YWx1ZSxob21vZ2VuZWl0eSBwLXZhbHVlLG1ldGhvZCxwb3N0aG9jIG1ldGhvZCxwb3N0aG9jIHAtdmFsdWUKRlVMTCxPYnNlcnZlZCwwLDAuMjY4MiwwLjU5OTIsQU5PVkEsVHVrZXkgcG9zdC1ob2MgdGVzdCwwLjA2MDUwNzM5OTMwNDE4ODEzNgpGVUxMLFNoYW5ub24gZXhwb25lbnRpYWwsMSwwLjk5NCwwLjQ5MDQsQU5PVkEsVHVrZXkgcG9zdC1ob2MgdGVzdCw2LjI3MTQ1MjExODg2MTMyNWUtNQpGVUxMLEludmVyc2UgU2ltcHNvbiwyLDAuNTg0LDAuMTU0NixBTk9WQSxUdWtleSBwb3N0LWhvYyB0ZXN0LDQuNjc0MjQzNzcyODg5NzMyNmUtNQpGSUxULE9ic2VydmVkLDAsMC4zNjcxLDAuNTg1MSxBTk9WQSxUdWtleSBwb3N0LWhvYyB0ZXN0LDAuMDA2MDc5NzkzMjU1NzQxMzQ1CkZJTFQsU2hhbm5vbiBleHBvbmVudGlhbCwxLDAuMjI5OSwwLjEwNjUsQU5PVkEsVHVrZXkgcG9zdC1ob2MgdGVzdCwyLjYxNjE5MjIyNjY5MTUwOGUtNgpGSUxULEludmVyc2UgU2ltcHNvbiwyLDAuNjczMywwLjQxMyxBTk9WQSxUdWtleSBwb3N0LWhvYyB0ZXN0LDEuMzAzNTc5ODA2MzcwMDc2OWUtNgpQSU1FLE9ic2VydmVkLDAsMC4wNTUyLDAuNzU1MSxBTk9WQSxUdWtleSBwb3N0LWhvYyB0ZXN0LDEuNDQxNTg1NTYzMTM3MDQxNGUtNgpQSU1FLFNoYW5ub24gZXhwb25lbnRpYWwsMSwwLjAzNjYsMC44OCxLcnVza2FsLVdhbGxpcyBUZXN0LER1bm4gdGVzdCB3aXRoIEJlbmphbWluaS1Ib2NoYmVyZyBjb3JyZWN0aW9uLDAuMDAxOTMwNDU0MTM2MjI3NzA5MwpQSU1FLEludmVyc2UgU2ltcHNvbiwyLDAuMzcwMSwwLjM3MTQsQU5PVkEsVHVrZXkgcG9zdC1ob2MgdGVzdCw5Ljg3MTU4MjcxNDAwNjcxM2UtNwpQRVJmZWN0LE9ic2VydmVkLDAsMC4xOTA1LDAuNDM3LEFOT1ZBLFR1a2V5IHBvc3QtaG9jIHRlc3QsMC4wNTA0NTQyNjg1OTE2NTQyNgpQRVJmZWN0LFNoYW5ub24gZXhwb25lbnRpYWwsMSwwLjMzMTEsMC4xNTkyLEFOT1ZBLFR1a2V5IHBvc3QtaG9jIHRlc3QsNy4xMzYwNzE2NDE2MTgyNjNlLTYKUEVSZmVjdCxJbnZlcnNlIFNpbXBzb24sMiwwLjc0NjgsMC4zNDg0LEFOT1ZBLFR1a2V5IHBvc3QtaG9jIHRlc3QsMy4xOTcxOTEwMzEzNzIyNjQ1ZS02Cg==&#39;).then(res =&gt; res.blob()).then(blob =&gt; {&#10;      const downloadURL = window.URL.createObjectURL(blob);&#10;      const a = document.createElement(&#39;a&#39;);&#10;      document.body.appendChild(a);&#10;      a.href = downloadURL;&#10;      a.download = &#39;ssu_sign_tests.csv&#39;; a.click();&#10;      window.URL.revokeObjectURL(downloadURL);&#10;      document.body.removeChild(a);&#10;    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-page">
<div id="htmlwidget-ef7caf86f93c66248b97" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ef7caf86f93c66248b97">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"dataset":["FULL","FULL","FULL","FILT","FILT","FILT","PIME","PIME","PIME","PERfect","PERfect","PERfect"],"metric":["Observed","Shannon exponential","Inverse Simpson","Observed","Shannon exponential","Inverse Simpson","Observed","Shannon exponential","Inverse Simpson","Observed","Shannon exponential","Inverse Simpson"],"q-value":["0","1","2","0","1","2","0","1","2","0","1","2"],"normality p-value":[0.2682,0.994,0.584,0.3671,0.2299,0.6733,0.0552,0.0366,0.3701,0.1905,0.3311,0.7468],"homogeneity p-value":[0.5992,0.4904,0.1546,0.5851,0.1065,0.413,0.7551,0.88,0.3714,0.437,0.1592,0.3484],"method":["ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","Kruskal-Wallis Test","ANOVA","ANOVA","ANOVA","ANOVA"],"posthoc method":["Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Dunn test with Benjamini-Hochberg correction","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test"],"posthoc p-value":[0.0605073993041881,6.27145211886132e-05,4.67424377288973e-05,0.00607979325574134,2.61619222669151e-06,1.30357980637008e-06,1.44158556313704e-06,0.00193045413622771,9.87158271400671e-07,0.0504542685916543,7.13607164161826e-06,3.19719103137226e-06]},"columns":[{"accessor":"dataset","name":"Dataset","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["FULL","FULL","FULL","FILT","FILT","FILT","PIME","PIME","PIME","PERfect","PERfect","PERfect"],"header":"Dataset","minWidth":80,"align":"left","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"metric","name":"metric","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["Observed","Shannon exponential","Inverse Simpson","Observed","Shannon exponential","Inverse Simpson","Observed","Shannon exponential","Inverse Simpson","Observed","Shannon exponential","Inverse Simpson"],"header":"metric","minWidth":140,"align":"left","footerStyle":{"fontWeight":"bold"}},{"accessor":"q-value","name":"q-value","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["0","1","2","0","1","2","0","1","2","0","1","2"],"header":"q-value","minWidth":70,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"normality p-value","name":"p-value (normality)","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["0.2682","0.994","0.584","0.3671","0.2299","0.6733","0.0552","0.0366","0.3701","0.1905","0.3311","0.7468"],"header":"p-value (normality)","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"homogeneity p-value","name":"p-value (homogeneity)","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["0.5992","0.4904","0.1546","0.5851","0.1065","0.413","0.7551","0.88","0.3714","0.437","0.1592","0.3484"],"header":"p-value (homogeneity)","minWidth":130,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"method","name":"method","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","Kruskal-Wallis Test","ANOVA","ANOVA","ANOVA","ANOVA"],"header":"method","minWidth":120,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"posthoc method","name":"posthoc method","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Dunn test with Benjamini-Hochberg correction","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test"],"header":"posthoc method","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"posthoc p-value","name":"posthoc p-value","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["6.05e-02","6.27e-05","4.67e-05","6.08e-03","2.62e-06","1.3e-06","1.44e-06","1.93e-03","9.87e-07","5.05e-02","7.14e-06","3.2e-06"],"header":"posthoc p-value","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}]}],"defaultPageSize":12,"showPageSizeOptions":true,"pageSizeOptions":[5,10,12],"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"nowrap":true,"theme":{"style":{"fontSize":"0.7em"}},"dataKey":"690f29c658c374e839a7e63b38ea970a"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
<h2 id="posthoc-analyses">PostHoc Analyses</h2>
<p>First lets check the results of each posthoc analysis.</p>
<div class="l-body">
<div class="panelset">
<section id="observed-q-value-0" class="panel">
<h3>Observed (q-value = 0)</h3>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;FULL (Observed)&quot;
[1] &quot;Tukey post-hoc test&quot;
        diff       lwr       upr      p.adj
3C-0C -219.6  -956.946 517.74597 0.71329233
8C-0C -721.0 -1458.346  16.34597 0.05546473
8C-3C -501.4 -1238.746 235.94597 0.20658985</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;FILT (Observed)&quot;
[1] &quot;Tukey post-hoc test&quot;
        diff       lwr      upr      p.adj
3C-0C   -6.0 -221.6963 209.6963 0.99696904
8C-0C -283.8 -499.4963 -68.1037 0.01109212
8C-3C -277.8 -493.4963 -62.1037 0.01267274</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;PIME (Observed)&quot;
[1] &quot;Tukey post-hoc test&quot;
        diff       lwr       upr          p.adj
3C-0C  -46.6 -140.1306   46.9306 0.406733563435
8C-0C -325.6 -419.1306 -232.0694 0.000002206313
8C-3C -279.0 -372.5306 -185.4694 0.000011005557</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;PERfect (Observed)&quot;
[1] &quot;Tukey post-hoc test&quot;
        diff       lwr       upr      p.adj
3C-0C    2.4 -191.8227 196.62267 0.99940103
8C-0C -174.2 -368.4227  20.02267 0.08059254
8C-3C -176.6 -370.8227  17.62267 0.07616681</code></pre>
</div>
</section>
<section id="shannon-exp-q-value-1" class="panel">
<h3>Shannon exp (q-value = 1)</h3>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;FULL (Shannon exponential)&quot;
[1] &quot;Tukey post-hoc test&quot;
           diff       lwr        upr         p.adj
3C-0C -246.4375 -442.9939  -49.88101 0.01492642034
8C-0C -511.4071 -707.9635 -314.85066 0.00004295672
8C-3C -264.9696 -461.5261  -68.41321 0.00950353162</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;FILT (Shannon exponential)&quot;
[1] &quot;Tukey post-hoc test&quot;
           diff       lwr        upr          p.adj
3C-0C  -96.4627 -177.3128  -15.61258 0.019965252769
8C-0C -283.1154 -363.9656 -202.26533 0.000002072881
8C-3C -186.6527 -267.5029 -105.80262 0.000133670064</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;PIME (Shannon exponential)&quot;
[1] &quot;Dunn test with Benjamini-Hochberg correction&quot;
             Z     P.unadj       P.adj
0C-3C 1.767767 0.077099872 0.077099872
0C-8C 3.535534 0.000406952 0.001220856
3C-8C 1.767767 0.077099872 0.115649808</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;PERfect (Shannon exponential)&quot;
[1] &quot;Tukey post-hoc test&quot;
            diff       lwr        upr          p.adj
3C-0C  -82.87871 -155.6134  -10.14407 0.025808833615
8C-0C -232.15045 -304.8851 -159.41581 0.000005488596
8C-3C -149.27174 -222.0064  -76.53710 0.000385348747</code></pre>
</div>
</section>
<section id="inverse-simpson-q-value-2" class="panel">
<h3>Inverse Simpson (q-value = 2)</h3>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;FULL (Inverse Simpson)&quot;
[1] &quot;Tukey post-hoc test&quot;
           diff       lwr        upr         p.adj
3C-0C -157.7331 -251.7760  -63.69027 0.00202540187
8C-0C -249.3938 -343.4367 -155.35097 0.00003566326
8C-3C  -91.6607 -185.7036    2.38217 0.05628905499</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;FILT (Inverse Simpson)&quot;
[1] &quot;Tukey post-hoc test&quot;
            diff       lwr        upr           p.adj
3C-0C  -84.38551 -126.1609  -42.61016 0.0004422940804
8C-0C -158.66732 -200.4427 -116.89197 0.0000008678593
8C-3C  -74.28181 -116.0572  -32.50646 0.0012804120273</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;PIME (Inverse Simpson)&quot;
[1] &quot;Tukey post-hoc test&quot;
            diff        lwr       upr          p.adj
3C-0C  -61.82853  -93.26452 -30.39255 0.000556162049
8C-0C -122.59975 -154.03574 -91.16377 0.000000651479
8C-3C  -60.77122  -92.20720 -29.33524 0.000643758728</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;PERfect (Inverse Simpson)&quot;
[1] &quot;Tukey post-hoc test&quot;
            diff       lwr        upr          p.adj
3C-0C  -80.17581 -122.1927  -38.15894 0.000718157610
8C-0C -146.57155 -188.5884 -104.55467 0.000002158578
8C-3C  -66.39574 -108.4126  -24.37887 0.003172202683</code></pre>
</div>
</section>
</div>
</div>
<p>Now we can plot the results from the posthoc analyses for each metric
and data set using the function <code>div_test_plot_jjs</code>. I
modified the original function (<code>div_test_plot</code>) to control a
little of the formatting.</p>
<p>The command is as follows:</p>
<pre><code>div_test_plot(divtest = x, chart = &quot;type&quot;, colour = col.pal, 
              posthoc = TRUE, threshold = value))</code></pre>
<p>where <code>x</code> is the results from the <code>div_test</code>
function, <code>"type"</code> is chart type (box, jitter, or violin),
<code>colour</code> is is a color palette, <code>posthoc</code>
indicates whether to run posthoc pairwise analyses, and
<code>value</code> is the maximum p-value to show in pairwise posthoc
results. <strong>WARNING</strong> if none of the posthoc results are
below the specified threshold, the function will throw an error.
Therefore, until this is fixed, all posthoc values are shown.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;hack_code/div_test_plot_jjs.R&quot;</span>)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern=</span><span class="st">&quot;_adt_plot&quot;</span>))</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">objects</span>(<span class="at">pattern=</span><span class="st">&quot;_adt&quot;</span>)) {</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_plot&quot;</span>))</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>     tmp_df <span class="ot">&lt;-</span> <span class="fu">div_test_plot_jjs</span>(tmp_get, <span class="at">chart =</span> <span class="st">&quot;box&quot;</span>,</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">colour =</span> swel_col, <span class="at">posthoc =</span> <span class="cn">TRUE</span>)</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>     tmp_df <span class="ot">&lt;-</span> <span class="fu">ggpar</span>(tmp_df, <span class="at">legend =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_name)</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_df)</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_q0_adt_plot</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="alpha-diversity-plots">Alpha Diversity Plots</h2>
<p>Posthoc adjusted p-values given for each pairwise comparison.</p>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/alpha/ssu18_alph_div_plots_asv.png" width="7086" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 1 | Alpha diversity plots. Top row =
Observed; middle row = Shannon exponential; bottom row = Inverse
Simpson. From left to right, Full (unfiltered) data, Arbitrary filtered,
PERfect filtered, PIME filtered.</strong></small></p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="its">ITS</h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>To account for presence of rare sequence variants caused by
sequencing errors or other technical artifacts, we use Hill numbers
<span class="citation" data-cites="alberdi2019guide">(<a
href="#ref-alberdi2019guide" role="doc-biblioref">Alberdi and Gilbert
2019a</a>)</span>. Hill numbers allow the weight put on rare versus
abundant sequence variants to be scaled while providing intuitive
comparisons of diversity levels using effective number of ASVs as a
measuring unit. This approach allows for balancing the over
representation of rare ASVs that might be inflated due to sequencing
errors.</p>
<p>We will then use Shapiro-Wilk tests to test for normalcy and then,
depending on the results, either use parametric ANOVA or non-parametric
Kruskal-Wallis to compare alpha diversity among treatments.</p>
<h2 id="calculate-hill-numbers-1">Calculate Hill Numbers</h2>
<p>To calculate Hill numbers, we use the R package <code>hilldiv</code>
<span class="citation" data-cites="alberdi2019hilldiv">(<a
href="#ref-alberdi2019hilldiv" role="doc-biblioref">Alberdi and Gilbert
2019b</a>)</span>. We calculate three metrics that put more or less
weight on common species:</p>
<ol type="1">
<li>Observed richness, where <code>q-value = 0</code>.</li>
<li>Shannon exponential, which weighs ASVs by their frequency, where
<code>q-value = 1</code>.</li>
<li>Simpson multiplicative inverse, which over weighs abundant ASVs,
where <code>q-value = 2</code>.</li>
</ol>
<p>We perform each analysis against the Full (unfiltered) data set as
well as the Arbitrary, PERfect and PIME, filtered data sets using the
function <code>hill_div</code>.</p>
<p>The command is as follows:</p>
<pre><code>hill_div(count = x, qvalue = i, tree = ultrametric_tree)</code></pre>
<p>where <code>x</code> is the sample by ASV table, <code>i</code> is
the q-value corresponding to the metric of interest and
<code>tree</code> is an ultrametric formatted phylogenetic tree, however
this really doesnt apply for ITS data anyway.</p>
<p>We first transform all the data to relative abundance values, and
compute new trees.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>its18_alpha_ds <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;its18_ps_work&quot;</span>, <span class="st">&quot;its18_ps_filt&quot;</span>, <span class="st">&quot;its18_ps_perfect&quot;</span>, <span class="st">&quot;its18_ps_pime&quot;</span>)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its18_alpha_ds) {</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  tmp_ps <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(<span class="fu">get</span>(i), <span class="cf">function</span>(otu) otu<span class="sc">/</span><span class="fu">sum</span>(otu))</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>  tmp_ps<span class="sc">@</span>phy_tree <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>  tmp_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(tmp_ps), <span class="at">rooted =</span> <span class="cn">TRUE</span>,</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(tmp_ps))</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>  tmp_ps_norm <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(tmp_ps, sample_data, tmp_tree)</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>  tmp_asv <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(tmp_ps_norm)))</span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>  tmp_ps_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_norm&quot;</span>))</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_ps_name, tmp_ps_norm)</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>  tmp_asv_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_tu&quot;</span>))</span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_asv_name, tmp_asv)</span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<p>Next, we run the analysis for all three metrics on the data sets
(without a tree).</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>qvalue <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> qvalue) {</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (j <span class="cf">in</span> its18_alpha_ds) {</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>     tmp_asv <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(j, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_tu&quot;</span>)))</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>     tmp_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">hill_div</span>(tmp_asv, <span class="at">qvalue =</span> i))</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>     tmp_df <span class="ot">&lt;-</span> tmp_df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;tmp_name&quot;</span> <span class="ot">=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>                              tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SamName&quot;</span>)</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(j, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_h&quot;</span>, i))</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_name)</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_df)</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;_h&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>And make summary tables to add back into each <code>ps</code>
object.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its18_alpha_ds) {</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>     tmp_obs <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_h0&quot;</span>)))</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>     tmp_sha <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_h1&quot;</span>)))</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>     tmp_sim <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_h2&quot;</span>)))</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>     tmp_hill <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_obs, tmp_sha, <span class="at">by =</span> <span class="st">&quot;SamName&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>       dplyr<span class="sc">::</span><span class="fu">left_join</span>(., tmp_sim, <span class="at">by =</span> <span class="st">&quot;SamName&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>       dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;Observed&quot;</span> <span class="ot">=</span> <span class="dv">2</span>, <span class="st">&quot;Shannon_exp&quot;</span> <span class="ot">=</span> <span class="dv">3</span>, <span class="st">&quot;InvSimpson&quot;</span> <span class="ot">=</span> <span class="dv">4</span>)</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_hill&quot;</span>))</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_hill)</span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;_hill&quot;</span>)</span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span></code></pre></div>
</details>
</div>
<p>And then create the new objects with the diversity data.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its18_alpha_ds) {</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>     tmp_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(tmp_ps), <span class="at">rooted =</span> <span class="cn">TRUE</span>,</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(tmp_ps))</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(tmp_ps, sample_data, tmp_tree)</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>     tmp_hill <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_hill&quot;</span>)))</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>     tmp_hill_samp <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(<span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_ps)),</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>                                         tmp_hill, <span class="at">by =</span> <span class="st">&quot;SamName&quot;</span>)</span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>     tmp_hill_samp<span class="sc">$</span>ID <span class="ot">&lt;-</span> tmp_hill_samp<span class="sc">$</span>SamName</span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>     tmp_hill_samp <span class="ot">&lt;-</span> tmp_hill_samp <span class="sc">%&gt;%</span>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;ID&quot;</span>)</span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>     tmp_ps2 <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(<span class="fu">otu_table</span>(tmp_ps),</span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">sample_data</span>(tmp_hill_samp),</span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">tax_table</span>(tmp_ps),</span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">phy_tree</span>(tmp_ps))</span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(i, tmp_ps2)</span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>     tmp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&quot;files/alpha/rdata/&quot;</span>)</span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>     <span class="fu">saveRDS</span>(tmp_ps2, <span class="fu">paste</span>(tmp_path, i, <span class="st">&quot;.rds&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span></code></pre></div>
</details>
</div>
<h2 id="hill-numbers-summary-1">Hill Numbers Summary</h2>
<p>Now we summarize the data for each sample against all three metrics.
The table contains the results of ASV diversity estimates from the full
data set and the three filtered data sets.</p>
<p>The suffix <code>_fi</code> indicates metrics for the Arbitrary data
set. The suffix <code>_pe</code> indicates metrics for the PERfect data
set and the suffix <code>_pet</code> indicates the lineage diversity for
the PERfect data set.</p>
<p>The suffix <code>_pi</code> indicates metrics for the PIME data set
and the suffix <code>_pit</code> indicates the lineage diversity for the
PIME data set.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="hill-summary-1">Hill Summary</h3>
<p><small><strong>(ITS) Table 1 | Hill numbers for samples in the
unfiltered &amp; filtered data sets.</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a onclick="fetch(&#39;data:text/csv;base64,U2FtcGxlX0lELE9ic2VydmVkLE9ic2VydmVkX2ZpLE9ic2VydmVkX3BlLE9ic2VydmVkX3BpLFNoYW5ub25fZXhwLFNoYW5ub25fZXhwX2ZpLFNoYW5ub25fZXhwX3BlLFNoYW5ub25fZXhwX3BpLEludlNpbXBzb24sSW52U2ltcHNvbl9maSxJbnZTaW1wc29uX3BlLEludlNpbXBzb25fcGkKUDAxX0QwMF8wMTBfVzNBLDk0NSw0OTksMTQ4LDE4MSw1NS45LDMzLjgsMTguNSwxNC44LDEwLjcsOC40LDYuOCw1LjYKUDAxX0QwMF8wMTBfVzhBLDcxOSwzNDQsOTgsMTIzLDE0MS41LDc5LDMxLjgsMzEuNiw0Mi40LDI4LjEsMTcuMywxMi41ClAwMl9EMDBfMDEwX0MwQSwxMDExLDQ5NCwxNDcsMjM4LDExMi41LDY0LDM4LjEsNDMuMywzNi4xLDI2LjIsMjEuOSwxOS45ClAwM19EMDBfMDEwX1czQiw3NjUsNDIzLDEzNiwxOTYsMTMyLjksODMsNDUuNyw1Ni41LDQ0LjEsMjkuMSwyNC40LDI2LjkKUDAzX0QwMF8wMTBfVzhCLDM0OSwyMDQsNjQsMTA5LDExMi4xLDcyLjIsMjYuMiw0Mi44LDQ5LDM2LjQsMTcuOCwyMy43ClAwNF9EMDBfMDEwX0MwQiwxMDE3LDQ3MSwxNjEsMjI5LDYwLjksNDMuMywyNS4yLDM5LjUsMTcuNiwxNS41LDEyLjUsMjAuNgpQMDVfRDAwXzAxMF9XOEMsNjE2LDI5Niw5MCwxMzEsMTU3LjUsNzYuNiwzNC4yLDM1LjEsNDEuOCwyMC43LDE1LjMsMTEuNQpQMDZfRDAwXzAxMF9DMEMsOTU0LDQ4NiwxNTcsMjQ2LDk5LjUsNTguNCwzNC4yLDM1LjIsMjMuOSwxNi41LDE0LjgsMTMuMQpQMDdfRDAwXzAxMF9XM0QsODY3LDQ5MiwxNDcsMjA1LDE2MC40LDEwMi45LDQ2LjMsNDgsNTYuMiw0Mi44LDI3LjIsMjIuNQpQMDdfRDAwXzAxMF9XOEQsMzM1LDIyMyw3Niw5NywzNSwyOCwxNi45LDE4LjEsMTMuNywxMi4zLDkuMSw5LjkKUDA4X0QwMF8wMTBfQzBELDcwMiwzNDUsMTA1LDE5OCw4Ny4yLDQ2LjcsMjUuNywzMC43LDI1LjUsMTcuOSwxNC4yLDE0LjIKUDA5X0QwMF8wMTBfVzNFLDc0NSwzOTAsMTE4LDE5NiwyMzIuNywxMzMuNiw1MS45LDY2LjYsOTYsNjEuNywzMy45LDM0LjkKUDEwX0QwMF8wMTBfQzBFLDgxMiw0MDQsMTQzLDIwMywyNC42LDE0LjcsMTEuMSw4LjEsNC45LDQsMy44LDIuNQo=&#39;).then(res =&gt; res.blob()).then(blob =&gt; {&#10;      const downloadURL = window.URL.createObjectURL(blob);&#10;      const a = document.createElement(&#39;a&#39;);&#10;      document.body.appendChild(a);&#10;      a.href = downloadURL;&#10;      a.download = &#39;its_tab_alpha_div.csv&#39;; a.click();&#10;      window.URL.revokeObjectURL(downloadURL);&#10;      document.body.removeChild(a);&#10;    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-page">
<div id="htmlwidget-7cc3b59c0f66a8380c40" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-7cc3b59c0f66a8380c40">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Sample_ID":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P10_D00_010_C0E"],"Observed":[945,719,1011,765,349,1017,616,954,867,335,702,745,812],"Observed_fi":[499,344,494,423,204,471,296,486,492,223,345,390,404],"Observed_pe":[148,98,147,136,64,161,90,157,147,76,105,118,143],"Observed_pi":[181,123,238,196,109,229,131,246,205,97,198,196,203],"Shannon_exp":[55.9,141.5,112.5,132.9,112.1,60.9,157.5,99.5,160.4,35,87.2,232.7,24.6],"Shannon_exp_fi":[33.8,79,64,83,72.2,43.3,76.6,58.4,102.9,28,46.7,133.6,14.7],"Shannon_exp_pe":[18.5,31.8,38.1,45.7,26.2,25.2,34.2,34.2,46.3,16.9,25.7,51.9,11.1],"Shannon_exp_pi":[14.8,31.6,43.3,56.5,42.8,39.5,35.1,35.2,48,18.1,30.7,66.6,8.1],"InvSimpson":[10.7,42.4,36.1,44.1,49,17.6,41.8,23.9,56.2,13.7,25.5,96,4.9],"InvSimpson_fi":[8.4,28.1,26.2,29.1,36.4,15.5,20.7,16.5,42.8,12.3,17.9,61.7,4],"InvSimpson_pe":[6.8,17.3,21.9,24.4,17.8,12.5,15.3,14.8,27.2,9.1,14.2,33.9,3.8],"InvSimpson_pi":[5.6,12.5,19.9,26.9,23.7,20.6,11.5,13.1,22.5,9.9,14.2,34.9,2.5]},"columns":[{"accessor":"Sample_ID","name":"Sample_ID","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P10_D00_010_C0E"],"header":"Sample ID","minWidth":150,"align":"left","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"5px solid #eee"},"headerStyle":{"borderRight":"5px solid #eee"}},{"accessor":"Observed","name":"FULL","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["945","719","1011","765","349","1017","616","954","867","335","702","745","812"],"header":"FULL","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Observed_fi","name":"Arbitrary","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["499","344","494","423","204","471","296","486","492","223","345","390","404"],"header":"Arbitrary","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Observed_pe","name":"PERfect","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["148","98","147","136","64","161","90","157","147","76","105","118","143"],"header":"PERfect","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Observed_pi","name":"PIME","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["181","123","238","196","109","229","131","246","205","97","198","196","203"],"header":"PIME","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"},"style":{"borderRight":"5px solid #eee"},"headerStyle":{"borderRight":"5px solid #eee"}},{"accessor":"Shannon_exp","name":"FULL","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["55.9","141.5","112.5","132.9","112.1","60.9","157.5","99.5","160.4","35","87.2","232.7","24.6"],"header":"FULL","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Shannon_exp_fi","name":"Arbitrary","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["33.8","79","64","83","72.2","43.3","76.6","58.4","102.9","28","46.7","133.6","14.7"],"header":"Arbitrary","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Shannon_exp_pe","name":"PERfect","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["18.5","31.8","38.1","45.7","26.2","25.2","34.2","34.2","46.3","16.9","25.7","51.9","11.1"],"header":"PERfect","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Shannon_exp_pi","name":"PIME","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["14.8","31.6","43.3","56.5","42.8","39.5","35.1","35.2","48","18.1","30.7","66.6","8.1"],"header":"PIME","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"},"style":{"borderRight":"5px solid #eee"},"headerStyle":{"borderRight":"5px solid #eee"}},{"accessor":"InvSimpson","name":"FULL","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["10.7","42.4","36.1","44.1","49","17.6","41.8","23.9","56.2","13.7","25.5","96","4.9"],"header":"FULL","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"InvSimpson_fi","name":"Arbitrary","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["8.4","28.1","26.2","29.1","36.4","15.5","20.7","16.5","42.8","12.3","17.9","61.7","4"],"header":"Arbitrary","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"InvSimpson_pe","name":"PERfect","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["6.8","17.3","21.9","24.4","17.8","12.5","15.3","14.8","27.2","9.1","14.2","33.9","3.8"],"header":"PERfect","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"InvSimpson_pi","name":"PIME","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["5.6","12.5","19.9","26.9","23.7","20.6","11.5","13.1","22.5","9.9","14.2","34.9","2.5"],"header":"PIME","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}}],"columnGroups":[{"name":"Observed richness","headerStyle":{"borderRight":"5px solid #eee","fontSize":"1.1em"},"columns":["Observed","Observed_fi","Observed_pe","Observed_pi"]},{"name":"Shannon exponential","headerStyle":{"fontSize":"1.1em"},"columns":["Shannon_exp","Shannon_exp_fi","Shannon_exp_pe","Shannon_exp_pi"]},{"name":"Simpson multiplicative inverse","headerStyle":{"fontSize":"1.1em"},"columns":["InvSimpson","InvSimpson_fi","InvSimpson_pe","InvSimpson_pi"]}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,13],"paginationType":"numbers","showPagination":true,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"484a2a1d115ced7d1614beaeae762c62"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
<h2 id="normality-tests-1">Normality Tests</h2>
<p>Before running significance tests, we need to know if data is
normally distributed, which will tell use whether to use a parametric or
non-parametric test. To test if the data are normally distributed, we
use the Shapiro-Wilk Normality test and the Bartlett Test of Homogeneity
of Variances.</p>
<p>If the p-values are both <strong>not significant</strong> (p &gt;
0.05) from the tests, we accept the null hypothesis (that the results
are normally distributed) and test for significance between samples
using an ANOVA. If the p-values are both <strong>significant</strong> (p
&lt; 0.05), we reject the null hypothesis (that the results are normally
distributed) and test for significance between samples using
Kruskal-Wallis (non-parametric equivalent of ANOVA).</p>
<p>The commands are as follows:</p>
<p><code>shapiro.test(x)</code>, where <code>x</code> is a numeric
vector of alpha diversity values from the sample data table.</p>
<p><code>bartlett.test(Value ~ Group, data = df)</code> Where
<code>Value</code> is the metric of interest, <code>Group</code> in the
treatment to compare, and <code>df</code> is the data frame.</p>
<p>First the Shapiro-Wilk Normality test.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>its_tab_alpha_div</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>its18_div_tab <span class="ot">&lt;-</span> its_tab_alpha_div</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>its18_shap_tests_asv <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">colnames</span>(its18_div_tab[,<span class="dv">7</span><span class="sc">:</span><span class="fu">ncol</span>(its18_div_tab)])) {</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>   tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">&quot;its18_shap_asv_&quot;</span>, .))</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>   its18_shap_tests_asv <span class="ot">&lt;-</span> <span class="fu">append</span>(its18_shap_tests_asv, tmp_name)</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>   tmp_test <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">shapiro.test</span>(its18_div_tab[[i]]))</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>   tmp_test<span class="sc">$</span>data.name <span class="ot">&lt;-</span> tmp_name</span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>   <span class="fu">assign</span>(tmp_name, tmp_test)</span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>   <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;shap_&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>And then the Bartlett Test of Homogeneity of Variances.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>its18_div_tab_asv <span class="ot">&lt;-</span> its_tab_alpha_div</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>its18_bart_tests_asv <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">colnames</span>(its18_div_tab_asv[,<span class="dv">7</span><span class="sc">:</span><span class="fu">ncol</span>(its18_div_tab_asv)])) {</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>   tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">&quot;its18_bart_asv_&quot;</span>, .))</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>   its18_bart_tests_asv <span class="ot">&lt;-</span> <span class="fu">append</span>(its18_bart_tests_asv, tmp_name)</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>   tmp_test <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">bartlett.test</span>(its18_div_tab_asv[[i]] <span class="sc">~</span> TEMP, <span class="at">data =</span> its18_div_tab_asv))</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>   tmp_test<span class="sc">$</span>data.name <span class="ot">&lt;-</span> tmp_name</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>   <span class="fu">assign</span>(tmp_name, tmp_test)</span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>   <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;_bart_&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>Here we see which Shapiro-Wilk Normality <strong>and</strong>
Bartlett tests were significant and which were not. So wherever the
value of both p-values in <code>&gt; 0.05</code> we can use an ANOVA,
otherwise we use Kruskal-Wallis.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><small><strong>(ITS) Table 2 | Summary of Normality Tests for
unfiltered &amp; filtered data sets. P-values in red indicate
significance (p-value &lt; 0.05).</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a onclick="fetch(&#39;data:text/csv;base64,bWV0cmljLGRhdGFzZXQscC12YWx1ZSAobm9ybWFsaXR5KSxwLXZhbHVlIChob21vZ2VuZWl0eSksbWV0aG9kCk9ic2VydmVkLEZVTEwsMC4xMjg0LDAuNTE2MSxBTk9WQQpPYnNlcnZlZCxGSUxULDAuMTE0OCwwLjkzODEsQU5PVkEKT2JzZXJ2ZWQsUEVSZmVjdCwwLjE2MTYsMC42Njc0LEFOT1ZBCk9ic2VydmVkLFBJTUUsMC4xMzQzLDAuNDU0MyxBTk9WQQpTaGFubm9uLEZVTEwsMC44NDY1LDAuNDQ1NSxBTk9WQQpTaGFubm9uLEZJTFQsMC45MzM5LDAuMzYzMixBTk9WQQpTaGFubm9uLFBFUmZlY3QsMC45MjEzLDAuNTU1MSxBTk9WQQpTaGFubm9uLFBJTUUsMC45NzI0LDAuNDMyNSxBTk9WQQpJbnZTaW1wc29uLEZVTEwsMC4xODM3LDAuMTI1NSxBTk9WQQpJbnZTaW1wc29uLEZJTFQsMC4zNTAzLDAuMTYwMSxBTk9WQQpJbnZTaW1wc29uLFBFUmZlY3QsMC45NjA3LDAuMjM3MixBTk9WQQpJbnZTaW1wc29uLFBJTUUsMC45NTgxLDAuNDc2MyxBTk9WQQo=&#39;).then(res =&gt; res.blob()).then(blob =&gt; {&#10;      const downloadURL = window.URL.createObjectURL(blob);&#10;      const a = document.createElement(&#39;a&#39;);&#10;      document.body.appendChild(a);&#10;      a.href = downloadURL;&#10;      a.download = &#39;its_norm_tests.csv&#39;; a.click();&#10;      window.URL.revokeObjectURL(downloadURL);&#10;      document.body.removeChild(a);&#10;    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-564573b3d9000a38e67a" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-564573b3d9000a38e67a">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"metric":["Observed","Observed","Observed","Observed","Shannon","Shannon","Shannon","Shannon","InvSimpson","InvSimpson","InvSimpson","InvSimpson"],"dataset":["FULL","FILT","PERfect","PIME","FULL","FILT","PERfect","PIME","FULL","FILT","PERfect","PIME"],"p-value (normality)":[0.1284,0.1148,0.1616,0.1343,0.8465,0.9339,0.9213,0.9724,0.1837,0.3503,0.9607,0.9581],"p-value (homogeneity)":[0.5161,0.9381,0.6674,0.4543,0.4455,0.3632,0.5551,0.4325,0.1255,0.1601,0.2372,0.4763],"method":["ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA"]},"columns":[{"accessor":"metric","name":"Metric","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["Observed","Observed","Observed","Observed","Shannon","Shannon","Shannon","Shannon","InvSimpson","InvSimpson","InvSimpson","InvSimpson"],"header":"Metric","minWidth":100,"align":"left","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"dataset","name":"dataset","type":"character","sortable":true,"resizable":false,"filterable":false,"cell":["FULL","FILT","PERfect","PIME","FULL","FILT","PERfect","PIME","FULL","FILT","PERfect","PIME"],"header":"dataset","minWidth":80,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"p-value (normality)","name":"p-value (normality)","type":"numeric","sortable":true,"resizable":false,"filterable":false,"cell":["0.1284","0.1148","0.1616","0.1343","0.8465","0.9339","0.9213","0.9724","0.1837","0.3503","0.9607","0.9581"],"header":"p-value (normality)","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"p-value (homogeneity)","name":"p-value (homogeneity)","type":"numeric","sortable":true,"resizable":false,"filterable":false,"cell":["0.5161","0.9381","0.6674","0.4543","0.4455","0.3632","0.5551","0.4325","0.1255","0.1601","0.2372","0.4763"],"header":"p-value (homogeneity)","minWidth":130,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"method","name":"method","type":"character","sortable":true,"resizable":false,"filterable":false,"cell":["ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA"],"header":"method","minWidth":120,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"}}],"defaultPageSize":12,"showPageSizeOptions":true,"pageSizeOptions":[5,10,12],"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"57ccf5f6f62b87e32127f70dae384abd"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
<h2 id="significance-tests-1">Significance Tests</h2>
<p>To begin, we need to create a hierarchy variable; a two-column matrix
specifying the relationship between samples (first column) and groups
(second column).</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>its18_hill_hier <span class="ot">&lt;-</span> its_tab_alpha_div</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>its18_hill_hier <span class="ot">&lt;-</span> its18_hill_hier <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="st">&quot;Sample_ID&quot;</span>, <span class="st">&quot;TEMP&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">remove_rownames</span>()</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>its18_hill_hier <span class="ot">&lt;-</span> its18_hill_hier[<span class="fu">order</span>(its18_hill_hier<span class="sc">$</span>TEMP), ]</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>its18_hill_hier<span class="sc">$</span>TEMP <span class="ot">=</span> <span class="fu">paste</span>(its18_hill_hier<span class="sc">$</span>TEMP, <span class="st">&#39;C&#39;</span>, <span class="at">sep=</span><span class="st">&#39;&#39;</span>)</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>its18_hill_hier <span class="ot">&lt;-</span> its18_hill_hier <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">remove_rownames</span>()</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(its18_hill_hier, <span class="st">&quot;files/alpha/rdata/its18_hill_hier.rds&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>Again, we start by testing significance of ASV diversity for the data
sets against each of the three metrics using the <code>div_test</code>
function.</p>
<p>The command is as follows:</p>
<pre><code>div_test(countable = x, qvalue = i, hierarchy = hier, 
         tree = ultrametric_tree, posthoc = TRUE)</code></pre>
<p>where <code>x</code> is ASV by sample table, <code>i</code> is the
q-value corresponding to the metric of interest, <code>hier</code> is
the hierarchy matrix, <code>tree</code> is an ultrametric formatted
phylogenetic tree (if you choose to explore lineage diversity), and
<code>posthoc</code> indicates whether to run post hoc pairwise
analyses.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>qvalue <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>)</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its18_alpha_ds) {</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>     <span class="cf">for</span> (j <span class="cf">in</span> qvalue) {</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>         tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_tu&quot;</span>)))</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>         tmp_test <span class="ot">&lt;-</span> <span class="fu">div_test</span>(tmp_get, <span class="at">qvalue =</span> j,</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">hierarchy =</span> its18_hill_hier,</span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">posthoc =</span> <span class="cn">TRUE</span>)</span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>         tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_q&quot;</span>, j, <span class="st">&quot;_adt&quot;</span>))</span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>         <span class="fu">print</span>(tmp_name)</span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>         <span class="fu">assign</span>(tmp_name, tmp_test)</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>         <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a> }</span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="summary-1">Summary</h2>
<p><small><strong>(ITS) Table 3 | Summary of significance tests for
unfiltered &amp; filtered data sets. P-values in red for post hoc
analysis indicate significance (p-value &lt; 0.05).</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a onclick="fetch(&#39;data:text/csv;base64,ZGF0YXNldCxtZXRyaWMscS12YWx1ZSxub3JtYWxpdHkgcC12YWx1ZSxob21vZ2VuZWl0eSBwLXZhbHVlLG1ldGhvZCxwb3N0aG9jIG1ldGhvZCxwb3N0aG9jIHAtdmFsdWUKRlVMTCxPYnNlcnZlZCwwLDAuMTI4NCwwLjUxNjEsQU5PVkEsVHVrZXkgcG9zdC1ob2MgdGVzdCwwLjAwNjE0NDk1NTM3NjQ2MjIwMDUKRlVMTCxTaGFubm9uIGV4cG9uZW50aWFsLDEsMC44NDU2LDAuNDQ1NCxBTk9WQSxUdWtleSBwb3N0LWhvYyB0ZXN0LDAuMjIxMjE2MjgxMzQzMTM0MDYKRlVMTCxJbnZlcnNlIFNpbXBzb24sMiwwLjE4MzcsMC4xMjU4LEFOT1ZBLFR1a2V5IHBvc3QtaG9jIHRlc3QsMC4xODE4NzIyNjcwNjIwMDI5CkZJTFQsT2JzZXJ2ZWQsMCwwLjExNDgsMC45MzgxLEFOT1ZBLFR1a2V5IHBvc3QtaG9jIHRlc3QsMC4wMDIzNjM0NTAxMzk1MzMyNApGSUxULFNoYW5ub24gZXhwb25lbnRpYWwsMSwwLjkzNDQsMC4zNjMyLEFOT1ZBLFR1a2V5IHBvc3QtaG9jIHRlc3QsMC4xMzk0ODI0NDU4MzExNjQ3CkZJTFQsSW52ZXJzZSBTaW1wc29uLDIsMC4zNDkxLDAuMTU5NCxBTk9WQSxUdWtleSBwb3N0LWhvYyB0ZXN0LDAuMTgzMjM4OTkwNzcyODQ2MgpQSU1FLE9ic2VydmVkLDAsMC4xMzQzLDAuNDU0MyxBTk9WQSxUdWtleSBwb3N0LWhvYyB0ZXN0LDcuMzYyNjU1Nzk1OTY4ODUzZS02ClBJTUUsU2hhbm5vbiBleHBvbmVudGlhbCwxLDAuOTcyMSwwLjQzNjMsQU5PVkEsVHVrZXkgcG9zdC1ob2MgdGVzdCwwLjM0NjQ0MTkzNDQyMjY4NDgKUElNRSxJbnZlcnNlIFNpbXBzb24sMiwwLjk1NDksMC40NzY4LEFOT1ZBLFR1a2V5IHBvc3QtaG9jIHRlc3QsMC4zNDE3ODgzNDU5MzU5NjcKUEVSZmVjdCxPYnNlcnZlZCwwLDAuMTYxNiwwLjY2NzQsQU5PVkEsVHVrZXkgcG9zdC1ob2MgdGVzdCwwLjAwMTA4NzA1MDMwNDEwMTIzNgpQRVJmZWN0LFNoYW5ub24gZXhwb25lbnRpYWwsMSwwLjkxOTQsMC41NTU0LEFOT1ZBLFR1a2V5IHBvc3QtaG9jIHRlc3QsMC4xODY4OTMxMTQ4NDIzNjQ4NwpQRVJmZWN0LEludmVyc2UgU2ltcHNvbiwyLDAuOTYwOSwwLjIzNjIsQU5PVkEsVHVrZXkgcG9zdC1ob2MgdGVzdCwwLjIwNDg1MDM4MzcyNTYzMTMK&#39;).then(res =&gt; res.blob()).then(blob =&gt; {&#10;      const downloadURL = window.URL.createObjectURL(blob);&#10;      const a = document.createElement(&#39;a&#39;);&#10;      document.body.appendChild(a);&#10;      a.href = downloadURL;&#10;      a.download = &#39;its_sign_tests.csv&#39;; a.click();&#10;      window.URL.revokeObjectURL(downloadURL);&#10;      document.body.removeChild(a);&#10;    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-page">
<div id="htmlwidget-de4d7ad236b03a9308f5" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-de4d7ad236b03a9308f5">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"dataset":["FULL","FULL","FULL","FILT","FILT","FILT","PIME","PIME","PIME","PERfect","PERfect","PERfect"],"metric":["Observed","Shannon exponential","Inverse Simpson","Observed","Shannon exponential","Inverse Simpson","Observed","Shannon exponential","Inverse Simpson","Observed","Shannon exponential","Inverse Simpson"],"q-value":["0","1","2","0","1","2","0","1","2","0","1","2"],"normality p-value":[0.1284,0.8456,0.1837,0.1148,0.9344,0.3491,0.1343,0.9721,0.9549,0.1616,0.9194,0.9609],"homogeneity p-value":[0.5161,0.4454,0.1258,0.9381,0.3632,0.1594,0.4543,0.4363,0.4768,0.6674,0.5554,0.2362],"method":["ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA"],"posthoc method":["Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test"],"posthoc p-value":[0.0061449553764622,0.221216281343134,0.181872267062003,0.00236345013953324,0.139482445831165,0.183238990772846,7.36265579596885e-06,0.346441934422685,0.341788345935967,0.00108705030410124,0.186893114842365,0.204850383725631]},"columns":[{"accessor":"dataset","name":"Dataset","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["FULL","FULL","FULL","FILT","FILT","FILT","PIME","PIME","PIME","PERfect","PERfect","PERfect"],"header":"Dataset","minWidth":100,"align":"left","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"metric","name":"metric","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["Observed","Shannon exponential","Inverse Simpson","Observed","Shannon exponential","Inverse Simpson","Observed","Shannon exponential","Inverse Simpson","Observed","Shannon exponential","Inverse Simpson"],"header":"metric","minWidth":140,"align":"left","footerStyle":{"fontWeight":"bold"}},{"accessor":"q-value","name":"q-value","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["0","1","2","0","1","2","0","1","2","0","1","2"],"header":"q-value","minWidth":70,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"normality p-value","name":"p-value (normality)","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["0.1284","0.8456","0.1837","0.1148","0.9344","0.3491","0.1343","0.9721","0.9549","0.1616","0.9194","0.9609"],"header":"p-value (normality)","minWidth":100,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"homogeneity p-value","name":"p-value (homogeneity)","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["0.5161","0.4454","0.1258","0.9381","0.3632","0.1594","0.4543","0.4363","0.4768","0.6674","0.5554","0.2362"],"header":"p-value (homogeneity)","minWidth":130,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"method","name":"method","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA","ANOVA"],"header":"method","minWidth":120,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"posthoc method","name":"posthoc method","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test","Tukey post-hoc test"],"header":"posthoc method","minWidth":80,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"posthoc p-value","name":"posthoc p-value","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["6.14e-03","2.21e-01","1.82e-01","2.36e-03","1.39e-01","1.83e-01","7.36e-06","3.46e-01","3.42e-01","1.09e-03","1.87e-01","2.05e-01"],"header":"posthoc p-value","minWidth":80,"align":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]}],"defaultPageSize":12,"showPageSizeOptions":true,"pageSizeOptions":[5,10,12],"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"nowrap":true,"theme":{"style":{"fontSize":"0.7em"}},"dataKey":"2d57270ad06466148b20e356d69dc825"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
<h2 id="posthoc-analyses-1">PostHoc Analyses</h2>
<p>First lets check the results of each posthoc analysis.</p>
<div class="l-body">
<div class="panelset">
<section id="observed-q-value-0-1" class="panel">
<h3>Observed (q-value = 0)</h3>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;FULL (Observed)&quot;
[1] &quot;Tukey post-hoc test&quot;
         diff       lwr        upr       p.adj
3C-0C  -68.70 -337.2489  199.84893 0.768173881
8C-0C -394.45 -662.9989 -125.90107 0.006201167
8C-3C -325.75 -608.8254  -42.67457 0.025372568</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;FILT (Observed)&quot;
[1] &quot;Tukey post-hoc test&quot;
         diff       lwr       upr       p.adj
3C-0C   11.00 -101.5783 123.57828 0.961373753
8C-0C -173.25 -285.8283 -60.67172 0.004588355
8C-3C -184.25 -302.9179 -65.58208 0.004327805</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;PIME (Observed)&quot;
[1] &quot;Tukey post-hoc test&quot;
        diff        lwr        upr          p.adj
3C-0C  -28.3  -59.02587   2.425871 0.071121744588
8C-0C -107.8 -138.52587 -77.074129 0.000006170051
8C-3C  -79.5 -111.88791 -47.112088 0.000138893963</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;PERfect (Observed)&quot;
[1] &quot;Tukey post-hoc test&quot;
        diff       lwr       upr       p.adj
3C-0C  -5.35 -38.46046  27.76046 0.898558605
8C-0C -60.60 -93.71046 -27.48954 0.001378091
8C-3C -55.25 -90.15149 -20.34851 0.003804791</code></pre>
</div>
</section>
<section id="shannon-exp-q-value-1-1" class="panel">
<h3>Shannon exp (q-value = 1)</h3>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;FULL (Shannon exponential)&quot;
[1] &quot;Tukey post-hoc test&quot;
           diff        lwr       upr     p.adj
3C-0C  68.52857  -31.79969 168.85684 0.1967805
8C-0C  34.59038  -65.73789 134.91864 0.6257554
8C-3C -33.93820 -139.69348  71.81708 0.6644870</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;FILT (Shannon exponential)&quot;
[1] &quot;Tukey post-hoc test&quot;
           diff       lwr      upr     p.adj
3C-0C  42.89239 -10.62278 96.40756 0.1199483
8C-0C  18.50838 -35.00679 72.02355 0.6240029
8C-3C -24.38401 -80.79395 32.02593 0.4878559</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;PIME (Shannon exponential)&quot;
[1] &quot;Tukey post-hoc test&quot;
             diff       lwr      upr     p.adj
3C-0C  15.1283988 -14.51712 44.77392 0.3777877
8C-0C   0.5613833 -29.08414 30.20690 0.9985157
8C-3C -14.5670155 -45.81614 16.68211 0.4381419</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;PERfect (Shannon exponential)&quot;
[1] &quot;Tukey post-hoc test&quot;
             diff        lwr       upr     p.adj
3C-0C  13.7449005  -7.058747 34.548549 0.2152611
8C-0C   0.3995566 -20.404091 21.203205 0.9984731
8C-3C -13.3453440 -35.274314  8.583626 0.2639185</code></pre>
</div>
</section>
<section id="inverse-simpson-q-value-2-1" class="panel">
<h3>Inverse Simpson (q-value = 2)</h3>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;FULL (Inverse Simpson)&quot;
[1] &quot;Tukey post-hoc test&quot;
           diff       lwr      upr     p.adj
3C-0C  30.12853 -10.93247 71.18953 0.1597780
8C-0C  15.09562 -25.96538 56.15662 0.5888250
8C-3C -15.03291 -58.31500 28.24918 0.6216188</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;FILT (Inverse Simpson)&quot;
[1] &quot;Tukey post-hoc test&quot;
            diff       lwr      upr     p.adj
3C-0C  19.474580  -7.08637 46.03553 0.1601425
8C-0C   8.362193 -18.19876 34.92314 0.6743506
8C-3C -11.112388 -39.11009 16.88531 0.5422962</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;PIME (Inverse Simpson)&quot;
[1] &quot;Tukey post-hoc test&quot;
            diff        lwr       upr     p.adj
3C-0C  8.4231399  -7.904659 24.750938 0.3705643
8C-0C  0.3894783 -15.938320 16.717277 0.9976460
8C-3C -8.0336616 -25.244672  9.177349 0.4372626</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;PERfect (Inverse Simpson)&quot;
[1] &quot;Tukey post-hoc test&quot;
           diff        lwr       upr     p.adj
3C-0C  9.662568  -4.771842 24.096979 0.2078889
8C-0C  1.449038 -12.985373 15.883449 0.9592809
8C-3C -8.213530 -23.428735  7.001674 0.3405328</code></pre>
</div>
</section>
</div>
</div>
<p>Now we can plot the results from the posthoc analyses for each metric
and data set using the function <code>div_test_plot_jjs</code>. I
modified the original function (<code>div_test_plot</code>) to control a
little of the formatting.</p>
<p>The command is as follows:</p>
<pre><code>div_test_plot(divtest = x, chart = &quot;type&quot;, colour = col.pal, 
              posthoc = TRUE, threshold = value))</code></pre>
<p>where <code>x</code> is the results from the <code>div_test</code>
function, <code>"type"</code> is chart type (box, jitter, or violin),
<code>colour</code> is is a color palette, <code>posthoc</code>
indicates whether to run posthoc pairwise analyses, and
<code>value</code> is the maximum p-value to show in pairwise posthoc
results. <strong>WARNING</strong> if none of the posthoc results are
below the specified threshold, the function will throw an error.
Therefore, until this is fixed, all posthoc values are shown.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;hack_code/div_test_plot_jjs.R&quot;</span>)</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span><span class="fu">ls</span>(<span class="at">pattern=</span><span class="st">&quot;_adt_plot&quot;</span>))</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">objects</span>(<span class="at">pattern=</span><span class="st">&quot;_adt&quot;</span>)) {</span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_plot&quot;</span>))</span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>     tmp_df <span class="ot">&lt;-</span> <span class="fu">div_test_plot_jjs</span>(tmp_get, <span class="at">chart =</span> <span class="st">&quot;box&quot;</span>,</span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">colour =</span> swel_col, <span class="at">posthoc =</span> <span class="cn">TRUE</span>)</span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>     tmp_df <span class="ot">&lt;-</span> <span class="fu">ggpar</span>(tmp_df, <span class="at">legend =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_name)</span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_df)</span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a>its18_ps_work_q0_adt_plot</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="alpha-diversity-plots-1">Alpha Diversity Plots</h2>
<p>Posthoc adjusted p-values given for each pairwise comparison.</p>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/alpha/its18_alph_div_plots_asv.png" width="7086" /></p>
</div>
<p><small><strong>(ITS) Figure 1 | Alpha diversity plots. Top row =
Observed; middle row = Shannon exponential; bottom row = Inverse
Simpson. From left to right, Full (unfiltered) data, Arbitrary filtered,
PERfect filtered, PIME filtered.</strong></small></p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="workflow-output">Workflow Output</h1>
<p>Data products generated in this workflow can be downloaded from
figshare.</p>
<iframe src="https://widgets.figshare.com/articles/16826779/embed?show_title=1" width="568" height="251" allowfullscreen frameborder="0">
</iframe>
<p></br></p>
<p><a href="beta.html" class="btnnav button_round" style="float: right;">Next
workflow: <br> 6. Beta Diversity &amp; Dispersion Estimates</a>
<a href="taxa.html" class="btnnav button_round" style="float: left;">Previous
workflow: <br> 4. Taxonomic Diversity</a></p>
<h2 class="appendix" id="source-code">Source Code</h2>
<p>The source code for this page can be accessed on GitHub by <a
href="https://github.com/sweltr/high-temp/blob/master/alpha.Rmd">clicking
this link</a>.</p>
<h2 class="appendix" id="data-availability">Data Availability</h2>
<p>Data generated in this workflow and the Rdata need to run the
workflow can be accessed on figshare at <a
href="https://doi.org/10.25573/data.16826779">10.25573/data.16826779</a>.</p>
<h2 class="appendix" id="last-updated-on">Last updated on</h2>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;2022-06-02 13:50:16 EST&quot;</code></pre>
</div>
<div id="refs" class="references csl-bib-body hanging-indent"
role="doc-bibliography">
<div id="ref-alberdi2019guide" class="csl-entry" role="doc-biblioentry">
Alberdi, Antton, and M Thomas P Gilbert. 2019a. <span>A Guide to the
Application of Hill Numbers to DNA-Based Diversity Analyses.</span>
<em>Molecular Ecology Resources</em> 19 (4): 80417. <a
href="https://doi.org/10.1111/17550998.13014">https://doi.org/10.1111/17550998.13014</a>.
</div>
<div id="ref-alberdi2019hilldiv" class="csl-entry"
role="doc-biblioentry">
. 2019b. <span>Hilldiv: An r Package for the Integral Analysis of
Diversity Based on Hill Numbers.</span> <em>bioRxiv</em>, 545665. <a
href="https://doi.org/10.1101/545665">https://doi.org/10.1101/545665</a>.
</div>
</div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/sweltr/high-temp/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/sweltr/high-temp/">https://github.com/sweltr/high-temp/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<div class="distill-site-nav distill-site-footer">
<p> Copyright 2022 metacrobe and The SWELTR Project.</p>
<p>Site constructed using <a
href="https://rstudio.github.io/distill/">Distill for R
Markdown</a>.</p>
</div>
<!--/radix_placeholder_navigation_after_body-->


</body>

</html>
