<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */
</style>

  <!--radix_placeholder_meta_tags-->
  <title>sweltr: 7 Beta Diversity &amp; Dispersion Estimates</title>

  <meta property="description" itemprop="description" content="Reproducible workflow for ... In this workflow, ...."/>

  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>


  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="sweltr: 7 Beta Diversity &amp; Dispersion Estimates"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="Reproducible workflow for ... In this workflow, ...."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="sweltr"/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="sweltr: 7 Beta Diversity &amp; Dispersion Estimates"/>
  <meta property="twitter:description" content="Reproducible workflow for ... In this workflow, ...."/>

  <!--/radix_placeholder_meta_tags-->
  
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","bibliography","output"]}},"value":[{"type":"character","attributes":{},"value":["7 Beta Diversity & Dispersion Estimates"]},{"type":"character","attributes":{},"value":["Reproducible workflow for ... In this workflow, ....\n"]},{"type":"character","attributes":{},"value":["assets/cite.bib"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["distill::distill_article"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["css","toc","toc_depth"]}},"value":[{"type":"character","attributes":{},"value":["assets/styles.css"]},{"type":"logical","attributes":{},"value":[true]},{"type":"integer","attributes":{},"value":[3]}]}]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }

  </style>

  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure img {
    width: 100%;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refs = $(this).attr('data-cites').split(" ");
        var refHtml = refs.map(function(ref) {
          return "<p>" + $('#ref-' + ref).html() + "</p>";
        }).join("\n");
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="site_libs/header-attrs-2.8/header-attrs.js"></script>
  <link href="site_libs/panelset-0.2.6/panelset.css" rel="stylesheet" />
  <script src="site_libs/panelset-0.2.6/panelset.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Roboto:300|Nunito+Sans:300|Inconsolata|Source+Sans+Pro|Work+Sans:200,300,400,500');
  /********************************/
  /* controls general properties of site*/
  /********************************/

  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
    background-color: #FEFEFE;
  }

  d-title h1{
  font-size: 2.1em;
  font-family: 'Work Sans';
  font-weight: 300;
  }

  d-article h1 {
    font-size: 1.8em;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 300;
  }

  d-article h2 {
    font-size: 1.65em;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 500;
  }

  d-article h3 {
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
    margin-top: 1em;
  }

  d-article p {
    font-size: 18px;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
  }

  d-byline .authors-affiliations {
    grid-column-end: span 3;
    grid-column-gap: 8px;
  }

  img.orcid-id {
    margin: 0;
  }

  a.name:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  a.affiliation:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  blockquote p {
  	font-size: 1.3em;
  	font-family: 'Lato';
    font-style: normal;
    margin: 0.5em 0 0.5em;
  }

  d-article li {
    margin: 0px 0px 4px;
    font-size: 18px;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
  }

  /********************************/
  /* controls CODE*/
  /********************************/

  /** pre.sourceCode controls code block without lang **/

  /** Controls r code  **/

  pre.sourceCode.r {
    margin-top: 0px;
     color: #333333;
     background-color: #E8E8E8;
     border-radius: 3px;
     font-size: 0.80em;
     font-family: 'Monaco', 'PT Mono', monospace;
     overflow-x: scroll !important;
     border-left: 6px solid #323232;
     border-top: 1px solid #B7B7B7;
     border-right: 1px solid #B7B7B7;
     border-bottom: 1px solid #B7B7B7;
     padding: 18px 0px 18px 20px;
   }

   pre.sourceCode.bash {
     margin-top: 0px;
      color: #333333;
      background-color: #E8E8E8;
      border-radius: 3px;
      font-size: 0.80em;
      font-family: 'Monaco', 'PT Mono', monospace;
      overflow-x: scroll !important;
      border-left: 6px solid #FC6464;
      border-top: 1px solid #B7B7B7;
      border-right: 1px solid #B7B7B7;
      border-bottom: 1px solid #B7B7B7;
      padding: 18px 0px 18px 20px;
    }
  /** Controls code output **/

  d-article pre {
     margin-top: 0px;
      color: #333333;
      background-color: #FAFAFA;
      font-size: 0.8em;
      border: 1px solid #ccc;
      overflow-x: scroll !important;
      padding:  10px 10px 10px 18px;
      border-radius: 5px;

    }

  /** Controls inline code  **/

  p code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }

  d-article aside code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-family: 'Monaco', 'PT Mono', monospace;
    color: #BF2727;
  }

  /********************************/
  /* controls NAV BAR and HEADERS*/
  /********************************/

  .nav-dropdown-content .nav-dropdown-header {
    text-transform: none;
  }

  .distill-site-header .logo img {
    margin-top: 10px;
    max-height: 30px;
    margin-bottom: 10px;
  }

  .distill-site-nav {
      background-color: #323232;
      font-size: 16px;
      font-weight: 300;
  }

  img {
    margin: 5px;
  }

  /** LIGHTBOX MARKUP **/
  .thumbnail {
    max-width: 100%;
    position: relative;
  }

  /** DETAILS fold down**/

  details {
    background-color: hsla(206, 95%, 20%, 0.2);
    cursor: pointer;
    color: #2d2d2d;
    margin-bottom:1vh;
    padding: 1em;
    border: none;
    text-align: left;
    outline: none;
  }

  .active, details:hover {
    transition: max-height 0.2s ease-out;
    outline: none;

    /* background-color: #acc2c2; */
  }


  details {
    display: flex;
    justify-content: space-between;
    align-content: center;
    border-radius: 4px;
    outline: none;

  }

  details.active {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    outline: none;

  }

  summary::-webkit-details-marker {
    display: none;
    outline: none;

  }
  summary:focus {
    outline: none;
  }
  summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f0fe";
  }

  details[open] summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f146";
  }

  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"7 Beta Diversity & Dispersion Estimates","description":"Reproducible workflow for ... In this workflow, ....","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title"></a>
<a href="index.html">
<i class="fa fa-home" aria-hidden="true"></i>
</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
A. Processing Workflows
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="dada2.html">1. DADA2</a>
<a href="data-prep.html">2. Data Set Prep</a>
<a href="otu.html">3. OTU Clustering</a>
<hr/>
<span class="nav-dropdown-header">Filtering</span>
<a href="pime.html">4a. PIME Filtering</a>
<a href="perfect.html">4b. PERFect Filtering</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
B. Diversity Workflows
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="taxa.html">5. Taxonomic</a>
<a href="alpha.html">6. Alpha Diversity</a>
<a href="beta.html">7. Beta Diversity</a>
<a href="da.html">8. DA ASVs &amp; OTUs</a>
<a href="metadata.html">9. Metadata</a>
</div>
</div>
</div>
<div class="nav-right">
<a href="temp.html">Data</a>
<a href="https://github.com/sweltr/high-temp/">
<i class="fa fa-github fa-lg" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>7 Beta Diversity &amp; Dispersion Estimates</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->
<p><p>Reproducible workflow for  In this workflow, .</p></p>
</div>


<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#synopsis">Synopsis</a></li>
<li><a href="#s-rrna">16s rRNA</a>
<ul>
<li><a href="#significance-tests">Significance Tests</a>
<ul>
<li><a href="#beta-dispersion">Beta Dispersion</a></li>
<li><a href="#significance-tests-full-asv">Significance Tests (FULL ASV)</a></li>
<li><a href="#significance-tests-pime-asv">Significance Tests (PIME ASV)</a></li>
<li><a href="#significance-tests-perfect-asv">Significance Tests (PERfect ASV)</a></li>
<li><a href="#summary-asv">Summary ASV</a></li>
<li><a href="#significance-tests-full-otu">Significance Tests (FULL OTU)</a></li>
<li><a href="#significance-tests-pime-otu">Significance Tests (PIME OTU)</a></li>
<li><a href="#significance-tests-perfect-otu">Significance Tests (PERfect OTU)</a></li>
<li><a href="#summary-otu">Summary OTU</a></li>
</ul></li>
<li><a href="#beta-diversity-plots">Beta Diversity Plots</a>
<ul>
<li><a href="#plot-code">Plot Code</a></li>
</ul></li>
</ul></li>
<li><a href="#its">ITS</a>
<ul>
<li><a href="#significance-tests-1">Significance Tests</a>
<ul>
<li><a href="#beta-dispersion-1">Beta Dispersion</a></li>
<li><a href="#significance-tests-full-asv-1">Significance Tests (FULL ASV)</a></li>
<li><a href="#significance-tests-pime-asv-1">Significance Tests (PIME ASV)</a></li>
<li><a href="#significance-tests-perfect-asv-1">Significance Tests (PERfect ASV)</a></li>
<li><a href="#summary-asv-1">Summary ASV</a></li>
<li><a href="#significance-tests-full-otu-1">Significance Tests (FULL OTU)</a></li>
<li><a href="#significance-tests-pime-otu-1">Significance Tests (PIME OTU)</a></li>
<li><a href="#significance-tests-perfect-otu-1">Significance Tests (PERfect OTU)</a></li>
<li><a href="#summary-otu-1">Summary OTU</a></li>
</ul></li>
<li><a href="#beta-diversity-plots-1">Beta Diversity Plots</a>
<ul>
<li><a href="#plot-code-1">Plot Code</a></li>
</ul></li>
<li><a href="#source-code">Source Code</a></li>
</ul></li>
</ul>
</nav>
</div>
<details>
<summary>
Click here for setup information.
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">echo =</span> <span class="cn">TRUE</span>, <span class="at">eval =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">119</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq); <span class="fu">packageVersion</span>(<span class="st">&quot;phyloseq&quot;</span>)</span></code></pre></div>
<pre><code>[1] &#39;1.34.0&#39;</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings); <span class="fu">packageVersion</span>(<span class="st">&quot;Biostrings&quot;</span>)</span></code></pre></div>
<pre><code>[1] &#39;2.58.0&#39;</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(tidyverse, DT, ampvis2, hilldiv, </span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>               microbiome, phytools, phangorn, </span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>               pairwiseAdonis, codefolder, naniar, </span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>               labdsv, patchwork, agricolae, </span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">install =</span> <span class="cn">FALSE</span>, <span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends(agricolae, local = TRUE)  </span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends_reverse(agricolae, local = TRUE)  </span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">scipen=</span><span class="dv">999</span>)</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_current<span class="sc">$</span><span class="fu">get</span>(<span class="fu">c</span>(</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache&quot;</span>,</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.path&quot;</span>,</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.rebuild&quot;</span>,</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;dependson&quot;</span>,</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;autodep&quot;</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>))</span></code></pre></div>
<pre><code>$cache
[1] 0

$cache.path
[1] &quot;beta_cache/html5/&quot;

$cache.rebuild
[1] FALSE

$dependson
NULL

$autodep
[1] FALSE</code></pre>
</div>
</details>
<div class="layout-chunk" data-layout="l-body">

</div>
<blockquote>
<p>Hit the <em>Hide Code</em> button to hide the R code (shown by default).</p>
</blockquote>
<aside>
<div class="layout-chunk" data-layout="l-body">
<input onclick="codefolder(&#39;pre.sourceCode&#39;);" type="button" value="Hide Code" id="codefolder-button" style="position: absolute; right: 14%; z-index: 200"/>
<script>
  function codefolder(query) {

    var x = document.querySelectorAll(query);
    if (x.length === 0) return;

    function toggle_vis(o) {
      var d = o.style.display;
      o.style.display = (d === 'block' || d === '') ? 'none':'block';
    }

    for (i = 0; i < x.length; i++) {
      var y = x[i];
      toggle_vis(y);
    }

    var elem = document.getElementById("codefolder-button");
    if (elem.value === "Hide Code") elem.value = "Show Code";
    else elem.value = "Hide Code";
  }
</script>
</div>
</aside>
<h1 id="synopsis">Synopsis</h1>
<p>This workflow contains diversity assessments for the 2018 high temperature data sets. In order to run the workflow, you either need to first run the <a href="dada2.html">DADA2 Workflow for 2018 High Temp samples</a> <strong>and</strong> the <a href="data-prep.html">Data Preparation workflow</a> <strong>or</strong> begin with the output files from the Data Preparation workflow. See the <a href="data-availability.html">Data Availability</a> page for complete details.</p>
<p>In this workflow</p>
<h1 id="s-rrna">16s rRNA</h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="significance-tests">Significance Tests</h2>
<p>In order to test for significance between sample groups, we must perform the following steps:</p>
<ol type="1">
<li>Create distance matrices based on some metrics, for example <code>wunifrac</code>, <code>unifrac</code>, and <code>jsd</code>. For this we use the function <code>phyloseq::distance</code>.</li>
<li>Next, calculate beta dispersion using the <code>betadisper</code> function from the <code>vegan</code> package.</li>
<li>Then, use the function <code>permutest</code> to run a Permutation test for homogeneity of multivariate dispersions.</li>
<li>If the beta dispersion tests are not significant we will run a PERMANOVA (since PERMANOVA assumes equal dispersion), otherwise we will use Analysis of Similarity (ANOSIM).</li>
</ol>
<p>Steps 1, 2, and 3 are analyzed in a <code>for</code> loop that tests all three distance metrics.</p>
<p>Before we begin, we must first transform sample counts to relative abundance for both data sets.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>samp_ps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ssu18_ps_work&quot;</span>, <span class="st">&quot;ssu18_ps_pime&quot;</span>, <span class="st">&quot;ssu18_ps_perfect&quot;</span>, </span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;ssu18_ps_work_otu&quot;</span>, <span class="st">&quot;ssu18_ps_pime_otu&quot;</span>, <span class="st">&quot;ssu18_ps_perfect_otu&quot;</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> samp_ps) {</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>))</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(tmp_get, <span class="cf">function</span>(otu) <span class="fl">1e5</span> <span class="sc">*</span> otu<span class="sc">/</span><span class="fu">sum</span>(otu))</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>     tmp_ps<span class="sc">@</span>phy_tree <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">prune_samples</span>(<span class="fu">sample_sums</span>(tmp_ps) <span class="sc">&gt;</span> <span class="dv">0</span>, tmp_ps)</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>     tmp_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(tmp_ps), <span class="at">rooted =</span> <span class="cn">TRUE</span>, </span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(tmp_ps))</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(tmp_ps, sample_data, tmp_tree)</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_name)</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_ps)</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<h3 id="beta-dispersion">Beta Dispersion</h3>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>dist <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;jsd&quot;</span>, <span class="st">&quot;unifrac&quot;</span>, <span class="st">&quot;wunifrac&quot;</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> samp_ps) {</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (d <span class="cf">in</span> dist){</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>             tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_prop&quot;</span>)))</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>             tmp_samp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>             tmp_df <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(tmp_get, <span class="at">method =</span> d)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>             tmp_df_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_beta_dist_&quot;</span>, .))</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>             <span class="fu">assign</span>(tmp_df_name, tmp_df)</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>             tmp_df2 <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(tmp_df, tmp_samp<span class="sc">$</span>TEMP, <span class="at">bias.adjust =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>             tmp_df_name2 <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_beta_dispersion_&quot;</span>, .))</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>             <span class="fu">assign</span>(tmp_df_name2, tmp_df2)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>             tmp_df3 <span class="ot">&lt;-</span> <span class="fu">permutest</span>(tmp_df2, <span class="at">pairwise =</span> <span class="cn">TRUE</span>,</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">permutations =</span> <span class="dv">1000</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>             tmp_df_name3 <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_permutest_&quot;</span>, .))</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>             <span class="fu">assign</span>(tmp_df_name3, tmp_df3)</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>             <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>       }</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span></code></pre></div>
</div>
<details>
<summary>
Detailed results of Beta Dispersion &amp; Permutation tests for ASV data
</summary>
<h5 id="permutation-tests-asv">Permutation tests (ASV)</h5>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
            ***** ssu18_ps_work ***** 

#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm  Pr(&gt;F)  
Groups     2 0.018639 0.0093195 2.9882   1000 0.07992 .
Residuals 12 0.037425 0.0031188                        
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3      8
0          0.431568 0.1638
3 0.412469          0.0559
8 0.159005 0.075347       

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq    Mean Sq      F N.Perm   Pr(&gt;F)   
Groups     2 0.0053980 0.00269899 5.6763   1000 0.006993 **
Residuals 12 0.0057058 0.00047549                          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3     8
0          0.795205 0.015
3 0.711058          0.014
8 0.035676 0.040101      

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq  Mean Sq     F N.Perm  Pr(&gt;F)  
Groups     2 0.026765 0.013383 8.266   1000 0.01099 *
Residuals 12 0.019428 0.001619                       
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3     8
0          0.806194 0.005
3 0.801585          0.006
8 0.019428 0.015409      

            ***** ssu18_ps_pime ***** 

#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.007283 0.0036417 0.9198   1000 0.4496
Residuals 12 0.047509 0.0039591                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.83217 0.4056
3 0.82435         0.3437
8 0.37620 0.31970       

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.010489 0.0052444 1.0181   1000 0.4176
Residuals 12 0.061816 0.0051514                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.98601 0.3516
3 0.98180         0.3397
8 0.31844 0.31157       

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm   Pr(&gt;F)    
Groups     2 0.027278 0.0136388 8.4161   1000 0.000999 ***
Residuals 12 0.019447 0.0016206                           
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3     8
0          0.976024 0.006
3 0.973601          0.001
8 0.017720 0.017128      

            ***** ssu18_ps_perfect ***** 

#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm  Pr(&gt;F)  
Groups     2 0.027097 0.0135487 4.5971   1000 0.01698 *
Residuals 12 0.035367 0.0029472                        
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3      8
0          0.780220 0.0629
3 0.771551          0.0290
8 0.061934 0.045555       

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm   Pr(&gt;F)   
Groups     2 0.035481 0.0177405 9.5079   1000 0.005994 **
Residuals 12 0.022390 0.0018659                          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
          0         3     8
0           0.3666334 0.004
3 0.3698728           0.011
8 0.0079006 0.0138310      

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq  Mean Sq      F N.Perm   Pr(&gt;F)   
Groups     2 0.028288 0.014144 8.4691   1000 0.002997 **
Residuals 12 0.020040 0.001670                          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3     8
0          0.896104 0.005
3 0.892442          0.003
8 0.017530 0.014806      </code></pre>
</div>
</details>
<details>
<summary>
Detailed results of Beta Dispersion &amp; Permutation tests for OTU data
</summary>
<h5 id="permutation-tests-otu">Permutation tests (OTU)</h5>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***** ssu18_ps_work_otu ***** 

#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm  Pr(&gt;F)  
Groups     2 0.029193 0.0145966 4.7465   1000 0.02498 *
Residuals 12 0.036903 0.0030752                        
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3     8
0          0.759241 0.041
3 0.721920          0.031
8 0.062819 0.046864      

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq   Mean Sq      F N.Perm   Pr(&gt;F)    
Groups     2 0.0089036 0.0044518 6.4055   1000 0.000999 ***
Residuals 12 0.0083400 0.0006950                           
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3     8
0          0.841159 0.008
3 0.742505          0.002
8 0.025194 0.032555      

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm   Pr(&gt;F)   
Groups     2 0.028394 0.0141969 11.107   1000 0.001998 **
Residuals 12 0.015338 0.0012781                          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
          0         3     8
0           0.5574426 0.002
3 0.5277804           0.003
8 0.0069554 0.0093532      

 ***** ssu18_ps_pime_otu ***** 

#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)  
Groups     2 0.016228 0.0081142 2.7323   1000 0.0969 .
Residuals 12 0.035636 0.0029697                       
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.93207 0.1209
3 0.93170         0.1349
8 0.12762 0.12575       

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.007974 0.0039868 1.2081   1000 0.3447
Residuals 12 0.039599 0.0032999                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.73227 0.2577
3 0.67071         0.3407
8 0.24531 0.32074       

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm   Pr(&gt;F)    
Groups     2 0.023264 0.0116321 10.671   1000 0.000999 ***
Residuals 12 0.013081 0.0010901                           
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
          0         3     8
0           0.5984016 0.001
3 0.5789562           0.002
8 0.0076016 0.0104283      

 ***** ssu18_ps_perfect_otu ***** 

#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm  Pr(&gt;F)  
Groups     2 0.033941 0.0169704 5.6989   1000 0.01099 *
Residuals 12 0.035734 0.0029778                        
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3     8
0          0.875125 0.020
3 0.870410          0.024
8 0.038832 0.039103      

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm   Pr(&gt;F)   
Groups     2 0.025389 0.0126943 8.3272   1000 0.002997 **
Residuals 12 0.018293 0.0015244                          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
          0         3     8
0           0.3106893 0.002
3 0.2991447           0.021
8 0.0077328 0.0274748      

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm   Pr(&gt;F)   
Groups     2 0.018599 0.0092993 10.196   1000 0.001998 **
Residuals 12 0.010945 0.0009121                          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
          0         3     8
0           0.5654346 0.002
3 0.5178978           0.006
8 0.0091143 0.0102264      </code></pre>
</div>
</details>
<p>Remember, if the beta dispersion p-value is greater than <code>0.05</code> we use PERMANOVA, otherwise we use ANOSIM.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">file.remove</span>(<span class="st">&quot;files/beta/tables/tmp.txt&quot;</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> samp_ps) {</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (d <span class="cf">in</span> dist){</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>          tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_permutest_&quot;</span>, d)))</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>          tmp_res <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">isTRUE</span>(tmp_get<span class="sc">$</span>tab<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>          tmp_print <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Permutation test p-value of  &quot;</span>, i, d,</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>          <span class="st">&quot; &lt; 0.05?&quot;</span>, tmp_res)</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>          <span class="fu">cat</span>(tmp_print,<span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="at">append =</span> <span class="cn">TRUE</span>, <span class="at">file =</span> <span class="st">&quot;files/beta/tables/tmp.txt&quot;</span>)</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>          <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>ssu18_perm_pvalues <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">&quot;files/beta/tables/tmp.txt&quot;</span>, <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="at">col_names =</span> <span class="st">&quot;permutation results&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="significance-tests-full-asv">Significance Tests (FULL ASV)</h3>
<div class="layout-chunk" data-layout="l-body">
<pre><code># A tibble: 3 x 1
  `permutation results`                                               
  &lt;chr&gt;                                                               
1 &quot;Permutation test p-value of   ssu18_ps_work jsd  &lt; 0.05? FALSE &quot;   
2 &quot;Permutation test p-value of   ssu18_ps_work unifrac  &lt; 0.05? TRUE &quot;
3 &quot;Permutation test p-value of   ssu18_ps_work wunifrac  &lt; 0.05? TRUE</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_adonis_jsd <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(ssu18_ps_work_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_adonis2_jsd <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(ssu18_ps_work_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_anosim_unifrac <span class="ot">&lt;-</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_work_prop, <span class="st">&quot;unifrac&quot;</span>),</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_work_groups)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_anosim_wunifrac <span class="ot">&lt;-</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_work_prop, <span class="st">&quot;wunifrac&quot;</span>),</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_work_groups)</span></code></pre></div>
</div>
<details>
<summary>
Detailed results of Significance tests for FULL data set
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***PERMANOVA for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = ssu18_ps_work_beta_dist_jsd ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)   
TEMP      2  0.21809 0.23826 1.8767 0.004995 **
Residual 12  0.69727 0.76174                   
Total    14  0.91536 1.00000                   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***ANOSIM for Unweighted UniFrac distance, `unifrac`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_work_prop, &quot;unifrac&quot;),      grouping = ssu18_ps_work_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.2133 
      Significance: 0.006 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
   90%    95%  97.5%    99% 
0.0853 0.1156 0.1469 0.1876 

Dissimilarity ranks between and within classes:
        0%  25%  50%   75% 100%  N
Between  1 32.0 58.0 81.50  103 75
0        4 10.0 19.5 30.75   56 10
3        7 15.5 31.5 37.50   61 10
8       35 72.0 84.5 93.75  105 10</code></pre>
<pre><code>
 ***ANOSIM for Weighted-UniFrac distance, `wunifrac`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_work_prop, &quot;wunifrac&quot;),      grouping = ssu18_ps_work_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.3689 
      Significance: 0.001 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.113 0.147 0.166 0.212 

Dissimilarity ranks between and within classes:
        0%   25%  50%    75% 100%  N
Between  5 37.50 60.0  80.50  102 75
0        1 10.25 22.5  33.75   41 10
3        2  6.25 17.5  27.00   43 10
8        7 62.25 89.0 101.75  105 10</code></pre>
</div>
</details>
<h3 id="significance-tests-pime-asv">Significance Tests (PIME ASV)</h3>
<div class="layout-chunk" data-layout="l-body">
<pre><code># A tibble: 3 x 1
  `permutation results`                                               
  &lt;chr&gt;                                                               
1 &quot;Permutation test p-value of   ssu18_ps_pime jsd  &lt; 0.05? FALSE &quot;   
2 &quot;Permutation test p-value of   ssu18_ps_pime unifrac  &lt; 0.05? FALSE
3 &quot;Permutation test p-value of   ssu18_ps_pime wunifrac  &lt; 0.05? TRUE</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_adonis_jsd <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(ssu18_ps_pime_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_adonis2_jsd <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(ssu18_ps_pime_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_adonis_unifrac <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(ssu18_ps_pime_beta_dist_unifrac <span class="sc">~</span> TEMP,</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_adonis2_unifrac <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(ssu18_ps_pime_beta_dist_unifrac <span class="sc">~</span> TEMP,</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_anosim_wunifrac <span class="ot">&lt;-</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_pime_prop, <span class="st">&quot;wunifrac&quot;</span>),</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_pime_groups)</span></code></pre></div>
</div>
<details>
<summary>
Detailed results of Significance tests for PIME data set
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***PERMANOVA for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = ssu18_ps_pime_beta_dist_jsd ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
TEMP      2  0.23994 0.59595 8.8496 0.000999 ***
Residual 12  0.16268 0.40405                    
Total    14  0.40262 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***PERMANOVA for Unweighted UniFrac distance, `unifrac`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = ssu18_ps_pime_beta_dist_unifrac ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs     R2      F   Pr(&gt;F)    
TEMP      2  0.78223 0.6556 11.421 0.000999 ***
Residual 12  0.41093 0.3444                    
Total    14  1.19316 1.0000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***ANOSIM for Weighted-UniFrac distance, `wunifrac`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_pime_prop, &quot;wunifrac&quot;),      grouping = ssu18_ps_pime_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.5244 
      Significance: 0.001 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.102 0.147 0.187 0.221 

Dissimilarity ranks between and within classes:
        0%   25%  50%   75% 100%  N
Between  6 40.50 63.0 83.50  105 75
0        2  8.25 14.5 22.75   36 10
3        1  5.50 15.0 23.25   39 10
8       11 58.75 66.0 91.00  104 10</code></pre>
</div>
</details>
<h3 id="significance-tests-perfect-asv">Significance Tests (PERfect ASV)</h3>
<div class="layout-chunk" data-layout="l-body">
<pre><code># A tibble: 3 x 1
  `permutation results`                                               
  &lt;chr&gt;                                                               
1 &quot;Permutation test p-value of   ssu18_ps_perfect jsd  &lt; 0.05? TRUE &quot; 
2 &quot;Permutation test p-value of   ssu18_ps_perfect unifrac  &lt; 0.05? TR
3 &quot;Permutation test p-value of   ssu18_ps_perfect wunifrac  &lt; 0.05? T</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_anosim_jsd <span class="ot">&lt;-</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_perfect_prop, <span class="st">&quot;jsd&quot;</span>),</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_perfect_groups)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_anosim_unifrac <span class="ot">&lt;-</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_perfect_prop, <span class="st">&quot;unifrac&quot;</span>),</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_perfect_groups)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_anosim_wunifrac <span class="ot">&lt;-</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_perfect_prop, <span class="st">&quot;wunifrac&quot;</span>),</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_perfect_groups)</span></code></pre></div>
</div>
<details>
<summary>
Detailed results of Significance tests for PIME data set
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***ANOSIM for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_perfect_prop, &quot;jsd&quot;),      grouping = ssu18_ps_perfect_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.2996 
      Significance: 0.002 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.107 0.135 0.181 0.228 

Dissimilarity ranks between and within classes:
        0%  25%  50%   75% 100%  N
Between  2 33.5 61.0 83.50  104 75
0        5 15.5 19.5 44.75   59 10
3        1  9.5 23.0 29.75   57 10
8       39 53.5 75.0 95.50  105 10</code></pre>
<pre><code>
 ***ANOSIM for Unweighted UniFrac distance, `unifrac`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_perfect_prop, &quot;unifrac&quot;),      grouping = ssu18_ps_perfect_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.2018 
      Significance: 0.004 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
   90%    95%  97.5%    99% 
0.0844 0.1103 0.1307 0.1565 

Dissimilarity ranks between and within classes:
        0%   25%  50%    75% 100%  N
Between  1 30.50 58.0  82.50  102 75
0        2  6.75 13.0  38.00   53 10
3       13 24.50 34.0  41.75   60 10
8       50 68.00 78.5 100.00  105 10</code></pre>
<pre><code>
 ***ANOSIM for Weighted-UniFrac distance, `wunifrac`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_perfect_prop, &quot;wunifrac&quot;),      grouping = ssu18_ps_perfect_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.3582 
      Significance: 0.001 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
   90%    95%  97.5%    99% 
0.0827 0.1218 0.1441 0.1831 

Dissimilarity ranks between and within classes:
        0%   25%  50%    75% 100%  N
Between  4 36.00 59.0  81.50  102 75
0        1 11.75 21.0  33.50   41 10
3        2  7.00 16.5  27.00   42 10
8       18 63.00 84.5 101.25  105 10</code></pre>
</div>
</details>
<h3 id="summary-asv">Summary ASV</h3>
<p>Here is a quick summary of significance tests for the FULL and PIME ASV data sets against the three distance matrices.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th style="text-align: left;">distance metric</th>
<th style="text-align: right;">p-value (FULL)</th>
<th style="text-align: right;">p-value (PIME)</th>
<th style="text-align: right;">p-value (PERfect)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Jensen-Shannon Divergence</td>
<td style="text-align: right;">0.003996</td>
<td style="text-align: right;">0.000999</td>
<td style="text-align: right;">0.002</td>
</tr>
<tr class="even">
<td style="text-align: left;">unweighted UniFrac</td>
<td style="text-align: right;">0.006000</td>
<td style="text-align: right;">0.000999</td>
<td style="text-align: right;">0.004</td>
</tr>
<tr class="odd">
<td style="text-align: left;">weighted UniFrac</td>
<td style="text-align: right;">0.001000</td>
<td style="text-align: right;">0.001000</td>
<td style="text-align: right;">0.001</td>
</tr>
</tbody>
</table>
</div>
<p><em>Summary of significant tests for the FULL, PIME, &amp; PERfect ASV data sets.</em></p>
<h3 id="significance-tests-full-otu">Significance Tests (FULL OTU)</h3>
<p>Remember, if the beta dispersion p-value is greater than <code>0.05</code> we use PERMANOVA, otherwise we use ANOSIM.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code># A tibble: 3 x 1
  `permutation results`                                               
  &lt;chr&gt;                                                               
1 &quot;Permutation test p-value of   ssu18_ps_work_otu jsd  &lt; 0.05? TRUE &quot;
2 &quot;Permutation test p-value of   ssu18_ps_work_otu unifrac  &lt; 0.05? T
3 &quot;Permutation test p-value of   ssu18_ps_work_otu wunifrac  &lt; 0.05? </code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_otu_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_otu_anosim_jsd <span class="ot">&lt;-</span></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_work_otu_prop, <span class="st">&quot;jsd&quot;</span>),</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_work_otu_groups)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_otu_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_otu_anosim_unifrac <span class="ot">&lt;-</span></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_work_otu_prop, <span class="st">&quot;unifrac&quot;</span>),</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_work_otu_groups)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_otu_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_otu_anosim_wunifrac <span class="ot">&lt;-</span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_work_otu_prop, <span class="st">&quot;wunifrac&quot;</span>),</span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_work_otu_groups)</span></code></pre></div>
</div>
<details>
<summary>
Detailed results of Significance tests for FULL data set
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***ANOSIM for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_work_otu_prop, &quot;jsd&quot;),      grouping = ssu18_ps_work_otu_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.2836 
      Significance: 0.003 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.096 0.122 0.153 0.207 

Dissimilarity ranks between and within classes:
        0%   25%  50%   75% 100%  N
Between  2 32.50 61.0 82.50  103 75
0        6 10.75 19.5 41.75   58 10
3        1  9.25 23.5 33.50   50 10
8       40 54.50 77.5 99.75  105 10</code></pre>
<pre><code>
 ***ANOSIM for Unweighted UniFrac distance, `unifrac`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_work_otu_prop, &quot;unifrac&quot;),      grouping = ssu18_ps_work_otu_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.2871 
      Significance: 0.003 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
   90%    95%  97.5%    99% 
0.0889 0.1200 0.1414 0.1841 

Dissimilarity ranks between and within classes:
        0%   25%  50%   75% 100%  N
Between  3 34.50 59.0 80.50  104 75
0        2  6.75 17.5 25.50   50 10
3        1 12.25 24.0 38.50   56 10
8       20 74.50 87.5 96.75  105 10</code></pre>
<pre><code>
 ***ANOSIM for Weighted-UniFrac distance, `wunifrac`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_work_otu_prop, &quot;wunifrac&quot;),      grouping = ssu18_ps_work_otu_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.3493 
      Significance: 0.001 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
   90%    95%  97.5%    99% 
0.0926 0.1308 0.1556 0.1831 

Dissimilarity ranks between and within classes:
        0%   25%  50%    75% 100%  N
Between  3 37.50 60.0  81.50  102 75
0        1  5.50 11.0  28.25   34 10
3        2 17.25 20.0  30.50   47 10
8       40 62.75 86.5 102.00  105 10</code></pre>
</div>
</details>
<h3 id="significance-tests-pime-otu">Significance Tests (PIME OTU)</h3>
<div class="layout-chunk" data-layout="l-body">
<pre><code># A tibble: 3 x 1
  `permutation results`                                               
  &lt;chr&gt;                                                               
1 &quot;Permutation test p-value of   ssu18_ps_pime_otu jsd  &lt; 0.05? FALSE
2 &quot;Permutation test p-value of   ssu18_ps_pime_otu unifrac  &lt; 0.05? F
3 &quot;Permutation test p-value of   ssu18_ps_pime_otu wunifrac  &lt; 0.05? </code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_otu_adonis_jsd <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(ssu18_ps_pime_otu_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_otu_adonis2_jsd <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(ssu18_ps_pime_otu_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_otu_adonis_unifrac <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(ssu18_ps_pime_otu_beta_dist_unifrac <span class="sc">~</span> TEMP,</span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_otu_adonis2_unifrac <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(ssu18_ps_pime_otu_beta_dist_unifrac <span class="sc">~</span> TEMP,</span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<details>
<summary>
Detailed results of Significance tests for PIME data set
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***PERMANOVA for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = ssu18_ps_pime_otu_beta_dist_jsd ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
TEMP      2  0.13956 0.50032 6.0076 0.000999 ***
Residual 12  0.13938 0.49968                    
Total    14  0.27894 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***PERMANOVA for Unweighted UniFrac distance, `unifrac`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = ssu18_ps_pime_otu_beta_dist_unifrac ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
TEMP      2  0.60098 0.66525 11.924 0.000999 ***
Residual 12  0.30241 0.33475                    
Total    14  0.90339 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***ANOSIM for Weighted-UniFrac distance, `wunifrac`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_pime_otu_prop, &quot;wunifrac&quot;),      grouping = ssu18_ps_pime_otu_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.4391 
      Significance: 0.001 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.114 0.163 0.197 0.243 

Dissimilarity ranks between and within classes:
        0%   25%  50%   75% 100%  N
Between  6 40.50 60.0 81.00  105 75
0        1  5.75 13.0 24.25   30 10
3        2  9.25 19.0 27.50   38 10
8       11 66.75 84.5 95.50  103 10</code></pre>
</div>
</details>
<h3 id="significance-tests-perfect-otu">Significance Tests (PERfect OTU)</h3>
<p>Remember, if the beta dispersion p-value is greater than <code>0.05</code> we use PERMANOVA, otherwise we use ANOSIM.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code># A tibble: 3 x 1
  `permutation results`                                               
  &lt;chr&gt;                                                               
1 &quot;Permutation test p-value of   ssu18_ps_perfect_otu jsd  &lt; 0.05? TR
2 &quot;Permutation test p-value of   ssu18_ps_perfect_otu unifrac  &lt; 0.05
3 &quot;Permutation test p-value of   ssu18_ps_perfect_otu wunifrac  &lt; 0.0</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_otu_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_otu_anosim_jsd <span class="ot">&lt;-</span></span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_perfect_otu_prop, <span class="st">&quot;jsd&quot;</span>),</span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_perfect_otu_groups)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_otu_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_otu_anosim_unifrac <span class="ot">&lt;-</span></span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_perfect_otu_prop, <span class="st">&quot;unifrac&quot;</span>),</span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_perfect_otu_groups)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_otu_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_otu_anosim_wunifrac <span class="ot">&lt;-</span></span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_perfect_otu_prop, <span class="st">&quot;wunifrac&quot;</span>),</span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_perfect_otu_groups)</span></code></pre></div>
</div>
<details>
<summary>
Detailed results of Significance tests for PERfect data set
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***ANOSIM for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_perfect_otu_prop, &quot;jsd&quot;),      grouping = ssu18_ps_perfect_otu_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.3004 
      Significance: 0.001 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
   90%    95%  97.5%    99% 
0.0926 0.1324 0.1620 0.2028 

Dissimilarity ranks between and within classes:
        0%   25%  50%    75% 100%  N
Between  3 33.50 61.0  82.50  102 75
0        2  8.25 13.0  35.50   54 10
3        1 14.75 23.5  37.75   51 10
8       46 56.50 79.0 100.00  105 10</code></pre>
<pre><code>
 ***ANOSIM for Unweighted UniFrac distance, `unifrac`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_perfect_otu_prop, &quot;unifrac&quot;),      grouping = ssu18_ps_perfect_otu_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.224 
      Significance: 0.007 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
   90%    95%  97.5%    99% 
0.0855 0.1192 0.1487 0.1915 

Dissimilarity ranks between and within classes:
        0%   25%  50%    75% 100%  N
Between  1 33.50 58.0  81.50  103 75
0        2  5.50 12.5  20.25   48 10
3       12 17.00 37.0  53.75   59 10
8       26 69.75 81.5 101.00  105 10</code></pre>
<pre><code>
 ***ANOSIM for Weighted-UniFrac distance, `wunifrac`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_perfect_otu_prop, &quot;wunifrac&quot;),      grouping = ssu18_ps_perfect_otu_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.3831 
      Significance: 0.001 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.102 0.150 0.188 0.234 

Dissimilarity ranks between and within classes:
        0%   25%  50%   75% 100%  N
Between  3 39.50 60.0 81.50  104 75
0        1  6.00 15.5 24.25   35 10
3        6 11.50 22.5 29.25   40 10
8       28 64.75 82.5 97.50  105 10</code></pre>
</div>
</details>
<h3 id="summary-otu">Summary OTU</h3>
<p>Here is a quick summary of significance tests for the FULL and PIME OTU data sets against the three distance matrices.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th style="text-align: left;">distance metric</th>
<th style="text-align: right;">p-value (FULL)</th>
<th style="text-align: right;">p-value (PIME)</th>
<th style="text-align: right;">p-value (PERfect)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Jensen-Shannon Divergence</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: right;">0.000999</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">unweighted UniFrac</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: right;">0.000999</td>
<td style="text-align: right;">0.007</td>
</tr>
<tr class="odd">
<td style="text-align: left;">weighted UniFrac</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.001000</td>
<td style="text-align: right;">0.001</td>
</tr>
</tbody>
</table>
</div>
<p><em>Summary of significant tests for the FULL, PIME, &amp; PERfect OTU data sets.</em></p>
<p>TO DOAdd posthoc analysis</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="beta-diversity-plots">Beta Diversity Plots</h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Here we visualize the different distance matrices using several ordination methods on the FULL, PIME filtered, and PERfect filtered data sets to access dissimilarity among sample.</p>
<h3 id="plot-code">Plot Code</h3>
<details>
<summary>
Code to reproduce plots
</summary>
<p>First, we inspect different ordination methods to see how the samples cluster using <strong>weighted-UniFrac</strong>, <strong>unweighted-UniFrac</strong>, and <strong>Jensen-Shannon Divergence</strong> distance measurements. We could also test different diversity metrics, different transformations, etc.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>gdata<span class="sc">::</span><span class="fu">keep</span>(ssu18_ps_work, ssu18_ps_pime, ssu18_ps_perfect, </span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>            ssu18_ps_work_otu, ssu18_ps_pime_otu, ssu18_ps_perfect_otu, </span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>            ssu18_ps_work_prop, ssu18_ps_pime_prop, ssu18_ps_perfect_prop, </span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>            ssu18_ps_work_otu_prop, ssu18_ps_pime_otu_prop, ssu18_ps_perfect_otu_prop, </span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a>            swel_col, <span class="at">sure =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>list_of_samps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ssu18_ps_work&quot;</span>, <span class="st">&quot;ssu18_ps_pime&quot;</span>, <span class="st">&quot;ssu18_ps_perfect&quot;</span>, </span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot;ssu18_ps_work_otu&quot;</span>, <span class="st">&quot;ssu18_ps_pime_otu&quot;</span>, <span class="st">&quot;ssu18_ps_perfect_otu&quot;</span>)</span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>dist <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;wunifrac&quot;</span>, <span class="st">&quot;unifrac&quot;</span>, <span class="st">&quot;jsd&quot;</span>)</span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (samp_ps <span class="cf">in</span> list_of_samps) {</span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (d <span class="cf">in</span> dist){</span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(samp_ps, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>)))</span>
<span id="cb72-7"><a href="#cb72-7" aria-hidden="true" tabindex="-1"></a>     ord_meths <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;NMDS&quot;</span>, <span class="st">&quot;PCoA&quot;</span>, <span class="st">&quot;CCA&quot;</span>, <span class="st">&quot;DCA&quot;</span>) <span class="co"># MDS = PCoA, &quot;CCA&quot;, &quot;DCA&quot;, &quot;DPCoA&quot;, &quot;RDA&quot;</span></span>
<span id="cb72-8"><a href="#cb72-8" aria-hidden="true" tabindex="-1"></a>     tmp_plist <span class="ot">&lt;-</span> <span class="fu">llply</span>(<span class="fu">as.list</span>(ord_meths), <span class="cf">function</span>(i, physeq, d) {</span>
<span id="cb72-9"><a href="#cb72-9" aria-hidden="true" tabindex="-1"></a>        ordi <span class="ot">=</span> <span class="fu">ordinate</span>(physeq, <span class="at">method =</span> i, <span class="at">distance =</span> d)</span>
<span id="cb72-10"><a href="#cb72-10" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plot_ordination</span>(physeq, ordi, <span class="st">&quot;samples&quot;</span>, <span class="at">color =</span> <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb72-11"><a href="#cb72-11" aria-hidden="true" tabindex="-1"></a>   }, tmp_get, d)</span>
<span id="cb72-12"><a href="#cb72-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-13"><a href="#cb72-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(tmp_plist) <span class="ot">&lt;-</span> ord_meths</span>
<span id="cb72-14"><a href="#cb72-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-15"><a href="#cb72-15" aria-hidden="true" tabindex="-1"></a>  tmp_df <span class="ot">&lt;-</span> <span class="fu">ldply</span>(tmp_plist, <span class="cf">function</span>(x){</span>
<span id="cb72-16"><a href="#cb72-16" aria-hidden="true" tabindex="-1"></a>      df <span class="ot">=</span> x<span class="sc">$</span>data[, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span>
<span id="cb72-17"><a href="#cb72-17" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(df) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Axis_1&quot;</span>, <span class="st">&quot;Axis_2&quot;</span>)</span>
<span id="cb72-18"><a href="#cb72-18" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">cbind</span>(df, x<span class="sc">$</span>data))})</span>
<span id="cb72-19"><a href="#cb72-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(tmp_df)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">&quot;method&quot;</span></span>
<span id="cb72-20"><a href="#cb72-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb72-21"><a href="#cb72-21" aria-hidden="true" tabindex="-1"></a>  tmp_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(tmp_df, <span class="fu">aes</span>(Axis_1, Axis_2, <span class="at">color =</span> TEMP, <span class="at">shape =</span> TEMP, <span class="at">fill =</span> TEMP))</span>
<span id="cb72-22"><a href="#cb72-22" aria-hidden="true" tabindex="-1"></a>  tmp_plot <span class="ot">&lt;-</span> tmp_plot <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>)</span>
<span id="cb72-23"><a href="#cb72-23" aria-hidden="true" tabindex="-1"></a>  tmp_plot <span class="ot">&lt;-</span> tmp_plot <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>method, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>)</span>
<span id="cb72-24"><a href="#cb72-24" aria-hidden="true" tabindex="-1"></a>  tmp_plot <span class="ot">&lt;-</span> tmp_plot <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> swel_col)</span>
<span id="cb72-25"><a href="#cb72-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb72-26"><a href="#cb72-26" aria-hidden="true" tabindex="-1"></a>  tmp_df_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(samp_ps, <span class="st">&quot;_dist_&quot;</span>, .))</span>
<span id="cb72-27"><a href="#cb72-27" aria-hidden="true" tabindex="-1"></a>  tmp_plist_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(samp_ps, <span class="st">&quot;_&quot;</span>, ., <span class="st">&quot;_plist&quot;</span>))</span>
<span id="cb72-28"><a href="#cb72-28" aria-hidden="true" tabindex="-1"></a>  tmp_plot_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(samp_ps, <span class="st">&quot;_dist_&quot;</span>, ., <span class="st">&quot;_plot&quot;</span>))</span>
<span id="cb72-29"><a href="#cb72-29" aria-hidden="true" tabindex="-1"></a>  tmp_list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="st">&quot;tmp_df_name&quot;</span> <span class="ot">=</span> tmp_df, <span class="at">tmp_plist_name =</span> tmp_plist, <span class="at">tmp_plot_name =</span> tmp_plot)</span>
<span id="cb72-30"><a href="#cb72-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(<span class="fu">paste0</span>(samp_ps, <span class="st">&quot;_&quot;</span>,  d, <span class="st">&quot;_ord_results&quot;</span>), tmp_list)</span>
<span id="cb72-31"><a href="#cb72-31" aria-hidden="true" tabindex="-1"></a>  <span class="co">#assign(purrr::map_chr(d, ~ paste0(samp_ps, &quot;_dist_&quot;, .)), tmp_df)</span></span>
<span id="cb72-32"><a href="#cb72-32" aria-hidden="true" tabindex="-1"></a>  <span class="co">#assign(purrr::map_chr(d, ~ paste0(samp_ps, &quot;_&quot;, ., &quot;_plist&quot;)), tmp_plist)</span></span>
<span id="cb72-33"><a href="#cb72-33" aria-hidden="true" tabindex="-1"></a>  <span class="co">#assign(purrr::map_chr(d, ~ paste0(samp_ps, &quot;_dist_&quot;, ., &quot;_plot&quot;)), tmp_plot)</span></span>
<span id="cb72-34"><a href="#cb72-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;_tmp&quot;</span>))</span>
<span id="cb72-35"><a href="#cb72-35" aria-hidden="true" tabindex="-1"></a> }</span>
<span id="cb72-36"><a href="#cb72-36" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb72-37"><a href="#cb72-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-38"><a href="#cb72-38" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;_ord_results&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>plist_name <span class="ot">&lt;-</span> <span class="fu">objects</span>(<span class="at">pattern=</span><span class="st">&quot;_ord_results&quot;</span>)</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>plot_num <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>)</span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> plist_name) {</span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (j <span class="cf">in</span> plot_num) {</span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a>       tmp_get_i <span class="ot">&lt;-</span> <span class="fu">get</span>(i)<span class="sc">$</span>tmp_plist_name</span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a>       tmp_ord <span class="ot">&lt;-</span> <span class="fu">names</span>(tmp_get_i)[j]</span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a>       tmp_name <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">str_replace</span>(i, <span class="st">&quot;ord_results&quot;</span>, tmp_ord)</span>
<span id="cb73-8"><a href="#cb73-8" aria-hidden="true" tabindex="-1"></a>       tmp_plot <span class="ot">&lt;-</span> tmp_get_i[[j]] <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> swel_col)</span>
<span id="cb73-9"><a href="#cb73-9" aria-hidden="true" tabindex="-1"></a>       tmp_plot <span class="ot">&lt;-</span> tmp_plot <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>, <span class="fu">aes</span>(<span class="at">shape =</span> TEMP)) <span class="sc">+</span></span>
<span id="cb73-10"><a href="#cb73-10" aria-hidden="true" tabindex="-1"></a>         <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>)</span>
<span id="cb73-11"><a href="#cb73-11" aria-hidden="true" tabindex="-1"></a>       tmp_plot<span class="sc">$</span>labels<span class="sc">$</span>shape <span class="ot">&lt;-</span> <span class="st">&quot;TEMP&quot;</span></span>
<span id="cb73-12"><a href="#cb73-12" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_name, tmp_plot)</span>
<span id="cb73-13"><a href="#cb73-13" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb73-14"><a href="#cb73-14" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb73-15"><a href="#cb73-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb73-16"><a href="#cb73-16" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
</details>
<p><br/></p>
<div class="l-body-outset">
<div class="panelset">
<section id="full-asv" class="panel">
<h4>FULL ASV</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:unnamed-chunk-55"></span>
<img src="include/beta/ssu18_ps_work_beta_div_ord_plots.png" alt="**Left** = Jensen-Shannon Divergence, **middle** = UniFrac (unwieghted), **right** = UniFrac (wieghted); **Top** = NMDS, **bottom** = PCoA." width="4252" />
<p class="caption">
Figure 1: <strong>Left</strong> = Jensen-Shannon Divergence, <strong>middle</strong> = UniFrac (unwieghted), <strong>right</strong> = UniFrac (wieghted); <strong>Top</strong> = NMDS, <strong>bottom</strong> = PCoA.
</p>
</div>
</div>
</section>
<section id="pime-asv" class="panel">
<h4>PIME ASV</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:unnamed-chunk-56"></span>
<img src="include/beta/ssu18_ps_pime_beta_div_ord_plots.png" alt="**Left** = Jensen-Shannon Divergence, **middle** = UniFrac (unwieghted), **right** = UniFrac (wieghted); **Top** = NMDS, **bottom** = PCoA." width="4252" />
<p class="caption">
Figure 2: <strong>Left</strong> = Jensen-Shannon Divergence, <strong>middle</strong> = UniFrac (unwieghted), <strong>right</strong> = UniFrac (wieghted); <strong>Top</strong> = NMDS, <strong>bottom</strong> = PCoA.
</p>
</div>
</div>
</section>
<section id="perfect-asv" class="panel">
<h4>PERfect ASV</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:unnamed-chunk-57"></span>
<img src="include/beta/ssu18_ps_perfect_beta_div_ord_plots.png" alt="**Left** = Jensen-Shannon Divergence, **middle** = UniFrac (unwieghted), **right** = UniFrac (wieghted); **Top** = NMDS, **bottom** = PCoA." width="4252" />
<p class="caption">
Figure 3: <strong>Left</strong> = Jensen-Shannon Divergence, <strong>middle</strong> = UniFrac (unwieghted), <strong>right</strong> = UniFrac (wieghted); <strong>Top</strong> = NMDS, <strong>bottom</strong> = PCoA.
</p>
</div>
</div>
</section>
<section id="full-otu" class="panel">
<h4>FULL OTU</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:unnamed-chunk-58"></span>
<img src="include/beta/ssu18_ps_work_otu_beta_div_ord_plots.png" alt="**Left** = Jensen-Shannon Divergence, **middle** = UniFrac (unwieghted), **right** = UniFrac (wieghted); **Top** = NMDS, **bottom** = PCoA." width="4252" />
<p class="caption">
Figure 4: <strong>Left</strong> = Jensen-Shannon Divergence, <strong>middle</strong> = UniFrac (unwieghted), <strong>right</strong> = UniFrac (wieghted); <strong>Top</strong> = NMDS, <strong>bottom</strong> = PCoA.
</p>
</div>
</div>
</section>
<section id="pime-otu" class="panel">
<h4>PIME OTU</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:unnamed-chunk-59"></span>
<img src="include/beta/ssu18_ps_pime_otu_beta_div_ord_plots.png" alt="**Left** = Jensen-Shannon Divergence, **middle** = UniFrac (unwieghted), **right** = UniFrac (wieghted); **Top** = NMDS, **bottom** = PCoA." width="4252" />
<p class="caption">
Figure 5: <strong>Left</strong> = Jensen-Shannon Divergence, <strong>middle</strong> = UniFrac (unwieghted), <strong>right</strong> = UniFrac (wieghted); <strong>Top</strong> = NMDS, <strong>bottom</strong> = PCoA.
</p>
</div>
</div>
</section>
<section id="perfect-otu" class="panel">
<h4>PERfect OTU</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:unnamed-chunk-60"></span>
<img src="include/beta/ssu18_ps_perfect_otu_beta_div_ord_plots.png" alt="**Left** = Jensen-Shannon Divergence, **middle** = UniFrac (unwieghted), **right** = UniFrac (wieghted); **Top** = NMDS, **bottom** = PCoA." width="4252" />
<p class="caption">
Figure 6: <strong>Left</strong> = Jensen-Shannon Divergence, <strong>middle</strong> = UniFrac (unwieghted), <strong>right</strong> = UniFrac (wieghted); <strong>Top</strong> = NMDS, <strong>bottom</strong> = PCoA.
</p>
</div>
</div>
</section>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body-outset">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body-outset">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="its">ITS</h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="significance-tests-1">Significance Tests</h2>
<p>In order to test for significance between sample groups, we must perform the following steps:</p>
<ol type="1">
<li>Create distance matrices based on some metrics, for example <code>gower</code>, <code>bray</code>, and <code>jsd</code>. For this we use the function <code>phyloseq::distance</code>.</li>
<li>Next, calculate beta dispersion using the <code>betadisper</code> function from the <code>vegan</code> package.</li>
<li>Then, use the function <code>permutest</code> to run a Permutation test for homogeneity of multivariate dispersions.</li>
<li>If the beta dispersion tests are not significant we will run a PERMANOVA (since PERMANOVA assumes equal dispersion), otherwise we will use Analysis of Similarity (ANOSIM).</li>
</ol>
<p>Steps 1, 2, and 3 are analyzed in a <code>for</code> loop that tests all three distance metrics.</p>
<p>Before we begin, we must first transform sample counts to relative abundance for both data sets.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>samp_ps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;its18_ps_work&quot;</span>, <span class="st">&quot;its18_ps_pime&quot;</span>, <span class="st">&quot;its18_ps_perfect&quot;</span>, </span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;its18_ps_work_otu&quot;</span>, <span class="st">&quot;its18_ps_pime_otu&quot;</span>, <span class="st">&quot;its18_ps_perfect_otu&quot;</span>)</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> samp_ps) {</span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>))</span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(tmp_get, <span class="cf">function</span>(otu) <span class="fl">1e5</span> <span class="sc">*</span> otu<span class="sc">/</span><span class="fu">sum</span>(otu))</span>
<span id="cb74-7"><a href="#cb74-7" aria-hidden="true" tabindex="-1"></a>     tmp_ps<span class="sc">@</span>phy_tree <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb74-8"><a href="#cb74-8" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">prune_samples</span>(<span class="fu">sample_sums</span>(tmp_ps) <span class="sc">&gt;</span> <span class="dv">0</span>, tmp_ps)</span>
<span id="cb74-9"><a href="#cb74-9" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(tmp_ps, sample_data)</span>
<span id="cb74-10"><a href="#cb74-10" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_name)</span>
<span id="cb74-11"><a href="#cb74-11" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_ps)</span>
<span id="cb74-12"><a href="#cb74-12" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb74-13"><a href="#cb74-13" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<h3 id="beta-dispersion-1">Beta Dispersion</h3>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>dist <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;jsd&quot;</span>, <span class="st">&quot;bray&quot;</span>, <span class="st">&quot;gower&quot;</span>)</span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> samp_ps) {</span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (d <span class="cf">in</span> dist){</span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a>             tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_prop&quot;</span>)))</span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a>             tmp_samp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a>             tmp_df <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(tmp_get, <span class="at">method =</span> d)</span>
<span id="cb75-7"><a href="#cb75-7" aria-hidden="true" tabindex="-1"></a>             tmp_df_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_beta_dist_&quot;</span>, .))</span>
<span id="cb75-8"><a href="#cb75-8" aria-hidden="true" tabindex="-1"></a>             <span class="fu">assign</span>(tmp_df_name, tmp_df)</span>
<span id="cb75-9"><a href="#cb75-9" aria-hidden="true" tabindex="-1"></a>             tmp_df2 <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(tmp_df, tmp_samp<span class="sc">$</span>TEMP, <span class="at">bias.adjust =</span> <span class="cn">TRUE</span>)</span>
<span id="cb75-10"><a href="#cb75-10" aria-hidden="true" tabindex="-1"></a>             tmp_df_name2 <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_beta_dispersion_&quot;</span>, .))</span>
<span id="cb75-11"><a href="#cb75-11" aria-hidden="true" tabindex="-1"></a>             <span class="fu">assign</span>(tmp_df_name2, tmp_df2)</span>
<span id="cb75-12"><a href="#cb75-12" aria-hidden="true" tabindex="-1"></a>             tmp_df3 <span class="ot">&lt;-</span> <span class="fu">permutest</span>(tmp_df2, <span class="at">pairwise =</span> <span class="cn">TRUE</span>,</span>
<span id="cb75-13"><a href="#cb75-13" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">permutations =</span> <span class="dv">1000</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>)</span>
<span id="cb75-14"><a href="#cb75-14" aria-hidden="true" tabindex="-1"></a>             tmp_df_name3 <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_permutest_&quot;</span>, .))</span>
<span id="cb75-15"><a href="#cb75-15" aria-hidden="true" tabindex="-1"></a>             <span class="fu">assign</span>(tmp_df_name3, tmp_df3)</span>
<span id="cb75-16"><a href="#cb75-16" aria-hidden="true" tabindex="-1"></a>             <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb75-17"><a href="#cb75-17" aria-hidden="true" tabindex="-1"></a>       }</span>
<span id="cb75-18"><a href="#cb75-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb75-19"><a href="#cb75-19" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span></code></pre></div>
</div>
<details>
<summary>
Detailed results of Beta Dispersion &amp; Permutation tests for ASV data
</summary>
<h5 id="permutation-tests-asv-1">Permutation tests (ASV)</h5>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
            ***** its18_ps_work ***** 

#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq   Mean Sq     F N.Perm Pr(&gt;F)
Groups     2 0.0016626 0.0008313 0.272   1000 0.7453
Residuals 10 0.0305600 0.0030560                    

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.64635 0.7493
3 0.65549         0.4705
8 0.75785 0.49503       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.002828 0.0014140 0.3342   1000 0.6963
Residuals 10 0.042306 0.0042306                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.53147 0.8372
3 0.53577         0.4805
8 0.85014 0.48472       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.0029899 0.0014950 1.3038   1000 0.3167
Residuals 10 0.0114662 0.0011466                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.80619 0.2358
3 0.75923         0.3576
8 0.22060 0.33265       

            ***** its18_ps_pime ***** 

#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.007678 0.0038389 0.3436   1000 0.7043
Residuals 10 0.111712 0.0111712                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.46454 0.4585
3 0.49362         0.9820
8 0.47799 0.97911       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.008276 0.0041378 0.2627   1000 0.7562
Residuals 10 0.157522 0.0157522                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.52947 0.5594
3 0.51164         0.8691
8 0.58541 0.88566       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq   Mean Sq      F N.Perm   Pr(&gt;F)    
Groups     2 0.0165873 0.0082936 15.047   1000 0.000999 ***
Residuals 10 0.0055118 0.0005512                           
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
           0          3     8
0            0.05394605 0.005
3 0.04179289            0.050
8 0.00072543 0.03074504      

            ***** its18_ps_perfect ***** 

#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.001059 0.0005293 0.1168   1000 0.8651
Residuals 10 0.045317 0.0045317                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.68731 0.9261
3 0.71118         0.6753
8 0.92002 0.67220       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.002522 0.0012611 0.2056   1000 0.8192
Residuals 10 0.061340 0.0061340                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.58242 0.9970
3 0.58316         0.6164
8 0.99819 0.62424       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.0023433 0.0011716 2.0201   1000 0.2018
Residuals 10 0.0058000 0.0005800                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3      8
0          0.851149 0.1029
3 0.869601          0.2577
8 0.091826 0.229457       </code></pre>
</div>
</details>
<details>
<summary>
Detailed results of Beta Dispersion &amp; Permutation tests for OTU data
</summary>
<h5 id="permutation-tests-otu-1">Permutation tests (OTU)</h5>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***** its18_ps_work_otu ***** 

#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.002419 0.0012097 0.2615   1000 0.7522
Residuals 10 0.046257 0.0046257                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.64635 0.7582
3 0.66923         0.4985
8 0.76033 0.50204       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.003931 0.0019657 0.3156   1000 0.7413
Residuals 10 0.062280 0.0062280                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.53946 0.8591
3 0.55841         0.4895
8 0.85261 0.48869       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.003782 0.0018910 1.6905   1000 0.2707
Residuals 10 0.011187 0.0011186                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.59041 0.1638
3 0.53049         0.3546
8 0.15716 0.31297       

 ***** its18_ps_pime_otu ***** 

#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.001693 0.0008466 0.0477   1000 0.9451
Residuals 10 0.177468 0.0177468                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.98501 0.8242
3 0.99146         0.7582
8 0.80232 0.77638       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq     F N.Perm Pr(&gt;F)
Groups     2 0.003233 0.0016163 0.077   1000 0.9211
Residuals 10 0.209821 0.0209821                    

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.82817 0.8591
3 0.84802         0.6953
8 0.84633 0.68287       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq   Mean Sq      F N.Perm   Pr(&gt;F)    
Groups     2 0.0227403 0.0113702 15.565   1000 0.000999 ***
Residuals 10 0.0073052 0.0007305                           
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
          0         3     8
0           0.0009990 0.041
3 0.0018318           0.009
8 0.0452647 0.0184460      

 ***** its18_ps_perfect_otu ***** 

#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.002185 0.0010926 0.1546   1000 0.8651
Residuals 10 0.070679 0.0070679                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.61938 0.9610
3 0.63967         0.6723
8 0.97318 0.65640       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.004550 0.0022752 0.2411   1000 0.7782
Residuals 10 0.094378 0.0094378                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.52348 0.9590
3 0.53828         0.5974
8 0.97023 0.60901       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.0016062 0.00080310 1.0557   1000 0.3906
Residuals 10 0.0076075 0.00076075                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.53147 0.2847
3 0.52398         0.4396
8 0.25374 0.42905       </code></pre>
</div>
</details>
<p>Remember, if the beta dispersion p-value is greater than <code>0.05</code> we use PERMANOVA, otherwise we use ANOSIM.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">file.remove</span>(<span class="st">&quot;files/beta/tables/tmp.txt&quot;</span>)</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> samp_ps) {</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (d <span class="cf">in</span> dist){</span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a>          tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_permutest_&quot;</span>, d)))</span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a>          tmp_res <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">isTRUE</span>(tmp_get<span class="sc">$</span>tab<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a>          tmp_print <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Permutation test p-value of  &quot;</span>, i, d,</span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a>          <span class="st">&quot; &lt; 0.05?&quot;</span>, tmp_res)</span>
<span id="cb78-8"><a href="#cb78-8" aria-hidden="true" tabindex="-1"></a>          <span class="fu">cat</span>(tmp_print,<span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="at">append =</span> <span class="cn">TRUE</span>, <span class="at">file =</span> <span class="st">&quot;files/beta/tables/tmp.txt&quot;</span>)</span>
<span id="cb78-9"><a href="#cb78-9" aria-hidden="true" tabindex="-1"></a>          <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb78-10"><a href="#cb78-10" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb78-11"><a href="#cb78-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb78-12"><a href="#cb78-12" aria-hidden="true" tabindex="-1"></a>its18_perm_pvalues <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">&quot;files/beta/tables/tmp.txt&quot;</span>, <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="at">col_names =</span> <span class="st">&quot;permutation results&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="significance-tests-full-asv-1">Significance Tests (FULL ASV)</h3>
<div class="layout-chunk" data-layout="l-body">
<pre><code># A tibble: 3 x 1
  `permutation results`                                              
  &lt;chr&gt;                                                              
1 &quot;Permutation test p-value of   its18_ps_work jsd  &lt; 0.05? FALSE &quot;  
2 &quot;Permutation test p-value of   its18_ps_work bray  &lt; 0.05? FALSE &quot; 
3 &quot;Permutation test p-value of   its18_ps_work gower  &lt; 0.05? FALSE &quot;</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>its18_ps_work_adonis_jsd <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_work_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>its18_ps_work_adonis2_jsd <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_work_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>its18_ps_work_adonis_bray <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_work_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a>its18_ps_work_adonis2_bray <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_work_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb81-4"><a href="#cb81-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>its18_ps_work_adonis_gower <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_work_beta_dist_gower <span class="sc">~</span> TEMP,</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>its18_ps_work_adonis2_gower <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_work_beta_dist_gower <span class="sc">~</span> TEMP,</span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<details>
<summary>
Detailed results of Significance tests for FULL data set
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***PERMANOVA for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = its18_ps_work_beta_dist_jsd ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs     R2      F  Pr(&gt;F)  
TEMP      2  0.38087 0.2287 1.4826 0.03596 *
Residual 10  1.28448 0.7713                 
Total    12  1.66534 1.0000                 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***PERMANOVA for Bray distance, `bray`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = its18_ps_work_beta_dist_bray ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F  Pr(&gt;F)  
TEMP      2   0.9003 0.20704 1.3055 0.03696 *
Residual 10   3.4484 0.79296                 
Total    12   4.3488 1.00000                 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***PERMANOVA for Gower distance, `gower`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = its18_ps_work_beta_dist_gower ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F  Pr(&gt;F)   
TEMP      2 0.052187 0.20163 1.2628 0.00999 **
Residual 10 0.206637 0.79837                  
Total    12 0.258824 1.00000                  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
</details>
<h3 id="significance-tests-pime-asv-1">Significance Tests (PIME ASV)</h3>
<div class="layout-chunk" data-layout="l-body">
<pre><code># A tibble: 3 x 1
  `permutation results`                                             
  &lt;chr&gt;                                                             
1 &quot;Permutation test p-value of   its18_ps_pime jsd  &lt; 0.05? FALSE &quot; 
2 &quot;Permutation test p-value of   its18_ps_pime bray  &lt; 0.05? FALSE &quot;
3 &quot;Permutation test p-value of   its18_ps_pime gower  &lt; 0.05? TRUE &quot;</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_adonis_jsd <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_pime_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_adonis2_jsd <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_pime_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_adonis_bray <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_pime_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_adonis2_bray <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_pime_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_anosim_gower <span class="ot">&lt;-</span></span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(its18_ps_pime_prop, <span class="st">&quot;gower&quot;</span>),</span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> its18_ps_pime_groups)</span></code></pre></div>
</div>
<details>
<summary>
Detailed results of Significance tests for PIME data set
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***PERMANOVA for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = its18_ps_pime_beta_dist_jsd ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs     R2      F   Pr(&gt;F)    
TEMP      2  0.54824 0.4002 3.3361 0.000999 ***
Residual 10  0.82167 0.5998                    
Total    12  1.36991 1.0000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***PERMANOVA for Bray distance, `bray`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = its18_ps_pime_beta_dist_bray ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2     F   Pr(&gt;F)   
TEMP      2   1.1998 0.30872 2.233 0.001998 **
Residual 10   2.6865 0.69128                  
Total    12   3.8862 1.00000                  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***ANOSIM for Gower distance, `gower`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(its18_ps_pime_prop, &quot;gower&quot;), grouping = its18_ps_pime_groups) 
Dissimilarity: gower 

ANOSIM statistic R: 0.6542 
      Significance: 0.001 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.147 0.203 0.258 0.317 

Dissimilarity ranks between and within classes:
        0%   25%  50%   75% 100%  N
Between  4 31.50 48.5 64.25   78 56
0       22 25.00 34.5 40.75   57 10
3        8 12.75 15.5 16.75   19  6
8        1  2.25  4.0  5.75    7  6</code></pre>
</div>
</details>
<h3 id="significance-tests-perfect-asv-1">Significance Tests (PERfect ASV)</h3>
<div class="layout-chunk" data-layout="l-body">
<pre><code># A tibble: 3 x 1
  `permutation results`                                               
  &lt;chr&gt;                                                               
1 &quot;Permutation test p-value of   its18_ps_perfect jsd  &lt; 0.05? FALSE &quot;
2 &quot;Permutation test p-value of   its18_ps_perfect bray  &lt; 0.05? FALSE
3 &quot;Permutation test p-value of   its18_ps_perfect gower  &lt; 0.05? FALS</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect_adonis_jsd <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_perfect_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect_adonis2_jsd <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_perfect_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect_adonis_bray <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_perfect_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect_adonis2_bray <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_perfect_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect_adonis_gower <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_perfect_beta_dist_gower <span class="sc">~</span> TEMP,</span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect_adonis2_gower <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_perfect_beta_dist_gower <span class="sc">~</span> TEMP,</span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<details>
<summary>
Detailed results of Significance tests for PIME data set
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***PERMANOVA for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>
Call:
adonis(formula = its18_ps_perfect_beta_dist_jsd ~ TEMP, data = adonis_sampledf,      permutations = 1000) 

Permutation: free
Number of permutations: 1000

Terms added sequentially (first to last)

          Df SumsOfSqs MeanSqs F.Model      R2  Pr(&gt;F)  
TEMP       2   0.38311 0.19156  1.5099 0.23194 0.04995 *
Residuals 10   1.26864 0.12686         0.76806          
Total     12   1.65176                 1.00000          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***PERMANOVA for Bray distance, `bray`*** </code></pre>
<pre><code>
Call:
adonis(formula = its18_ps_perfect_beta_dist_bray ~ TEMP, data = adonis_sampledf,      permutations = 1000) 

Permutation: free
Number of permutations: 1000

Terms added sequentially (first to last)

          Df SumsOfSqs MeanSqs F.Model      R2  Pr(&gt;F)  
TEMP       2    0.8938 0.44689  1.2942 0.20562 0.06294 .
Residuals 10    3.4530 0.34530         0.79438          
Total     12    4.3468                 1.00000          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***PERMANOVA for Gower distance, `gower`*** </code></pre>
<pre><code>
Call:
adonis(formula = its18_ps_perfect_beta_dist_gower ~ TEMP, data = adonis_sampledf,      permutations = 1000) 

Permutation: free
Number of permutations: 1000

Terms added sequentially (first to last)

          Df SumsOfSqs  MeanSqs F.Model      R2   Pr(&gt;F)   
TEMP       2  0.064697 0.032349  1.5095 0.23189 0.001998 **
Residuals 10  0.214300 0.021430         0.76811            
Total     12  0.278998                  1.00000            
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
</details>
<h3 id="summary-asv-1">Summary ASV</h3>
<p>Here is a quick summary of significance tests for the FULL and PIME ASV data sets against the three distance matrices.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th style="text-align: left;">distance metric</th>
<th style="text-align: right;">p-value (FULL)</th>
<th style="text-align: right;">p-value (PIME)</th>
<th style="text-align: right;">p-value (PERfect)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Jensen-Shannon Divergence</td>
<td style="text-align: right;">0.043956</td>
<td style="text-align: right;">0.001998</td>
<td style="text-align: right;">0.0499500</td>
</tr>
<tr class="even">
<td style="text-align: left;">Bray</td>
<td style="text-align: right;">0.041958</td>
<td style="text-align: right;">0.002997</td>
<td style="text-align: right;">0.0629371</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Gower</td>
<td style="text-align: right;">0.004995</td>
<td style="text-align: right;">0.001000</td>
<td style="text-align: right;">0.0019980</td>
</tr>
</tbody>
</table>
</div>
<p><em>Summary of significant tests for the FULL, PIME, &amp; PERfect ASV data sets.</em></p>
<h3 id="significance-tests-full-otu-1">Significance Tests (FULL OTU)</h3>
<p>Remember, if the beta dispersion p-value is greater than <code>0.05</code> we use PERMANOVA, otherwise we use ANOSIM.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code># A tibble: 3 x 1
  `permutation results`                                               
  &lt;chr&gt;                                                               
1 &quot;Permutation test p-value of   its18_ps_work_otu jsd  &lt; 0.05? FALSE
2 &quot;Permutation test p-value of   its18_ps_work_otu bray  &lt; 0.05? FALS
3 &quot;Permutation test p-value of   its18_ps_work_otu gower  &lt; 0.05? FAL</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a>its18_ps_work_otu_adonis_jsd <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_work_otu_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a>its18_ps_work_otu_adonis2_jsd <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_work_otu_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>its18_ps_work_otu_adonis_bray <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_work_otu_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a>its18_ps_work_otu_adonis2_bray <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_work_otu_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a>its18_ps_work_otu_adonis_gower <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_work_otu_beta_dist_gower <span class="sc">~</span> TEMP,</span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a>its18_ps_work_otu_adonis2_gower <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_work_otu_beta_dist_gower <span class="sc">~</span> TEMP,</span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<details>
<summary>
Detailed results of Significance tests for FULL data set
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***PERMANOVA for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>
Call:
adonis(formula = its18_ps_work_otu_beta_dist_jsd ~ TEMP, data = adonis_sampledf,      permutations = 1000) 

Permutation: free
Number of permutations: 1000

Terms added sequentially (first to last)

          Df SumsOfSqs MeanSqs F.Model      R2  Pr(&gt;F)  
TEMP       2   0.37585 0.18792  1.6422 0.24724 0.03896 *
Residuals 10   1.14433 0.11443         0.75276          
Total     12   1.52018                 1.00000          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***PERMANOVA for Bray distance, `bray`*** </code></pre>
<pre><code>
Call:
adonis(formula = its18_ps_work_otu_beta_dist_bray ~ TEMP, data = adonis_sampledf,      permutations = 1000) 

Permutation: free
Number of permutations: 1000

Terms added sequentially (first to last)

          Df SumsOfSqs MeanSqs F.Model      R2  Pr(&gt;F)  
TEMP       2    0.9070 0.45349  1.4081 0.21973 0.03497 *
Residuals 10    3.2206 0.32206         0.78027          
Total     12    4.1276                 1.00000          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***PERMANOVA for Gower distance, `gower`*** </code></pre>
<pre><code>
Call:
adonis(formula = its18_ps_work_otu_beta_dist_gower ~ TEMP, data = adonis_sampledf,      permutations = 1000) 

Permutation: free
Number of permutations: 1000

Terms added sequentially (first to last)

          Df SumsOfSqs  MeanSqs F.Model     R2  Pr(&gt;F)  
TEMP       2  0.051668 0.025834  1.2461 0.1995 0.01399 *
Residuals 10  0.207322 0.020732         0.8005          
Total     12  0.258990                  1.0000          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
</details>
<h3 id="significance-tests-pime-otu-1">Significance Tests (PIME OTU)</h3>
<div class="layout-chunk" data-layout="l-body">
<pre><code># A tibble: 3 x 1
  `permutation results`                                               
  &lt;chr&gt;                                                               
1 &quot;Permutation test p-value of   its18_ps_pime_otu jsd  &lt; 0.05? FALSE
2 &quot;Permutation test p-value of   its18_ps_pime_otu bray  &lt; 0.05? FALS
3 &quot;Permutation test p-value of   its18_ps_pime_otu gower  &lt; 0.05? TRU</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_otu_adonis_jsd <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_pime_otu_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_otu_adonis2_jsd <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_pime_otu_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb120-4"><a href="#cb120-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_otu_adonis_bray <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_pime_otu_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_otu_adonis2_bray <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_pime_otu_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb121-4"><a href="#cb121-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<details>
<summary>
Detailed results of Significance tests for PIME data set
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***PERMANOVA for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = its18_ps_pime_otu_beta_dist_jsd ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
TEMP      2  0.51149 0.40961 3.4689 0.000999 ***
Residual 10  0.73725 0.59039                    
Total    12  1.24874 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***PERMANOVA for Bray distance, `bray`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = its18_ps_pime_otu_beta_dist_bray ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2     F   Pr(&gt;F)    
TEMP      2   1.1739 0.32121 2.366 0.000999 ***
Residual 10   2.4807 0.67879                   
Total    12   3.6546 1.00000                   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***ANOSIM for Gower distance, `gower`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(its18_ps_pime_otu_prop, &quot;gower&quot;),      grouping = its18_ps_pime_otu_groups) 
Dissimilarity: gower 

ANOSIM statistic R: 0.7159 
      Significance: 0.001 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.136 0.185 0.227 0.289 

Dissimilarity ranks between and within classes:
        0%   25%  50%   75% 100%  N
Between  4 32.75 47.5 64.25   78 56
0        1  3.50  6.5  8.75   11 10
3       28 34.25 46.5 54.25   57  6
8       12 14.50 16.5 18.50   23  6</code></pre>
</div>
</details>
<h3 id="significance-tests-perfect-otu-1">Significance Tests (PERfect OTU)</h3>
<p>Remember, if the beta dispersion p-value is greater than <code>0.05</code> we use PERMANOVA, otherwise we use ANOSIM.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code># A tibble: 3 x 1
  `permutation results`                                               
  &lt;chr&gt;                                                               
1 &quot;Permutation test p-value of   its18_ps_perfect_otu jsd  &lt; 0.05? FA
2 &quot;Permutation test p-value of   its18_ps_perfect_otu bray  &lt; 0.05? F
3 &quot;Permutation test p-value of   its18_ps_perfect_otu gower  &lt; 0.05? </code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect_otu_adonis_jsd <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_perfect_otu_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb129-2"><a href="#cb129-2" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb129-3"><a href="#cb129-3" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect_otu_adonis2_jsd <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_perfect_otu_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb129-4"><a href="#cb129-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect_otu_adonis_bray <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_perfect_otu_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb130-2"><a href="#cb130-2" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb130-3"><a href="#cb130-3" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect_otu_adonis2_bray <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_perfect_otu_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb130-4"><a href="#cb130-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<details>
<summary>
Detailed results of Significance tests for PERfect data set
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***ANOSIM for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>
Call:
adonis(formula = its18_ps_perfect_otu_beta_dist_jsd ~ TEMP, data = adonis_sampledf,      permutations = 1000) 

Permutation: free
Number of permutations: 1000

Terms added sequentially (first to last)

          Df SumsOfSqs MeanSqs F.Model      R2  Pr(&gt;F)  
TEMP       2   0.37797 0.18898  1.7172 0.25564 0.04496 *
Residuals 10   1.10054 0.11005         0.74436          
Total     12   1.47851                 1.00000          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***ANOSIM for Bray distance, `bray`*** </code></pre>
<pre><code>
Call:
adonis(formula = its18_ps_perfect_otu_beta_dist_bray ~ TEMP,      data = adonis_sampledf, permutations = 1000) 

Permutation: free
Number of permutations: 1000

Terms added sequentially (first to last)

          Df SumsOfSqs MeanSqs F.Model      R2  Pr(&gt;F)  
TEMP       2    0.9140 0.45698   1.443 0.22396 0.03097 *
Residuals 10    3.1668 0.31668         0.77604          
Total     12    4.0808                 1.00000          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***ANOSIM for Gower distance, `gower`*** </code></pre>
<pre><code>
Call:
adonis(formula = its18_ps_perfect_otu_beta_dist_gower ~ TEMP,      data = adonis_sampledf, permutations = 1000) 

Permutation: free
Number of permutations: 1000

Terms added sequentially (first to last)

          Df SumsOfSqs  MeanSqs F.Model      R2   Pr(&gt;F)   
TEMP       2  0.076444 0.038222  1.6352 0.24645 0.001998 **
Residuals 10  0.233742 0.023374         0.75355            
Total     12  0.310186                  1.00000            
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
</details>
<h3 id="summary-otu-1">Summary OTU</h3>
<p>Here is a quick summary of significance tests for the FULL and PIME OTU data sets against the three distance matrices.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th style="text-align: left;">distance metric</th>
<th style="text-align: right;">p-value (FULL)</th>
<th style="text-align: right;">p-value (PIME)</th>
<th style="text-align: right;">p-value (PERfect)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Jensen-Shannon Divergence</td>
<td style="text-align: right;">0.038961</td>
<td style="text-align: right;">0.001998</td>
<td style="text-align: right;">0.044955</td>
</tr>
<tr class="even">
<td style="text-align: left;">Bray</td>
<td style="text-align: right;">0.034965</td>
<td style="text-align: right;">0.001998</td>
<td style="text-align: right;">0.030969</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Gower</td>
<td style="text-align: right;">0.013986</td>
<td style="text-align: right;">0.001000</td>
<td style="text-align: right;">0.001998</td>
</tr>
</tbody>
</table>
</div>
<p><em>Summary of significant tests for the FULL, PIME, &amp; PERfect OTU data sets.</em></p>
<p>TO DOAdd posthoc analysis</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="beta-diversity-plots-1">Beta Diversity Plots</h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Here we visualize the different distance matrices using several ordination methods on the FULL, PIME filtered, and PERfect filtered data sets to access dissimilarity among sample.</p>
<h3 id="plot-code-1">Plot Code</h3>
<details>
<summary>
Code to reproduce plots
</summary>
<p>First, we inspect different ordination methods to see how the samples cluster using <strong>Gower</strong>, <strong>unGower</strong>, and <strong>Jensen-Shannon Divergence</strong> distance measurements. We could also test different diversity metrics, different transformations, etc.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" aria-hidden="true" tabindex="-1"></a>list_of_samps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;its18_ps_work&quot;</span>, <span class="st">&quot;its18_ps_pime&quot;</span>, <span class="st">&quot;its18_ps_perfect&quot;</span>, </span>
<span id="cb137-2"><a href="#cb137-2" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot;its18_ps_work_otu&quot;</span>, <span class="st">&quot;its18_ps_pime_otu&quot;</span>, <span class="st">&quot;its18_ps_perfect_otu&quot;</span>)</span>
<span id="cb137-3"><a href="#cb137-3" aria-hidden="true" tabindex="-1"></a>dist <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;gower&quot;</span>, <span class="st">&quot;bray&quot;</span>, <span class="st">&quot;jsd&quot;</span>)</span>
<span id="cb137-4"><a href="#cb137-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (samp_ps <span class="cf">in</span> list_of_samps) {</span>
<span id="cb137-5"><a href="#cb137-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (d <span class="cf">in</span> dist){</span>
<span id="cb137-6"><a href="#cb137-6" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(samp_ps, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>)))</span>
<span id="cb137-7"><a href="#cb137-7" aria-hidden="true" tabindex="-1"></a>     ord_meths <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;NMDS&quot;</span>, <span class="st">&quot;PCoA&quot;</span>, <span class="st">&quot;CCA&quot;</span>, <span class="st">&quot;DCA&quot;</span>) <span class="co"># MDS = PCoA, &quot;CCA&quot;, &quot;DCA&quot;, &quot;DPCoA&quot;, &quot;RDA&quot;</span></span>
<span id="cb137-8"><a href="#cb137-8" aria-hidden="true" tabindex="-1"></a>     tmp_plist <span class="ot">&lt;-</span> <span class="fu">llply</span>(<span class="fu">as.list</span>(ord_meths), <span class="cf">function</span>(i, physeq, d) {</span>
<span id="cb137-9"><a href="#cb137-9" aria-hidden="true" tabindex="-1"></a>        ordi <span class="ot">=</span> <span class="fu">ordinate</span>(physeq, <span class="at">method =</span> i, <span class="at">distance =</span> d)</span>
<span id="cb137-10"><a href="#cb137-10" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plot_ordination</span>(physeq, ordi, <span class="st">&quot;samples&quot;</span>, <span class="at">color =</span> <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb137-11"><a href="#cb137-11" aria-hidden="true" tabindex="-1"></a>   }, tmp_get, d)</span>
<span id="cb137-12"><a href="#cb137-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-13"><a href="#cb137-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(tmp_plist) <span class="ot">&lt;-</span> ord_meths</span>
<span id="cb137-14"><a href="#cb137-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-15"><a href="#cb137-15" aria-hidden="true" tabindex="-1"></a>  tmp_df <span class="ot">&lt;-</span> <span class="fu">ldply</span>(tmp_plist, <span class="cf">function</span>(x){</span>
<span id="cb137-16"><a href="#cb137-16" aria-hidden="true" tabindex="-1"></a>      df <span class="ot">=</span> x<span class="sc">$</span>data[, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span>
<span id="cb137-17"><a href="#cb137-17" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(df) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Axis_1&quot;</span>, <span class="st">&quot;Axis_2&quot;</span>)</span>
<span id="cb137-18"><a href="#cb137-18" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">cbind</span>(df, x<span class="sc">$</span>data))})</span>
<span id="cb137-19"><a href="#cb137-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(tmp_df)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">&quot;method&quot;</span></span>
<span id="cb137-20"><a href="#cb137-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb137-21"><a href="#cb137-21" aria-hidden="true" tabindex="-1"></a>  tmp_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(tmp_df, <span class="fu">aes</span>(Axis_1, Axis_2, <span class="at">color =</span> TEMP, <span class="at">shape =</span> TEMP, <span class="at">fill =</span> TEMP))</span>
<span id="cb137-22"><a href="#cb137-22" aria-hidden="true" tabindex="-1"></a>  tmp_plot <span class="ot">&lt;-</span> tmp_plot <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>)</span>
<span id="cb137-23"><a href="#cb137-23" aria-hidden="true" tabindex="-1"></a>  tmp_plot <span class="ot">&lt;-</span> tmp_plot <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>method, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>)</span>
<span id="cb137-24"><a href="#cb137-24" aria-hidden="true" tabindex="-1"></a>  tmp_plot <span class="ot">&lt;-</span> tmp_plot <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> swel_col)</span>
<span id="cb137-25"><a href="#cb137-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb137-26"><a href="#cb137-26" aria-hidden="true" tabindex="-1"></a>  tmp_df_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(samp_ps, <span class="st">&quot;_dist_&quot;</span>, .))</span>
<span id="cb137-27"><a href="#cb137-27" aria-hidden="true" tabindex="-1"></a>  tmp_plist_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(samp_ps, <span class="st">&quot;_&quot;</span>, ., <span class="st">&quot;_plist&quot;</span>))</span>
<span id="cb137-28"><a href="#cb137-28" aria-hidden="true" tabindex="-1"></a>  tmp_plot_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(samp_ps, <span class="st">&quot;_dist_&quot;</span>, ., <span class="st">&quot;_plot&quot;</span>))</span>
<span id="cb137-29"><a href="#cb137-29" aria-hidden="true" tabindex="-1"></a>  tmp_list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="st">&quot;tmp_df_name&quot;</span> <span class="ot">=</span> tmp_df, <span class="at">tmp_plist_name =</span> tmp_plist, <span class="at">tmp_plot_name =</span> tmp_plot)</span>
<span id="cb137-30"><a href="#cb137-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(<span class="fu">paste0</span>(samp_ps, <span class="st">&quot;_&quot;</span>,  d, <span class="st">&quot;_ord_results&quot;</span>), tmp_list)</span>
<span id="cb137-31"><a href="#cb137-31" aria-hidden="true" tabindex="-1"></a>  <span class="co">#assign(purrr::map_chr(d, ~ paste0(samp_ps, &quot;_dist_&quot;, .)), tmp_df)</span></span>
<span id="cb137-32"><a href="#cb137-32" aria-hidden="true" tabindex="-1"></a>  <span class="co">#assign(purrr::map_chr(d, ~ paste0(samp_ps, &quot;_&quot;, ., &quot;_plist&quot;)), tmp_plist)</span></span>
<span id="cb137-33"><a href="#cb137-33" aria-hidden="true" tabindex="-1"></a>  <span class="co">#assign(purrr::map_chr(d, ~ paste0(samp_ps, &quot;_dist_&quot;, ., &quot;_plot&quot;)), tmp_plot)</span></span>
<span id="cb137-34"><a href="#cb137-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;_tmp&quot;</span>))</span>
<span id="cb137-35"><a href="#cb137-35" aria-hidden="true" tabindex="-1"></a> }</span>
<span id="cb137-36"><a href="#cb137-36" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="#cb138-1" aria-hidden="true" tabindex="-1"></a>plist_name <span class="ot">&lt;-</span> <span class="fu">objects</span>(<span class="at">pattern=</span><span class="st">&quot;_ord_results&quot;</span>)</span>
<span id="cb138-2"><a href="#cb138-2" aria-hidden="true" tabindex="-1"></a>plot_num <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>)</span>
<span id="cb138-3"><a href="#cb138-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> plist_name) {</span>
<span id="cb138-4"><a href="#cb138-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (j <span class="cf">in</span> plot_num) {</span>
<span id="cb138-5"><a href="#cb138-5" aria-hidden="true" tabindex="-1"></a>       tmp_get_i <span class="ot">&lt;-</span> <span class="fu">get</span>(i)<span class="sc">$</span>tmp_plist_name</span>
<span id="cb138-6"><a href="#cb138-6" aria-hidden="true" tabindex="-1"></a>       tmp_ord <span class="ot">&lt;-</span> <span class="fu">names</span>(tmp_get_i)[j]</span>
<span id="cb138-7"><a href="#cb138-7" aria-hidden="true" tabindex="-1"></a>       tmp_name <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">str_replace</span>(i, <span class="st">&quot;ord_results&quot;</span>, tmp_ord)</span>
<span id="cb138-8"><a href="#cb138-8" aria-hidden="true" tabindex="-1"></a>       tmp_plot <span class="ot">&lt;-</span> tmp_get_i[[j]] <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> swel_col)</span>
<span id="cb138-9"><a href="#cb138-9" aria-hidden="true" tabindex="-1"></a>       tmp_plot <span class="ot">&lt;-</span> tmp_plot <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>, <span class="fu">aes</span>(<span class="at">shape =</span> TEMP)) <span class="sc">+</span></span>
<span id="cb138-10"><a href="#cb138-10" aria-hidden="true" tabindex="-1"></a>         <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>)</span>
<span id="cb138-11"><a href="#cb138-11" aria-hidden="true" tabindex="-1"></a>       tmp_plot<span class="sc">$</span>labels<span class="sc">$</span>shape <span class="ot">&lt;-</span> <span class="st">&quot;TEMP&quot;</span></span>
<span id="cb138-12"><a href="#cb138-12" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_name, tmp_plot)</span>
<span id="cb138-13"><a href="#cb138-13" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb138-14"><a href="#cb138-14" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb138-15"><a href="#cb138-15" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
</details>
<p><br/></p>
<div class="l-body-outset">
<div class="panelset">
<section id="full-asv-1" class="panel">
<h4>FULL ASV</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:unnamed-chunk-129"></span>
<img src="include/beta/its18_ps_work_beta_div_ord_plots.png" alt="**Left** = Jensen-Shannon Divergence, **middle** = Bray-Curtis, **right** = Gower; **Top** = NMDS, **bottom** = PCoA." width="4252" />
<p class="caption">
Figure 7: <strong>Left</strong> = Jensen-Shannon Divergence, <strong>middle</strong> = Bray-Curtis, <strong>right</strong> = Gower; <strong>Top</strong> = NMDS, <strong>bottom</strong> = PCoA.
</p>
</div>
</div>
</section>
<section id="pime-asv-1" class="panel">
<h4>PIME ASV</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:unnamed-chunk-130"></span>
<img src="include/beta/its18_ps_pime_beta_div_ord_plots.png" alt="**Left** = Jensen-Shannon Divergence, **middle** = Bray-Curtis, **right** = Gower; **Top** = NMDS, **bottom** = PCoA." width="4252" />
<p class="caption">
Figure 8: <strong>Left</strong> = Jensen-Shannon Divergence, <strong>middle</strong> = Bray-Curtis, <strong>right</strong> = Gower; <strong>Top</strong> = NMDS, <strong>bottom</strong> = PCoA.
</p>
</div>
</div>
</section>
<section id="perfect-asv-1" class="panel">
<h4>PERfect ASV</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:unnamed-chunk-131"></span>
<img src="include/beta/its18_ps_perfect_beta_div_ord_plots.png" alt="**Left** = Jensen-Shannon Divergence, **middle** = Bray-Curtis, **right** = Gower; **Top** = NMDS, **bottom** = PCoA." width="4252" />
<p class="caption">
Figure 9: <strong>Left</strong> = Jensen-Shannon Divergence, <strong>middle</strong> = Bray-Curtis, <strong>right</strong> = Gower; <strong>Top</strong> = NMDS, <strong>bottom</strong> = PCoA.
</p>
</div>
</div>
</section>
<section id="full-otu-1" class="panel">
<h4>FULL OTU</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:unnamed-chunk-132"></span>
<img src="include/beta/its18_ps_work_otu_beta_div_ord_plots.png" alt="**Left** = Jensen-Shannon Divergence, **middle** = Bray-Curtis, **right** = Gower; **Top** = NMDS, **bottom** = PCoA." width="4252" />
<p class="caption">
Figure 10: <strong>Left</strong> = Jensen-Shannon Divergence, <strong>middle</strong> = Bray-Curtis, <strong>right</strong> = Gower; <strong>Top</strong> = NMDS, <strong>bottom</strong> = PCoA.
</p>
</div>
</div>
</section>
<section id="pime-otu-1" class="panel">
<h4>PIME OTU</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:unnamed-chunk-133"></span>
<img src="include/beta/its18_ps_pime_otu_beta_div_ord_plots.png" alt="**Left** = Jensen-Shannon Divergence, **middle** = Bray-Curtis, **right** = Gower; **Top** = NMDS, **bottom** = PCoA." width="4252" />
<p class="caption">
Figure 11: <strong>Left</strong> = Jensen-Shannon Divergence, <strong>middle</strong> = Bray-Curtis, <strong>right</strong> = Gower; <strong>Top</strong> = NMDS, <strong>bottom</strong> = PCoA.
</p>
</div>
</div>
</section>
<section id="perfect-otu-1" class="panel">
<h4>PERfect OTU</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:unnamed-chunk-134"></span>
<img src="include/beta/its18_ps_perfect_otu_beta_div_ord_plots.png" alt="**Left** = Jensen-Shannon Divergence, **middle** = Bray-Curtis, **right** = Gower; **Top** = NMDS, **bottom** = PCoA." width="4252" />
<p class="caption">
Figure 12: <strong>Left</strong> = Jensen-Shannon Divergence, <strong>middle</strong> = Bray-Curtis, <strong>right</strong> = Gower; <strong>Top</strong> = NMDS, <strong>bottom</strong> = PCoA.
</p>
</div>
</div>
</section>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body-outset">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body-outset">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 class="appendix" id="source-code">Source Code</h2>
<p>The source code for this page can be accessed on GitHub by <a href="https://github.com/sweltr/high-temp/blob/master/beta.Rmd">clicking this link</a>. Please note, that in order to process the data <em>and</em> build the website, we needed to run the workflow and get the results. Then hard code the results and turn off the individual commands. So the raw file for this page is a bit messyyou have been warned.</p>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/sweltr/high-temp/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/sweltr/high-temp/">https://github.com/sweltr/high-temp/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<div class="distill-site-nav distill-site-footer">
<p> Copyright 2021 The SWELTR Project.</p>
<p>Site constructed using <a href="https://rstudio.github.io/distill/">Distill for R Markdown</a>.</p>
</div>
<!--/radix_placeholder_navigation_after_body-->


</body>

</html>
