<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */
</style>

<style>
  div.csl-bib-body { }
  div.csl-entry {
    clear: both;
    }
  .hanging div.csl-entry {
    margin-left:2em;
    text-indent:-2em;
  }
  div.csl-left-margin {
    min-width:2em;
    float:left;
  }
  div.csl-right-inline {
    margin-left:2em;
    padding-left:1em;
  }
  div.csl-indent {
    margin-left: 2em;
  }
</style>

  <!--radix_placeholder_meta_tags-->
  <title>: 6. Beta Diversity &amp; Dispersion Estimates</title>

  <meta property="description" itemprop="description" content="Reproducible workflow for assessing community dissimilarity across temperature treatments."/>

  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>


  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content=": 6. Beta Diversity &amp; Dispersion Estimates"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="Reproducible workflow for assessing community dissimilarity across temperature treatments."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content=""/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content=": 6. Beta Diversity &amp; Dispersion Estimates"/>
  <meta property="twitter:description" content="Reproducible workflow for assessing community dissimilarity across temperature treatments."/>

  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Vegan: Community ecology package;citation_publication_date=2012;citation_volume=2;citation_author=Jari Oksanen;citation_author=F. Guillaume Blanchet;citation_author=Roeland Kindt;citation_author=Pierre Legendre;citation_author=Peter R. Minchin;citation_author=R. B. O’Hara;citation_author=Gavin L. Simpson;citation_author=Peter Sólymos;citation_author=M. Henry H. Stevens;citation_author=Helene Wagner"/>
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","bibliography","output"]}},"value":[{"type":"character","attributes":{},"value":["6. Beta Diversity & Dispersion Estimates"]},{"type":"character","attributes":{},"value":["Reproducible workflow for assessing community dissimilarity across temperature treatments.\n"]},{"type":"character","attributes":{},"value":["assets/cite.bib"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["distill::distill_article"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["css","toc","toc_depth"]}},"value":[{"type":"character","attributes":{},"value":["assets/styles.css"]},{"type":"logical","attributes":{},"value":[true]},{"type":"integer","attributes":{},"value":[3]}]}]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }

  </style>

  <script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <script src="site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
  <script src="site_libs/fuse-6.4.1/fuse.min.js"></script>

  <script type="application/javascript">

  function getMeta(metaName) {
    var metas = document.getElementsByTagName('meta');
    for (let i = 0; i < metas.length; i++) {
      if (metas[i].getAttribute('name') === metaName) {
        return metas[i].getAttribute('content');
      }
    }
    return '';
  }

  function offsetURL(url) {
    var offset = getMeta('distill:offset');
    return offset ? offset + '/' + url : url;
  }

  function createFuseIndex() {

    // create fuse index
    var options = {
      keys: [
        { name: 'title', weight: 20 },
        { name: 'categories', weight: 15 },
        { name: 'description', weight: 10 },
        { name: 'contents', weight: 5 },
      ],
      ignoreLocation: true,
      threshold: 0
    };
    var fuse = new window.Fuse([], options);

    // fetch the main search.json
    return fetch(offsetURL('search.json'))
      .then(function(response) {
        if (response.status == 200) {
          return response.json().then(function(json) {
            // index main articles
            json.articles.forEach(function(article) {
              fuse.add(article);
            });
            // download collections and index their articles
            return Promise.all(json.collections.map(function(collection) {
              return fetch(offsetURL(collection)).then(function(response) {
                if (response.status === 200) {
                  return response.json().then(function(articles) {
                    articles.forEach(function(article) {
                      fuse.add(article);
                    });
                  })
                } else {
                  return Promise.reject(
                    new Error('Unexpected status from search index request: ' +
                              response.status)
                  );
                }
              });
            })).then(function() {
              return fuse;
            });
          });

        } else {
          return Promise.reject(
            new Error('Unexpected status from search index request: ' +
                        response.status)
          );
        }
      });
  }

  window.document.addEventListener("DOMContentLoaded", function (event) {

    // get search element (bail if we don't have one)
    var searchEl = window.document.getElementById('distill-search');
    if (!searchEl)
      return;

    createFuseIndex()
      .then(function(fuse) {

        // make search box visible
        searchEl.classList.remove('hidden');

        // initialize autocomplete
        var options = {
          autoselect: true,
          hint: false,
          minLength: 2,
        };
        window.autocomplete(searchEl, options, [{
          source: function(query, callback) {
            const searchOptions = {
              isCaseSensitive: false,
              shouldSort: true,
              minMatchCharLength: 2,
              limit: 10,
            };
            var results = fuse.search(query, searchOptions);
            callback(results
              .map(function(result) { return result.item; })
            );
          },
          templates: {
            suggestion: function(suggestion) {
              var img = suggestion.preview && Object.keys(suggestion.preview).length > 0
                ? `<img src="${offsetURL(suggestion.preview)}"</img>`
                : '';
              var html = `
                <div class="search-item">
                  <h3>${suggestion.title}</h3>
                  <div class="search-item-description">
                    ${suggestion.description || ''}
                  </div>
                  <div class="search-item-preview">
                    ${img}
                  </div>
                </div>
              `;
              return html;
            }
          }
        }]).on('autocomplete:selected', function(event, suggestion) {
          window.location.href = offsetURL(suggestion.path);
        });
        // remove inline display style on autocompleter (we want to
        // manage responsive display via css)
        $('.algolia-autocomplete').css("display", "");
      })
      .catch(function(error) {
        console.log(error);
      });

  });

  </script>

  <style type="text/css">

  .nav-search {
    font-size: x-small;
  }

  /* Algolioa Autocomplete */

  .algolia-autocomplete {
    display: inline-block;
    margin-left: 10px;
    vertical-align: sub;
    background-color: white;
    color: black;
    padding: 6px;
    padding-top: 8px;
    padding-bottom: 0;
    border-radius: 6px;
    border: 1px #0F2E3D solid;
    width: 180px;
  }


  @media screen and (max-width: 768px) {
    .distill-site-nav .algolia-autocomplete {
      display: none;
      visibility: hidden;
    }
    .distill-site-nav.responsive .algolia-autocomplete {
      display: inline-block;
      visibility: visible;
    }
    .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
      margin-left: 0;
      width: 400px;
      max-height: 400px;
    }
  }

  .algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
    width: 90%;
    outline: none;
    border: none;
  }

  .algolia-autocomplete .aa-hint {
    color: #999;
  }
  .algolia-autocomplete .aa-dropdown-menu {
    width: 550px;
    max-height: 70vh;
    overflow-x: visible;
    overflow-y: scroll;
    padding: 5px;
    margin-top: 3px;
    margin-left: -150px;
    background-color: #fff;
    border-radius: 5px;
    border: 1px solid #999;
    border-top: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
    cursor: pointer;
    padding: 5px 4px;
    border-bottom: 1px solid #eee;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
    border-bottom: none;
    margin-bottom: 2px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
    overflow: hidden;
    font-size: 0.8em;
    line-height: 1.4em;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
    font-size: 1rem;
    margin-block-start: 0;
    margin-block-end: 5px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
    display: inline-block;
    overflow: hidden;
    height: 2.8em;
    width: 80%;
    margin-right: 4%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
    display: inline-block;
    width: 15%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
    height: 3em;
    width: auto;
    display: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
    display: initial;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
    background-color: #eee;
  }
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
    font-weight: bold;
    font-style: normal;
  }

  </style>


  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure img {
    width: 100%;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refs = $(this).attr('data-cites').split(" ");
        var refHtml = refs.map(function(ref) {
          return "<p>" + $('#ref-' + ref).html() + "</p>";
        }).join("\n");
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
  <script src="site_libs/clipboard-2.0.6/clipboard.min.js"></script>
  <link href="site_libs/xaringanExtra-clipboard-0.2.6/xaringanExtra-clipboard.css" rel="stylesheet" />
  <script src="site_libs/xaringanExtra-clipboard-0.2.6/xaringanExtra-clipboard.js"></script>
  <script>window.xaringanExtraClipboard(null, {"button":"<i class=\"fa fa-clone fa-2x\" style=\"color: #004276\"><\/i>","success":"<i class=\"fa fa-check fa-2x\" style=\"color: #90BE6D\"><\/i>","error":"<i class=\"fa fa-times fa-2x\" style=\"color: #F94144\"><\/i>"})</script>
  <script src="site_libs/font-awesome-5.13.0/js/script.js"></script>
  <link href="site_libs/panelset-0.2.6/panelset.css" rel="stylesheet" />
  <script src="site_libs/panelset-0.2.6/panelset.js"></script>
  <script src="site_libs/core-js-2.5.3/shim.min.js"></script>
  <script src="site_libs/react-17.0.0/react.min.js"></script>
  <script src="site_libs/react-17.0.0/react-dom.min.js"></script>
  <script src="site_libs/reactwidget-1.0.0/react-tools.js"></script>
  <script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
  <script src="site_libs/reactable-binding-0.2.3.9000/reactable.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Inconsolata|Source+Sans+Pro|Work+Sans:200,300,400,500|Karla:400,400i,700,700i');
  /********************************/
  /* controls general properties of site*/
  /********************************/

  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    background-color: #FEFEFE;
  }

  d-title h1{
  font-size: 2.1em;
  font-family: 'Work Sans';
  font-weight: 300;
  }

  d-article h1 {
    font-size: 1.8em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 300;
  }

  d-article h2 {
    font-size: 1.65em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 500;
  }

  d-article h3 {
    margin-top: 1em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
  }

  d-byline .authors-affiliations {
    grid-column-end: span 3;
    grid-column-gap: 8px;
  }

  img.orcid-id {
    margin: 0;
  }

  a.name:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  a.affiliation:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  blockquote p {
  	font-size: 1em;
  	font-family: 'Work Sans';
    font-style: normal;
    margin: 0.5em 0 0.5em;
  }

  d-article li {
    margin: 0px 0px 4px;
    font-size: 18px;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
  }

  /********************************/
  /* controls CODE*/
  /********************************/

  /** pre.sourceCode controls code block without lang **/

  /** Controls r code  **/

  pre.sourceCode.r {
    margin-top: 0px;
     color: #333333;
     background-color: #E8E8E8;
     border-radius: 3px;
     font-size: 0.80em;
     font-family: 'Monaco', 'PT Mono', monospace;
     overflow-x: scroll !important;
     border-left: 6px solid #2271B2;
     border-top: 1px solid #B7B7B7;
     border-right: 1px solid #B7B7B7;
     border-bottom: 1px solid #B7B7B7;
     padding: 18px 0px 18px 20px;
   }

   pre.sourceCode.bash {
     margin-top: 0px;
      color: #333333;
      background-color: #E8E8E8;
      border-radius: 3px;
      font-size: 0.80em;
      font-family: 'Monaco', 'PT Mono', monospace;
      overflow-x: scroll !important;
      border-left: 6px solid #B22271;
      border-top: 1px solid #B7B7B7;
      border-right: 1px solid #B7B7B7;
      border-bottom: 1px solid #B7B7B7;
      padding: 18px 0px 18px 20px;
    }
  /** Controls code output **/

  d-article pre {
     margin-top: 0px;
      color: #333333;
      background-color: #FAFAFA;
      font-size: 0.8em;
      border: 1px solid #ccc;
      overflow-x: scroll !important;
      padding:  10px 10px 10px 18px;
      border-radius: 5px;
    }

  /** Controls inline code  **/

  p code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }

  li code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }

  d-article aside code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-family: 'Monaco', 'PT Mono', monospace;
    color: #BF2727;
  }

  /********************************/
  /* controls NAV BAR and HEADERS*/
  /********************************/

  .nav-dropdown-content .nav-dropdown-header {
    text-transform: uppercase;
    color: #686868;
  }

  .nav-dropdown-content a {
    font-family: 'Work Sans'
  }
  .nav-dropdown .nav-dropbtn {
    color: #FEFEFE;
    font-weight: 400;
  }

  .distill-site-header .logo img {
    margin-top: 10px;
    max-height: 30px;
    margin-bottom: auto;
  }

  .distill-site-nav {
      color: #FEFEFE;
      background-color: #113858;
      font-size: 0.95em;
      font-weight: 400;
      min-width: 165px;
      position: absolute;
      font-family: 'Karla';
      letter-spacing: .1em;
      text-transform: uppercase;
  }

  img {
    margin: 5px;
  }

  /** LIGHTBOX MARKUP **/
  .thumbnail {
    max-width: 100%;
    position: relative;
  }

  /** DETAILS fold down**/

  details {
    background-color: #f6f6f6;
    cursor: pointer;
    color: #2d2d2d;
    margin-bottom:1vh;
    padding: 1em;
    border: none;
    text-align: left;
    outline: none;
  }

  .active, details:hover {
    transition: max-height 0.2s ease-out;
    outline: none;

    /* background-color: #acc2c2; */
  }

  details {
    display: flex;
    justify-content: space-between;
    align-content: center;
    border-radius: 4px;
    outline: none;
  }

  details.active {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    outline: none;
  }

  summary::-webkit-details-marker {
    display: none;
    outline: none;
  }

  summary:focus {
    outline: none;
  }

  summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f0fe";
  }

  details[open] summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f146";
  }

  /********************************/
  /* controls downloadthis  */
  /********************************/

  button.btn.btn-default {
    font-size: 0.8em;
    font-family: "PT Sans";
    border-color: #B22271;
    border-width: medium;
    margin-bottom: 15px;
  }

  /********************************/
  /* controls reactable  */
  /********************************/

  div.rt-text-content {
    font-size: 1.2em;
  }

  /********************************/
  /* next and previous buttons at the end of the article */
  /********************************/

  .btnnav {
    border: none;
    background-color: #113858;
    color: #f2f2f2;
    padding: 14px 28px;
    font-size: 16px;
    cursor: pointer;
    display: inline-block;
  }

  .button_round {border-radius: 12px;}

  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"6. Beta Diversity & Dispersion Estimates","description":"Reproducible workflow for assessing community dissimilarity across temperature treatments.","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a class="logo" href="index.html">
<img src="assets/icon.png" alt="Logo"/>
</a>
<a href="index.html" class="title"></a>
</div>
<div class="nav-right">
<a href="index.html">
<i class="fa fa-home" aria-hidden="true"></i>
</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
A. Processing
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">Workflows</span>
<hr/>
<a href="dada2.html">1. DADA2</a>
<a href="data-prep.html">2. Data Set Prep</a>
<a href="filtering.html">3. Filtering</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
B. Diversity
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">Workflows</span>
<hr/>
<a href="taxa.html">4. Taxonomic</a>
<a href="alpha.html">5. Alpha Diversity</a>
<a href="beta.html">6. Beta Diversity</a>
<a href="da.html">7. DA ASVs &amp; Taxa</a>
<a href="metadata.html">8. Metadata</a>
</div>
</div>
<a href="data-availability.html">Data</a>
<a href="som.html">SOM</a>
<a href="https://github.com/sweltr/high-temp/">
<i class="fa fa-github fa-lg" aria-hidden="true"></i>
</a>
<a href="about.html">
<i class="fa fa-question" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>6. Beta Diversity &amp; Dispersion Estimates</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->
<p><p>Reproducible workflow for assessing community dissimilarity across
temperature treatments.</p></p>
</div>


<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#synopsis">Synopsis</a>
<ul>
<li><a href="#workflow-input">Workflow Input</a></li>
</ul></li>
<li><a href="#s-rrna">16s rRNA</a>
<ul>
<li><a href="#beta-dispersion">Beta Dispersion</a></li>
<li><a href="#significance-tests">Significance Tests</a></li>
<li><a href="#summaries">Summaries</a></li>
</ul></li>
<li><a href="#its">ITS</a>
<ul>
<li><a href="#beta-dispersion-1">Beta Dispersion</a></li>
<li><a href="#significance-tests-1">Significance Tests</a></li>
<li><a href="#summaries-1">Summaries</a></li>
</ul></li>
<li><a href="#ordinations">Ordinations</a>
<ul>
<li><a href="#s-rrna-1">16S rRNA</a></li>
<li><a href="#its-1">ITS</a></li>
</ul></li>
<li><a href="#within-between-group-distances">Within &amp; Between Group
Distances</a>
<ul>
<li><a href="#s-rrna-2">16S rRNA</a></li>
<li><a href="#its-2">ITS</a></li>
</ul></li>
<li><a href="#workflow-output">Workflow Output</a>
<ul>
<li><a href="#source-code">Source Code</a></li>
<li><a href="#data-availability">Data Availability</a></li>
<li><a href="#last-updated-on">Last updated on</a></li>
</ul></li>
</ul>
</nav>
</div>
<details>
<summary>
<strong>Click here</strong> for setup information.
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">echo =</span> <span class="cn">TRUE</span>, <span class="at">eval =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">119</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq); <span class="fu">packageVersion</span>(<span class="st">&quot;phyloseq&quot;</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings); <span class="fu">packageVersion</span>(<span class="st">&quot;Biostrings&quot;</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(tidyverse, DT, ampvis2, hilldiv, </span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>               microbiome, phytools, phangorn, reactable, Matrix, </span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>               pairwiseAdonis, naniar, downloadthis,</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>               labdsv, patchwork, agricolae, captioner, microeco,</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">install =</span> <span class="cn">FALSE</span>, <span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends(agricolae, local = TRUE)  </span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends_reverse(agricolae, local = TRUE)  </span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">scipen=</span><span class="dv">999</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_current<span class="sc">$</span><span class="fu">get</span>(<span class="fu">c</span>(</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache&quot;</span>,</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.path&quot;</span>,</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.rebuild&quot;</span>,</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;dependson&quot;</span>,</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;autodep&quot;</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>))</span></code></pre></div>
</div>
</details>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="synopsis">Synopsis</h1>
<p>This workflow contains beta diversity assessments for the high
temperature data sets. In order to run the workflow, you either need to
first run the <a href="dada2.html">DADA2 Workflow for 2018 High Temp
samples</a>, the <a href="data-prep.html">Data Preparation workflow</a>,
and the <a href="alpha.html">Alpha diversity workflow</a>
<strong>or</strong> download the files linked below. See the <a
href="data-availability.html">Data Availability</a> page for complete
details.</p>
<h2 id="workflow-input">Workflow Input</h2>
<p>Files needed to run this workflow can be downloaded from
figshare.</p>
<iframe src="https://widgets.figshare.com/articles/16826779/embed?show_title=1" width="568" height="251" allowfullscreen frameborder="0">
</iframe>
<h1 id="s-rrna">16s rRNA</h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>In order to test for significance between sample groups, we must
perform the following steps:</p>
<ol type="1">
<li>Transform sample counts to relative abundance.</li>
<li>Create distance matrices based on some metrics, for example
<code>wunifrac</code>, <code>unifrac</code>, and <code>jsd</code>. For
this we use the function <code>phyloseq::distance</code>.</li>
<li>Next, calculate beta dispersion using the <code>betadisper</code>
function from the <code>vegan</code> package <span class="citation"
data-cites="oksanen2013community">(<a href="#ref-oksanen2013community"
role="doc-biblioref">Oksanen et al. 2012</a>)</span>.</li>
<li>Then, use the function <code>permutest</code> to run a Permutation
test for homogeneity of multivariate dispersions.</li>
<li>If the beta dispersion tests are not significant we will run a
PERMANOVA using <code>adonis</code> (since PERMANOVA assumes equal
dispersion), otherwise we will use Analysis of Similarity (ANOSIM), both
available in the <code>vegan</code> package <span class="citation"
data-cites="oksanen2013community">(<a href="#ref-oksanen2013community"
role="doc-biblioref">Oksanen et al. 2012</a>)</span>.</li>
</ol>
<p>Steps 2–4 are combined in a <code>for</code> loop that tests all
three distance metrics.</p>
<p>Before we begin, we need to first transform sample counts to relative
abundance and generate some new trees.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>ssu_samp_ps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ssu18_ps_work&quot;</span>, <span class="st">&quot;ssu18_ps_filt&quot;</span>, </span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;ssu18_ps_perfect&quot;</span>, <span class="st">&quot;ssu18_ps_pime&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu_samp_ps) {</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>))</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(tmp_get, <span class="cf">function</span>(otu) </span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>                                          <span class="fl">1e5</span> <span class="sc">*</span> otu<span class="sc">/</span><span class="fu">sum</span>(otu))</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>     tmp_ps<span class="sc">@</span>phy_tree <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">prune_samples</span>(<span class="fu">sample_sums</span>(tmp_ps) <span class="sc">&gt;</span> <span class="dv">0</span>, tmp_ps)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>     tmp_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(tmp_ps), <span class="at">rooted =</span> <span class="cn">TRUE</span>, </span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>                       <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(tmp_ps))</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(tmp_ps, sample_data, tmp_tree)</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_name)</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_ps)</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<h2 id="beta-dispersion">Beta Dispersion</h2>
<p>Here we create distance matrices for each metric, calculate the beta
dispersion, and run a permutation test for homogeneity of multivariate
dispersions.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>ssu_dist <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;jsd&quot;</span>, <span class="st">&quot;unifrac&quot;</span>, <span class="st">&quot;wunifrac&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu_samp_ps) {</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (d <span class="cf">in</span> ssu_dist){</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>       tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_prop&quot;</span>)))</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>       tmp_samp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>       tmp_df <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(tmp_get, <span class="at">method =</span> d)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>       tmp_df_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_beta_dist_&quot;</span>, .))</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_df_name, tmp_df)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>       tmp_df2 <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(tmp_df, tmp_samp<span class="sc">$</span>TEMP, <span class="at">bias.adjust =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>       tmp_df_name2 <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_beta_dispersion_&quot;</span>, .))</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_df_name2, tmp_df2)</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>       tmp_df3 <span class="ot">&lt;-</span> <span class="fu">permutest</span>(tmp_df2, <span class="at">pairwise =</span> <span class="cn">TRUE</span>,</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>                            <span class="at">permutations =</span> <span class="dv">1000</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>       tmp_df_name3 <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_permutest_&quot;</span>, .))</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_df_name3, tmp_df3)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span></code></pre></div>
</details>
</div>
<details>
<summary>
<strong>Detailed results</strong> of Beta Dispersion &amp; Permutation
tests
</summary>
<div class="l-body">
<div class="panelset">
<div class="layout-chunk" data-layout="l-body">

</div>
<section id="full" class="panel">
<h4>FULL</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.018639 0.0093195 2.9882   1000 0.1089
Residuals 12 0.037425 0.0031188                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3      8
0          0.465534 0.1638
3 0.412469          0.0599
8 0.159005 0.075347       

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq    Mean Sq      F N.Perm  Pr(&gt;F)   
Groups     2 0.0062292 0.00311458 5.3484   1000 0.00999 **
Residuals 12 0.0069881 0.00058234                         
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3     8
0          0.831169 0.012
3 0.727926          0.032
8 0.033203 0.048150      

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm   Pr(&gt;F)   
Groups     2 0.018783 0.0093917 7.3178   1000 0.005994 **
Residuals 12 0.015401 0.0012834                          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3     8
0          0.957043 0.015
3 0.938214          0.004
8 0.024555 0.019472      </code></pre>
</div>
</section>
<section id="arbitrary" class="panel">
<h4>Arbitrary</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm  Pr(&gt;F)  
Groups     2 0.019867 0.0099333 3.0824   1000 0.08891 .
Residuals 12 0.038671 0.0032226                        
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3      8
0          0.854146 0.0859
3 0.843567          0.0869
8 0.112793 0.090799       

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm   Pr(&gt;F)   
Groups     2 0.025430 0.0127152 7.0788   1000 0.005994 **
Residuals 12 0.021555 0.0017962                          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3     8
0          0.583417 0.011
3 0.568432          0.018
8 0.018174 0.023627      

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm   Pr(&gt;F)   
Groups     2 0.023408 0.0117041 6.7591   1000 0.004995 **
Residuals 12 0.020779 0.0017316                          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3     8
0          0.565435 0.008
3 0.528192          0.013
8 0.023244 0.032852      </code></pre>
</div>
</section>
<section id="perfect" class="panel">
<h4>PERfect</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm  Pr(&gt;F)  
Groups     2 0.027097 0.0135487 4.5971   1000 0.02098 *
Residuals 12 0.035367 0.0029472                        
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3     8
0          0.778222 0.044
3 0.771551          0.027
8 0.061934 0.045555      

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm   Pr(&gt;F)   
Groups     2 0.034618 0.0173088 9.4781   1000 0.003996 **
Residuals 12 0.021914 0.0018262                          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
          0         3     8
0           0.5684316 0.008
3 0.5529721           0.008
8 0.0088715 0.0096445      

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm   Pr(&gt;F)   
Groups     2 0.020738 0.0103692 7.5898   1000 0.002997 **
Residuals 12 0.016395 0.0013662                          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3     8
0          0.547453 0.005
3 0.516927          0.003
8 0.027061 0.017081      </code></pre>
</div>
</section>
<section id="pime" class="panel">
<h4>PIME</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.007284 0.0036418 0.9199   1000 0.4356
Residuals 12 0.047509 0.0039591                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.84515 0.3876
3 0.82435         0.3407
8 0.37620 0.31969       

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.016868 0.0084341 1.4936   1000 0.2398
Residuals 12 0.067761 0.0056468                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.97902 0.2617
3 0.97444         0.2058
8 0.24180 0.21934       

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm   Pr(&gt;F)   
Groups     2 0.025802 0.0129010 8.6861   1000 0.008991 **
Residuals 12 0.017823 0.0014852                          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3     8
0          0.982018 0.006
3 0.982289          0.004
8 0.015248 0.014089      </code></pre>
</div>
</section>
</div>
</div>
</details>
<p>Remember, if the beta dispersion p-value is greater than
<code>0.05</code> we use PERMANOVA, otherwise we use ANOSIM.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><small><strong>(16S rRNA) Table 1 | Summary of Beta Dispersion Tests
for unfiltered &amp; filtered data sets against various distance
metrics. The <code>test</code> column shows the method chosen (based on
the test results) to assess differences in beta diversity. If p-values
are significant (red, p-value &lt; 0.05) we used ANOSIM, otherwise we
used PERMANOVA (blue, p-value &gt; 0.05).</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a onclick="fetch(&#39;data:text/csv;base64,ZGF0YXNldCxkaXN0LHB2YWwsdGVzdApGVUxMLEplbnNlbuKAk1NoYW5ub24gZGl2ZXJnZW5jZSwwLjEwODksQURPTklTCkZVTEwsdW53ZWlnaHRlZCBVbmlGcmFjLDAuMDEsQU5PU0lNCkZVTEwsd2VpZ2h0ZWQgVW5pRnJhYywwLjAwNixBTk9TSU0KRklMVCxKZW5zZW7igJNTaGFubm9uIGRpdmVyZ2VuY2UsMC4wODg5LEFET05JUwpGSUxULHVud2VpZ2h0ZWQgVW5pRnJhYywwLjAwNixBTk9TSU0KRklMVCx3ZWlnaHRlZCBVbmlGcmFjLDAuMDA1LEFOT1NJTQpQRVJmZWN0LEplbnNlbuKAk1NoYW5ub24gZGl2ZXJnZW5jZSwwLjAyMSxBTk9TSU0KUEVSZmVjdCx1bndlaWdodGVkIFVuaUZyYWMsMC4wMDQsQU5PU0lNClBFUmZlY3Qsd2VpZ2h0ZWQgVW5pRnJhYywwLjAwMyxBTk9TSU0KUElNRSxKZW5zZW7igJNTaGFubm9uIGRpdmVyZ2VuY2UsMC40MzU2LEFET05JUwpQSU1FLHVud2VpZ2h0ZWQgVW5pRnJhYywwLjIzOTgsQURPTklTClBJTUUsd2VpZ2h0ZWQgVW5pRnJhYywwLjAwOSxBTk9TSU0K&#39;).then(res =&gt; res.blob()).then(blob =&gt; {&#10;      const downloadURL = window.URL.createObjectURL(blob);&#10;      const a = document.createElement(&#39;a&#39;);&#10;      document.body.appendChild(a);&#10;      a.href = downloadURL;&#10;      a.download = &#39;ssu_beta_disp_tests.csv&#39;; a.click();&#10;      window.URL.revokeObjectURL(downloadURL);&#10;      document.body.removeChild(a);&#10;    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-fc5f7b00899a1383a230" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-fc5f7b00899a1383a230">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"dataset":["FULL","FULL","FULL","FILT","FILT","FILT","PERfect","PERfect","PERfect","PIME","PIME","PIME"],"dist":["Jensen–Shannon divergence","unweighted UniFrac","weighted UniFrac","Jensen–Shannon divergence","unweighted UniFrac","weighted UniFrac","Jensen–Shannon divergence","unweighted UniFrac","weighted UniFrac","Jensen–Shannon divergence","unweighted UniFrac","weighted UniFrac"],"pval":[0.1089,0.01,0.006,0.0889,0.006,0.005,0.021,0.004,0.003,0.4356,0.2398,0.009],"test":["ADONIS","ANOSIM","ANOSIM","ADONIS","ANOSIM","ANOSIM","ANOSIM","ANOSIM","ANOSIM","ADONIS","ADONIS","ANOSIM"]},"columns":[{"accessor":"dataset","name":"Dataset","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["FULL","FULL","FULL","FILT","FILT","FILT","PERfect","PERfect","PERfect","PIME","PIME","PIME"],"header":"Dataset","minWidth":50,"align":"left","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"dist","name":"distance","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["Jensen–Shannon divergence","unweighted UniFrac","weighted UniFrac","Jensen–Shannon divergence","unweighted UniFrac","weighted UniFrac","Jensen–Shannon divergence","unweighted UniFrac","weighted UniFrac","Jensen–Shannon divergence","unweighted UniFrac","weighted UniFrac"],"header":"distance","minWidth":100,"align":"left","headerVAlign":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"pval","name":"p-value","type":"numeric","sortable":true,"resizable":false,"filterable":true,"cell":["0.1089","0.0100","0.0060","0.0889","0.0060","0.0050","0.0210","0.0040","0.0030","0.4356","0.2398","0.0090"],"header":"p-value","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}]},{"accessor":"test","name":"test","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["ADONIS","ANOSIM","ANOSIM","ADONIS","ANOSIM","ANOSIM","ANOSIM","ANOSIM","ANOSIM","ADONIS","ADONIS","ANOSIM"],"header":"test","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"}}],"defaultPageSize":12,"showPageSizeOptions":true,"pageSizeOptions":[5,10,12],"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"125489211532ae68fedb7a51e9633aab"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
<p>Now we to create sample data frames for each data set.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu_samp_ps) {</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>)))</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_sampledf&quot;</span>))</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_samp)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern=</span><span class="st">&quot;_sampledf&quot;</span>)</span></code></pre></div>
</details>
</div>
<h2 id="significance-tests">Significance Tests</h2>
<p>Depending on the test, we need different data structures. For
<code>adonis</code> we use the sample metadata and for
<code>anosim</code> we will start with a phyloseq object. So for each
data set we create two variables: <code>adonis_sampledf</code> and
<code>anosim_data</code></p>
<div class="l-body">
<div class="panelset">
<section id="full-1" class="panel">
<h4>FULL</h4>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Is the p-value of the jsd distance metric less than
<code>0.05</code>? FALSE.<br />
Then we use <strong>ADONIS</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_adonis_jsd <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(ssu18_ps_work_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_adonis2_jsd <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(ssu18_ps_work_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<p>Is the p-value of the unifrac distance metric less than
<code>0.05</code>? TRUE.<br />
Then we use <strong>ANOSIM</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_anosim_unifrac <span class="ot">&lt;-</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_work_prop, <span class="st">&quot;unifrac&quot;</span>),</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_work_groups)</span></code></pre></div>
</div>
<p>Is the p-value of the wunifrac distance metric less than
<code>0.05</code>? TRUE.<br />
Then we use <strong>ANOSIM</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_anosim_wunifrac <span class="ot">&lt;-</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_work_prop, <span class="st">&quot;wunifrac&quot;</span>),</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_work_groups)</span></code></pre></div>
</div>
</section>
<section id="arbitrary-1" class="panel">
<h4>Arbitrary</h4>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Is the p-value of the jsd distance metric less than
<code>0.05</code>? FALSE.<br />
Then we use <strong>ADONIS</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_filt_adonis_jsd <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(ssu18_ps_filt_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>ssu18_ps_filt_adonis2_jsd <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(ssu18_ps_filt_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<p>Is the p-value of the unifrac distance metric less than
<code>0.05</code>? TRUE.<br />
Then we use <strong>ANOSIM</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_filt_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_filt_anosim_unifrac <span class="ot">&lt;-</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_filt_prop, <span class="st">&quot;unifrac&quot;</span>),</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_filt_groups)</span></code></pre></div>
</div>
<p>Is the p-value of the wunifrac distance metric less than
<code>0.05</code>? TRUE.<br />
Then we use <strong>ANOSIM</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_filt_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_filt_anosim_wunifrac <span class="ot">&lt;-</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_filt_prop, <span class="st">&quot;wunifrac&quot;</span>),</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_filt_groups)</span></code></pre></div>
</div>
</section>
<section id="perfect-1" class="panel">
<h4>PERfect</h4>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Is the p-value of the jsd distance metric less than
<code>0.05</code>? TRUE.<br />
Then we use <strong>ANOSIM</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_anosim_jsd <span class="ot">&lt;-</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_perfect_prop, <span class="st">&quot;jsd&quot;</span>),</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_perfect_groups)</span></code></pre></div>
</div>
<p>Is the p-value of the unifrac distance metric less than
<code>0.05</code>? TRUE.<br />
Then we use <strong>ANOSIM</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_anosim_unifrac <span class="ot">&lt;-</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_perfect_prop, <span class="st">&quot;unifrac&quot;</span>),</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_perfect_groups)</span></code></pre></div>
</div>
<p>Is the p-value of the wunifrac distance metric less than
<code>0.05</code>? TRUE.<br />
Then we use <strong>ANOSIM</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_anosim_wunifrac <span class="ot">&lt;-</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_perfect_prop, <span class="st">&quot;wunifrac&quot;</span>),</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_perfect_groups)</span></code></pre></div>
</div>
</section>
<section id="pime-1" class="panel">
<h4>PIME</h4>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Is the p-value of the jsd distance metric less than
<code>0.05</code>? FALSE.<br />
Then we use <strong>ADONIS</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_adonis_jsd <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(ssu18_ps_pime_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_adonis2_jsd <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(ssu18_ps_pime_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<p>Is the p-value of the unifrac distance metric less than
<code>0.05</code>? FALSE.<br />
Then we use <strong>ADONIS</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_adonis_unifrac <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(ssu18_ps_pime_beta_dist_unifrac <span class="sc">~</span> TEMP,</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_adonis2_unifrac <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(ssu18_ps_pime_beta_dist_unifrac <span class="sc">~</span> TEMP,</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<p>Is the p-value of the wunifrac distance metric less than
<code>0.05</code>? TRUE.<br />
Then we use <strong>ANOSIM</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_anosim_wunifrac <span class="ot">&lt;-</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(ssu18_ps_pime_prop, <span class="st">&quot;wunifrac&quot;</span>),</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> ssu18_ps_pime_groups)</span></code></pre></div>
</div>
</section>
</div>
</div>
<details>
<summary>
<strong>Detailed results</strong> of Significance tests
</summary>
<div class="l-body">
<div class="panelset">
<section id="full-2" class="panel">
<h4>FULL</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***PERMANOVA for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = ssu18_ps_work_beta_dist_jsd ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)   
TEMP      2  0.21809 0.23826 1.8767 0.003996 **
Residual 12  0.69727 0.76174                   
Total    14  0.91536 1.00000                   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***ANOSIM for Unweighted UniFrac distance, `unifrac`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_work_prop, &quot;unifrac&quot;),      grouping = ssu18_ps_work_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.2356 
      Significance: 0.003 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
   90%    95%  97.5%    99% 
0.0873 0.1130 0.1396 0.1725 

Dissimilarity ranks between and within classes:
        0%   25%  50%   75% 100%  N
Between  1 32.50 58.0 81.00  103 75
0        2 10.25 17.5 33.25   56 10
3        4 14.75 26.5 41.00   65 10
8       39 69.75 85.5 94.25  105 10</code></pre>
<pre><code>
 ***ANOSIM for Weighted-UniFrac distance, `wunifrac`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_work_prop, &quot;wunifrac&quot;),      grouping = ssu18_ps_work_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.3218 
      Significance: 0.001 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.090 0.120 0.149 0.176 

Dissimilarity ranks between and within classes:
        0%   25%  50%    75% 100%  N
Between  2 36.50 59.0  81.50  102 75
0        3  9.75 20.5  35.00   50 10
3        1 14.50 19.0  27.75   48 10
8       17 66.75 80.5 102.00  105 10</code></pre>
</div>
</section>
<section id="arbitrary-2" class="panel">
<h4>Arbitrary</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***PERMANOVA for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = ssu18_ps_filt_beta_dist_jsd ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F  Pr(&gt;F)   
TEMP      2  0.15050 0.29049 2.4566 0.00999 **
Residual 12  0.36758 0.70951                  
Total    14  0.51808 1.00000                  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***ANOSIM for Unweighted UniFrac distance, `unifrac`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_filt_prop, &quot;unifrac&quot;),      grouping = ssu18_ps_filt_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.2427 
      Significance: 0.009 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.109 0.136 0.169 0.225 

Dissimilarity ranks between and within classes:
        0%   25%  50%   75% 100%  N
Between  2 35.00 58.0 81.50  102 75
0        1  7.75 17.5 34.75   59 10
3       13 23.00 28.5 34.50   57 10
8       11 66.50 81.5 99.50  105 10</code></pre>
<pre><code>
 ***ANOSIM for Weighted-UniFrac distance, `wunifrac`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_filt_prop, &quot;wunifrac&quot;),      grouping = ssu18_ps_filt_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.3618 
      Significance: 0.001 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.103 0.140 0.173 0.216 

Dissimilarity ranks between and within classes:
        0%   25% 50%   75% 100%  N
Between  3 37.50  61 82.50  104 75
0        1  7.25  14 27.25   36 10
3        2 13.00  27 37.75   43 10
8       23 59.25  76 99.00  105 10</code></pre>
</div>
</section>
<section id="perfect-2" class="panel">
<h4>PERfect</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***ANOSIM for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_perfect_prop, &quot;jsd&quot;),      grouping = ssu18_ps_perfect_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.2996 
      Significance: 0.001 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
   90%    95%  97.5%    99% 
0.0999 0.1450 0.1770 0.2081 

Dissimilarity ranks between and within classes:
        0%  25%  50%   75% 100%  N
Between  2 33.5 61.0 83.50  104 75
0        5 15.5 19.5 44.75   59 10
3        1  9.5 23.0 29.75   57 10
8       39 53.5 75.0 95.50  105 10</code></pre>
<pre><code>
 ***ANOSIM for Unweighted UniFrac distance, `unifrac`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_perfect_prop, &quot;unifrac&quot;),      grouping = ssu18_ps_perfect_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.2053 
      Significance: 0.007 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
   90%    95%  97.5%    99% 
0.0908 0.1191 0.1441 0.1814 

Dissimilarity ranks between and within classes:
        0%   25% 50%    75% 100%  N
Between  5 31.00  59  81.50  102 75
0        1  4.75  14  36.25   56 10
3        4 21.75  34  43.50   53 10
8       52 70.50  80 100.00  105 10</code></pre>
<pre><code>
 ***ANOSIM for Weighted-UniFrac distance, `wunifrac`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_perfect_prop, &quot;wunifrac&quot;),      grouping = ssu18_ps_perfect_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.3378 
      Significance: 0.001 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
   90%    95%  97.5%    99% 
0.0942 0.1213 0.1538 0.1831 

Dissimilarity ranks between and within classes:
        0%   25%  50%   75% 100%  N
Between  2 36.00 61.0 80.50  105 75
0        1 20.00 26.5 35.50   40 10
3        3  6.25 15.5 20.75   42 10
8       14 61.00 92.5 99.25  104 10</code></pre>
</div>
</section>
<section id="pime-2" class="panel">
<h4>PIME</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***PERMANOVA for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = ssu18_ps_pime_beta_dist_jsd ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
TEMP      2  0.23994 0.59595 8.8496 0.000999 ***
Residual 12  0.16268 0.40405                    
Total    14  0.40262 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***PERMANOVA for Unweighted UniFrac distance, `unifrac`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = ssu18_ps_pime_beta_dist_unifrac ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
TEMP      2  0.78789 0.64853 11.071 0.000999 ***
Residual 12  0.42699 0.35147                    
Total    14  1.21488 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***ANOSIM for Weighted-UniFrac distance, `wunifrac`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(ssu18_ps_pime_prop, &quot;wunifrac&quot;),      grouping = ssu18_ps_pime_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.5316 
      Significance: 0.001 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.114 0.157 0.207 0.262 

Dissimilarity ranks between and within classes:
        0%   25% 50%  75% 100%  N
Between  9 41.00  62 85.5  105 75
0        1  8.75  17 28.5   37 10
3        2  5.50  17 26.5   39 10
8        4 50.50  73 81.0   97 10</code></pre>
</div>
</section>
</div>
</div>
</details>
<h2 id="summaries">Summaries</h2>
<p>Here is a quick summary of significance tests for the data sets
against three distance matrices.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><small><strong>(16S rRNA) Table 2 | Summary of significant tests for
the unfiltered &amp; filtered data sets. P-values in red indicate
significant differences (p-value &lt; 0.05).</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a onclick="fetch(&#39;data:text/csv;base64,ZGlzdGFuY2UgbWV0cmljLEZVTEwsRklMVCxQRVJmZWN0LFBJTUUKSmVuc2VuLVNoYW5ub24gRGl2ZXJnZW5jZSwwLjAwNiwwLjAxLDAuMDAxLDAuMDAxCnVud2VpZ2h0ZWQgVW5pRnJhYywwLjAwMywwLjAwOSwwLjAwNywwLjAwMQp3ZWlnaHRlZCBVbmlGcmFjLDAuMDAxLDAuMDAxLDAuMDAxLDAuMDAxCg==&#39;).then(res =&gt; res.blob()).then(blob =&gt; {&#10;      const downloadURL = window.URL.createObjectURL(blob);&#10;      const a = document.createElement(&#39;a&#39;);&#10;      document.body.appendChild(a);&#10;      a.href = downloadURL;&#10;      a.download = &#39;ssu_sign_tests.csv&#39;; a.click();&#10;      window.URL.revokeObjectURL(downloadURL);&#10;      document.body.removeChild(a);&#10;    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-cecec87b5299f1f465f5" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-cecec87b5299f1f465f5">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"distance metric":["Jensen-Shannon Divergence","unweighted UniFrac","weighted UniFrac"],"FULL":[0.006,0.003,0.001],"FILT":[0.01,0.009,0.001],"PERfect":[0.001,0.007,0.001],"PIME":[0.001,0.001,0.001]},"columns":[{"accessor":"distance metric","name":"Distance metric","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["Jensen-Shannon Divergence","unweighted UniFrac","weighted UniFrac"],"header":"Distance metric","minWidth":200,"align":"left","headerVAlign":"center","style":{"borderRight":"1px solid #eee"},"footerStyle":{"fontWeight":"bold"},"sticky":"left","headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"FULL","name":"FULL","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.006","0.003","0.001"],"header":"FULL","minWidth":100,"align":"center","headerVAlign":"center","style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}],"footerStyle":{"fontWeight":"bold"}},{"accessor":"FILT","name":"FILT","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.010","0.009","0.001"],"header":"FILT","minWidth":100,"align":"center","headerVAlign":"center","style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}],"footerStyle":{"fontWeight":"bold"}},{"accessor":"PERfect","name":"PERfect","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.001","0.007","0.001"],"header":"PERfect","minWidth":100,"align":"center","headerVAlign":"center","style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}],"footerStyle":{"fontWeight":"bold"}},{"accessor":"PIME","name":"PIME","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.001","0.001","0.001"],"header":"PIME","minWidth":100,"align":"center","headerVAlign":"center","style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}],"footerStyle":{"fontWeight":"bold"}}],"columnGroups":[{"name":"p-values","headerStyle":{"fontSize":"1.3em"},"columns":["FULL","FILT","PERfect","PIME"]}],"defaultPageSize":12,"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"theme":{"style":{"fontSize":"0.9em"}},"dataKey":"bd26b561fb9059d6b0e32a959b06d95c"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
<h1 id="its">ITS</h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>In order to test for significance between sample groups, we must
perform the following steps:</p>
<ol type="1">
<li>Transform sample counts to relative abundance.</li>
<li>Create distance matrices based on some metrics, for example
<code>gower</code>, <code>bray</code>, and <code>jsd</code>. For this we
use the function <code>phyloseq::distance</code>.</li>
<li>Next, calculate beta dispersion using the <code>betadisper</code>
function from the <code>vegan</code> package <span class="citation"
data-cites="oksanen2013community">(<a href="#ref-oksanen2013community"
role="doc-biblioref">Oksanen et al. 2012</a>)</span>.</li>
<li>Then, use the function <code>permutest</code> to run a Permutation
test for homogeneity of multivariate dispersions.</li>
<li>If the beta dispersion tests are not significant we will run a
PERMANOVA using <code>adonis</code> (since PERMANOVA aitsmes equal
dispersion), otherwise we will use Analysis of Similarity (ANOSIM), both
available in the <code>vegan</code> package <span class="citation"
data-cites="oksanen2013community">(<a href="#ref-oksanen2013community"
role="doc-biblioref">Oksanen et al. 2012</a>)</span>.</li>
</ol>
<p>Steps 2–4 are combined in a <code>for</code> loop that tests all
three distance metrics.</p>
<p>Before we begin, we need to first transform sample counts to relative
abundance and generate some new trees.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>its_samp_ps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;its18_ps_work&quot;</span>, <span class="st">&quot;its18_ps_filt&quot;</span>, </span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;its18_ps_perfect&quot;</span>, <span class="st">&quot;its18_ps_pime&quot;</span>)</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its_samp_ps) {</span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>))</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(tmp_get, <span class="cf">function</span>(otu) </span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>                                          <span class="fl">1e5</span> <span class="sc">*</span> otu<span class="sc">/</span><span class="fu">sum</span>(otu))</span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a>     tmp_ps<span class="sc">@</span>phy_tree <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">prune_samples</span>(<span class="fu">sample_sums</span>(tmp_ps) <span class="sc">&gt;</span> <span class="dv">0</span>, tmp_ps)</span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(tmp_ps, sample_data)</span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_name)</span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_ps)</span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<h3 id="beta-dispersion-1">Beta Dispersion</h3>
<p>Here we create distance matrices for each metric, calculate the beta
dispersion, and run a permutation test for homogeneity of multivariate
dispersions.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>its_dist <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;jsd&quot;</span>, <span class="st">&quot;bray&quot;</span>, <span class="st">&quot;gower&quot;</span>)</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its_samp_ps) {</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (d <span class="cf">in</span> its_dist){</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>       tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_prop&quot;</span>)))</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>       tmp_samp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>       tmp_df <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(tmp_get, <span class="at">method =</span> d)</span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>       tmp_df_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_beta_dist_&quot;</span>, .))</span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_df_name, tmp_df)</span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a>       tmp_df2 <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(tmp_df, tmp_samp<span class="sc">$</span>TEMP, <span class="at">bias.adjust =</span> <span class="cn">TRUE</span>)</span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a>       tmp_df_name2 <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_beta_dispersion_&quot;</span>, .))</span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_df_name2, tmp_df2)</span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a>       tmp_df3 <span class="ot">&lt;-</span> <span class="fu">permutest</span>(tmp_df2, <span class="at">pairwise =</span> <span class="cn">TRUE</span>,</span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a>                            <span class="at">permutations =</span> <span class="dv">1000</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>)</span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a>       tmp_df_name3 <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_permutest_&quot;</span>, .))</span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_df_name3, tmp_df3)</span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb46-18"><a href="#cb46-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb46-19"><a href="#cb46-19" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span></code></pre></div>
</details>
</div>
<details>
<summary>
<strong>Detailed results</strong> of Beta Dispersion &amp; Permutation
tests
</summary>
<div class="l-body">
<div class="panelset">
<div class="layout-chunk" data-layout="l-body">

</div>
<section id="full-3" class="panel">
<h4>FULL</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq   Mean Sq     F N.Perm Pr(&gt;F)
Groups     2 0.0016626 0.0008313 0.272   1000 0.7522
Residuals 10 0.0305600 0.0030560                    

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.65734 0.7612
3 0.65549         0.4925
8 0.75785 0.49503       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.002828 0.0014140 0.3342   1000 0.7093
Residuals 10 0.042306 0.0042306                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.51548 0.8501
3 0.53577         0.4525
8 0.85014 0.48472       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.0029899 0.0014950 1.3038   1000 0.3247
Residuals 10 0.0114662 0.0011466                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.78921 0.2058
3 0.75923         0.3506
8 0.22060 0.33265       </code></pre>
</div>
</section>
<section id="arbitrary-3" class="panel">
<h4>Arbitrary</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.001023 0.0005116 0.0745   1000 0.9101
Residuals 10 0.068640 0.0068640                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.68831 0.9301
3 0.67472         0.8062
8 0.92769 0.81555       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq  Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.002408 0.001204 0.1338   1000 0.8801
Residuals 10 0.090000 0.009000                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.58042 0.9061
3 0.58847         0.7582
8 0.90501 0.75350       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.0058376 0.0029188 2.2494   1000 0.1419
Residuals 10 0.0129758 0.0012976                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3      8
0          0.061938 0.4555
3 0.074044          0.2847
8 0.464903 0.264060       </code></pre>
</div>
</section>
<section id="perfect-3" class="panel">
<h4>PERfect</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.001059 0.0005293 0.1168   1000 0.8831
Residuals 10 0.045317 0.0045317                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.69231 0.9131
3 0.71118         0.6683
8 0.92002 0.67220       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.002522 0.0012611 0.2056   1000 0.8062
Residuals 10 0.061340 0.0061340                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.57443 0.9950
3 0.58316         0.6224
8 0.99819 0.62424       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.0023433 0.0011716 2.0201   1000 0.2138
Residuals 10 0.0058000 0.0005800                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         0        3      8
0          0.868132 0.1029
3 0.869601          0.2677
8 0.091826 0.229457       </code></pre>
</div>
</section>
<section id="pime-3" class="panel">
<h4>PIME</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.007678 0.0038389 0.3436   1000 0.7163
Residuals 10 0.111712 0.0111712                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.47253 0.4805
3 0.49362         0.9760
8 0.47799 0.97911       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     2 0.008276 0.0041378 0.2627   1000 0.7602
Residuals 10 0.157522 0.0157522                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        0       3      8
0         0.46753 0.5874
3 0.51164         0.8821
8 0.58541 0.88566       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq   Mean Sq      F N.Perm   Pr(&gt;F)   
Groups     2 0.0165873 0.0082936 15.047   1000 0.001998 **
Residuals 10 0.0055118 0.0005512                          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
           0          3     8
0            0.05594406 0.002
3 0.04179289            0.040
8 0.00072543 0.03074504      </code></pre>
</div>
</section>
</div>
</div>
</details>
<p>Remember, if the beta dispersion p-value is greater than
<code>0.05</code> we use PERMANOVA, otherwise we use ANOSIM.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><small><strong>(ITS) Table 1 | Summary of Beta Dispersion Tests for
unfiltered &amp; filtered data sets against various distance metrics.
The <code>test</code> column shows the method chosen (based on the test
results) to assess differences in beta diversity. If p-values are
significant (red, p-value &lt; 0.05) we used ANOSIM, otherwise we used
PERMANOVA (blue, p-value &gt; 0.05).</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a onclick="fetch(&#39;data:text/csv;base64,ZGF0YXNldCxkaXN0LHB2YWwsdGVzdApGVUxMLEplbnNlbuKAk1NoYW5ub24gZGl2ZXJnZW5jZSwwLjc1MjIsQURPTklTCkZVTEwsQnJheS1DdXJ0aXMgZGlzc2ltaWxhcml0eSwwLjcwOTMsQURPTklTCkZVTEwsR293ZXIgZGlzdGFuY2UsMC4zMjQ3LEFET05JUwpGSUxULEplbnNlbuKAk1NoYW5ub24gZGl2ZXJnZW5jZSwwLjkxMDEsQURPTklTCkZJTFQsQnJheS1DdXJ0aXMgZGlzc2ltaWxhcml0eSwwLjg4MDEsQURPTklTCkZJTFQsR293ZXIgZGlzdGFuY2UsMC4xNDE5LEFET05JUwpQRVJmZWN0LEplbnNlbuKAk1NoYW5ub24gZGl2ZXJnZW5jZSwwLjg4MzEsQURPTklTClBFUmZlY3QsQnJheS1DdXJ0aXMgZGlzc2ltaWxhcml0eSwwLjgwNjIsQURPTklTClBFUmZlY3QsR293ZXIgZGlzdGFuY2UsMC4yMTM4LEFET05JUwpQSU1FLEplbnNlbuKAk1NoYW5ub24gZGl2ZXJnZW5jZSwwLjcxNjMsQURPTklTClBJTUUsQnJheS1DdXJ0aXMgZGlzc2ltaWxhcml0eSwwLjc2MDIsQURPTklTClBJTUUsR293ZXIgZGlzdGFuY2UsMC4wMDIsQU5PU0lNCg==&#39;).then(res =&gt; res.blob()).then(blob =&gt; {&#10;      const downloadURL = window.URL.createObjectURL(blob);&#10;      const a = document.createElement(&#39;a&#39;);&#10;      document.body.appendChild(a);&#10;      a.href = downloadURL;&#10;      a.download = &#39;its_beta_disp_tests.csv&#39;; a.click();&#10;      window.URL.revokeObjectURL(downloadURL);&#10;      document.body.removeChild(a);&#10;    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-0206c46563879bcb8d51" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-0206c46563879bcb8d51">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"dataset":["FULL","FULL","FULL","FILT","FILT","FILT","PERfect","PERfect","PERfect","PIME","PIME","PIME"],"dist":["Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance"],"pval":[0.7522,0.7093,0.3247,0.9101,0.8801,0.1419,0.8831,0.8062,0.2138,0.7163,0.7602,0.002],"test":["ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ANOSIM"]},"columns":[{"accessor":"dataset","name":"Dataset","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["FULL","FULL","FULL","FILT","FILT","FILT","PERfect","PERfect","PERfect","PIME","PIME","PIME"],"header":"Dataset","minWidth":50,"align":"left","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"dist","name":"distance","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance"],"header":"distance","minWidth":100,"align":"left","headerVAlign":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"pval","name":"p-value","type":"numeric","sortable":true,"resizable":false,"filterable":true,"cell":["0.7522","0.7093","0.3247","0.9101","0.8801","0.1419","0.8831","0.8062","0.2138","0.7163","0.7602","0.0020"],"header":"p-value","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}]},{"accessor":"test","name":"test","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ANOSIM"],"header":"test","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"}}],"defaultPageSize":12,"showPageSizeOptions":true,"pageSizeOptions":[5,10,12],"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"0e627cd3ad28b39af93d6fd48b0e16b6"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
<p>Now we to create sample data frames for each data set.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its_samp_ps) {</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>)))</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_sampledf&quot;</span>))</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_samp)</span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern=</span><span class="st">&quot;_sampledf&quot;</span>)</span></code></pre></div>
</details>
</div>
<h2 id="significance-tests-1">Significance Tests</h2>
<p>Depending on the test, we need different data structures. For
<code>adonis</code> we use the sample metadata and for
<code>anosim</code> we will start with a phyloseq object. So for each
data set we create two variables: <code>adonis_sampledf</code> and
<code>anosim_data</code></p>
<div class="l-body">
<div class="panelset">
<section id="full-4" class="panel">
<h4>FULL</h4>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Is the p-value of the jsd distance metric less than
<code>0.05</code>? FALSE.<br />
Then we use <strong>ADONIS</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>its18_ps_work_adonis_jsd <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_work_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>its18_ps_work_adonis2_jsd <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_work_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<p>Is the p-value of the bray distance metric less than
<code>0.05</code>? FALSE.<br />
Then we use <strong>ADONIS</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>its18_ps_work_adonis_bray <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_work_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>its18_ps_work_adonis2_bray <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_work_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<p>Is the p-value of the gower distance metric less than
<code>0.05</code>? FALSE.<br />
Then we use <strong>ADONIS</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>its18_ps_work_adonis_gower <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_work_beta_dist_gower <span class="sc">~</span> TEMP,</span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>its18_ps_work_adonis2_gower <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_work_beta_dist_gower <span class="sc">~</span> TEMP,</span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
</section>
<section id="arbitrary-4" class="panel">
<h4>Arbitrary</h4>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Is the p-value of the jsd distance metric less than
<code>0.05</code>? FALSE.<br />
Then we use <strong>ADONIS</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>its18_ps_filt_adonis_jsd <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_filt_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>its18_ps_filt_adonis2_jsd <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_filt_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<p>Is the p-value of the bray distance metric less than
<code>0.05</code>? FALSE.<br />
Then we use <strong>ADONIS</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>its18_ps_filt_adonis_bray <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_filt_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>its18_ps_filt_adonis2_bray <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_filt_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<p>Is the p-value of the gower distance metric less than
<code>0.05</code>? FALSE.<br />
Then we use <strong>ADONIS</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>its18_ps_filt_adonis_gower <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_filt_beta_dist_gower <span class="sc">~</span> TEMP,</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>its18_ps_filt_adonis2_gower <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_filt_beta_dist_gower <span class="sc">~</span> TEMP,</span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
</section>
<section id="perfect-4" class="panel">
<h4>PERfect</h4>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Is the p-value of the jsd distance metric less than
<code>0.05</code>? FALSE.<br />
Then we use <strong>ADONIS</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect_adonis_jsd <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_perfect_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect_adonis2_jsd <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_perfect_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<p>Is the p-value of the bray distance metric less than
<code>0.05</code>? FALSE.<br />
Then we use <strong>ADONIS</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect_adonis_bray <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_perfect_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect_adonis2_bray <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_perfect_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<p>Is the p-value of the gower distance metric less than
<code>0.05</code>? FALSE.<br />
Then we use <strong>ADONIS</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect_adonis_gower <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_perfect_beta_dist_gower <span class="sc">~</span> TEMP,</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect_adonis2_gower <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_perfect_beta_dist_gower <span class="sc">~</span> TEMP,</span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
</section>
<section id="pime-4" class="panel">
<h4>PIME</h4>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Is the p-value of the jsd distance metric less than
<code>0.05</code>? FALSE.<br />
Then we use <strong>ADONIS</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_adonis_jsd <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_pime_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_adonis2_jsd <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_pime_beta_dist_jsd <span class="sc">~</span> TEMP,</span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<p>Is the p-value of the bray distance metric less than
<code>0.05</code>? FALSE.<br />
Then we use <strong>ADONIS</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_adonis_bray <span class="ot">&lt;-</span>  <span class="fu">adonis</span>(its18_ps_pime_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_adonis2_bray <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(its18_ps_pime_beta_dist_bray <span class="sc">~</span> TEMP,</span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<p>Is the p-value of the gower distance metric less than
<code>0.05</code>? TRUE.<br />
Then we use <strong>ANOSIM</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(anosim_data, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_anosim_gower <span class="ot">&lt;-</span></span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(its18_ps_pime_prop, <span class="st">&quot;gower&quot;</span>),</span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">grouping =</span> its18_ps_pime_groups)</span></code></pre></div>
</div>
</section>
</div>
</div>
<details>
<summary>
<strong>Detailed results</strong> of Significance tests.
</summary>
<div class="l-body">
<div class="panelset">
<section id="full-5" class="panel">
<h4>FULL</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***PERMANOVA for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = its18_ps_work_beta_dist_jsd ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs     R2      F  Pr(&gt;F)  
TEMP      2  0.38087 0.2287 1.4826 0.03796 *
Residual 10  1.28448 0.7713                 
Total    12  1.66534 1.0000                 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***PERMANOVA for Bray-Curtis dissimilarity, `bray`*** </code></pre>
<pre><code>       Df        SumOfSqs            R2               F        
 Min.   : 2   Min.   :0.9003   Min.   :0.2070   Min.   :1.305  
 1st Qu.: 6   1st Qu.:2.1744   1st Qu.:0.5000   1st Qu.:1.305  
 Median :10   Median :3.4484   Median :0.7930   Median :1.305  
 Mean   : 8   Mean   :2.8992   Mean   :0.6667   Mean   :1.305  
 3rd Qu.:11   3rd Qu.:3.8986   3rd Qu.:0.8965   3rd Qu.:1.305  
 Max.   :12   Max.   :4.3488   Max.   :1.0000   Max.   :1.305  
                                                NA&#39;s   :2      
     Pr(&gt;F)       
 Min.   :0.05095  
 1st Qu.:0.05095  
 Median :0.05095  
 Mean   :0.05095  
 3rd Qu.:0.05095  
 Max.   :0.05095  
 NA&#39;s   :2        </code></pre>
<pre><code>
 ***PERMANOVA for Gower distance, `gower`*** </code></pre>
<pre><code>       Df        SumOfSqs             R2               F        
 Min.   : 2   Min.   :0.05219   Min.   :0.2016   Min.   :1.263  
 1st Qu.: 6   1st Qu.:0.12941   1st Qu.:0.5000   1st Qu.:1.263  
 Median :10   Median :0.20664   Median :0.7984   Median :1.263  
 Mean   : 8   Mean   :0.17255   Mean   :0.6667   Mean   :1.263  
 3rd Qu.:11   3rd Qu.:0.23273   3rd Qu.:0.8992   3rd Qu.:1.263  
 Max.   :12   Max.   :0.25882   Max.   :1.0000   Max.   :1.263  
                                                 NA&#39;s   :2      
     Pr(&gt;F)       
 Min.   :0.00999  
 1st Qu.:0.00999  
 Median :0.00999  
 Mean   :0.00999  
 3rd Qu.:0.00999  
 Max.   :0.00999  
 NA&#39;s   :2        </code></pre>
</div>
</section>
<section id="arbitrary-5" class="panel">
<h4>Arbitrary</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***PERMANOVA for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = its18_ps_filt_beta_dist_jsd ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F  Pr(&gt;F)  
TEMP      2  0.36598 0.25398 1.7022 0.02498 *
Residual 10  1.07499 0.74602                 
Total    12  1.44097 1.00000                 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***PERMANOVA for Bray-Curtis dissimilarity, `bray`*** </code></pre>
<pre><code>       Df        SumOfSqs            R2               F        
 Min.   : 2   Min.   :0.8899   Min.   :0.2216   Min.   :1.423  
 1st Qu.: 6   1st Qu.:2.0078   1st Qu.:0.5000   1st Qu.:1.423  
 Median :10   Median :3.1257   Median :0.7784   Median :1.423  
 Mean   : 8   Mean   :2.6770   Mean   :0.6667   Mean   :1.423  
 3rd Qu.:11   3rd Qu.:3.5706   3rd Qu.:0.8892   3rd Qu.:1.423  
 Max.   :12   Max.   :4.0155   Max.   :1.0000   Max.   :1.423  
                                                NA&#39;s   :2      
     Pr(&gt;F)       
 Min.   :0.03996  
 1st Qu.:0.03996  
 Median :0.03996  
 Mean   :0.03996  
 3rd Qu.:0.03996  
 Max.   :0.03996  
 NA&#39;s   :2        </code></pre>
<pre><code>
 ***PERMANOVA for Gower distance, `gower`*** </code></pre>
<pre><code>       Df        SumOfSqs            R2               F        
 Min.   : 2   Min.   :0.1076   Min.   :0.2416   Min.   :1.593  
 1st Qu.: 6   1st Qu.:0.2225   1st Qu.:0.5000   1st Qu.:1.593  
 Median :10   Median :0.3375   Median :0.7584   Median :1.593  
 Mean   : 8   Mean   :0.2967   Mean   :0.6667   Mean   :1.593  
 3rd Qu.:11   3rd Qu.:0.3913   3rd Qu.:0.8792   3rd Qu.:1.593  
 Max.   :12   Max.   :0.4451   Max.   :1.0000   Max.   :1.593  
                                                NA&#39;s   :2      
     Pr(&gt;F)        
 Min.   :0.006993  
 1st Qu.:0.006993  
 Median :0.006993  
 Mean   :0.006993  
 3rd Qu.:0.006993  
 Max.   :0.006993  
 NA&#39;s   :2         </code></pre>
</div>
</section>
<section id="perfect-5" class="panel">
<h4>PERfect</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***PERMANOVA for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = its18_ps_perfect_beta_dist_jsd ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F  Pr(&gt;F)  
TEMP      2  0.38311 0.23194 1.5099 0.04296 *
Residual 10  1.26864 0.76806                 
Total    12  1.65176 1.00000                 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***PERMANOVA for Bray-Curtis dissimilarity, `bray`*** </code></pre>
<pre><code>       Df        SumOfSqs            R2               F        
 Min.   : 2   Min.   :0.8938   Min.   :0.2056   Min.   :1.294  
 1st Qu.: 6   1st Qu.:2.1734   1st Qu.:0.5000   1st Qu.:1.294  
 Median :10   Median :3.4530   Median :0.7944   Median :1.294  
 Mean   : 8   Mean   :2.8978   Mean   :0.6667   Mean   :1.294  
 3rd Qu.:11   3rd Qu.:3.8999   3rd Qu.:0.8972   3rd Qu.:1.294  
 Max.   :12   Max.   :4.3468   Max.   :1.0000   Max.   :1.294  
                                                NA&#39;s   :2      
     Pr(&gt;F)       
 Min.   :0.07293  
 1st Qu.:0.07293  
 Median :0.07293  
 Mean   :0.07293  
 3rd Qu.:0.07293  
 Max.   :0.07293  
 NA&#39;s   :2        </code></pre>
<pre><code>
 ***PERMANOVA for Gower distance, `gower`*** </code></pre>
<pre><code>       Df        SumOfSqs            R2               F        
 Min.   : 2   Min.   :0.0647   Min.   :0.2319   Min.   :1.509  
 1st Qu.: 6   1st Qu.:0.1395   1st Qu.:0.5000   1st Qu.:1.509  
 Median :10   Median :0.2143   Median :0.7681   Median :1.509  
 Mean   : 8   Mean   :0.1860   Mean   :0.6667   Mean   :1.509  
 3rd Qu.:11   3rd Qu.:0.2466   3rd Qu.:0.8841   3rd Qu.:1.509  
 Max.   :12   Max.   :0.2790   Max.   :1.0000   Max.   :1.509  
                                                NA&#39;s   :2      
     Pr(&gt;F)        
 Min.   :0.001998  
 1st Qu.:0.001998  
 Median :0.001998  
 Mean   :0.001998  
 3rd Qu.:0.001998  
 Max.   :0.001998  
 NA&#39;s   :2         </code></pre>
</div>
</section>
<section id="pime-5" class="panel">
<h4>PIME</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
 ***PERMANOVA for Jensen-Shannon Divergence, `jsd`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = its18_ps_pime_beta_dist_jsd ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs     R2      F   Pr(&gt;F)    
TEMP      2  0.54824 0.4002 3.3361 0.000999 ***
Residual 10  0.82167 0.5998                    
Total    12  1.36991 1.0000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***PERMANOVA for Bray-Curtis dissimilarity, `bray`*** </code></pre>
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = its18_ps_pime_beta_dist_bray ~ TEMP, data = adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2     F   Pr(&gt;F)    
TEMP      2   1.1998 0.30872 2.233 0.000999 ***
Residual 10   2.6865 0.69128                   
Total    12   3.8862 1.00000                   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>
 ***ANOSIM for Gower distance, `gower`*** </code></pre>
<pre><code>
Call:
anosim(x = phyloseq::distance(its18_ps_pime_prop, &quot;gower&quot;), grouping = its18_ps_pime_groups) 
Dissimilarity: gower 

ANOSIM statistic R: 0.6542 
      Significance: 0.001 

Permutation: free
Number of permutations: 999

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.151 0.221 0.265 0.346 

Dissimilarity ranks between and within classes:
        0%   25%  50%   75% 100%  N
Between  4 31.50 48.5 64.25   78 56
0       22 25.00 34.5 40.75   57 10
3        8 12.75 15.5 16.75   19  6
8        1  2.25  4.0  5.75    7  6</code></pre>
</div>
</section>
</div>
</div>
</details>
<h2 id="summaries-1">Summaries</h2>
<p>Here is a quick summary of significance tests for the data sets
against three distance matrices.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><small><strong>(ITS) Table 2 | Summary of significant tests for the
unfiltered &amp; filtered data sets. P-values in red indicate
significant differences (p-value &lt; 0.05).</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a onclick="fetch(&#39;data:text/csv;base64,ZGlzdGFuY2UgbWV0cmljLEZVTEwsRklMVCxQRVJmZWN0LFBJTUUKSmVuc2VuLVNoYW5ub24gRGl2ZXJnZW5jZSwwLjA0LDAuMDM3LDAuMDU5OSwwLjAwMQpCcmF5LUN1cnRpcyBkaXNzaW1pbGFyaXR5LDAuMDQ2LDAuMDMzLDAuMDgxOSwwLjAwMQpHb3dlciBkaXN0YW5jZSwwLjAxNCwwLjAwOCwwLjAwMiwwLjAwMQo=&#39;).then(res =&gt; res.blob()).then(blob =&gt; {&#10;      const downloadURL = window.URL.createObjectURL(blob);&#10;      const a = document.createElement(&#39;a&#39;);&#10;      document.body.appendChild(a);&#10;      a.href = downloadURL;&#10;      a.download = &#39;its_sign_tests.csv&#39;; a.click();&#10;      window.URL.revokeObjectURL(downloadURL);&#10;      document.body.removeChild(a);&#10;    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-b85f030edf0c79c9706a" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-b85f030edf0c79c9706a">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"distance metric":["Jensen-Shannon Divergence","Bray-Curtis dissimilarity","Gower distance"],"FULL":[0.04,0.046,0.014],"FILT":[0.037,0.033,0.008],"PERfect":[0.0599,0.0819,0.002],"PIME":[0.001,0.001,0.001]},"columns":[{"accessor":"distance metric","name":"Distance metric","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["Jensen-Shannon Divergence","Bray-Curtis dissimilarity","Gower distance"],"header":"Distance metric","minWidth":200,"align":"left","headerVAlign":"center","style":{"borderRight":"1px solid #eee"},"footerStyle":{"fontWeight":"bold"},"sticky":"left","headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"FULL","name":"FULL","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.040","0.046","0.014"],"header":"FULL","minWidth":100,"align":"center","headerVAlign":"center","style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}],"footerStyle":{"fontWeight":"bold"}},{"accessor":"FILT","name":"FILT","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.037","0.033","0.008"],"header":"FILT","minWidth":100,"align":"center","headerVAlign":"center","style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}],"footerStyle":{"fontWeight":"bold"}},{"accessor":"PERfect","name":"PERfect","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.0599","0.0819","0.002"],"header":"PERfect","minWidth":100,"align":"center","headerVAlign":"center","style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}],"footerStyle":{"fontWeight":"bold"}},{"accessor":"PIME","name":"PIME","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.001","0.001","0.001"],"header":"PIME","minWidth":100,"align":"center","headerVAlign":"center","style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}],"footerStyle":{"fontWeight":"bold"}}],"columnGroups":[{"name":"p-values","headerStyle":{"fontSize":"1.3em"},"columns":["FULL","FILT","PERfect","PIME"]}],"defaultPageSize":12,"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"theme":{"style":{"fontSize":"0.9em"}},"dataKey":"079db4e6c2e9ed19842eac9085e5d884"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="ordinations">Ordinations</h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Now we can visualize the <strong>Jensen-Shannon Divergence</strong>,
<strong>unweighted-UniFrac</strong>, and
<strong>weighted-UniFrac</strong> distance matrices to access
dissimilarity among sample. Here we ordinate using Principal Coordinate
Analysis (PCoA) but we could also test other methods like NMDS, CCA,
etc. We will also test the whether there are any differences in
ordination between the <code>phyloseq</code> package and the <a
href="https://github.com/ChiLiubio/microeco"><code>microeco</code>
package</a>.</p>
<h4 id="plot-code">Plot Code</h4>
<details>
<summary>
<strong>Code to reproduce plots</strong>
</summary>
<p><strong>We begin with the code for the 16S rRNA data.</strong></p>
<p>First the code for ordination implementation in
<code>phyloseq</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>ssu18_data_sets <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ssu18_ps_work&quot;</span>, <span class="st">&quot;ssu18_ps_filt&quot;</span>, <span class="st">&quot;ssu18_ps_perfect&quot;</span>, <span class="st">&quot;ssu18_ps_pime&quot;</span>)</span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>ssu_dist <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;jsd&quot;</span>, <span class="st">&quot;unifrac&quot;</span>, <span class="st">&quot;wunifrac&quot;</span>)</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (samp_ps <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (d <span class="cf">in</span> ssu_dist){</span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(samp_ps, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>)))</span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a>     ord_meths <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;NMDS&quot;</span>, <span class="st">&quot;PCoA&quot;</span>, <span class="st">&quot;CCA&quot;</span>, <span class="st">&quot;DCA&quot;</span>) <span class="co"># MDS = PCoA, &quot;CCA&quot;, &quot;DCA&quot;, &quot;DPCoA&quot;, &quot;RDA&quot;</span></span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>     tmp_plist <span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">llply</span>(<span class="fu">as.list</span>(ord_meths), <span class="cf">function</span>(i, physeq, d) {</span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a>        ordi <span class="ot">=</span> <span class="fu">ordinate</span>(physeq, <span class="at">method =</span> i, <span class="at">distance =</span> d)</span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plot_ordination</span>(physeq, ordi, <span class="st">&quot;samples&quot;</span>, <span class="at">color =</span> <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a>   }, tmp_get, d)</span>
<span id="cb88-11"><a href="#cb88-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-12"><a href="#cb88-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(tmp_plist) <span class="ot">&lt;-</span> ord_meths</span>
<span id="cb88-13"><a href="#cb88-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-14"><a href="#cb88-14" aria-hidden="true" tabindex="-1"></a>  tmp_df <span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">ldply</span>(tmp_plist, <span class="cf">function</span>(x){</span>
<span id="cb88-15"><a href="#cb88-15" aria-hidden="true" tabindex="-1"></a>      df <span class="ot">=</span> x<span class="sc">$</span>data[, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span>
<span id="cb88-16"><a href="#cb88-16" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(df) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Axis_1&quot;</span>, <span class="st">&quot;Axis_2&quot;</span>)</span>
<span id="cb88-17"><a href="#cb88-17" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">cbind</span>(df, x<span class="sc">$</span>data))})</span>
<span id="cb88-18"><a href="#cb88-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(tmp_df)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">&quot;method&quot;</span></span>
<span id="cb88-19"><a href="#cb88-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb88-20"><a href="#cb88-20" aria-hidden="true" tabindex="-1"></a>  tmp_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(tmp_df, <span class="fu">aes</span>(Axis_1, Axis_2, <span class="at">color =</span> TEMP, <span class="at">shape =</span> TEMP, <span class="at">fill =</span> TEMP))</span>
<span id="cb88-21"><a href="#cb88-21" aria-hidden="true" tabindex="-1"></a>  tmp_plot <span class="ot">&lt;-</span> tmp_plot <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>)</span>
<span id="cb88-22"><a href="#cb88-22" aria-hidden="true" tabindex="-1"></a>  tmp_plot <span class="ot">&lt;-</span> tmp_plot <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>method, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>)</span>
<span id="cb88-23"><a href="#cb88-23" aria-hidden="true" tabindex="-1"></a>  tmp_plot <span class="ot">&lt;-</span> tmp_plot <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> swel_col)</span>
<span id="cb88-24"><a href="#cb88-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb88-25"><a href="#cb88-25" aria-hidden="true" tabindex="-1"></a>  tmp_df_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(samp_ps, <span class="st">&quot;_dist_&quot;</span>, .))</span>
<span id="cb88-26"><a href="#cb88-26" aria-hidden="true" tabindex="-1"></a>  tmp_plist_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(samp_ps, <span class="st">&quot;_&quot;</span>, ., <span class="st">&quot;_plist&quot;</span>))</span>
<span id="cb88-27"><a href="#cb88-27" aria-hidden="true" tabindex="-1"></a>  tmp_plot_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(samp_ps, <span class="st">&quot;_dist_&quot;</span>, ., <span class="st">&quot;_plot&quot;</span>))</span>
<span id="cb88-28"><a href="#cb88-28" aria-hidden="true" tabindex="-1"></a>  tmp_list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="st">&quot;tmp_df_name&quot;</span> <span class="ot">=</span> tmp_df, <span class="at">tmp_plist_name =</span> tmp_plist, <span class="at">tmp_plot_name =</span> tmp_plot)</span>
<span id="cb88-29"><a href="#cb88-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(<span class="fu">paste0</span>(samp_ps, <span class="st">&quot;_&quot;</span>,  d, <span class="st">&quot;_ord_results&quot;</span>), tmp_list)</span>
<span id="cb88-30"><a href="#cb88-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;_tmp&quot;</span>))</span>
<span id="cb88-31"><a href="#cb88-31" aria-hidden="true" tabindex="-1"></a> }</span>
<span id="cb88-32"><a href="#cb88-32" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>plist_name <span class="ot">&lt;-</span> <span class="fu">objects</span>(<span class="at">pattern=</span><span class="st">&quot;_ord_results&quot;</span>)</span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>plot_num <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>)</span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> plist_name) {</span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (j <span class="cf">in</span> plot_num) {</span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a>       tmp_get_i <span class="ot">&lt;-</span> <span class="fu">get</span>(i)<span class="sc">$</span>tmp_plist_name</span>
<span id="cb89-6"><a href="#cb89-6" aria-hidden="true" tabindex="-1"></a>       tmp_ord <span class="ot">&lt;-</span> <span class="fu">names</span>(tmp_get_i)[j]</span>
<span id="cb89-7"><a href="#cb89-7" aria-hidden="true" tabindex="-1"></a>       tmp_name <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">str_replace</span>(i, <span class="st">&quot;ord_results&quot;</span>, tmp_ord)</span>
<span id="cb89-8"><a href="#cb89-8" aria-hidden="true" tabindex="-1"></a>       tmp_plot <span class="ot">&lt;-</span> tmp_get_i[[j]] <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> swel_col)</span>
<span id="cb89-9"><a href="#cb89-9" aria-hidden="true" tabindex="-1"></a>       tmp_plot <span class="ot">&lt;-</span> tmp_plot <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>, <span class="fu">aes</span>(<span class="at">shape =</span> TEMP)) <span class="sc">+</span></span>
<span id="cb89-10"><a href="#cb89-10" aria-hidden="true" tabindex="-1"></a>         <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>)</span>
<span id="cb89-11"><a href="#cb89-11" aria-hidden="true" tabindex="-1"></a>       tmp_plot<span class="sc">$</span>labels<span class="sc">$</span>shape <span class="ot">&lt;-</span> <span class="st">&quot;TEMP&quot;</span></span>
<span id="cb89-12"><a href="#cb89-12" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_name, tmp_plot)</span>
<span id="cb89-13"><a href="#cb89-13" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb89-14"><a href="#cb89-14" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb89-15"><a href="#cb89-15" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<p>And now the code for ordination implementation in
<code>microeco</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>microeco_path <span class="ot">&lt;-</span> <span class="st">&quot;files/beta/microeco/&quot;</span></span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>  tmp_dataset <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_me&quot;</span>)))</span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span><span class="fu">cal_betadiv</span>(<span class="at">unifrac =</span> <span class="cn">TRUE</span>)</span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a><span class="do">#### CODE TO ADD JSD DISTANCE ####  </span></span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>  tmp_jsd <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(<span class="fu">get</span>(i), <span class="at">method =</span> <span class="st">&quot;jsd&quot;</span>) </span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a>  tmp_jsd <span class="ot">&lt;-</span> <span class="fu">forceSymmetric</span>(<span class="fu">as.matrix</span>(tmp_jsd), <span class="at">uplo =</span> <span class="st">&quot;L&quot;</span>)</span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a>  tmp_jsd <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_jsd)</span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span>beta_diversity<span class="sc">$</span>jsd <span class="ot">&lt;-</span> tmp_jsd</span>
<span id="cb90-10"><a href="#cb90-10" aria-hidden="true" tabindex="-1"></a><span class="do">#### #### #### #### #### #### #### ####  </span></span>
<span id="cb90-11"><a href="#cb90-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="fu">paste</span>(microeco_path, i, <span class="st">&quot;/metrics/&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb90-12"><a href="#cb90-12" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span><span class="fu">save_betadiv</span>(<span class="at">dirpath =</span> <span class="fu">paste</span>(microeco_path, i, <span class="st">&quot;/metrics/&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb90-13"><a href="#cb90-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb90-14"><a href="#cb90-14" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(ssu18_ps_perfect_me<span class="sc">$</span>beta_diversity)</span></code></pre></div>
<pre><code>[1] &quot;list&quot;</code></pre>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(ssu18_ps_perfect_me<span class="sc">$</span>beta_diversity)</span></code></pre></div>
<pre><code>              Length Class  Mode   
bray          225    -none- numeric
jaccard       225    -none- numeric
wei_unifrac   225    -none- numeric
unwei_unifrac 225    -none- numeric
jsd           225    -none- numeric</code></pre>
</div>
<p>Here I made a custom “function” to run the analysis, plot the graphs,
save graph objects, and save plots (as <code>.png</code> and
<code>.pdf</code> files). I am sure an actual programmer would be
shocked, but it works.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>microeco_beta_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(choose_input, choose_metric, choose_ord) {  </span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>  tmp_dataset <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(choose_input, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_me&quot;</span>)))</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>  tmp_t1 <span class="ot">&lt;-</span> trans_beta<span class="sc">$</span><span class="fu">new</span>(<span class="at">dataset =</span> tmp_dataset, <span class="at">group =</span> <span class="st">&quot;TEMP&quot;</span>, <span class="at">measure =</span> choose_metric)</span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a>  tmp_t1<span class="sc">$</span><span class="fu">cal_ordination</span>(<span class="at">ordination =</span> choose_ord)</span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a>  tmp_t1_ord_plot <span class="ot">&lt;-</span> tmp_t1<span class="sc">$</span><span class="fu">plot_ordination</span>(<span class="at">plot_color =</span> <span class="st">&quot;TEMP&quot;</span>, </span>
<span id="cb95-6"><a href="#cb95-6" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">plot_shape =</span> <span class="st">&quot;TEMP&quot;</span>, </span>
<span id="cb95-7"><a href="#cb95-7" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">plot_group_ellipse =</span> <span class="cn">FALSE</span>, </span>
<span id="cb95-8"><a href="#cb95-8" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">color_values =</span> swel_col, </span>
<span id="cb95-9"><a href="#cb95-9" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">shape_values =</span> <span class="fu">c</span>(<span class="dv">16</span>, <span class="dv">17</span>, <span class="dv">15</span>)) <span class="sc">+</span> </span>
<span id="cb95-10"><a href="#cb95-10" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>)</span>
<span id="cb95-11"><a href="#cb95-11" aria-hidden="true" tabindex="-1"></a>  tmp_t1<span class="sc">$</span><span class="fu">cal_group_distance</span>()</span>
<span id="cb95-12"><a href="#cb95-12" aria-hidden="true" tabindex="-1"></a>  tmp_t1_within_group_plot <span class="ot">&lt;-</span> tmp_t1<span class="sc">$</span><span class="fu">plot_group_distance</span>(<span class="at">distance_pair_stat =</span> <span class="cn">TRUE</span>, <span class="at">color_values =</span> swel_col)</span>
<span id="cb95-13"><a href="#cb95-13" aria-hidden="true" tabindex="-1"></a>  tmp_t1<span class="sc">$</span>res_group_distance</span>
<span id="cb95-14"><a href="#cb95-14" aria-hidden="true" tabindex="-1"></a>  tmp_t1<span class="sc">$</span><span class="fu">cal_group_distance</span>(<span class="at">within_group =</span> <span class="cn">FALSE</span>)</span>
<span id="cb95-15"><a href="#cb95-15" aria-hidden="true" tabindex="-1"></a>  tmp_t1_btwn_group_plot <span class="ot">&lt;-</span> tmp_t1<span class="sc">$</span><span class="fu">plot_group_distance</span>(<span class="at">distance_pair_stat =</span> <span class="cn">TRUE</span>, <span class="at">color_values =</span> swel_col) </span>
<span id="cb95-16"><a href="#cb95-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-17"><a href="#cb95-17" aria-hidden="true" tabindex="-1"></a><span class="do">###### SET names</span></span>
<span id="cb95-18"><a href="#cb95-18" aria-hidden="true" tabindex="-1"></a>  tmp_name_ord <span class="ot">&lt;-</span> <span class="fu">paste</span>(choose_input, <span class="st">&quot;_me_&quot;</span>, choose_metric, <span class="st">&quot;_&quot;</span>, choose_ord, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb95-19"><a href="#cb95-19" aria-hidden="true" tabindex="-1"></a>  tmp_name_wg <span class="ot">&lt;-</span> <span class="fu">paste</span>(choose_input, <span class="st">&quot;_me_wg_&quot;</span>, choose_metric, <span class="st">&quot;_&quot;</span>, choose_ord, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb95-20"><a href="#cb95-20" aria-hidden="true" tabindex="-1"></a>  tmp_name_bg <span class="ot">&lt;-</span> <span class="fu">paste</span>(choose_input, <span class="st">&quot;_me_bg_&quot;</span>, choose_metric, <span class="st">&quot;_&quot;</span>, choose_ord, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb95-21"><a href="#cb95-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb95-22"><a href="#cb95-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name_ord, tmp_t1_ord_plot, <span class="at">envir =</span> <span class="fu">parent.frame</span>() )</span>
<span id="cb95-23"><a href="#cb95-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name_wg, tmp_t1_within_group_plot, <span class="at">envir =</span> <span class="fu">parent.frame</span>() )</span>
<span id="cb95-24"><a href="#cb95-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name_bg, tmp_t1_btwn_group_plot, <span class="at">envir =</span> <span class="fu">parent.frame</span>() )</span>
<span id="cb95-25"><a href="#cb95-25" aria-hidden="true" tabindex="-1"></a><span class="do">###### SET paths</span></span>
<span id="cb95-26"><a href="#cb95-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="fu">paste</span>(microeco_path, choose_input, <span class="st">&quot;/ordination/&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb95-27"><a href="#cb95-27" aria-hidden="true" tabindex="-1"></a>  tmp_path <span class="ot">&lt;-</span> <span class="fu">paste</span>(microeco_path, choose_input, <span class="st">&quot;/ordination/&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb95-28"><a href="#cb95-28" aria-hidden="true" tabindex="-1"></a>  tmp_path_ord <span class="ot">&lt;-</span> <span class="fu">paste</span>(tmp_path, <span class="st">&quot;ord_plots/&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb95-29"><a href="#cb95-29" aria-hidden="true" tabindex="-1"></a>  tmp_path_wg <span class="ot">&lt;-</span> <span class="fu">paste</span>(tmp_path, <span class="st">&quot;within_group/&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb95-30"><a href="#cb95-30" aria-hidden="true" tabindex="-1"></a>  tmp_path_bg <span class="ot">&lt;-</span> <span class="fu">paste</span>(tmp_path, <span class="st">&quot;between_group/&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb95-31"><a href="#cb95-31" aria-hidden="true" tabindex="-1"></a><span class="do">###### SAVE plots  </span></span>
<span id="cb95-32"><a href="#cb95-32" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_t1_ord_plot, <span class="at">path =</span> <span class="fu">paste</span>(tmp_path_ord, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">filename =</span> <span class="fu">paste0</span>(tmp_name_ord, <span class="st">&quot;.png&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb95-33"><a href="#cb95-33" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_t1_ord_plot, <span class="at">path =</span> <span class="fu">paste</span>(tmp_path_ord, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">filename =</span> <span class="fu">paste0</span>(tmp_name_ord, <span class="st">&quot;.pdf&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))                    </span>
<span id="cb95-34"><a href="#cb95-34" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_t1_within_group_plot, <span class="at">path =</span> <span class="fu">paste</span>(tmp_path_wg, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">filename =</span> <span class="fu">paste0</span>(tmp_name_wg, <span class="st">&quot;.png&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb95-35"><a href="#cb95-35" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_t1_within_group_plot, <span class="at">path =</span> <span class="fu">paste</span>(tmp_path_wg, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">filename =</span> <span class="fu">paste0</span>(tmp_name_wg, <span class="st">&quot;.pdf&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))                  </span>
<span id="cb95-36"><a href="#cb95-36" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_t1_btwn_group_plot, <span class="at">path =</span> <span class="fu">paste</span>(tmp_path_bg, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">filename =</span> <span class="fu">paste0</span>(tmp_name_bg, <span class="st">&quot;.png&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb95-37"><a href="#cb95-37" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_t1_btwn_group_plot, <span class="at">path =</span> <span class="fu">paste</span>(tmp_path_bg, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">filename =</span> <span class="fu">paste0</span>(tmp_name_bg, <span class="st">&quot;.pdf&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))                  </span>
<span id="cb95-38"><a href="#cb95-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb95-39"><a href="#cb95-39" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<p>And then run it like so:</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="fu">microeco_beta_plot</span>(<span class="at">choose_input =</span> <span class="st">&quot;ssu18_ps_pime&quot;</span>, <span class="at">choose_metric =</span> <span class="st">&quot;bray&quot;</span>, <span class="at">choose_ord =</span> <span class="st">&quot;PCoA&quot;</span>)</span></code></pre></div>
</div>
<p>It would be super cool to have a function that looped through a bunch
of data sets, distance metrics, and ordination methods.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">get</span>(<span class="fu">paste</span>(<span class="st">&quot;ssu18_ps_work&quot;</span>, <span class="st">&quot;_me&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))<span class="sc">$</span>beta_diversity)) {</span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>  tmp_metric <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">get</span>(<span class="fu">paste</span>(<span class="st">&quot;ssu18_ps_work&quot;</span>, <span class="st">&quot;_me&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))<span class="sc">$</span>beta_diversity[j])</span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">microeco_beta_plot</span>(<span class="at">choose_input =</span> <span class="st">&quot;ssu18_ps_work&quot;</span>, <span class="at">choose_metric =</span> tmp_metric, <span class="at">choose_ord =</span> <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-7"><a href="#cb97-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">get</span>(<span class="fu">paste</span>(<span class="st">&quot;ssu18_ps_filt&quot;</span>, <span class="st">&quot;_me&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))<span class="sc">$</span>beta_diversity)) {</span>
<span id="cb97-8"><a href="#cb97-8" aria-hidden="true" tabindex="-1"></a>  tmp_metric <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">get</span>(<span class="fu">paste</span>(<span class="st">&quot;ssu18_ps_filt&quot;</span>, <span class="st">&quot;_me&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))<span class="sc">$</span>beta_diversity[j])</span>
<span id="cb97-9"><a href="#cb97-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">microeco_beta_plot</span>(<span class="at">choose_input =</span> <span class="st">&quot;ssu18_ps_filt&quot;</span>, <span class="at">choose_metric =</span> tmp_metric, <span class="at">choose_ord =</span> <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb97-10"><a href="#cb97-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb97-11"><a href="#cb97-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb97-12"><a href="#cb97-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-13"><a href="#cb97-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">get</span>(<span class="fu">paste</span>(<span class="st">&quot;ssu18_ps_perfect&quot;</span>, <span class="st">&quot;_me&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))<span class="sc">$</span>beta_diversity)) {</span>
<span id="cb97-14"><a href="#cb97-14" aria-hidden="true" tabindex="-1"></a>  tmp_metric <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">get</span>(<span class="fu">paste</span>(<span class="st">&quot;ssu18_ps_perfect&quot;</span>, <span class="st">&quot;_me&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))<span class="sc">$</span>beta_diversity[j])</span>
<span id="cb97-15"><a href="#cb97-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">microeco_beta_plot</span>(<span class="at">choose_input =</span> <span class="st">&quot;ssu18_ps_perfect&quot;</span>, <span class="at">choose_metric =</span> tmp_metric, <span class="at">choose_ord =</span> <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb97-16"><a href="#cb97-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb97-17"><a href="#cb97-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb97-18"><a href="#cb97-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-19"><a href="#cb97-19" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">get</span>(<span class="fu">paste</span>(<span class="st">&quot;ssu18_ps_pime&quot;</span>, <span class="st">&quot;_me&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))<span class="sc">$</span>beta_diversity)) {</span>
<span id="cb97-20"><a href="#cb97-20" aria-hidden="true" tabindex="-1"></a>  tmp_metric <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">get</span>(<span class="fu">paste</span>(<span class="st">&quot;ssu18_ps_pime&quot;</span>, <span class="st">&quot;_me&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))<span class="sc">$</span>beta_diversity[j])</span>
<span id="cb97-21"><a href="#cb97-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">microeco_beta_plot</span>(<span class="at">choose_input =</span> <span class="st">&quot;ssu18_ps_pime&quot;</span>, <span class="at">choose_metric =</span> tmp_metric, <span class="at">choose_ord =</span> <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb97-22"><a href="#cb97-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb97-23"><a href="#cb97-23" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><strong>And now the ITS code.</strong></p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>its18_data_sets <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;its18_ps_work&quot;</span>, <span class="st">&quot;its18_ps_filt&quot;</span>, <span class="st">&quot;its18_ps_perfect&quot;</span>, <span class="st">&quot;its18_ps_pime&quot;</span>)</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>its_dist <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;jsd&quot;</span>, <span class="st">&quot;bray&quot;</span>, <span class="st">&quot;gower&quot;</span>)</span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (samp_ps <span class="cf">in</span> its18_data_sets) {</span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (d <span class="cf">in</span> its_dist){</span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(samp_ps, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>)))</span>
<span id="cb98-6"><a href="#cb98-6" aria-hidden="true" tabindex="-1"></a>     ord_meths <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;NMDS&quot;</span>, <span class="st">&quot;PCoA&quot;</span>, <span class="st">&quot;CCA&quot;</span>, <span class="st">&quot;DCA&quot;</span>) <span class="co"># MDS = PCoA, &quot;CCA&quot;, &quot;DCA&quot;, &quot;DPCoA&quot;, &quot;RDA&quot;</span></span>
<span id="cb98-7"><a href="#cb98-7" aria-hidden="true" tabindex="-1"></a>     tmp_plist <span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">llply</span>(<span class="fu">as.list</span>(ord_meths), <span class="cf">function</span>(i, physeq, d) {</span>
<span id="cb98-8"><a href="#cb98-8" aria-hidden="true" tabindex="-1"></a>        ordi <span class="ot">=</span> <span class="fu">ordinate</span>(physeq, <span class="at">method =</span> i, <span class="at">distance =</span> d)</span>
<span id="cb98-9"><a href="#cb98-9" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plot_ordination</span>(physeq, ordi, <span class="st">&quot;samples&quot;</span>, <span class="at">color =</span> <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb98-10"><a href="#cb98-10" aria-hidden="true" tabindex="-1"></a>   }, tmp_get, d)</span>
<span id="cb98-11"><a href="#cb98-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-12"><a href="#cb98-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(tmp_plist) <span class="ot">&lt;-</span> ord_meths</span>
<span id="cb98-13"><a href="#cb98-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-14"><a href="#cb98-14" aria-hidden="true" tabindex="-1"></a>  tmp_df <span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">ldply</span>(tmp_plist, <span class="cf">function</span>(x){</span>
<span id="cb98-15"><a href="#cb98-15" aria-hidden="true" tabindex="-1"></a>      df <span class="ot">=</span> x<span class="sc">$</span>data[, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span>
<span id="cb98-16"><a href="#cb98-16" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(df) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Axis_1&quot;</span>, <span class="st">&quot;Axis_2&quot;</span>)</span>
<span id="cb98-17"><a href="#cb98-17" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">cbind</span>(df, x<span class="sc">$</span>data))})</span>
<span id="cb98-18"><a href="#cb98-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(tmp_df)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">&quot;method&quot;</span></span>
<span id="cb98-19"><a href="#cb98-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb98-20"><a href="#cb98-20" aria-hidden="true" tabindex="-1"></a>  tmp_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(tmp_df, <span class="fu">aes</span>(Axis_1, Axis_2, <span class="at">color =</span> TEMP, <span class="at">shape =</span> TEMP, <span class="at">fill =</span> TEMP))</span>
<span id="cb98-21"><a href="#cb98-21" aria-hidden="true" tabindex="-1"></a>  tmp_plot <span class="ot">&lt;-</span> tmp_plot <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>)</span>
<span id="cb98-22"><a href="#cb98-22" aria-hidden="true" tabindex="-1"></a>  tmp_plot <span class="ot">&lt;-</span> tmp_plot <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>method, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>)</span>
<span id="cb98-23"><a href="#cb98-23" aria-hidden="true" tabindex="-1"></a>  tmp_plot <span class="ot">&lt;-</span> tmp_plot <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> swel_col)</span>
<span id="cb98-24"><a href="#cb98-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb98-25"><a href="#cb98-25" aria-hidden="true" tabindex="-1"></a>  tmp_df_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(samp_ps, <span class="st">&quot;_dist_&quot;</span>, .))</span>
<span id="cb98-26"><a href="#cb98-26" aria-hidden="true" tabindex="-1"></a>  tmp_plist_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(samp_ps, <span class="st">&quot;_&quot;</span>, ., <span class="st">&quot;_plist&quot;</span>))</span>
<span id="cb98-27"><a href="#cb98-27" aria-hidden="true" tabindex="-1"></a>  tmp_plot_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(samp_ps, <span class="st">&quot;_dist_&quot;</span>, ., <span class="st">&quot;_plot&quot;</span>))</span>
<span id="cb98-28"><a href="#cb98-28" aria-hidden="true" tabindex="-1"></a>  tmp_list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="st">&quot;tmp_df_name&quot;</span> <span class="ot">=</span> tmp_df, <span class="at">tmp_plist_name =</span> tmp_plist, <span class="at">tmp_plot_name =</span> tmp_plot)</span>
<span id="cb98-29"><a href="#cb98-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(<span class="fu">paste0</span>(samp_ps, <span class="st">&quot;_&quot;</span>,  d, <span class="st">&quot;_ord_results&quot;</span>), tmp_list)</span>
<span id="cb98-30"><a href="#cb98-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;_tmp&quot;</span>))</span>
<span id="cb98-31"><a href="#cb98-31" aria-hidden="true" tabindex="-1"></a> }</span>
<span id="cb98-32"><a href="#cb98-32" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>plist_name <span class="ot">&lt;-</span> <span class="fu">objects</span>(<span class="at">pattern=</span><span class="st">&quot;_ord_results&quot;</span>)</span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>plot_num <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>)</span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> plist_name) {</span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (j <span class="cf">in</span> plot_num) {</span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a>       tmp_get_i <span class="ot">&lt;-</span> <span class="fu">get</span>(i)<span class="sc">$</span>tmp_plist_name</span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a>       tmp_ord <span class="ot">&lt;-</span> <span class="fu">names</span>(tmp_get_i)[j]</span>
<span id="cb99-7"><a href="#cb99-7" aria-hidden="true" tabindex="-1"></a>       tmp_name <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">str_replace</span>(i, <span class="st">&quot;ord_results&quot;</span>, tmp_ord)</span>
<span id="cb99-8"><a href="#cb99-8" aria-hidden="true" tabindex="-1"></a>       tmp_plot <span class="ot">&lt;-</span> tmp_get_i[[j]] <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> swel_col)</span>
<span id="cb99-9"><a href="#cb99-9" aria-hidden="true" tabindex="-1"></a>       tmp_plot <span class="ot">&lt;-</span> tmp_plot <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>, <span class="fu">aes</span>(<span class="at">shape =</span> TEMP)) <span class="sc">+</span></span>
<span id="cb99-10"><a href="#cb99-10" aria-hidden="true" tabindex="-1"></a>         <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>)</span>
<span id="cb99-11"><a href="#cb99-11" aria-hidden="true" tabindex="-1"></a>       tmp_plot<span class="sc">$</span>labels<span class="sc">$</span>shape <span class="ot">&lt;-</span> <span class="st">&quot;TEMP&quot;</span></span>
<span id="cb99-12"><a href="#cb99-12" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_name, tmp_plot)</span>
<span id="cb99-13"><a href="#cb99-13" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb99-14"><a href="#cb99-14" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb99-15"><a href="#cb99-15" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<p>And now the code for ordination implementation in
<code>microeco</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb100"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>microeco_path <span class="ot">&lt;-</span> <span class="st">&quot;files/beta/microeco/&quot;</span></span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its18_data_sets) {</span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>  tmp_dataset <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_me&quot;</span>)))</span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span><span class="fu">cal_betadiv</span>(<span class="at">unifrac =</span> <span class="cn">FALSE</span>)</span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a><span class="do">#### CODE TO ADD JSD DISTANCE ####  </span></span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a>  tmp_jsd <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(<span class="fu">get</span>(i), <span class="at">method =</span> <span class="st">&quot;jsd&quot;</span>) </span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a>  tmp_jsd <span class="ot">&lt;-</span> <span class="fu">forceSymmetric</span>(<span class="fu">as.matrix</span>(tmp_jsd), <span class="at">uplo =</span> <span class="st">&quot;L&quot;</span>)</span>
<span id="cb100-8"><a href="#cb100-8" aria-hidden="true" tabindex="-1"></a>  tmp_jsd <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_jsd)</span>
<span id="cb100-9"><a href="#cb100-9" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span>beta_diversity<span class="sc">$</span>jsd <span class="ot">&lt;-</span> tmp_jsd</span>
<span id="cb100-10"><a href="#cb100-10" aria-hidden="true" tabindex="-1"></a><span class="do">#### CODE TO ADD GOWER DISTANCE ####  </span></span>
<span id="cb100-11"><a href="#cb100-11" aria-hidden="true" tabindex="-1"></a>  tmp_gower <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(<span class="fu">get</span>(i), <span class="at">method =</span> <span class="st">&quot;gower&quot;</span>) </span>
<span id="cb100-12"><a href="#cb100-12" aria-hidden="true" tabindex="-1"></a>  tmp_gower <span class="ot">&lt;-</span> <span class="fu">forceSymmetric</span>(<span class="fu">as.matrix</span>(tmp_gower), <span class="at">uplo =</span> <span class="st">&quot;L&quot;</span>)</span>
<span id="cb100-13"><a href="#cb100-13" aria-hidden="true" tabindex="-1"></a>  tmp_gower <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_gower)</span>
<span id="cb100-14"><a href="#cb100-14" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span>beta_diversity<span class="sc">$</span>gower <span class="ot">&lt;-</span> tmp_gower</span>
<span id="cb100-15"><a href="#cb100-15" aria-hidden="true" tabindex="-1"></a><span class="do">#### #### #### #### #### #### #### ####  </span></span>
<span id="cb100-16"><a href="#cb100-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="fu">paste</span>(microeco_path, i, <span class="st">&quot;/metrics/&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb100-17"><a href="#cb100-17" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span><span class="fu">save_betadiv</span>(<span class="at">dirpath =</span> <span class="fu">paste</span>(microeco_path, i, <span class="st">&quot;/metrics/&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb100-18"><a href="#cb100-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb100-19"><a href="#cb100-19" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb101"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(its18_ps_perfect_me<span class="sc">$</span>beta_diversity)</span></code></pre></div>
<pre><code>[1] &quot;list&quot;</code></pre>
<div class="sourceCode" id="cb103"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(its18_ps_perfect_me<span class="sc">$</span>beta_diversity)</span></code></pre></div>
<pre><code>        Length Class  Mode   
bray    169    -none- numeric
jaccard 169    -none- numeric
jsd     169    -none- numeric
gower   169    -none- numeric</code></pre>
</div>
<p>Here we call the custom function described above.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb105"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">get</span>(<span class="fu">paste</span>(<span class="st">&quot;its18_ps_work&quot;</span>, <span class="st">&quot;_me&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))<span class="sc">$</span>beta_diversity)) {</span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a>  tmp_metric <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">get</span>(<span class="fu">paste</span>(<span class="st">&quot;its18_ps_work&quot;</span>, <span class="st">&quot;_me&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))<span class="sc">$</span>beta_diversity[j])</span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">microeco_beta_plot</span>(<span class="at">choose_input =</span> <span class="st">&quot;its18_ps_work&quot;</span>, <span class="at">choose_metric =</span> tmp_metric, <span class="at">choose_ord =</span> <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb105-5"><a href="#cb105-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb105-6"><a href="#cb105-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-7"><a href="#cb105-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">get</span>(<span class="fu">paste</span>(<span class="st">&quot;its18_ps_filt&quot;</span>, <span class="st">&quot;_me&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))<span class="sc">$</span>beta_diversity)) {</span>
<span id="cb105-8"><a href="#cb105-8" aria-hidden="true" tabindex="-1"></a>  tmp_metric <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">get</span>(<span class="fu">paste</span>(<span class="st">&quot;its18_ps_filt&quot;</span>, <span class="st">&quot;_me&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))<span class="sc">$</span>beta_diversity[j])</span>
<span id="cb105-9"><a href="#cb105-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">microeco_beta_plot</span>(<span class="at">choose_input =</span> <span class="st">&quot;its18_ps_filt&quot;</span>, <span class="at">choose_metric =</span> tmp_metric, <span class="at">choose_ord =</span> <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb105-10"><a href="#cb105-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb105-11"><a href="#cb105-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb105-12"><a href="#cb105-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-13"><a href="#cb105-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">get</span>(<span class="fu">paste</span>(<span class="st">&quot;its18_ps_perfect&quot;</span>, <span class="st">&quot;_me&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))<span class="sc">$</span>beta_diversity)) {</span>
<span id="cb105-14"><a href="#cb105-14" aria-hidden="true" tabindex="-1"></a>  tmp_metric <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">get</span>(<span class="fu">paste</span>(<span class="st">&quot;its18_ps_perfect&quot;</span>, <span class="st">&quot;_me&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))<span class="sc">$</span>beta_diversity[j])</span>
<span id="cb105-15"><a href="#cb105-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">microeco_beta_plot</span>(<span class="at">choose_input =</span> <span class="st">&quot;its18_ps_perfect&quot;</span>, <span class="at">choose_metric =</span> tmp_metric, <span class="at">choose_ord =</span> <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb105-16"><a href="#cb105-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb105-17"><a href="#cb105-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb105-18"><a href="#cb105-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-19"><a href="#cb105-19" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">get</span>(<span class="fu">paste</span>(<span class="st">&quot;its18_ps_pime&quot;</span>, <span class="st">&quot;_me&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))<span class="sc">$</span>beta_diversity)) {</span>
<span id="cb105-20"><a href="#cb105-20" aria-hidden="true" tabindex="-1"></a>  tmp_metric <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">get</span>(<span class="fu">paste</span>(<span class="st">&quot;its18_ps_pime&quot;</span>, <span class="st">&quot;_me&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))<span class="sc">$</span>beta_diversity[j])</span>
<span id="cb105-21"><a href="#cb105-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">microeco_beta_plot</span>(<span class="at">choose_input =</span> <span class="st">&quot;its18_ps_pime&quot;</span>, <span class="at">choose_metric =</span> tmp_metric, <span class="at">choose_ord =</span> <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb105-22"><a href="#cb105-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb105-23"><a href="#cb105-23" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
</details>
<p><br/></p>
<h2 id="s-rrna-1">16S rRNA</h2>
<div class="l-body-outset">
<div class="panelset">
<section id="full-6" class="panel">
<h4>FULL</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/beta/ssu18_ps_work_beta_div_ord_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 1 | PCoA ordination plots of the
FULL data set. Top = phyloseq, bottom = microeco; Left = Jensen-Shannon
Divergence, middle = UniFrac (unwieghted), right = UniFrac
(wieghted).</strong></small></p>
</section>
<section id="arbitrary-6" class="panel">
<h4>Arbitrary</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/beta/ssu18_ps_filt_beta_div_ord_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 2 | PCoA ordination plots of the
Arbitrary filtered data set. Top = phyloseq, bottom = microeco; Left =
Jensen-Shannon Divergence, middle = UniFrac (unwieghted), right =
UniFrac (wieghted).</strong></small></p>
</section>
<section id="perfect-6" class="panel">
<h4>PERfect</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/beta/ssu18_ps_perfect_beta_div_ord_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 3 | PCoA ordination plots of the
PERfect filtered data set. Top = phyloseq, bottom = microeco; Left =
Jensen-Shannon Divergence, middle = UniFrac (unwieghted), right =
UniFrac (wieghted).</strong></small></p>
</section>
<section id="pime-6" class="panel">
<h4>PIME</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/beta/ssu18_ps_pime_beta_div_ord_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 4 | PCoA ordination plots of the
PIME filtered data set. Top = phyloseq, bottom = microeco; Left =
Jensen-Shannon Divergence, middle = UniFrac (unwieghted), right =
UniFrac (wieghted).</strong></small></p>
</section>
</div>
</div>
<h2 id="its-1">ITS</h2>
<div class="l-body-outset">
<div class="panelset">
<section id="arbitrary-7" class="panel">
<h4>Arbitrary</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/beta/its18_ps_work_beta_div_ord_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 1 | PCoA ordination plots of the FULL
data set. Top = phyloseq, bottom = microeco; Left = Jensen-Shannon
Divergence, middle = Bray-Curtis dissimilarity, right = Gower
distance.</strong></small></p>
</section>
<section id="filt" class="panel">
<h4>FILT</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/beta/its18_ps_filt_beta_div_ord_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 2 | PCoA ordination plots of the
Arbitrary filtered data set. Top = phyloseq, bottom = microeco; Left =
Jensen-Shannon Divergence, middle = Bray-Curtis dissimilarity, right =
Gower distance.</strong></small></p>
</section>
<section id="perfect-7" class="panel">
<h4>PERfect</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/beta/its18_ps_perfect_beta_div_ord_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 3 | PCoA ordination plots of the PERfect
filtered data set. Top = phyloseq, bottom = microeco; Left =
Jensen-Shannon Divergence, middle = Bray-Curtis dissimilarity, right =
Gower distance.</strong></small></p>
</section>
<section id="pime-7" class="panel">
<h4>PIME</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/beta/its18_ps_pime_beta_div_ord_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 4 | PCoA ordination plots of the PIME
filtered data set. Top = phyloseq, bottom = microeco; Left =
Jensen-Shannon Divergence, middle = Bray-Curtis dissimilarity, right =
Gower distance.</strong></small></p>
</section>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="within-between-group-distances">Within &amp; Between Group
Distances</h1>
<p>Above we calculated the sample distances within groups. Let’s see
what those plots look like.</p>
<h2 id="s-rrna-2">16S rRNA</h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="l-body-outset">
<div class="panelset">
<section id="full-7" class="panel">
<h4>FULL</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/beta/ssu18_ps_work_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 5 | Within (top) and between
(bottom) group variation plots of the FULL data set. Left =
Jensen-Shannon Divergence, middle = UniFrac (unwieghted), right =
UniFrac (wieghted). Significant differences denoted by asterisks (* p
&lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001). ns =
not significant. </strong></small></p>
</section>
<section id="arbitrary-8" class="panel">
<h4>Arbitrary</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/beta/ssu18_ps_filt_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 6 | Within (top) and between
(bottom) group variation plots of the Arbitrary filtered data set. Left
= Jensen-Shannon Divergence, middle = UniFrac (unwieghted), right =
UniFrac (wieghted). Significant differences denoted by asterisks (* p
&lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001). ns =
not significant. </strong></small></p>
</section>
<section id="perfect-8" class="panel">
<h4>PERfect</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/beta/ssu18_ps_perfect_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 7 | Within (top) and between
(bottom) group variation plots of the PERfect filtered data set. Left =
Jensen-Shannon Divergence, middle = UniFrac (unwieghted), right =
UniFrac (wieghted). Significant differences denoted by asterisks (* p
&lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001). ns =
not significant. </strong></small></p>
</section>
<section id="pime-8" class="panel">
<h4>PIME</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/beta/ssu18_ps_pime_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 8 | Within (top) and between
(bottom) group variation plots of the PIME filtered data set. Left =
Jensen-Shannon Divergence, middle = UniFrac (unwieghted), right =
UniFrac (wieghted). Significant differences denoted by asterisks (* p
&lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001). ns =
not significant. </strong></small></p>
</section>
</div>
</div>
<h2 id="its-2">ITS</h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="l-body-outset">
<div class="panelset">
<section id="full-8" class="panel">
<h4>FULL</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/beta/its18_ps_work_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 5 | Within (top) and between (bottom)
group variation plots of the FULL data set. Left = Jensen-Shannon
Divergence, middle = Bray-Curtis dissimilarity, right = Gower distance.
Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt;
0.01, *** p &lt; 0.001, **** p &lt; 0.0001). ns = not significant.
</strong></small></p>
</section>
<section id="arbitrary-9" class="panel">
<h4>Arbitrary</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/beta/its18_ps_filt_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 6 | Within (top) and between (bottom)
group variation plots of the Arbitrary filtered data set. Left =
Jensen-Shannon Divergence, middle = Bray-Curtis dissimilarity, right =
Gower distance. Significant differences denoted by asterisks (* p &lt;
0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001). ns = not
significant. </strong></small></p>
</section>
<section id="perfect-9" class="panel">
<h4>PERfect</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/beta/its18_ps_perfect_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 7 | Within (top) and between (bottom)
group variation plots of the PERfect filtered data set. Left =
Jensen-Shannon Divergence, middle = Bray-Curtis dissimilarity, right =
Gower distance. Significant differences denoted by asterisks (* p &lt;
0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001). ns = not
significant. </strong></small></p>
</section>
<section id="pime-9" class="panel">
<h4>PIME</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/beta/its18_ps_pime_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 8 | Within (top) and between (bottom)
group variation plots of the PIME filtered data set. Left =
Jensen-Shannon Divergence, middle = Bray-Curtis dissimilarity, right =
Gower distance. Significant differences denoted by asterisks (* p &lt;
0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001). ns = not
significant. </strong></small></p>
</section>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="workflow-output">Workflow Output</h1>
<p>Data products generated in this workflow can be downloaded from
figshare.</p>
<iframe src="https://widgets.figshare.com/articles/16828063/embed?show_title=1" width="568" height="251" allowfullscreen frameborder="0">
</iframe>
<p></br></p>
<p><a href="da.html" class="btnnav button_round" style="float: right;">Next
workflow: <br> 7. Differentially Abundant ASVs &amp; Taxa</a>
<a href="alpha.html" class="btnnav button_round" style="float: left;">Previous
workflow: <br> 5. Alpha Diversity Estimates</a></p>
<h2 class="appendix" id="source-code">Source Code</h2>
<p>The source code for this page can be accessed on GitHub by <a
href="https://github.com/sweltr/high-temp/blob/master/beta.Rmd">clicking
this link</a>.</p>
<h2 class="appendix" id="data-availability">Data Availability</h2>
<p>Data generated in this workflow and the Rdata need to run the
workflow can be accessed on figshare at <a
href="https://doi.org/10.25573/data.16828063">10.25573/data.16828063</a>.</p>
<h2 class="appendix" id="last-updated-on">Last updated on</h2>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;2022-06-02 13:50:47 EST&quot;</code></pre>
</div>
<div id="refs" class="references csl-bib-body hanging-indent"
role="doc-bibliography">
<div id="ref-oksanen2013community" class="csl-entry"
role="doc-biblioentry">
Oksanen, Jari, F. Guillaume Blanchet, Roeland Kindt, Pierre Legendre,
Peter R. Minchin, R. B. O’Hara, Gavin L. Simpson, Peter Sólymos, M.
Henry H. Stevens, and Helene Wagner. 2012. <span>“Vegan: Community
Ecology Package.”</span> <em>R Package Version</em> 2 (0). <a
href="https://cran.r-project.org/web/packages/vegan/index.html">https://cran.r-project.org/web/packages/vegan/index.html</a>.
</div>
</div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/sweltr/high-temp/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/sweltr/high-temp/">https://github.com/sweltr/high-temp/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<div class="distill-site-nav distill-site-footer">
<p>© Copyright 2022 metacrobe and The SWELTR Project.</p>
<p>Site constructed using <a
href="https://rstudio.github.io/distill/">Distill for R
Markdown</a>.</p>
</div>
<!--/radix_placeholder_navigation_after_body-->


</body>

</html>
