<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */
</style>

  <!--radix_placeholder_meta_tags-->
  <title>sweltr: 1 DADA2 Workflow</title>

  <meta property="description" itemprop="description" content="DADA2 workflow for processing the **2018 16S rRNA** and the **2018 ITS** high temperature data sets. Workflow uses paired end reads, beginning with raw fastq files, ending with sequence and taxonomy tables."/>

  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>


  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="sweltr: 1 DADA2 Workflow"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="DADA2 workflow for processing the **2018 16S rRNA** and the **2018 ITS** high temperature data sets. Workflow uses paired end reads, beginning with raw fastq files, ending with sequence and taxonomy tables."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="sweltr"/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="sweltr: 1 DADA2 Workflow"/>
  <meta property="twitter:description" content="DADA2 workflow for processing the **2018 16S rRNA** and the **2018 ITS** high temperature data sets. Workflow uses paired end reads, beginning with raw fastq files, ending with sequence and taxonomy tables."/>

  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Cutadapt removes adapter sequences from high-throughput sequencing reads;citation_publication_date=2011;citation_volume=17;citation_author=Marcel Martin"/>
  <meta name="citation_reference" content="citation_title=The SILVA ribosomal RNA gene database project: Improved data processing and web-based tools;citation_publication_date=2012;citation_volume=41;citation_author=Christian Quast;citation_author=Elmar Pruesse;citation_author=Pelin Yilmaz;citation_author=Jan Gerken;citation_author=Timmy Schweer;citation_author=Pablo Yarza;citation_author=Jörg Peplies;citation_author=Frank Oliver Glöckner"/>
  <meta name="citation_reference" content="citation_title=UNITE general FASTA release for fungi;citation_publication_date=2020;citation_publisher=UNITE Community;citation_author=Kessy Abarenkov;citation_author=Allan Zirk;citation_author=Timo Piirmann;citation_author=Raivo Pöhönen;citation_author=Filipp Ivanov;citation_author=R. Henrik Nilsson;citation_author=Urmas Kõljalg"/>
  <meta name="citation_reference" content="citation_title=The UNITE database for molecular identification of fungi: Handling dark taxa and parallel taxonomic classifications;citation_publication_date=2019;citation_volume=47;citation_author=Rolf Henrik Nilsson;citation_author=Karl-Henrik Larsson;citation_author=Andy F S Taylor;citation_author=Johan Bengtsson-Palme;citation_author=Thomas S Jeppesen;citation_author=Dmitry Schigel;citation_author=Peter Kennedy;citation_author=Kathryn Picard;citation_author=Frank Oliver Glöckner;citation_author=Leho Tedersoo;citation_author=The UNITE database for molecular identification of fungi: Handling dark taxa and parallel taxonomic classifications"/>
  <meta name="citation_reference" content="citation_title=Global patterns of 16S rRNA diversity at a depth of millions of sequences per sample;citation_publication_date=2011;citation_volume=108;citation_author=J Gregory Caporaso;citation_author=Christian L Lauber;citation_author=William A Walters;citation_author=Donna Berg-Lyons;citation_author=Catherine A Lozupone;citation_author=Peter J Turnbaugh;citation_author=Noah Fierer;citation_author=Rob Knight"/>
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["1 DADA2 Workflow"]},{"type":"character","attributes":{},"value":["DADA2 workflow for processing the **2018 16S rRNA** and the **2018 ITS** high temperature data sets. Workflow uses paired end reads, beginning with raw fastq files, ending with sequence and taxonomy tables.\n"]},{"type":"character","attributes":{},"value":["assets/cite.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }

  </style>

  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure img {
    width: 100%;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refs = $(this).attr('data-cites').split(" ");
        var refHtml = refs.map(function(ref) {
          return "<p>" + $('#ref-' + ref).html() + "</p>";
        }).join("\n");
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="site_libs/header-attrs-2.8/header-attrs.js"></script>
  <script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
  <script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
  <link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
  <script src="site_libs/datatables-binding-0.18/datatables.js"></script>
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
  <script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
  <script src="site_libs/jszip-1.10.20/jszip.min.js"></script>
  <link href="site_libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
  <script src="site_libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
  <link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
  <script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Roboto:300|Nunito+Sans:300|Inconsolata|Source+Sans+Pro|Work+Sans:200,300,400,500');
  /********************************/
  /* controls general properties of site*/
  /********************************/

  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
    background-color: #FEFEFE;
  }

  d-title h1{
  font-size: 2.1em;
  font-family: 'Work Sans';
  font-weight: 300;
  }

  d-article h1 {
    font-size: 1.8em;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 300;
  }

  d-article h2 {
    font-size: 1.65em;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 500;
  }

  d-article h3 {
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
    margin-top: 1em;
  }

  d-article p {
    font-size: 18px;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
  }

  d-byline .authors-affiliations {
    grid-column-end: span 3;
    grid-column-gap: 8px;
  }

  img.orcid-id {
    margin: 0;
  }

  a.name:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  a.affiliation:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  blockquote p {
  	font-size: 1.3em;
  	font-family: 'Lato';
    font-style: normal;
    margin: 0.5em 0 0.5em;
  }

  d-article li {
    margin: 0px 0px 4px;
    font-size: 18px;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
  }

  /********************************/
  /* controls CODE*/
  /********************************/

  /** pre.sourceCode controls code block without lang **/

  /** Controls r code  **/

  pre.sourceCode.r {
    margin-top: 0px;
     color: #333333;
     background-color: #E8E8E8;
     border-radius: 3px;
     font-size: 0.80em;
     font-family: 'Monaco', 'PT Mono', monospace;
     overflow-x: scroll !important;
     border-left: 6px solid #2271B2;
     border-top: 1px solid #B7B7B7;
     border-right: 1px solid #B7B7B7;
     border-bottom: 1px solid #B7B7B7;
     padding: 18px 0px 18px 20px;
   }

   pre.sourceCode.bash {
     margin-top: 0px;
      color: #333333;
      background-color: #E8E8E8;
      border-radius: 3px;
      font-size: 0.80em;
      font-family: 'Monaco', 'PT Mono', monospace;
      overflow-x: scroll !important;
      border-left: 6px solid #B22271;
      border-top: 1px solid #B7B7B7;
      border-right: 1px solid #B7B7B7;
      border-bottom: 1px solid #B7B7B7;
      padding: 18px 0px 18px 20px;
    }
  /** Controls code output **/

  d-article pre {
     margin-top: 0px;
      color: #333333;
      background-color: #FAFAFA;
      font-size: 0.8em;
      border: 1px solid #ccc;
      overflow-x: scroll !important;
      padding:  10px 10px 10px 18px;
      border-radius: 5px;

    }

  /** Controls inline code  **/

  p code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }

  d-article aside code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-family: 'Monaco', 'PT Mono', monospace;
    color: #BF2727;
  }

  /********************************/
  /* controls NAV BAR and HEADERS*/
  /********************************/

  .nav-dropdown-content .nav-dropdown-header {
    text-transform: none;
  }

  .distill-site-header .logo img {
    margin-top: 10px;
    max-height: 30px;
    margin-bottom: 10px;
  }

  .distill-site-nav {
      background-color: #113858;
      font-size: 16px;
      font-weight: 300;
  }

  img {
    margin: 5px;
  }

  /** LIGHTBOX MARKUP **/
  .thumbnail {
    max-width: 100%;
    position: relative;
  }

  /** DETAILS fold down**/

  details {
    background-color: #f6f6f6;
    cursor: pointer;
    color: #2d2d2d;
    margin-bottom:1vh;
    padding: 1em;
    border: none;
    text-align: left;
    outline: none;
  }

  .active, details:hover {
    transition: max-height 0.2s ease-out;
    outline: none;

    /* background-color: #acc2c2; */
  }


  details {
    display: flex;
    justify-content: space-between;
    align-content: center;
    border-radius: 4px;
    outline: none;

  }

  details.active {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    outline: none;

  }

  summary::-webkit-details-marker {
    display: none;
    outline: none;

  }
  summary:focus {
    outline: none;
  }
  summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f0fe";
  }

  details[open] summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f146";
  }

  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"1 DADA2 Workflow","description":"DADA2 workflow for processing the **2018 16S rRNA** and the **2018 ITS** high temperature data sets. Workflow uses paired end reads, beginning with raw fastq files, ending with sequence and taxonomy tables.","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title"></a>
<a href="index.html">
<i class="fa fa-home" aria-hidden="true"></i>
</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
A. Processing Workflows
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="dada2.html">1. DADA2</a>
<a href="data-prep.html">2. Data Set Prep</a>
<a href="otu.html">3. OTU Clustering</a>
<hr/>
<span class="nav-dropdown-header">Filtering</span>
<a href="pime.html">4a. PIME Filtering</a>
<a href="perfect.html">4b. PERFect Filtering</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
B. Diversity Workflows
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="taxa.html">5. Taxonomic</a>
<a href="alpha.html">6. Alpha Diversity</a>
<a href="beta.html">7. Beta Diversity</a>
<a href="da.html">8. DA ASVs &amp; OTUs</a>
<a href="metadata.html">9. Metadata</a>
</div>
</div>
<a href="data-availability.html">Data Products</a>
</div>
<div class="nav-right">
<a href="https://github.com/sweltr/high-temp/">
<i class="fa fa-github fa-lg" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>1 DADA2 Workflow</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->
<p><p>DADA2 workflow for processing the <strong>2018 16S rRNA</strong> and the <strong>2018 ITS</strong> high temperature data sets. Workflow uses paired end reads, beginning with raw fastq files, ending with sequence and taxonomy tables.</p></p>
</div>


<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#prerequisites">Prerequisites</a></li>
<li><a href="#s-rrna-data">16s rRNA Data</a>
<ul>
<li><a href="#overview">Overview</a></li>
<li><a href="#set-working-environment">1. Set Working Environment</a></li>
<li><a href="#define-primers">2. Define Primers</a></li>
<li><a href="#remove-primers">3. Remove Primers</a></li>
<li><a href="#quality-assessment-filtering">4. Quality Assessment &amp; Filtering</a></li>
<li><a href="#learn-error-rates">5. Learn Error Rates</a></li>
<li><a href="#dereplicate-reads">6. Dereplicate Reads</a></li>
<li><a href="#dada2-asv-inference">7. DADA2 &amp; ASV Inference</a></li>
<li><a href="#construct-sequence-table">8. Construct Sequence Table</a></li>
<li><a href="#remove-chimeras">9. Remove Chimeras</a></li>
<li><a href="#track-reads-through-workflow">10. Track Reads through Workflow</a></li>
<li><a href="#assign-taxonomy">11. Assign Taxonomy</a></li>
<li><a href="#save-image">12. Save Image</a></li>
<li><a href="#r-session-information-code">R Session Information &amp; Code</a></li>
</ul></li>
<li><a href="#its-data">ITS Data</a>
<ul>
<li><a href="#overview-1">Overview</a></li>
<li><a href="#set-working-environment-1">1. Set Working Environment</a></li>
<li><a href="#define-primers-1">2. Define Primers</a></li>
<li><a href="#remove-primers-1">3. Remove Primers</a></li>
<li><a href="#quality-assessment-filtering-1">4. Quality Assessment &amp; Filtering</a></li>
<li><a href="#learn-error-rates-1">5. Learn Error Rates</a></li>
<li><a href="#dereplicate-reads-1">6. Dereplicate Reads</a></li>
<li><a href="#dada2-asv-inference-1">7. DADA2 &amp; ASV Inference</a></li>
<li><a href="#merge-paired-reads">8. Merge Paired Reads</a></li>
<li><a href="#construct-sequence-table-1">9. Construct Sequence Table</a></li>
<li><a href="#remove-chimeras-1">10. Remove Chimeras</a></li>
<li><a href="#track-reads-through-workflow-1">11. Track Reads through Workflow</a></li>
<li><a href="#assign-taxonomy-1">12. Assign Taxonomy</a></li>
<li><a href="#save-image-1">13. Save Image</a></li>
<li><a href="#r-session-information-code-1">R Session Information &amp; Code</a></li>
<li><a href="#source-code">Source Code</a></li>
<li><a href="#data-availability">Data Availability</a></li>
</ul></li>
</ul>
</nav>
</div>
<details>
<summary>
Click here for setup information.
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">echo =</span> <span class="cn">TRUE</span>, <span class="at">eval =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">919191</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends(ape, local = TRUE)  </span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends_reverse(ape, local = TRUE)  </span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dada2); <span class="fu">packageVersion</span>(<span class="st">&quot;dada2&quot;</span>)</span></code></pre></div>
<pre><code>[1] &#39;1.18.0&#39;</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ShortRead); <span class="fu">packageVersion</span>(<span class="st">&quot;ShortRead&quot;</span>)</span></code></pre></div>
<pre><code>[1] &#39;1.48.0&#39;</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2); <span class="fu">packageVersion</span>(<span class="st">&quot;ggplot2&quot;</span>)</span></code></pre></div>
<pre><code>[1] &#39;3.3.3&#39;</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(rmarkdown, DT, gridExtra, grid, </span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">install =</span> <span class="cn">FALSE</span>, <span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">scipen=</span><span class="dv">999</span>)</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_current<span class="sc">$</span><span class="fu">get</span>(<span class="fu">c</span>(</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache&quot;</span>,</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.path&quot;</span>,</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.rebuild&quot;</span>,</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;dependson&quot;</span>,</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;autodep&quot;</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>))</span></code></pre></div>
<pre><code>$cache
[1] 0

$cache.path
[1] &quot;dada2_cache/html5/&quot;

$cache.rebuild
[1] FALSE

$dependson
NULL

$autodep
[1] FALSE</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code></pre></div>
</div>
</details>
<h1 id="prerequisites">Prerequisites</h1>
<p>In order to run this workflow, you either need the <em>raw</em> data, available on the figshare project site, or the <em>trimmed</em> data, available from the ENA under project accession number … See the <a href="data-availability.html">Data Availability</a> page for complete details.</p>
<p>This workflow is how we processed our 16S rRNA and ITS data sets using <a href="https://benjjneb.github.io/dada2/">DADA2</a> <span class="citation" data-cites="callahan2016dada2">(<a href="#ref-callahan2016dada2" role="doc-biblioref">Callahan et al. 2016</a>)</span> from samples collected in <strong>2018</strong>. Workflow construction is based on the <a href="https://benjjneb.github.io/dada2/tutorial_1_8.html">DADA2 Pipeline Tutorial (1.8)</a> and the primer identification section of the <a href="https://benjjneb.github.io/dada2/ITS_workflow.html">DADA2 ITS Pipeline Workflow (1.8)</a>.</p>
<aside>
Because the workflows share many common variable names, you must either split the workflow into a script for each data set OR run the command <code>remove(list = ls())</code> before beginning the next workflow.
</aside>
<h1 id="s-rrna-data">16s rRNA Data</h1>
<blockquote>
<p>NOTE: The reverse reads for 16s rRNA data set are unusable, therefore this workflow deals with forward reads <em>only</em>.</p>
</blockquote>
<p>We will screen the forward reads for reverse primers and show the quality plots of the reverse reads at the beginning of the workflow. After that, reverse reads are not used.</p>
<h2 id="overview">Overview</h2>
<h3 id="sequence-files-samples">Sequence Files &amp; Samples</h3>
<p>We sequenced a total of 15 samples collected from 10 different plots at a single depth in each plot.</p>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="htmlwidget-75d662642cd4c45aefa9" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-75d662642cd4c45aefa9">{"x":{"filter":"none","extensions":["Buttons"],"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15"],["P1","P2","P3","P4","P5","P6","P7","P8","P9","P10","P1-8C","P3-8C","P5-8C","P7-8C","P9-8C"],["P01","P02","P03","P04","P05","P06","P07","P08","P09","P10","P01","P03","P05","P07","P09"],["Warm","Control","Warm","Control","Warm","Control","Warm","Control","Warm","Control","Warm","Warm","Warm","Warm","Warm"],[3,0,3,0,3,0,3,0,3,0,8,8,8,8,8],["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],["A","A","B","B","C","C","D","D","E","E","A","B","C","D","E"],["P1","P2","P3","P4","P5","P6","P7","P8","P9","P10","P1-8C","P3-8C","P5-8C","P7-8C","P9-8C"],["SD701","SD701","SD701","SD701","SD701","SD701","SD701","SD701","SD702","SD702","SD702","SD702","SD702","SD702","SD702"],["tactaggt","tactaggt","tactaggt","tactaggt","tactaggt","tactaggt","tactaggt","tactaggt","acgtacgt","acgtacgt","acgtacgt","acgtacgt","acgtacgt","acgtacgt","acgtacgt"],["SD501","SD502","SD503","SD504","SD505","SD506","SD507","SD508","SD501","SD502","SD503","SD504","SD505","SD506","SD507"],["AAGCAGCA","ACGCGTGA","CGATCTAC","TGCGTCAC","GTCTAGTG","CTAGTATG","GATAGCGT","TCTACACT","AAGCAGCA","ACGCGTGA","CGATCTAC","TGCGTCAC","GTCTAGTG","CTAGTATG","GATAGCGT"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Sample_ID<\/th>\n      <th>Plot<\/th>\n      <th>Treatment<\/th>\n      <th>Temp<\/th>\n      <th>Depth<\/th>\n      <th>Pairing<\/th>\n      <th>Fastq_Files_ID<\/th>\n      <th>I7_Index_ID<\/th>\n      <th>index<\/th>\n      <th>I5_Index_ID<\/th>\n      <th>index2<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"className":"dt-right","targets":4},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<h3 id="workflow">Workflow</h3>
<p>The workflow consists of the following steps:</p>
<table>
<thead>
<tr class="header">
<th>Step</th>
<th>Command</th>
<th>What we’re doing</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>multiple</td>
<td>prepare input file names &amp; paths</td>
</tr>
<tr class="even">
<td>2</td>
<td>multiple</td>
<td>Define primers (all orientations)</td>
</tr>
<tr class="odd">
<td>3</td>
<td><code>cutadapt</code></td>
<td>Remove primers</td>
</tr>
<tr class="even">
<td>4</td>
<td><code>filterAndTrim()</code></td>
<td>assess quality &amp; filter reads</td>
</tr>
<tr class="odd">
<td>5</td>
<td><code>learnErrors()</code></td>
<td>generate an error model for the data</td>
</tr>
<tr class="even">
<td>6</td>
<td><code>derepFastq</code></td>
<td>dereplicate sequences</td>
</tr>
<tr class="odd">
<td>7</td>
<td><code>dada()</code></td>
<td>infer ASVs on both forward &amp; reverse reads independently</td>
</tr>
<tr class="even">
<td>8</td>
<td><code>makeSequenceTable()</code></td>
<td>generate a count table</td>
</tr>
<tr class="odd">
<td>9</td>
<td><code>removeBimeraDenovo()</code></td>
<td>screen for &amp; remove chimeras</td>
</tr>
<tr class="even">
<td>10</td>
<td></td>
<td>track reads through workflow</td>
</tr>
<tr class="odd">
<td>11</td>
<td><code>assignTaxonomy()</code></td>
<td>assign taxonomy &amp; finish workflow</td>
</tr>
<tr class="even">
<td>12</td>
<td><code>save.image()</code></td>
<td>save an image of seqtab &amp; taxtab for next part of workflow</td>
</tr>
</tbody>
</table>
<h2 id="set-working-environment">1. Set Working Environment</h2>
<p>Next, we need to setup the working environment by renaming the fastq files and defining a path for the working directory.</p>
<h3 id="rename-samples">Rename samples</h3>
<p>To make the parsing easier, we will eliminate the lane and well number from each sample. If you do not wish to do that you will need to adjust the code accordingly. I am <em>positive</em> there are more elegant ways of doing this.</p>
<blockquote>
<p>CAUTION. If you use this for your own data, please check that this code works on a few backup files before proceeding.</p>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;RAW&quot;</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>orig_fastq <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">pattern =</span> <span class="st">&quot;*.fastq.gz&quot;</span>)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>newname_fastq <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_S.*_L001&quot;</span>, <span class="st">&quot;&quot;</span>, orig_fastq)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>newname_fastq <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_001&quot;</span>, <span class="st">&quot;&quot;</span>, newname_fastq)</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="fu">file.rename</span>(orig_fastq, newname_fastq)</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;../&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>path <span class="ot">&lt;-</span> <span class="st">&quot;RAW&quot;</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">list.files</span>(path))</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;P1_R1.fastq.gz&quot;    &quot;P1_R2.fastq.gz&quot;    &quot;P1-8C_R1.fastq.gz&quot;
[4] &quot;P1-8C_R2.fastq.gz&quot; &quot;P10_R1.fastq.gz&quot;   &quot;P10_R2.fastq.gz&quot;  </code></pre>
</div>
<p>Then, we generate matched lists of the forward and reverse read files. We also parse out the sample name.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>fnFs <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern =</span> <span class="st">&quot;_R1.fastq.gz&quot;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>fnRs <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern =</span> <span class="st">&quot;_R2.fastq.gz&quot;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>p1a <span class="ot">&lt;-</span> <span class="fu">plotQualityProfile</span>(fnFs[<span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>], <span class="at">aggregate =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>p2a <span class="ot">&lt;-</span> <span class="fu">plotQualityProfile</span>(fnRs[<span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>], <span class="at">aggregate =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>p3a <span class="ot">&lt;-</span> <span class="fu">grid.arrange</span>(p1a, p2a, <span class="at">nrow =</span> <span class="dv">1</span>)</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;plot_qscores_2018a.png&quot;</span>, p3a, <span class="at">width =</span> <span class="dv">7</span>, <span class="at">height =</span> <span class="dv">3</span>)</span></code></pre></div>
</div>
<p>Unfortunately, the reverse reads are unusable.</p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="include/dada2/ssu18_plot_qscores_a.png" width="1050" /></p>
</div>
<p><small><em>Aggregated quality score plots for forward reads &amp; reverse reads.</em></small></p>
<h2 id="define-primers">2. Define Primers</h2>
<p>Before we start the DADA2 workflow we need to run <a href="https://github.com/marcelm/cutadapt">catadapt</a> <span class="citation" data-cites="martin2011cutadapt">(<a href="#ref-martin2011cutadapt" role="doc-biblioref">Martin 2011</a>)</span> on all <code>fastq.gz</code> files to trim the primers. For this part of the study we used the primer pair 505F (GTGCCAGCMGCCGCGGTAA) and 806R (GGACTACHVGGGTWTCTAAT) <span class="citation" data-cites="caporaso2011global">(<a href="#ref-caporaso2011global" role="doc-biblioref">Caporaso et al. 2011</a>)</span> which should yield an amplicon length of about 253 bp.</p>
<p>First we define the primers.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>FWD <span class="ot">&lt;-</span> <span class="st">&quot;GTGCCAGCMGCCGCGGTAA&quot;</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>REV <span class="ot">&lt;-</span> <span class="st">&quot;GGACTACHVGGGTWTCTAAT&quot;</span></span></code></pre></div>
</div>
<p>Next, we check the presence and orientation of these primers in the data. I started doing this for ITS data because of primer read-through but I really like the general idea of doing it just to make sure nothing funny is going of with the data.</p>
<p>To do this we will create all orientations of the input primer sequences. In other words the <em>Forward</em>, <em>Complement</em>, <em>Reverse</em>, and <em>Reverse Complement</em> variations.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>allOrients <span class="ot">&lt;-</span> <span class="cf">function</span>(primer) {</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">require</span>(Biostrings)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>    dna <span class="ot">&lt;-</span> <span class="fu">DNAString</span>(primer)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>    orients <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">Forward =</span> dna,</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>                 <span class="at">Complement =</span> <span class="fu">complement</span>(dna),</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">Reverse =</span> <span class="fu">reverse</span>(dna),</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">RevComp =</span> <span class="fu">reverseComplement</span>(dna))</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">sapply</span>(orients, toString))</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>FWD.orients <span class="ot">&lt;-</span> <span class="fu">allOrients</span>(FWD)</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>REV.orients <span class="ot">&lt;-</span> <span class="fu">allOrients</span>(REV)</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>FWD.orients</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>REV.orients</span></code></pre></div>
</div>
<pre><code>              Forward            Complement               Reverse
&quot;GTGCCAGCMGCCGCGGTAA&quot; &quot;CACGGTCGKCGGCGCCATT&quot; &quot;AATGGCGCCGMCGACCGTG&quot;
              RevComp
&quot;TTACCGCGGCKGCTGGCAC&quot;
               Forward             Complement                Reverse
&quot;GGACTACHVGGGTWTCTAAT&quot; &quot;CCTGATGDBCCCAWAGATTA&quot; &quot;TAATCTWTGGGVHCATCAGG&quot;
               RevComp
&quot;ATTAGAWACCCBDGTAGTCC&quot;</code></pre>
<p>Now we do a little pre-filter step to eliminate ambiguous bases (<em>Ns</em>) because <em>Ns</em> make mapping of short primer sequences difficult. This step removes any reads with <em>Ns</em>. Again, set some files paths, this time for the filtered reads.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>fnFs.filtN <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">&quot;filtN&quot;</span>, <span class="fu">basename</span>(fnFs))</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">filterAndTrim</span>(fnFs, fnFs.filtN,</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxN =</span> <span class="dv">0</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<p>Nice. Time to assess the number of times a primer (and all possible primer orientation) appear in the forward and reverse reads. According to the workflow, counting the primers on one set of paired end fastq files is sufficient to see if there is a problem. This assumes that all the files were created using the same library prep. Basically for both primers, we will search for all four orientations in both forward and reverse reads. Since this is 16S rRNA we do not anticipate any issues but it is worth checking.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>primerHits <span class="ot">&lt;-</span> <span class="cf">function</span>(primer, fn) {</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Counts number of reads in which the primer is found</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>    nhits <span class="ot">&lt;-</span> <span class="fu">vcountPattern</span>(primer, <span class="fu">sread</span>(<span class="fu">readFastq</span>(fn)), <span class="at">fixed =</span> <span class="cn">FALSE</span>)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">sum</span>(nhits <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<h4 id="forward-primers">Forward primers</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits,</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">fn =</span> fnFs.filtN[[sampnum]]))</span></code></pre></div>
</div>
<pre><code>                 Forward Complement Reverse RevComp
FWD.ForwardReads   74707          0       0       0</code></pre>
<h4 id="reverse-primers">Reverse primers</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits,</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">fn =</span> fnFs.filtN[[sampnum]]))</span></code></pre></div>
</div>
<pre><code>                 Forward Complement Reverse RevComp
REV.ForwardReads       0          0       0     165</code></pre>
<p>As expected, forward primers predominantly in the forward reads and very little evidence of reverse primers.</p>
<h2 id="remove-primers">3. Remove Primers</h2>
<p>Now we can run <a href="https://github.com/marcelm/cutadapt">catadapt</a> <span class="citation" data-cites="martin2011cutadapt">(<a href="#ref-martin2011cutadapt" role="doc-biblioref">Martin 2011</a>)</span> to remove the primers from the fastq sequences. A little setup first. If this command executes successfully it means R has found cutadapt.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>cutadapt <span class="ot">&lt;-</span> <span class="st">&quot;/Users/rad/miniconda3/envs/cutadapt/bin/cutadapt&quot;</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="st">&quot;--version&quot;</span>) <span class="co"># Run shell commands from R</span></span></code></pre></div>
</div>
<pre><code>2.8</code></pre>
<p>We set paths and trim the forward primer and the reverse-complement of the reverse primer off of R1 (forward reads) and trim the reverse primer and the reverse-complement of the forward primer off of R2 (reverse reads).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>path.cut <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">&quot;cutadapt&quot;</span>)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(path.cut)) <span class="fu">dir.create</span>(path.cut)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>fnFs.cut <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(fnFs))</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>FWD.RC <span class="ot">&lt;-</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(FWD)</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>REV.RC <span class="ot">&lt;-</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(REV)</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>R1.flags <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;-g&quot;</span>, FWD, <span class="st">&quot;-a&quot;</span>, REV.RC)</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(fnFs)) {<span class="fu">system2</span>(cutadapt,</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">args =</span> <span class="fu">c</span>(R1.flags,</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">&quot;-m&quot;</span>, <span class="dv">20</span>, <span class="st">&quot;-n&quot;</span>, <span class="dv">2</span>, <span class="st">&quot;-e&quot;</span>, <span class="fl">0.12</span>,</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">&quot;-o&quot;</span>, fnFs.cut[i],</span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>                                            fnFs.filtN[i]))}</span></code></pre></div>
</div>
<pre><code>This is cutadapt 2.8 with Python 3.7.6
Command line parameters: -g GTGCCAGCMGCCGCGGTAA -a ATTAGAWACCCBDGTAGTCC -m 20 -n 2 -e 0.12 -o RAW/cutadapt/P1_R1.fastq.gz RAW/filtN/P1_R1.fastq.gz</code></pre>
<p><em>Note</em>. If the code above removes all of the base pairs in a sequence, you will get downstream errors unless you set the <code>-m</code> flag. This flag sets the minimum length and reads shorter than this will be discarded. Without this flag, reads of length 0 will be kept and cause issues. <em>Also, a lot of output will be written to the screen by cutadapt!</em>.</p>
<p>We can now count the number of primers in the sequences from the output of cutadapt.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[sampnum]]),</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[sampnum]]))</span></code></pre></div>
</div>
<pre><code>                 Forward Complement Reverse RevComp
FWD.ForwardReads       0          0       0       0
REV.ForwardReads       0          0       0       0</code></pre>
<p>Basically, primers are no longer detected in the cutadapted reads. Now, for each sample, we can take a look at how the pre-filtering step and primer removal affected the total number of raw reads.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-189bf7e795965c842df7" style="width:80%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-189bf7e795965c842df7">{"x":{"filter":"none","extensions":["Buttons"],"data":[["P1","P1-8C","P10","P2","P3","P3-8C","P4","P5","P5-8C","P6","P7","P7-8C","P8","P9","P9-8C"],[85489,77645,67583,87091,93065,66401,67810,58266,120892,101599,76781,79032,77599,95566,95182],[85482,77642,67579,87085,93053,66367,67800,58259,120842,101592,76776,79022,77591,95564,94457],[85402,77470,67432,86878,92851,65498,67617,58006,117015,101363,76583,78724,77133,95310,44250]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>raw<\/th>\n      <th>pre_filt<\/th>\n      <th>cut<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<h2 id="quality-assessment-filtering">4. Quality Assessment &amp; Filtering</h2>
<p>We need the forward and reverse fastq file names and the sample names.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>cutFs <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path.cut, <span class="at">pattern =</span> <span class="st">&quot;_R1.fastq.gz&quot;</span>,</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>get.sample.name <span class="ot">&lt;-</span> <span class="cf">function</span>(fname) <span class="fu">strsplit</span>(<span class="fu">basename</span>(fname), <span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>sample.names <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">sapply</span>(cutFs, get.sample.name))</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sample.names)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="st">&quot;P1&quot;</span>, <span class="st">&quot;P1-8C&quot;</span>, <span class="st">&quot;P10&quot;</span>, <span class="st">&quot;P2&quot;</span>, <span class="st">&quot;P3&quot;</span>, <span class="st">&quot;P3-8C&quot;</span>)</span></code></pre></div>
<pre><code>[1] &quot;P1&quot;    &quot;P1-8C&quot; &quot;P10&quot;   &quot;P2&quot;    &quot;P3&quot;    &quot;P3-8C&quot;</code></pre>
</div>
<p>First let’s look at the quality of our reads. The numbers in brackets specify which samples to view. Here we are looking at an aggregate plot of all data (except the negative control)</p>
<h3 id="quality-plots">Quality plots</h3>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">plotQualityProfile</span>(cutFs[<span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>], <span class="at">aggregate =</span> <span class="cn">TRUE</span>)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;ssu18_plot_qscores.png&quot;</span>, p1, <span class="at">width =</span> <span class="dv">7</span>, <span class="at">height =</span> <span class="dv">3</span>)</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;ssu18_plot_qscores.png&quot;</span>, p1)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="include/dada2/ssu18_plot_qscores_b.png" width="552" /></p>
</div>
<p><small><em>Aggregated quality score plots for forward reads.</em></small></p>
<p>The forward reads look good.</p>
<h3 id="filtering">Filtering</h3>
<p>First, we again make some path variables and setup a new directory of filtered reads.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>filtFs <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut, <span class="st">&quot;filtered&quot;</span>, <span class="fu">basename</span>(cutFs))</span></code></pre></div>
</div>
<p>And then we trim the reads.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>out <span class="ot">&lt;-</span> <span class="fu">filterAndTrim</span>(cutFs, filtFs,</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">maxN =</span> <span class="dv">0</span>, <span class="at">maxEE =</span> <span class="dv">2</span>,</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">truncQ =</span> <span class="dv">2</span>, <span class="at">rm.phix =</span> <span class="cn">TRUE</span>,</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">compress =</span> <span class="cn">TRUE</span>, <span class="at">multithread =</span> <span class="dv">20</span>)</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>out</span></code></pre></div>
</div>
<aside>
These parameters should be set based on the anticipated length of the amplicon and the read quality.
</aside>
<p>And here is a table of how the filtering step affected the number of reads in each sample. As you can see, there are a few samples that started with a low read count to begin with—we will likely remove those samples at some point.</p>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-4e280cda47fb19178af4" style="width:80%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4e280cda47fb19178af4">{"x":{"filter":"none","extensions":["Buttons"],"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15"],["P1_R1","P1-8C_R1","P10_R1","P2_R1","P3_R1","P3-8C_R1","P4_R1","P5_R1","P5-8C_R1","P6_R1","P7_R1","P7-8C_R1","P8_R1","P9_R1","P9-8C_R1"],[85402,77470,67432,86878,92851,65498,67617,58006,117015,101363,76583,78724,77133,95310,44250],[79804,72760,62598,81240,86732,60860,63037,54127,109421,94740,71678,73639,72263,89315,33000]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Sample<\/th>\n      <th>reads.in<\/th>\n      <th>reads.out<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"className":"dt-right","targets":[2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<h2 id="learn-error-rates">5. Learn Error Rates</h2>
<p>Now it is time to assess the error rate of the data. The DADA2 algorithm uses a parametric error model. Every amplicon data set has a different set of error rates and the <code>learnErrors</code> method learns this error model <em>from the data</em>. It does this by alternating estimation of the error rates and inference of sample composition until they converge on a jointly consistent solution. The algorithm begins with an initial guess, for which the maximum possible error rates in the data are used.</p>
<h4 id="forward-reads">Forward Reads</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>errF <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(filtFs, <span class="at">multithread =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<pre><code>103002523 total bases in 443994 reads from 6 samples will be used for learning the error rates.</code></pre>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errF, <span class="at">nominalQ=</span><span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/dada2/ssu18_plot_errorF_1.png" width="1050" /></p>
</div>
<p>The error rates for each possible transition (A to C, A to G, etc.) are shown. Points are the observed error rates for each consensus quality score. The black line shows the estimated error rates after convergence of the machine-learning algorithm. The red line shows the error rates expected under the nominal definition of the Q-score. Here the estimated error rates (black line) are a good fit to the observed rates (points), and the error rates drop with increased quality as expected.</p>
<h2 id="dereplicate-reads">6. Dereplicate Reads</h2>
<p>Now we can use <code>derepFastq</code> to identify the unique sequences in the forward and reverse fastq files.</p>
<h4 id="forward-reads-1">Forward Reads</h4>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>derepFs <span class="ot">&lt;-</span> <span class="fu">derepFastq</span>(filtFs)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(derepFs) <span class="ot">&lt;-</span> sample.names</span></code></pre></div>
</div>
<details>
<summary>
Detailed results of <code>derep</code> forward reads
</summary>
<pre><code>
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P1_R1.fastq.gz
Encountered 39950 unique sequences from 79804 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P1-8C_R1.fastq.gz
Encountered 31163 unique sequences from 72760 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P10_R1.fastq.gz
Encountered 35274 unique sequences from 62598 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P2_R1.fastq.gz
Encountered 42119 unique sequences from 81240 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P3_R1.fastq.gz
Encountered 40336 unique sequences from 86732 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P3-8C_R1.fastq.gz
Encountered 31104 unique sequences from 60860 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P4_R1.fastq.gz
Encountered 35169 unique sequences from 63037 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P5_R1.fastq.gz
Encountered 27752 unique sequences from 54127 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P5-8C_R1.fastq.gz
Encountered 49045 unique sequences from 109421 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P6_R1.fastq.gz
Encountered 46853 unique sequences from 94740 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P7_R1.fastq.gz
Encountered 35363 unique sequences from 71678 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P7-8C_R1.fastq.gz
Encountered 35567 unique sequences from 73639 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P8_R1.fastq.gz
Encountered 38518 unique sequences from 72263 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P9_R1.fastq.gz
Encountered 43426 unique sequences from 89315 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P9-8C_R1.fastq.gz
Encountered 13673 unique sequences from 33000 total sequences read.
</code></pre>
</details>
<p><br/></p>
<h2 id="dada2-asv-inference">7. DADA2 &amp; ASV Inference</h2>
<p>At this point we are ready to apply the core sample inference algorithm (dada) to the filtered and trimmed sequence data. DADA2 offers three options for whether and how to pool samples for ASV inference.</p>
<p>If <code>pool = TRUE</code>, the algorithm will pool together all samples prior to sample inference.<br />
If <code>pool = FALSE</code>, sample inference is performed on each sample individually.<br />
If <code>pool = "pseudo"</code>, the algorithm will perform pseudo-pooling between individually processed samples.</p>
<p>We tested all three methods through the full pipeline. Click the <code>+</code> to see the results of the test. For our final analysis, we chose <code>pool = FALSE</code> for this data set.</p>
<details>
<summary>
Show/hide Results of <code>dada</code> pooling options
</summary>
<p>Here are a few summary tables of results from the tests. Values are from the final sequence table prodcuced by each option.</p>
<p><br/></p>
<em>Total number of reads, total number of ASV, minimum and maximum ASVs, followed by the number of singletons, doubletons, etc.</em>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">unpooled</th>
<th style="text-align: right;">psuedo</th>
<th style="text-align: right;">pooled</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">no_reads</td>
<td style="text-align: right;">937761</td>
<td style="text-align: right;">942031</td>
<td style="text-align: right;">906077</td>
</tr>
<tr class="even">
<td style="text-align: left;">no_asvs</td>
<td style="text-align: right;">20332</td>
<td style="text-align: right;">21269</td>
<td style="text-align: right;">13206</td>
</tr>
<tr class="odd">
<td style="text-align: left;">min_asv</td>
<td style="text-align: right;">816</td>
<td style="text-align: right;">1248</td>
<td style="text-align: right;">2207</td>
</tr>
<tr class="even">
<td style="text-align: left;">max_asv</td>
<td style="text-align: right;">3082</td>
<td style="text-align: right;">4155</td>
<td style="text-align: right;">7522</td>
</tr>
<tr class="odd">
<td style="text-align: left;">n_samp_1</td>
<td style="text-align: right;">17122</td>
<td style="text-align: right;">17948</td>
<td style="text-align: right;">2430</td>
</tr>
<tr class="even">
<td style="text-align: left;">n_samp_2</td>
<td style="text-align: right;">1270</td>
<td style="text-align: right;">309</td>
<td style="text-align: right;">1283</td>
</tr>
<tr class="odd">
<td style="text-align: left;">n_samp_3</td>
<td style="text-align: right;">586</td>
<td style="text-align: right;">265</td>
<td style="text-align: right;">890</td>
</tr>
<tr class="even">
<td style="text-align: left;">n_samp_4</td>
<td style="text-align: right;">357</td>
<td style="text-align: right;">238</td>
<td style="text-align: right;">797</td>
</tr>
<tr class="odd">
<td style="text-align: left;">n_samp_5</td>
<td style="text-align: right;">215</td>
<td style="text-align: right;">250</td>
<td style="text-align: right;">752</td>
</tr>
<tr class="even">
<td style="text-align: left;">n_samp_6</td>
<td style="text-align: right;">161</td>
<td style="text-align: right;">245</td>
<td style="text-align: right;">694</td>
</tr>
<tr class="odd">
<td style="text-align: left;">n_samp_7</td>
<td style="text-align: right;">130</td>
<td style="text-align: right;">253</td>
<td style="text-align: right;">711</td>
</tr>
<tr class="even">
<td style="text-align: left;">n_samp_8</td>
<td style="text-align: right;">84</td>
<td style="text-align: right;">275</td>
<td style="text-align: right;">705</td>
</tr>
<tr class="odd">
<td style="text-align: left;">n_samp_9</td>
<td style="text-align: right;">102</td>
<td style="text-align: right;">229</td>
<td style="text-align: right;">717</td>
</tr>
<tr class="even">
<td style="text-align: left;">n_samp_10</td>
<td style="text-align: right;">63</td>
<td style="text-align: right;">235</td>
<td style="text-align: right;">733</td>
</tr>
<tr class="odd">
<td style="text-align: left;">n_samp_11</td>
<td style="text-align: right;">55</td>
<td style="text-align: right;">210</td>
<td style="text-align: right;">715</td>
</tr>
<tr class="even">
<td style="text-align: left;">n_samp_12</td>
<td style="text-align: right;">60</td>
<td style="text-align: right;">213</td>
<td style="text-align: right;">726</td>
</tr>
<tr class="odd">
<td style="text-align: left;">n_samp_13</td>
<td style="text-align: right;">36</td>
<td style="text-align: right;">225</td>
<td style="text-align: right;">750</td>
</tr>
<tr class="even">
<td style="text-align: left;">n_samp_14</td>
<td style="text-align: right;">42</td>
<td style="text-align: right;">205</td>
<td style="text-align: right;">730</td>
</tr>
<tr class="odd">
<td style="text-align: left;">n_samp_15</td>
<td style="text-align: right;">49</td>
<td style="text-align: right;">169</td>
<td style="text-align: right;">573</td>
</tr>
</tbody>
</table>
</div>
<em>Total number of reads by sample.</em>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">unpooled</th>
<th style="text-align: right;">psuedo</th>
<th style="text-align: right;">pooled</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">P1</td>
<td style="text-align: right;">68037</td>
<td style="text-align: right;">68529</td>
<td style="text-align: right;">65521</td>
</tr>
<tr class="even">
<td style="text-align: left;">P1-8C</td>
<td style="text-align: right;">60535</td>
<td style="text-align: right;">60699</td>
<td style="text-align: right;">57636</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P10</td>
<td style="text-align: right;">53625</td>
<td style="text-align: right;">53827</td>
<td style="text-align: right;">52032</td>
</tr>
<tr class="even">
<td style="text-align: left;">P2</td>
<td style="text-align: right;">70436</td>
<td style="text-align: right;">70436</td>
<td style="text-align: right;">67208</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P3</td>
<td style="text-align: right;">75183</td>
<td style="text-align: right;">75470</td>
<td style="text-align: right;">71954</td>
</tr>
<tr class="even">
<td style="text-align: left;">P3-8C</td>
<td style="text-align: right;">53437</td>
<td style="text-align: right;">53134</td>
<td style="text-align: right;">51893</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P4</td>
<td style="text-align: right;">53813</td>
<td style="text-align: right;">54115</td>
<td style="text-align: right;">52133</td>
</tr>
<tr class="even">
<td style="text-align: left;">P5</td>
<td style="text-align: right;">47070</td>
<td style="text-align: right;">47328</td>
<td style="text-align: right;">45012</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P5-8C</td>
<td style="text-align: right;">86841</td>
<td style="text-align: right;">87418</td>
<td style="text-align: right;">85245</td>
</tr>
<tr class="even">
<td style="text-align: left;">P6</td>
<td style="text-align: right;">81219</td>
<td style="text-align: right;">81416</td>
<td style="text-align: right;">78759</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P7</td>
<td style="text-align: right;">61742</td>
<td style="text-align: right;">62091</td>
<td style="text-align: right;">58740</td>
</tr>
<tr class="even">
<td style="text-align: left;">P7-8C</td>
<td style="text-align: right;">62202</td>
<td style="text-align: right;">61917</td>
<td style="text-align: right;">58300</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P8</td>
<td style="text-align: right;">62171</td>
<td style="text-align: right;">62527</td>
<td style="text-align: right;">60245</td>
</tr>
<tr class="even">
<td style="text-align: left;">P9</td>
<td style="text-align: right;">76299</td>
<td style="text-align: right;">76671</td>
<td style="text-align: right;">72833</td>
</tr>
</tbody>
</table>
</div>
<em>Total number of ASVs by sample.</em>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">unpooled</th>
<th style="text-align: right;">psuedo</th>
<th style="text-align: right;">pooled</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">P1</td>
<td style="text-align: right;">2383</td>
<td style="text-align: right;">3385</td>
<td style="text-align: right;">6542</td>
</tr>
<tr class="even">
<td style="text-align: left;">P1-8C</td>
<td style="text-align: right;">1618</td>
<td style="text-align: right;">2434</td>
<td style="text-align: right;">4927</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P10</td>
<td style="text-align: right;">2070</td>
<td style="text-align: right;">3055</td>
<td style="text-align: right;">6361</td>
</tr>
<tr class="even">
<td style="text-align: left;">P2</td>
<td style="text-align: right;">2278</td>
<td style="text-align: right;">3309</td>
<td style="text-align: right;">6809</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P3</td>
<td style="text-align: right;">2368</td>
<td style="text-align: right;">3277</td>
<td style="text-align: right;">6324</td>
</tr>
<tr class="even">
<td style="text-align: left;">P3-8C</td>
<td style="text-align: right;">1593</td>
<td style="text-align: right;">2246</td>
<td style="text-align: right;">4658</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P4</td>
<td style="text-align: right;">2052</td>
<td style="text-align: right;">3086</td>
<td style="text-align: right;">6669</td>
</tr>
<tr class="even">
<td style="text-align: left;">P5</td>
<td style="text-align: right;">1672</td>
<td style="text-align: right;">2565</td>
<td style="text-align: right;">5275</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P5-8C</td>
<td style="text-align: right;">2156</td>
<td style="text-align: right;">2946</td>
<td style="text-align: right;">5608</td>
</tr>
<tr class="even">
<td style="text-align: left;">P6</td>
<td style="text-align: right;">3082</td>
<td style="text-align: right;">4155</td>
<td style="text-align: right;">7522</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P7</td>
<td style="text-align: right;">2025</td>
<td style="text-align: right;">3049</td>
<td style="text-align: right;">6256</td>
</tr>
<tr class="even">
<td style="text-align: left;">P7-8C</td>
<td style="text-align: right;">2167</td>
<td style="text-align: right;">2998</td>
<td style="text-align: right;">5592</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P8</td>
<td style="text-align: right;">2453</td>
<td style="text-align: right;">3446</td>
<td style="text-align: right;">6791</td>
</tr>
<tr class="even">
<td style="text-align: left;">P9</td>
<td style="text-align: right;">2402</td>
<td style="text-align: right;">3412</td>
<td style="text-align: right;">6779</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P9-8C</td>
<td style="text-align: right;">816</td>
<td style="text-align: right;">1248</td>
<td style="text-align: right;">2207</td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>dadaFs <span class="ot">&lt;-</span> <span class="fu">dada</span>(derepFs, <span class="at">err =</span> errF, <span class="at">multithread =</span> <span class="cn">TRUE</span>, <span class="at">pool =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<details>
<summary>
Detailed results of <code>dada</code> on forward reads
</summary>
<pre><code>
Sample 1 - 79804 reads in 39950 unique sequences.
Sample 2 - 72760 reads in 31163 unique sequences.
Sample 3 - 62598 reads in 35274 unique sequences.
Sample 4 - 81240 reads in 42119 unique sequences.
Sample 5 - 86732 reads in 40336 unique sequences.
Sample 6 - 60860 reads in 31104 unique sequences.
Sample 7 - 63037 reads in 35169 unique sequences.
Sample 8 - 54127 reads in 27752 unique sequences.
Sample 9 - 109421 reads in 49045 unique sequences.
Sample 10 - 94740 reads in 46853 unique sequences.
Sample 11 - 71678 reads in 35363 unique sequences.
Sample 12 - 73639 reads in 35567 unique sequences.
Sample 13 - 72263 reads in 38518 unique sequences.
Sample 14 - 89315 reads in 43426 unique sequences.
Sample 15 - 33000 reads in 13673 unique sequences.
</code></pre>
</details>
<p>As an example, we can inspect the returned <code>dada-class</code> object for the forward and reverse reads from the sample #2:</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>dadaFs[[sampnum]]</span></code></pre></div>
</div>
<pre><code>dada-class: object describing DADA2 denoising results
1759 sequence variants were inferred from 31163 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
<p>This output tells us how many true sequence variants the DADA2 algorithm inferred from the unique sequences, in this case the sample 2.</p>
<h2 id="construct-sequence-table">8. Construct Sequence Table</h2>
<p>Now we construct an amplicon sequence variant (ASV) table.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>seqtab <span class="ot">&lt;-</span> <span class="fu">makeSequenceTable</span>(dadaFs)</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1]    15 14985</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab)))</span></code></pre></div>
</div>
<pre><code>   22    23    24    25    27    41    66    73   120   149   185   190   193 
    2     1     1     2     1     1     1     1     1     1     1     9     1 
  198   200   201   202   208   209   211   215   217   220   221   222   223 
    1     1     1     1     1     1     1     1     2     4    14    21     8 
  224   225   227   228   229   230   231   232   233   234   238   251 
    3     4     6     4     5     9    20 21342   190    12     2     8 </code></pre>
<p>The sequence table is a matrix with rows corresponding to (and named by) the samples, and columns corresponding to (and named by) the sequence variants. We have 14985 sequence variants but also a range of sequence lengths. Since many of these sequence variants are singletons or doubletons, we can just select a range that corresponds to the expected amplicon length and eliminate the spurious reads.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/dada2/ssu18_read_length_before.png" width="1050" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>seqtab<span class="fl">.2</span> <span class="ot">&lt;-</span> seqtab[,<span class="fu">nchar</span>(<span class="fu">colnames</span>(seqtab)) <span class="sc">%in%</span> <span class="fu">seq</span>(<span class="dv">230</span>,<span class="dv">235</span>)]</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab<span class="fl">.2</span>)</span></code></pre></div>
</div>
<aside>
The values you select should be based on the sequence table generated above.
</aside>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1]    15 21573</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab<span class="fl">.2</span>)))</span></code></pre></div>
</div>
<pre><code>  230   231   232   233   234 
    9    20 21342   190    12 </code></pre>
<p>After removing the extreme length variants, we have 21573, a reduction of -6588 sequence variants.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="remove-chimeras">9. Remove Chimeras</h2>
<p>Even though the <code>dada</code> method corrects substitution and indel errors, chimeric sequences remain. According to the DADA2 documentation, the accuracy of sequence variants after denoising makes identifying chimeric ASVs simpler than when dealing with fuzzy OTUs. Chimeric sequences are identified if they can be <em>exactly reconstructed</em> by combining a left-segment and a right-segment from two more abundant <em>parent</em> sequences.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>seqtab.nochim <span class="ot">&lt;-</span> <span class="fu">removeBimeraDenovo</span>(seqtab<span class="fl">.2</span>,</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">method =</span> <span class="st">&quot;consensus&quot;</span>,</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">multithread =</span> <span class="dv">20</span>,</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab.nochim)</span></code></pre></div>
</div>
<pre><code>Identified 1241 bimeras out of 21573 input sequences.</code></pre>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1]    15 20332</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(seqtab.nochim)<span class="sc">/</span><span class="fu">sum</span>(seqtab<span class="fl">.2</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] 0.9596595</code></pre>
</div>
<p>Chimera checking removed an additional 1241 sequence variants however, when we account for the abundances of each variant, we see chimeras accounts for about 4.03405% of the merged sequence reads. Not bad.</p>
<p>The last thing we want to do is write the sequence table to an <code>RDS</code> file.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(seqtab.nochim, <span class="st">&quot;ssu18_seqtab.nochim.rds&quot;</span>)</span></code></pre></div>
</div>
<h2 id="track-reads-through-workflow">10. Track Reads through Workflow</h2>
<p>At this point we can look at the number of reads that made it through each step of the workflow for every sample.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>getN <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">sum</span>(<span class="fu">getUniques</span>(x))</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>track <span class="ot">&lt;-</span> <span class="fu">cbind</span>(out,</span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>               <span class="fu">sapply</span>(dadaFs, getN),</span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>               <span class="fu">rowSums</span>(seqtab.nochim))</span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(track) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;input&quot;</span>, <span class="st">&quot;filtered&quot;</span>, <span class="st">&quot;denoisedF&quot;</span>, <span class="st">&quot;nonchim&quot;</span>)</span>
<span id="cb57-6"><a href="#cb57-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(track) <span class="ot">&lt;-</span> sample.names</span></code></pre></div>
</div>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="5becfm0ylbqxfrglu40z" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="5becfm0ylbqxfrglu40z">{"x":{"filter":"none","extensions":["Buttons"],"data":[["P1","P1-8C","P10","P2","P3","P3-8C","P4","P5","P5-8C","P6","P7","P7-8C","P8","P9","P9-8C"],[85489,77645,67583,87091,93065,66401,67810,58266,120892,101599,76781,79032,77599,95566,95182],[85482,77642,67579,87085,93053,66367,67800,58259,120842,101592,76776,79022,77591,95564,94457],[85402,77470,67432,86878,92851,65498,67617,58006,117015,101363,76583,78724,77133,95310,44250],[79804,72760,62598,81240,86732,60860,63037,54127,109421,94740,71678,73639,72263,89315,33000],[70434,65950,54075,71961,78067,54822,54678,47868,100026,83190,64031,65125,63248,79799,31620],[68037,60535,53625,70436,75183,53437,53813,47070,86841,81219,61742,62202,62171,76299,25151]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>raw<\/th>\n      <th>pre_filt<\/th>\n      <th>cut<\/th>\n      <th>filtered<\/th>\n      <th>denoisedF<\/th>\n      <th>nonchim<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<p>And we can save a copy of the table for later use.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(track, <span class="st">&quot;ssu18_read_changes.txt&quot;</span>,</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="assign-taxonomy">11. Assign Taxonomy</h2>
<p>The <code>assignTaxonomy</code> command implements the naive Bayesian classifier, so for reproducible results you need to set a random number seed (see issue <a href="https://github.com/benjjneb/dada2/issues/538">#538</a>). We did this at the beginning of the workflow. For taxonomic assignment, we are using the Silva version 138 <span class="citation" data-cites="quast2012silva">(<a href="#ref-quast2012silva" role="doc-biblioref">Quast et al. 2012</a>)</span>. The developers of DADA2 maintain a <a href="https://benjjneb.github.io/dada2/training.html">formatted version of the database</a>.</p>
<p>We will read in the <code>RDS</code> file containing the sequence table saved above. We also need to run <code>remove(list = ls())</code> command, otherwise the final image we save will be huge. This way, the image only contains the sample data, seqtab, and taxtab <em>after</em> running <code>removeBimeraDenovo</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>seqtab <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;ssu18_seqtab.nochim.rds&quot;</span>)</span></code></pre></div>
</div>
<p>And then native Bayesian classifier against the Silva database.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>tax_silva <span class="ot">&lt;-</span> <span class="fu">assignTaxonomy</span>(</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>  seqtab, <span class="st">&quot;silva_nr_v138_train_set.fa.gz&quot;</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<h2 id="save-image">12. Save Image</h2>
<p>And finally, we save an image for use in the analytic part of the workflow. This R data file will be needed as the input for the phyloseq portion of the workflow. See the <a href="data-availability.html">Data Availability</a> page for complete details on where to get this file.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="st">&quot;ssu18_dada2_wf.rdata&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="r-session-information-code">R Session Information &amp; Code</h2>
<p>This workflow was run on the <a href="https://doi.org/10.25572/SIHPC">Smithsonian High Performance Cluster (SI/HPC)</a>, Smithsonian Institution. Below are the specific packages and versions used in this workflow using both <code>sessionInfo()</code> and <code>devtools::session_info()</code>. Click the arrow to see the details.</p>
<details>
<summary>
Show/hide R Session Info
</summary>
<pre><code>sessionInfo()
R version 4.0.0 (2020-04-24)
Platform: x86_64-conda_cos6-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /home/scottjj/miniconda3/envs/R-4/lib/libopenblasp-r0.3.9.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
 [1] grid      stats4    parallel  stats     graphics  grDevices utils
 [8] datasets  methods   base

other attached packages:
 [1] gridExtra_2.3               ggplot2_3.3.0
 [3] ShortRead_1.46.0            GenomicAlignments_1.24.0
 [5] SummarizedExperiment_1.18.1 DelayedArray_0.14.0
 [7] matrixStats_0.56.0          Biobase_2.48.0
 [9] Rsamtools_2.4.0             GenomicRanges_1.40.0
[11] GenomeInfoDb_1.24.0         Biostrings_2.56.0
[13] XVector_0.28.0              IRanges_2.22.1
[15] S4Vectors_0.26.0            BiocParallel_1.22.0
[17] BiocGenerics_0.34.0         dada2_1.16.0
[19] Rcpp_1.0.4.6

loaded via a namespace (and not attached):
 [1] plyr_1.8.6             RColorBrewer_1.1-2     pillar_1.4.4
 [4] compiler_4.0.0         bitops_1.0-6           tools_4.0.0
 [7] zlibbioc_1.34.0        lifecycle_0.2.0        tibble_3.0.1
[10] gtable_0.3.0           lattice_0.20-41        png_0.1-7
[13] pkgconfig_2.0.3        rlang_0.4.6            Matrix_1.2-18
[16] GenomeInfoDbData_1.2.3 withr_2.2.0            stringr_1.4.0
[19] hwriter_1.3.2          vctrs_0.3.0            glue_1.4.1
[22] R6_2.4.1               jpeg_0.1-8.1           latticeExtra_0.6-29
[25] reshape2_1.4.4         magrittr_1.5           scales_1.1.1
[28] ellipsis_0.3.1         colorspace_1.4-1       stringi_1.4.6
[31] RCurl_1.98-1.2         RcppParallel_5.0.1     munsell_0.5.0
[34] crayon_1.3.4

devtools::session_info()
─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.0.0 (2020-04-24)
 os       CentOS Linux 7 (Core)
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       America/New_York
 date     2020-06-17

─ Packages ───────────────────────────────────────────────────────────────────
 package              * version  date       lib source
 assertthat             0.2.1    2019-03-21 [1] CRAN (R 4.0.0)
 backports              1.1.7    2020-05-13 [1] CRAN (R 4.0.0)
 Biobase              * 2.48.0   2020-04-27 [1] Bioconductor
 BiocGenerics         * 0.34.0   2020-04-27 [1] Bioconductor
 BiocParallel         * 1.22.0   2020-04-27 [1] Bioconductor
 Biostrings           * 2.56.0   2020-04-27 [1] Bioconductor
 bitops                 1.0-6    2013-08-17 [1] CRAN (R 4.0.0)
 callr                  3.4.3    2020-03-28 [1] CRAN (R 4.0.0)
 cli                    2.0.2    2020-02-28 [1] CRAN (R 4.0.0)
 colorspace             1.4-1    2019-03-18 [1] CRAN (R 4.0.0)
 crayon                 1.3.4    2017-09-16 [1] CRAN (R 4.0.0)
 dada2                * 1.16.0   2020-04-27 [1] Bioconductor
 DelayedArray         * 0.14.0   2020-04-27 [1] Bioconductor
 desc                   1.2.0    2018-05-01 [1] CRAN (R 4.0.0)
 devtools               2.3.0    2020-04-10 [1] CRAN (R 4.0.0)
 digest                 0.6.25   2020-02-23 [1] CRAN (R 4.0.0)
 ellipsis               0.3.1    2020-05-15 [1] CRAN (R 4.0.0)
 fansi                  0.4.1    2020-01-08 [1] CRAN (R 4.0.0)
 fs                     1.4.1    2020-04-04 [1] CRAN (R 4.0.0)
 GenomeInfoDb         * 1.24.0   2020-04-27 [1] Bioconductor
 GenomeInfoDbData       1.2.3    2020-05-22 [1] Bioconductor
 GenomicAlignments    * 1.24.0   2020-04-27 [1] Bioconductor
 GenomicRanges        * 1.40.0   2020-04-27 [1] Bioconductor
 ggplot2              * 3.3.0    2020-03-05 [1] CRAN (R 4.0.0)
 glue                   1.4.1    2020-05-13 [1] CRAN (R 4.0.0)
 gridExtra            * 2.3      2017-09-09 [1] CRAN (R 4.0.0)
 gtable                 0.3.0    2019-03-25 [1] CRAN (R 4.0.0)
 hwriter                1.3.2    2014-09-10 [1] CRAN (R 4.0.0)
 IRanges              * 2.22.1   2020-04-28 [1] Bioconductor
 jpeg                   0.1-8.1  2019-10-24 [1] CRAN (R 4.0.0)
 lattice                0.20-41  2020-04-02 [1] CRAN (R 4.0.0)
 latticeExtra           0.6-29   2019-12-19 [1] CRAN (R 4.0.0)
 lifecycle              0.2.0    2020-03-06 [1] CRAN (R 4.0.0)
 magrittr               1.5      2014-11-22 [1] CRAN (R 4.0.0)
 Matrix                 1.2-18   2019-11-27 [1] CRAN (R 4.0.0)
 matrixStats          * 0.56.0   2020-03-13 [1] CRAN (R 4.0.0)
 memoise                1.1.0    2017-04-21 [1] CRAN (R 4.0.0)
 munsell                0.5.0    2018-06-12 [1] CRAN (R 4.0.0)
 pillar                 1.4.4    2020-05-05 [1] CRAN (R 4.0.0)
 pkgbuild               1.0.8    2020-05-07 [1] CRAN (R 4.0.0)
 pkgconfig              2.0.3    2019-09-22 [1] CRAN (R 4.0.0)
 pkgload                1.0.2    2018-10-29 [1] CRAN (R 4.0.0)
 plyr                   1.8.6    2020-03-03 [1] CRAN (R 4.0.0)
 png                    0.1-7    2013-12-03 [1] CRAN (R 4.0.0)
 prettyunits            1.1.1    2020-01-24 [1] CRAN (R 4.0.0)
 processx               3.4.2    2020-02-09 [1] CRAN (R 4.0.0)
 ps                     1.3.3    2020-05-08 [1] CRAN (R 4.0.0)
 R6                     2.4.1    2019-11-12 [1] CRAN (R 4.0.0)
 RColorBrewer           1.1-2    2014-12-07 [1] CRAN (R 4.0.0)
 Rcpp                 * 1.0.4.6  2020-04-09 [1] CRAN (R 4.0.0)
 RcppParallel           5.0.1    2020-05-06 [1] CRAN (R 4.0.0)
 RCurl                  1.98-1.2 2020-04-18 [1] CRAN (R 4.0.0)
 remotes                2.1.1    2020-02-15 [1] CRAN (R 4.0.0)
 reshape2               1.4.4    2020-04-09 [1] CRAN (R 4.0.0)
 rlang                  0.4.6    2020-05-02 [1] CRAN (R 4.0.0)
 rprojroot              1.3-2    2018-01-03 [1] CRAN (R 4.0.0)
 Rsamtools            * 2.4.0    2020-04-27 [1] Bioconductor
 S4Vectors            * 0.26.0   2020-04-27 [1] Bioconductor
 scales                 1.1.1    2020-05-11 [1] CRAN (R 4.0.0)
 sessioninfo            1.1.1    2018-11-05 [1] CRAN (R 4.0.0)
 ShortRead            * 1.46.0   2020-04-27 [1] Bioconductor
 stringi                1.4.6    2020-02-17 [1] CRAN (R 4.0.0)
 stringr                1.4.0    2019-02-10 [1] CRAN (R 4.0.0)
 SummarizedExperiment * 1.18.1   2020-04-30 [1] Bioconductor
 testthat               2.3.2    2020-03-02 [1] CRAN (R 4.0.0)
 tibble                 3.0.1    2020-04-20 [1] CRAN (R 4.0.0)
 usethis                1.6.1    2020-04-29 [1] CRAN (R 4.0.0)
 vctrs                  0.3.0    2020-05-11 [1] CRAN (R 4.0.0)
 withr                  2.2.0    2020-04-20 [1] CRAN (R 4.0.0)
 XVector              * 0.28.0   2020-04-27 [1] Bioconductor
 zlibbioc               1.34.0   2020-04-27 [1] Bioconductor</code></pre>
</details>
<p><br/></p>
<h1 id="its-data">ITS Data</h1>
<h2 id="overview-1">Overview</h2>
<h3 id="sequence-files-samples-1">Sequence Files &amp; Samples</h3>
<p>We sequenced a total of 15 samples collected from 10 different plots at a single depth in each plot.</p>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="htmlwidget-3a0ad04372aab487f689" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-3a0ad04372aab487f689">{"x":{"filter":"none","extensions":["Buttons"],"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15"],["P1","P2","P3","P4","P5","P6","P7","P8","P9","P10","P1-8C","P3-8C","P5-8C","P7-8C","P9-8C"],["P01","P02","P03","P04","P05","P06","P07","P08","P09","P10","P01","P03","P05","P07","P09"],["Warm","Control","Warm","Control","Warm","Control","Warm","Control","Warm","Control","Warm","Warm","Warm","Warm","Warm"],[3,0,3,0,3,0,3,0,3,0,8,8,8,8,8],["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],["A","A","B","B","C","C","D","D","E","E","A","B","C","D","E"],["P1","P2","P3","P4","P5","P6","P7","P8","P9","P10","P1-8C","P3-8C","P5-8C","P7-8C","P9-8C"],["SC701R","SC702R","SC701R","SC701R","SC701R","SC701R","SC701R","SC701R","SC701R","SC702R","SC702R","SC702R","SC704R","SC708R","SC702R"],["cagtaggt","atagcgct","cagtaggt","cagtaggt","cagtaggt","cagtaggt","cagtaggt","cagtaggt","cagtaggt","atagcgct","atagcgct","atagcgct","tcctcatg","agcatacc","atagcgct"],["SC508","SC501","SC506","SC507","SC501","SC502","SC503","SC504","SC505","SC507","SC506","SC503","SC502","SC508","SC502"],["TAGTGTAG","ACGACGTG","GACACTGA","TGCGTACG","ACGACGTG","ATATACAC","CGTCGCTA","CTAGAGCT","GCTCTAGT","TGCGTACG","GACACTGA","CGTCGCTA","ATATACAC","TAGTGTAG","ATATACAC"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Sample_ID<\/th>\n      <th>Plot<\/th>\n      <th>Treatment<\/th>\n      <th>Temp<\/th>\n      <th>Depth<\/th>\n      <th>Pairing<\/th>\n      <th>Fastq_Files_ID<\/th>\n      <th>I7_Index_ID<\/th>\n      <th>index<\/th>\n      <th>I5_Index_ID<\/th>\n      <th>index2<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"className":"dt-right","targets":4},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<h3 id="workflow-1">Workflow</h3>
<p>Our workflow is pretty much ripped from the <a href="https://benjjneb.github.io/dada2/ITS_workflow.html">DADA2 ITS Workflow (1.8)</a> on the DADA2 website. That webpage contains thorough explanations for each step so we will not repeat most of that here. For more details, check out the post. The workflow consists of the following steps:</p>
<table>
<thead>
<tr class="header">
<th>Step</th>
<th>Command</th>
<th>What we’re doing</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>multiple</td>
<td>prepare input file names &amp; paths</td>
</tr>
<tr class="even">
<td>2</td>
<td>multiple</td>
<td>Define primers (all orientations)</td>
</tr>
<tr class="odd">
<td>3</td>
<td><code>cutadapt</code></td>
<td>Remove primers</td>
</tr>
<tr class="even">
<td>4</td>
<td><code>filterAndTrim()</code></td>
<td>assess quality &amp; filter reads</td>
</tr>
<tr class="odd">
<td>5</td>
<td><code>learnErrors()</code></td>
<td>generate an error model for the data</td>
</tr>
<tr class="even">
<td>6</td>
<td><code>derepFastq</code></td>
<td>dereplicate sequences</td>
</tr>
<tr class="odd">
<td>7</td>
<td><code>dada()</code></td>
<td>infer ASVs on both forward &amp; reverse reads independently</td>
</tr>
<tr class="even">
<td>8</td>
<td><code>mergePairs()</code></td>
<td>merge forward &amp; reverse reads to further refine ASVs</td>
</tr>
<tr class="odd">
<td>9</td>
<td><code>makeSequenceTable()</code></td>
<td>generate a count table</td>
</tr>
<tr class="even">
<td>10</td>
<td><code>removeBimeraDenovo()</code></td>
<td>screen for &amp; remove chimeras</td>
</tr>
<tr class="odd">
<td>11</td>
<td></td>
<td>track reads through workflow</td>
</tr>
<tr class="even">
<td>12</td>
<td><code>assignTaxonomy()</code></td>
<td>assign taxonomy &amp; finish workflow</td>
</tr>
<tr class="odd">
<td>13</td>
<td><code>save.image()</code></td>
<td>save an image of seqtab &amp; taxtab for next part of workflow</td>
</tr>
</tbody>
</table>
<h2 id="set-working-environment-1">1. Set Working Environment</h2>
<p>Next, we need to setup the working environment by renaming the fastq files and defining a path for the working directory.</p>
<h4 id="rename-samples-1">Rename samples</h4>
<p>To make the parsing easier, we will eliminate the lane and well number from each sample. If you do not wish to do that you will need to adjust the code accordingly. I am <em>positive</em> there are more elegant ways of doing this.</p>
<blockquote>
<p>CAUTION. You should check that this code works on a few backup files before proceeding.</p>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;RAW&quot;</span>)</span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>orig_fastq <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">pattern =</span> <span class="st">&quot;*.fastq.gz&quot;</span>)</span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>newname_fastq <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_S.*_L001&quot;</span>, <span class="st">&quot;&quot;</span>, orig_fastq)</span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>newname_fastq <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_001&quot;</span>, <span class="st">&quot;&quot;</span>, newname_fastq)</span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a><span class="fu">file.rename</span>(orig_fastq, newname_fastq)</span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;../&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>path <span class="ot">&lt;-</span> <span class="st">&quot;RAW&quot;</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">list.files</span>(path))</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;P1_R1.fastq.gz&quot;    &quot;P1_R2.fastq.gz&quot;    &quot;P1-8C_R1.fastq.gz&quot;
[4] &quot;P1-8C_R2.fastq.gz&quot; &quot;P10_R1.fastq.gz&quot;   &quot;P10_R2.fastq.gz&quot;  </code></pre>
</div>
<p>Then, we generate matched lists of the forward and reverse read files. We also parse out the sample name.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>fnFs <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern =</span> <span class="st">&quot;_R1.fastq.gz&quot;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>fnRs <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern =</span> <span class="st">&quot;_R2.fastq.gz&quot;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span></code></pre></div>
</div>
<h2 id="define-primers-1">2. Define Primers</h2>
<p>Before we start the DADA2 workflow we need to run <a href="https://github.com/marcelm/cutadapt">catadapt</a> <span class="citation" data-cites="martin2011cutadapt">(<a href="#ref-martin2011cutadapt" role="doc-biblioref">Martin 2011</a>)</span> on all <code>fastq.gz</code> files to trim the primers. For this part of the study we used the primer pair ITS1f (CTTGGTCATTTAGAGGAAGTAA) and ITS2 (GCTGCGTTCTTCATCGATGC)[REF] which should yield variable amplicon lengths between 100 to 400 bp. What we are looking for is potential read-through scenarios that are possible when sequencing the length variable ITS region as described in the <a href="https://benjjneb.github.io/dada2/ITS_workflow.html">DADA2 ITS Workflow (1.8)</a>. Please refer to this document for a complete explanation.</p>
<p>First we define the primers.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>FWD <span class="ot">&lt;-</span> <span class="st">&quot;CTTGGTCATTTAGAGGAAGTAA&quot;</span></span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>REV <span class="ot">&lt;-</span> <span class="st">&quot;GCTGCGTTCTTCATCGATGC&quot;</span></span></code></pre></div>
</div>
<p>Next, we check the presence and orientation of these primers in the data. To do this we will create all orientations of the input primer sequences. In other words the <em>Forward</em>, <em>Complement</em>, <em>Reverse</em>, and <em>Reverse Complement</em> variations.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>allOrients <span class="ot">&lt;-</span> <span class="cf">function</span>(primer) {</span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">require</span>(Biostrings)</span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>    dna <span class="ot">&lt;-</span> <span class="fu">DNAString</span>(primer)</span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a>    orients <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">Forward =</span> dna,</span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a>                 <span class="at">Complement =</span> <span class="fu">complement</span>(dna),</span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">Reverse =</span> <span class="fu">reverse</span>(dna),</span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">RevComp =</span> <span class="fu">reverseComplement</span>(dna))</span>
<span id="cb68-8"><a href="#cb68-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">sapply</span>(orients, toString))</span>
<span id="cb68-9"><a href="#cb68-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb68-10"><a href="#cb68-10" aria-hidden="true" tabindex="-1"></a>FWD.orients <span class="ot">&lt;-</span> <span class="fu">allOrients</span>(FWD)</span>
<span id="cb68-11"><a href="#cb68-11" aria-hidden="true" tabindex="-1"></a>REV.orients <span class="ot">&lt;-</span> <span class="fu">allOrients</span>(REV)</span>
<span id="cb68-12"><a href="#cb68-12" aria-hidden="true" tabindex="-1"></a>FWD.orients</span>
<span id="cb68-13"><a href="#cb68-13" aria-hidden="true" tabindex="-1"></a>REV.orients</span></code></pre></div>
</div>
<pre><code>                 Forward               Complement                  Reverse
&quot;CTTGGTCATTTAGAGGAAGTAA&quot; &quot;GAACCAGTAAATCTCCTTCATT&quot; &quot;AATGAAGGAGATTTACTGGTTC&quot;
                 RevComp
&quot;TTACTTCCTCTAAATGACCAAG&quot;
               Forward             Complement                Reverse
&quot;GCTGCGTTCTTCATCGATGC&quot; &quot;CGACGCAAGAAGTAGCTACG&quot; &quot;CGTAGCTACTTCTTGCGTCG&quot;
               RevComp
&quot;GCATCGATGAAGAACGCAGC&quot;</code></pre>
<p>Now we do a little pre-filter step to eliminate ambiguous bases (<em>Ns</em>) because <em>Ns</em> make mapping of short primer sequences difficult. This step removes any reads with <em>Ns</em>. Again, set some files paths, this time for the filtered reads.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>fnFs.filtN <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">&quot;filtN&quot;</span>, <span class="fu">basename</span>(fnFs))</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>fnRs.filtN <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">&quot;filtN&quot;</span>, <span class="fu">basename</span>(fnRs))</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a><span class="fu">filterAndTrim</span>(fnFs, fnFs.filtN, fnRs, fnRs.filtN,</span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxN =</span> <span class="dv">0</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<p>Sweet. Time to assess the number of times a primer (and all possible primer orientation) appear in the forward and reverse reads. According to the workflow, counting the primers on one set of paired end fastq files is sufficient to see if there is a problem. This assumes that all the files were created using the same library prep. Basically for both primers, we will search for all four orientations in both forward and reverse reads</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>sampnum <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>primerHits <span class="ot">&lt;-</span> <span class="cf">function</span>(primer, fn) {</span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Counts number of reads in which the primer is found</span></span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>    nhits <span class="ot">&lt;-</span> <span class="fu">vcountPattern</span>(primer, <span class="fu">sread</span>(<span class="fu">readFastq</span>(fn)), <span class="at">fixed =</span> <span class="cn">FALSE</span>)</span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">sum</span>(nhits <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<h4 id="forward-primers-1">Forward primers</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits,</span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">fn =</span> fnFs.filtN[[sampnum]]),</span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">FWD.ReverseReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits,</span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">fn =</span> fnRs.filtN[[sampnum]]))</span></code></pre></div>
</div>
<pre><code>                 Forward Complement Reverse RevComp
FWD.ForwardReads   48467          0       0       0
FWD.ReverseReads       0          0       0    5078</code></pre>
<h4 id="reverse-primers-1">Reverse primers</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits,</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">fn =</span> fnFs.filtN[[sampnum]]),</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">REV.ReverseReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits,</span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">fn =</span> fnRs.filtN[[sampnum]]))</span></code></pre></div>
</div>
<pre><code>                 Forward Complement Reverse RevComp
REV.ForwardReads       0          0       0    6237
REV.ReverseReads   43902          0       0       0</code></pre>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>What does this table mean? I wondered the same thing. Let’s break it down. Sample P1 had 50,894 sequences after the original filtering described earlier. The code searched the forward and reverse fastq files for all 8 primers. If we look at the two outputs, we see the forward primer is found in the forward reads in its forward orientation but also in some reverse reads in its reverse-complement orientation. The reverse primer is found in the reverse reads in its forward orientation but also in some forward reads in its reverse-complement orientation. This is due to read-through when the ITS region is short.</p>
<h2 id="remove-primers-1">3. Remove Primers</h2>
<p>Now we can run <a href="https://github.com/marcelm/cutadapt">catadapt</a> <span class="citation" data-cites="martin2011cutadapt">(<a href="#ref-martin2011cutadapt" role="doc-biblioref">Martin 2011</a>)</span> to remove the primers from the fastq sequences. A little setup first. If this command executes successfully it means R has found cutadapt.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>cutadapt <span class="ot">&lt;-</span> <span class="st">&quot;/PATH/to/cutadapt&quot;</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a><span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="st">&quot;--version&quot;</span>) <span class="co"># Run shell commands from R</span></span></code></pre></div>
</div>
<pre><code>2.8</code></pre>
<p>We set paths and trim the forward primer and the reverse-complement of the reverse primer off of R1 (forward reads) and trim the reverse primer and the reverse-complement of the forward primer off of R2 (reverse reads).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>path.cut <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">&quot;cutadapt&quot;</span>)</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(path.cut)) <span class="fu">dir.create</span>(path.cut)</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>fnFs.cut <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(fnFs))</span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a>fnRs.cut <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(fnRs))</span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a>FWD.RC <span class="ot">&lt;-</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(FWD)</span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a>REV.RC <span class="ot">&lt;-</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(REV)</span>
<span id="cb78-8"><a href="#cb78-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-9"><a href="#cb78-9" aria-hidden="true" tabindex="-1"></a>R1.flags <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;-g&quot;</span>, FWD, <span class="st">&quot;-a&quot;</span>, REV.RC)</span>
<span id="cb78-10"><a href="#cb78-10" aria-hidden="true" tabindex="-1"></a>R2.flags <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;-G&quot;</span>, REV, <span class="st">&quot;-A&quot;</span>, FWD.RC)</span>
<span id="cb78-11"><a href="#cb78-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-12"><a href="#cb78-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(fnFs)) {<span class="fu">system2</span>(cutadapt,</span>
<span id="cb78-13"><a href="#cb78-13" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">args =</span> <span class="fu">c</span>(R1.flags, R2.flags,</span>
<span id="cb78-14"><a href="#cb78-14" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">&quot;-m&quot;</span>, <span class="dv">20</span>, <span class="st">&quot;-n&quot;</span>, <span class="dv">2</span>,</span>
<span id="cb78-15"><a href="#cb78-15" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">&quot;-o&quot;</span>, fnFs.cut[i],</span>
<span id="cb78-16"><a href="#cb78-16" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">&quot;-p&quot;</span>, fnRs.cut[i],</span>
<span id="cb78-17"><a href="#cb78-17" aria-hidden="true" tabindex="-1"></a>                                            fnFs.filtN[i], fnRs.filtN[i]))}</span></code></pre></div>
</div>
<pre><code>This is cutadapt 2.8 with Python 3.7.6
Command line parameters: -g CTTGGTCATTTAGAGGAAGTAA -a GCATCGATGAAGAACGCAGC -G GCTGCGTTCTTCATCGATGC -A TTACTTCCTCTAAATGACCAAG -m 20 -n 2 -o RAW/cutadapt/P1_R1.fastq.gz -p RAW/cutadapt/P1_R2.fastq.gz RAW/filtN/P1_R1.fastq.gz RAW/filtN/P1_R2.fastq.gz</code></pre>
<p><em>Note</em>. If the code above removes all of the base pairs in a sequence, you will get downstream errors unless you set the <code>-m</code> flag. This flag sets the minimum length and reads shorter than this will be discarded. Without this flag, reads of length 0 will be kept and cause issues. <em>Also, a lot of output will be written to the screen by cutadapt!</em></p>
<p>We can now count the number of primers in the sequences from the output of cutadapt.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(</span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[sampnum]]),</span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">FWD.ReverseReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnRs.cut[[sampnum]]),</span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[sampnum]]),</span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">REV.ReverseReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnRs.cut[[sampnum]]))</span></code></pre></div>
</div>
<pre><code>                 Forward Complement Reverse RevComp
FWD.ForwardReads       0          0       0       0
FWD.ReverseReads       0          0       0       0
REV.ForwardReads       0          0       0       0
REV.ReverseReads       0          0       0       0</code></pre>
<p>Basically, primers are no longer detected in the cutadapted reads. Now, for each sample, we can take a look at how the pre-filtering step and primer removal affected the total number of raw reads.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-eed293ac19a22cf1972a" style="width:80%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-eed293ac19a22cf1972a">{"x":{"filter":"none","extensions":["Buttons"],"data":[["P1_R1","P1-8C_R1","P10_R1","P2_R1","P3_R1","P3-8C_R1","P4_R1","P5_R1","P5-8C_R1","P6_R1","P7_R1","P7-8C_R1","P8_R1","P9_R1","P9-8C_R1"],[50894,70734,96531,74581,60006,31174,88732,809,81908,67419,70123,15435,60799,55927,1418],[50882,70649,96492,74557,59953,31029,88694,797,81671,67399,70073,15417,60604,55764,1383],[50419,68075,95256,73994,58146,25782,87532,412,74669,66656,68474,14955,54534,51382,141]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>raw<\/th>\n      <th>pre_filt<\/th>\n      <th>cut<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<h2 id="quality-assessment-filtering-1">4. Quality Assessment &amp; Filtering</h2>
<p>We need the forward and reverse fastq file names and the sample names.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>cutFs <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path.cut, <span class="at">pattern =</span> <span class="st">&quot;_R1.fastq.gz&quot;</span>,</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>cutRs <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path.cut, <span class="at">pattern =</span> <span class="st">&quot;_R2.fastq.gz&quot;</span>,</span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a>get.sample.name <span class="ot">&lt;-</span> <span class="cf">function</span>(fname) <span class="fu">strsplit</span>(<span class="fu">basename</span>(fname), <span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]</span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a>sample.names <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">sapply</span>(cutFs, get.sample.name))</span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sample.names)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;P1&quot;    &quot;P1-8C&quot; &quot;P10&quot;   &quot;P2&quot;    &quot;P3&quot;    &quot;P3-8C&quot;</code></pre>
</div>
<h3 id="quality-plots-1">Quality plots</h3>
<p>Then we inspect read quality.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">plotQualityProfile</span>(cutFs[<span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>], <span class="at">aggregate =</span> <span class="cn">TRUE</span>)</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">plotQualityProfile</span>(cutRs[<span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>], <span class="at">aggregate =</span> <span class="cn">TRUE</span>)</span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">grid.arrange</span>(p1, p2, <span class="at">nrow =</span> <span class="dv">1</span>)</span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;plot_qscores_2018.png&quot;</span>, p3, <span class="at">width =</span> <span class="dv">7</span>, <span class="at">height =</span> <span class="dv">3</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/dada2/its18_plot_qscores.png" width="1050" /></p>
</div>
<p><small><em>Aggregated quality score plots for forward (left) &amp; reverse (right) reads.</em></small></p>
<h3 id="filtering-1">Filtering</h3>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>filtFs <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut, <span class="st">&quot;filtered&quot;</span>, <span class="fu">basename</span>(cutFs))</span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a>filtRs <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut, <span class="st">&quot;filtered&quot;</span>, <span class="fu">basename</span>(cutRs))</span></code></pre></div>
</div>
<p>We use a <code>minLen</code> value to get rid of very short length sequences.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>out <span class="ot">&lt;-</span> <span class="fu">filterAndTrim</span>(cutFs, filtFs, cutRs, filtRs,</span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">maxN =</span> <span class="dv">0</span>, <span class="at">maxEE =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>), <span class="at">truncQ =</span> <span class="dv">2</span>,</span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">minLen =</span> <span class="dv">50</span>, <span class="at">rm.phix =</span> <span class="cn">TRUE</span>,</span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">compress =</span> <span class="cn">TRUE</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a>out</span></code></pre></div>
</div>
<aside>
These parameters should be set based on the anticipated length of the amplicon and the read quality.
</aside>
<p>And here is a table of how the filtering step affected the number of reads in each sample.</p>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-f6d5de2b538c2ee39d90" style="width:80%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f6d5de2b538c2ee39d90">{"x":{"filter":"none","extensions":["Buttons"],"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15"],["P1_R1","P1-8C_R1","P10_R1","P2_R1","P3_R1","P3-8C_R1","P4_R1","P5_R1","P5-8C_R1","P6_R1","P7_R1","P7-8C_R1","P8_R1","P9_R1","P9-8C_R1"],[50419,68075,95256,73994,58146,25782,87532,412,74669,66656,68474,14955,54534,51382,141],[35999,48230,80877,58956,42563,17548,68259,308,48518,51055,43471,9488,41563,35722,85]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Sample<\/th>\n      <th>reads.in<\/th>\n      <th>reads.out<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"className":"dt-right","targets":[2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<h2 id="learn-error-rates-1">5. Learn Error Rates</h2>
<p>Time to assess the error rate of the data. The rest of the workflow is very similar to the 16S workflows presented previously. So I will basically stop talking.</p>
<h4 id="forward-reads-2">Forward Reads</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a>errF <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(filtFs, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<pre><code>108788735 total bases in 495784 reads from 11 samples will be used for learning the error rates.
109358094 total bases in 495784 reads from 11 samples will be used for learning the error rates.</code></pre>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errF, <span class="at">nominalQ =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/dada2/its18_plot_errorF_1.png" width="1050" /></p>
</div>
<h4 id="reverse-reads">Reverse Reads</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>errR <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(filtRs, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<pre><code>109358094 total bases in 495784 reads from 11 samples will be used for learning the error rates.</code></pre>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errR, <span class="at">nominalQ =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/dada2/its18_plot_errorR_1.png" width="1050" /></p>
</div>
<h2 id="dereplicate-reads-1">6. Dereplicate Reads</h2>
<h4 id="forward-reads-3">Forward Reads</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a>derepFs <span class="ot">&lt;-</span> <span class="fu">derepFastq</span>(filtFs, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(derepFs) <span class="ot">&lt;-</span> sample.names</span></code></pre></div>
</div>
<p><br/></p>
<details>
<summary>
Detailed results of derep forward reads
</summary>
<pre><code>
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P1_R1.fastq.gz
Encountered 7277 unique sequences from 35999 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P1-8C_R1.fastq.gz
Encountered 11366 unique sequences from 48230 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P10_R1.fastq.gz
Encountered 11714 unique sequences from 80877 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P2_R1.fastq.gz
Encountered 13088 unique sequences from 58956 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P3_R1.fastq.gz
Encountered 10180 unique sequences from 42563 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P3-8C_R1.fastq.gz
Encountered 4480 unique sequences from 17548 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P4_R1.fastq.gz
Encountered 12259 unique sequences from 68259 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P5_R1.fastq.gz
Encountered 160 unique sequences from 308 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P5-8C_R1.fastq.gz
Encountered 10436 unique sequences from 48518 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P6_R1.fastq.gz
Encountered 11230 unique sequences from 51055 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P7_R1.fastq.gz
Encountered 10316 unique sequences from 43471 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P7-8C_R1.fastq.gz
Encountered 2121 unique sequences from 9488 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P8_R1.fastq.gz
Encountered 9236 unique sequences from 41563 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P9_R1.fastq.gz
Encountered 9000 unique sequences from 35722 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P9-8C_R1.fastq.gz
Encountered 42 unique sequences from 85 total sequences read.
</code></pre>
</details>
<p><br/></p>
<h4 id="reverse-reads-1">Reverse Reads</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a>derepRs <span class="ot">&lt;-</span> <span class="fu">derepFastq</span>(filtRs, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(derepRs) <span class="ot">&lt;-</span> sample.names</span></code></pre></div>
</div>
<p><br/></p>
<details>
<summary>
Detailed results of derep reverse reads
</summary>
<pre><code>
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P1_R2.fastq.gz
Encountered 13081 unique sequences from 35999 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P1-8C_R2.fastq.gz
Encountered 17965 unique sequences from 48230 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P10_R2.fastq.gz
Encountered 18610 unique sequences from 80877 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P2_R2.fastq.gz
Encountered 19600 unique sequences from 58956 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P3_R2.fastq.gz
Encountered 16145 unique sequences from 42563 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P3-8C_R2.fastq.gz
Encountered 7496 unique sequences from 17548 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P4_R2.fastq.gz
Encountered 19933 unique sequences from 68259 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P5_R2.fastq.gz
Encountered 199 unique sequences from 308 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P5-8C_R2.fastq.gz
Encountered 18495 unique sequences from 48518 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P6_R2.fastq.gz
Encountered 17216 unique sequences from 51055 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P7_R2.fastq.gz
Encountered 17111 unique sequences from 43471 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P7-8C_R2.fastq.gz
Encountered 3809 unique sequences from 9488 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P8_R2.fastq.gz
Encountered 14140 unique sequences from 41563 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P9_R2.fastq.gz
Encountered 14856 unique sequences from 35722 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P9-8C_R2.fastq.gz
Encountered 45 unique sequences from 85 total sequences read.
</code></pre>
</details>
<p><br/></p>
<h2 id="dada2-asv-inference-1">7. DADA2 &amp; ASV Inference</h2>
<p>At this point we are ready to apply the core sample inference algorithm (dada) to the filtered and trimmed sequence data. DADA2 offers three options for whether and how to pool samples for ASV inference.</p>
<p>If <code>pool = TRUE</code>, the algorithm will pool together all samples prior to sample inference.<br />
If <code>pool = FALSE</code>, sample inference is performed on each sample individually.<br />
If <code>pool = "pseudo"</code>, the algorithm will perform pseudo-pooling between individually processed samples.</p>
<p>We tested all three methods through the full pipeline. Click the <code>+</code> to see the results of the test. For our final analysis, we chose <code>pool = TRUE</code> for this data set.</p>
<details>
<summary>
Show/hide Results of <code>dada</code> pooling options
</summary>
<p>Here are a few summary tables of results from the tests. Values are from the final sequence table prodcuced by each option.</p>
<p><br/></p>
<em>Total number of reads, total number of ASV, minimum and maximum ASVs, followed by the number of singletons, doubletons, etc.</em>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">unpooled</th>
<th style="text-align: right;">psuedo</th>
<th style="text-align: right;">pooled</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">no_reads</td>
<td style="text-align: right;">500415</td>
<td style="text-align: right;">521309</td>
<td style="text-align: right;">491143</td>
</tr>
<tr class="even">
<td style="text-align: left;">no_asvs</td>
<td style="text-align: right;">4322</td>
<td style="text-align: right;">4348</td>
<td style="text-align: right;">3357</td>
</tr>
<tr class="odd">
<td style="text-align: left;">min_asv</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">14</td>
</tr>
<tr class="even">
<td style="text-align: left;">max_asv</td>
<td style="text-align: right;">692</td>
<td style="text-align: right;">817</td>
<td style="text-align: right;">1017</td>
</tr>
<tr class="odd">
<td style="text-align: left;">n_samp_1</td>
<td style="text-align: right;">3186</td>
<td style="text-align: right;">3176</td>
<td style="text-align: right;">1247</td>
</tr>
<tr class="even">
<td style="text-align: left;">n_samp_2</td>
<td style="text-align: right;">598</td>
<td style="text-align: right;">401</td>
<td style="text-align: right;">711</td>
</tr>
<tr class="odd">
<td style="text-align: left;">n_samp_3</td>
<td style="text-align: right;">239</td>
<td style="text-align: right;">250</td>
<td style="text-align: right;">443</td>
</tr>
<tr class="even">
<td style="text-align: left;">n_samp_4</td>
<td style="text-align: right;">123</td>
<td style="text-align: right;">172</td>
<td style="text-align: right;">289</td>
</tr>
<tr class="odd">
<td style="text-align: left;">n_samp_5</td>
<td style="text-align: right;">48</td>
<td style="text-align: right;">118</td>
<td style="text-align: right;">204</td>
</tr>
<tr class="even">
<td style="text-align: left;">n_samp_6</td>
<td style="text-align: right;">45</td>
<td style="text-align: right;">82</td>
<td style="text-align: right;">123</td>
</tr>
<tr class="odd">
<td style="text-align: left;">n_samp_7</td>
<td style="text-align: right;">23</td>
<td style="text-align: right;">42</td>
<td style="text-align: right;">102</td>
</tr>
<tr class="even">
<td style="text-align: left;">n_samp_8</td>
<td style="text-align: right;">22</td>
<td style="text-align: right;">32</td>
<td style="text-align: right;">86</td>
</tr>
<tr class="odd">
<td style="text-align: left;">n_samp_9</td>
<td style="text-align: right;">16</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">49</td>
</tr>
<tr class="even">
<td style="text-align: left;">n_samp_10</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">32</td>
</tr>
<tr class="odd">
<td style="text-align: left;">n_samp_11</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">24</td>
</tr>
<tr class="even">
<td style="text-align: left;">n_samp_12</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">15</td>
</tr>
<tr class="odd">
<td style="text-align: left;">n_samp_13</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">23</td>
</tr>
<tr class="even">
<td style="text-align: left;">n_samp_14</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">n_samp_15</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">2</td>
</tr>
</tbody>
</table>
</div>
<em>Total number of reads by sample.</em>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">unpooled</th>
<th style="text-align: right;">psuedo</th>
<th style="text-align: right;">pooled</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">P1</td>
<td style="text-align: right;">29683</td>
<td style="text-align: right;">32793</td>
<td style="text-align: right;">29794</td>
</tr>
<tr class="even">
<td style="text-align: left;">P1-8C</td>
<td style="text-align: right;">42959</td>
<td style="text-align: right;">43591</td>
<td style="text-align: right;">41880</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P10</td>
<td style="text-align: right;">64981</td>
<td style="text-align: right;">78039</td>
<td style="text-align: right;">64636</td>
</tr>
<tr class="even">
<td style="text-align: left;">P2</td>
<td style="text-align: right;">51414</td>
<td style="text-align: right;">51894</td>
<td style="text-align: right;">51144</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P3</td>
<td style="text-align: right;">38437</td>
<td style="text-align: right;">38589</td>
<td style="text-align: right;">38374</td>
</tr>
<tr class="even">
<td style="text-align: left;">P3-8C</td>
<td style="text-align: right;">14909</td>
<td style="text-align: right;">15045</td>
<td style="text-align: right;">14188</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P4</td>
<td style="text-align: right;">56662</td>
<td style="text-align: right;">57911</td>
<td style="text-align: right;">56593</td>
</tr>
<tr class="even">
<td style="text-align: left;">P5</td>
<td style="text-align: right;">226</td>
<td style="text-align: right;">237</td>
<td style="text-align: right;">296</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P5-8C</td>
<td style="text-align: right;">41453</td>
<td style="text-align: right;">42395</td>
<td style="text-align: right;">35600</td>
</tr>
<tr class="even">
<td style="text-align: left;">P6</td>
<td style="text-align: right;">45755</td>
<td style="text-align: right;">45913</td>
<td style="text-align: right;">45850</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P7</td>
<td style="text-align: right;">37637</td>
<td style="text-align: right;">37946</td>
<td style="text-align: right;">36063</td>
</tr>
<tr class="even">
<td style="text-align: left;">P7-8C</td>
<td style="text-align: right;">8953</td>
<td style="text-align: right;">8984</td>
<td style="text-align: right;">9172</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P8</td>
<td style="text-align: right;">38087</td>
<td style="text-align: right;">38297</td>
<td style="text-align: right;">38371</td>
</tr>
<tr class="even">
<td style="text-align: left;">P9</td>
<td style="text-align: right;">29184</td>
<td style="text-align: right;">29601</td>
<td style="text-align: right;">29102</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P9-8C</td>
<td style="text-align: right;">75</td>
<td style="text-align: right;">74</td>
<td style="text-align: right;">80</td>
</tr>
</tbody>
</table>
</div>
<em>Total number of ASVs by sample.</em>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">unpooled</th>
<th style="text-align: right;">psuedo</th>
<th style="text-align: right;">pooled</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">P1</td>
<td style="text-align: right;">439</td>
<td style="text-align: right;">575</td>
<td style="text-align: right;">945</td>
</tr>
<tr class="even">
<td style="text-align: left;">P1-8C</td>
<td style="text-align: right;">607</td>
<td style="text-align: right;">655</td>
<td style="text-align: right;">719</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P10</td>
<td style="text-align: right;">561</td>
<td style="text-align: right;">642</td>
<td style="text-align: right;">812</td>
</tr>
<tr class="even">
<td style="text-align: left;">P2</td>
<td style="text-align: right;">692</td>
<td style="text-align: right;">817</td>
<td style="text-align: right;">1011</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P3</td>
<td style="text-align: right;">538</td>
<td style="text-align: right;">624</td>
<td style="text-align: right;">765</td>
</tr>
<tr class="even">
<td style="text-align: left;">P3-8C</td>
<td style="text-align: right;">282</td>
<td style="text-align: right;">307</td>
<td style="text-align: right;">349</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P4</td>
<td style="text-align: right;">641</td>
<td style="text-align: right;">771</td>
<td style="text-align: right;">1017</td>
</tr>
<tr class="even">
<td style="text-align: left;">P5</td>
<td style="text-align: right;">29</td>
<td style="text-align: right;">49</td>
<td style="text-align: right;">95</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P5-8C</td>
<td style="text-align: right;">535</td>
<td style="text-align: right;">548</td>
<td style="text-align: right;">616</td>
</tr>
<tr class="even">
<td style="text-align: left;">P6</td>
<td style="text-align: right;">650</td>
<td style="text-align: right;">745</td>
<td style="text-align: right;">954</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P7</td>
<td style="text-align: right;">615</td>
<td style="text-align: right;">708</td>
<td style="text-align: right;">867</td>
</tr>
<tr class="even">
<td style="text-align: left;">P7-8C</td>
<td style="text-align: right;">119</td>
<td style="text-align: right;">145</td>
<td style="text-align: right;">335</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P8</td>
<td style="text-align: right;">546</td>
<td style="text-align: right;">607</td>
<td style="text-align: right;">702</td>
</tr>
<tr class="even">
<td style="text-align: left;">P9</td>
<td style="text-align: right;">589</td>
<td style="text-align: right;">653</td>
<td style="text-align: right;">745</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P9-8C</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">14</td>
</tr>
</tbody>
</table>
</div>
</details>
<h4 id="forward-reads-4">Forward Reads</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>dadaFs <span class="ot">&lt;-</span> <span class="fu">dada</span>(derepFs, <span class="at">err =</span> errF, <span class="at">pool =</span> <span class="cn">TRUE</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<pre><code>15 samples were pooled: 582642 reads in 102887 unique sequences.</code></pre>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>dadaFs[[sampnum]]</span></code></pre></div>
</div>
<pre><code>dada-class: object describing DADA2 denoising results
1175 sequence variants were inferred from 7277 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
<h4 id="reverse-reads-2">Reverse Reads</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>dadaRs <span class="ot">&lt;-</span> <span class="fu">dada</span>(derepRs, <span class="at">err =</span> errR, <span class="at">pool =</span> <span class="cn">TRUE</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<pre><code>15 samples were pooled: 582642 reads in 174866 unique sequences.</code></pre>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a>dadaRs[[sampnum]]</span></code></pre></div>
</div>
<pre><code>dada-class: object describing DADA2 denoising results
1116 sequence variants were inferred from 13081 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="st">&quot;pre_merge_its_2018.rdata&quot;</span>)</span></code></pre></div>
</div>
<h2 id="merge-paired-reads">8. Merge Paired Reads</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>mergers <span class="ot">&lt;-</span> <span class="fu">mergePairs</span>(dadaFs, derepFs, dadaRs, derepRs, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<details>
<summary>
Detailed results of merging reads
</summary>
<pre><code>
29795 paired-reads (in 946 unique pairings) successfully merged out of 32193 (in 1177 pairings) input.
41880 paired-reads (in 719 unique pairings) successfully merged out of 46715 (in 941 pairings) input.
64805 paired-reads (in 815 unique pairings) successfully merged out of 66771 (in 1015 pairings) input.
Duplicate sequences in merged output.
51144 paired-reads (in 1011 unique pairings) successfully merged out of 57613 (in 1325 pairings) input.
38442 paired-reads (in 767 unique pairings) successfully merged out of 41985 (in 994 pairings) input.
14188 paired-reads (in 349 unique pairings) successfully merged out of 16981 (in 452 pairings) input.
56594 paired-reads (in 1018 unique pairings) successfully merged out of 64411 (in 1306 pairings) input.
296 paired-reads (in 95 unique pairings) successfully merged out of 301 (in 100 pairings) input.
35713 paired-reads (in 619 unique pairings) successfully merged out of 41006 (in 786 pairings) input.
45851 paired-reads (in 955 unique pairings) successfully merged out of 50174 (in 1240 pairings) input.
36121 paired-reads (in 870 unique pairings) successfully merged out of 41541 (in 1135 pairings) input.
Duplicate sequences in merged output.
9172 paired-reads (in 335 unique pairings) successfully merged out of 9403 (in 402 pairings) input.
38392 paired-reads (in 704 unique pairings) successfully merged out of 40897 (in 886 pairings) input.
29155 paired-reads (in 748 unique pairings) successfully merged out of 34979 (in 990 pairings) input.
Duplicate sequences in merged output.
80 paired-reads (in 14 unique pairings) successfully merged out of 82 (in 16 pairings) input.
</code></pre>
</details>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="construct-sequence-table-1">9. Construct Sequence Table</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a>seqtabF <span class="ot">&lt;-</span> <span class="fu">makeSequenceTable</span>(mergers)</span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtabF)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1]   15 3364</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtabF)))</span></code></pre></div>
</div>
<pre><code>130 131 132 133 136 139 140 141 143 144 146 147 148 149 150 151 152 153 154 155 
  2   2   1   1   1   1   1   3   2   1   3   5   5   4   5   5   3   2   5   4 
156 159 160 161 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 
  1   2   1   1   2   2   4   9   2   1   2   5   5   5   8   9   2   5  10  13 
179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 
 13  14  11  14  12  15  19  12  12  13   9  22  11  11  12  15  17  13  10  11 
199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 
 22  16  10  11  13  17  10   8  10  12  13   7  20  18  24  30  39  38  20  19 
219 220 221 222 223 224 225 226 228 229 230 231 232 233 234 235 236 237 238 239 
 12  18  21  23  23  29  32  32   6  48   4  73  20  27  29  19  31  33  30  29 
240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255 256 257 258 259 
 35  21  36  16  16  18  17  25  22  29  17  27  28  22  20  18  28  19  28  25 
260 261 262 263 264 265 266 267 268 269 270 271 272 273 274 275 276 277 278 279 
 23  11  15  13  16  18  23  17  12  12  16  15  32  17  22  21  21  18  16  17 
280 281 282 283 284 285 286 287 288 289 290 291 292 293 294 295 296 297 298 299 
 13  22  13  18  12  13  13  18  17  15  14  18   8  11  13  17  12  11  13   9 
300 301 302 303 304 305 306 307 308 309 310 311 312 313 314 315 316 317 318 319 
 16  10   9   3   8  12   8  13  20  15   8  13  16  11  10   8  18  10   8  12 
320 321 322 323 324 325 326 327 328 329 330 331 332 333 334 335 336 337 338 339 
  8   5  12  17  13   9   8   8   9  12  16  13  14  11  12  13  18  10  11  10 
340 341 342 343 344 345 346 347 348 349 350 351 352 353 354 355 356 357 358 359 
  9  11   8   8   5  15   8  16  11  15   4   9   7   9  12  14   5  12  11   5 
360 361 362 363 364 365 366 367 368 369 370 371 372 373 374 375 376 377 378 379 
  2   4  11   5   2  14   6  12   7   5   6   4   3   9   5  10   8   9   4   6 
380 381 382 383 384 385 386 387 388 389 390 391 392 393 394 395 396 397 398 399 
  5   3   7   7   4   2   3   4   3   5   5   1   3   4   3   6   5   4   3   2 
400 401 402 403 404 405 406 407 408 409 410 411 412 413 414 415 416 417 418 421 
 11   5   3   3   3   3   2   1   2   3   3   1   3   3   1   2   2   1   2   2 
422 423 424 427 430 431 432 434 436 437 439 440 441 442 443 446 447 448 
  2   3   2   1   3   2   4   2   1   2   4   3   1   2   1   1   2   1 
</code></pre>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/dada2/its18_read_length_before.png" width="1050" /></p>
</div>
<p>As expected, a wide range of length variability in the the amplified ITS region.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a>seqtab <span class="ot">&lt;-</span> seqtabF[,<span class="fu">nchar</span>(<span class="fu">colnames</span>(seqtabF)) <span class="sc">%in%</span> <span class="fu">seq</span>(<span class="dv">100</span>,<span class="dv">450</span>)]</span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1]   15 3364</code></pre>
</div>
<p>After removing the extreme length variants, we have 3364, a reduction of 0 sequence variants.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab)))</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="remove-chimeras-1">10. Remove Chimeras</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a>seqtab.nochim <span class="ot">&lt;-</span> <span class="fu">removeBimeraDenovo</span>(seqtab,</span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">method =</span> <span class="st">&quot;consensus&quot;</span>,</span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">multithread =</span> <span class="cn">TRUE</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab.nochim)</span></code></pre></div>
</div>
<pre><code>Identified 7 bimeras out of 3364 input sequences.</code></pre>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1]   15 3357</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(seqtab.nochim)<span class="sc">/</span><span class="fu">sum</span>(seqtab<span class="fl">.2</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] 0.9990135</code></pre>
</div>
<p>Chimera checking removed an additional 7 sequence variants however, when we account for the abundances of each variant, we see chimeras accounts for about 0.09865% of the merged sequence reads. Curious.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">colSums</span>(seqtab.nochim<span class="sc">&gt;</span><span class="dv">0</span>))</span></code></pre></div>
</div>
<pre><code>   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15
1247  711  443  289  204  123  102   86   49   32   24   15   23    7    2</code></pre>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">colSums</span>(seqtab.nochim<span class="sc">&gt;</span><span class="dv">0</span>))</span></code></pre></div>
</div>
<pre><code>  14   95  335  349  616  702  719  745  765  812  867  945  954 1011 1017
   1    1    1    1    1    1    1    1    1    1    1    1    1    1    1</code></pre>
<p>The last thing we want to do is write the sequence table to an <code>RDS</code> file.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(seqtab.nochim, <span class="st">&quot;its18_seqtab.nochim.rds&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="track-reads-through-workflow-1">11. Track Reads through Workflow</h2>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="htmlwidget-bb983d864ca15ad3e051" style="width:80%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-bb983d864ca15ad3e051">{"x":{"filter":"none","extensions":["Buttons"],"data":[["P1","P1-8C","P10","P2","P3","P3-8C","P4","P5","P5-8C","P6","P7","P7-8C","P8","P9","P9-8C"],[50894,70734,96531,74581,60006,31174,88732,809,81908,67419,70123,15435,60799,55927,1418],[50882,70649,96492,74557,59953,31029,88694,797,81671,67399,70073,15417,60604,55764,1383],[50419,68075,95256,73994,58146,25782,87532,412,74669,66656,68474,14955,54534,51382,141],[35999,48230,80877,58956,42563,17548,68259,308,48518,51055,43471,9488,41563,35722,85],[32635,47150,68681,58018,42253,17132,64797,302,41357,50634,42194,9426,41243,35410,84],[33389,47059,70013,58130,42151,17210,67613,302,46321,50362,42074,9427,41077,35168,82],[29795,41880,64805,51144,38442,14188,56594,296,35713,45851,36121,9172,38392,29155,80],[29794,41880,64636,51144,38374,14188,56593,296,35600,45850,36063,9172,38371,29102,80]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>raw<\/th>\n      <th>pre_filt<\/th>\n      <th>cut<\/th>\n      <th>filtered<\/th>\n      <th>denoisedF<\/th>\n      <th>denoisedR<\/th>\n      <th>merged<\/th>\n      <th>nonchim<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="assign-taxonomy-1">12. Assign Taxonomy</h2>
<p>For taxonomic classification, we used the UNITE <span class="citation" data-cites="nilsson2019unite">(<a href="#ref-nilsson2019unite" role="doc-biblioref">Nilsson et al. 2019</a>)</span> database, specifically the <a href="https://plutof.ut.ee/#/doi/10.15156/BIO/786368">UNITE general FASTA release for Fungi (v. 04.02.2020)</a> <span class="citation" data-cites="abarenkov2020unite">(<a href="#ref-abarenkov2020unite" role="doc-biblioref">Abarenkov et al. 2020</a>)</span>.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a>tax <span class="ot">&lt;-</span> <span class="fu">assignTaxonomy</span>(seqtab,</span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a>                      <span class="st">&quot;sh_general_release_dynamic_s_04.02.2020_dev.fasta&quot;</span>,</span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">multithread =</span> <span class="dv">20</span>, <span class="at">tryRC =</span> <span class="cn">TRUE</span>)</span>
<span id="cb122-4"><a href="#cb122-4" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(tax, <span class="st">&quot;its18_tax_its.rds&quot;</span>)</span></code></pre></div>
</div>
<pre><code>UNITE fungal taxonomic reference detected.</code></pre>
<h2 id="save-image-1">13. Save Image</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="st">&quot;its18_dada2_wf.rdata&quot;</span>)</span></code></pre></div>
</div>
<h2 id="r-session-information-code-1">R Session Information &amp; Code</h2>
<p>This workflow was run on the <a href="https://doi.org/10.25572/SIHPC">Smithsonian High Performance Cluster (SI/HPC)</a>, Smithsonian Institution. Below are the specific packages and versions used in this workflow using both <code>sessionInfo()</code> and <code>devtools::session_info()</code>. Click the arrow to see the details.</p>
<details>
<summary>
Show/hide R Session Info
</summary>
<pre><code>sessionInfo()
R version 4.0.0 (2020-04-24)
Platform: x86_64-conda_cos6-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /home/scottjj/miniconda3/envs/R-4/lib/libopenblasp-r0.3.9.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
 [1] grid      stats4    parallel  stats     graphics  grDevices utils
 [8] datasets  methods   base

other attached packages:
 [1] gridExtra_2.3               ggplot2_3.3.0
 [3] ShortRead_1.46.0            GenomicAlignments_1.24.0
 [5] SummarizedExperiment_1.18.1 DelayedArray_0.14.0
 [7] matrixStats_0.56.0          Biobase_2.48.0
 [9] Rsamtools_2.4.0             GenomicRanges_1.40.0
[11] GenomeInfoDb_1.24.0         Biostrings_2.56.0
[13] XVector_0.28.0              IRanges_2.22.1
[15] S4Vectors_0.26.0            BiocParallel_1.22.0
[17] BiocGenerics_0.34.0         dada2_1.16.0
[19] Rcpp_1.0.4.6

loaded via a namespace (and not attached):
 [1] plyr_1.8.6             RColorBrewer_1.1-2     pillar_1.4.4
 [4] compiler_4.0.0         bitops_1.0-6           tools_4.0.0
 [7] zlibbioc_1.34.0        lifecycle_0.2.0        tibble_3.0.1
[10] gtable_0.3.0           lattice_0.20-41        png_0.1-7
[13] pkgconfig_2.0.3        rlang_0.4.6            Matrix_1.2-18
[16] GenomeInfoDbData_1.2.3 withr_2.2.0            stringr_1.4.0
[19] hwriter_1.3.2          vctrs_0.3.0            glue_1.4.1
[22] R6_2.4.1               jpeg_0.1-8.1           latticeExtra_0.6-29
[25] reshape2_1.4.4         magrittr_1.5           scales_1.1.1
[28] ellipsis_0.3.1         colorspace_1.4-1       stringi_1.4.6
[31] RCurl_1.98-1.2         RcppParallel_5.0.1     munsell_0.5.0
[34] crayon_1.3.4

devtools::session_info()
─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.0.0 (2020-04-24)
 os       CentOS Linux 7 (Core)
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       America/New_York
 date     2020-06-04

─ Packages ───────────────────────────────────────────────────────────────────
 package              * version  date       lib source
 assertthat             0.2.1    2019-03-21 [1] CRAN (R 4.0.0)
 backports              1.1.7    2020-05-13 [1] CRAN (R 4.0.0)
 Biobase              * 2.48.0   2020-04-27 [1] Bioconductor
 BiocGenerics         * 0.34.0   2020-04-27 [1] Bioconductor
 BiocParallel         * 1.22.0   2020-04-27 [1] Bioconductor
 Biostrings           * 2.56.0   2020-04-27 [1] Bioconductor
 bitops                 1.0-6    2013-08-17 [1] CRAN (R 4.0.0)
 callr                  3.4.3    2020-03-28 [1] CRAN (R 4.0.0)
 cli                    2.0.2    2020-02-28 [1] CRAN (R 4.0.0)
 colorspace             1.4-1    2019-03-18 [1] CRAN (R 4.0.0)
 crayon                 1.3.4    2017-09-16 [1] CRAN (R 4.0.0)
 dada2                * 1.16.0   2020-04-27 [1] Bioconductor
 DelayedArray         * 0.14.0   2020-04-27 [1] Bioconductor
 desc                   1.2.0    2018-05-01 [1] CRAN (R 4.0.0)
 devtools               2.3.0    2020-04-10 [1] CRAN (R 4.0.0)
 digest                 0.6.25   2020-02-23 [1] CRAN (R 4.0.0)
 ellipsis               0.3.1    2020-05-15 [1] CRAN (R 4.0.0)
 fansi                  0.4.1    2020-01-08 [1] CRAN (R 4.0.0)
 fs                     1.4.1    2020-04-04 [1] CRAN (R 4.0.0)
 GenomeInfoDb         * 1.24.0   2020-04-27 [1] Bioconductor
 GenomeInfoDbData       1.2.3    2020-05-22 [1] Bioconductor
 GenomicAlignments    * 1.24.0   2020-04-27 [1] Bioconductor
 GenomicRanges        * 1.40.0   2020-04-27 [1] Bioconductor
 ggplot2              * 3.3.0    2020-03-05 [1] CRAN (R 4.0.0)
 glue                   1.4.1    2020-05-13 [1] CRAN (R 4.0.0)
 gridExtra            * 2.3      2017-09-09 [1] CRAN (R 4.0.0)
 gtable                 0.3.0    2019-03-25 [1] CRAN (R 4.0.0)
 hwriter                1.3.2    2014-09-10 [1] CRAN (R 4.0.0)
 IRanges              * 2.22.1   2020-04-28 [1] Bioconductor
 jpeg                   0.1-8.1  2019-10-24 [1] CRAN (R 4.0.0)
 lattice                0.20-41  2020-04-02 [1] CRAN (R 4.0.0)
 latticeExtra           0.6-29   2019-12-19 [1] CRAN (R 4.0.0)
 lifecycle              0.2.0    2020-03-06 [1] CRAN (R 4.0.0)
 magrittr               1.5      2014-11-22 [1] CRAN (R 4.0.0)
 Matrix                 1.2-18   2019-11-27 [1] CRAN (R 4.0.0)
 matrixStats          * 0.56.0   2020-03-13 [1] CRAN (R 4.0.0)
 memoise                1.1.0    2017-04-21 [1] CRAN (R 4.0.0)
 munsell                0.5.0    2018-06-12 [1] CRAN (R 4.0.0)
 pillar                 1.4.4    2020-05-05 [1] CRAN (R 4.0.0)
 pkgbuild               1.0.8    2020-05-07 [1] CRAN (R 4.0.0)
 pkgconfig              2.0.3    2019-09-22 [1] CRAN (R 4.0.0)
 pkgload                1.0.2    2018-10-29 [1] CRAN (R 4.0.0)
 plyr                   1.8.6    2020-03-03 [1] CRAN (R 4.0.0)
 png                    0.1-7    2013-12-03 [1] CRAN (R 4.0.0)
 prettyunits            1.1.1    2020-01-24 [1] CRAN (R 4.0.0)
 processx               3.4.2    2020-02-09 [1] CRAN (R 4.0.0)
 ps                     1.3.3    2020-05-08 [1] CRAN (R 4.0.0)
 R6                     2.4.1    2019-11-12 [1] CRAN (R 4.0.0)
 RColorBrewer           1.1-2    2014-12-07 [1] CRAN (R 4.0.0)
 Rcpp                 * 1.0.4.6  2020-04-09 [1] CRAN (R 4.0.0)
 RcppParallel           5.0.1    2020-05-06 [1] CRAN (R 4.0.0)
 RCurl                  1.98-1.2 2020-04-18 [1] CRAN (R 4.0.0)
 remotes                2.1.1    2020-02-15 [1] CRAN (R 4.0.0)
 reshape2               1.4.4    2020-04-09 [1] CRAN (R 4.0.0)
 rlang                  0.4.6    2020-05-02 [1] CRAN (R 4.0.0)
 rprojroot              1.3-2    2018-01-03 [1] CRAN (R 4.0.0)
 Rsamtools            * 2.4.0    2020-04-27 [1] Bioconductor
 S4Vectors            * 0.26.0   2020-04-27 [1] Bioconductor
 scales                 1.1.1    2020-05-11 [1] CRAN (R 4.0.0)
 sessioninfo            1.1.1    2018-11-05 [1] CRAN (R 4.0.0)
 ShortRead            * 1.46.0   2020-04-27 [1] Bioconductor
 stringi                1.4.6    2020-02-17 [1] CRAN (R 4.0.0)
 stringr                1.4.0    2019-02-10 [1] CRAN (R 4.0.0)
 SummarizedExperiment * 1.18.1   2020-04-30 [1] Bioconductor
 testthat               2.3.2    2020-03-02 [1] CRAN (R 4.0.0)
 tibble                 3.0.1    2020-04-20 [1] CRAN (R 4.0.0)
 usethis                1.6.1    2020-04-29 [1] CRAN (R 4.0.0)
 vctrs                  0.3.0    2020-05-11 [1] CRAN (R 4.0.0)
 withr                  2.2.0    2020-04-20 [1] CRAN (R 4.0.0)
 XVector              * 0.28.0   2020-04-27 [1] Bioconductor
 zlibbioc               1.34.0   2020-04-27 [1] Bioconductor</code></pre>
</details>
<p><br/></p>
<p>The end!</p>
<h2 class="appendix" id="source-code">Source Code</h2>
<p>The source code for this page can be accessed on GitHub by <a href="https://github.com/sweltr/high-temp/blob/master/dada2.Rmd">clicking this link</a>. Please note, that in order to process the data <em>and</em> build the website, we needed to run the workflow and get the results. Then hard code the results and turn off the individual commands. So the raw file for this page is a bit messy—you have been warned.</p>
<h2 class="appendix" id="data-availability">Data Availability</h2>
<p>Raw fastq files available on figshare at XXXXXXXX. Trimmed fastq files (primers removed) available through the ENA under project accession number <a href="">XXXXXXXX</a>. Output files from this workflow available on figshare 5at <a href="">XXXXXXXX.</a>.</p>
<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-abarenkov2020unite" class="csl-entry" role="doc-biblioentry">
Abarenkov, Kessy, Allan Zirk, Timo Piirmann, Raivo Pöhönen, Filipp Ivanov, R. Henrik Nilsson, and Urmas Kõljalg. 2020. <span>“UNITE General FASTA Release for Fungi.”</span> <a href="https://dx.doi.org/10.15156/BIO/786368">https://dx.doi.org/10.15156/BIO/786368</a>.
</div>
<div id="ref-callahan2016dada2" class="csl-entry" role="doc-biblioentry">
Callahan, Benjamin J, Paul J McMurdie, Michael J Rosen, Andrew W Han, Amy Jo A Johnson, and Susan P Holmes. 2016. <span>“Dada2: High-Resolution Sample Inference from Illumina Amplicon Data.”</span> <em>Nature Methods</em> 13 (7): 581. <a href="https://doi.org/10.1038/nmeth.3869">https://doi.org/10.1038/nmeth.3869</a>.
</div>
<div id="ref-caporaso2011global" class="csl-entry" role="doc-biblioentry">
Caporaso, J Gregory, Christian L Lauber, William A Walters, Donna Berg-Lyons, Catherine A Lozupone, Peter J Turnbaugh, Noah Fierer, and Rob Knight. 2011. <span>“Global Patterns of 16s rRNA Diversity at a Depth of Millions of Sequences Per Sample.”</span> <em>Proceedings of the National Academy of Sciences</em> 108: 4516–22. <a href="https://doi.org/10.1073/pnas.1000080107">https://doi.org/10.1073/pnas.1000080107</a>.
</div>
<div id="ref-martin2011cutadapt" class="csl-entry" role="doc-biblioentry">
Martin, Marcel. 2011. <span>“Cutadapt Removes Adapter Sequences from High-Throughput Sequencing Reads.”</span> <em>EMBnet. Journal</em> 17 (1): 10–12. <a href="https://doi.org/10.14806/ej.17.1.200">https://doi.org/10.14806/ej.17.1.200</a>.
</div>
<div id="ref-nilsson2019unite" class="csl-entry" role="doc-biblioentry">
Nilsson, Rolf Henrik, Karl-Henrik Larsson, Andy F S Taylor, Johan Bengtsson-Palme, Thomas S Jeppesen, Dmitry Schigel, Peter Kennedy, et al. 2019. <span>“The UNITE Database for Molecular Identification of Fungi: Handling Dark Taxa and Parallel Taxonomic Classifications.”</span> <em>Nucleic Acids Research</em> 47 (D1): D259–64. <a href="https://doi.org/10.1093/nar/gky1022">https://doi.org/10.1093/nar/gky1022</a>.
</div>
<div id="ref-quast2012silva" class="csl-entry" role="doc-biblioentry">
Quast, Christian, Elmar Pruesse, Pelin Yilmaz, Jan Gerken, Timmy Schweer, Pablo Yarza, Jörg Peplies, and Frank Oliver Glöckner. 2012. <span>“The SILVA Ribosomal RNA Gene Database Project: Improved Data Processing and Web-Based Tools.”</span> <em>Nucleic Acids Research</em> 41 (D1): D590–96. <a href="https://doi.org/10.1093/nar/gks1219">https://doi.org/10.1093/nar/gks1219</a>.
</div>
</div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/sweltr/high-temp/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/sweltr/high-temp/">https://github.com/sweltr/high-temp/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<div class="distill-site-nav distill-site-footer">
<p>© Copyright 2021 The SWELTR Project.</p>
<p>Site constructed using <a href="https://rstudio.github.io/distill/">Distill for R Markdown</a>.</p>
</div>
<!--/radix_placeholder_navigation_after_body-->


</body>

</html>
