<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-0.9.598">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="description" content="DADA2 workflow for processing the 16S rRNA and the ITS high temperature data sets. Workflow uses paired end reads, beginning with raw fastq files, ending with sequence and taxonomy tables.">
<title>SWELTR - 1. DADA2 Workflow</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./data-prep.html" rel="next">
<link href="./assets/favicon.ico" rel="icon">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script><style>
    .quarto-title-block .quarto-title-banner {
      background: #113858;
    }
    </style>
<script src="site_libs/font-awesome-5.13.0/js/script.js"></script><script src="site_libs/core-js-2.5.3/shim.min.js"></script><script src="site_libs/react-17.0.0/react.min.js"></script><script src="site_libs/react-17.0.0/react-dom.min.js"></script><script src="site_libs/reactwidget-1.0.0/react-tools.js"></script><script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><script src="site_libs/reactable-binding-0.2.3.9000/reactable.js"></script><link rel="stylesheet" href="assets/css/styles.css">
</head>
<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg navbar-dark "><div class="navbar-container container-fluid">
      <a class="navbar-brand" href="./index.html">
    <img src="./assets/icon.png" alt=""></a>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item compact">
    <a class="nav-link" href="./index.html"><i class="bi bi-home" role="img">
</i> 
 </a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-a.-processing" role="button" data-bs-toggle="dropdown" aria-expanded="false">A. Processing</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-a.-processing">
<li class="dropdown-header">Workflows</li>
        <li><hr class="dropdown-divider"></li>
        <li>
    <a class="dropdown-item" href="./dada2.html">
 <span class="dropdown-text">1. DADA2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./data-prep.html">
 <span class="dropdown-text">2. Data Set Prep</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./filtering.html">
 <span class="dropdown-text">3. Filtering</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-b.-diversity" role="button" data-bs-toggle="dropdown" aria-expanded="false">B. Diversity</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-b.-diversity">
<li class="dropdown-header">Workflows</li>
        <li><hr class="dropdown-divider"></li>
        <li>
    <a class="dropdown-item" href="./taxa.html">
 <span class="dropdown-text">4. Taxonomic</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./alpha.html">
 <span class="dropdown-text">5. Alpha Diversity</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./beta.html">
 <span class="dropdown-text">6. Beta Diversity</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./da.html">
 <span class="dropdown-text">7. DA ASVs &amp; Taxa</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./metadata.html">
 <span class="dropdown-text">8. Metadata</span></a>
  </li>  
    </ul>
</li>
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item">
    <a class="nav-link active" href="./data-availability.html" aria-current="page">DATA</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./pub.html">PAPER</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./som.html">SOM</a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/sweltr/high-temp/"><i class="bi bi-github" role="img" aria-label="Agua Salud GitHub">
</i> 
 </a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="./about.html"><i class="bi bi-question" role="img">
</i> 
 </a>
  </li>  
</ul>
<div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }"><div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">1. DADA2 Workflow</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title d-none d-lg-block">1. DADA2 Workflow</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i></button></div></div>
                  <div>
        <div class="description">
          <p>DADA2 workflow for processing the 16S rRNA and the ITS high temperature data sets. Workflow uses paired end reads, beginning with raw fastq files, ending with sequence and taxonomy tables.</p>
        </div>
      </div>
                </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto"><div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">A. Processing</a>
      <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./dada2.html" class="sidebar-item-text sidebar-link active">1. DADA2 Workflow</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./data-prep.html" class="sidebar-item-text sidebar-link">2. Data Set Preparation</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./filtering.html" class="sidebar-item-text sidebar-link">3. Filtering</a>
  </div>
</li>
    </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">B. Diversity</a>
      <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./taxa.html" class="sidebar-item-text sidebar-link">4. Taxonomic Diversity</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./alpha.html" class="sidebar-item-text sidebar-link">5. Alpha Diversity Estimates</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./beta.html" class="sidebar-item-text sidebar-link">6. Beta Diversity &amp; Dispersion Estimates</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./da.html" class="sidebar-item-text sidebar-link">7. Differentially Abundant ASVs &amp; Taxa</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./metadata.html" class="sidebar-item-text sidebar-link">8. Metadata Analysis</a>
  </div>
</li>
    </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">More</a>
      <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./data-availability.html" class="sidebar-item-text sidebar-link">Data Availability</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./som.html" class="sidebar-item-text sidebar-link">Electronic Supplementary Material</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./about.html" class="sidebar-item-text sidebar-link">Website Contruction</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pub.html" class="sidebar-item-text sidebar-link">Code for Publication Figures</a>
  </div>
</li>
    </ul>
</li>
    </ul>
</div>
</nav><!-- margin-sidebar --><div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#prerequisites" id="toc-prerequisites" class="nav-link active" data-scroll-target="#prerequisites">Prerequisites</a></li>
  <li>
<a href="#s-rrna-data" id="toc-s-rrna-data" class="nav-link" data-scroll-target="#s-rrna-data">16s rRNA Data</a>
  <ul class="collapse">
<li><a href="#workflow-input" id="toc-workflow-input" class="nav-link" data-scroll-target="#workflow-input">Workflow Input</a></li>
  <li><a href="#overview" id="toc-overview" class="nav-link" data-scroll-target="#overview">Overview</a></li>
  <li><a href="#set-working-environment" id="toc-set-working-environment" class="nav-link" data-scroll-target="#set-working-environment">1. Set Working Environment</a></li>
  <li><a href="#define-primers" id="toc-define-primers" class="nav-link" data-scroll-target="#define-primers">2. Define Primers</a></li>
  <li><a href="#remove-primers" id="toc-remove-primers" class="nav-link" data-scroll-target="#remove-primers">3. Remove Primers</a></li>
  <li><a href="#quality-assessment-filtering" id="toc-quality-assessment-filtering" class="nav-link" data-scroll-target="#quality-assessment-filtering">4. Quality Assessment &amp; Filtering</a></li>
  <li><a href="#learn-error-rates" id="toc-learn-error-rates" class="nav-link" data-scroll-target="#learn-error-rates">5. Learn Error Rates</a></li>
  <li><a href="#dereplicate-reads" id="toc-dereplicate-reads" class="nav-link" data-scroll-target="#dereplicate-reads">6. Dereplicate Reads</a></li>
  <li><a href="#dada2-asv-inference" id="toc-dada2-asv-inference" class="nav-link" data-scroll-target="#dada2-asv-inference">7. DADA2 &amp; ASV Inference</a></li>
  <li><a href="#construct-sequence-table" id="toc-construct-sequence-table" class="nav-link" data-scroll-target="#construct-sequence-table">8. Construct Sequence Table</a></li>
  <li><a href="#remove-chimeras" id="toc-remove-chimeras" class="nav-link" data-scroll-target="#remove-chimeras">9. Remove Chimeras</a></li>
  <li><a href="#track-reads-through-workflow" id="toc-track-reads-through-workflow" class="nav-link" data-scroll-target="#track-reads-through-workflow">10. Track Reads through Workflow</a></li>
  <li><a href="#assign-taxonomy" id="toc-assign-taxonomy" class="nav-link" data-scroll-target="#assign-taxonomy">11. Assign Taxonomy</a></li>
  <li><a href="#save-image" id="toc-save-image" class="nav-link" data-scroll-target="#save-image">12. Save Image</a></li>
  <li><a href="#r-session-information-code" id="toc-r-session-information-code" class="nav-link" data-scroll-target="#r-session-information-code">R Session Information &amp; Code</a></li>
  </ul>
</li>
  <li>
<a href="#its-data" id="toc-its-data" class="nav-link" data-scroll-target="#its-data">ITS Data</a>
  <ul class="collapse">
<li><a href="#workflow-input-1" id="toc-workflow-input-1" class="nav-link" data-scroll-target="#workflow-input-1">Workflow Input</a></li>
  <li><a href="#overview-1" id="toc-overview-1" class="nav-link" data-scroll-target="#overview-1">Overview</a></li>
  <li><a href="#set-working-environment-1" id="toc-set-working-environment-1" class="nav-link" data-scroll-target="#set-working-environment-1">1. Set Working Environment</a></li>
  <li><a href="#define-primers-1" id="toc-define-primers-1" class="nav-link" data-scroll-target="#define-primers-1">2. Define Primers</a></li>
  <li><a href="#remove-primers-1" id="toc-remove-primers-1" class="nav-link" data-scroll-target="#remove-primers-1">3. Remove Primers</a></li>
  <li><a href="#quality-assessment-filtering-1" id="toc-quality-assessment-filtering-1" class="nav-link" data-scroll-target="#quality-assessment-filtering-1">4. Quality Assessment &amp; Filtering</a></li>
  <li><a href="#learn-error-rates-1" id="toc-learn-error-rates-1" class="nav-link" data-scroll-target="#learn-error-rates-1">5. Learn Error Rates</a></li>
  <li><a href="#dereplicate-reads-1" id="toc-dereplicate-reads-1" class="nav-link" data-scroll-target="#dereplicate-reads-1">6. Dereplicate Reads</a></li>
  <li><a href="#dada2-asv-inference-1" id="toc-dada2-asv-inference-1" class="nav-link" data-scroll-target="#dada2-asv-inference-1">7. DADA2 &amp; ASV Inference</a></li>
  <li><a href="#merge-paired-reads" id="toc-merge-paired-reads" class="nav-link" data-scroll-target="#merge-paired-reads">8. Merge Paired Reads</a></li>
  <li><a href="#construct-sequence-table-1" id="toc-construct-sequence-table-1" class="nav-link" data-scroll-target="#construct-sequence-table-1">9. Construct Sequence Table</a></li>
  <li><a href="#remove-chimeras-1" id="toc-remove-chimeras-1" class="nav-link" data-scroll-target="#remove-chimeras-1">10. Remove Chimeras</a></li>
  <li><a href="#track-reads-through-workflow-1" id="toc-track-reads-through-workflow-1" class="nav-link" data-scroll-target="#track-reads-through-workflow-1">11. Track Reads through Workflow</a></li>
  <li><a href="#assign-taxonomy-1" id="toc-assign-taxonomy-1" class="nav-link" data-scroll-target="#assign-taxonomy-1">12. Assign Taxonomy</a></li>
  <li><a href="#save-image-1" id="toc-save-image-1" class="nav-link" data-scroll-target="#save-image-1">13. Save Image</a></li>
  <li><a href="#r-session-information-code-1" id="toc-r-session-information-code-1" class="nav-link" data-scroll-target="#r-session-information-code-1">R Session Information &amp; Code</a></li>
  </ul>
</li>
  <li><a href="#workflow-output" id="toc-workflow-output" class="nav-link" data-scroll-target="#workflow-output">Workflow Output</a></li>
  </ul><div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/sweltr/high-temp/edit/main/dada2.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/sweltr/high-temp/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content"><details><summary><strong>Click here</strong> for setup information.
</summary><div class="cell">
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>echo <span class="op">=</span> <span class="cn">TRUE</span>, eval <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">919191</span><span class="op">)</span>
<span class="co">#pacman::p_depends(ape, local = TRUE)  </span>
<span class="co">#pacman::p_depends_reverse(ape, local = TRUE)  </span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://benjjneb.github.io/dada2/">dada2</a></span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/utils/packageDescription.html">packageVersion</a></span><span class="op">(</span><span class="st">"dada2"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">ShortRead</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/utils/packageDescription.html">packageVersion</a></span><span class="op">(</span><span class="st">"ShortRead"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/utils/packageDescription.html">packageVersion</a></span><span class="op">(</span><span class="st">"ggplot2"</span><span class="op">)</span>
<span class="fu">pacman</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pacman/man/p_load.html">p_load</a></span><span class="op">(</span><span class="va">tidyverse</span>, <span class="va">gridExtra</span>, <span class="va">grid</span>, 
               <span class="va">formatR</span>, <span class="va">reactable</span>, <span class="va">downloadthis</span>,
               <span class="va">captioner</span>, <span class="va">sparkline</span>, 
               install <span class="op">=</span> <span class="cn">FALSE</span>, update <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>scipen<span class="op">=</span><span class="fl">999</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html">opts_current</a></span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
  <span class="st">"cache"</span>,
  <span class="st">"cache.path"</span>,
  <span class="st">"cache.rebuild"</span>,
  <span class="st">"dependson"</span>,
  <span class="st">"autodep"</span>
<span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">remove</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="st">"assets"</span>, <span class="st">"functions.R"</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</details><div class="cell">

</div>
<div class="cell">

</div>
<section id="prerequisites" class="level1"><h1>Prerequisites</h1>
<p>In order to run this workflow, you either need the <em>raw</em> data, available on the figshare project site (see below), or the <em>trimmed</em> data, available from the European Nucleotide Archive (ENA) under project accession number <a href="https://www.ebi.ac.uk/ena/browser/view/PRJEB45074">PRJEB45074</a>. See the <a href="data-availability.html">Data Availability</a> page for more details.</p>
<p>This workflow contains the code we used to process the 16S rRNA and ITS data sets using <a href="https://benjjneb.github.io/dada2/">DADA2</a> <span class="citation" data-cites="callahan2016dada2">(<a href="#ref-callahan2016dada2" role="doc-biblioref">Callahan et al. 2016</a>)</span>. Workflow construction is based on the <a href="https://benjjneb.github.io/dada2/tutorial_1_8.html">DADA2 Pipeline Tutorial (1.8)</a> and the primer identification section of the <a href="https://benjjneb.github.io/dada2/ITS_workflow.html">DADA2 ITS Pipeline Workflow (1.8)</a>.</p>
</section><section id="s-rrna-data" class="level1"><h1>16s rRNA Data</h1>
<section id="workflow-input" class="level2"><h2 class="anchored" data-anchor-id="workflow-input">Workflow Input</h2>
<p>All files needed to run this workflow can be downloaded from figshare.</p>
<iframe src="https://widgets.figshare.com/articles/14686665/embed?show_title=1" width="100%" height="351" allowfullscreen="" frameborder="0">
</iframe>
<blockquote class="blockquote">
<p>NOTE: The reverse reads for 16S rRNA data set are unusable (see below), therefore this workflow deals with forward reads <em>only</em>.</p>
</blockquote>
<p>We will screen the forward reads for reverse primers and show the quality plots of the reverse reads at the beginning of the workflow. After that, reverse reads are not used.</p>
</section><section id="overview" class="level2"><h2 class="anchored" data-anchor-id="overview">Overview</h2>
<section id="sequence-files-samples" class="level3"><h3 class="anchored" data-anchor-id="sequence-files-samples">Sequence Files &amp; Samples</h3>
<p>We sequenced a total of 15 samples collected from 10 different plots at a single depth in each plot.</p>
<div class="cell">

</div>
<p><small><strong>(16S rRNA) Table 1 | Sample data &amp; associated sequencing information.</strong></small></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,U2FtcGxlX0lELFBsb3QsVHJlYXRtZW50LFRlbXAsRGVwdGgsUGFpcixyYXdfcmVhZHMsRmFzdHFfSURfRm9yd2FyZCxGYXN0cV9JRF9SZXZlcnNlLEk3X0luZGV4X0lELGluZGV4LEk1X0luZGV4X0lELGluZGV4MgpQMDFfRDAwXzAxMF9XM0EsUDAxLFdhcm0sMywwMF8wMTAsQSw4NTQ4OSxQMV9TMV9MMDAxX1IxXzAwMV8xNlMuZmFzdHEuZ3osUDFfUzFfTDAwMV9SMl8wMDFfMTZTLmZhc3RxLmd6LFNENzAxLHRhY3RhZ2d0LFNENTAxLEFBR0NBR0NBClAwMV9EMDBfMDEwX1c4QSxQMDEsV2FybSw4LDAwXzAxMCxBLDc3NjQ1LFAxLThDX1MxMV9MMDAxX1IxXzAwMV8xNlMuZmFzdHEuZ3osUDEtOENfUzExX0wwMDFfUjJfMDAxXzE2Uy5mYXN0cS5neixTRDcwMixhY2d0YWNndCxTRDUwMyxDR0FUQ1RBQwpQMDJfRDAwXzAxMF9DMEEsUDAyLENvbnRyb2wsMCwwMF8wMTAsQSw4NzA5MSxQMl9TMl9MMDAxX1IxXzAwMV8xNlMuZmFzdHEuZ3osUDJfUzJfTDAwMV9SMl8wMDFfMTZTLmZhc3RxLmd6LFNENzAxLHRhY3RhZ2d0LFNENTAyLEFDR0NHVEdBClAwM19EMDBfMDEwX1czQixQMDMsV2FybSwzLDAwXzAxMCxCLDkzMDY1LFAzX1MzX0wwMDFfUjFfMDAxXzE2Uy5mYXN0cS5neixQM19TM19MMDAxX1IyXzAwMV8xNlMuZmFzdHEuZ3osU0Q3MDEsdGFjdGFnZ3QsU0Q1MDMsQ0dBVENUQUMKUDAzX0QwMF8wMTBfVzhCLFAwMyxXYXJtLDgsMDBfMDEwLEIsNjY0MDEsUDMtOENfUzEyX0wwMDFfUjFfMDAxXzE2Uy5mYXN0cS5neixQMy04Q19TMTJfTDAwMV9SMl8wMDFfMTZTLmZhc3RxLmd6LFNENzAyLGFjZ3RhY2d0LFNENTA0LFRHQ0dUQ0FDClAwNF9EMDBfMDEwX0MwQixQMDQsQ29udHJvbCwwLDAwXzAxMCxCLDY3ODEwLFA0X1M0X0wwMDFfUjFfMDAxXzE2Uy5mYXN0cS5neixQNF9TNF9MMDAxX1IyXzAwMV8xNlMuZmFzdHEuZ3osU0Q3MDEsdGFjdGFnZ3QsU0Q1MDQsVEdDR1RDQUMKUDA1X0QwMF8wMTBfVzNDLFAwNSxXYXJtLDMsMDBfMDEwLEMsNTgyNjYsUDVfUzVfTDAwMV9SMV8wMDFfMTZTLmZhc3RxLmd6LFA1X1M1X0wwMDFfUjJfMDAxXzE2Uy5mYXN0cS5neixTRDcwMSx0YWN0YWdndCxTRDUwNSxHVENUQUdURwpQMDVfRDAwXzAxMF9XOEMsUDA1LFdhcm0sOCwwMF8wMTAsQywxMjA4OTIsUDUtOENfUzEzX0wwMDFfUjFfMDAxXzE2Uy5mYXN0cS5neixQNS04Q19TMTNfTDAwMV9SMl8wMDFfMTZTLmZhc3RxLmd6LFNENzAyLGFjZ3RhY2d0LFNENTA1LEdUQ1RBR1RHClAwNl9EMDBfMDEwX0MwQyxQMDYsQ29udHJvbCwwLDAwXzAxMCxDLDEwMTU5OSxQNl9TNl9MMDAxX1IxXzAwMV8xNlMuZmFzdHEuZ3osUDZfUzZfTDAwMV9SMl8wMDFfMTZTLmZhc3RxLmd6LFNENzAxLHRhY3RhZ2d0LFNENTA2LENUQUdUQVRHClAwN19EMDBfMDEwX1czRCxQMDcsV2FybSwzLDAwXzAxMCxELDc2NzgxLFA3X1M3X0wwMDFfUjFfMDAxXzE2Uy5mYXN0cS5neixQN19TN19MMDAxX1IyXzAwMV8xNlMuZmFzdHEuZ3osU0Q3MDEsdGFjdGFnZ3QsU0Q1MDcsR0FUQUdDR1QKUDA3X0QwMF8wMTBfVzhELFAwNyxXYXJtLDgsMDBfMDEwLEQsNzkwMzIsUDctOENfUzE0X0wwMDFfUjFfMDAxXzE2Uy5mYXN0cS5neixQNy04Q19TMTRfTDAwMV9SMl8wMDFfMTZTLmZhc3RxLmd6LFNENzAyLGFjZ3RhY2d0LFNENTA2LENUQUdUQVRHClAwOF9EMDBfMDEwX0MwRCxQMDgsQ29udHJvbCwwLDAwXzAxMCxELDc3NTk5LFA4X1M4X0wwMDFfUjFfMDAxXzE2Uy5mYXN0cS5neixQOF9TOF9MMDAxX1IyXzAwMV8xNlMuZmFzdHEuZ3osU0Q3MDEsdGFjdGFnZ3QsU0Q1MDgsVENUQUNBQ1QKUDA5X0QwMF8wMTBfVzNFLFAwOSxXYXJtLDMsMDBfMDEwLEUsOTU1NjYsUDlfUzlfTDAwMV9SMV8wMDFfMTZTLmZhc3RxLmd6LFA5X1M5X0wwMDFfUjJfMDAxXzE2Uy5mYXN0cS5neixTRDcwMixhY2d0YWNndCxTRDUwMSxBQUdDQUdDQQpQMDlfRDAwXzAxMF9XOEUsUDA5LFdhcm0sOCwwMF8wMTAsRSw5NTE4MixQOS04Q19TMTVfTDAwMV9SMV8wMDFfMTZTLmZhc3RxLmd6LFA5LThDX1MxNV9MMDAxX1IyXzAwMV8xNlMuZmFzdHEuZ3osU0Q3MDIsYWNndGFjZ3QsU0Q1MDcsR0FUQUdDR1QKUDEwX0QwMF8wMTBfQzBFLFAxMCxDb250cm9sLDAsMDBfMDEwLEUsNjc1ODMsUDEwX1MxMF9MMDAxX1IxXzAwMV8xNlMuZmFzdHEuZ3osUDEwX1MxMF9MMDAxX1IyXzAwMV8xNlMuZmFzdHEuZ3osU0Q3MDIsYWNndGFjZ3QsU0Q1MDIsQUNHQ0dUR0EK').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'ssu18_sample_seq_info.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="cell-output-display">

<div id="htmlwidget-75d662642cd4c45aefa9" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-75d662642cd4c45aefa9">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Sample_ID":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"Plot":["P01","P01","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09","P10"],"Treatment":["Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control"],"Temp":[3,8,0,3,8,0,3,8,0,3,8,0,3,8,0],"Depth":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"Pair":["A","A","A","B","B","B","C","C","C","D","D","D","E","E","E"],"raw_reads":[85489,77645,87091,93065,66401,67810,58266,120892,101599,76781,79032,77599,95566,95182,67583],"Fastq_ID_Forward":["P1_S1_L001_R1_001_16S.fastq.gz","P1-8C_S11_L001_R1_001_16S.fastq.gz","P2_S2_L001_R1_001_16S.fastq.gz","P3_S3_L001_R1_001_16S.fastq.gz","P3-8C_S12_L001_R1_001_16S.fastq.gz","P4_S4_L001_R1_001_16S.fastq.gz","P5_S5_L001_R1_001_16S.fastq.gz","P5-8C_S13_L001_R1_001_16S.fastq.gz","P6_S6_L001_R1_001_16S.fastq.gz","P7_S7_L001_R1_001_16S.fastq.gz","P7-8C_S14_L001_R1_001_16S.fastq.gz","P8_S8_L001_R1_001_16S.fastq.gz","P9_S9_L001_R1_001_16S.fastq.gz","P9-8C_S15_L001_R1_001_16S.fastq.gz","P10_S10_L001_R1_001_16S.fastq.gz"],"Fastq_ID_Reverse":["P1_S1_L001_R2_001_16S.fastq.gz","P1-8C_S11_L001_R2_001_16S.fastq.gz","P2_S2_L001_R2_001_16S.fastq.gz","P3_S3_L001_R2_001_16S.fastq.gz","P3-8C_S12_L001_R2_001_16S.fastq.gz","P4_S4_L001_R2_001_16S.fastq.gz","P5_S5_L001_R2_001_16S.fastq.gz","P5-8C_S13_L001_R2_001_16S.fastq.gz","P6_S6_L001_R2_001_16S.fastq.gz","P7_S7_L001_R2_001_16S.fastq.gz","P7-8C_S14_L001_R2_001_16S.fastq.gz","P8_S8_L001_R2_001_16S.fastq.gz","P9_S9_L001_R2_001_16S.fastq.gz","P9-8C_S15_L001_R2_001_16S.fastq.gz","P10_S10_L001_R2_001_16S.fastq.gz"],"I7_Index_ID":["SD701","SD702","SD701","SD701","SD702","SD701","SD701","SD702","SD701","SD701","SD702","SD701","SD702","SD702","SD702"],"index":["tactaggt","acgtacgt","tactaggt","tactaggt","acgtacgt","tactaggt","tactaggt","acgtacgt","tactaggt","tactaggt","acgtacgt","tactaggt","acgtacgt","acgtacgt","acgtacgt"],"I5_Index_ID":["SD501","SD503","SD502","SD503","SD504","SD504","SD505","SD505","SD506","SD507","SD506","SD508","SD501","SD507","SD502"],"index2":["AAGCAGCA","CGATCTAC","ACGCGTGA","CGATCTAC","TGCGTCAC","TGCGTCAC","GTCTAGTG","GTCTAGTG","CTAGTATG","GATAGCGT","CTAGTATG","TCTACACT","AAGCAGCA","GATAGCGT","ACGCGTGA"]},"columns":[{"accessor":"Sample_ID","name":"Sample ID","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"header":"Sample ID","align":"left","footerStyle":{"fontWeight":"bold"},"footer":"Total reads","minWidth":150,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"Plot","name":"Plot","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01","P01","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09","P10"],"header":"Plot","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Treatment","name":"Treatment","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control"],"header":"Treatment","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Temp","name":"Temp","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["3","8","0","3","8","0","3","8","0","3","8","0","3","8","0"],"header":"Temp","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Depth","name":"Depth","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"header":"Depth","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Pair","name":"Pair","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["A","A","A","B","B","B","C","C","C","D","D","D","E","E","E"],"header":"Pair","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"raw_reads","name":"raw_reads","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["85489","77645","87091","93065","66401","67810","58266","120892","101599","76781","79032","77599","95566","95182","67583"],"header":"raw reads","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"1250001"},{"accessor":"Fastq_ID_Forward","name":"Fastq_ID_Forward","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P1_S1_L001_R1_001_16S.fastq.gz","P1-8C_S11_L001_R1_001_16S.fastq.gz","P2_S2_L001_R1_001_16S.fastq.gz","P3_S3_L001_R1_001_16S.fastq.gz","P3-8C_S12_L001_R1_001_16S.fastq.gz","P4_S4_L001_R1_001_16S.fastq.gz","P5_S5_L001_R1_001_16S.fastq.gz","P5-8C_S13_L001_R1_001_16S.fastq.gz","P6_S6_L001_R1_001_16S.fastq.gz","P7_S7_L001_R1_001_16S.fastq.gz","P7-8C_S14_L001_R1_001_16S.fastq.gz","P8_S8_L001_R1_001_16S.fastq.gz","P9_S9_L001_R1_001_16S.fastq.gz","P9-8C_S15_L001_R1_001_16S.fastq.gz","P10_S10_L001_R1_001_16S.fastq.gz"],"header":"Fastq ID Forward","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Fastq_ID_Reverse","name":"Fastq_ID_Reverse","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P1_S1_L001_R2_001_16S.fastq.gz","P1-8C_S11_L001_R2_001_16S.fastq.gz","P2_S2_L001_R2_001_16S.fastq.gz","P3_S3_L001_R2_001_16S.fastq.gz","P3-8C_S12_L001_R2_001_16S.fastq.gz","P4_S4_L001_R2_001_16S.fastq.gz","P5_S5_L001_R2_001_16S.fastq.gz","P5-8C_S13_L001_R2_001_16S.fastq.gz","P6_S6_L001_R2_001_16S.fastq.gz","P7_S7_L001_R2_001_16S.fastq.gz","P7-8C_S14_L001_R2_001_16S.fastq.gz","P8_S8_L001_R2_001_16S.fastq.gz","P9_S9_L001_R2_001_16S.fastq.gz","P9-8C_S15_L001_R2_001_16S.fastq.gz","P10_S10_L001_R2_001_16S.fastq.gz"],"header":"Fastq ID Reverse","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"I7_Index_ID","name":"I7_Index_ID","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["SD701","SD702","SD701","SD701","SD702","SD701","SD701","SD702","SD701","SD701","SD702","SD701","SD702","SD702","SD702"],"header":"I7 Index ID","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"index","name":"index","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["tactaggt","acgtacgt","tactaggt","tactaggt","acgtacgt","tactaggt","tactaggt","acgtacgt","tactaggt","tactaggt","acgtacgt","tactaggt","acgtacgt","acgtacgt","acgtacgt"],"header":"index","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"I5_Index_ID","name":"I5_Index_ID","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["SD501","SD503","SD502","SD503","SD504","SD504","SD505","SD505","SD506","SD507","SD506","SD508","SD501","SD507","SD502"],"header":"I5 Index ID","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"index2","name":"index2","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["AAGCAGCA","CGATCTAC","ACGCGTGA","CGATCTAC","TGCGTCAC","TGCGTCAC","GTCTAGTG","GTCTAGTG","CTAGTATG","GATAGCGT","CTAGTATG","TCTACACT","AAGCAGCA","GATAGCGT","ACGCGTGA"],"header":"index2","align":"center","footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"d0a61580e78aeaf7492f0e095e84ee27"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p></p>
</section><section id="workflow" class="level3"><h3 class="anchored" data-anchor-id="workflow">Workflow</h3>
<p>The workflow consists of the following steps:</p>
<table class="table">
<colgroup>
<col style="width: 6%">
<col style="width: 26%">
<col style="width: 66%">
</colgroup>
<thead><tr class="header">
<th>Step</th>
<th>Command</th>
<th>What weâ€™re doing</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>multiple</td>
<td>prepare input file names &amp; paths</td>
</tr>
<tr class="even">
<td>2</td>
<td>multiple</td>
<td>Define primers (all orientations)</td>
</tr>
<tr class="odd">
<td>3</td>
<td><code>cutadapt</code></td>
<td>Remove primers</td>
</tr>
<tr class="even">
<td>4</td>
<td><code><a href="https://rdrr.io/pkg/dada2/man/filterAndTrim.html">filterAndTrim()</a></code></td>
<td>assess quality &amp; filter reads</td>
</tr>
<tr class="odd">
<td>5</td>
<td><code><a href="https://rdrr.io/pkg/dada2/man/learnErrors.html">learnErrors()</a></code></td>
<td>generate an error model for the data</td>
</tr>
<tr class="even">
<td>6</td>
<td><code>derepFastq</code></td>
<td>dereplicate sequences</td>
</tr>
<tr class="odd">
<td>7</td>
<td><code><a href="https://rdrr.io/pkg/dada2/man/dada.html">dada()</a></code></td>
<td>infer ASVs on both forward &amp; reverse reads independently</td>
</tr>
<tr class="even">
<td>8</td>
<td><code><a href="https://rdrr.io/pkg/dada2/man/makeSequenceTable.html">makeSequenceTable()</a></code></td>
<td>generate a count table</td>
</tr>
<tr class="odd">
<td>9</td>
<td><code><a href="https://rdrr.io/pkg/dada2/man/removeBimeraDenovo.html">removeBimeraDenovo()</a></code></td>
<td>screen for &amp; remove chimeras</td>
</tr>
<tr class="even">
<td>10</td>
<td></td>
<td>track reads through workflow</td>
</tr>
<tr class="odd">
<td>11</td>
<td><code><a href="https://rdrr.io/pkg/dada2/man/assignTaxonomy.html">assignTaxonomy()</a></code></td>
<td>assign taxonomy &amp; finish workflow</td>
</tr>
<tr class="even">
<td>12</td>
<td><code><a href="https://rdrr.io/r/base/save.html">save.image()</a></code></td>
<td>save an image of seqtab &amp; taxtab for next part of workflow</td>
</tr>
</tbody>
</table></section></section><section id="set-working-environment" class="level2"><h2 class="anchored" data-anchor-id="set-working-environment">1. Set Working Environment</h2>
<p>Next, we need to setup the working environment by renaming the fastq files and defining a path for the working directory.</p>
<section id="rename-samples" class="level3"><h3 class="anchored" data-anchor-id="rename-samples">Rename samples</h3>
<p>To make the parsing easier, we will eliminate the lane and well number from each sample. If you do not wish to do that you will need to adjust the code accordingly. I am <em>positive</em> there are more elegant ways of doing this.</p>
<blockquote class="blockquote">
<p>CAUTION. If you use this for your own data, please check that this code works on a few backup files before proceeding.</p>
</blockquote>
<div class="cell">
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span><span class="op">(</span><span class="st">"RAW"</span><span class="op">)</span>
<span class="va">orig_fastq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"*.fastq.gz"</span><span class="op">)</span>
<span class="va">newname_fastq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_S.*_L001"</span>, <span class="st">""</span>, <span class="va">orig_fastq</span><span class="op">)</span>
<span class="va">newname_fastq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_001"</span>, <span class="st">""</span>, <span class="va">newname_fastq</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.rename</a></span><span class="op">(</span><span class="va">orig_fastq</span>, <span class="va">newname_fastq</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span><span class="op">(</span><span class="st">"../"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">path</span> <span class="op">&lt;-</span> <span class="st">"RAW"</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "P1_R1.fastq.gz"    "P1_R2.fastq.gz"    "P1-8C_R1.fastq.gz"
[4] "P1-8C_R2.fastq.gz" "P10_R1.fastq.gz"   "P10_R2.fastq.gz"  </code></pre>
</div>
</div>
<p>Then, we generate matched lists of the forward and reverse read files. We also parse out the sample name.</p>
<div class="cell">
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">fnFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span>, pattern <span class="op">=</span> <span class="st">"_R1.fastq.gz"</span>, full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
<span class="va">fnRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span>, pattern <span class="op">=</span> <span class="st">"_R2.fastq.gz"</span>, full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">p1a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/plotQualityProfile.html">plotQualityProfile</a></span><span class="op">(</span><span class="va">fnFs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, aggregate <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">p2a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/plotQualityProfile.html">plotQualityProfile</a></span><span class="op">(</span><span class="va">fnRs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, aggregate <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="va">p3a</span> <span class="op">&lt;-</span> <span class="fu">grid.arrange</span><span class="op">(</span><span class="va">p1a</span>, <span class="va">p2a</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"plot_qscores_2018a.png"</span>, <span class="va">p3a</span>, width <span class="op">=</span> <span class="fl">7</span>, height <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">

</div>
<p>Unfortunately, the reverse reads are unusable.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="include/dada2/ssu18_plot_qscores_a.png" class="img-fluid" width="1050"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 1 | Aggregated quality score plots for forward (left) &amp; reverse (right) reads.</strong></small></p>
</section></section><section id="define-primers" class="level2"><h2 class="anchored" data-anchor-id="define-primers">2. Define Primers</h2>
<p>Before we start the DADA2 workflow we need to run <a href="https://github.com/marcelm/cutadapt">catadapt</a> <span class="citation" data-cites="martin2011cutadapt">(<a href="#ref-martin2011cutadapt" role="doc-biblioref">Martin 2011</a>)</span> on all <code>fastq.gz</code> files to trim the primers. For bacteria and archaea, we amplified the V4 hypervariable region of the 16S rRNA gene using the primer pair 515F (GTGCCAGCMGCCGCGGTAA) and 806R (GGACTACHVGGGTWTCTAAT) <span class="citation" data-cites="caporaso2011global">(<a href="#ref-caporaso2011global" role="doc-biblioref">Caporaso et al. 2011</a>)</span>, which should yield an amplicon length of about 253 bp.</p>
<p>First we define the primers.</p>
<div class="cell">
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">FWD</span> <span class="op">&lt;-</span> <span class="st">"GTGCCAGCMGCCGCGGTAA"</span>
<span class="va">REV</span> <span class="op">&lt;-</span> <span class="st">"GGACTACHVGGGTWTCTAAT"</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Next, we check the presence and orientation of these primers in the data. I started doing this for ITS data because of primer read-through but I really like the general idea of doing it just to make sure nothing funny is going of with the data. To do this, we will create all orientations of the input primer sequences. In other words the <em>Forward</em>, <em>Complement</em>, <em>Reverse</em>, and <em>Reverse Complement</em> variations.</p>
<div class="cell">
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">allOrients</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">primer</span><span class="op">)</span> <span class="op">{</span>
    <span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings">Biostrings</a></span><span class="op">)</span>
    <span class="va">dna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/DNAString-class.html">DNAString</a></span><span class="op">(</span><span class="va">primer</span><span class="op">)</span>
    <span class="va">orients</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span>Forward <span class="op">=</span> <span class="va">dna</span>,
                 Complement <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html">complement</a></span><span class="op">(</span><span class="va">dna</span><span class="op">)</span>,
                 Reverse <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/reverse-methods.html">reverse</a></span><span class="op">(</span><span class="va">dna</span><span class="op">)</span>,
                 RevComp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html">reverseComplement</a></span><span class="op">(</span><span class="va">dna</span><span class="op">)</span><span class="op">)</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">sapply</a></span><span class="op">(</span><span class="va">orients</span>, <span class="va">toString</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="va">FWD.orients</span> <span class="op">&lt;-</span> <span class="fu">allOrients</span><span class="op">(</span><span class="va">FWD</span><span class="op">)</span>
<span class="va">REV.orients</span> <span class="op">&lt;-</span> <span class="fu">allOrients</span><span class="op">(</span><span class="va">REV</span><span class="op">)</span>
<span class="va">FWD.orients</span>
<span class="va">REV.orients</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>              Forward            Complement               Reverse
"GTGCCAGCMGCCGCGGTAA" "CACGGTCGKCGGCGCCATT" "AATGGCGCCGMCGACCGTG"
              RevComp
"TTACCGCGGCKGCTGGCAC"
               Forward             Complement                Reverse
"GGACTACHVGGGTWTCTAAT" "CCTGATGDBCCCAWAGATTA" "TAATCTWTGGGVHCATCAGG"
               RevComp
"ATTAGAWACCCBDGTAGTCC"</code></pre>
<p>Now we do a little pre-filter step to eliminate ambiguous bases (<em>Ns</em>) because <em>Ns</em> make mapping of short primer sequences difficult. This step removes any reads with <em>Ns</em>. Again, set some files paths, this time for the filtered reads.</p>
<div class="cell">
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">fnFs.filtN</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"filtN"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html">basename</a></span><span class="op">(</span><span class="va">fnFs</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/dada2/man/filterAndTrim.html">filterAndTrim</a></span><span class="op">(</span><span class="va">fnFs</span>, <span class="va">fnFs.filtN</span>,
              maxN <span class="op">=</span> <span class="fl">0</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Nice. Time to assess the number of times a primer (and all primer orientations) appear in the forward and reverse reads. According to the workflow, counting the primers on one set of paired end fastq files is sufficient to see if there is a problem. This assumes that all the files were created using the same library prep. Basically for both primers, we will search for all four orientations in both forward and reverse reads. Since this is 16S rRNA we do not anticipate any issues but it is worth checking anyway.</p>
<div class="cell">
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">sampnum</span> <span class="op">&lt;-</span> <span class="fl">2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">primerHits</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">primer</span>, <span class="va">fn</span><span class="op">)</span> <span class="op">{</span>
    <span class="co"># Counts number of reads in which the primer is found</span>
    <span class="va">nhits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/matchPattern.html">vcountPattern</a></span><span class="op">(</span><span class="va">primer</span>, <span class="fu"><a href="https://rdrr.io/pkg/ShortRead/man/ShortRead-class.html">sread</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ShortRead/man/readFastq.html">readFastq</a></span><span class="op">(</span><span class="va">fn</span><span class="op">)</span><span class="op">)</span>, fixed <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">nhits</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="forward-primers" class="level4"><h4 class="anchored" data-anchor-id="forward-primers">Forward primers</h4>
<div class="cell">
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html">rbind</a></span><span class="op">(</span>FWD.ForwardReads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">sapply</a></span><span class="op">(</span><span class="va">FWD.orients</span>, <span class="va">primerHits</span>,
                                fn <span class="op">=</span> <span class="va">fnFs.filtN</span><span class="op">[[</span><span class="va">sampnum</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>                 Forward Complement Reverse RevComp
FWD.ForwardReads   74707          0       0       0</code></pre>
</section><section id="reverse-primers" class="level4"><h4 class="anchored" data-anchor-id="reverse-primers">Reverse primers</h4>
<div class="cell">
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html">rbind</a></span><span class="op">(</span>REV.ForwardReads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">sapply</a></span><span class="op">(</span><span class="va">REV.orients</span>, <span class="va">primerHits</span>,
                                fn <span class="op">=</span> <span class="va">fnFs.filtN</span><span class="op">[[</span><span class="va">sampnum</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>                 Forward Complement Reverse RevComp
REV.ForwardReads       0          0       0     165</code></pre>
<p>As expected, forward primers predominantly in the forward reads and very little evidence of reverse primers.</p>
</section></section><section id="remove-primers" class="level2"><h2 class="anchored" data-anchor-id="remove-primers">3. Remove Primers</h2>
<p>Now we can run <a href="https://github.com/marcelm/cutadapt">catadapt</a> <span class="citation" data-cites="martin2011cutadapt">(<a href="#ref-martin2011cutadapt" role="doc-biblioref">Martin 2011</a>)</span> to remove the primers from the fastq sequences. A little setup first. If this command executes successfully it means R has found cutadapt.</p>
<div class="cell">
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">cutadapt</span> <span class="op">&lt;-</span> <span class="st">"/Users/rad/miniconda3/envs/cutadapt/bin/cutadapt"</span>
<span class="fu"><a href="https://rdrr.io/r/base/system2.html">system2</a></span><span class="op">(</span><span class="va">cutadapt</span>, args <span class="op">=</span> <span class="st">"--version"</span><span class="op">)</span> <span class="co"># Run shell commands from R</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>2.8</code></pre>
<p>We set paths and trim the forward primer and the reverse-complement of the reverse primer off of R1 (forward reads) and trim the reverse primer and the reverse-complement of the forward primer off of R2 (reverse reads).</p>
<div class="cell">
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">path.cut</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"cutadapt"</span><span class="op">)</span>
<span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files2.html">dir.exists</a></span><span class="op">(</span><span class="va">path.cut</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/files2.html">dir.create</a></span><span class="op">(</span><span class="va">path.cut</span><span class="op">)</span>
<span class="va">fnFs.cut</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path.cut</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html">basename</a></span><span class="op">(</span><span class="va">fnFs</span><span class="op">)</span><span class="op">)</span>

<span class="va">FWD.RC</span> <span class="op">&lt;-</span> <span class="fu">dada2</span><span class="fu">:::</span><span class="fu"><a href="https://rdrr.io/pkg/dada2/man/rc.html">rc</a></span><span class="op">(</span><span class="va">FWD</span><span class="op">)</span>
<span class="va">REV.RC</span> <span class="op">&lt;-</span> <span class="fu">dada2</span><span class="fu">:::</span><span class="fu"><a href="https://rdrr.io/pkg/dada2/man/rc.html">rc</a></span><span class="op">(</span><span class="va">REV</span><span class="op">)</span>

<span class="va">R1.flags</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="st">"-g"</span>, <span class="va">FWD</span>, <span class="st">"-a"</span>, <span class="va">REV.RC</span><span class="op">)</span>

<span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_along</a></span><span class="op">(</span><span class="va">fnFs</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/system2.html">system2</a></span><span class="op">(</span><span class="va">cutadapt</span>,
                                   args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="va">R1.flags</span>,
                                            <span class="st">"-m"</span>, <span class="fl">20</span>, <span class="st">"-n"</span>, <span class="fl">2</span>, <span class="st">"-e"</span>, <span class="fl">0.12</span>,
                                            <span class="st">"-o"</span>, <span class="va">fnFs.cut</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>,
                                            <span class="va">fnFs.filtN</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>This is cutadapt 2.8 with Python 3.7.6
Command line parameters: -g GTGCCAGCMGCCGCGGTAA -a ATTAGAWACCCBDGTAGTCC \
                         -m 20 -n 2 -e 0.12 \
                         -o RAW/cutadapt/P1_R1.fastq.gz RAW/filtN/P1_R1.fastq.gz</code></pre>
<p><em>Note</em>. If the code above removes all of the base pairs in a sequence, you will get downstream errors unless you set the <code>-m</code> flag. This flag sets the minimum length and reads shorter than this will be discarded. Without this flag, reads of length 0 will be kept and cause issues. <em>Also, a lot of output will be written to the screen by cutadapt!</em>.</p>
<p>We can now count the number of primers in the sequences from the output of cutadapt.</p>
<div class="cell">
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html">rbind</a></span><span class="op">(</span>
  FWD.ForwardReads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">sapply</a></span><span class="op">(</span><span class="va">FWD.orients</span>, <span class="va">primerHits</span>, fn <span class="op">=</span> <span class="va">fnFs.cut</span><span class="op">[[</span><span class="va">sampnum</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
  REV.ForwardReads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">sapply</a></span><span class="op">(</span><span class="va">REV.orients</span>, <span class="va">primerHits</span>, fn <span class="op">=</span> <span class="va">fnFs.cut</span><span class="op">[[</span><span class="va">sampnum</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>                 Forward Complement Reverse RevComp
FWD.ForwardReads       0          0       0       0
REV.ForwardReads       0          0       0       0</code></pre>
<p>Basically, primers are no longer detected in the cutadapted reads. Now, for each sample, we can take a look at how the pre-filtering step and primer removal affected the total number of raw reads.</p>
<div class="cell">

</div>
<div class="cell">

</div>
<p><small><strong>(16S rRNA) Table 2 | Total reads per sample after prefiltering and primer removal (using cutadapt).</strong></small></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,U2FtcGxlX0lELHJhd19yZWFkcyxwcmVfZmlsdF9yZWFkcyxjdXRhZGFwdF9yZWFkcyxwZXJfcmVhZHNfa2VwdApQMDFfRDAwXzAxMF9XM0EsODU0ODksODU0ODIsODU0MDIsMC45OTkKUDAxX0QwMF8wMTBfVzhBLDc3NjQ1LDc3NjQyLDc3NDcwLDAuOTk4ClAwMl9EMDBfMDEwX0MwQSw4NzA5MSw4NzA4NSw4Njg3OCwwLjk5OApQMDNfRDAwXzAxMF9XM0IsOTMwNjUsOTMwNTMsOTI4NTEsMC45OTgKUDAzX0QwMF8wMTBfVzhCLDY2NDAxLDY2MzY3LDY1NDk4LDAuOTg2ClAwNF9EMDBfMDEwX0MwQiw2NzgxMCw2NzgwMCw2NzYxNywwLjk5NwpQMDVfRDAwXzAxMF9XM0MsNTgyNjYsNTgyNTksNTgwMDYsMC45OTYKUDA1X0QwMF8wMTBfVzhDLDEyMDg5MiwxMjA4NDIsMTE3MDE1LDAuOTY4ClAwNl9EMDBfMDEwX0MwQywxMDE1OTksMTAxNTkyLDEwMTM2MywwLjk5OApQMDdfRDAwXzAxMF9XM0QsNzY3ODEsNzY3NzYsNzY1ODMsMC45OTcKUDA3X0QwMF8wMTBfVzhELDc5MDMyLDc5MDIyLDc4NzI0LDAuOTk2ClAwOF9EMDBfMDEwX0MwRCw3NzU5OSw3NzU5MSw3NzEzMywwLjk5NApQMDlfRDAwXzAxMF9XM0UsOTU1NjYsOTU1NjQsOTUzMTAsMC45OTcKUDA5X0QwMF8wMTBfVzhFLDk1MTgyLDk0NDU3LDQ0MjUwLDAuNDY1ClAxMF9EMDBfMDEwX0MwRSw2NzU4Myw2NzU3OSw2NzQzMiwwLjk5OAo=').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'ssu18_cutadapt.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="cell-output-display">

<div id="htmlwidget-189bf7e795965c842df7" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-189bf7e795965c842df7">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Sample_ID":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"raw_reads":[85489,77645,87091,93065,66401,67810,58266,120892,101599,76781,79032,77599,95566,95182,67583],"pre_filt_reads":[85482,77642,87085,93053,66367,67800,58259,120842,101592,76776,79022,77591,95564,94457,67579],"cutadapt_reads":[85402,77470,86878,92851,65498,67617,58006,117015,101363,76583,78724,77133,95310,44250,67432],"per_reads_kept":[0.999,0.998,0.998,0.998,0.986,0.997,0.996,0.968,0.998,0.997,0.996,0.994,0.997,0.465,0.998]},"columns":[{"accessor":"Sample_ID","name":"Sample ID","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"header":"Sample ID","align":"left","footerStyle":{"fontWeight":"bold"},"footer":"Total reads","minWidth":150,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"raw_reads","name":"raw_reads","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["85489","77645","87091","93065","66401","67810","58266","120892","101599","76781","79032","77599","95566","95182","67583"],"header":"raw reads","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"1250001"},{"accessor":"pre_filt_reads","name":"pre_filt_reads","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["85482","77642","87085","93053","66367","67800","58259","120842","101592","76776","79022","77591","95564","94457","67579"],"header":"pre filt reads","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"1249111"},{"accessor":"cutadapt_reads","name":"cutadapt_reads","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["85402","77470","86878","92851","65498","67617","58006","117015","101363","76583","78724","77133","95310","44250","67432"],"header":"cutadapt reads","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"1191532"},{"accessor":"per_reads_kept","name":"per_reads_kept","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["0.999","0.998","0.998","0.998","0.986","0.997","0.996","0.968","0.998","0.997","0.996","0.994","0.997","0.465","0.998"],"header":"per reads kept","align":"center","footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"cf6f2904b0768c9d8cc3684f58d63f4c"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section><section id="quality-assessment-filtering" class="level2"><h2 class="anchored" data-anchor-id="quality-assessment-filtering">4. Quality Assessment &amp; Filtering</h2>
<p>We need the forward and reverse fastq file names plus the sample names.</p>
<div class="cell">
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">cutFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path.cut</span>, pattern <span class="op">=</span> <span class="st">"_R1.fastq.gz"</span>,
                         full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
<span class="va">get.sample.name</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">fname</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html">basename</a></span><span class="op">(</span><span class="va">fname</span><span class="op">)</span>, <span class="st">"_"</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>
<span class="va">sample.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unname.html">unname</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">sapply</a></span><span class="op">(</span><span class="va">cutFs</span>, <span class="va">get.sample.name</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">head</a></span><span class="op">(</span><span class="va">sample.names</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"P1"</span>, <span class="st">"P1-8C"</span>, <span class="st">"P10"</span>, <span class="st">"P2"</span>, <span class="st">"P3"</span>, <span class="st">"P3-8C"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "P1"    "P1-8C" "P10"   "P2"    "P3"    "P3-8C"</code></pre>
</div>
</div>
<p>First letâ€™s look at the quality of our reads. The numbers in brackets specify which samples to view. Here we are looking at an aggregate plot of all data (except the negative control).</p>
<section id="quality-plots" class="level3"><h3 class="anchored" data-anchor-id="quality-plots">Quality plots</h3>
<div class="cell">
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/plotQualityProfile.html">plotQualityProfile</a></span><span class="op">(</span><span class="va">cutFs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, aggregate <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"ssu18_plot_qscores.png"</span>, <span class="va">p1</span>, width <span class="op">=</span> <span class="fl">7</span>, height <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"ssu18_plot_qscores.png"</span>, <span class="va">p1</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">

</div>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/dada2/ssu18_plot_qscores_b.png" class="img-fluid figure-img" style="width:55.0%"></p>
</figure>
</div>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 2 | Aggregated quality score plot for forward reads after prefilter &amp; primer removal.</strong></small></p>
<p>The forward reads look good.</p>
</section><section id="filtering" class="level3"><h3 class="anchored" data-anchor-id="filtering">Filtering</h3>
<p>First, we again make some path variables and setup a new directory of filtered reads.</p>
<div class="cell">
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">filtFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path.cut</span>, <span class="st">"filtered"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html">basename</a></span><span class="op">(</span><span class="va">cutFs</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>And then we trim the reads.</p>
<div class="cell">
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/filterAndTrim.html">filterAndTrim</a></span><span class="op">(</span><span class="va">cutFs</span>, <span class="va">filtFs</span>,
                     maxN <span class="op">=</span> <span class="fl">0</span>, maxEE <span class="op">=</span> <span class="fl">2</span>,
                     truncQ <span class="op">=</span> <span class="fl">2</span>, rm.phix <span class="op">=</span> <span class="cn">TRUE</span>,
                     compress <span class="op">=</span> <span class="cn">TRUE</span>, multithread <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>
<span class="va">out</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout-note callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>These parameters should be set based on the anticipated length of the amplicon and the read quality.</p>
</div>
</div>
<p>And here is a table of how the filtering step affected the number of reads in each sample. As you can see, there are a few samples that started with a low read count to begin withâ€”we will likely remove those samples at some point.</p>
<div class="cell">

</div>
<p><small><strong>(16S rRNA) Table 3 | Total reads per sample after filtering.</strong></small></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,U2FtcGxlX0lELHJlYWRzX2luLHJlYWRzX291dCxwZXJfcmVhZHNfa2VwdApQMDFfRDAwXzAxMF9XM0EsODU0MDIsNzk4MDQsMC45MzQKUDAxX0QwMF8wMTBfVzhBLDc3NDcwLDcyNzYwLDAuOTM5ClAwMl9EMDBfMDEwX0MwQSw4Njg3OCw4MTI0MCwwLjkzNQpQMDNfRDAwXzAxMF9XM0IsOTI4NTEsODY3MzIsMC45MzQKUDAzX0QwMF8wMTBfVzhCLDY1NDk4LDYwODYwLDAuOTI5ClAwNF9EMDBfMDEwX0MwQiw2NzYxNyw2MzAzNywwLjkzMgpQMDVfRDAwXzAxMF9XM0MsNTgwMDYsNTQxMjcsMC45MzMKUDA1X0QwMF8wMTBfVzhDLDExNzAxNSwxMDk0MjEsMC45MzUKUDA2X0QwMF8wMTBfQzBDLDEwMTM2Myw5NDc0MCwwLjkzNQpQMDdfRDAwXzAxMF9XM0QsNzY1ODMsNzE2NzgsMC45MzYKUDA3X0QwMF8wMTBfVzhELDc4NzI0LDczNjM5LDAuOTM1ClAwOF9EMDBfMDEwX0MwRCw3NzEzMyw3MjI2MywwLjkzNwpQMDlfRDAwXzAxMF9XM0UsOTUzMTAsODkzMTUsMC45MzcKUDA5X0QwMF8wMTBfVzhFLDQ0MjUwLDMzMDAwLDAuNzQ2ClAxMF9EMDBfMDEwX0MwRSw2NzQzMiw2MjU5OCwwLjkyOAo=').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'ssu18_filter.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="cell-output-display">

<div id="htmlwidget-4e280cda47fb19178af4" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-4e280cda47fb19178af4">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Sample_ID":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"reads_in":[85402,77470,86878,92851,65498,67617,58006,117015,101363,76583,78724,77133,95310,44250,67432],"reads_out":[79804,72760,81240,86732,60860,63037,54127,109421,94740,71678,73639,72263,89315,33000,62598],"per_reads_kept":[0.934,0.939,0.935,0.934,0.929,0.932,0.933,0.935,0.935,0.936,0.935,0.937,0.937,0.746,0.928]},"columns":[{"accessor":"Sample_ID","name":"Sample ID","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"header":"Sample ID","align":"left","footerStyle":{"fontWeight":"bold"},"footer":"Total reads","minWidth":150,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"reads_in","name":"reads_in","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["85402","77470","86878","92851","65498","67617","58006","117015","101363","76583","78724","77133","95310","44250","67432"],"header":"reads in","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"1191532"},{"accessor":"reads_out","name":"reads_out","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["79804","72760","81240","86732","60860","63037","54127","109421","94740","71678","73639","72263","89315","33000","62598"],"header":"reads out","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"1105214"},{"accessor":"per_reads_kept","name":"per_reads_kept","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["0.934","0.939","0.935","0.934","0.929","0.932","0.933","0.935","0.935","0.936","0.935","0.937","0.937","0.746","0.928"],"header":"per reads kept","align":"center","footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"4ce7cbb181333ed4728cdf7cbfc942b7"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section></section><section id="learn-error-rates" class="level2"><h2 class="anchored" data-anchor-id="learn-error-rates">5. Learn Error Rates</h2>
<p>Now it is time to assess the error rate of the data. The DADA2 algorithm uses a parametric error model. Every amplicon data set has a different set of error rates and the <code>learnErrors</code> method learns this error model <em>from the data</em>. It does this by alternating estimation of the error rates and inference of sample composition until they converge on a jointly consistent solution. The algorithm begins with an initial guess, for which the maximum possible error rates in the data are used.</p>
<section id="forward-reads" class="level4"><h4 class="anchored" data-anchor-id="forward-reads">Forward Reads</h4>
<div class="cell">
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">errF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/learnErrors.html">learnErrors</a></span><span class="op">(</span><span class="va">filtFs</span>, multithread <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>103002523 total bases in 443994 reads from 6 samples will be used for learning the error rates.</code></pre>
<div class="cell">
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/dada2/man/plotErrors.html">plotErrors</a></span><span class="op">(</span><span class="va">errF</span>, nominalQ<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="include/dada2/ssu18_plot_errorF.png" class="img-fluid" width="1050"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 3 | Forward reads: Observed frequency of each transition (e.g., T -&gt; G) as a function of the associated quality score.</strong></small></p>
<p>The error rates for each possible transition (A to C, A to G, etc.) are shown. Points are the observed error rates for each consensus quality score. The black line shows the estimated error rates after convergence of the machine-learning algorithm. The red line shows the error rates expected under the nominal definition of the Q-score. Here the estimated error rates (black line) are a good fit to the observed rates (points), and the error rates drop with increased quality as expected.</p>
</section></section><section id="dereplicate-reads" class="level2"><h2 class="anchored" data-anchor-id="dereplicate-reads">6. Dereplicate Reads</h2>
<p>Now we can use <code>derepFastq</code> to identify the unique sequences in the forward and reverse fastq files.</p>
<section id="forward-reads-1" class="level4"><h4 class="anchored" data-anchor-id="forward-reads-1">Forward Reads</h4>
<div class="cell">

</div>
<div class="cell">
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">derepFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/derepFastq.html">derepFastq</a></span><span class="op">(</span><span class="va">filtFs</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">derepFs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample.names</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<details><summary>
Detailed results of <code>derep</code> forward reads
</summary><pre><code>
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P1_R1.fastq.gz
Encountered 39950 unique sequences from 79804 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P1-8C_R1.fastq.gz
Encountered 31163 unique sequences from 72760 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P10_R1.fastq.gz
Encountered 35274 unique sequences from 62598 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P2_R1.fastq.gz
Encountered 42119 unique sequences from 81240 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P3_R1.fastq.gz
Encountered 40336 unique sequences from 86732 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P3-8C_R1.fastq.gz
Encountered 31104 unique sequences from 60860 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P4_R1.fastq.gz
Encountered 35169 unique sequences from 63037 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P5_R1.fastq.gz
Encountered 27752 unique sequences from 54127 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P5-8C_R1.fastq.gz
Encountered 49045 unique sequences from 109421 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P6_R1.fastq.gz
Encountered 46853 unique sequences from 94740 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P7_R1.fastq.gz
Encountered 35363 unique sequences from 71678 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P7-8C_R1.fastq.gz
Encountered 35567 unique sequences from 73639 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P8_R1.fastq.gz
Encountered 38518 unique sequences from 72263 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P9_R1.fastq.gz
Encountered 43426 unique sequences from 89315 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P9-8C_R1.fastq.gz
Encountered 13673 unique sequences from 33000 total sequences read.
</code></pre>
</details><p><br></p>
</section></section><section id="dada2-asv-inference" class="level2"><h2 class="anchored" data-anchor-id="dada2-asv-inference">7. DADA2 &amp; ASV Inference</h2>
<p>At this point we are ready to apply the core sample inference algorithm (<code>dada</code>) to the filtered and trimmed sequence data. DADA2 offers three options for whether and how to pool samples for ASV inference.</p>
<p>If <code>pool = TRUE</code>, the algorithm will pool together all samples prior to sample inference.<br>
If <code>pool = FALSE</code>, sample inference is performed on each sample individually.<br>
If <code>pool = "pseudo"</code>, the algorithm will perform pseudo-pooling between individually processed samples.</p>
<p>We tested all three methods through the full pipeline. Click the <code>+</code> to see the results of the test. For our final analysis, we chose <code>pool = FALSE</code> for this data set.</p>
<details><summary>
Show/hide Results of <code>dada</code> pooling options
</summary><p><br></p>
<p>Here are summary tables of results from the tests. Values are from the final sequence table produced by each option.</p>
<div class="cell">

</div>
<p><small><strong>(16S rRNA) Table 4 | Total number of reads, total number of ASVs, minimum and maximum ASVs, followed by the number of singletons, doubletons, etc. for pooling options.</strong></small></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,dW5wb29sZWQscHN1ZWRvLHBvb2xlZAo5Mzc3NjEsOTQyMDMxLDkwNjA3NwoyMDMzMiwyMTI2OSwxMzIwNgo4MTYsMTI0OCwyMjA3CjMwODIsNDE1NSw3NTIyCjE3MTIyLDE3OTQ4LDI0MzAKMTI3MCwzMDksMTI4Mwo1ODYsMjY1LDg5MAozNTcsMjM4LDc5NwoyMTUsMjUwLDc1MgoxNjEsMjQ1LDY5NAoxMzAsMjUzLDcxMQo4NCwyNzUsNzA1CjEwMiwyMjksNzE3CjYzLDIzNSw3MzMKNTUsMjEwLDcxNQo2MCwyMTMsNzI2CjM2LDIyNSw3NTAKNDIsMjA1LDczMAo0OSwxNjksNTczCg==').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'ssu18_pooling_summary.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="cell-output-display">

<div id="htmlwidget-3a0ad04372aab487f689" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-3a0ad04372aab487f689">{"x":{"tag":{"name":"Reactable","attribs":{"data":{".rownames":["no_reads","no_asvs","min_asv","max_asv","n_samp_1","n_samp_2","n_samp_3","n_samp_4","n_samp_5","n_samp_6","n_samp_7","n_samp_8","n_samp_9","n_samp_10","n_samp_11","n_samp_12","n_samp_13","n_samp_14","n_samp_15"],"unpooled":[937761,20332,816,3082,17122,1270,586,357,215,161,130,84,102,63,55,60,36,42,49],"psuedo":[942031,21269,1248,4155,17948,309,265,238,250,245,253,275,229,235,210,213,225,205,169],"pooled":[906077,13206,2207,7522,2430,1283,890,797,752,694,711,705,717,733,715,726,750,730,573]},"columns":[{"accessor":".rownames","name":"","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["no_reads","no_asvs","min_asv","max_asv","n_samp_1","n_samp_2","n_samp_3","n_samp_4","n_samp_5","n_samp_6","n_samp_7","n_samp_8","n_samp_9","n_samp_10","n_samp_11","n_samp_12","n_samp_13","n_samp_14","n_samp_15"],"header":"","align":"center","footerStyle":{"fontWeight":"bold"},"rowHeader":true},{"accessor":"unpooled","name":"unpooled","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["937761","20332","816","3082","17122","1270","586","357","215","161","130","84","102","63","55","60","36","42","49"],"header":"unpooled","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"psuedo","name":"psuedo","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["942031","21269","1248","4155","17948","309","265","238","250","245","253","275","229","235","210","213","225","205","169"],"header":"psuedo","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"pooled","name":"pooled","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["906077","13206","2207","7522","2430","1283","890","797","752","694","711","705","717","733","715","726","750","730","573"],"header":"pooled","align":"center","footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,19],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"574a6ceef2b12b399c47c69ab625d6c5"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="cell">

</div>
<p><small><strong>(16S rRNA) Table 5 | Total number of reads and ASVs by sample for pooling options.</strong></small></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,U2FtcGxlX0lELHVucG9vbGVkX3JlYWRzLHBzdWVkb19yZWFkcyxwb29sZWRfcmVhZHMsdW5wb29sZWRfYXN2LHBzdWVkb19hc3YscG9vbGVkX2FzdgpQMDFfRDAwXzAxMF9XM0EsNjgwMzcsNjg1MjksNjU1MjEsMjM4MywzMzg1LDY1NDIKUDAxX0QwMF8wMTBfVzhBLDYwNTM1LDYwNjk5LDU3NjM2LDE2MTgsMjQzNCw0OTI3ClAwMl9EMDBfMDEwX0MwQSw3MDQzNiw3MDQzNiw2NzIwOCwyMjc4LDMzMDksNjgwOQpQMDNfRDAwXzAxMF9XM0IsNzUxODMsNzU0NzAsNzE5NTQsMjM2OCwzMjc3LDYzMjQKUDAzX0QwMF8wMTBfVzhCLDUzNDM3LDUzMTM0LDUxODkzLDE1OTMsMjI0Niw0NjU4ClAwNF9EMDBfMDEwX0MwQiw1MzgxMyw1NDExNSw1MjEzMywyMDUyLDMwODYsNjY2OQpQMDVfRDAwXzAxMF9XM0MsNDcwNzAsNDczMjgsNDUwMTIsMTY3MiwyNTY1LDUyNzUKUDA1X0QwMF8wMTBfVzhDLDg2ODQxLDg3NDE4LDg1MjQ1LDIxNTYsMjk0Niw1NjA4ClAwNl9EMDBfMDEwX0MwQyw4MTIxOSw4MTQxNiw3ODc1OSwzMDgyLDQxNTUsNzUyMgpQMDdfRDAwXzAxMF9XM0QsNjE3NDIsNjIwOTEsNTg3NDAsMjAyNSwzMDQ5LDYyNTYKUDA3X0QwMF8wMTBfVzhELDYyMjAyLDYxOTE3LDU4MzAwLDIxNjcsMjk5OCw1NTkyClAwOF9EMDBfMDEwX0MwRCw2MjE3MSw2MjUyNyw2MDI0NSwyNDUzLDM0NDYsNjc5MQpQMDlfRDAwXzAxMF9XM0UsNzYyOTksNzY2NzEsNzI4MzMsMjQwMiwzNDEyLDY3NzkKUDA5X0QwMF8wMTBfVzhFLDI1MTUxLDI2NDUzLDI4NTY2LDgxNiwxMjQ4LDIyMDcKUDEwX0QwMF8wMTBfQzBFLDUzNjI1LDUzODI3LDUyMDMyLDIwNzAsMzA1NSw2MzYxCg==').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'ssu18_pooling_results.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="cell-output-display">

<div id="htmlwidget-eed293ac19a22cf1972a" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-eed293ac19a22cf1972a">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Sample_ID":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"unpooled_reads":[68037,60535,70436,75183,53437,53813,47070,86841,81219,61742,62202,62171,76299,25151,53625],"psuedo_reads":[68529,60699,70436,75470,53134,54115,47328,87418,81416,62091,61917,62527,76671,26453,53827],"pooled_reads":[65521,57636,67208,71954,51893,52133,45012,85245,78759,58740,58300,60245,72833,28566,52032],"unpooled_asv":[2383,1618,2278,2368,1593,2052,1672,2156,3082,2025,2167,2453,2402,816,2070],"psuedo_asv":[3385,2434,3309,3277,2246,3086,2565,2946,4155,3049,2998,3446,3412,1248,3055],"pooled_asv":[6542,4927,6809,6324,4658,6669,5275,5608,7522,6256,5592,6791,6779,2207,6361]},"columns":[{"accessor":"Sample_ID","name":"Sample ID","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"header":"Sample ID","minWidth":150,"align":"left","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"unpooled_reads","name":"unpooled_reads","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["68037","60535","70436","75183","53437","53813","47070","86841","81219","61742","62202","62171","76299","25151","53625"],"header":"unpooled reads","minWidth":150,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"psuedo_reads","name":"psuedo_reads","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["68529","60699","70436","75470","53134","54115","47328","87418","81416","62091","61917","62527","76671","26453","53827"],"header":"psuedo reads","minWidth":150,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"pooled_reads","name":"pooled_reads","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["65521","57636","67208","71954","51893","52133","45012","85245","78759","58740","58300","60245","72833","28566","52032"],"header":"pooled reads","minWidth":150,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"unpooled_asv","name":"unpooled_asv","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["2383","1618","2278","2368","1593","2052","1672","2156","3082","2025","2167","2453","2402","816","2070"],"header":"unpooled asv","minWidth":150,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"psuedo_asv","name":"psuedo_asv","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["3385","2434","3309","3277","2246","3086","2565","2946","4155","3049","2998","3446","3412","1248","3055"],"header":"psuedo asv","minWidth":150,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"pooled_asv","name":"pooled_asv","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["6542","4927","6809","6324","4658","6669","5275","5608","7522","6256","5592","6791","6779","2207","6361"],"header":"pooled asv","minWidth":150,"align":"center","footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"d9032bb36d5f991e71c433729341e8e8"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</details><div class="cell">
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">dadaFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/dada.html">dada</a></span><span class="op">(</span><span class="va">derepFs</span>, err <span class="op">=</span> <span class="va">errF</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span>, pool <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<details><summary>
Detailed results of <code>dada</code> on forward reads
</summary><pre><code>
Sample 1 - 79804 reads in 39950 unique sequences.
Sample 2 - 72760 reads in 31163 unique sequences.
Sample 3 - 62598 reads in 35274 unique sequences.
Sample 4 - 81240 reads in 42119 unique sequences.
Sample 5 - 86732 reads in 40336 unique sequences.
Sample 6 - 60860 reads in 31104 unique sequences.
Sample 7 - 63037 reads in 35169 unique sequences.
Sample 8 - 54127 reads in 27752 unique sequences.
Sample 9 - 109421 reads in 49045 unique sequences.
Sample 10 - 94740 reads in 46853 unique sequences.
Sample 11 - 71678 reads in 35363 unique sequences.
Sample 12 - 73639 reads in 35567 unique sequences.
Sample 13 - 72263 reads in 38518 unique sequences.
Sample 14 - 89315 reads in 43426 unique sequences.
Sample 15 - 33000 reads in 13673 unique sequences.
</code></pre>
</details><p>As an example, we can inspect the returned <code>dada-class</code> object for the forward and reverse reads from the sample #2:</p>
<div class="cell">
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">dadaFs</span><span class="op">[[</span><span class="va">sampnum</span><span class="op">]</span><span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>dada-class: object describing DADA2 denoising results
1759 sequence variants were inferred from 31163 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
<p>This output tells us how many true sequence variants the DADA2 algorithm inferred from the unique sequences, in this case the sample 2.</p>
</section><section id="construct-sequence-table" class="level2"><h2 class="anchored" data-anchor-id="construct-sequence-table">8. Construct Sequence Table</h2>
<p>Now we construct an amplicon sequence variant (ASV) table.</p>
<div class="cell">
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">seqtab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/makeSequenceTable.html">makeSequenceTable</a></span><span class="op">(</span><span class="va">dadaFs</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">seqtab</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1]    15 21685</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/dada2/man/getSequences.html">getSequences</a></span><span class="op">(</span><span class="va">seqtab</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>   22    23    24    25    27    41    66    73   120   149   185   190   193 
    2     1     1     2     1     1     1     1     1     1     1     9     1 
  198   200   201   202   208   209   211   215   217   220   221   222   223 
    1     1     1     1     1     1     1     1     2     4    14    21     8 
  224   225   227   228   229   230   231   232   233   234   238   251 
    3     4     6     4     5     9    20 21342   190    12     2     8 </code></pre>
<p>The sequence table is a matrix with rows corresponding to (and named by) the samples, and columns corresponding to (and named by) the sequence variants. We have 21685 sequence variants but also a range of sequence lengths. Since many of these sequence variants are singletons or doubletons, we can just select a range that corresponds to the expected amplicon length and eliminate the spurious reads.</p>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="include/dada2/ssu18_read_length_before.png" class="img-fluid" width="1050"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 4 | Distribution of read length by total ASVs before removing extreme length variants.</strong></small></p>
<div class="cell">
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">seqtab.2</span> <span class="op">&lt;-</span> <span class="va">seqtab</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">seqtab</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">230</span>,<span class="fl">235</span><span class="op">)</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">seqtab.2</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout-note callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The values you select should be based on the sequence table generated above.</p>
</div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1]    15 21573</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/dada2/man/getSequences.html">getSequences</a></span><span class="op">(</span><span class="va">seqtab.2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>  230   231   232   233   234 
    9    20 21342   190    12 </code></pre>
<p>After removing the extreme length variants, we have 21573, a reduction of 112 sequence variants.</p>
<div class="cell">

</div>
</section><section id="remove-chimeras" class="level2"><h2 class="anchored" data-anchor-id="remove-chimeras">9. Remove Chimeras</h2>
<p>Even though the <code>dada</code> method corrects substitution and indel errors, chimeric sequences remain. According to the DADA2 documentation, the accuracy of sequence variants after denoising makes identifying chimeric ASVs simpler than when dealing with fuzzy OTUs. Chimeric sequences are identified if they can be <em>exactly reconstructed</em> by combining a left-segment and a right-segment from two more abundant <em>parent</em> sequences.</p>
<div class="cell">
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">seqtab.nochim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/removeBimeraDenovo.html">removeBimeraDenovo</a></span><span class="op">(</span><span class="va">seqtab.2</span>,
                                    method <span class="op">=</span> <span class="st">"consensus"</span>,
                                    multithread <span class="op">=</span> <span class="fl">20</span>,
                                    verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">seqtab.nochim</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>Identified 1241 bimeras out of 21573 input sequences.</code></pre>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1]    15 20332</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">seqtab.nochim</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">seqtab.2</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.9596595</code></pre>
</div>
</div>
<p>Chimera checking removed an additional 1241 sequence variants however, when we account for the abundances of each variant, we see chimeras accounts for about 4.03405% of the merged sequence reads. Not bad.</p>
<p>The last thing we want to do is write the sequence table to an <code>RDS</code> file.</p>
<div class="cell">
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">seqtab.nochim</span>, <span class="st">"ssu18_seqtab.nochim.rds"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="track-reads-through-workflow" class="level2"><h2 class="anchored" data-anchor-id="track-reads-through-workflow">10. Track Reads through Workflow</h2>
<p>At this point we can look at the number of reads that made it through each step of the workflow for every sample.</p>
<div class="cell">
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">getN</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/dada2/man/getUniques.html">getUniques</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span>
<span class="va">track</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html">cbind</a></span><span class="op">(</span><span class="va">out</span>,
               <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">sapply</a></span><span class="op">(</span><span class="va">dadaFs</span>, <span class="va">getN</span><span class="op">)</span>,
               <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">seqtab.nochim</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">track</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"input"</span>, <span class="st">"filtered"</span>, <span class="st">"denoisedF"</span>, <span class="st">"nonchim"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">track</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample.names</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">

</div>
<p><small><strong>(16S rRNA) Table 6 | Tracking reads changes at each step of the DADA2 workflow.</strong></small></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,U2FtcGxlX0lELHJhdyxwcmVfZmlsdCxjdXQsZmlsdGVyZWQsZGVub2lzZWRGLG5vbmNoaW0KUDAxX0QwMF8wMTBfVzNBLDg1NDg5LDg1NDgyLDg1NDAyLDc5ODA0LDcwNDM0LDY4MDM3ClAwMV9EMDBfMDEwX1c4QSw3NzY0NSw3NzY0Miw3NzQ3MCw3Mjc2MCw2NTk1MCw2MDUzNQpQMDJfRDAwXzAxMF9DMEEsODcwOTEsODcwODUsODY4NzgsODEyNDAsNzE5NjEsNzA0MzYKUDAzX0QwMF8wMTBfVzNCLDkzMDY1LDkzMDUzLDkyODUxLDg2NzMyLDc4MDY3LDc1MTgzClAwM19EMDBfMDEwX1c4Qiw2NjQwMSw2NjM2Nyw2NTQ5OCw2MDg2MCw1NDgyMiw1MzQzNwpQMDRfRDAwXzAxMF9DMEIsNjc4MTAsNjc4MDAsNjc2MTcsNjMwMzcsNTQ2NzgsNTM4MTMKUDA1X0QwMF8wMTBfVzNDLDU4MjY2LDU4MjU5LDU4MDA2LDU0MTI3LDQ3ODY4LDQ3MDcwClAwNV9EMDBfMDEwX1c4QywxMjA4OTIsMTIwODQyLDExNzAxNSwxMDk0MjEsMTAwMDI2LDg2ODQxClAwNl9EMDBfMDEwX0MwQywxMDE1OTksMTAxNTkyLDEwMTM2Myw5NDc0MCw4MzE5MCw4MTIxOQpQMDdfRDAwXzAxMF9XM0QsNzY3ODEsNzY3NzYsNzY1ODMsNzE2NzgsNjQwMzEsNjE3NDIKUDA3X0QwMF8wMTBfVzhELDc5MDMyLDc5MDIyLDc4NzI0LDczNjM5LDY1MTI1LDYyMjAyClAwOF9EMDBfMDEwX0MwRCw3NzU5OSw3NzU5MSw3NzEzMyw3MjI2Myw2MzI0OCw2MjE3MQpQMDlfRDAwXzAxMF9XM0UsOTU1NjYsOTU1NjQsOTUzMTAsODkzMTUsNzk3OTksNzYyOTkKUDA5X0QwMF8wMTBfVzhFLDk1MTgyLDk0NDU3LDQ0MjUwLDMzMDAwLDMxNjIwLDI1MTUxClAxMF9EMDBfMDEwX0MwRSw2NzU4Myw2NzU3OSw2NzQzMiw2MjU5OCw1NDA3NSw1MzYyNQo=').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'ssu18_track_read_changes.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="cell-output-display">

<div id="htmlwidget-f6d5de2b538c2ee39d90" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f6d5de2b538c2ee39d90">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Sample_ID":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"raw":[85489,77645,87091,93065,66401,67810,58266,120892,101599,76781,79032,77599,95566,95182,67583],"pre_filt":[85482,77642,87085,93053,66367,67800,58259,120842,101592,76776,79022,77591,95564,94457,67579],"cut":[85402,77470,86878,92851,65498,67617,58006,117015,101363,76583,78724,77133,95310,44250,67432],"filtered":[79804,72760,81240,86732,60860,63037,54127,109421,94740,71678,73639,72263,89315,33000,62598],"denoisedF":[70434,65950,71961,78067,54822,54678,47868,100026,83190,64031,65125,63248,79799,31620,54075],"nonchim":[68037,60535,70436,75183,53437,53813,47070,86841,81219,61742,62202,62171,76299,25151,53625]},"columns":[{"accessor":"Sample_ID","name":"Sample ID","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"header":"Sample ID","align":"left","footerStyle":{"fontWeight":"bold"},"footer":"Total reads","minWidth":150,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"raw","name":"raw","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["85489","77645","87091","93065","66401","67810","58266","120892","101599","76781","79032","77599","95566","95182","67583"],"header":"raw","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"1250001"},{"accessor":"pre_filt","name":"pre_filt","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["85482","77642","87085","93053","66367","67800","58259","120842","101592","76776","79022","77591","95564","94457","67579"],"header":"pre filt","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"1249111"},{"accessor":"cut","name":"cut","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["85402","77470","86878","92851","65498","67617","58006","117015","101363","76583","78724","77133","95310","44250","67432"],"header":"cut","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"1191532"},{"accessor":"filtered","name":"filtered","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["79804","72760","81240","86732","60860","63037","54127","109421","94740","71678","73639","72263","89315","33000","62598"],"header":"filtered","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"1105214"},{"accessor":"denoisedF","name":"denoisedF","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["70434","65950","71961","78067","54822","54678","47868","100026","83190","64031","65125","63248","79799","31620","54075"],"header":"denoisedF","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"984894"},{"accessor":"nonchim","name":"nonchim","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["68037","60535","70436","75183","53437","53813","47070","86841","81219","61742","62202","62171","76299","25151","53625"],"header":"nonchim","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"937761"}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"1acd74285d20b349eeba325d60426cec"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>And we can save a copy of the table for later use.</p>
<div class="cell">
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="va">track</span>, <span class="st">"ssu18_read_changes.txt"</span>,
            sep <span class="op">=</span> <span class="st">"\t"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, col.names<span class="op">=</span><span class="cn">NA</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">

</div>
</section><section id="assign-taxonomy" class="level2"><h2 class="anchored" data-anchor-id="assign-taxonomy">11. Assign Taxonomy</h2>
<p>The <code>assignTaxonomy</code> command implements the naive Bayesian classifier, so for reproducible results you need to set a random number seed (see issue <a href="https://github.com/benjjneb/dada2/issues/538">#538</a>). We did this at the beginning of the workflow. For taxonomic assignment, we are using the Silva version 138 <span class="citation" data-cites="quast2012silva">(<a href="#ref-quast2012silva" role="doc-biblioref">Quast et al. 2012</a>)</span>. The developers of DADA2 maintain a <a href="https://benjjneb.github.io/dada2/training.html">formatted version of the database</a>.</p>
<p>We will read in the <code>RDS</code> file containing the sequence table saved above. We also need to run <code>remove(list = ls())</code> command, otherwise the final image we save will be huge. This way, the image only contains the sample data, seqtab, and taxtab <em>after</em> running <code>removeBimeraDenovo</code>.</p>
<div class="cell">
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/rm.html">remove</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="va">seqtab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"ssu18_seqtab.nochim.rds"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>And then native Bayesian <span class="citation" data-cites="wang2007naive">(<a href="#ref-wang2007naive" role="doc-biblioref">Wang et al. 2007</a>)</span> classifier against the Silva database.</p>
<div class="cell">
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tax_silva</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/assignTaxonomy.html">assignTaxonomy</a></span><span class="op">(</span>
  <span class="va">seqtab</span>, <span class="st">"silva_nr_v138_train_set.fa.gz"</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="save-image" class="level2"><h2 class="anchored" data-anchor-id="save-image">12. Save Image</h2>
<p>And finally, we save an image for use in the analytic part of the workflow. This R data file will be needed as the input for the phyloseq portion of the workflow. See the <a href="data-availability.html">Data Availability</a> page for complete details on where to get this file.</p>
<div class="cell">
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/save.html">save.image</a></span><span class="op">(</span><span class="st">"ssu18_dada2_wf.rdata"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">

</div>
</section><section id="r-session-information-code" class="level2"><h2 class="anchored" data-anchor-id="r-session-information-code">R Session Information &amp; Code</h2>
<p>This workflow was run on the <a href="https://doi.org/10.25572/SIHPC">Smithsonian High Performance Cluster (SI/HPC)</a>, Smithsonian Institution. Below are the specific packages and versions used in this workflow using both <code><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo()</a></code> and <code><a href="https://r-lib.github.io/sessioninfo/reference/session_info.html">devtools::session_info()</a></code>. Click the arrow to see the details.</p>
<details><summary>
Show/hide R Session Info
</summary><pre><code>sessionInfo()
R version 4.0.0 (2020-04-24)
Platform: x86_64-conda_cos6-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /home/scottjj/miniconda3/envs/R-4/lib/libopenblasp-r0.3.9.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
 [1] grid      stats4    parallel  stats     graphics  grDevices utils
 [8] datasets  methods   base

other attached packages:
 [1] gridExtra_2.3               ggplot2_3.3.0
 [3] ShortRead_1.46.0            GenomicAlignments_1.24.0
 [5] SummarizedExperiment_1.18.1 DelayedArray_0.14.0
 [7] matrixStats_0.56.0          Biobase_2.48.0
 [9] Rsamtools_2.4.0             GenomicRanges_1.40.0
[11] GenomeInfoDb_1.24.0         Biostrings_2.56.0
[13] XVector_0.28.0              IRanges_2.22.1
[15] S4Vectors_0.26.0            BiocParallel_1.22.0
[17] BiocGenerics_0.34.0         dada2_1.16.0
[19] Rcpp_1.0.4.6

loaded via a namespace (and not attached):
 [1] plyr_1.8.6             RColorBrewer_1.1-2     pillar_1.4.4
 [4] compiler_4.0.0         bitops_1.0-6           tools_4.0.0
 [7] zlibbioc_1.34.0        lifecycle_0.2.0        tibble_3.0.1
[10] gtable_0.3.0           lattice_0.20-41        png_0.1-7
[13] pkgconfig_2.0.3        rlang_0.4.6            Matrix_1.2-18
[16] GenomeInfoDbData_1.2.3 withr_2.2.0            stringr_1.4.0
[19] hwriter_1.3.2          vctrs_0.3.0            glue_1.4.1
[22] R6_2.4.1               jpeg_0.1-8.1           latticeExtra_0.6-29
[25] reshape2_1.4.4         magrittr_1.5           scales_1.1.1
[28] ellipsis_0.3.1         colorspace_1.4-1       stringi_1.4.6
[31] RCurl_1.98-1.2         RcppParallel_5.0.1     munsell_0.5.0
[34] crayon_1.3.4

devtools::session_info()
â”€ Session info â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 setting  value
 version  R version 4.0.0 (2020-04-24)
 os       CentOS Linux 7 (Core)
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       America/New_York
 date     2020-06-17

â”€ Packages â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 package              * version  date       lib source
 assertthat             0.2.1    2019-03-21 [1] CRAN (R 4.0.0)
 backports              1.1.7    2020-05-13 [1] CRAN (R 4.0.0)
 Biobase              * 2.48.0   2020-04-27 [1] Bioconductor
 BiocGenerics         * 0.34.0   2020-04-27 [1] Bioconductor
 BiocParallel         * 1.22.0   2020-04-27 [1] Bioconductor
 Biostrings           * 2.56.0   2020-04-27 [1] Bioconductor
 bitops                 1.0-6    2013-08-17 [1] CRAN (R 4.0.0)
 callr                  3.4.3    2020-03-28 [1] CRAN (R 4.0.0)
 cli                    2.0.2    2020-02-28 [1] CRAN (R 4.0.0)
 colorspace             1.4-1    2019-03-18 [1] CRAN (R 4.0.0)
 crayon                 1.3.4    2017-09-16 [1] CRAN (R 4.0.0)
 dada2                * 1.16.0   2020-04-27 [1] Bioconductor
 DelayedArray         * 0.14.0   2020-04-27 [1] Bioconductor
 desc                   1.2.0    2018-05-01 [1] CRAN (R 4.0.0)
 devtools               2.3.0    2020-04-10 [1] CRAN (R 4.0.0)
 digest                 0.6.25   2020-02-23 [1] CRAN (R 4.0.0)
 ellipsis               0.3.1    2020-05-15 [1] CRAN (R 4.0.0)
 fansi                  0.4.1    2020-01-08 [1] CRAN (R 4.0.0)
 fs                     1.4.1    2020-04-04 [1] CRAN (R 4.0.0)
 GenomeInfoDb         * 1.24.0   2020-04-27 [1] Bioconductor
 GenomeInfoDbData       1.2.3    2020-05-22 [1] Bioconductor
 GenomicAlignments    * 1.24.0   2020-04-27 [1] Bioconductor
 GenomicRanges        * 1.40.0   2020-04-27 [1] Bioconductor
 ggplot2              * 3.3.0    2020-03-05 [1] CRAN (R 4.0.0)
 glue                   1.4.1    2020-05-13 [1] CRAN (R 4.0.0)
 gridExtra            * 2.3      2017-09-09 [1] CRAN (R 4.0.0)
 gtable                 0.3.0    2019-03-25 [1] CRAN (R 4.0.0)
 hwriter                1.3.2    2014-09-10 [1] CRAN (R 4.0.0)
 IRanges              * 2.22.1   2020-04-28 [1] Bioconductor
 jpeg                   0.1-8.1  2019-10-24 [1] CRAN (R 4.0.0)
 lattice                0.20-41  2020-04-02 [1] CRAN (R 4.0.0)
 latticeExtra           0.6-29   2019-12-19 [1] CRAN (R 4.0.0)
 lifecycle              0.2.0    2020-03-06 [1] CRAN (R 4.0.0)
 magrittr               1.5      2014-11-22 [1] CRAN (R 4.0.0)
 Matrix                 1.2-18   2019-11-27 [1] CRAN (R 4.0.0)
 matrixStats          * 0.56.0   2020-03-13 [1] CRAN (R 4.0.0)
 memoise                1.1.0    2017-04-21 [1] CRAN (R 4.0.0)
 munsell                0.5.0    2018-06-12 [1] CRAN (R 4.0.0)
 pillar                 1.4.4    2020-05-05 [1] CRAN (R 4.0.0)
 pkgbuild               1.0.8    2020-05-07 [1] CRAN (R 4.0.0)
 pkgconfig              2.0.3    2019-09-22 [1] CRAN (R 4.0.0)
 pkgload                1.0.2    2018-10-29 [1] CRAN (R 4.0.0)
 plyr                   1.8.6    2020-03-03 [1] CRAN (R 4.0.0)
 png                    0.1-7    2013-12-03 [1] CRAN (R 4.0.0)
 prettyunits            1.1.1    2020-01-24 [1] CRAN (R 4.0.0)
 processx               3.4.2    2020-02-09 [1] CRAN (R 4.0.0)
 ps                     1.3.3    2020-05-08 [1] CRAN (R 4.0.0)
 R6                     2.4.1    2019-11-12 [1] CRAN (R 4.0.0)
 RColorBrewer           1.1-2    2014-12-07 [1] CRAN (R 4.0.0)
 Rcpp                 * 1.0.4.6  2020-04-09 [1] CRAN (R 4.0.0)
 RcppParallel           5.0.1    2020-05-06 [1] CRAN (R 4.0.0)
 RCurl                  1.98-1.2 2020-04-18 [1] CRAN (R 4.0.0)
 remotes                2.1.1    2020-02-15 [1] CRAN (R 4.0.0)
 reshape2               1.4.4    2020-04-09 [1] CRAN (R 4.0.0)
 rlang                  0.4.6    2020-05-02 [1] CRAN (R 4.0.0)
 rprojroot              1.3-2    2018-01-03 [1] CRAN (R 4.0.0)
 Rsamtools            * 2.4.0    2020-04-27 [1] Bioconductor
 S4Vectors            * 0.26.0   2020-04-27 [1] Bioconductor
 scales                 1.1.1    2020-05-11 [1] CRAN (R 4.0.0)
 sessioninfo            1.1.1    2018-11-05 [1] CRAN (R 4.0.0)
 ShortRead            * 1.46.0   2020-04-27 [1] Bioconductor
 stringi                1.4.6    2020-02-17 [1] CRAN (R 4.0.0)
 stringr                1.4.0    2019-02-10 [1] CRAN (R 4.0.0)
 SummarizedExperiment * 1.18.1   2020-04-30 [1] Bioconductor
 testthat               2.3.2    2020-03-02 [1] CRAN (R 4.0.0)
 tibble                 3.0.1    2020-04-20 [1] CRAN (R 4.0.0)
 usethis                1.6.1    2020-04-29 [1] CRAN (R 4.0.0)
 vctrs                  0.3.0    2020-05-11 [1] CRAN (R 4.0.0)
 withr                  2.2.0    2020-04-20 [1] CRAN (R 4.0.0)
 XVector              * 0.28.0   2020-04-27 [1] Bioconductor
 zlibbioc               1.34.0   2020-04-27 [1] Bioconductor</code></pre>
</details><p><br></p>
</section></section><section id="its-data" class="level1"><h1>ITS Data</h1>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<section id="workflow-input-1" class="level2"><h2 class="anchored" data-anchor-id="workflow-input-1">Workflow Input</h2>
<p>All files needed to run this workflow can be downloaded from figshare.</p>
<iframe src="https://widgets.figshare.com/articles/14686755/embed?show_title=1" width="568" height="351" allowfullscreen="" frameborder="0">
</iframe>
</section><section id="overview-1" class="level2"><h2 class="anchored" data-anchor-id="overview-1">Overview</h2>
<section id="sequence-files-samples-1" class="level3"><h3 class="anchored" data-anchor-id="sequence-files-samples-1">Sequence Files &amp; Samples</h3>
<p>We sequenced a total of 15 samples collected from 10 different plots at a single depth in each plot.</p>
<div class="cell">

</div>
<p><small><strong>(ITS) Table 1 | Sample data &amp; associated sequencing information.</strong></small></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,U2FtcGxlX0lELFBsb3QsVHJlYXRtZW50LFRlbXAsRGVwdGgsUGFpcixyYXdfcmVhZHMsRmFzdHFfSURfRm9yd2FyZCxGYXN0cV9JRF9SZXZlcnNlLEk3X0luZGV4X0lELGluZGV4LEk1X0luZGV4X0lELGluZGV4MgpQMDFfRDAwXzAxMF9XM0EsUDAxLFdhcm0sMywwMF8wMTAsQSw1MDg5NCxQMV9TOF9MMDAxX1IxXzAwMV9JVFMuZmFzdHEuZ3osUDFfUzhfTDAwMV9SMl8wMDFfSVRTLmZhc3RxLmd6LFNDNzAxUixjYWd0YWdndCxTQzUwOCxUQUdUR1RBRwpQMDFfRDAwXzAxMF9XOEEsUDAxLFdhcm0sOCwwMF8wMTAsQSw3MDczNCxQMS04Q19TMTRfTDAwMV9SMV8wMDFfSVRTLmZhc3RxLmd6LFAxLThDX1MxNF9MMDAxX1IyXzAwMV9JVFMuZmFzdHEuZ3osU0M3MDJSLGF0YWdjZ2N0LFNDNTA2LEdBQ0FDVEdBClAwMl9EMDBfMDEwX0MwQSxQMDIsQ29udHJvbCwwLDAwXzAxMCxBLDc0NTgxLFAyX1M5X0wwMDFfUjFfMDAxX0lUUy5mYXN0cS5neixQMl9TOV9MMDAxX1IyXzAwMV9JVFMuZmFzdHEuZ3osU0M3MDJSLGF0YWdjZ2N0LFNDNTAxLEFDR0FDR1RHClAwM19EMDBfMDEwX1czQixQMDMsV2FybSwzLDAwXzAxMCxCLDYwMDA2LFAzX1M2X0wwMDFfUjFfMDAxX0lUUy5mYXN0cS5neixQM19TNl9MMDAxX1IyXzAwMV9JVFMuZmFzdHEuZ3osU0M3MDFSLGNhZ3RhZ2d0LFNDNTA2LEdBQ0FDVEdBClAwM19EMDBfMDEwX1c4QixQMDMsV2FybSw4LDAwXzAxMCxCLDMxMTc0LFAzLThDX1MxMV9MMDAxX1IxXzAwMV9JVFMuZmFzdHEuZ3osUDMtOENfUzExX0wwMDFfUjJfMDAxX0lUUy5mYXN0cS5neixTQzcwMlIsYXRhZ2NnY3QsU0M1MDMsQ0dUQ0dDVEEKUDA0X0QwMF8wMTBfQzBCLFAwNCxDb250cm9sLDAsMDBfMDEwLEIsODg3MzIsUDRfUzdfTDAwMV9SMV8wMDFfSVRTLmZhc3RxLmd6LFA0X1M3X0wwMDFfUjJfMDAxX0lUUy5mYXN0cS5neixTQzcwMVIsY2FndGFnZ3QsU0M1MDcsVEdDR1RBQ0cKUDA1X0QwMF8wMTBfVzNDLFAwNSxXYXJtLDMsMDBfMDEwLEMsODA5LFA1X1MxX0wwMDFfUjFfMDAxX0lUUy5mYXN0cS5neixQNV9TMV9MMDAxX1IyXzAwMV9JVFMuZmFzdHEuZ3osU0M3MDFSLGNhZ3RhZ2d0LFNDNTAxLEFDR0FDR1RHClAwNV9EMDBfMDEwX1c4QyxQMDUsV2FybSw4LDAwXzAxMCxDLDgxOTA4LFA1LThDX1MyNl9MMDAxX1IxXzAwMV9JVFMuZmFzdHEuZ3osUDUtOENfUzI2X0wwMDFfUjJfMDAxX0lUUy5mYXN0cS5neixTQzcwNFIsdGNjdGNhdGcsU0M1MDIsQVRBVEFDQUMKUDA2X0QwMF8wMTBfQzBDLFAwNixDb250cm9sLDAsMDBfMDEwLEMsNjc0MTksUDZfUzJfTDAwMV9SMV8wMDFfSVRTLmZhc3RxLmd6LFA2X1MyX0wwMDFfUjJfMDAxX0lUUy5mYXN0cS5neixTQzcwMVIsY2FndGFnZ3QsU0M1MDIsQVRBVEFDQUMKUDA3X0QwMF8wMTBfVzNELFAwNyxXYXJtLDMsMDBfMDEwLEQsNzAxMjMsUDdfUzNfTDAwMV9SMV8wMDFfSVRTLmZhc3RxLmd6LFA3X1MzX0wwMDFfUjJfMDAxX0lUUy5mYXN0cS5neixTQzcwMVIsY2FndGFnZ3QsU0M1MDMsQ0dUQ0dDVEEKUDA3X0QwMF8wMTBfVzhELFAwNyxXYXJtLDgsMDBfMDEwLEQsMTU0MzUsUDctOENfUzY0X0wwMDFfUjFfMDAxX0lUUy5mYXN0cS5neixQNy04Q19TNjRfTDAwMV9SMl8wMDFfSVRTLmZhc3RxLmd6LFNDNzA4UixhZ2NhdGFjYyxTQzUwOCxUQUdUR1RBRwpQMDhfRDAwXzAxMF9DMEQsUDA4LENvbnRyb2wsMCwwMF8wMTAsRCw2MDc5OSxQOF9TNF9MMDAxX1IxXzAwMV9JVFMuZmFzdHEuZ3osUDhfUzRfTDAwMV9SMl8wMDFfSVRTLmZhc3RxLmd6LFNDNzAxUixjYWd0YWdndCxTQzUwNCxDVEFHQUdDVApQMDlfRDAwXzAxMF9XM0UsUDA5LFdhcm0sMywwMF8wMTAsRSw1NTkyNyxQOV9TNV9MMDAxX1IxXzAwMV9JVFMuZmFzdHEuZ3osUDlfUzVfTDAwMV9SMl8wMDFfSVRTLmZhc3RxLmd6LFNDNzAxUixjYWd0YWdndCxTQzUwNSxHQ1RDVEFHVApQMDlfRDAwXzAxMF9XOEUsUDA5LFdhcm0sOCwwMF8wMTAsRSwxNDE4LFA5LThDX1MxMF9MMDAxX1IxXzAwMV9JVFMuZmFzdHEuZ3osUDktOENfUzEwX0wwMDFfUjJfMDAxX0lUUy5mYXN0cS5neixTQzcwMlIsYXRhZ2NnY3QsU0M1MDIsQVRBVEFDQUMKUDEwX0QwMF8wMTBfQzBFLFAxMCxDb250cm9sLDAsMDBfMDEwLEUsOTY1MzEsUDEwX1MxNV9MMDAxX1IxXzAwMV9JVFMuZmFzdHEuZ3osUDEwX1MxNV9MMDAxX1IyXzAwMV9JVFMuZmFzdHEuZ3osU0M3MDJSLGF0YWdjZ2N0LFNDNTA3LFRHQ0dUQUNHCg==').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'its18_sample_seq_info.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="cell-output-display">

<div id="htmlwidget-bb983d864ca15ad3e051" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-bb983d864ca15ad3e051">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Sample_ID":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"Plot":["P01","P01","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09","P10"],"Treatment":["Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control"],"Temp":[3,8,0,3,8,0,3,8,0,3,8,0,3,8,0],"Depth":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"Pair":["A","A","A","B","B","B","C","C","C","D","D","D","E","E","E"],"raw_reads":[50894,70734,74581,60006,31174,88732,809,81908,67419,70123,15435,60799,55927,1418,96531],"Fastq_ID_Forward":["P1_S8_L001_R1_001_ITS.fastq.gz","P1-8C_S14_L001_R1_001_ITS.fastq.gz","P2_S9_L001_R1_001_ITS.fastq.gz","P3_S6_L001_R1_001_ITS.fastq.gz","P3-8C_S11_L001_R1_001_ITS.fastq.gz","P4_S7_L001_R1_001_ITS.fastq.gz","P5_S1_L001_R1_001_ITS.fastq.gz","P5-8C_S26_L001_R1_001_ITS.fastq.gz","P6_S2_L001_R1_001_ITS.fastq.gz","P7_S3_L001_R1_001_ITS.fastq.gz","P7-8C_S64_L001_R1_001_ITS.fastq.gz","P8_S4_L001_R1_001_ITS.fastq.gz","P9_S5_L001_R1_001_ITS.fastq.gz","P9-8C_S10_L001_R1_001_ITS.fastq.gz","P10_S15_L001_R1_001_ITS.fastq.gz"],"Fastq_ID_Reverse":["P1_S8_L001_R2_001_ITS.fastq.gz","P1-8C_S14_L001_R2_001_ITS.fastq.gz","P2_S9_L001_R2_001_ITS.fastq.gz","P3_S6_L001_R2_001_ITS.fastq.gz","P3-8C_S11_L001_R2_001_ITS.fastq.gz","P4_S7_L001_R2_001_ITS.fastq.gz","P5_S1_L001_R2_001_ITS.fastq.gz","P5-8C_S26_L001_R2_001_ITS.fastq.gz","P6_S2_L001_R2_001_ITS.fastq.gz","P7_S3_L001_R2_001_ITS.fastq.gz","P7-8C_S64_L001_R2_001_ITS.fastq.gz","P8_S4_L001_R2_001_ITS.fastq.gz","P9_S5_L001_R2_001_ITS.fastq.gz","P9-8C_S10_L001_R2_001_ITS.fastq.gz","P10_S15_L001_R2_001_ITS.fastq.gz"],"I7_Index_ID":["SC701R","SC702R","SC702R","SC701R","SC702R","SC701R","SC701R","SC704R","SC701R","SC701R","SC708R","SC701R","SC701R","SC702R","SC702R"],"index":["cagtaggt","atagcgct","atagcgct","cagtaggt","atagcgct","cagtaggt","cagtaggt","tcctcatg","cagtaggt","cagtaggt","agcatacc","cagtaggt","cagtaggt","atagcgct","atagcgct"],"I5_Index_ID":["SC508","SC506","SC501","SC506","SC503","SC507","SC501","SC502","SC502","SC503","SC508","SC504","SC505","SC502","SC507"],"index2":["TAGTGTAG","GACACTGA","ACGACGTG","GACACTGA","CGTCGCTA","TGCGTACG","ACGACGTG","ATATACAC","ATATACAC","CGTCGCTA","TAGTGTAG","CTAGAGCT","GCTCTAGT","ATATACAC","TGCGTACG"]},"columns":[{"accessor":"Sample_ID","name":"Sample ID","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"header":"Sample ID","align":"left","footerStyle":{"fontWeight":"bold"},"footer":"Total reads","minWidth":150,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"Plot","name":"Plot","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01","P01","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09","P10"],"header":"Plot","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Treatment","name":"Treatment","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control"],"header":"Treatment","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Temp","name":"Temp","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["3","8","0","3","8","0","3","8","0","3","8","0","3","8","0"],"header":"Temp","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Depth","name":"Depth","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"header":"Depth","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Pair","name":"Pair","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["A","A","A","B","B","B","C","C","C","D","D","D","E","E","E"],"header":"Pair","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"raw_reads","name":"raw_reads","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["50894","70734","74581","60006","31174","88732","809","81908","67419","70123","15435","60799","55927","1418","96531"],"header":"raw reads","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"826490"},{"accessor":"Fastq_ID_Forward","name":"Fastq_ID_Forward","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P1_S8_L001_R1_001_ITS.fastq.gz","P1-8C_S14_L001_R1_001_ITS.fastq.gz","P2_S9_L001_R1_001_ITS.fastq.gz","P3_S6_L001_R1_001_ITS.fastq.gz","P3-8C_S11_L001_R1_001_ITS.fastq.gz","P4_S7_L001_R1_001_ITS.fastq.gz","P5_S1_L001_R1_001_ITS.fastq.gz","P5-8C_S26_L001_R1_001_ITS.fastq.gz","P6_S2_L001_R1_001_ITS.fastq.gz","P7_S3_L001_R1_001_ITS.fastq.gz","P7-8C_S64_L001_R1_001_ITS.fastq.gz","P8_S4_L001_R1_001_ITS.fastq.gz","P9_S5_L001_R1_001_ITS.fastq.gz","P9-8C_S10_L001_R1_001_ITS.fastq.gz","P10_S15_L001_R1_001_ITS.fastq.gz"],"header":"Fastq ID Forward","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Fastq_ID_Reverse","name":"Fastq_ID_Reverse","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P1_S8_L001_R2_001_ITS.fastq.gz","P1-8C_S14_L001_R2_001_ITS.fastq.gz","P2_S9_L001_R2_001_ITS.fastq.gz","P3_S6_L001_R2_001_ITS.fastq.gz","P3-8C_S11_L001_R2_001_ITS.fastq.gz","P4_S7_L001_R2_001_ITS.fastq.gz","P5_S1_L001_R2_001_ITS.fastq.gz","P5-8C_S26_L001_R2_001_ITS.fastq.gz","P6_S2_L001_R2_001_ITS.fastq.gz","P7_S3_L001_R2_001_ITS.fastq.gz","P7-8C_S64_L001_R2_001_ITS.fastq.gz","P8_S4_L001_R2_001_ITS.fastq.gz","P9_S5_L001_R2_001_ITS.fastq.gz","P9-8C_S10_L001_R2_001_ITS.fastq.gz","P10_S15_L001_R2_001_ITS.fastq.gz"],"header":"Fastq ID Reverse","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"I7_Index_ID","name":"I7_Index_ID","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["SC701R","SC702R","SC702R","SC701R","SC702R","SC701R","SC701R","SC704R","SC701R","SC701R","SC708R","SC701R","SC701R","SC702R","SC702R"],"header":"I7 Index ID","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"index","name":"index","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["cagtaggt","atagcgct","atagcgct","cagtaggt","atagcgct","cagtaggt","cagtaggt","tcctcatg","cagtaggt","cagtaggt","agcatacc","cagtaggt","cagtaggt","atagcgct","atagcgct"],"header":"index","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"I5_Index_ID","name":"I5_Index_ID","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["SC508","SC506","SC501","SC506","SC503","SC507","SC501","SC502","SC502","SC503","SC508","SC504","SC505","SC502","SC507"],"header":"I5 Index ID","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"index2","name":"index2","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["TAGTGTAG","GACACTGA","ACGACGTG","GACACTGA","CGTCGCTA","TGCGTACG","ACGACGTG","ATATACAC","ATATACAC","CGTCGCTA","TAGTGTAG","CTAGAGCT","GCTCTAGT","ATATACAC","TGCGTACG"],"header":"index2","align":"center","footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"692464a2864e6551a524023856fa71f1"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section><section id="workflow-1" class="level3"><h3 class="anchored" data-anchor-id="workflow-1">Workflow</h3>
<p>Our workflow is pretty much ripped from the <a href="https://benjjneb.github.io/dada2/ITS_workflow.html">DADA2 ITS Workflow (1.8)</a> on the DADA2 website. That webpage contains thorough explanations for each step so we will not repeat most of that here. For more details, check out the post. The workflow consists of the following steps:</p>
<table class="table">
<colgroup>
<col style="width: 6%">
<col style="width: 26%">
<col style="width: 66%">
</colgroup>
<thead><tr class="header">
<th>Step</th>
<th>Command</th>
<th>What weâ€™re doing</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>multiple</td>
<td>prepare input file names &amp; paths</td>
</tr>
<tr class="even">
<td>2</td>
<td>multiple</td>
<td>Define primers (all orientations)</td>
</tr>
<tr class="odd">
<td>3</td>
<td><code>cutadapt</code></td>
<td>Remove primers</td>
</tr>
<tr class="even">
<td>4</td>
<td><code><a href="https://rdrr.io/pkg/dada2/man/filterAndTrim.html">filterAndTrim()</a></code></td>
<td>assess quality &amp; filter reads</td>
</tr>
<tr class="odd">
<td>5</td>
<td><code><a href="https://rdrr.io/pkg/dada2/man/learnErrors.html">learnErrors()</a></code></td>
<td>generate an error model for the data</td>
</tr>
<tr class="even">
<td>6</td>
<td><code>derepFastq</code></td>
<td>dereplicate sequences</td>
</tr>
<tr class="odd">
<td>7</td>
<td><code><a href="https://rdrr.io/pkg/dada2/man/dada.html">dada()</a></code></td>
<td>infer ASVs on both forward &amp; reverse reads independently</td>
</tr>
<tr class="even">
<td>8</td>
<td><code><a href="https://rdrr.io/pkg/dada2/man/mergePairs.html">mergePairs()</a></code></td>
<td>merge forward &amp; reverse reads to further refine ASVs</td>
</tr>
<tr class="odd">
<td>9</td>
<td><code><a href="https://rdrr.io/pkg/dada2/man/makeSequenceTable.html">makeSequenceTable()</a></code></td>
<td>generate a count table</td>
</tr>
<tr class="even">
<td>10</td>
<td><code><a href="https://rdrr.io/pkg/dada2/man/removeBimeraDenovo.html">removeBimeraDenovo()</a></code></td>
<td>screen for &amp; remove chimeras</td>
</tr>
<tr class="odd">
<td>11</td>
<td></td>
<td>track reads through workflow</td>
</tr>
<tr class="even">
<td>12</td>
<td><code><a href="https://rdrr.io/pkg/dada2/man/assignTaxonomy.html">assignTaxonomy()</a></code></td>
<td>assign taxonomy &amp; finish workflow</td>
</tr>
<tr class="odd">
<td>13</td>
<td><code><a href="https://rdrr.io/r/base/save.html">save.image()</a></code></td>
<td>save an image of seqtab &amp; taxtab for next part of workflow</td>
</tr>
</tbody>
</table></section></section><section id="set-working-environment-1" class="level2"><h2 class="anchored" data-anchor-id="set-working-environment-1">1. Set Working Environment</h2>
<p>Next, we need to setup the working environment by renaming the fastq files and defining a path for the working directory.</p>
<section id="rename-samples-1" class="level4"><h4 class="anchored" data-anchor-id="rename-samples-1">Rename samples</h4>
<p>To make the parsing easier, we will eliminate the lane and well number from each sample. If you do not wish to do that you will need to adjust the code accordingly. I am <em>positive</em> there are more elegant ways of doing this.</p>
<blockquote class="blockquote">
<p>CAUTION. You should check that this code works on a few backup files before proceeding.</p>
</blockquote>
<div class="cell">
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span><span class="op">(</span><span class="st">"RAW"</span><span class="op">)</span>
<span class="va">orig_fastq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"*.fastq.gz"</span><span class="op">)</span>
<span class="va">newname_fastq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_S.*_L001"</span>, <span class="st">""</span>, <span class="va">orig_fastq</span><span class="op">)</span>
<span class="va">newname_fastq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_001"</span>, <span class="st">""</span>, <span class="va">newname_fastq</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.rename</a></span><span class="op">(</span><span class="va">orig_fastq</span>, <span class="va">newname_fastq</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span><span class="op">(</span><span class="st">"../"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">path</span> <span class="op">&lt;-</span> <span class="st">"RAW"</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "P1_R1.fastq.gz"    "P1_R2.fastq.gz"    "P1-8C_R1.fastq.gz"
[4] "P1-8C_R2.fastq.gz" "P10_R1.fastq.gz"   "P10_R2.fastq.gz"  </code></pre>
</div>
</div>
<p>Then, we generate matched lists of the forward and reverse read files. We also parse out the sample name.</p>
<div class="cell">
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">fnFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span>, pattern <span class="op">=</span> <span class="st">"_R1.fastq.gz"</span>, full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
<span class="va">fnRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span>, pattern <span class="op">=</span> <span class="st">"_R2.fastq.gz"</span>, full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section></section><section id="define-primers-1" class="level2"><h2 class="anchored" data-anchor-id="define-primers-1">2. Define Primers</h2>
<p>Before we start the DADA2 workflow we need to run <a href="https://github.com/marcelm/cutadapt">catadapt</a> <span class="citation" data-cites="martin2011cutadapt">(<a href="#ref-martin2011cutadapt" role="doc-biblioref">Martin 2011</a>)</span> on all <code>fastq.gz</code> files to trim the primers. For this part of the study we used the primer pair ITS1f (CTTGGTCATTTAGAGGAAGTAA) <span class="citation" data-cites="gardes1993its">(<a href="#ref-gardes1993its" role="doc-biblioref">Gardes and Bruns 1993</a>)</span> and ITS2 (GCTGCGTTCTTCATCGATGC) <span class="citation" data-cites="white1990amplification">(<a href="#ref-white1990amplification" role="doc-biblioref">White et al. 1990</a>)</span> which should yield variable amplicon lengths between 100 to 400 bp. What we are looking for is potential read-through scenarios that are possible when sequencing the length variable ITS region as described in the <a href="https://benjjneb.github.io/dada2/ITS_workflow.html">DADA2 ITS Workflow (1.8)</a>. Please refer to this document for a complete explanation.</p>
<p>First we define the primers.</p>
<div class="cell">
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">FWD</span> <span class="op">&lt;-</span> <span class="st">"CTTGGTCATTTAGAGGAAGTAA"</span>
<span class="va">REV</span> <span class="op">&lt;-</span> <span class="st">"GCTGCGTTCTTCATCGATGC"</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Next, we check the presence and orientation of these primers in the data. To do this we will create all orientations of the input primer sequences. In other words the <em>Forward</em>, <em>Complement</em>, <em>Reverse</em>, and <em>Reverse Complement</em> variations.</p>
<div class="cell">
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">allOrients</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">primer</span><span class="op">)</span> <span class="op">{</span>
    <span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings">Biostrings</a></span><span class="op">)</span>
    <span class="va">dna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/DNAString-class.html">DNAString</a></span><span class="op">(</span><span class="va">primer</span><span class="op">)</span>
    <span class="va">orients</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span>Forward <span class="op">=</span> <span class="va">dna</span>,
                 Complement <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html">complement</a></span><span class="op">(</span><span class="va">dna</span><span class="op">)</span>,
                 Reverse <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/reverse-methods.html">reverse</a></span><span class="op">(</span><span class="va">dna</span><span class="op">)</span>,
                 RevComp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html">reverseComplement</a></span><span class="op">(</span><span class="va">dna</span><span class="op">)</span><span class="op">)</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">sapply</a></span><span class="op">(</span><span class="va">orients</span>, <span class="va">toString</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="va">FWD.orients</span> <span class="op">&lt;-</span> <span class="fu">allOrients</span><span class="op">(</span><span class="va">FWD</span><span class="op">)</span>
<span class="va">REV.orients</span> <span class="op">&lt;-</span> <span class="fu">allOrients</span><span class="op">(</span><span class="va">REV</span><span class="op">)</span>
<span class="va">FWD.orients</span>
<span class="va">REV.orients</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>                 Forward               Complement                  Reverse
"CTTGGTCATTTAGAGGAAGTAA" "GAACCAGTAAATCTCCTTCATT" "AATGAAGGAGATTTACTGGTTC"
                 RevComp
"TTACTTCCTCTAAATGACCAAG"
               Forward             Complement                Reverse
"GCTGCGTTCTTCATCGATGC" "CGACGCAAGAAGTAGCTACG" "CGTAGCTACTTCTTGCGTCG"
               RevComp
"GCATCGATGAAGAACGCAGC"</code></pre>
<p>Now we do a little pre-filter step to eliminate ambiguous bases (<em>Ns</em>) because <em>Ns</em> make mapping of short primer sequences difficult. This step removes any reads with <em>Ns</em>. Again, set some files paths, this time for the filtered reads.</p>
<div class="cell">
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">fnFs.filtN</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"filtN"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html">basename</a></span><span class="op">(</span><span class="va">fnFs</span><span class="op">)</span><span class="op">)</span>
<span class="va">fnRs.filtN</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"filtN"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html">basename</a></span><span class="op">(</span><span class="va">fnRs</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/dada2/man/filterAndTrim.html">filterAndTrim</a></span><span class="op">(</span><span class="va">fnFs</span>, <span class="va">fnFs.filtN</span>, <span class="va">fnRs</span>, <span class="va">fnRs.filtN</span>,
              maxN <span class="op">=</span> <span class="fl">0</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Sweet. Time to assess the number of times a primer (and all possible primer orientation) appear in the forward and reverse reads. According to the workflow, counting the primers on one set of paired end fastq files is sufficient to see if there is a problem. This assumes that all the files were created using the same library prep. Basically for both primers, we will search for all four orientations in both forward and reverse reads</p>
<div class="cell">
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">sampnum</span> <span class="op">&lt;-</span> <span class="fl">1</span>
<span class="va">primerHits</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">primer</span>, <span class="va">fn</span><span class="op">)</span> <span class="op">{</span>
    <span class="co"># Counts number of reads in which the primer is found</span>
    <span class="va">nhits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/matchPattern.html">vcountPattern</a></span><span class="op">(</span><span class="va">primer</span>, <span class="fu"><a href="https://rdrr.io/pkg/ShortRead/man/ShortRead-class.html">sread</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ShortRead/man/readFastq.html">readFastq</a></span><span class="op">(</span><span class="va">fn</span><span class="op">)</span><span class="op">)</span>, fixed <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">nhits</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="forward-primers-1" class="level4"><h4 class="anchored" data-anchor-id="forward-primers-1">Forward primers</h4>
<div class="cell">
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html">rbind</a></span><span class="op">(</span>FWD.ForwardReads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">sapply</a></span><span class="op">(</span><span class="va">FWD.orients</span>, <span class="va">primerHits</span>,
                                fn <span class="op">=</span> <span class="va">fnFs.filtN</span><span class="op">[[</span><span class="va">sampnum</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
    FWD.ReverseReads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">sapply</a></span><span class="op">(</span><span class="va">FWD.orients</span>, <span class="va">primerHits</span>,
                              fn <span class="op">=</span> <span class="va">fnRs.filtN</span><span class="op">[[</span><span class="va">sampnum</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>                 Forward Complement Reverse RevComp
FWD.ForwardReads   48467          0       0       0
FWD.ReverseReads       0          0       0    5078</code></pre>
</section><section id="reverse-primers-1" class="level4"><h4 class="anchored" data-anchor-id="reverse-primers-1">Reverse primers</h4>
<div class="cell">
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html">rbind</a></span><span class="op">(</span>REV.ForwardReads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">sapply</a></span><span class="op">(</span><span class="va">REV.orients</span>, <span class="va">primerHits</span>,
                                fn <span class="op">=</span> <span class="va">fnFs.filtN</span><span class="op">[[</span><span class="va">sampnum</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
       REV.ReverseReads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">sapply</a></span><span class="op">(</span><span class="va">REV.orients</span>, <span class="va">primerHits</span>,
                                 fn <span class="op">=</span> <span class="va">fnRs.filtN</span><span class="op">[[</span><span class="va">sampnum</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>                 Forward Complement Reverse RevComp
REV.ForwardReads       0          0       0    6237
REV.ReverseReads   43902          0       0       0</code></pre>
<div class="cell">

</div>
<p>What does this table mean? I wondered the same thing. Letâ€™s break it down. Sample P1 had 50,894 sequences after the original filtering described earlier. The code searched the forward and reverse fastq files for all 8 primers. If we look at the two outputs, we see the forward primer is found in the forward reads in its forward orientation but also in some reverse reads in its reverse-complement orientation. The reverse primer is found in the reverse reads in its forward orientation but also in some forward reads in its reverse-complement orientation. This is due to read-through when the ITS region is short.</p>
</section></section><section id="remove-primers-1" class="level2"><h2 class="anchored" data-anchor-id="remove-primers-1">3. Remove Primers</h2>
<p>Now we can run <a href="https://github.com/marcelm/cutadapt">catadapt</a> <span class="citation" data-cites="martin2011cutadapt">(<a href="#ref-martin2011cutadapt" role="doc-biblioref">Martin 2011</a>)</span> to remove the primers from the fastq sequences. A little setup first. If this command executes successfully it means R has found cutadapt.</p>
<div class="cell">
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">cutadapt</span> <span class="op">&lt;-</span> <span class="st">"/PATH/to/cutadapt"</span>
<span class="fu"><a href="https://rdrr.io/r/base/system2.html">system2</a></span><span class="op">(</span><span class="va">cutadapt</span>, args <span class="op">=</span> <span class="st">"--version"</span><span class="op">)</span> <span class="co"># Run shell commands from R</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>2.8</code></pre>
<p>We set paths and trim the forward primer and the reverse-complement of the reverse primer off of R1 (forward reads) and trim the reverse primer and the reverse-complement of the forward primer off of R2 (reverse reads).</p>
<div class="cell">
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">path.cut</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"cutadapt"</span><span class="op">)</span>
<span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files2.html">dir.exists</a></span><span class="op">(</span><span class="va">path.cut</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/files2.html">dir.create</a></span><span class="op">(</span><span class="va">path.cut</span><span class="op">)</span>
<span class="va">fnFs.cut</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path.cut</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html">basename</a></span><span class="op">(</span><span class="va">fnFs</span><span class="op">)</span><span class="op">)</span>
<span class="va">fnRs.cut</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path.cut</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html">basename</a></span><span class="op">(</span><span class="va">fnRs</span><span class="op">)</span><span class="op">)</span>

<span class="va">FWD.RC</span> <span class="op">&lt;-</span> <span class="fu">dada2</span><span class="fu">:::</span><span class="fu"><a href="https://rdrr.io/pkg/dada2/man/rc.html">rc</a></span><span class="op">(</span><span class="va">FWD</span><span class="op">)</span>
<span class="va">REV.RC</span> <span class="op">&lt;-</span> <span class="fu">dada2</span><span class="fu">:::</span><span class="fu"><a href="https://rdrr.io/pkg/dada2/man/rc.html">rc</a></span><span class="op">(</span><span class="va">REV</span><span class="op">)</span>

<span class="va">R1.flags</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="st">"-g"</span>, <span class="va">FWD</span>, <span class="st">"-a"</span>, <span class="va">REV.RC</span><span class="op">)</span>
<span class="va">R2.flags</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="st">"-G"</span>, <span class="va">REV</span>, <span class="st">"-A"</span>, <span class="va">FWD.RC</span><span class="op">)</span>

<span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_along</a></span><span class="op">(</span><span class="va">fnFs</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/system2.html">system2</a></span><span class="op">(</span><span class="va">cutadapt</span>,
                                   args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="va">R1.flags</span>, <span class="va">R2.flags</span>,
                                            <span class="st">"-m"</span>, <span class="fl">20</span>, <span class="st">"-n"</span>, <span class="fl">2</span>,
                                            <span class="st">"-o"</span>, <span class="va">fnFs.cut</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>,
                                            <span class="st">"-p"</span>, <span class="va">fnRs.cut</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>,
                                            <span class="va">fnFs.filtN</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, <span class="va">fnRs.filtN</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>This is cutadapt 2.8 with Python 3.7.6
Command line parameters: -g CTTGGTCATTTAGAGGAAGTAA -a GCATCGATGAAGAACGCAGC \
                         -G GCTGCGTTCTTCATCGATGC -A TTACTTCCTCTAAATGACCAAG \
                         -m 20 -n 2 \
                         -o RAW/cutadapt/P1_R1.fastq.gz \
                         -p RAW/cutadapt/P1_R2.fastq.gz RAW/filtN/P1_R1.fastq.gz RAW/filtN/P1_R2.fastq.gz</code></pre>
<p><em>Note</em>. If the code above removes all of the base pairs in a sequence, you will get downstream errors unless you set the <code>-m</code> flag. This flag sets the minimum length and reads shorter than this will be discarded. Without this flag, reads of length 0 will be kept and cause issues. <em>Also, a lot of output will be written to the screen by cutadapt!</em></p>
<p>We can now count the number of primers in the sequences from the output of cutadapt.</p>
<div class="cell">
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html">rbind</a></span><span class="op">(</span>
  FWD.ForwardReads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">sapply</a></span><span class="op">(</span><span class="va">FWD.orients</span>, <span class="va">primerHits</span>, fn <span class="op">=</span> <span class="va">fnFs.cut</span><span class="op">[[</span><span class="va">sampnum</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
  FWD.ReverseReads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">sapply</a></span><span class="op">(</span><span class="va">FWD.orients</span>, <span class="va">primerHits</span>, fn <span class="op">=</span> <span class="va">fnRs.cut</span><span class="op">[[</span><span class="va">sampnum</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
  REV.ForwardReads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">sapply</a></span><span class="op">(</span><span class="va">REV.orients</span>, <span class="va">primerHits</span>, fn <span class="op">=</span> <span class="va">fnFs.cut</span><span class="op">[[</span><span class="va">sampnum</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
  REV.ReverseReads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">sapply</a></span><span class="op">(</span><span class="va">REV.orients</span>, <span class="va">primerHits</span>, fn <span class="op">=</span> <span class="va">fnRs.cut</span><span class="op">[[</span><span class="va">sampnum</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>                 Forward Complement Reverse RevComp
FWD.ForwardReads       0          0       0       0
FWD.ReverseReads       0          0       0       0
REV.ForwardReads       0          0       0       0
REV.ReverseReads       0          0       0       0</code></pre>
<p>Basically, primers are no longer detected in the cutadapted reads. Now, for each sample, we can take a look at how the pre-filtering step and primer removal affected the total number of raw reads.</p>
<div class="cell">

</div>
<div class="cell">

</div>
<p><small><strong>(ITS) Table 2 | Total reads per sample after prefiltering and primer removal (using cutadapt).</strong></small></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,U2FtcGxlX0lELHJhd19yZWFkcyxwcmVfZmlsdF9yZWFkcyxjdXRhZGFwdF9yZWFkcyxwZXJfcmVhZHNfa2VwdApQMDFfRDAwXzAxMF9XM0EsNTA4OTQsNTA4ODIsNTA0MTksMC45OTEKUDAxX0QwMF8wMTBfVzhBLDcwNzM0LDcwNjQ5LDY4MDc1LDAuOTYyClAwMl9EMDBfMDEwX0MwQSw3NDU4MSw3NDU1Nyw3Mzk5NCwwLjk5MgpQMDNfRDAwXzAxMF9XM0IsNjAwMDYsNTk5NTMsNTgxNDYsMC45NjkKUDAzX0QwMF8wMTBfVzhCLDMxMTc0LDMxMDI5LDI1NzgyLDAuODI3ClAwNF9EMDBfMDEwX0MwQiw4ODczMiw4ODY5NCw4NzUzMiwwLjk4NgpQMDVfRDAwXzAxMF9XM0MsODA5LDc5Nyw0MTIsMC41MDkKUDA1X0QwMF8wMTBfVzhDLDgxOTA4LDgxNjcxLDc0NjY5LDAuOTEyClAwNl9EMDBfMDEwX0MwQyw2NzQxOSw2NzM5OSw2NjY1NiwwLjk4OQpQMDdfRDAwXzAxMF9XM0QsNzAxMjMsNzAwNzMsNjg0NzQsMC45NzYKUDA3X0QwMF8wMTBfVzhELDE1NDM1LDE1NDE3LDE0OTU1LDAuOTY5ClAwOF9EMDBfMDEwX0MwRCw2MDc5OSw2MDYwNCw1NDUzNCwwLjg5NwpQMDlfRDAwXzAxMF9XM0UsNTU5MjcsNTU3NjQsNTEzODIsMC45MTkKUDA5X0QwMF8wMTBfVzhFLDE0MTgsMTM4MywxNDEsMC4wOTkKUDEwX0QwMF8wMTBfQzBFLDk2NTMxLDk2NDkyLDk1MjU2LDAuOTg3Cg==').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'its18_cutadapt.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="cell-output-display">

<div id="htmlwidget-08d6a2e44bdb2d67d0ae" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-08d6a2e44bdb2d67d0ae">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Sample_ID":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"raw_reads":[50894,70734,74581,60006,31174,88732,809,81908,67419,70123,15435,60799,55927,1418,96531],"pre_filt_reads":[50882,70649,74557,59953,31029,88694,797,81671,67399,70073,15417,60604,55764,1383,96492],"cutadapt_reads":[50419,68075,73994,58146,25782,87532,412,74669,66656,68474,14955,54534,51382,141,95256],"per_reads_kept":[0.991,0.962,0.992,0.969,0.827,0.986,0.509,0.912,0.989,0.976,0.969,0.897,0.919,0.099,0.987]},"columns":[{"accessor":"Sample_ID","name":"Sample ID","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"header":"Sample ID","align":"left","footerStyle":{"fontWeight":"bold"},"footer":"Total reads","minWidth":150,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"raw_reads","name":"raw_reads","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["50894","70734","74581","60006","31174","88732","809","81908","67419","70123","15435","60799","55927","1418","96531"],"header":"raw reads","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"826490"},{"accessor":"pre_filt_reads","name":"pre_filt_reads","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["50882","70649","74557","59953","31029","88694","797","81671","67399","70073","15417","60604","55764","1383","96492"],"header":"pre filt reads","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"825364"},{"accessor":"cutadapt_reads","name":"cutadapt_reads","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["50419","68075","73994","58146","25782","87532","412","74669","66656","68474","14955","54534","51382","141","95256"],"header":"cutadapt reads","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"790427"},{"accessor":"per_reads_kept","name":"per_reads_kept","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["0.991","0.962","0.992","0.969","0.827","0.986","0.509","0.912","0.989","0.976","0.969","0.897","0.919","0.099","0.987"],"header":"per reads kept","align":"center","footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"0da35be3575d2a3e53c55f657133b005"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section><section id="quality-assessment-filtering-1" class="level2"><h2 class="anchored" data-anchor-id="quality-assessment-filtering-1">4. Quality Assessment &amp; Filtering</h2>
<p>We need the forward and reverse fastq file names and the sample names.</p>
<div class="cell">
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">cutFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path.cut</span>, pattern <span class="op">=</span> <span class="st">"_R1.fastq.gz"</span>,
                         full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
<span class="va">cutRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path.cut</span>, pattern <span class="op">=</span> <span class="st">"_R2.fastq.gz"</span>,
                         full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>

<span class="va">get.sample.name</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">fname</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html">basename</a></span><span class="op">(</span><span class="va">fname</span><span class="op">)</span>, <span class="st">"_"</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>
<span class="va">sample.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unname.html">unname</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">sapply</a></span><span class="op">(</span><span class="va">cutFs</span>, <span class="va">get.sample.name</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">head</a></span><span class="op">(</span><span class="va">sample.names</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "P1"    "P1-8C" "P10"   "P2"    "P3"    "P3-8C"</code></pre>
</div>
</div>
<section id="quality-plots-1" class="level3"><h3 class="anchored" data-anchor-id="quality-plots-1">Quality plots</h3>
<p>Then we inspect read quality.</p>
<div class="cell">
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/plotQualityProfile.html">plotQualityProfile</a></span><span class="op">(</span><span class="va">cutFs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, aggregate <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/plotQualityProfile.html">plotQualityProfile</a></span><span class="op">(</span><span class="va">cutRs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, aggregate <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu">grid.arrange</span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"plot_qscores_2018.png"</span>, <span class="va">p3</span>, width <span class="op">=</span> <span class="fl">7</span>, height <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="include/dada2/its18_plot_qscores.png" class="img-fluid" width="1050"></p>
</div>
</div>
<p><small><strong>(ITS) Figure 1 | Aggregated quality score plots for forward (left) &amp; reverse (right) reads before filtering.</strong></small></p>
</section><section id="filtering-1" class="level3"><h3 class="anchored" data-anchor-id="filtering-1">Filtering</h3>
<div class="cell">
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">filtFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path.cut</span>, <span class="st">"filtered"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html">basename</a></span><span class="op">(</span><span class="va">cutFs</span><span class="op">)</span><span class="op">)</span>
<span class="va">filtRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path.cut</span>, <span class="st">"filtered"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html">basename</a></span><span class="op">(</span><span class="va">cutRs</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We use a <code>minLen</code> value to get rid of very short length sequences.</p>
<div class="cell">
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/filterAndTrim.html">filterAndTrim</a></span><span class="op">(</span><span class="va">cutFs</span>, <span class="va">filtFs</span>, <span class="va">cutRs</span>, <span class="va">filtRs</span>,
                     maxN <span class="op">=</span> <span class="fl">0</span>, maxEE <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span>, truncQ <span class="op">=</span> <span class="fl">2</span>,
                     minLen <span class="op">=</span> <span class="fl">50</span>, rm.phix <span class="op">=</span> <span class="cn">TRUE</span>,
                     compress <span class="op">=</span> <span class="cn">TRUE</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">out</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout-note callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>These parameters should be set based on the anticipated length of the amplicon and the read quality.</p>
</div>
</div>
<p>And here is a table of how the filtering step affected the number of reads in each sample.</p>
<div class="cell">

</div>
<p><small><strong>(ITS) Table 3 | Total reads per sample after filtering.</strong></small></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,U2FtcGxlX0lELHJlYWRzX2luLHJlYWRzX291dCxwZXJfcmVhZHNfa2VwdApQMDFfRDAwXzAxMF9XM0EsNTA0MTksMzU5OTksMC43MTQKUDAxX0QwMF8wMTBfVzhBLDY4MDc1LDQ4MjMwLDAuNzA4ClAwMl9EMDBfMDEwX0MwQSw3Mzk5NCw1ODk1NiwwLjc5NwpQMDNfRDAwXzAxMF9XM0IsNTgxNDYsNDI1NjMsMC43MzIKUDAzX0QwMF8wMTBfVzhCLDI1NzgyLDE3NTQ4LDAuNjgxClAwNF9EMDBfMDEwX0MwQiw4NzUzMiw2ODI1OSwwLjc4ClAwNV9EMDBfMDEwX1czQyw0MTIsMzA4LDAuNzQ4ClAwNV9EMDBfMDEwX1c4Qyw3NDY2OSw0ODUxOCwwLjY1ClAwNl9EMDBfMDEwX0MwQyw2NjY1Niw1MTA1NSwwLjc2NgpQMDdfRDAwXzAxMF9XM0QsNjg0NzQsNDM0NzEsMC42MzUKUDA3X0QwMF8wMTBfVzhELDE0OTU1LDk0ODgsMC42MzQKUDA4X0QwMF8wMTBfQzBELDU0NTM0LDQxNTYzLDAuNzYyClAwOV9EMDBfMDEwX1czRSw1MTM4MiwzNTcyMiwwLjY5NQpQMDlfRDAwXzAxMF9XOEUsMTQxLDg1LDAuNjAzClAxMF9EMDBfMDEwX0MwRSw5NTI1Niw4MDg3NywwLjg0OQo=').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'its18_filter.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="cell-output-display">

<div id="htmlwidget-1c0439fe13d4aafd3c73" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-1c0439fe13d4aafd3c73">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Sample_ID":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"reads_in":[50419,68075,73994,58146,25782,87532,412,74669,66656,68474,14955,54534,51382,141,95256],"reads_out":[35999,48230,58956,42563,17548,68259,308,48518,51055,43471,9488,41563,35722,85,80877],"per_reads_kept":[0.714,0.708,0.797,0.732,0.681,0.78,0.748,0.65,0.766,0.635,0.634,0.762,0.695,0.603,0.849]},"columns":[{"accessor":"Sample_ID","name":"Sample ID","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"header":"Sample ID","align":"left","footerStyle":{"fontWeight":"bold"},"footer":"Total reads","minWidth":150,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"reads_in","name":"reads_in","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["50419","68075","73994","58146","25782","87532","412","74669","66656","68474","14955","54534","51382","141","95256"],"header":"reads in","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"790427"},{"accessor":"reads_out","name":"reads_out","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["35999","48230","58956","42563","17548","68259","308","48518","51055","43471","9488","41563","35722","85","80877"],"header":"reads out","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"582642"},{"accessor":"per_reads_kept","name":"per_reads_kept","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["0.714","0.708","0.797","0.732","0.681","0.78","0.748","0.65","0.766","0.635","0.634","0.762","0.695","0.603","0.849"],"header":"per reads kept","align":"center","footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"2f16225aef9ee575b20e864ad14abadb"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section></section><section id="learn-error-rates-1" class="level2"><h2 class="anchored" data-anchor-id="learn-error-rates-1">5. Learn Error Rates</h2>
<p>Time to assess the error rate of the data. The rest of the workflow is very similar to the 16S workflows presented previously. So I will basically stop talking.</p>
<section id="forward-reads-2" class="level4"><h4 class="anchored" data-anchor-id="forward-reads-2">Forward Reads</h4>
<div class="cell">
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">errF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/learnErrors.html">learnErrors</a></span><span class="op">(</span><span class="va">filtFs</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>108788735 total bases in 495784 reads from 11 samples will be used for learning the error rates.
109358094 total bases in 495784 reads from 11 samples will be used for learning the error rates.</code></pre>
<div class="cell">
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/dada2/man/plotErrors.html">plotErrors</a></span><span class="op">(</span><span class="va">errF</span>, nominalQ <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="include/dada2/its18_plot_errorF.png" class="img-fluid" width="1050"></p>
</div>
</div>
<p><small><strong>(ITS) Figure 3 | Forward reads: Observed frequency of each transition (e.g., T -&gt; G) as a function of the associated quality score.</strong></small></p>
</section><section id="reverse-reads" class="level4"><h4 class="anchored" data-anchor-id="reverse-reads">Reverse Reads</h4>
<div class="cell">
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">errR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/learnErrors.html">learnErrors</a></span><span class="op">(</span><span class="va">filtRs</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>109358094 total bases in 495784 reads from 11 samples will be used for learning the error rates.</code></pre>
<div class="cell">
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/dada2/man/plotErrors.html">plotErrors</a></span><span class="op">(</span><span class="va">errR</span>, nominalQ <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="include/dada2/its18_plot_errorR.png" class="img-fluid" width="1050"></p>
</div>
</div>
<p><small><strong>(ITS) Figure 4 | Reverse reads: Observed frequency of each transition (e.g., T -&gt; G) as a function of the associated quality score.</strong></small></p>
</section></section><section id="dereplicate-reads-1" class="level2"><h2 class="anchored" data-anchor-id="dereplicate-reads-1">6. Dereplicate Reads</h2>
<section id="forward-reads-3" class="level4"><h4 class="anchored" data-anchor-id="forward-reads-3">Forward Reads</h4>
<div class="cell">
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">derepFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/derepFastq.html">derepFastq</a></span><span class="op">(</span><span class="va">filtFs</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">derepFs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample.names</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br></p>
<details><summary>
Detailed results of derep forward reads
</summary><pre><code>
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P1_R1.fastq.gz
Encountered 7277 unique sequences from 35999 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P1-8C_R1.fastq.gz
Encountered 11366 unique sequences from 48230 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P10_R1.fastq.gz
Encountered 11714 unique sequences from 80877 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P2_R1.fastq.gz
Encountered 13088 unique sequences from 58956 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P3_R1.fastq.gz
Encountered 10180 unique sequences from 42563 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P3-8C_R1.fastq.gz
Encountered 4480 unique sequences from 17548 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P4_R1.fastq.gz
Encountered 12259 unique sequences from 68259 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P5_R1.fastq.gz
Encountered 160 unique sequences from 308 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P5-8C_R1.fastq.gz
Encountered 10436 unique sequences from 48518 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P6_R1.fastq.gz
Encountered 11230 unique sequences from 51055 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P7_R1.fastq.gz
Encountered 10316 unique sequences from 43471 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P7-8C_R1.fastq.gz
Encountered 2121 unique sequences from 9488 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P8_R1.fastq.gz
Encountered 9236 unique sequences from 41563 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P9_R1.fastq.gz
Encountered 9000 unique sequences from 35722 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P9-8C_R1.fastq.gz
Encountered 42 unique sequences from 85 total sequences read.
</code></pre>
</details><p><br></p>
</section><section id="reverse-reads-1" class="level4"><h4 class="anchored" data-anchor-id="reverse-reads-1">Reverse Reads</h4>
<div class="cell">
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">derepRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/derepFastq.html">derepFastq</a></span><span class="op">(</span><span class="va">filtRs</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">derepRs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample.names</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br></p>
<details><summary>
Detailed results of derep reverse reads
</summary><pre><code>
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P1_R2.fastq.gz
Encountered 13081 unique sequences from 35999 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P1-8C_R2.fastq.gz
Encountered 17965 unique sequences from 48230 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P10_R2.fastq.gz
Encountered 18610 unique sequences from 80877 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P2_R2.fastq.gz
Encountered 19600 unique sequences from 58956 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P3_R2.fastq.gz
Encountered 16145 unique sequences from 42563 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P3-8C_R2.fastq.gz
Encountered 7496 unique sequences from 17548 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P4_R2.fastq.gz
Encountered 19933 unique sequences from 68259 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P5_R2.fastq.gz
Encountered 199 unique sequences from 308 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P5-8C_R2.fastq.gz
Encountered 18495 unique sequences from 48518 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P6_R2.fastq.gz
Encountered 17216 unique sequences from 51055 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P7_R2.fastq.gz
Encountered 17111 unique sequences from 43471 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P7-8C_R2.fastq.gz
Encountered 3809 unique sequences from 9488 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P8_R2.fastq.gz
Encountered 14140 unique sequences from 41563 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P9_R2.fastq.gz
Encountered 14856 unique sequences from 35722 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P9-8C_R2.fastq.gz
Encountered 45 unique sequences from 85 total sequences read.
</code></pre>
</details><p><br></p>
</section></section><section id="dada2-asv-inference-1" class="level2"><h2 class="anchored" data-anchor-id="dada2-asv-inference-1">7. DADA2 &amp; ASV Inference</h2>
<p>At this point we are ready to apply the core sample inference algorithm (dada) to the filtered and trimmed sequence data. DADA2 offers three options for whether and how to pool samples for ASV inference.</p>
<p>If <code>pool = TRUE</code>, the algorithm will pool together all samples prior to sample inference.<br>
If <code>pool = FALSE</code>, sample inference is performed on each sample individually.<br>
If <code>pool = "pseudo"</code>, the algorithm will perform pseudo-pooling between individually processed samples.</p>
<p>We tested all three methods through the full pipeline. Click the <code>+</code> to see the results of the test. For our final analysis, we chose <code>pool = TRUE</code> for this data set.</p>
<details><summary>
Show/hide Results of <code>dada</code> pooling options
</summary><p>Here are a few summary tables of results from the tests. Values are from the final sequence table prodcuced by each option.</p>
<div class="cell">

</div>
<p><small><strong>(ITS) Table 4 | Total number of reads, total number of ASVs, minimum and maximum ASVs, followed by the number of singletons, doubletons, etc. for pooling options.</strong></small></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,dW5wb29sZWQscHN1ZWRvLHBvb2xlZAo1MDA0MTUsNTIxMzA5LDQ5MTE0Mwo0MzIyLDQzNDgsMzM1Nwo2LDEwLDE0CjY5Miw4MTcsMTAxNwozMTg2LDMxNzYsMTI0Nwo1OTgsNDAxLDcxMQoyMzksMjUwLDQ0MwoxMjMsMTcyLDI4OQo0OCwxMTgsMjA0CjQ1LDgyLDEyMwoyMyw0MiwxMDIKMjIsMzIsODYKMTYsMjYsNDkKMywxMSwzMgo0LDExLDI0CjcsMTUsMTUKNCw3LDIzCjMsMyw3CjEsMiwyCg==').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'its18_pooling_summary.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="cell-output-display">

<div id="htmlwidget-e6b5fb577cda0c457c55" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e6b5fb577cda0c457c55">{"x":{"tag":{"name":"Reactable","attribs":{"data":{".rownames":["no_reads","no_asvs","min_asv","max_asv","n_samp_1","n_samp_2","n_samp_3","n_samp_4","n_samp_5","n_samp_6","n_samp_7","n_samp_8","n_samp_9","n_samp_10","n_samp_11","n_samp_12","n_samp_13","n_samp_14","n_samp_15"],"unpooled":[500415,4322,6,692,3186,598,239,123,48,45,23,22,16,3,4,7,4,3,1],"psuedo":[521309,4348,10,817,3176,401,250,172,118,82,42,32,26,11,11,15,7,3,2],"pooled":[491143,3357,14,1017,1247,711,443,289,204,123,102,86,49,32,24,15,23,7,2]},"columns":[{"accessor":".rownames","name":"","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["no_reads","no_asvs","min_asv","max_asv","n_samp_1","n_samp_2","n_samp_3","n_samp_4","n_samp_5","n_samp_6","n_samp_7","n_samp_8","n_samp_9","n_samp_10","n_samp_11","n_samp_12","n_samp_13","n_samp_14","n_samp_15"],"header":"","align":"center","footerStyle":{"fontWeight":"bold"},"rowHeader":true},{"accessor":"unpooled","name":"unpooled","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["500415","4322","6","692","3186","598","239","123","48","45","23","22","16","3","4","7","4","3","1"],"header":"unpooled","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"psuedo","name":"psuedo","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["521309","4348","10","817","3176","401","250","172","118","82","42","32","26","11","11","15","7","3","2"],"header":"psuedo","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"pooled","name":"pooled","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["491143","3357","14","1017","1247","711","443","289","204","123","102","86","49","32","24","15","23","7","2"],"header":"pooled","align":"center","footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,19],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"0331c0e7a396ed95156c4aa348e1f131"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="cell">

</div>
<p><small><strong>(ITS) Table 5 | Total number of reads and ASVs by sample for pooling options.</strong></small></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,U2FtcGxlX0lELHVucG9vbGVkX3JlYWRzLHBzdWVkb19yZWFkcyxwb29sZWRfcmVhZHMsdW5wb29sZWRfYXN2LHBzdWVkb19hc3YscG9vbGVkX2FzdgpQMDFfRDAwXzAxMF9XM0EsMjk2ODMsMzI3OTMsMjk3OTQsNDM5LDU3NSw5NDUKUDAxX0QwMF8wMTBfVzhBLDQyOTU5LDQzNTkxLDQxODgwLDYwNyw2NTUsNzE5ClAwMl9EMDBfMDEwX0MwQSw1MTQxNCw1MTg5NCw1MTE0NCw2OTIsODE3LDEwMTEKUDAzX0QwMF8wMTBfVzNCLDM4NDM3LDM4NTg5LDM4Mzc0LDUzOCw2MjQsNzY1ClAwM19EMDBfMDEwX1c4QiwxNDkwOSwxNTA0NSwxNDE4OCwyODIsMzA3LDM0OQpQMDRfRDAwXzAxMF9DMEIsNTY2NjIsNTc5MTEsNTY1OTMsNjQxLDc3MSwxMDE3ClAwNV9EMDBfMDEwX1czQywyMjYsMjM3LDI5NiwyOSw0OSw5NQpQMDVfRDAwXzAxMF9XOEMsNDE0NTMsNDIzOTUsMzU2MDAsNTM1LDU0OCw2MTYKUDA2X0QwMF8wMTBfQzBDLDQ1NzU1LDQ1OTEzLDQ1ODUwLDY1MCw3NDUsOTU0ClAwN19EMDBfMDEwX1czRCwzNzYzNywzNzk0NiwzNjA2Myw2MTUsNzA4LDg2NwpQMDdfRDAwXzAxMF9XOEQsODk1Myw4OTg0LDkxNzIsMTE5LDE0NSwzMzUKUDA4X0QwMF8wMTBfQzBELDM4MDg3LDM4Mjk3LDM4MzcxLDU0Niw2MDcsNzAyClAwOV9EMDBfMDEwX1czRSwyOTE4NCwyOTYwMSwyOTEwMiw1ODksNjUzLDc0NQpQMDlfRDAwXzAxMF9XOEUsNzUsNzQsODAsNiwxMCwxNApQMTBfRDAwXzAxMF9DMEUsNjQ5ODEsNzgwMzksNjQ2MzYsNTYxLDY0Miw4MTIK').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'its18_pooling_results.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="cell-output-display">

<div id="htmlwidget-a3e3a742d1fe4442b0ba" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-a3e3a742d1fe4442b0ba">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Sample_ID":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"unpooled_reads":[29683,42959,51414,38437,14909,56662,226,41453,45755,37637,8953,38087,29184,75,64981],"psuedo_reads":[32793,43591,51894,38589,15045,57911,237,42395,45913,37946,8984,38297,29601,74,78039],"pooled_reads":[29794,41880,51144,38374,14188,56593,296,35600,45850,36063,9172,38371,29102,80,64636],"unpooled_asv":[439,607,692,538,282,641,29,535,650,615,119,546,589,6,561],"psuedo_asv":[575,655,817,624,307,771,49,548,745,708,145,607,653,10,642],"pooled_asv":[945,719,1011,765,349,1017,95,616,954,867,335,702,745,14,812]},"columns":[{"accessor":"Sample_ID","name":"Sample ID","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"header":"Sample ID","minWidth":150,"align":"left","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"unpooled_reads","name":"unpooled_reads","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["29683","42959","51414","38437","14909","56662","226","41453","45755","37637","8953","38087","29184","75","64981"],"header":"unpooled reads","minWidth":150,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"psuedo_reads","name":"psuedo_reads","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["32793","43591","51894","38589","15045","57911","237","42395","45913","37946","8984","38297","29601","74","78039"],"header":"psuedo reads","minWidth":150,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"pooled_reads","name":"pooled_reads","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["29794","41880","51144","38374","14188","56593","296","35600","45850","36063","9172","38371","29102","80","64636"],"header":"pooled reads","minWidth":150,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"unpooled_asv","name":"unpooled_asv","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["439","607","692","538","282","641","29","535","650","615","119","546","589","6","561"],"header":"unpooled asv","minWidth":150,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"psuedo_asv","name":"psuedo_asv","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["575","655","817","624","307","771","49","548","745","708","145","607","653","10","642"],"header":"psuedo asv","minWidth":150,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"pooled_asv","name":"pooled_asv","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["945","719","1011","765","349","1017","95","616","954","867","335","702","745","14","812"],"header":"pooled asv","minWidth":150,"align":"center","footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"3070768176cc2b5b7639bf1a54040863"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</details><section id="forward-reads-4" class="level4"><h4 class="anchored" data-anchor-id="forward-reads-4">Forward Reads</h4>
<div class="cell">
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">dadaFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/dada.html">dada</a></span><span class="op">(</span><span class="va">derepFs</span>, err <span class="op">=</span> <span class="va">errF</span>, pool <span class="op">=</span> <span class="cn">TRUE</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>15 samples were pooled: 582642 reads in 102887 unique sequences.</code></pre>
<div class="cell">
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">dadaFs</span><span class="op">[[</span><span class="va">sampnum</span><span class="op">]</span><span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>dada-class: object describing DADA2 denoising results
1175 sequence variants were inferred from 7277 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
</section><section id="reverse-reads-2" class="level4"><h4 class="anchored" data-anchor-id="reverse-reads-2">Reverse Reads</h4>
<div class="cell">
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">dadaRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/dada.html">dada</a></span><span class="op">(</span><span class="va">derepRs</span>, err <span class="op">=</span> <span class="va">errR</span>, pool <span class="op">=</span> <span class="cn">TRUE</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>15 samples were pooled: 582642 reads in 174866 unique sequences.</code></pre>
<div class="cell">
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">dadaRs</span><span class="op">[[</span><span class="va">sampnum</span><span class="op">]</span><span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>dada-class: object describing DADA2 denoising results
1116 sequence variants were inferred from 13081 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
<div class="cell">
<div class="sourceCode" id="cb96"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/save.html">save.image</a></span><span class="op">(</span><span class="st">"pre_merge_its_2018.rdata"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section></section><section id="merge-paired-reads" class="level2"><h2 class="anchored" data-anchor-id="merge-paired-reads">8. Merge Paired Reads</h2>
<div class="cell">
<div class="sourceCode" id="cb97"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">mergers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/mergePairs.html">mergePairs</a></span><span class="op">(</span><span class="va">dadaFs</span>, <span class="va">derepFs</span>, <span class="va">dadaRs</span>, <span class="va">derepRs</span>, verbose<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<details><summary>
Detailed results of merging reads
</summary><pre><code>
29795 paired-reads (in 946 unique pairings) successfully merged out of 32193 (in 1177 pairings) input.
41880 paired-reads (in 719 unique pairings) successfully merged out of 46715 (in 941 pairings) input.
64805 paired-reads (in 815 unique pairings) successfully merged out of 66771 (in 1015 pairings) input.
Duplicate sequences in merged output.
51144 paired-reads (in 1011 unique pairings) successfully merged out of 57613 (in 1325 pairings) input.
38442 paired-reads (in 767 unique pairings) successfully merged out of 41985 (in 994 pairings) input.
14188 paired-reads (in 349 unique pairings) successfully merged out of 16981 (in 452 pairings) input.
56594 paired-reads (in 1018 unique pairings) successfully merged out of 64411 (in 1306 pairings) input.
296 paired-reads (in 95 unique pairings) successfully merged out of 301 (in 100 pairings) input.
35713 paired-reads (in 619 unique pairings) successfully merged out of 41006 (in 786 pairings) input.
45851 paired-reads (in 955 unique pairings) successfully merged out of 50174 (in 1240 pairings) input.
36121 paired-reads (in 870 unique pairings) successfully merged out of 41541 (in 1135 pairings) input.
Duplicate sequences in merged output.
9172 paired-reads (in 335 unique pairings) successfully merged out of 9403 (in 402 pairings) input.
38392 paired-reads (in 704 unique pairings) successfully merged out of 40897 (in 886 pairings) input.
29155 paired-reads (in 748 unique pairings) successfully merged out of 34979 (in 990 pairings) input.
Duplicate sequences in merged output.
80 paired-reads (in 14 unique pairings) successfully merged out of 82 (in 16 pairings) input.
</code></pre>
</details><p><br></p>
<div class="cell">

</div>
</section><section id="construct-sequence-table-1" class="level2"><h2 class="anchored" data-anchor-id="construct-sequence-table-1">9. Construct Sequence Table</h2>
<div class="cell">
<div class="sourceCode" id="cb98"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">seqtabF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/makeSequenceTable.html">makeSequenceTable</a></span><span class="op">(</span><span class="va">mergers</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">seqtabF</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1]   15 3364</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/dada2/man/getSequences.html">getSequences</a></span><span class="op">(</span><span class="va">seqtabF</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>130 131 132 133 136 139 140 141 143 144 146 147 148 149 150 151 152 153 154 155 
  2   2   1   1   1   1   1   3   2   1   3   5   5   4   5   5   3   2   5   4 
156 159 160 161 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 
  1   2   1   1   2   2   4   9   2   1   2   5   5   5   8   9   2   5  10  13 
179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 
 13  14  11  14  12  15  19  12  12  13   9  22  11  11  12  15  17  13  10  11 
199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 
 22  16  10  11  13  17  10   8  10  12  13   7  20  18  24  30  39  38  20  19 
219 220 221 222 223 224 225 226 228 229 230 231 232 233 234 235 236 237 238 239 
 12  18  21  23  23  29  32  32   6  48   4  73  20  27  29  19  31  33  30  29 
240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255 256 257 258 259 
 35  21  36  16  16  18  17  25  22  29  17  27  28  22  20  18  28  19  28  25 
260 261 262 263 264 265 266 267 268 269 270 271 272 273 274 275 276 277 278 279 
 23  11  15  13  16  18  23  17  12  12  16  15  32  17  22  21  21  18  16  17 
280 281 282 283 284 285 286 287 288 289 290 291 292 293 294 295 296 297 298 299 
 13  22  13  18  12  13  13  18  17  15  14  18   8  11  13  17  12  11  13   9 
300 301 302 303 304 305 306 307 308 309 310 311 312 313 314 315 316 317 318 319 
 16  10   9   3   8  12   8  13  20  15   8  13  16  11  10   8  18  10   8  12 
320 321 322 323 324 325 326 327 328 329 330 331 332 333 334 335 336 337 338 339 
  8   5  12  17  13   9   8   8   9  12  16  13  14  11  12  13  18  10  11  10 
340 341 342 343 344 345 346 347 348 349 350 351 352 353 354 355 356 357 358 359 
  9  11   8   8   5  15   8  16  11  15   4   9   7   9  12  14   5  12  11   5 
360 361 362 363 364 365 366 367 368 369 370 371 372 373 374 375 376 377 378 379 
  2   4  11   5   2  14   6  12   7   5   6   4   3   9   5  10   8   9   4   6 
380 381 382 383 384 385 386 387 388 389 390 391 392 393 394 395 396 397 398 399 
  5   3   7   7   4   2   3   4   3   5   5   1   3   4   3   6   5   4   3   2 
400 401 402 403 404 405 406 407 408 409 410 411 412 413 414 415 416 417 418 421 
 11   5   3   3   3   3   2   1   2   3   3   1   3   3   1   2   2   1   2   2 
422 423 424 427 430 431 432 434 436 437 439 440 441 442 443 446 447 448 
  2   3   2   1   3   2   4   2   1   2   4   3   1   2   1   1   2   1 
</code></pre>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="include/dada2/its18_read_length_before.png" class="img-fluid" width="1050"></p>
</div>
</div>
<p><small><strong>(ITS) Figure 5 | Distribution of read length by total ASVs before removing extreme length variants.</strong></small></p>
<p>As expected, a wide range of length variability in the the amplified ITS region.</p>
<div class="cell">
<div class="sourceCode" id="cb102"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">seqtab</span> <span class="op">&lt;-</span> <span class="va">seqtabF</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">seqtabF</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">100</span>,<span class="fl">450</span><span class="op">)</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">seqtab</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1]   15 3364</code></pre>
</div>
</div>
<p>After removing the extreme length variants, we have 3364, a reduction of 0 sequence variants.</p>
<div class="cell">
<div class="sourceCode" id="cb104"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/dada2/man/getSequences.html">getSequences</a></span><span class="op">(</span><span class="va">seqtab</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">

</div>
</section><section id="remove-chimeras-1" class="level2"><h2 class="anchored" data-anchor-id="remove-chimeras-1">10. Remove Chimeras</h2>
<div class="cell">
<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">seqtab.nochim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/removeBimeraDenovo.html">removeBimeraDenovo</a></span><span class="op">(</span><span class="va">seqtab</span>,
                                    method <span class="op">=</span> <span class="st">"consensus"</span>,
                                    multithread <span class="op">=</span> <span class="cn">TRUE</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">seqtab.nochim</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>Identified 7 bimeras out of 3364 input sequences.</code></pre>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1]   15 3357</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb108"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">seqtab.nochim</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">seqtab.2</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.9990135</code></pre>
</div>
</div>
<p>Chimera checking removed an additional 7 sequence variants however, when we account for the abundances of each variant, we see chimeras accounts for about 0.09865% of the merged sequence reads. Curious.</p>
<div class="cell">
<div class="sourceCode" id="cb110"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html">colSums</a></span><span class="op">(</span><span class="va">seqtab.nochim</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15
1247  711  443  289  204  123  102   86   49   32   24   15   23    7    2</code></pre>
<div class="cell">
<div class="sourceCode" id="cb112"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html">colSums</a></span><span class="op">(</span><span class="va">seqtab.nochim</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>  14   95  335  349  616  702  719  745  765  812  867  945  954 1011 1017
   1    1    1    1    1    1    1    1    1    1    1    1    1    1    1</code></pre>
<p>The last thing we want to do is write the sequence table to an <code>RDS</code> file.</p>
<div class="cell">
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">seqtab.nochim</span>, <span class="st">"its18_seqtab.nochim.rds"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">

</div>
</section><section id="track-reads-through-workflow-1" class="level2"><h2 class="anchored" data-anchor-id="track-reads-through-workflow-1">11. Track Reads through Workflow</h2>
<div class="cell">

</div>
<p><small><strong>(ITS) Table 6 | Tracking reads changes at each step of the DADA2 workflow.</strong></small></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,U2FtcGxlX0lELHJhdyxwcmVfZmlsdCxjdXQsZmlsdGVyZWQsZGVub2lzZWRGLGRlbm9pc2VkUixtZXJnZWQsbm9uY2hpbQpQMDFfRDAwXzAxMF9XM0EsNTA4OTQsNTA4ODIsNTA0MTksMzU5OTksMzI2MzUsMzMzODksMjk3OTUsMjk3OTQKUDAxX0QwMF8wMTBfVzhBLDcwNzM0LDcwNjQ5LDY4MDc1LDQ4MjMwLDQ3MTUwLDQ3MDU5LDQxODgwLDQxODgwClAwMl9EMDBfMDEwX0MwQSw3NDU4MSw3NDU1Nyw3Mzk5NCw1ODk1Niw1ODAxOCw1ODEzMCw1MTE0NCw1MTE0NApQMDNfRDAwXzAxMF9XM0IsNjAwMDYsNTk5NTMsNTgxNDYsNDI1NjMsNDIyNTMsNDIxNTEsMzg0NDIsMzgzNzQKUDAzX0QwMF8wMTBfVzhCLDMxMTc0LDMxMDI5LDI1NzgyLDE3NTQ4LDE3MTMyLDE3MjEwLDE0MTg4LDE0MTg4ClAwNF9EMDBfMDEwX0MwQiw4ODczMiw4ODY5NCw4NzUzMiw2ODI1OSw2NDc5Nyw2NzYxMyw1NjU5NCw1NjU5MwpQMDVfRDAwXzAxMF9XM0MsODA5LDc5Nyw0MTIsMzA4LDMwMiwzMDIsMjk2LDI5NgpQMDVfRDAwXzAxMF9XOEMsODE5MDgsODE2NzEsNzQ2NjksNDg1MTgsNDEzNTcsNDYzMjEsMzU3MTMsMzU2MDAKUDA2X0QwMF8wMTBfQzBDLDY3NDE5LDY3Mzk5LDY2NjU2LDUxMDU1LDUwNjM0LDUwMzYyLDQ1ODUxLDQ1ODUwClAwN19EMDBfMDEwX1czRCw3MDEyMyw3MDA3Myw2ODQ3NCw0MzQ3MSw0MjE5NCw0MjA3NCwzNjEyMSwzNjA2MwpQMDdfRDAwXzAxMF9XOEQsMTU0MzUsMTU0MTcsMTQ5NTUsOTQ4OCw5NDI2LDk0MjcsOTE3Miw5MTcyClAwOF9EMDBfMDEwX0MwRCw2MDc5OSw2MDYwNCw1NDUzNCw0MTU2Myw0MTI0Myw0MTA3NywzODM5MiwzODM3MQpQMDlfRDAwXzAxMF9XM0UsNTU5MjcsNTU3NjQsNTEzODIsMzU3MjIsMzU0MTAsMzUxNjgsMjkxNTUsMjkxMDIKUDA5X0QwMF8wMTBfVzhFLDE0MTgsMTM4MywxNDEsODUsODQsODIsODAsODAKUDEwX0QwMF8wMTBfQzBFLDk2NTMxLDk2NDkyLDk1MjU2LDgwODc3LDY4NjgxLDcwMDEzLDY0ODA1LDY0NjM2Cg==').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'its18_track_read_changes.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="cell-output-display">

<div id="htmlwidget-26b01c9158019c69a9e9" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-26b01c9158019c69a9e9">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Sample_ID":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"raw":[50894,70734,74581,60006,31174,88732,809,81908,67419,70123,15435,60799,55927,1418,96531],"pre_filt":[50882,70649,74557,59953,31029,88694,797,81671,67399,70073,15417,60604,55764,1383,96492],"cut":[50419,68075,73994,58146,25782,87532,412,74669,66656,68474,14955,54534,51382,141,95256],"filtered":[35999,48230,58956,42563,17548,68259,308,48518,51055,43471,9488,41563,35722,85,80877],"denoisedF":[32635,47150,58018,42253,17132,64797,302,41357,50634,42194,9426,41243,35410,84,68681],"denoisedR":[33389,47059,58130,42151,17210,67613,302,46321,50362,42074,9427,41077,35168,82,70013],"merged":[29795,41880,51144,38442,14188,56594,296,35713,45851,36121,9172,38392,29155,80,64805],"nonchim":[29794,41880,51144,38374,14188,56593,296,35600,45850,36063,9172,38371,29102,80,64636]},"columns":[{"accessor":"Sample_ID","name":"Sample ID","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"header":"Sample ID","align":"left","footerStyle":{"fontWeight":"bold"},"footer":"Total reads","minWidth":150,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"raw","name":"raw","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["50894","70734","74581","60006","31174","88732","809","81908","67419","70123","15435","60799","55927","1418","96531"],"header":"raw","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"826490"},{"accessor":"pre_filt","name":"pre_filt","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["50882","70649","74557","59953","31029","88694","797","81671","67399","70073","15417","60604","55764","1383","96492"],"header":"pre filt","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"825364"},{"accessor":"cut","name":"cut","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["50419","68075","73994","58146","25782","87532","412","74669","66656","68474","14955","54534","51382","141","95256"],"header":"cut","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"790427"},{"accessor":"filtered","name":"filtered","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["35999","48230","58956","42563","17548","68259","308","48518","51055","43471","9488","41563","35722","85","80877"],"header":"filtered","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"582642"},{"accessor":"denoisedF","name":"denoisedF","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["32635","47150","58018","42253","17132","64797","302","41357","50634","42194","9426","41243","35410","84","68681"],"header":"denoisedF","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"551316"},{"accessor":"denoisedR","name":"denoisedR","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["33389","47059","58130","42151","17210","67613","302","46321","50362","42074","9427","41077","35168","82","70013"],"header":"denoisedR","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"560378"},{"accessor":"merged","name":"merged","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["29795","41880","51144","38442","14188","56594","296","35713","45851","36121","9172","38392","29155","80","64805"],"header":"merged","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"491628"},{"accessor":"nonchim","name":"nonchim","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["29794","41880","51144","38374","14188","56593","296","35600","45850","36063","9172","38371","29102","80","64636"],"header":"nonchim","align":"center","footerStyle":{"fontWeight":"bold"},"footer":"491143"}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"37165b537a6de76186ee802667fbb823"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="cell">

</div>
</section><section id="assign-taxonomy-1" class="level2"><h2 class="anchored" data-anchor-id="assign-taxonomy-1">12. Assign Taxonomy</h2>
<p>For taxonomic classification, we used the UNITE <span class="citation" data-cites="nilsson2019unite">(<a href="#ref-nilsson2019unite" role="doc-biblioref">Nilsson et al. 2019</a>)</span> database, specifically the <a href="https://plutof.ut.ee/#/doi/10.15156/BIO/786368">UNITE general FASTA release for Fungi (v. 04.02.2020)</a> <span class="citation" data-cites="abarenkov2020unite">(<a href="#ref-abarenkov2020unite" role="doc-biblioref">Abarenkov et al. 2020</a>)</span>.</p>
<div class="cell">

</div>
<div class="cell">
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dada2/man/assignTaxonomy.html">assignTaxonomy</a></span><span class="op">(</span><span class="va">seqtab</span>,
                      <span class="st">"sh_general_release_dynamic_s_04.02.2020_dev.fasta"</span>,
                      multithread <span class="op">=</span> <span class="fl">20</span>, tryRC <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">tax</span>, <span class="st">"its18_tax_its.rds"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>UNITE fungal taxonomic reference detected.</code></pre>
</section><section id="save-image-1" class="level2"><h2 class="anchored" data-anchor-id="save-image-1">13. Save Image</h2>
<div class="cell">
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/save.html">save.image</a></span><span class="op">(</span><span class="st">"its18_dada2_wf.rdata"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="r-session-information-code-1" class="level2"><h2 class="anchored" data-anchor-id="r-session-information-code-1">R Session Information &amp; Code</h2>
<p>This workflow was run on the <a href="https://doi.org/10.25572/SIHPC">Smithsonian High Performance Cluster (SI/HPC)</a>, Smithsonian Institution. Below are the specific packages and versions used in this workflow using both <code><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo()</a></code> and <code><a href="https://r-lib.github.io/sessioninfo/reference/session_info.html">devtools::session_info()</a></code>. Click the arrow to see the details.</p>
<details><summary>
Show/hide R Session Info
</summary><pre><code>sessionInfo()
R version 4.0.0 (2020-04-24)
Platform: x86_64-conda_cos6-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /home/scottjj/miniconda3/envs/R-4/lib/libopenblasp-r0.3.9.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
 [1] grid      stats4    parallel  stats     graphics  grDevices utils
 [8] datasets  methods   base

other attached packages:
 [1] gridExtra_2.3               ggplot2_3.3.0
 [3] ShortRead_1.46.0            GenomicAlignments_1.24.0
 [5] SummarizedExperiment_1.18.1 DelayedArray_0.14.0
 [7] matrixStats_0.56.0          Biobase_2.48.0
 [9] Rsamtools_2.4.0             GenomicRanges_1.40.0
[11] GenomeInfoDb_1.24.0         Biostrings_2.56.0
[13] XVector_0.28.0              IRanges_2.22.1
[15] S4Vectors_0.26.0            BiocParallel_1.22.0
[17] BiocGenerics_0.34.0         dada2_1.16.0
[19] Rcpp_1.0.4.6

loaded via a namespace (and not attached):
 [1] plyr_1.8.6             RColorBrewer_1.1-2     pillar_1.4.4
 [4] compiler_4.0.0         bitops_1.0-6           tools_4.0.0
 [7] zlibbioc_1.34.0        lifecycle_0.2.0        tibble_3.0.1
[10] gtable_0.3.0           lattice_0.20-41        png_0.1-7
[13] pkgconfig_2.0.3        rlang_0.4.6            Matrix_1.2-18
[16] GenomeInfoDbData_1.2.3 withr_2.2.0            stringr_1.4.0
[19] hwriter_1.3.2          vctrs_0.3.0            glue_1.4.1
[22] R6_2.4.1               jpeg_0.1-8.1           latticeExtra_0.6-29
[25] reshape2_1.4.4         magrittr_1.5           scales_1.1.1
[28] ellipsis_0.3.1         colorspace_1.4-1       stringi_1.4.6
[31] RCurl_1.98-1.2         RcppParallel_5.0.1     munsell_0.5.0
[34] crayon_1.3.4

devtools::session_info()
â”€ Session info â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 setting  value
 version  R version 4.0.0 (2020-04-24)
 os       CentOS Linux 7 (Core)
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       America/New_York
 date     2020-06-04

â”€ Packages â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 package              * version  date       lib source
 assertthat             0.2.1    2019-03-21 [1] CRAN (R 4.0.0)
 backports              1.1.7    2020-05-13 [1] CRAN (R 4.0.0)
 Biobase              * 2.48.0   2020-04-27 [1] Bioconductor
 BiocGenerics         * 0.34.0   2020-04-27 [1] Bioconductor
 BiocParallel         * 1.22.0   2020-04-27 [1] Bioconductor
 Biostrings           * 2.56.0   2020-04-27 [1] Bioconductor
 bitops                 1.0-6    2013-08-17 [1] CRAN (R 4.0.0)
 callr                  3.4.3    2020-03-28 [1] CRAN (R 4.0.0)
 cli                    2.0.2    2020-02-28 [1] CRAN (R 4.0.0)
 colorspace             1.4-1    2019-03-18 [1] CRAN (R 4.0.0)
 crayon                 1.3.4    2017-09-16 [1] CRAN (R 4.0.0)
 dada2                * 1.16.0   2020-04-27 [1] Bioconductor
 DelayedArray         * 0.14.0   2020-04-27 [1] Bioconductor
 desc                   1.2.0    2018-05-01 [1] CRAN (R 4.0.0)
 devtools               2.3.0    2020-04-10 [1] CRAN (R 4.0.0)
 digest                 0.6.25   2020-02-23 [1] CRAN (R 4.0.0)
 ellipsis               0.3.1    2020-05-15 [1] CRAN (R 4.0.0)
 fansi                  0.4.1    2020-01-08 [1] CRAN (R 4.0.0)
 fs                     1.4.1    2020-04-04 [1] CRAN (R 4.0.0)
 GenomeInfoDb         * 1.24.0   2020-04-27 [1] Bioconductor
 GenomeInfoDbData       1.2.3    2020-05-22 [1] Bioconductor
 GenomicAlignments    * 1.24.0   2020-04-27 [1] Bioconductor
 GenomicRanges        * 1.40.0   2020-04-27 [1] Bioconductor
 ggplot2              * 3.3.0    2020-03-05 [1] CRAN (R 4.0.0)
 glue                   1.4.1    2020-05-13 [1] CRAN (R 4.0.0)
 gridExtra            * 2.3      2017-09-09 [1] CRAN (R 4.0.0)
 gtable                 0.3.0    2019-03-25 [1] CRAN (R 4.0.0)
 hwriter                1.3.2    2014-09-10 [1] CRAN (R 4.0.0)
 IRanges              * 2.22.1   2020-04-28 [1] Bioconductor
 jpeg                   0.1-8.1  2019-10-24 [1] CRAN (R 4.0.0)
 lattice                0.20-41  2020-04-02 [1] CRAN (R 4.0.0)
 latticeExtra           0.6-29   2019-12-19 [1] CRAN (R 4.0.0)
 lifecycle              0.2.0    2020-03-06 [1] CRAN (R 4.0.0)
 magrittr               1.5      2014-11-22 [1] CRAN (R 4.0.0)
 Matrix                 1.2-18   2019-11-27 [1] CRAN (R 4.0.0)
 matrixStats          * 0.56.0   2020-03-13 [1] CRAN (R 4.0.0)
 memoise                1.1.0    2017-04-21 [1] CRAN (R 4.0.0)
 munsell                0.5.0    2018-06-12 [1] CRAN (R 4.0.0)
 pillar                 1.4.4    2020-05-05 [1] CRAN (R 4.0.0)
 pkgbuild               1.0.8    2020-05-07 [1] CRAN (R 4.0.0)
 pkgconfig              2.0.3    2019-09-22 [1] CRAN (R 4.0.0)
 pkgload                1.0.2    2018-10-29 [1] CRAN (R 4.0.0)
 plyr                   1.8.6    2020-03-03 [1] CRAN (R 4.0.0)
 png                    0.1-7    2013-12-03 [1] CRAN (R 4.0.0)
 prettyunits            1.1.1    2020-01-24 [1] CRAN (R 4.0.0)
 processx               3.4.2    2020-02-09 [1] CRAN (R 4.0.0)
 ps                     1.3.3    2020-05-08 [1] CRAN (R 4.0.0)
 R6                     2.4.1    2019-11-12 [1] CRAN (R 4.0.0)
 RColorBrewer           1.1-2    2014-12-07 [1] CRAN (R 4.0.0)
 Rcpp                 * 1.0.4.6  2020-04-09 [1] CRAN (R 4.0.0)
 RcppParallel           5.0.1    2020-05-06 [1] CRAN (R 4.0.0)
 RCurl                  1.98-1.2 2020-04-18 [1] CRAN (R 4.0.0)
 remotes                2.1.1    2020-02-15 [1] CRAN (R 4.0.0)
 reshape2               1.4.4    2020-04-09 [1] CRAN (R 4.0.0)
 rlang                  0.4.6    2020-05-02 [1] CRAN (R 4.0.0)
 rprojroot              1.3-2    2018-01-03 [1] CRAN (R 4.0.0)
 Rsamtools            * 2.4.0    2020-04-27 [1] Bioconductor
 S4Vectors            * 0.26.0   2020-04-27 [1] Bioconductor
 scales                 1.1.1    2020-05-11 [1] CRAN (R 4.0.0)
 sessioninfo            1.1.1    2018-11-05 [1] CRAN (R 4.0.0)
 ShortRead            * 1.46.0   2020-04-27 [1] Bioconductor
 stringi                1.4.6    2020-02-17 [1] CRAN (R 4.0.0)
 stringr                1.4.0    2019-02-10 [1] CRAN (R 4.0.0)
 SummarizedExperiment * 1.18.1   2020-04-30 [1] Bioconductor
 testthat               2.3.2    2020-03-02 [1] CRAN (R 4.0.0)
 tibble                 3.0.1    2020-04-20 [1] CRAN (R 4.0.0)
 usethis                1.6.1    2020-04-29 [1] CRAN (R 4.0.0)
 vctrs                  0.3.0    2020-05-11 [1] CRAN (R 4.0.0)
 withr                  2.2.0    2020-04-20 [1] CRAN (R 4.0.0)
 XVector              * 0.28.0   2020-04-27 [1] Bioconductor
 zlibbioc               1.34.0   2020-04-27 [1] Bioconductor</code></pre>
</details><p><br></p>
</section></section><section id="workflow-output" class="level1"><h1>Workflow Output</h1>
<p>Data products generated in this workflow can be downloaded from figshare.</p>
<iframe src="https://widgets.figshare.com/articles/14687184/embed?show_title=1" width="100%" height="351" frameborder="0">
</iframe>
<p>The end!</p>
<p><a href="data-prep.html" class="btnnav button_round" style="float: right;">Next workflow: <br> 2. Data Set Preparation</a><br><br></p>
<div class="cell">

</div>
</section><div id="quarto-appendix" class="default"><section id="source-code" class="level4 appendix"><h2 class="quarto-appendix-heading">Source Code</h2><div class="quarto-appendix-contents">
<p>The source code for this page can be accessed on GitHub <svg aria-hidden="true" role="img" viewbox="0 0 496 512" style="height:1em;width:0.97em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:currentColor;overflow:visible;position:relative;"><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"></path></svg> by <a href="https://github.com/sweltr/high-temp/blob/main/dada2.qmd">clicking this link</a>.</p>
</div></section><section id="data-availability" class="level4 appendix"><h2 class="quarto-appendix-heading">Data Availability</h2><div class="quarto-appendix-contents">
<p>Raw fastq files available on figshare at <a href="https://doi.org/10.25573/data.14686665">10.25573/data.14686665</a> for the 16S rRNA data and <a href="https://doi.org/10.25573/data.14686755">10.25573/data.14686755</a> for the ITS data. Trimmed fastq files (primers removed) available through the ENA under project accession number <a href="https://www.ebi.ac.uk/ena/browser/view/PRJEB45074">PRJEB45074 (ERP129199)</a>. Data generated in this workflow can be accessed on figshare at <a href="https://doi.org/10.25573/data.14687184">10.25573/data.14687184</a>.</p>
</div></section><section id="last-updated-on" class="level4 appendix"><h2 class="quarto-appendix-heading">Last updated on</h2><div class="quarto-appendix-contents">
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "2022-06-29 07:30:39 EST"</code></pre>
</div>
</div>


<!-- -->


</div></section><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-abarenkov2020unite" class="csl-entry" role="doc-biblioentry">
Abarenkov, Kessy, Allan Zirk, Timo Piirmann, Raivo PÃ¶hÃ¶nen, Filipp Ivanov, R. Henrik Nilsson, and Urmas KÃµljalg. 2020. <span>â€œUNITE General FASTA Release for Fungi.â€</span> <a href="https://dx.doi.org/10.15156/BIO/786368">https://dx.doi.org/10.15156/BIO/786368</a>.
</div>
<div id="ref-callahan2016dada2" class="csl-entry" role="doc-biblioentry">
Callahan, Benjamin J, Paul J McMurdie, Michael J Rosen, Andrew W Han, Amy Jo A Johnson, and Susan P Holmes. 2016. <span>â€œDada2: High-Resolution Sample Inference from Illumina Amplicon Data.â€</span> <em>Nature Methods</em> 13 (7): 581. <a href="https://doi.org/10.1038/nmeth.3869">https://doi.org/10.1038/nmeth.3869</a>.
</div>
<div id="ref-caporaso2011global" class="csl-entry" role="doc-biblioentry">
Caporaso, J Gregory, Christian L Lauber, William A Walters, Donna Berg-Lyons, Catherine A Lozupone, Peter J Turnbaugh, Noah Fierer, and Rob Knight. 2011. <span>â€œGlobal Patterns of 16s rRNA Diversity at a Depth of Millions of Sequences Per Sample.â€</span> <em>Proceedings of the National Academy of Sciences</em> 108: 4516â€“22. https://doi.org/<a href="https://doi.org/10.1073/pnas.1000080107">https://doi.org/10.1073/pnas.1000080107</a>.
</div>
<div id="ref-gardes1993its" class="csl-entry" role="doc-biblioentry">
Gardes, Monique, and Thomas D Bruns. 1993. <span>â€œITS Primers with Enhanced Specificity for Basidiomycetes-Application to the Identification of Mycorrhizae and Rusts.â€</span> <em>Molecular Ecology</em> 2 (2): 113â€“18. <a href="https://doi.org/10.1111/j.1365-294X.1993.tb00005.x">https://doi.org/10.1111/j.1365-294X.1993.tb00005.x</a>.
</div>
<div id="ref-martin2011cutadapt" class="csl-entry" role="doc-biblioentry">
Martin, Marcel. 2011. <span>â€œCutadapt Removes Adapter Sequences from High-Throughput Sequencing Reads.â€</span> <em>EMBnet. Journal</em> 17 (1): 10â€“12. <a href="https://doi.org/10.14806/ej.17.1.200">https://doi.org/10.14806/ej.17.1.200</a>.
</div>
<div id="ref-nilsson2019unite" class="csl-entry" role="doc-biblioentry">
Nilsson, Rolf Henrik, Karl-Henrik Larsson, Andy F S Taylor, Johan Bengtsson-Palme, Thomas S Jeppesen, Dmitry Schigel, Peter Kennedy, et al. 2019. <span>â€œThe UNITE Database for Molecular Identification of Fungi: Handling Dark Taxa and Parallel Taxonomic Classifications.â€</span> <em>Nucleic Acids Research</em> 47 (D1): D259â€“64. <a href="https://doi.org/10.1093/nar/gky1022">https://doi.org/10.1093/nar/gky1022</a>.
</div>
<div id="ref-quast2012silva" class="csl-entry" role="doc-biblioentry">
Quast, Christian, Elmar Pruesse, Pelin Yilmaz, Jan Gerken, Timmy Schweer, Pablo Yarza, JÃ¶rg Peplies, and Frank Oliver GlÃ¶ckner. 2012. <span>â€œThe SILVA Ribosomal RNA Gene Database Project: Improved Data Processing and Web-Based Tools.â€</span> <em>Nucleic Acids Research</em> 41 (D1): D590â€“96. <a href="https://doi.org/10.1093/nar/gks1219">https://doi.org/10.1093/nar/gks1219</a>.
</div>
<div id="ref-wang2007naive" class="csl-entry" role="doc-biblioentry">
Wang, Qiong, George M Garrity, James M Tiedje, and James R Cole. 2007. <span>â€œNaive Bayesian Classifier for Rapid Assignment of rRNA Sequences into the New Bacterial Taxonomy.â€</span> <em>Applied and Environmental Microbiology</em> 73 (16): 5261â€“67. <a href="https://doi.org/10.1128/AEM.00062-07">https://doi.org/10.1128/AEM.00062-07</a>.
</div>
<div id="ref-white1990amplification" class="csl-entry" role="doc-biblioentry">
White, Thomas J, Thomas Bruns, SJWT Lee, John Taylor, et al. 1990. <span>â€œAmplification and Direct Sequencing of Fungal Ribosomal RNA Genes for Phylogenetics.â€</span> <em>PCR Protocols: A Guide to Methods and Applications</em> 18 (1): 315â€“22.
</div>
</div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
  </div>
  <div class="nav-page nav-page-next">
      <a href="./data-prep.html" class="pagination-link">
        <span class="nav-page-text">2. Data Set Preparation</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb120" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "1. DADA2 Workflow"</span></span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a><span class="an">description:</span><span class="co"> |</span></span>
<span id="cb120-4"><a href="#cb120-4" aria-hidden="true" tabindex="-1"></a><span class="co">  DADA2 workflow for processing the 16S rRNA and the ITS high temperature data sets. Workflow uses paired end reads, beginning with raw fastq files, ending with sequence and taxonomy tables.</span></span>
<span id="cb120-5"><a href="#cb120-5" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb120-6"><a href="#cb120-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-7"><a href="#cb120-7" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb120-8"><a href="#cb120-8" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;&lt;strong&gt;</span>Click here<span class="kw">&lt;/strong&gt;</span> for setup information.<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb120-9"><a href="#cb120-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-12"><a href="#cb120-12" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-13"><a href="#cb120-13" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb120-14"><a href="#cb120-14" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb120-15"><a href="#cb120-15" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">echo =</span> <span class="cn">TRUE</span>, <span class="at">eval =</span> <span class="cn">FALSE</span>)</span>
<span id="cb120-16"><a href="#cb120-16" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">919191</span>)</span>
<span id="cb120-17"><a href="#cb120-17" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends(ape, local = TRUE)  </span></span>
<span id="cb120-18"><a href="#cb120-18" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends_reverse(ape, local = TRUE)  </span></span>
<span id="cb120-19"><a href="#cb120-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dada2); <span class="fu">packageVersion</span>(<span class="st">"dada2"</span>)</span>
<span id="cb120-20"><a href="#cb120-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ShortRead); <span class="fu">packageVersion</span>(<span class="st">"ShortRead"</span>)</span>
<span id="cb120-21"><a href="#cb120-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2); <span class="fu">packageVersion</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb120-22"><a href="#cb120-22" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(tidyverse, gridExtra, grid, </span>
<span id="cb120-23"><a href="#cb120-23" aria-hidden="true" tabindex="-1"></a>               formatR, reactable, downloadthis,</span>
<span id="cb120-24"><a href="#cb120-24" aria-hidden="true" tabindex="-1"></a>               captioner, sparkline, </span>
<span id="cb120-25"><a href="#cb120-25" aria-hidden="true" tabindex="-1"></a>               <span class="at">install =</span> <span class="cn">FALSE</span>, <span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb120-26"><a href="#cb120-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-27"><a href="#cb120-27" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">scipen=</span><span class="dv">999</span>)</span>
<span id="cb120-28"><a href="#cb120-28" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_current<span class="sc">$</span><span class="fu">get</span>(<span class="fu">c</span>(</span>
<span id="cb120-29"><a href="#cb120-29" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cache"</span>,</span>
<span id="cb120-30"><a href="#cb120-30" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cache.path"</span>,</span>
<span id="cb120-31"><a href="#cb120-31" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cache.rebuild"</span>,</span>
<span id="cb120-32"><a href="#cb120-32" aria-hidden="true" tabindex="-1"></a>  <span class="st">"dependson"</span>,</span>
<span id="cb120-33"><a href="#cb120-33" aria-hidden="true" tabindex="-1"></a>  <span class="st">"autodep"</span></span>
<span id="cb120-34"><a href="#cb120-34" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb120-35"><a href="#cb120-35" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb120-36"><a href="#cb120-36" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">"assets"</span>, <span class="st">"functions.R"</span>))</span>
<span id="cb120-37"><a href="#cb120-37" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-38"><a href="#cb120-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-39"><a href="#cb120-39" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb120-40"><a href="#cb120-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-43"><a href="#cb120-43" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-44"><a href="#cb120-44" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-45"><a href="#cb120-45" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-46"><a href="#cb120-46" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb120-47"><a href="#cb120-47" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the caption(s) with captioner</span></span>
<span id="cb120-48"><a href="#cb120-48" aria-hidden="true" tabindex="-1"></a>caption_tab <span class="ot">&lt;-</span> <span class="fu">captioner</span>(<span class="at">prefix =</span> <span class="st">"(16S rRNA) Table"</span>, <span class="at">suffix =</span> <span class="st">" |"</span>, <span class="at">style =</span> <span class="st">"b"</span>)</span>
<span id="cb120-49"><a href="#cb120-49" aria-hidden="true" tabindex="-1"></a>caption_fig <span class="ot">&lt;-</span> <span class="fu">captioner</span>(<span class="at">prefix =</span> <span class="st">"(16S rRNA) Figure"</span>, <span class="at">suffix =</span> <span class="st">" |"</span>, <span class="at">style =</span> <span class="st">"b"</span>)</span>
<span id="cb120-50"><a href="#cb120-50" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a function for referring to the tables in text</span></span>
<span id="cb120-51"><a href="#cb120-51" aria-hidden="true" tabindex="-1"></a>ref <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">str_extract</span>(x, <span class="st">"[^|]*"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb120-52"><a href="#cb120-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">trimws</span>(<span class="at">which =</span> <span class="st">"right"</span>, <span class="at">whitespace =</span> <span class="st">"[ ]"</span>)</span>
<span id="cb120-53"><a href="#cb120-53" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-54"><a href="#cb120-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-57"><a href="#cb120-57" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-58"><a href="#cb120-58" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-59"><a href="#cb120-59" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-60"><a href="#cb120-60" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"assets/captions/captions_dada2.R"</span>)</span>
<span id="cb120-61"><a href="#cb120-61" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-62"><a href="#cb120-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-63"><a href="#cb120-63" aria-hidden="true" tabindex="-1"></a><span class="fu"># Prerequisites</span></span>
<span id="cb120-64"><a href="#cb120-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-65"><a href="#cb120-65" aria-hidden="true" tabindex="-1"></a>In order to run this workflow, you either need the *raw* data, available on the figshare project site (see below), or the *trimmed* data, available from the European Nucleotide Archive (ENA) under project accession number <span class="co">[</span><span class="ot">PRJEB45074</span><span class="co">](https://www.ebi.ac.uk/ena/browser/view/PRJEB45074)</span>. See the <span class="co">[</span><span class="ot">Data Availability</span><span class="co">](data-availability.html)</span> page for more details.</span>
<span id="cb120-66"><a href="#cb120-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-67"><a href="#cb120-67" aria-hidden="true" tabindex="-1"></a>This workflow contains the code we used to process the 16S rRNA and ITS data sets using <span class="co">[</span><span class="ot">DADA2</span><span class="co">](https://benjjneb.github.io/dada2/)</span> <span class="co">[</span><span class="ot">@callahan2016dada2</span><span class="co">]</span>. Workflow construction is based on the <span class="co">[</span><span class="ot">DADA2 Pipeline Tutorial (1.8)</span><span class="co">](https://benjjneb.github.io/dada2/tutorial_1_8.html)</span> and the primer identification section of the <span class="co">[</span><span class="ot">DADA2 ITS Pipeline Workflow (1.8)</span><span class="co">](https://benjjneb.github.io/dada2/ITS_workflow.html)</span>.</span>
<span id="cb120-68"><a href="#cb120-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-69"><a href="#cb120-69" aria-hidden="true" tabindex="-1"></a><span class="fu"># 16s rRNA Data</span></span>
<span id="cb120-70"><a href="#cb120-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-71"><a href="#cb120-71" aria-hidden="true" tabindex="-1"></a><span class="fu">## Workflow Input</span></span>
<span id="cb120-72"><a href="#cb120-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-73"><a href="#cb120-73" aria-hidden="true" tabindex="-1"></a>All files needed to run this workflow can be downloaded from figshare. </span>
<span id="cb120-74"><a href="#cb120-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-75"><a href="#cb120-75" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;iframe</span> <span class="er">src</span><span class="ot">=</span><span class="st">"https://widgets.figshare.com/articles/14686665/embed?show_title=1"</span> <span class="er">width</span><span class="ot">=</span><span class="st">"100%"</span> <span class="er">height</span><span class="ot">=</span><span class="st">"351"</span> <span class="er">allowfullscreen</span> <span class="er">frameborder</span><span class="ot">=</span><span class="st">"0"</span><span class="kw">&gt;&lt;/iframe&gt;</span></span>
<span id="cb120-76"><a href="#cb120-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-77"><a href="#cb120-77" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; NOTE: The reverse reads for 16S rRNA data set are unusable (see below), therefore this workflow deals with forward reads *only*.</span></span>
<span id="cb120-78"><a href="#cb120-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-79"><a href="#cb120-79" aria-hidden="true" tabindex="-1"></a>We will screen the forward reads for reverse primers and show the quality plots of the reverse reads at the beginning of the workflow. After that, reverse reads are not used.</span>
<span id="cb120-80"><a href="#cb120-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-81"><a href="#cb120-81" aria-hidden="true" tabindex="-1"></a><span class="fu">## Overview</span></span>
<span id="cb120-82"><a href="#cb120-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-83"><a href="#cb120-83" aria-hidden="true" tabindex="-1"></a><span class="fu">### Sequence Files &amp; Samples</span></span>
<span id="cb120-84"><a href="#cb120-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-85"><a href="#cb120-85" aria-hidden="true" tabindex="-1"></a>We sequenced a total of 15 samples collected from 10 different plots at a single depth in each plot.</span>
<span id="cb120-86"><a href="#cb120-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-89"><a href="#cb120-89" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-90"><a href="#cb120-90" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-91"><a href="#cb120-91" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-92"><a href="#cb120-92" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"files/dada2/tables/ssu18_sample_seq_info.txt"</span>,</span>
<span id="cb120-93"><a href="#cb120-93" aria-hidden="true" tabindex="-1"></a>                        <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb120-94"><a href="#cb120-94" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-95"><a href="#cb120-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-96"><a href="#cb120-96" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab("seq_dets_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-97"><a href="#cb120-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-100"><a href="#cb120-100" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-101"><a href="#cb120-101" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-102"><a href="#cb120-102" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-103"><a href="#cb120-103" aria-hidden="true" tabindex="-1"></a>seq_table <span class="sc">%&gt;%</span></span>
<span id="cb120-104"><a href="#cb120-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb120-105"><a href="#cb120-105" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"ssu18_sample_seq_info"</span>,</span>
<span id="cb120-106"><a href="#cb120-106" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb120-107"><a href="#cb120-107" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb120-108"><a href="#cb120-108" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb120-109"><a href="#cb120-109" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb120-110"><a href="#cb120-110" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-111"><a href="#cb120-111" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb120-112"><a href="#cb120-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-113"><a href="#cb120-113" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb120-114"><a href="#cb120-114" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb120-115"><a href="#cb120-115" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb120-116"><a href="#cb120-116" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">1</span>),</span>
<span id="cb120-117"><a href="#cb120-117" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">TRUE</span>, <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-118"><a href="#cb120-118" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb120-119"><a href="#cb120-119" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-120"><a href="#cb120-120" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb120-121"><a href="#cb120-121" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Sample ID"</span>, </span>
<span id="cb120-122"><a href="#cb120-122" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb120-123"><a href="#cb120-123" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>),</span>
<span id="cb120-124"><a href="#cb120-124" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>), </span>
<span id="cb120-125"><a href="#cb120-125" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb120-126"><a href="#cb120-126" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">150</span>, <span class="at">footer =</span> <span class="st">"Total reads"</span>), </span>
<span id="cb120-127"><a href="#cb120-127" aria-hidden="true" tabindex="-1"></a>    <span class="at">raw_reads =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values)))</span>
<span id="cb120-128"><a href="#cb120-128" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-129"><a href="#cb120-129" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">5</span>, </span>
<span id="cb120-130"><a href="#cb120-130" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb120-131"><a href="#cb120-131" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb120-132"><a href="#cb120-132" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb120-133"><a href="#cb120-133" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-134"><a href="#cb120-134" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb120-135"><a href="#cb120-135" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-136"><a href="#cb120-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-137"><a href="#cb120-137" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb120-138"><a href="#cb120-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-139"><a href="#cb120-139" aria-hidden="true" tabindex="-1"></a><span class="fu">### Workflow</span></span>
<span id="cb120-140"><a href="#cb120-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-141"><a href="#cb120-141" aria-hidden="true" tabindex="-1"></a>The workflow consists of the following steps:</span>
<span id="cb120-142"><a href="#cb120-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-143"><a href="#cb120-143" aria-hidden="true" tabindex="-1"></a>| Step | Command                | What weâ€™re doing                                           |</span>
<span id="cb120-144"><a href="#cb120-144" aria-hidden="true" tabindex="-1"></a>|------|------------------------|------------------------------------------------------------|</span>
<span id="cb120-145"><a href="#cb120-145" aria-hidden="true" tabindex="-1"></a>| 1    | multiple               | prepare input file names &amp; paths                           |</span>
<span id="cb120-146"><a href="#cb120-146" aria-hidden="true" tabindex="-1"></a>| 2    | multiple               | Define  primers (all orientations)                         |</span>
<span id="cb120-147"><a href="#cb120-147" aria-hidden="true" tabindex="-1"></a>| 3    | <span class="in">`cutadapt`</span>             | Remove  primers                                            |</span>
<span id="cb120-148"><a href="#cb120-148" aria-hidden="true" tabindex="-1"></a>| 4    | <span class="in">`filterAndTrim()`</span>      | assess quality &amp; filter reads                              |</span>
<span id="cb120-149"><a href="#cb120-149" aria-hidden="true" tabindex="-1"></a>| 5    | <span class="in">`learnErrors()`</span>        | generate an error model for the data                       |</span>
<span id="cb120-150"><a href="#cb120-150" aria-hidden="true" tabindex="-1"></a>| 6    | <span class="in">`derepFastq`</span>           | dereplicate sequences                                      |</span>
<span id="cb120-151"><a href="#cb120-151" aria-hidden="true" tabindex="-1"></a>| 7    | <span class="in">`dada()`</span>               | infer ASVs on both forward &amp; reverse reads independently   |</span>
<span id="cb120-152"><a href="#cb120-152" aria-hidden="true" tabindex="-1"></a>| 8    | <span class="in">`makeSequenceTable()`</span>  | generate a count table                                     |</span>
<span id="cb120-153"><a href="#cb120-153" aria-hidden="true" tabindex="-1"></a>| 9    | <span class="in">`removeBimeraDenovo()`</span> | screen for &amp; remove chimeras                               |</span>
<span id="cb120-154"><a href="#cb120-154" aria-hidden="true" tabindex="-1"></a>| 10   |                        | track reads through workflow                               |</span>
<span id="cb120-155"><a href="#cb120-155" aria-hidden="true" tabindex="-1"></a>| 11   | <span class="in">`assignTaxonomy()`</span>     | assign taxonomy &amp; finish workflow                          |</span>
<span id="cb120-156"><a href="#cb120-156" aria-hidden="true" tabindex="-1"></a>| 12   | <span class="in">`save.image()`</span>         | save an image of seqtab &amp; taxtab for next part of workflow |</span>
<span id="cb120-157"><a href="#cb120-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-158"><a href="#cb120-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-159"><a href="#cb120-159" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1. Set Working Environment</span></span>
<span id="cb120-160"><a href="#cb120-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-161"><a href="#cb120-161" aria-hidden="true" tabindex="-1"></a>Next, we need to setup the working environment by renaming the fastq files and defining a path for the working directory.</span>
<span id="cb120-162"><a href="#cb120-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-163"><a href="#cb120-163" aria-hidden="true" tabindex="-1"></a><span class="fu">### Rename samples</span></span>
<span id="cb120-164"><a href="#cb120-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-165"><a href="#cb120-165" aria-hidden="true" tabindex="-1"></a>To make the parsing easier, we will eliminate the lane and well number from each sample. If you do not wish to do that you will need to adjust the code accordingly. I am *positive* there are more elegant ways of doing this.</span>
<span id="cb120-166"><a href="#cb120-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-167"><a href="#cb120-167" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; CAUTION. If you use this for your own data, please check that this code works on a few backup files before proceeding.</span></span>
<span id="cb120-168"><a href="#cb120-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-171"><a href="#cb120-171" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-172"><a href="#cb120-172" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb120-173"><a href="#cb120-173" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"RAW"</span>)</span>
<span id="cb120-174"><a href="#cb120-174" aria-hidden="true" tabindex="-1"></a>orig_fastq <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">pattern =</span> <span class="st">"*.fastq.gz"</span>)</span>
<span id="cb120-175"><a href="#cb120-175" aria-hidden="true" tabindex="-1"></a>newname_fastq <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_S.*_L001"</span>, <span class="st">""</span>, orig_fastq)</span>
<span id="cb120-176"><a href="#cb120-176" aria-hidden="true" tabindex="-1"></a>newname_fastq <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_001"</span>, <span class="st">""</span>, newname_fastq)</span>
<span id="cb120-177"><a href="#cb120-177" aria-hidden="true" tabindex="-1"></a><span class="fu">file.rename</span>(orig_fastq, newname_fastq)</span>
<span id="cb120-178"><a href="#cb120-178" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../"</span>)</span>
<span id="cb120-179"><a href="#cb120-179" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-180"><a href="#cb120-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-183"><a href="#cb120-183" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-184"><a href="#cb120-184" aria-hidden="true" tabindex="-1"></a>path <span class="ot">&lt;-</span> <span class="st">"RAW"</span></span>
<span id="cb120-185"><a href="#cb120-185" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">list.files</span>(path))</span>
<span id="cb120-186"><a href="#cb120-186" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-187"><a href="#cb120-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-190"><a href="#cb120-190" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-191"><a href="#cb120-191" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-192"><a href="#cb120-192" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-193"><a href="#cb120-193" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="st">"P1_R1.fastq.gz"</span>, <span class="st">"P1_R2.fastq.gz"</span>, <span class="st">"P1-8C_R1.fastq.gz"</span>,</span>
<span id="cb120-194"><a href="#cb120-194" aria-hidden="true" tabindex="-1"></a>  <span class="st">"P1-8C_R2.fastq.gz"</span>, <span class="st">"P10_R1.fastq.gz"</span>, <span class="st">"P10_R2.fastq.gz"</span>)</span>
<span id="cb120-195"><a href="#cb120-195" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-196"><a href="#cb120-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-197"><a href="#cb120-197" aria-hidden="true" tabindex="-1"></a>Then, we generate matched lists of the forward and reverse read files. We also parse out the sample name.</span>
<span id="cb120-198"><a href="#cb120-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-201"><a href="#cb120-201" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-202"><a href="#cb120-202" aria-hidden="true" tabindex="-1"></a>fnFs <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern =</span> <span class="st">"_R1.fastq.gz"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb120-203"><a href="#cb120-203" aria-hidden="true" tabindex="-1"></a>fnRs <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern =</span> <span class="st">"_R2.fastq.gz"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb120-204"><a href="#cb120-204" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-205"><a href="#cb120-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-206"><a href="#cb120-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-209"><a href="#cb120-209" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-210"><a href="#cb120-210" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb120-211"><a href="#cb120-211" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 3</span></span>
<span id="cb120-212"><a href="#cb120-212" aria-hidden="true" tabindex="-1"></a>p1a <span class="ot">&lt;-</span> <span class="fu">plotQualityProfile</span>(fnFs[<span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>], <span class="at">aggregate =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-213"><a href="#cb120-213" aria-hidden="true" tabindex="-1"></a>p2a <span class="ot">&lt;-</span> <span class="fu">plotQualityProfile</span>(fnRs[<span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>], <span class="at">aggregate =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-214"><a href="#cb120-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-215"><a href="#cb120-215" aria-hidden="true" tabindex="-1"></a>p3a <span class="ot">&lt;-</span> <span class="fu">grid.arrange</span>(p1a, p2a, <span class="at">nrow =</span> <span class="dv">1</span>)</span>
<span id="cb120-216"><a href="#cb120-216" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"plot_qscores_2018a.png"</span>, p3a, <span class="at">width =</span> <span class="dv">7</span>, <span class="at">height =</span> <span class="dv">3</span>)</span>
<span id="cb120-217"><a href="#cb120-217" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-218"><a href="#cb120-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-221"><a href="#cb120-221" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-222"><a href="#cb120-222" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-223"><a href="#cb120-223" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">"cp files/dada2/figures/ssu18_plot_qscores_a.png include/dada2/ssu18_plot_qscores_a.png"</span>)</span>
<span id="cb120-224"><a href="#cb120-224" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-225"><a href="#cb120-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-226"><a href="#cb120-226" aria-hidden="true" tabindex="-1"></a>Unfortunately, the reverse reads are unusable.</span>
<span id="cb120-227"><a href="#cb120-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-230"><a href="#cb120-230" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-231"><a href="#cb120-231" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-232"><a href="#cb120-232" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-233"><a href="#cb120-233" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb120-234"><a href="#cb120-234" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 4</span></span>
<span id="cb120-235"><a href="#cb120-235" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb120-236"><a href="#cb120-236" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/dada2/ssu18_plot_qscores_a.png"</span>)</span>
<span id="cb120-237"><a href="#cb120-237" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-238"><a href="#cb120-238" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("qual_scores_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-239"><a href="#cb120-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-240"><a href="#cb120-240" aria-hidden="true" tabindex="-1"></a><span class="fu">## 2. Define Primers</span></span>
<span id="cb120-241"><a href="#cb120-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-242"><a href="#cb120-242" aria-hidden="true" tabindex="-1"></a>Before we start the DADA2 workflow we need to run <span class="co">[</span><span class="ot">catadapt</span><span class="co">](https://github.com/marcelm/cutadapt)</span> <span class="co">[</span><span class="ot">@martin2011cutadapt</span><span class="co">]</span> on all <span class="in">`fastq.gz`</span> files to trim the primers. For bacteria and archaea, we amplified the V4 hypervariable region of the 16S rRNA gene using the primer pair 515F (GTGCCAGCMGCCGCGGTAA) and 806R (GGACTACHVGGGTWTCTAAT) <span class="co">[</span><span class="ot">@caporaso2011global</span><span class="co">]</span>, which should yield an amplicon length of about 253 bp.</span>
<span id="cb120-243"><a href="#cb120-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-244"><a href="#cb120-244" aria-hidden="true" tabindex="-1"></a>First we define the primers.</span>
<span id="cb120-245"><a href="#cb120-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-248"><a href="#cb120-248" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-249"><a href="#cb120-249" aria-hidden="true" tabindex="-1"></a>FWD <span class="ot">&lt;-</span> <span class="st">"GTGCCAGCMGCCGCGGTAA"</span></span>
<span id="cb120-250"><a href="#cb120-250" aria-hidden="true" tabindex="-1"></a>REV <span class="ot">&lt;-</span> <span class="st">"GGACTACHVGGGTWTCTAAT"</span></span>
<span id="cb120-251"><a href="#cb120-251" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-252"><a href="#cb120-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-253"><a href="#cb120-253" aria-hidden="true" tabindex="-1"></a>Next, we check the presence and orientation of these primers in the data. I started doing this for ITS data because of primer read-through but I really like the general idea of doing it just to make sure nothing funny is going of with the data. To do this, we will create all orientations of the input primer sequences. In other words the *Forward*, *Complement*, *Reverse*, and *Reverse Complement* variations.</span>
<span id="cb120-254"><a href="#cb120-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-257"><a href="#cb120-257" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-258"><a href="#cb120-258" aria-hidden="true" tabindex="-1"></a>allOrients <span class="ot">&lt;-</span> <span class="cf">function</span>(primer) {</span>
<span id="cb120-259"><a href="#cb120-259" aria-hidden="true" tabindex="-1"></a>    <span class="fu">require</span>(Biostrings)</span>
<span id="cb120-260"><a href="#cb120-260" aria-hidden="true" tabindex="-1"></a>    dna <span class="ot">&lt;-</span> <span class="fu">DNAString</span>(primer)</span>
<span id="cb120-261"><a href="#cb120-261" aria-hidden="true" tabindex="-1"></a>    orients <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">Forward =</span> dna,</span>
<span id="cb120-262"><a href="#cb120-262" aria-hidden="true" tabindex="-1"></a>                 <span class="at">Complement =</span> <span class="fu">complement</span>(dna),</span>
<span id="cb120-263"><a href="#cb120-263" aria-hidden="true" tabindex="-1"></a>                 <span class="at">Reverse =</span> <span class="fu">reverse</span>(dna),</span>
<span id="cb120-264"><a href="#cb120-264" aria-hidden="true" tabindex="-1"></a>                 <span class="at">RevComp =</span> <span class="fu">reverseComplement</span>(dna))</span>
<span id="cb120-265"><a href="#cb120-265" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">sapply</span>(orients, toString))</span>
<span id="cb120-266"><a href="#cb120-266" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb120-267"><a href="#cb120-267" aria-hidden="true" tabindex="-1"></a>FWD.orients <span class="ot">&lt;-</span> <span class="fu">allOrients</span>(FWD)</span>
<span id="cb120-268"><a href="#cb120-268" aria-hidden="true" tabindex="-1"></a>REV.orients <span class="ot">&lt;-</span> <span class="fu">allOrients</span>(REV)</span>
<span id="cb120-269"><a href="#cb120-269" aria-hidden="true" tabindex="-1"></a>FWD.orients</span>
<span id="cb120-270"><a href="#cb120-270" aria-hidden="true" tabindex="-1"></a>REV.orients</span>
<span id="cb120-271"><a href="#cb120-271" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-272"><a href="#cb120-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-273"><a href="#cb120-273" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-274"><a href="#cb120-274" aria-hidden="true" tabindex="-1"></a><span class="in">              Forward            Complement               Reverse</span></span>
<span id="cb120-275"><a href="#cb120-275" aria-hidden="true" tabindex="-1"></a><span class="in">"GTGCCAGCMGCCGCGGTAA" "CACGGTCGKCGGCGCCATT" "AATGGCGCCGMCGACCGTG"</span></span>
<span id="cb120-276"><a href="#cb120-276" aria-hidden="true" tabindex="-1"></a><span class="in">              RevComp</span></span>
<span id="cb120-277"><a href="#cb120-277" aria-hidden="true" tabindex="-1"></a><span class="in">"TTACCGCGGCKGCTGGCAC"</span></span>
<span id="cb120-278"><a href="#cb120-278" aria-hidden="true" tabindex="-1"></a><span class="in">               Forward             Complement                Reverse</span></span>
<span id="cb120-279"><a href="#cb120-279" aria-hidden="true" tabindex="-1"></a><span class="in">"GGACTACHVGGGTWTCTAAT" "CCTGATGDBCCCAWAGATTA" "TAATCTWTGGGVHCATCAGG"</span></span>
<span id="cb120-280"><a href="#cb120-280" aria-hidden="true" tabindex="-1"></a><span class="in">               RevComp</span></span>
<span id="cb120-281"><a href="#cb120-281" aria-hidden="true" tabindex="-1"></a><span class="in">"ATTAGAWACCCBDGTAGTCC"</span></span>
<span id="cb120-282"><a href="#cb120-282" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-283"><a href="#cb120-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-284"><a href="#cb120-284" aria-hidden="true" tabindex="-1"></a>Now we do a little pre-filter step to eliminate ambiguous bases (*Ns*) because *Ns* make mapping of short primer sequences difficult. This step removes any reads with *Ns*. Again, set some files paths, this time for the filtered reads.</span>
<span id="cb120-285"><a href="#cb120-285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-288"><a href="#cb120-288" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-289"><a href="#cb120-289" aria-hidden="true" tabindex="-1"></a>fnFs.filtN <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">"filtN"</span>, <span class="fu">basename</span>(fnFs))</span>
<span id="cb120-290"><a href="#cb120-290" aria-hidden="true" tabindex="-1"></a><span class="fu">filterAndTrim</span>(fnFs, fnFs.filtN,</span>
<span id="cb120-291"><a href="#cb120-291" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxN =</span> <span class="dv">0</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-292"><a href="#cb120-292" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-293"><a href="#cb120-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-294"><a href="#cb120-294" aria-hidden="true" tabindex="-1"></a>Nice.  Time to assess the number of times a primer (and all primer orientations) appear in the forward and reverse reads. According to the workflow, counting the primers on one set of paired end fastq files is sufficient to see if there is a problem. This assumes that all the files were created using the same library prep. Basically for both primers, we will search for all four orientations in both forward and reverse reads. Since this is 16S rRNA we do not anticipate any issues but it is worth checking anyway.</span>
<span id="cb120-295"><a href="#cb120-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-298"><a href="#cb120-298" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-299"><a href="#cb120-299" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-300"><a href="#cb120-300" aria-hidden="true" tabindex="-1"></a>sampnum <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb120-301"><a href="#cb120-301" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-302"><a href="#cb120-302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-303"><a href="#cb120-303" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-306"><a href="#cb120-306" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-307"><a href="#cb120-307" aria-hidden="true" tabindex="-1"></a>primerHits <span class="ot">&lt;-</span> <span class="cf">function</span>(primer, fn) {</span>
<span id="cb120-308"><a href="#cb120-308" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Counts number of reads in which the primer is found</span></span>
<span id="cb120-309"><a href="#cb120-309" aria-hidden="true" tabindex="-1"></a>    nhits <span class="ot">&lt;-</span> <span class="fu">vcountPattern</span>(primer, <span class="fu">sread</span>(<span class="fu">readFastq</span>(fn)), <span class="at">fixed =</span> <span class="cn">FALSE</span>)</span>
<span id="cb120-310"><a href="#cb120-310" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">sum</span>(nhits <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb120-311"><a href="#cb120-311" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb120-312"><a href="#cb120-312" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-313"><a href="#cb120-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-314"><a href="#cb120-314" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Forward primers</span></span>
<span id="cb120-315"><a href="#cb120-315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-318"><a href="#cb120-318" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-319"><a href="#cb120-319" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits,</span>
<span id="cb120-320"><a href="#cb120-320" aria-hidden="true" tabindex="-1"></a>                                <span class="at">fn =</span> fnFs.filtN[[sampnum]]))</span>
<span id="cb120-321"><a href="#cb120-321" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-322"><a href="#cb120-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-323"><a href="#cb120-323" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-324"><a href="#cb120-324" aria-hidden="true" tabindex="-1"></a><span class="in">                 Forward Complement Reverse RevComp</span></span>
<span id="cb120-325"><a href="#cb120-325" aria-hidden="true" tabindex="-1"></a><span class="in">FWD.ForwardReads   74707          0       0       0</span></span>
<span id="cb120-326"><a href="#cb120-326" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-327"><a href="#cb120-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-328"><a href="#cb120-328" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Reverse primers</span></span>
<span id="cb120-329"><a href="#cb120-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-332"><a href="#cb120-332" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-333"><a href="#cb120-333" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits,</span>
<span id="cb120-334"><a href="#cb120-334" aria-hidden="true" tabindex="-1"></a>                                <span class="at">fn =</span> fnFs.filtN[[sampnum]]))</span>
<span id="cb120-335"><a href="#cb120-335" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-336"><a href="#cb120-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-337"><a href="#cb120-337" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-338"><a href="#cb120-338" aria-hidden="true" tabindex="-1"></a><span class="in">                 Forward Complement Reverse RevComp</span></span>
<span id="cb120-339"><a href="#cb120-339" aria-hidden="true" tabindex="-1"></a><span class="in">REV.ForwardReads       0          0       0     165</span></span>
<span id="cb120-340"><a href="#cb120-340" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-341"><a href="#cb120-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-342"><a href="#cb120-342" aria-hidden="true" tabindex="-1"></a>As expected, forward primers predominantly in the forward reads and very little evidence of reverse primers.</span>
<span id="cb120-343"><a href="#cb120-343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-344"><a href="#cb120-344" aria-hidden="true" tabindex="-1"></a><span class="fu">## 3. Remove Primers</span></span>
<span id="cb120-345"><a href="#cb120-345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-346"><a href="#cb120-346" aria-hidden="true" tabindex="-1"></a>Now we can run <span class="co">[</span><span class="ot">catadapt</span><span class="co">](https://github.com/marcelm/cutadapt)</span> <span class="co">[</span><span class="ot">@martin2011cutadapt</span><span class="co">]</span> to remove the primers from the fastq sequences. A little setup first. If this command executes successfully it means R has found cutadapt.</span>
<span id="cb120-347"><a href="#cb120-347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-350"><a href="#cb120-350" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-351"><a href="#cb120-351" aria-hidden="true" tabindex="-1"></a>cutadapt <span class="ot">&lt;-</span> <span class="st">"/Users/rad/miniconda3/envs/cutadapt/bin/cutadapt"</span></span>
<span id="cb120-352"><a href="#cb120-352" aria-hidden="true" tabindex="-1"></a><span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="st">"--version"</span>) <span class="co"># Run shell commands from R</span></span>
<span id="cb120-353"><a href="#cb120-353" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-354"><a href="#cb120-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-355"><a href="#cb120-355" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-356"><a href="#cb120-356" aria-hidden="true" tabindex="-1"></a><span class="in">2.8</span></span>
<span id="cb120-357"><a href="#cb120-357" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-358"><a href="#cb120-358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-359"><a href="#cb120-359" aria-hidden="true" tabindex="-1"></a>We set paths and trim the forward primer and the reverse-complement of the reverse primer off of R1 (forward reads) and trim the reverse primer and the reverse-complement of the forward primer off of R2 (reverse reads).</span>
<span id="cb120-360"><a href="#cb120-360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-363"><a href="#cb120-363" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-364"><a href="#cb120-364" aria-hidden="true" tabindex="-1"></a>path.cut <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">"cutadapt"</span>)</span>
<span id="cb120-365"><a href="#cb120-365" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(path.cut)) <span class="fu">dir.create</span>(path.cut)</span>
<span id="cb120-366"><a href="#cb120-366" aria-hidden="true" tabindex="-1"></a>fnFs.cut <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(fnFs))</span>
<span id="cb120-367"><a href="#cb120-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-368"><a href="#cb120-368" aria-hidden="true" tabindex="-1"></a>FWD.RC <span class="ot">&lt;-</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(FWD)</span>
<span id="cb120-369"><a href="#cb120-369" aria-hidden="true" tabindex="-1"></a>REV.RC <span class="ot">&lt;-</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(REV)</span>
<span id="cb120-370"><a href="#cb120-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-371"><a href="#cb120-371" aria-hidden="true" tabindex="-1"></a>R1.flags <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"-g"</span>, FWD, <span class="st">"-a"</span>, REV.RC)</span>
<span id="cb120-372"><a href="#cb120-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-373"><a href="#cb120-373" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(fnFs)) {<span class="fu">system2</span>(cutadapt,</span>
<span id="cb120-374"><a href="#cb120-374" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">args =</span> <span class="fu">c</span>(R1.flags,</span>
<span id="cb120-375"><a href="#cb120-375" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">"-m"</span>, <span class="dv">20</span>, <span class="st">"-n"</span>, <span class="dv">2</span>, <span class="st">"-e"</span>, <span class="fl">0.12</span>,</span>
<span id="cb120-376"><a href="#cb120-376" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">"-o"</span>, fnFs.cut[i],</span>
<span id="cb120-377"><a href="#cb120-377" aria-hidden="true" tabindex="-1"></a>                                            fnFs.filtN[i]))}</span>
<span id="cb120-378"><a href="#cb120-378" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-379"><a href="#cb120-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-380"><a href="#cb120-380" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-381"><a href="#cb120-381" aria-hidden="true" tabindex="-1"></a><span class="in">This is cutadapt 2.8 with Python 3.7.6</span></span>
<span id="cb120-382"><a href="#cb120-382" aria-hidden="true" tabindex="-1"></a><span class="in">Command line parameters: -g GTGCCAGCMGCCGCGGTAA -a ATTAGAWACCCBDGTAGTCC \</span></span>
<span id="cb120-383"><a href="#cb120-383" aria-hidden="true" tabindex="-1"></a><span class="in">                         -m 20 -n 2 -e 0.12 \</span></span>
<span id="cb120-384"><a href="#cb120-384" aria-hidden="true" tabindex="-1"></a><span class="in">                         -o RAW/cutadapt/P1_R1.fastq.gz RAW/filtN/P1_R1.fastq.gz</span></span>
<span id="cb120-385"><a href="#cb120-385" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-386"><a href="#cb120-386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-387"><a href="#cb120-387" aria-hidden="true" tabindex="-1"></a>*Note*. If the code above removes all of the base pairs in a sequence, you will get downstream errors unless you set the `-m` flag. This flag sets the minimum length and reads shorter than this will be discarded. Without this flag, reads of length 0 will be kept and cause issues. *Also, a lot of output will be written to the screen by cutadapt!*.</span>
<span id="cb120-388"><a href="#cb120-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-389"><a href="#cb120-389" aria-hidden="true" tabindex="-1"></a>We can now count the number of primers in the sequences from the output of cutadapt.</span>
<span id="cb120-390"><a href="#cb120-390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-393"><a href="#cb120-393" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-394"><a href="#cb120-394" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(</span>
<span id="cb120-395"><a href="#cb120-395" aria-hidden="true" tabindex="-1"></a>  <span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[sampnum]]),</span>
<span id="cb120-396"><a href="#cb120-396" aria-hidden="true" tabindex="-1"></a>  <span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[sampnum]]))</span>
<span id="cb120-397"><a href="#cb120-397" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-398"><a href="#cb120-398" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-399"><a href="#cb120-399" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-400"><a href="#cb120-400" aria-hidden="true" tabindex="-1"></a><span class="in">                 Forward Complement Reverse RevComp</span></span>
<span id="cb120-401"><a href="#cb120-401" aria-hidden="true" tabindex="-1"></a><span class="in">FWD.ForwardReads       0          0       0       0</span></span>
<span id="cb120-402"><a href="#cb120-402" aria-hidden="true" tabindex="-1"></a><span class="in">REV.ForwardReads       0          0       0       0</span></span>
<span id="cb120-403"><a href="#cb120-403" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-404"><a href="#cb120-404" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-405"><a href="#cb120-405" aria-hidden="true" tabindex="-1"></a>Basically, primers are no longer detected in the cutadapted reads. Now, for each sample, we can take a look at how the pre-filtering step and primer removal affected the total number of raw reads.</span>
<span id="cb120-406"><a href="#cb120-406" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-409"><a href="#cb120-409" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-410"><a href="#cb120-410" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-411"><a href="#cb120-411" aria-hidden="true" tabindex="-1"></a>temp1 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"files/dada2/tables/ssu18_read_changes.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-412"><a href="#cb120-412" aria-hidden="true" tabindex="-1"></a>temp1[, <span class="dv">4</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="cn">NULL</span>)</span>
<span id="cb120-413"><a href="#cb120-413" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(temp1, <span class="st">"files/dada2/tables/ssu18_cut.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb120-414"><a href="#cb120-414" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-415"><a href="#cb120-415" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-418"><a href="#cb120-418" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-419"><a href="#cb120-419" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-420"><a href="#cb120-420" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-421"><a href="#cb120-421" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"files/dada2/tables/ssu18_cut.txt"</span>,</span>
<span id="cb120-422"><a href="#cb120-422" aria-hidden="true" tabindex="-1"></a>                       <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb120-423"><a href="#cb120-423" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-424"><a href="#cb120-424" aria-hidden="true" tabindex="-1"></a>seq_table<span class="sc">$</span>per_reads_kept <span class="ot">&lt;-</span> <span class="fu">round</span>(seq_table<span class="sc">$</span>cutadapt_reads<span class="sc">/</span>seq_table<span class="sc">$</span>raw_reads, </span>
<span id="cb120-425"><a href="#cb120-425" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb120-426"><a href="#cb120-426" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-427"><a href="#cb120-427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-428"><a href="#cb120-428" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab("seq_cutadapt_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-429"><a href="#cb120-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-432"><a href="#cb120-432" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-433"><a href="#cb120-433" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-434"><a href="#cb120-434" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-435"><a href="#cb120-435" aria-hidden="true" tabindex="-1"></a>seq_table <span class="sc">%&gt;%</span></span>
<span id="cb120-436"><a href="#cb120-436" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb120-437"><a href="#cb120-437" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"ssu18_cutadapt"</span>,</span>
<span id="cb120-438"><a href="#cb120-438" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb120-439"><a href="#cb120-439" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb120-440"><a href="#cb120-440" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb120-441"><a href="#cb120-441" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb120-442"><a href="#cb120-442" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-443"><a href="#cb120-443" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb120-444"><a href="#cb120-444" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-445"><a href="#cb120-445" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb120-446"><a href="#cb120-446" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb120-447"><a href="#cb120-447" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb120-448"><a href="#cb120-448" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">1</span>),</span>
<span id="cb120-449"><a href="#cb120-449" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">FALSE</span>, <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-450"><a href="#cb120-450" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb120-451"><a href="#cb120-451" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-452"><a href="#cb120-452" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb120-453"><a href="#cb120-453" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Sample ID"</span>, </span>
<span id="cb120-454"><a href="#cb120-454" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb120-455"><a href="#cb120-455" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>),</span>
<span id="cb120-456"><a href="#cb120-456" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>), </span>
<span id="cb120-457"><a href="#cb120-457" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb120-458"><a href="#cb120-458" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">150</span>, <span class="at">footer =</span> <span class="st">"Total reads"</span>), </span>
<span id="cb120-459"><a href="#cb120-459" aria-hidden="true" tabindex="-1"></a>    <span class="at">raw_reads =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-460"><a href="#cb120-460" aria-hidden="true" tabindex="-1"></a>    <span class="at">pre_filt_reads =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-461"><a href="#cb120-461" aria-hidden="true" tabindex="-1"></a>    <span class="at">cutadapt_reads =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values)))</span>
<span id="cb120-462"><a href="#cb120-462" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-463"><a href="#cb120-463" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">5</span>, </span>
<span id="cb120-464"><a href="#cb120-464" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb120-465"><a href="#cb120-465" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb120-466"><a href="#cb120-466" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb120-467"><a href="#cb120-467" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-468"><a href="#cb120-468" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb120-469"><a href="#cb120-469" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-470"><a href="#cb120-470" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-471"><a href="#cb120-471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-472"><a href="#cb120-472" aria-hidden="true" tabindex="-1"></a><span class="fu">## 4. Quality Assessment &amp; Filtering</span></span>
<span id="cb120-473"><a href="#cb120-473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-474"><a href="#cb120-474" aria-hidden="true" tabindex="-1"></a>We need the forward and reverse fastq file names plus the sample names.</span>
<span id="cb120-475"><a href="#cb120-475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-478"><a href="#cb120-478" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-479"><a href="#cb120-479" aria-hidden="true" tabindex="-1"></a>cutFs <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path.cut, <span class="at">pattern =</span> <span class="st">"_R1.fastq.gz"</span>,</span>
<span id="cb120-480"><a href="#cb120-480" aria-hidden="true" tabindex="-1"></a>                         <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb120-481"><a href="#cb120-481" aria-hidden="true" tabindex="-1"></a>get.sample.name <span class="ot">&lt;-</span> <span class="cf">function</span>(fname) <span class="fu">strsplit</span>(<span class="fu">basename</span>(fname), <span class="st">"_"</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]</span>
<span id="cb120-482"><a href="#cb120-482" aria-hidden="true" tabindex="-1"></a>sample.names <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">sapply</span>(cutFs, get.sample.name))</span>
<span id="cb120-483"><a href="#cb120-483" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sample.names)</span>
<span id="cb120-484"><a href="#cb120-484" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-485"><a href="#cb120-485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-488"><a href="#cb120-488" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-489"><a href="#cb120-489" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-490"><a href="#cb120-490" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="st">"P1"</span>, <span class="st">"P1-8C"</span>, <span class="st">"P10"</span>, <span class="st">"P2"</span>, <span class="st">"P3"</span>, <span class="st">"P3-8C"</span>)</span>
<span id="cb120-491"><a href="#cb120-491" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-492"><a href="#cb120-492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-493"><a href="#cb120-493" aria-hidden="true" tabindex="-1"></a>First let's look at the quality of our reads. The numbers in brackets specify which samples to view. Here we are looking at an aggregate plot of all data (except the negative control).</span>
<span id="cb120-494"><a href="#cb120-494" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-495"><a href="#cb120-495" aria-hidden="true" tabindex="-1"></a><span class="fu">### Quality plots</span></span>
<span id="cb120-496"><a href="#cb120-496" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-499"><a href="#cb120-499" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-500"><a href="#cb120-500" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb120-501"><a href="#cb120-501" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">plotQualityProfile</span>(cutFs[<span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>], <span class="at">aggregate =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-502"><a href="#cb120-502" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"ssu18_plot_qscores.png"</span>, p1, <span class="at">width =</span> <span class="dv">7</span>, <span class="at">height =</span> <span class="dv">3</span>)</span>
<span id="cb120-503"><a href="#cb120-503" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"ssu18_plot_qscores.png"</span>, p1)</span>
<span id="cb120-504"><a href="#cb120-504" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-505"><a href="#cb120-505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-508"><a href="#cb120-508" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-509"><a href="#cb120-509" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-510"><a href="#cb120-510" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">"cp files/dada2/figures/ssu18_plot_qscores_b.png include/dada2/ssu18_plot_qscores_b.png"</span>)</span>
<span id="cb120-511"><a href="#cb120-511" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-512"><a href="#cb120-512" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-515"><a href="#cb120-515" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-516"><a href="#cb120-516" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-517"><a href="#cb120-517" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-518"><a href="#cb120-518" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb120-519"><a href="#cb120-519" aria-hidden="true" tabindex="-1"></a><span class="co">#| out-width: 55%</span></span>
<span id="cb120-520"><a href="#cb120-520" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-align: center</span></span>
<span id="cb120-521"><a href="#cb120-521" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-522"><a href="#cb120-522" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-523"><a href="#cb120-523" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/dada2/ssu18_plot_qscores_b.png"</span>)</span>
<span id="cb120-524"><a href="#cb120-524" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-525"><a href="#cb120-525" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("qual_scores_after_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-526"><a href="#cb120-526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-527"><a href="#cb120-527" aria-hidden="true" tabindex="-1"></a>The forward reads look good.</span>
<span id="cb120-528"><a href="#cb120-528" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-529"><a href="#cb120-529" aria-hidden="true" tabindex="-1"></a><span class="fu">### Filtering</span></span>
<span id="cb120-530"><a href="#cb120-530" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-531"><a href="#cb120-531" aria-hidden="true" tabindex="-1"></a>First, we again make some path variables and setup a new directory of filtered reads.</span>
<span id="cb120-532"><a href="#cb120-532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-535"><a href="#cb120-535" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-536"><a href="#cb120-536" aria-hidden="true" tabindex="-1"></a>filtFs <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut, <span class="st">"filtered"</span>, <span class="fu">basename</span>(cutFs))</span>
<span id="cb120-537"><a href="#cb120-537" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-538"><a href="#cb120-538" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-539"><a href="#cb120-539" aria-hidden="true" tabindex="-1"></a>And then we trim the reads.</span>
<span id="cb120-540"><a href="#cb120-540" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-543"><a href="#cb120-543" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-544"><a href="#cb120-544" aria-hidden="true" tabindex="-1"></a>out <span class="ot">&lt;-</span> <span class="fu">filterAndTrim</span>(cutFs, filtFs,</span>
<span id="cb120-545"><a href="#cb120-545" aria-hidden="true" tabindex="-1"></a>                     <span class="at">maxN =</span> <span class="dv">0</span>, <span class="at">maxEE =</span> <span class="dv">2</span>,</span>
<span id="cb120-546"><a href="#cb120-546" aria-hidden="true" tabindex="-1"></a>                     <span class="at">truncQ =</span> <span class="dv">2</span>, <span class="at">rm.phix =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-547"><a href="#cb120-547" aria-hidden="true" tabindex="-1"></a>                     <span class="at">compress =</span> <span class="cn">TRUE</span>, <span class="at">multithread =</span> <span class="dv">20</span>)</span>
<span id="cb120-548"><a href="#cb120-548" aria-hidden="true" tabindex="-1"></a>out</span>
<span id="cb120-549"><a href="#cb120-549" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-550"><a href="#cb120-550" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-551"><a href="#cb120-551" aria-hidden="true" tabindex="-1"></a>:::{.callout-note}</span>
<span id="cb120-552"><a href="#cb120-552" aria-hidden="true" tabindex="-1"></a>These parameters should be set based on the anticipated length of the amplicon and the read quality.</span>
<span id="cb120-553"><a href="#cb120-553" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb120-554"><a href="#cb120-554" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-555"><a href="#cb120-555" aria-hidden="true" tabindex="-1"></a>And here is a table of how the filtering step affected the number of reads in each sample. As you can see, there are a few samples that started with a low read count to begin with---we will likely remove those samples at some point.</span>
<span id="cb120-556"><a href="#cb120-556" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-559"><a href="#cb120-559" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-560"><a href="#cb120-560" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-561"><a href="#cb120-561" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-562"><a href="#cb120-562" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"files/dada2/tables/ssu18_filter.txt"</span>,</span>
<span id="cb120-563"><a href="#cb120-563" aria-hidden="true" tabindex="-1"></a>                       <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb120-564"><a href="#cb120-564" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-565"><a href="#cb120-565" aria-hidden="true" tabindex="-1"></a>seq_table<span class="sc">$</span>per_reads_kept <span class="ot">&lt;-</span> <span class="fu">round</span>(seq_table<span class="sc">$</span>reads_out<span class="sc">/</span>seq_table<span class="sc">$</span>reads_in, </span>
<span id="cb120-566"><a href="#cb120-566" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb120-567"><a href="#cb120-567" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-568"><a href="#cb120-568" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab("filter_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-569"><a href="#cb120-569" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-572"><a href="#cb120-572" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-573"><a href="#cb120-573" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-574"><a href="#cb120-574" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-575"><a href="#cb120-575" aria-hidden="true" tabindex="-1"></a>seq_table <span class="sc">%&gt;%</span></span>
<span id="cb120-576"><a href="#cb120-576" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb120-577"><a href="#cb120-577" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"ssu18_filter"</span>,</span>
<span id="cb120-578"><a href="#cb120-578" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb120-579"><a href="#cb120-579" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb120-580"><a href="#cb120-580" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb120-581"><a href="#cb120-581" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb120-582"><a href="#cb120-582" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-583"><a href="#cb120-583" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb120-584"><a href="#cb120-584" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-585"><a href="#cb120-585" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb120-586"><a href="#cb120-586" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb120-587"><a href="#cb120-587" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb120-588"><a href="#cb120-588" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">1</span>),</span>
<span id="cb120-589"><a href="#cb120-589" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">FALSE</span>, <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-590"><a href="#cb120-590" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb120-591"><a href="#cb120-591" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-592"><a href="#cb120-592" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb120-593"><a href="#cb120-593" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Sample ID"</span>, </span>
<span id="cb120-594"><a href="#cb120-594" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb120-595"><a href="#cb120-595" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>),</span>
<span id="cb120-596"><a href="#cb120-596" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>), </span>
<span id="cb120-597"><a href="#cb120-597" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb120-598"><a href="#cb120-598" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">150</span>, <span class="at">footer =</span> <span class="st">"Total reads"</span>), </span>
<span id="cb120-599"><a href="#cb120-599" aria-hidden="true" tabindex="-1"></a>    <span class="at">reads_in =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-600"><a href="#cb120-600" aria-hidden="true" tabindex="-1"></a>    <span class="at">reads_out =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values)))</span>
<span id="cb120-601"><a href="#cb120-601" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-602"><a href="#cb120-602" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">5</span>, </span>
<span id="cb120-603"><a href="#cb120-603" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb120-604"><a href="#cb120-604" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb120-605"><a href="#cb120-605" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb120-606"><a href="#cb120-606" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-607"><a href="#cb120-607" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb120-608"><a href="#cb120-608" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-609"><a href="#cb120-609" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-610"><a href="#cb120-610" aria-hidden="true" tabindex="-1"></a><span class="fu">## 5. Learn Error Rates</span></span>
<span id="cb120-611"><a href="#cb120-611" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-612"><a href="#cb120-612" aria-hidden="true" tabindex="-1"></a>Now it is time to assess the error rate of the data. The DADA2 algorithm uses a parametric error model. Every amplicon data set has a different set of error rates and the <span class="in">`learnErrors`</span> method learns this error model *from the data*. It does this by alternating estimation of the error rates and inference of sample composition until they converge on a jointly consistent solution. The algorithm begins with an initial guess, for which the maximum possible error rates in the data are used.</span>
<span id="cb120-613"><a href="#cb120-613" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-614"><a href="#cb120-614" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Forward Reads</span></span>
<span id="cb120-615"><a href="#cb120-615" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-618"><a href="#cb120-618" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-619"><a href="#cb120-619" aria-hidden="true" tabindex="-1"></a>errF <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(filtFs, <span class="at">multithread =</span> <span class="cn">FALSE</span>)</span>
<span id="cb120-620"><a href="#cb120-620" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-621"><a href="#cb120-621" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-622"><a href="#cb120-622" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-623"><a href="#cb120-623" aria-hidden="true" tabindex="-1"></a><span class="in">103002523 total bases in 443994 reads from 6 samples will be used for learning the error rates.</span></span>
<span id="cb120-624"><a href="#cb120-624" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-625"><a href="#cb120-625" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-628"><a href="#cb120-628" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-629"><a href="#cb120-629" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb120-630"><a href="#cb120-630" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errF, <span class="at">nominalQ=</span><span class="cn">TRUE</span>)</span>
<span id="cb120-631"><a href="#cb120-631" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-632"><a href="#cb120-632" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-635"><a href="#cb120-635" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-636"><a href="#cb120-636" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-637"><a href="#cb120-637" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">"cp files/dada2/figures/ssu18_plot_errorF.png include/dada2/ssu18_plot_errorF.png"</span>)</span>
<span id="cb120-638"><a href="#cb120-638" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-639"><a href="#cb120-639" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-642"><a href="#cb120-642" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-643"><a href="#cb120-643" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-644"><a href="#cb120-644" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-645"><a href="#cb120-645" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/dada2/ssu18_plot_errorF.png"</span>)</span>
<span id="cb120-646"><a href="#cb120-646" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-647"><a href="#cb120-647" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("error_F_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-648"><a href="#cb120-648" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-649"><a href="#cb120-649" aria-hidden="true" tabindex="-1"></a>The error rates for each possible transition (A to C, A to G, etc.) are shown. Points are the observed error rates for each consensus quality score. The black line shows the estimated error rates after convergence of the machine-learning algorithm. The red line shows the error rates expected under the nominal definition of the Q-score. Here the estimated error rates (black line) are a good fit to the observed rates (points), and the error rates drop with increased quality as expected.</span>
<span id="cb120-650"><a href="#cb120-650" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-651"><a href="#cb120-651" aria-hidden="true" tabindex="-1"></a><span class="fu">## 6. Dereplicate Reads</span></span>
<span id="cb120-652"><a href="#cb120-652" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-653"><a href="#cb120-653" aria-hidden="true" tabindex="-1"></a>Now we can use <span class="in">`derepFastq`</span> to identify the unique sequences in the forward and reverse fastq files.</span>
<span id="cb120-654"><a href="#cb120-654" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-655"><a href="#cb120-655" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Forward Reads</span></span>
<span id="cb120-656"><a href="#cb120-656" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-659"><a href="#cb120-659" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-660"><a href="#cb120-660" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-661"><a href="#cb120-661" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-662"><a href="#cb120-662" aria-hidden="true" tabindex="-1"></a><span class="co"># TO run and cache the derepFastq you must set</span></span>
<span id="cb120-663"><a href="#cb120-663" aria-hidden="true" tabindex="-1"></a><span class="co"># `cache.lazy=FALSE`</span></span>
<span id="cb120-664"><a href="#cb120-664" aria-hidden="true" tabindex="-1"></a><span class="co"># For both code blocks</span></span>
<span id="cb120-665"><a href="#cb120-665" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-666"><a href="#cb120-666" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-669"><a href="#cb120-669" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-670"><a href="#cb120-670" aria-hidden="true" tabindex="-1"></a>derepFs <span class="ot">&lt;-</span> <span class="fu">derepFastq</span>(filtFs)</span>
<span id="cb120-671"><a href="#cb120-671" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(derepFs) <span class="ot">&lt;-</span> sample.names</span>
<span id="cb120-672"><a href="#cb120-672" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-673"><a href="#cb120-673" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-674"><a href="#cb120-674" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb120-675"><a href="#cb120-675" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Detailed results of <span class="in">`derep`</span> forward reads<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb120-676"><a href="#cb120-676" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;pre&gt;&lt;code&gt;</span></span>
<span id="cb120-677"><a href="#cb120-677" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P1_R1.fastq.gz</span>
<span id="cb120-678"><a href="#cb120-678" aria-hidden="true" tabindex="-1"></a>Encountered 39950 unique sequences from 79804 total sequences read.</span>
<span id="cb120-679"><a href="#cb120-679" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P1-8C_R1.fastq.gz</span>
<span id="cb120-680"><a href="#cb120-680" aria-hidden="true" tabindex="-1"></a>Encountered 31163 unique sequences from 72760 total sequences read.</span>
<span id="cb120-681"><a href="#cb120-681" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P10_R1.fastq.gz</span>
<span id="cb120-682"><a href="#cb120-682" aria-hidden="true" tabindex="-1"></a>Encountered 35274 unique sequences from 62598 total sequences read.</span>
<span id="cb120-683"><a href="#cb120-683" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P2_R1.fastq.gz</span>
<span id="cb120-684"><a href="#cb120-684" aria-hidden="true" tabindex="-1"></a>Encountered 42119 unique sequences from 81240 total sequences read.</span>
<span id="cb120-685"><a href="#cb120-685" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P3_R1.fastq.gz</span>
<span id="cb120-686"><a href="#cb120-686" aria-hidden="true" tabindex="-1"></a>Encountered 40336 unique sequences from 86732 total sequences read.</span>
<span id="cb120-687"><a href="#cb120-687" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P3-8C_R1.fastq.gz</span>
<span id="cb120-688"><a href="#cb120-688" aria-hidden="true" tabindex="-1"></a>Encountered 31104 unique sequences from 60860 total sequences read.</span>
<span id="cb120-689"><a href="#cb120-689" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P4_R1.fastq.gz</span>
<span id="cb120-690"><a href="#cb120-690" aria-hidden="true" tabindex="-1"></a>Encountered 35169 unique sequences from 63037 total sequences read.</span>
<span id="cb120-691"><a href="#cb120-691" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P5_R1.fastq.gz</span>
<span id="cb120-692"><a href="#cb120-692" aria-hidden="true" tabindex="-1"></a>Encountered 27752 unique sequences from 54127 total sequences read.</span>
<span id="cb120-693"><a href="#cb120-693" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P5-8C_R1.fastq.gz</span>
<span id="cb120-694"><a href="#cb120-694" aria-hidden="true" tabindex="-1"></a>Encountered 49045 unique sequences from 109421 total sequences read.</span>
<span id="cb120-695"><a href="#cb120-695" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P6_R1.fastq.gz</span>
<span id="cb120-696"><a href="#cb120-696" aria-hidden="true" tabindex="-1"></a>Encountered 46853 unique sequences from 94740 total sequences read.</span>
<span id="cb120-697"><a href="#cb120-697" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P7_R1.fastq.gz</span>
<span id="cb120-698"><a href="#cb120-698" aria-hidden="true" tabindex="-1"></a>Encountered 35363 unique sequences from 71678 total sequences read.</span>
<span id="cb120-699"><a href="#cb120-699" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P7-8C_R1.fastq.gz</span>
<span id="cb120-700"><a href="#cb120-700" aria-hidden="true" tabindex="-1"></a>Encountered 35567 unique sequences from 73639 total sequences read.</span>
<span id="cb120-701"><a href="#cb120-701" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P8_R1.fastq.gz</span>
<span id="cb120-702"><a href="#cb120-702" aria-hidden="true" tabindex="-1"></a>Encountered 38518 unique sequences from 72263 total sequences read.</span>
<span id="cb120-703"><a href="#cb120-703" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P9_R1.fastq.gz</span>
<span id="cb120-704"><a href="#cb120-704" aria-hidden="true" tabindex="-1"></a>Encountered 43426 unique sequences from 89315 total sequences read.</span>
<span id="cb120-705"><a href="#cb120-705" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P9-8C_R1.fastq.gz</span>
<span id="cb120-706"><a href="#cb120-706" aria-hidden="true" tabindex="-1"></a>Encountered 13673 unique sequences from 33000 total sequences read.</span>
<span id="cb120-707"><a href="#cb120-707" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/code&gt;&lt;/pre&gt;</span></span>
<span id="cb120-708"><a href="#cb120-708" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb120-709"><a href="#cb120-709" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-710"><a href="#cb120-710" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br/&gt;</span></span>
<span id="cb120-711"><a href="#cb120-711" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-712"><a href="#cb120-712" aria-hidden="true" tabindex="-1"></a><span class="fu">## 7. DADA2 &amp; ASV Inference</span></span>
<span id="cb120-713"><a href="#cb120-713" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-714"><a href="#cb120-714" aria-hidden="true" tabindex="-1"></a>At this point we are ready to apply the core sample inference algorithm (<span class="in">`dada`</span>) to the filtered and trimmed sequence data. DADA2 offers three options for whether and how to pool samples for ASV inference. </span>
<span id="cb120-715"><a href="#cb120-715" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-716"><a href="#cb120-716" aria-hidden="true" tabindex="-1"></a>If <span class="in">`pool = TRUE`</span>, the algorithm will pool together all samples prior to sample inference.  </span>
<span id="cb120-717"><a href="#cb120-717" aria-hidden="true" tabindex="-1"></a>If <span class="in">`pool = FALSE`</span>, sample inference is performed on each sample individually.  </span>
<span id="cb120-718"><a href="#cb120-718" aria-hidden="true" tabindex="-1"></a>If <span class="in">`pool = "pseudo"`</span>, the algorithm will perform pseudo-pooling between individually processed samples.</span>
<span id="cb120-719"><a href="#cb120-719" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-720"><a href="#cb120-720" aria-hidden="true" tabindex="-1"></a>We tested all three methods through the full pipeline. Click the <span class="in">`+`</span> to see the results of the test. For our final analysis, we chose <span class="in">`pool = FALSE`</span> for this data set. </span>
<span id="cb120-721"><a href="#cb120-721" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-722"><a href="#cb120-722" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb120-723"><a href="#cb120-723" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show/hide  Results of <span class="in">`dada`</span> pooling options <span class="kw">&lt;/summary&gt;</span></span>
<span id="cb120-724"><a href="#cb120-724" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-725"><a href="#cb120-725" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br/&gt;</span></span>
<span id="cb120-726"><a href="#cb120-726" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-727"><a href="#cb120-727" aria-hidden="true" tabindex="-1"></a>Here are summary tables of results from the tests. Values are from the final sequence table produced by each option. </span>
<span id="cb120-728"><a href="#cb120-728" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-729"><a href="#cb120-729" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-732"><a href="#cb120-732" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-733"><a href="#cb120-733" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-734"><a href="#cb120-734" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-735"><a href="#cb120-735" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"files/dada2/tables/ssu18_dada_compare.txt"</span>,</span>
<span id="cb120-736"><a href="#cb120-736" aria-hidden="true" tabindex="-1"></a>                        <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb120-737"><a href="#cb120-737" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-738"><a href="#cb120-738" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab("pooling_summary_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-739"><a href="#cb120-739" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-742"><a href="#cb120-742" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-743"><a href="#cb120-743" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-744"><a href="#cb120-744" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-745"><a href="#cb120-745" aria-hidden="true" tabindex="-1"></a>seq_table <span class="sc">%&gt;%</span></span>
<span id="cb120-746"><a href="#cb120-746" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb120-747"><a href="#cb120-747" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"ssu18_pooling_summary"</span>,</span>
<span id="cb120-748"><a href="#cb120-748" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb120-749"><a href="#cb120-749" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb120-750"><a href="#cb120-750" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb120-751"><a href="#cb120-751" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb120-752"><a href="#cb120-752" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-753"><a href="#cb120-753" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb120-754"><a href="#cb120-754" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-755"><a href="#cb120-755" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb120-756"><a href="#cb120-756" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb120-757"><a href="#cb120-757" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb120-758"><a href="#cb120-758" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">1</span>),</span>
<span id="cb120-759"><a href="#cb120-759" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">FALSE</span>, <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-760"><a href="#cb120-760" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb120-761"><a href="#cb120-761" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-762"><a href="#cb120-762" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">5</span>, </span>
<span id="cb120-763"><a href="#cb120-763" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb120-764"><a href="#cb120-764" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb120-765"><a href="#cb120-765" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb120-766"><a href="#cb120-766" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-767"><a href="#cb120-767" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb120-768"><a href="#cb120-768" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-769"><a href="#cb120-769" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-770"><a href="#cb120-770" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-773"><a href="#cb120-773" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-774"><a href="#cb120-774" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-775"><a href="#cb120-775" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-776"><a href="#cb120-776" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"files/dada2/tables/ssu18_dada_compare_by_samp.txt"</span>,</span>
<span id="cb120-777"><a href="#cb120-777" aria-hidden="true" tabindex="-1"></a>                        <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb120-778"><a href="#cb120-778" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-779"><a href="#cb120-779" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab("pooling_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-780"><a href="#cb120-780" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-783"><a href="#cb120-783" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-784"><a href="#cb120-784" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-785"><a href="#cb120-785" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-786"><a href="#cb120-786" aria-hidden="true" tabindex="-1"></a>seq_table <span class="sc">%&gt;%</span></span>
<span id="cb120-787"><a href="#cb120-787" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb120-788"><a href="#cb120-788" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"ssu18_pooling_results"</span>,</span>
<span id="cb120-789"><a href="#cb120-789" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb120-790"><a href="#cb120-790" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb120-791"><a href="#cb120-791" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb120-792"><a href="#cb120-792" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb120-793"><a href="#cb120-793" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-794"><a href="#cb120-794" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb120-795"><a href="#cb120-795" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-796"><a href="#cb120-796" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb120-797"><a href="#cb120-797" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb120-798"><a href="#cb120-798" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb120-799"><a href="#cb120-799" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">1</span>),</span>
<span id="cb120-800"><a href="#cb120-800" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">minWidth =</span> <span class="dv">150</span>, <span class="at">filterable =</span> <span class="cn">FALSE</span>, </span>
<span id="cb120-801"><a href="#cb120-801" aria-hidden="true" tabindex="-1"></a>    <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-802"><a href="#cb120-802" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb120-803"><a href="#cb120-803" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-804"><a href="#cb120-804" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb120-805"><a href="#cb120-805" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Sample ID"</span>, </span>
<span id="cb120-806"><a href="#cb120-806" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb120-807"><a href="#cb120-807" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>),</span>
<span id="cb120-808"><a href="#cb120-808" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>), </span>
<span id="cb120-809"><a href="#cb120-809" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb120-810"><a href="#cb120-810" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">150</span>) </span>
<span id="cb120-811"><a href="#cb120-811" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-812"><a href="#cb120-812" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">5</span>, </span>
<span id="cb120-813"><a href="#cb120-813" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb120-814"><a href="#cb120-814" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb120-815"><a href="#cb120-815" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb120-816"><a href="#cb120-816" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-817"><a href="#cb120-817" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb120-818"><a href="#cb120-818" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-819"><a href="#cb120-819" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-820"><a href="#cb120-820" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb120-821"><a href="#cb120-821" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-822"><a href="#cb120-822" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-825"><a href="#cb120-825" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-826"><a href="#cb120-826" aria-hidden="true" tabindex="-1"></a>dadaFs <span class="ot">&lt;-</span> <span class="fu">dada</span>(derepFs, <span class="at">err =</span> errF, <span class="at">multithread =</span> <span class="cn">TRUE</span>, <span class="at">pool =</span> <span class="cn">FALSE</span>)</span>
<span id="cb120-827"><a href="#cb120-827" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-828"><a href="#cb120-828" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-829"><a href="#cb120-829" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb120-830"><a href="#cb120-830" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Detailed results of <span class="in">`dada`</span> on forward reads<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb120-831"><a href="#cb120-831" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;pre&gt;&lt;code&gt;</span></span>
<span id="cb120-832"><a href="#cb120-832" aria-hidden="true" tabindex="-1"></a>Sample 1 - 79804 reads in 39950 unique sequences.</span>
<span id="cb120-833"><a href="#cb120-833" aria-hidden="true" tabindex="-1"></a>Sample 2 - 72760 reads in 31163 unique sequences.</span>
<span id="cb120-834"><a href="#cb120-834" aria-hidden="true" tabindex="-1"></a>Sample 3 - 62598 reads in 35274 unique sequences.</span>
<span id="cb120-835"><a href="#cb120-835" aria-hidden="true" tabindex="-1"></a>Sample 4 - 81240 reads in 42119 unique sequences.</span>
<span id="cb120-836"><a href="#cb120-836" aria-hidden="true" tabindex="-1"></a>Sample 5 - 86732 reads in 40336 unique sequences.</span>
<span id="cb120-837"><a href="#cb120-837" aria-hidden="true" tabindex="-1"></a>Sample 6 - 60860 reads in 31104 unique sequences.</span>
<span id="cb120-838"><a href="#cb120-838" aria-hidden="true" tabindex="-1"></a>Sample 7 - 63037 reads in 35169 unique sequences.</span>
<span id="cb120-839"><a href="#cb120-839" aria-hidden="true" tabindex="-1"></a>Sample 8 - 54127 reads in 27752 unique sequences.</span>
<span id="cb120-840"><a href="#cb120-840" aria-hidden="true" tabindex="-1"></a>Sample 9 - 109421 reads in 49045 unique sequences.</span>
<span id="cb120-841"><a href="#cb120-841" aria-hidden="true" tabindex="-1"></a>Sample 10 - 94740 reads in 46853 unique sequences.</span>
<span id="cb120-842"><a href="#cb120-842" aria-hidden="true" tabindex="-1"></a>Sample 11 - 71678 reads in 35363 unique sequences.</span>
<span id="cb120-843"><a href="#cb120-843" aria-hidden="true" tabindex="-1"></a>Sample 12 - 73639 reads in 35567 unique sequences.</span>
<span id="cb120-844"><a href="#cb120-844" aria-hidden="true" tabindex="-1"></a>Sample 13 - 72263 reads in 38518 unique sequences.</span>
<span id="cb120-845"><a href="#cb120-845" aria-hidden="true" tabindex="-1"></a>Sample 14 - 89315 reads in 43426 unique sequences.</span>
<span id="cb120-846"><a href="#cb120-846" aria-hidden="true" tabindex="-1"></a>Sample 15 - 33000 reads in 13673 unique sequences.</span>
<span id="cb120-847"><a href="#cb120-847" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/code&gt;&lt;/pre&gt;</span></span>
<span id="cb120-848"><a href="#cb120-848" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb120-849"><a href="#cb120-849" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-850"><a href="#cb120-850" aria-hidden="true" tabindex="-1"></a>As an example, we can inspect the returned <span class="in">`dada-class`</span> object for the forward and reverse reads from the sample #<span class="in">`r sampnum`</span>:</span>
<span id="cb120-851"><a href="#cb120-851" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-854"><a href="#cb120-854" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-855"><a href="#cb120-855" aria-hidden="true" tabindex="-1"></a>dadaFs[[sampnum]]</span>
<span id="cb120-856"><a href="#cb120-856" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-857"><a href="#cb120-857" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-858"><a href="#cb120-858" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-859"><a href="#cb120-859" aria-hidden="true" tabindex="-1"></a><span class="in">dada-class: object describing DADA2 denoising results</span></span>
<span id="cb120-860"><a href="#cb120-860" aria-hidden="true" tabindex="-1"></a><span class="in">1759 sequence variants were inferred from 31163 input unique sequences.</span></span>
<span id="cb120-861"><a href="#cb120-861" aria-hidden="true" tabindex="-1"></a><span class="in">Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</span></span>
<span id="cb120-862"><a href="#cb120-862" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-863"><a href="#cb120-863" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-864"><a href="#cb120-864" aria-hidden="true" tabindex="-1"></a>This output tells us how many true sequence variants the DADA2 algorithm inferred from the  unique sequences, in this case the sample 2.</span>
<span id="cb120-865"><a href="#cb120-865" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-866"><a href="#cb120-866" aria-hidden="true" tabindex="-1"></a><span class="fu">## 8. Construct Sequence Table</span></span>
<span id="cb120-867"><a href="#cb120-867" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-868"><a href="#cb120-868" aria-hidden="true" tabindex="-1"></a>Now we construct an amplicon sequence variant (ASV) table.</span>
<span id="cb120-869"><a href="#cb120-869" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-872"><a href="#cb120-872" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-873"><a href="#cb120-873" aria-hidden="true" tabindex="-1"></a>seqtab <span class="ot">&lt;-</span> <span class="fu">makeSequenceTable</span>(dadaFs)</span>
<span id="cb120-874"><a href="#cb120-874" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab)</span>
<span id="cb120-875"><a href="#cb120-875" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-876"><a href="#cb120-876" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-879"><a href="#cb120-879" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-880"><a href="#cb120-880" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-881"><a href="#cb120-881" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-882"><a href="#cb120-882" aria-hidden="true" tabindex="-1"></a>seqtab <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">15</span>, <span class="dv">21685</span>)</span>
<span id="cb120-883"><a href="#cb120-883" aria-hidden="true" tabindex="-1"></a>seqtab</span>
<span id="cb120-884"><a href="#cb120-884" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-885"><a href="#cb120-885" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-888"><a href="#cb120-888" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-889"><a href="#cb120-889" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab)))</span>
<span id="cb120-890"><a href="#cb120-890" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-891"><a href="#cb120-891" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-892"><a href="#cb120-892" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-893"><a href="#cb120-893" aria-hidden="true" tabindex="-1"></a><span class="in">   22    23    24    25    27    41    66    73   120   149   185   190   193 </span></span>
<span id="cb120-894"><a href="#cb120-894" aria-hidden="true" tabindex="-1"></a><span class="in">    2     1     1     2     1     1     1     1     1     1     1     9     1 </span></span>
<span id="cb120-895"><a href="#cb120-895" aria-hidden="true" tabindex="-1"></a><span class="in">  198   200   201   202   208   209   211   215   217   220   221   222   223 </span></span>
<span id="cb120-896"><a href="#cb120-896" aria-hidden="true" tabindex="-1"></a><span class="in">    1     1     1     1     1     1     1     1     2     4    14    21     8 </span></span>
<span id="cb120-897"><a href="#cb120-897" aria-hidden="true" tabindex="-1"></a><span class="in">  224   225   227   228   229   230   231   232   233   234   238   251 </span></span>
<span id="cb120-898"><a href="#cb120-898" aria-hidden="true" tabindex="-1"></a><span class="in">    3     4     6     4     5     9    20 21342   190    12     2     8 </span></span>
<span id="cb120-899"><a href="#cb120-899" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-900"><a href="#cb120-900" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-901"><a href="#cb120-901" aria-hidden="true" tabindex="-1"></a>The sequence table is a matrix with rows corresponding to (and named by) the samples, and columns corresponding to (and named by) the sequence variants. We have <span class="in">`r (seqtab)[2]`</span> sequence variants but also a range of sequence lengths. Since many of these sequence variants are singletons or doubletons, we can just select a range that corresponds to the expected amplicon length and eliminate the spurious reads.</span>
<span id="cb120-902"><a href="#cb120-902" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-905"><a href="#cb120-905" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-906"><a href="#cb120-906" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-907"><a href="#cb120-907" aria-hidden="true" tabindex="-1"></a>read_length <span class="ot">&lt;-</span>  <span class="fu">data.frame</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab)))</span>
<span id="cb120-908"><a href="#cb120-908" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(read_length) <span class="ot">&lt;-</span> <span class="st">"length"</span></span>
<span id="cb120-909"><a href="#cb120-909" aria-hidden="true" tabindex="-1"></a>plot_a <span class="ot">&lt;-</span> <span class="fu">qplot</span>(length, <span class="at">data =</span> read_length,</span>
<span id="cb120-910"><a href="#cb120-910" aria-hidden="true" tabindex="-1"></a>                <span class="at">geom =</span> <span class="st">"histogram"</span>, <span class="at">binwidth =</span> <span class="dv">1</span>,</span>
<span id="cb120-911"><a href="#cb120-911" aria-hidden="true" tabindex="-1"></a>                <span class="at">xlab =</span> <span class="st">"read length"</span>,</span>
<span id="cb120-912"><a href="#cb120-912" aria-hidden="true" tabindex="-1"></a>                <span class="at">ylab =</span> <span class="st">"total variants"</span>,</span>
<span id="cb120-913"><a href="#cb120-913" aria-hidden="true" tabindex="-1"></a>                <span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">200</span>, <span class="dv">275</span>))</span>
<span id="cb120-914"><a href="#cb120-914" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"read_length_before.png"</span>, plot_a, <span class="at">width =</span> <span class="dv">7</span>, <span class="at">height =</span> <span class="dv">3</span>)</span>
<span id="cb120-915"><a href="#cb120-915" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-916"><a href="#cb120-916" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-919"><a href="#cb120-919" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-920"><a href="#cb120-920" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-921"><a href="#cb120-921" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">"cp files/dada2/figures/ssu18_read_length_before.png include/dada2/ssu18_read_length_before.png"</span>)</span>
<span id="cb120-922"><a href="#cb120-922" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-923"><a href="#cb120-923" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-926"><a href="#cb120-926" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-927"><a href="#cb120-927" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-928"><a href="#cb120-928" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-929"><a href="#cb120-929" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/dada2/ssu18_read_length_before.png"</span>)</span>
<span id="cb120-930"><a href="#cb120-930" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-931"><a href="#cb120-931" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("read_length_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-932"><a href="#cb120-932" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-935"><a href="#cb120-935" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-936"><a href="#cb120-936" aria-hidden="true" tabindex="-1"></a>seqtab<span class="fl">.2</span> <span class="ot">&lt;-</span> seqtab[,<span class="fu">nchar</span>(<span class="fu">colnames</span>(seqtab)) <span class="sc">%in%</span> <span class="fu">seq</span>(<span class="dv">230</span>,<span class="dv">235</span>)]</span>
<span id="cb120-937"><a href="#cb120-937" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab<span class="fl">.2</span>)</span>
<span id="cb120-938"><a href="#cb120-938" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-939"><a href="#cb120-939" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-940"><a href="#cb120-940" aria-hidden="true" tabindex="-1"></a>:::{.callout-note}</span>
<span id="cb120-941"><a href="#cb120-941" aria-hidden="true" tabindex="-1"></a>The values you select should be based on the sequence table generated above.</span>
<span id="cb120-942"><a href="#cb120-942" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb120-943"><a href="#cb120-943" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-946"><a href="#cb120-946" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-947"><a href="#cb120-947" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-948"><a href="#cb120-948" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-949"><a href="#cb120-949" aria-hidden="true" tabindex="-1"></a>seqtab2 <span class="ot">&lt;-</span>  <span class="fu">c</span>(<span class="dv">15</span>, <span class="dv">21573</span>)</span>
<span id="cb120-950"><a href="#cb120-950" aria-hidden="true" tabindex="-1"></a>seqtab2</span>
<span id="cb120-951"><a href="#cb120-951" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-952"><a href="#cb120-952" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-955"><a href="#cb120-955" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-956"><a href="#cb120-956" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab<span class="fl">.2</span>)))</span>
<span id="cb120-957"><a href="#cb120-957" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-958"><a href="#cb120-958" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-959"><a href="#cb120-959" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-960"><a href="#cb120-960" aria-hidden="true" tabindex="-1"></a><span class="in">  230   231   232   233   234 </span></span>
<span id="cb120-961"><a href="#cb120-961" aria-hidden="true" tabindex="-1"></a><span class="in">    9    20 21342   190    12 </span></span>
<span id="cb120-962"><a href="#cb120-962" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-963"><a href="#cb120-963" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-964"><a href="#cb120-964" aria-hidden="true" tabindex="-1"></a>After removing the extreme length variants, we have <span class="in">`r (seqtab2)[2]`</span>, a reduction of <span class="in">`r (seqtab)[2] - (seqtab2)[2]`</span> sequence variants.</span>
<span id="cb120-965"><a href="#cb120-965" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-968"><a href="#cb120-968" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-969"><a href="#cb120-969" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-970"><a href="#cb120-970" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(seqtab<span class="fl">.2</span>, <span class="st">"ssu18_seqtab.rds"</span>)</span>
<span id="cb120-971"><a href="#cb120-971" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-972"><a href="#cb120-972" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-973"><a href="#cb120-973" aria-hidden="true" tabindex="-1"></a><span class="fu">## 9. Remove Chimeras</span></span>
<span id="cb120-974"><a href="#cb120-974" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-975"><a href="#cb120-975" aria-hidden="true" tabindex="-1"></a>Even though the <span class="in">`dada`</span> method corrects substitution and indel errors,  chimeric sequences remain. According to the DADA2 documentation, the accuracy of sequence variants after denoising makes identifying chimeric ASVs simpler than when dealing with fuzzy OTUs. Chimeric sequences are identified if they can be *exactly reconstructed* by combining a left-segment and a right-segment from two more abundant *parent* sequences.</span>
<span id="cb120-976"><a href="#cb120-976" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-979"><a href="#cb120-979" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-980"><a href="#cb120-980" aria-hidden="true" tabindex="-1"></a>seqtab.nochim <span class="ot">&lt;-</span> <span class="fu">removeBimeraDenovo</span>(seqtab<span class="fl">.2</span>,</span>
<span id="cb120-981"><a href="#cb120-981" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">method =</span> <span class="st">"consensus"</span>,</span>
<span id="cb120-982"><a href="#cb120-982" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">multithread =</span> <span class="dv">20</span>,</span>
<span id="cb120-983"><a href="#cb120-983" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-984"><a href="#cb120-984" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab.nochim)</span>
<span id="cb120-985"><a href="#cb120-985" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-986"><a href="#cb120-986" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-987"><a href="#cb120-987" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-988"><a href="#cb120-988" aria-hidden="true" tabindex="-1"></a><span class="in">Identified 1241 bimeras out of 21573 input sequences.</span></span>
<span id="cb120-989"><a href="#cb120-989" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-990"><a href="#cb120-990" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-993"><a href="#cb120-993" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-994"><a href="#cb120-994" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-995"><a href="#cb120-995" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-996"><a href="#cb120-996" aria-hidden="true" tabindex="-1"></a>seqtab3 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">15</span>, <span class="dv">20332</span>)</span>
<span id="cb120-997"><a href="#cb120-997" aria-hidden="true" tabindex="-1"></a>seqtab3</span>
<span id="cb120-998"><a href="#cb120-998" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-999"><a href="#cb120-999" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1002"><a href="#cb120-1002" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1003"><a href="#cb120-1003" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(seqtab.nochim)<span class="sc">/</span><span class="fu">sum</span>(seqtab<span class="fl">.2</span>)</span>
<span id="cb120-1004"><a href="#cb120-1004" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1005"><a href="#cb120-1005" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1008"><a href="#cb120-1008" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1009"><a href="#cb120-1009" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1010"><a href="#cb120-1010" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-1011"><a href="#cb120-1011" aria-hidden="true" tabindex="-1"></a>chim_rem <span class="ot">&lt;-</span> <span class="fl">0.9596595</span></span>
<span id="cb120-1012"><a href="#cb120-1012" aria-hidden="true" tabindex="-1"></a><span class="fl">0.9596595</span></span>
<span id="cb120-1013"><a href="#cb120-1013" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1014"><a href="#cb120-1014" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1015"><a href="#cb120-1015" aria-hidden="true" tabindex="-1"></a>Chimera checking removed an additional <span class="in">`r (seqtab2)[2] - (seqtab3)[2]`</span> sequence variants however, when we account for the abundances of each variant, we see chimeras accounts for about <span class="in">`r 100*(1-chim_rem)`</span>% of the merged sequence reads. Not bad.</span>
<span id="cb120-1016"><a href="#cb120-1016" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1017"><a href="#cb120-1017" aria-hidden="true" tabindex="-1"></a>The last thing we want to do is write the sequence table to an <span class="in">`RDS`</span> file.</span>
<span id="cb120-1018"><a href="#cb120-1018" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1021"><a href="#cb120-1021" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1022"><a href="#cb120-1022" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(seqtab.nochim, <span class="st">"ssu18_seqtab.nochim.rds"</span>)</span>
<span id="cb120-1023"><a href="#cb120-1023" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1024"><a href="#cb120-1024" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1025"><a href="#cb120-1025" aria-hidden="true" tabindex="-1"></a><span class="fu">## 10. Track Reads through Workflow</span></span>
<span id="cb120-1026"><a href="#cb120-1026" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1027"><a href="#cb120-1027" aria-hidden="true" tabindex="-1"></a>At this point we can look at the number of reads that made it through each step of the workflow for every sample.</span>
<span id="cb120-1028"><a href="#cb120-1028" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1031"><a href="#cb120-1031" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1032"><a href="#cb120-1032" aria-hidden="true" tabindex="-1"></a>getN <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">sum</span>(<span class="fu">getUniques</span>(x))</span>
<span id="cb120-1033"><a href="#cb120-1033" aria-hidden="true" tabindex="-1"></a>track <span class="ot">&lt;-</span> <span class="fu">cbind</span>(out,</span>
<span id="cb120-1034"><a href="#cb120-1034" aria-hidden="true" tabindex="-1"></a>               <span class="fu">sapply</span>(dadaFs, getN),</span>
<span id="cb120-1035"><a href="#cb120-1035" aria-hidden="true" tabindex="-1"></a>               <span class="fu">rowSums</span>(seqtab.nochim))</span>
<span id="cb120-1036"><a href="#cb120-1036" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(track) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"input"</span>, <span class="st">"filtered"</span>, <span class="st">"denoisedF"</span>, <span class="st">"nonchim"</span>)</span>
<span id="cb120-1037"><a href="#cb120-1037" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(track) <span class="ot">&lt;-</span> sample.names</span>
<span id="cb120-1038"><a href="#cb120-1038" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1039"><a href="#cb120-1039" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1040"><a href="#cb120-1040" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1043"><a href="#cb120-1043" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1044"><a href="#cb120-1044" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1045"><a href="#cb120-1045" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-1046"><a href="#cb120-1046" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"files/dada2/tables/ssu18_read_changes.txt"</span>,</span>
<span id="cb120-1047"><a href="#cb120-1047" aria-hidden="true" tabindex="-1"></a>                        <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb120-1048"><a href="#cb120-1048" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1049"><a href="#cb120-1049" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1050"><a href="#cb120-1050" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab("track_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-1051"><a href="#cb120-1051" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1054"><a href="#cb120-1054" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1055"><a href="#cb120-1055" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1056"><a href="#cb120-1056" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-1057"><a href="#cb120-1057" aria-hidden="true" tabindex="-1"></a>seq_table <span class="sc">%&gt;%</span></span>
<span id="cb120-1058"><a href="#cb120-1058" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb120-1059"><a href="#cb120-1059" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"ssu18_track_read_changes"</span>,</span>
<span id="cb120-1060"><a href="#cb120-1060" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb120-1061"><a href="#cb120-1061" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb120-1062"><a href="#cb120-1062" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb120-1063"><a href="#cb120-1063" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb120-1064"><a href="#cb120-1064" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-1065"><a href="#cb120-1065" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb120-1066"><a href="#cb120-1066" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1067"><a href="#cb120-1067" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb120-1068"><a href="#cb120-1068" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb120-1069"><a href="#cb120-1069" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb120-1070"><a href="#cb120-1070" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">1</span>),</span>
<span id="cb120-1071"><a href="#cb120-1071" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">TRUE</span>, <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-1072"><a href="#cb120-1072" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb120-1073"><a href="#cb120-1073" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-1074"><a href="#cb120-1074" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb120-1075"><a href="#cb120-1075" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Sample ID"</span>, </span>
<span id="cb120-1076"><a href="#cb120-1076" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb120-1077"><a href="#cb120-1077" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>),</span>
<span id="cb120-1078"><a href="#cb120-1078" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>), </span>
<span id="cb120-1079"><a href="#cb120-1079" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb120-1080"><a href="#cb120-1080" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">150</span>, <span class="at">footer =</span> <span class="st">"Total reads"</span>), </span>
<span id="cb120-1081"><a href="#cb120-1081" aria-hidden="true" tabindex="-1"></a>    <span class="at">raw =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-1082"><a href="#cb120-1082" aria-hidden="true" tabindex="-1"></a>    <span class="at">pre_filt =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-1083"><a href="#cb120-1083" aria-hidden="true" tabindex="-1"></a>    <span class="at">pre_filt =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-1084"><a href="#cb120-1084" aria-hidden="true" tabindex="-1"></a>    <span class="at">cut =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-1085"><a href="#cb120-1085" aria-hidden="true" tabindex="-1"></a>    <span class="at">filtered =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-1086"><a href="#cb120-1086" aria-hidden="true" tabindex="-1"></a>    <span class="at">denoisedF =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-1087"><a href="#cb120-1087" aria-hidden="true" tabindex="-1"></a>    <span class="at">nonchim =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values)))</span>
<span id="cb120-1088"><a href="#cb120-1088" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-1089"><a href="#cb120-1089" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">5</span>, </span>
<span id="cb120-1090"><a href="#cb120-1090" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb120-1091"><a href="#cb120-1091" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb120-1092"><a href="#cb120-1092" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb120-1093"><a href="#cb120-1093" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-1094"><a href="#cb120-1094" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb120-1095"><a href="#cb120-1095" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1096"><a href="#cb120-1096" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1097"><a href="#cb120-1097" aria-hidden="true" tabindex="-1"></a>And we can save a copy of the table for later use.</span>
<span id="cb120-1098"><a href="#cb120-1098" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1101"><a href="#cb120-1101" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1102"><a href="#cb120-1102" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(track, <span class="st">"ssu18_read_changes.txt"</span>,</span>
<span id="cb120-1103"><a href="#cb120-1103" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span>
<span id="cb120-1104"><a href="#cb120-1104" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1105"><a href="#cb120-1105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1108"><a href="#cb120-1108" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1109"><a href="#cb120-1109" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1110"><a href="#cb120-1110" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="st">"pre_tax_16s_2018.rdata"</span>)</span>
<span id="cb120-1111"><a href="#cb120-1111" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1112"><a href="#cb120-1112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1113"><a href="#cb120-1113" aria-hidden="true" tabindex="-1"></a><span class="fu">## 11. Assign Taxonomy</span></span>
<span id="cb120-1114"><a href="#cb120-1114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1115"><a href="#cb120-1115" aria-hidden="true" tabindex="-1"></a>The <span class="in">`assignTaxonomy`</span> command implements the naive Bayesian classifier, so for reproducible results you need to set a random number seed (see issue <span class="co">[</span><span class="ot">#538</span><span class="co">](https://github.com/benjjneb/dada2/issues/538)</span>). We did this at the beginning of the workflow. For taxonomic assignment, we are using the Silva version 138 <span class="co">[</span><span class="ot">@quast2012silva</span><span class="co">]</span>. The developers of DADA2 maintain a <span class="co">[</span><span class="ot">formatted version of the database</span><span class="co">](https://benjjneb.github.io/dada2/training.html)</span>.</span>
<span id="cb120-1116"><a href="#cb120-1116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1117"><a href="#cb120-1117" aria-hidden="true" tabindex="-1"></a>We will read in the <span class="in">`RDS`</span> file containing the sequence table saved above. We also need to run <span class="in">`remove(list = ls())`</span> command, otherwise the final image we save will be huge. This way, the image only contains the sample data, seqtab, and taxtab *after* running <span class="in">`removeBimeraDenovo`</span>.</span>
<span id="cb120-1118"><a href="#cb120-1118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1121"><a href="#cb120-1121" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1122"><a href="#cb120-1122" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb120-1123"><a href="#cb120-1123" aria-hidden="true" tabindex="-1"></a>seqtab <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"ssu18_seqtab.nochim.rds"</span>)</span>
<span id="cb120-1124"><a href="#cb120-1124" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1125"><a href="#cb120-1125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1126"><a href="#cb120-1126" aria-hidden="true" tabindex="-1"></a>And then native Bayesian <span class="co">[</span><span class="ot">@wang2007naive</span><span class="co">]</span> classifier against the Silva database.</span>
<span id="cb120-1127"><a href="#cb120-1127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1130"><a href="#cb120-1130" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1131"><a href="#cb120-1131" aria-hidden="true" tabindex="-1"></a>tax_silva <span class="ot">&lt;-</span> <span class="fu">assignTaxonomy</span>(</span>
<span id="cb120-1132"><a href="#cb120-1132" aria-hidden="true" tabindex="-1"></a>  seqtab, <span class="st">"silva_nr_v138_train_set.fa.gz"</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-1133"><a href="#cb120-1133" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1134"><a href="#cb120-1134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1135"><a href="#cb120-1135" aria-hidden="true" tabindex="-1"></a><span class="fu">## 12. Save Image</span></span>
<span id="cb120-1136"><a href="#cb120-1136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1137"><a href="#cb120-1137" aria-hidden="true" tabindex="-1"></a>And finally, we save an image for use in the analytic part of the workflow. This R data file will be needed as the input for the phyloseq portion of the workflow. See the <span class="co">[</span><span class="ot">Data Availability</span><span class="co">](data-availability.html)</span> page for complete details on where to get this file.</span>
<span id="cb120-1138"><a href="#cb120-1138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1141"><a href="#cb120-1141" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1142"><a href="#cb120-1142" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="st">"ssu18_dada2_wf.rdata"</span>)</span>
<span id="cb120-1143"><a href="#cb120-1143" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1144"><a href="#cb120-1144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1147"><a href="#cb120-1147" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1148"><a href="#cb120-1148" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1149"><a href="#cb120-1149" aria-hidden="true" tabindex="-1"></a><span class="co"># Classify with IdTAX db</span></span>
<span id="cb120-1150"><a href="#cb120-1150" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb120-1151"><a href="#cb120-1151" aria-hidden="true" tabindex="-1"></a>seqtab <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"ssu18_seqtab.nochim.rds"</span>)</span>
<span id="cb120-1152"><a href="#cb120-1152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1153"><a href="#cb120-1153" aria-hidden="true" tabindex="-1"></a>tax_silva <span class="ot">&lt;-</span> <span class="fu">assignTaxonomy</span>(seqtab, <span class="st">"silva_nr_v138_train_set.fa.gz"</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-1154"><a href="#cb120-1154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1155"><a href="#cb120-1155" aria-hidden="true" tabindex="-1"></a>dna <span class="ot">&lt;-</span> <span class="fu">DNAStringSet</span>(<span class="fu">getSequences</span>(seqtab)) <span class="co"># Create a DNAStringSet from the ASVs</span></span>
<span id="cb120-1156"><a href="#cb120-1156" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"SILVA_SSU_r138_2018.RData"</span>)</span>
<span id="cb120-1157"><a href="#cb120-1157" aria-hidden="true" tabindex="-1"></a>ids <span class="ot">&lt;-</span> <span class="fu">IdTaxa</span>(dna, trainingSet, <span class="at">strand =</span> <span class="st">"top"</span>, <span class="at">processors =</span> <span class="cn">NULL</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>) <span class="co"># use all processors</span></span>
<span id="cb120-1158"><a href="#cb120-1158" aria-hidden="true" tabindex="-1"></a>ranks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"domain"</span>, <span class="st">"phylum"</span>, <span class="st">"class"</span>, <span class="st">"order"</span>, <span class="st">"family"</span>, <span class="st">"genus"</span>) <span class="co"># ranks of interest</span></span>
<span id="cb120-1159"><a href="#cb120-1159" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the output object of class "Taxa" to a matrix analogous to the output from assignTaxonomy</span></span>
<span id="cb120-1160"><a href="#cb120-1160" aria-hidden="true" tabindex="-1"></a>tax_id <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">sapply</span>(ids, <span class="cf">function</span>(x) {</span>
<span id="cb120-1161"><a href="#cb120-1161" aria-hidden="true" tabindex="-1"></a>        m <span class="ot">&lt;-</span> <span class="fu">match</span>(ranks, x<span class="sc">$</span>rank)</span>
<span id="cb120-1162"><a href="#cb120-1162" aria-hidden="true" tabindex="-1"></a>        taxa <span class="ot">&lt;-</span> x<span class="sc">$</span>taxon[m]</span>
<span id="cb120-1163"><a href="#cb120-1163" aria-hidden="true" tabindex="-1"></a>        taxa[<span class="fu">startsWith</span>(taxa, <span class="st">"unclassified_"</span>)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb120-1164"><a href="#cb120-1164" aria-hidden="true" tabindex="-1"></a>        taxa</span>
<span id="cb120-1165"><a href="#cb120-1165" aria-hidden="true" tabindex="-1"></a>}))</span>
<span id="cb120-1166"><a href="#cb120-1166" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tax_id) <span class="ot">&lt;-</span> ranks; <span class="fu">rownames</span>(tax_id) <span class="ot">&lt;-</span> <span class="fu">getSequences</span>(seqtab)</span>
<span id="cb120-1167"><a href="#cb120-1167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1168"><a href="#cb120-1168" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="st">"ssu18_data_prep_wf_ALT.rdata"</span>)</span>
<span id="cb120-1169"><a href="#cb120-1169" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(seqtab, <span class="st">"ssu18_seqtab.nochim.2.rds"</span>)</span>
<span id="cb120-1170"><a href="#cb120-1170" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(tax_silva, <span class="st">"ssu18_tax_silva.rds"</span>)</span>
<span id="cb120-1171"><a href="#cb120-1171" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(tax_id, <span class="st">"ssu18_tax_id.rds"</span>)</span>
<span id="cb120-1172"><a href="#cb120-1172" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1173"><a href="#cb120-1173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1174"><a href="#cb120-1174" aria-hidden="true" tabindex="-1"></a><span class="fu">## R Session Information &amp; Code</span></span>
<span id="cb120-1175"><a href="#cb120-1175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1176"><a href="#cb120-1176" aria-hidden="true" tabindex="-1"></a>This workflow was run on the <span class="co">[</span><span class="ot">Smithsonian High Performance Cluster (SI/HPC)</span><span class="co">](https://doi.org/10.25572/SIHPC)</span>, Smithsonian Institution. Below are the specific packages and versions used in this workflow using both <span class="in">`sessionInfo()`</span> and <span class="in">`devtools::session_info()`</span>. Click the arrow to see the details.</span>
<span id="cb120-1177"><a href="#cb120-1177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1178"><a href="#cb120-1178" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb120-1179"><a href="#cb120-1179" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show/hide  R Session Info<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb120-1180"><a href="#cb120-1180" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1181"><a href="#cb120-1181" aria-hidden="true" tabindex="-1"></a><span class="in">sessionInfo()</span></span>
<span id="cb120-1182"><a href="#cb120-1182" aria-hidden="true" tabindex="-1"></a><span class="in">R version 4.0.0 (2020-04-24)</span></span>
<span id="cb120-1183"><a href="#cb120-1183" aria-hidden="true" tabindex="-1"></a><span class="in">Platform: x86_64-conda_cos6-linux-gnu (64-bit)</span></span>
<span id="cb120-1184"><a href="#cb120-1184" aria-hidden="true" tabindex="-1"></a><span class="in">Running under: CentOS Linux 7 (Core)</span></span>
<span id="cb120-1185"><a href="#cb120-1185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1186"><a href="#cb120-1186" aria-hidden="true" tabindex="-1"></a><span class="in">Matrix products: default</span></span>
<span id="cb120-1187"><a href="#cb120-1187" aria-hidden="true" tabindex="-1"></a><span class="in">BLAS/LAPACK: /home/scottjj/miniconda3/envs/R-4/lib/libopenblasp-r0.3.9.so</span></span>
<span id="cb120-1188"><a href="#cb120-1188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1189"><a href="#cb120-1189" aria-hidden="true" tabindex="-1"></a><span class="in">locale:</span></span>
<span id="cb120-1190"><a href="#cb120-1190" aria-hidden="true" tabindex="-1"></a><span class="in"> [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C</span></span>
<span id="cb120-1191"><a href="#cb120-1191" aria-hidden="true" tabindex="-1"></a><span class="in"> [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8</span></span>
<span id="cb120-1192"><a href="#cb120-1192" aria-hidden="true" tabindex="-1"></a><span class="in"> [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8</span></span>
<span id="cb120-1193"><a href="#cb120-1193" aria-hidden="true" tabindex="-1"></a><span class="in"> [7] LC_PAPER=en_US.UTF-8       LC_NAME=C</span></span>
<span id="cb120-1194"><a href="#cb120-1194" aria-hidden="true" tabindex="-1"></a><span class="in"> [9] LC_ADDRESS=C               LC_TELEPHONE=C</span></span>
<span id="cb120-1195"><a href="#cb120-1195" aria-hidden="true" tabindex="-1"></a><span class="in">[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C</span></span>
<span id="cb120-1196"><a href="#cb120-1196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1197"><a href="#cb120-1197" aria-hidden="true" tabindex="-1"></a><span class="in">attached base packages:</span></span>
<span id="cb120-1198"><a href="#cb120-1198" aria-hidden="true" tabindex="-1"></a><span class="in"> [1] grid      stats4    parallel  stats     graphics  grDevices utils</span></span>
<span id="cb120-1199"><a href="#cb120-1199" aria-hidden="true" tabindex="-1"></a><span class="in"> [8] datasets  methods   base</span></span>
<span id="cb120-1200"><a href="#cb120-1200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1201"><a href="#cb120-1201" aria-hidden="true" tabindex="-1"></a><span class="in">other attached packages:</span></span>
<span id="cb120-1202"><a href="#cb120-1202" aria-hidden="true" tabindex="-1"></a><span class="in"> [1] gridExtra_2.3               ggplot2_3.3.0</span></span>
<span id="cb120-1203"><a href="#cb120-1203" aria-hidden="true" tabindex="-1"></a><span class="in"> [3] ShortRead_1.46.0            GenomicAlignments_1.24.0</span></span>
<span id="cb120-1204"><a href="#cb120-1204" aria-hidden="true" tabindex="-1"></a><span class="in"> [5] SummarizedExperiment_1.18.1 DelayedArray_0.14.0</span></span>
<span id="cb120-1205"><a href="#cb120-1205" aria-hidden="true" tabindex="-1"></a><span class="in"> [7] matrixStats_0.56.0          Biobase_2.48.0</span></span>
<span id="cb120-1206"><a href="#cb120-1206" aria-hidden="true" tabindex="-1"></a><span class="in"> [9] Rsamtools_2.4.0             GenomicRanges_1.40.0</span></span>
<span id="cb120-1207"><a href="#cb120-1207" aria-hidden="true" tabindex="-1"></a><span class="in">[11] GenomeInfoDb_1.24.0         Biostrings_2.56.0</span></span>
<span id="cb120-1208"><a href="#cb120-1208" aria-hidden="true" tabindex="-1"></a><span class="in">[13] XVector_0.28.0              IRanges_2.22.1</span></span>
<span id="cb120-1209"><a href="#cb120-1209" aria-hidden="true" tabindex="-1"></a><span class="in">[15] S4Vectors_0.26.0            BiocParallel_1.22.0</span></span>
<span id="cb120-1210"><a href="#cb120-1210" aria-hidden="true" tabindex="-1"></a><span class="in">[17] BiocGenerics_0.34.0         dada2_1.16.0</span></span>
<span id="cb120-1211"><a href="#cb120-1211" aria-hidden="true" tabindex="-1"></a><span class="in">[19] Rcpp_1.0.4.6</span></span>
<span id="cb120-1212"><a href="#cb120-1212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1213"><a href="#cb120-1213" aria-hidden="true" tabindex="-1"></a><span class="in">loaded via a namespace (and not attached):</span></span>
<span id="cb120-1214"><a href="#cb120-1214" aria-hidden="true" tabindex="-1"></a><span class="in"> [1] plyr_1.8.6             RColorBrewer_1.1-2     pillar_1.4.4</span></span>
<span id="cb120-1215"><a href="#cb120-1215" aria-hidden="true" tabindex="-1"></a><span class="in"> [4] compiler_4.0.0         bitops_1.0-6           tools_4.0.0</span></span>
<span id="cb120-1216"><a href="#cb120-1216" aria-hidden="true" tabindex="-1"></a><span class="in"> [7] zlibbioc_1.34.0        lifecycle_0.2.0        tibble_3.0.1</span></span>
<span id="cb120-1217"><a href="#cb120-1217" aria-hidden="true" tabindex="-1"></a><span class="in">[10] gtable_0.3.0           lattice_0.20-41        png_0.1-7</span></span>
<span id="cb120-1218"><a href="#cb120-1218" aria-hidden="true" tabindex="-1"></a><span class="in">[13] pkgconfig_2.0.3        rlang_0.4.6            Matrix_1.2-18</span></span>
<span id="cb120-1219"><a href="#cb120-1219" aria-hidden="true" tabindex="-1"></a><span class="in">[16] GenomeInfoDbData_1.2.3 withr_2.2.0            stringr_1.4.0</span></span>
<span id="cb120-1220"><a href="#cb120-1220" aria-hidden="true" tabindex="-1"></a><span class="in">[19] hwriter_1.3.2          vctrs_0.3.0            glue_1.4.1</span></span>
<span id="cb120-1221"><a href="#cb120-1221" aria-hidden="true" tabindex="-1"></a><span class="in">[22] R6_2.4.1               jpeg_0.1-8.1           latticeExtra_0.6-29</span></span>
<span id="cb120-1222"><a href="#cb120-1222" aria-hidden="true" tabindex="-1"></a><span class="in">[25] reshape2_1.4.4         magrittr_1.5           scales_1.1.1</span></span>
<span id="cb120-1223"><a href="#cb120-1223" aria-hidden="true" tabindex="-1"></a><span class="in">[28] ellipsis_0.3.1         colorspace_1.4-1       stringi_1.4.6</span></span>
<span id="cb120-1224"><a href="#cb120-1224" aria-hidden="true" tabindex="-1"></a><span class="in">[31] RCurl_1.98-1.2         RcppParallel_5.0.1     munsell_0.5.0</span></span>
<span id="cb120-1225"><a href="#cb120-1225" aria-hidden="true" tabindex="-1"></a><span class="in">[34] crayon_1.3.4</span></span>
<span id="cb120-1226"><a href="#cb120-1226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1227"><a href="#cb120-1227" aria-hidden="true" tabindex="-1"></a><span class="in">devtools::session_info()</span></span>
<span id="cb120-1228"><a href="#cb120-1228" aria-hidden="true" tabindex="-1"></a><span class="in">â”€ Session info â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span>
<span id="cb120-1229"><a href="#cb120-1229" aria-hidden="true" tabindex="-1"></a><span class="in"> setting  value</span></span>
<span id="cb120-1230"><a href="#cb120-1230" aria-hidden="true" tabindex="-1"></a><span class="in"> version  R version 4.0.0 (2020-04-24)</span></span>
<span id="cb120-1231"><a href="#cb120-1231" aria-hidden="true" tabindex="-1"></a><span class="in"> os       CentOS Linux 7 (Core)</span></span>
<span id="cb120-1232"><a href="#cb120-1232" aria-hidden="true" tabindex="-1"></a><span class="in"> system   x86_64, linux-gnu</span></span>
<span id="cb120-1233"><a href="#cb120-1233" aria-hidden="true" tabindex="-1"></a><span class="in"> ui       X11</span></span>
<span id="cb120-1234"><a href="#cb120-1234" aria-hidden="true" tabindex="-1"></a><span class="in"> language (EN)</span></span>
<span id="cb120-1235"><a href="#cb120-1235" aria-hidden="true" tabindex="-1"></a><span class="in"> collate  en_US.UTF-8</span></span>
<span id="cb120-1236"><a href="#cb120-1236" aria-hidden="true" tabindex="-1"></a><span class="in"> ctype    en_US.UTF-8</span></span>
<span id="cb120-1237"><a href="#cb120-1237" aria-hidden="true" tabindex="-1"></a><span class="in"> tz       America/New_York</span></span>
<span id="cb120-1238"><a href="#cb120-1238" aria-hidden="true" tabindex="-1"></a><span class="in"> date     2020-06-17</span></span>
<span id="cb120-1239"><a href="#cb120-1239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1240"><a href="#cb120-1240" aria-hidden="true" tabindex="-1"></a><span class="in">â”€ Packages â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span>
<span id="cb120-1241"><a href="#cb120-1241" aria-hidden="true" tabindex="-1"></a><span class="in"> package              * version  date       lib source</span></span>
<span id="cb120-1242"><a href="#cb120-1242" aria-hidden="true" tabindex="-1"></a><span class="in"> assertthat             0.2.1    2019-03-21 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1243"><a href="#cb120-1243" aria-hidden="true" tabindex="-1"></a><span class="in"> backports              1.1.7    2020-05-13 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1244"><a href="#cb120-1244" aria-hidden="true" tabindex="-1"></a><span class="in"> Biobase              * 2.48.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-1245"><a href="#cb120-1245" aria-hidden="true" tabindex="-1"></a><span class="in"> BiocGenerics         * 0.34.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-1246"><a href="#cb120-1246" aria-hidden="true" tabindex="-1"></a><span class="in"> BiocParallel         * 1.22.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-1247"><a href="#cb120-1247" aria-hidden="true" tabindex="-1"></a><span class="in"> Biostrings           * 2.56.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-1248"><a href="#cb120-1248" aria-hidden="true" tabindex="-1"></a><span class="in"> bitops                 1.0-6    2013-08-17 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1249"><a href="#cb120-1249" aria-hidden="true" tabindex="-1"></a><span class="in"> callr                  3.4.3    2020-03-28 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1250"><a href="#cb120-1250" aria-hidden="true" tabindex="-1"></a><span class="in"> cli                    2.0.2    2020-02-28 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1251"><a href="#cb120-1251" aria-hidden="true" tabindex="-1"></a><span class="in"> colorspace             1.4-1    2019-03-18 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1252"><a href="#cb120-1252" aria-hidden="true" tabindex="-1"></a><span class="in"> crayon                 1.3.4    2017-09-16 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1253"><a href="#cb120-1253" aria-hidden="true" tabindex="-1"></a><span class="in"> dada2                * 1.16.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-1254"><a href="#cb120-1254" aria-hidden="true" tabindex="-1"></a><span class="in"> DelayedArray         * 0.14.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-1255"><a href="#cb120-1255" aria-hidden="true" tabindex="-1"></a><span class="in"> desc                   1.2.0    2018-05-01 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1256"><a href="#cb120-1256" aria-hidden="true" tabindex="-1"></a><span class="in"> devtools               2.3.0    2020-04-10 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1257"><a href="#cb120-1257" aria-hidden="true" tabindex="-1"></a><span class="in"> digest                 0.6.25   2020-02-23 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1258"><a href="#cb120-1258" aria-hidden="true" tabindex="-1"></a><span class="in"> ellipsis               0.3.1    2020-05-15 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1259"><a href="#cb120-1259" aria-hidden="true" tabindex="-1"></a><span class="in"> fansi                  0.4.1    2020-01-08 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1260"><a href="#cb120-1260" aria-hidden="true" tabindex="-1"></a><span class="in"> fs                     1.4.1    2020-04-04 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1261"><a href="#cb120-1261" aria-hidden="true" tabindex="-1"></a><span class="in"> GenomeInfoDb         * 1.24.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-1262"><a href="#cb120-1262" aria-hidden="true" tabindex="-1"></a><span class="in"> GenomeInfoDbData       1.2.3    2020-05-22 [1] Bioconductor</span></span>
<span id="cb120-1263"><a href="#cb120-1263" aria-hidden="true" tabindex="-1"></a><span class="in"> GenomicAlignments    * 1.24.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-1264"><a href="#cb120-1264" aria-hidden="true" tabindex="-1"></a><span class="in"> GenomicRanges        * 1.40.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-1265"><a href="#cb120-1265" aria-hidden="true" tabindex="-1"></a><span class="in"> ggplot2              * 3.3.0    2020-03-05 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1266"><a href="#cb120-1266" aria-hidden="true" tabindex="-1"></a><span class="in"> glue                   1.4.1    2020-05-13 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1267"><a href="#cb120-1267" aria-hidden="true" tabindex="-1"></a><span class="in"> gridExtra            * 2.3      2017-09-09 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1268"><a href="#cb120-1268" aria-hidden="true" tabindex="-1"></a><span class="in"> gtable                 0.3.0    2019-03-25 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1269"><a href="#cb120-1269" aria-hidden="true" tabindex="-1"></a><span class="in"> hwriter                1.3.2    2014-09-10 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1270"><a href="#cb120-1270" aria-hidden="true" tabindex="-1"></a><span class="in"> IRanges              * 2.22.1   2020-04-28 [1] Bioconductor</span></span>
<span id="cb120-1271"><a href="#cb120-1271" aria-hidden="true" tabindex="-1"></a><span class="in"> jpeg                   0.1-8.1  2019-10-24 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1272"><a href="#cb120-1272" aria-hidden="true" tabindex="-1"></a><span class="in"> lattice                0.20-41  2020-04-02 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1273"><a href="#cb120-1273" aria-hidden="true" tabindex="-1"></a><span class="in"> latticeExtra           0.6-29   2019-12-19 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1274"><a href="#cb120-1274" aria-hidden="true" tabindex="-1"></a><span class="in"> lifecycle              0.2.0    2020-03-06 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1275"><a href="#cb120-1275" aria-hidden="true" tabindex="-1"></a><span class="in"> magrittr               1.5      2014-11-22 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1276"><a href="#cb120-1276" aria-hidden="true" tabindex="-1"></a><span class="in"> Matrix                 1.2-18   2019-11-27 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1277"><a href="#cb120-1277" aria-hidden="true" tabindex="-1"></a><span class="in"> matrixStats          * 0.56.0   2020-03-13 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1278"><a href="#cb120-1278" aria-hidden="true" tabindex="-1"></a><span class="in"> memoise                1.1.0    2017-04-21 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1279"><a href="#cb120-1279" aria-hidden="true" tabindex="-1"></a><span class="in"> munsell                0.5.0    2018-06-12 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1280"><a href="#cb120-1280" aria-hidden="true" tabindex="-1"></a><span class="in"> pillar                 1.4.4    2020-05-05 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1281"><a href="#cb120-1281" aria-hidden="true" tabindex="-1"></a><span class="in"> pkgbuild               1.0.8    2020-05-07 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1282"><a href="#cb120-1282" aria-hidden="true" tabindex="-1"></a><span class="in"> pkgconfig              2.0.3    2019-09-22 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1283"><a href="#cb120-1283" aria-hidden="true" tabindex="-1"></a><span class="in"> pkgload                1.0.2    2018-10-29 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1284"><a href="#cb120-1284" aria-hidden="true" tabindex="-1"></a><span class="in"> plyr                   1.8.6    2020-03-03 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1285"><a href="#cb120-1285" aria-hidden="true" tabindex="-1"></a><span class="in"> png                    0.1-7    2013-12-03 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1286"><a href="#cb120-1286" aria-hidden="true" tabindex="-1"></a><span class="in"> prettyunits            1.1.1    2020-01-24 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1287"><a href="#cb120-1287" aria-hidden="true" tabindex="-1"></a><span class="in"> processx               3.4.2    2020-02-09 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1288"><a href="#cb120-1288" aria-hidden="true" tabindex="-1"></a><span class="in"> ps                     1.3.3    2020-05-08 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1289"><a href="#cb120-1289" aria-hidden="true" tabindex="-1"></a><span class="in"> R6                     2.4.1    2019-11-12 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1290"><a href="#cb120-1290" aria-hidden="true" tabindex="-1"></a><span class="in"> RColorBrewer           1.1-2    2014-12-07 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1291"><a href="#cb120-1291" aria-hidden="true" tabindex="-1"></a><span class="in"> Rcpp                 * 1.0.4.6  2020-04-09 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1292"><a href="#cb120-1292" aria-hidden="true" tabindex="-1"></a><span class="in"> RcppParallel           5.0.1    2020-05-06 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1293"><a href="#cb120-1293" aria-hidden="true" tabindex="-1"></a><span class="in"> RCurl                  1.98-1.2 2020-04-18 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1294"><a href="#cb120-1294" aria-hidden="true" tabindex="-1"></a><span class="in"> remotes                2.1.1    2020-02-15 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1295"><a href="#cb120-1295" aria-hidden="true" tabindex="-1"></a><span class="in"> reshape2               1.4.4    2020-04-09 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1296"><a href="#cb120-1296" aria-hidden="true" tabindex="-1"></a><span class="in"> rlang                  0.4.6    2020-05-02 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1297"><a href="#cb120-1297" aria-hidden="true" tabindex="-1"></a><span class="in"> rprojroot              1.3-2    2018-01-03 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1298"><a href="#cb120-1298" aria-hidden="true" tabindex="-1"></a><span class="in"> Rsamtools            * 2.4.0    2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-1299"><a href="#cb120-1299" aria-hidden="true" tabindex="-1"></a><span class="in"> S4Vectors            * 0.26.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-1300"><a href="#cb120-1300" aria-hidden="true" tabindex="-1"></a><span class="in"> scales                 1.1.1    2020-05-11 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1301"><a href="#cb120-1301" aria-hidden="true" tabindex="-1"></a><span class="in"> sessioninfo            1.1.1    2018-11-05 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1302"><a href="#cb120-1302" aria-hidden="true" tabindex="-1"></a><span class="in"> ShortRead            * 1.46.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-1303"><a href="#cb120-1303" aria-hidden="true" tabindex="-1"></a><span class="in"> stringi                1.4.6    2020-02-17 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1304"><a href="#cb120-1304" aria-hidden="true" tabindex="-1"></a><span class="in"> stringr                1.4.0    2019-02-10 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1305"><a href="#cb120-1305" aria-hidden="true" tabindex="-1"></a><span class="in"> SummarizedExperiment * 1.18.1   2020-04-30 [1] Bioconductor</span></span>
<span id="cb120-1306"><a href="#cb120-1306" aria-hidden="true" tabindex="-1"></a><span class="in"> testthat               2.3.2    2020-03-02 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1307"><a href="#cb120-1307" aria-hidden="true" tabindex="-1"></a><span class="in"> tibble                 3.0.1    2020-04-20 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1308"><a href="#cb120-1308" aria-hidden="true" tabindex="-1"></a><span class="in"> usethis                1.6.1    2020-04-29 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1309"><a href="#cb120-1309" aria-hidden="true" tabindex="-1"></a><span class="in"> vctrs                  0.3.0    2020-05-11 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1310"><a href="#cb120-1310" aria-hidden="true" tabindex="-1"></a><span class="in"> withr                  2.2.0    2020-04-20 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-1311"><a href="#cb120-1311" aria-hidden="true" tabindex="-1"></a><span class="in"> XVector              * 0.28.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-1312"><a href="#cb120-1312" aria-hidden="true" tabindex="-1"></a><span class="in"> zlibbioc               1.34.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-1313"><a href="#cb120-1313" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1314"><a href="#cb120-1314" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb120-1315"><a href="#cb120-1315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1316"><a href="#cb120-1316" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br/&gt;</span></span>
<span id="cb120-1317"><a href="#cb120-1317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1318"><a href="#cb120-1318" aria-hidden="true" tabindex="-1"></a><span class="fu"># ITS Data</span></span>
<span id="cb120-1319"><a href="#cb120-1319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1322"><a href="#cb120-1322" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1323"><a href="#cb120-1323" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb120-1324"><a href="#cb120-1324" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb120-1325"><a href="#cb120-1325" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1326"><a href="#cb120-1326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1329"><a href="#cb120-1329" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1330"><a href="#cb120-1330" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1331"><a href="#cb120-1331" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-1332"><a href="#cb120-1332" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb120-1333"><a href="#cb120-1333" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the caption(s) with captioner</span></span>
<span id="cb120-1334"><a href="#cb120-1334" aria-hidden="true" tabindex="-1"></a>caption_tab <span class="ot">&lt;-</span> <span class="fu">captioner</span>(<span class="at">prefix =</span> <span class="st">"(ITS) Table"</span>, <span class="at">suffix =</span> <span class="st">" |"</span>, <span class="at">style =</span> <span class="st">"b"</span>)</span>
<span id="cb120-1335"><a href="#cb120-1335" aria-hidden="true" tabindex="-1"></a>caption_fig <span class="ot">&lt;-</span> <span class="fu">captioner</span>(<span class="at">prefix =</span> <span class="st">"(ITS) Figure"</span>, <span class="at">suffix =</span> <span class="st">" |"</span>, <span class="at">style =</span> <span class="st">"b"</span>)</span>
<span id="cb120-1336"><a href="#cb120-1336" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a function for referring to the tables in text</span></span>
<span id="cb120-1337"><a href="#cb120-1337" aria-hidden="true" tabindex="-1"></a>ref <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">str_extract</span>(x, <span class="st">"[^|]*"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb120-1338"><a href="#cb120-1338" aria-hidden="true" tabindex="-1"></a>  <span class="fu">trimws</span>(<span class="at">which =</span> <span class="st">"right"</span>, <span class="at">whitespace =</span> <span class="st">"[ ]"</span>)</span>
<span id="cb120-1339"><a href="#cb120-1339" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1340"><a href="#cb120-1340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1343"><a href="#cb120-1343" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1344"><a href="#cb120-1344" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1345"><a href="#cb120-1345" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-1346"><a href="#cb120-1346" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb120-1347"><a href="#cb120-1347" aria-hidden="true" tabindex="-1"></a><span class="do">## Table captions</span></span>
<span id="cb120-1348"><a href="#cb120-1348" aria-hidden="true" tabindex="-1"></a><span class="fu">caption_tab</span>(<span class="st">"seq_dets_its"</span>, <span class="st">"Sample data &amp; associated sequencing information."</span>)</span>
<span id="cb120-1349"><a href="#cb120-1349" aria-hidden="true" tabindex="-1"></a><span class="fu">caption_tab</span>(<span class="st">"seq_cutadapt_its"</span>, <span class="st">"Total reads per sample after prefiltering and primer removal (using cutadapt)."</span>)</span>
<span id="cb120-1350"><a href="#cb120-1350" aria-hidden="true" tabindex="-1"></a><span class="fu">caption_tab</span>(<span class="st">"filter_its"</span>, <span class="st">"Total reads per sample after filtering."</span>)</span>
<span id="cb120-1351"><a href="#cb120-1351" aria-hidden="true" tabindex="-1"></a><span class="fu">caption_tab</span>(<span class="st">"pooling_summary_its"</span>, <span class="st">"Total number of reads, total number of ASVs, minimum and maximum ASVs, followed by the number of singletons, doubletons, etc. for pooling options."</span>)</span>
<span id="cb120-1352"><a href="#cb120-1352" aria-hidden="true" tabindex="-1"></a><span class="fu">caption_tab</span>(<span class="st">"pooling_its"</span>, <span class="st">"Total number of reads and ASVs by sample for pooling options."</span>)</span>
<span id="cb120-1353"><a href="#cb120-1353" aria-hidden="true" tabindex="-1"></a><span class="fu">caption_tab</span>(<span class="st">"track_its"</span>, <span class="st">"Tracking reads changes at each step of the DADA2 workflow."</span>)</span>
<span id="cb120-1354"><a href="#cb120-1354" aria-hidden="true" tabindex="-1"></a><span class="co">#&lt;small&gt;`r caption_tab("qual_scores_after_its")`&lt;/small&gt;</span></span>
<span id="cb120-1355"><a href="#cb120-1355" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1356"><a href="#cb120-1356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1359"><a href="#cb120-1359" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1360"><a href="#cb120-1360" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1361"><a href="#cb120-1361" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-1362"><a href="#cb120-1362" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb120-1363"><a href="#cb120-1363" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure captions</span></span>
<span id="cb120-1364"><a href="#cb120-1364" aria-hidden="true" tabindex="-1"></a><span class="fu">caption_fig</span>(<span class="st">"qual_scores_its"</span>, <span class="st">"Aggregated quality score plots for forward (left) &amp; reverse (right) reads before filtering."</span>)</span>
<span id="cb120-1365"><a href="#cb120-1365" aria-hidden="true" tabindex="-1"></a><span class="fu">caption_fig</span>(<span class="st">"qual_scores_after_its"</span>, <span class="st">"Aggregated quality score plot for forward reads after prefilter &amp; primer removal."</span>)</span>
<span id="cb120-1366"><a href="#cb120-1366" aria-hidden="true" tabindex="-1"></a><span class="fu">caption_fig</span>(<span class="st">"error_F_its"</span>, <span class="st">"Forward reads: Observed frequency of each transition (e.g., T -&gt; G) as a function of the associated quality score."</span>)</span>
<span id="cb120-1367"><a href="#cb120-1367" aria-hidden="true" tabindex="-1"></a><span class="fu">caption_fig</span>(<span class="st">"error_R_its"</span>, <span class="st">"Reverse reads: Observed frequency of each transition (e.g., T -&gt; G) as a function of the associated quality score."</span>)</span>
<span id="cb120-1368"><a href="#cb120-1368" aria-hidden="true" tabindex="-1"></a><span class="fu">caption_fig</span>(<span class="st">"read_length_its"</span>, <span class="st">"Distribution of read length by total ASVs before removing extreme length variants."</span>)</span>
<span id="cb120-1369"><a href="#cb120-1369" aria-hidden="true" tabindex="-1"></a><span class="co">#&lt;small&gt;`r caption_fig("read_length_its")`&lt;/small&gt;</span></span>
<span id="cb120-1370"><a href="#cb120-1370" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1371"><a href="#cb120-1371" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1372"><a href="#cb120-1372" aria-hidden="true" tabindex="-1"></a><span class="fu">## Workflow Input</span></span>
<span id="cb120-1373"><a href="#cb120-1373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1374"><a href="#cb120-1374" aria-hidden="true" tabindex="-1"></a>All files needed to run this workflow can be downloaded from figshare. </span>
<span id="cb120-1375"><a href="#cb120-1375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1376"><a href="#cb120-1376" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;iframe</span> <span class="er">src</span><span class="ot">=</span><span class="st">"https://widgets.figshare.com/articles/14686755/embed?show_title=1"</span> <span class="er">width</span><span class="ot">=</span><span class="st">"568"</span> <span class="er">height</span><span class="ot">=</span><span class="st">"351"</span> <span class="er">allowfullscreen</span> <span class="er">frameborder</span><span class="ot">=</span><span class="st">"0"</span><span class="kw">&gt;&lt;/iframe&gt;</span></span>
<span id="cb120-1377"><a href="#cb120-1377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1378"><a href="#cb120-1378" aria-hidden="true" tabindex="-1"></a><span class="fu">## Overview</span></span>
<span id="cb120-1379"><a href="#cb120-1379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1380"><a href="#cb120-1380" aria-hidden="true" tabindex="-1"></a><span class="fu">### Sequence Files &amp; Samples</span></span>
<span id="cb120-1381"><a href="#cb120-1381" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1382"><a href="#cb120-1382" aria-hidden="true" tabindex="-1"></a>We sequenced a total of 15 samples collected from 10 different plots at a single depth in each plot.</span>
<span id="cb120-1383"><a href="#cb120-1383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1386"><a href="#cb120-1386" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1387"><a href="#cb120-1387" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1388"><a href="#cb120-1388" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-1389"><a href="#cb120-1389" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"files/dada2/tables/its18_sample_seq_info.txt"</span>,</span>
<span id="cb120-1390"><a href="#cb120-1390" aria-hidden="true" tabindex="-1"></a>                        <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb120-1391"><a href="#cb120-1391" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1392"><a href="#cb120-1392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1393"><a href="#cb120-1393" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab("seq_dets_its")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-1394"><a href="#cb120-1394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1397"><a href="#cb120-1397" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1398"><a href="#cb120-1398" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1399"><a href="#cb120-1399" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-1400"><a href="#cb120-1400" aria-hidden="true" tabindex="-1"></a>seq_table <span class="sc">%&gt;%</span></span>
<span id="cb120-1401"><a href="#cb120-1401" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb120-1402"><a href="#cb120-1402" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"its18_sample_seq_info"</span>,</span>
<span id="cb120-1403"><a href="#cb120-1403" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb120-1404"><a href="#cb120-1404" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb120-1405"><a href="#cb120-1405" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb120-1406"><a href="#cb120-1406" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb120-1407"><a href="#cb120-1407" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-1408"><a href="#cb120-1408" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb120-1409"><a href="#cb120-1409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1410"><a href="#cb120-1410" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb120-1411"><a href="#cb120-1411" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb120-1412"><a href="#cb120-1412" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb120-1413"><a href="#cb120-1413" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">1</span>),</span>
<span id="cb120-1414"><a href="#cb120-1414" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">TRUE</span>, <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-1415"><a href="#cb120-1415" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb120-1416"><a href="#cb120-1416" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-1417"><a href="#cb120-1417" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb120-1418"><a href="#cb120-1418" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Sample ID"</span>, </span>
<span id="cb120-1419"><a href="#cb120-1419" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb120-1420"><a href="#cb120-1420" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>),</span>
<span id="cb120-1421"><a href="#cb120-1421" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>), </span>
<span id="cb120-1422"><a href="#cb120-1422" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb120-1423"><a href="#cb120-1423" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">150</span>, <span class="at">footer =</span> <span class="st">"Total reads"</span>), </span>
<span id="cb120-1424"><a href="#cb120-1424" aria-hidden="true" tabindex="-1"></a>    <span class="at">raw_reads =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values)))</span>
<span id="cb120-1425"><a href="#cb120-1425" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-1426"><a href="#cb120-1426" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">5</span>, </span>
<span id="cb120-1427"><a href="#cb120-1427" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb120-1428"><a href="#cb120-1428" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb120-1429"><a href="#cb120-1429" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb120-1430"><a href="#cb120-1430" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-1431"><a href="#cb120-1431" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb120-1432"><a href="#cb120-1432" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1433"><a href="#cb120-1433" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1434"><a href="#cb120-1434" aria-hidden="true" tabindex="-1"></a><span class="fu">### Workflow</span></span>
<span id="cb120-1435"><a href="#cb120-1435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1436"><a href="#cb120-1436" aria-hidden="true" tabindex="-1"></a>Our workflow is pretty much ripped from the <span class="co">[</span><span class="ot">DADA2 ITS Workflow (1.8)</span><span class="co">](https://benjjneb.github.io/dada2/ITS_workflow.html)</span> on the DADA2 website. That webpage contains thorough explanations for each step so we will not repeat most of that here. For more details, check out the post. The workflow consists of the following steps:</span>
<span id="cb120-1437"><a href="#cb120-1437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1438"><a href="#cb120-1438" aria-hidden="true" tabindex="-1"></a>| Step | Command                | What weâ€™re doing                                           |</span>
<span id="cb120-1439"><a href="#cb120-1439" aria-hidden="true" tabindex="-1"></a>|------|------------------------|------------------------------------------------------------|</span>
<span id="cb120-1440"><a href="#cb120-1440" aria-hidden="true" tabindex="-1"></a>| 1    | multiple               | prepare input file names &amp; paths                           |</span>
<span id="cb120-1441"><a href="#cb120-1441" aria-hidden="true" tabindex="-1"></a>| 2    | multiple               | Define  primers (all orientations)                         |</span>
<span id="cb120-1442"><a href="#cb120-1442" aria-hidden="true" tabindex="-1"></a>| 3    | <span class="in">`cutadapt`</span>             | Remove  primers                                            |</span>
<span id="cb120-1443"><a href="#cb120-1443" aria-hidden="true" tabindex="-1"></a>| 4    | <span class="in">`filterAndTrim()`</span>      | assess quality &amp; filter reads                              |</span>
<span id="cb120-1444"><a href="#cb120-1444" aria-hidden="true" tabindex="-1"></a>| 5    | <span class="in">`learnErrors()`</span>        | generate an error model for the data                       |</span>
<span id="cb120-1445"><a href="#cb120-1445" aria-hidden="true" tabindex="-1"></a>| 6    | <span class="in">`derepFastq`</span>           | dereplicate sequences                                      |</span>
<span id="cb120-1446"><a href="#cb120-1446" aria-hidden="true" tabindex="-1"></a>| 7    | <span class="in">`dada()`</span>               | infer ASVs on both forward &amp; reverse reads independently   |</span>
<span id="cb120-1447"><a href="#cb120-1447" aria-hidden="true" tabindex="-1"></a>| 8    | <span class="in">`mergePairs()`</span>         | merge forward &amp; reverse reads to further refine ASVs       |</span>
<span id="cb120-1448"><a href="#cb120-1448" aria-hidden="true" tabindex="-1"></a>| 9    | <span class="in">`makeSequenceTable()`</span>  | generate a count table                                     |</span>
<span id="cb120-1449"><a href="#cb120-1449" aria-hidden="true" tabindex="-1"></a>| 10   | <span class="in">`removeBimeraDenovo()`</span> | screen for &amp; remove chimeras                               |</span>
<span id="cb120-1450"><a href="#cb120-1450" aria-hidden="true" tabindex="-1"></a>| 11   |                        | track reads through workflow                               |</span>
<span id="cb120-1451"><a href="#cb120-1451" aria-hidden="true" tabindex="-1"></a>| 12   | <span class="in">`assignTaxonomy()`</span>     | assign taxonomy &amp; finish workflow                          |</span>
<span id="cb120-1452"><a href="#cb120-1452" aria-hidden="true" tabindex="-1"></a>| 13   | <span class="in">`save.image()`</span>         | save an image of seqtab &amp; taxtab for next part of workflow |</span>
<span id="cb120-1453"><a href="#cb120-1453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1454"><a href="#cb120-1454" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1455"><a href="#cb120-1455" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1. Set Working Environment</span></span>
<span id="cb120-1456"><a href="#cb120-1456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1457"><a href="#cb120-1457" aria-hidden="true" tabindex="-1"></a>Next, we need to setup the working environment by renaming the fastq files and defining a path for the working directory.</span>
<span id="cb120-1458"><a href="#cb120-1458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1459"><a href="#cb120-1459" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Rename samples</span></span>
<span id="cb120-1460"><a href="#cb120-1460" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1461"><a href="#cb120-1461" aria-hidden="true" tabindex="-1"></a>To make the parsing easier, we will eliminate the lane and well number from each sample. If you do not wish to do that you will need to adjust the code accordingly. I am *positive* there are more elegant ways of doing this.</span>
<span id="cb120-1462"><a href="#cb120-1462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1463"><a href="#cb120-1463" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; CAUTION. You should check that this code works on a few backup files before proceeding.</span></span>
<span id="cb120-1464"><a href="#cb120-1464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1467"><a href="#cb120-1467" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1468"><a href="#cb120-1468" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb120-1469"><a href="#cb120-1469" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"RAW"</span>)</span>
<span id="cb120-1470"><a href="#cb120-1470" aria-hidden="true" tabindex="-1"></a>orig_fastq <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">pattern =</span> <span class="st">"*.fastq.gz"</span>)</span>
<span id="cb120-1471"><a href="#cb120-1471" aria-hidden="true" tabindex="-1"></a>newname_fastq <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_S.*_L001"</span>, <span class="st">""</span>, orig_fastq)</span>
<span id="cb120-1472"><a href="#cb120-1472" aria-hidden="true" tabindex="-1"></a>newname_fastq <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_001"</span>, <span class="st">""</span>, newname_fastq)</span>
<span id="cb120-1473"><a href="#cb120-1473" aria-hidden="true" tabindex="-1"></a><span class="fu">file.rename</span>(orig_fastq, newname_fastq)</span>
<span id="cb120-1474"><a href="#cb120-1474" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../"</span>)</span>
<span id="cb120-1475"><a href="#cb120-1475" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1476"><a href="#cb120-1476" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1479"><a href="#cb120-1479" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1480"><a href="#cb120-1480" aria-hidden="true" tabindex="-1"></a>path <span class="ot">&lt;-</span> <span class="st">"RAW"</span></span>
<span id="cb120-1481"><a href="#cb120-1481" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">list.files</span>(path))</span>
<span id="cb120-1482"><a href="#cb120-1482" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1483"><a href="#cb120-1483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1486"><a href="#cb120-1486" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1487"><a href="#cb120-1487" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1488"><a href="#cb120-1488" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-1489"><a href="#cb120-1489" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="st">"P1_R1.fastq.gz"</span>, <span class="st">"P1_R2.fastq.gz"</span>, <span class="st">"P1-8C_R1.fastq.gz"</span>,</span>
<span id="cb120-1490"><a href="#cb120-1490" aria-hidden="true" tabindex="-1"></a>  <span class="st">"P1-8C_R2.fastq.gz"</span>, <span class="st">"P10_R1.fastq.gz"</span>, <span class="st">"P10_R2.fastq.gz"</span>)</span>
<span id="cb120-1491"><a href="#cb120-1491" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1492"><a href="#cb120-1492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1493"><a href="#cb120-1493" aria-hidden="true" tabindex="-1"></a>Then, we generate matched lists of the forward and reverse read files. We also parse out the sample name.</span>
<span id="cb120-1494"><a href="#cb120-1494" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1497"><a href="#cb120-1497" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1498"><a href="#cb120-1498" aria-hidden="true" tabindex="-1"></a>fnFs <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern =</span> <span class="st">"_R1.fastq.gz"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb120-1499"><a href="#cb120-1499" aria-hidden="true" tabindex="-1"></a>fnRs <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern =</span> <span class="st">"_R2.fastq.gz"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb120-1500"><a href="#cb120-1500" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1501"><a href="#cb120-1501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1502"><a href="#cb120-1502" aria-hidden="true" tabindex="-1"></a><span class="fu">## 2. Define Primers</span></span>
<span id="cb120-1503"><a href="#cb120-1503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1504"><a href="#cb120-1504" aria-hidden="true" tabindex="-1"></a>Before we start the DADA2 workflow we need to run <span class="co">[</span><span class="ot">catadapt</span><span class="co">](https://github.com/marcelm/cutadapt)</span> <span class="co">[</span><span class="ot">@martin2011cutadapt</span><span class="co">]</span> on all <span class="in">`fastq.gz`</span> files to trim the primers. For this part of the study we used  the primer pair ITS1f (CTTGGTCATTTAGAGGAAGTAA) <span class="co">[</span><span class="ot">@gardes1993its</span><span class="co">]</span> and ITS2   (GCTGCGTTCTTCATCGATGC) <span class="co">[</span><span class="ot">@white1990amplification</span><span class="co">]</span> which should yield variable amplicon lengths between 100 to 400 bp. What we are looking for is potential read-through scenarios that are possible when sequencing the length variable ITS region as described in the <span class="co">[</span><span class="ot">DADA2 ITS Workflow (1.8)</span><span class="co">](https://benjjneb.github.io/dada2/ITS_workflow.html)</span>. Please refer to this document for a complete explanation.</span>
<span id="cb120-1505"><a href="#cb120-1505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1506"><a href="#cb120-1506" aria-hidden="true" tabindex="-1"></a>First we define the primers.</span>
<span id="cb120-1507"><a href="#cb120-1507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1510"><a href="#cb120-1510" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1511"><a href="#cb120-1511" aria-hidden="true" tabindex="-1"></a>FWD <span class="ot">&lt;-</span> <span class="st">"CTTGGTCATTTAGAGGAAGTAA"</span></span>
<span id="cb120-1512"><a href="#cb120-1512" aria-hidden="true" tabindex="-1"></a>REV <span class="ot">&lt;-</span> <span class="st">"GCTGCGTTCTTCATCGATGC"</span></span>
<span id="cb120-1513"><a href="#cb120-1513" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1514"><a href="#cb120-1514" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1515"><a href="#cb120-1515" aria-hidden="true" tabindex="-1"></a>Next, we check the presence and orientation of these primers in the data. To do this we will create all orientations of the input primer sequences. In other words the *Forward*, *Complement*, *Reverse*, and *Reverse Complement* variations.</span>
<span id="cb120-1516"><a href="#cb120-1516" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1519"><a href="#cb120-1519" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1520"><a href="#cb120-1520" aria-hidden="true" tabindex="-1"></a>allOrients <span class="ot">&lt;-</span> <span class="cf">function</span>(primer) {</span>
<span id="cb120-1521"><a href="#cb120-1521" aria-hidden="true" tabindex="-1"></a>    <span class="fu">require</span>(Biostrings)</span>
<span id="cb120-1522"><a href="#cb120-1522" aria-hidden="true" tabindex="-1"></a>    dna <span class="ot">&lt;-</span> <span class="fu">DNAString</span>(primer)</span>
<span id="cb120-1523"><a href="#cb120-1523" aria-hidden="true" tabindex="-1"></a>    orients <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">Forward =</span> dna,</span>
<span id="cb120-1524"><a href="#cb120-1524" aria-hidden="true" tabindex="-1"></a>                 <span class="at">Complement =</span> <span class="fu">complement</span>(dna),</span>
<span id="cb120-1525"><a href="#cb120-1525" aria-hidden="true" tabindex="-1"></a>                 <span class="at">Reverse =</span> <span class="fu">reverse</span>(dna),</span>
<span id="cb120-1526"><a href="#cb120-1526" aria-hidden="true" tabindex="-1"></a>                 <span class="at">RevComp =</span> <span class="fu">reverseComplement</span>(dna))</span>
<span id="cb120-1527"><a href="#cb120-1527" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">sapply</span>(orients, toString))</span>
<span id="cb120-1528"><a href="#cb120-1528" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb120-1529"><a href="#cb120-1529" aria-hidden="true" tabindex="-1"></a>FWD.orients <span class="ot">&lt;-</span> <span class="fu">allOrients</span>(FWD)</span>
<span id="cb120-1530"><a href="#cb120-1530" aria-hidden="true" tabindex="-1"></a>REV.orients <span class="ot">&lt;-</span> <span class="fu">allOrients</span>(REV)</span>
<span id="cb120-1531"><a href="#cb120-1531" aria-hidden="true" tabindex="-1"></a>FWD.orients</span>
<span id="cb120-1532"><a href="#cb120-1532" aria-hidden="true" tabindex="-1"></a>REV.orients</span>
<span id="cb120-1533"><a href="#cb120-1533" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1534"><a href="#cb120-1534" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1535"><a href="#cb120-1535" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1536"><a href="#cb120-1536" aria-hidden="true" tabindex="-1"></a><span class="in">                 Forward               Complement                  Reverse</span></span>
<span id="cb120-1537"><a href="#cb120-1537" aria-hidden="true" tabindex="-1"></a><span class="in">"CTTGGTCATTTAGAGGAAGTAA" "GAACCAGTAAATCTCCTTCATT" "AATGAAGGAGATTTACTGGTTC"</span></span>
<span id="cb120-1538"><a href="#cb120-1538" aria-hidden="true" tabindex="-1"></a><span class="in">                 RevComp</span></span>
<span id="cb120-1539"><a href="#cb120-1539" aria-hidden="true" tabindex="-1"></a><span class="in">"TTACTTCCTCTAAATGACCAAG"</span></span>
<span id="cb120-1540"><a href="#cb120-1540" aria-hidden="true" tabindex="-1"></a><span class="in">               Forward             Complement                Reverse</span></span>
<span id="cb120-1541"><a href="#cb120-1541" aria-hidden="true" tabindex="-1"></a><span class="in">"GCTGCGTTCTTCATCGATGC" "CGACGCAAGAAGTAGCTACG" "CGTAGCTACTTCTTGCGTCG"</span></span>
<span id="cb120-1542"><a href="#cb120-1542" aria-hidden="true" tabindex="-1"></a><span class="in">               RevComp</span></span>
<span id="cb120-1543"><a href="#cb120-1543" aria-hidden="true" tabindex="-1"></a><span class="in">"GCATCGATGAAGAACGCAGC"</span></span>
<span id="cb120-1544"><a href="#cb120-1544" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1545"><a href="#cb120-1545" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1546"><a href="#cb120-1546" aria-hidden="true" tabindex="-1"></a>Now we do a little pre-filter step to eliminate ambiguous bases (*Ns*) because *Ns* make mapping of short primer sequences difficult. This step removes any reads with *Ns*. Again, set some files paths, this time for the filtered reads.</span>
<span id="cb120-1547"><a href="#cb120-1547" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1550"><a href="#cb120-1550" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1551"><a href="#cb120-1551" aria-hidden="true" tabindex="-1"></a>fnFs.filtN <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">"filtN"</span>, <span class="fu">basename</span>(fnFs))</span>
<span id="cb120-1552"><a href="#cb120-1552" aria-hidden="true" tabindex="-1"></a>fnRs.filtN <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">"filtN"</span>, <span class="fu">basename</span>(fnRs))</span>
<span id="cb120-1553"><a href="#cb120-1553" aria-hidden="true" tabindex="-1"></a><span class="fu">filterAndTrim</span>(fnFs, fnFs.filtN, fnRs, fnRs.filtN,</span>
<span id="cb120-1554"><a href="#cb120-1554" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxN =</span> <span class="dv">0</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-1555"><a href="#cb120-1555" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1556"><a href="#cb120-1556" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1557"><a href="#cb120-1557" aria-hidden="true" tabindex="-1"></a>Sweet. Time to assess the number of times a primer (and all possible primer orientation) appear in the forward and reverse reads. According to the workflow, counting the primers on one set of paired end fastq files is sufficient to see if there is a problem. This assumes that all the files were created using the same library prep. Basically for both primers, we will search for all four orientations in both forward and reverse reads</span>
<span id="cb120-1558"><a href="#cb120-1558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1561"><a href="#cb120-1561" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1562"><a href="#cb120-1562" aria-hidden="true" tabindex="-1"></a>sampnum <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb120-1563"><a href="#cb120-1563" aria-hidden="true" tabindex="-1"></a>primerHits <span class="ot">&lt;-</span> <span class="cf">function</span>(primer, fn) {</span>
<span id="cb120-1564"><a href="#cb120-1564" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Counts number of reads in which the primer is found</span></span>
<span id="cb120-1565"><a href="#cb120-1565" aria-hidden="true" tabindex="-1"></a>    nhits <span class="ot">&lt;-</span> <span class="fu">vcountPattern</span>(primer, <span class="fu">sread</span>(<span class="fu">readFastq</span>(fn)), <span class="at">fixed =</span> <span class="cn">FALSE</span>)</span>
<span id="cb120-1566"><a href="#cb120-1566" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">sum</span>(nhits <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb120-1567"><a href="#cb120-1567" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb120-1568"><a href="#cb120-1568" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1569"><a href="#cb120-1569" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1570"><a href="#cb120-1570" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Forward primers</span></span>
<span id="cb120-1571"><a href="#cb120-1571" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1574"><a href="#cb120-1574" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1575"><a href="#cb120-1575" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits,</span>
<span id="cb120-1576"><a href="#cb120-1576" aria-hidden="true" tabindex="-1"></a>                                <span class="at">fn =</span> fnFs.filtN[[sampnum]]),</span>
<span id="cb120-1577"><a href="#cb120-1577" aria-hidden="true" tabindex="-1"></a>    <span class="at">FWD.ReverseReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits,</span>
<span id="cb120-1578"><a href="#cb120-1578" aria-hidden="true" tabindex="-1"></a>                              <span class="at">fn =</span> fnRs.filtN[[sampnum]]))</span>
<span id="cb120-1579"><a href="#cb120-1579" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1580"><a href="#cb120-1580" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1581"><a href="#cb120-1581" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1582"><a href="#cb120-1582" aria-hidden="true" tabindex="-1"></a><span class="in">                 Forward Complement Reverse RevComp</span></span>
<span id="cb120-1583"><a href="#cb120-1583" aria-hidden="true" tabindex="-1"></a><span class="in">FWD.ForwardReads   48467          0       0       0</span></span>
<span id="cb120-1584"><a href="#cb120-1584" aria-hidden="true" tabindex="-1"></a><span class="in">FWD.ReverseReads       0          0       0    5078</span></span>
<span id="cb120-1585"><a href="#cb120-1585" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1586"><a href="#cb120-1586" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1587"><a href="#cb120-1587" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Reverse primers</span></span>
<span id="cb120-1588"><a href="#cb120-1588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1591"><a href="#cb120-1591" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1592"><a href="#cb120-1592" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits,</span>
<span id="cb120-1593"><a href="#cb120-1593" aria-hidden="true" tabindex="-1"></a>                                <span class="at">fn =</span> fnFs.filtN[[sampnum]]),</span>
<span id="cb120-1594"><a href="#cb120-1594" aria-hidden="true" tabindex="-1"></a>       <span class="at">REV.ReverseReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits,</span>
<span id="cb120-1595"><a href="#cb120-1595" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">fn =</span> fnRs.filtN[[sampnum]]))</span>
<span id="cb120-1596"><a href="#cb120-1596" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1597"><a href="#cb120-1597" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1598"><a href="#cb120-1598" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1599"><a href="#cb120-1599" aria-hidden="true" tabindex="-1"></a><span class="in">                 Forward Complement Reverse RevComp</span></span>
<span id="cb120-1600"><a href="#cb120-1600" aria-hidden="true" tabindex="-1"></a><span class="in">REV.ForwardReads       0          0       0    6237</span></span>
<span id="cb120-1601"><a href="#cb120-1601" aria-hidden="true" tabindex="-1"></a><span class="in">REV.ReverseReads   43902          0       0       0</span></span>
<span id="cb120-1602"><a href="#cb120-1602" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1603"><a href="#cb120-1603" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1606"><a href="#cb120-1606" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1607"><a href="#cb120-1607" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1608"><a href="#cb120-1608" aria-hidden="true" tabindex="-1"></a>fnFsXX <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern =</span> <span class="st">"_R1.fastq.gz"</span>, <span class="at">full.names =</span> <span class="cn">FALSE</span>))</span>
<span id="cb120-1609"><a href="#cb120-1609" aria-hidden="true" tabindex="-1"></a>qsamp <span class="ot">&lt;-</span> fnFsXX[sampnum]</span>
<span id="cb120-1610"><a href="#cb120-1610" aria-hidden="true" tabindex="-1"></a>qsamp <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(qsamp, <span class="st">"_"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb120-1611"><a href="#cb120-1611" aria-hidden="true" tabindex="-1"></a><span class="do">### add this for inline calc `r countLines(fnFs.filtN[[sampnum]])/4`</span></span>
<span id="cb120-1612"><a href="#cb120-1612" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1613"><a href="#cb120-1613" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1614"><a href="#cb120-1614" aria-hidden="true" tabindex="-1"></a>What does this table mean? I wondered the same thing. Let's break it down. Sample P1 had  50,894 sequences after the original filtering described earlier. The code searched the forward and reverse fastq files for all 8 primers. If we look at the two outputs, we see the forward primer is found in the forward reads in its forward orientation but also in some reverse reads in its reverse-complement orientation. The reverse primer is found in the reverse reads in its forward orientation but also in some forward reads in its reverse-complement orientation. This is due to read-through when the ITS region is short.</span>
<span id="cb120-1615"><a href="#cb120-1615" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1616"><a href="#cb120-1616" aria-hidden="true" tabindex="-1"></a><span class="fu">## 3. Remove Primers</span></span>
<span id="cb120-1617"><a href="#cb120-1617" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1618"><a href="#cb120-1618" aria-hidden="true" tabindex="-1"></a>Now we can run <span class="co">[</span><span class="ot">catadapt</span><span class="co">](https://github.com/marcelm/cutadapt)</span> <span class="co">[</span><span class="ot">@martin2011cutadapt</span><span class="co">]</span> to remove the primers from the fastq sequences. A little setup first. If this command executes successfully it means R has found cutadapt.</span>
<span id="cb120-1619"><a href="#cb120-1619" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1622"><a href="#cb120-1622" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1623"><a href="#cb120-1623" aria-hidden="true" tabindex="-1"></a>cutadapt <span class="ot">&lt;-</span> <span class="st">"/PATH/to/cutadapt"</span></span>
<span id="cb120-1624"><a href="#cb120-1624" aria-hidden="true" tabindex="-1"></a><span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="st">"--version"</span>) <span class="co"># Run shell commands from R</span></span>
<span id="cb120-1625"><a href="#cb120-1625" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1626"><a href="#cb120-1626" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1627"><a href="#cb120-1627" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1628"><a href="#cb120-1628" aria-hidden="true" tabindex="-1"></a><span class="in">2.8</span></span>
<span id="cb120-1629"><a href="#cb120-1629" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1630"><a href="#cb120-1630" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1631"><a href="#cb120-1631" aria-hidden="true" tabindex="-1"></a>We set paths and trim the forward primer and the reverse-complement of the reverse primer off of R1 (forward reads) and trim the reverse primer and the reverse-complement of the forward primer off of R2 (reverse reads).</span>
<span id="cb120-1632"><a href="#cb120-1632" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1635"><a href="#cb120-1635" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1636"><a href="#cb120-1636" aria-hidden="true" tabindex="-1"></a>path.cut <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">"cutadapt"</span>)</span>
<span id="cb120-1637"><a href="#cb120-1637" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(path.cut)) <span class="fu">dir.create</span>(path.cut)</span>
<span id="cb120-1638"><a href="#cb120-1638" aria-hidden="true" tabindex="-1"></a>fnFs.cut <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(fnFs))</span>
<span id="cb120-1639"><a href="#cb120-1639" aria-hidden="true" tabindex="-1"></a>fnRs.cut <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(fnRs))</span>
<span id="cb120-1640"><a href="#cb120-1640" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1641"><a href="#cb120-1641" aria-hidden="true" tabindex="-1"></a>FWD.RC <span class="ot">&lt;-</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(FWD)</span>
<span id="cb120-1642"><a href="#cb120-1642" aria-hidden="true" tabindex="-1"></a>REV.RC <span class="ot">&lt;-</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(REV)</span>
<span id="cb120-1643"><a href="#cb120-1643" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1644"><a href="#cb120-1644" aria-hidden="true" tabindex="-1"></a>R1.flags <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"-g"</span>, FWD, <span class="st">"-a"</span>, REV.RC)</span>
<span id="cb120-1645"><a href="#cb120-1645" aria-hidden="true" tabindex="-1"></a>R2.flags <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"-G"</span>, REV, <span class="st">"-A"</span>, FWD.RC)</span>
<span id="cb120-1646"><a href="#cb120-1646" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1647"><a href="#cb120-1647" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(fnFs)) {<span class="fu">system2</span>(cutadapt,</span>
<span id="cb120-1648"><a href="#cb120-1648" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">args =</span> <span class="fu">c</span>(R1.flags, R2.flags,</span>
<span id="cb120-1649"><a href="#cb120-1649" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">"-m"</span>, <span class="dv">20</span>, <span class="st">"-n"</span>, <span class="dv">2</span>,</span>
<span id="cb120-1650"><a href="#cb120-1650" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">"-o"</span>, fnFs.cut[i],</span>
<span id="cb120-1651"><a href="#cb120-1651" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">"-p"</span>, fnRs.cut[i],</span>
<span id="cb120-1652"><a href="#cb120-1652" aria-hidden="true" tabindex="-1"></a>                                            fnFs.filtN[i], fnRs.filtN[i]))}</span>
<span id="cb120-1653"><a href="#cb120-1653" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1654"><a href="#cb120-1654" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1655"><a href="#cb120-1655" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1656"><a href="#cb120-1656" aria-hidden="true" tabindex="-1"></a><span class="in">This is cutadapt 2.8 with Python 3.7.6</span></span>
<span id="cb120-1657"><a href="#cb120-1657" aria-hidden="true" tabindex="-1"></a><span class="in">Command line parameters: -g CTTGGTCATTTAGAGGAAGTAA -a GCATCGATGAAGAACGCAGC \</span></span>
<span id="cb120-1658"><a href="#cb120-1658" aria-hidden="true" tabindex="-1"></a><span class="in">                         -G GCTGCGTTCTTCATCGATGC -A TTACTTCCTCTAAATGACCAAG \</span></span>
<span id="cb120-1659"><a href="#cb120-1659" aria-hidden="true" tabindex="-1"></a><span class="in">                         -m 20 -n 2 \</span></span>
<span id="cb120-1660"><a href="#cb120-1660" aria-hidden="true" tabindex="-1"></a><span class="in">                         -o RAW/cutadapt/P1_R1.fastq.gz \</span></span>
<span id="cb120-1661"><a href="#cb120-1661" aria-hidden="true" tabindex="-1"></a><span class="in">                         -p RAW/cutadapt/P1_R2.fastq.gz RAW/filtN/P1_R1.fastq.gz RAW/filtN/P1_R2.fastq.gz</span></span>
<span id="cb120-1662"><a href="#cb120-1662" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1663"><a href="#cb120-1663" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1664"><a href="#cb120-1664" aria-hidden="true" tabindex="-1"></a>*Note*. If the code above removes all of the base pairs in a sequence, you will get downstream errors unless you set the `-m` flag. This flag sets the minimum length and reads shorter than this will be discarded. Without this flag, reads of length 0 will be kept and cause issues. *Also, a lot of output will be written to the screen by cutadapt!*</span>
<span id="cb120-1665"><a href="#cb120-1665" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1666"><a href="#cb120-1666" aria-hidden="true" tabindex="-1"></a>We can now count the number of primers in the sequences from the output of cutadapt.</span>
<span id="cb120-1667"><a href="#cb120-1667" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1670"><a href="#cb120-1670" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1671"><a href="#cb120-1671" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(</span>
<span id="cb120-1672"><a href="#cb120-1672" aria-hidden="true" tabindex="-1"></a>  <span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[sampnum]]),</span>
<span id="cb120-1673"><a href="#cb120-1673" aria-hidden="true" tabindex="-1"></a>  <span class="at">FWD.ReverseReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnRs.cut[[sampnum]]),</span>
<span id="cb120-1674"><a href="#cb120-1674" aria-hidden="true" tabindex="-1"></a>  <span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[sampnum]]),</span>
<span id="cb120-1675"><a href="#cb120-1675" aria-hidden="true" tabindex="-1"></a>  <span class="at">REV.ReverseReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnRs.cut[[sampnum]]))</span>
<span id="cb120-1676"><a href="#cb120-1676" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1677"><a href="#cb120-1677" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1678"><a href="#cb120-1678" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1679"><a href="#cb120-1679" aria-hidden="true" tabindex="-1"></a><span class="in">                 Forward Complement Reverse RevComp</span></span>
<span id="cb120-1680"><a href="#cb120-1680" aria-hidden="true" tabindex="-1"></a><span class="in">FWD.ForwardReads       0          0       0       0</span></span>
<span id="cb120-1681"><a href="#cb120-1681" aria-hidden="true" tabindex="-1"></a><span class="in">FWD.ReverseReads       0          0       0       0</span></span>
<span id="cb120-1682"><a href="#cb120-1682" aria-hidden="true" tabindex="-1"></a><span class="in">REV.ForwardReads       0          0       0       0</span></span>
<span id="cb120-1683"><a href="#cb120-1683" aria-hidden="true" tabindex="-1"></a><span class="in">REV.ReverseReads       0          0       0       0</span></span>
<span id="cb120-1684"><a href="#cb120-1684" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1685"><a href="#cb120-1685" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1686"><a href="#cb120-1686" aria-hidden="true" tabindex="-1"></a>Basically, primers are no longer detected in the cutadapted reads. Now, for each sample, we can take a look at how the pre-filtering step and primer removal affected the total number of raw reads.</span>
<span id="cb120-1687"><a href="#cb120-1687" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1690"><a href="#cb120-1690" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1691"><a href="#cb120-1691" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1692"><a href="#cb120-1692" aria-hidden="true" tabindex="-1"></a>temp1 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"files/dada2/tables/its18_read_changes.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-1693"><a href="#cb120-1693" aria-hidden="true" tabindex="-1"></a>temp1[, <span class="dv">4</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="cn">NULL</span>)</span>
<span id="cb120-1694"><a href="#cb120-1694" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(temp1, <span class="st">"files/dada2/tables/its18_cut.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb120-1695"><a href="#cb120-1695" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1696"><a href="#cb120-1696" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1699"><a href="#cb120-1699" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1700"><a href="#cb120-1700" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1701"><a href="#cb120-1701" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-1702"><a href="#cb120-1702" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"files/dada2/tables/its18_cut.txt"</span>,</span>
<span id="cb120-1703"><a href="#cb120-1703" aria-hidden="true" tabindex="-1"></a>                       <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb120-1704"><a href="#cb120-1704" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1705"><a href="#cb120-1705" aria-hidden="true" tabindex="-1"></a>seq_table<span class="sc">$</span>per_reads_kept <span class="ot">&lt;-</span> <span class="fu">round</span>(seq_table<span class="sc">$</span>cutadapt_reads<span class="sc">/</span>seq_table<span class="sc">$</span>raw_reads, </span>
<span id="cb120-1706"><a href="#cb120-1706" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb120-1707"><a href="#cb120-1707" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1708"><a href="#cb120-1708" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1709"><a href="#cb120-1709" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab("seq_cutadapt_its")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-1710"><a href="#cb120-1710" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1713"><a href="#cb120-1713" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1714"><a href="#cb120-1714" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1715"><a href="#cb120-1715" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-1716"><a href="#cb120-1716" aria-hidden="true" tabindex="-1"></a>seq_table <span class="sc">%&gt;%</span></span>
<span id="cb120-1717"><a href="#cb120-1717" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb120-1718"><a href="#cb120-1718" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"its18_cutadapt"</span>,</span>
<span id="cb120-1719"><a href="#cb120-1719" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb120-1720"><a href="#cb120-1720" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb120-1721"><a href="#cb120-1721" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb120-1722"><a href="#cb120-1722" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb120-1723"><a href="#cb120-1723" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-1724"><a href="#cb120-1724" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb120-1725"><a href="#cb120-1725" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1726"><a href="#cb120-1726" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb120-1727"><a href="#cb120-1727" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb120-1728"><a href="#cb120-1728" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb120-1729"><a href="#cb120-1729" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">1</span>),</span>
<span id="cb120-1730"><a href="#cb120-1730" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">FALSE</span>, <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-1731"><a href="#cb120-1731" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb120-1732"><a href="#cb120-1732" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-1733"><a href="#cb120-1733" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb120-1734"><a href="#cb120-1734" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Sample ID"</span>, </span>
<span id="cb120-1735"><a href="#cb120-1735" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb120-1736"><a href="#cb120-1736" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>),</span>
<span id="cb120-1737"><a href="#cb120-1737" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>), </span>
<span id="cb120-1738"><a href="#cb120-1738" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb120-1739"><a href="#cb120-1739" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">150</span>, <span class="at">footer =</span> <span class="st">"Total reads"</span>), </span>
<span id="cb120-1740"><a href="#cb120-1740" aria-hidden="true" tabindex="-1"></a>    <span class="at">raw_reads =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-1741"><a href="#cb120-1741" aria-hidden="true" tabindex="-1"></a>    <span class="at">pre_filt_reads =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-1742"><a href="#cb120-1742" aria-hidden="true" tabindex="-1"></a>    <span class="at">cutadapt_reads =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values)))</span>
<span id="cb120-1743"><a href="#cb120-1743" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-1744"><a href="#cb120-1744" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">5</span>, </span>
<span id="cb120-1745"><a href="#cb120-1745" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb120-1746"><a href="#cb120-1746" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb120-1747"><a href="#cb120-1747" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb120-1748"><a href="#cb120-1748" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-1749"><a href="#cb120-1749" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb120-1750"><a href="#cb120-1750" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1751"><a href="#cb120-1751" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1752"><a href="#cb120-1752" aria-hidden="true" tabindex="-1"></a><span class="fu">## 4. Quality Assessment &amp; Filtering</span></span>
<span id="cb120-1753"><a href="#cb120-1753" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1754"><a href="#cb120-1754" aria-hidden="true" tabindex="-1"></a>We need the forward and reverse fastq file names and the sample names.</span>
<span id="cb120-1755"><a href="#cb120-1755" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1758"><a href="#cb120-1758" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1759"><a href="#cb120-1759" aria-hidden="true" tabindex="-1"></a>cutFs <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path.cut, <span class="at">pattern =</span> <span class="st">"_R1.fastq.gz"</span>,</span>
<span id="cb120-1760"><a href="#cb120-1760" aria-hidden="true" tabindex="-1"></a>                         <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb120-1761"><a href="#cb120-1761" aria-hidden="true" tabindex="-1"></a>cutRs <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path.cut, <span class="at">pattern =</span> <span class="st">"_R2.fastq.gz"</span>,</span>
<span id="cb120-1762"><a href="#cb120-1762" aria-hidden="true" tabindex="-1"></a>                         <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb120-1763"><a href="#cb120-1763" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1764"><a href="#cb120-1764" aria-hidden="true" tabindex="-1"></a>get.sample.name <span class="ot">&lt;-</span> <span class="cf">function</span>(fname) <span class="fu">strsplit</span>(<span class="fu">basename</span>(fname), <span class="st">"_"</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]</span>
<span id="cb120-1765"><a href="#cb120-1765" aria-hidden="true" tabindex="-1"></a>sample.names <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">sapply</span>(cutFs, get.sample.name))</span>
<span id="cb120-1766"><a href="#cb120-1766" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sample.names)</span>
<span id="cb120-1767"><a href="#cb120-1767" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1768"><a href="#cb120-1768" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1771"><a href="#cb120-1771" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1772"><a href="#cb120-1772" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1773"><a href="#cb120-1773" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-1774"><a href="#cb120-1774" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="st">"P1"</span>, <span class="st">"P1-8C"</span>, <span class="st">"P10"</span>, <span class="st">"P2"</span>, <span class="st">"P3"</span>, <span class="st">"P3-8C"</span>)</span>
<span id="cb120-1775"><a href="#cb120-1775" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1776"><a href="#cb120-1776" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1777"><a href="#cb120-1777" aria-hidden="true" tabindex="-1"></a><span class="fu">### Quality plots</span></span>
<span id="cb120-1778"><a href="#cb120-1778" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1779"><a href="#cb120-1779" aria-hidden="true" tabindex="-1"></a>Then we inspect read quality.</span>
<span id="cb120-1780"><a href="#cb120-1780" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1783"><a href="#cb120-1783" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1784"><a href="#cb120-1784" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">plotQualityProfile</span>(cutFs[<span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>], <span class="at">aggregate =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-1785"><a href="#cb120-1785" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">plotQualityProfile</span>(cutRs[<span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>], <span class="at">aggregate =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-1786"><a href="#cb120-1786" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1787"><a href="#cb120-1787" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">grid.arrange</span>(p1, p2, <span class="at">nrow =</span> <span class="dv">1</span>)</span>
<span id="cb120-1788"><a href="#cb120-1788" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"plot_qscores_2018.png"</span>, p3, <span class="at">width =</span> <span class="dv">7</span>, <span class="at">height =</span> <span class="dv">3</span>)</span>
<span id="cb120-1789"><a href="#cb120-1789" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1790"><a href="#cb120-1790" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1793"><a href="#cb120-1793" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1794"><a href="#cb120-1794" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1795"><a href="#cb120-1795" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">"cp files/dada2/figures/its18_plot_qscores.png include/dada2/its18_plot_qscores.png"</span>)</span>
<span id="cb120-1796"><a href="#cb120-1796" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1797"><a href="#cb120-1797" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1800"><a href="#cb120-1800" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1801"><a href="#cb120-1801" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1802"><a href="#cb120-1802" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-1803"><a href="#cb120-1803" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 3</span></span>
<span id="cb120-1804"><a href="#cb120-1804" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb120-1805"><a href="#cb120-1805" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/dada2/its18_plot_qscores.png"</span>)</span>
<span id="cb120-1806"><a href="#cb120-1806" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1807"><a href="#cb120-1807" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("qual_scores_its")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-1808"><a href="#cb120-1808" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1809"><a href="#cb120-1809" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1810"><a href="#cb120-1810" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1811"><a href="#cb120-1811" aria-hidden="true" tabindex="-1"></a><span class="fu">### Filtering</span></span>
<span id="cb120-1812"><a href="#cb120-1812" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1815"><a href="#cb120-1815" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1816"><a href="#cb120-1816" aria-hidden="true" tabindex="-1"></a>filtFs <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut, <span class="st">"filtered"</span>, <span class="fu">basename</span>(cutFs))</span>
<span id="cb120-1817"><a href="#cb120-1817" aria-hidden="true" tabindex="-1"></a>filtRs <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut, <span class="st">"filtered"</span>, <span class="fu">basename</span>(cutRs))</span>
<span id="cb120-1818"><a href="#cb120-1818" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1819"><a href="#cb120-1819" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1820"><a href="#cb120-1820" aria-hidden="true" tabindex="-1"></a>We use a <span class="in">`minLen`</span> value to get rid of very short length sequences.</span>
<span id="cb120-1821"><a href="#cb120-1821" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1824"><a href="#cb120-1824" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1825"><a href="#cb120-1825" aria-hidden="true" tabindex="-1"></a>out <span class="ot">&lt;-</span> <span class="fu">filterAndTrim</span>(cutFs, filtFs, cutRs, filtRs,</span>
<span id="cb120-1826"><a href="#cb120-1826" aria-hidden="true" tabindex="-1"></a>                     <span class="at">maxN =</span> <span class="dv">0</span>, <span class="at">maxEE =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>), <span class="at">truncQ =</span> <span class="dv">2</span>,</span>
<span id="cb120-1827"><a href="#cb120-1827" aria-hidden="true" tabindex="-1"></a>                     <span class="at">minLen =</span> <span class="dv">50</span>, <span class="at">rm.phix =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-1828"><a href="#cb120-1828" aria-hidden="true" tabindex="-1"></a>                     <span class="at">compress =</span> <span class="cn">TRUE</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-1829"><a href="#cb120-1829" aria-hidden="true" tabindex="-1"></a>out</span>
<span id="cb120-1830"><a href="#cb120-1830" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1831"><a href="#cb120-1831" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1832"><a href="#cb120-1832" aria-hidden="true" tabindex="-1"></a>:::{.callout-note}</span>
<span id="cb120-1833"><a href="#cb120-1833" aria-hidden="true" tabindex="-1"></a>These parameters should be set based on the anticipated length of the amplicon and the read quality.</span>
<span id="cb120-1834"><a href="#cb120-1834" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb120-1835"><a href="#cb120-1835" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1836"><a href="#cb120-1836" aria-hidden="true" tabindex="-1"></a>And here is a table of how the filtering step affected the number of reads in each sample.</span>
<span id="cb120-1837"><a href="#cb120-1837" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1838"><a href="#cb120-1838" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1841"><a href="#cb120-1841" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1842"><a href="#cb120-1842" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1843"><a href="#cb120-1843" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-1844"><a href="#cb120-1844" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"files/dada2/tables/its18_filter.txt"</span>,</span>
<span id="cb120-1845"><a href="#cb120-1845" aria-hidden="true" tabindex="-1"></a>                       <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb120-1846"><a href="#cb120-1846" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1847"><a href="#cb120-1847" aria-hidden="true" tabindex="-1"></a>seq_table<span class="sc">$</span>per_reads_kept <span class="ot">&lt;-</span> <span class="fu">round</span>(seq_table<span class="sc">$</span>reads_out<span class="sc">/</span>seq_table<span class="sc">$</span>reads_in, </span>
<span id="cb120-1848"><a href="#cb120-1848" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb120-1849"><a href="#cb120-1849" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1850"><a href="#cb120-1850" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab("filter_its")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-1851"><a href="#cb120-1851" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1854"><a href="#cb120-1854" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1855"><a href="#cb120-1855" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1856"><a href="#cb120-1856" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-1857"><a href="#cb120-1857" aria-hidden="true" tabindex="-1"></a>seq_table <span class="sc">%&gt;%</span></span>
<span id="cb120-1858"><a href="#cb120-1858" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb120-1859"><a href="#cb120-1859" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"its18_filter"</span>,</span>
<span id="cb120-1860"><a href="#cb120-1860" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb120-1861"><a href="#cb120-1861" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb120-1862"><a href="#cb120-1862" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb120-1863"><a href="#cb120-1863" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb120-1864"><a href="#cb120-1864" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-1865"><a href="#cb120-1865" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb120-1866"><a href="#cb120-1866" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1867"><a href="#cb120-1867" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb120-1868"><a href="#cb120-1868" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb120-1869"><a href="#cb120-1869" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb120-1870"><a href="#cb120-1870" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">1</span>),</span>
<span id="cb120-1871"><a href="#cb120-1871" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">FALSE</span>, <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-1872"><a href="#cb120-1872" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb120-1873"><a href="#cb120-1873" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-1874"><a href="#cb120-1874" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb120-1875"><a href="#cb120-1875" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Sample ID"</span>, </span>
<span id="cb120-1876"><a href="#cb120-1876" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb120-1877"><a href="#cb120-1877" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>),</span>
<span id="cb120-1878"><a href="#cb120-1878" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>), </span>
<span id="cb120-1879"><a href="#cb120-1879" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb120-1880"><a href="#cb120-1880" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">150</span>, <span class="at">footer =</span> <span class="st">"Total reads"</span>), </span>
<span id="cb120-1881"><a href="#cb120-1881" aria-hidden="true" tabindex="-1"></a>    <span class="at">reads_in =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-1882"><a href="#cb120-1882" aria-hidden="true" tabindex="-1"></a>    <span class="at">reads_out =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values)))</span>
<span id="cb120-1883"><a href="#cb120-1883" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-1884"><a href="#cb120-1884" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">5</span>, </span>
<span id="cb120-1885"><a href="#cb120-1885" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb120-1886"><a href="#cb120-1886" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb120-1887"><a href="#cb120-1887" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb120-1888"><a href="#cb120-1888" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-1889"><a href="#cb120-1889" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb120-1890"><a href="#cb120-1890" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1891"><a href="#cb120-1891" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1892"><a href="#cb120-1892" aria-hidden="true" tabindex="-1"></a><span class="fu">## 5. Learn Error Rates</span></span>
<span id="cb120-1893"><a href="#cb120-1893" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1894"><a href="#cb120-1894" aria-hidden="true" tabindex="-1"></a>Time to assess the error rate of the data. The rest of the workflow is very similar to the 16S workflows presented previously. So I will basically stop talking.</span>
<span id="cb120-1895"><a href="#cb120-1895" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1896"><a href="#cb120-1896" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Forward Reads</span></span>
<span id="cb120-1897"><a href="#cb120-1897" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1900"><a href="#cb120-1900" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1901"><a href="#cb120-1901" aria-hidden="true" tabindex="-1"></a>errF <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(filtFs, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-1902"><a href="#cb120-1902" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1903"><a href="#cb120-1903" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1904"><a href="#cb120-1904" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1905"><a href="#cb120-1905" aria-hidden="true" tabindex="-1"></a><span class="in">108788735 total bases in 495784 reads from 11 samples will be used for learning the error rates.</span></span>
<span id="cb120-1906"><a href="#cb120-1906" aria-hidden="true" tabindex="-1"></a><span class="in">109358094 total bases in 495784 reads from 11 samples will be used for learning the error rates.</span></span>
<span id="cb120-1907"><a href="#cb120-1907" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1908"><a href="#cb120-1908" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1911"><a href="#cb120-1911" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1912"><a href="#cb120-1912" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errF, <span class="at">nominalQ =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-1913"><a href="#cb120-1913" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1914"><a href="#cb120-1914" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1917"><a href="#cb120-1917" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1918"><a href="#cb120-1918" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1919"><a href="#cb120-1919" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">plotErrors</span>(errF, <span class="at">nominalQ =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-1920"><a href="#cb120-1920" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"plot_errorF_1.png"</span>, p3, <span class="at">width =</span> <span class="dv">7</span>, <span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb120-1921"><a href="#cb120-1921" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"plot_errorF_2.png"</span>, p3)</span>
<span id="cb120-1922"><a href="#cb120-1922" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1923"><a href="#cb120-1923" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1926"><a href="#cb120-1926" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1927"><a href="#cb120-1927" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1928"><a href="#cb120-1928" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">"cp files/dada2/figures/its18_plot_errorF.png include/dada2/its18_plot_errorF.png"</span>)</span>
<span id="cb120-1929"><a href="#cb120-1929" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1930"><a href="#cb120-1930" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1933"><a href="#cb120-1933" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1934"><a href="#cb120-1934" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1935"><a href="#cb120-1935" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-1936"><a href="#cb120-1936" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/dada2/its18_plot_errorF.png"</span>)</span>
<span id="cb120-1937"><a href="#cb120-1937" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1938"><a href="#cb120-1938" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("error_F_its")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-1939"><a href="#cb120-1939" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1940"><a href="#cb120-1940" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Reverse Reads</span></span>
<span id="cb120-1941"><a href="#cb120-1941" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1944"><a href="#cb120-1944" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1945"><a href="#cb120-1945" aria-hidden="true" tabindex="-1"></a>errR <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(filtRs, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-1946"><a href="#cb120-1946" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1947"><a href="#cb120-1947" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1948"><a href="#cb120-1948" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1949"><a href="#cb120-1949" aria-hidden="true" tabindex="-1"></a><span class="in">109358094 total bases in 495784 reads from 11 samples will be used for learning the error rates.</span></span>
<span id="cb120-1950"><a href="#cb120-1950" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1951"><a href="#cb120-1951" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1954"><a href="#cb120-1954" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1955"><a href="#cb120-1955" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errR, <span class="at">nominalQ =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-1956"><a href="#cb120-1956" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1957"><a href="#cb120-1957" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1960"><a href="#cb120-1960" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1961"><a href="#cb120-1961" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1962"><a href="#cb120-1962" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">plotErrors</span>(errR, <span class="at">nominalQ =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-1963"><a href="#cb120-1963" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"plot_errorR_1.png"</span>, p4, <span class="at">width =</span> <span class="dv">7</span>, <span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb120-1964"><a href="#cb120-1964" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"plot_errorR_2.png"</span>, p4)</span>
<span id="cb120-1965"><a href="#cb120-1965" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1966"><a href="#cb120-1966" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1969"><a href="#cb120-1969" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1970"><a href="#cb120-1970" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1971"><a href="#cb120-1971" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">"cp files/dada2/figures/its18_plot_errorR.png include/dada2/its18_plot_errorR.png"</span>)</span>
<span id="cb120-1972"><a href="#cb120-1972" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1973"><a href="#cb120-1973" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1976"><a href="#cb120-1976" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1977"><a href="#cb120-1977" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-1978"><a href="#cb120-1978" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-1979"><a href="#cb120-1979" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/dada2/its18_plot_errorR.png"</span>)</span>
<span id="cb120-1980"><a href="#cb120-1980" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1981"><a href="#cb120-1981" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("error_R_its")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-1982"><a href="#cb120-1982" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1983"><a href="#cb120-1983" aria-hidden="true" tabindex="-1"></a><span class="fu">## 6. Dereplicate Reads</span></span>
<span id="cb120-1984"><a href="#cb120-1984" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1985"><a href="#cb120-1985" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Forward Reads</span></span>
<span id="cb120-1986"><a href="#cb120-1986" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1989"><a href="#cb120-1989" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-1990"><a href="#cb120-1990" aria-hidden="true" tabindex="-1"></a>derepFs <span class="ot">&lt;-</span> <span class="fu">derepFastq</span>(filtFs, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-1991"><a href="#cb120-1991" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(derepFs) <span class="ot">&lt;-</span> sample.names</span>
<span id="cb120-1992"><a href="#cb120-1992" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-1993"><a href="#cb120-1993" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1994"><a href="#cb120-1994" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br/&gt;</span></span>
<span id="cb120-1995"><a href="#cb120-1995" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-1996"><a href="#cb120-1996" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb120-1997"><a href="#cb120-1997" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Detailed results of derep forward reads<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb120-1998"><a href="#cb120-1998" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;pre&gt;&lt;code&gt;</span></span>
<span id="cb120-1999"><a href="#cb120-1999" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P1_R1.fastq.gz</span>
<span id="cb120-2000"><a href="#cb120-2000" aria-hidden="true" tabindex="-1"></a>Encountered 7277 unique sequences from 35999 total sequences read.</span>
<span id="cb120-2001"><a href="#cb120-2001" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P1-8C_R1.fastq.gz</span>
<span id="cb120-2002"><a href="#cb120-2002" aria-hidden="true" tabindex="-1"></a>Encountered 11366 unique sequences from 48230 total sequences read.</span>
<span id="cb120-2003"><a href="#cb120-2003" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P10_R1.fastq.gz</span>
<span id="cb120-2004"><a href="#cb120-2004" aria-hidden="true" tabindex="-1"></a>Encountered 11714 unique sequences from 80877 total sequences read.</span>
<span id="cb120-2005"><a href="#cb120-2005" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P2_R1.fastq.gz</span>
<span id="cb120-2006"><a href="#cb120-2006" aria-hidden="true" tabindex="-1"></a>Encountered 13088 unique sequences from 58956 total sequences read.</span>
<span id="cb120-2007"><a href="#cb120-2007" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P3_R1.fastq.gz</span>
<span id="cb120-2008"><a href="#cb120-2008" aria-hidden="true" tabindex="-1"></a>Encountered 10180 unique sequences from 42563 total sequences read.</span>
<span id="cb120-2009"><a href="#cb120-2009" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P3-8C_R1.fastq.gz</span>
<span id="cb120-2010"><a href="#cb120-2010" aria-hidden="true" tabindex="-1"></a>Encountered 4480 unique sequences from 17548 total sequences read.</span>
<span id="cb120-2011"><a href="#cb120-2011" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P4_R1.fastq.gz</span>
<span id="cb120-2012"><a href="#cb120-2012" aria-hidden="true" tabindex="-1"></a>Encountered 12259 unique sequences from 68259 total sequences read.</span>
<span id="cb120-2013"><a href="#cb120-2013" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P5_R1.fastq.gz</span>
<span id="cb120-2014"><a href="#cb120-2014" aria-hidden="true" tabindex="-1"></a>Encountered 160 unique sequences from 308 total sequences read.</span>
<span id="cb120-2015"><a href="#cb120-2015" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P5-8C_R1.fastq.gz</span>
<span id="cb120-2016"><a href="#cb120-2016" aria-hidden="true" tabindex="-1"></a>Encountered 10436 unique sequences from 48518 total sequences read.</span>
<span id="cb120-2017"><a href="#cb120-2017" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P6_R1.fastq.gz</span>
<span id="cb120-2018"><a href="#cb120-2018" aria-hidden="true" tabindex="-1"></a>Encountered 11230 unique sequences from 51055 total sequences read.</span>
<span id="cb120-2019"><a href="#cb120-2019" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P7_R1.fastq.gz</span>
<span id="cb120-2020"><a href="#cb120-2020" aria-hidden="true" tabindex="-1"></a>Encountered 10316 unique sequences from 43471 total sequences read.</span>
<span id="cb120-2021"><a href="#cb120-2021" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P7-8C_R1.fastq.gz</span>
<span id="cb120-2022"><a href="#cb120-2022" aria-hidden="true" tabindex="-1"></a>Encountered 2121 unique sequences from 9488 total sequences read.</span>
<span id="cb120-2023"><a href="#cb120-2023" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P8_R1.fastq.gz</span>
<span id="cb120-2024"><a href="#cb120-2024" aria-hidden="true" tabindex="-1"></a>Encountered 9236 unique sequences from 41563 total sequences read.</span>
<span id="cb120-2025"><a href="#cb120-2025" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P9_R1.fastq.gz</span>
<span id="cb120-2026"><a href="#cb120-2026" aria-hidden="true" tabindex="-1"></a>Encountered 9000 unique sequences from 35722 total sequences read.</span>
<span id="cb120-2027"><a href="#cb120-2027" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P9-8C_R1.fastq.gz</span>
<span id="cb120-2028"><a href="#cb120-2028" aria-hidden="true" tabindex="-1"></a>Encountered 42 unique sequences from 85 total sequences read.</span>
<span id="cb120-2029"><a href="#cb120-2029" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/code&gt;&lt;/pre&gt;</span></span>
<span id="cb120-2030"><a href="#cb120-2030" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb120-2031"><a href="#cb120-2031" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2032"><a href="#cb120-2032" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br/&gt;</span></span>
<span id="cb120-2033"><a href="#cb120-2033" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2034"><a href="#cb120-2034" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Reverse Reads</span></span>
<span id="cb120-2035"><a href="#cb120-2035" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2038"><a href="#cb120-2038" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2039"><a href="#cb120-2039" aria-hidden="true" tabindex="-1"></a>derepRs <span class="ot">&lt;-</span> <span class="fu">derepFastq</span>(filtRs, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-2040"><a href="#cb120-2040" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(derepRs) <span class="ot">&lt;-</span> sample.names</span>
<span id="cb120-2041"><a href="#cb120-2041" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2042"><a href="#cb120-2042" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2043"><a href="#cb120-2043" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br/&gt;</span></span>
<span id="cb120-2044"><a href="#cb120-2044" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2045"><a href="#cb120-2045" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb120-2046"><a href="#cb120-2046" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Detailed results of derep reverse reads<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb120-2047"><a href="#cb120-2047" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;pre&gt;&lt;code&gt;</span></span>
<span id="cb120-2048"><a href="#cb120-2048" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P1_R2.fastq.gz</span>
<span id="cb120-2049"><a href="#cb120-2049" aria-hidden="true" tabindex="-1"></a>Encountered 13081 unique sequences from 35999 total sequences read.</span>
<span id="cb120-2050"><a href="#cb120-2050" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P1-8C_R2.fastq.gz</span>
<span id="cb120-2051"><a href="#cb120-2051" aria-hidden="true" tabindex="-1"></a>Encountered 17965 unique sequences from 48230 total sequences read.</span>
<span id="cb120-2052"><a href="#cb120-2052" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P10_R2.fastq.gz</span>
<span id="cb120-2053"><a href="#cb120-2053" aria-hidden="true" tabindex="-1"></a>Encountered 18610 unique sequences from 80877 total sequences read.</span>
<span id="cb120-2054"><a href="#cb120-2054" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P2_R2.fastq.gz</span>
<span id="cb120-2055"><a href="#cb120-2055" aria-hidden="true" tabindex="-1"></a>Encountered 19600 unique sequences from 58956 total sequences read.</span>
<span id="cb120-2056"><a href="#cb120-2056" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P3_R2.fastq.gz</span>
<span id="cb120-2057"><a href="#cb120-2057" aria-hidden="true" tabindex="-1"></a>Encountered 16145 unique sequences from 42563 total sequences read.</span>
<span id="cb120-2058"><a href="#cb120-2058" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P3-8C_R2.fastq.gz</span>
<span id="cb120-2059"><a href="#cb120-2059" aria-hidden="true" tabindex="-1"></a>Encountered 7496 unique sequences from 17548 total sequences read.</span>
<span id="cb120-2060"><a href="#cb120-2060" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P4_R2.fastq.gz</span>
<span id="cb120-2061"><a href="#cb120-2061" aria-hidden="true" tabindex="-1"></a>Encountered 19933 unique sequences from 68259 total sequences read.</span>
<span id="cb120-2062"><a href="#cb120-2062" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P5_R2.fastq.gz</span>
<span id="cb120-2063"><a href="#cb120-2063" aria-hidden="true" tabindex="-1"></a>Encountered 199 unique sequences from 308 total sequences read.</span>
<span id="cb120-2064"><a href="#cb120-2064" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P5-8C_R2.fastq.gz</span>
<span id="cb120-2065"><a href="#cb120-2065" aria-hidden="true" tabindex="-1"></a>Encountered 18495 unique sequences from 48518 total sequences read.</span>
<span id="cb120-2066"><a href="#cb120-2066" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P6_R2.fastq.gz</span>
<span id="cb120-2067"><a href="#cb120-2067" aria-hidden="true" tabindex="-1"></a>Encountered 17216 unique sequences from 51055 total sequences read.</span>
<span id="cb120-2068"><a href="#cb120-2068" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P7_R2.fastq.gz</span>
<span id="cb120-2069"><a href="#cb120-2069" aria-hidden="true" tabindex="-1"></a>Encountered 17111 unique sequences from 43471 total sequences read.</span>
<span id="cb120-2070"><a href="#cb120-2070" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P7-8C_R2.fastq.gz</span>
<span id="cb120-2071"><a href="#cb120-2071" aria-hidden="true" tabindex="-1"></a>Encountered 3809 unique sequences from 9488 total sequences read.</span>
<span id="cb120-2072"><a href="#cb120-2072" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P8_R2.fastq.gz</span>
<span id="cb120-2073"><a href="#cb120-2073" aria-hidden="true" tabindex="-1"></a>Encountered 14140 unique sequences from 41563 total sequences read.</span>
<span id="cb120-2074"><a href="#cb120-2074" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P9_R2.fastq.gz</span>
<span id="cb120-2075"><a href="#cb120-2075" aria-hidden="true" tabindex="-1"></a>Encountered 14856 unique sequences from 35722 total sequences read.</span>
<span id="cb120-2076"><a href="#cb120-2076" aria-hidden="true" tabindex="-1"></a>Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/P9-8C_R2.fastq.gz</span>
<span id="cb120-2077"><a href="#cb120-2077" aria-hidden="true" tabindex="-1"></a>Encountered 45 unique sequences from 85 total sequences read.</span>
<span id="cb120-2078"><a href="#cb120-2078" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/code&gt;&lt;/pre&gt;</span></span>
<span id="cb120-2079"><a href="#cb120-2079" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb120-2080"><a href="#cb120-2080" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2081"><a href="#cb120-2081" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br/&gt;</span></span>
<span id="cb120-2082"><a href="#cb120-2082" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2083"><a href="#cb120-2083" aria-hidden="true" tabindex="-1"></a><span class="fu">## 7. DADA2 &amp; ASV Inference</span></span>
<span id="cb120-2084"><a href="#cb120-2084" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2085"><a href="#cb120-2085" aria-hidden="true" tabindex="-1"></a>At this point we are ready to apply the core sample inference algorithm (dada) to the filtered and trimmed sequence data. DADA2 offers three options for whether and how to pool samples for ASV inference. </span>
<span id="cb120-2086"><a href="#cb120-2086" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2087"><a href="#cb120-2087" aria-hidden="true" tabindex="-1"></a>If <span class="in">`pool = TRUE`</span>, the algorithm will pool together all samples prior to sample inference.  </span>
<span id="cb120-2088"><a href="#cb120-2088" aria-hidden="true" tabindex="-1"></a>If <span class="in">`pool = FALSE`</span>, sample inference is performed on each sample individually.  </span>
<span id="cb120-2089"><a href="#cb120-2089" aria-hidden="true" tabindex="-1"></a>If <span class="in">`pool = "pseudo"`</span>, the algorithm will perform pseudo-pooling between individually processed samples.</span>
<span id="cb120-2090"><a href="#cb120-2090" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2091"><a href="#cb120-2091" aria-hidden="true" tabindex="-1"></a>We tested all three methods through the full pipeline. Click the <span class="in">`+`</span> to see the results of the test. For our final analysis, we chose <span class="in">`pool = TRUE`</span> for this data set. </span>
<span id="cb120-2092"><a href="#cb120-2092" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2093"><a href="#cb120-2093" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb120-2094"><a href="#cb120-2094" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show/hide  Results of <span class="in">`dada`</span> pooling options <span class="kw">&lt;/summary&gt;</span></span>
<span id="cb120-2095"><a href="#cb120-2095" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2096"><a href="#cb120-2096" aria-hidden="true" tabindex="-1"></a>Here are a few summary tables of results from the tests. Values are from the final sequence table prodcuced by each option. </span>
<span id="cb120-2097"><a href="#cb120-2097" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2100"><a href="#cb120-2100" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2101"><a href="#cb120-2101" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2102"><a href="#cb120-2102" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-2103"><a href="#cb120-2103" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"files/dada2/tables/its18_dada_compare.txt"</span>,</span>
<span id="cb120-2104"><a href="#cb120-2104" aria-hidden="true" tabindex="-1"></a>                        <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb120-2105"><a href="#cb120-2105" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2106"><a href="#cb120-2106" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab("pooling_summary_its")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-2107"><a href="#cb120-2107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2110"><a href="#cb120-2110" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2111"><a href="#cb120-2111" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2112"><a href="#cb120-2112" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-2113"><a href="#cb120-2113" aria-hidden="true" tabindex="-1"></a>seq_table <span class="sc">%&gt;%</span></span>
<span id="cb120-2114"><a href="#cb120-2114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb120-2115"><a href="#cb120-2115" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"its18_pooling_summary"</span>,</span>
<span id="cb120-2116"><a href="#cb120-2116" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb120-2117"><a href="#cb120-2117" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb120-2118"><a href="#cb120-2118" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb120-2119"><a href="#cb120-2119" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb120-2120"><a href="#cb120-2120" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-2121"><a href="#cb120-2121" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb120-2122"><a href="#cb120-2122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2123"><a href="#cb120-2123" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb120-2124"><a href="#cb120-2124" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb120-2125"><a href="#cb120-2125" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb120-2126"><a href="#cb120-2126" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">1</span>),</span>
<span id="cb120-2127"><a href="#cb120-2127" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">FALSE</span>, <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-2128"><a href="#cb120-2128" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb120-2129"><a href="#cb120-2129" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-2130"><a href="#cb120-2130" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">5</span>, </span>
<span id="cb120-2131"><a href="#cb120-2131" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb120-2132"><a href="#cb120-2132" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb120-2133"><a href="#cb120-2133" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb120-2134"><a href="#cb120-2134" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-2135"><a href="#cb120-2135" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb120-2136"><a href="#cb120-2136" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2137"><a href="#cb120-2137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2138"><a href="#cb120-2138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2141"><a href="#cb120-2141" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2142"><a href="#cb120-2142" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2143"><a href="#cb120-2143" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-2144"><a href="#cb120-2144" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"files/dada2/tables/its18_dada_compare_by_samp.txt"</span>,</span>
<span id="cb120-2145"><a href="#cb120-2145" aria-hidden="true" tabindex="-1"></a>                        <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb120-2146"><a href="#cb120-2146" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2147"><a href="#cb120-2147" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab("pooling_its")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-2148"><a href="#cb120-2148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2151"><a href="#cb120-2151" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2152"><a href="#cb120-2152" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2153"><a href="#cb120-2153" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-2154"><a href="#cb120-2154" aria-hidden="true" tabindex="-1"></a>seq_table <span class="sc">%&gt;%</span></span>
<span id="cb120-2155"><a href="#cb120-2155" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb120-2156"><a href="#cb120-2156" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"its18_pooling_results"</span>,</span>
<span id="cb120-2157"><a href="#cb120-2157" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb120-2158"><a href="#cb120-2158" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb120-2159"><a href="#cb120-2159" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb120-2160"><a href="#cb120-2160" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb120-2161"><a href="#cb120-2161" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-2162"><a href="#cb120-2162" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb120-2163"><a href="#cb120-2163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2164"><a href="#cb120-2164" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb120-2165"><a href="#cb120-2165" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb120-2166"><a href="#cb120-2166" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb120-2167"><a href="#cb120-2167" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">1</span>),</span>
<span id="cb120-2168"><a href="#cb120-2168" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">minWidth =</span> <span class="dv">150</span>, <span class="at">filterable =</span> <span class="cn">FALSE</span>, </span>
<span id="cb120-2169"><a href="#cb120-2169" aria-hidden="true" tabindex="-1"></a>    <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-2170"><a href="#cb120-2170" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb120-2171"><a href="#cb120-2171" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-2172"><a href="#cb120-2172" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb120-2173"><a href="#cb120-2173" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Sample ID"</span>, </span>
<span id="cb120-2174"><a href="#cb120-2174" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb120-2175"><a href="#cb120-2175" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>),</span>
<span id="cb120-2176"><a href="#cb120-2176" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>), </span>
<span id="cb120-2177"><a href="#cb120-2177" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb120-2178"><a href="#cb120-2178" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">150</span>) </span>
<span id="cb120-2179"><a href="#cb120-2179" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-2180"><a href="#cb120-2180" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">5</span>, </span>
<span id="cb120-2181"><a href="#cb120-2181" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb120-2182"><a href="#cb120-2182" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb120-2183"><a href="#cb120-2183" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb120-2184"><a href="#cb120-2184" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-2185"><a href="#cb120-2185" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb120-2186"><a href="#cb120-2186" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2187"><a href="#cb120-2187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2188"><a href="#cb120-2188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2189"><a href="#cb120-2189" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb120-2190"><a href="#cb120-2190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2191"><a href="#cb120-2191" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Forward Reads</span></span>
<span id="cb120-2192"><a href="#cb120-2192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2195"><a href="#cb120-2195" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2196"><a href="#cb120-2196" aria-hidden="true" tabindex="-1"></a>dadaFs <span class="ot">&lt;-</span> <span class="fu">dada</span>(derepFs, <span class="at">err =</span> errF, <span class="at">pool =</span> <span class="cn">TRUE</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-2197"><a href="#cb120-2197" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2198"><a href="#cb120-2198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2199"><a href="#cb120-2199" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2200"><a href="#cb120-2200" aria-hidden="true" tabindex="-1"></a><span class="in">15 samples were pooled: 582642 reads in 102887 unique sequences.</span></span>
<span id="cb120-2201"><a href="#cb120-2201" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2202"><a href="#cb120-2202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2205"><a href="#cb120-2205" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2206"><a href="#cb120-2206" aria-hidden="true" tabindex="-1"></a>dadaFs[[sampnum]]</span>
<span id="cb120-2207"><a href="#cb120-2207" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2208"><a href="#cb120-2208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2209"><a href="#cb120-2209" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2210"><a href="#cb120-2210" aria-hidden="true" tabindex="-1"></a><span class="in">dada-class: object describing DADA2 denoising results</span></span>
<span id="cb120-2211"><a href="#cb120-2211" aria-hidden="true" tabindex="-1"></a><span class="in">1175 sequence variants were inferred from 7277 input unique sequences.</span></span>
<span id="cb120-2212"><a href="#cb120-2212" aria-hidden="true" tabindex="-1"></a><span class="in">Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</span></span>
<span id="cb120-2213"><a href="#cb120-2213" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2214"><a href="#cb120-2214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2215"><a href="#cb120-2215" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Reverse  Reads</span></span>
<span id="cb120-2216"><a href="#cb120-2216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2219"><a href="#cb120-2219" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2220"><a href="#cb120-2220" aria-hidden="true" tabindex="-1"></a>dadaRs <span class="ot">&lt;-</span> <span class="fu">dada</span>(derepRs, <span class="at">err =</span> errR, <span class="at">pool =</span> <span class="cn">TRUE</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-2221"><a href="#cb120-2221" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2222"><a href="#cb120-2222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2223"><a href="#cb120-2223" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2224"><a href="#cb120-2224" aria-hidden="true" tabindex="-1"></a><span class="in">15 samples were pooled: 582642 reads in 174866 unique sequences.</span></span>
<span id="cb120-2225"><a href="#cb120-2225" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2226"><a href="#cb120-2226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2227"><a href="#cb120-2227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2230"><a href="#cb120-2230" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2231"><a href="#cb120-2231" aria-hidden="true" tabindex="-1"></a>dadaRs[[sampnum]]</span>
<span id="cb120-2232"><a href="#cb120-2232" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2233"><a href="#cb120-2233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2234"><a href="#cb120-2234" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2235"><a href="#cb120-2235" aria-hidden="true" tabindex="-1"></a><span class="in">dada-class: object describing DADA2 denoising results</span></span>
<span id="cb120-2236"><a href="#cb120-2236" aria-hidden="true" tabindex="-1"></a><span class="in">1116 sequence variants were inferred from 13081 input unique sequences.</span></span>
<span id="cb120-2237"><a href="#cb120-2237" aria-hidden="true" tabindex="-1"></a><span class="in">Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</span></span>
<span id="cb120-2238"><a href="#cb120-2238" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2239"><a href="#cb120-2239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2242"><a href="#cb120-2242" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2243"><a href="#cb120-2243" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="st">"pre_merge_its_2018.rdata"</span>)</span>
<span id="cb120-2244"><a href="#cb120-2244" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2245"><a href="#cb120-2245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2246"><a href="#cb120-2246" aria-hidden="true" tabindex="-1"></a><span class="fu">## 8. Merge Paired Reads</span></span>
<span id="cb120-2247"><a href="#cb120-2247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2250"><a href="#cb120-2250" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2251"><a href="#cb120-2251" aria-hidden="true" tabindex="-1"></a>mergers <span class="ot">&lt;-</span> <span class="fu">mergePairs</span>(dadaFs, derepFs, dadaRs, derepRs, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb120-2252"><a href="#cb120-2252" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2253"><a href="#cb120-2253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2254"><a href="#cb120-2254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2255"><a href="#cb120-2255" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb120-2256"><a href="#cb120-2256" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Detailed results of merging reads<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb120-2257"><a href="#cb120-2257" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;pre&gt;&lt;code&gt;</span></span>
<span id="cb120-2258"><a href="#cb120-2258" aria-hidden="true" tabindex="-1"></a>29795 paired-reads (in 946 unique pairings) successfully merged out of 32193 (in 1177 pairings) input.</span>
<span id="cb120-2259"><a href="#cb120-2259" aria-hidden="true" tabindex="-1"></a>41880 paired-reads (in 719 unique pairings) successfully merged out of 46715 (in 941 pairings) input.</span>
<span id="cb120-2260"><a href="#cb120-2260" aria-hidden="true" tabindex="-1"></a>64805 paired-reads (in 815 unique pairings) successfully merged out of 66771 (in 1015 pairings) input.</span>
<span id="cb120-2261"><a href="#cb120-2261" aria-hidden="true" tabindex="-1"></a>Duplicate sequences in merged output.</span>
<span id="cb120-2262"><a href="#cb120-2262" aria-hidden="true" tabindex="-1"></a>51144 paired-reads (in 1011 unique pairings) successfully merged out of 57613 (in 1325 pairings) input.</span>
<span id="cb120-2263"><a href="#cb120-2263" aria-hidden="true" tabindex="-1"></a>38442 paired-reads (in 767 unique pairings) successfully merged out of 41985 (in 994 pairings) input.</span>
<span id="cb120-2264"><a href="#cb120-2264" aria-hidden="true" tabindex="-1"></a>14188 paired-reads (in 349 unique pairings) successfully merged out of 16981 (in 452 pairings) input.</span>
<span id="cb120-2265"><a href="#cb120-2265" aria-hidden="true" tabindex="-1"></a>56594 paired-reads (in 1018 unique pairings) successfully merged out of 64411 (in 1306 pairings) input.</span>
<span id="cb120-2266"><a href="#cb120-2266" aria-hidden="true" tabindex="-1"></a>296 paired-reads (in 95 unique pairings) successfully merged out of 301 (in 100 pairings) input.</span>
<span id="cb120-2267"><a href="#cb120-2267" aria-hidden="true" tabindex="-1"></a>35713 paired-reads (in 619 unique pairings) successfully merged out of 41006 (in 786 pairings) input.</span>
<span id="cb120-2268"><a href="#cb120-2268" aria-hidden="true" tabindex="-1"></a>45851 paired-reads (in 955 unique pairings) successfully merged out of 50174 (in 1240 pairings) input.</span>
<span id="cb120-2269"><a href="#cb120-2269" aria-hidden="true" tabindex="-1"></a>36121 paired-reads (in 870 unique pairings) successfully merged out of 41541 (in 1135 pairings) input.</span>
<span id="cb120-2270"><a href="#cb120-2270" aria-hidden="true" tabindex="-1"></a>Duplicate sequences in merged output.</span>
<span id="cb120-2271"><a href="#cb120-2271" aria-hidden="true" tabindex="-1"></a>9172 paired-reads (in 335 unique pairings) successfully merged out of 9403 (in 402 pairings) input.</span>
<span id="cb120-2272"><a href="#cb120-2272" aria-hidden="true" tabindex="-1"></a>38392 paired-reads (in 704 unique pairings) successfully merged out of 40897 (in 886 pairings) input.</span>
<span id="cb120-2273"><a href="#cb120-2273" aria-hidden="true" tabindex="-1"></a>29155 paired-reads (in 748 unique pairings) successfully merged out of 34979 (in 990 pairings) input.</span>
<span id="cb120-2274"><a href="#cb120-2274" aria-hidden="true" tabindex="-1"></a>Duplicate sequences in merged output.</span>
<span id="cb120-2275"><a href="#cb120-2275" aria-hidden="true" tabindex="-1"></a>80 paired-reads (in 14 unique pairings) successfully merged out of 82 (in 16 pairings) input.</span>
<span id="cb120-2276"><a href="#cb120-2276" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/code&gt;&lt;/pre&gt;</span></span>
<span id="cb120-2277"><a href="#cb120-2277" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb120-2278"><a href="#cb120-2278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2279"><a href="#cb120-2279" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br/&gt;</span></span>
<span id="cb120-2280"><a href="#cb120-2280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2283"><a href="#cb120-2283" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2284"><a href="#cb120-2284" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2285"><a href="#cb120-2285" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mergers[[<span class="dv">1</span>]])</span>
<span id="cb120-2286"><a href="#cb120-2286" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2287"><a href="#cb120-2287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2288"><a href="#cb120-2288" aria-hidden="true" tabindex="-1"></a><span class="fu">## 9. Construct Sequence Table</span></span>
<span id="cb120-2289"><a href="#cb120-2289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2292"><a href="#cb120-2292" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2293"><a href="#cb120-2293" aria-hidden="true" tabindex="-1"></a>seqtabF <span class="ot">&lt;-</span> <span class="fu">makeSequenceTable</span>(mergers)</span>
<span id="cb120-2294"><a href="#cb120-2294" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtabF)</span>
<span id="cb120-2295"><a href="#cb120-2295" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2296"><a href="#cb120-2296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2299"><a href="#cb120-2299" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2300"><a href="#cb120-2300" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2301"><a href="#cb120-2301" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-2302"><a href="#cb120-2302" aria-hidden="true" tabindex="-1"></a>seqtabX <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">15</span>, <span class="dv">3364</span>)</span>
<span id="cb120-2303"><a href="#cb120-2303" aria-hidden="true" tabindex="-1"></a>seqtabX</span>
<span id="cb120-2304"><a href="#cb120-2304" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2305"><a href="#cb120-2305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2308"><a href="#cb120-2308" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2309"><a href="#cb120-2309" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtabF)))</span>
<span id="cb120-2310"><a href="#cb120-2310" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2311"><a href="#cb120-2311" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2312"><a href="#cb120-2312" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2313"><a href="#cb120-2313" aria-hidden="true" tabindex="-1"></a><span class="in">130 131 132 133 136 139 140 141 143 144 146 147 148 149 150 151 152 153 154 155 </span></span>
<span id="cb120-2314"><a href="#cb120-2314" aria-hidden="true" tabindex="-1"></a><span class="in">  2   2   1   1   1   1   1   3   2   1   3   5   5   4   5   5   3   2   5   4 </span></span>
<span id="cb120-2315"><a href="#cb120-2315" aria-hidden="true" tabindex="-1"></a><span class="in">156 159 160 161 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 </span></span>
<span id="cb120-2316"><a href="#cb120-2316" aria-hidden="true" tabindex="-1"></a><span class="in">  1   2   1   1   2   2   4   9   2   1   2   5   5   5   8   9   2   5  10  13 </span></span>
<span id="cb120-2317"><a href="#cb120-2317" aria-hidden="true" tabindex="-1"></a><span class="in">179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 </span></span>
<span id="cb120-2318"><a href="#cb120-2318" aria-hidden="true" tabindex="-1"></a><span class="in"> 13  14  11  14  12  15  19  12  12  13   9  22  11  11  12  15  17  13  10  11 </span></span>
<span id="cb120-2319"><a href="#cb120-2319" aria-hidden="true" tabindex="-1"></a><span class="in">199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 </span></span>
<span id="cb120-2320"><a href="#cb120-2320" aria-hidden="true" tabindex="-1"></a><span class="in"> 22  16  10  11  13  17  10   8  10  12  13   7  20  18  24  30  39  38  20  19 </span></span>
<span id="cb120-2321"><a href="#cb120-2321" aria-hidden="true" tabindex="-1"></a><span class="in">219 220 221 222 223 224 225 226 228 229 230 231 232 233 234 235 236 237 238 239 </span></span>
<span id="cb120-2322"><a href="#cb120-2322" aria-hidden="true" tabindex="-1"></a><span class="in"> 12  18  21  23  23  29  32  32   6  48   4  73  20  27  29  19  31  33  30  29 </span></span>
<span id="cb120-2323"><a href="#cb120-2323" aria-hidden="true" tabindex="-1"></a><span class="in">240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255 256 257 258 259 </span></span>
<span id="cb120-2324"><a href="#cb120-2324" aria-hidden="true" tabindex="-1"></a><span class="in"> 35  21  36  16  16  18  17  25  22  29  17  27  28  22  20  18  28  19  28  25 </span></span>
<span id="cb120-2325"><a href="#cb120-2325" aria-hidden="true" tabindex="-1"></a><span class="in">260 261 262 263 264 265 266 267 268 269 270 271 272 273 274 275 276 277 278 279 </span></span>
<span id="cb120-2326"><a href="#cb120-2326" aria-hidden="true" tabindex="-1"></a><span class="in"> 23  11  15  13  16  18  23  17  12  12  16  15  32  17  22  21  21  18  16  17 </span></span>
<span id="cb120-2327"><a href="#cb120-2327" aria-hidden="true" tabindex="-1"></a><span class="in">280 281 282 283 284 285 286 287 288 289 290 291 292 293 294 295 296 297 298 299 </span></span>
<span id="cb120-2328"><a href="#cb120-2328" aria-hidden="true" tabindex="-1"></a><span class="in"> 13  22  13  18  12  13  13  18  17  15  14  18   8  11  13  17  12  11  13   9 </span></span>
<span id="cb120-2329"><a href="#cb120-2329" aria-hidden="true" tabindex="-1"></a><span class="in">300 301 302 303 304 305 306 307 308 309 310 311 312 313 314 315 316 317 318 319 </span></span>
<span id="cb120-2330"><a href="#cb120-2330" aria-hidden="true" tabindex="-1"></a><span class="in"> 16  10   9   3   8  12   8  13  20  15   8  13  16  11  10   8  18  10   8  12 </span></span>
<span id="cb120-2331"><a href="#cb120-2331" aria-hidden="true" tabindex="-1"></a><span class="in">320 321 322 323 324 325 326 327 328 329 330 331 332 333 334 335 336 337 338 339 </span></span>
<span id="cb120-2332"><a href="#cb120-2332" aria-hidden="true" tabindex="-1"></a><span class="in">  8   5  12  17  13   9   8   8   9  12  16  13  14  11  12  13  18  10  11  10 </span></span>
<span id="cb120-2333"><a href="#cb120-2333" aria-hidden="true" tabindex="-1"></a><span class="in">340 341 342 343 344 345 346 347 348 349 350 351 352 353 354 355 356 357 358 359 </span></span>
<span id="cb120-2334"><a href="#cb120-2334" aria-hidden="true" tabindex="-1"></a><span class="in">  9  11   8   8   5  15   8  16  11  15   4   9   7   9  12  14   5  12  11   5 </span></span>
<span id="cb120-2335"><a href="#cb120-2335" aria-hidden="true" tabindex="-1"></a><span class="in">360 361 362 363 364 365 366 367 368 369 370 371 372 373 374 375 376 377 378 379 </span></span>
<span id="cb120-2336"><a href="#cb120-2336" aria-hidden="true" tabindex="-1"></a><span class="in">  2   4  11   5   2  14   6  12   7   5   6   4   3   9   5  10   8   9   4   6 </span></span>
<span id="cb120-2337"><a href="#cb120-2337" aria-hidden="true" tabindex="-1"></a><span class="in">380 381 382 383 384 385 386 387 388 389 390 391 392 393 394 395 396 397 398 399 </span></span>
<span id="cb120-2338"><a href="#cb120-2338" aria-hidden="true" tabindex="-1"></a><span class="in">  5   3   7   7   4   2   3   4   3   5   5   1   3   4   3   6   5   4   3   2 </span></span>
<span id="cb120-2339"><a href="#cb120-2339" aria-hidden="true" tabindex="-1"></a><span class="in">400 401 402 403 404 405 406 407 408 409 410 411 412 413 414 415 416 417 418 421 </span></span>
<span id="cb120-2340"><a href="#cb120-2340" aria-hidden="true" tabindex="-1"></a><span class="in"> 11   5   3   3   3   3   2   1   2   3   3   1   3   3   1   2   2   1   2   2 </span></span>
<span id="cb120-2341"><a href="#cb120-2341" aria-hidden="true" tabindex="-1"></a><span class="in">422 423 424 427 430 431 432 434 436 437 439 440 441 442 443 446 447 448 </span></span>
<span id="cb120-2342"><a href="#cb120-2342" aria-hidden="true" tabindex="-1"></a><span class="in">  2   3   2   1   3   2   4   2   1   2   4   3   1   2   1   1   2   1 </span></span>
<span id="cb120-2343"><a href="#cb120-2343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2344"><a href="#cb120-2344" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2345"><a href="#cb120-2345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2348"><a href="#cb120-2348" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2349"><a href="#cb120-2349" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2350"><a href="#cb120-2350" aria-hidden="true" tabindex="-1"></a>read_length <span class="ot">&lt;-</span>  <span class="fu">data.frame</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab)))</span>
<span id="cb120-2351"><a href="#cb120-2351" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(read_length) <span class="ot">&lt;-</span> <span class="st">"length"</span></span>
<span id="cb120-2352"><a href="#cb120-2352" aria-hidden="true" tabindex="-1"></a>plot_a <span class="ot">&lt;-</span> <span class="fu">qplot</span>(length, <span class="at">data =</span> read_length,</span>
<span id="cb120-2353"><a href="#cb120-2353" aria-hidden="true" tabindex="-1"></a>                <span class="at">geom =</span> <span class="st">"histogram"</span>, <span class="at">binwidth =</span> <span class="dv">1</span>,</span>
<span id="cb120-2354"><a href="#cb120-2354" aria-hidden="true" tabindex="-1"></a>                <span class="at">xlab =</span> <span class="st">"read length"</span>,</span>
<span id="cb120-2355"><a href="#cb120-2355" aria-hidden="true" tabindex="-1"></a>                <span class="at">ylab =</span> <span class="st">"total variants"</span>,</span>
<span id="cb120-2356"><a href="#cb120-2356" aria-hidden="true" tabindex="-1"></a>                <span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">200</span>, <span class="dv">300</span>))</span>
<span id="cb120-2357"><a href="#cb120-2357" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"read_length_before.png"</span>, plot_a, <span class="at">width =</span> <span class="dv">7</span>, <span class="at">height =</span> <span class="dv">3</span>)</span>
<span id="cb120-2358"><a href="#cb120-2358" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2359"><a href="#cb120-2359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2362"><a href="#cb120-2362" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2363"><a href="#cb120-2363" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2364"><a href="#cb120-2364" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">"cp files/dada2/figures/its18_read_length_before.png include/dada2/its18_read_length_before.png"</span>)</span>
<span id="cb120-2365"><a href="#cb120-2365" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2366"><a href="#cb120-2366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2369"><a href="#cb120-2369" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2370"><a href="#cb120-2370" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2371"><a href="#cb120-2371" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-2372"><a href="#cb120-2372" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/dada2/its18_read_length_before.png"</span>)</span>
<span id="cb120-2373"><a href="#cb120-2373" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2374"><a href="#cb120-2374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2375"><a href="#cb120-2375" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("read_length_its")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-2376"><a href="#cb120-2376" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2377"><a href="#cb120-2377" aria-hidden="true" tabindex="-1"></a>As expected, a wide range of length variability in the the amplified ITS region.</span>
<span id="cb120-2378"><a href="#cb120-2378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2381"><a href="#cb120-2381" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2382"><a href="#cb120-2382" aria-hidden="true" tabindex="-1"></a>seqtab <span class="ot">&lt;-</span> seqtabF[,<span class="fu">nchar</span>(<span class="fu">colnames</span>(seqtabF)) <span class="sc">%in%</span> <span class="fu">seq</span>(<span class="dv">100</span>,<span class="dv">450</span>)]</span>
<span id="cb120-2383"><a href="#cb120-2383" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab)</span>
<span id="cb120-2384"><a href="#cb120-2384" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2385"><a href="#cb120-2385" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2388"><a href="#cb120-2388" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2389"><a href="#cb120-2389" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2390"><a href="#cb120-2390" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-2391"><a href="#cb120-2391" aria-hidden="true" tabindex="-1"></a>seqtabY <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">15</span>, <span class="dv">3364</span>)</span>
<span id="cb120-2392"><a href="#cb120-2392" aria-hidden="true" tabindex="-1"></a>seqtabY</span>
<span id="cb120-2393"><a href="#cb120-2393" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2394"><a href="#cb120-2394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2395"><a href="#cb120-2395" aria-hidden="true" tabindex="-1"></a>After removing the extreme length variants, we have <span class="in">`r (seqtabY)[2]`</span>, a reduction of <span class="in">`r (seqtabX)[2] - (seqtabY)[2]`</span> sequence variants.</span>
<span id="cb120-2396"><a href="#cb120-2396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2399"><a href="#cb120-2399" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2400"><a href="#cb120-2400" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab)))</span>
<span id="cb120-2401"><a href="#cb120-2401" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2402"><a href="#cb120-2402" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2405"><a href="#cb120-2405" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2406"><a href="#cb120-2406" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2407"><a href="#cb120-2407" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(seqtab, <span class="st">"its18_seqtab.rds"</span>)</span>
<span id="cb120-2408"><a href="#cb120-2408" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2409"><a href="#cb120-2409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2410"><a href="#cb120-2410" aria-hidden="true" tabindex="-1"></a><span class="fu">## 10. Remove Chimeras</span></span>
<span id="cb120-2411"><a href="#cb120-2411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2414"><a href="#cb120-2414" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2415"><a href="#cb120-2415" aria-hidden="true" tabindex="-1"></a>seqtab.nochim <span class="ot">&lt;-</span> <span class="fu">removeBimeraDenovo</span>(seqtab,</span>
<span id="cb120-2416"><a href="#cb120-2416" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">method =</span> <span class="st">"consensus"</span>,</span>
<span id="cb120-2417"><a href="#cb120-2417" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">multithread =</span> <span class="cn">TRUE</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-2418"><a href="#cb120-2418" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab.nochim)</span>
<span id="cb120-2419"><a href="#cb120-2419" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2420"><a href="#cb120-2420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2421"><a href="#cb120-2421" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2422"><a href="#cb120-2422" aria-hidden="true" tabindex="-1"></a><span class="in">Identified 7 bimeras out of 3364 input sequences.</span></span>
<span id="cb120-2423"><a href="#cb120-2423" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2424"><a href="#cb120-2424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2427"><a href="#cb120-2427" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2428"><a href="#cb120-2428" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2429"><a href="#cb120-2429" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-2430"><a href="#cb120-2430" aria-hidden="true" tabindex="-1"></a>seqtabZ <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">15</span>, <span class="dv">3357</span>)</span>
<span id="cb120-2431"><a href="#cb120-2431" aria-hidden="true" tabindex="-1"></a>seqtabZ</span>
<span id="cb120-2432"><a href="#cb120-2432" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2433"><a href="#cb120-2433" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2436"><a href="#cb120-2436" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2437"><a href="#cb120-2437" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(seqtab.nochim)<span class="sc">/</span><span class="fu">sum</span>(seqtab<span class="fl">.2</span>)</span>
<span id="cb120-2438"><a href="#cb120-2438" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2439"><a href="#cb120-2439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2442"><a href="#cb120-2442" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2443"><a href="#cb120-2443" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2444"><a href="#cb120-2444" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-2445"><a href="#cb120-2445" aria-hidden="true" tabindex="-1"></a>chim_rem <span class="ot">&lt;-</span> <span class="fl">0.9990135</span></span>
<span id="cb120-2446"><a href="#cb120-2446" aria-hidden="true" tabindex="-1"></a><span class="fl">0.9990135</span></span>
<span id="cb120-2447"><a href="#cb120-2447" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2448"><a href="#cb120-2448" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2449"><a href="#cb120-2449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2450"><a href="#cb120-2450" aria-hidden="true" tabindex="-1"></a>Chimera checking removed an additional <span class="in">`r (seqtabY)[2] - (seqtabZ)[2]`</span> sequence variants however, when we account for the abundances of each variant, we see chimeras accounts for about <span class="in">`r 100*(1-chim_rem)`</span>% of the merged sequence reads. Curious.</span>
<span id="cb120-2451"><a href="#cb120-2451" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2454"><a href="#cb120-2454" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2455"><a href="#cb120-2455" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">colSums</span>(seqtab.nochim<span class="sc">&gt;</span><span class="dv">0</span>))</span>
<span id="cb120-2456"><a href="#cb120-2456" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2457"><a href="#cb120-2457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2458"><a href="#cb120-2458" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2459"><a href="#cb120-2459" aria-hidden="true" tabindex="-1"></a><span class="in">   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15</span></span>
<span id="cb120-2460"><a href="#cb120-2460" aria-hidden="true" tabindex="-1"></a><span class="in">1247  711  443  289  204  123  102   86   49   32   24   15   23    7    2</span></span>
<span id="cb120-2461"><a href="#cb120-2461" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2462"><a href="#cb120-2462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2465"><a href="#cb120-2465" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2466"><a href="#cb120-2466" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">colSums</span>(seqtab.nochim<span class="sc">&gt;</span><span class="dv">0</span>))</span>
<span id="cb120-2467"><a href="#cb120-2467" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2468"><a href="#cb120-2468" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2469"><a href="#cb120-2469" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2470"><a href="#cb120-2470" aria-hidden="true" tabindex="-1"></a><span class="in">  14   95  335  349  616  702  719  745  765  812  867  945  954 1011 1017</span></span>
<span id="cb120-2471"><a href="#cb120-2471" aria-hidden="true" tabindex="-1"></a><span class="in">   1    1    1    1    1    1    1    1    1    1    1    1    1    1    1</span></span>
<span id="cb120-2472"><a href="#cb120-2472" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2473"><a href="#cb120-2473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2474"><a href="#cb120-2474" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2475"><a href="#cb120-2475" aria-hidden="true" tabindex="-1"></a>The last thing we want to do is write the sequence table to an <span class="in">`RDS`</span> file.</span>
<span id="cb120-2476"><a href="#cb120-2476" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2479"><a href="#cb120-2479" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2480"><a href="#cb120-2480" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(seqtab.nochim, <span class="st">"its18_seqtab.nochim.rds"</span>)</span>
<span id="cb120-2481"><a href="#cb120-2481" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2482"><a href="#cb120-2482" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2485"><a href="#cb120-2485" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2486"><a href="#cb120-2486" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2487"><a href="#cb120-2487" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab.nochim)))</span>
<span id="cb120-2488"><a href="#cb120-2488" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2489"><a href="#cb120-2489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2490"><a href="#cb120-2490" aria-hidden="true" tabindex="-1"></a><span class="fu">## 11. Track Reads through Workflow</span></span>
<span id="cb120-2491"><a href="#cb120-2491" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2494"><a href="#cb120-2494" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2495"><a href="#cb120-2495" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2496"><a href="#cb120-2496" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-2497"><a href="#cb120-2497" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"files/dada2/tables/its18_read_changes.txt"</span>,</span>
<span id="cb120-2498"><a href="#cb120-2498" aria-hidden="true" tabindex="-1"></a>                        <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb120-2499"><a href="#cb120-2499" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2500"><a href="#cb120-2500" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2501"><a href="#cb120-2501" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab("track_its")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb120-2502"><a href="#cb120-2502" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2505"><a href="#cb120-2505" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2506"><a href="#cb120-2506" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2507"><a href="#cb120-2507" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-2508"><a href="#cb120-2508" aria-hidden="true" tabindex="-1"></a>seq_table <span class="sc">%&gt;%</span></span>
<span id="cb120-2509"><a href="#cb120-2509" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb120-2510"><a href="#cb120-2510" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"its18_track_read_changes"</span>,</span>
<span id="cb120-2511"><a href="#cb120-2511" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb120-2512"><a href="#cb120-2512" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb120-2513"><a href="#cb120-2513" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb120-2514"><a href="#cb120-2514" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb120-2515"><a href="#cb120-2515" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-2516"><a href="#cb120-2516" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb120-2517"><a href="#cb120-2517" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2518"><a href="#cb120-2518" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb120-2519"><a href="#cb120-2519" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb120-2520"><a href="#cb120-2520" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb120-2521"><a href="#cb120-2521" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">1</span>),</span>
<span id="cb120-2522"><a href="#cb120-2522" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">TRUE</span>, <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-2523"><a href="#cb120-2523" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb120-2524"><a href="#cb120-2524" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-2525"><a href="#cb120-2525" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb120-2526"><a href="#cb120-2526" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Sample ID"</span>, </span>
<span id="cb120-2527"><a href="#cb120-2527" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb120-2528"><a href="#cb120-2528" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>),</span>
<span id="cb120-2529"><a href="#cb120-2529" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>), </span>
<span id="cb120-2530"><a href="#cb120-2530" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb120-2531"><a href="#cb120-2531" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">150</span>, <span class="at">footer =</span> <span class="st">"Total reads"</span>), </span>
<span id="cb120-2532"><a href="#cb120-2532" aria-hidden="true" tabindex="-1"></a>    <span class="at">raw =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-2533"><a href="#cb120-2533" aria-hidden="true" tabindex="-1"></a>    <span class="at">pre_filt =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-2534"><a href="#cb120-2534" aria-hidden="true" tabindex="-1"></a>    <span class="at">pre_filt =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-2535"><a href="#cb120-2535" aria-hidden="true" tabindex="-1"></a>    <span class="at">cut =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-2536"><a href="#cb120-2536" aria-hidden="true" tabindex="-1"></a>    <span class="at">filtered =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-2537"><a href="#cb120-2537" aria-hidden="true" tabindex="-1"></a>    <span class="at">denoisedF =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-2538"><a href="#cb120-2538" aria-hidden="true" tabindex="-1"></a>    <span class="at">denoisedR =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-2539"><a href="#cb120-2539" aria-hidden="true" tabindex="-1"></a>    <span class="at">merged =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values))),</span>
<span id="cb120-2540"><a href="#cb120-2540" aria-hidden="true" tabindex="-1"></a>    <span class="at">nonchim =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values)))</span>
<span id="cb120-2541"><a href="#cb120-2541" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb120-2542"><a href="#cb120-2542" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">5</span>, </span>
<span id="cb120-2543"><a href="#cb120-2543" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb120-2544"><a href="#cb120-2544" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb120-2545"><a href="#cb120-2545" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb120-2546"><a href="#cb120-2546" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-2547"><a href="#cb120-2547" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb120-2548"><a href="#cb120-2548" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2549"><a href="#cb120-2549" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2552"><a href="#cb120-2552" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2553"><a href="#cb120-2553" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2554"><a href="#cb120-2554" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="st">"pre_tax_its_2018.rdata"</span>)</span>
<span id="cb120-2555"><a href="#cb120-2555" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2556"><a href="#cb120-2556" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2557"><a href="#cb120-2557" aria-hidden="true" tabindex="-1"></a><span class="fu">## 12. Assign Taxonomy</span></span>
<span id="cb120-2558"><a href="#cb120-2558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2559"><a href="#cb120-2559" aria-hidden="true" tabindex="-1"></a>For taxonomic classification, we used the UNITE <span class="co">[</span><span class="ot">@nilsson2019unite</span><span class="co">]</span> database, specifically the <span class="co">[</span><span class="ot">UNITE general FASTA release for Fungi (v. 04.02.2020)</span><span class="co">](https://plutof.ut.ee/#/doi/10.15156/BIO/786368)</span> <span class="co">[</span><span class="ot">@abarenkov2020unite</span><span class="co">]</span>.</span>
<span id="cb120-2560"><a href="#cb120-2560" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2563"><a href="#cb120-2563" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2564"><a href="#cb120-2564" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2565"><a href="#cb120-2565" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb120-2566"><a href="#cb120-2566" aria-hidden="true" tabindex="-1"></a>seqtab <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"its18_seqtab.nochim.rds"</span>)</span>
<span id="cb120-2567"><a href="#cb120-2567" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2568"><a href="#cb120-2568" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2571"><a href="#cb120-2571" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2572"><a href="#cb120-2572" aria-hidden="true" tabindex="-1"></a>tax <span class="ot">&lt;-</span> <span class="fu">assignTaxonomy</span>(seqtab,</span>
<span id="cb120-2573"><a href="#cb120-2573" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"sh_general_release_dynamic_s_04.02.2020_dev.fasta"</span>,</span>
<span id="cb120-2574"><a href="#cb120-2574" aria-hidden="true" tabindex="-1"></a>                      <span class="at">multithread =</span> <span class="dv">20</span>, <span class="at">tryRC =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-2575"><a href="#cb120-2575" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(tax, <span class="st">"its18_tax_its.rds"</span>)</span>
<span id="cb120-2576"><a href="#cb120-2576" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2577"><a href="#cb120-2577" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2578"><a href="#cb120-2578" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2579"><a href="#cb120-2579" aria-hidden="true" tabindex="-1"></a><span class="in">UNITE fungal taxonomic reference detected.</span></span>
<span id="cb120-2580"><a href="#cb120-2580" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2581"><a href="#cb120-2581" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2582"><a href="#cb120-2582" aria-hidden="true" tabindex="-1"></a><span class="fu">## 13. Save Image</span></span>
<span id="cb120-2583"><a href="#cb120-2583" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2586"><a href="#cb120-2586" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2587"><a href="#cb120-2587" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="st">"its18_dada2_wf.rdata"</span>)</span>
<span id="cb120-2588"><a href="#cb120-2588" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2589"><a href="#cb120-2589" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2590"><a href="#cb120-2590" aria-hidden="true" tabindex="-1"></a><span class="fu">## R Session Information &amp; Code</span></span>
<span id="cb120-2591"><a href="#cb120-2591" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2592"><a href="#cb120-2592" aria-hidden="true" tabindex="-1"></a>This workflow was run on the <span class="co">[</span><span class="ot">Smithsonian High Performance Cluster (SI/HPC)</span><span class="co">](https://doi.org/10.25572/SIHPC)</span>, Smithsonian Institution. Below are the specific packages and versions used in this workflow using both <span class="in">`sessionInfo()`</span> and <span class="in">`devtools::session_info()`</span>. Click the arrow to see the details.</span>
<span id="cb120-2593"><a href="#cb120-2593" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2594"><a href="#cb120-2594" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb120-2595"><a href="#cb120-2595" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show/hide  R Session Info<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb120-2596"><a href="#cb120-2596" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2597"><a href="#cb120-2597" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2598"><a href="#cb120-2598" aria-hidden="true" tabindex="-1"></a><span class="in">sessionInfo()</span></span>
<span id="cb120-2599"><a href="#cb120-2599" aria-hidden="true" tabindex="-1"></a><span class="in">R version 4.0.0 (2020-04-24)</span></span>
<span id="cb120-2600"><a href="#cb120-2600" aria-hidden="true" tabindex="-1"></a><span class="in">Platform: x86_64-conda_cos6-linux-gnu (64-bit)</span></span>
<span id="cb120-2601"><a href="#cb120-2601" aria-hidden="true" tabindex="-1"></a><span class="in">Running under: CentOS Linux 7 (Core)</span></span>
<span id="cb120-2602"><a href="#cb120-2602" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2603"><a href="#cb120-2603" aria-hidden="true" tabindex="-1"></a><span class="in">Matrix products: default</span></span>
<span id="cb120-2604"><a href="#cb120-2604" aria-hidden="true" tabindex="-1"></a><span class="in">BLAS/LAPACK: /home/scottjj/miniconda3/envs/R-4/lib/libopenblasp-r0.3.9.so</span></span>
<span id="cb120-2605"><a href="#cb120-2605" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2606"><a href="#cb120-2606" aria-hidden="true" tabindex="-1"></a><span class="in">locale:</span></span>
<span id="cb120-2607"><a href="#cb120-2607" aria-hidden="true" tabindex="-1"></a><span class="in"> [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C</span></span>
<span id="cb120-2608"><a href="#cb120-2608" aria-hidden="true" tabindex="-1"></a><span class="in"> [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8</span></span>
<span id="cb120-2609"><a href="#cb120-2609" aria-hidden="true" tabindex="-1"></a><span class="in"> [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8</span></span>
<span id="cb120-2610"><a href="#cb120-2610" aria-hidden="true" tabindex="-1"></a><span class="in"> [7] LC_PAPER=en_US.UTF-8       LC_NAME=C</span></span>
<span id="cb120-2611"><a href="#cb120-2611" aria-hidden="true" tabindex="-1"></a><span class="in"> [9] LC_ADDRESS=C               LC_TELEPHONE=C</span></span>
<span id="cb120-2612"><a href="#cb120-2612" aria-hidden="true" tabindex="-1"></a><span class="in">[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C</span></span>
<span id="cb120-2613"><a href="#cb120-2613" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2614"><a href="#cb120-2614" aria-hidden="true" tabindex="-1"></a><span class="in">attached base packages:</span></span>
<span id="cb120-2615"><a href="#cb120-2615" aria-hidden="true" tabindex="-1"></a><span class="in"> [1] grid      stats4    parallel  stats     graphics  grDevices utils</span></span>
<span id="cb120-2616"><a href="#cb120-2616" aria-hidden="true" tabindex="-1"></a><span class="in"> [8] datasets  methods   base</span></span>
<span id="cb120-2617"><a href="#cb120-2617" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2618"><a href="#cb120-2618" aria-hidden="true" tabindex="-1"></a><span class="in">other attached packages:</span></span>
<span id="cb120-2619"><a href="#cb120-2619" aria-hidden="true" tabindex="-1"></a><span class="in"> [1] gridExtra_2.3               ggplot2_3.3.0</span></span>
<span id="cb120-2620"><a href="#cb120-2620" aria-hidden="true" tabindex="-1"></a><span class="in"> [3] ShortRead_1.46.0            GenomicAlignments_1.24.0</span></span>
<span id="cb120-2621"><a href="#cb120-2621" aria-hidden="true" tabindex="-1"></a><span class="in"> [5] SummarizedExperiment_1.18.1 DelayedArray_0.14.0</span></span>
<span id="cb120-2622"><a href="#cb120-2622" aria-hidden="true" tabindex="-1"></a><span class="in"> [7] matrixStats_0.56.0          Biobase_2.48.0</span></span>
<span id="cb120-2623"><a href="#cb120-2623" aria-hidden="true" tabindex="-1"></a><span class="in"> [9] Rsamtools_2.4.0             GenomicRanges_1.40.0</span></span>
<span id="cb120-2624"><a href="#cb120-2624" aria-hidden="true" tabindex="-1"></a><span class="in">[11] GenomeInfoDb_1.24.0         Biostrings_2.56.0</span></span>
<span id="cb120-2625"><a href="#cb120-2625" aria-hidden="true" tabindex="-1"></a><span class="in">[13] XVector_0.28.0              IRanges_2.22.1</span></span>
<span id="cb120-2626"><a href="#cb120-2626" aria-hidden="true" tabindex="-1"></a><span class="in">[15] S4Vectors_0.26.0            BiocParallel_1.22.0</span></span>
<span id="cb120-2627"><a href="#cb120-2627" aria-hidden="true" tabindex="-1"></a><span class="in">[17] BiocGenerics_0.34.0         dada2_1.16.0</span></span>
<span id="cb120-2628"><a href="#cb120-2628" aria-hidden="true" tabindex="-1"></a><span class="in">[19] Rcpp_1.0.4.6</span></span>
<span id="cb120-2629"><a href="#cb120-2629" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2630"><a href="#cb120-2630" aria-hidden="true" tabindex="-1"></a><span class="in">loaded via a namespace (and not attached):</span></span>
<span id="cb120-2631"><a href="#cb120-2631" aria-hidden="true" tabindex="-1"></a><span class="in"> [1] plyr_1.8.6             RColorBrewer_1.1-2     pillar_1.4.4</span></span>
<span id="cb120-2632"><a href="#cb120-2632" aria-hidden="true" tabindex="-1"></a><span class="in"> [4] compiler_4.0.0         bitops_1.0-6           tools_4.0.0</span></span>
<span id="cb120-2633"><a href="#cb120-2633" aria-hidden="true" tabindex="-1"></a><span class="in"> [7] zlibbioc_1.34.0        lifecycle_0.2.0        tibble_3.0.1</span></span>
<span id="cb120-2634"><a href="#cb120-2634" aria-hidden="true" tabindex="-1"></a><span class="in">[10] gtable_0.3.0           lattice_0.20-41        png_0.1-7</span></span>
<span id="cb120-2635"><a href="#cb120-2635" aria-hidden="true" tabindex="-1"></a><span class="in">[13] pkgconfig_2.0.3        rlang_0.4.6            Matrix_1.2-18</span></span>
<span id="cb120-2636"><a href="#cb120-2636" aria-hidden="true" tabindex="-1"></a><span class="in">[16] GenomeInfoDbData_1.2.3 withr_2.2.0            stringr_1.4.0</span></span>
<span id="cb120-2637"><a href="#cb120-2637" aria-hidden="true" tabindex="-1"></a><span class="in">[19] hwriter_1.3.2          vctrs_0.3.0            glue_1.4.1</span></span>
<span id="cb120-2638"><a href="#cb120-2638" aria-hidden="true" tabindex="-1"></a><span class="in">[22] R6_2.4.1               jpeg_0.1-8.1           latticeExtra_0.6-29</span></span>
<span id="cb120-2639"><a href="#cb120-2639" aria-hidden="true" tabindex="-1"></a><span class="in">[25] reshape2_1.4.4         magrittr_1.5           scales_1.1.1</span></span>
<span id="cb120-2640"><a href="#cb120-2640" aria-hidden="true" tabindex="-1"></a><span class="in">[28] ellipsis_0.3.1         colorspace_1.4-1       stringi_1.4.6</span></span>
<span id="cb120-2641"><a href="#cb120-2641" aria-hidden="true" tabindex="-1"></a><span class="in">[31] RCurl_1.98-1.2         RcppParallel_5.0.1     munsell_0.5.0</span></span>
<span id="cb120-2642"><a href="#cb120-2642" aria-hidden="true" tabindex="-1"></a><span class="in">[34] crayon_1.3.4</span></span>
<span id="cb120-2643"><a href="#cb120-2643" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2644"><a href="#cb120-2644" aria-hidden="true" tabindex="-1"></a><span class="in">devtools::session_info()</span></span>
<span id="cb120-2645"><a href="#cb120-2645" aria-hidden="true" tabindex="-1"></a><span class="in">â”€ Session info â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span>
<span id="cb120-2646"><a href="#cb120-2646" aria-hidden="true" tabindex="-1"></a><span class="in"> setting  value</span></span>
<span id="cb120-2647"><a href="#cb120-2647" aria-hidden="true" tabindex="-1"></a><span class="in"> version  R version 4.0.0 (2020-04-24)</span></span>
<span id="cb120-2648"><a href="#cb120-2648" aria-hidden="true" tabindex="-1"></a><span class="in"> os       CentOS Linux 7 (Core)</span></span>
<span id="cb120-2649"><a href="#cb120-2649" aria-hidden="true" tabindex="-1"></a><span class="in"> system   x86_64, linux-gnu</span></span>
<span id="cb120-2650"><a href="#cb120-2650" aria-hidden="true" tabindex="-1"></a><span class="in"> ui       X11</span></span>
<span id="cb120-2651"><a href="#cb120-2651" aria-hidden="true" tabindex="-1"></a><span class="in"> language (EN)</span></span>
<span id="cb120-2652"><a href="#cb120-2652" aria-hidden="true" tabindex="-1"></a><span class="in"> collate  en_US.UTF-8</span></span>
<span id="cb120-2653"><a href="#cb120-2653" aria-hidden="true" tabindex="-1"></a><span class="in"> ctype    en_US.UTF-8</span></span>
<span id="cb120-2654"><a href="#cb120-2654" aria-hidden="true" tabindex="-1"></a><span class="in"> tz       America/New_York</span></span>
<span id="cb120-2655"><a href="#cb120-2655" aria-hidden="true" tabindex="-1"></a><span class="in"> date     2020-06-04</span></span>
<span id="cb120-2656"><a href="#cb120-2656" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2657"><a href="#cb120-2657" aria-hidden="true" tabindex="-1"></a><span class="in">â”€ Packages â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span>
<span id="cb120-2658"><a href="#cb120-2658" aria-hidden="true" tabindex="-1"></a><span class="in"> package              * version  date       lib source</span></span>
<span id="cb120-2659"><a href="#cb120-2659" aria-hidden="true" tabindex="-1"></a><span class="in"> assertthat             0.2.1    2019-03-21 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2660"><a href="#cb120-2660" aria-hidden="true" tabindex="-1"></a><span class="in"> backports              1.1.7    2020-05-13 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2661"><a href="#cb120-2661" aria-hidden="true" tabindex="-1"></a><span class="in"> Biobase              * 2.48.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-2662"><a href="#cb120-2662" aria-hidden="true" tabindex="-1"></a><span class="in"> BiocGenerics         * 0.34.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-2663"><a href="#cb120-2663" aria-hidden="true" tabindex="-1"></a><span class="in"> BiocParallel         * 1.22.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-2664"><a href="#cb120-2664" aria-hidden="true" tabindex="-1"></a><span class="in"> Biostrings           * 2.56.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-2665"><a href="#cb120-2665" aria-hidden="true" tabindex="-1"></a><span class="in"> bitops                 1.0-6    2013-08-17 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2666"><a href="#cb120-2666" aria-hidden="true" tabindex="-1"></a><span class="in"> callr                  3.4.3    2020-03-28 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2667"><a href="#cb120-2667" aria-hidden="true" tabindex="-1"></a><span class="in"> cli                    2.0.2    2020-02-28 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2668"><a href="#cb120-2668" aria-hidden="true" tabindex="-1"></a><span class="in"> colorspace             1.4-1    2019-03-18 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2669"><a href="#cb120-2669" aria-hidden="true" tabindex="-1"></a><span class="in"> crayon                 1.3.4    2017-09-16 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2670"><a href="#cb120-2670" aria-hidden="true" tabindex="-1"></a><span class="in"> dada2                * 1.16.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-2671"><a href="#cb120-2671" aria-hidden="true" tabindex="-1"></a><span class="in"> DelayedArray         * 0.14.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-2672"><a href="#cb120-2672" aria-hidden="true" tabindex="-1"></a><span class="in"> desc                   1.2.0    2018-05-01 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2673"><a href="#cb120-2673" aria-hidden="true" tabindex="-1"></a><span class="in"> devtools               2.3.0    2020-04-10 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2674"><a href="#cb120-2674" aria-hidden="true" tabindex="-1"></a><span class="in"> digest                 0.6.25   2020-02-23 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2675"><a href="#cb120-2675" aria-hidden="true" tabindex="-1"></a><span class="in"> ellipsis               0.3.1    2020-05-15 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2676"><a href="#cb120-2676" aria-hidden="true" tabindex="-1"></a><span class="in"> fansi                  0.4.1    2020-01-08 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2677"><a href="#cb120-2677" aria-hidden="true" tabindex="-1"></a><span class="in"> fs                     1.4.1    2020-04-04 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2678"><a href="#cb120-2678" aria-hidden="true" tabindex="-1"></a><span class="in"> GenomeInfoDb         * 1.24.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-2679"><a href="#cb120-2679" aria-hidden="true" tabindex="-1"></a><span class="in"> GenomeInfoDbData       1.2.3    2020-05-22 [1] Bioconductor</span></span>
<span id="cb120-2680"><a href="#cb120-2680" aria-hidden="true" tabindex="-1"></a><span class="in"> GenomicAlignments    * 1.24.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-2681"><a href="#cb120-2681" aria-hidden="true" tabindex="-1"></a><span class="in"> GenomicRanges        * 1.40.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-2682"><a href="#cb120-2682" aria-hidden="true" tabindex="-1"></a><span class="in"> ggplot2              * 3.3.0    2020-03-05 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2683"><a href="#cb120-2683" aria-hidden="true" tabindex="-1"></a><span class="in"> glue                   1.4.1    2020-05-13 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2684"><a href="#cb120-2684" aria-hidden="true" tabindex="-1"></a><span class="in"> gridExtra            * 2.3      2017-09-09 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2685"><a href="#cb120-2685" aria-hidden="true" tabindex="-1"></a><span class="in"> gtable                 0.3.0    2019-03-25 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2686"><a href="#cb120-2686" aria-hidden="true" tabindex="-1"></a><span class="in"> hwriter                1.3.2    2014-09-10 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2687"><a href="#cb120-2687" aria-hidden="true" tabindex="-1"></a><span class="in"> IRanges              * 2.22.1   2020-04-28 [1] Bioconductor</span></span>
<span id="cb120-2688"><a href="#cb120-2688" aria-hidden="true" tabindex="-1"></a><span class="in"> jpeg                   0.1-8.1  2019-10-24 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2689"><a href="#cb120-2689" aria-hidden="true" tabindex="-1"></a><span class="in"> lattice                0.20-41  2020-04-02 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2690"><a href="#cb120-2690" aria-hidden="true" tabindex="-1"></a><span class="in"> latticeExtra           0.6-29   2019-12-19 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2691"><a href="#cb120-2691" aria-hidden="true" tabindex="-1"></a><span class="in"> lifecycle              0.2.0    2020-03-06 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2692"><a href="#cb120-2692" aria-hidden="true" tabindex="-1"></a><span class="in"> magrittr               1.5      2014-11-22 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2693"><a href="#cb120-2693" aria-hidden="true" tabindex="-1"></a><span class="in"> Matrix                 1.2-18   2019-11-27 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2694"><a href="#cb120-2694" aria-hidden="true" tabindex="-1"></a><span class="in"> matrixStats          * 0.56.0   2020-03-13 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2695"><a href="#cb120-2695" aria-hidden="true" tabindex="-1"></a><span class="in"> memoise                1.1.0    2017-04-21 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2696"><a href="#cb120-2696" aria-hidden="true" tabindex="-1"></a><span class="in"> munsell                0.5.0    2018-06-12 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2697"><a href="#cb120-2697" aria-hidden="true" tabindex="-1"></a><span class="in"> pillar                 1.4.4    2020-05-05 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2698"><a href="#cb120-2698" aria-hidden="true" tabindex="-1"></a><span class="in"> pkgbuild               1.0.8    2020-05-07 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2699"><a href="#cb120-2699" aria-hidden="true" tabindex="-1"></a><span class="in"> pkgconfig              2.0.3    2019-09-22 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2700"><a href="#cb120-2700" aria-hidden="true" tabindex="-1"></a><span class="in"> pkgload                1.0.2    2018-10-29 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2701"><a href="#cb120-2701" aria-hidden="true" tabindex="-1"></a><span class="in"> plyr                   1.8.6    2020-03-03 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2702"><a href="#cb120-2702" aria-hidden="true" tabindex="-1"></a><span class="in"> png                    0.1-7    2013-12-03 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2703"><a href="#cb120-2703" aria-hidden="true" tabindex="-1"></a><span class="in"> prettyunits            1.1.1    2020-01-24 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2704"><a href="#cb120-2704" aria-hidden="true" tabindex="-1"></a><span class="in"> processx               3.4.2    2020-02-09 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2705"><a href="#cb120-2705" aria-hidden="true" tabindex="-1"></a><span class="in"> ps                     1.3.3    2020-05-08 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2706"><a href="#cb120-2706" aria-hidden="true" tabindex="-1"></a><span class="in"> R6                     2.4.1    2019-11-12 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2707"><a href="#cb120-2707" aria-hidden="true" tabindex="-1"></a><span class="in"> RColorBrewer           1.1-2    2014-12-07 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2708"><a href="#cb120-2708" aria-hidden="true" tabindex="-1"></a><span class="in"> Rcpp                 * 1.0.4.6  2020-04-09 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2709"><a href="#cb120-2709" aria-hidden="true" tabindex="-1"></a><span class="in"> RcppParallel           5.0.1    2020-05-06 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2710"><a href="#cb120-2710" aria-hidden="true" tabindex="-1"></a><span class="in"> RCurl                  1.98-1.2 2020-04-18 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2711"><a href="#cb120-2711" aria-hidden="true" tabindex="-1"></a><span class="in"> remotes                2.1.1    2020-02-15 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2712"><a href="#cb120-2712" aria-hidden="true" tabindex="-1"></a><span class="in"> reshape2               1.4.4    2020-04-09 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2713"><a href="#cb120-2713" aria-hidden="true" tabindex="-1"></a><span class="in"> rlang                  0.4.6    2020-05-02 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2714"><a href="#cb120-2714" aria-hidden="true" tabindex="-1"></a><span class="in"> rprojroot              1.3-2    2018-01-03 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2715"><a href="#cb120-2715" aria-hidden="true" tabindex="-1"></a><span class="in"> Rsamtools            * 2.4.0    2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-2716"><a href="#cb120-2716" aria-hidden="true" tabindex="-1"></a><span class="in"> S4Vectors            * 0.26.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-2717"><a href="#cb120-2717" aria-hidden="true" tabindex="-1"></a><span class="in"> scales                 1.1.1    2020-05-11 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2718"><a href="#cb120-2718" aria-hidden="true" tabindex="-1"></a><span class="in"> sessioninfo            1.1.1    2018-11-05 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2719"><a href="#cb120-2719" aria-hidden="true" tabindex="-1"></a><span class="in"> ShortRead            * 1.46.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-2720"><a href="#cb120-2720" aria-hidden="true" tabindex="-1"></a><span class="in"> stringi                1.4.6    2020-02-17 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2721"><a href="#cb120-2721" aria-hidden="true" tabindex="-1"></a><span class="in"> stringr                1.4.0    2019-02-10 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2722"><a href="#cb120-2722" aria-hidden="true" tabindex="-1"></a><span class="in"> SummarizedExperiment * 1.18.1   2020-04-30 [1] Bioconductor</span></span>
<span id="cb120-2723"><a href="#cb120-2723" aria-hidden="true" tabindex="-1"></a><span class="in"> testthat               2.3.2    2020-03-02 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2724"><a href="#cb120-2724" aria-hidden="true" tabindex="-1"></a><span class="in"> tibble                 3.0.1    2020-04-20 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2725"><a href="#cb120-2725" aria-hidden="true" tabindex="-1"></a><span class="in"> usethis                1.6.1    2020-04-29 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2726"><a href="#cb120-2726" aria-hidden="true" tabindex="-1"></a><span class="in"> vctrs                  0.3.0    2020-05-11 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2727"><a href="#cb120-2727" aria-hidden="true" tabindex="-1"></a><span class="in"> withr                  2.2.0    2020-04-20 [1] CRAN (R 4.0.0)</span></span>
<span id="cb120-2728"><a href="#cb120-2728" aria-hidden="true" tabindex="-1"></a><span class="in"> XVector              * 0.28.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-2729"><a href="#cb120-2729" aria-hidden="true" tabindex="-1"></a><span class="in"> zlibbioc               1.34.0   2020-04-27 [1] Bioconductor</span></span>
<span id="cb120-2730"><a href="#cb120-2730" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2731"><a href="#cb120-2731" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb120-2732"><a href="#cb120-2732" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2733"><a href="#cb120-2733" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br/&gt;</span></span>
<span id="cb120-2734"><a href="#cb120-2734" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2735"><a href="#cb120-2735" aria-hidden="true" tabindex="-1"></a><span class="fu"># Workflow Output </span></span>
<span id="cb120-2736"><a href="#cb120-2736" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2737"><a href="#cb120-2737" aria-hidden="true" tabindex="-1"></a>Data products generated in this workflow can be downloaded from figshare.</span>
<span id="cb120-2738"><a href="#cb120-2738" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2739"><a href="#cb120-2739" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;iframe</span> <span class="er">src</span><span class="ot">=</span><span class="st">"https://widgets.figshare.com/articles/14687184/embed?show_title=1"</span> <span class="er">width</span><span class="ot">=</span><span class="st">"100%"</span> <span class="er">height</span><span class="ot">=</span><span class="st">"351"</span> <span class="er">frameborder</span><span class="ot">=</span><span class="st">"0"</span><span class="kw">&gt;&lt;/iframe&gt;</span></span>
<span id="cb120-2740"><a href="#cb120-2740" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2741"><a href="#cb120-2741" aria-hidden="true" tabindex="-1"></a>The end!</span>
<span id="cb120-2742"><a href="#cb120-2742" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2745"><a href="#cb120-2745" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2746"><a href="#cb120-2746" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb120-2747"><a href="#cb120-2747" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-2748"><a href="#cb120-2748" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb120-2749"><a href="#cb120-2749" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2750"><a href="#cb120-2750" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2751"><a href="#cb120-2751" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"data-prep.html"</span> <span class="er">class</span><span class="ot">=</span><span class="st">"btnnav button_round"</span> <span class="er">style</span><span class="ot">=</span><span class="st">"float: right;"</span><span class="kw">&gt;</span>Next workflow:</span>
<span id="cb120-2752"><a href="#cb120-2752" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br&gt;</span> 2. Data Set Preparation<span class="kw">&lt;/a&gt;&lt;br&gt;&lt;br&gt;</span></span>
<span id="cb120-2753"><a href="#cb120-2753" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2756"><a href="#cb120-2756" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2757"><a href="#cb120-2757" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false </span></span>
<span id="cb120-2758"><a href="#cb120-2758" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb120-2759"><a href="#cb120-2759" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-2760"><a href="#cb120-2760" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2761"><a href="#cb120-2761" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb120-2762"><a href="#cb120-2762" aria-hidden="true" tabindex="-1"></a><span class="do">### COmmon formatting scripts</span></span>
<span id="cb120-2763"><a href="#cb120-2763" aria-hidden="true" tabindex="-1"></a><span class="do">### </span><span class="al">NOTE</span><span class="do">: captioner.R must be read BEFORE captions_XYZ.R</span></span>
<span id="cb120-2764"><a href="#cb120-2764" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">"assets"</span>, <span class="st">"functions.R"</span>))</span>
<span id="cb120-2765"><a href="#cb120-2765" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb120-2766"><a href="#cb120-2766" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2767"><a href="#cb120-2767" aria-hidden="true" tabindex="-1"></a><span class="fu">####  Source Code {.appendix}</span></span>
<span id="cb120-2768"><a href="#cb120-2768" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2769"><a href="#cb120-2769" aria-hidden="true" tabindex="-1"></a>The source code for this page can be accessed on GitHub <span class="in">`r fa(name = "github")`</span> by <span class="co">[</span><span class="ot">clicking this link</span><span class="co">]</span>(<span class="in">`r source_code()`</span>). </span>
<span id="cb120-2770"><a href="#cb120-2770" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2771"><a href="#cb120-2771" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Data Availability {.appendix}</span></span>
<span id="cb120-2772"><a href="#cb120-2772" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2773"><a href="#cb120-2773" aria-hidden="true" tabindex="-1"></a>Raw fastq files available on figshare at <span class="co">[</span><span class="ot">10.25573/data.14686665</span><span class="co">](https://doi.org/10.25573/data.14686665)</span> for the 16S rRNA data and <span class="co">[</span><span class="ot">10.25573/data.14686755</span><span class="co">](https://doi.org/10.25573/data.14686755)</span> for the ITS data. Trimmed fastq files (primers removed) available through the ENA under project accession number <span class="co">[</span><span class="ot">PRJEB45074 (ERP129199)</span><span class="co">](https://www.ebi.ac.uk/ena/browser/view/PRJEB45074)</span>. Data generated in this workflow can be accessed on figshare at <span class="co">[</span><span class="ot">10.25573/data.14687184</span><span class="co">](https://doi.org/10.25573/data.14687184)</span>.</span>
<span id="cb120-2774"><a href="#cb120-2774" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2775"><a href="#cb120-2775" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Last updated on {.appendix}</span></span>
<span id="cb120-2776"><a href="#cb120-2776" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-2779"><a href="#cb120-2779" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb120-2780"><a href="#cb120-2780" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb120-2781"><a href="#cb120-2781" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb120-2782"><a href="#cb120-2782" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb120-2783"><a href="#cb120-2783" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
      <div class="nav-footer-center">Â© Copyright 2022 <a href="https://metacrobe.github.io/">metacrobe</a> &amp; The SWELTR Project. Site constructed using <a href="https://quarto.org/">Quarto</a>.</div>
  </div>
</footer>


</body></html>