<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */
</style>

<style>
  div.csl-bib-body { }
  div.csl-entry {
    clear: both;
    }
  .hanging div.csl-entry {
    margin-left:2em;
    text-indent:-2em;
  }
  div.csl-left-margin {
    min-width:2em;
    float:left;
  }
  div.csl-right-inline {
    margin-left:2em;
    padding-left:1em;
  }
  div.csl-indent {
    margin-left: 2em;
  }
</style>

  <!--radix_placeholder_meta_tags-->
  <title>: 2. Data Set Preparation</title>

  <meta property="description" itemprop="description" content="Workflow for preparation  of the **16S rRNA** and **ITS** data sets. These steps are needed before analyzing the data. In this workflow, sample groups are defined, and phyloseq objects are created and curated."/>

  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>


  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content=": 2. Data Set Preparation"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="Workflow for preparation  of the **16S rRNA** and **ITS** data sets. These steps are needed before analyzing the data. In this workflow, sample groups are defined, and phyloseq objects are created and curated."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content=""/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content=": 2. Data Set Preparation"/>
  <meta property="twitter:description" content="Workflow for preparation  of the **16S rRNA** and **ITS** data sets. These steps are needed before analyzing the data. In this workflow, sample groups are defined, and phyloseq objects are created and curated."/>

  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Phyloseq: An r package for reproducible interactive analysis and graphics of microbiome census data;citation_publication_date=2013;citation_volume=8;citation_author=Paul J McMurdie;citation_author=Susan Holmes"/>
  <meta name="citation_reference" content="citation_title=Tools for microbiome analysis in r. Version 1.9.97;citation_publication_date=2017;citation_author=Leo Lahti;citation_author=Shetty Sudarshan;citation_author=Tools for microbiome analysis in r. Version 1.9.97"/>
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["2. Data Set Preparation"]},{"type":"character","attributes":{},"value":["Workflow for preparation  of the **16S rRNA** and **ITS** data sets. These steps are needed before analyzing the data. In this workflow, sample groups are defined, and phyloseq objects are created and curated.\n"]},{"type":"character","attributes":{},"value":["assets/cite.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }

  </style>

  <script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <script src="site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
  <script src="site_libs/fuse-6.4.1/fuse.min.js"></script>

  <script type="application/javascript">

  function getMeta(metaName) {
    var metas = document.getElementsByTagName('meta');
    for (let i = 0; i < metas.length; i++) {
      if (metas[i].getAttribute('name') === metaName) {
        return metas[i].getAttribute('content');
      }
    }
    return '';
  }

  function offsetURL(url) {
    var offset = getMeta('distill:offset');
    return offset ? offset + '/' + url : url;
  }

  function createFuseIndex() {

    // create fuse index
    var options = {
      keys: [
        { name: 'title', weight: 20 },
        { name: 'categories', weight: 15 },
        { name: 'description', weight: 10 },
        { name: 'contents', weight: 5 },
      ],
      ignoreLocation: true,
      threshold: 0
    };
    var fuse = new window.Fuse([], options);

    // fetch the main search.json
    return fetch(offsetURL('search.json'))
      .then(function(response) {
        if (response.status == 200) {
          return response.json().then(function(json) {
            // index main articles
            json.articles.forEach(function(article) {
              fuse.add(article);
            });
            // download collections and index their articles
            return Promise.all(json.collections.map(function(collection) {
              return fetch(offsetURL(collection)).then(function(response) {
                if (response.status === 200) {
                  return response.json().then(function(articles) {
                    articles.forEach(function(article) {
                      fuse.add(article);
                    });
                  })
                } else {
                  return Promise.reject(
                    new Error('Unexpected status from search index request: ' +
                              response.status)
                  );
                }
              });
            })).then(function() {
              return fuse;
            });
          });

        } else {
          return Promise.reject(
            new Error('Unexpected status from search index request: ' +
                        response.status)
          );
        }
      });
  }

  window.document.addEventListener("DOMContentLoaded", function (event) {

    // get search element (bail if we don't have one)
    var searchEl = window.document.getElementById('distill-search');
    if (!searchEl)
      return;

    createFuseIndex()
      .then(function(fuse) {

        // make search box visible
        searchEl.classList.remove('hidden');

        // initialize autocomplete
        var options = {
          autoselect: true,
          hint: false,
          minLength: 2,
        };
        window.autocomplete(searchEl, options, [{
          source: function(query, callback) {
            const searchOptions = {
              isCaseSensitive: false,
              shouldSort: true,
              minMatchCharLength: 2,
              limit: 10,
            };
            var results = fuse.search(query, searchOptions);
            callback(results
              .map(function(result) { return result.item; })
            );
          },
          templates: {
            suggestion: function(suggestion) {
              var img = suggestion.preview && Object.keys(suggestion.preview).length > 0
                ? `<img src="${offsetURL(suggestion.preview)}"</img>`
                : '';
              var html = `
                <div class="search-item">
                  <h3>${suggestion.title}</h3>
                  <div class="search-item-description">
                    ${suggestion.description || ''}
                  </div>
                  <div class="search-item-preview">
                    ${img}
                  </div>
                </div>
              `;
              return html;
            }
          }
        }]).on('autocomplete:selected', function(event, suggestion) {
          window.location.href = offsetURL(suggestion.path);
        });
        // remove inline display style on autocompleter (we want to
        // manage responsive display via css)
        $('.algolia-autocomplete').css("display", "");
      })
      .catch(function(error) {
        console.log(error);
      });

  });

  </script>

  <style type="text/css">

  .nav-search {
    font-size: x-small;
  }

  /* Algolioa Autocomplete */

  .algolia-autocomplete {
    display: inline-block;
    margin-left: 10px;
    vertical-align: sub;
    background-color: white;
    color: black;
    padding: 6px;
    padding-top: 8px;
    padding-bottom: 0;
    border-radius: 6px;
    border: 1px #0F2E3D solid;
    width: 180px;
  }


  @media screen and (max-width: 768px) {
    .distill-site-nav .algolia-autocomplete {
      display: none;
      visibility: hidden;
    }
    .distill-site-nav.responsive .algolia-autocomplete {
      display: inline-block;
      visibility: visible;
    }
    .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
      margin-left: 0;
      width: 400px;
      max-height: 400px;
    }
  }

  .algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
    width: 90%;
    outline: none;
    border: none;
  }

  .algolia-autocomplete .aa-hint {
    color: #999;
  }
  .algolia-autocomplete .aa-dropdown-menu {
    width: 550px;
    max-height: 70vh;
    overflow-x: visible;
    overflow-y: scroll;
    padding: 5px;
    margin-top: 3px;
    margin-left: -150px;
    background-color: #fff;
    border-radius: 5px;
    border: 1px solid #999;
    border-top: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
    cursor: pointer;
    padding: 5px 4px;
    border-bottom: 1px solid #eee;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
    border-bottom: none;
    margin-bottom: 2px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
    overflow: hidden;
    font-size: 0.8em;
    line-height: 1.4em;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
    font-size: 1rem;
    margin-block-start: 0;
    margin-block-end: 5px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
    display: inline-block;
    overflow: hidden;
    height: 2.8em;
    width: 80%;
    margin-right: 4%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
    display: inline-block;
    width: 15%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
    height: 3em;
    width: auto;
    display: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
    display: initial;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
    background-color: #eee;
  }
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
    font-weight: bold;
    font-style: normal;
  }

  </style>


  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure img {
    width: 100%;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refs = $(this).attr('data-cites').split(" ");
        var refHtml = refs.map(function(ref) {
          return "<p>" + $('#ref-' + ref).html() + "</p>";
        }).join("\n");
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
  <script src="site_libs/clipboard-2.0.6/clipboard.min.js"></script>
  <link href="site_libs/xaringanExtra-clipboard-0.2.6/xaringanExtra-clipboard.css" rel="stylesheet" />
  <script src="site_libs/xaringanExtra-clipboard-0.2.6/xaringanExtra-clipboard.js"></script>
  <script>window.xaringanExtraClipboard(null, {"button":"<i class=\"fa fa-clone fa-2x\" style=\"color: #004276\"><\/i>","success":"<i class=\"fa fa-check fa-2x\" style=\"color: #90BE6D\"><\/i>","error":"<i class=\"fa fa-times fa-2x\" style=\"color: #F94144\"><\/i>"})</script>
  <script src="site_libs/font-awesome-5.13.0/js/script.js"></script>
  <link href="site_libs/panelset-0.2.6/panelset.css" rel="stylesheet" />
  <script src="site_libs/panelset-0.2.6/panelset.js"></script>
  <script src="site_libs/core-js-2.5.3/shim.min.js"></script>
  <script src="site_libs/react-17.0.0/react.min.js"></script>
  <script src="site_libs/react-17.0.0/react-dom.min.js"></script>
  <script src="site_libs/reactwidget-1.0.0/react-tools.js"></script>
  <script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
  <script src="site_libs/reactable-binding-0.2.3.9000/reactable.js"></script>
  <script>window.xaringanExtraClipboard(null, {"button":"<i class=\"fa fa-clone fa-2x\" style=\"color: #004276\"><\/i>","success":"<i class=\"fa fa-check fa-2x\" style=\"color: #90BE6D\"><\/i>","error":"<i class=\"fa fa-times fa-2x\" style=\"color: #F94144\"><\/i>"})</script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Inconsolata|Source+Sans+Pro|Work+Sans:200,300,400,500|Karla:400,400i,700,700i');
  /********************************/
  /* controls general properties of site*/
  /********************************/

  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    background-color: #FEFEFE;
  }

  d-title h1{
  font-size: 2.1em;
  font-family: 'Work Sans';
  font-weight: 300;
  }

  d-article h1 {
    font-size: 1.8em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 300;
  }

  d-article h2 {
    font-size: 1.65em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 500;
  }

  d-article h3 {
    margin-top: 1em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
  }

  d-byline .authors-affiliations {
    grid-column-end: span 3;
    grid-column-gap: 8px;
  }

  img.orcid-id {
    margin: 0;
  }

  a.name:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  a.affiliation:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  blockquote p {
  	font-size: 1em;
  	font-family: 'Work Sans';
    font-style: normal;
    margin: 0.5em 0 0.5em;
  }

  d-article li {
    margin: 0px 0px 4px;
    font-size: 18px;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
  }

  /********************************/
  /* controls CODE*/
  /********************************/

  /** pre.sourceCode controls code block without lang **/

  /** Controls r code  **/

  pre.sourceCode.r {
    margin-top: 0px;
     color: #333333;
     background-color: #E8E8E8;
     border-radius: 3px;
     font-size: 0.80em;
     font-family: 'Monaco', 'PT Mono', monospace;
     overflow-x: scroll !important;
     border-left: 6px solid #2271B2;
     border-top: 1px solid #B7B7B7;
     border-right: 1px solid #B7B7B7;
     border-bottom: 1px solid #B7B7B7;
     padding: 18px 0px 18px 20px;
   }

   pre.sourceCode.bash {
     margin-top: 0px;
      color: #333333;
      background-color: #E8E8E8;
      border-radius: 3px;
      font-size: 0.80em;
      font-family: 'Monaco', 'PT Mono', monospace;
      overflow-x: scroll !important;
      border-left: 6px solid #B22271;
      border-top: 1px solid #B7B7B7;
      border-right: 1px solid #B7B7B7;
      border-bottom: 1px solid #B7B7B7;
      padding: 18px 0px 18px 20px;
    }
  /** Controls code output **/

  d-article pre {
     margin-top: 0px;
      color: #333333;
      background-color: #FAFAFA;
      font-size: 0.8em;
      border: 1px solid #ccc;
      overflow-x: scroll !important;
      padding:  10px 10px 10px 18px;
      border-radius: 5px;
    }

  /** Controls inline code  **/

  p code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }

  li code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }

  d-article aside code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-family: 'Monaco', 'PT Mono', monospace;
    color: #BF2727;
  }

  /********************************/
  /* controls NAV BAR and HEADERS*/
  /********************************/

  .nav-dropdown-content .nav-dropdown-header {
    text-transform: uppercase;
    color: #686868;
  }

  .nav-dropdown-content a {
    font-family: 'Work Sans'
  }
  .nav-dropdown .nav-dropbtn {
    color: #FEFEFE;
    font-weight: 400;
  }

  .distill-site-header .logo img {
    margin-top: 10px;
    max-height: 30px;
    margin-bottom: auto;
  }

  .distill-site-nav {
      color: #FEFEFE;
      background-color: #113858;
      font-size: 0.95em;
      font-weight: 400;
      min-width: 165px;
      position: absolute;
      font-family: 'Karla';
      letter-spacing: .1em;
      text-transform: uppercase;
  }

  img {
    margin: 5px;
  }

  /** LIGHTBOX MARKUP **/
  .thumbnail {
    max-width: 100%;
    position: relative;
  }

  /** DETAILS fold down**/

  details {
    background-color: #f6f6f6;
    cursor: pointer;
    color: #2d2d2d;
    margin-bottom:1vh;
    padding: 1em;
    border: none;
    text-align: left;
    outline: none;
  }

  .active, details:hover {
    transition: max-height 0.2s ease-out;
    outline: none;

    /* background-color: #acc2c2; */
  }

  details {
    display: flex;
    justify-content: space-between;
    align-content: center;
    border-radius: 4px;
    outline: none;
  }

  details.active {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    outline: none;
  }

  summary::-webkit-details-marker {
    display: none;
    outline: none;
  }

  summary:focus {
    outline: none;
  }

  summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f0fe";
  }

  details[open] summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f146";
  }

  /********************************/
  /* controls downloadthis  */
  /********************************/

  button.btn.btn-default {
    font-size: 0.8em;
    font-family: "PT Sans";
    border-color: #B22271;
    border-width: medium;
    margin-bottom: 15px;
  }

  /********************************/
  /* controls reactable  */
  /********************************/

  div.rt-text-content {
    font-size: 1.2em;
  }

  /********************************/
  /* next and previous buttons at the end of the article */
  /********************************/

  .btnnav {
    border: none;
    background-color: #113858;
    color: #f2f2f2;
    padding: 14px 28px;
    font-size: 16px;
    cursor: pointer;
    display: inline-block;
  }

  .button_round {border-radius: 12px;}

  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"2. Data Set Preparation","description":"Workflow for preparation  of the **16S rRNA** and **ITS** data sets. These steps are needed before analyzing the data. In this workflow, sample groups are defined, and phyloseq objects are created and curated.","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a class="logo" href="index.html">
<img src="assets/icon.png" alt="Logo"/>
</a>
<a href="index.html" class="title"></a>
</div>
<div class="nav-right">
<a href="index.html">
<i class="fa fa-home" aria-hidden="true"></i>
</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
A. Processing
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">Workflows</span>
<hr/>
<a href="dada2.html">1. DADA2</a>
<a href="data-prep.html">2. Data Set Prep</a>
<a href="filtering.html">3. Filtering</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
B. Diversity
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">Workflows</span>
<hr/>
<a href="taxa.html">4. Taxonomic</a>
<a href="alpha.html">5. Alpha Diversity</a>
<a href="beta.html">6. Beta Diversity</a>
<a href="da.html">7. DA ASVs &amp; Taxa</a>
<a href="metadata.html">8. Metadata</a>
</div>
</div>
<a href="data-availability.html">Data</a>
<a href="som.html">SOM</a>
<a href="https://github.com/sweltr/high-temp/">
<i class="fa fa-github fa-lg" aria-hidden="true"></i>
</a>
<a href="about.html">
<i class="fa fa-question" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>2. Data Set Preparation</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->
<p><p>Workflow for preparation of the <strong>16S rRNA</strong> and
<strong>ITS</strong> data sets. These steps are needed before analyzing
the data. In this workflow, sample groups are defined, and phyloseq
objects are created and curated.</p></p>
</div>


<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#workflow-input">Workflow Input</a></li>
<li><a href="#s-rrna-data">16s rRNA Data</a>
<ul>
<li><a href="#prerequisites">Prerequisites</a></li>
<li><a href="#read-counts-assessment">Read Counts Assessment</a></li>
<li><a href="#defining-groups">Defining Groups</a></li>
<li><a href="#create-a-phyloseq-object">Create a Phyloseq
Object</a></li>
<li><a href="#remove-contaminants-unwanted-taxa">Remove Contaminants
&amp; Unwanted Taxa</a></li>
<li><a href="#rename-na-taxonomic-ranks">Rename NA taxonomic
ranks</a></li>
<li><a href="#add-phylogenetic-tree">Add Phylogenetic Tree</a></li>
<li><a href="#data-set-summary">Data Set Summary</a></li>
<li><a href="#review">Review</a></li>
<li><a href="#final-steps">Final Steps</a></li>
</ul></li>
<li><a href="#its-data">ITS Data</a>
<ul>
<li><a href="#prerequisites-1">Prerequisites</a></li>
<li><a href="#read-counts-assessment-1">Read Counts Assessment</a></li>
<li><a href="#defining-groups-1">Defining Groups</a></li>
<li><a href="#create-a-phyloseq-object-1">Create a Phyloseq
Object</a></li>
<li><a href="#remove-low-count-samples">Remove Low-Count
Samples</a></li>
<li><a href="#rename-na-taxonomic-ranks-1">Rename NA taxonomic
ranks</a></li>
<li><a href="#data-set-summary-1">Data Set Summary</a></li>
<li><a href="#review-1">Review</a></li>
<li><a href="#final-steps-1">Final Steps</a></li>
</ul></li>
<li><a href="#workflow-output">Workflow Output</a>
<ul>
<li><a href="#source-code">Source Code</a></li>
<li><a href="#data-availability">Data Availability</a></li>
<li><a href="#last-updated-on">Last updated on</a></li>
</ul></li>
</ul>
</nav>
</div>
<details>
<summary>
<strong>Click here</strong> for setup information.
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">echo =</span> <span class="cn">TRUE</span>, <span class="at">eval =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">119</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#library(conflicted)</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends(ape, local = TRUE)  </span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends_reverse(ape, local = TRUE)  </span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq); <span class="fu">packageVersion</span>(<span class="st">&quot;phyloseq&quot;</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings); <span class="fu">packageVersion</span>(<span class="st">&quot;Biostrings&quot;</span>)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(tidyverse, DT, microbiome, taxa, formatR,</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>               captioner, reactable, downloadthis, fontawesome,</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>               metacoder, ampvis2, ape, statnet.common, </span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">install =</span> <span class="cn">FALSE</span>, <span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">scipen=</span><span class="dv">999</span>)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_current<span class="sc">$</span><span class="fu">get</span>(<span class="fu">c</span>(</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache&quot;</span>,</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.path&quot;</span>,</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.rebuild&quot;</span>,</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;dependson&quot;</span>,</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;autodep&quot;</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>))</span></code></pre></div>
</div>
</details>
<h1 id="workflow-input">Workflow Input</h1>
<blockquote>
<p>You will either need to run the <a href="dada2.html">DADA2
workflow</a> or download the files linked below. See the <a
href="data-availability.html">Data Availability</a> page for complete
details.</p>
</blockquote>
<p>Files needed to run this workflow can be downloaded from figshare.
These are the output files from the <a href="dada2.html">DADA2
workflow</a> page.</p>
<iframe src="https://widgets.figshare.com/articles/14687184/embed?show_title=1" width="568" height="351" allowfullscreen frameborder="0">
</iframe>
<p>These workflows share many common variable names so you must split
the workflow into a script for each data set OR run the command
<code>remove(list = ls())</code> before beginning the next workflow.</p>
<h1 id="s-rrna-data">16s rRNA Data</h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="prerequisites">Prerequisites</h2>
<p>In order to run this workflow, you either need to run the
corresponding <a href="dada2-high-temp.html#s-rrna-data.html">DADA2
Workflow for 16S rRNA</a> or begin with the output from that workflow,
<code>data_prep_ssu18_wf.rdata</code>. See the <a
href="data-availability.html">Data Availability</a> page for complete
details.</p>
<p>Unless otherwise noted, we primarily use <a
href="https://joey711.github.io/phyloseq/">phyloseq</a> <span
class="citation" data-cites="mcmurdie2013phyloseq">(<a
href="#ref-mcmurdie2013phyloseq" role="doc-biblioref">McMurdie and
Holmes 2013</a>)</span> in this section of the workflow to prepare the
2018 16S rRNA data set for community analyses. We prepare the data set
by curating samples, removing contaminants, and creating phyloseq
objects.</p>
<h2 id="read-counts-assessment">Read Counts Assessment</h2>
<p>Before we begin, lets create a summary table containing some basic
sample metadata and the read count data from the <a
href="dada2-high-temp.html#s-rrna-data.html">DADA2 workflow</a>. We need
to inspect how total reads changed through the workflow. While we are at
it, lets create more intuitive Sample IDs. For more details on how
reads changed at each step of the workflow, see the <a
href="dada2-high-temp.html#track-reads-through-workflow">summary
table</a> at the end of the DADA2 section. Table headers are as
follows:</p>
<table>
<colgroup>
<col style="width: 23%" />
<col style="width: 76%" />
</colgroup>
<thead>
<tr class="header">
<th>Header</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>Sample ID</code></td>
<td>the new sample ID based on Plot number, Depth, Treatment,
Temperature, &amp; Pair</td>
</tr>
<tr class="even">
<td><code>Plot</code></td>
<td>the experimental plot number</td>
</tr>
<tr class="odd">
<td><code>Depth</code></td>
<td>the depth where the sample was collected</td>
</tr>
<tr class="even">
<td><code>Treatment</code></td>
<td>warm or control</td>
</tr>
<tr class="odd">
<td><code>Temp</code></td>
<td>temperature of soil heating (0C, +4C, or +8C)</td>
</tr>
<tr class="even">
<td><code>Pair</code></td>
<td>control/treatment coupling</td>
</tr>
<tr class="odd">
<td><code>input</code></td>
<td>number of raw reads</td>
</tr>
<tr class="even">
<td><code>nochim</code></td>
<td>final read count after removing chimeras</td>
</tr>
<tr class="odd">
<td><code>Remain</code></td>
<td>percent of reads remaining from <code>input</code> to
<code>nonchim</code></td>
</tr>
<tr class="even">
<td><code>FastqID</code></td>
<td>base name of the fastq file</td>
</tr>
</tbody>
</table>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><small><strong>(16S rRNA) Table 1 | Sample summary table including
read changes at start and end of DADA2 workflow.</strong></small></p>
<div class="layout-chunk" data-layout="l-body-outset">
<a onclick="fetch(&#39;data:text/csv;base64,U2FtcGxlX0lELHJvb3QgbmFtZSxQbG90LFRyZWF0bWVudCxUZW1wLERlcHRoLFBhaXIsaW5wdXQgcmVhZHMsZmluYWwgcmVhZHMsbm8uIEFTVnMscGVyLiByZWFkcyByZW1haW4KUDAxX0QwMF8wMTBfVzNBLFAxLFAwMSxXYXJtLDMsMDBfMDEwLEEsODU0ODksNjgwMzcsMjM4MywwLjc5NgpQMDFfRDAwXzAxMF9XOEEsUDEtOEMsUDAxLFdhcm0sOCwwMF8wMTAsQSw3NzY0NSw2MDUzNSwxNjE4LDAuNzgKUDEwX0QwMF8wMTBfQzBFLFAxMCxQMTAsQ29udHJvbCwwLDAwXzAxMCxFLDY3NTgzLDUzNjI1LDIwNzAsMC43OTMKUDAyX0QwMF8wMTBfQzBBLFAyLFAwMixDb250cm9sLDAsMDBfMDEwLEEsODcwOTEsNzA0MzYsMjI3OCwwLjgwOQpQMDNfRDAwXzAxMF9XM0IsUDMsUDAzLFdhcm0sMywwMF8wMTAsQiw5MzA2NSw3NTE4MywyMzY4LDAuODA4ClAwM19EMDBfMDEwX1c4QixQMy04QyxQMDMsV2FybSw4LDAwXzAxMCxCLDY2NDAxLDUzNDM3LDE1OTMsMC44MDUKUDA0X0QwMF8wMTBfQzBCLFA0LFAwNCxDb250cm9sLDAsMDBfMDEwLEIsNjc4MTAsNTM4MTMsMjA1MiwwLjc5NApQMDVfRDAwXzAxMF9XM0MsUDUsUDA1LFdhcm0sMywwMF8wMTAsQyw1ODI2Niw0NzA3MCwxNjcyLDAuODA4ClAwNV9EMDBfMDEwX1c4QyxQNS04QyxQMDUsV2FybSw4LDAwXzAxMCxDLDEyMDg5Miw4Njg0MSwyMTU2LDAuNzE4ClAwNl9EMDBfMDEwX0MwQyxQNixQMDYsQ29udHJvbCwwLDAwXzAxMCxDLDEwMTU5OSw4MTIxOSwzMDgyLDAuNzk5ClAwN19EMDBfMDEwX1czRCxQNyxQMDcsV2FybSwzLDAwXzAxMCxELDc2NzgxLDYxNzQyLDIwMjUsMC44MDQKUDA3X0QwMF8wMTBfVzhELFA3LThDLFAwNyxXYXJtLDgsMDBfMDEwLEQsNzkwMzIsNjIyMDIsMjE2NywwLjc4NwpQMDhfRDAwXzAxMF9DMEQsUDgsUDA4LENvbnRyb2wsMCwwMF8wMTAsRCw3NzU5OSw2MjE3MSwyNDUzLDAuODAxClAwOV9EMDBfMDEwX1czRSxQOSxQMDksV2FybSwzLDAwXzAxMCxFLDk1NTY2LDc2Mjk5LDI0MDIsMC43OTgKUDA5X0QwMF8wMTBfVzhFLFA5LThDLFAwOSxXYXJtLDgsMDBfMDEwLEUsOTUxODIsMjUxNTEsODE2LDAuMjY0Cg==&#39;).then(res =&gt; res.blob()).then(blob =&gt; {&#10;      const downloadURL = window.URL.createObjectURL(blob);&#10;      const a = document.createElement(&#39;a&#39;);&#10;      document.body.appendChild(a);&#10;      a.href = downloadURL;&#10;      a.download = &#39;ssu18_sample_summary.csv&#39;; a.click();&#10;      window.URL.revokeObjectURL(downloadURL);&#10;      document.body.removeChild(a);&#10;    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
<div id="htmlwidget-b85ebd9764e80c0724f5" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-b85ebd9764e80c0724f5">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Sample_ID":["P01_D00_010_W3A","P01_D00_010_W8A","P10_D00_010_C0E","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E"],"root name":["P1","P1-8C","P10","P2","P3","P3-8C","P4","P5","P5-8C","P6","P7","P7-8C","P8","P9","P9-8C"],"Plot":["P01","P01","P10","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09"],"Treatment":["Warm","Warm","Control","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm"],"Temp":[3,8,0,0,3,8,0,3,8,0,3,8,0,3,8],"Depth":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"Pair":["A","A","E","A","B","B","B","C","C","C","D","D","D","E","E"],"input reads":[85489,77645,67583,87091,93065,66401,67810,58266,120892,101599,76781,79032,77599,95566,95182],"final reads":[68037,60535,53625,70436,75183,53437,53813,47070,86841,81219,61742,62202,62171,76299,25151],"no. ASVs":[2383,1618,2070,2278,2368,1593,2052,1672,2156,3082,2025,2167,2453,2402,816],"per. reads remain":[0.796,0.78,0.793,0.809,0.808,0.805,0.794,0.808,0.718,0.799,0.804,0.787,0.801,0.798,0.264]},"columns":[{"accessor":"Sample_ID","name":"Sample ID","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P10_D00_010_C0E","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E"],"header":"Sample ID","align":"left","footerStyle":{"fontWeight":"bold"},"footer":"Total reads","minWidth":150,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"root name","name":"root name","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P1","P1-8C","P10","P2","P3","P3-8C","P4","P5","P5-8C","P6","P7","P7-8C","P8","P9","P9-8C"],"header":"root name","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Plot","name":"Plot","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01","P01","P10","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09"],"header":"Plot","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Treatment","name":"Treatment","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["Warm","Warm","Control","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm"],"header":"Treatment","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Temp","name":"Temp","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["3","8","0","0","3","8","0","3","8","0","3","8","0","3","8"],"header":"Temp","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Depth","name":"Depth","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"header":"Depth","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Pair","name":"Pair","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["A","A","E","A","B","B","B","C","C","C","D","D","D","E","E"],"header":"Pair","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"input reads","name":"input reads","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["85489","77645","67583","87091","93065","66401","67810","58266","120892","101599","76781","79032","77599","95566","95182"],"header":"input reads","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"final reads","name":"final reads","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["68037","60535","53625","70436","75183","53437","53813","47070","86841","81219","61742","62202","62171","76299","25151"],"header":"final reads","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"no. ASVs","name":"no. ASVs","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["2383","1618","2070","2278","2368","1593","2052","1672","2156","3082","2025","2167","2453","2402","816"],"header":"no. ASVs","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"per. reads remain","name":"per. reads remain","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.796","0.78","0.793","0.809","0.808","0.805","0.794","0.808","0.718","0.799","0.804","0.787","0.801","0.798","0.264"],"header":"per. reads remain","align":"center","footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"4336b1e377481b6e4af6b56079537876"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="defining-groups">Defining Groups</h2>
<ol type="1">
<li><p>Load the data packet produced in the final step of the DADA2
workflow. This packet (<code>ssu18_dada2_wf.rdata</code>) contains the
ASV-by-sample table and the ASV taxonomy table.</p></li>
<li><p>Rename the samples so names have plot and Depth info.</p></li>
<li><p>After we load the data packet, we next need to format sample
names and define groups.</p></li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> seq_table[<span class="fu">order</span>(seq_table<span class="sc">$</span><span class="st">`</span><span class="at">root name</span><span class="st">`</span>), ]</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>tmp_names <span class="ot">&lt;-</span> seq_table<span class="sc">$</span><span class="st">`</span><span class="at">Sample_ID</span><span class="st">`</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;files/dada2/rdata/ssu18_dada2_wf.rdata&quot;</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(seq_table<span class="sc">$</span><span class="st">`</span><span class="at">root name</span><span class="st">`</span>, <span class="fu">rownames</span>(seqtab))</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(seqtab) <span class="ot">&lt;-</span> tmp_names</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>samples.out <span class="ot">&lt;-</span> <span class="fu">rownames</span>(seqtab)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>sample_name <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">1</span>, <span class="dv">999</span>)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>plot <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">0</span>, <span class="dv">3</span>)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>depth <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">6</span>, <span class="dv">11</span>)</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>treatment <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">13</span>, <span class="dv">13</span>)</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">14</span>, <span class="dv">14</span>)</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>pair <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">15</span>, <span class="dv">15</span>)</span></code></pre></div>
</details>
</div>
<p>We have a total of <strong>15</strong> samples, <strong>5</strong>
sample pairs, from <strong>10</strong> plots (<strong>1</strong> depth).
There are <strong>2</strong> treatments corresponding to
<strong>3</strong> different temperature regimes.</p>
<ol start="4" type="1">
<li>And finally we define a sample data frame that holds the different
groups we extracted from the sample names.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">#define a sample data frame</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>samdf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">SamName =</span> sample_name,</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">PLOT =</span> plot,</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">DEPTH =</span> depth,</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">TREAT =</span> treatment,</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">TEMP =</span> temp,</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">PAIR =</span> pair)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(samdf) <span class="ot">&lt;-</span> samples.out</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>For example, P07_D00_010_W3D is the sample heated at
<strong>+3C</strong> from plot <strong>P07</strong>, collected from a
depth of 00_010cm. This sample is in group <strong>D</strong> which
consists of P07_D00_010_W3D, P07_D00_010_W8D, P08_D00_010_C0D.</p>
<h2 id="create-a-phyloseq-object">Create a Phyloseq Object</h2>
<p><strong>A</strong>. The first step is to rename the amplicon sequence
variants (ASVs) so the designations are a bit more user friendly. By
default, DADA2 names each ASV by its unique sequence so that data can be
directly compared across studies (which is great). But this convention
can get cumbersome downstream, so we rename the ASVs using a simpler
conventionASV1, ASV2, ASV3, and so on.</p>
<aside>
A phyloseq object contains ASV table (taxa abundances), sample metadata,
and taxonomy table (mapping between ASVs and higher-level taxonomic
classifications).
</aside>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># this create the phyloseq object</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(<span class="fu">otu_table</span>(seqtab, <span class="at">taxa_are_rows =</span> <span class="cn">FALSE</span>),</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">sample_data</span>(samdf), <span class="fu">tax_table</span>(tax_silva))</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">tax_table</span>(ps),</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">rownames</span>(<span class="fu">tax_table</span>(ps)))</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co"># adding unique ASV names</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="fu">taxa_names</span>(ps) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;ASV&quot;</span>, <span class="fu">seq</span>(<span class="fu">ntaxa</span>(ps)))</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">tax_table</span>(ps),</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">rownames</span>(<span class="fu">tax_table</span>(ps)))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;ASV1&quot; &quot;ASV2&quot; &quot;ASV3&quot; &quot;ASV4&quot; &quot;ASV5&quot; &quot;ASV6&quot;</code></pre>
</div>
<p>So the complete data set contains 20332 ASVs. We can use the <a
href="https://github.com/microbiome/microbiome/">microbiome R
package</a> <span class="citation" data-cites="lahti2017microbiome">(<a
href="#ref-lahti2017microbiome" role="doc-biblioref">Lahti, Sudarshan,
et al. 2017</a>)</span> to get some additional summary data from the
phyloseq object.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<table>
<colgroup>
<col style="width: 40%" />
<col style="width: 59%" />
</colgroup>
<thead>
<tr class="header">
<th>Metric</th>
<th>Results</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>25151</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>86841</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>937761</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>62517</td>
</tr>
<tr class="odd">
<td>Median number of reads</td>
<td>62171</td>
</tr>
<tr class="even">
<td>Total ASVs</td>
<td>20332</td>
</tr>
<tr class="odd">
<td>Sparsity</td>
<td>0.898</td>
</tr>
<tr class="even">
<td>Any ASVs sum to 1 or less?</td>
<td>FALSE</td>
</tr>
<tr class="odd">
<td>Number of singleton ASVs</td>
<td>NA</td>
</tr>
<tr class="even">
<td>Percent of ASVs that are singletons</td>
<td>NA</td>
</tr>
<tr class="odd">
<td>Number of sample variables are:</td>
<td>6 (SamName, PLOT, DEPTH, TREAT, TEMP, PAIR)</td>
</tr>
</tbody>
</table>
<p><strong>B</strong>. Add two final columns containing the ASV
sequences and ASV IDs. This will be useful later when trying to export a
fasta file. We can also take a look at the phyloseq object.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(<span class="fu">tax_table</span>(ps)) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Kingdom&quot;</span>, <span class="st">&quot;Phylum&quot;</span>, <span class="st">&quot;Class&quot;</span>, <span class="st">&quot;Order&quot;</span>,</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Genus&quot;</span>, <span class="st">&quot;ASV_SEQ&quot;</span>, <span class="st">&quot;ASV_ID&quot;</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>ps</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20332 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20332 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<p><strong>C</strong>. Export sequence and taxonomy tables for the
unadulterated phyloseq object for later use. We will use the prefix
<code>full</code> to indicate that these are the <em>raw</em>
outputs.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">tax_table</span>(ps),</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;files/data-prep/tables/ssu18_full_tax_table.txt&quot;</span>,</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">NA</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ps)),</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;files/data-prep/tables/ssu18_full_seq_table.txt&quot;</span>,</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">NA</span>)</span></code></pre></div>
</div>
<h2 id="remove-contaminants-unwanted-taxa">Remove Contaminants &amp;
Unwanted Taxa</h2>
<p>Lets see if we have any potential contaminants. We can use some <a
href="https://rmarkdown.rstudio.com/lesson-4.html">inline R code</a> to
see the taxonomy table for any taxa of interest.</p>
<aside>
The code (hidden by default) is written as
<code>`r "Chloroplast" %in% tax_table(ps)`</code>.
</aside>
<ul>
<li>Are Mitochondria present? TRUE</li>
<li>Are Chloroplast present? TRUE</li>
<li>Are Eukaryota present? TRUE</li>
</ul>
<p>Lets remove these taxaEukaryota because we used bacterial/archaeal
primers, Mitochondria because those are likely from eukaryotes, and
Chloroplast because those are likely from plants. We must do each of
these in turn using phyloseq and it gets a little messy.</p>
<p>Why messy? The <code>subset_taxa</code> command removes anything that
is <code>NA</code> for the specified taxonomic level or above. For
example, lets say you run the <code>subset_taxa</code> command using
<code>Family != "Mitochondria</code>. Seems like you should get a
phyloseq object with everything except Mitochondria. But actually the
command not only gets rid of Mitochondria but everything else that has
<code>NA</code> for Family and above. In my experience this is not well
documented and I had to dig through the files to figure out what was
happening.</p>
<p>Anyway, to remove the taxa we do the following:</p>
<ul>
<li>Subset the taxa and generate a <code>ps</code> object of just the
taxa of interest,</li>
<li>Select the ASV column only, turn it into a factor, and use this to
remove <INSERT TAXA> from the <code>ps</code> object.</li>
</ul>
<h3 id="remove-mitochondria-asvs">Remove Mitochondria ASVs</h3>
<p>Remember the original data set contained 20332 ASVs. Here we generate
a file with mitochondria ASVs only.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>MT1 <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(ps, Family <span class="sc">==</span> <span class="st">&quot;Mitochondria&quot;</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>MT1 <span class="ot">&lt;-</span>  <span class="fu">as</span>(<span class="fu">tax_table</span>(MT1), <span class="st">&quot;matrix&quot;</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>MT1 <span class="ot">&lt;-</span> MT1[, <span class="dv">8</span>]</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>MT1df <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(MT1)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>goodTaxa <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">taxa_names</span>(ps), MT1df)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>ps_no_mito <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(goodTaxa, ps)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>ps_no_mito</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20291 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20291 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<p>Looks like this removed <strong>41 Mitochondria ASVs</strong>. We
will duplicate the code block to remove other groups.</p>
<h3 id="remove-chloroplast-asvs">Remove Chloroplast ASVs</h3>
<p>And again with Chloroplast ASVs only.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>CH1 <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(ps_no_mito, Order <span class="sc">==</span> <span class="st">&quot;Chloroplast&quot;</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>CH1 <span class="ot">&lt;-</span>  <span class="fu">as</span>(<span class="fu">tax_table</span>(CH1), <span class="st">&quot;matrix&quot;</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>CH1 <span class="ot">&lt;-</span> CH1[, <span class="dv">8</span>]</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>CH1df <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(CH1)</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>goodTaxa <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">taxa_names</span>(ps_no_mito), CH1df)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>ps_no_chloro <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(goodTaxa, ps_no_mito)</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>ps_no_chloro</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20274 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20274 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<p>The code removed an additional <strong>17 Chloroplast
ASVs</strong>.</p>
<h3 id="remove-eukaryota-asvs">Remove Eukaryota ASVs</h3>
<p>And again with Eukaryota ASVs only.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>EU1 <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(ps_no_chloro, Kingdom <span class="sc">==</span> <span class="st">&quot;Eukaryota&quot;</span>)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>EU1 <span class="ot">&lt;-</span>  <span class="fu">as</span>(<span class="fu">tax_table</span>(EU1), <span class="st">&quot;matrix&quot;</span>)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>EU1 <span class="ot">&lt;-</span> EU1[, <span class="dv">8</span>]</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>EU1df <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(EU1)</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>goodTaxa <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">taxa_names</span>(ps_no_chloro), EU1df)</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>ps_no_euk <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(goodTaxa, ps_no_chloro)</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>ps_no_euk</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20223 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20223 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<p>The code removed an additional <strong>51 Eukaryota ASVs</strong>
from the <code>ps</code> object.</p>
<h3 id="remove-any-kingdom-nas">Remove any Kingdom NAs</h3>
<p>Here we can just use the straight up <code>subset_taxa</code> command
since we do not need to worry about any ranks above Kingdom also being
removed.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>ps_filt <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(ps_no_euk, <span class="sc">!</span><span class="fu">is.na</span>(Kingdom))</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>ps_work_o <span class="ot">&lt;-</span> ps_filt</span></code></pre></div>
</div>
<p>The code eliminated an additional <strong>50 Kingdom level NA
ASVs</strong> from the phyloseq object.</p>
<h2 id="rename-na-taxonomic-ranks">Rename NA taxonomic ranks</h2>
<p>Phyloseq has an odd way of dealing with taxonomic ranks that have no
valuein other words, <strong>NA</strong> in the tax table. The first
thing we are going to do before moving forward is to change all of the
<em>NA</em>s to have a value of the next highest classified rank. For
example, <code>ASV26</code> is not classified at the Genus level but is
at Family level (Xanthobacteraceae). So we change the Genus name to
<em>Family_Xanthobacteraceae</em>. The code for comes from these two
posts on the phyloseq GitHub, both by <a
href="https://github.com/MSMortensen">MSMortensen</a>: issue <a
href="https://github.com/joey711/phyloseq/issues/850#issuecomment-394771087">#850</a>
and issue <a
href="https://github.com/joey711/phyloseq/issues/990#issuecomment-424618425">#990</a>.</p>
<blockquote>
<p>One thing this code does is reassign the functions <code>class</code>
and <code>order</code> to taxonomic ranks. This can cause issues if you
need these functions.</p>
</blockquote>
<p>So you need to run something like this
<code>rm(class, order, phylum, kingdom)</code> at the end of the code to
remove these as variables. For now, I have not come up with a better
solution.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>tax.clean <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ps_filt))</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>){ tax.clean[,i] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tax.clean[,i])}</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>tax.clean[<span class="fu">is.na</span>(tax.clean)] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(tax.clean)){</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (tax.clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>        kingdom <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;k_&quot;</span>, tax.clean[i,<span class="dv">1</span>], <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> kingdom</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>        phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;p_&quot;</span>, tax.clean[i,<span class="dv">2</span>], <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">3</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">4</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>        class <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;c_&quot;</span>, tax.clean[i,<span class="dv">3</span>], <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">4</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> class</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">5</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>        order <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;o_&quot;</span>, tax.clean[i,<span class="dv">4</span>], <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">5</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> order</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">6</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>        tax.clean<span class="sc">$</span>Genus[i] <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;f&quot;</span>,tax.clean<span class="sc">$</span>Family[i], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps_filt) <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tax.clean)</span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a><span class="fu">rank_names</span>(ps_filt)</span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(class, order, phylum, kingdom)</span></code></pre></div>
</details>
</div>
<p>Still the same ranks. Thats good. What about the new groups? Lets
take a peak at some families.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_filt, <span class="st">&quot;Family&quot;</span>), <span class="dv">16</span>)</span></code></pre></div>
<pre><code> [1] &quot;Planococcaceae&quot;         &quot;Chthoniobacteraceae&quot;   
 [3] &quot;Nitrososphaeraceae&quot;     &quot;p_RCP2-54&quot;             
 [5] &quot;o_Rokubacteriales&quot;      &quot;Methyloligellaceae&quot;    
 [7] &quot;Nitrospiraceae&quot;         &quot;Xanthobacteraceae&quot;     
 [9] &quot;o_Vicinamibacterales&quot;   &quot;Vicinamibacteraceae&quot;   
[11] &quot;c_Subgroup_5&quot;           &quot;Xiphinematobacteraceae&quot;
[13] &quot;Entotheonellaceae&quot;      &quot;Chitinophagaceae&quot;      
[15] &quot;Nitrosomonadaceae&quot;      &quot;o_Acidobacteriales&quot;    </code></pre>
</div>
<p>Nice. Bye-bye <em>NA</em></p>
<blockquote>
<p><em>Note</em>. The original code mentioned above is written for data
sets that have species-level designations.</p>
</blockquote>
<p>Since this data set does not contain species, I modified the code to
stop at the genus level. If your data set has species, my modifications
will not work for you.</p>
<p>Finally, we rename the ps object. This is now our working data
set.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>ps_work <span class="ot">&lt;-</span> ps_filt</span></code></pre></div>
</div>
<h2 id="add-phylogenetic-tree">Add Phylogenetic Tree</h2>
<p>One final task is to add a phylogenetic tree to the phyloseq
object.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>ps_work_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(ps_work), <span class="at">rooted =</span> <span class="cn">TRUE</span>,</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(ps_work))</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>ps_work <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(ps_work,</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>                          sample_data,</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>                          ps_work_tree)</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>ps_work_o_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(ps_work_o), <span class="at">rooted =</span> <span class="cn">TRUE</span>,</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(ps_work_o))</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>ps_work_o <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(ps_work_o,</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>                          sample_data,</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>                          ps_work_o_tree)</span></code></pre></div>
</div>
<h2 id="data-set-summary">Data Set Summary</h2>
<p>Now that we have removed the NC and sample with zero reads, we can
summarize the data set. Lets start with the in our working phyloseq
object using the <code>summarize_phyloseq</code> from the <a
href="https://github.com/microbiome/microbiome/">microbiome R
package</a> <span class="citation" data-cites="lahti2017microbiome">(<a
href="#ref-lahti2017microbiome" role="doc-biblioref">Lahti, Sudarshan,
et al. 2017</a>)</span> as we did above.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<table>
<colgroup>
<col style="width: 38%" />
<col style="width: 61%" />
</colgroup>
<thead>
<tr class="header">
<th>Metric</th>
<th>Results</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>25088</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>86600</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>936640</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>62443</td>
</tr>
<tr class="odd">
<td>Median number of reads</td>
<td>62117</td>
</tr>
<tr class="even">
<td>Total ASVs</td>
<td>20173</td>
</tr>
<tr class="odd">
<td>Sparsity</td>
<td>0.891</td>
</tr>
<tr class="even">
<td>Any ASVs sum to 1 or less?</td>
<td>FALSE</td>
</tr>
<tr class="odd">
<td>Number of singleton ASVs</td>
<td>NA</td>
</tr>
<tr class="even">
<td>Percent of ASVs that are singletons</td>
<td>NA</td>
</tr>
<tr class="odd">
<td>Number of sample variables are:</td>
<td>6 (SamName, PLOT, DEPTH, TREAT, TEMP, PAIR)</td>
</tr>
</tbody>
</table>
<p>We can also generate a summary table of total reads &amp; ASVs for
each sample. You can sort the table, download a copy, or filter by
search term. Here is the code to generate the data for the table. First,
we create data frames that hold total reads and ASVs for each
sample.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> <span class="fu">sample_sums</span>(ps_work)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(total_reads, <span class="at">make.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> total_reads <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;Sample_ID&quot;</span>)</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>total_asvs <span class="ot">&lt;-</span> <span class="fu">estimate_richness</span>(ps_work,</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>                                <span class="at">measures =</span> <span class="st">&quot;Observed&quot;</span>)</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>total_asvs <span class="ot">&lt;-</span> total_asvs <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;Sample_ID&quot;</span>)</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>total_asvs<span class="sc">$</span>Sample_ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>, <span class="st">&#39;-&#39;</span>, total_asvs<span class="sc">$</span>Sample_ID)</span></code></pre></div>
</details>
</div>
<p>And then we merge these two data frames with the sample data
frame.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>sam_details <span class="ot">&lt;-</span> samdf</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sam_details) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(sam_details) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Sample_ID&quot;</span>, <span class="st">&quot;Plot&quot;</span>, <span class="st">&quot;Depth&quot;</span>,</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;Treatment&quot;</span>, <span class="st">&quot;Temp&quot;</span>, <span class="st">&quot;Pair&quot;</span>)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>tmp_tab1 <span class="ot">&lt;-</span> seq_table</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>tmp_tab1[, <span class="fu">c</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">7</span>,<span class="dv">11</span>)] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>tmp_tab2 <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(sam_details, total_reads) <span class="sc">%&gt;%</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>                <span class="fu">left_join</span>(., total_asvs) <span class="sc">%&gt;%</span></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>                <span class="fu">left_join</span>(., tmp_tab1)</span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>reads_lost <span class="ot">&lt;-</span> tmp_tab2<span class="sc">$</span><span class="st">`</span><span class="at">final reads</span><span class="st">`</span> <span class="sc">-</span> tmp_tab2<span class="sc">$</span>total_reads </span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>asvs_lost <span class="ot">&lt;-</span> tmp_tab2<span class="sc">$</span><span class="st">`</span><span class="at">no. ASVs</span><span class="st">`</span> <span class="sc">-</span> tmp_tab2<span class="sc">$</span>Observed </span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>tmp_tab2<span class="sc">$</span>reads_lost <span class="ot">&lt;-</span> reads_lost</span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>tmp_tab2<span class="sc">$</span>asvs_lost <span class="ot">&lt;-</span> asvs_lost</span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>seq_table2 <span class="ot">&lt;-</span> tmp_tab2[, <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>,<span class="dv">12</span>,<span class="dv">13</span>)]</span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(seq_table2) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;sample_id&quot;</span>, <span class="st">&quot;PLOT&quot;</span>, <span class="st">&quot;DEPTH&quot;</span>,</span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a>                          <span class="st">&quot;TREAT&quot;</span>, <span class="st">&quot;TEMP&quot;</span>, <span class="st">&quot;PAIR&quot;</span>, <span class="st">&quot;total_reads&quot;</span>,</span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a>                          <span class="st">&quot;total_ASVs&quot;</span>, <span class="st">&quot;reads_lost&quot;</span>, <span class="st">&quot;ASVs_lost&quot;</span>)</span></code></pre></div>
</details>
</div>
<p><small><strong>(16S rRNA) Table 2 | Sample summary table showing the
final number of reads and ASVs (per sample) as well as the number of
reads/ASVs lost during curation.</strong></small></p>
<div class="layout-chunk" data-layout="l-body-outset">
<a onclick="fetch(&#39;data:text/csv;base64,c2FtcGxlX2lkLFBMT1QsREVQVEgsVFJFQVQsVEVNUCxQQUlSLHRvdGFsX3JlYWRzLHRvdGFsX0FTVnMscmVhZHNfbG9zdCxBU1ZzX2xvc3QKUDAxX0QwMF8wMTBfVzNBLFAwMSwwMF8wMTAsVywzLEEsNjc5MDIsMjM3MSwxMzUsMTIKUDAxX0QwMF8wMTBfVzhBLFAwMSwwMF8wMTAsVyw4LEEsNjA1MDksMTYxMCwyNiw4ClAxMF9EMDBfMDEwX0MwRSxQMTAsMDBfMDEwLEMsMCxFLDUzNjAwLDIwNjYsMjUsNApQMDJfRDAwXzAxMF9DMEEsUDAyLDAwXzAxMCxDLDAsQSw3MDQxMiwyMjcyLDI0LDYKUDAzX0QwMF8wMTBfVzNCLFAwMywwMF8wMTAsVywzLEIsNzUxMjMsMjM1Niw2MCwxMgpQMDNfRDAwXzAxMF9XOEIsUDAzLDAwXzAxMCxXLDgsQiw1MzM0NywxNTc1LDkwLDE4ClAwNF9EMDBfMDEwX0MwQixQMDQsMDBfMDEwLEMsMCxCLDUzNzcwLDIwNDAsNDMsMTIKUDA1X0QwMF8wMTBfVzNDLFAwNSwwMF8wMTAsVywzLEMsNDcwMjMsMTY2Myw0Nyw5ClAwNV9EMDBfMDEwX1c4QyxQMDUsMDBfMDEwLFcsOCxDLDg2NjAwLDIxMjQsMjQxLDMyClAwNl9EMDBfMDEwX0MwQyxQMDYsMDBfMDEwLEMsMCxDLDgxMTUwLDMwNjUsNjksMTcKUDA3X0QwMF8wMTBfVzNELFAwNywwMF8wMTAsVywzLEQsNjE2MzIsMjAxMCwxMTAsMTUKUDA3X0QwMF8wMTBfVzhELFAwNywwMF8wMTAsVyw4LEQsNjIxMTcsMjE1Nyw4NSwxMApQMDhfRDAwXzAxMF9DMEQsUDA4LDAwXzAxMCxDLDAsRCw2MjEzNiwyNDQxLDM1LDEyClAwOV9EMDBfMDEwX1czRSxQMDksMDBfMDEwLFcsMyxFLDc2MjMxLDIzODYsNjgsMTYKUDA5X0QwMF8wMTBfVzhFLFAwOSwwMF8wMTAsVyw4LEUsMjUwODgsODEzLDYzLDMK&#39;).then(res =&gt; res.blob()).then(blob =&gt; {&#10;      const downloadURL = window.URL.createObjectURL(blob);&#10;      const a = document.createElement(&#39;a&#39;);&#10;      document.body.appendChild(a);&#10;      a.href = downloadURL;&#10;      a.download = &#39;ssu18_sample_summary_final.csv&#39;; a.click();&#10;      window.URL.revokeObjectURL(downloadURL);&#10;      document.body.removeChild(a);&#10;    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
<div id="htmlwidget-58e4fdc74beb9d94e52c" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-58e4fdc74beb9d94e52c">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"sample_id":["P01_D00_010_W3A","P01_D00_010_W8A","P10_D00_010_C0E","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E"],"PLOT":["P01","P01","P10","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09"],"DEPTH":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"TREAT":["W","W","C","C","W","W","C","W","W","C","W","W","C","W","W"],"TEMP":["3","8","0","0","3","8","0","3","8","0","3","8","0","3","8"],"PAIR":["A","A","E","A","B","B","B","C","C","C","D","D","D","E","E"],"total_reads":[67902,60509,53600,70412,75123,53347,53770,47023,86600,81150,61632,62117,62136,76231,25088],"total_ASVs":[2371,1610,2066,2272,2356,1575,2040,1663,2124,3065,2010,2157,2441,2386,813],"reads_lost":[135,26,25,24,60,90,43,47,241,69,110,85,35,68,63],"ASVs_lost":[12,8,4,6,12,18,12,9,32,17,15,10,12,16,3]},"columns":[{"accessor":"sample_id","name":"sample_id","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P10_D00_010_C0E","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E"],"header":"sample id","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"PLOT","name":"PLOT","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01","P01","P10","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09"],"header":"PLOT","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"DEPTH","name":"DEPTH","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"header":"DEPTH","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"TREAT","name":"TREAT","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["W","W","C","C","W","W","C","W","W","C","W","W","C","W","W"],"header":"TREAT","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"TEMP","name":"TEMP","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["3","8","0","0","3","8","0","3","8","0","3","8","0","3","8"],"header":"TEMP","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"PAIR","name":"PAIR","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["A","A","E","A","B","B","B","C","C","C","D","D","D","E","E"],"header":"PAIR","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"total_reads","name":"total_reads","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["67902","60509","53600","70412","75123","53347","53770","47023","86600","81150","61632","62117","62136","76231","25088"],"header":"total reads","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"total_ASVs","name":"total_ASVs","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["2371","1610","2066","2272","2356","1575","2040","1663","2124","3065","2010","2157","2441","2386","813"],"header":"total ASVs","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"reads_lost","name":"reads_lost","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["135","26","25","24","60","90","43","47","241","69","110","85","35","68","63"],"header":"reads lost","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"ASVs_lost","name":"ASVs_lost","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["12","8","4","6","12","18","12","9","32","17","15","10","12","16","3"],"header":"ASVs lost","align":"center","footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"551a45caf403b7a1a1de2911a76de0c1"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
<h2 id="review">Review</h2>
<p>We have a few different options to play with. At this point it is
difficult to say which we will use or whether additional objects need to
be created. Here is a summary of the objects we have.</p>
<p>The full phyloseq object <em>before</em> removing contaminants.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>ps</span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20332 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20332 taxa by 8 taxonomic ranks ]</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps, <span class="st">&quot;Family&quot;</span>), <span class="dv">16</span>)</span></code></pre></div>
<pre><code> [1] &quot;Planococcaceae&quot;         &quot;Chthoniobacteraceae&quot;   
 [3] &quot;Nitrososphaeraceae&quot;     NA                      
 [5] &quot;Methyloligellaceae&quot;     &quot;Nitrospiraceae&quot;        
 [7] &quot;Xanthobacteraceae&quot;      &quot;Vicinamibacteraceae&quot;   
 [9] &quot;Xiphinematobacteraceae&quot; &quot;Entotheonellaceae&quot;     
[11] &quot;Chitinophagaceae&quot;       &quot;Nitrosomonadaceae&quot;     
[13] &quot;B1-7BS&quot;                 &quot;WX65&quot;                  
[15] &quot;Gaiellaceae&quot;            &quot;Steroidobacteraceae&quot;   </code></pre>
</div>
<p>The phyloseq object <em>before</em> changing NA ranks.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>ps_work_o</span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20173 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20173 taxa by 8 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 20173 tips and 20172 internal nodes ]</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_work_o, <span class="st">&quot;Family&quot;</span>), <span class="dv">16</span>)</span></code></pre></div>
<pre><code> [1] &quot;A4b&quot;                            &quot;Burkholderiaceae&quot;              
 [3] &quot;Peptococcaceae&quot;                 NA                              
 [5] &quot;Polyangiaceae&quot;                  &quot;Pedosphaeraceae&quot;               
 [7] &quot;AKYH767&quot;                        &quot;Micromonosporaceae&quot;            
 [9] &quot;Acidobacteriaceae_(Subgroup_1)&quot; &quot;Pyrinomonadaceae&quot;              
[11] &quot;Clostridiaceae&quot;                 &quot;Myxococcaceae&quot;                 
[13] &quot;Hyphomonadaceae&quot;                &quot;Roseiflexaceae&quot;                
[15] &quot;type_III&quot;                       &quot;Ktedonobacteraceae&quot;            </code></pre>
</div>
<p>The phyloseq object <em>after</em> changing NA ranks.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>ps_work</span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20173 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20173 taxa by 8 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 20173 tips and 20172 internal nodes ]</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_work, <span class="st">&quot;Family&quot;</span>), <span class="dv">16</span>)</span></code></pre></div>
<pre><code> [1] &quot;c_Subgroup_22&quot;         &quot;c_OM190&quot;              
 [3] &quot;A4b&quot;                   &quot;Diplorickettsiaceae&quot;  
 [5] &quot;k_Bacteria&quot;            &quot;Gemmataceae&quot;          
 [7] &quot;c_Gammaproteobacteria&quot; &quot;Microscillaceae&quot;      
 [9] &quot;Entotheonellaceae&quot;     &quot;Syntrophaceae&quot;        
[11] &quot;p_WS2&quot;                 &quot;Verrucomicrobiaceae&quot;  
[13] &quot;Pyrinomonadaceae&quot;      &quot;Caulobacteraceae&quot;     
[15] &quot;Parachlamydiaceae&quot;     &quot;c_Phycisphaerae&quot;      </code></pre>
</div>
<h2 id="final-steps">Final Steps</h2>
<p>The last thing tasks to complete are to <strong>a</strong>) save
copies of the taxonomy and sequence tables and <strong>b</strong>) save
an image of the workflow for the next section.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">tax_table</span>(ps_work),</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;files/data-prep/tables/ssu18_work_tax_table.txt&quot;</span>,</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">NA</span>)</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ps_work)),</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;files/data-prep/tables/ssu18_work_seq_table.txt&quot;</span>,</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">NA</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="its-data">ITS Data</h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="prerequisites-1">Prerequisites</h2>
<p>In order to run this workflow, you either need to run the
corresponding <a href="dada2-high-temp.html#its-data">DADA2 Workflow for
2018 ITS</a> or begin with the output from that workflow,
<code>data_prep_its18_wf.rdata</code>. See the <a
href="data-availability.html">Data Availability</a> page for complete
details.</p>
<p>Unless otherwise noted, we primarily use <a
href="https://joey711.github.io/phyloseq/">phyloseq</a> <span
class="citation" data-cites="mcmurdie2013phyloseq">(<a
href="#ref-mcmurdie2013phyloseq" role="doc-biblioref">McMurdie and
Holmes 2013</a>)</span> in this section of the workflow to prepare the
2018 ITS data set for community analyses. We prepare the data set by
curating samples, removing contaminants, and creating phyloseq
objects.</p>
<h2 id="read-counts-assessment-1">Read Counts Assessment</h2>
<p>Before we begin, lets create a summary table containing some basic
sample metadata and the read count data from the <a
href="dada2-high-temp.html#its-data">DADA2 workflow</a>. We need to
inspect how total reads changed from to beginning to the end of the
workflow. While we are at it, lets create more intuitive Sample IDs.
For more details on how reads changed at each step of the workflow, see
the <a
href="dada2-high-temp.html#track-reads-through-workflow-1">summary
table</a> at the end of the DADA2 section.</p>
<p>Table headers are as follows:</p>
<table>
<colgroup>
<col style="width: 23%" />
<col style="width: 76%" />
</colgroup>
<thead>
<tr class="header">
<th>Header</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>Sample ID</code></td>
<td>the new sample ID based on Plot number, Depth, Treatment,
Temperature, &amp; Pair</td>
</tr>
<tr class="even">
<td><code>Plot</code></td>
<td>the experimental plot number</td>
</tr>
<tr class="odd">
<td><code>Depth</code></td>
<td>the depth where the sample was collected</td>
</tr>
<tr class="even">
<td><code>Treatment</code></td>
<td>warm or control</td>
</tr>
<tr class="odd">
<td><code>Temp</code></td>
<td>temperature of soil heating (0C, +4C, or +8C)</td>
</tr>
<tr class="even">
<td><code>Pair</code></td>
<td>control/treatment coupling</td>
</tr>
<tr class="odd">
<td><code>input</code></td>
<td>number of raw reads</td>
</tr>
<tr class="even">
<td><code>nochim</code></td>
<td>final read count after removing chimeras</td>
</tr>
<tr class="odd">
<td><code>Remain</code></td>
<td>percent of reads remaining from <code>input</code> to
<code>nonchim</code></td>
</tr>
<tr class="even">
<td><code>FastqID</code></td>
<td>base name of the fastq file</td>
</tr>
</tbody>
</table>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><small><strong>(ITS) Table 1 | Sample summary table including read
changes at start and end of DADA2 workflow.</strong></small></p>
<div class="layout-chunk" data-layout="l-body-outset">
<a onclick="fetch(&#39;data:text/csv;base64,U2FtcGxlX0lELHJvb3QgbmFtZSxQbG90LFRyZWF0bWVudCxUZW1wLERlcHRoLFBhaXIsaW5wdXQgcmVhZHMsZmluYWwgcmVhZHMsbm8uIEFTVnMscGVyLiByZWFkcyByZW1haW4KUDAxX0QwMF8wMTBfVzNBLFAxLFAwMSxXYXJtLDMsMDBfMDEwLEEsNTA4OTQsMjk3OTQsOTQ1LDAuNTg1ClAwMV9EMDBfMDEwX1c4QSxQMS04QyxQMDEsV2FybSw4LDAwXzAxMCxBLDcwNzM0LDQxODgwLDcxOSwwLjU5MgpQMTBfRDAwXzAxMF9DMEUsUDEwLFAxMCxDb250cm9sLDAsMDBfMDEwLEUsOTY1MzEsNjQ2MzYsODEyLDAuNjcKUDAyX0QwMF8wMTBfQzBBLFAyLFAwMixDb250cm9sLDAsMDBfMDEwLEEsNzQ1ODEsNTExNDQsMTAxMSwwLjY4NgpQMDNfRDAwXzAxMF9XM0IsUDMsUDAzLFdhcm0sMywwMF8wMTAsQiw2MDAwNiwzODM3NCw3NjUsMC42NApQMDNfRDAwXzAxMF9XOEIsUDMtOEMsUDAzLFdhcm0sOCwwMF8wMTAsQiwzMTE3NCwxNDE4OCwzNDksMC40NTUKUDA0X0QwMF8wMTBfQzBCLFA0LFAwNCxDb250cm9sLDAsMDBfMDEwLEIsODg3MzIsNTY1OTMsMTAxNywwLjYzOApQMDVfRDAwXzAxMF9XM0MsUDUsUDA1LFdhcm0sMywwMF8wMTAsQyw4MDksMjk2LDk1LDAuMzY2ClAwNV9EMDBfMDEwX1c4QyxQNS04QyxQMDUsV2FybSw4LDAwXzAxMCxDLDgxOTA4LDM1NjAwLDYxNiwwLjQzNQpQMDZfRDAwXzAxMF9DMEMsUDYsUDA2LENvbnRyb2wsMCwwMF8wMTAsQyw2NzQxOSw0NTg1MCw5NTQsMC42OApQMDdfRDAwXzAxMF9XM0QsUDcsUDA3LFdhcm0sMywwMF8wMTAsRCw3MDEyMywzNjA2Myw4NjcsMC41MTQKUDA3X0QwMF8wMTBfVzhELFA3LThDLFAwNyxXYXJtLDgsMDBfMDEwLEQsMTU0MzUsOTE3MiwzMzUsMC41OTQKUDA4X0QwMF8wMTBfQzBELFA4LFAwOCxDb250cm9sLDAsMDBfMDEwLEQsNjA3OTksMzgzNzEsNzAyLDAuNjMxClAwOV9EMDBfMDEwX1czRSxQOSxQMDksV2FybSwzLDAwXzAxMCxFLDU1OTI3LDI5MTAyLDc0NSwwLjUyClAwOV9EMDBfMDEwX1c4RSxQOS04QyxQMDksV2FybSw4LDAwXzAxMCxFLDE0MTgsODAsMTQsMC4wNTYK&#39;).then(res =&gt; res.blob()).then(blob =&gt; {&#10;      const downloadURL = window.URL.createObjectURL(blob);&#10;      const a = document.createElement(&#39;a&#39;);&#10;      document.body.appendChild(a);&#10;      a.href = downloadURL;&#10;      a.download = &#39;its18_sample_summary.csv&#39;; a.click();&#10;      window.URL.revokeObjectURL(downloadURL);&#10;      document.body.removeChild(a);&#10;    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
<div id="htmlwidget-bd7eedfa163b72c55333" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-bd7eedfa163b72c55333">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Sample_ID":["P01_D00_010_W3A","P01_D00_010_W8A","P10_D00_010_C0E","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E"],"root name":["P1","P1-8C","P10","P2","P3","P3-8C","P4","P5","P5-8C","P6","P7","P7-8C","P8","P9","P9-8C"],"Plot":["P01","P01","P10","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09"],"Treatment":["Warm","Warm","Control","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm"],"Temp":[3,8,0,0,3,8,0,3,8,0,3,8,0,3,8],"Depth":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"Pair":["A","A","E","A","B","B","B","C","C","C","D","D","D","E","E"],"input reads":[50894,70734,96531,74581,60006,31174,88732,809,81908,67419,70123,15435,60799,55927,1418],"final reads":[29794,41880,64636,51144,38374,14188,56593,296,35600,45850,36063,9172,38371,29102,80],"no. ASVs":[945,719,812,1011,765,349,1017,95,616,954,867,335,702,745,14],"per. reads remain":[0.585,0.592,0.67,0.686,0.64,0.455,0.638,0.366,0.435,0.68,0.514,0.594,0.631,0.52,0.056]},"columns":[{"accessor":"Sample_ID","name":"Sample ID","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P10_D00_010_C0E","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E"],"header":"Sample ID","align":"left","footerStyle":{"fontWeight":"bold"},"footer":"Total reads","minWidth":150,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"root name","name":"root name","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P1","P1-8C","P10","P2","P3","P3-8C","P4","P5","P5-8C","P6","P7","P7-8C","P8","P9","P9-8C"],"header":"root name","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Plot","name":"Plot","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01","P01","P10","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09"],"header":"Plot","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Treatment","name":"Treatment","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["Warm","Warm","Control","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm"],"header":"Treatment","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Temp","name":"Temp","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["3","8","0","0","3","8","0","3","8","0","3","8","0","3","8"],"header":"Temp","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Depth","name":"Depth","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"header":"Depth","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Pair","name":"Pair","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["A","A","E","A","B","B","B","C","C","C","D","D","D","E","E"],"header":"Pair","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"input reads","name":"input reads","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["50894","70734","96531","74581","60006","31174","88732","809","81908","67419","70123","15435","60799","55927","1418"],"header":"input reads","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"final reads","name":"final reads","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["29794","41880","64636","51144","38374","14188","56593","296","35600","45850","36063","9172","38371","29102","80"],"header":"final reads","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"no. ASVs","name":"no. ASVs","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["945","719","812","1011","765","349","1017","95","616","954","867","335","702","745","14"],"header":"no. ASVs","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"per. reads remain","name":"per. reads remain","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.585","0.592","0.67","0.686","0.64","0.455","0.638","0.366","0.435","0.68","0.514","0.594","0.631","0.52","0.056"],"header":"per. reads remain","align":"center","footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"97e38721ca4829afb80720154cf9d5a7"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
<p>The following samples have fewer than <strong>1000</strong> reads at
the end of the workflow:</p>
<p><strong>P05_D00_010_W3C, P09_D00_010_W8E</strong>.</p>
<p>We can also plot the final read count by the temperature where the
sample was collected.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="defining-groups-1">Defining Groups</h2>
<ol type="1">
<li><p>Load the data packet produced in the final step of the DADA2
workflow. This packet (<code>its18_dada2_wf.rdata</code>) contains the
ASV-by-sample table and the ASV taxonomy table.</p></li>
<li><p>Rename the samples so names have plot and Depth info. For
convenience, we renamed any Negative Controls (NC) to
<code>P00_D00-000_NNN</code>.</p></li>
<li><p>After we load the data packet we next need to format sample names
and define groups.</p></li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> seq_table[<span class="fu">order</span>(seq_table<span class="sc">$</span><span class="st">`</span><span class="at">root name</span><span class="st">`</span>), ]</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>tmp_names <span class="ot">&lt;-</span> seq_table<span class="sc">$</span><span class="st">`</span><span class="at">Sample_ID</span><span class="st">`</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;files/dada2/rdata/its18_dada2_wf.rdata&quot;</span>)</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(seq_table<span class="sc">$</span><span class="st">`</span><span class="at">root name</span><span class="st">`</span>, <span class="fu">rownames</span>(seqtab))</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(seqtab) <span class="ot">&lt;-</span> tmp_names</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>samples.out <span class="ot">&lt;-</span> <span class="fu">rownames</span>(seqtab)</span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>sample_name <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">1</span>, <span class="dv">999</span>)</span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>plot <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">0</span>, <span class="dv">3</span>)</span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>depth <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">6</span>, <span class="dv">11</span>)</span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>treatment <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">13</span>, <span class="dv">13</span>)</span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">14</span>, <span class="dv">14</span>)</span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a>pair <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">15</span>, <span class="dv">15</span>)</span></code></pre></div>
</details>
</div>
<p>We have a total of <strong>15</strong> samples, <strong>5</strong>
sample pairs, from <strong>10</strong> plots (<strong>1</strong> depth).
There are <strong>2</strong> treatments corresponding to
<strong>3</strong> different temperature regimes.</p>
<ol start="4" type="1">
<li>And finally we define a sample data frame that holds the different
groups we extracted from the sample names.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>samdf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">SamName =</span> sample_name,</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">PLOT =</span> plot,</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">DEPTH =</span> depth,</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">TREAT =</span> treatment,</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">TEMP =</span> temp,</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">PAIR =</span> pair)</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(samdf) <span class="ot">&lt;-</span> samples.out</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>For example, P08_D00_010_C0D is the sample heated at
<strong>+0C</strong> from plot <strong>P08</strong>, collected from a
depth of 00_010cm. This sample is in group <strong>D</strong> which
consists of P07_D00_010_W3D, P07_D00_010_W8D, P08_D00_010_C0D.</p>
<h2 id="create-a-phyloseq-object-1">Create a Phyloseq Object</h2>
<h3 id="fix-taxon-names">Fix Taxon Names</h3>
<p>The UNITE database we used for classification adds a little prefix to
every rank call, like <code>k__</code> for Kingdom, <code>o__</code> for
Order, etc. To keep in step with our other analyses, we will remove
these tags. Of course, we also make a ps object with the original names
just in case. At this point we will also remove the Species rank since
it is essentially useless.</p>
<aside>
<em>I KNOW there is an easier way to do this</em>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>tax_mod_o <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(tax)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>tax_mod_o<span class="sc">$</span>Species <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>tax_mod_o <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tax_mod_o)</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>tax_mod <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;k__&quot;</span>, <span class="st">&quot;&quot;</span>, tax_mod_o)</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>tax_mod <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;p__&quot;</span>, <span class="st">&quot;&quot;</span>, tax_mod)</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>tax_mod <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;c__&quot;</span>, <span class="st">&quot;&quot;</span>, tax_mod)</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>tax_mod <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;o__&quot;</span>, <span class="st">&quot;&quot;</span>, tax_mod)</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>tax_mod <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;f__&quot;</span>, <span class="st">&quot;&quot;</span>, tax_mod)</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>tax_mod <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;g__&quot;</span>, <span class="st">&quot;&quot;</span>, tax_mod)</span></code></pre></div>
</div>
<p><strong>A</strong>. The first step is to rename the amplicon sequence
variants (ASVs) so the designations are a bit more user friendly. By
default, DADA2 names each ASV by its unique sequence so that data can be
directly compared across studies (which is great). But this convention
can get cumbersome downstream, so we rename the ASVs using a simpler
conventionASV1, ASV2, ASV3, and so on.</p>
<aside>
A phyloseq object contains ASV table (taxa abundances), sample metadata,
and taxonomy table (mapping between ASVs and higher-level taxonomic
classifications).
</aside>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># this create the phyloseq object</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(<span class="fu">otu_table</span>(seqtab, <span class="at">taxa_are_rows =</span> <span class="cn">FALSE</span>),</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">sample_data</span>(samdf), <span class="fu">tax_table</span>(tax_mod))</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">tax_table</span>(ps),</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">rownames</span>(<span class="fu">tax_table</span>(ps)))</span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a><span class="co"># adding unique ASV names</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a><span class="fu">taxa_names</span>(ps) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;ASV&quot;</span>, <span class="fu">seq</span>(<span class="fu">ntaxa</span>(ps)))</span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">tax_table</span>(ps),</span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">rownames</span>(<span class="fu">tax_table</span>(ps)))</span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">taxa_names</span>(ps))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;ASV1&quot; &quot;ASV2&quot; &quot;ASV3&quot; &quot;ASV4&quot; &quot;ASV5&quot; &quot;ASV6&quot;</code></pre>
</div>
<p>So the complete data set contains 3357 ASVs. We can also use the <a
href="https://github.com/microbiome/microbiome/">microbiome R
package</a> <span class="citation" data-cites="lahti2017microbiome">(<a
href="#ref-lahti2017microbiome" role="doc-biblioref">Lahti, Sudarshan,
et al. 2017</a>)</span> to get some additional summary data from the
phyloseq object.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<table>
<colgroup>
<col style="width: 40%" />
<col style="width: 59%" />
</colgroup>
<thead>
<tr class="header">
<th>Metric</th>
<th>Results</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>80</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>64636</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>491143</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>32743</td>
</tr>
<tr class="odd">
<td>Median number of reads</td>
<td>36063</td>
</tr>
<tr class="even">
<td>Total ASVs</td>
<td>3357</td>
</tr>
<tr class="odd">
<td>Sparsity</td>
<td>0.802</td>
</tr>
<tr class="even">
<td>Any ASVs sum to 1 or less?</td>
<td>TRUE</td>
</tr>
<tr class="odd">
<td>Number of singleton ASVs</td>
<td>3</td>
</tr>
<tr class="even">
<td>Percent of ASVs that are singletons</td>
<td>0.089</td>
</tr>
<tr class="odd">
<td>Number of sample variables are:</td>
<td>6 (SamName, PLOT, DEPTH, TREAT, TEMP, PAIR)</td>
</tr>
</tbody>
</table>
<p><strong>B</strong>. Add two final columns containing the ASV
sequences and ASV IDs. This will be useful later when trying to export a
fasta file. We can also take a look at the phyloseq object.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(<span class="fu">tax_table</span>(ps)) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Kingdom&quot;</span>, <span class="st">&quot;Phylum&quot;</span>, <span class="st">&quot;Class&quot;</span>, <span class="st">&quot;Order&quot;</span>,</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Genus&quot;</span>, <span class="st">&quot;ASV_SEQ&quot;</span>, <span class="st">&quot;ASV_ID&quot;</span>)</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rank_names</span>(ps)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3357 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3357 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><strong>C</strong>. Export sequence and taxonomy tables for the
unadulterated phyloseq object for later use. We will use the prefix
<code>full</code> to indicate that these are the <em>raw</em>
outputs.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">tax_table</span>(ps),</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;files/data-prep/tables/its18_full_tax_table.txt&quot;</span>,</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ps)),</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;files/data-prep/tables/its18_full_seq_table.txt&quot;</span>,</span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>At this point we would normally remove unwanted taxa and/or negative
controls. Since we do not have any of these for this dataset, we are
going to save the phyloseq object with a new name. That way, if we do
decide later on to remove some taxa it will be easier to plug in the
code here and then continue with the rest of the workflow.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>ps_filt <span class="ot">&lt;-</span> ps</span></code></pre></div>
</div>
<h2 id="remove-low-count-samples">Remove Low-Count Samples</h2>
<p>Next, we can remove samples with really low read counts, say less
than 500 reads.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>ps_filt_no_low <span class="ot">&lt;-</span> <span class="fu">prune_samples</span>(<span class="fu">sample_sums</span>(ps_filt) <span class="sc">&gt;</span> <span class="dv">500</span>, ps_filt)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>ps_filt_no_low</span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3357 taxa and 13 samples ]
sample_data() Sample Data:       [ 13 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3357 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<p>We lost 2 sample(s). After removing samples we need to check whether
any ASVs ended up with no reads.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>no_reads <span class="ot">&lt;-</span> <span class="fu">taxa_sums</span>(ps_filt_no_low) <span class="sc">==</span> <span class="dv">0</span></span></code></pre></div>
</div>
<p>And we lost 2 ASV(s). So now we must remove these from the ps
object.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>ps_filt_no_nc <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(<span class="fu">taxa_sums</span>(ps_filt_no_low) <span class="sc">&gt;</span> <span class="dv">0</span>, ps_filt_no_low)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>ps_filt_no_nc</span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3355 taxa and 13 samples ]
sample_data() Sample Data:       [ 13 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3355 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<h2 id="rename-na-taxonomic-ranks-1">Rename NA taxonomic ranks</h2>
<p>Phyloseq has an odd way of dealing with taxonomic ranks that have no
valuein other words, <em>NA</em> in the tax table. The first thing we
are going to do before moving forward is to change all of the
<em>NA</em>s to have a value of the next highest classified rank. For
example, <code>ASV4</code> is not classified at the Genus level but is
at Family level (Nitrososphaeraceae). So we change the Genus name to
<em>Family_Nitrososphaeraceae</em>. The code for comes from these two
posts on the phyloseq GitHub, both by <a
href="https://github.com/MSMortensen">MSMortensen</a>: issue <a
href="https://github.com/joey711/phyloseq/issues/850#issuecomment-394771087">#850</a>
and issue <a
href="https://github.com/joey711/phyloseq/issues/990#issuecomment-424618425">#990</a>.</p>
<blockquote>
<p>One thing this code does is reassign the functions <code>class</code>
and <code>order</code> to taxonomic ranks. This can cause issues if you
need these functions.</p>
</blockquote>
<p>So you need to run something like this
<code>rm(class, order, phylum, kingdom)</code> at the end of the code to
remove these as variables. For now, I have not come up with a better
solution.</p>
<p>Before doing that, we will save a copy of the phyloseq object with
the original taxa names.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>ps_work_o <span class="ot">&lt;-</span> ps_filt_no_nc</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>tax.clean <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ps_filt_no_nc))</span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>){ tax.clean[,i] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tax.clean[,i])}</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>tax.clean[<span class="fu">is.na</span>(tax.clean)] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(tax.clean)){</span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (tax.clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>        kingdom <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;k_&quot;</span>, tax.clean[i,<span class="dv">1</span>], <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> kingdom</span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a>        phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;p_&quot;</span>, tax.clean[i,<span class="dv">2</span>], <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb54-11"><a href="#cb54-11" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">3</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb54-12"><a href="#cb54-12" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">4</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb54-13"><a href="#cb54-13" aria-hidden="true" tabindex="-1"></a>        class <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;c_&quot;</span>, tax.clean[i,<span class="dv">3</span>], <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb54-14"><a href="#cb54-14" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">4</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> class</span>
<span id="cb54-15"><a href="#cb54-15" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">5</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb54-16"><a href="#cb54-16" aria-hidden="true" tabindex="-1"></a>        order <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;o_&quot;</span>, tax.clean[i,<span class="dv">4</span>], <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb54-17"><a href="#cb54-17" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">5</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> order</span>
<span id="cb54-18"><a href="#cb54-18" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">6</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb54-19"><a href="#cb54-19" aria-hidden="true" tabindex="-1"></a>        tax.clean<span class="sc">$</span>Genus[i] <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;f&quot;</span>,tax.clean<span class="sc">$</span>Family[i], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb54-20"><a href="#cb54-20" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb54-21"><a href="#cb54-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb54-22"><a href="#cb54-22" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps_filt_no_nc) <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tax.clean)</span>
<span id="cb54-23"><a href="#cb54-23" aria-hidden="true" tabindex="-1"></a><span class="fu">rank_names</span>(ps_filt_no_nc)</span>
<span id="cb54-24"><a href="#cb54-24" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(class, order, phylum, kingdom)</span></code></pre></div>
</details>
</div>
<p>Still the same ranks. Thats good. What about the new groups? Lets
take a peak at some families.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_filt_no_nc, <span class="st">&quot;Family&quot;</span>), <span class="dv">16</span>)</span></code></pre></div>
<pre><code> [1] &quot;Hygrophoraceae&quot;         &quot;Mycosphaerellaceae&quot;    
 [3] &quot;p_Ascomycota&quot;           &quot;Trichosporonaceae&quot;     
 [5] &quot;c_Agaricomycetes&quot;       &quot;Archaeorhizomycetaceae&quot;
 [7] &quot;Chaetomellaceae&quot;        &quot;Geastraceae&quot;           
 [9] &quot;k_Fungi&quot;                &quot;Metschnikowiaceae&quot;     
[11] &quot;Clavariaceae&quot;           &quot;o_Agaricales&quot;          
[13] &quot;Ascobolaceae&quot;           &quot;Trichocomaceae&quot;        
[15] &quot;Piptocephalidaceae&quot;     &quot;p_Chytridiomycota&quot;     </code></pre>
</div>
<p>Nice. Adios <em>NA</em>.</p>
<blockquote>
<p><em>Note</em>. The original code is written for data sets that have
species-level designations.</p>
</blockquote>
<p>Since this data set does not contain species, I modified the code to
stop at the genus level. If your data set has species, my modifications
will not work for you.</p>
<p>Finally, we rename the ps object. This is now our working data
set.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>ps_work <span class="ot">&lt;-</span> ps_filt_no_nc</span></code></pre></div>
</div>
<h2 id="data-set-summary-1">Data Set Summary</h2>
<p>Now that we have removed the NC and samples with low reads, we can
summarize the data in our working phyloseq object. Lets start with the
data set as a whole using the <code>summarize_phyloseq</code> from the
<a href="https://github.com/microbiome/microbiome/">microbiome R
package</a> <span class="citation" data-cites="lahti2017microbiome">(<a
href="#ref-lahti2017microbiome" role="doc-biblioref">Lahti, Sudarshan,
et al. 2017</a>)</span> as we did above.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<table>
<colgroup>
<col style="width: 38%" />
<col style="width: 61%" />
</colgroup>
<thead>
<tr class="header">
<th>Metric</th>
<th>Results</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>9172</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>64636</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>490767</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>37751</td>
</tr>
<tr class="odd">
<td>Median number of reads</td>
<td>38371</td>
</tr>
<tr class="even">
<td>Total ASVs</td>
<td>3355</td>
</tr>
<tr class="odd">
<td>Sparsity</td>
<td>0.671</td>
</tr>
<tr class="even">
<td>Any ASVs sum to 1 or less?</td>
<td>TRUE</td>
</tr>
<tr class="odd">
<td>Number of singleton ASVs</td>
<td>5</td>
</tr>
<tr class="even">
<td>Percent of ASVs that are singletons</td>
<td>0.149</td>
</tr>
<tr class="odd">
<td>Number of sample variables are:</td>
<td>6 (SamName, PLOT, DEPTH, TREAT, TEMP, PAIR)</td>
</tr>
</tbody>
</table>
<p>We can also generate a summary table of total reads &amp; ASVs for
each sample. You can sort the table, download a copy, or filter by
search term. Here is the code to generate the data for the table. First,
we create data frames that hold total reads and ASVs for each
sample.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> <span class="fu">sample_sums</span>(ps_work)</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(total_reads, <span class="at">make.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> total_reads <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;Sample_ID&quot;</span>)</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>total_asvs <span class="ot">&lt;-</span> <span class="fu">estimate_richness</span>(ps_work, <span class="at">measures =</span> <span class="st">&quot;Observed&quot;</span>)</span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>total_asvs <span class="ot">&lt;-</span> total_asvs <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;Sample_ID&quot;</span>)</span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>total_asvs<span class="sc">$</span>Sample_ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>, <span class="st">&#39;-&#39;</span>, total_asvs<span class="sc">$</span>Sample_ID)</span></code></pre></div>
</details>
</div>
<p>And then we merge these two data frames with the sample data
frame.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>sam_details <span class="ot">&lt;-</span> samdf</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sam_details) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(sam_details) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Sample_ID&quot;</span>, <span class="st">&quot;Plot&quot;</span>, <span class="st">&quot;Depth&quot;</span>,</span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;Treatment&quot;</span>, <span class="st">&quot;Temp&quot;</span>, <span class="st">&quot;Pair&quot;</span>)</span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a>tmp_tab1 <span class="ot">&lt;-</span> seq_table</span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a>tmp_tab1[, <span class="fu">c</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">7</span>,<span class="dv">11</span>)] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a>tmp_tab2 <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(sam_details, total_reads) <span class="sc">%&gt;%</span></span>
<span id="cb59-10"><a href="#cb59-10" aria-hidden="true" tabindex="-1"></a>                <span class="fu">left_join</span>(., total_asvs) <span class="sc">%&gt;%</span></span>
<span id="cb59-11"><a href="#cb59-11" aria-hidden="true" tabindex="-1"></a>                <span class="fu">left_join</span>(., tmp_tab1)</span>
<span id="cb59-12"><a href="#cb59-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-13"><a href="#cb59-13" aria-hidden="true" tabindex="-1"></a>reads_lost <span class="ot">&lt;-</span> tmp_tab2<span class="sc">$</span><span class="st">`</span><span class="at">final reads</span><span class="st">`</span> <span class="sc">-</span> tmp_tab2<span class="sc">$</span>total_reads </span>
<span id="cb59-14"><a href="#cb59-14" aria-hidden="true" tabindex="-1"></a>asvs_lost <span class="ot">&lt;-</span> tmp_tab2<span class="sc">$</span><span class="st">`</span><span class="at">no. ASVs</span><span class="st">`</span> <span class="sc">-</span> tmp_tab2<span class="sc">$</span>Observed </span>
<span id="cb59-15"><a href="#cb59-15" aria-hidden="true" tabindex="-1"></a>tmp_tab2<span class="sc">$</span>reads_lost <span class="ot">&lt;-</span> reads_lost</span>
<span id="cb59-16"><a href="#cb59-16" aria-hidden="true" tabindex="-1"></a>tmp_tab2<span class="sc">$</span>asvs_lost <span class="ot">&lt;-</span> asvs_lost</span>
<span id="cb59-17"><a href="#cb59-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-18"><a href="#cb59-18" aria-hidden="true" tabindex="-1"></a>seq_table2 <span class="ot">&lt;-</span> tmp_tab2[, <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>,<span class="dv">12</span>,<span class="dv">13</span>)]</span>
<span id="cb59-19"><a href="#cb59-19" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(seq_table2) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;sample_id&quot;</span>, <span class="st">&quot;PLOT&quot;</span>, <span class="st">&quot;DEPTH&quot;</span>,</span>
<span id="cb59-20"><a href="#cb59-20" aria-hidden="true" tabindex="-1"></a>                          <span class="st">&quot;TREAT&quot;</span>, <span class="st">&quot;TEMP&quot;</span>, <span class="st">&quot;PAIR&quot;</span>, <span class="st">&quot;total_reads&quot;</span>,</span>
<span id="cb59-21"><a href="#cb59-21" aria-hidden="true" tabindex="-1"></a>                          <span class="st">&quot;total_ASVs&quot;</span>, <span class="st">&quot;reads_lost&quot;</span>, <span class="st">&quot;ASVs_lost&quot;</span>)</span></code></pre></div>
</details>
</div>
<p><small><strong>(ITS) Table 2 | Sample summary table showing the final
number of reads and ASVs (per sample) as well as the number of
reads/ASVs lost during curation.</strong></small></p>
<div class="layout-chunk" data-layout="l-body-outset">
<a onclick="fetch(&#39;data:text/csv;base64,c2FtcGxlX2lkLFBMT1QsREVQVEgsVFJFQVQsVEVNUCxQQUlSLHRvdGFsX3JlYWRzLHRvdGFsX0FTVnMscmVhZHNfbG9zdCxBU1ZzX2xvc3QKUDAxX0QwMF8wMTBfVzNBLFAwMSwwMF8wMTAsVywzLEEsMjk3OTQsOTQ1LDAsMApQMDFfRDAwXzAxMF9XOEEsUDAxLDAwXzAxMCxXLDgsQSw0MTg4MCw3MTksMCwwClAxMF9EMDBfMDEwX0MwRSxQMTAsMDBfMDEwLEMsMCxFLDY0NjM2LDgxMiwwLDAKUDAyX0QwMF8wMTBfQzBBLFAwMiwwMF8wMTAsQywwLEEsNTExNDQsMTAxMSwwLDAKUDAzX0QwMF8wMTBfVzNCLFAwMywwMF8wMTAsVywzLEIsMzgzNzQsNzY1LDAsMApQMDNfRDAwXzAxMF9XOEIsUDAzLDAwXzAxMCxXLDgsQiwxNDE4OCwzNDksMCwwClAwNF9EMDBfMDEwX0MwQixQMDQsMDBfMDEwLEMsMCxCLDU2NTkzLDEwMTcsMCwwClAwNV9EMDBfMDEwX1czQyxQMDUsMDBfMDEwLFcsMyxDLE5BLE5BLE5BLE5BClAwNV9EMDBfMDEwX1c4QyxQMDUsMDBfMDEwLFcsOCxDLDM1NjAwLDYxNiwwLDAKUDA2X0QwMF8wMTBfQzBDLFAwNiwwMF8wMTAsQywwLEMsNDU4NTAsOTU0LDAsMApQMDdfRDAwXzAxMF9XM0QsUDA3LDAwXzAxMCxXLDMsRCwzNjA2Myw4NjcsMCwwClAwN19EMDBfMDEwX1c4RCxQMDcsMDBfMDEwLFcsOCxELDkxNzIsMzM1LDAsMApQMDhfRDAwXzAxMF9DMEQsUDA4LDAwXzAxMCxDLDAsRCwzODM3MSw3MDIsMCwwClAwOV9EMDBfMDEwX1czRSxQMDksMDBfMDEwLFcsMyxFLDI5MTAyLDc0NSwwLDAKUDA5X0QwMF8wMTBfVzhFLFAwOSwwMF8wMTAsVyw4LEUsTkEsTkEsTkEsTkEK&#39;).then(res =&gt; res.blob()).then(blob =&gt; {&#10;      const downloadURL = window.URL.createObjectURL(blob);&#10;      const a = document.createElement(&#39;a&#39;);&#10;      document.body.appendChild(a);&#10;      a.href = downloadURL;&#10;      a.download = &#39;its18_sample_summary_final.csv&#39;; a.click();&#10;      window.URL.revokeObjectURL(downloadURL);&#10;      document.body.removeChild(a);&#10;    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
<div id="htmlwidget-f5df62365d825fffd82e" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f5df62365d825fffd82e">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"sample_id":["P01_D00_010_W3A","P01_D00_010_W8A","P10_D00_010_C0E","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E"],"PLOT":["P01","P01","P10","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09"],"DEPTH":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"TREAT":["W","W","C","C","W","W","C","W","W","C","W","W","C","W","W"],"TEMP":["3","8","0","0","3","8","0","3","8","0","3","8","0","3","8"],"PAIR":["A","A","E","A","B","B","B","C","C","C","D","D","D","E","E"],"total_reads":[29794,41880,64636,51144,38374,14188,56593,"NA",35600,45850,36063,9172,38371,29102,"NA"],"total_ASVs":[945,719,812,1011,765,349,1017,"NA",616,954,867,335,702,745,"NA"],"reads_lost":[0,0,0,0,0,0,0,"NA",0,0,0,0,0,0,"NA"],"ASVs_lost":[0,0,0,0,0,0,0,"NA",0,0,0,0,0,0,"NA"]},"columns":[{"accessor":"sample_id","name":"sample_id","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P10_D00_010_C0E","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E"],"header":"sample id","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"PLOT","name":"PLOT","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01","P01","P10","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09"],"header":"PLOT","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"DEPTH","name":"DEPTH","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"header":"DEPTH","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"TREAT","name":"TREAT","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["W","W","C","C","W","W","C","W","W","C","W","W","C","W","W"],"header":"TREAT","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"TEMP","name":"TEMP","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["3","8","0","0","3","8","0","3","8","0","3","8","0","3","8"],"header":"TEMP","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"PAIR","name":"PAIR","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["A","A","E","A","B","B","B","C","C","C","D","D","D","E","E"],"header":"PAIR","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"total_reads","name":"total_reads","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["29794","41880","64636","51144","38374","14188","56593","NA","35600","45850","36063","9172","38371","29102","NA"],"header":"total reads","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"total_ASVs","name":"total_ASVs","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["945","719","812","1011","765","349","1017","NA","616","954","867","335","702","745","NA"],"header":"total ASVs","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"reads_lost","name":"reads_lost","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0","0","0","0","0","0","0","NA","0","0","0","0","0","0","NA"],"header":"reads lost","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"ASVs_lost","name":"ASVs_lost","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0","0","0","0","0","0","0","NA","0","0","0","0","0","0","NA"],"header":"ASVs lost","align":"center","footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"a2dd250c4eb4bd5fe9c909c2c8dd3f2f"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
<h2 id="review-1">Review</h2>
<p>We have a few different options to play with. At this point it is
difficult to say which we will use or whether additional objects need to
be created. Here is a summary of the objects we have.</p>
<p>The full phyloseq object <em>before</em> removing contaminants.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>ps</span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3357 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3357 taxa by 8 taxonomic ranks ]</code></pre>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps, <span class="st">&quot;Family&quot;</span>), <span class="dv">16</span>)</span></code></pre></div>
<pre><code> [1] &quot;Hygrophoraceae&quot;                 
 [2] &quot;Mycosphaerellaceae&quot;             
 [3] NA                               
 [4] &quot;Trichosporonaceae&quot;              
 [5] &quot;Archaeorhizomycetaceae&quot;         
 [6] &quot;Chaetomellaceae&quot;                
 [7] &quot;Geastraceae&quot;                    
 [8] &quot;Metschnikowiaceae&quot;              
 [9] &quot;Clavariaceae&quot;                   
[10] &quot;Ascobolaceae&quot;                   
[11] &quot;Trichocomaceae&quot;                 
[12] &quot;Piptocephalidaceae&quot;             
[13] &quot;Mortierellaceae&quot;                
[14] &quot;Pleosporales_fam_Incertae_sedis&quot;
[15] &quot;Glomeraceae&quot;                    
[16] &quot;Ceratobasidiaceae&quot;              </code></pre>
</div>
<p>The phyloseq object <em>before</em> removing the prefix of taxon
ranks. This still has low coverage samples.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>ps_o</span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3357 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3357 taxa by 8 taxonomic ranks ]</code></pre>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_o, <span class="st">&quot;Family&quot;</span>), <span class="dv">16</span>)</span></code></pre></div>
<pre><code> [1] &quot;f__Hygrophoraceae&quot;                 
 [2] &quot;f__Mycosphaerellaceae&quot;             
 [3] NA                                  
 [4] &quot;f__Trichosporonaceae&quot;              
 [5] &quot;f__Archaeorhizomycetaceae&quot;         
 [6] &quot;f__Chaetomellaceae&quot;                
 [7] &quot;f__Geastraceae&quot;                    
 [8] &quot;f__Metschnikowiaceae&quot;              
 [9] &quot;f__Clavariaceae&quot;                   
[10] &quot;f__Ascobolaceae&quot;                   
[11] &quot;f__Trichocomaceae&quot;                 
[12] &quot;f__Piptocephalidaceae&quot;             
[13] &quot;f__Mortierellaceae&quot;                
[14] &quot;f__Pleosporales_fam_Incertae_sedis&quot;
[15] &quot;f__Glomeraceae&quot;                    
[16] &quot;f__Ceratobasidiaceae&quot;              </code></pre>
</div>
<p>The phyloseq object <em>before</em> changing NA ranks.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>ps_work_o</span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3355 taxa and 13 samples ]
sample_data() Sample Data:       [ 13 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3355 taxa by 8 taxonomic ranks ]</code></pre>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_work_o, <span class="st">&quot;Family&quot;</span>), <span class="dv">16</span>)</span></code></pre></div>
<pre><code> [1] &quot;Hygrophoraceae&quot;                 
 [2] &quot;Mycosphaerellaceae&quot;             
 [3] NA                               
 [4] &quot;Trichosporonaceae&quot;              
 [5] &quot;Archaeorhizomycetaceae&quot;         
 [6] &quot;Chaetomellaceae&quot;                
 [7] &quot;Geastraceae&quot;                    
 [8] &quot;Metschnikowiaceae&quot;              
 [9] &quot;Clavariaceae&quot;                   
[10] &quot;Ascobolaceae&quot;                   
[11] &quot;Trichocomaceae&quot;                 
[12] &quot;Piptocephalidaceae&quot;             
[13] &quot;Mortierellaceae&quot;                
[14] &quot;Pleosporales_fam_Incertae_sedis&quot;
[15] &quot;Glomeraceae&quot;                    
[16] &quot;Ceratobasidiaceae&quot;              </code></pre>
</div>
<p>The phyloseq object <em>after</em> changing NA ranks.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>ps_work</span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3355 taxa and 13 samples ]
sample_data() Sample Data:       [ 13 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3355 taxa by 8 taxonomic ranks ]</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_work, <span class="st">&quot;Family&quot;</span>), <span class="dv">16</span>)</span></code></pre></div>
<pre><code> [1] &quot;Hygrophoraceae&quot;         &quot;Mycosphaerellaceae&quot;    
 [3] &quot;p_Ascomycota&quot;           &quot;Trichosporonaceae&quot;     
 [5] &quot;c_Agaricomycetes&quot;       &quot;Archaeorhizomycetaceae&quot;
 [7] &quot;Chaetomellaceae&quot;        &quot;Geastraceae&quot;           
 [9] &quot;k_Fungi&quot;                &quot;Metschnikowiaceae&quot;     
[11] &quot;Clavariaceae&quot;           &quot;o_Agaricales&quot;          
[13] &quot;Ascobolaceae&quot;           &quot;Trichocomaceae&quot;        
[15] &quot;Piptocephalidaceae&quot;     &quot;p_Chytridiomycota&quot;     </code></pre>
</div>
<h2 id="final-steps-1">Final Steps</h2>
<p>The last thing tasks to complete are to <strong>a</strong>) save
copies of the taxonomy and sequence tables and <strong>b</strong>) save
an image of the workflow for the next part of the workflow.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">tax_table</span>(ps_work),</span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;files/data-prep/tables/its18_work_tax_table.txt&quot;</span>,</span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ps_work)),</span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;files/data-prep/tables/its18_work_seq_table.txt&quot;</span>,</span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="workflow-output">Workflow Output</h1>
<p>Data products generated in this workflow can be downloaded from
figshare.</p>
<iframe src="https://widgets.figshare.com/articles/14690739/embed?show_title=1" width="568" height="351" allowfullscreen frameborder="0">
</iframe>
<p>Thats all for this part!</p>
<p></br></p>
<p><a href="filtering.html" class="btnnav button_round" style="float: right;">Next
workflow: <br> 3. Filtering</a>
<a href="dada2.html" class="btnnav button_round" style="float: left;">Previous
workflow: <br> 1. DADA2 Workflow</a></p>
<h2 class="appendix" id="source-code">Source Code</h2>
<p>You can find the source code for this page by <a
href="https://github.com/sweltr/high-temp/blob/master/data-prep.Rmd">clicking
this link</a>.</p>
<h2 class="appendix" id="data-availability">Data Availability</h2>
<p>Data generated in this workflow and the <code>.Rdata</code> need to
run the workflow can be accessed on figshare at <a
href="https://doi.org/10.25573/data.14690739">10.25573/data.14690739</a>.</p>
<h2 class="appendix" id="last-updated-on">Last updated on</h2>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;2022-06-02 13:48:36 EST&quot;</code></pre>
</div>
<div id="refs" class="references csl-bib-body hanging-indent"
role="doc-bibliography">
<div id="ref-lahti2017microbiome" class="csl-entry"
role="doc-biblioentry">
Lahti, Leo, Shetty Sudarshan, et al. 2017. <span>Tools for Microbiome
Analysis in r. Version 1.9.97.</span> <a
href="https://github.com/microbiome/microbiome">https://github.com/microbiome/microbiome</a>.
</div>
<div id="ref-mcmurdie2013phyloseq" class="csl-entry"
role="doc-biblioentry">
McMurdie, Paul J, and Susan Holmes. 2013. <span>Phyloseq: An r Package
for Reproducible Interactive Analysis and Graphics of Microbiome Census
Data.</span> <em>PLoS One</em> 8 (4): e61217. <a
href="https://doi.org/10.1371/journal.pone.0061217">https://doi.org/10.1371/journal.pone.0061217</a>.
</div>
</div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/sweltr/high-temp/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/sweltr/high-temp/">https://github.com/sweltr/high-temp/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<div class="distill-site-nav distill-site-footer">
<p> Copyright 2022 metacrobe and The SWELTR Project.</p>
<p>Site constructed using <a
href="https://rstudio.github.io/distill/">Distill for R
Markdown</a>.</p>
</div>
<!--/radix_placeholder_navigation_after_body-->


</body>

</html>
