<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-0.9.514">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="description" content="Workflow for preparation of the 16S rRNA and ITS data sets. These steps are needed before analyzing the data. In this workflow, sample groups are defined, and phyloseq objects are created and curated.">
<title>- 2. Data Set Preparation</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./filtering.html" rel="next">
<link href="./dada2.html" rel="prev">
<link href="./assets/favicon.ico" rel="icon">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script><style>
    .quarto-title-block .quarto-title-banner {
      background: #FFFFFF;
    }
    </style>
<script src="site_libs/font-awesome-5.13.0/js/script.js"></script><script src="site_libs/core-js-2.5.3/shim.min.js"></script><script src="site_libs/react-17.0.0/react.min.js"></script><script src="site_libs/react-17.0.0/react-dom.min.js"></script><script src="site_libs/reactwidget-1.0.0/react-tools.js"></script><script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><script src="site_libs/reactable-binding-0.2.3.9000/reactable.js"></script><link rel="stylesheet" href="assets/styles.css">
</head>
<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg navbar-dark "><div class="navbar-container container-fluid">
      <a class="navbar-brand" href="./index.html">
    <img src="./assets/icon.png" alt=""></a>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item compact">
    <a class="nav-link" href="./index.html"><i class="bi bi-home" role="img">
</i> 
 </a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-a.-processing" role="button" data-bs-toggle="dropdown" aria-expanded="false">A. Processing</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-a.-processing">
<li class="dropdown-header">Workflows</li>
        <li><hr class="dropdown-divider"></li>
        <li>
    <a class="dropdown-item" href="./dada2.html">
 <span class="dropdown-text">1. DADA2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./data-prep.html">
 <span class="dropdown-text">2. Data Set Prep</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./filtering.html">
 <span class="dropdown-text">3. Filtering</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-b.-diversity" role="button" data-bs-toggle="dropdown" aria-expanded="false">B. Diversity</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-b.-diversity">
<li class="dropdown-header">Workflows</li>
        <li><hr class="dropdown-divider"></li>
        <li>
    <a class="dropdown-item" href="./taxa.html">
 <span class="dropdown-text">4. Taxonomic</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./alpha.html">
 <span class="dropdown-text">5. Alpha Diversity</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./beta.html">
 <span class="dropdown-text">6. Beta Diversity</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./da.html">
 <span class="dropdown-text">7. DA ASVs &amp; Taxa</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./metadata.html">
 <span class="dropdown-text">8. Metadata</span></a>
  </li>  
    </ul>
</li>
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item">
    <a class="nav-link active" href="./data-availability.html" aria-current="page">DATA</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./som.html">SOM</a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/sweltr/high-temp/"><i class="bi bi-github" role="img" aria-label="Agua Salud GitHub">
</i> 
 </a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="./about.html"><i class="bi bi-question" role="img">
</i> 
 </a>
  </li>  
</ul>
<div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }"><div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">2. Data Set Preparation</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title d-none d-lg-block">2. Data Set Preparation</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i></button></div></div>
                  <div>
        <div class="description">
          <p>Workflow for preparation of the <strong>16S rRNA</strong> and <strong>ITS</strong> data sets. These steps are needed before analyzing the data. In this workflow, sample groups are defined, and phyloseq objects are created and curated.</p>
        </div>
      </div>
                </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto"><div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">A. Processing</a>
      <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./dada2.html" class="sidebar-item-text sidebar-link">1. DADA2 Workflow</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./data-prep.html" class="sidebar-item-text sidebar-link active">2. Data Set Preparation</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./filtering.html" class="sidebar-item-text sidebar-link">3. Filtering</a>
  </div>
</li>
    </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">B. Diversity</a>
      <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./taxa.html" class="sidebar-item-text sidebar-link">4. Taxonomic Diversity</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./alpha.html" class="sidebar-item-text sidebar-link">5. Alpha Diversity Estimates</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./beta.html" class="sidebar-item-text sidebar-link">6. Beta Diversity &amp; Dispersion Estimates</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./da.html" class="sidebar-item-text sidebar-link">7. Differentially Abundant ASVs &amp; Taxa</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./metadata.html" class="sidebar-item-text sidebar-link">8. Metadata Analysis</a>
  </div>
</li>
    </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">More</a>
      <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./data-availability.html" class="sidebar-item-text sidebar-link">Data Availability</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./about.html" class="sidebar-item-text sidebar-link">Website Contruction</a>
  </div>
</li>
    </ul>
</li>
    </ul>
</div>
</nav><!-- margin-sidebar --><div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#workflow-input" id="toc-workflow-input" class="nav-link active" data-scroll-target="#workflow-input">Workflow Input</a></li>
  <li>
<a href="#s-rrna-data" id="toc-s-rrna-data" class="nav-link" data-scroll-target="#s-rrna-data">16s rRNA Data</a>
  <ul class="collapse">
<li><a href="#prerequisites" id="toc-prerequisites" class="nav-link" data-scroll-target="#prerequisites">Prerequisites</a></li>
  <li><a href="#read-counts-assessment" id="toc-read-counts-assessment" class="nav-link" data-scroll-target="#read-counts-assessment">Read Counts Assessment</a></li>
  <li><a href="#defining-groups" id="toc-defining-groups" class="nav-link" data-scroll-target="#defining-groups">Defining Groups</a></li>
  <li><a href="#create-a-phyloseq-object" id="toc-create-a-phyloseq-object" class="nav-link" data-scroll-target="#create-a-phyloseq-object">Create a Phyloseq Object</a></li>
  <li><a href="#remove-contaminants-unwanted-taxa" id="toc-remove-contaminants-unwanted-taxa" class="nav-link" data-scroll-target="#remove-contaminants-unwanted-taxa">Remove Contaminants &amp; Unwanted Taxa</a></li>
  <li><a href="#rename-na-taxonomic-ranks" id="toc-rename-na-taxonomic-ranks" class="nav-link" data-scroll-target="#rename-na-taxonomic-ranks">Rename NA taxonomic ranks</a></li>
  <li><a href="#add-phylogenetic-tree" id="toc-add-phylogenetic-tree" class="nav-link" data-scroll-target="#add-phylogenetic-tree">Add Phylogenetic Tree</a></li>
  <li><a href="#data-set-summary" id="toc-data-set-summary" class="nav-link" data-scroll-target="#data-set-summary">Data Set Summary</a></li>
  <li><a href="#review" id="toc-review" class="nav-link" data-scroll-target="#review">Review</a></li>
  <li><a href="#final-steps" id="toc-final-steps" class="nav-link" data-scroll-target="#final-steps">Final Steps</a></li>
  </ul>
</li>
  <li>
<a href="#its-data" id="toc-its-data" class="nav-link" data-scroll-target="#its-data">ITS Data</a>
  <ul class="collapse">
<li><a href="#prerequisites-1" id="toc-prerequisites-1" class="nav-link" data-scroll-target="#prerequisites-1">Prerequisites</a></li>
  <li><a href="#read-counts-assessment-1" id="toc-read-counts-assessment-1" class="nav-link" data-scroll-target="#read-counts-assessment-1">Read Counts Assessment</a></li>
  <li><a href="#defining-groups-1" id="toc-defining-groups-1" class="nav-link" data-scroll-target="#defining-groups-1">Defining Groups</a></li>
  <li><a href="#create-a-phyloseq-object-1" id="toc-create-a-phyloseq-object-1" class="nav-link" data-scroll-target="#create-a-phyloseq-object-1">Create a Phyloseq Object</a></li>
  <li><a href="#remove-low-count-samples" id="toc-remove-low-count-samples" class="nav-link" data-scroll-target="#remove-low-count-samples">Remove Low-Count Samples</a></li>
  <li><a href="#rename-na-taxonomic-ranks-1" id="toc-rename-na-taxonomic-ranks-1" class="nav-link" data-scroll-target="#rename-na-taxonomic-ranks-1">Rename NA taxonomic ranks</a></li>
  <li><a href="#data-set-summary-1" id="toc-data-set-summary-1" class="nav-link" data-scroll-target="#data-set-summary-1">Data Set Summary</a></li>
  <li><a href="#review-1" id="toc-review-1" class="nav-link" data-scroll-target="#review-1">Review</a></li>
  <li><a href="#final-steps-1" id="toc-final-steps-1" class="nav-link" data-scroll-target="#final-steps-1">Final Steps</a></li>
  </ul>
</li>
  <li>
<a href="#workflow-output" id="toc-workflow-output" class="nav-link" data-scroll-target="#workflow-output">Workflow Output</a>
  <ul class="collapse">
<li><a href="#data-availability" id="toc-data-availability" class="nav-link" data-scroll-target="#data-availability">Data Availability</a></li>
  <li><a href="#last-updated-on" id="toc-last-updated-on" class="nav-link" data-scroll-target="#last-updated-on">Last updated on</a></li>
  </ul>
</li>
  </ul><div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/sweltr/high-temp/edit/main/data-prep.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/sweltr/high-temp/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content"><details><summary><strong>Click here</strong> for setup information.
</summary><div class="cell">
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>echo <span class="op">=</span> <span class="cn">TRUE</span>, eval <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">119</span><span class="op">)</span>
<span class="co">#library(conflicted)</span>
<span class="co">#pacman::p_depends(ape, local = TRUE)  </span>
<span class="co">#pacman::p_depends_reverse(ape, local = TRUE)  </span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/utils/packageDescription.html">packageVersion</a></span><span class="op">(</span><span class="st">"phyloseq"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings">Biostrings</a></span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/utils/packageDescription.html">packageVersion</a></span><span class="op">(</span><span class="st">"Biostrings"</span><span class="op">)</span>
<span class="fu">pacman</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pacman/man/p_load.html">p_load</a></span><span class="op">(</span><span class="va">tidyverse</span>, <span class="va">DT</span>, <span class="va">microbiome</span>, <span class="va">taxa</span>, <span class="va">formatR</span>,
               <span class="va">captioner</span>, <span class="va">reactable</span>, <span class="va">downloadthis</span>, <span class="va">fontawesome</span>,
               <span class="va">metacoder</span>, <span class="va">ampvis2</span>, <span class="va">ape</span>, <span class="va">statnet.common</span>, 
               install <span class="op">=</span> <span class="cn">FALSE</span>, update <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>scipen<span class="op">=</span><span class="fl">999</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html">opts_current</a></span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span>
  <span class="st">"cache"</span>,
  <span class="st">"cache.path"</span>,
  <span class="st">"cache.rebuild"</span>,
  <span class="st">"dependson"</span>,
  <span class="st">"autodep"</span>
<span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</details><section id="workflow-input" class="level1"><h1>Workflow Input</h1>
<blockquote class="blockquote">
<p>You will either need to run the <a href="dada2.html">DADA2 workflow</a> or download the files linked below. See the <a href="data-availability.html">Data Availability</a> page for complete details.</p>
</blockquote>
<p>Files needed to run this workflow can be downloaded from figshare. These are the output files from the <a href="dada2.html">DADA2 workflow</a> page.</p>
<iframe src="https://widgets.figshare.com/articles/14687184/embed?show_title=1" width="100%" height="351" allowfullscreen="" frameborder="0">
</iframe>
<p>These workflows share many common variable names so you must split the workflow into a script for each data set OR run the command <code>remove(list = ls())</code> before beginning the next workflow.</p>
</section><section id="s-rrna-data" class="level1"><h1>16s rRNA Data</h1>
<div class="cell">

</div>
<div class="cell">

</div>
<section id="prerequisites" class="level2"><h2 class="anchored" data-anchor-id="prerequisites">Prerequisites</h2>
<p>In order to run this workflow, you either need to run the corresponding <a href="dada2-high-temp.html#s-rrna-data.html">DADA2 Workflow for 16S rRNA</a> or begin with the output from that workflow, <code>data_prep_ssu18_wf.rdata</code>. See the <a href="data-availability.html">Data Availability</a> page for complete details.</p>
<p>Unless otherwise noted, we primarily use <a href="https://joey711.github.io/phyloseq/">phyloseq</a> <span class="citation" data-cites="mcmurdie2013phyloseq">(<a href="#ref-mcmurdie2013phyloseq" role="doc-biblioref">McMurdie and Holmes 2013</a>)</span> in this section of the workflow to prepare the 2018 16S rRNA data set for community analyses. We prepare the data set by curating samples, removing contaminants, and creating phyloseq objects.</p>
</section><section id="read-counts-assessment" class="level2"><h2 class="anchored" data-anchor-id="read-counts-assessment">Read Counts Assessment</h2>
<p>Before we begin, let’s create a summary table containing some basic sample metadata and the read count data from the <a href="dada2-high-temp.html#s-rrna-data.html">DADA2 workflow</a>. We need to inspect how total reads changed through the workflow. While we are at it, let’s create more intuitive Sample IDs. For more details on how reads changed at each step of the workflow, see the <a href="dada2-high-temp.html#track-reads-through-workflow">summary table</a> at the end of the DADA2 section. Table headers are as follows:</p>
<table class="table">
<colgroup>
<col style="width: 23%">
<col style="width: 76%">
</colgroup>
<thead><tr class="header">
<th>Header</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>Sample ID</code></td>
<td>the new sample ID based on Plot number, Depth, Treatment, Temperature, &amp; Pair</td>
</tr>
<tr class="even">
<td><code>Plot</code></td>
<td>the experimental plot number</td>
</tr>
<tr class="odd">
<td><code>Depth</code></td>
<td>the depth where the sample was collected</td>
</tr>
<tr class="even">
<td><code>Treatment</code></td>
<td>warm or control</td>
</tr>
<tr class="odd">
<td><code>Temp</code></td>
<td>temperature of soil heating (0C, +4C, or +8C)</td>
</tr>
<tr class="even">
<td><code>Pair</code></td>
<td>control/treatment coupling</td>
</tr>
<tr class="odd">
<td><code>input</code></td>
<td>number of raw reads</td>
</tr>
<tr class="even">
<td><code>nochim</code></td>
<td>final read count after removing chimeras</td>
</tr>
<tr class="odd">
<td><code>Remain</code></td>
<td>percent of reads remaining from <code>input</code> to <code>nonchim</code>
</td>
</tr>
<tr class="even">
<td><code>FastqID</code></td>
<td>base name of the fastq file</td>
</tr>
</tbody>
</table>
<p><br></p>
<div class="cell">

</div>
<p><small><strong>(16S rRNA) Table 1 | Sample summary table including read changes at start and end of DADA2 workflow.</strong></small></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,U2FtcGxlX0lELHJvb3QgbmFtZSxQbG90LFRyZWF0bWVudCxUZW1wLERlcHRoLFBhaXIsaW5wdXQgcmVhZHMsZmluYWwgcmVhZHMsbm8uIEFTVnMscGVyLiByZWFkcyByZW1haW4KUDAxX0QwMF8wMTBfVzNBLFAxLFAwMSxXYXJtLDMsMDBfMDEwLEEsODU0ODksNjgwMzcsMjM4MywwLjc5NgpQMDFfRDAwXzAxMF9XOEEsUDEtOEMsUDAxLFdhcm0sOCwwMF8wMTAsQSw3NzY0NSw2MDUzNSwxNjE4LDAuNzgKUDEwX0QwMF8wMTBfQzBFLFAxMCxQMTAsQ29udHJvbCwwLDAwXzAxMCxFLDY3NTgzLDUzNjI1LDIwNzAsMC43OTMKUDAyX0QwMF8wMTBfQzBBLFAyLFAwMixDb250cm9sLDAsMDBfMDEwLEEsODcwOTEsNzA0MzYsMjI3OCwwLjgwOQpQMDNfRDAwXzAxMF9XM0IsUDMsUDAzLFdhcm0sMywwMF8wMTAsQiw5MzA2NSw3NTE4MywyMzY4LDAuODA4ClAwM19EMDBfMDEwX1c4QixQMy04QyxQMDMsV2FybSw4LDAwXzAxMCxCLDY2NDAxLDUzNDM3LDE1OTMsMC44MDUKUDA0X0QwMF8wMTBfQzBCLFA0LFAwNCxDb250cm9sLDAsMDBfMDEwLEIsNjc4MTAsNTM4MTMsMjA1MiwwLjc5NApQMDVfRDAwXzAxMF9XM0MsUDUsUDA1LFdhcm0sMywwMF8wMTAsQyw1ODI2Niw0NzA3MCwxNjcyLDAuODA4ClAwNV9EMDBfMDEwX1c4QyxQNS04QyxQMDUsV2FybSw4LDAwXzAxMCxDLDEyMDg5Miw4Njg0MSwyMTU2LDAuNzE4ClAwNl9EMDBfMDEwX0MwQyxQNixQMDYsQ29udHJvbCwwLDAwXzAxMCxDLDEwMTU5OSw4MTIxOSwzMDgyLDAuNzk5ClAwN19EMDBfMDEwX1czRCxQNyxQMDcsV2FybSwzLDAwXzAxMCxELDc2NzgxLDYxNzQyLDIwMjUsMC44MDQKUDA3X0QwMF8wMTBfVzhELFA3LThDLFAwNyxXYXJtLDgsMDBfMDEwLEQsNzkwMzIsNjIyMDIsMjE2NywwLjc4NwpQMDhfRDAwXzAxMF9DMEQsUDgsUDA4LENvbnRyb2wsMCwwMF8wMTAsRCw3NzU5OSw2MjE3MSwyNDUzLDAuODAxClAwOV9EMDBfMDEwX1czRSxQOSxQMDksV2FybSwzLDAwXzAxMCxFLDk1NTY2LDc2Mjk5LDI0MDIsMC43OTgKUDA5X0QwMF8wMTBfVzhFLFA5LThDLFAwOSxXYXJtLDgsMDBfMDEwLEUsOTUxODIsMjUxNTEsODE2LDAuMjY0Cg==').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'ssu18_sample_summary.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="cell-output-display">

<div id="htmlwidget-6456cbde9d228c420acc" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-6456cbde9d228c420acc">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Sample_ID":["P01_D00_010_W3A","P01_D00_010_W8A","P10_D00_010_C0E","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E"],"root name":["P1","P1-8C","P10","P2","P3","P3-8C","P4","P5","P5-8C","P6","P7","P7-8C","P8","P9","P9-8C"],"Plot":["P01","P01","P10","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09"],"Treatment":["Warm","Warm","Control","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm"],"Temp":[3,8,0,0,3,8,0,3,8,0,3,8,0,3,8],"Depth":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"Pair":["A","A","E","A","B","B","B","C","C","C","D","D","D","E","E"],"input reads":[85489,77645,67583,87091,93065,66401,67810,58266,120892,101599,76781,79032,77599,95566,95182],"final reads":[68037,60535,53625,70436,75183,53437,53813,47070,86841,81219,61742,62202,62171,76299,25151],"no. ASVs":[2383,1618,2070,2278,2368,1593,2052,1672,2156,3082,2025,2167,2453,2402,816],"per. reads remain":[0.796,0.78,0.793,0.809,0.808,0.805,0.794,0.808,0.718,0.799,0.804,0.787,0.801,0.798,0.264]},"columns":[{"accessor":"Sample_ID","name":"Sample ID","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P10_D00_010_C0E","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E"],"header":"Sample ID","align":"left","footerStyle":{"fontWeight":"bold"},"footer":"Total reads","minWidth":150,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"root name","name":"root name","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P1","P1-8C","P10","P2","P3","P3-8C","P4","P5","P5-8C","P6","P7","P7-8C","P8","P9","P9-8C"],"header":"root name","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Plot","name":"Plot","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01","P01","P10","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09"],"header":"Plot","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Treatment","name":"Treatment","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["Warm","Warm","Control","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm"],"header":"Treatment","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Temp","name":"Temp","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["3","8","0","0","3","8","0","3","8","0","3","8","0","3","8"],"header":"Temp","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Depth","name":"Depth","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"header":"Depth","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Pair","name":"Pair","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["A","A","E","A","B","B","B","C","C","C","D","D","D","E","E"],"header":"Pair","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"input reads","name":"input reads","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["85489","77645","67583","87091","93065","66401","67810","58266","120892","101599","76781","79032","77599","95566","95182"],"header":"input reads","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"final reads","name":"final reads","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["68037","60535","53625","70436","75183","53437","53813","47070","86841","81219","61742","62202","62171","76299","25151"],"header":"final reads","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"no. ASVs","name":"no. ASVs","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["2383","1618","2070","2278","2368","1593","2052","1672","2156","3082","2025","2167","2453","2402","816"],"header":"no. ASVs","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"per. reads remain","name":"per. reads remain","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.796","0.78","0.793","0.809","0.808","0.805","0.794","0.808","0.718","0.799","0.804","0.787","0.801","0.798","0.264"],"header":"per. reads remain","align":"center","footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"4336b1e377481b6e4af6b56079537876"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="cell">

</div>
</section><section id="defining-groups" class="level2"><h2 class="anchored" data-anchor-id="defining-groups">Defining Groups</h2>
<ol type="1">
<li><p>Load the data packet produced in the final step of the DADA2 workflow. This packet (<code>ssu18_dada2_wf.rdata</code>) contains the ASV-by-sample table and the ASV taxonomy table.</p></li>
<li><p>Rename the samples so names have plot and Depth info.</p></li>
<li><p>After we load the data packet, we next need to format sample names and define groups.</p></li>
</ol>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">seq_table</span> <span class="op">&lt;-</span> <span class="va">seq_table</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html">order</a></span><span class="op">(</span><span class="va">seq_table</span><span class="op">$</span><span class="va">`root name`</span><span class="op">)</span>, <span class="op">]</span>
<span class="va">tmp_names</span> <span class="op">&lt;-</span> <span class="va">seq_table</span><span class="op">$</span><span class="va">`Sample_ID`</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"files/dada2/rdata/ssu18_dada2_wf.rdata"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span><span class="op">(</span><span class="va">seq_table</span><span class="op">$</span><span class="va">`root name`</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">seqtab</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">seqtab</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">tmp_names</span>
<span class="va">samples.out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">seqtab</span><span class="op">)</span>

<span class="va">sample_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span><span class="op">(</span><span class="va">samples.out</span>, <span class="fl">1</span>, <span class="fl">999</span><span class="op">)</span>
<span class="va">plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span><span class="op">(</span><span class="va">samples.out</span>, <span class="fl">0</span>, <span class="fl">3</span><span class="op">)</span>
<span class="va">depth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span><span class="op">(</span><span class="va">samples.out</span>, <span class="fl">6</span>, <span class="fl">11</span><span class="op">)</span>
<span class="va">treatment</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span><span class="op">(</span><span class="va">samples.out</span>, <span class="fl">13</span>, <span class="fl">13</span><span class="op">)</span>
<span class="va">temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span><span class="op">(</span><span class="va">samples.out</span>, <span class="fl">14</span>, <span class="fl">14</span><span class="op">)</span>
<span class="va">pair</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span><span class="op">(</span><span class="va">samples.out</span>, <span class="fl">15</span>, <span class="fl">15</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We have a total of <strong>15</strong> samples, <strong>5</strong> sample pairs, from <strong>10</strong> plots (<strong>1</strong> depth). There are <strong>2</strong> treatments corresponding to <strong>3</strong> different temperature regimes.</p>
<ol start="4" type="1">
<li>And finally we define a sample data frame that holds the different groups we extracted from the sample names.</li>
</ol>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="co">#define a sample data frame</span>
<span class="va">samdf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>SamName <span class="op">=</span> <span class="va">sample_name</span>,
                    PLOT <span class="op">=</span> <span class="va">plot</span>,
                    DEPTH <span class="op">=</span> <span class="va">depth</span>,
                    TREAT <span class="op">=</span> <span class="va">treatment</span>,
                    TEMP <span class="op">=</span> <span class="va">temp</span>,
                    PAIR <span class="op">=</span> <span class="va">pair</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">samdf</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">samples.out</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">

</div>
<p>For example, P07_D00_010_W3D is the sample heated at <strong>+3C</strong> from plot <strong>P07</strong>, collected from a depth of 00_010cm. This sample is in group <strong>D</strong> which consists of P07_D00_010_W3D, P07_D00_010_W8D, P08_D00_010_C0D.</p>
</section><section id="create-a-phyloseq-object" class="level2"><h2 class="anchored" data-anchor-id="create-a-phyloseq-object">Create a Phyloseq Object</h2>
<p><strong>A</strong>. The first step is to rename the amplicon sequence variants (ASVs) so the designations are a bit more user friendly. By default, DADA2 names each ASV by its unique sequence so that data can be directly compared across studies (which is great). But this convention can get cumbersome downstream, so we rename the ASVs using a simpler convention—ASV1, ASV2, ASV3, and so on.</p>
<div class="callout-note callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>A phyloseq object contains ASV table (taxa abundances), sample metadata, and taxonomy table (mapping between ASVs and higher-level taxonomic classifications).</p>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="co"># this create the phyloseq object</span>
<span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phyloseq.html">phyloseq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">seqtab</span>, taxa_are_rows <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>,
                   <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">samdf</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">tax_silva</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html">cbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>,
                           <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># adding unique ASV names</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_names-methods.html">taxa_names</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"ASV"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/ntaxa-methods.html">ntaxa</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html">cbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>,
                           <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ASV1" "ASV2" "ASV3" "ASV4" "ASV5" "ASV6"</code></pre>
</div>
</div>
<p>So the complete data set contains 20332 ASVs. We can use the <a href="https://github.com/microbiome/microbiome/">microbiome R package</a> <span class="citation" data-cites="lahti2017microbiome">(<a href="#ref-lahti2017microbiome" role="doc-biblioref">Lahti, Sudarshan, et al. 2017</a>)</span> to get some additional summary data from the phyloseq object.</p>
<div class="cell">

</div>
<table class="table">
<colgroup>
<col style="width: 40%">
<col style="width: 59%">
</colgroup>
<thead><tr class="header">
<th>Metric</th>
<th>Results</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>25151</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>86841</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>937761</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>62517</td>
</tr>
<tr class="odd">
<td>Median number of reads</td>
<td>62171</td>
</tr>
<tr class="even">
<td>Total ASVs</td>
<td>20332</td>
</tr>
<tr class="odd">
<td>Sparsity</td>
<td>0.898</td>
</tr>
<tr class="even">
<td>Any ASVs sum to 1 or less?</td>
<td>FALSE</td>
</tr>
<tr class="odd">
<td>Number of singleton ASVs</td>
<td>NA</td>
</tr>
<tr class="even">
<td>Percent of ASVs that are singletons</td>
<td>NA</td>
</tr>
<tr class="odd">
<td>Number of sample variables are:</td>
<td>6 (SamName, PLOT, DEPTH, TREAT, TEMP, PAIR)</td>
</tr>
</tbody>
</table>
<p><strong>B</strong>. Add two final columns containing the ASV sequences and ASV IDs. This will be useful later when trying to export a fasta file. We can also take a look at the phyloseq object.</p>
<div class="cell">
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"Kingdom"</span>, <span class="st">"Phylum"</span>, <span class="st">"Class"</span>, <span class="st">"Order"</span>,
    <span class="st">"Family"</span>, <span class="st">"Genus"</span>, <span class="st">"ASV_SEQ"</span>, <span class="st">"ASV_ID"</span><span class="op">)</span>
<span class="va">ps</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20332 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20332 taxa by 8 taxonomic ranks ]</code></pre>
</div>
</div>
<p><strong>C</strong>. Export sequence and taxonomy tables for the unadulterated phyloseq object for later use. We will use the prefix <code>full</code> to indicate that these are the <em>raw</em> outputs.</p>
<div class="cell">
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>,
            <span class="st">"files/data-prep/tables/ssu18_full_tax_table.txt"</span>,
            sep <span class="op">=</span> <span class="st">"\t"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, col.names <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">)</span>,
            <span class="st">"files/data-prep/tables/ssu18_full_seq_table.txt"</span>,
            sep <span class="op">=</span> <span class="st">"\t"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, col.names <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="remove-contaminants-unwanted-taxa" class="level2"><h2 class="anchored" data-anchor-id="remove-contaminants-unwanted-taxa">Remove Contaminants &amp; Unwanted Taxa</h2>
<p>Let’s see if we have any potential contaminants. We can use some <a href="https://rmarkdown.rstudio.com/lesson-4.html">inline R code</a> to see the taxonomy table for any taxa of interest.</p>
<div class="callout-note callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The code (hidden by default) is written as <code>`r "Chloroplast" %in% tax_table(ps)`</code>.</p>
</div>
</div>
<ul>
<li>Are Mitochondria present? TRUE</li>
<li>Are Chloroplast present? TRUE</li>
<li>Are Eukaryota present? TRUE</li>
</ul>
<p>Let’s remove these taxa—Eukaryota because we used bacterial/archaeal primers, Mitochondria because those are likely from eukaryotes, and Chloroplast because those are likely from plants. We must do each of these in turn using phyloseq and it gets a little messy.</p>
<p>Why messy? The <code>subset_taxa</code> command removes anything that is <code>NA</code> for the specified taxonomic level or above. For example, lets say you run the <code>subset_taxa</code> command using <code>Family != "Mitochondria</code>“. Seems like you should get a phyloseq object with everything except Mitochondria. But actually the command not only gets rid of Mitochondria but everything else that has <code>NA</code> for Family and above. In my experience this is not well documented and I had to dig through the files to figure out what was happening.</p>
<p>Anyway, to remove the taxa we do the following:</p>
<ul>
<li>Subset the taxa and generate a <code>ps</code> object of just the taxa of interest,</li>
<li>Select the ASV column only, turn it into a factor, and use this to remove <insert taxa=""> from the <code>ps</code> object.</insert>
</li>
</ul>
<section id="remove-mitochondria-asvs" class="level3"><h3 class="anchored" data-anchor-id="remove-mitochondria-asvs">Remove Mitochondria ASVs</h3>
<p>Remember the original data set contained 20332 ASVs. Here we generate a file with mitochondria ASVs only.</p>
<div class="cell">
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">MT1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_taxa-methods.html">subset_taxa</a></span><span class="op">(</span><span class="va">ps</span>, <span class="va">Family</span> <span class="op">==</span> <span class="st">"Mitochondria"</span><span class="op">)</span>
<span class="va">MT1</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/methods/as.html">as</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">MT1</span><span class="op">)</span>, <span class="st">"matrix"</span><span class="op">)</span>
<span class="va">MT1</span> <span class="op">&lt;-</span> <span class="va">MT1</span><span class="op">[</span>, <span class="fl">8</span><span class="op">]</span>
<span class="va">MT1df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">MT1</span><span class="op">)</span>
<span class="va">goodTaxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_names-methods.html">taxa_names</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>, <span class="va">MT1df</span><span class="op">)</span>
<span class="va">ps_no_mito</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_taxa-methods.html">prune_taxa</a></span><span class="op">(</span><span class="va">goodTaxa</span>, <span class="va">ps</span><span class="op">)</span>
<span class="va">ps_no_mito</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20291 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20291 taxa by 8 taxonomic ranks ]</code></pre>
</div>
</div>
<p>Looks like this removed <strong>41 Mitochondria ASVs</strong>. We will duplicate the code block to remove other groups.</p>
</section><section id="remove-chloroplast-asvs" class="level3"><h3 class="anchored" data-anchor-id="remove-chloroplast-asvs">Remove Chloroplast ASVs</h3>
<p>And again with Chloroplast ASVs only.</p>
<div class="cell">
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">CH1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_taxa-methods.html">subset_taxa</a></span><span class="op">(</span><span class="va">ps_no_mito</span>, <span class="va">Order</span> <span class="op">==</span> <span class="st">"Chloroplast"</span><span class="op">)</span>
<span class="va">CH1</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/methods/as.html">as</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">CH1</span><span class="op">)</span>, <span class="st">"matrix"</span><span class="op">)</span>
<span class="va">CH1</span> <span class="op">&lt;-</span> <span class="va">CH1</span><span class="op">[</span>, <span class="fl">8</span><span class="op">]</span>
<span class="va">CH1df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">CH1</span><span class="op">)</span>
<span class="va">goodTaxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_names-methods.html">taxa_names</a></span><span class="op">(</span><span class="va">ps_no_mito</span><span class="op">)</span>, <span class="va">CH1df</span><span class="op">)</span>
<span class="va">ps_no_chloro</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_taxa-methods.html">prune_taxa</a></span><span class="op">(</span><span class="va">goodTaxa</span>, <span class="va">ps_no_mito</span><span class="op">)</span>
<span class="va">ps_no_chloro</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20274 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20274 taxa by 8 taxonomic ranks ]</code></pre>
</div>
</div>
<p>The code removed an additional <strong>17 Chloroplast ASVs</strong>.</p>
</section><section id="remove-eukaryota-asvs" class="level3"><h3 class="anchored" data-anchor-id="remove-eukaryota-asvs">Remove Eukaryota ASVs</h3>
<p>And again with Eukaryota ASVs only.</p>
<div class="cell">
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">EU1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_taxa-methods.html">subset_taxa</a></span><span class="op">(</span><span class="va">ps_no_chloro</span>, <span class="va">Kingdom</span> <span class="op">==</span> <span class="st">"Eukaryota"</span><span class="op">)</span>
<span class="va">EU1</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/methods/as.html">as</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">EU1</span><span class="op">)</span>, <span class="st">"matrix"</span><span class="op">)</span>
<span class="va">EU1</span> <span class="op">&lt;-</span> <span class="va">EU1</span><span class="op">[</span>, <span class="fl">8</span><span class="op">]</span>
<span class="va">EU1df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">EU1</span><span class="op">)</span>
<span class="va">goodTaxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_names-methods.html">taxa_names</a></span><span class="op">(</span><span class="va">ps_no_chloro</span><span class="op">)</span>, <span class="va">EU1df</span><span class="op">)</span>
<span class="va">ps_no_euk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_taxa-methods.html">prune_taxa</a></span><span class="op">(</span><span class="va">goodTaxa</span>, <span class="va">ps_no_chloro</span><span class="op">)</span>
<span class="va">ps_no_euk</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20223 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20223 taxa by 8 taxonomic ranks ]</code></pre>
</div>
</div>
<p>The code removed an additional <strong>51 Eukaryota ASVs</strong> from the <code>ps</code> object.</p>
</section><section id="remove-any-kingdom-nas" class="level3"><h3 class="anchored" data-anchor-id="remove-any-kingdom-nas">Remove any Kingdom NAs</h3>
<p>Here we can just use the straight up <code>subset_taxa</code> command since we do not need to worry about any ranks above Kingdom also being removed.</p>
<div class="cell">
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ps_filt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_taxa-methods.html">subset_taxa</a></span><span class="op">(</span><span class="va">ps_no_euk</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">Kingdom</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ps_work_o</span> <span class="op">&lt;-</span> <span class="va">ps_filt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The code eliminated an additional <strong>50 Kingdom level NA ASVs</strong> from the phyloseq object.</p>
</section></section><section id="rename-na-taxonomic-ranks" class="level2"><h2 class="anchored" data-anchor-id="rename-na-taxonomic-ranks">Rename NA taxonomic ranks</h2>
<p>Phyloseq has an odd way of dealing with taxonomic ranks that have no value—in other words, <strong>NA</strong> in the tax table. The first thing we are going to do before moving forward is to change all of the <em>NA</em>s to have a value of the next highest classified rank. For example, <code>ASV26</code> is not classified at the Genus level but is at Family level (Xanthobacteraceae). So we change the Genus name to <em>Family_Xanthobacteraceae</em>. The code for comes from these two posts on the phyloseq GitHub, both by <a href="https://github.com/MSMortensen">MSMortensen</a>: issue <a href="https://github.com/joey711/phyloseq/issues/850#issuecomment-394771087">#850</a> and issue <a href="https://github.com/joey711/phyloseq/issues/990#issuecomment-424618425">#990</a>.</p>
<blockquote class="blockquote">
<p>One thing this code does is reassign the functions <code>class</code> and <code>order</code> to taxonomic ranks. This can cause issues if you need these functions.</p>
</blockquote>
<p>So you need to run something like this <code>rm(class, order, phylum, kingdom)</code> at the end of the code to remove these as variables. For now, I have not come up with a better solution.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tax.clean</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps_filt</span><span class="op">)</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">)</span><span class="op">{</span> <span class="va">tax.clean</span><span class="op">[</span>,<span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">tax.clean</span><span class="op">[</span>,<span class="va">i</span><span class="op">]</span><span class="op">)</span><span class="op">}</span>
<span class="va">tax.clean</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">tax.clean</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">""</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html">nrow</a></span><span class="op">(</span><span class="va">tax.clean</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
    <span class="kw">if</span> <span class="op">(</span><span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">==</span> <span class="st">""</span><span class="op">)</span><span class="op">{</span>
        <span class="va">kingdom</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"k_"</span>, <span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">1</span><span class="op">]</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
        <span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>, <span class="fl">2</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">kingdom</span>
    <span class="op">}</span> <span class="kw">else</span> <span class="kw">if</span> <span class="op">(</span><span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">3</span><span class="op">]</span> <span class="op">==</span> <span class="st">""</span><span class="op">)</span><span class="op">{</span>
        <span class="va">phylum</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"p_"</span>, <span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">2</span><span class="op">]</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
        <span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>, <span class="fl">3</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">phylum</span>
    <span class="op">}</span> <span class="kw">else</span> <span class="kw">if</span> <span class="op">(</span><span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">4</span><span class="op">]</span> <span class="op">==</span> <span class="st">""</span><span class="op">)</span><span class="op">{</span>
        <span class="va">class</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"c_"</span>, <span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">3</span><span class="op">]</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
        <span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>, <span class="fl">4</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">class</span>
    <span class="op">}</span> <span class="kw">else</span> <span class="kw">if</span> <span class="op">(</span><span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">5</span><span class="op">]</span> <span class="op">==</span> <span class="st">""</span><span class="op">)</span><span class="op">{</span>
        <span class="va">order</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"o_"</span>, <span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">4</span><span class="op">]</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
        <span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>, <span class="fl">5</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">order</span>
    <span class="op">}</span> <span class="kw">else</span> <span class="kw">if</span> <span class="op">(</span><span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">6</span><span class="op">]</span> <span class="op">==</span> <span class="st">""</span><span class="op">)</span><span class="op">{</span>
        <span class="va">tax.clean</span><span class="op">$</span><span class="va">Genus</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"f"</span>,<span class="va">tax.clean</span><span class="op">$</span><span class="va">Family</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span>
        <span class="op">}</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps_filt</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">tax.clean</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/rank_names.html">rank_names</a></span><span class="op">(</span><span class="va">ps_filt</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">class</span>, <span class="va">order</span>, <span class="va">phylum</span>, <span class="va">kingdom</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Still the same ranks. That’s good. What about the new groups? Let’s take a peak at some families.</p>
<div class="cell">
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/get_taxa_unique.html">get_taxa_unique</a></span><span class="op">(</span><span class="va">ps_filt</span>, <span class="st">"Family"</span><span class="op">)</span>, <span class="fl">16</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Planococcaceae"         "Chthoniobacteraceae"    "Nitrososphaeraceae"    
 [4] "p_RCP2-54"              "o_Rokubacteriales"      "Methyloligellaceae"    
 [7] "Nitrospiraceae"         "Xanthobacteraceae"      "o_Vicinamibacterales"  
[10] "Vicinamibacteraceae"    "c_Subgroup_5"           "Xiphinematobacteraceae"
[13] "Entotheonellaceae"      "Chitinophagaceae"       "Nitrosomonadaceae"     
[16] "o_Acidobacteriales"    </code></pre>
</div>
</div>
<p>Nice. Bye-bye <em>NA</em></p>
<blockquote class="blockquote">
<p><em>Note</em>. The original code mentioned above is written for data sets that have species-level designations.</p>
</blockquote>
<p>Since this data set does not contain species, I modified the code to stop at the genus level. If your data set has species, my modifications will not work for you.</p>
<p>Finally, we rename the ps object. This is now our working data set.</p>
<div class="cell">
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ps_work</span> <span class="op">&lt;-</span> <span class="va">ps_filt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="add-phylogenetic-tree" class="level2"><h2 class="anchored" data-anchor-id="add-phylogenetic-tree">Add Phylogenetic Tree</h2>
<p>One final task is to add a phylogenetic tree to the phyloseq object.</p>
<div class="cell">
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ps_work_tree</span> <span class="op">&lt;-</span> <span class="fu">rtree</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/ntaxa-methods.html">ntaxa</a></span><span class="op">(</span><span class="va">ps_work</span><span class="op">)</span>, rooted <span class="op">=</span> <span class="cn">TRUE</span>,
                      tip.label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_names-methods.html">taxa_names</a></span><span class="op">(</span><span class="va">ps_work</span><span class="op">)</span><span class="op">)</span>
<span class="va">ps_work</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/merge_phyloseq.html">merge_phyloseq</a></span><span class="op">(</span><span class="va">ps_work</span>,
                          <span class="va">sample_data</span>,
                          <span class="va">ps_work_tree</span><span class="op">)</span>
<span class="va">ps_work_o_tree</span> <span class="op">&lt;-</span> <span class="fu">rtree</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/ntaxa-methods.html">ntaxa</a></span><span class="op">(</span><span class="va">ps_work_o</span><span class="op">)</span>, rooted <span class="op">=</span> <span class="cn">TRUE</span>,
                      tip.label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_names-methods.html">taxa_names</a></span><span class="op">(</span><span class="va">ps_work_o</span><span class="op">)</span><span class="op">)</span>
<span class="va">ps_work_o</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/merge_phyloseq.html">merge_phyloseq</a></span><span class="op">(</span><span class="va">ps_work_o</span>,
                          <span class="va">sample_data</span>,
                          <span class="va">ps_work_o_tree</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="data-set-summary" class="level2"><h2 class="anchored" data-anchor-id="data-set-summary">Data Set Summary</h2>
<p>Now that we have removed the NC and sample with zero reads, we can summarize the data set. Let’s start with the in our working phyloseq object using the <code>summarize_phyloseq</code> from the <a href="https://github.com/microbiome/microbiome/">microbiome R package</a> <span class="citation" data-cites="lahti2017microbiome">(<a href="#ref-lahti2017microbiome" role="doc-biblioref">Lahti, Sudarshan, et al. 2017</a>)</span> as we did above.</p>
<div class="cell">

</div>
<table class="table">
<colgroup>
<col style="width: 38%">
<col style="width: 61%">
</colgroup>
<thead><tr class="header">
<th>Metric</th>
<th>Results</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>25088</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>86600</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>936640</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>62443</td>
</tr>
<tr class="odd">
<td>Median number of reads</td>
<td>62117</td>
</tr>
<tr class="even">
<td>Total ASVs</td>
<td>20173</td>
</tr>
<tr class="odd">
<td>Sparsity</td>
<td>0.891</td>
</tr>
<tr class="even">
<td>Any ASVs sum to 1 or less?</td>
<td>FALSE</td>
</tr>
<tr class="odd">
<td>Number of singleton ASVs</td>
<td>NA</td>
</tr>
<tr class="even">
<td>Percent of ASVs that are singletons</td>
<td>NA</td>
</tr>
<tr class="odd">
<td>Number of sample variables are:</td>
<td>6 (SamName, PLOT, DEPTH, TREAT, TEMP, PAIR)</td>
</tr>
</tbody>
</table>
<p>We can also generate a summary table of total reads &amp; ASVs for each sample. You can sort the table, download a copy, or filter by search term. Here is the code to generate the data for the table. First, we create data frames that hold total reads and ASVs for each sample.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">total_reads</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_sums.html">sample_sums</a></span><span class="op">(</span><span class="va">ps_work</span><span class="op">)</span>
<span class="va">total_reads</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">total_reads</span>, make.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">total_reads</span> <span class="op">&lt;-</span> <span class="va">total_reads</span> <span class="op">%&gt;%</span> <span class="fu">rownames_to_column</span><span class="op">(</span><span class="st">"Sample_ID"</span><span class="op">)</span>

<span class="va">total_asvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/estimate_richness.html">estimate_richness</a></span><span class="op">(</span><span class="va">ps_work</span>,
                                measures <span class="op">=</span> <span class="st">"Observed"</span><span class="op">)</span>
<span class="va">total_asvs</span> <span class="op">&lt;-</span> <span class="va">total_asvs</span> <span class="op">%&gt;%</span> <span class="fu">rownames_to_column</span><span class="op">(</span><span class="st">"Sample_ID"</span><span class="op">)</span>
<span class="va">total_asvs</span><span class="op">$</span><span class="va">Sample_ID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">'\\.'</span>, <span class="st">'-'</span>, <span class="va">total_asvs</span><span class="op">$</span><span class="va">Sample_ID</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>And then we merge these two data frames with the sample data frame.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">sam_details</span> <span class="op">&lt;-</span> <span class="va">samdf</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">sam_details</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">sam_details</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"Sample_ID"</span>, <span class="st">"Plot"</span>, <span class="st">"Depth"</span>,
                                  <span class="st">"Treatment"</span>, <span class="st">"Temp"</span>, <span class="st">"Pair"</span><span class="op">)</span>
<span class="va">tmp_tab1</span> <span class="op">&lt;-</span> <span class="va">seq_table</span>
<span class="va">tmp_tab1</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="fl">2</span><span class="op">:</span><span class="fl">7</span>,<span class="fl">11</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">tmp_tab2</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">sam_details</span>, <span class="va">total_reads</span><span class="op">)</span> <span class="op">%&gt;%</span>
                <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">total_asvs</span><span class="op">)</span> <span class="op">%&gt;%</span>
                <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">tmp_tab1</span><span class="op">)</span>

<span class="va">reads_lost</span> <span class="op">&lt;-</span> <span class="va">tmp_tab2</span><span class="op">$</span><span class="va">`final reads`</span> <span class="op">-</span> <span class="va">tmp_tab2</span><span class="op">$</span><span class="va">total_reads</span> 
<span class="va">asvs_lost</span> <span class="op">&lt;-</span> <span class="va">tmp_tab2</span><span class="op">$</span><span class="va">`no. ASVs`</span> <span class="op">-</span> <span class="va">tmp_tab2</span><span class="op">$</span><span class="va">Observed</span> 
<span class="va">tmp_tab2</span><span class="op">$</span><span class="va">reads_lost</span> <span class="op">&lt;-</span> <span class="va">reads_lost</span>
<span class="va">tmp_tab2</span><span class="op">$</span><span class="va">asvs_lost</span> <span class="op">&lt;-</span> <span class="va">asvs_lost</span>

<span class="va">seq_table2</span> <span class="op">&lt;-</span> <span class="va">tmp_tab2</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span>,<span class="fl">12</span>,<span class="fl">13</span><span class="op">)</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">seq_table2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"sample_id"</span>, <span class="st">"PLOT"</span>, <span class="st">"DEPTH"</span>,
                          <span class="st">"TREAT"</span>, <span class="st">"TEMP"</span>, <span class="st">"PAIR"</span>, <span class="st">"total_reads"</span>,
                          <span class="st">"total_ASVs"</span>, <span class="st">"reads_lost"</span>, <span class="st">"ASVs_lost"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><small><strong>(16S rRNA) Table 2 | Sample summary table showing the final number of reads and ASVs (per sample) as well as the number of reads/ASVs lost during curation.</strong></small></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,c2FtcGxlX2lkLFBMT1QsREVQVEgsVFJFQVQsVEVNUCxQQUlSLHRvdGFsX3JlYWRzLHRvdGFsX0FTVnMscmVhZHNfbG9zdCxBU1ZzX2xvc3QKUDAxX0QwMF8wMTBfVzNBLFAwMSwwMF8wMTAsVywzLEEsNjc5MDIsMjM3MSwxMzUsMTIKUDAxX0QwMF8wMTBfVzhBLFAwMSwwMF8wMTAsVyw4LEEsNjA1MDksMTYxMCwyNiw4ClAxMF9EMDBfMDEwX0MwRSxQMTAsMDBfMDEwLEMsMCxFLDUzNjAwLDIwNjYsMjUsNApQMDJfRDAwXzAxMF9DMEEsUDAyLDAwXzAxMCxDLDAsQSw3MDQxMiwyMjcyLDI0LDYKUDAzX0QwMF8wMTBfVzNCLFAwMywwMF8wMTAsVywzLEIsNzUxMjMsMjM1Niw2MCwxMgpQMDNfRDAwXzAxMF9XOEIsUDAzLDAwXzAxMCxXLDgsQiw1MzM0NywxNTc1LDkwLDE4ClAwNF9EMDBfMDEwX0MwQixQMDQsMDBfMDEwLEMsMCxCLDUzNzcwLDIwNDAsNDMsMTIKUDA1X0QwMF8wMTBfVzNDLFAwNSwwMF8wMTAsVywzLEMsNDcwMjMsMTY2Myw0Nyw5ClAwNV9EMDBfMDEwX1c4QyxQMDUsMDBfMDEwLFcsOCxDLDg2NjAwLDIxMjQsMjQxLDMyClAwNl9EMDBfMDEwX0MwQyxQMDYsMDBfMDEwLEMsMCxDLDgxMTUwLDMwNjUsNjksMTcKUDA3X0QwMF8wMTBfVzNELFAwNywwMF8wMTAsVywzLEQsNjE2MzIsMjAxMCwxMTAsMTUKUDA3X0QwMF8wMTBfVzhELFAwNywwMF8wMTAsVyw4LEQsNjIxMTcsMjE1Nyw4NSwxMApQMDhfRDAwXzAxMF9DMEQsUDA4LDAwXzAxMCxDLDAsRCw2MjEzNiwyNDQxLDM1LDEyClAwOV9EMDBfMDEwX1czRSxQMDksMDBfMDEwLFcsMyxFLDc2MjMxLDIzODYsNjgsMTYKUDA5X0QwMF8wMTBfVzhFLFAwOSwwMF8wMTAsVyw4LEUsMjUwODgsODEzLDYzLDMK').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'ssu18_sample_summary_final.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="cell-output-display">

<div id="htmlwidget-b6e7351c34343571148c" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-b6e7351c34343571148c">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"sample_id":["P01_D00_010_W3A","P01_D00_010_W8A","P10_D00_010_C0E","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E"],"PLOT":["P01","P01","P10","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09"],"DEPTH":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"TREAT":["W","W","C","C","W","W","C","W","W","C","W","W","C","W","W"],"TEMP":["3","8","0","0","3","8","0","3","8","0","3","8","0","3","8"],"PAIR":["A","A","E","A","B","B","B","C","C","C","D","D","D","E","E"],"total_reads":[67902,60509,53600,70412,75123,53347,53770,47023,86600,81150,61632,62117,62136,76231,25088],"total_ASVs":[2371,1610,2066,2272,2356,1575,2040,1663,2124,3065,2010,2157,2441,2386,813],"reads_lost":[135,26,25,24,60,90,43,47,241,69,110,85,35,68,63],"ASVs_lost":[12,8,4,6,12,18,12,9,32,17,15,10,12,16,3]},"columns":[{"accessor":"sample_id","name":"sample_id","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P10_D00_010_C0E","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E"],"header":"sample id","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"PLOT","name":"PLOT","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01","P01","P10","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09"],"header":"PLOT","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"DEPTH","name":"DEPTH","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"header":"DEPTH","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"TREAT","name":"TREAT","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["W","W","C","C","W","W","C","W","W","C","W","W","C","W","W"],"header":"TREAT","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"TEMP","name":"TEMP","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["3","8","0","0","3","8","0","3","8","0","3","8","0","3","8"],"header":"TEMP","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"PAIR","name":"PAIR","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["A","A","E","A","B","B","B","C","C","C","D","D","D","E","E"],"header":"PAIR","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"total_reads","name":"total_reads","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["67902","60509","53600","70412","75123","53347","53770","47023","86600","81150","61632","62117","62136","76231","25088"],"header":"total reads","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"total_ASVs","name":"total_ASVs","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["2371","1610","2066","2272","2356","1575","2040","1663","2124","3065","2010","2157","2441","2386","813"],"header":"total ASVs","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"reads_lost","name":"reads_lost","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["135","26","25","24","60","90","43","47","241","69","110","85","35","68","63"],"header":"reads lost","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"ASVs_lost","name":"ASVs_lost","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["12","8","4","6","12","18","12","9","32","17","15","10","12","16","3"],"header":"ASVs lost","align":"center","footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"551a45caf403b7a1a1de2911a76de0c1"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section><section id="review" class="level2"><h2 class="anchored" data-anchor-id="review">Review</h2>
<p>We have a few different options to play with. At this point it is difficult to say which we will use or whether additional objects need to be created. Here is a summary of the objects we have.</p>
<p>The full phyloseq object <em>before</em> removing contaminants.</p>
<div class="cell">
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ps</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20332 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20332 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/get_taxa_unique.html">get_taxa_unique</a></span><span class="op">(</span><span class="va">ps</span>, <span class="st">"Family"</span><span class="op">)</span>, <span class="fl">16</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Planococcaceae"         "Chthoniobacteraceae"    "Nitrososphaeraceae"    
 [4] NA                       "Methyloligellaceae"     "Nitrospiraceae"        
 [7] "Xanthobacteraceae"      "Vicinamibacteraceae"    "Xiphinematobacteraceae"
[10] "Entotheonellaceae"      "Chitinophagaceae"       "Nitrosomonadaceae"     
[13] "B1-7BS"                 "WX65"                   "Gaiellaceae"           
[16] "Steroidobacteraceae"   </code></pre>
</div>
</div>
<p>The phyloseq object <em>before</em> changing NA ranks.</p>
<div class="cell">
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ps_work_o</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20173 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20173 taxa by 8 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 20173 tips and 20172 internal nodes ]</code></pre>
</div>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/get_taxa_unique.html">get_taxa_unique</a></span><span class="op">(</span><span class="va">ps_work_o</span>, <span class="st">"Family"</span><span class="op">)</span>, <span class="fl">16</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "A4b"                            "Burkholderiaceae"              
 [3] "Peptococcaceae"                 NA                              
 [5] "Polyangiaceae"                  "Pedosphaeraceae"               
 [7] "AKYH767"                        "Micromonosporaceae"            
 [9] "Acidobacteriaceae_(Subgroup_1)" "Pyrinomonadaceae"              
[11] "Clostridiaceae"                 "Myxococcaceae"                 
[13] "Hyphomonadaceae"                "Roseiflexaceae"                
[15] "type_III"                       "Ktedonobacteraceae"            </code></pre>
</div>
</div>
<p>The phyloseq object <em>after</em> changing NA ranks.</p>
<div class="cell">
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ps_work</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20173 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20173 taxa by 8 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 20173 tips and 20172 internal nodes ]</code></pre>
</div>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/get_taxa_unique.html">get_taxa_unique</a></span><span class="op">(</span><span class="va">ps_work</span>, <span class="st">"Family"</span><span class="op">)</span>, <span class="fl">16</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "c_Subgroup_22"         "c_OM190"               "A4b"                  
 [4] "Diplorickettsiaceae"   "k_Bacteria"            "Gemmataceae"          
 [7] "c_Gammaproteobacteria" "Microscillaceae"       "Entotheonellaceae"    
[10] "Syntrophaceae"         "p_WS2"                 "Verrucomicrobiaceae"  
[13] "Pyrinomonadaceae"      "Caulobacteraceae"      "Parachlamydiaceae"    
[16] "c_Phycisphaerae"      </code></pre>
</div>
</div>
</section><section id="final-steps" class="level2"><h2 class="anchored" data-anchor-id="final-steps">Final Steps</h2>
<p>The last thing tasks to complete are to <strong>a</strong>) save copies of the taxonomy and sequence tables and <strong>b</strong>) save an image of the workflow for the next section.</p>
<div class="cell">
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps_work</span><span class="op">)</span>,
            <span class="st">"files/data-prep/tables/ssu18_work_tax_table.txt"</span>,
            sep <span class="op">=</span> <span class="st">"\t"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, col.names <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">ps_work</span><span class="op">)</span><span class="op">)</span>,
            <span class="st">"files/data-prep/tables/ssu18_work_seq_table.txt"</span>,
            sep <span class="op">=</span> <span class="st">"\t"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, col.names <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">

</div>
<div class="cell">

</div>
</section></section><section id="its-data" class="level1"><h1>ITS Data</h1>
<div class="cell">

</div>
<div class="cell">

</div>
<section id="prerequisites-1" class="level2"><h2 class="anchored" data-anchor-id="prerequisites-1">Prerequisites</h2>
<p>In order to run this workflow, you either need to run the corresponding <a href="dada2-high-temp.html#its-data">DADA2 Workflow for 2018 ITS</a> or begin with the output from that workflow, <code>data_prep_its18_wf.rdata</code>. See the <a href="data-availability.html">Data Availability</a> page for complete details.</p>
<p>Unless otherwise noted, we primarily use <a href="https://joey711.github.io/phyloseq/">phyloseq</a> <span class="citation" data-cites="mcmurdie2013phyloseq">(<a href="#ref-mcmurdie2013phyloseq" role="doc-biblioref">McMurdie and Holmes 2013</a>)</span> in this section of the workflow to prepare the 2018 ITS data set for community analyses. We prepare the data set by curating samples, removing contaminants, and creating phyloseq objects.</p>
</section><section id="read-counts-assessment-1" class="level2"><h2 class="anchored" data-anchor-id="read-counts-assessment-1">Read Counts Assessment</h2>
<p>Before we begin, let’s create a summary table containing some basic sample metadata and the read count data from the <a href="dada2-high-temp.html#its-data">DADA2 workflow</a>. We need to inspect how total reads changed from to beginning to the end of the workflow. While we are at it, let’s create more intuitive Sample IDs. For more details on how reads changed at each step of the workflow, see the <a href="dada2-high-temp.html#track-reads-through-workflow-1">summary table</a> at the end of the DADA2 section.</p>
<p>Table headers are as follows:</p>
<table class="table">
<colgroup>
<col style="width: 23%">
<col style="width: 76%">
</colgroup>
<thead><tr class="header">
<th>Header</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>Sample ID</code></td>
<td>the new sample ID based on Plot number, Depth, Treatment, Temperature, &amp; Pair</td>
</tr>
<tr class="even">
<td><code>Plot</code></td>
<td>the experimental plot number</td>
</tr>
<tr class="odd">
<td><code>Depth</code></td>
<td>the depth where the sample was collected</td>
</tr>
<tr class="even">
<td><code>Treatment</code></td>
<td>warm or control</td>
</tr>
<tr class="odd">
<td><code>Temp</code></td>
<td>temperature of soil heating (0C, +4C, or +8C)</td>
</tr>
<tr class="even">
<td><code>Pair</code></td>
<td>control/treatment coupling</td>
</tr>
<tr class="odd">
<td><code>input</code></td>
<td>number of raw reads</td>
</tr>
<tr class="even">
<td><code>nochim</code></td>
<td>final read count after removing chimeras</td>
</tr>
<tr class="odd">
<td><code>Remain</code></td>
<td>percent of reads remaining from <code>input</code> to <code>nonchim</code>
</td>
</tr>
<tr class="even">
<td><code>FastqID</code></td>
<td>base name of the fastq file</td>
</tr>
</tbody>
</table>
<div class="cell">

</div>
<p><small><strong>(ITS) Table 1 | Sample summary table including read changes at start and end of DADA2 workflow.</strong></small></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,U2FtcGxlX0lELHJvb3QgbmFtZSxQbG90LFRyZWF0bWVudCxUZW1wLERlcHRoLFBhaXIsaW5wdXQgcmVhZHMsZmluYWwgcmVhZHMsbm8uIEFTVnMscGVyLiByZWFkcyByZW1haW4KUDAxX0QwMF8wMTBfVzNBLFAxLFAwMSxXYXJtLDMsMDBfMDEwLEEsNTA4OTQsMjk3OTQsOTQ1LDAuNTg1ClAwMV9EMDBfMDEwX1c4QSxQMS04QyxQMDEsV2FybSw4LDAwXzAxMCxBLDcwNzM0LDQxODgwLDcxOSwwLjU5MgpQMTBfRDAwXzAxMF9DMEUsUDEwLFAxMCxDb250cm9sLDAsMDBfMDEwLEUsOTY1MzEsNjQ2MzYsODEyLDAuNjcKUDAyX0QwMF8wMTBfQzBBLFAyLFAwMixDb250cm9sLDAsMDBfMDEwLEEsNzQ1ODEsNTExNDQsMTAxMSwwLjY4NgpQMDNfRDAwXzAxMF9XM0IsUDMsUDAzLFdhcm0sMywwMF8wMTAsQiw2MDAwNiwzODM3NCw3NjUsMC42NApQMDNfRDAwXzAxMF9XOEIsUDMtOEMsUDAzLFdhcm0sOCwwMF8wMTAsQiwzMTE3NCwxNDE4OCwzNDksMC40NTUKUDA0X0QwMF8wMTBfQzBCLFA0LFAwNCxDb250cm9sLDAsMDBfMDEwLEIsODg3MzIsNTY1OTMsMTAxNywwLjYzOApQMDVfRDAwXzAxMF9XM0MsUDUsUDA1LFdhcm0sMywwMF8wMTAsQyw4MDksMjk2LDk1LDAuMzY2ClAwNV9EMDBfMDEwX1c4QyxQNS04QyxQMDUsV2FybSw4LDAwXzAxMCxDLDgxOTA4LDM1NjAwLDYxNiwwLjQzNQpQMDZfRDAwXzAxMF9DMEMsUDYsUDA2LENvbnRyb2wsMCwwMF8wMTAsQyw2NzQxOSw0NTg1MCw5NTQsMC42OApQMDdfRDAwXzAxMF9XM0QsUDcsUDA3LFdhcm0sMywwMF8wMTAsRCw3MDEyMywzNjA2Myw4NjcsMC41MTQKUDA3X0QwMF8wMTBfVzhELFA3LThDLFAwNyxXYXJtLDgsMDBfMDEwLEQsMTU0MzUsOTE3MiwzMzUsMC41OTQKUDA4X0QwMF8wMTBfQzBELFA4LFAwOCxDb250cm9sLDAsMDBfMDEwLEQsNjA3OTksMzgzNzEsNzAyLDAuNjMxClAwOV9EMDBfMDEwX1czRSxQOSxQMDksV2FybSwzLDAwXzAxMCxFLDU1OTI3LDI5MTAyLDc0NSwwLjUyClAwOV9EMDBfMDEwX1c4RSxQOS04QyxQMDksV2FybSw4LDAwXzAxMCxFLDE0MTgsODAsMTQsMC4wNTYK').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'its18_sample_summary.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="cell-output-display">

<div id="htmlwidget-87ebba9d4a5eaba8f1df" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-87ebba9d4a5eaba8f1df">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Sample_ID":["P01_D00_010_W3A","P01_D00_010_W8A","P10_D00_010_C0E","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E"],"root name":["P1","P1-8C","P10","P2","P3","P3-8C","P4","P5","P5-8C","P6","P7","P7-8C","P8","P9","P9-8C"],"Plot":["P01","P01","P10","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09"],"Treatment":["Warm","Warm","Control","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm"],"Temp":[3,8,0,0,3,8,0,3,8,0,3,8,0,3,8],"Depth":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"Pair":["A","A","E","A","B","B","B","C","C","C","D","D","D","E","E"],"input reads":[50894,70734,96531,74581,60006,31174,88732,809,81908,67419,70123,15435,60799,55927,1418],"final reads":[29794,41880,64636,51144,38374,14188,56593,296,35600,45850,36063,9172,38371,29102,80],"no. ASVs":[945,719,812,1011,765,349,1017,95,616,954,867,335,702,745,14],"per. reads remain":[0.585,0.592,0.67,0.686,0.64,0.455,0.638,0.366,0.435,0.68,0.514,0.594,0.631,0.52,0.056]},"columns":[{"accessor":"Sample_ID","name":"Sample ID","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P10_D00_010_C0E","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E"],"header":"Sample ID","align":"left","footerStyle":{"fontWeight":"bold"},"footer":"Total reads","minWidth":150,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"root name","name":"root name","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P1","P1-8C","P10","P2","P3","P3-8C","P4","P5","P5-8C","P6","P7","P7-8C","P8","P9","P9-8C"],"header":"root name","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Plot","name":"Plot","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01","P01","P10","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09"],"header":"Plot","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Treatment","name":"Treatment","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["Warm","Warm","Control","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm","Control","Warm","Warm"],"header":"Treatment","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Temp","name":"Temp","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["3","8","0","0","3","8","0","3","8","0","3","8","0","3","8"],"header":"Temp","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Depth","name":"Depth","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"header":"Depth","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"Pair","name":"Pair","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["A","A","E","A","B","B","B","C","C","C","D","D","D","E","E"],"header":"Pair","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"input reads","name":"input reads","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["50894","70734","96531","74581","60006","31174","88732","809","81908","67419","70123","15435","60799","55927","1418"],"header":"input reads","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"final reads","name":"final reads","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["29794","41880","64636","51144","38374","14188","56593","296","35600","45850","36063","9172","38371","29102","80"],"header":"final reads","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"no. ASVs","name":"no. ASVs","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["945","719","812","1011","765","349","1017","95","616","954","867","335","702","745","14"],"header":"no. ASVs","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"per. reads remain","name":"per. reads remain","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.585","0.592","0.67","0.686","0.64","0.455","0.638","0.366","0.435","0.68","0.514","0.594","0.631","0.52","0.056"],"header":"per. reads remain","align":"center","footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"97e38721ca4829afb80720154cf9d5a7"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>The following samples have fewer than <strong>1000</strong> reads at the end of the workflow:</p>
<p><strong>P05_D00_010_W3C, P09_D00_010_W8E</strong>.</p>
<p>We can also plot the final read count by the temperature where the sample was collected.</p>
<div class="cell">

</div>
</section><section id="defining-groups-1" class="level2"><h2 class="anchored" data-anchor-id="defining-groups-1">Defining Groups</h2>
<ol type="1">
<li><p>Load the data packet produced in the final step of the DADA2 workflow. This packet (<code>its18_dada2_wf.rdata</code>) contains the ASV-by-sample table and the ASV taxonomy table.</p></li>
<li><p>Rename the samples so names have plot and Depth info. For convenience, we renamed any Negative Controls (NC) to <code>P00_D00-000_NNN</code>.</p></li>
<li><p>After we load the data packet we next need to format sample names and define groups.</p></li>
</ol>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb37"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">seq_table</span> <span class="op">&lt;-</span> <span class="va">seq_table</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html">order</a></span><span class="op">(</span><span class="va">seq_table</span><span class="op">$</span><span class="va">`root name`</span><span class="op">)</span>, <span class="op">]</span>
<span class="va">tmp_names</span> <span class="op">&lt;-</span> <span class="va">seq_table</span><span class="op">$</span><span class="va">`Sample_ID`</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"files/dada2/rdata/its18_dada2_wf.rdata"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span><span class="op">(</span><span class="va">seq_table</span><span class="op">$</span><span class="va">`root name`</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">seqtab</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">seqtab</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">tmp_names</span>
<span class="va">samples.out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">seqtab</span><span class="op">)</span>
<span class="va">sample_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span><span class="op">(</span><span class="va">samples.out</span>, <span class="fl">1</span>, <span class="fl">999</span><span class="op">)</span>
<span class="va">plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span><span class="op">(</span><span class="va">samples.out</span>, <span class="fl">0</span>, <span class="fl">3</span><span class="op">)</span>
<span class="va">depth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span><span class="op">(</span><span class="va">samples.out</span>, <span class="fl">6</span>, <span class="fl">11</span><span class="op">)</span>
<span class="va">treatment</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span><span class="op">(</span><span class="va">samples.out</span>, <span class="fl">13</span>, <span class="fl">13</span><span class="op">)</span>
<span class="va">temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span><span class="op">(</span><span class="va">samples.out</span>, <span class="fl">14</span>, <span class="fl">14</span><span class="op">)</span>
<span class="va">pair</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span><span class="op">(</span><span class="va">samples.out</span>, <span class="fl">15</span>, <span class="fl">15</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We have a total of <strong>15</strong> samples, <strong>5</strong> sample pairs, from <strong>10</strong> plots (<strong>1</strong> depth). There are <strong>2</strong> treatments corresponding to <strong>3</strong> different temperature regimes.</p>
<ol start="4" type="1">
<li>And finally we define a sample data frame that holds the different groups we extracted from the sample names.</li>
</ol>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb38"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">samdf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>SamName <span class="op">=</span> <span class="va">sample_name</span>,
                    PLOT <span class="op">=</span> <span class="va">plot</span>,
                    DEPTH <span class="op">=</span> <span class="va">depth</span>,
                    TREAT <span class="op">=</span> <span class="va">treatment</span>,
                    TEMP <span class="op">=</span> <span class="va">temp</span>,
                    PAIR <span class="op">=</span> <span class="va">pair</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">samdf</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">samples.out</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">

</div>
<p>For example, P08_D00_010_C0D is the sample heated at <strong>+0C</strong> from plot <strong>P08</strong>, collected from a depth of 00_010cm. This sample is in group <strong>D</strong> which consists of P07_D00_010_W3D, P07_D00_010_W8D, P08_D00_010_C0D.</p>
</section><section id="create-a-phyloseq-object-1" class="level2"><h2 class="anchored" data-anchor-id="create-a-phyloseq-object-1">Create a Phyloseq Object</h2>
<section id="fix-taxon-names" class="level3"><h3 class="anchored" data-anchor-id="fix-taxon-names">Fix Taxon Names</h3>
<p>The UNITE database we used for classification adds a little prefix to every rank call, like <code>k__</code> for Kingdom, <code>o__</code> for Order, etc. To keep in step with our other analyses, we will remove these tags. Of course, we also make a ps object with the original names just in case. At this point we will also remove the Species rank since it is essentially useless.</p>
<div class="cell">
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tax_mod_o</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">tax</span><span class="op">)</span>
<span class="va">tax_mod_o</span><span class="op">$</span><span class="va">Species</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">tax_mod_o</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">tax_mod_o</span><span class="op">)</span>
<span class="va">tax_mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"k__"</span>, <span class="st">""</span>, <span class="va">tax_mod_o</span><span class="op">)</span>
<span class="va">tax_mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"p__"</span>, <span class="st">""</span>, <span class="va">tax_mod</span><span class="op">)</span>
<span class="va">tax_mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"c__"</span>, <span class="st">""</span>, <span class="va">tax_mod</span><span class="op">)</span>
<span class="va">tax_mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"o__"</span>, <span class="st">""</span>, <span class="va">tax_mod</span><span class="op">)</span>
<span class="va">tax_mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"f__"</span>, <span class="st">""</span>, <span class="va">tax_mod</span><span class="op">)</span>
<span class="va">tax_mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"g__"</span>, <span class="st">""</span>, <span class="va">tax_mod</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>A</strong>. The first step is to rename the amplicon sequence variants (ASVs) so the designations are a bit more user friendly. By default, DADA2 names each ASV by its unique sequence so that data can be directly compared across studies (which is great). But this convention can get cumbersome downstream, so we rename the ASVs using a simpler convention—ASV1, ASV2, ASV3, and so on.</p>
<div class="callout-note callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>A phyloseq object contains ASV table (taxa abundances), sample metadata, and taxonomy table (mapping between ASVs and higher-level taxonomic classifications).</p>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb40"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="co"># this create the phyloseq object</span>
<span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phyloseq.html">phyloseq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">seqtab</span>, taxa_are_rows <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>,
                   <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">samdf</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">tax_mod</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html">cbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>,
                           <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># adding unique ASV names</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_names-methods.html">taxa_names</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"ASV"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/ntaxa-methods.html">ntaxa</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html">cbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>,
                           <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_names-methods.html">taxa_names</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ASV1" "ASV2" "ASV3" "ASV4" "ASV5" "ASV6"</code></pre>
</div>
</div>
<p>So the complete data set contains 3357 ASVs. We can also use the <a href="https://github.com/microbiome/microbiome/">microbiome R package</a> <span class="citation" data-cites="lahti2017microbiome">(<a href="#ref-lahti2017microbiome" role="doc-biblioref">Lahti, Sudarshan, et al. 2017</a>)</span> to get some additional summary data from the phyloseq object.</p>
<div class="cell">

</div>
<table class="table">
<colgroup>
<col style="width: 40%">
<col style="width: 59%">
</colgroup>
<thead><tr class="header">
<th>Metric</th>
<th>Results</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>80</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>64636</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>491143</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>32743</td>
</tr>
<tr class="odd">
<td>Median number of reads</td>
<td>36063</td>
</tr>
<tr class="even">
<td>Total ASVs</td>
<td>3357</td>
</tr>
<tr class="odd">
<td>Sparsity</td>
<td>0.802</td>
</tr>
<tr class="even">
<td>Any ASVs sum to 1 or less?</td>
<td>TRUE</td>
</tr>
<tr class="odd">
<td>Number of singleton ASVs</td>
<td>3</td>
</tr>
<tr class="even">
<td>Percent of ASVs that are singletons</td>
<td>0.089</td>
</tr>
<tr class="odd">
<td>Number of sample variables are:</td>
<td>6 (SamName, PLOT, DEPTH, TREAT, TEMP, PAIR)</td>
</tr>
</tbody>
</table>
<p><strong>B</strong>. Add two final columns containing the ASV sequences and ASV IDs. This will be useful later when trying to export a fasta file. We can also take a look at the phyloseq object.</p>
<div class="cell">
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"Kingdom"</span>, <span class="st">"Phylum"</span>, <span class="st">"Class"</span>, <span class="st">"Order"</span>,
    <span class="st">"Family"</span>, <span class="st">"Genus"</span>, <span class="st">"ASV_SEQ"</span>, <span class="st">"ASV_ID"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/rank_names.html">rank_names</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3357 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3357 taxa by 8 taxonomic ranks ]</code></pre>
</div>
</div>
<div class="cell">

</div>
<p><strong>C</strong>. Export sequence and taxonomy tables for the unadulterated phyloseq object for later use. We will use the prefix <code>full</code> to indicate that these are the <em>raw</em> outputs.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb44"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>,
            <span class="st">"files/data-prep/tables/its18_full_tax_table.txt"</span>,
            sep<span class="op">=</span><span class="st">"\t"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, col.names<span class="op">=</span><span class="cn">NA</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">)</span>,
            <span class="st">"files/data-prep/tables/its18_full_seq_table.txt"</span>,
            sep<span class="op">=</span><span class="st">"\t"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, col.names<span class="op">=</span><span class="cn">NA</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">

</div>
<p>At this point we would normally remove unwanted taxa and/or negative controls. Since we do not have any of these for this dataset, we are going to save the phyloseq object with a new name. That way, if we do decide later on to remove some taxa it will be easier to plug in the code here and then continue with the rest of the workflow.</p>
<div class="cell">
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ps_filt</span> <span class="op">&lt;-</span> <span class="va">ps</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section></section><section id="remove-low-count-samples" class="level2"><h2 class="anchored" data-anchor-id="remove-low-count-samples">Remove Low-Count Samples</h2>
<p>Next, we can remove samples with really low read counts, say less than 500 reads.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ps_filt_no_low</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_samples-methods.html">prune_samples</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_sums.html">sample_sums</a></span><span class="op">(</span><span class="va">ps_filt</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">500</span>, <span class="va">ps_filt</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell">
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ps_filt_no_low</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3357 taxa and 13 samples ]
sample_data() Sample Data:       [ 13 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3357 taxa by 8 taxonomic ranks ]</code></pre>
</div>
</div>
<p>We lost 2 sample(s). After removing samples we need to check whether any ASVs ended up with no reads.</p>
<div class="cell">
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">no_reads</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_sums.html">taxa_sums</a></span><span class="op">(</span><span class="va">ps_filt_no_low</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>And we lost 2 ASV(s). So now we must remove these from the ps object.</p>
<div class="cell">
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ps_filt_no_nc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_taxa-methods.html">prune_taxa</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_sums.html">taxa_sums</a></span><span class="op">(</span><span class="va">ps_filt_no_low</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="va">ps_filt_no_low</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ps_filt_no_nc</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3355 taxa and 13 samples ]
sample_data() Sample Data:       [ 13 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3355 taxa by 8 taxonomic ranks ]</code></pre>
</div>
</div>
</section><section id="rename-na-taxonomic-ranks-1" class="level2"><h2 class="anchored" data-anchor-id="rename-na-taxonomic-ranks-1">Rename NA taxonomic ranks</h2>
<p>Phyloseq has an odd way of dealing with taxonomic ranks that have no value—in other words, <em>NA</em> in the tax table. The first thing we are going to do before moving forward is to change all of the <em>NA</em>s to have a value of the next highest classified rank. For example, <code>ASV4</code> is not classified at the Genus level but is at Family level (Nitrososphaeraceae). So we change the Genus name to <em>Family_Nitrososphaeraceae</em>. The code for comes from these two posts on the phyloseq GitHub, both by <a href="https://github.com/MSMortensen">MSMortensen</a>: issue <a href="https://github.com/joey711/phyloseq/issues/850#issuecomment-394771087">#850</a> and issue <a href="https://github.com/joey711/phyloseq/issues/990#issuecomment-424618425">#990</a>.</p>
<blockquote class="blockquote">
<p>One thing this code does is reassign the functions <code>class</code> and <code>order</code> to taxonomic ranks. This can cause issues if you need these functions.</p>
</blockquote>
<p>So you need to run something like this <code>rm(class, order, phylum, kingdom)</code> at the end of the code to remove these as variables. For now, I have not come up with a better solution.</p>
<p>Before doing that, we will save a copy of the phyloseq object with the original taxa names.</p>
<div class="cell">
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ps_work_o</span> <span class="op">&lt;-</span> <span class="va">ps_filt_no_nc</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb54"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tax.clean</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps_filt_no_nc</span><span class="op">)</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">)</span><span class="op">{</span> <span class="va">tax.clean</span><span class="op">[</span>,<span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">tax.clean</span><span class="op">[</span>,<span class="va">i</span><span class="op">]</span><span class="op">)</span><span class="op">}</span>
<span class="va">tax.clean</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">tax.clean</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">""</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html">nrow</a></span><span class="op">(</span><span class="va">tax.clean</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
    <span class="kw">if</span> <span class="op">(</span><span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">==</span> <span class="st">""</span><span class="op">)</span><span class="op">{</span>
        <span class="va">kingdom</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"k_"</span>, <span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">1</span><span class="op">]</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
        <span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>, <span class="fl">2</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">kingdom</span>
    <span class="op">}</span> <span class="kw">else</span> <span class="kw">if</span> <span class="op">(</span><span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">3</span><span class="op">]</span> <span class="op">==</span> <span class="st">""</span><span class="op">)</span><span class="op">{</span>
        <span class="va">phylum</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"p_"</span>, <span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">2</span><span class="op">]</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
        <span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>, <span class="fl">3</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">phylum</span>
    <span class="op">}</span> <span class="kw">else</span> <span class="kw">if</span> <span class="op">(</span><span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">4</span><span class="op">]</span> <span class="op">==</span> <span class="st">""</span><span class="op">)</span><span class="op">{</span>
        <span class="va">class</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"c_"</span>, <span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">3</span><span class="op">]</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
        <span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>, <span class="fl">4</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">class</span>
    <span class="op">}</span> <span class="kw">else</span> <span class="kw">if</span> <span class="op">(</span><span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">5</span><span class="op">]</span> <span class="op">==</span> <span class="st">""</span><span class="op">)</span><span class="op">{</span>
        <span class="va">order</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"o_"</span>, <span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">4</span><span class="op">]</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
        <span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>, <span class="fl">5</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">order</span>
    <span class="op">}</span> <span class="kw">else</span> <span class="kw">if</span> <span class="op">(</span><span class="va">tax.clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">6</span><span class="op">]</span> <span class="op">==</span> <span class="st">""</span><span class="op">)</span><span class="op">{</span>
        <span class="va">tax.clean</span><span class="op">$</span><span class="va">Genus</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"f"</span>,<span class="va">tax.clean</span><span class="op">$</span><span class="va">Family</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span>
        <span class="op">}</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps_filt_no_nc</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">tax.clean</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/rank_names.html">rank_names</a></span><span class="op">(</span><span class="va">ps_filt_no_nc</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">class</span>, <span class="va">order</span>, <span class="va">phylum</span>, <span class="va">kingdom</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Still the same ranks. That’s good. What about the new groups? Let’s take a peak at some families.</p>
<div class="cell">
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/get_taxa_unique.html">get_taxa_unique</a></span><span class="op">(</span><span class="va">ps_filt_no_nc</span>, <span class="st">"Family"</span><span class="op">)</span>, <span class="fl">16</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Hygrophoraceae"         "Mycosphaerellaceae"     "p_Ascomycota"          
 [4] "Trichosporonaceae"      "c_Agaricomycetes"       "Archaeorhizomycetaceae"
 [7] "Chaetomellaceae"        "Geastraceae"            "k_Fungi"               
[10] "Metschnikowiaceae"      "Clavariaceae"           "o_Agaricales"          
[13] "Ascobolaceae"           "Trichocomaceae"         "Piptocephalidaceae"    
[16] "p_Chytridiomycota"     </code></pre>
</div>
</div>
<p>Nice. Adios <em>NA</em>.</p>
<blockquote class="blockquote">
<p><em>Note</em>. The original code is written for data sets that have species-level designations.</p>
</blockquote>
<p>Since this data set does not contain species, I modified the code to stop at the genus level. If your data set has species, my modifications will not work for you.</p>
<p>Finally, we rename the ps object. This is now our working data set.</p>
<div class="cell">
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ps_work</span> <span class="op">&lt;-</span> <span class="va">ps_filt_no_nc</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="data-set-summary-1" class="level2"><h2 class="anchored" data-anchor-id="data-set-summary-1">Data Set Summary</h2>
<p>Now that we have removed the NC and samples with low reads, we can summarize the data in our working phyloseq object. Let’s start with the data set as a whole using the <code>summarize_phyloseq</code> from the <a href="https://github.com/microbiome/microbiome/">microbiome R package</a> <span class="citation" data-cites="lahti2017microbiome">(<a href="#ref-lahti2017microbiome" role="doc-biblioref">Lahti, Sudarshan, et al. 2017</a>)</span> as we did above.</p>
<div class="cell">

</div>
<table class="table">
<colgroup>
<col style="width: 38%">
<col style="width: 61%">
</colgroup>
<thead><tr class="header">
<th>Metric</th>
<th>Results</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>9172</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>64636</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>490767</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>37751</td>
</tr>
<tr class="odd">
<td>Median number of reads</td>
<td>38371</td>
</tr>
<tr class="even">
<td>Total ASVs</td>
<td>3355</td>
</tr>
<tr class="odd">
<td>Sparsity</td>
<td>0.671</td>
</tr>
<tr class="even">
<td>Any ASVs sum to 1 or less?</td>
<td>TRUE</td>
</tr>
<tr class="odd">
<td>Number of singleton ASVs</td>
<td>5</td>
</tr>
<tr class="even">
<td>Percent of ASVs that are singletons</td>
<td>0.149</td>
</tr>
<tr class="odd">
<td>Number of sample variables are:</td>
<td>6 (SamName, PLOT, DEPTH, TREAT, TEMP, PAIR)</td>
</tr>
</tbody>
</table>
<p>We can also generate a summary table of total reads &amp; ASVs for each sample. You can sort the table, download a copy, or filter by search term. Here is the code to generate the data for the table. First, we create data frames that hold total reads and ASVs for each sample.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb58"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">total_reads</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_sums.html">sample_sums</a></span><span class="op">(</span><span class="va">ps_work</span><span class="op">)</span>
<span class="va">total_reads</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">total_reads</span>, make.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">total_reads</span> <span class="op">&lt;-</span> <span class="va">total_reads</span> <span class="op">%&gt;%</span> <span class="fu">rownames_to_column</span><span class="op">(</span><span class="st">"Sample_ID"</span><span class="op">)</span>

<span class="va">total_asvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/estimate_richness.html">estimate_richness</a></span><span class="op">(</span><span class="va">ps_work</span>, measures <span class="op">=</span> <span class="st">"Observed"</span><span class="op">)</span>
<span class="va">total_asvs</span> <span class="op">&lt;-</span> <span class="va">total_asvs</span> <span class="op">%&gt;%</span> <span class="fu">rownames_to_column</span><span class="op">(</span><span class="st">"Sample_ID"</span><span class="op">)</span>
<span class="va">total_asvs</span><span class="op">$</span><span class="va">Sample_ID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">'\\.'</span>, <span class="st">'-'</span>, <span class="va">total_asvs</span><span class="op">$</span><span class="va">Sample_ID</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>And then we merge these two data frames with the sample data frame.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb59"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>

<span class="va">sam_details</span> <span class="op">&lt;-</span> <span class="va">samdf</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">sam_details</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">sam_details</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"Sample_ID"</span>, <span class="st">"Plot"</span>, <span class="st">"Depth"</span>,
                                  <span class="st">"Treatment"</span>, <span class="st">"Temp"</span>, <span class="st">"Pair"</span><span class="op">)</span>
<span class="va">tmp_tab1</span> <span class="op">&lt;-</span> <span class="va">seq_table</span>
<span class="va">tmp_tab1</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="fl">2</span><span class="op">:</span><span class="fl">7</span>,<span class="fl">11</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">tmp_tab2</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">sam_details</span>, <span class="va">total_reads</span><span class="op">)</span> <span class="op">%&gt;%</span>
                <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">total_asvs</span><span class="op">)</span> <span class="op">%&gt;%</span>
                <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">tmp_tab1</span><span class="op">)</span>

<span class="va">reads_lost</span> <span class="op">&lt;-</span> <span class="va">tmp_tab2</span><span class="op">$</span><span class="va">`final reads`</span> <span class="op">-</span> <span class="va">tmp_tab2</span><span class="op">$</span><span class="va">total_reads</span> 
<span class="va">asvs_lost</span> <span class="op">&lt;-</span> <span class="va">tmp_tab2</span><span class="op">$</span><span class="va">`no. ASVs`</span> <span class="op">-</span> <span class="va">tmp_tab2</span><span class="op">$</span><span class="va">Observed</span> 
<span class="va">tmp_tab2</span><span class="op">$</span><span class="va">reads_lost</span> <span class="op">&lt;-</span> <span class="va">reads_lost</span>
<span class="va">tmp_tab2</span><span class="op">$</span><span class="va">asvs_lost</span> <span class="op">&lt;-</span> <span class="va">asvs_lost</span>

<span class="va">seq_table2</span> <span class="op">&lt;-</span> <span class="va">tmp_tab2</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span>,<span class="fl">12</span>,<span class="fl">13</span><span class="op">)</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">seq_table2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"sample_id"</span>, <span class="st">"PLOT"</span>, <span class="st">"DEPTH"</span>,
                          <span class="st">"TREAT"</span>, <span class="st">"TEMP"</span>, <span class="st">"PAIR"</span>, <span class="st">"total_reads"</span>,
                          <span class="st">"total_ASVs"</span>, <span class="st">"reads_lost"</span>, <span class="st">"ASVs_lost"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><small><strong>(ITS) Table 2 | Sample summary table showing the final number of reads and ASVs (per sample) as well as the number of reads/ASVs lost during curation.</strong></small></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,c2FtcGxlX2lkLFBMT1QsREVQVEgsVFJFQVQsVEVNUCxQQUlSLHRvdGFsX3JlYWRzLHRvdGFsX0FTVnMscmVhZHNfbG9zdCxBU1ZzX2xvc3QKUDAxX0QwMF8wMTBfVzNBLFAwMSwwMF8wMTAsVywzLEEsMjk3OTQsOTQ1LDAsMApQMDFfRDAwXzAxMF9XOEEsUDAxLDAwXzAxMCxXLDgsQSw0MTg4MCw3MTksMCwwClAxMF9EMDBfMDEwX0MwRSxQMTAsMDBfMDEwLEMsMCxFLDY0NjM2LDgxMiwwLDAKUDAyX0QwMF8wMTBfQzBBLFAwMiwwMF8wMTAsQywwLEEsNTExNDQsMTAxMSwwLDAKUDAzX0QwMF8wMTBfVzNCLFAwMywwMF8wMTAsVywzLEIsMzgzNzQsNzY1LDAsMApQMDNfRDAwXzAxMF9XOEIsUDAzLDAwXzAxMCxXLDgsQiwxNDE4OCwzNDksMCwwClAwNF9EMDBfMDEwX0MwQixQMDQsMDBfMDEwLEMsMCxCLDU2NTkzLDEwMTcsMCwwClAwNV9EMDBfMDEwX1czQyxQMDUsMDBfMDEwLFcsMyxDLE5BLE5BLE5BLE5BClAwNV9EMDBfMDEwX1c4QyxQMDUsMDBfMDEwLFcsOCxDLDM1NjAwLDYxNiwwLDAKUDA2X0QwMF8wMTBfQzBDLFAwNiwwMF8wMTAsQywwLEMsNDU4NTAsOTU0LDAsMApQMDdfRDAwXzAxMF9XM0QsUDA3LDAwXzAxMCxXLDMsRCwzNjA2Myw4NjcsMCwwClAwN19EMDBfMDEwX1c4RCxQMDcsMDBfMDEwLFcsOCxELDkxNzIsMzM1LDAsMApQMDhfRDAwXzAxMF9DMEQsUDA4LDAwXzAxMCxDLDAsRCwzODM3MSw3MDIsMCwwClAwOV9EMDBfMDEwX1czRSxQMDksMDBfMDEwLFcsMyxFLDI5MTAyLDc0NSwwLDAKUDA5X0QwMF8wMTBfVzhFLFAwOSwwMF8wMTAsVyw4LEUsTkEsTkEsTkEsTkEK').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'its18_sample_summary_final.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="cell-output-display">

<div id="htmlwidget-ddad56c6977ddacc3e3f" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ddad56c6977ddacc3e3f">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"sample_id":["P01_D00_010_W3A","P01_D00_010_W8A","P10_D00_010_C0E","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E"],"PLOT":["P01","P01","P10","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09"],"DEPTH":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"TREAT":["W","W","C","C","W","W","C","W","W","C","W","W","C","W","W"],"TEMP":["3","8","0","0","3","8","0","3","8","0","3","8","0","3","8"],"PAIR":["A","A","E","A","B","B","B","C","C","C","D","D","D","E","E"],"total_reads":[29794,41880,64636,51144,38374,14188,56593,"NA",35600,45850,36063,9172,38371,29102,"NA"],"total_ASVs":[945,719,812,1011,765,349,1017,"NA",616,954,867,335,702,745,"NA"],"reads_lost":[0,0,0,0,0,0,0,"NA",0,0,0,0,0,0,"NA"],"ASVs_lost":[0,0,0,0,0,0,0,"NA",0,0,0,0,0,0,"NA"]},"columns":[{"accessor":"sample_id","name":"sample_id","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P10_D00_010_C0E","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E"],"header":"sample id","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"PLOT","name":"PLOT","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01","P01","P10","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09"],"header":"PLOT","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"DEPTH","name":"DEPTH","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],"header":"DEPTH","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"TREAT","name":"TREAT","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["W","W","C","C","W","W","C","W","W","C","W","W","C","W","W"],"header":"TREAT","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"TEMP","name":"TEMP","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["3","8","0","0","3","8","0","3","8","0","3","8","0","3","8"],"header":"TEMP","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"PAIR","name":"PAIR","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["A","A","E","A","B","B","B","C","C","C","D","D","D","E","E"],"header":"PAIR","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"total_reads","name":"total_reads","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["29794","41880","64636","51144","38374","14188","56593","NA","35600","45850","36063","9172","38371","29102","NA"],"header":"total reads","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"total_ASVs","name":"total_ASVs","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["945","719","812","1011","765","349","1017","NA","616","954","867","335","702","745","NA"],"header":"total ASVs","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"reads_lost","name":"reads_lost","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0","0","0","0","0","0","0","NA","0","0","0","0","0","0","NA"],"header":"reads lost","align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"ASVs_lost","name":"ASVs_lost","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0","0","0","0","0","0","0","NA","0","0","0","0","0","0","NA"],"header":"ASVs lost","align":"center","footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"a2dd250c4eb4bd5fe9c909c2c8dd3f2f"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section><section id="review-1" class="level2"><h2 class="anchored" data-anchor-id="review-1">Review</h2>
<p>We have a few different options to play with. At this point it is difficult to say which we will use or whether additional objects need to be created. Here is a summary of the objects we have.</p>
<p>The full phyloseq object <em>before</em> removing contaminants.</p>
<div class="cell">
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ps</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3357 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3357 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/get_taxa_unique.html">get_taxa_unique</a></span><span class="op">(</span><span class="va">ps</span>, <span class="st">"Family"</span><span class="op">)</span>, <span class="fl">16</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Hygrophoraceae"                  "Mycosphaerellaceae"             
 [3] NA                                "Trichosporonaceae"              
 [5] "Archaeorhizomycetaceae"          "Chaetomellaceae"                
 [7] "Geastraceae"                     "Metschnikowiaceae"              
 [9] "Clavariaceae"                    "Ascobolaceae"                   
[11] "Trichocomaceae"                  "Piptocephalidaceae"             
[13] "Mortierellaceae"                 "Pleosporales_fam_Incertae_sedis"
[15] "Glomeraceae"                     "Ceratobasidiaceae"              </code></pre>
</div>
</div>
<p>The phyloseq object <em>before</em> removing the prefix of taxon ranks. This still has low coverage samples.</p>
<div class="cell">
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ps_o</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3357 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3357 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/get_taxa_unique.html">get_taxa_unique</a></span><span class="op">(</span><span class="va">ps_o</span>, <span class="st">"Family"</span><span class="op">)</span>, <span class="fl">16</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "f__Hygrophoraceae"                  "f__Mycosphaerellaceae"             
 [3] NA                                   "f__Trichosporonaceae"              
 [5] "f__Archaeorhizomycetaceae"          "f__Chaetomellaceae"                
 [7] "f__Geastraceae"                     "f__Metschnikowiaceae"              
 [9] "f__Clavariaceae"                    "f__Ascobolaceae"                   
[11] "f__Trichocomaceae"                  "f__Piptocephalidaceae"             
[13] "f__Mortierellaceae"                 "f__Pleosporales_fam_Incertae_sedis"
[15] "f__Glomeraceae"                     "f__Ceratobasidiaceae"              </code></pre>
</div>
</div>
<p>The phyloseq object <em>before</em> changing NA ranks.</p>
<div class="cell">
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ps_work_o</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3355 taxa and 13 samples ]
sample_data() Sample Data:       [ 13 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3355 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/get_taxa_unique.html">get_taxa_unique</a></span><span class="op">(</span><span class="va">ps_work_o</span>, <span class="st">"Family"</span><span class="op">)</span>, <span class="fl">16</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Hygrophoraceae"                  "Mycosphaerellaceae"             
 [3] NA                                "Trichosporonaceae"              
 [5] "Archaeorhizomycetaceae"          "Chaetomellaceae"                
 [7] "Geastraceae"                     "Metschnikowiaceae"              
 [9] "Clavariaceae"                    "Ascobolaceae"                   
[11] "Trichocomaceae"                  "Piptocephalidaceae"             
[13] "Mortierellaceae"                 "Pleosporales_fam_Incertae_sedis"
[15] "Glomeraceae"                     "Ceratobasidiaceae"              </code></pre>
</div>
</div>
<p>The phyloseq object <em>after</em> changing NA ranks.</p>
<div class="cell">
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ps_work</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3355 taxa and 13 samples ]
sample_data() Sample Data:       [ 13 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3355 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/get_taxa_unique.html">get_taxa_unique</a></span><span class="op">(</span><span class="va">ps_work</span>, <span class="st">"Family"</span><span class="op">)</span>, <span class="fl">16</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Hygrophoraceae"         "Mycosphaerellaceae"     "p_Ascomycota"          
 [4] "Trichosporonaceae"      "c_Agaricomycetes"       "Archaeorhizomycetaceae"
 [7] "Chaetomellaceae"        "Geastraceae"            "k_Fungi"               
[10] "Metschnikowiaceae"      "Clavariaceae"           "o_Agaricales"          
[13] "Ascobolaceae"           "Trichocomaceae"         "Piptocephalidaceae"    
[16] "p_Chytridiomycota"     </code></pre>
</div>
</div>
</section><section id="final-steps-1" class="level2"><h2 class="anchored" data-anchor-id="final-steps-1">Final Steps</h2>
<p>The last thing tasks to complete are to <strong>a</strong>) save copies of the taxonomy and sequence tables and <strong>b</strong>) save an image of the workflow for the next part of the workflow.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb76"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps_work</span><span class="op">)</span>,
            <span class="st">"files/data-prep/tables/its18_work_tax_table.txt"</span>,
            sep<span class="op">=</span><span class="st">"\t"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, col.names<span class="op">=</span><span class="cn">NA</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">ps_work</span><span class="op">)</span><span class="op">)</span>,
            <span class="st">"files/data-prep/tables/its18_work_seq_table.txt"</span>,
            sep<span class="op">=</span><span class="st">"\t"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, col.names<span class="op">=</span><span class="cn">NA</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">

</div>
<div class="cell">

</div>
</section></section><section id="workflow-output" class="level1"><h1>Workflow Output</h1>
<p>Data products generated in this workflow can be downloaded from figshare.</p>
<iframe src="https://widgets.figshare.com/articles/14690739/embed?show_title=1" width="100%" height="351" allowfullscreen="" frameborder="0">
</iframe>
<p>That’s all for this part!</p>
<p></p>
<p><a href="filtering.html" class="btnnav button_round" style="float: right;">Next workflow: <br> 3. Filtering</a> <a href="dada2.html" class="btnnav button_round" style="float: left;">Previous workflow: <br> 1. DADA2 Workflow</a><br><br></p>
<div class="cell">

</div>
</section><div id="quarto-appendix" class="default"><section id="source-code" class="level3 appendix"><h2 class="quarto-appendix-heading">Source Code</h2><div class="quarto-appendix-contents">
<p>The source code for this page can be accessed on GitHub <svg aria-hidden="true" role="img" viewbox="0 0 496 512" style="height:1em;width:0.97em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:currentColor;overflow:visible;position:relative;"><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"></path></svg> by <a href="https://github.com/sweltr/high-temp/blob/main/data-prep.qmd">clicking this link</a>.</p>
</div></section><section id="data-availability" class="level2 appendix"><h2 class="quarto-appendix-heading">Data Availability</h2><div class="quarto-appendix-contents">
<p>Data generated in this workflow and the <code>.Rdata</code> need to run the workflow can be accessed on figshare at <a href="https://doi.org/10.25573/data.14690739">10.25573/data.14690739</a>.</p>
</div></section><section id="last-updated-on" class="level2 appendix"><h2 class="quarto-appendix-heading">Last updated on</h2><div class="quarto-appendix-contents">
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "2022-06-19 13:17:22 EST"</code></pre>
</div>
</div>


<!-- -->


</div></section><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-lahti2017microbiome" class="csl-entry" role="doc-biblioentry">
Lahti, Leo, Shetty Sudarshan, et al. 2017. <span>“Tools for Microbiome Analysis in r. Version 1.9.97.”</span> <a href="https://github.com/microbiome/microbiome">https://github.com/microbiome/microbiome</a>.
</div>
<div id="ref-mcmurdie2013phyloseq" class="csl-entry" role="doc-biblioentry">
McMurdie, Paul J, and Susan Holmes. 2013. <span>“Phyloseq: An r Package for Reproducible Interactive Analysis and Graphics of Microbiome Census Data.”</span> <em>PLoS One</em> 8 (4): e61217. <a href="https://doi.org/10.1371/journal.pone.0061217">https://doi.org/10.1371/journal.pone.0061217</a>.
</div>
</div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./dada2.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">1. DADA2 Workflow</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./filtering.html" class="pagination-link">
        <span class="nav-page-text">3. Filtering</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb78" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "2. Data Set Preparation"</span></span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a><span class="an">description:</span><span class="co"> |</span></span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a><span class="co">  Workflow for preparation  of the **16S rRNA** and **ITS** data sets. These steps are needed before analyzing the data. In this workflow, sample groups are defined, and phyloseq objects are created and curated.</span></span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb78-8"><a href="#cb78-8" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;&lt;strong&gt;</span>Click here<span class="kw">&lt;/strong&gt;</span> for setup information.<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb78-9"><a href="#cb78-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-12"><a href="#cb78-12" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-13"><a href="#cb78-13" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb78-14"><a href="#cb78-14" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb78-15"><a href="#cb78-15" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">echo =</span> <span class="cn">TRUE</span>, <span class="at">eval =</span> <span class="cn">FALSE</span>)</span>
<span id="cb78-16"><a href="#cb78-16" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">119</span>)</span>
<span id="cb78-17"><a href="#cb78-17" aria-hidden="true" tabindex="-1"></a><span class="co">#library(conflicted)</span></span>
<span id="cb78-18"><a href="#cb78-18" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends(ape, local = TRUE)  </span></span>
<span id="cb78-19"><a href="#cb78-19" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends_reverse(ape, local = TRUE)  </span></span>
<span id="cb78-20"><a href="#cb78-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq); <span class="fu">packageVersion</span>(<span class="st">"phyloseq"</span>)</span>
<span id="cb78-21"><a href="#cb78-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings); <span class="fu">packageVersion</span>(<span class="st">"Biostrings"</span>)</span>
<span id="cb78-22"><a href="#cb78-22" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(tidyverse, DT, microbiome, taxa, formatR,</span>
<span id="cb78-23"><a href="#cb78-23" aria-hidden="true" tabindex="-1"></a>               captioner, reactable, downloadthis, fontawesome,</span>
<span id="cb78-24"><a href="#cb78-24" aria-hidden="true" tabindex="-1"></a>               metacoder, ampvis2, ape, statnet.common, </span>
<span id="cb78-25"><a href="#cb78-25" aria-hidden="true" tabindex="-1"></a>               <span class="at">install =</span> <span class="cn">FALSE</span>, <span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb78-26"><a href="#cb78-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-27"><a href="#cb78-27" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">scipen=</span><span class="dv">999</span>)</span>
<span id="cb78-28"><a href="#cb78-28" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_current<span class="sc">$</span><span class="fu">get</span>(<span class="fu">c</span>(</span>
<span id="cb78-29"><a href="#cb78-29" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cache"</span>,</span>
<span id="cb78-30"><a href="#cb78-30" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cache.path"</span>,</span>
<span id="cb78-31"><a href="#cb78-31" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cache.rebuild"</span>,</span>
<span id="cb78-32"><a href="#cb78-32" aria-hidden="true" tabindex="-1"></a>  <span class="st">"dependson"</span>,</span>
<span id="cb78-33"><a href="#cb78-33" aria-hidden="true" tabindex="-1"></a>  <span class="st">"autodep"</span></span>
<span id="cb78-34"><a href="#cb78-34" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb78-35"><a href="#cb78-35" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-36"><a href="#cb78-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-37"><a href="#cb78-37" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb78-38"><a href="#cb78-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-39"><a href="#cb78-39" aria-hidden="true" tabindex="-1"></a><span class="fu"># Workflow Input</span></span>
<span id="cb78-40"><a href="#cb78-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-41"><a href="#cb78-41" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; You will either need to run the </span><span class="co">[</span><span class="ot">DADA2 workflow</span><span class="co">](dada2.html)</span><span class="at"> or download the files linked below. See  the </span><span class="co">[</span><span class="ot">Data Availability</span><span class="co">](data-availability.html)</span><span class="at"> page for complete details.</span></span>
<span id="cb78-42"><a href="#cb78-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-43"><a href="#cb78-43" aria-hidden="true" tabindex="-1"></a>Files needed to run this workflow can be downloaded from figshare. These are the output files from the <span class="co">[</span><span class="ot">DADA2 workflow</span><span class="co">](dada2.html)</span> page.</span>
<span id="cb78-44"><a href="#cb78-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-45"><a href="#cb78-45" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;iframe</span> <span class="er">src</span><span class="ot">=</span><span class="st">"https://widgets.figshare.com/articles/14687184/embed?show_title=1"</span> <span class="er">width</span><span class="ot">=</span><span class="st">"100%"</span> <span class="er">height</span><span class="ot">=</span><span class="st">"351"</span> <span class="er">allowfullscreen</span> <span class="er">frameborder</span><span class="ot">=</span><span class="st">"0"</span><span class="kw">&gt;&lt;/iframe&gt;</span></span>
<span id="cb78-46"><a href="#cb78-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-47"><a href="#cb78-47" aria-hidden="true" tabindex="-1"></a>These workflows share many common variable names so you must split the workflow into a script for each data set OR run the command <span class="in">`remove(list = ls())`</span> before beginning the next workflow.</span>
<span id="cb78-48"><a href="#cb78-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-49"><a href="#cb78-49" aria-hidden="true" tabindex="-1"></a><span class="fu"># 16s rRNA Data</span></span>
<span id="cb78-50"><a href="#cb78-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-53"><a href="#cb78-53" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-54"><a href="#cb78-54" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb78-55"><a href="#cb78-55" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-56"><a href="#cb78-56" aria-hidden="true" tabindex="-1"></a><span class="do">## ONLY FOR PAGE BUILD</span></span>
<span id="cb78-57"><a href="#cb78-57" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb78-58"><a href="#cb78-58" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"page_build/data_prep_ssu18_wf.rdata"</span>)</span>
<span id="cb78-59"><a href="#cb78-59" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-60"><a href="#cb78-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-63"><a href="#cb78-63" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-64"><a href="#cb78-64" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-65"><a href="#cb78-65" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-66"><a href="#cb78-66" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb78-67"><a href="#cb78-67" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the caption(s) with captioner</span></span>
<span id="cb78-68"><a href="#cb78-68" aria-hidden="true" tabindex="-1"></a>caption_tab <span class="ot">&lt;-</span> <span class="fu">captioner</span>(<span class="at">prefix =</span> <span class="st">"(16S rRNA) Table"</span>, <span class="at">suffix =</span> <span class="st">" |"</span>, <span class="at">style =</span> <span class="st">"b"</span>)</span>
<span id="cb78-69"><a href="#cb78-69" aria-hidden="true" tabindex="-1"></a>caption_fig <span class="ot">&lt;-</span> <span class="fu">captioner</span>(<span class="at">prefix =</span> <span class="st">"(16S rRNA) Figure"</span>, <span class="at">suffix =</span> <span class="st">" |"</span>, <span class="at">style =</span> <span class="st">"b"</span>)</span>
<span id="cb78-70"><a href="#cb78-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a function for referring to the tables in text</span></span>
<span id="cb78-71"><a href="#cb78-71" aria-hidden="true" tabindex="-1"></a>ref <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">str_extract</span>(x, <span class="st">"[^|]*"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb78-72"><a href="#cb78-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">trimws</span>(<span class="at">which =</span> <span class="st">"right"</span>, <span class="at">whitespace =</span> <span class="st">"[ ]"</span>)</span>
<span id="cb78-73"><a href="#cb78-73" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-74"><a href="#cb78-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-77"><a href="#cb78-77" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-78"><a href="#cb78-78" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-79"><a href="#cb78-79" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-80"><a href="#cb78-80" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"assets/captions/captions_data_prep_ssu.R"</span>)</span>
<span id="cb78-81"><a href="#cb78-81" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-82"><a href="#cb78-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-83"><a href="#cb78-83" aria-hidden="true" tabindex="-1"></a><span class="fu">## Prerequisites</span></span>
<span id="cb78-84"><a href="#cb78-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-85"><a href="#cb78-85" aria-hidden="true" tabindex="-1"></a>In order to run this workflow, you either need to run the corresponding <span class="co">[</span><span class="ot">DADA2 Workflow for 16S rRNA</span><span class="co">](dada2-high-temp.html#s-rrna-data.html)</span> or begin with the output from that workflow, <span class="in">`data_prep_ssu18_wf.rdata`</span>. See the <span class="co">[</span><span class="ot">Data Availability</span><span class="co">](data-availability.html)</span> page for complete details.</span>
<span id="cb78-86"><a href="#cb78-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-87"><a href="#cb78-87" aria-hidden="true" tabindex="-1"></a>Unless otherwise noted, we primarily use <span class="co">[</span><span class="ot">phyloseq</span><span class="co">](https://joey711.github.io/phyloseq/)</span>  <span class="co">[</span><span class="ot">@mcmurdie2013phyloseq</span><span class="co">]</span> in this section of the workflow to prepare the 2018 16S rRNA data set for community analyses. We prepare the data set by curating samples, removing contaminants, and creating phyloseq objects.</span>
<span id="cb78-88"><a href="#cb78-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-89"><a href="#cb78-89" aria-hidden="true" tabindex="-1"></a><span class="fu">## Read Counts Assessment</span></span>
<span id="cb78-90"><a href="#cb78-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-91"><a href="#cb78-91" aria-hidden="true" tabindex="-1"></a>Before we begin, let's create a summary table containing some basic sample metadata and the read count data from the <span class="co">[</span><span class="ot">DADA2 workflow</span><span class="co">](dada2-high-temp.html#s-rrna-data.html)</span>. We need to inspect how total reads changed through the workflow. While we are at it, let's create more intuitive Sample IDs. For more details on how reads changed at each step of the workflow, see the <span class="co">[</span><span class="ot">summary table</span><span class="co">](dada2-high-temp.html#track-reads-through-workflow)</span> at the end of the DADA2 section. Table headers are as follows:</span>
<span id="cb78-92"><a href="#cb78-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-93"><a href="#cb78-93" aria-hidden="true" tabindex="-1"></a>| Header                    | Description                                                                              |</span>
<span id="cb78-94"><a href="#cb78-94" aria-hidden="true" tabindex="-1"></a>|---------------------------|------------------------------------------------------------------------------------------|</span>
<span id="cb78-95"><a href="#cb78-95" aria-hidden="true" tabindex="-1"></a>| <span class="in">`Sample ID`</span>               | the new sample ID based on Plot number, Depth, Treatment, Temperature, &amp; Pair            |</span>
<span id="cb78-96"><a href="#cb78-96" aria-hidden="true" tabindex="-1"></a>| <span class="in">`Plot`</span>                    | the experimental plot number                                                             |</span>
<span id="cb78-97"><a href="#cb78-97" aria-hidden="true" tabindex="-1"></a>| <span class="in">`Depth`</span>                   | the depth where the sample was collected                                                 |</span>
<span id="cb78-98"><a href="#cb78-98" aria-hidden="true" tabindex="-1"></a>| <span class="in">`Treatment`</span>               | warm or control                                                                          |</span>
<span id="cb78-99"><a href="#cb78-99" aria-hidden="true" tabindex="-1"></a>| <span class="in">`Temp`</span>                    | temperature of soil heating (0C, +4C, or +8C)                                            |</span>
<span id="cb78-100"><a href="#cb78-100" aria-hidden="true" tabindex="-1"></a>| <span class="in">`Pair`</span>                    | control/treatment coupling                                                               |</span>
<span id="cb78-101"><a href="#cb78-101" aria-hidden="true" tabindex="-1"></a>| <span class="in">`input`</span>                   | number of raw reads                                                                      |</span>
<span id="cb78-102"><a href="#cb78-102" aria-hidden="true" tabindex="-1"></a>| <span class="in">`nochim`</span>                  | final read count after removing chimeras                                                 |</span>
<span id="cb78-103"><a href="#cb78-103" aria-hidden="true" tabindex="-1"></a>| <span class="in">`Remain`</span>                  | percent of reads remaining from <span class="in">`input`</span> to <span class="in">`nonchim`</span>                                     |</span>
<span id="cb78-104"><a href="#cb78-104" aria-hidden="true" tabindex="-1"></a>| <span class="in">`FastqID`</span>                 | base name of the fastq file                                                              |</span>
<span id="cb78-105"><a href="#cb78-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-106"><a href="#cb78-106" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br/&gt;</span></span>
<span id="cb78-107"><a href="#cb78-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-110"><a href="#cb78-110" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-111"><a href="#cb78-111" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-112"><a href="#cb78-112" aria-hidden="true" tabindex="-1"></a>tmp_tab1 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb78-113"><a href="#cb78-113" aria-hidden="true" tabindex="-1"></a>  <span class="st">"files/dada2/tables/ssu18_sample_seq_info.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb78-114"><a href="#cb78-114" aria-hidden="true" tabindex="-1"></a>tmp_tab1<span class="sc">$</span>root <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, tmp_tab1<span class="sc">$</span>Fastq_ID_Forward)</span>
<span id="cb78-115"><a href="#cb78-115" aria-hidden="true" tabindex="-1"></a>tmp_tab1[<span class="dv">9</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_tab1)<span class="sc">-</span><span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb78-116"><a href="#cb78-116" aria-hidden="true" tabindex="-1"></a>tmp_tab1 <span class="ot">&lt;-</span> tmp_tab1[, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">8</span>,<span class="dv">2</span><span class="sc">:</span><span class="dv">7</span>)]</span>
<span id="cb78-117"><a href="#cb78-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-118"><a href="#cb78-118" aria-hidden="true" tabindex="-1"></a>tmp_tab2 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb78-119"><a href="#cb78-119" aria-hidden="true" tabindex="-1"></a>  <span class="st">"files/dada2/tables/ssu18_read_changes.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>,  <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb78-120"><a href="#cb78-120" aria-hidden="true" tabindex="-1"></a>tmp_tab2[<span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb78-121"><a href="#cb78-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-122"><a href="#cb78-122" aria-hidden="true" tabindex="-1"></a>tmp_tab3 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb78-123"><a href="#cb78-123" aria-hidden="true" tabindex="-1"></a>  <span class="st">"files/dada2/tables/ssu18_dada_compare_by_samp.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>,  <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb78-124"><a href="#cb78-124" aria-hidden="true" tabindex="-1"></a>tmp_tab3 <span class="ot">&lt;-</span> tmp_tab3[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">5</span>)]</span>
<span id="cb78-125"><a href="#cb78-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-126"><a href="#cb78-126" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> <span class="fu">left_join</span>(tmp_tab1, tmp_tab2, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Sample_ID"</span> <span class="ot">=</span> <span class="st">"Sample_ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb78-127"><a href="#cb78-127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., tmp_tab3, <span class="at">by =</span> <span class="st">"Sample_ID"</span>)</span>
<span id="cb78-128"><a href="#cb78-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-129"><a href="#cb78-129" aria-hidden="true" tabindex="-1"></a>seq_table<span class="sc">$</span>per_reads_kept <span class="ot">&lt;-</span> <span class="fu">round</span>(seq_table<span class="sc">$</span>nonchim<span class="sc">/</span>seq_table<span class="sc">$</span>raw_reads, <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb78-130"><a href="#cb78-130" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(seq_table) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Sample_ID"</span>, <span class="st">"root name"</span>, <span class="st">"Plot"</span>, <span class="st">"Treatment"</span>, </span>
<span id="cb78-131"><a href="#cb78-131" aria-hidden="true" tabindex="-1"></a>                         <span class="st">"Temp"</span>, <span class="st">"Depth"</span>, <span class="st">"Pair"</span>, <span class="st">"input reads"</span>, </span>
<span id="cb78-132"><a href="#cb78-132" aria-hidden="true" tabindex="-1"></a>                         <span class="st">"final reads"</span>, <span class="st">"no. ASVs"</span>, <span class="st">"per. reads remain"</span>)</span>
<span id="cb78-133"><a href="#cb78-133" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb78-134"><a href="#cb78-134" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-135"><a href="#cb78-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-136"><a href="#cb78-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-137"><a href="#cb78-137" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab("seq_summary_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb78-138"><a href="#cb78-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-141"><a href="#cb78-141" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-142"><a href="#cb78-142" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-143"><a href="#cb78-143" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-144"><a href="#cb78-144" aria-hidden="true" tabindex="-1"></a>seq_table <span class="sc">%&gt;%</span></span>
<span id="cb78-145"><a href="#cb78-145" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb78-146"><a href="#cb78-146" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"ssu18_sample_summary"</span>,</span>
<span id="cb78-147"><a href="#cb78-147" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb78-148"><a href="#cb78-148" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb78-149"><a href="#cb78-149" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb78-150"><a href="#cb78-150" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb78-151"><a href="#cb78-151" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb78-152"><a href="#cb78-152" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb78-153"><a href="#cb78-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-154"><a href="#cb78-154" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb78-155"><a href="#cb78-155" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb78-156"><a href="#cb78-156" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb78-157"><a href="#cb78-157" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">1</span>),</span>
<span id="cb78-158"><a href="#cb78-158" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">TRUE</span>, <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb78-159"><a href="#cb78-159" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb78-160"><a href="#cb78-160" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb78-161"><a href="#cb78-161" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb78-162"><a href="#cb78-162" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Sample ID"</span>, </span>
<span id="cb78-163"><a href="#cb78-163" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb78-164"><a href="#cb78-164" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>),</span>
<span id="cb78-165"><a href="#cb78-165" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>), </span>
<span id="cb78-166"><a href="#cb78-166" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb78-167"><a href="#cb78-167" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">150</span>, <span class="at">footer =</span> <span class="st">"Total reads"</span>)</span>
<span id="cb78-168"><a href="#cb78-168" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb78-169"><a href="#cb78-169" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">5</span>, </span>
<span id="cb78-170"><a href="#cb78-170" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb78-171"><a href="#cb78-171" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb78-172"><a href="#cb78-172" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb78-173"><a href="#cb78-173" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb78-174"><a href="#cb78-174" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb78-175"><a href="#cb78-175" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-176"><a href="#cb78-176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-177"><a href="#cb78-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-180"><a href="#cb78-180" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-181"><a href="#cb78-181" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-182"><a href="#cb78-182" aria-hidden="true" tabindex="-1"></a>wong_pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#D55E00"</span>, <span class="st">"#56B4E9"</span>, <span class="st">"#009E73"</span>, <span class="st">"#0072B2"</span>,</span>
<span id="cb78-183"><a href="#cb78-183" aria-hidden="true" tabindex="-1"></a>              <span class="st">"#F0E442"</span>, <span class="st">"#CC79A7"</span>,  <span class="st">"#E69F00"</span>, <span class="st">"#7F7F7F"</span>,</span>
<span id="cb78-184"><a href="#cb78-184" aria-hidden="true" tabindex="-1"></a>              <span class="st">"#000000"</span>)</span>
<span id="cb78-185"><a href="#cb78-185" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-186"><a href="#cb78-186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-187"><a href="#cb78-187" aria-hidden="true" tabindex="-1"></a><span class="fu">## Defining Groups</span></span>
<span id="cb78-188"><a href="#cb78-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-189"><a href="#cb78-189" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>Load the data packet produced in the final step of the DADA2 workflow. This packet (<span class="in">`ssu18_dada2_wf.rdata`</span>) contains the ASV-by-sample table and the ASV taxonomy table.</span>
<span id="cb78-190"><a href="#cb78-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-191"><a href="#cb78-191" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>Rename the samples so names have plot and Depth info.</span>
<span id="cb78-192"><a href="#cb78-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-193"><a href="#cb78-193" aria-hidden="true" tabindex="-1"></a><span class="ss">3. </span>After we load the data packet, we next need to format sample names and define groups.</span>
<span id="cb78-194"><a href="#cb78-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-197"><a href="#cb78-197" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-198"><a href="#cb78-198" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb78-199"><a href="#cb78-199" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> seq_table[<span class="fu">order</span>(seq_table<span class="sc">$</span><span class="st">`</span><span class="at">root name</span><span class="st">`</span>), ]</span>
<span id="cb78-200"><a href="#cb78-200" aria-hidden="true" tabindex="-1"></a>tmp_names <span class="ot">&lt;-</span> seq_table<span class="sc">$</span><span class="st">`</span><span class="at">Sample_ID</span><span class="st">`</span></span>
<span id="cb78-201"><a href="#cb78-201" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"files/dada2/rdata/ssu18_dada2_wf.rdata"</span>)</span>
<span id="cb78-202"><a href="#cb78-202" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(seq_table<span class="sc">$</span><span class="st">`</span><span class="at">root name</span><span class="st">`</span>, <span class="fu">rownames</span>(seqtab))</span>
<span id="cb78-203"><a href="#cb78-203" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(seqtab) <span class="ot">&lt;-</span> tmp_names</span>
<span id="cb78-204"><a href="#cb78-204" aria-hidden="true" tabindex="-1"></a>samples.out <span class="ot">&lt;-</span> <span class="fu">rownames</span>(seqtab)</span>
<span id="cb78-205"><a href="#cb78-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-206"><a href="#cb78-206" aria-hidden="true" tabindex="-1"></a>sample_name <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">1</span>, <span class="dv">999</span>)</span>
<span id="cb78-207"><a href="#cb78-207" aria-hidden="true" tabindex="-1"></a>plot <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">0</span>, <span class="dv">3</span>)</span>
<span id="cb78-208"><a href="#cb78-208" aria-hidden="true" tabindex="-1"></a>depth <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">6</span>, <span class="dv">11</span>)</span>
<span id="cb78-209"><a href="#cb78-209" aria-hidden="true" tabindex="-1"></a>treatment <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">13</span>, <span class="dv">13</span>)</span>
<span id="cb78-210"><a href="#cb78-210" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">14</span>, <span class="dv">14</span>)</span>
<span id="cb78-211"><a href="#cb78-211" aria-hidden="true" tabindex="-1"></a>pair <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">15</span>, <span class="dv">15</span>)</span>
<span id="cb78-212"><a href="#cb78-212" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-213"><a href="#cb78-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-214"><a href="#cb78-214" aria-hidden="true" tabindex="-1"></a>We have a total of **`r length(unique(sample_name))`** samples, **`r length(unique(pair))`** sample pairs, from **`r length(unique(plot))`** plots (**`r length(unique(depth))`** depth). There are **`r length(unique(treatment))`** treatments corresponding to **`r length(unique(temp))`** different temperature regimes.</span>
<span id="cb78-215"><a href="#cb78-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-216"><a href="#cb78-216" aria-hidden="true" tabindex="-1"></a><span class="ss">4. </span>And finally we define a sample data frame that holds the different groups we extracted from the sample names.</span>
<span id="cb78-217"><a href="#cb78-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-220"><a href="#cb78-220" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-221"><a href="#cb78-221" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb78-222"><a href="#cb78-222" aria-hidden="true" tabindex="-1"></a><span class="co">#define a sample data frame</span></span>
<span id="cb78-223"><a href="#cb78-223" aria-hidden="true" tabindex="-1"></a>samdf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">SamName =</span> sample_name,</span>
<span id="cb78-224"><a href="#cb78-224" aria-hidden="true" tabindex="-1"></a>                    <span class="at">PLOT =</span> plot,</span>
<span id="cb78-225"><a href="#cb78-225" aria-hidden="true" tabindex="-1"></a>                    <span class="at">DEPTH =</span> depth,</span>
<span id="cb78-226"><a href="#cb78-226" aria-hidden="true" tabindex="-1"></a>                    <span class="at">TREAT =</span> treatment,</span>
<span id="cb78-227"><a href="#cb78-227" aria-hidden="true" tabindex="-1"></a>                    <span class="at">TEMP =</span> temp,</span>
<span id="cb78-228"><a href="#cb78-228" aria-hidden="true" tabindex="-1"></a>                    <span class="at">PAIR =</span> pair)</span>
<span id="cb78-229"><a href="#cb78-229" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(samdf) <span class="ot">&lt;-</span> samples.out</span>
<span id="cb78-230"><a href="#cb78-230" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-231"><a href="#cb78-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-234"><a href="#cb78-234" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-235"><a href="#cb78-235" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-236"><a href="#cb78-236" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-237"><a href="#cb78-237" aria-hidden="true" tabindex="-1"></a>example_sam <span class="ot">&lt;-</span> <span class="st">"P07_D00_010_W3D"</span></span>
<span id="cb78-238"><a href="#cb78-238" aria-hidden="true" tabindex="-1"></a>qpair <span class="ot">&lt;-</span> samdf[example_sam,]<span class="sc">$</span>PAIR</span>
<span id="cb78-239"><a href="#cb78-239" aria-hidden="true" tabindex="-1"></a>qgroup <span class="ot">&lt;-</span> samdf <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(PAIR <span class="sc">==</span> qpair)</span>
<span id="cb78-240"><a href="#cb78-240" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-241"><a href="#cb78-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-242"><a href="#cb78-242" aria-hidden="true" tabindex="-1"></a>For example, <span class="in">`r example_sam`</span> is the sample heated at **+`r samdf[example_sam,]$TEMP`C** from plot **`r samdf[example_sam,]$PLOT`**, collected from  a depth of `r samdf[example_sam,]$DEPTH`cm. This sample is in group **`r qpair`** which consists of <span class="in">`r qgroup$SamName`</span>.</span>
<span id="cb78-243"><a href="#cb78-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-244"><a href="#cb78-244" aria-hidden="true" tabindex="-1"></a><span class="fu">## Create a Phyloseq Object</span></span>
<span id="cb78-245"><a href="#cb78-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-246"><a href="#cb78-246" aria-hidden="true" tabindex="-1"></a>**A**. The first step is to rename the amplicon sequence variants (ASVs) so the designations are a bit more user friendly. By default, DADA2 names each ASV by its unique sequence so that data can be directly compared across studies (which is great). But this convention can get cumbersome downstream, so we rename the ASVs using a simpler convention---ASV1, ASV2, ASV3, and so on.</span>
<span id="cb78-247"><a href="#cb78-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-248"><a href="#cb78-248" aria-hidden="true" tabindex="-1"></a>:::{.callout-note}</span>
<span id="cb78-249"><a href="#cb78-249" aria-hidden="true" tabindex="-1"></a>A phyloseq object contains ASV table (taxa abundances), sample metadata, and taxonomy table (mapping between ASVs and higher-level taxonomic classifications).</span>
<span id="cb78-250"><a href="#cb78-250" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb78-251"><a href="#cb78-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-254"><a href="#cb78-254" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-255"><a href="#cb78-255" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb78-256"><a href="#cb78-256" aria-hidden="true" tabindex="-1"></a><span class="co"># this create the phyloseq object</span></span>
<span id="cb78-257"><a href="#cb78-257" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(<span class="fu">otu_table</span>(seqtab, <span class="at">taxa_are_rows =</span> <span class="cn">FALSE</span>),</span>
<span id="cb78-258"><a href="#cb78-258" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">sample_data</span>(samdf), <span class="fu">tax_table</span>(tax_silva))</span>
<span id="cb78-259"><a href="#cb78-259" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">tax_table</span>(ps),</span>
<span id="cb78-260"><a href="#cb78-260" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">rownames</span>(<span class="fu">tax_table</span>(ps)))</span>
<span id="cb78-261"><a href="#cb78-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-262"><a href="#cb78-262" aria-hidden="true" tabindex="-1"></a><span class="co"># adding unique ASV names</span></span>
<span id="cb78-263"><a href="#cb78-263" aria-hidden="true" tabindex="-1"></a><span class="fu">taxa_names</span>(ps) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"ASV"</span>, <span class="fu">seq</span>(<span class="fu">ntaxa</span>(ps)))</span>
<span id="cb78-264"><a href="#cb78-264" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">tax_table</span>(ps),</span>
<span id="cb78-265"><a href="#cb78-265" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">rownames</span>(<span class="fu">tax_table</span>(ps)))</span>
<span id="cb78-266"><a href="#cb78-266" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-267"><a href="#cb78-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-268"><a href="#cb78-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-271"><a href="#cb78-271" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-272"><a href="#cb78-272" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-273"><a href="#cb78-273" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-274"><a href="#cb78-274" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">taxa_names</span>(ps))</span>
<span id="cb78-275"><a href="#cb78-275" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-276"><a href="#cb78-276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-277"><a href="#cb78-277" aria-hidden="true" tabindex="-1"></a>So the complete data set contains <span class="in">`r ntaxa(ps)`</span> ASVs. We can use the <span class="co">[</span><span class="ot">microbiome R package</span><span class="co">](https://github.com/microbiome/microbiome/)</span>  <span class="co">[</span><span class="ot">@lahti2017microbiome</span><span class="co">]</span> to get some additional summary data from the phyloseq object.</span>
<span id="cb78-278"><a href="#cb78-278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-281"><a href="#cb78-281" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-282"><a href="#cb78-282" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-283"><a href="#cb78-283" aria-hidden="true" tabindex="-1"></a>min_read_ps <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">readcount</span>(ps))</span>
<span id="cb78-284"><a href="#cb78-284" aria-hidden="true" tabindex="-1"></a>max_read_ps <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">readcount</span>(ps))</span>
<span id="cb78-285"><a href="#cb78-285" aria-hidden="true" tabindex="-1"></a>total_reads_ps <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">readcount</span>(ps))</span>
<span id="cb78-286"><a href="#cb78-286" aria-hidden="true" tabindex="-1"></a>mean_reads_ps <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">mean</span>(<span class="fu">readcount</span>(ps)), <span class="at">digits =</span> <span class="dv">0</span>)</span>
<span id="cb78-287"><a href="#cb78-287" aria-hidden="true" tabindex="-1"></a>median_reads_ps <span class="ot">&lt;-</span> <span class="fu">median</span>(<span class="fu">readcount</span>(ps))</span>
<span id="cb78-288"><a href="#cb78-288" aria-hidden="true" tabindex="-1"></a>total_asvs_ps <span class="ot">&lt;-</span> <span class="fu">ntaxa</span>(ps)</span>
<span id="cb78-289"><a href="#cb78-289" aria-hidden="true" tabindex="-1"></a>singleton_ps <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>(<span class="fu">ntaxa</span>(<span class="fu">rare</span>(ps, <span class="at">detection =</span> <span class="dv">1</span>, <span class="at">prevalence =</span> <span class="dv">0</span>)), <span class="at">error=</span><span class="cf">function</span>(err) <span class="cn">NA</span>)</span>
<span id="cb78-290"><a href="#cb78-290" aria-hidden="true" tabindex="-1"></a>singleton_ps_perc <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>(<span class="fu">round</span>((<span class="dv">100</span><span class="sc">*</span>(<span class="fu">ntaxa</span>(<span class="fu">rare</span>(ps, <span class="at">detection =</span> <span class="dv">1</span>, <span class="at">prevalence =</span> <span class="dv">0</span>)) <span class="sc">/</span></span>
<span id="cb78-291"><a href="#cb78-291" aria-hidden="true" tabindex="-1"></a>                                   <span class="fu">ntaxa</span>(ps))), <span class="at">digits =</span> <span class="dv">3</span>), <span class="at">error=</span><span class="cf">function</span>(err) <span class="cn">NA</span>)</span>
<span id="cb78-292"><a href="#cb78-292" aria-hidden="true" tabindex="-1"></a>sparsity_ps <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">length</span>(<span class="fu">which</span>(<span class="fu">abundances</span>(ps) <span class="sc">==</span> <span class="dv">0</span>))<span class="sc">/</span><span class="fu">length</span>(<span class="fu">abundances</span>(ps)),</span>
<span id="cb78-293"><a href="#cb78-293" aria-hidden="true" tabindex="-1"></a>                     <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb78-294"><a href="#cb78-294" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-295"><a href="#cb78-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-296"><a href="#cb78-296" aria-hidden="true" tabindex="-1"></a>| Metric                              | Results                                              |</span>
<span id="cb78-297"><a href="#cb78-297" aria-hidden="true" tabindex="-1"></a>|-------------------------------------|------------------------------------------------------|</span>
<span id="cb78-298"><a href="#cb78-298" aria-hidden="true" tabindex="-1"></a>| Min. number of reads                | <span class="in">`r min_read_ps`</span>                                      |</span>
<span id="cb78-299"><a href="#cb78-299" aria-hidden="true" tabindex="-1"></a>| Max. number of reads                | <span class="in">`r max_read_ps`</span>                                      |</span>
<span id="cb78-300"><a href="#cb78-300" aria-hidden="true" tabindex="-1"></a>| Total number of reads               | <span class="in">`r total_reads_ps`</span>                                   |</span>
<span id="cb78-301"><a href="#cb78-301" aria-hidden="true" tabindex="-1"></a>| Average number of reads             | <span class="in">`r mean_reads_ps`</span>                                    |</span>
<span id="cb78-302"><a href="#cb78-302" aria-hidden="true" tabindex="-1"></a>| Median number of reads              | <span class="in">`r median_reads_ps`</span>                                  |</span>
<span id="cb78-303"><a href="#cb78-303" aria-hidden="true" tabindex="-1"></a>| Total ASVs                          | <span class="in">`r total_asvs_ps`</span>                                    |</span>
<span id="cb78-304"><a href="#cb78-304" aria-hidden="true" tabindex="-1"></a>| Sparsity                            | <span class="in">`r sparsity_ps`</span>                                      |</span>
<span id="cb78-305"><a href="#cb78-305" aria-hidden="true" tabindex="-1"></a>| Any ASVs sum to 1 or less?          | <span class="in">`r isTRUE(singleton_ps &gt;= 1)`</span>                        |</span>
<span id="cb78-306"><a href="#cb78-306" aria-hidden="true" tabindex="-1"></a>| Number of singleton ASVs            | <span class="in">`r singleton_ps`</span>                                     |</span>
<span id="cb78-307"><a href="#cb78-307" aria-hidden="true" tabindex="-1"></a>| Percent of ASVs that are singletons | <span class="in">`r singleton_ps_perc`</span>                                |</span>
<span id="cb78-308"><a href="#cb78-308" aria-hidden="true" tabindex="-1"></a>| Number of sample variables are:     | <span class="in">`r length(sample_data(ps))`</span>    (<span class="in">`r colnames(samdf)`</span>) |</span>
<span id="cb78-309"><a href="#cb78-309" aria-hidden="true" tabindex="-1"></a>**B**. Add two final columns containing the ASV sequences and ASV IDs. This will be useful later when trying to export a fasta file. We can also take a look at the phyloseq object.</span>
<span id="cb78-310"><a href="#cb78-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-313"><a href="#cb78-313" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-314"><a href="#cb78-314" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(<span class="fu">tax_table</span>(ps)) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Kingdom"</span>, <span class="st">"Phylum"</span>, <span class="st">"Class"</span>, <span class="st">"Order"</span>,</span>
<span id="cb78-315"><a href="#cb78-315" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Family"</span>, <span class="st">"Genus"</span>, <span class="st">"ASV_SEQ"</span>, <span class="st">"ASV_ID"</span>)</span>
<span id="cb78-316"><a href="#cb78-316" aria-hidden="true" tabindex="-1"></a>ps</span>
<span id="cb78-317"><a href="#cb78-317" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-318"><a href="#cb78-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-321"><a href="#cb78-321" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-322"><a href="#cb78-322" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-323"><a href="#cb78-323" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-324"><a href="#cb78-324" aria-hidden="true" tabindex="-1"></a>ps</span>
<span id="cb78-325"><a href="#cb78-325" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-326"><a href="#cb78-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-327"><a href="#cb78-327" aria-hidden="true" tabindex="-1"></a>**C**. Export sequence and taxonomy tables for the unadulterated phyloseq object for later use. We will use the prefix <span class="in">`full`</span> to indicate that these are the *raw* outputs.</span>
<span id="cb78-328"><a href="#cb78-328" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-331"><a href="#cb78-331" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-332"><a href="#cb78-332" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">tax_table</span>(ps),</span>
<span id="cb78-333"><a href="#cb78-333" aria-hidden="true" tabindex="-1"></a>            <span class="st">"files/data-prep/tables/ssu18_full_tax_table.txt"</span>,</span>
<span id="cb78-334"><a href="#cb78-334" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">NA</span>)</span>
<span id="cb78-335"><a href="#cb78-335" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ps)),</span>
<span id="cb78-336"><a href="#cb78-336" aria-hidden="true" tabindex="-1"></a>            <span class="st">"files/data-prep/tables/ssu18_full_seq_table.txt"</span>,</span>
<span id="cb78-337"><a href="#cb78-337" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">NA</span>)</span>
<span id="cb78-338"><a href="#cb78-338" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-339"><a href="#cb78-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-340"><a href="#cb78-340" aria-hidden="true" tabindex="-1"></a><span class="fu">## Remove Contaminants &amp; Unwanted Taxa</span></span>
<span id="cb78-341"><a href="#cb78-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-342"><a href="#cb78-342" aria-hidden="true" tabindex="-1"></a>Let's see if we have any potential contaminants. We can use some <span class="co">[</span><span class="ot">inline R code</span><span class="co">](https://rmarkdown.rstudio.com/lesson-4.html)</span> to see the taxonomy table for any taxa of interest.</span>
<span id="cb78-343"><a href="#cb78-343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-344"><a href="#cb78-344" aria-hidden="true" tabindex="-1"></a>:::{.callout-note}</span>
<span id="cb78-345"><a href="#cb78-345" aria-hidden="true" tabindex="-1"></a>The code (hidden by default) is written as `<span class="in">` `</span>r</span>
<span id="cb78-346"><a href="#cb78-346" aria-hidden="true" tabindex="-1"></a>"Chloroplast" %in% tax_table(ps)` ``.</span>
<span id="cb78-347"><a href="#cb78-347" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb78-348"><a href="#cb78-348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-349"><a href="#cb78-349" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Are Mitochondria present? <span class="in">`r "Mitochondria" %in% tax_table(ps)`</span></span>
<span id="cb78-350"><a href="#cb78-350" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Are Chloroplast present? <span class="in">`r "Chloroplast" %in% tax_table(ps)`</span></span>
<span id="cb78-351"><a href="#cb78-351" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Are Eukaryota present? <span class="in">`r "Eukaryota" %in% tax_table(ps)`</span></span>
<span id="cb78-352"><a href="#cb78-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-353"><a href="#cb78-353" aria-hidden="true" tabindex="-1"></a>Let's remove these taxa---Eukaryota because we used bacterial/archaeal primers, Mitochondria because those are likely from  eukaryotes, and Chloroplast because those are likely from plants. We must do each of these in turn using phyloseq and it gets a little messy.</span>
<span id="cb78-354"><a href="#cb78-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-355"><a href="#cb78-355" aria-hidden="true" tabindex="-1"></a>Why messy? The <span class="in">`subset_taxa`</span> command removes anything that is <span class="in">`NA`</span> for the specified taxonomic level or above. For example, lets say you run the <span class="in">`subset_taxa`</span> command using <span class="in">`Family != "Mitochondria`</span>". Seems like you should get a phyloseq object with everything except Mitochondria. But actually the command not only gets rid of Mitochondria but everything else that has <span class="in">`NA`</span> for Family and above. In my experience this is not well documented and I had to dig through the files to figure out what was happening.</span>
<span id="cb78-356"><a href="#cb78-356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-357"><a href="#cb78-357" aria-hidden="true" tabindex="-1"></a>Anyway, to remove the taxa we do the following:</span>
<span id="cb78-358"><a href="#cb78-358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-359"><a href="#cb78-359" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Subset the taxa and generate a <span class="in">`ps`</span> object of just the taxa of interest,</span>
<span id="cb78-360"><a href="#cb78-360" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Select the ASV column only, turn it into a factor, and use this to remove <span class="kw">&lt;INSERT</span> <span class="er">TAXA</span><span class="kw">&gt;</span> from the <span class="in">`ps`</span> object.</span>
<span id="cb78-361"><a href="#cb78-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-362"><a href="#cb78-362" aria-hidden="true" tabindex="-1"></a><span class="fu">### Remove  Mitochondria ASVs</span></span>
<span id="cb78-363"><a href="#cb78-363" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-364"><a href="#cb78-364" aria-hidden="true" tabindex="-1"></a>Remember the original data set contained <span class="in">`r ntaxa(ps)`</span> ASVs. Here we generate a file with mitochondria ASVs only.</span>
<span id="cb78-365"><a href="#cb78-365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-368"><a href="#cb78-368" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-369"><a href="#cb78-369" aria-hidden="true" tabindex="-1"></a>MT1 <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(ps, Family <span class="sc">==</span> <span class="st">"Mitochondria"</span>)</span>
<span id="cb78-370"><a href="#cb78-370" aria-hidden="true" tabindex="-1"></a>MT1 <span class="ot">&lt;-</span>  <span class="fu">as</span>(<span class="fu">tax_table</span>(MT1), <span class="st">"matrix"</span>)</span>
<span id="cb78-371"><a href="#cb78-371" aria-hidden="true" tabindex="-1"></a>MT1 <span class="ot">&lt;-</span> MT1[, <span class="dv">8</span>]</span>
<span id="cb78-372"><a href="#cb78-372" aria-hidden="true" tabindex="-1"></a>MT1df <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(MT1)</span>
<span id="cb78-373"><a href="#cb78-373" aria-hidden="true" tabindex="-1"></a>goodTaxa <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">taxa_names</span>(ps), MT1df)</span>
<span id="cb78-374"><a href="#cb78-374" aria-hidden="true" tabindex="-1"></a>ps_no_mito <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(goodTaxa, ps)</span>
<span id="cb78-375"><a href="#cb78-375" aria-hidden="true" tabindex="-1"></a>ps_no_mito</span>
<span id="cb78-376"><a href="#cb78-376" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-377"><a href="#cb78-377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-380"><a href="#cb78-380" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-381"><a href="#cb78-381" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-382"><a href="#cb78-382" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-383"><a href="#cb78-383" aria-hidden="true" tabindex="-1"></a>ps_no_mito</span>
<span id="cb78-384"><a href="#cb78-384" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-385"><a href="#cb78-385" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-386"><a href="#cb78-386" aria-hidden="true" tabindex="-1"></a>Looks like this removed **`r ntaxa(ps) - ntaxa(ps_no_mito)` Mitochondria ASVs**. We will duplicate the code block to remove other groups.</span>
<span id="cb78-387"><a href="#cb78-387" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-388"><a href="#cb78-388" aria-hidden="true" tabindex="-1"></a><span class="fu">### Remove Chloroplast ASVs</span></span>
<span id="cb78-389"><a href="#cb78-389" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-390"><a href="#cb78-390" aria-hidden="true" tabindex="-1"></a>And again with Chloroplast ASVs only.</span>
<span id="cb78-391"><a href="#cb78-391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-394"><a href="#cb78-394" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-395"><a href="#cb78-395" aria-hidden="true" tabindex="-1"></a>CH1 <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(ps_no_mito, Order <span class="sc">==</span> <span class="st">"Chloroplast"</span>)</span>
<span id="cb78-396"><a href="#cb78-396" aria-hidden="true" tabindex="-1"></a>CH1 <span class="ot">&lt;-</span>  <span class="fu">as</span>(<span class="fu">tax_table</span>(CH1), <span class="st">"matrix"</span>)</span>
<span id="cb78-397"><a href="#cb78-397" aria-hidden="true" tabindex="-1"></a>CH1 <span class="ot">&lt;-</span> CH1[, <span class="dv">8</span>]</span>
<span id="cb78-398"><a href="#cb78-398" aria-hidden="true" tabindex="-1"></a>CH1df <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(CH1)</span>
<span id="cb78-399"><a href="#cb78-399" aria-hidden="true" tabindex="-1"></a>goodTaxa <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">taxa_names</span>(ps_no_mito), CH1df)</span>
<span id="cb78-400"><a href="#cb78-400" aria-hidden="true" tabindex="-1"></a>ps_no_chloro <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(goodTaxa, ps_no_mito)</span>
<span id="cb78-401"><a href="#cb78-401" aria-hidden="true" tabindex="-1"></a>ps_no_chloro</span>
<span id="cb78-402"><a href="#cb78-402" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-403"><a href="#cb78-403" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-406"><a href="#cb78-406" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-407"><a href="#cb78-407" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-408"><a href="#cb78-408" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-409"><a href="#cb78-409" aria-hidden="true" tabindex="-1"></a>ps_no_chloro</span>
<span id="cb78-410"><a href="#cb78-410" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-411"><a href="#cb78-411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-412"><a href="#cb78-412" aria-hidden="true" tabindex="-1"></a>The code removed an additional **`r ntaxa(ps_no_mito) - ntaxa(ps_no_chloro)` Chloroplast ASVs**.</span>
<span id="cb78-413"><a href="#cb78-413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-414"><a href="#cb78-414" aria-hidden="true" tabindex="-1"></a><span class="fu">### Remove Eukaryota ASVs</span></span>
<span id="cb78-415"><a href="#cb78-415" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-416"><a href="#cb78-416" aria-hidden="true" tabindex="-1"></a>And again with Eukaryota ASVs only.</span>
<span id="cb78-417"><a href="#cb78-417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-420"><a href="#cb78-420" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-421"><a href="#cb78-421" aria-hidden="true" tabindex="-1"></a>EU1 <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(ps_no_chloro, Kingdom <span class="sc">==</span> <span class="st">"Eukaryota"</span>)</span>
<span id="cb78-422"><a href="#cb78-422" aria-hidden="true" tabindex="-1"></a>EU1 <span class="ot">&lt;-</span>  <span class="fu">as</span>(<span class="fu">tax_table</span>(EU1), <span class="st">"matrix"</span>)</span>
<span id="cb78-423"><a href="#cb78-423" aria-hidden="true" tabindex="-1"></a>EU1 <span class="ot">&lt;-</span> EU1[, <span class="dv">8</span>]</span>
<span id="cb78-424"><a href="#cb78-424" aria-hidden="true" tabindex="-1"></a>EU1df <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(EU1)</span>
<span id="cb78-425"><a href="#cb78-425" aria-hidden="true" tabindex="-1"></a>goodTaxa <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">taxa_names</span>(ps_no_chloro), EU1df)</span>
<span id="cb78-426"><a href="#cb78-426" aria-hidden="true" tabindex="-1"></a>ps_no_euk <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(goodTaxa, ps_no_chloro)</span>
<span id="cb78-427"><a href="#cb78-427" aria-hidden="true" tabindex="-1"></a>ps_no_euk</span>
<span id="cb78-428"><a href="#cb78-428" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-429"><a href="#cb78-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-432"><a href="#cb78-432" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-433"><a href="#cb78-433" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-434"><a href="#cb78-434" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-435"><a href="#cb78-435" aria-hidden="true" tabindex="-1"></a>ps_no_euk</span>
<span id="cb78-436"><a href="#cb78-436" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-437"><a href="#cb78-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-438"><a href="#cb78-438" aria-hidden="true" tabindex="-1"></a>The code removed an additional **`r ntaxa(ps_no_chloro) - ntaxa(ps_no_euk)` Eukaryota ASVs** from the <span class="in">`ps`</span> object.</span>
<span id="cb78-439"><a href="#cb78-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-440"><a href="#cb78-440" aria-hidden="true" tabindex="-1"></a><span class="fu">### Remove any Kingdom NAs</span></span>
<span id="cb78-441"><a href="#cb78-441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-442"><a href="#cb78-442" aria-hidden="true" tabindex="-1"></a>Here we can just use the straight up <span class="in">`subset_taxa`</span> command since we do not need to worry about any ranks above Kingdom also being removed.</span>
<span id="cb78-443"><a href="#cb78-443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-446"><a href="#cb78-446" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-447"><a href="#cb78-447" aria-hidden="true" tabindex="-1"></a>ps_filt <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(ps_no_euk, <span class="sc">!</span><span class="fu">is.na</span>(Kingdom))</span>
<span id="cb78-448"><a href="#cb78-448" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-449"><a href="#cb78-449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-450"><a href="#cb78-450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-453"><a href="#cb78-453" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-454"><a href="#cb78-454" aria-hidden="true" tabindex="-1"></a>ps_work_o <span class="ot">&lt;-</span> ps_filt</span>
<span id="cb78-455"><a href="#cb78-455" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-456"><a href="#cb78-456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-457"><a href="#cb78-457" aria-hidden="true" tabindex="-1"></a>The code eliminated an additional **`r ntaxa(ps_no_euk) - ntaxa(ps_filt)` Kingdom level NA ASVs** from the phyloseq object.</span>
<span id="cb78-458"><a href="#cb78-458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-459"><a href="#cb78-459" aria-hidden="true" tabindex="-1"></a><span class="fu">## Rename NA taxonomic ranks</span></span>
<span id="cb78-460"><a href="#cb78-460" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-461"><a href="#cb78-461" aria-hidden="true" tabindex="-1"></a>Phyloseq has an odd way of dealing with taxonomic ranks that have no value---in other words,  **NA** in the tax table. The first thing we are going to do before moving forward is to change all of the *NA*s to have a value of the next highest classified rank. For example, `ASV26` is not classified at the Genus level but is at Family level (Xanthobacteraceae). So we change the Genus name to *Family_Xanthobacteraceae*. The code for comes from these two posts on the phyloseq GitHub, both by <span class="co">[</span><span class="ot">MSMortensen</span><span class="co">](https://github.com/MSMortensen)</span>: issue <span class="co">[</span><span class="ot">#850</span><span class="co">](https://github.com/joey711/phyloseq/issues/850#issuecomment-394771087)</span> and issue <span class="co">[</span><span class="ot">#990</span><span class="co">](https://github.com/joey711/phyloseq/issues/990#issuecomment-424618425)</span>.</span>
<span id="cb78-462"><a href="#cb78-462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-463"><a href="#cb78-463" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; One thing this code does is reassign the functions </span><span class="in">`class`</span><span class="at"> and </span><span class="in">`order`</span><span class="at"> to taxonomic ranks. This can cause issues if you need these functions.</span></span>
<span id="cb78-464"><a href="#cb78-464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-465"><a href="#cb78-465" aria-hidden="true" tabindex="-1"></a>So you need to run something like this <span class="in">`rm(class, order, phylum, kingdom)`</span> at the end of the code to remove these as variables. For now, I have not come up with a better solution.</span>
<span id="cb78-466"><a href="#cb78-466" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-469"><a href="#cb78-469" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-470"><a href="#cb78-470" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb78-471"><a href="#cb78-471" aria-hidden="true" tabindex="-1"></a>tax.clean <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ps_filt))</span>
<span id="cb78-472"><a href="#cb78-472" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>){ tax.clean[,i] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tax.clean[,i])}</span>
<span id="cb78-473"><a href="#cb78-473" aria-hidden="true" tabindex="-1"></a>tax.clean[<span class="fu">is.na</span>(tax.clean)] <span class="ot">&lt;-</span> <span class="st">""</span></span>
<span id="cb78-474"><a href="#cb78-474" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-475"><a href="#cb78-475" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(tax.clean)){</span>
<span id="cb78-476"><a href="#cb78-476" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (tax.clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">""</span>){</span>
<span id="cb78-477"><a href="#cb78-477" aria-hidden="true" tabindex="-1"></a>        kingdom <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">"k_"</span>, tax.clean[i,<span class="dv">1</span>], <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb78-478"><a href="#cb78-478" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> kingdom</span>
<span id="cb78-479"><a href="#cb78-479" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">""</span>){</span>
<span id="cb78-480"><a href="#cb78-480" aria-hidden="true" tabindex="-1"></a>        phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">"p_"</span>, tax.clean[i,<span class="dv">2</span>], <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb78-481"><a href="#cb78-481" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">3</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb78-482"><a href="#cb78-482" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">4</span>] <span class="sc">==</span> <span class="st">""</span>){</span>
<span id="cb78-483"><a href="#cb78-483" aria-hidden="true" tabindex="-1"></a>        class <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">"c_"</span>, tax.clean[i,<span class="dv">3</span>], <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb78-484"><a href="#cb78-484" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">4</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> class</span>
<span id="cb78-485"><a href="#cb78-485" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">5</span>] <span class="sc">==</span> <span class="st">""</span>){</span>
<span id="cb78-486"><a href="#cb78-486" aria-hidden="true" tabindex="-1"></a>        order <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">"o_"</span>, tax.clean[i,<span class="dv">4</span>], <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb78-487"><a href="#cb78-487" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">5</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> order</span>
<span id="cb78-488"><a href="#cb78-488" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">6</span>] <span class="sc">==</span> <span class="st">""</span>){</span>
<span id="cb78-489"><a href="#cb78-489" aria-hidden="true" tabindex="-1"></a>        tax.clean<span class="sc">$</span>Genus[i] <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">"f"</span>,tax.clean<span class="sc">$</span>Family[i], <span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb78-490"><a href="#cb78-490" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb78-491"><a href="#cb78-491" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb78-492"><a href="#cb78-492" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps_filt) <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tax.clean)</span>
<span id="cb78-493"><a href="#cb78-493" aria-hidden="true" tabindex="-1"></a><span class="fu">rank_names</span>(ps_filt)</span>
<span id="cb78-494"><a href="#cb78-494" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(class, order, phylum, kingdom)</span>
<span id="cb78-495"><a href="#cb78-495" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-496"><a href="#cb78-496" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-497"><a href="#cb78-497" aria-hidden="true" tabindex="-1"></a>Still the same ranks. That's good. What about the new groups? Let's take a peak at some families.</span>
<span id="cb78-498"><a href="#cb78-498" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-501"><a href="#cb78-501" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-502"><a href="#cb78-502" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-503"><a href="#cb78-503" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_filt, <span class="st">"Family"</span>), <span class="dv">16</span>)</span>
<span id="cb78-504"><a href="#cb78-504" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-505"><a href="#cb78-505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-506"><a href="#cb78-506" aria-hidden="true" tabindex="-1"></a>Nice. Bye-bye *NA*</span>
<span id="cb78-507"><a href="#cb78-507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-508"><a href="#cb78-508" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; *Note*. The original code mentioned above is written for data sets that have species-level designations.</span></span>
<span id="cb78-509"><a href="#cb78-509" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-510"><a href="#cb78-510" aria-hidden="true" tabindex="-1"></a>Since this data set does not contain species, I modified the code to stop at the genus level. If your data set has species, my modifications will not work for you.</span>
<span id="cb78-511"><a href="#cb78-511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-512"><a href="#cb78-512" aria-hidden="true" tabindex="-1"></a>Finally, we rename the ps object. This is now our working data set.</span>
<span id="cb78-513"><a href="#cb78-513" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-516"><a href="#cb78-516" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-517"><a href="#cb78-517" aria-hidden="true" tabindex="-1"></a>ps_work <span class="ot">&lt;-</span> ps_filt</span>
<span id="cb78-518"><a href="#cb78-518" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-519"><a href="#cb78-519" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-520"><a href="#cb78-520" aria-hidden="true" tabindex="-1"></a><span class="fu">## Add Phylogenetic Tree</span></span>
<span id="cb78-521"><a href="#cb78-521" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-522"><a href="#cb78-522" aria-hidden="true" tabindex="-1"></a>One final task is to add a phylogenetic tree to the phyloseq object.</span>
<span id="cb78-523"><a href="#cb78-523" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-526"><a href="#cb78-526" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-527"><a href="#cb78-527" aria-hidden="true" tabindex="-1"></a>ps_work_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(ps_work), <span class="at">rooted =</span> <span class="cn">TRUE</span>,</span>
<span id="cb78-528"><a href="#cb78-528" aria-hidden="true" tabindex="-1"></a>                      <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(ps_work))</span>
<span id="cb78-529"><a href="#cb78-529" aria-hidden="true" tabindex="-1"></a>ps_work <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(ps_work,</span>
<span id="cb78-530"><a href="#cb78-530" aria-hidden="true" tabindex="-1"></a>                          sample_data,</span>
<span id="cb78-531"><a href="#cb78-531" aria-hidden="true" tabindex="-1"></a>                          ps_work_tree)</span>
<span id="cb78-532"><a href="#cb78-532" aria-hidden="true" tabindex="-1"></a>ps_work_o_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(ps_work_o), <span class="at">rooted =</span> <span class="cn">TRUE</span>,</span>
<span id="cb78-533"><a href="#cb78-533" aria-hidden="true" tabindex="-1"></a>                      <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(ps_work_o))</span>
<span id="cb78-534"><a href="#cb78-534" aria-hidden="true" tabindex="-1"></a>ps_work_o <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(ps_work_o,</span>
<span id="cb78-535"><a href="#cb78-535" aria-hidden="true" tabindex="-1"></a>                          sample_data,</span>
<span id="cb78-536"><a href="#cb78-536" aria-hidden="true" tabindex="-1"></a>                          ps_work_o_tree)</span>
<span id="cb78-537"><a href="#cb78-537" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-538"><a href="#cb78-538" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-539"><a href="#cb78-539" aria-hidden="true" tabindex="-1"></a><span class="fu">## Data Set Summary</span></span>
<span id="cb78-540"><a href="#cb78-540" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-541"><a href="#cb78-541" aria-hidden="true" tabindex="-1"></a>Now that we have removed the NC and sample with zero reads, we can summarize the data set. Let's start with the in our working phyloseq object using the <span class="in">`summarize_phyloseq`</span> from the <span class="co">[</span><span class="ot">microbiome R package</span><span class="co">](https://github.com/microbiome/microbiome/)</span> <span class="co">[</span><span class="ot">@lahti2017microbiome</span><span class="co">]</span> as we did above.</span>
<span id="cb78-542"><a href="#cb78-542" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-545"><a href="#cb78-545" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-546"><a href="#cb78-546" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-547"><a href="#cb78-547" aria-hidden="true" tabindex="-1"></a>min_read_ps_work <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">readcount</span>(ps_work))</span>
<span id="cb78-548"><a href="#cb78-548" aria-hidden="true" tabindex="-1"></a>max_read_ps_work <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">readcount</span>(ps_work))</span>
<span id="cb78-549"><a href="#cb78-549" aria-hidden="true" tabindex="-1"></a>total_reads_ps_work <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">readcount</span>(ps_work))</span>
<span id="cb78-550"><a href="#cb78-550" aria-hidden="true" tabindex="-1"></a>mean_reads_ps_work <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">mean</span>(<span class="fu">readcount</span>(ps_work)), <span class="at">digits =</span> <span class="dv">0</span>)</span>
<span id="cb78-551"><a href="#cb78-551" aria-hidden="true" tabindex="-1"></a>median_reads_ps_work <span class="ot">&lt;-</span> <span class="fu">median</span>(<span class="fu">readcount</span>(ps_work))</span>
<span id="cb78-552"><a href="#cb78-552" aria-hidden="true" tabindex="-1"></a>total_asvs_ps_work <span class="ot">&lt;-</span> <span class="fu">ntaxa</span>(ps_work)</span>
<span id="cb78-553"><a href="#cb78-553" aria-hidden="true" tabindex="-1"></a>singleton_ps_work <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>(<span class="fu">ntaxa</span>(<span class="fu">rare</span>(ps_work,</span>
<span id="cb78-554"><a href="#cb78-554" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">detection =</span> <span class="dv">1</span>,</span>
<span id="cb78-555"><a href="#cb78-555" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">prevalence =</span> <span class="dv">0</span>)),</span>
<span id="cb78-556"><a href="#cb78-556" aria-hidden="true" tabindex="-1"></a>                              <span class="at">error=</span><span class="cf">function</span>(err) <span class="cn">NA</span>)</span>
<span id="cb78-557"><a href="#cb78-557" aria-hidden="true" tabindex="-1"></a>singleton_ps_work_perc <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>(<span class="fu">round</span>((<span class="dv">100</span><span class="sc">*</span>(<span class="fu">ntaxa</span>(<span class="fu">rare</span>(</span>
<span id="cb78-558"><a href="#cb78-558" aria-hidden="true" tabindex="-1"></a>  ps_work, <span class="at">detection =</span> <span class="dv">1</span>, <span class="at">prevalence =</span> <span class="dv">0</span>)) <span class="sc">/</span> <span class="fu">ntaxa</span>(ps_work))), <span class="at">digits =</span> <span class="dv">3</span>),</span>
<span id="cb78-559"><a href="#cb78-559" aria-hidden="true" tabindex="-1"></a>  <span class="at">error =</span> <span class="cf">function</span>(err) <span class="cn">NA</span>)</span>
<span id="cb78-560"><a href="#cb78-560" aria-hidden="true" tabindex="-1"></a>sparsity_ps_work <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">length</span>(<span class="fu">which</span>(</span>
<span id="cb78-561"><a href="#cb78-561" aria-hidden="true" tabindex="-1"></a>  <span class="fu">abundances</span>(ps_work) <span class="sc">==</span> <span class="dv">0</span>))<span class="sc">/</span><span class="fu">length</span>(<span class="fu">abundances</span>(ps)), <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb78-562"><a href="#cb78-562" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-563"><a href="#cb78-563" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-564"><a href="#cb78-564" aria-hidden="true" tabindex="-1"></a>| Metric                              | Results                                                   |</span>
<span id="cb78-565"><a href="#cb78-565" aria-hidden="true" tabindex="-1"></a>|-------------------------------------|-----------------------------------------------------------|</span>
<span id="cb78-566"><a href="#cb78-566" aria-hidden="true" tabindex="-1"></a>| Min. number of reads                | <span class="in">`r min_read_ps_work`</span>                                      |</span>
<span id="cb78-567"><a href="#cb78-567" aria-hidden="true" tabindex="-1"></a>| Max. number of reads                | <span class="in">`r max_read_ps_work`</span>                                      |</span>
<span id="cb78-568"><a href="#cb78-568" aria-hidden="true" tabindex="-1"></a>| Total number of reads               | <span class="in">`r total_reads_ps_work`</span>                                   |</span>
<span id="cb78-569"><a href="#cb78-569" aria-hidden="true" tabindex="-1"></a>| Average number of reads             | <span class="in">`r mean_reads_ps_work`</span>                                    |</span>
<span id="cb78-570"><a href="#cb78-570" aria-hidden="true" tabindex="-1"></a>| Median number of reads              | <span class="in">`r median_reads_ps_work`</span>                                  |</span>
<span id="cb78-571"><a href="#cb78-571" aria-hidden="true" tabindex="-1"></a>| Total ASVs                          | <span class="in">`r total_asvs_ps_work`</span>                                    |</span>
<span id="cb78-572"><a href="#cb78-572" aria-hidden="true" tabindex="-1"></a>| Sparsity                            | <span class="in">`r sparsity_ps_work`</span>                                      |</span>
<span id="cb78-573"><a href="#cb78-573" aria-hidden="true" tabindex="-1"></a>| Any ASVs sum to 1 or less?          | <span class="in">`r isTRUE(singleton_ps_work &gt;= 1)`</span>                        |</span>
<span id="cb78-574"><a href="#cb78-574" aria-hidden="true" tabindex="-1"></a>| Number of singleton ASVs            | <span class="in">`r singleton_ps_work`</span>                                     |</span>
<span id="cb78-575"><a href="#cb78-575" aria-hidden="true" tabindex="-1"></a>| Percent of ASVs that are singletons | <span class="in">`r singleton_ps_work_perc`</span>                                |</span>
<span id="cb78-576"><a href="#cb78-576" aria-hidden="true" tabindex="-1"></a>| Number of sample variables are:     | <span class="in">`r length(sample_data(ps_work))`</span>    (<span class="in">`r colnames(samdf)`</span>) |</span>
<span id="cb78-577"><a href="#cb78-577" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-578"><a href="#cb78-578" aria-hidden="true" tabindex="-1"></a>We can also generate a summary table of total reads &amp; ASVs for each sample. You can sort the table, download a copy, or filter by search term. Here is the code to generate the data for the table. First, we create data frames that hold total reads and ASVs for each sample.</span>
<span id="cb78-579"><a href="#cb78-579" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-582"><a href="#cb78-582" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-583"><a href="#cb78-583" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb78-584"><a href="#cb78-584" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> <span class="fu">sample_sums</span>(ps_work)</span>
<span id="cb78-585"><a href="#cb78-585" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(total_reads, <span class="at">make.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb78-586"><a href="#cb78-586" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> total_reads <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"Sample_ID"</span>)</span>
<span id="cb78-587"><a href="#cb78-587" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-588"><a href="#cb78-588" aria-hidden="true" tabindex="-1"></a>total_asvs <span class="ot">&lt;-</span> <span class="fu">estimate_richness</span>(ps_work,</span>
<span id="cb78-589"><a href="#cb78-589" aria-hidden="true" tabindex="-1"></a>                                <span class="at">measures =</span> <span class="st">"Observed"</span>)</span>
<span id="cb78-590"><a href="#cb78-590" aria-hidden="true" tabindex="-1"></a>total_asvs <span class="ot">&lt;-</span> total_asvs <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"Sample_ID"</span>)</span>
<span id="cb78-591"><a href="#cb78-591" aria-hidden="true" tabindex="-1"></a>total_asvs<span class="sc">$</span>Sample_ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">'</span><span class="sc">\\</span><span class="st">.'</span>, <span class="st">'-'</span>, total_asvs<span class="sc">$</span>Sample_ID)</span>
<span id="cb78-592"><a href="#cb78-592" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-593"><a href="#cb78-593" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-594"><a href="#cb78-594" aria-hidden="true" tabindex="-1"></a>And then we merge these two data frames with the sample data frame.</span>
<span id="cb78-595"><a href="#cb78-595" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-598"><a href="#cb78-598" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-599"><a href="#cb78-599" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb78-600"><a href="#cb78-600" aria-hidden="true" tabindex="-1"></a>sam_details <span class="ot">&lt;-</span> samdf</span>
<span id="cb78-601"><a href="#cb78-601" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sam_details) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb78-602"><a href="#cb78-602" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(sam_details) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Sample_ID"</span>, <span class="st">"Plot"</span>, <span class="st">"Depth"</span>,</span>
<span id="cb78-603"><a href="#cb78-603" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">"Treatment"</span>, <span class="st">"Temp"</span>, <span class="st">"Pair"</span>)</span>
<span id="cb78-604"><a href="#cb78-604" aria-hidden="true" tabindex="-1"></a>tmp_tab1 <span class="ot">&lt;-</span> seq_table</span>
<span id="cb78-605"><a href="#cb78-605" aria-hidden="true" tabindex="-1"></a>tmp_tab1[, <span class="fu">c</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">7</span>,<span class="dv">11</span>)] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb78-606"><a href="#cb78-606" aria-hidden="true" tabindex="-1"></a>tmp_tab2 <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(sam_details, total_reads) <span class="sc">%&gt;%</span></span>
<span id="cb78-607"><a href="#cb78-607" aria-hidden="true" tabindex="-1"></a>                <span class="fu">left_join</span>(., total_asvs) <span class="sc">%&gt;%</span></span>
<span id="cb78-608"><a href="#cb78-608" aria-hidden="true" tabindex="-1"></a>                <span class="fu">left_join</span>(., tmp_tab1)</span>
<span id="cb78-609"><a href="#cb78-609" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-610"><a href="#cb78-610" aria-hidden="true" tabindex="-1"></a>reads_lost <span class="ot">&lt;-</span> tmp_tab2<span class="sc">$</span><span class="st">`</span><span class="at">final reads</span><span class="st">`</span> <span class="sc">-</span> tmp_tab2<span class="sc">$</span>total_reads </span>
<span id="cb78-611"><a href="#cb78-611" aria-hidden="true" tabindex="-1"></a>asvs_lost <span class="ot">&lt;-</span> tmp_tab2<span class="sc">$</span><span class="st">`</span><span class="at">no. ASVs</span><span class="st">`</span> <span class="sc">-</span> tmp_tab2<span class="sc">$</span>Observed </span>
<span id="cb78-612"><a href="#cb78-612" aria-hidden="true" tabindex="-1"></a>tmp_tab2<span class="sc">$</span>reads_lost <span class="ot">&lt;-</span> reads_lost</span>
<span id="cb78-613"><a href="#cb78-613" aria-hidden="true" tabindex="-1"></a>tmp_tab2<span class="sc">$</span>asvs_lost <span class="ot">&lt;-</span> asvs_lost</span>
<span id="cb78-614"><a href="#cb78-614" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-615"><a href="#cb78-615" aria-hidden="true" tabindex="-1"></a>seq_table2 <span class="ot">&lt;-</span> tmp_tab2[, <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>,<span class="dv">12</span>,<span class="dv">13</span>)]</span>
<span id="cb78-616"><a href="#cb78-616" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(seq_table2) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"sample_id"</span>, <span class="st">"PLOT"</span>, <span class="st">"DEPTH"</span>,</span>
<span id="cb78-617"><a href="#cb78-617" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"TREAT"</span>, <span class="st">"TEMP"</span>, <span class="st">"PAIR"</span>, <span class="st">"total_reads"</span>,</span>
<span id="cb78-618"><a href="#cb78-618" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"total_ASVs"</span>, <span class="st">"reads_lost"</span>, <span class="st">"ASVs_lost"</span>)</span>
<span id="cb78-619"><a href="#cb78-619" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-620"><a href="#cb78-620" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-621"><a href="#cb78-621" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab("seq_summary_final_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb78-622"><a href="#cb78-622" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-625"><a href="#cb78-625" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-626"><a href="#cb78-626" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-627"><a href="#cb78-627" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-628"><a href="#cb78-628" aria-hidden="true" tabindex="-1"></a>seq_table2 <span class="sc">%&gt;%</span></span>
<span id="cb78-629"><a href="#cb78-629" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb78-630"><a href="#cb78-630" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"ssu18_sample_summary_final"</span>,</span>
<span id="cb78-631"><a href="#cb78-631" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb78-632"><a href="#cb78-632" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb78-633"><a href="#cb78-633" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb78-634"><a href="#cb78-634" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb78-635"><a href="#cb78-635" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb78-636"><a href="#cb78-636" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb78-637"><a href="#cb78-637" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-638"><a href="#cb78-638" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table2,</span>
<span id="cb78-639"><a href="#cb78-639" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb78-640"><a href="#cb78-640" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb78-641"><a href="#cb78-641" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">0</span>),</span>
<span id="cb78-642"><a href="#cb78-642" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">TRUE</span>, <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb78-643"><a href="#cb78-643" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb78-644"><a href="#cb78-644" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb78-645"><a href="#cb78-645" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb78-646"><a href="#cb78-646" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Sample ID"</span>, </span>
<span id="cb78-647"><a href="#cb78-647" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb78-648"><a href="#cb78-648" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>),</span>
<span id="cb78-649"><a href="#cb78-649" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>), </span>
<span id="cb78-650"><a href="#cb78-650" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb78-651"><a href="#cb78-651" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">150</span>, <span class="at">footer =</span> <span class="st">"Total reads"</span>)</span>
<span id="cb78-652"><a href="#cb78-652" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb78-653"><a href="#cb78-653" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">5</span>, </span>
<span id="cb78-654"><a href="#cb78-654" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb78-655"><a href="#cb78-655" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb78-656"><a href="#cb78-656" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb78-657"><a href="#cb78-657" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb78-658"><a href="#cb78-658" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb78-659"><a href="#cb78-659" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-660"><a href="#cb78-660" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-661"><a href="#cb78-661" aria-hidden="true" tabindex="-1"></a><span class="fu">## Review</span></span>
<span id="cb78-662"><a href="#cb78-662" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-663"><a href="#cb78-663" aria-hidden="true" tabindex="-1"></a>We have a few different options to play with. At this point it is difficult to say which we will use or whether additional objects need to be created. Here is a summary of the objects we have.</span>
<span id="cb78-664"><a href="#cb78-664" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-665"><a href="#cb78-665" aria-hidden="true" tabindex="-1"></a>The full phyloseq object *before* removing contaminants.</span>
<span id="cb78-666"><a href="#cb78-666" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-669"><a href="#cb78-669" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-670"><a href="#cb78-670" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-671"><a href="#cb78-671" aria-hidden="true" tabindex="-1"></a>ps</span>
<span id="cb78-672"><a href="#cb78-672" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps, <span class="st">"Family"</span>), <span class="dv">16</span>)</span>
<span id="cb78-673"><a href="#cb78-673" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-674"><a href="#cb78-674" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-675"><a href="#cb78-675" aria-hidden="true" tabindex="-1"></a>The phyloseq object *before* changing NA ranks.</span>
<span id="cb78-676"><a href="#cb78-676" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-679"><a href="#cb78-679" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-680"><a href="#cb78-680" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-681"><a href="#cb78-681" aria-hidden="true" tabindex="-1"></a>ps_work_o</span>
<span id="cb78-682"><a href="#cb78-682" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_work_o, <span class="st">"Family"</span>), <span class="dv">16</span>)</span>
<span id="cb78-683"><a href="#cb78-683" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-684"><a href="#cb78-684" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-685"><a href="#cb78-685" aria-hidden="true" tabindex="-1"></a>The phyloseq object *after* changing NA ranks.</span>
<span id="cb78-686"><a href="#cb78-686" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-689"><a href="#cb78-689" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-690"><a href="#cb78-690" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-691"><a href="#cb78-691" aria-hidden="true" tabindex="-1"></a>ps_work</span>
<span id="cb78-692"><a href="#cb78-692" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_work, <span class="st">"Family"</span>), <span class="dv">16</span>)</span>
<span id="cb78-693"><a href="#cb78-693" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-694"><a href="#cb78-694" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-695"><a href="#cb78-695" aria-hidden="true" tabindex="-1"></a><span class="fu">## Final Steps</span></span>
<span id="cb78-696"><a href="#cb78-696" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-697"><a href="#cb78-697" aria-hidden="true" tabindex="-1"></a>The last thing tasks to complete are to **a**) save copies of the taxonomy and sequence tables and **b**) save an image of the workflow for the next section.</span>
<span id="cb78-698"><a href="#cb78-698" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-701"><a href="#cb78-701" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-702"><a href="#cb78-702" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">tax_table</span>(ps_work),</span>
<span id="cb78-703"><a href="#cb78-703" aria-hidden="true" tabindex="-1"></a>            <span class="st">"files/data-prep/tables/ssu18_work_tax_table.txt"</span>,</span>
<span id="cb78-704"><a href="#cb78-704" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">NA</span>)</span>
<span id="cb78-705"><a href="#cb78-705" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ps_work)),</span>
<span id="cb78-706"><a href="#cb78-706" aria-hidden="true" tabindex="-1"></a>            <span class="st">"files/data-prep/tables/ssu18_work_seq_table.txt"</span>,</span>
<span id="cb78-707"><a href="#cb78-707" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">NA</span>)</span>
<span id="cb78-708"><a href="#cb78-708" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-709"><a href="#cb78-709" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-712"><a href="#cb78-712" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-713"><a href="#cb78-713" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-714"><a href="#cb78-714" aria-hidden="true" tabindex="-1"></a><span class="co"># PAGE BUILD</span></span>
<span id="cb78-715"><a href="#cb78-715" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="st">"page_build/data_prep_ssu18_wf.rdata"</span>)</span>
<span id="cb78-716"><a href="#cb78-716" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-717"><a href="#cb78-717" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-720"><a href="#cb78-720" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-721"><a href="#cb78-721" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-722"><a href="#cb78-722" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ps, <span class="st">"files/data-prep/rdata/ssu18_ps.rds"</span>)</span>
<span id="cb78-723"><a href="#cb78-723" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ps_work, <span class="st">"files/data-prep/rdata/ssu18_ps_work.rds"</span>)</span>
<span id="cb78-724"><a href="#cb78-724" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ps_work_o, <span class="st">"files/data-prep/rdata/ssu18_ps_work_o.rds"</span>)</span>
<span id="cb78-725"><a href="#cb78-725" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-726"><a href="#cb78-726" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-727"><a href="#cb78-727" aria-hidden="true" tabindex="-1"></a><span class="fu"># ITS Data</span></span>
<span id="cb78-728"><a href="#cb78-728" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-731"><a href="#cb78-731" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-732"><a href="#cb78-732" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb78-733"><a href="#cb78-733" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-734"><a href="#cb78-734" aria-hidden="true" tabindex="-1"></a><span class="do">## ONLY FOR PAGE BUILD</span></span>
<span id="cb78-735"><a href="#cb78-735" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb78-736"><a href="#cb78-736" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"page_build/data_prep_its18_wf.rdata"</span>)</span>
<span id="cb78-737"><a href="#cb78-737" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-738"><a href="#cb78-738" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-741"><a href="#cb78-741" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-742"><a href="#cb78-742" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-743"><a href="#cb78-743" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-744"><a href="#cb78-744" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb78-745"><a href="#cb78-745" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the caption(s) with captioner</span></span>
<span id="cb78-746"><a href="#cb78-746" aria-hidden="true" tabindex="-1"></a>caption_tab <span class="ot">&lt;-</span> <span class="fu">captioner</span>(<span class="at">prefix =</span> <span class="st">"(ITS) Table"</span>, <span class="at">suffix =</span> <span class="st">" |"</span>, <span class="at">style =</span> <span class="st">"b"</span>)</span>
<span id="cb78-747"><a href="#cb78-747" aria-hidden="true" tabindex="-1"></a>caption_fig <span class="ot">&lt;-</span> <span class="fu">captioner</span>(<span class="at">prefix =</span> <span class="st">"(ITS) Figure"</span>, <span class="at">suffix =</span> <span class="st">" |"</span>, <span class="at">style =</span> <span class="st">"b"</span>)</span>
<span id="cb78-748"><a href="#cb78-748" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a function for referring to the tables in text</span></span>
<span id="cb78-749"><a href="#cb78-749" aria-hidden="true" tabindex="-1"></a>ref <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">str_extract</span>(x, <span class="st">"[^|]*"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb78-750"><a href="#cb78-750" aria-hidden="true" tabindex="-1"></a>  <span class="fu">trimws</span>(<span class="at">which =</span> <span class="st">"right"</span>, <span class="at">whitespace =</span> <span class="st">"[ ]"</span>)</span>
<span id="cb78-751"><a href="#cb78-751" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-752"><a href="#cb78-752" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-755"><a href="#cb78-755" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-756"><a href="#cb78-756" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-757"><a href="#cb78-757" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-758"><a href="#cb78-758" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"assets/captions/captions_data_prep_its.R"</span>)</span>
<span id="cb78-759"><a href="#cb78-759" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-760"><a href="#cb78-760" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-761"><a href="#cb78-761" aria-hidden="true" tabindex="-1"></a><span class="fu">## Prerequisites</span></span>
<span id="cb78-762"><a href="#cb78-762" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-763"><a href="#cb78-763" aria-hidden="true" tabindex="-1"></a>In order to run this workflow, you either need to run the corresponding <span class="co">[</span><span class="ot">DADA2 Workflow for 2018 ITS</span><span class="co">](dada2-high-temp.html#its-data)</span> or begin with the output from that workflow, <span class="in">`data_prep_its18_wf.rdata`</span>. See the <span class="co">[</span><span class="ot">Data Availability</span><span class="co">](data-availability.html)</span> page for complete details.</span>
<span id="cb78-764"><a href="#cb78-764" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-765"><a href="#cb78-765" aria-hidden="true" tabindex="-1"></a>Unless otherwise noted, we primarily use <span class="co">[</span><span class="ot">phyloseq</span><span class="co">](https://joey711.github.io/phyloseq/)</span> <span class="co">[</span><span class="ot">@mcmurdie2013phyloseq</span><span class="co">]</span> in this section of the workflow to prepare the 2018 ITS data set for community analyses. We prepare the data set by curating samples, removing contaminants, and creating phyloseq objects.</span>
<span id="cb78-766"><a href="#cb78-766" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-767"><a href="#cb78-767" aria-hidden="true" tabindex="-1"></a><span class="fu">## Read Counts Assessment</span></span>
<span id="cb78-768"><a href="#cb78-768" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-769"><a href="#cb78-769" aria-hidden="true" tabindex="-1"></a>Before we begin, let's create a summary table containing some basic sample metadata and the read count data from the <span class="co">[</span><span class="ot">DADA2 workflow</span><span class="co">](dada2-high-temp.html#its-data)</span>. We need to inspect how total reads changed from to beginning to the end of the workflow. While we are at it, let's create more intuitive Sample IDs. For more details on how reads changed at each step of the workflow, see the <span class="co">[</span><span class="ot">summary table</span><span class="co">](dada2-high-temp.html#track-reads-through-workflow-1)</span> at the end of the DADA2 section.</span>
<span id="cb78-770"><a href="#cb78-770" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-771"><a href="#cb78-771" aria-hidden="true" tabindex="-1"></a>Table headers are as follows:</span>
<span id="cb78-772"><a href="#cb78-772" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-773"><a href="#cb78-773" aria-hidden="true" tabindex="-1"></a>| Header                    | Description                                                                              |</span>
<span id="cb78-774"><a href="#cb78-774" aria-hidden="true" tabindex="-1"></a>|---------------------------|------------------------------------------------------------------------------------------|</span>
<span id="cb78-775"><a href="#cb78-775" aria-hidden="true" tabindex="-1"></a>| <span class="in">`Sample ID`</span>               | the new sample ID based on Plot number, Depth, Treatment, Temperature, &amp; Pair            |</span>
<span id="cb78-776"><a href="#cb78-776" aria-hidden="true" tabindex="-1"></a>| <span class="in">`Plot`</span>                    | the experimental plot number                                                             |</span>
<span id="cb78-777"><a href="#cb78-777" aria-hidden="true" tabindex="-1"></a>| <span class="in">`Depth`</span>                   | the depth where the sample was collected                                                 |</span>
<span id="cb78-778"><a href="#cb78-778" aria-hidden="true" tabindex="-1"></a>| <span class="in">`Treatment`</span>               | warm or control                                                                          |</span>
<span id="cb78-779"><a href="#cb78-779" aria-hidden="true" tabindex="-1"></a>| <span class="in">`Temp`</span>                    | temperature of soil heating (0C, +4C, or +8C)                                            |</span>
<span id="cb78-780"><a href="#cb78-780" aria-hidden="true" tabindex="-1"></a>| <span class="in">`Pair`</span>                    | control/treatment coupling                                                               |</span>
<span id="cb78-781"><a href="#cb78-781" aria-hidden="true" tabindex="-1"></a>| <span class="in">`input`</span>                   | number of raw reads                                                                      |</span>
<span id="cb78-782"><a href="#cb78-782" aria-hidden="true" tabindex="-1"></a>| <span class="in">`nochim`</span>                  | final read count after removing chimeras                                                 |</span>
<span id="cb78-783"><a href="#cb78-783" aria-hidden="true" tabindex="-1"></a>| <span class="in">`Remain`</span>                  | percent of reads remaining from <span class="in">`input`</span> to <span class="in">`nonchim`</span>                                     |</span>
<span id="cb78-784"><a href="#cb78-784" aria-hidden="true" tabindex="-1"></a>| <span class="in">`FastqID`</span>                 | base name of the fastq file                                                              |</span>
<span id="cb78-785"><a href="#cb78-785" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-788"><a href="#cb78-788" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-789"><a href="#cb78-789" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-790"><a href="#cb78-790" aria-hidden="true" tabindex="-1"></a>tmp_tab1 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb78-791"><a href="#cb78-791" aria-hidden="true" tabindex="-1"></a>  <span class="st">"files/dada2/tables/its18_sample_seq_info.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb78-792"><a href="#cb78-792" aria-hidden="true" tabindex="-1"></a>tmp_tab1<span class="sc">$</span>root <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, tmp_tab1<span class="sc">$</span>Fastq_ID_Forward)</span>
<span id="cb78-793"><a href="#cb78-793" aria-hidden="true" tabindex="-1"></a>tmp_tab1[<span class="dv">9</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_tab1)<span class="sc">-</span><span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb78-794"><a href="#cb78-794" aria-hidden="true" tabindex="-1"></a>tmp_tab1 <span class="ot">&lt;-</span> tmp_tab1[, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">8</span>,<span class="dv">2</span><span class="sc">:</span><span class="dv">7</span>)]</span>
<span id="cb78-795"><a href="#cb78-795" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-796"><a href="#cb78-796" aria-hidden="true" tabindex="-1"></a>tmp_tab2 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb78-797"><a href="#cb78-797" aria-hidden="true" tabindex="-1"></a>  <span class="st">"files/dada2/tables/its18_read_changes.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>,  <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb78-798"><a href="#cb78-798" aria-hidden="true" tabindex="-1"></a>tmp_tab2[<span class="dv">2</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb78-799"><a href="#cb78-799" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-800"><a href="#cb78-800" aria-hidden="true" tabindex="-1"></a>tmp_tab3 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb78-801"><a href="#cb78-801" aria-hidden="true" tabindex="-1"></a>  <span class="st">"files/dada2/tables/its18_dada_compare_by_samp.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>,  <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb78-802"><a href="#cb78-802" aria-hidden="true" tabindex="-1"></a>tmp_tab3 <span class="ot">&lt;-</span> tmp_tab3[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">7</span>)]</span>
<span id="cb78-803"><a href="#cb78-803" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-804"><a href="#cb78-804" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> <span class="fu">left_join</span>(tmp_tab1, tmp_tab2, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Sample_ID"</span> <span class="ot">=</span> <span class="st">"Sample_ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb78-805"><a href="#cb78-805" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., tmp_tab3, <span class="at">by =</span> <span class="st">"Sample_ID"</span>)</span>
<span id="cb78-806"><a href="#cb78-806" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-807"><a href="#cb78-807" aria-hidden="true" tabindex="-1"></a>seq_table<span class="sc">$</span>per_reads_kept <span class="ot">&lt;-</span> <span class="fu">round</span>(seq_table<span class="sc">$</span>nonchim<span class="sc">/</span>seq_table<span class="sc">$</span>raw_reads, <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb78-808"><a href="#cb78-808" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(seq_table) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Sample_ID"</span>, <span class="st">"root name"</span>, <span class="st">"Plot"</span>, <span class="st">"Treatment"</span>, </span>
<span id="cb78-809"><a href="#cb78-809" aria-hidden="true" tabindex="-1"></a>                         <span class="st">"Temp"</span>, <span class="st">"Depth"</span>, <span class="st">"Pair"</span>, <span class="st">"input reads"</span>, </span>
<span id="cb78-810"><a href="#cb78-810" aria-hidden="true" tabindex="-1"></a>                         <span class="st">"final reads"</span>, <span class="st">"no. ASVs"</span>, <span class="st">"per. reads remain"</span>)</span>
<span id="cb78-811"><a href="#cb78-811" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb78-812"><a href="#cb78-812" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-813"><a href="#cb78-813" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-814"><a href="#cb78-814" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab("seq_summary_its")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb78-815"><a href="#cb78-815" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-818"><a href="#cb78-818" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-819"><a href="#cb78-819" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-820"><a href="#cb78-820" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-821"><a href="#cb78-821" aria-hidden="true" tabindex="-1"></a>seq_table <span class="sc">%&gt;%</span></span>
<span id="cb78-822"><a href="#cb78-822" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb78-823"><a href="#cb78-823" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"its18_sample_summary"</span>,</span>
<span id="cb78-824"><a href="#cb78-824" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb78-825"><a href="#cb78-825" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb78-826"><a href="#cb78-826" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb78-827"><a href="#cb78-827" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb78-828"><a href="#cb78-828" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb78-829"><a href="#cb78-829" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb78-830"><a href="#cb78-830" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-831"><a href="#cb78-831" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb78-832"><a href="#cb78-832" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb78-833"><a href="#cb78-833" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb78-834"><a href="#cb78-834" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">1</span>),</span>
<span id="cb78-835"><a href="#cb78-835" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">TRUE</span>, <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb78-836"><a href="#cb78-836" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb78-837"><a href="#cb78-837" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb78-838"><a href="#cb78-838" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb78-839"><a href="#cb78-839" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Sample ID"</span>, </span>
<span id="cb78-840"><a href="#cb78-840" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb78-841"><a href="#cb78-841" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>),</span>
<span id="cb78-842"><a href="#cb78-842" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>), </span>
<span id="cb78-843"><a href="#cb78-843" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb78-844"><a href="#cb78-844" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">150</span>, <span class="at">footer =</span> <span class="st">"Total reads"</span>)</span>
<span id="cb78-845"><a href="#cb78-845" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb78-846"><a href="#cb78-846" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">5</span>, </span>
<span id="cb78-847"><a href="#cb78-847" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb78-848"><a href="#cb78-848" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb78-849"><a href="#cb78-849" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb78-850"><a href="#cb78-850" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb78-851"><a href="#cb78-851" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb78-852"><a href="#cb78-852" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-853"><a href="#cb78-853" aria-hidden="true" tabindex="-1"></a>cutoff <span class="ot">&lt;-</span> <span class="dv">1000</span></span>
<span id="cb78-854"><a href="#cb78-854" aria-hidden="true" tabindex="-1"></a>remove_sam <span class="ot">&lt;-</span> seq_table[seq_table<span class="sc">$</span><span class="st">`</span><span class="at">final reads</span><span class="st">`</span> <span class="sc">&lt;=</span> cutoff, ]</span>
<span id="cb78-855"><a href="#cb78-855" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-856"><a href="#cb78-856" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-857"><a href="#cb78-857" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-858"><a href="#cb78-858" aria-hidden="true" tabindex="-1"></a>The following samples have fewer than **`r cutoff`** reads at the end of the workflow:</span>
<span id="cb78-859"><a href="#cb78-859" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-860"><a href="#cb78-860" aria-hidden="true" tabindex="-1"></a>**`r remove_sam$Sample_ID`**.</span>
<span id="cb78-861"><a href="#cb78-861" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-862"><a href="#cb78-862" aria-hidden="true" tabindex="-1"></a>We can also plot the final read count by the temperature where the sample was collected.</span>
<span id="cb78-863"><a href="#cb78-863" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-866"><a href="#cb78-866" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-867"><a href="#cb78-867" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-868"><a href="#cb78-868" aria-hidden="true" tabindex="-1"></a>wong_pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#D55E00"</span>, <span class="st">"#56B4E9"</span>, <span class="st">"#009E73"</span>, <span class="st">"#0072B2"</span>,</span>
<span id="cb78-869"><a href="#cb78-869" aria-hidden="true" tabindex="-1"></a>              <span class="st">"#F0E442"</span>, <span class="st">"#CC79A7"</span>,  <span class="st">"#E69F00"</span>, <span class="st">"#7F7F7F"</span>,</span>
<span id="cb78-870"><a href="#cb78-870" aria-hidden="true" tabindex="-1"></a>              <span class="st">"#000000"</span>)</span>
<span id="cb78-871"><a href="#cb78-871" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-872"><a href="#cb78-872" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-873"><a href="#cb78-873" aria-hidden="true" tabindex="-1"></a><span class="fu">## Defining Groups</span></span>
<span id="cb78-874"><a href="#cb78-874" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-875"><a href="#cb78-875" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>Load the data packet produced in the final step of the DADA2 workflow. This packet (<span class="in">`its18_dada2_wf.rdata`</span>) contains the ASV-by-sample table and the ASV taxonomy table.</span>
<span id="cb78-876"><a href="#cb78-876" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-877"><a href="#cb78-877" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>Rename the samples so names have plot and Depth info. For convenience, we renamed any Negative Controls (NC) to <span class="in">`P00_D00-000_NNN`</span>.</span>
<span id="cb78-878"><a href="#cb78-878" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-879"><a href="#cb78-879" aria-hidden="true" tabindex="-1"></a><span class="ss">3. </span>After we load the data packet we next need to format sample names and define groups.</span>
<span id="cb78-880"><a href="#cb78-880" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-883"><a href="#cb78-883" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-884"><a href="#cb78-884" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb78-885"><a href="#cb78-885" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> seq_table[<span class="fu">order</span>(seq_table<span class="sc">$</span><span class="st">`</span><span class="at">root name</span><span class="st">`</span>), ]</span>
<span id="cb78-886"><a href="#cb78-886" aria-hidden="true" tabindex="-1"></a>tmp_names <span class="ot">&lt;-</span> seq_table<span class="sc">$</span><span class="st">`</span><span class="at">Sample_ID</span><span class="st">`</span></span>
<span id="cb78-887"><a href="#cb78-887" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"files/dada2/rdata/its18_dada2_wf.rdata"</span>)</span>
<span id="cb78-888"><a href="#cb78-888" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(seq_table<span class="sc">$</span><span class="st">`</span><span class="at">root name</span><span class="st">`</span>, <span class="fu">rownames</span>(seqtab))</span>
<span id="cb78-889"><a href="#cb78-889" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(seqtab) <span class="ot">&lt;-</span> tmp_names</span>
<span id="cb78-890"><a href="#cb78-890" aria-hidden="true" tabindex="-1"></a>samples.out <span class="ot">&lt;-</span> <span class="fu">rownames</span>(seqtab)</span>
<span id="cb78-891"><a href="#cb78-891" aria-hidden="true" tabindex="-1"></a>sample_name <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">1</span>, <span class="dv">999</span>)</span>
<span id="cb78-892"><a href="#cb78-892" aria-hidden="true" tabindex="-1"></a>plot <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">0</span>, <span class="dv">3</span>)</span>
<span id="cb78-893"><a href="#cb78-893" aria-hidden="true" tabindex="-1"></a>depth <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">6</span>, <span class="dv">11</span>)</span>
<span id="cb78-894"><a href="#cb78-894" aria-hidden="true" tabindex="-1"></a>treatment <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">13</span>, <span class="dv">13</span>)</span>
<span id="cb78-895"><a href="#cb78-895" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">14</span>, <span class="dv">14</span>)</span>
<span id="cb78-896"><a href="#cb78-896" aria-hidden="true" tabindex="-1"></a>pair <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">15</span>, <span class="dv">15</span>)</span>
<span id="cb78-897"><a href="#cb78-897" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-898"><a href="#cb78-898" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-899"><a href="#cb78-899" aria-hidden="true" tabindex="-1"></a>We have a total of **`r length(unique(sample_name))`** samples, **`r length(unique(pair))`** sample pairs, from **`r length(unique(plot))`** plots (**`r length(unique(depth))`** depth). There are **`r length(unique(treatment))`** treatments corresponding to **`r length(unique(temp))`** different temperature regimes.</span>
<span id="cb78-900"><a href="#cb78-900" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-901"><a href="#cb78-901" aria-hidden="true" tabindex="-1"></a><span class="ss">4. </span>And finally we define a sample data frame that holds the different groups we extracted from the sample names.</span>
<span id="cb78-902"><a href="#cb78-902" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-905"><a href="#cb78-905" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-906"><a href="#cb78-906" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb78-907"><a href="#cb78-907" aria-hidden="true" tabindex="-1"></a>samdf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">SamName =</span> sample_name,</span>
<span id="cb78-908"><a href="#cb78-908" aria-hidden="true" tabindex="-1"></a>                    <span class="at">PLOT =</span> plot,</span>
<span id="cb78-909"><a href="#cb78-909" aria-hidden="true" tabindex="-1"></a>                    <span class="at">DEPTH =</span> depth,</span>
<span id="cb78-910"><a href="#cb78-910" aria-hidden="true" tabindex="-1"></a>                    <span class="at">TREAT =</span> treatment,</span>
<span id="cb78-911"><a href="#cb78-911" aria-hidden="true" tabindex="-1"></a>                    <span class="at">TEMP =</span> temp,</span>
<span id="cb78-912"><a href="#cb78-912" aria-hidden="true" tabindex="-1"></a>                    <span class="at">PAIR =</span> pair)</span>
<span id="cb78-913"><a href="#cb78-913" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(samdf) <span class="ot">&lt;-</span> samples.out</span>
<span id="cb78-914"><a href="#cb78-914" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-915"><a href="#cb78-915" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-918"><a href="#cb78-918" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-919"><a href="#cb78-919" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-920"><a href="#cb78-920" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-921"><a href="#cb78-921" aria-hidden="true" tabindex="-1"></a>example_sam <span class="ot">&lt;-</span> <span class="st">"P08_D00_010_C0D"</span></span>
<span id="cb78-922"><a href="#cb78-922" aria-hidden="true" tabindex="-1"></a>qpair <span class="ot">&lt;-</span> samdf[example_sam,]<span class="sc">$</span>PAIR</span>
<span id="cb78-923"><a href="#cb78-923" aria-hidden="true" tabindex="-1"></a>qgroup <span class="ot">&lt;-</span> samdf <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(PAIR <span class="sc">==</span> qpair)</span>
<span id="cb78-924"><a href="#cb78-924" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-925"><a href="#cb78-925" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-926"><a href="#cb78-926" aria-hidden="true" tabindex="-1"></a>For example, <span class="in">`r example_sam`</span> is the sample heated at **+`r samdf[example_sam,]$TEMP`C** from plot **`r samdf[example_sam,]$PLOT`**, collected from  a depth of `r samdf[example_sam,]$DEPTH`cm. This sample is in group **`r qpair`** which consists of <span class="in">`r qgroup$SamName`</span>.</span>
<span id="cb78-927"><a href="#cb78-927" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-928"><a href="#cb78-928" aria-hidden="true" tabindex="-1"></a><span class="fu">## Create a Phyloseq Object</span></span>
<span id="cb78-929"><a href="#cb78-929" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-930"><a href="#cb78-930" aria-hidden="true" tabindex="-1"></a><span class="fu">###  Fix Taxon Names</span></span>
<span id="cb78-931"><a href="#cb78-931" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-932"><a href="#cb78-932" aria-hidden="true" tabindex="-1"></a>The UNITE database we used for classification adds a little prefix to every rank call, like <span class="in">`k__`</span> for Kingdom, <span class="in">`o__`</span> for Order, etc. To keep in step with our other analyses, we will remove these tags. Of course, we also make a ps object with the original names just in case. At this point we will also remove the Species rank since it is essentially useless.</span>
<span id="cb78-933"><a href="#cb78-933" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-936"><a href="#cb78-936" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-937"><a href="#cb78-937" aria-hidden="true" tabindex="-1"></a>tax_mod_o <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(tax)</span>
<span id="cb78-938"><a href="#cb78-938" aria-hidden="true" tabindex="-1"></a>tax_mod_o<span class="sc">$</span>Species <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb78-939"><a href="#cb78-939" aria-hidden="true" tabindex="-1"></a>tax_mod_o <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tax_mod_o)</span>
<span id="cb78-940"><a href="#cb78-940" aria-hidden="true" tabindex="-1"></a>tax_mod <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"k__"</span>, <span class="st">""</span>, tax_mod_o)</span>
<span id="cb78-941"><a href="#cb78-941" aria-hidden="true" tabindex="-1"></a>tax_mod <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"p__"</span>, <span class="st">""</span>, tax_mod)</span>
<span id="cb78-942"><a href="#cb78-942" aria-hidden="true" tabindex="-1"></a>tax_mod <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"c__"</span>, <span class="st">""</span>, tax_mod)</span>
<span id="cb78-943"><a href="#cb78-943" aria-hidden="true" tabindex="-1"></a>tax_mod <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"o__"</span>, <span class="st">""</span>, tax_mod)</span>
<span id="cb78-944"><a href="#cb78-944" aria-hidden="true" tabindex="-1"></a>tax_mod <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"f__"</span>, <span class="st">""</span>, tax_mod)</span>
<span id="cb78-945"><a href="#cb78-945" aria-hidden="true" tabindex="-1"></a>tax_mod <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"g__"</span>, <span class="st">""</span>, tax_mod)</span>
<span id="cb78-946"><a href="#cb78-946" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-947"><a href="#cb78-947" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-948"><a href="#cb78-948" aria-hidden="true" tabindex="-1"></a>**A**. The first step is to rename the amplicon sequence variants (ASVs) so the designations are a bit more user friendly. By default, DADA2 names each ASV by its unique sequence so that data can be directly compared across studies (which is great). But this convention can get cumbersome downstream, so we rename the ASVs using a simpler convention---ASV1, ASV2, ASV3, and so on.</span>
<span id="cb78-949"><a href="#cb78-949" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-950"><a href="#cb78-950" aria-hidden="true" tabindex="-1"></a>:::{.callout-note}</span>
<span id="cb78-951"><a href="#cb78-951" aria-hidden="true" tabindex="-1"></a>A phyloseq object contains ASV table (taxa abundances), sample metadata, and taxonomy table (mapping between ASVs and higher-level taxonomic classifications).</span>
<span id="cb78-952"><a href="#cb78-952" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb78-953"><a href="#cb78-953" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-956"><a href="#cb78-956" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-957"><a href="#cb78-957" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb78-958"><a href="#cb78-958" aria-hidden="true" tabindex="-1"></a><span class="co"># this create the phyloseq object</span></span>
<span id="cb78-959"><a href="#cb78-959" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(<span class="fu">otu_table</span>(seqtab, <span class="at">taxa_are_rows =</span> <span class="cn">FALSE</span>),</span>
<span id="cb78-960"><a href="#cb78-960" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">sample_data</span>(samdf), <span class="fu">tax_table</span>(tax_mod))</span>
<span id="cb78-961"><a href="#cb78-961" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">tax_table</span>(ps),</span>
<span id="cb78-962"><a href="#cb78-962" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">rownames</span>(<span class="fu">tax_table</span>(ps)))</span>
<span id="cb78-963"><a href="#cb78-963" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-964"><a href="#cb78-964" aria-hidden="true" tabindex="-1"></a><span class="co"># adding unique ASV names</span></span>
<span id="cb78-965"><a href="#cb78-965" aria-hidden="true" tabindex="-1"></a><span class="fu">taxa_names</span>(ps) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"ASV"</span>, <span class="fu">seq</span>(<span class="fu">ntaxa</span>(ps)))</span>
<span id="cb78-966"><a href="#cb78-966" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">tax_table</span>(ps),</span>
<span id="cb78-967"><a href="#cb78-967" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">rownames</span>(<span class="fu">tax_table</span>(ps)))</span>
<span id="cb78-968"><a href="#cb78-968" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">taxa_names</span>(ps))</span>
<span id="cb78-969"><a href="#cb78-969" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-970"><a href="#cb78-970" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-973"><a href="#cb78-973" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-974"><a href="#cb78-974" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-975"><a href="#cb78-975" aria-hidden="true" tabindex="-1"></a><span class="co"># this create the phyloseq object</span></span>
<span id="cb78-976"><a href="#cb78-976" aria-hidden="true" tabindex="-1"></a>ps_o <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(<span class="fu">otu_table</span>(seqtab, <span class="at">taxa_are_rows =</span> <span class="cn">FALSE</span>),</span>
<span id="cb78-977"><a href="#cb78-977" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">sample_data</span>(samdf), <span class="fu">tax_table</span>(tax_mod_o))</span>
<span id="cb78-978"><a href="#cb78-978" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps_o) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">tax_table</span>(ps_o),</span>
<span id="cb78-979"><a href="#cb78-979" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">rownames</span>(<span class="fu">tax_table</span>(ps_o)))</span>
<span id="cb78-980"><a href="#cb78-980" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-981"><a href="#cb78-981" aria-hidden="true" tabindex="-1"></a><span class="co"># adding unique ASV names</span></span>
<span id="cb78-982"><a href="#cb78-982" aria-hidden="true" tabindex="-1"></a><span class="fu">taxa_names</span>(ps_o) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"ASV"</span>, <span class="fu">seq</span>(<span class="fu">ntaxa</span>(ps_o)))</span>
<span id="cb78-983"><a href="#cb78-983" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps_o) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">tax_table</span>(ps_o),</span>
<span id="cb78-984"><a href="#cb78-984" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">rownames</span>(<span class="fu">tax_table</span>(ps_o)))</span>
<span id="cb78-985"><a href="#cb78-985" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">taxa_names</span>(ps_o))</span>
<span id="cb78-986"><a href="#cb78-986" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-987"><a href="#cb78-987" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-988"><a href="#cb78-988" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-991"><a href="#cb78-991" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-992"><a href="#cb78-992" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-993"><a href="#cb78-993" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-994"><a href="#cb78-994" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">taxa_names</span>(ps))</span>
<span id="cb78-995"><a href="#cb78-995" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-996"><a href="#cb78-996" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-997"><a href="#cb78-997" aria-hidden="true" tabindex="-1"></a>So the complete data set contains <span class="in">`r ntaxa(ps)`</span> ASVs. We can also use the <span class="co">[</span><span class="ot">microbiome R package</span><span class="co">](https://github.com/microbiome/microbiome/)</span> <span class="co">[</span><span class="ot">@lahti2017microbiome</span><span class="co">]</span> to get some additional summary data from the phyloseq object.</span>
<span id="cb78-998"><a href="#cb78-998" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1001"><a href="#cb78-1001" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1002"><a href="#cb78-1002" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-1003"><a href="#cb78-1003" aria-hidden="true" tabindex="-1"></a>min_read_ps <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">readcount</span>(ps))</span>
<span id="cb78-1004"><a href="#cb78-1004" aria-hidden="true" tabindex="-1"></a>max_read_ps <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">readcount</span>(ps))</span>
<span id="cb78-1005"><a href="#cb78-1005" aria-hidden="true" tabindex="-1"></a>total_reads_ps <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">readcount</span>(ps))</span>
<span id="cb78-1006"><a href="#cb78-1006" aria-hidden="true" tabindex="-1"></a>mean_reads_ps <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">mean</span>(<span class="fu">readcount</span>(ps)), <span class="at">digits =</span> <span class="dv">0</span>)</span>
<span id="cb78-1007"><a href="#cb78-1007" aria-hidden="true" tabindex="-1"></a>median_reads_ps <span class="ot">&lt;-</span> <span class="fu">median</span>(<span class="fu">readcount</span>(ps))</span>
<span id="cb78-1008"><a href="#cb78-1008" aria-hidden="true" tabindex="-1"></a>total_asvs_ps <span class="ot">&lt;-</span> <span class="fu">ntaxa</span>(ps)</span>
<span id="cb78-1009"><a href="#cb78-1009" aria-hidden="true" tabindex="-1"></a>singleton_ps <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>(<span class="fu">ntaxa</span>(<span class="fu">rare</span>(ps, <span class="at">detection =</span> <span class="dv">1</span>, <span class="at">prevalence =</span> <span class="dv">0</span>)),</span>
<span id="cb78-1010"><a href="#cb78-1010" aria-hidden="true" tabindex="-1"></a>                              <span class="at">error=</span><span class="cf">function</span>(err) <span class="cn">NA</span>)</span>
<span id="cb78-1011"><a href="#cb78-1011" aria-hidden="true" tabindex="-1"></a>singleton_ps_perc <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>(<span class="fu">round</span>((<span class="dv">100</span><span class="sc">*</span>(<span class="fu">ntaxa</span>(<span class="fu">rare</span>(ps, <span class="at">detection =</span> <span class="dv">1</span>, <span class="at">prevalence =</span> <span class="dv">0</span>)) <span class="sc">/</span></span>
<span id="cb78-1012"><a href="#cb78-1012" aria-hidden="true" tabindex="-1"></a>                                   <span class="fu">ntaxa</span>(ps))), <span class="at">digits =</span> <span class="dv">3</span>), <span class="at">error =</span> <span class="cf">function</span>(err) <span class="cn">NA</span>)</span>
<span id="cb78-1013"><a href="#cb78-1013" aria-hidden="true" tabindex="-1"></a>sparsity_ps <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">length</span>(<span class="fu">which</span>(<span class="fu">abundances</span>(ps) <span class="sc">==</span> <span class="dv">0</span>))<span class="sc">/</span><span class="fu">length</span>(<span class="fu">abundances</span>(ps)),</span>
<span id="cb78-1014"><a href="#cb78-1014" aria-hidden="true" tabindex="-1"></a>                     <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb78-1015"><a href="#cb78-1015" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1016"><a href="#cb78-1016" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1017"><a href="#cb78-1017" aria-hidden="true" tabindex="-1"></a>| Metric                              | Results                                              |</span>
<span id="cb78-1018"><a href="#cb78-1018" aria-hidden="true" tabindex="-1"></a>|-------------------------------------|------------------------------------------------------|</span>
<span id="cb78-1019"><a href="#cb78-1019" aria-hidden="true" tabindex="-1"></a>| Min. number of reads                | <span class="in">`r min_read_ps`</span>                                      |</span>
<span id="cb78-1020"><a href="#cb78-1020" aria-hidden="true" tabindex="-1"></a>| Max. number of reads                | <span class="in">`r max_read_ps`</span>                                      |</span>
<span id="cb78-1021"><a href="#cb78-1021" aria-hidden="true" tabindex="-1"></a>| Total number of reads               | <span class="in">`r total_reads_ps`</span>                                   |</span>
<span id="cb78-1022"><a href="#cb78-1022" aria-hidden="true" tabindex="-1"></a>| Average number of reads             | <span class="in">`r mean_reads_ps`</span>                                    |</span>
<span id="cb78-1023"><a href="#cb78-1023" aria-hidden="true" tabindex="-1"></a>| Median number of reads              | <span class="in">`r median_reads_ps`</span>                                  |</span>
<span id="cb78-1024"><a href="#cb78-1024" aria-hidden="true" tabindex="-1"></a>| Total ASVs                          | <span class="in">`r total_asvs_ps`</span>                                    |</span>
<span id="cb78-1025"><a href="#cb78-1025" aria-hidden="true" tabindex="-1"></a>| Sparsity                            | <span class="in">`r sparsity_ps`</span>                                      |</span>
<span id="cb78-1026"><a href="#cb78-1026" aria-hidden="true" tabindex="-1"></a>| Any ASVs sum to 1 or less?          | <span class="in">`r isTRUE(singleton_ps &gt;= 1)`</span>                        |</span>
<span id="cb78-1027"><a href="#cb78-1027" aria-hidden="true" tabindex="-1"></a>| Number of singleton ASVs            | <span class="in">`r singleton_ps`</span>                                     |</span>
<span id="cb78-1028"><a href="#cb78-1028" aria-hidden="true" tabindex="-1"></a>| Percent of ASVs that are singletons | <span class="in">`r singleton_ps_perc`</span>                                |</span>
<span id="cb78-1029"><a href="#cb78-1029" aria-hidden="true" tabindex="-1"></a>| Number of sample variables are:     | <span class="in">`r length(sample_data(ps))`</span>    (<span class="in">`r colnames(samdf)`</span>) |</span>
<span id="cb78-1030"><a href="#cb78-1030" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1031"><a href="#cb78-1031" aria-hidden="true" tabindex="-1"></a>**B**. Add two final columns containing the ASV sequences and ASV IDs. This will be useful later when trying to export a fasta file. We can also take a look at the phyloseq object.</span>
<span id="cb78-1032"><a href="#cb78-1032" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1035"><a href="#cb78-1035" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1036"><a href="#cb78-1036" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(<span class="fu">tax_table</span>(ps)) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Kingdom"</span>, <span class="st">"Phylum"</span>, <span class="st">"Class"</span>, <span class="st">"Order"</span>,</span>
<span id="cb78-1037"><a href="#cb78-1037" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Family"</span>, <span class="st">"Genus"</span>, <span class="st">"ASV_SEQ"</span>, <span class="st">"ASV_ID"</span>)</span>
<span id="cb78-1038"><a href="#cb78-1038" aria-hidden="true" tabindex="-1"></a><span class="fu">rank_names</span>(ps)</span>
<span id="cb78-1039"><a href="#cb78-1039" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1040"><a href="#cb78-1040" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1043"><a href="#cb78-1043" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1044"><a href="#cb78-1044" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-1045"><a href="#cb78-1045" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-1046"><a href="#cb78-1046" aria-hidden="true" tabindex="-1"></a>ps</span>
<span id="cb78-1047"><a href="#cb78-1047" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1048"><a href="#cb78-1048" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1051"><a href="#cb78-1051" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1052"><a href="#cb78-1052" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-1053"><a href="#cb78-1053" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(<span class="fu">tax_table</span>(ps_o)) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Kingdom"</span>, <span class="st">"Phylum"</span>, <span class="st">"Class"</span>, <span class="st">"Order"</span>,</span>
<span id="cb78-1054"><a href="#cb78-1054" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Family"</span>, <span class="st">"Genus"</span>, <span class="st">"ASV_SEQ"</span>, <span class="st">"ASV_ID"</span>)</span>
<span id="cb78-1055"><a href="#cb78-1055" aria-hidden="true" tabindex="-1"></a>ps_o</span>
<span id="cb78-1056"><a href="#cb78-1056" aria-hidden="true" tabindex="-1"></a><span class="fu">rank_names</span>(ps_o)</span>
<span id="cb78-1057"><a href="#cb78-1057" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1058"><a href="#cb78-1058" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1059"><a href="#cb78-1059" aria-hidden="true" tabindex="-1"></a>**C**. Export sequence and taxonomy tables for the unadulterated phyloseq object for later use. We will use the prefix <span class="in">`full`</span> to indicate that these are the *raw* outputs.</span>
<span id="cb78-1060"><a href="#cb78-1060" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1063"><a href="#cb78-1063" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1064"><a href="#cb78-1064" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb78-1065"><a href="#cb78-1065" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">tax_table</span>(ps),</span>
<span id="cb78-1066"><a href="#cb78-1066" aria-hidden="true" tabindex="-1"></a>            <span class="st">"files/data-prep/tables/its18_full_tax_table.txt"</span>,</span>
<span id="cb78-1067"><a href="#cb78-1067" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span>
<span id="cb78-1068"><a href="#cb78-1068" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ps)),</span>
<span id="cb78-1069"><a href="#cb78-1069" aria-hidden="true" tabindex="-1"></a>            <span class="st">"files/data-prep/tables/its18_full_seq_table.txt"</span>,</span>
<span id="cb78-1070"><a href="#cb78-1070" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span>
<span id="cb78-1071"><a href="#cb78-1071" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1072"><a href="#cb78-1072" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1073"><a href="#cb78-1073" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1076"><a href="#cb78-1076" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1077"><a href="#cb78-1077" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-1078"><a href="#cb78-1078" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">tax_table</span>(ps_o),</span>
<span id="cb78-1079"><a href="#cb78-1079" aria-hidden="true" tabindex="-1"></a>            <span class="st">"files/data-prep/tables/its18_full_tax_table_o.txt"</span>,</span>
<span id="cb78-1080"><a href="#cb78-1080" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span>
<span id="cb78-1081"><a href="#cb78-1081" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ps_o)),</span>
<span id="cb78-1082"><a href="#cb78-1082" aria-hidden="true" tabindex="-1"></a>            <span class="st">"files/data-prep/tables/its18_full_seq_table_o.txt"</span>,</span>
<span id="cb78-1083"><a href="#cb78-1083" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span>
<span id="cb78-1084"><a href="#cb78-1084" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1085"><a href="#cb78-1085" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1086"><a href="#cb78-1086" aria-hidden="true" tabindex="-1"></a>At this point we would normally remove unwanted taxa and/or negative controls. Since we do not have any of these for this dataset, we are going to save the phyloseq object with a new name. That way, if we do decide later on to remove some taxa it will be easier to plug in the code here and then continue with the rest of the workflow.</span>
<span id="cb78-1087"><a href="#cb78-1087" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1090"><a href="#cb78-1090" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1091"><a href="#cb78-1091" aria-hidden="true" tabindex="-1"></a>ps_filt <span class="ot">&lt;-</span> ps</span>
<span id="cb78-1092"><a href="#cb78-1092" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1093"><a href="#cb78-1093" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1094"><a href="#cb78-1094" aria-hidden="true" tabindex="-1"></a><span class="fu">## Remove Low-Count Samples</span></span>
<span id="cb78-1095"><a href="#cb78-1095" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1096"><a href="#cb78-1096" aria-hidden="true" tabindex="-1"></a>Next, we can remove samples with really low read counts, say less than 500 reads.</span>
<span id="cb78-1097"><a href="#cb78-1097" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1100"><a href="#cb78-1100" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1101"><a href="#cb78-1101" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: asis</span></span>
<span id="cb78-1102"><a href="#cb78-1102" aria-hidden="true" tabindex="-1"></a>ps_filt_no_low <span class="ot">&lt;-</span> <span class="fu">prune_samples</span>(<span class="fu">sample_sums</span>(ps_filt) <span class="sc">&gt;</span> <span class="dv">500</span>, ps_filt)</span>
<span id="cb78-1103"><a href="#cb78-1103" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1104"><a href="#cb78-1104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1107"><a href="#cb78-1107" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1108"><a href="#cb78-1108" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-1109"><a href="#cb78-1109" aria-hidden="true" tabindex="-1"></a>ps_filt_no_low</span>
<span id="cb78-1110"><a href="#cb78-1110" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1111"><a href="#cb78-1111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1112"><a href="#cb78-1112" aria-hidden="true" tabindex="-1"></a>We lost <span class="in">`r nsamples(ps) - nsamples(ps_filt_no_low)`</span> sample(s). After removing samples we need to check whether any ASVs ended up with no reads.</span>
<span id="cb78-1113"><a href="#cb78-1113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1116"><a href="#cb78-1116" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1117"><a href="#cb78-1117" aria-hidden="true" tabindex="-1"></a>no_reads <span class="ot">&lt;-</span> <span class="fu">taxa_sums</span>(ps_filt_no_low) <span class="sc">==</span> <span class="dv">0</span></span>
<span id="cb78-1118"><a href="#cb78-1118" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1119"><a href="#cb78-1119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1120"><a href="#cb78-1120" aria-hidden="true" tabindex="-1"></a>And we lost <span class="in">`r sum(no_reads)`</span> ASV(s). So now we must remove these from the ps object.</span>
<span id="cb78-1121"><a href="#cb78-1121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1124"><a href="#cb78-1124" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1125"><a href="#cb78-1125" aria-hidden="true" tabindex="-1"></a>ps_filt_no_nc <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(<span class="fu">taxa_sums</span>(ps_filt_no_low) <span class="sc">&gt;</span> <span class="dv">0</span>, ps_filt_no_low)</span>
<span id="cb78-1126"><a href="#cb78-1126" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1127"><a href="#cb78-1127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1130"><a href="#cb78-1130" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1131"><a href="#cb78-1131" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-1132"><a href="#cb78-1132" aria-hidden="true" tabindex="-1"></a>ps_filt_no_nc</span>
<span id="cb78-1133"><a href="#cb78-1133" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1134"><a href="#cb78-1134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1135"><a href="#cb78-1135" aria-hidden="true" tabindex="-1"></a><span class="fu">## Rename NA taxonomic ranks</span></span>
<span id="cb78-1136"><a href="#cb78-1136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1137"><a href="#cb78-1137" aria-hidden="true" tabindex="-1"></a>Phyloseq has an odd way of dealing with taxonomic ranks that have no value---in other words,  *NA* in the tax table. The first thing we are going to do before moving forward is to change all of the *NA*s to have a value of the next highest classified rank. For example, `ASV4` is not classified at the Genus level but is at Family level (Nitrososphaeraceae). So we change the Genus name to *Family_Nitrososphaeraceae*. The code for comes from these two posts on the phyloseq GitHub, both by <span class="co">[</span><span class="ot">MSMortensen</span><span class="co">](https://github.com/MSMortensen)</span>: issue <span class="co">[</span><span class="ot">#850</span><span class="co">](https://github.com/joey711/phyloseq/issues/850#issuecomment-394771087)</span> and issue <span class="co">[</span><span class="ot">#990</span><span class="co">](https://github.com/joey711/phyloseq/issues/990#issuecomment-424618425)</span>.</span>
<span id="cb78-1138"><a href="#cb78-1138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1139"><a href="#cb78-1139" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; One thing this code does is reassign the functions </span><span class="in">`class`</span><span class="at"> and </span><span class="in">`order`</span><span class="at"> to taxonomic ranks. This can cause issues if you need these functions.</span></span>
<span id="cb78-1140"><a href="#cb78-1140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1141"><a href="#cb78-1141" aria-hidden="true" tabindex="-1"></a>So you need to run something like this <span class="in">`rm(class, order, phylum, kingdom)`</span> at the end of the code to remove these as variables. For now, I have not come up with a better solution.</span>
<span id="cb78-1142"><a href="#cb78-1142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1143"><a href="#cb78-1143" aria-hidden="true" tabindex="-1"></a>Before doing that, we will save a copy of the phyloseq object with the original taxa names.</span>
<span id="cb78-1144"><a href="#cb78-1144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1147"><a href="#cb78-1147" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1148"><a href="#cb78-1148" aria-hidden="true" tabindex="-1"></a>ps_work_o <span class="ot">&lt;-</span> ps_filt_no_nc</span>
<span id="cb78-1149"><a href="#cb78-1149" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1150"><a href="#cb78-1150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1153"><a href="#cb78-1153" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1154"><a href="#cb78-1154" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb78-1155"><a href="#cb78-1155" aria-hidden="true" tabindex="-1"></a>tax.clean <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ps_filt_no_nc))</span>
<span id="cb78-1156"><a href="#cb78-1156" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>){ tax.clean[,i] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tax.clean[,i])}</span>
<span id="cb78-1157"><a href="#cb78-1157" aria-hidden="true" tabindex="-1"></a>tax.clean[<span class="fu">is.na</span>(tax.clean)] <span class="ot">&lt;-</span> <span class="st">""</span></span>
<span id="cb78-1158"><a href="#cb78-1158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1159"><a href="#cb78-1159" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(tax.clean)){</span>
<span id="cb78-1160"><a href="#cb78-1160" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (tax.clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">""</span>){</span>
<span id="cb78-1161"><a href="#cb78-1161" aria-hidden="true" tabindex="-1"></a>        kingdom <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">"k_"</span>, tax.clean[i,<span class="dv">1</span>], <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb78-1162"><a href="#cb78-1162" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> kingdom</span>
<span id="cb78-1163"><a href="#cb78-1163" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">""</span>){</span>
<span id="cb78-1164"><a href="#cb78-1164" aria-hidden="true" tabindex="-1"></a>        phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">"p_"</span>, tax.clean[i,<span class="dv">2</span>], <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb78-1165"><a href="#cb78-1165" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">3</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb78-1166"><a href="#cb78-1166" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">4</span>] <span class="sc">==</span> <span class="st">""</span>){</span>
<span id="cb78-1167"><a href="#cb78-1167" aria-hidden="true" tabindex="-1"></a>        class <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">"c_"</span>, tax.clean[i,<span class="dv">3</span>], <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb78-1168"><a href="#cb78-1168" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">4</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> class</span>
<span id="cb78-1169"><a href="#cb78-1169" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">5</span>] <span class="sc">==</span> <span class="st">""</span>){</span>
<span id="cb78-1170"><a href="#cb78-1170" aria-hidden="true" tabindex="-1"></a>        order <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">"o_"</span>, tax.clean[i,<span class="dv">4</span>], <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb78-1171"><a href="#cb78-1171" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">5</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> order</span>
<span id="cb78-1172"><a href="#cb78-1172" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">6</span>] <span class="sc">==</span> <span class="st">""</span>){</span>
<span id="cb78-1173"><a href="#cb78-1173" aria-hidden="true" tabindex="-1"></a>        tax.clean<span class="sc">$</span>Genus[i] <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">"f"</span>,tax.clean<span class="sc">$</span>Family[i], <span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb78-1174"><a href="#cb78-1174" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb78-1175"><a href="#cb78-1175" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb78-1176"><a href="#cb78-1176" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps_filt_no_nc) <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tax.clean)</span>
<span id="cb78-1177"><a href="#cb78-1177" aria-hidden="true" tabindex="-1"></a><span class="fu">rank_names</span>(ps_filt_no_nc)</span>
<span id="cb78-1178"><a href="#cb78-1178" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(class, order, phylum, kingdom)</span>
<span id="cb78-1179"><a href="#cb78-1179" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1180"><a href="#cb78-1180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1181"><a href="#cb78-1181" aria-hidden="true" tabindex="-1"></a>Still the same ranks. That's good. What about the new groups? Let's take a peak at some families.</span>
<span id="cb78-1182"><a href="#cb78-1182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1185"><a href="#cb78-1185" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1186"><a href="#cb78-1186" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-1187"><a href="#cb78-1187" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_filt_no_nc, <span class="st">"Family"</span>), <span class="dv">16</span>)</span>
<span id="cb78-1188"><a href="#cb78-1188" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1189"><a href="#cb78-1189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1190"><a href="#cb78-1190" aria-hidden="true" tabindex="-1"></a>Nice. Adios *NA*.</span>
<span id="cb78-1191"><a href="#cb78-1191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1192"><a href="#cb78-1192" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; *Note*. The original code is written for data sets that have species-level designations.</span></span>
<span id="cb78-1193"><a href="#cb78-1193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1194"><a href="#cb78-1194" aria-hidden="true" tabindex="-1"></a>Since this data set does not contain species, I modified the code to stop at the genus level. If your data set has species, my modifications will not work for you.</span>
<span id="cb78-1195"><a href="#cb78-1195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1196"><a href="#cb78-1196" aria-hidden="true" tabindex="-1"></a>Finally, we rename the ps object. This is now our working data set.</span>
<span id="cb78-1197"><a href="#cb78-1197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1200"><a href="#cb78-1200" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1201"><a href="#cb78-1201" aria-hidden="true" tabindex="-1"></a>ps_work <span class="ot">&lt;-</span> ps_filt_no_nc</span>
<span id="cb78-1202"><a href="#cb78-1202" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1203"><a href="#cb78-1203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1204"><a href="#cb78-1204" aria-hidden="true" tabindex="-1"></a><span class="fu">## Data Set Summary</span></span>
<span id="cb78-1205"><a href="#cb78-1205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1206"><a href="#cb78-1206" aria-hidden="true" tabindex="-1"></a>Now that we have removed the NC and samples with low reads, we can summarize the data in our working phyloseq object. Let's start with the data set as a whole using the <span class="in">`summarize_phyloseq`</span> from the <span class="co">[</span><span class="ot">microbiome R package</span><span class="co">](https://github.com/microbiome/microbiome/)</span> <span class="co">[</span><span class="ot">@lahti2017microbiome</span><span class="co">]</span> as we did above.</span>
<span id="cb78-1207"><a href="#cb78-1207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1210"><a href="#cb78-1210" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1211"><a href="#cb78-1211" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-1212"><a href="#cb78-1212" aria-hidden="true" tabindex="-1"></a>min_read_ps_work <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">readcount</span>(ps_work))</span>
<span id="cb78-1213"><a href="#cb78-1213" aria-hidden="true" tabindex="-1"></a>max_read_ps_work <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">readcount</span>(ps_work))</span>
<span id="cb78-1214"><a href="#cb78-1214" aria-hidden="true" tabindex="-1"></a>total_reads_ps_work <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">readcount</span>(ps_work))</span>
<span id="cb78-1215"><a href="#cb78-1215" aria-hidden="true" tabindex="-1"></a>mean_reads_ps_work <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">mean</span>(<span class="fu">readcount</span>(ps_work)), <span class="at">digits =</span> <span class="dv">0</span>)</span>
<span id="cb78-1216"><a href="#cb78-1216" aria-hidden="true" tabindex="-1"></a>median_reads_ps_work <span class="ot">&lt;-</span> <span class="fu">median</span>(<span class="fu">readcount</span>(ps_work))</span>
<span id="cb78-1217"><a href="#cb78-1217" aria-hidden="true" tabindex="-1"></a>total_asvs_ps_work <span class="ot">&lt;-</span> <span class="fu">ntaxa</span>(ps_work)</span>
<span id="cb78-1218"><a href="#cb78-1218" aria-hidden="true" tabindex="-1"></a>singleton_ps_work <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>(<span class="fu">ntaxa</span>(<span class="fu">rare</span>(ps_work,</span>
<span id="cb78-1219"><a href="#cb78-1219" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">detection =</span> <span class="dv">1</span>,</span>
<span id="cb78-1220"><a href="#cb78-1220" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">prevalence =</span> <span class="dv">0</span>)),</span>
<span id="cb78-1221"><a href="#cb78-1221" aria-hidden="true" tabindex="-1"></a>                              <span class="at">error=</span><span class="cf">function</span>(err) <span class="cn">NA</span>)</span>
<span id="cb78-1222"><a href="#cb78-1222" aria-hidden="true" tabindex="-1"></a>singleton_ps_work_perc <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>(<span class="fu">round</span>((<span class="dv">100</span><span class="sc">*</span>(<span class="fu">ntaxa</span>(<span class="fu">rare</span>(ps_work,</span>
<span id="cb78-1223"><a href="#cb78-1223" aria-hidden="true" tabindex="-1"></a>                                                          <span class="at">detection =</span> <span class="dv">1</span>,</span>
<span id="cb78-1224"><a href="#cb78-1224" aria-hidden="true" tabindex="-1"></a>                                                          <span class="at">prevalence =</span> <span class="dv">0</span>)) <span class="sc">/</span></span>
<span id="cb78-1225"><a href="#cb78-1225" aria-hidden="true" tabindex="-1"></a>                                   <span class="fu">ntaxa</span>(ps_work))), <span class="at">digits =</span> <span class="dv">3</span>),</span>
<span id="cb78-1226"><a href="#cb78-1226" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">error=</span><span class="cf">function</span>(err) <span class="cn">NA</span>)</span>
<span id="cb78-1227"><a href="#cb78-1227" aria-hidden="true" tabindex="-1"></a>sparsity_ps_work <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">length</span>(<span class="fu">which</span>(<span class="fu">abundances</span>(ps_work) <span class="sc">==</span> <span class="dv">0</span>))<span class="sc">/</span></span>
<span id="cb78-1228"><a href="#cb78-1228" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">length</span>(<span class="fu">abundances</span>(ps)), <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb78-1229"><a href="#cb78-1229" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1230"><a href="#cb78-1230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1231"><a href="#cb78-1231" aria-hidden="true" tabindex="-1"></a>| Metric                              | Results                                                  |</span>
<span id="cb78-1232"><a href="#cb78-1232" aria-hidden="true" tabindex="-1"></a>|-------------------------------------|----------------------------------------------------------|</span>
<span id="cb78-1233"><a href="#cb78-1233" aria-hidden="true" tabindex="-1"></a>| Min. number of reads                | <span class="in">`r min_read_ps_work`</span>                                     |</span>
<span id="cb78-1234"><a href="#cb78-1234" aria-hidden="true" tabindex="-1"></a>| Max. number of reads                | <span class="in">`r max_read_ps_work`</span>                                     |</span>
<span id="cb78-1235"><a href="#cb78-1235" aria-hidden="true" tabindex="-1"></a>| Total number of reads               | <span class="in">`r total_reads_ps_work`</span>                                  |</span>
<span id="cb78-1236"><a href="#cb78-1236" aria-hidden="true" tabindex="-1"></a>| Average number of reads             | <span class="in">`r mean_reads_ps_work`</span>                                   |</span>
<span id="cb78-1237"><a href="#cb78-1237" aria-hidden="true" tabindex="-1"></a>| Median number of reads              | <span class="in">`r median_reads_ps_work`</span>                                 |</span>
<span id="cb78-1238"><a href="#cb78-1238" aria-hidden="true" tabindex="-1"></a>| Total ASVs                          | <span class="in">`r total_asvs_ps_work`</span>                                   |</span>
<span id="cb78-1239"><a href="#cb78-1239" aria-hidden="true" tabindex="-1"></a>| Sparsity                            | <span class="in">`r sparsity_ps_work`</span>                                     |</span>
<span id="cb78-1240"><a href="#cb78-1240" aria-hidden="true" tabindex="-1"></a>| Any ASVs sum to 1 or less?          | <span class="in">`r isTRUE(singleton_ps_work &gt;= 1)`</span>                       |</span>
<span id="cb78-1241"><a href="#cb78-1241" aria-hidden="true" tabindex="-1"></a>| Number of singleton ASVs            | <span class="in">`r singleton_ps_work`</span>                                    |</span>
<span id="cb78-1242"><a href="#cb78-1242" aria-hidden="true" tabindex="-1"></a>| Percent of ASVs that are singletons | <span class="in">`r singleton_ps_work_perc`</span>                               |</span>
<span id="cb78-1243"><a href="#cb78-1243" aria-hidden="true" tabindex="-1"></a>| Number of sample variables are:     | <span class="in">`r length(sample_data(ps_work))`</span>   (<span class="in">`r colnames(samdf)`</span>) |</span>
<span id="cb78-1244"><a href="#cb78-1244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1245"><a href="#cb78-1245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1246"><a href="#cb78-1246" aria-hidden="true" tabindex="-1"></a>We can also generate a summary table of total reads &amp; ASVs for each sample. You can sort the table, download a copy, or filter by search term. Here is the code to generate the data for the table. First, we create data frames that hold total reads and ASVs for each sample.</span>
<span id="cb78-1247"><a href="#cb78-1247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1250"><a href="#cb78-1250" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1251"><a href="#cb78-1251" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb78-1252"><a href="#cb78-1252" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> <span class="fu">sample_sums</span>(ps_work)</span>
<span id="cb78-1253"><a href="#cb78-1253" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(total_reads, <span class="at">make.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb78-1254"><a href="#cb78-1254" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> total_reads <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"Sample_ID"</span>)</span>
<span id="cb78-1255"><a href="#cb78-1255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1256"><a href="#cb78-1256" aria-hidden="true" tabindex="-1"></a>total_asvs <span class="ot">&lt;-</span> <span class="fu">estimate_richness</span>(ps_work, <span class="at">measures =</span> <span class="st">"Observed"</span>)</span>
<span id="cb78-1257"><a href="#cb78-1257" aria-hidden="true" tabindex="-1"></a>total_asvs <span class="ot">&lt;-</span> total_asvs <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"Sample_ID"</span>)</span>
<span id="cb78-1258"><a href="#cb78-1258" aria-hidden="true" tabindex="-1"></a>total_asvs<span class="sc">$</span>Sample_ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">'</span><span class="sc">\\</span><span class="st">.'</span>, <span class="st">'-'</span>, total_asvs<span class="sc">$</span>Sample_ID)</span>
<span id="cb78-1259"><a href="#cb78-1259" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1260"><a href="#cb78-1260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1261"><a href="#cb78-1261" aria-hidden="true" tabindex="-1"></a>And then we merge these two data frames with the sample data frame.</span>
<span id="cb78-1262"><a href="#cb78-1262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1265"><a href="#cb78-1265" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1266"><a href="#cb78-1266" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb78-1267"><a href="#cb78-1267" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb78-1268"><a href="#cb78-1268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1269"><a href="#cb78-1269" aria-hidden="true" tabindex="-1"></a>sam_details <span class="ot">&lt;-</span> samdf</span>
<span id="cb78-1270"><a href="#cb78-1270" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sam_details) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb78-1271"><a href="#cb78-1271" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(sam_details) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Sample_ID"</span>, <span class="st">"Plot"</span>, <span class="st">"Depth"</span>,</span>
<span id="cb78-1272"><a href="#cb78-1272" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">"Treatment"</span>, <span class="st">"Temp"</span>, <span class="st">"Pair"</span>)</span>
<span id="cb78-1273"><a href="#cb78-1273" aria-hidden="true" tabindex="-1"></a>tmp_tab1 <span class="ot">&lt;-</span> seq_table</span>
<span id="cb78-1274"><a href="#cb78-1274" aria-hidden="true" tabindex="-1"></a>tmp_tab1[, <span class="fu">c</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">7</span>,<span class="dv">11</span>)] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb78-1275"><a href="#cb78-1275" aria-hidden="true" tabindex="-1"></a>tmp_tab2 <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(sam_details, total_reads) <span class="sc">%&gt;%</span></span>
<span id="cb78-1276"><a href="#cb78-1276" aria-hidden="true" tabindex="-1"></a>                <span class="fu">left_join</span>(., total_asvs) <span class="sc">%&gt;%</span></span>
<span id="cb78-1277"><a href="#cb78-1277" aria-hidden="true" tabindex="-1"></a>                <span class="fu">left_join</span>(., tmp_tab1)</span>
<span id="cb78-1278"><a href="#cb78-1278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1279"><a href="#cb78-1279" aria-hidden="true" tabindex="-1"></a>reads_lost <span class="ot">&lt;-</span> tmp_tab2<span class="sc">$</span><span class="st">`</span><span class="at">final reads</span><span class="st">`</span> <span class="sc">-</span> tmp_tab2<span class="sc">$</span>total_reads </span>
<span id="cb78-1280"><a href="#cb78-1280" aria-hidden="true" tabindex="-1"></a>asvs_lost <span class="ot">&lt;-</span> tmp_tab2<span class="sc">$</span><span class="st">`</span><span class="at">no. ASVs</span><span class="st">`</span> <span class="sc">-</span> tmp_tab2<span class="sc">$</span>Observed </span>
<span id="cb78-1281"><a href="#cb78-1281" aria-hidden="true" tabindex="-1"></a>tmp_tab2<span class="sc">$</span>reads_lost <span class="ot">&lt;-</span> reads_lost</span>
<span id="cb78-1282"><a href="#cb78-1282" aria-hidden="true" tabindex="-1"></a>tmp_tab2<span class="sc">$</span>asvs_lost <span class="ot">&lt;-</span> asvs_lost</span>
<span id="cb78-1283"><a href="#cb78-1283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1284"><a href="#cb78-1284" aria-hidden="true" tabindex="-1"></a>seq_table2 <span class="ot">&lt;-</span> tmp_tab2[, <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>,<span class="dv">12</span>,<span class="dv">13</span>)]</span>
<span id="cb78-1285"><a href="#cb78-1285" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(seq_table2) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"sample_id"</span>, <span class="st">"PLOT"</span>, <span class="st">"DEPTH"</span>,</span>
<span id="cb78-1286"><a href="#cb78-1286" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"TREAT"</span>, <span class="st">"TEMP"</span>, <span class="st">"PAIR"</span>, <span class="st">"total_reads"</span>,</span>
<span id="cb78-1287"><a href="#cb78-1287" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"total_ASVs"</span>, <span class="st">"reads_lost"</span>, <span class="st">"ASVs_lost"</span>)</span>
<span id="cb78-1288"><a href="#cb78-1288" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1289"><a href="#cb78-1289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1290"><a href="#cb78-1290" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab("seq_summary_final_its")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb78-1291"><a href="#cb78-1291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1294"><a href="#cb78-1294" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1295"><a href="#cb78-1295" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-1296"><a href="#cb78-1296" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-1297"><a href="#cb78-1297" aria-hidden="true" tabindex="-1"></a>seq_table2 <span class="sc">%&gt;%</span></span>
<span id="cb78-1298"><a href="#cb78-1298" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb78-1299"><a href="#cb78-1299" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"its18_sample_summary_final"</span>,</span>
<span id="cb78-1300"><a href="#cb78-1300" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb78-1301"><a href="#cb78-1301" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb78-1302"><a href="#cb78-1302" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb78-1303"><a href="#cb78-1303" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb78-1304"><a href="#cb78-1304" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb78-1305"><a href="#cb78-1305" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb78-1306"><a href="#cb78-1306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1307"><a href="#cb78-1307" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table2,</span>
<span id="cb78-1308"><a href="#cb78-1308" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb78-1309"><a href="#cb78-1309" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb78-1310"><a href="#cb78-1310" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">0</span>),</span>
<span id="cb78-1311"><a href="#cb78-1311" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">TRUE</span>, <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb78-1312"><a href="#cb78-1312" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb78-1313"><a href="#cb78-1313" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb78-1314"><a href="#cb78-1314" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb78-1315"><a href="#cb78-1315" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Sample ID"</span>, </span>
<span id="cb78-1316"><a href="#cb78-1316" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb78-1317"><a href="#cb78-1317" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>),</span>
<span id="cb78-1318"><a href="#cb78-1318" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>), </span>
<span id="cb78-1319"><a href="#cb78-1319" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb78-1320"><a href="#cb78-1320" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">150</span>, <span class="at">footer =</span> <span class="st">"Total reads"</span>)</span>
<span id="cb78-1321"><a href="#cb78-1321" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb78-1322"><a href="#cb78-1322" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">5</span>, </span>
<span id="cb78-1323"><a href="#cb78-1323" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb78-1324"><a href="#cb78-1324" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb78-1325"><a href="#cb78-1325" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb78-1326"><a href="#cb78-1326" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb78-1327"><a href="#cb78-1327" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb78-1328"><a href="#cb78-1328" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1329"><a href="#cb78-1329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1330"><a href="#cb78-1330" aria-hidden="true" tabindex="-1"></a><span class="fu">## Review</span></span>
<span id="cb78-1331"><a href="#cb78-1331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1332"><a href="#cb78-1332" aria-hidden="true" tabindex="-1"></a>We have a few different options to play with. At this point it is difficult to say which we will use or whether additional objects need to be created. Here is a summary of the objects we have.</span>
<span id="cb78-1333"><a href="#cb78-1333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1334"><a href="#cb78-1334" aria-hidden="true" tabindex="-1"></a>The full phyloseq object *before* removing contaminants.</span>
<span id="cb78-1335"><a href="#cb78-1335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1338"><a href="#cb78-1338" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1339"><a href="#cb78-1339" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-1340"><a href="#cb78-1340" aria-hidden="true" tabindex="-1"></a>ps</span>
<span id="cb78-1341"><a href="#cb78-1341" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps, <span class="st">"Family"</span>), <span class="dv">16</span>)</span>
<span id="cb78-1342"><a href="#cb78-1342" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1343"><a href="#cb78-1343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1344"><a href="#cb78-1344" aria-hidden="true" tabindex="-1"></a>The phyloseq object *before* removing the prefix of taxon ranks. This still has low coverage samples.</span>
<span id="cb78-1345"><a href="#cb78-1345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1348"><a href="#cb78-1348" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1349"><a href="#cb78-1349" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-1350"><a href="#cb78-1350" aria-hidden="true" tabindex="-1"></a>ps_o</span>
<span id="cb78-1351"><a href="#cb78-1351" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_o, <span class="st">"Family"</span>), <span class="dv">16</span>)</span>
<span id="cb78-1352"><a href="#cb78-1352" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1353"><a href="#cb78-1353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1354"><a href="#cb78-1354" aria-hidden="true" tabindex="-1"></a>The phyloseq object *before* changing NA ranks.</span>
<span id="cb78-1355"><a href="#cb78-1355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1358"><a href="#cb78-1358" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1359"><a href="#cb78-1359" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-1360"><a href="#cb78-1360" aria-hidden="true" tabindex="-1"></a>ps_work_o</span>
<span id="cb78-1361"><a href="#cb78-1361" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_work_o, <span class="st">"Family"</span>), <span class="dv">16</span>)</span>
<span id="cb78-1362"><a href="#cb78-1362" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1363"><a href="#cb78-1363" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1364"><a href="#cb78-1364" aria-hidden="true" tabindex="-1"></a>The phyloseq object *after* changing NA ranks.</span>
<span id="cb78-1365"><a href="#cb78-1365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1368"><a href="#cb78-1368" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1369"><a href="#cb78-1369" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-1370"><a href="#cb78-1370" aria-hidden="true" tabindex="-1"></a>ps_work</span>
<span id="cb78-1371"><a href="#cb78-1371" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_work, <span class="st">"Family"</span>), <span class="dv">16</span>)</span>
<span id="cb78-1372"><a href="#cb78-1372" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1373"><a href="#cb78-1373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1374"><a href="#cb78-1374" aria-hidden="true" tabindex="-1"></a><span class="fu">## Final Steps</span></span>
<span id="cb78-1375"><a href="#cb78-1375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1376"><a href="#cb78-1376" aria-hidden="true" tabindex="-1"></a>The last thing tasks to complete are to **a**) save copies of the taxonomy and sequence tables and **b**) save an image of the workflow for the next part of the workflow.</span>
<span id="cb78-1377"><a href="#cb78-1377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1380"><a href="#cb78-1380" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1381"><a href="#cb78-1381" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb78-1382"><a href="#cb78-1382" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">tax_table</span>(ps_work),</span>
<span id="cb78-1383"><a href="#cb78-1383" aria-hidden="true" tabindex="-1"></a>            <span class="st">"files/data-prep/tables/its18_work_tax_table.txt"</span>,</span>
<span id="cb78-1384"><a href="#cb78-1384" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span>
<span id="cb78-1385"><a href="#cb78-1385" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ps_work)),</span>
<span id="cb78-1386"><a href="#cb78-1386" aria-hidden="true" tabindex="-1"></a>            <span class="st">"files/data-prep/tables/its18_work_seq_table.txt"</span>,</span>
<span id="cb78-1387"><a href="#cb78-1387" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span>
<span id="cb78-1388"><a href="#cb78-1388" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1389"><a href="#cb78-1389" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1392"><a href="#cb78-1392" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1393"><a href="#cb78-1393" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-1394"><a href="#cb78-1394" aria-hidden="true" tabindex="-1"></a><span class="co"># PAGE BUILD</span></span>
<span id="cb78-1395"><a href="#cb78-1395" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="st">"page_build/data_prep_its18_wf.rdata"</span>)</span>
<span id="cb78-1396"><a href="#cb78-1396" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1397"><a href="#cb78-1397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1400"><a href="#cb78-1400" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1401"><a href="#cb78-1401" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-1402"><a href="#cb78-1402" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ps, <span class="st">"files/data-prep/rdata/its18_ps.rds"</span>)</span>
<span id="cb78-1403"><a href="#cb78-1403" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ps_work, <span class="st">"files/data-prep/rdata/its18_ps_work.rds"</span>)</span>
<span id="cb78-1404"><a href="#cb78-1404" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ps_work_o, <span class="st">"files/data-prep/rdata/its18_ps_work_o.rds"</span>)</span>
<span id="cb78-1405"><a href="#cb78-1405" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1406"><a href="#cb78-1406" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1409"><a href="#cb78-1409" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1410"><a href="#cb78-1410" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb78-1411"><a href="#cb78-1411" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-1412"><a href="#cb78-1412" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb78-1413"><a href="#cb78-1413" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1414"><a href="#cb78-1414" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1415"><a href="#cb78-1415" aria-hidden="true" tabindex="-1"></a><span class="fu"># Workflow Output </span></span>
<span id="cb78-1416"><a href="#cb78-1416" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1417"><a href="#cb78-1417" aria-hidden="true" tabindex="-1"></a>Data products generated in this workflow can be downloaded from figshare.</span>
<span id="cb78-1418"><a href="#cb78-1418" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1419"><a href="#cb78-1419" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;iframe</span> <span class="er">src</span><span class="ot">=</span><span class="st">"https://widgets.figshare.com/articles/14690739/embed?show_title=1"</span> <span class="er">width</span><span class="ot">=</span><span class="st">"100%"</span> <span class="er">height</span><span class="ot">=</span><span class="st">"351"</span> <span class="er">allowfullscreen</span> <span class="er">frameborder</span><span class="ot">=</span><span class="st">"0"</span><span class="kw">&gt;&lt;/iframe&gt;</span></span>
<span id="cb78-1420"><a href="#cb78-1420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1421"><a href="#cb78-1421" aria-hidden="true" tabindex="-1"></a>That's all for this part!</span>
<span id="cb78-1422"><a href="#cb78-1422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1423"><a href="#cb78-1423" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb78-1424"><a href="#cb78-1424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1425"><a href="#cb78-1425" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"filtering.html"</span> <span class="er">class</span><span class="ot">=</span><span class="st">"btnnav button_round"</span> <span class="er">style</span><span class="ot">=</span><span class="st">"float: right;"</span><span class="kw">&gt;</span>Next workflow:</span>
<span id="cb78-1426"><a href="#cb78-1426" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br&gt;</span> 3. Filtering<span class="kw">&lt;/a&gt;</span></span>
<span id="cb78-1427"><a href="#cb78-1427" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"dada2.html"</span> <span class="er">class</span><span class="ot">=</span><span class="st">"btnnav button_round"</span> <span class="er">style</span><span class="ot">=</span><span class="st">"float: left;"</span><span class="kw">&gt;</span>Previous workflow:</span>
<span id="cb78-1428"><a href="#cb78-1428" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br&gt;</span> 1. DADA2 Workflow<span class="kw">&lt;/a&gt;&lt;br&gt;&lt;br&gt;</span></span>
<span id="cb78-1429"><a href="#cb78-1429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1432"><a href="#cb78-1432" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1433"><a href="#cb78-1433" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false </span></span>
<span id="cb78-1434"><a href="#cb78-1434" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb78-1435"><a href="#cb78-1435" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-1436"><a href="#cb78-1436" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-1437"><a href="#cb78-1437" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb78-1438"><a href="#cb78-1438" aria-hidden="true" tabindex="-1"></a><span class="do">### COmmon formatting scripts</span></span>
<span id="cb78-1439"><a href="#cb78-1439" aria-hidden="true" tabindex="-1"></a><span class="do">### </span><span class="al">NOTE</span><span class="do">: captioner.R must be read BEFORE captions_XYZ.R</span></span>
<span id="cb78-1440"><a href="#cb78-1440" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">"assets"</span>, <span class="st">"functions.R"</span>))</span>
<span id="cb78-1441"><a href="#cb78-1441" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb78-1442"><a href="#cb78-1442" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1443"><a href="#cb78-1443" aria-hidden="true" tabindex="-1"></a><span class="fu">###  Source Code {.appendix}</span></span>
<span id="cb78-1444"><a href="#cb78-1444" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1445"><a href="#cb78-1445" aria-hidden="true" tabindex="-1"></a>The source code for this page can be accessed on GitHub <span class="in">`r fa(name = "github")`</span> by <span class="co">[</span><span class="ot">clicking this link</span><span class="co">]</span>(<span class="in">`r source_code()`</span>). </span>
<span id="cb78-1446"><a href="#cb78-1446" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1447"><a href="#cb78-1447" aria-hidden="true" tabindex="-1"></a><span class="fu">## Data Availability {.appendix}</span></span>
<span id="cb78-1448"><a href="#cb78-1448" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1449"><a href="#cb78-1449" aria-hidden="true" tabindex="-1"></a>Data generated in this workflow and the <span class="in">`.Rdata`</span> need to run the workflow can be accessed on figshare at <span class="co">[</span><span class="ot">10.25573/data.14690739</span><span class="co">](https://doi.org/10.25573/data.14690739)</span>.</span>
<span id="cb78-1450"><a href="#cb78-1450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1451"><a href="#cb78-1451" aria-hidden="true" tabindex="-1"></a><span class="fu">## Last updated on {.appendix}</span></span>
<span id="cb78-1452"><a href="#cb78-1452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-1455"><a href="#cb78-1455" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb78-1456"><a href="#cb78-1456" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb78-1457"><a href="#cb78-1457" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb78-1458"><a href="#cb78-1458" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb78-1459"><a href="#cb78-1459" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
      <div class="nav-footer-center">© Copyright 2022 <a href="https://metacrobe.github.io/">metacrobe</a> &amp; The SWELTR Project. Site constructed using <a href="https://quarto.org/">Quarto</a>.</div>
  </div>
</footer>


</body></html>