<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */
</style>

  <!--radix_placeholder_meta_tags-->
  <title>sweltr: 2 Data Set Preparation</title>

  <meta property="description" itemprop="description" content="Workflow for preparation  of the **2018 16S rRNA** and **2018 ITS** data sets. These steps are needed before analyzing the data. In this workflow, sample groups are defined, and phyloseq objects are created and curated."/>

  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>


  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="sweltr: 2 Data Set Preparation"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="Workflow for preparation  of the **2018 16S rRNA** and **2018 ITS** data sets. These steps are needed before analyzing the data. In this workflow, sample groups are defined, and phyloseq objects are created and curated."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="sweltr"/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="sweltr: 2 Data Set Preparation"/>
  <meta property="twitter:description" content="Workflow for preparation  of the **2018 16S rRNA** and **2018 ITS** data sets. These steps are needed before analyzing the data. In this workflow, sample groups are defined, and phyloseq objects are created and curated."/>

  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Phyloseq: An r package for reproducible interactive analysis and graphics of microbiome census data;citation_publication_date=2013;citation_volume=8;citation_author=Paul J McMurdie;citation_author=Susan Holmes"/>
  <meta name="citation_reference" content="citation_title=Tools for microbiome analysis in r. Version 1.9.97;citation_publication_date=2017;citation_author=Leo Lahti;citation_author=Shetty Sudarshan;citation_author=Tools for microbiome analysis in r. Version 1.9.97"/>
  <meta name="citation_reference" content="citation_title=Metacoder: An r package for visualization and manipulation of community taxonomic diversity data;citation_publication_date=2017;citation_volume=13;citation_author=Zachary SL Foster;citation_author=Thomas J Sharpton;citation_author=Niklaus J GrÃ¼nwald"/>
  <meta name="citation_reference" content="citation_title=ampvis2: An r package to analyse and visualise 16S rRNA amplicon data;citation_publication_date=2018;citation_author=Kasper S. Andersen;citation_author=Rasmus H. Kirkegaard;citation_author=Soren M. Karst;citation_author=Mads Albertsen"/>
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["2 Data Set Preparation"]},{"type":"character","attributes":{},"value":["Workflow for preparation  of the **2018 16S rRNA** and **2018 ITS** data sets. These steps are needed before analyzing the data. In this workflow, sample groups are defined, and phyloseq objects are created and curated.\n"]},{"type":"character","attributes":{},"value":["assets/cite.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }

  </style>

  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure img {
    width: 100%;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refs = $(this).attr('data-cites').split(" ");
        var refHtml = refs.map(function(ref) {
          return "<p>" + $('#ref-' + ref).html() + "</p>";
        }).join("\n");
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="site_libs/header-attrs-2.8/header-attrs.js"></script>
  <script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
  <script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
  <link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
  <script src="site_libs/datatables-binding-0.18/datatables.js"></script>
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
  <script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
  <script src="site_libs/jszip-1.10.20/jszip.min.js"></script>
  <link href="site_libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
  <script src="site_libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
  <link href="site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
  <script src="site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
  <link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
  <script src="site_libs/selectize-0.12.0/selectize.min.js"></script>
  <link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
  <script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Roboto:300|Nunito+Sans:300|Inconsolata|Source+Sans+Pro|Work+Sans:200,300,400,500');
  /********************************/
  /* controls general properties of site*/
  /********************************/

  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
    background-color: #FEFEFE;
  }

  d-title h1{
  font-size: 2.1em;
  font-family: 'Work Sans';
  font-weight: 300;
  }

  d-article h1 {
    font-size: 1.8em;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 300;
  }

  d-article h2 {
    font-size: 1.65em;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 500;
  }

  d-article h3 {
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
    margin-top: 1em;
  }

  d-article p {
    font-size: 18px;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
  }

  d-byline .authors-affiliations {
    grid-column-end: span 3;
    grid-column-gap: 8px;
  }

  img.orcid-id {
    margin: 0;
  }

  a.name:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  a.affiliation:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  blockquote p {
  	font-size: 1.3em;
  	font-family: 'Lato';
    font-style: normal;
    margin: 0.5em 0 0.5em;
  }

  d-article li {
    margin: 0px 0px 4px;
    font-size: 18px;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
  }

  /********************************/
  /* controls CODE*/
  /********************************/

  /** pre.sourceCode controls code block without lang **/

  /** Controls r code  **/

  pre.sourceCode.r {
    margin-top: 0px;
     color: #333333;
     background-color: #E8E8E8;
     border-radius: 3px;
     font-size: 0.80em;
     font-family: 'Monaco', 'PT Mono', monospace;
     overflow-x: scroll !important;
     border-left: 6px solid #2271B2;
     border-top: 1px solid #B7B7B7;
     border-right: 1px solid #B7B7B7;
     border-bottom: 1px solid #B7B7B7;
     padding: 18px 0px 18px 20px;
   }

   pre.sourceCode.bash {
     margin-top: 0px;
      color: #333333;
      background-color: #E8E8E8;
      border-radius: 3px;
      font-size: 0.80em;
      font-family: 'Monaco', 'PT Mono', monospace;
      overflow-x: scroll !important;
      border-left: 6px solid #B22271;
      border-top: 1px solid #B7B7B7;
      border-right: 1px solid #B7B7B7;
      border-bottom: 1px solid #B7B7B7;
      padding: 18px 0px 18px 20px;
    }
  /** Controls code output **/

  d-article pre {
     margin-top: 0px;
      color: #333333;
      background-color: #FAFAFA;
      font-size: 0.8em;
      border: 1px solid #ccc;
      overflow-x: scroll !important;
      padding:  10px 10px 10px 18px;
      border-radius: 5px;

    }

  /** Controls inline code  **/

  p code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }

  d-article aside code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-family: 'Monaco', 'PT Mono', monospace;
    color: #BF2727;
  }

  /********************************/
  /* controls NAV BAR and HEADERS*/
  /********************************/

  .nav-dropdown-content .nav-dropdown-header {
    text-transform: none;
  }

  .distill-site-header .logo img {
    margin-top: 10px;
    max-height: 30px;
    margin-bottom: 10px;
  }

  .distill-site-nav {
      background-color: #113858;
      font-size: 16px;
      font-weight: 300;
  }

  img {
    margin: 5px;
  }

  /** LIGHTBOX MARKUP **/
  .thumbnail {
    max-width: 100%;
    position: relative;
  }

  /** DETAILS fold down**/

  details {
    background-color: #f6f6f6;
    cursor: pointer;
    color: #2d2d2d;
    margin-bottom:1vh;
    padding: 1em;
    border: none;
    text-align: left;
    outline: none;
  }

  .active, details:hover {
    transition: max-height 0.2s ease-out;
    outline: none;

    /* background-color: #acc2c2; */
  }


  details {
    display: flex;
    justify-content: space-between;
    align-content: center;
    border-radius: 4px;
    outline: none;

  }

  details.active {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    outline: none;

  }

  summary::-webkit-details-marker {
    display: none;
    outline: none;

  }
  summary:focus {
    outline: none;
  }
  summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f0fe";
  }

  details[open] summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f146";
  }

  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"2 Data Set Preparation","description":"Workflow for preparation  of the **2018 16S rRNA** and **2018 ITS** data sets. These steps are needed before analyzing the data. In this workflow, sample groups are defined, and phyloseq objects are created and curated.","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title"></a>
<a href="index.html">
<i class="fa fa-home" aria-hidden="true"></i>
</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
A. Processing Workflows
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="dada2.html">1. DADA2</a>
<a href="data-prep.html">2. Data Set Prep</a>
<a href="otu.html">3. OTU Clustering</a>
<hr/>
<span class="nav-dropdown-header">Filtering</span>
<a href="pime.html">4a. PIME Filtering</a>
<a href="perfect.html">4b. PERFect Filtering</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
B. Diversity Workflows
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="taxa.html">5. Taxonomic</a>
<a href="alpha.html">6. Alpha Diversity</a>
<a href="beta.html">7. Beta Diversity</a>
<a href="da.html">8. DA ASVs &amp; OTUs</a>
<a href="metadata.html">9. Metadata</a>
</div>
</div>
<a href="data-availability.html">Data Products</a>
</div>
<div class="nav-right">
<a href="https://github.com/sweltr/high-temp/">
<i class="fa fa-github fa-lg" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>2 Data Set Preparation</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->
<p><p>Workflow for preparation of the <strong>2018 16S rRNA</strong> and <strong>2018 ITS</strong> data sets. These steps are needed before analyzing the data. In this workflow, sample groups are defined, and phyloseq objects are created and curated.</p></p>
</div>


<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#s-rrna-data">16s rRNA Data</a>
<ul>
<li><a href="#prerequisites">Prerequisites</a></li>
<li><a href="#read-counts-assessment">Read Counts Assessment</a></li>
<li><a href="#defining-groups">Defining Groups</a></li>
<li><a href="#create-a-phyloseq-object">Create a Phyloseq Object</a></li>
<li><a href="#remove-contaminants-unwanted-taxa">Remove Contaminants &amp; Unwanted Taxa</a></li>
<li><a href="#rename-na-taxonomic-ranks">Rename <em>NA</em> taxonomic ranks</a></li>
<li><a href="#add-phylogenetic-tree">Add Phylogenetic Tree</a></li>
<li><a href="#sample-summary">Sample Summary</a></li>
<li><a href="#taxmap-object">Taxmap Object</a></li>
<li><a href="#ampvis2-object">Ampvis2 Object</a></li>
<li><a href="#review">Review</a></li>
<li><a href="#final-steps">Final Steps</a></li>
</ul></li>
<li><a href="#its-data">ITS Data</a>
<ul>
<li><a href="#prerequisites-1">Prerequisites</a></li>
<li><a href="#read-counts-assessment-1">Read Counts Assessment</a></li>
<li><a href="#defining-groups-1">Defining Groups</a></li>
<li><a href="#create-a-phyloseq-object-1">Create a Phyloseq Object</a></li>
<li><a href="#remove-unwanted-samples">Remove Unwanted Samples</a></li>
<li><a href="#rename-na-taxonomic-ranks-1">Rename <em>NA</em> taxonomic ranks</a></li>
<li><a href="#sample-summary-1">Sample Summary</a></li>
<li><a href="#taxmap-object-1">Taxmap Object</a></li>
<li><a href="#ampvis2-object-1">Ampvis2 Object</a></li>
<li><a href="#review-1">Review</a></li>
<li><a href="#final-steps-1">Final Steps</a></li>
<li><a href="#source-code">Source Code</a></li>
<li><a href="#data-availability">Data Availability</a></li>
</ul></li>
</ul>
</nav>
</div>
<details>
<summary>
Click here for setup information.
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">echo =</span> <span class="cn">TRUE</span>, <span class="at">eval =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">119</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#library(conflicted)</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends(ape, local = TRUE)  </span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends_reverse(ape, local = TRUE)  </span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq); <span class="fu">packageVersion</span>(<span class="st">&quot;phyloseq&quot;</span>)</span></code></pre></div>
<pre><code>[1] &#39;1.34.0&#39;</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings); <span class="fu">packageVersion</span>(<span class="st">&quot;Biostrings&quot;</span>)</span></code></pre></div>
<pre><code>[1] &#39;2.58.0&#39;</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(tidyverse, DT, microbiome, taxa, </span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>               metacoder, ampvis2, ape, statnet.common, </span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">install =</span> <span class="cn">FALSE</span>, <span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">scipen=</span><span class="dv">999</span>)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_current<span class="sc">$</span><span class="fu">get</span>(<span class="fu">c</span>(</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache&quot;</span>,</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.path&quot;</span>,</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.rebuild&quot;</span>,</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;dependson&quot;</span>,</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;autodep&quot;</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>))</span></code></pre></div>
<pre><code>$cache
[1] 0

$cache.path
[1] &quot;data-prep_cache/html5/&quot;

$cache.rebuild
[1] FALSE

$dependson
NULL

$autodep
[1] FALSE</code></pre>
</div>
</details>
<p>These workflows share many common variable names so you must split the workflow into a script for each data set OR run the command <code>remove(list = ls())</code> before beginning the next workflow.</p>
<h1 id="s-rrna-data">16s rRNA Data</h1>
<h2 id="prerequisites">Prerequisites</h2>
<p>In order to run this workflow, you either need to run the corresponding <a href="dada2-high-temp.html#s-rrna-data.html">DADA2 Workflow for 2018 16S rRNA</a> or begin with the output from that workflow, <code>data_prep_ssu18_wf.rdata</code>. See the <a href="data-availability.html">Data Availability</a> page for complete details.</p>
<p>Unless otherwise noted, we primarily use <a href="https://joey711.github.io/phyloseq/">phyloseq</a> <span class="citation" data-cites="mcmurdie2013phyloseq">(<a href="#ref-mcmurdie2013phyloseq" role="doc-biblioref">McMurdie and Holmes 2013</a>)</span> in this section of the workflow to prepare the 2018 16S rRNA data set for community analyses. We prepare the data set by curating samples, removing contaminants, and creating phyloseq objects.</p>
<h2 id="read-counts-assessment">Read Counts Assessment</h2>
<p>Before we begin, letâs create a summary table containing some basic sample metadata and the read count data from the <a href="dada2-high-temp.html#s-rrna-data.html">DADA2 workflow</a>. We need to inspect how total reads changed through the workflow. While we are at it, letâs create more intuitive Sample IDs. For more details on how reads changed at each step of the workflow, see the <a href="dada2-high-temp.html#track-reads-through-workflow">summary table</a> at the end of the DADA2 section. Table headers are as follows:</p>
<table>
<colgroup>
<col style="width: 23%" />
<col style="width: 76%" />
</colgroup>
<thead>
<tr class="header">
<th>Header</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>Sample ID</code></td>
<td>the new sample ID based on Plot number, Depth, Treatment, Temperature, &amp; Pair</td>
</tr>
<tr class="even">
<td><code>Plot</code></td>
<td>the experimental plot number</td>
</tr>
<tr class="odd">
<td><code>Depth</code></td>
<td>the depth where the sample was collected</td>
</tr>
<tr class="even">
<td><code>Treatment</code></td>
<td>warm or control</td>
</tr>
<tr class="odd">
<td><code>Temp</code></td>
<td>temperature of soil heating (0C, +4C, or +8C)</td>
</tr>
<tr class="even">
<td><code>Pair</code></td>
<td>control/treatment coupling</td>
</tr>
<tr class="odd">
<td><code>input</code></td>
<td>number of raw reads</td>
</tr>
<tr class="even">
<td><code>nochim</code></td>
<td>final read count after removing chimeras</td>
</tr>
<tr class="odd">
<td><code>Remain</code></td>
<td>percent of reads remaining from <code>input</code> to <code>nonchim</code></td>
</tr>
<tr class="even">
<td><code>FastqID</code></td>
<td>base name of the fastq file</td>
</tr>
</tbody>
</table>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-page">
<div id="po4ycztlg0tzswm5sssm" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="po4ycztlg0tzswm5sssm">{"x":{"filter":"top","filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"disabled\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0\" data-max=\"8\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"58266\" data-max=\"120892\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"25151\" data-max=\"86841\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.264\" data-max=\"0.809\" data-scale=\"3\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Sample  summary table including\n            read changes at start and end of DADA2 workflow. Use the buttons to\n            navigate through the table or download a copy.<\/em>\n<\/caption>","data":[["P01_D00_010_W3A","P02_D00_010_C0A","P03_D00_010_W3B","P04_D00_010_C0B","P05_D00_010_W3C","P06_D00_010_C0C","P07_D00_010_W3D","P08_D00_010_C0D","P09_D00_010_W3E","P10_D00_010_C0E","P01_D00_010_W8A","P03_D00_010_W8B","P05_D00_010_W8C","P07_D00_010_W8D","P09_D00_010_W8E"],["P01","P02","P03","P04","P05","P06","P07","P08","P09","P10","P01","P03","P05","P07","P09"],["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],["Warm","Control","Warm","Control","Warm","Control","Warm","Control","Warm","Control","Warm","Warm","Warm","Warm","Warm"],[3,0,3,0,3,0,3,0,3,0,8,8,8,8,8],["A","A","B","B","C","C","D","D","E","E","A","B","C","D","E"],[85489,87091,93065,67810,58266,101599,76781,77599,95566,67583,77645,66401,120892,79032,95182],[68037,70436,75183,53813,47070,81219,61742,62171,76299,53625,60535,53437,86841,62202,25151],[0.796,0.809,0.808,0.794,0.808,0.799,0.804,0.801,0.798,0.793,0.78,0.805,0.718,0.787,0.264],["P1","P2","P3","P4","P5","P6","P7","P8","P9","P10","P1-8C","P3-8C","P5-8C","P7-8C","P9-8C"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Sample_ID<\/th>\n      <th>Plot<\/th>\n      <th>Depth<\/th>\n      <th>Treatment<\/th>\n      <th>Temp<\/th>\n      <th>Pair<\/th>\n      <th>input<\/th>\n      <th>nochim<\/th>\n      <th>Remain<\/th>\n      <th>FastqID<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"className":"dt-right","targets":[4,6,7,8]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-size':'80%'});\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\nvar value=data[8]; $(this.api().cell(row, 8).node()).css({'font-size':'80%'});\nvar value=data[9]; $(this.api().cell(row, 9).node()).css({'font-size':'80%'});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
<p>We can also plot the final read count by the <em>Treatment</em> the sample was subjected to.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body-outset">

</div>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/data-prep/ssu18_samp_summary_plot.png" width="1050" /></p>
</div>
<p><small>Final read count by temperature treatment. Each point represents a different sample.</small></p>
<h2 id="defining-groups">Defining Groups</h2>
<ol type="1">
<li><p>Load the data packet produced in the final step of the DADA2 workflow. This packet (<code>ssu18_dada2_wf.rdata</code>) contains the ASV-by-sample table and the ASV taxonomy table.</p></li>
<li><p>Rename the samples so names have plot and Depth info.</p></li>
<li><p>After we load the data packet, we next need to format sample names and define groups.</p></li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>joined_tab_ro <span class="ot">&lt;-</span> joined_tab_percent[<span class="fu">order</span>(joined_tab_percent<span class="sc">$</span>FastqID), ]</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>new_names <span class="ot">&lt;-</span> joined_tab_ro<span class="sc">$</span><span class="st">`</span><span class="at">Sample_ID</span><span class="st">`</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;files/dada2/rdata/ssu18_dada2_wf.rdata&quot;</span>)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(seqtab) <span class="ot">&lt;-</span> new_names</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>samples.out <span class="ot">&lt;-</span> <span class="fu">rownames</span>(seqtab)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>sample_name <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">1</span>, <span class="dv">999</span>)</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>plot <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">0</span>, <span class="dv">3</span>)</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>depth <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">6</span>, <span class="dv">11</span>)</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>treatment <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">13</span>, <span class="dv">13</span>)</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">14</span>, <span class="dv">14</span>)</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>pair <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">15</span>, <span class="dv">15</span>)</span></code></pre></div>
</details>
</div>
<p>We have a total of <strong>15</strong> samples, <strong>5</strong> sample pairs, from <strong>10</strong> plots (<strong>1</strong> depth). There are <strong>2</strong> treatments corresponding to <strong>3</strong> different temperature regimes.</p>
<ol start="4" type="1">
<li>And finally we define a sample data frame that holds the different groups we extracted from the sample names.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">#define a sample data frame</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>samdf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">SamName =</span> sample_name,</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">PLOT =</span> plot,</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">DEPTH =</span> depth,</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">TREAT =</span> treatment,</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">TEMP =</span> temp,</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">PAIR =</span> pair)</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(samdf) <span class="ot">&lt;-</span> samples.out</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>For example, P07_D00-010_W4D is the sample heated at +NAC from plot NA and a depth of NAcm. This sample is in group NA along with .</p>
<h2 id="create-a-phyloseq-object">Create a Phyloseq Object</h2>
<p><strong>A</strong>. The first step is to rename the amplicon sequence variants (ASVs) so the designations are a bit more user friendly. By default, DADA2 names each ASV by its unique sequence so that data can be directly compared across studies (which is great). But this convention can get cumbersome downstream, so we rename the ASVs using a simpler conventionâASV1, ASV2, ASV3, and so on.</p>
<aside>
A phyloseq object contains ASV table (taxa abundances), sample metadata, and taxonomy table (mapping between ASVs and higher-level taxonomic classifications).
</aside>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># this create the phyloseq object</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(<span class="fu">otu_table</span>(seqtab, <span class="at">taxa_are_rows =</span> <span class="cn">FALSE</span>),</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">sample_data</span>(samdf), <span class="fu">tax_table</span>(tax_silva))</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">tax_table</span>(ps),</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">rownames</span>(<span class="fu">tax_table</span>(ps)))</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="co"># adding unique ASV names</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="fu">taxa_names</span>(ps) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;ASV&quot;</span>, <span class="fu">seq</span>(<span class="fu">ntaxa</span>(ps)))</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">tax_table</span>(ps),</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">rownames</span>(<span class="fu">tax_table</span>(ps)))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;ASV1&quot; &quot;ASV2&quot; &quot;ASV3&quot; &quot;ASV4&quot; &quot;ASV5&quot; &quot;ASV6&quot;</code></pre>
</div>
<p>So the complete data set contains 20332 ASVs. We can also use the <a href="https://github.com/microbiome/microbiome/">microbiome R package</a> <span class="citation" data-cites="lahti2017microbiome">(<a href="#ref-lahti2017microbiome" role="doc-biblioref">Lahti, Sudarshan, and others 2017</a>)</span> to get some additional summary data from the phyloseq object.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<table>
<colgroup>
<col style="width: 40%" />
<col style="width: 59%" />
</colgroup>
<thead>
<tr class="header">
<th>Metric</th>
<th>Results</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>25151</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>86841</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>937761</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>62517</td>
</tr>
<tr class="odd">
<td>Median number of reads</td>
<td>62171</td>
</tr>
<tr class="even">
<td>Sparsity</td>
<td>0.898</td>
</tr>
<tr class="odd">
<td>Any ASVs sum to 1 or less?</td>
<td>FALSE</td>
</tr>
<tr class="even">
<td>Number of singleton ASVs</td>
<td>NA</td>
</tr>
<tr class="odd">
<td>Percent of ASVs that are singletons</td>
<td>NA</td>
</tr>
<tr class="even">
<td>Number of sample variables are:</td>
<td>6 (SamName, PLOT, DEPTH, TREAT, TEMP, PAIR)</td>
</tr>
</tbody>
</table>
<p><strong>B</strong>. Add two final columns containing the ASV sequences and ASV IDs. This will be useful later when trying to export a fasta file. We can also take a look at the phyloseq object.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(<span class="fu">tax_table</span>(ps)) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Kingdom&quot;</span>, <span class="st">&quot;Phylum&quot;</span>, <span class="st">&quot;Class&quot;</span>, <span class="st">&quot;Order&quot;</span>,</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Genus&quot;</span>, <span class="st">&quot;ASV_SEQ&quot;</span>, <span class="st">&quot;ASV_ID&quot;</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>ps</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20332 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20332 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<p><strong>C</strong>. Export sequence and taxonomy tables for the unadulterated phyloseq object for later use. We will use the prefix <code>full</code> to indicate that these are the <em>raw</em> outputs.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">tax_table</span>(ps),</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;files/data-prep/tables/ssu18_full_tax_table.txt&quot;</span>,</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ps)),</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;files/data-prep/tables/ssu18_full_seq_table.txt&quot;</span>,</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span></code></pre></div>
</div>
<h2 id="remove-contaminants-unwanted-taxa">Remove Contaminants &amp; Unwanted Taxa</h2>
<p>Letâs see if we have any potential contaminants. We can use some <a href="https://rmarkdown.rstudio.com/lesson-4.html">inline R code</a> to see the taxonomy table for any taxa of interest.</p>
<aside>
The code (hidden by default) is written as <code>`r "Chloroplast" %in% tax_table(ps)`</code>.
</aside>
<ul>
<li>Are Mitochondria present? TRUE</li>
<li>Are Chloroplast present? TRUE</li>
<li>Are Eukaryota present? TRUE</li>
</ul>
<p>Letâs remove these taxaâEukaryota because we used bacterial/archaeal primers, Mitochondria because those are likely from eukaryotes, and Chloroplast because those are likely from plants. We must do each of these in turn using phyloseq and it gets a little messy.</p>
<p>Why messy? The <code>subset_taxa</code> command removes anything that is <code>NA</code> for the specified taxonomic level or above. For example, lets say you run the <code>subset_taxa</code> command using <code>Family != "Mitochondria</code>". Seems like you should get a phyloseq object with everything except Mitochondria. But actually the command not only gets rid of Mitochondria but everything else that has <code>NA</code> for Family and above. In my experience this is not well documented and I had to dig through the files to figure out what was happening.</p>
<p>Anyway, to remove the taxa we do the following:</p>
<ul>
<li>Subset the taxa and generate a <code>ps</code> object of just the taxa of interest,</li>
<li>Select the ASV column only, turn it into a factor, and use this to remove <INSERT TAXA> from the <code>ps</code> object.</li>
</ul>
<h3 id="remove-mitochondria-asvs">Remove Mitochondria ASVs</h3>
<p>Remember the original data set contained 20332 ASVs. Here we generate a file with mitochondria ASVs only.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>MT1 <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(ps, Family <span class="sc">==</span> <span class="st">&quot;Mitochondria&quot;</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>MT1 <span class="ot">&lt;-</span>  <span class="fu">as</span>(<span class="fu">tax_table</span>(MT1), <span class="st">&quot;matrix&quot;</span>)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>MT1 <span class="ot">&lt;-</span> MT1[, <span class="dv">8</span>]</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>MT1df <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(MT1)</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>goodTaxa <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">taxa_names</span>(ps), MT1df)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>ps_no_mito <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(goodTaxa, ps)</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>ps_no_mito</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20291 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20291 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<p>Looks like this removed <strong>41 Mitochondria ASVs</strong>. We will duplicate the code block to remove other groups.</p>
<h3 id="remove-chloroplast-asvs">Remove Chloroplast ASVs</h3>
<p>And again with Chloroplast ASVs only.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>CH1 <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(ps_no_mito, Order <span class="sc">==</span> <span class="st">&quot;Chloroplast&quot;</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>CH1 <span class="ot">&lt;-</span>  <span class="fu">as</span>(<span class="fu">tax_table</span>(CH1), <span class="st">&quot;matrix&quot;</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>CH1 <span class="ot">&lt;-</span> CH1[, <span class="dv">8</span>]</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>CH1df <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(CH1)</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>goodTaxa <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">taxa_names</span>(ps_no_mito), CH1df)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>ps_no_chloro <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(goodTaxa, ps_no_mito)</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>ps_no_chloro</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20274 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20274 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<p>The code removed an additional <strong>17 Chloroplast ASVs</strong>.</p>
<h3 id="remove-eukaryota-asvs">Remove Eukaryota ASVs</h3>
<p>And again with Eukaryota ASVs only.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>EU1 <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(ps_no_chloro, Kingdom <span class="sc">==</span> <span class="st">&quot;Eukaryota&quot;</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>EU1 <span class="ot">&lt;-</span>  <span class="fu">as</span>(<span class="fu">tax_table</span>(EU1), <span class="st">&quot;matrix&quot;</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>EU1 <span class="ot">&lt;-</span> EU1[, <span class="dv">8</span>]</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>EU1df <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(EU1)</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>goodTaxa <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">taxa_names</span>(ps_no_chloro), EU1df)</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>ps_no_euk <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(goodTaxa, ps_no_chloro)</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>ps_no_euk</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20223 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20223 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<p>The code removed an additional <strong>51 Eukaryota ASVs</strong> from the <code>ps</code> object.</p>
<h3 id="remove-any-kingdom-nas">Remove any Kingdom NAs</h3>
<p>Here we can just use the straight up <code>subset_taxa</code> command since we do not need to worry about any ranks above Kingdom also being removed.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>ps_filt <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(ps_no_euk, <span class="sc">!</span><span class="fu">is.na</span>(Kingdom))</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>ps_work_o <span class="ot">&lt;-</span> ps_filt</span></code></pre></div>
</div>
<p>The code eliminated an additional <strong>50 Kingdom level NA ASVs</strong> from the phyloseq object.</p>
<h2 id="rename-na-taxonomic-ranks">Rename <em>NA</em> taxonomic ranks</h2>
<p>Phyloseq has an odd way of dealing with taxonomic ranks that have no valueâin other words, <em>NA</em> in the tax table. The first thing we are going to do before moving forward is to change all of the <em>NA</em>s to have a value of the next highest classified rank. For example, <code>ASV26</code> is not classified at the Genus level but is at Family level (Xanthobacteraceae). So we change the Genus name to <em>Family_Xanthobacteraceae</em>. The code for comes from these two posts on the phyloseq GitHub, both by <a href="https://github.com/MSMortensen">MSMortensen</a>: issue <a href="https://github.com/joey711/phyloseq/issues/850#issuecomment-394771087">#850</a> and issue <a href="https://github.com/joey711/phyloseq/issues/990#issuecomment-424618425">#990</a>.</p>
<blockquote>
<p>One thing this code does is reassign the functions <code>class</code> and <code>order</code> to taxonomic ranks. This can cause issues if you need these functions.</p>
</blockquote>
<p>So you need to run something like this <code>rm(class, order, phylum, kingdom)</code> at the end of the code to remove these as variables. For now, I have not come up with a better solution.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>tax.clean <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ps_filt))</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>){ tax.clean[,i] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tax.clean[,i])}</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>tax.clean[<span class="fu">is.na</span>(tax.clean)] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(tax.clean)){</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (tax.clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>        kingdom <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;k_&quot;</span>, tax.clean[i,<span class="dv">1</span>], <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> kingdom</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>        phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;p_&quot;</span>, tax.clean[i,<span class="dv">2</span>], <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">3</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">4</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>        class <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;c_&quot;</span>, tax.clean[i,<span class="dv">3</span>], <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">4</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> class</span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">5</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>        order <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;o_&quot;</span>, tax.clean[i,<span class="dv">4</span>], <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">5</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> order</span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">6</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>        tax.clean<span class="sc">$</span>Genus[i] <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;f&quot;</span>,tax.clean<span class="sc">$</span>Family[i], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps_filt) <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tax.clean)</span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a><span class="fu">rank_names</span>(ps_filt)</span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(class, order, phylum, kingdom)</span></code></pre></div>
</details>
</div>
<p>Still the same ranks. Thatâs good. What about the new groups? Letâs take a peak at some families.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_filt, <span class="st">&quot;Family&quot;</span>), <span class="dv">16</span>)</span></code></pre></div>
<pre><code> [1] &quot;Planococcaceae&quot;         &quot;Chthoniobacteraceae&quot;   
 [3] &quot;Nitrososphaeraceae&quot;     &quot;p_RCP2-54&quot;             
 [5] &quot;o_Rokubacteriales&quot;      &quot;Methyloligellaceae&quot;    
 [7] &quot;Nitrospiraceae&quot;         &quot;Xanthobacteraceae&quot;     
 [9] &quot;o_Vicinamibacterales&quot;   &quot;Vicinamibacteraceae&quot;   
[11] &quot;c_Subgroup_5&quot;           &quot;Xiphinematobacteraceae&quot;
[13] &quot;Entotheonellaceae&quot;      &quot;Chitinophagaceae&quot;      
[15] &quot;Nitrosomonadaceae&quot;      &quot;o_Acidobacteriales&quot;    </code></pre>
</div>
<p>Nice. Bye-bye <em>NA</em></p>
<blockquote>
<p><em>Note</em>. The original code mentioned above is written for data sets that have species-level designations.</p>
</blockquote>
<p>Since this data set does not contain species, I modified the code to stop at the genus level. If your data set has species, my modifications will not work for you.</p>
<p>Finally, we rename the ps object. This is now our working data set.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>ps_work <span class="ot">&lt;-</span> ps_filt</span></code></pre></div>
</div>
<h2 id="add-phylogenetic-tree">Add Phylogenetic Tree</h2>
<p>One final task is to add a phylogenetic tree to the phyloseq object.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>ps_work_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(ps_work), <span class="at">rooted =</span> <span class="cn">TRUE</span>,</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(ps_work))</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>ps_work <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(ps_work,</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>                          sample_data,</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>                          ps_work_tree)</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>ps_work_o_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(ps_work_o), <span class="at">rooted =</span> <span class="cn">TRUE</span>,</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(ps_work_o))</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>ps_work_o <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(ps_work_o,</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>                          sample_data,</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>                          ps_work_o_tree)</span></code></pre></div>
</div>
<p>Later in the workflow, we will generate an ampvis2 object from the working phyloseq object, but first we need to fix the OTU table from the phyloseq object. For some reason, values in otu table are <code>int</code> but need to be <code>dbl</code> or else ampvis2 will not work properly.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>otu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ps_work)))</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>otu[] <span class="ot">&lt;-</span> <span class="fu">lapply</span>(otu, as.numeric)</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>otu <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(otu)</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>tax <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ps_work)))</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(ps_work))</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>ps_amp <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(<span class="fu">otu_table</span>(otu, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>),</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">tax_table</span>(tax, tax),</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">sample_data</span>(samples))</span></code></pre></div>
</div>
<h2 id="sample-summary">Sample Summary</h2>
<p>Now that we have removed the NC and sample with zero reads, we can summarize the data set so far. Letâs start with the in our working phyloseq object using the <code>summarize_phyloseq</code> from the <a href="https://github.com/microbiome/microbiome/">microbiome R package</a> <span class="citation" data-cites="lahti2017microbiome">(<a href="#ref-lahti2017microbiome" role="doc-biblioref">Lahti, Sudarshan, and others 2017</a>)</span> as we did above.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<table>
<colgroup>
<col style="width: 38%" />
<col style="width: 61%" />
</colgroup>
<thead>
<tr class="header">
<th>Metric</th>
<th>Results</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>25088</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>86600</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>936640</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>62443</td>
</tr>
<tr class="odd">
<td>Median number of reads</td>
<td>62117</td>
</tr>
<tr class="even">
<td>Sparsity</td>
<td>0.891</td>
</tr>
<tr class="odd">
<td>Any ASVs sum to 1 or less?</td>
<td>FALSE</td>
</tr>
<tr class="even">
<td>Number of singleton ASVs</td>
<td>NA</td>
</tr>
<tr class="odd">
<td>Percent of ASVs that are singletons</td>
<td>NA</td>
</tr>
<tr class="even">
<td>Number of sample variables are:</td>
<td>6 (SamName, PLOT, DEPTH, TREAT, TEMP, PAIR)</td>
</tr>
</tbody>
</table>
<p>We can also generate a summary table of total reads &amp; ASVs for each sample. You can sort the table, download a copy, or filter by search term. Here is the code to generate the data for the table. First, we create data frames that hold total reads and ASVs for each sample.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> <span class="fu">sample_sums</span>(ps_work)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(total_reads, <span class="at">make.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> total_reads <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;Sample_ID&quot;</span>)</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>total_asvs <span class="ot">&lt;-</span> <span class="fu">estimate_richness</span>(ps_work,</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>                                <span class="at">measures =</span> <span class="st">&quot;Observed&quot;</span>)</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>total_asvs <span class="ot">&lt;-</span> total_asvs <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;Sample_ID&quot;</span>)</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>total_asvs<span class="sc">$</span>Sample_ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>, <span class="st">&#39;-&#39;</span>, total_asvs<span class="sc">$</span>Sample_ID)</span></code></pre></div>
</details>
</div>
<p>And then we merge these two data frames with the sample data frame.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>sam_details <span class="ot">&lt;-</span> samdf</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sam_details) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(sam_details) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Sample_ID&quot;</span>, <span class="st">&quot;Plot&quot;</span>, <span class="st">&quot;Depth&quot;</span>,</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;Treatment&quot;</span>, <span class="st">&quot;Temp&quot;</span>, <span class="st">&quot;Pair&quot;</span>)</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>joined_percent <span class="ot">&lt;-</span> joined_tab_percent</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>joined_percent[, <span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>merge_tab <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(sam_details, total_reads) <span class="sc">%&gt;%</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>                <span class="fu">left_join</span>(., joined_percent)</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>percent_change2 <span class="ot">&lt;-</span> <span class="dv">100</span><span class="sc">*</span>(<span class="dv">1</span><span class="sc">-</span>(merge_tab<span class="sc">$</span>total_reads<span class="sc">/</span>merge_tab<span class="sc">$</span>nochim)) <span class="sc">%&gt;%</span></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">round</span>(<span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>merge_tab[<span class="dv">11</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>merge_tab<span class="sc">$</span>Change <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">sprintf</span>(<span class="st">&quot;%.3f&quot;</span>, percent_change2))</span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>merge_tab2 <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(merge_tab, total_asvs, <span class="at">by =</span> <span class="st">&quot;Sample_ID&quot;</span>)</span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>merge_tab2 <span class="ot">&lt;-</span> merge_tab2[, <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>,<span class="dv">11</span>,<span class="dv">12</span>)]</span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>merge_tab2<span class="sc">$</span>coverage <span class="ot">=</span> <span class="fu">cut</span>(merge_tab2<span class="sc">$</span>total_reads,</span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">5000</span>, <span class="dv">25000</span>, <span class="dv">100000</span>, <span class="dv">200000</span>),</span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a>                          <span class="at">labels =</span> <span class="fu">c</span>(</span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&quot;low (&lt; 5k)&quot;</span>, <span class="st">&quot;medium (5-25k)&quot;</span>,</span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&quot;high (25-100k)&quot;</span>, <span class="st">&quot;extra_high (&gt; 100k)&quot;</span>))</span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(merge_tab2) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;sample_id&quot;</span>, <span class="st">&quot;PLOT&quot;</span>, <span class="st">&quot;DEPTH&quot;</span>,</span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a>                          <span class="st">&quot;TREAT&quot;</span>, <span class="st">&quot;TEMP&quot;</span>, <span class="st">&quot;PAIR&quot;</span>,</span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a>                          <span class="st">&quot;final_reads&quot;</span>, <span class="st">&quot;reads_lost&quot;</span>, <span class="st">&quot;total_asvs&quot;</span>, <span class="st">&quot;coverage&quot;</span>)</span></code></pre></div>
</details>
</div>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="y2yczq8hn32js8mwqmp6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="y2yczq8hn32js8mwqmp6">{"x":{"filter":"top","filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"disabled\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"25088\" data-max=\"86600\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0\" data-max=\"0.3\" data-scale=\"1\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"813\" data-max=\"3065\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"disabled\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Total reads &amp; ASVs by sample.<\/em>\n<\/caption>","data":[["P01_D00_010_W3A","P01_D00_010_W8A","P10_D00_010_C0E","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E"],["P01","P01","P10","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09"],["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],["W","W","C","C","W","W","C","W","W","C","W","W","C","W","W"],["3","8","0","0","3","8","0","3","8","0","3","8","0","3","8"],["A","A","E","A","B","B","B","C","C","C","D","D","D","E","E"],[67902,60509,53600,70412,75123,53347,53770,47023,86600,81150,61632,62117,62136,76231,25088],[0.2,0,0,0,0.1,0.2,0.1,0.1,0.3,0.1,0.2,0.1,0.1,0.1,0.3],[2371,1610,2066,2272,2356,1575,2040,1663,2124,3065,2010,2157,2441,2386,813],["high (25-100k)","high (25-100k)","high (25-100k)","high (25-100k)","high (25-100k)","high (25-100k)","high (25-100k)","high (25-100k)","high (25-100k)","high (25-100k)","high (25-100k)","high (25-100k)","high (25-100k)","high (25-100k)","high (25-100k)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>sample_id<\/th>\n      <th>PLOT<\/th>\n      <th>DEPTH<\/th>\n      <th>TREAT<\/th>\n      <th>TEMP<\/th>\n      <th>PAIR<\/th>\n      <th>final_reads<\/th>\n      <th>reads_lost<\/th>\n      <th>total_asvs<\/th>\n      <th>coverage<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"targets":7,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\");\n  }"},{"className":"dt-right","targets":[6,7,8]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-size':'80%'});\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\nvar value=data[8]; $(this.api().cell(row, 8).node()).css({'font-size':'80%'});\nvar value=data[9]; $(this.api().cell(row, 9).node()).css({'font-size':'80%'});\n}"}},"evals":["options.columnDefs.0.render","options.rowCallback"],"jsHooks":[]}</script>
</div>
<p><br/></p>
<h2 id="taxmap-object">Taxmap Object</h2>
<p>We have phyloseq objects but we also want to make <code>taxmap</code> objects so we can use the <a href="https://github.com/grunwaldlab/metacoder">metacoder</a> <span class="citation" data-cites="foster2017metacoder">(<a href="#ref-foster2017metacoder" role="doc-biblioref">Foster, Sharpton, and GrÃ¼nwald 2017</a>)</span> and <a href="https://github.com/ropensci/taxa">taxa</a> packages. We will jump back and forth between between different tools and packages because each has functionality not available in the others. That said, each package also requires a specially formatted <em>object</em> to work with.</p>
<p>Confused? I was too until I realized that a phyloseq object contains only three type of tables: an <code>otu_table</code>, a <code>sample_data</code>, and a <code>tax_table</code><a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>. So any time you want to manipulate your dataâsay look at relative ASV abundanceâyou need a <em>new</em> phyloseq object. A taxmap object on the other hand can contain many different types of data tables, and as such, is generally more flexible.</p>
<h3 id="data-prep">Data Prep</h3>
<p>First, we can take a look at the phyloseq object generated during the previous workflow.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>ps_work</span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20173 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20173 taxa by 8 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 20173 tips and 20172 internal nodes ]</code></pre>
</div>
<p>In order to use the taxa package, we need to have our data in a specific format. Metacoder has a command called <code>parse_phyloseq</code>, which should do the work for us, but I was unable to get this to work for our phyloseq object. Instead, we generate the necessary tables directly from the phyloseq object and then create a basic taxmap object. I apologize for the circuitous route, but at least it is relatively simple. We need the <strong>a</strong>) taxonomy table, <strong>b</strong>) ASV table, and <strong>c</strong>) sample metadata table. We will first create the tables and then save a local copy.</p>
<h4 id="taxonomy-table">Taxonomy table</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>tax_tab <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">tax_table</span>(ps_work))</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>tax_tab <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(tax_tab, <span class="st">&quot;asv_id&quot;</span>)</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>tax_tab<span class="sc">$</span>ASV_SEQ <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tax_tab)[<span class="fu">colnames</span>(tax_tab) <span class="sc">==</span> <span class="st">&quot;ASV_ID&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;ASV&quot;</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(tax_tab, <span class="st">&quot;files/data-prep/tables/ssu18_tax_tab_mc.txt&quot;</span>,</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<h4 id="asv-table">ASV table</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>asv_tab <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">otu_table</span>(ps_work))</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="co"># this orders in decending fashion</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>asv_tab <span class="ot">&lt;-</span> asv_tab[ statnet.common<span class="sc">::</span><span class="fu">order</span>(<span class="fu">row.names</span>(asv_tab)), ]</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>asv_tab <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(asv_tab))</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>asv_tab <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(asv_tab, <span class="st">&quot;asv_id&quot;</span>)</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(asv_tab, <span class="st">&quot;files/data-prep/tables/ssu18_asv_tab_mc.txt&quot;</span>,</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<h4 id="sample-table">Sample table</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> merge_tab2</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>sam_tab[,<span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> sam_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;total_reads&quot;</span> <span class="ot">=</span> <span class="st">&quot;final_reads&quot;</span>)</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="co"># this orders in decending fashion</span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> sam_tab[statnet.common<span class="sc">::</span><span class="fu">order</span>(sam_tab<span class="sc">$</span>sample_id), ]</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(sam_tab, <span class="st">&quot;files/data-prep/tables/ssu18_sam_tab_mc.txt&quot;</span>,</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<h3 id="create-the-taxmap-object">Create the Taxmap Object</h3>
<p>Now that we have a local copy of each table (<code>tax_tab</code>, <code>asv_tab</code>, &amp; <code>sam_tab</code>), we can read those tables back in using the <a href="https://readr.tidyverse.org/articles/readr.html">readr</a> package. Why are we doing it this way instead of just setting each table to a variable? Great question. We do it this way because <code>readr</code> returns tibbles instead of data frames. Metacoder/taxa works best when the data is in tibble format. Instead of writing and reading the tables, you could use the command <code>tibble()</code>, which would convert the data frame to a tibble.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>asv_data_i <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;files/data-prep/tables/ssu18_asv_tab_mc.txt&quot;</span>)</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(asv_data_i)</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>tax_data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;files/data-prep/tables/ssu18_tax_tab_mc.txt&quot;</span>)</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(tax_data)</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>samp_data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;files/data-prep/tables/ssu18_sam_tab_mc.txt&quot;</span>,</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">col_types =</span> <span class="st">&quot;ccccccddc&quot;</span>)</span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(samp_data)</span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>asv_data <span class="ot">&lt;-</span> <span class="fu">left_join</span>(asv_data_i, tax_data,</span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>                      <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;asv_id&quot;</span> <span class="ot">=</span> <span class="st">&quot;asv_id&quot;</span>))</span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(asv_data)</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(asv_data, <span class="st">&quot;files/data-prep/tables/ssu18_asv_tax_tab.csv&quot;</span>)</span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(samp_data, <span class="st">&quot;files/data-prep/tables/ssu18_sam_tab_mc.csv&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>With all the data in hand that we need, time to create a taxmap object.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">parse_tax_data</span>(asv_data,</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">class_cols =</span> <span class="dv">17</span><span class="sc">:</span><span class="dv">23</span>,</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">named_by_rank =</span> <span class="cn">TRUE</span>,</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">include_tax_data =</span> <span class="cn">TRUE</span>) <span class="co"># makes `taxon_ranks()` work</span></span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(obj<span class="sc">$</span>data) <span class="ot">&lt;-</span> <span class="st">&quot;asv_counts&quot;</span></span></code></pre></div>
</div>
<h2 id="ampvis2-object">Ampvis2 Object</h2>
<p>The last task is to create an <a href="https://madsalbertsen.github.io/ampvis2/index.html">ampvis2</a> object so we can use the ampvis2 <span class="citation" data-cites="kasper2018ampvis2">(<a href="#ref-kasper2018ampvis2" role="doc-biblioref">Andersen et al. 2018</a>)</span> R package for some of the <a href="https://madsalbertsen.github.io/ampvis2/reference/index.html">analyses and visualizations</a>. An ampvis2 object is similar in structure to a phyloseq object except that you need to provide a combined ASV/taxonomy table. To create the object we use the modified phyloseq object, <code>ps_amp</code>.</p>
<h3 id="create-dataframes">Create Dataframes</h3>
<p>Start by generating and modifying dataframes from the tables in the phyloseq object. Ampvis2 requires the column header âOTUâ instead of âASVâ or any other name.</p>
<h4 id="sample-metadata">Sample metadata</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>amp_meta_1 <span class="ot">&lt;-</span> sam_tab</span></code></pre></div>
</div>
<h4 id="asv-data">ASV data</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>amp_asv  <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">otu_table</span>(ps_amp))</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>amp_asv <span class="ot">&lt;-</span> amp_asv <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;OTU&quot;</span>)</span></code></pre></div>
</div>
<h4 id="taxonomy-data">Taxonomy data</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>amp_tax  <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ps_amp))</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>amp_tax <span class="ot">&lt;-</span> amp_tax <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;OTU&quot;</span>)</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>amp_tax<span class="sc">$</span>ASV_SEQ <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(amp_tax)[<span class="fu">colnames</span>(amp_tax) <span class="sc">==</span> <span class="st">&quot;ASV_ID&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Species&quot;</span></span></code></pre></div>
</div>
<h4 id="merge-asv-taxonomy-tables">Merge ASV &amp; taxonomy tables</h4>
<p>Here, rows must be OTUs (ASVs) and the last 7 columns must be taxonomic lineage.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>amp_asv_tax <span class="ot">&lt;-</span> <span class="fu">left_join</span>(amp_asv, amp_tax, <span class="at">by =</span> <span class="st">&quot;OTU&quot;</span>)</span></code></pre></div>
</div>
<h3 id="create-the-ampvis2-object">Create the ampvis2 object</h3>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>amp_data <span class="ot">&lt;-</span> <span class="fu">amp_load</span>(amp_asv_tax, <span class="at">metadata =</span> amp_meta_1, <span class="at">tree =</span> ps_work_tree)</span></code></pre></div>
</div>
<p>You can now access the different parts of the ampvis2 object using <code>amp_data$abund</code> for the ASV table, <code>amp_data$tax</code> for the taxonomy table, and <code>amp_data$metadata</code> for the sample data.</p>
<h2 id="review">Review</h2>
<p>We have a few different options to play with. At this point it is difficult to say which we will use or whether additional objects need to be created. Here is a summary of the objects we have.</p>
<p>The phyloseq object <em>before</em> changing NA ranks.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>ps_work_o</span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20173 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20173 taxa by 8 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 20173 tips and 20172 internal nodes ]</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_work_o, <span class="st">&quot;Family&quot;</span>), <span class="dv">16</span>)</span></code></pre></div>
<pre><code> [1] &quot;A4b&quot;                            &quot;Burkholderiaceae&quot;              
 [3] &quot;Peptococcaceae&quot;                 NA                              
 [5] &quot;Polyangiaceae&quot;                  &quot;Pedosphaeraceae&quot;               
 [7] &quot;AKYH767&quot;                        &quot;Micromonosporaceae&quot;            
 [9] &quot;Acidobacteriaceae_(Subgroup_1)&quot; &quot;Pyrinomonadaceae&quot;              
[11] &quot;Clostridiaceae&quot;                 &quot;Myxococcaceae&quot;                 
[13] &quot;Hyphomonadaceae&quot;                &quot;Roseiflexaceae&quot;                
[15] &quot;type_III&quot;                       &quot;Ktedonobacteraceae&quot;            </code></pre>
</div>
<p>The phyloseq object <em>after</em> changing NA ranks.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>ps_work</span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 20173 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 20173 taxa by 8 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 20173 tips and 20172 internal nodes ]</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_work, <span class="st">&quot;Family&quot;</span>), <span class="dv">16</span>)</span></code></pre></div>
<pre><code> [1] &quot;c_Subgroup_22&quot;         &quot;c_OM190&quot;              
 [3] &quot;A4b&quot;                   &quot;Diplorickettsiaceae&quot;  
 [5] &quot;k_Bacteria&quot;            &quot;Gemmataceae&quot;          
 [7] &quot;c_Gammaproteobacteria&quot; &quot;Microscillaceae&quot;      
 [9] &quot;Entotheonellaceae&quot;     &quot;Syntrophaceae&quot;        
[11] &quot;p_WS2&quot;                 &quot;Verrucomicrobiaceae&quot;  
[13] &quot;Pyrinomonadaceae&quot;      &quot;Caulobacteraceae&quot;     
[15] &quot;Parachlamydiaceae&quot;     &quot;c_Phycisphaerae&quot;      </code></pre>
</div>
<p>The taxmap object.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(obj)</span></code></pre></div>
<pre><code>&lt;Taxmap&gt;
  22454 taxa: aaab. Bacteria ... bhfq. ASV7647
  22454 edges: NA-&gt;aaab, NA-&gt;aaac ... aceb-&gt;bhfq
  1 data sets:
    asv_counts:
      # A tibble: 20,173 x 24
        taxon_id asv_id  P01_D00_010_W3A P01_D00_010_W8A
        &lt;chr&gt;    &lt;chr&gt;             &lt;dbl&gt;           &lt;dbl&gt;
      1 adju     ASV9128               0               0
      2 adjv     ASV7870               0               0
      3 adjw     ASV8537               0               0
      # â¦ with 20,170 more rows, and 20 more variables:
      #   P02_D00_010_C0A &lt;dbl&gt;, P03_D00_010_W3B &lt;dbl&gt;,
      #   P03_D00_010_W8B &lt;dbl&gt;, P04_D00_010_C0B &lt;dbl&gt;,
      #   P05_D00_010_W3C &lt;dbl&gt;, P05_D00_010_W8C &lt;dbl&gt;,
      #   P06_D00_010_C0C &lt;dbl&gt;, P07_D00_010_W3D &lt;dbl&gt;,
      #   P07_D00_010_W8D &lt;dbl&gt;, P08_D00_010_C0D &lt;dbl&gt;, â¦
  0 functions:</code></pre>
</div>
<p>The ampvis2 object.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>amp_data</span></code></pre></div>
<pre><code>ampvis2 object with 4 elements. 
Summary of OTU table:
     Samples         OTUs  Total#Reads    Min#Reads    Max#Reads 
          15        20173       936640        25088        86600 
Median#Reads    Avg#Reads 
       62117     62442.67 

Assigned taxonomy:
      Kingdom        Phylum         Class         Order        Family 
  20173(100%) 20166(99.97%) 20163(99.95%) 20163(99.95%) 20065(99.46%) 
        Genus       Species 
20165(99.96%)   20173(100%) 

Metadata variables: 9 
 sample_id, PLOT, DEPTH, TREAT, TEMP, PAIR, total_reads, total_asvs, coverage</code></pre>
</div>
<h2 id="final-steps">Final Steps</h2>
<p>The last thing tasks to complete are to <strong>a</strong>) save copies of the taxonomy and sequence tables and <strong>b</strong>) save an image of the workflow for the next section.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">tax_table</span>(ps_work),</span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;files/data-prep/tables/ssu18_work_tax_table.txt&quot;</span>,</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ps_work)),</span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;files/data-prep/tables/ssu18_work_seq_table.txt&quot;</span>,</span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="its-data">ITS Data</h1>
<aside>
<div class="layout-chunk" data-layout="l-body">
<input onclick="codefolder(&#39;pre.sourceCode&#39;);" type="button" value="Hide Code" id="codefolder-button" style="position: absolute; right: 14%; z-index: 200"/>
<script>
  function codefolder(query) {

    var x = document.querySelectorAll(query);
    if (x.length === 0) return;

    function toggle_vis(o) {
      var d = o.style.display;
      o.style.display = (d === 'block' || d === '') ? 'none':'block';
    }

    for (i = 0; i < x.length; i++) {
      var y = x[i];
      toggle_vis(y);
    }

    var elem = document.getElementById("codefolder-button");
    if (elem.value === "Hide Code") elem.value = "Show Code";
    else elem.value = "Hide Code";
  }
</script>
</div>
</aside>
<h2 id="prerequisites-1">Prerequisites</h2>
<p>In order to run this workflow, you either need to run the corresponding <a href="dada2-high-temp.html#its-data">DADA2 Workflow for 2018 ITS</a> or begin with the output from that workflow, <code>data_prep_its18_wf.rdata</code>. See the <a href="data-availability.html">Data Availability</a> page for complete details.</p>
<p>Unless otherwise noted, we primarily use <a href="https://joey711.github.io/phyloseq/">phyloseq</a> <span class="citation" data-cites="mcmurdie2013phyloseq">(<a href="#ref-mcmurdie2013phyloseq" role="doc-biblioref">McMurdie and Holmes 2013</a>)</span> in this section of the workflow to prepare the 2018 ITS data set for community analyses. We prepare the data set by curating samples, removing contaminants, and creating phyloseq objects.</p>
<h2 id="read-counts-assessment-1">Read Counts Assessment</h2>
<p>Before we begin, letâs create a summary table containing some basic sample metadata and the read count data from the <a href="dada2-high-temp.html#its-data">DADA2 workflow</a>. We need to inspect how total reads changed from to beginning to the end of the workflow. While we are at it, letâs create more intuitive Sample IDs. For more details on how reads changed at each step of the workflow, see the <a href="dada2-high-temp.html#track-reads-through-workflow-1">summary table</a> at the end of the DADA2 section.</p>
<p>Table headers are as follows:</p>
<table>
<colgroup>
<col style="width: 23%" />
<col style="width: 76%" />
</colgroup>
<thead>
<tr class="header">
<th>Header</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>Sample ID</code></td>
<td>the new sample ID based on Plot number, Depth, Treatment, Temperature, &amp; Pair</td>
</tr>
<tr class="even">
<td><code>Plot</code></td>
<td>the experimental plot number</td>
</tr>
<tr class="odd">
<td><code>Depth</code></td>
<td>the depth where the sample was collected</td>
</tr>
<tr class="even">
<td><code>Treatment</code></td>
<td>warm or control</td>
</tr>
<tr class="odd">
<td><code>Temp</code></td>
<td>temperature of soil heating (0C, +4C, or +8C)</td>
</tr>
<tr class="even">
<td><code>Pair</code></td>
<td>Control/Warm plot pairs</td>
</tr>
<tr class="odd">
<td><code>input</code></td>
<td>number of raw reads</td>
</tr>
<tr class="even">
<td><code>nochim</code></td>
<td>final read count after removing chimeras</td>
</tr>
<tr class="odd">
<td><code>Remain</code></td>
<td>percent of reads remaining from <code>input</code> to <code>nonchim</code></td>
</tr>
<tr class="even">
<td><code>FastqID</code></td>
<td>base name of the fastq file</td>
</tr>
</tbody>
</table>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-page">
<div id="33fgpgd1k5ugbiw74hvx" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="33fgpgd1k5ugbiw74hvx">{"x":{"filter":"top","filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"disabled\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0\" data-max=\"8\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"809\" data-max=\"96531\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"80\" data-max=\"64636\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.056\" data-max=\"0.686\" data-scale=\"3\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Sample summary table including\n            read changes at start and end of DADA2 workflow. Use the buttons to\n            navigate through the table or download a copy.<\/em>\n<\/caption>","data":[["P01_D00_010_W3A","P02_D00_010_C0A","P03_D00_010_W3B","P04_D00_010_C0B","P05_D00_010_W3C","P06_D00_010_C0C","P07_D00_010_W3D","P08_D00_010_C0D","P09_D00_010_W3E","P10_D00_010_C0E","P01_D00_010_W8A","P03_D00_010_W8B","P05_D00_010_W8C","P07_D00_010_W8D","P09_D00_010_W8E"],["P01","P02","P03","P04","P05","P06","P07","P08","P09","P10","P01","P03","P05","P07","P09"],["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],["Warm","Control","Warm","Control","Warm","Control","Warm","Control","Warm","Control","Warm","Warm","Warm","Warm","Warm"],[3,0,3,0,3,0,3,0,3,0,8,8,8,8,8],["A","A","B","B","C","C","D","D","E","E","A","B","C","D","E"],[50894,74581,60006,88732,809,67419,70123,60799,55927,96531,70734,31174,81908,15435,1418],[29794,51144,38374,56593,296,45850,36063,38371,29102,64636,41880,14188,35600,9172,80],[0.585,0.686,0.64,0.638,0.366,0.68,0.514,0.631,0.52,0.67,0.592,0.455,0.435,0.594,0.056],["P1","P2","P3","P4","P5","P6","P7","P8","P9","P10","P1-8C","P3-8C","P5-8C","P7-8C","P9-8C"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Sample_ID<\/th>\n      <th>Plot<\/th>\n      <th>Depth<\/th>\n      <th>Treatment<\/th>\n      <th>Temp<\/th>\n      <th>Pair<\/th>\n      <th>input<\/th>\n      <th>nochim<\/th>\n      <th>Remain<\/th>\n      <th>FastqID<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[5,15],"columnDefs":[{"className":"dt-right","targets":[4,6,7,8]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-size':'80%'});\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\nvar value=data[8]; $(this.api().cell(row, 8).node()).css({'font-size':'80%'});\nvar value=data[9]; $(this.api().cell(row, 9).node()).css({'font-size':'80%'});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
<p>The following samples have fewer than <strong>1000</strong> reads at the end of the workflow: <strong>P5, P9-8C</strong>.</p>
<p>We can also plot the final read count by the temperature where the sample was collected.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body-outset">

</div>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/data-prep/its18_samp_summary_plot.png" width="1050" /></p>
</div>
<p><small>Final read count by depth. Each point represents a differnt sample. Note the Negative control is not included.</small></p>
<h2 id="defining-groups-1">Defining Groups</h2>
<ol type="1">
<li><p>Load the data packet produced in the final step of the DADA2 workflow. This packet (<code>its18_dada2_wf.rdata</code>) contains the ASV-by-sample table and the ASV taxonomy table.</p></li>
<li><p>Rename the samples so names have plot and Depth info. For convenience, we renamed any Negative Controls (NC) to <code>P00_D00-000_NNN</code>.</p></li>
<li><p>After we load the data packet we next need to format sample names and define groups.</p></li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>joined_tab_ro <span class="ot">&lt;-</span> joined_tab_percent[<span class="fu">order</span>(joined_tab_percent<span class="sc">$</span>FastqID), ]</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>new_names <span class="ot">&lt;-</span> joined_tab_ro<span class="sc">$</span><span class="st">`</span><span class="at">Sample_ID</span><span class="st">`</span></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;files/dada2/rdata/its18_dada2_wf.rdata&quot;</span>)</span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(seqtab) <span class="ot">&lt;-</span> new_names</span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>samples.out <span class="ot">&lt;-</span> <span class="fu">rownames</span>(seqtab)</span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a>sample_name <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">1</span>, <span class="dv">999</span>)</span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a>plot <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">0</span>, <span class="dv">3</span>)</span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a>depth <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">6</span>, <span class="dv">11</span>)</span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a>treatment <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">13</span>, <span class="dv">13</span>)</span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">14</span>, <span class="dv">14</span>)</span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a>pair <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples.out, <span class="dv">15</span>, <span class="dv">15</span>)</span></code></pre></div>
</details>
</div>
<p>We have a total of <strong>15</strong> samples, <strong>5</strong> sample pairs, from <strong>10</strong> plots, and <strong>1</strong> different depths. There are <strong>2</strong> treatments corresponding to <strong>3</strong> different temperature regimes.</p>
<ol start="4" type="1">
<li>And finally we define a sample data frame that holds the different groups we extracted from the sample names.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>samdf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">SamName =</span> sample_name,</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">PLOT =</span> plot,</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">DEPTH =</span> depth,</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">TREAT =</span> treatment,</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">TEMP =</span> temp,</span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">PAIR =</span> pair)</span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(samdf) <span class="ot">&lt;-</span> samples.out</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>For example, P08_D00_010_C0D is the sample heated at +0C from plot P08 and a depth of 00_010cm. This sample is in group D along with P07_D00_010_W3D, P07_D00_010_W8D, P08_D00_010_C0D.</p>
<h2 id="create-a-phyloseq-object-1">Create a Phyloseq Object</h2>
<h3 id="fix-taxon-names">Fix Taxon Names</h3>
<p>The UNITE database we used for classification adds a little prefix to every rank call, like <code>k__</code> for Kingdom, <code>o__</code> for Order, etc. To keep in step with our other analyses, we will remove these tags. Of course, we also make a ps object with the original names just in case. At this point we will also remove the Species rank since it is essentially useless.</p>
<aside>
<em>I KNOW there is an easier way to do this</em>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>tax_mod_o <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(tax)</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>tax_mod_o<span class="sc">$</span>Species <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>tax_mod_o <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tax_mod_o)</span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>tax_mod <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;k__&quot;</span>, <span class="st">&quot;&quot;</span>, tax_mod_o)</span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a>tax_mod <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;p__&quot;</span>, <span class="st">&quot;&quot;</span>, tax_mod)</span>
<span id="cb57-6"><a href="#cb57-6" aria-hidden="true" tabindex="-1"></a>tax_mod <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;c__&quot;</span>, <span class="st">&quot;&quot;</span>, tax_mod)</span>
<span id="cb57-7"><a href="#cb57-7" aria-hidden="true" tabindex="-1"></a>tax_mod <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;o__&quot;</span>, <span class="st">&quot;&quot;</span>, tax_mod)</span>
<span id="cb57-8"><a href="#cb57-8" aria-hidden="true" tabindex="-1"></a>tax_mod <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;f__&quot;</span>, <span class="st">&quot;&quot;</span>, tax_mod)</span>
<span id="cb57-9"><a href="#cb57-9" aria-hidden="true" tabindex="-1"></a>tax_mod <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;g__&quot;</span>, <span class="st">&quot;&quot;</span>, tax_mod)</span></code></pre></div>
</div>
<p><strong>A</strong>. The first step is to rename the amplicon sequence variants (ASVs) so the designations are a bit more user friendly. By default, DADA2 names each ASV by its unique sequence so that data can be directly compared across studies (which is great). But this convention can get cumbersome downstream, so we rename the ASVs using a simpler conventionâASV1, ASV2, ASV3, and so on.</p>
<aside>
A phyloseq object contains ASV table (taxa abundances), sample metadata, and taxonomy table (mapping between ASVs and higher-level taxonomic classifications).
</aside>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># this create the phyloseq object</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(<span class="fu">otu_table</span>(seqtab, <span class="at">taxa_are_rows =</span> <span class="cn">FALSE</span>),</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">sample_data</span>(samdf), <span class="fu">tax_table</span>(tax_mod))</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">tax_table</span>(ps),</span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">rownames</span>(<span class="fu">tax_table</span>(ps)))</span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a><span class="co"># adding unique ASV names</span></span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a><span class="fu">taxa_names</span>(ps) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;ASV&quot;</span>, <span class="fu">seq</span>(<span class="fu">ntaxa</span>(ps)))</span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">tax_table</span>(ps),</span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">rownames</span>(<span class="fu">tax_table</span>(ps)))</span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">taxa_names</span>(ps))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;ASV1&quot; &quot;ASV2&quot; &quot;ASV3&quot; &quot;ASV4&quot; &quot;ASV5&quot; &quot;ASV6&quot;</code></pre>
</div>
<p>So the complete data set contains 3357 ASVs. We can also use the <a href="https://github.com/microbiome/microbiome/">microbiome R package</a> <span class="citation" data-cites="lahti2017microbiome">(<a href="#ref-lahti2017microbiome" role="doc-biblioref">Lahti, Sudarshan, and others 2017</a>)</span> to get some additional summary data from the phyloseq object.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<table>
<colgroup>
<col style="width: 40%" />
<col style="width: 59%" />
</colgroup>
<thead>
<tr class="header">
<th>Metric</th>
<th>Results</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>80</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>64636</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>491143</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>32743</td>
</tr>
<tr class="odd">
<td>Median number of reads</td>
<td>36063</td>
</tr>
<tr class="even">
<td>Sparsity</td>
<td>0.802</td>
</tr>
<tr class="odd">
<td>Any ASVs sum to 1 or less?</td>
<td>TRUE</td>
</tr>
<tr class="even">
<td>Number of singleton ASVs</td>
<td>3</td>
</tr>
<tr class="odd">
<td>Percent of ASVs that are singletons</td>
<td>0.089</td>
</tr>
<tr class="even">
<td>Number of sample variables are:</td>
<td>6 (SamName, PLOT, DEPTH, TREAT, TEMP, PAIR)</td>
</tr>
</tbody>
</table>
<p><strong>B</strong>. Add two final columns containing the ASV sequences and ASV IDs. This will be useful later when trying to export a fasta file. We can also take a look at the phyloseq object.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(<span class="fu">tax_table</span>(ps)) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Kingdom&quot;</span>, <span class="st">&quot;Phylum&quot;</span>, <span class="st">&quot;Class&quot;</span>, <span class="st">&quot;Order&quot;</span>,</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Genus&quot;</span>, <span class="st">&quot;ASV_SEQ&quot;</span>, <span class="st">&quot;ASV_ID&quot;</span>)</span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rank_names</span>(ps)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3357 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3357 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><strong>C</strong>. Export sequence and taxonomy tables for the unadulterated phyloseq object for later use. We will use the prefix <code>full</code> to indicate that these are the <em>raw</em> outputs.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">tax_table</span>(ps),</span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;files/data-prep/tables/its18_full_tax_table.txt&quot;</span>,</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ps)),</span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;files/data-prep/tables/its18_full_seq_table.txt&quot;</span>,</span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>At this point we would normally remove unwanted taxa and/or negative controls. Since we do not have any of these for this dataset, we are going to save the phyloseq object with a new name. That way, if we do decide later on to remove some taxa it will be easier to plug in the code here and then continue with the rest of the workflow.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>ps_filt <span class="ot">&lt;-</span> ps</span></code></pre></div>
</div>
<h2 id="remove-unwanted-samples">Remove Unwanted Samples</h2>
<p>Next, we can remove samples with really low read counts, say less than 500 reads.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>ps_filt_no_low <span class="ot">&lt;-</span> <span class="fu">prune_samples</span>(<span class="fu">sample_sums</span>(ps_filt) <span class="sc">&gt;</span> <span class="dv">500</span>, ps_filt)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>ps_filt_no_low</span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3357 taxa and 13 samples ]
sample_data() Sample Data:       [ 13 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3357 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<p>We lost 2 sample(s). After removing samples we need to check whether any ASVs ended up with no reads.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>no_reads <span class="ot">&lt;-</span> <span class="fu">taxa_sums</span>(ps_filt_no_low) <span class="sc">==</span> <span class="dv">0</span></span></code></pre></div>
</div>
<p>And we lost 2 ASV(s). So now we must remove these from the ps object.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>ps_filt_no_nc <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(<span class="fu">taxa_sums</span>(ps_filt_no_low) <span class="sc">&gt;</span> <span class="dv">0</span>, ps_filt_no_low)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>ps_filt_no_nc</span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3355 taxa and 13 samples ]
sample_data() Sample Data:       [ 13 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3355 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<h2 id="rename-na-taxonomic-ranks-1">Rename <em>NA</em> taxonomic ranks</h2>
<p>Phyloseq has an odd way of dealing with taxonomic ranks that have no valueâin other words, <em>NA</em> in the tax table. The first thing we are going to do before moving forward is to change all of the <em>NA</em>s to have a value of the next highest classified rank. For example, <code>ASV4</code> is not classified at the Genus level but is at Family level (Nitrososphaeraceae). So we change the Genus name to <em>Family_Nitrososphaeraceae</em>. The code for comes from these two posts on the phyloseq GitHub, both by <a href="https://github.com/MSMortensen">MSMortensen</a>: issue <a href="https://github.com/joey711/phyloseq/issues/850#issuecomment-394771087">#850</a> and issue <a href="https://github.com/joey711/phyloseq/issues/990#issuecomment-424618425">#990</a>.</p>
<blockquote>
<p>One thing this code does is reassign the functions <code>class</code> and <code>order</code> to taxonomic ranks. This can cause issues if you need these functions.</p>
</blockquote>
<p>So you need to run something like this <code>rm(class, order, phylum, kingdom)</code> at the end of the code to remove these as variables. For now, I have not come up with a better solution.</p>
<p>Before doing that, we will save a copy of the phyloseq object with the original taxa names.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>ps_work_o <span class="ot">&lt;-</span> ps_filt_no_nc</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>tax.clean <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ps_filt_no_nc))</span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>){ tax.clean[,i] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tax.clean[,i])}</span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>tax.clean[<span class="fu">is.na</span>(tax.clean)] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span></span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(tax.clean)){</span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (tax.clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb72-7"><a href="#cb72-7" aria-hidden="true" tabindex="-1"></a>        kingdom <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;k_&quot;</span>, tax.clean[i,<span class="dv">1</span>], <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb72-8"><a href="#cb72-8" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> kingdom</span>
<span id="cb72-9"><a href="#cb72-9" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb72-10"><a href="#cb72-10" aria-hidden="true" tabindex="-1"></a>        phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;p_&quot;</span>, tax.clean[i,<span class="dv">2</span>], <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb72-11"><a href="#cb72-11" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">3</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb72-12"><a href="#cb72-12" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">4</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb72-13"><a href="#cb72-13" aria-hidden="true" tabindex="-1"></a>        class <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;c_&quot;</span>, tax.clean[i,<span class="dv">3</span>], <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb72-14"><a href="#cb72-14" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">4</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> class</span>
<span id="cb72-15"><a href="#cb72-15" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">5</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb72-16"><a href="#cb72-16" aria-hidden="true" tabindex="-1"></a>        order <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;o_&quot;</span>, tax.clean[i,<span class="dv">4</span>], <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb72-17"><a href="#cb72-17" aria-hidden="true" tabindex="-1"></a>        tax.clean[i, <span class="dv">5</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> order</span>
<span id="cb72-18"><a href="#cb72-18" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (tax.clean[i,<span class="dv">6</span>] <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb72-19"><a href="#cb72-19" aria-hidden="true" tabindex="-1"></a>        tax.clean<span class="sc">$</span>Genus[i] <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;f&quot;</span>,tax.clean<span class="sc">$</span>Family[i], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb72-20"><a href="#cb72-20" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb72-21"><a href="#cb72-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb72-22"><a href="#cb72-22" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps_filt_no_nc) <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tax.clean)</span>
<span id="cb72-23"><a href="#cb72-23" aria-hidden="true" tabindex="-1"></a><span class="fu">rank_names</span>(ps_filt_no_nc)</span>
<span id="cb72-24"><a href="#cb72-24" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(class, order, phylum, kingdom)</span></code></pre></div>
</details>
</div>
<p>Still the same ranks. Thatâs good. What about the new groups? Letâs take a peak at some families.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_filt_no_nc, <span class="st">&quot;Family&quot;</span>), <span class="dv">16</span>)</span></code></pre></div>
<pre><code> [1] &quot;Hygrophoraceae&quot;         &quot;Mycosphaerellaceae&quot;    
 [3] &quot;p_Ascomycota&quot;           &quot;Trichosporonaceae&quot;     
 [5] &quot;c_Agaricomycetes&quot;       &quot;Archaeorhizomycetaceae&quot;
 [7] &quot;Chaetomellaceae&quot;        &quot;Geastraceae&quot;           
 [9] &quot;k_Fungi&quot;                &quot;Metschnikowiaceae&quot;     
[11] &quot;Clavariaceae&quot;           &quot;o_Agaricales&quot;          
[13] &quot;Ascobolaceae&quot;           &quot;Trichocomaceae&quot;        
[15] &quot;Piptocephalidaceae&quot;     &quot;p_Chytridiomycota&quot;     </code></pre>
</div>
<p>Nice. Adios <em>NA</em>.</p>
<blockquote>
<p><em>Note</em>. The original code is written for data sets that have species-level designations.</p>
</blockquote>
<p>Since this data set does not contain species, I modified the code to stop at the genus level. If your data set has species, my modifications will not work for you.</p>
<p>Finally, we rename the ps object. This is now our working data set.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>ps_work <span class="ot">&lt;-</span> ps_filt_no_nc</span></code></pre></div>
</div>
<p>Later in the workflow, we will generate an ampvis2 object from the working phyloseq object, but first we need to fix the OTU table from the phyloseq object. For some reason, values in otu table are <code>int</code> but need to be <code>dbl</code> or else ampvis2 will not work properly.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>otu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ps_work)))</span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>otu[] <span class="ot">&lt;-</span> <span class="fu">lapply</span>(otu, as.numeric)</span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>otu <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(otu)</span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a>tax <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ps_work)))</span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(ps_work))</span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a>ps_amp <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(<span class="fu">otu_table</span>(otu, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>),</span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">tax_table</span>(tax, tax),</span>
<span id="cb76-8"><a href="#cb76-8" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">sample_data</span>(samples))</span></code></pre></div>
</details>
</div>
<h2 id="sample-summary-1">Sample Summary</h2>
<p>Now that we have removed the NC and samples with low reads, we can summarize the data in our working phyloseq object. Letâs start with the data set as a whole using the <code>summarize_phyloseq</code> from the <a href="https://github.com/microbiome/microbiome/">microbiome R package</a> <span class="citation" data-cites="lahti2017microbiome">(<a href="#ref-lahti2017microbiome" role="doc-biblioref">Lahti, Sudarshan, and others 2017</a>)</span> as we did above.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<table>
<colgroup>
<col style="width: 38%" />
<col style="width: 61%" />
</colgroup>
<thead>
<tr class="header">
<th>Metric</th>
<th>Results</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>9172</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>64636</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>490767</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>37751</td>
</tr>
<tr class="odd">
<td>Median number of reads</td>
<td>38371</td>
</tr>
<tr class="even">
<td>Sparsity</td>
<td>0.671</td>
</tr>
<tr class="odd">
<td>Any ASVs sum to 1 or less?</td>
<td>TRUE</td>
</tr>
<tr class="even">
<td>Number of singleton ASVs</td>
<td>5</td>
</tr>
<tr class="odd">
<td>Percent of ASVs that are singletons</td>
<td>0.149</td>
</tr>
<tr class="even">
<td>Number of sample variables are:</td>
<td>6 (SamName, PLOT, DEPTH, TREAT, TEMP, PAIR)</td>
</tr>
</tbody>
</table>
<p>We can also generate a summary table of total reads &amp; ASVs for each sample. You can sort the table, download a copy, or filter by search term. Here is the code to generate the data for the table. First, we create data frames that hold total reads and ASVs for each sample.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> <span class="fu">sample_sums</span>(ps_work)</span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(total_reads, <span class="at">make.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>total_reads <span class="ot">&lt;-</span> total_reads <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;Sample_ID&quot;</span>)</span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-5"><a href="#cb77-5" aria-hidden="true" tabindex="-1"></a>total_asvs <span class="ot">&lt;-</span> <span class="fu">estimate_richness</span>(ps_work, <span class="at">measures =</span> <span class="st">&quot;Observed&quot;</span>)</span>
<span id="cb77-6"><a href="#cb77-6" aria-hidden="true" tabindex="-1"></a>total_asvs <span class="ot">&lt;-</span> total_asvs <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;Sample_ID&quot;</span>)</span>
<span id="cb77-7"><a href="#cb77-7" aria-hidden="true" tabindex="-1"></a>total_asvs<span class="sc">$</span>Sample_ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>, <span class="st">&#39;-&#39;</span>, total_asvs<span class="sc">$</span>Sample_ID)</span></code></pre></div>
</details>
</div>
<p>And then we merge these two data frames with the sample data frame.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>sam_details <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(ps_work))</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sam_details) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(sam_details) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Sample_ID&quot;</span>, <span class="st">&quot;Plot&quot;</span>, <span class="st">&quot;Depth&quot;</span>,</span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;Treatment&quot;</span>, <span class="st">&quot;Temp&quot;</span>, <span class="st">&quot;Pair&quot;</span>)</span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a>joined_percent <span class="ot">&lt;-</span> joined_tab_percent</span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a>joined_percent[, <span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb78-8"><a href="#cb78-8" aria-hidden="true" tabindex="-1"></a>merge_tab <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(sam_details, total_reads) <span class="sc">%&gt;%</span></span>
<span id="cb78-9"><a href="#cb78-9" aria-hidden="true" tabindex="-1"></a>                <span class="fu">left_join</span>(., joined_percent)</span>
<span id="cb78-10"><a href="#cb78-10" aria-hidden="true" tabindex="-1"></a>percent_change2 <span class="ot">&lt;-</span> <span class="dv">100</span><span class="sc">*</span>(<span class="dv">1</span><span class="sc">-</span>(merge_tab<span class="sc">$</span>total_reads<span class="sc">/</span>merge_tab<span class="sc">$</span>nochim)) <span class="sc">%&gt;%</span></span>
<span id="cb78-11"><a href="#cb78-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">round</span>(<span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb78-12"><a href="#cb78-12" aria-hidden="true" tabindex="-1"></a>merge_tab[<span class="dv">11</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb78-13"><a href="#cb78-13" aria-hidden="true" tabindex="-1"></a>merge_tab<span class="sc">$</span>Change <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">sprintf</span>(<span class="st">&quot;%.3f&quot;</span>, percent_change2))</span>
<span id="cb78-14"><a href="#cb78-14" aria-hidden="true" tabindex="-1"></a>merge_tab2 <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(merge_tab, total_asvs, <span class="at">by =</span> <span class="st">&quot;Sample_ID&quot;</span>)</span>
<span id="cb78-15"><a href="#cb78-15" aria-hidden="true" tabindex="-1"></a>merge_tab2 <span class="ot">&lt;-</span> merge_tab2[, <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>,<span class="dv">11</span>,<span class="dv">12</span>)]</span>
<span id="cb78-16"><a href="#cb78-16" aria-hidden="true" tabindex="-1"></a>merge_tab2<span class="sc">$</span>coverage <span class="ot">=</span> <span class="fu">cut</span>(merge_tab2<span class="sc">$</span>total_reads,</span>
<span id="cb78-17"><a href="#cb78-17" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">5000</span>, <span class="dv">25000</span>, <span class="dv">100000</span>, <span class="dv">200000</span>),</span>
<span id="cb78-18"><a href="#cb78-18" aria-hidden="true" tabindex="-1"></a>                          <span class="at">labels =</span> <span class="fu">c</span>(</span>
<span id="cb78-19"><a href="#cb78-19" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&quot;low (&lt; 5k)&quot;</span>, <span class="st">&quot;medium (5-25k)&quot;</span>,</span>
<span id="cb78-20"><a href="#cb78-20" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&quot;high (25-100k)&quot;</span>, <span class="st">&quot;extra_high (&gt; 100k)&quot;</span>))</span>
<span id="cb78-21"><a href="#cb78-21" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(merge_tab2) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;sample_id&quot;</span>, <span class="st">&quot;PLOT&quot;</span>, <span class="st">&quot;DEPTH&quot;</span>,</span>
<span id="cb78-22"><a href="#cb78-22" aria-hidden="true" tabindex="-1"></a>                          <span class="st">&quot;TREAT&quot;</span>, <span class="st">&quot;TEMP&quot;</span>, <span class="st">&quot;PAIR&quot;</span>,</span>
<span id="cb78-23"><a href="#cb78-23" aria-hidden="true" tabindex="-1"></a>                          <span class="st">&quot;final_reads&quot;</span>, <span class="st">&quot;reads_lost&quot;</span>, <span class="st">&quot;total_asvs&quot;</span>, <span class="st">&quot;coverage&quot;</span>)</span></code></pre></div>
</details>
</div>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="gakgruoovivy4mz28ah2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="gakgruoovivy4mz28ah2">{"x":{"filter":"top","filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"disabled\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"9172\" data-max=\"64636\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"disabled\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"335\" data-max=\"1017\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;low (&lt; 5k)&quot;,&quot;medium (5-25k)&quot;,&quot;high (25-100k)&quot;,&quot;extra_high (&gt; 100k)&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Total reads &amp; ASVs by sample.<\/em>\n<\/caption>","data":[["P01_D00_010_W3A","P01_D00_010_W8A","P10_D00_010_C0E","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E"],["P01","P01","P10","P02","P03","P03","P04","P05","P06","P07","P07","P08","P09"],["00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010","00_010"],["W","W","C","C","W","W","C","W","C","W","W","C","W"],["3","8","0","0","3","8","0","8","0","3","8","0","3"],["A","A","E","A","B","B","B","C","C","D","D","D","E"],[29794,41880,64636,51144,38374,14188,56593,35600,45850,36063,9172,38371,29102],[0,0,0,0,0,0,0,0,0,0,0,0,0],[945,719,812,1011,765,349,1017,616,954,867,335,702,745],["high (25-100k)","high (25-100k)","high (25-100k)","high (25-100k)","high (25-100k)","medium (5-25k)","high (25-100k)","high (25-100k)","high (25-100k)","high (25-100k)","medium (5-25k)","high (25-100k)","high (25-100k)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>sample_id<\/th>\n      <th>PLOT<\/th>\n      <th>DEPTH<\/th>\n      <th>TREAT<\/th>\n      <th>TEMP<\/th>\n      <th>PAIR<\/th>\n      <th>final_reads<\/th>\n      <th>reads_lost<\/th>\n      <th>total_asvs<\/th>\n      <th>coverage<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"targets":7,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\");\n  }"},{"className":"dt-right","targets":[6,7,8]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-size':'80%'});\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\nvar value=data[8]; $(this.api().cell(row, 8).node()).css({'font-size':'80%'});\nvar value=data[9]; $(this.api().cell(row, 9).node()).css({'font-size':'80%'});\n}"}},"evals":["options.columnDefs.0.render","options.rowCallback"],"jsHooks":[]}</script>
</div>
<p><br/></p>
<h2 id="taxmap-object-1">Taxmap Object</h2>
<p>We have phyloseq objects but we also want to make a <code>taxmap</code> object so we can use the <a href="https://github.com/grunwaldlab/metacoder">metacoder</a> <span class="citation" data-cites="foster2017metacoder">(<a href="#ref-foster2017metacoder" role="doc-biblioref">Foster, Sharpton, and GrÃ¼nwald 2017</a>)</span> and <a href="https://github.com/ropensci/taxa">taxa</a> packages. We will jump back and forth between between different tools and packages because each has functionality not available in the others. That said, each package also requires a specially formatted <em>object</em> to work with.</p>
<h3 id="data-prep-1">Data Prep</h3>
<p>First, we can take a look at the phyloseq object generated during the previous workflow.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>ps_work</span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3355 taxa and 13 samples ]
sample_data() Sample Data:       [ 13 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3355 taxa by 8 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 3355 tips and 3354 internal nodes ]</code></pre>
</div>
<p>In order to use the taxa package, we need to have our data in a specific format. Metacoder has a command called <code>parse_phyloseq</code>, which should do the work for us, but I was unable to get this to work for our phyloseq object. Instead, we generate the necessary tables directly from the phyloseq object and then create a basic taxmap object. I apologize for the circuitous route, but at least it is relatively simple. We need the <strong>a</strong>) taxonomy table, <strong>b</strong>) ASV table, and <strong>c</strong>) sample metadata table. We will first create the tables and then save a local copy.</p>
<h4 id="taxonomy-table-1">Taxonomy table</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>tax_tab <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">tax_table</span>(ps_work))</span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>tax_tab <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(tax_tab, <span class="st">&quot;asv_id&quot;</span>)</span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a>tax_tab<span class="sc">$</span>ASV_SEQ <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb81-4"><a href="#cb81-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tax_tab)[<span class="fu">colnames</span>(tax_tab) <span class="sc">==</span> <span class="st">&quot;ASV_ID&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;ASV&quot;</span></span>
<span id="cb81-5"><a href="#cb81-5" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(tax_tab, <span class="st">&quot;files/data-prep/tables/its18_tax_tab_mc.txt&quot;</span>,</span>
<span id="cb81-6"><a href="#cb81-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<h4 id="asv-table-1">ASV table</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>asv_tab <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">otu_table</span>(ps_work))</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a><span class="co"># this orders in decending fashion</span></span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>asv_tab <span class="ot">&lt;-</span> asv_tab[ statnet.common<span class="sc">::</span><span class="fu">order</span>(<span class="fu">row.names</span>(asv_tab)), ]</span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>asv_tab <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(asv_tab))</span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a>asv_tab <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(asv_tab, <span class="st">&quot;asv_id&quot;</span>)</span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(asv_tab, <span class="st">&quot;files/data-prep/tables/its18_asv_tab_mc.txt&quot;</span>,</span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<h4 id="sample-table-1">Sample table</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> merge_tab2</span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>sam_tab[,<span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> sam_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;total_reads&quot;</span> <span class="ot">=</span> <span class="st">&quot;final_reads&quot;</span>)</span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a><span class="co"># this orders in decending fashion</span></span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> sam_tab[statnet.common<span class="sc">::</span><span class="fu">order</span>(sam_tab<span class="sc">$</span>sample_id), ]</span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(sam_tab, <span class="st">&quot;files/data-prep/tables/its18_sam_tab_mc.txt&quot;</span>,</span>
<span id="cb83-7"><a href="#cb83-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<h3 id="create-the-taxmap-object-1">Create the Taxmap Object</h3>
<p>Now that we have a local copy of each table (<code>tax_tab</code>, <code>asv_tab</code>, &amp; <code>sam_tab</code>), we can read those tables back in using the <a href="https://readr.tidyverse.org/articles/readr.html">readr</a> package. Why are we doing it this way instead of just setting each table to a variable? Great question. We do it this way because <code>readr</code> returns tibbles instead of data frames. Metacoder/taxa works best when the data is in tibble format. Instead of writing and reading the tables, you could use the command <code>tibble()</code>, which would convert the data frame to a tibble.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>asv_data_i <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;files/data-prep/tables/its18_asv_tab_mc.txt&quot;</span>)</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(asv_data_i)</span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>tax_data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;files/data-prep/tables/its18_tax_tab_mc.txt&quot;</span>)</span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(tax_data)</span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>samp_data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;files/data-prep/tables/its18_sam_tab_mc.txt&quot;</span>,</span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">col_types =</span> <span class="st">&quot;cccccc&quot;</span>)</span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(samp_data)</span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a>asv_data <span class="ot">&lt;-</span> <span class="fu">left_join</span>(asv_data_i, tax_data,</span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a>                      <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;asv_id&quot;</span> <span class="ot">=</span> <span class="st">&quot;asv_id&quot;</span>))</span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(asv_data)</span>
<span id="cb84-11"><a href="#cb84-11" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(asv_data, <span class="st">&quot;files/data-prep/tables/its18_asv_tax_tab.csv&quot;</span>)</span>
<span id="cb84-12"><a href="#cb84-12" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(samp_data, <span class="st">&quot;files/data-prep/tables/its18_sam_tab_mc.csv&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>With all the data in hand that we need, time to create a taxmap object.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">parse_tax_data</span>(asv_data,</span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">class_cols =</span> <span class="dv">15</span><span class="sc">:</span><span class="dv">21</span>,</span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">named_by_rank =</span> <span class="cn">TRUE</span>,</span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">include_tax_data =</span> <span class="cn">TRUE</span>) <span class="co"># makes `taxon_ranks()` work</span></span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(obj<span class="sc">$</span>data) <span class="ot">&lt;-</span> <span class="st">&quot;asv_counts&quot;</span></span></code></pre></div>
</div>
<p>Now we have a taxmap object <em>and</em> a phyloseq object from the taxmap. Hopefully this will make it easier to generate data from one and import it into the other.</p>
<h2 id="ampvis2-object-1">Ampvis2 Object</h2>
<p>The last task is to create an <a href="https://madsalbertsen.github.io/ampvis2/index.html">ampvis2</a> object so we can use the ampvis2 <span class="citation" data-cites="kasper2018ampvis2">(<a href="#ref-kasper2018ampvis2" role="doc-biblioref">Andersen et al. 2018</a>)</span> R package for some of the <a href="https://madsalbertsen.github.io/ampvis2/reference/index.html">analyses and visualizations</a>. An ampvis2 object is similar in structure to a phyloseq object except that you need to provide a combined ASV/taxonomy table. To create the object we use the modified phyloseq object, <code>ps_amp</code>.</p>
<h3 id="create-dataframes-1">Create Dataframes</h3>
<p>Start by generating and modifying dataframes from the tables in the phyloseq object. Ampvis2 requires the column header âOTUâ instead of âASVâ or any other name.</p>
<h4 id="sample-metadata-1">Sample metadata</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>amp_meta_1 <span class="ot">&lt;-</span> sam_tab</span></code></pre></div>
</div>
<h4 id="asv-data-1">ASV data</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a>amp_asv <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">otu_table</span>(ps_amp))</span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>amp_asv <span class="ot">&lt;-</span> amp_asv <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;OTU&quot;</span>)</span></code></pre></div>
</div>
<h4 id="taxonomy-data-1">Taxonomy data</h4>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>amp_tax  <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ps_amp))</span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>amp_tax <span class="ot">&lt;-</span> amp_tax <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;OTU&quot;</span>)</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>amp_tax<span class="sc">$</span>ASV_SEQ <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(amp_tax)[<span class="fu">colnames</span>(amp_tax) <span class="sc">==</span> <span class="st">&quot;ASV_ID&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Species&quot;</span></span></code></pre></div>
</div>
<h4 id="merge-asv-taxonomy-tables-1">Merge ASV &amp; taxonomy tables</h4>
<p>Here, rows must be OTUs (ASVs) and the last 7 columns must be taxonomic lineage.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>amp_asv_tax <span class="ot">&lt;-</span> <span class="fu">left_join</span>(amp_asv, amp_tax, <span class="at">by =</span> <span class="st">&quot;OTU&quot;</span>)</span></code></pre></div>
</div>
<h3 id="create-the-ampvis2-object-1">Create the ampvis2 object</h3>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>amp_data <span class="ot">&lt;-</span> <span class="fu">amp_load</span>(amp_asv_tax, <span class="at">metadata =</span> amp_meta_1, <span class="at">tree =</span> ps_work_tree)</span></code></pre></div>
</div>
<p>You can now access the different parts of the ampvis2 object using <code>amp_data$abund</code> for the ASV table, <code>amp_data$tax</code> for the taxonomy table, and <code>amp_data$metadata</code> for the sample data.</p>
<h2 id="review-1">Review</h2>
<p>We have a few different options to play with. At this point it is difficult to say which we will use or whether additional objects need to be created. Here is a summary of the objects we have.</p>
<p>The phyloseq object <em>before</em> removing the prefix of taxon ranks. This still has low coverage samples.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>ps_o</span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3357 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3357 taxa by 8 taxonomic ranks ]</code></pre>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_o, <span class="st">&quot;Family&quot;</span>), <span class="dv">16</span>)</span></code></pre></div>
<pre><code> [1] &quot;f__Hygrophoraceae&quot;                 
 [2] &quot;f__Mycosphaerellaceae&quot;             
 [3] NA                                  
 [4] &quot;f__Trichosporonaceae&quot;              
 [5] &quot;f__Archaeorhizomycetaceae&quot;         
 [6] &quot;f__Chaetomellaceae&quot;                
 [7] &quot;f__Geastraceae&quot;                    
 [8] &quot;f__Metschnikowiaceae&quot;              
 [9] &quot;f__Clavariaceae&quot;                   
[10] &quot;f__Ascobolaceae&quot;                   
[11] &quot;f__Trichocomaceae&quot;                 
[12] &quot;f__Piptocephalidaceae&quot;             
[13] &quot;f__Mortierellaceae&quot;                
[14] &quot;f__Pleosporales_fam_Incertae_sedis&quot;
[15] &quot;f__Glomeraceae&quot;                    
[16] &quot;f__Ceratobasidiaceae&quot;              </code></pre>
</div>
<p>The phyloseq object <em>before</em> changing NA ranks.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>ps_work_o</span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3355 taxa and 13 samples ]
sample_data() Sample Data:       [ 13 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3355 taxa by 8 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 3355 tips and 3354 internal nodes ]</code></pre>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_work_o, <span class="st">&quot;Family&quot;</span>), <span class="dv">16</span>)</span></code></pre></div>
<pre><code> [1] &quot;Crepidotaceae&quot;                  &quot;Agaricaceae&quot;                   
 [3] &quot;Metschnikowiaceae&quot;              &quot;Sympoventuriaceae&quot;             
 [5] &quot;Hypocreales_fam_Incertae_sedis&quot; NA                              
 [7] &quot;Clavicipitaceae&quot;                &quot;Ganodermataceae&quot;               
 [9] &quot;Nectriaceae&quot;                    &quot;Herpotrichiellaceae&quot;           
[11] &quot;Tricholomataceae&quot;               &quot;Archaeorhizomycetaceae&quot;        
[13] &quot;Thelephoraceae&quot;                 &quot;Ceratobasidiaceae&quot;             
[15] &quot;Xylariaceae&quot;                    &quot;Diaporthaceae&quot;                 </code></pre>
</div>
<p>The phyloseq object <em>after</em> changing NA ranks.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>ps_work</span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3355 taxa and 13 samples ]
sample_data() Sample Data:       [ 13 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3355 taxa by 8 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 3355 tips and 3354 internal nodes ]</code></pre>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">get_taxa_unique</span>(ps_work, <span class="st">&quot;Family&quot;</span>), <span class="dv">16</span>)</span></code></pre></div>
<pre><code> [1] &quot;Gigasporaceae&quot;        &quot;c_Paraglomeromycetes&quot;
 [3] &quot;c_Agaricomycetes&quot;     &quot;p_Ascomycota&quot;        
 [5] &quot;c_Sordariomycetes&quot;    &quot;Minutisphaeraceae&quot;   
 [7] &quot;Carcinomycetaceae&quot;    &quot;k_Fungi&quot;             
 [9] &quot;Hydnodontaceae&quot;       &quot;c_Dothideomycetes&quot;   
[11] &quot;Claroideoglomeraceae&quot; &quot;Geastraceae&quot;         
[13] &quot;Teratosphaeriaceae&quot;   &quot;Metschnikowiaceae&quot;   
[15] &quot;Tricholomataceae&quot;     &quot;o_Agaricales&quot;        </code></pre>
</div>
<p>The taxmap object.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(obj)</span></code></pre></div>
<pre><code>&lt;Taxmap&gt;
  4277 taxa: aab. Fungi ... gin. ASV440
  4277 edges: NA-&gt;aab, aab-&gt;aac ... aqx-&gt;gim, bgj-&gt;gin
  1 data sets:
    asv_counts:
      # A tibble: 3,355 x 22
        taxon_id asv_id  P01_D00_010_W3A P01_D00_010_W8A
        &lt;chr&gt;    &lt;chr&gt;             &lt;dbl&gt;           &lt;dbl&gt;
      1 bjn      ASV2001               0               0
      2 bjo      ASV142                0               0
      3 bjp      ASV5                  1               0
      # â¦ with 3,352 more rows, and 18 more variables:
      #   P02_D00_010_C0A &lt;dbl&gt;, P03_D00_010_W3B &lt;dbl&gt;,
      #   P03_D00_010_W8B &lt;dbl&gt;, P04_D00_010_C0B &lt;dbl&gt;,
      #   P05_D00_010_W8C &lt;dbl&gt;, P06_D00_010_C0C &lt;dbl&gt;,
      #   P07_D00_010_W3D &lt;dbl&gt;, P07_D00_010_W8D &lt;dbl&gt;,
      #   P08_D00_010_C0D &lt;dbl&gt;, P09_D00_010_W3E &lt;dbl&gt;, â¦
  0 functions:</code></pre>
</div>
<p>The ampvis2 object.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a>amp_data</span></code></pre></div>
<pre><code>ampvis2 object with 4 elements. 
Summary of OTU table:
     Samples         OTUs  Total#Reads    Min#Reads    Max#Reads 
          13         3355       490767         9172        64636 
Median#Reads    Avg#Reads 
       38371     37751.31 

Assigned taxonomy:
   Kingdom     Phylum      Class      Order     Family      Genus 
3355(100%) 3355(100%) 3355(100%) 3355(100%) 3355(100%) 3355(100%) 
   Species 
3355(100%) 

Metadata variables: 9 
 sample_id, PLOT, DEPTH, TREAT, TEMP, PAIR, total_reads, total_asvs, coverage</code></pre>
</div>
<h2 id="final-steps-1">Final Steps</h2>
<p>The last thing tasks to complete are to <strong>a</strong>) save copies of the taxonomy and sequence tables and <strong>b</strong>) save an image of the workflow for the next part of the workflow.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">tax_table</span>(ps_work),</span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;files/data-prep/tables/its18_work_tax_table.txt&quot;</span>,</span>
<span id="cb107-3"><a href="#cb107-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span>
<span id="cb107-4"><a href="#cb107-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ps_work)),</span>
<span id="cb107-5"><a href="#cb107-5" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;files/data-prep/tables/its18_work_seq_table.txt&quot;</span>,</span>
<span id="cb107-6"><a href="#cb107-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Thatâs all for this part!</p>
<h2 class="appendix" id="source-code">Source Code</h2>
<p>You can find the source code for this page by <a href="https://github.com/sweltr/high-temp/blob/master/data-prep.Rmd">clicking this link</a>.</p>
<h2 class="appendix" id="data-availability">Data Availability</h2>
<p>Raw fastq files available on figshare at XXXXXXXX. Trimmed fastq files (primers removed) available through the ENA under project accession number <a href="">XXXXXXXX</a>. Output files from this workflow available on figshare at <a href="">XXXXXXXX.</a>.</p>
<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-kasper2018ampvis2" class="csl-entry" role="doc-biblioentry">
Andersen, Kasper S., Rasmus H. Kirkegaard, Soren M. Karst, and Mads Albertsen. 2018. <span>âAmpvis2: An r Package to Analyse and Visualise 16s rRNA Amplicon Data.â</span> <em>bioRxiv</em>. <a href="https://doi.org/10.1101/299537">https://doi.org/10.1101/299537</a>.
</div>
<div id="ref-foster2017metacoder" class="csl-entry" role="doc-biblioentry">
Foster, Zachary SL, Thomas J Sharpton, and Niklaus J GrÃ¼nwald. 2017. <span>âMetacoder: An r Package for Visualization and Manipulation of Community Taxonomic Diversity Data.â</span> <em>PLoS Computational Biology</em> 13 (2): e1005404. <a href="https://doi.org/10.1371/journal.pcbi.1005404">https://doi.org/10.1371/journal.pcbi.1005404</a>.
</div>
<div id="ref-lahti2017microbiome" class="csl-entry" role="doc-biblioentry">
Lahti, Leo, Shetty Sudarshan, and others. 2017. <span>âTools for Microbiome Analysis in r. Version 1.9.97.â</span> <a href="https://github.com/microbiome/microbiome">https://github.com/microbiome/microbiome</a>.
</div>
<div id="ref-mcmurdie2013phyloseq" class="csl-entry" role="doc-biblioentry">
McMurdie, Paul J, and Susan Holmes. 2013. <span>âPhyloseq: An r Package for Reproducible Interactive Analysis and Graphics of Microbiome Census Data.â</span> <em>PLoS One</em> 8 (4): e61217. <a href="https://doi.org/10.1371/journal.pone.0061217">https://doi.org/10.1371/journal.pone.0061217</a>.
</div>
</div>
<section class="footnotes" role="doc-endnotes">
<hr />
<ol>
<li id="fn1" role="doc-endnote"><p>it can also contain a <code>phy_tree</code> table but we will not use that here<a href="#fnref1" class="footnote-back" role="doc-backlink">â©ï¸</a></p></li>
</ol>
</section>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/sweltr/high-temp/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/sweltr/high-temp/">https://github.com/sweltr/high-temp/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<div class="distill-site-nav distill-site-footer">
<p>Â© Copyright 2021 The SWELTR Project.</p>
<p>Site constructed using <a href="https://rstudio.github.io/distill/">Distill for R Markdown</a>.</p>
</div>
<!--/radix_placeholder_navigation_after_body-->


</body>

</html>
