<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-0.9.598">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="description" content="On this page, you can find information on obtaining sequence data, data products, and processing scripts. Code is embedded in indivudual workflows on the website or can be accessed through the project GitHub repo.">
<title>SWELTR - Data Availability</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./som.html" rel="next">
<link href="./metadata.html" rel="prev">
<link href="./assets/favicon.ico" rel="icon">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script><style>
    .quarto-title-block .quarto-title-banner {
      background: #113858;
    }
    </style>
<link rel="stylesheet" href="assets/css/styles.css">
</head>
<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg navbar-dark "><div class="navbar-container container-fluid">
      <a class="navbar-brand" href="./index.html">
    <img src="./assets/icon.png" alt=""></a>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item compact">
    <a class="nav-link" href="./index.html"><i class="bi bi-home" role="img">
</i> 
 </a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-a.-processing" role="button" data-bs-toggle="dropdown" aria-expanded="false">A. Processing</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-a.-processing">
<li class="dropdown-header">Workflows</li>
        <li><hr class="dropdown-divider"></li>
        <li>
    <a class="dropdown-item" href="./dada2.html">
 <span class="dropdown-text">1. DADA2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./data-prep.html">
 <span class="dropdown-text">2. Data Set Prep</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./filtering.html">
 <span class="dropdown-text">3. Filtering</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-b.-diversity" role="button" data-bs-toggle="dropdown" aria-expanded="false">B. Diversity</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-b.-diversity">
<li class="dropdown-header">Workflows</li>
        <li><hr class="dropdown-divider"></li>
        <li>
    <a class="dropdown-item" href="./taxa.html">
 <span class="dropdown-text">4. Taxonomic</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./alpha.html">
 <span class="dropdown-text">5. Alpha Diversity</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./beta.html">
 <span class="dropdown-text">6. Beta Diversity</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./da.html">
 <span class="dropdown-text">7. DA ASVs &amp; Taxa</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./metadata.html">
 <span class="dropdown-text">8. Metadata</span></a>
  </li>  
    </ul>
</li>
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item">
    <a class="nav-link active" href="./data-availability.html" aria-current="page">DATA</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./pub.html">PAPER</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./som.html">SOM</a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/sweltr/high-temp/"><i class="bi bi-github" role="img" aria-label="Agua Salud GitHub">
</i> 
 </a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="./about.html"><i class="bi bi-question" role="img">
</i> 
 </a>
  </li>  
</ul>
<div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }"><div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">Data Availability</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title d-none d-lg-block">Data Availability</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i></button></div></div>
                  <div>
        <div class="description">
          <p>On this page, you can find information on obtaining sequence data, data products, and processing scripts. Code is embedded in indivudual workflows on the website or can be accessed through the project GitHub repo.</p>
        </div>
      </div>
                </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto"><div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">A. Processing</a>
      <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./dada2.html" class="sidebar-item-text sidebar-link">1. DADA2 Workflow</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./data-prep.html" class="sidebar-item-text sidebar-link">2. Data Set Preparation</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./filtering.html" class="sidebar-item-text sidebar-link">3. Filtering</a>
  </div>
</li>
    </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">B. Diversity</a>
      <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./taxa.html" class="sidebar-item-text sidebar-link">4. Taxonomic Diversity</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./alpha.html" class="sidebar-item-text sidebar-link">5. Alpha Diversity Estimates</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./beta.html" class="sidebar-item-text sidebar-link">6. Beta Diversity &amp; Dispersion Estimates</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./da.html" class="sidebar-item-text sidebar-link">7. Differentially Abundant ASVs &amp; Taxa</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./metadata.html" class="sidebar-item-text sidebar-link">8. Metadata Analysis</a>
  </div>
</li>
    </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">More</a>
      <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./data-availability.html" class="sidebar-item-text sidebar-link active">Data Availability</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./som.html" class="sidebar-item-text sidebar-link">Electronic Supplementary Material</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./about.html" class="sidebar-item-text sidebar-link">Website Contruction</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pub.html" class="sidebar-item-text sidebar-link">Code for Publication Figures</a>
  </div>
</li>
    </ul>
</li>
    </ul>
</div>
</nav><!-- margin-sidebar --><div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#archived-sequence-data" id="toc-archived-sequence-data" class="nav-link active" data-scroll-target="#archived-sequence-data">Archived Sequence Data</a></li>
  <li><a href="#pipeline-data" id="toc-pipeline-data" class="nav-link" data-scroll-target="#pipeline-data">Pipeline Data</a></li>
  <li><a href="#submitting-sequence-data" id="toc-submitting-sequence-data" class="nav-link" data-scroll-target="#submitting-sequence-data">Submitting Sequence Data</a></li>
  </ul><div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/sweltr/high-temp/edit/main/data-availability.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/sweltr/high-temp/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content"><section id="archived-sequence-data" class="level2"><h2 class="anchored" data-anchor-id="archived-sequence-data">Archived Sequence Data</h2>
<p>Raw <strong>16S rRNA</strong> data for each sample (before removing primers) is available on figshare at <a href="https://doi.org/10.25573/data.14686665">doi:10.25573/data.14686665</a>. Raw <strong>ITS</strong> data for each sample (before removing primers) is available on figshare at <a href="https://doi.org/10.25573/data.14686755">doi:10.25573/data.14686755</a>.</p>
<p>All trimmed sequence data (primers removed) is deposited at the European Nucleotide Archive (ENA) under the study accession number <a href="https://www.ebi.ac.uk/ena/browser/view/PRJEB45074">PRJEB45074 (ERP129199)</a>. The trimmed <strong>16S rRNA</strong> data (primers removed) are deposited under sample accession numbers <a href="https://www.ebi.ac.uk/ena/browser/view/ERS6485270-ERS6485284">ERS6485270-ERS6485284</a>. The trimmed <strong>ITS</strong> data (primers removed) are deposited under sample accession numbers <a href="https://www.ebi.ac.uk/ena/browser/view/ERS6485285-ERS6485299">ERS6485285-ERS6485299</a>.</p>
</section><section id="pipeline-data" class="level2"><h2 class="anchored" data-anchor-id="pipeline-data">Pipeline Data</h2>
<p>Data for each individual pipeline are available through the Smithsonian figshare under a single collection at <a href="https://doi.org/10.25573/data.c.5667571">doi:10.25573/data.c.5667571</a>. In addition, data from each pipeline are available for download from figshare using the links at the bottom of each workflow page (where applicable).</p>
<p></p>
<div class="callout-note callout callout-style-default no-icon callout-captioned">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-1-contents" aria-controls="callout-1" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
<strong>Show</strong> descriptions and DOI’s for each workflow
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-1" class="callout-1-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Scott, Jarrod (2022): (high-temp/16s) No 0. Raw 16S rRNA fastq data files for the high-temp SWELTR study. Smithsonian Tropical Research Institute. Dataset. <a href="https://doi.org/10.25573/data.14686665.v1">https://doi.org/10.25573/data.14686665.v1</a></p>
<p>Scott, Jarrod (2022): (high-temp/ITS) No 0. Raw ITS fastq data files for the high-temp SWELTR study. Smithsonian Tropical Research Institute. Dataset. <a href="https://doi.org/10.25573/data.14686755.v1">https://doi.org/10.25573/data.14686755.v1</a></p>
<p>Scott, Jarrod (2022): (high-temp) No 1. <a href="https://sweltr.github.io/high-temp/dada2.html">DADA2 Workflow (16S rRNA/ITS)</a> Output. Smithsonian Tropical Research Institute. Dataset. <a href="https://doi.org/10.25573/data.14687184.v1">https://doi.org/10.25573/data.14687184.v1</a></p>
<p>Scott, Jarrod (2022): (high-temp) No 2. <a href="https://sweltr.github.io/high-temp/data-prep.html">Data Preparation</a> (16S rRNA/ITS) Output. Smithsonian Tropical Research Institute. Dataset. <a href="https://doi.org/10.25573/data.14690739.v1">https://doi.org/10.25573/data.14690739.v1</a></p>
<p>Scott, Jarrod (2022): (high-temp) No 3. <a href="https://sweltr.github.io/high-temp/filtering.html">Filtering</a>: (16S rRNA/ITS) Output. Smithsonian Tropical Research Institute. Dataset. <a href="https://doi.org/10.25573/data.14701440.v1">https://doi.org/10.25573/data.14701440.v1</a></p>
<p>Scott, Jarrod (2022): (high-temp) No 4. <a href="https://sweltr.github.io/high-temp/taxa.html">Taxonomic diversity</a>: (16S rRNA/ITS) Output. Smithsonian Tropical Research Institute. Dataset. <a href="https://doi.org/10.25573/data.16826737.v1">https://doi.org/10.25573/data.16826737.v1</a></p>
<p>Scott, Jarrod (2022): (high-temp) No 5. <a href="https://sweltr.github.io/high-temp/alpha.html">Aplha diversity</a> (16S rRNA/ITS) Output. Smithsonian Tropical Research Institute. Dataset. <a href="https://doi.org/10.25573/data.16826779.v1">https://doi.org/10.25573/data.16826779.v1</a></p>
<p>Scott, Jarrod (2022): (high-temp) No 6. <a href="https://sweltr.github.io/high-temp/beta.html">Beta diversity</a> (16S rRNA/ITS) Output. Smithsonian Tropical Research Institute. Dataset. <a href="https://doi.org/10.25573/data.16828063.v1">https://doi.org/10.25573/data.16828063.v1</a></p>
<p>Scott, Jarrod (2022): (high-temp) No 7. <a href="https://sweltr.github.io/high-temp/da.html">Differentially Abundant ASVs &amp; Taxa</a> (16S rRNA/ITS) Output. Smithsonian Tropical Research Institute. Dataset. <a href="https://doi.org/10.25573/data.16828249.v1">https://doi.org/10.25573/data.16828249.v1</a></p>
<p>Scott, Jarrod (2022): (high-temp) No 8. <a href="https://sweltr.github.io/high-temp/metadata.html">Metadata Analysis</a> (16S rRNA/ITS) Output. Smithsonian Tropical Research Institute. Dataset. <a href="https://doi.org/10.25573/data.16828294.v1">https://doi.org/10.25573/data.16828294.v1</a></p>
</div>
</div>
</div>
</section><section id="submitting-sequence-data" class="level2"><h2 class="anchored" data-anchor-id="submitting-sequence-data">Submitting Sequence Data</h2>
<p>We submitted out data to the <a href="https://www.ebi.ac.uk/ena">European Nucleotide Archive (ENA)</a>. The ENA does not like RAW data and prefers to have primers removed. So we submitted the trimmed Fastq files to the ENA. You can find these data under the study accession number <strong>PRJEB45074 (ERP129199)</strong>. The RAW files on our figshare site (see above).</p>
<p>To submit to the ENA you need two data tables (plus your sequence data). One file describes the samples and the other file describes the sequencing data.</p>
<p>You can download our submission data tables here:</p>
<ul>
<li><a href="include/data-availability/ENA_Sample_SUBMISSION_SWELTR-high-temp.tsv">Sample submission form</a></li>
<li><a href="include/data-availability/ENA_PAIRED_FASTQ_SUBMISSION_SWELTR-high-temp.tsv">FASTq submission form</a></li>
</ul>
<p>Note, these forms are study specific, so please use these as guides only.</p>
<section id="instructions-for-submitting-to-the-ena" class="level3"><h3 class="anchored" data-anchor-id="instructions-for-submitting-to-the-ena">Instructions for Submitting to the ENA</h3>
<p>Even though I have done this dozens of times, the process of submitting sequence data to read archives still baffles me.</p>
<blockquote class="blockquote">
<p>Note: I submit data with primers, barcodes, etc <strong>removed</strong></p>
</blockquote>
<section id="register-project-upload-sample-data" class="level4"><h4 class="anchored" data-anchor-id="register-project-upload-sample-data">Register Project &amp; Upload Sample Data</h4>
<ol type="1">
<li>go to https://www.ebi.ac.uk/ena/submit and select <strong>Submit to ENA Interactively</strong>.</li>
<li>Login or Register.</li>
<li>Go to <strong>New Submission</strong> tab and, if this is a new project, select <strong>Register study (project)</strong>.</li>
<li>Hit Next</li>
<li>Enter details and hit <strong>Submit</strong>.</li>
<li>Next, <a href="https://www.ebi.ac.uk/ena/submit/checklists">Select Checklist</a>. This will be specific to the type of samples you have and basically will create a template so you can add your sample metadata. For this study I chose <strong>GSC MIxS soil</strong>, checklist accession number <a href="https://www.ebi.ac.uk/ena/data/view/ERC000022">ERC000022</a>
</li>
<li>Next</li>
<li>Now go through and select/deselect fields as needed. Note, some fields are mandatory.</li>
<li>Once finished, hit <strong>Next</strong> to fill in any details that apply to <em>All</em> samples and the download the template. Alternatively, you can download the template and fill in the data by hand.</li>
<li>Upload the sample sheet.</li>
<li>Once everything looks good and uploaded, click the <strong>New Submission</strong> tab.</li>
</ol></section><section id="upload-sequence-data" class="level4"><h4 class="anchored" data-anchor-id="upload-sequence-data">Upload Sequence Data</h4>
<ol start="12" type="1">
<li>Hit <strong>Skip</strong> and then select <strong>Two Fastq files (Paired)</strong>, Download the template and fill in the details.</li>
<li>Next, make sure all the trimmed fastq files are gzipped <strong>.gz</strong> (these are what you submit).</li>
<li>Navigate to the directory with the trimmed, compressed fastq files and run:</li>
</ol>
<pre><code>md5sum *.gz</code></pre>
<ol start="16" type="1">
<li>Add the checksums and file names to the fastq submission form. You can read more about <a href="https://ena-docs.readthedocs.io/en/latest/submit/fileprep/preparation.html">Preparing A File For Upload here</a>.</li>
</ol>
<p><strong>STOP</strong></p>
<p>Before uploading the sheet, you first must upload the fastq files. Documentation for Uploading Files To ENA can be found <a href="https://ena-docs.readthedocs.io/en/latest/submit/fileprep/upload.html">here</a>.</p>
<ol start="17" type="1">
<li>In the same directory, run:</li>
</ol>
<pre><code>lftp webin2.ebi.ac.uk -u Webin-XXXXX</code></pre>
<p>where <code>Webin-XXXXX</code> is your user name. Enter your password</p>
<p>at the prompt run:</p>
<pre><code>mput *.gz
# when finished
bye</code></pre>
<p>The files should begin uploading. Depending on internet speed and/or file sizes/numbers, this could take a while.</p>
<ol start="18" type="1">
<li>Once the upload is finished, upload and submit the fastq submission form. If the <code>sample alias</code> field dows not autofill you may need to upload the sample form again.</li>
</ol>
<div class="cell">

</div>
</section></section></section><div id="quarto-appendix" class="default"><section id="source-code" class="level4 appendix"><h2 class="quarto-appendix-heading">Source Code</h2><div class="quarto-appendix-contents">
<p>The source code for this page can be accessed on GitHub <svg aria-hidden="true" role="img" viewbox="0 0 496 512" style="height:1em;width:0.97em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:currentColor;overflow:visible;position:relative;"><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"></path></svg> by <a href="https://github.com/sweltr/high-temp/blob/main/data-availability.qmd">clicking this link</a>.</p>
</div></section><section id="last-updated-on" class="level4 appendix"><h2 class="quarto-appendix-heading">Last updated on</h2><div class="quarto-appendix-contents">
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "2022-06-29 07:31:27 EST"</code></pre>
</div>
</div>


<!-- -->

</div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./metadata.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">8. Metadata Analysis</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./som.html" class="pagination-link">
        <span class="nav-page-text">Electronic Supplementary Material</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb5" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Data Availability"</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="an">description:</span><span class="co"> |</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co">  On this page, you can find information on obtaining sequence data, data products, and processing scripts. Code is  embedded in indivudual workflows on the website or can be accessed through the project GitHub repo.</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="fu">## Archived Sequence Data</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>Raw **16S rRNA** data for each sample (before removing primers) is available on figshare at [doi:10.25573/data.14686665](https://doi.org/10.25573/data.14686665). Raw **ITS** data for each sample (before removing primers) is available on figshare at <span class="co">[</span><span class="ot">doi:10.25573/data.14686755</span><span class="co">](https://doi.org/10.25573/data.14686755)</span>. </span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>All trimmed sequence data (primers removed) is deposited at the European Nucleotide Archive (ENA) under the study accession number <span class="co">[</span><span class="ot">PRJEB45074 (ERP129199)</span><span class="co">](https://www.ebi.ac.uk/ena/browser/view/PRJEB45074)</span>. The trimmed **16S rRNA** data (primers removed) are deposited under sample accession numbers [ERS6485270-ERS6485284](https://www.ebi.ac.uk/ena/browser/view/ERS6485270-ERS6485284). The trimmed **ITS** data (primers removed) are deposited under sample accession numbers <span class="co">[</span><span class="ot">ERS6485285-ERS6485299</span><span class="co">](https://www.ebi.ac.uk/ena/browser/view/ERS6485285-ERS6485299)</span>.</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="fu">## Pipeline Data</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>Data for each individual pipeline are available through the Smithsonian figshare under a single collection at <span class="co">[</span><span class="ot">doi:10.25573/data.c.5667571</span><span class="co">](https://doi.org/10.25573/data.c.5667571)</span>. In addition, data from each pipeline are available for download from figshare using the links at the bottom of each workflow page (where applicable).</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>:::{.callout-note collapse="true" icon=false}</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a><span class="fu">## **Show** descriptions and DOI's for each workflow</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>Scott, Jarrod (2022): (high-temp/16s) No 0. Raw 16S rRNA fastq data files for the high-temp SWELTR study. Smithsonian Tropical Research Institute. Dataset. <span class="co">[</span><span class="ot">https://doi.org/10.25573/data.14686665.v1</span><span class="co">](https://doi.org/10.25573/data.14686665.v1)</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>Scott, Jarrod (2022): (high-temp/ITS) No 0. Raw ITS fastq data files for the high-temp SWELTR study. Smithsonian Tropical Research Institute. Dataset. <span class="co">[</span><span class="ot">https://doi.org/10.25573/data.14686755.v1</span><span class="co">](https://doi.org/10.25573/data.14686755.v1)</span></span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a>Scott, Jarrod (2022): (high-temp) No 1. <span class="co">[</span><span class="ot">DADA2 Workflow (16S rRNA/ITS)</span><span class="co">](https://sweltr.github.io/high-temp/dada2.html)</span> Output. Smithsonian Tropical Research Institute. Dataset. <span class="co">[</span><span class="ot">https://doi.org/10.25573/data.14687184.v1</span><span class="co">](https://doi.org/10.25573/data.14687184.v1)</span></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>Scott, Jarrod (2022): (high-temp) No 2. <span class="co">[</span><span class="ot">Data Preparation</span><span class="co">](https://sweltr.github.io/high-temp/data-prep.html)</span> (16S rRNA/ITS) Output. Smithsonian Tropical Research Institute. Dataset. <span class="co">[</span><span class="ot">https://doi.org/10.25573/data.14690739.v1</span><span class="co">](https://doi.org/10.25573/data.14690739.v1)</span></span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a>Scott, Jarrod (2022): (high-temp) No 3. <span class="co">[</span><span class="ot">Filtering</span><span class="co">](https://sweltr.github.io/high-temp/filtering.html)</span>: (16S rRNA/ITS) Output. Smithsonian Tropical Research Institute. Dataset. <span class="co">[</span><span class="ot">https://doi.org/10.25573/data.14701440.v1</span><span class="co">](https://doi.org/10.25573/data.14701440.v1)</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>Scott, Jarrod (2022): (high-temp) No 4. <span class="co">[</span><span class="ot">Taxonomic diversity</span><span class="co">](https://sweltr.github.io/high-temp/taxa.html)</span>: (16S rRNA/ITS) Output. Smithsonian Tropical Research Institute. Dataset. <span class="co">[</span><span class="ot">https://doi.org/10.25573/data.16826737.v1</span><span class="co">](https://doi.org/10.25573/data.16826737.v1)</span></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a>Scott, Jarrod (2022): (high-temp) No 5. <span class="co">[</span><span class="ot">Aplha diversity</span><span class="co">](https://sweltr.github.io/high-temp/alpha.html)</span> (16S rRNA/ITS) Output. Smithsonian Tropical Research Institute. Dataset. <span class="co">[</span><span class="ot">https://doi.org/10.25573/data.16826779.v1</span><span class="co">](https://doi.org/10.25573/data.16826779.v1)</span></span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a>Scott, Jarrod (2022): (high-temp) No 6. <span class="co">[</span><span class="ot">Beta diversity</span><span class="co">](https://sweltr.github.io/high-temp/beta.html)</span> (16S rRNA/ITS) Output. Smithsonian Tropical Research Institute. Dataset. <span class="co">[</span><span class="ot">https://doi.org/10.25573/data.16828063.v1</span><span class="co">](https://doi.org/10.25573/data.16828063.v1)</span></span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a>Scott, Jarrod (2022): (high-temp) No 7. <span class="co">[</span><span class="ot">Differentially Abundant ASVs &amp; Taxa</span><span class="co">](https://sweltr.github.io/high-temp/da.html)</span> (16S rRNA/ITS) Output. Smithsonian Tropical Research Institute. Dataset. <span class="co">[</span><span class="ot">https://doi.org/10.25573/data.16828249.v1</span><span class="co">](https://doi.org/10.25573/data.16828249.v1)</span></span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a>Scott, Jarrod (2022): (high-temp) No 8. <span class="co">[</span><span class="ot">Metadata Analysis</span><span class="co">](https://sweltr.github.io/high-temp/metadata.html)</span> (16S rRNA/ITS) Output. Smithsonian Tropical Research Institute. Dataset. <span class="co">[</span><span class="ot">https://doi.org/10.25573/data.16828294.v1</span><span class="co">](https://doi.org/10.25573/data.16828294.v1)</span></span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a><span class="fu">## Submitting Sequence Data</span></span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a>We submitted out data to the <span class="co">[</span><span class="ot">European Nucleotide Archive (ENA)</span><span class="co">](https://www.ebi.ac.uk/ena)</span>. The ENA does not like RAW data and prefers to have primers removed. So we submitted the trimmed Fastq files to the ENA. You can find these data under the study accession number **PRJEB45074 (ERP129199)**. The RAW files on our figshare site (see above).</span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a>To submit to the ENA you need two data tables (plus your sequence data). One file describes the samples and the other file describes the sequencing data.</span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a>You can download our submission data tables here:</span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="co">[</span><span class="ot">Sample submission form</span><span class="co">](include/data-availability/ENA_Sample_SUBMISSION_SWELTR-high-temp.tsv)</span></span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="co">[</span><span class="ot">FASTq submission form</span><span class="co">](include/data-availability/ENA_PAIRED_FASTQ_SUBMISSION_SWELTR-high-temp.tsv)</span></span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a>Note, these forms are study specific, so please use these as guides only.</span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-58"><a href="#cb5-58" aria-hidden="true" tabindex="-1"></a><span class="fu">### Instructions for Submitting to the ENA</span></span>
<span id="cb5-59"><a href="#cb5-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-60"><a href="#cb5-60" aria-hidden="true" tabindex="-1"></a>Even though I have done this dozens of times, the process of submitting sequence data to read archives still baffles me. </span>
<span id="cb5-61"><a href="#cb5-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; Note: I submit data with primers, barcodes, etc **removed**</span></span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-64"><a href="#cb5-64" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Register Project &amp; Upload Sample Data</span></span>
<span id="cb5-65"><a href="#cb5-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-66"><a href="#cb5-66" aria-hidden="true" tabindex="-1"></a>1) go to https://www.ebi.ac.uk/ena/submit and select **Submit to ENA Interactively**.</span>
<span id="cb5-67"><a href="#cb5-67" aria-hidden="true" tabindex="-1"></a>2) Login or Register.</span>
<span id="cb5-68"><a href="#cb5-68" aria-hidden="true" tabindex="-1"></a>3) Go to **New Submission** tab and, if this is a new project, select **Register study (project)**.</span>
<span id="cb5-69"><a href="#cb5-69" aria-hidden="true" tabindex="-1"></a>4) Hit Next</span>
<span id="cb5-70"><a href="#cb5-70" aria-hidden="true" tabindex="-1"></a>5) Enter details and hit **Submit**.</span>
<span id="cb5-71"><a href="#cb5-71" aria-hidden="true" tabindex="-1"></a>6) Next, <span class="co">[</span><span class="ot">Select Checklist</span><span class="co">](https://www.ebi.ac.uk/ena/submit/checklists)</span>. This will be specific to the type of samples you have and basically will create a template so you can add your sample metadata. For this study I chose **GSC MIxS soil**, checklist accession number <span class="co">[</span><span class="ot">ERC000022</span><span class="co">](https://www.ebi.ac.uk/ena/data/view/ERC000022)</span></span>
<span id="cb5-72"><a href="#cb5-72" aria-hidden="true" tabindex="-1"></a>7) Next</span>
<span id="cb5-73"><a href="#cb5-73" aria-hidden="true" tabindex="-1"></a>8) Now go through and select/deselect fields as needed. Note, some fields are mandatory.</span>
<span id="cb5-74"><a href="#cb5-74" aria-hidden="true" tabindex="-1"></a>9) Once finished, hit **Next** to fill in any details that apply to *All* samples and the download the template. Alternatively, you can download the template and fill in the data by hand.</span>
<span id="cb5-75"><a href="#cb5-75" aria-hidden="true" tabindex="-1"></a>10) Upload the sample sheet. </span>
<span id="cb5-76"><a href="#cb5-76" aria-hidden="true" tabindex="-1"></a>11) Once everything looks good and uploaded, click the **New Submission** tab.</span>
<span id="cb5-77"><a href="#cb5-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-78"><a href="#cb5-78" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Upload Sequence Data</span></span>
<span id="cb5-79"><a href="#cb5-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-80"><a href="#cb5-80" aria-hidden="true" tabindex="-1"></a>12) Hit **Skip** and then select **Two Fastq files (Paired)**, Download the template and fill in the details.</span>
<span id="cb5-81"><a href="#cb5-81" aria-hidden="true" tabindex="-1"></a>13) Next, make sure all the trimmed fastq files are gzipped **.gz**  (these are what you submit). </span>
<span id="cb5-82"><a href="#cb5-82" aria-hidden="true" tabindex="-1"></a>14) Navigate to the directory with  the trimmed, compressed fastq files and run:</span>
<span id="cb5-83"><a href="#cb5-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-84"><a href="#cb5-84" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb5-85"><a href="#cb5-85" aria-hidden="true" tabindex="-1"></a><span class="in">md5sum *.gz</span></span>
<span id="cb5-86"><a href="#cb5-86" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb5-87"><a href="#cb5-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-88"><a href="#cb5-88" aria-hidden="true" tabindex="-1"></a>16) Add the checksums and file names to the fastq submission form. You can read more about <span class="co">[</span><span class="ot">Preparing A File For Upload here</span><span class="co">](https://ena-docs.readthedocs.io/en/latest/submit/fileprep/preparation.html)</span>.</span>
<span id="cb5-89"><a href="#cb5-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-90"><a href="#cb5-90" aria-hidden="true" tabindex="-1"></a>**STOP**</span>
<span id="cb5-91"><a href="#cb5-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-92"><a href="#cb5-92" aria-hidden="true" tabindex="-1"></a>Before uploading the sheet, you first must upload the fastq files. Documentation for Uploading Files To ENA can be found <span class="co">[</span><span class="ot">here</span><span class="co">](https://ena-docs.readthedocs.io/en/latest/submit/fileprep/upload.html)</span>. </span>
<span id="cb5-93"><a href="#cb5-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-94"><a href="#cb5-94" aria-hidden="true" tabindex="-1"></a>17) In the same directory, run:</span>
<span id="cb5-95"><a href="#cb5-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-96"><a href="#cb5-96" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb5-97"><a href="#cb5-97" aria-hidden="true" tabindex="-1"></a><span class="in">lftp webin2.ebi.ac.uk -u Webin-XXXXX</span></span>
<span id="cb5-98"><a href="#cb5-98" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb5-99"><a href="#cb5-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-100"><a href="#cb5-100" aria-hidden="true" tabindex="-1"></a>where <span class="in">`Webin-XXXXX`</span> is your user name. Enter your password</span>
<span id="cb5-101"><a href="#cb5-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-102"><a href="#cb5-102" aria-hidden="true" tabindex="-1"></a>at the prompt run:</span>
<span id="cb5-103"><a href="#cb5-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-104"><a href="#cb5-104" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb5-105"><a href="#cb5-105" aria-hidden="true" tabindex="-1"></a><span class="in">mput *.gz</span></span>
<span id="cb5-106"><a href="#cb5-106" aria-hidden="true" tabindex="-1"></a><span class="in"># when finished</span></span>
<span id="cb5-107"><a href="#cb5-107" aria-hidden="true" tabindex="-1"></a><span class="in">bye</span></span>
<span id="cb5-108"><a href="#cb5-108" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb5-109"><a href="#cb5-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-110"><a href="#cb5-110" aria-hidden="true" tabindex="-1"></a>The files should begin uploading. Depending on internet speed and/or file sizes/numbers, this could take a while. </span>
<span id="cb5-111"><a href="#cb5-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-112"><a href="#cb5-112" aria-hidden="true" tabindex="-1"></a>18) Once the upload is finished, upload and submit the fastq submission form. If the <span class="in">`sample alias`</span> field dows not autofill you may need to upload the sample form again. </span>
<span id="cb5-113"><a href="#cb5-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-114"><a href="#cb5-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-117"><a href="#cb5-117" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb5-118"><a href="#cb5-118" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false </span></span>
<span id="cb5-119"><a href="#cb5-119" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb5-120"><a href="#cb5-120" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb5-121"><a href="#cb5-121" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb5-122"><a href="#cb5-122" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb5-123"><a href="#cb5-123" aria-hidden="true" tabindex="-1"></a><span class="do">### COmmon formatting scripts</span></span>
<span id="cb5-124"><a href="#cb5-124" aria-hidden="true" tabindex="-1"></a><span class="do">### </span><span class="al">NOTE</span><span class="do">: captioner.R must be read BEFORE captions_XYZ.R</span></span>
<span id="cb5-125"><a href="#cb5-125" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">"assets"</span>, <span class="st">"functions.R"</span>))</span>
<span id="cb5-126"><a href="#cb5-126" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb5-127"><a href="#cb5-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-128"><a href="#cb5-128" aria-hidden="true" tabindex="-1"></a><span class="fu">####  Source Code {.appendix}</span></span>
<span id="cb5-129"><a href="#cb5-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-130"><a href="#cb5-130" aria-hidden="true" tabindex="-1"></a>The source code for this page can be accessed on GitHub <span class="in">`r fa(name = "github")`</span> by <span class="co">[</span><span class="ot">clicking this link</span><span class="co">]</span>(<span class="in">`r source_code()`</span>). </span>
<span id="cb5-131"><a href="#cb5-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-132"><a href="#cb5-132" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Last updated on {.appendix}</span></span>
<span id="cb5-133"><a href="#cb5-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-134"><a href="#cb5-134" aria-hidden="true" tabindex="-1"></a><span class="in">```{r,echo=FALSE, eval=TRUE}</span></span>
<span id="cb5-135"><a href="#cb5-135" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb5-136"><a href="#cb5-136" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
      <div class="nav-footer-center">© Copyright 2022 <a href="https://metacrobe.github.io/">metacrobe</a> &amp; The SWELTR Project. Site constructed using <a href="https://quarto.org/">Quarto</a>.</div>
  </div>
</footer>


</body></html>