---
title: "The S.W.E.L.T.R. Microbial Cookbook"
description: |
  **S**oil **W**arming **E**xperiment in **L**owland **T**ropical **R**ainforest
author:
  - name: "Andrew Nottingham"
    affiliation: University of Edinburgh
    affiliation-url: https://www.ed.ac.uk/
  - name: "Jarrod J Scott"
    orcid: 0000-0001-9863-1318
    url: https://metacrobe.github.io/
    affiliation: Smithsonian Tropical Research Institute
    affiliation-url: https://stri.si.edu/
  - name: "Kristin Saltonstall"
    orcid: 0000-0002-1811-4087
    url: https://stri.si.edu/scientist/kristin-saltonstall
    affiliation: Smithsonian Tropical Research Institute
    affiliation-url: https://stri.si.edu/
  - name: "Kirk Broders"
    orcid: 0000-0002-8569-6005
    url: https://www.ars.usda.gov/midwest-area/peoria-il/national-center-for-agricultural-utilization-research/mycotoxin-prevention-and-applied-microbiology-research/people/kirk-broders/
    affiliation: USDA-ARS
  - name: "Maria Montero-Sanchez"
    affiliation: Smithsonian Tropical Research Institute
    affiliation-url: https://stri.si.edu/
  - name: "Johann PÃ¼spÃ¶k"
    affiliation: Smithsonian Tropical Research Institute
    affiliation-url: https://stri.si.edu/
  - name: "Erland BÃ¥Ã¥th"
    orcid: 0000-0002-2616-1342 
    affiliation: Lund University
  - name: "Patrick Meir"
    orcid: 0000-0002-2362-0398 
    url: https://biology.anu.edu.au/research/groups/meir-group-tropical-forest-ecosystems
    affiliation: Australian National University
format:
  html: 
    code-tools: false    
    css: ["assets/css/styles_index.css", "assets/css/styles.css"]
title-block-banner: assets/banner.png
---

```{r}
#| label: setup
#| include: false
knitr::opts_chunk$set(echo = TRUE, eval = FALSE)

pacman::p_load(tidyverse, tm, SnowballC, wordcloud, RColorBrewer, 
               wordcloud2, webshot, htmlwidgets, 
               leaflet, htmlwidgets,
               install = FALSE, update = FALSE)
```


<script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script>
<script async src="https://badge.dimensions.ai/badge.js" charset="utf-8"></script>

<div class="pub_float">
  <div class="__dimensions_badge_embed__" data-doi="10.1038/s41564-022-01200-1" data-hide-zero-citations="false" data-legend="hover-bottom" data-style="small_circle"></div>
  <div class="altmetric-embed" data-badge-type="donut" data-doi="10.1038/s41564-022-01200-1" data-altmetric-id="135469873"></div>
  <span class="pub-title"><a href=" https://doi.org/10.1038/s41564-022-01200-1" target="_new">Microbial diversity declines in warmed tropical soil and respiration rise exceed predictions as communities adapt</a></span>
  <span class="pub-authors">Nottingham AT, Scott JJ, Saltonstall K, Broders K, Montero-Sanchez M, PÃ¼spÃ¶k J, BÃ¥Ã¥th E, Meir P. </span>
<div class="pub-info">
<div class="pub-featured-image">
  <a href="assets/main_figs.png">
    <img src="assets/main_figs.png" style="max-width: 100px; max-height: 400px; width: auto; border: none; height: auto; margin: 0 auto; display: block; transform: translateY(10%);" />
  </a>
</div>
<div class="pub-highlights">
  <span style="display: inline-block; padding-bottom: 5px;"
  >- **Perturbation** of soil microbial communities by **rising temperatures** could have important consequences for biodiversity and future climate.</span><br><span style="display: inline-block; padding-bottom: 5px;"
  >- Particularly in **tropical forests** where high biological diversity coincides with a **vast store of soil carbon**. </span><br><span style="display: inline-block; padding-bottom: 5px;"
  >- We carried out a 2-year *in situ* soil **warming experiment** in a tropical forest in Panama.</span><br><span style="display: inline-block; padding-bottom: 5px;"
  >- Found **large changes** in the soil microbial community and its growth sensitivity.</span><br><span style="display: inline-block; padding-bottom: 5px;"
  >- Microbial **diversity declined** markedly with warming, demonstrating a breakdown in the positive temperature-diversity relationship observed elsewhere.</span><br><span style="display: inline-block; padding-bottom: 5px;"
  >- Microbial **community composition shifted** with warming---many taxa were no longer detected while others enriched, including thermophilic taxa.</span><br><span style="display: inline-block; padding-bottom: 5px;"
  >- Shifts resulted in community adaptation of growth to warmer temperatures, which we used to predict changes in soil CO~2~ emissions.</span><br><span style="display: inline-block; padding-bottom: 5px;"
  >- The *in situ* CO~2~ emissions **exceeded our model predictions** threefold.</span><br><span style="display: inline-block; padding-bottom: 5px;"
  >- Results suggest that warming of tropical forests will have **rapid, detrimental consequences** for soil microbial biodiversity and future climate.</span>
</div>
  </div>
  <span class="pub-journal">
    <b>Nature Microbiology 7 (2022)</b>
    <a href="https://doi.org/10.1038/s41564-022-01200-1" target="_blank">ðŸ”—</a>
  </span>
</div>

<small><em>Code for the highlight box above shamelessly stolen from the [Meren Lab website](https://merenlab.org/).</em></small>


> [Click here](https://www.nature.com/articles/s41564-022-01200-1/metrics) for article metrics on the  Nature Microbiology website. 

<br/>

# Site Overview

Welcome to the website of the **S**oil **W**arming **E**xperiment in **L**owland **T**ropical **R**ainforest project, or SWELTR for short. This site contains reproducible workflows and pipelines for all aspects of processing and analyzing the SWELTR high temperature experiment, entitled *Microbial diversity declines in warmed tropical soil and respiration rise exceed predictions as communities adapt*. Most (if not all) steps were performed in R. Though we ran some steps using the Smithsonian compute cluster (Hydra), all workflows can be run on a laptop. Workflows have a modular design, meaning they can each be run independently as long as you have the necessary input files described at the beginning of each workflow. Many of the tables have download or copy buttons that allow you to quickly export data in different formats. Citations, which are embedded in the workflows, can be copied for use in reference management software. Generally, all R code is available in each workflow. Code for displaying tables and other non-analytical features are always hidden but can be accessed from the `qmd` file on the site GitHub repo. At the bottom of each page there are a variety of Appendices including a bibliography, links to source code, instructions on accessing raw data, etc.

# Data Availability

In order to run a complete pipeline, you either need the *raw* fastq files, available on the [figshare project site](https://doi.org/10.25573/data.c.5667571), or the *trimmed* fastq files, available from the European Nucleotide Archive (ENA) under project accession number [PRJEB45074 (ERP129199)](https://www.ebi.ac.uk/ena/browser/view/PRJEB45074). See the specific pipeline or [Data Availability](data-availability.html) page for more details.

Each workflow page within a pipeline also contains a Data Availability appendix that links to specific data packets require for the workflow. Or again, you can visit the [Data Availability](data-availability.html) page for more information.

# Pipelines & Workflows

There are **8** workflows in total. Each workflow contains code for processing the 16S rRNA and ITS data sets. These workflows are complete, reproducible, and designed to be run in order. That said, each ends with the creation of R data packages that can be used as input for the next workflow, allowing you to skip any individual section. In order, these are:

## Processing Workflows

1. [DADA2](dada2.html).  Workflow begins with raw fastq files and goes through the process of primer removal (using [catadapt](https://github.com/marcelm/cutadapt) [@martin2011cutadapt]), filtering, quality screening, trimming, error correction, chimera removal, and taxonomic assignment using [DADA2](https://benjjneb.github.io/dada2/) [@callahan2016dada2]. The workflow ends with the generation of an Rdata file for the next step. Depending on the DADA2 version you have installed, you may get [slightly different results](https://github.com/benjjneb/dada2/issues/532) due to fundamental changes in the code-base. This is unavoidable at times and the developers do the best they can to maintain fidelity across versions. We set random number seeds at the beginning of each workflow to ensure reproducibility. To see the package versions of tools used in each workflow, please see the R Session Information section on the bottom of each page.
2. [Data Set Prep](data-prep.html). Workflow begins with the initial sequence and taxonomy tables from the DADA2 workflow and uses [phyloseq](https://joey711.github.io/phyloseq/) [@mcmurdie2013phyloseq] to define samples and sample groups, create phyloseq objects, and curate data sets. Summary data is provided throughout the workflow. Again, workflow ends with the generation of an Rdata file for the next step.
3. [Filtering](filtering.html). In [Part A](#a.-arbitrary-filtering), we apply arbitrary filtering to the 16S rRNA and ITS data sets. In [Part B](#b.-perfect-filtering) we use PERFect [(PERmutation Filtering test for microbiome data)](https://github.com/katiasmirn/PERFect) [@smirnova2019perfect] to filter the data sets. And in [Part C](#c.-pime-filtering) of this workflow, we use PIME [(Prevalence Interval for Microbiome Evaluation)](https://github.com/microEcology/pime) [@roesch2020pime] to filter the FULL 16S rRNA and ITS data sets.

## Diversity Workflows

4. [Taxonomic](taxa.html). Taxonomic diversity primarily focused on taxon bar charts.
5. [Alpha Diversity](alpha.html). This section covers the distribution of reads across samples and includes histograms of read counts & rarefaction curves. Also covers estimates of alpha diversity using Hill numbers [@alberdi2019guide] and includes normality tests, significance tests, & diversity plots. 
6. [Beta Diversity](beta.html) Estimates of beta diversity including significance tests, beta dispersion, and ordination plots.
7. [DA ASVs & OTUs](da.html). Results and visualizations from Indicator Analysis and LEfSe. 
8. [Metadata](metadata.html). Multivariate Analysis Overview of environmental metadata and community composition. Includes metadata Normality Tests, Normalization, Autocorrelation Tests, Dissimilarity Correlation Tests, Best Subset of Variables, Distance-based Redundancy, and Capscale analysis.

Though the workflows are annotated, they are not necessarily meant to be tutorials since we only provide minimal explanation. There are many great tutorials and explanations out there on amplicon processing that you can dive into.

## Manuscript Visualization

Here is a word cloud of the manuscript, just for the fun of it ;) The size of each word in proportional to the number of times it appeared in the manuscript.


```{r}
#| echo: false
#| eval: false

filePath <- "include/main.txt"
text <- readLines(filePath)
docs <- Corpus(VectorSource(text))
inspect(docs)
toSpace <- content_transformer(function (x , pattern ) gsub(pattern, " ", x))
docs <- tm_map(docs, toSpace, "/")
docs <- tm_map(docs, toSpace, "@")
docs <- tm_map(docs, toSpace, "\\|")
```

```{r}
#| echo: false
#| eval: false
# Convert the text to lower case
# Load the text as a corpus
# Convert the text to lower case
docs <- tm_map(docs, content_transformer(tolower))
# Remove numbers
docs <- tm_map(docs, removeNumbers)
# Remove stopwords for the language 
docs <- tm_map(docs, removeWords, stopwords("english"))
docs <- tm_map(docs, removeWords, c("used", "based", "samples", "see",
                                    "using", "sample", "one", "also",
                                    "two", "site", "within", "fig"))
# Remove punctuations
docs <- tm_map(docs, removePunctuation)
# Eliminate extra white spaces
docs <- tm_map(docs, stripWhitespace)
# Text stemming
#docs <- tm_map(docs, stemDocument)
```

```{r}
#| echo: false
#| eval: false
dtm <- TermDocumentMatrix(docs)
m <- as.matrix(dtm)
v <- sort(rowSums(m), decreasing = TRUE)
d <- data.frame(word = names(v), freq = v)
head(d, 200)
```

```{r}
#| echo: false
#| eval: false
set.seed(1234)
d1 <- d %>% filter(freq > 1)

#wordcloud(words = d$word, freq = d$freq, min.freq = 1, scale=c(2,.5),
#          max.words=200, random.order=FALSE, rot.per=0.35,
#          colors=brewer.pal(8, "Dark2"))
wc <- wordcloud2(data = d1, size = 1.0, color = 'random-dark', fontFamily = "serif", gridSize = 0)
saveWidget(wc, "tmp.html", selfcontained = FALSE)
webshot("tmp.html", "wordcloud.pdf", delay = 10, vwidth = 1180, vheight = 580)
```

```{r}
#| echo: false
#| eval: true
#| warning: false
#| fig-height: 1.5
knitr::include_graphics("assets/wordcloud.png")
```

### Acknowledgements {.appendix}

This study was supported by three fellowships to ATN---a UK NERC grant NE/T012226, a European Union Marie-Curie Fellowship FP7-2012-329360, and a STRI Tupper Fellowship. Further support came from a UK NERC grant NE/K01627X/1 to PM, an ANU Biology Innovation grant to PM and Simons Foundation grant No. 429440 to W. Wcislo, STRI, and support from the U.S. Department of Agriculture (USDA), Agricultural Research Service to KB. We thank Ben Turner for his contribution to SWELTR, especially during its initial phase of operation. For their support we further thank Oris Acevado, Dayana Agudo, Aleksandra Bielnicka, Gloria Broders, Melissa Cano, David Dominguez, Milton Garcia, Matthew Larsen, Julio Rodriguez, Hubert Szczygiel, Irene Torres, Esther Velasquez, William Wcislo, Klaus Winter and Joe Wright. For the purpose of open access, the author has applied a Creative Commons Attribution (CC BY) licence to any Author Accepted Manuscript version arising from this submission. Many of the computations were conducted on the Smithsonian High-Performance Cluster (SI/HPC), Smithsonian Institution ([doi:10.25572/SIHPC](https://doi.org/10.25572/SIHPC)). 

### Author Contributions {.appendix}

ATN conceived the study. ATN, JJS, MM, JP, EB, KB and KS performed the study. ATN and JJS analysed the data. ATN wrote the paper with input from JJS, EB, KS, KB and PM. Website created by J.J.S. with input and material from ATN and KS.

```{r}
#| message: false 
#| results: hide
#| eval: true
#| echo: false
#remove(list = ls())
### COmmon formatting scripts
### NOTE: captioner.R must be read BEFORE captions_XYZ.R
source(file.path("assets", "functions.R"))
```

####  Source Code {.appendix}

The source code for this page can be accessed on GitHub `r fa(name = "github")` by [clicking this link](`r source_code()`). 

<a href="https://zenodo.org/badge/latestdoi/368915237"><img src="https://zenodo.org/badge/368915237.svg" alt="DOI"></a>

#### Data Availability {.appendix}

All data associated with this project can be accessed through the figshare Collection at [doi:10.25573/data.c.5667571](https://doi.org/10.25573/data.c.5667571) with the exception of trimmed fastq files (primers removed), which are available through the ENA under project accession number [PRJEB45074 (ERP129199)](https://www.ebi.ac.uk/ena/browser/view/PRJEB45074). 

#### Last updated on {.appendix}

```{r,echo=FALSE, eval=TRUE}
Sys.time()
```
