<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */
</style>

  <!--radix_placeholder_meta_tags-->
  <title>sweltr: 9. Metadata Analysis</title>

  <meta property="description" itemprop="description" content="Workflow for ..."/>

  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>


  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="sweltr: 9. Metadata Analysis"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="Workflow for ..."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="sweltr"/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="sweltr: 9. Metadata Analysis"/>
  <meta property="twitter:description" content="Workflow for ..."/>

  <!--/radix_placeholder_meta_tags-->
  
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["9. Metadata Analysis"]},{"type":"character","attributes":{},"value":["Workflow for ...\n"]},{"type":"character","attributes":{},"value":["assets/cite.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }

  </style>

  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure img {
    width: 100%;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refHtml = $('#ref-' + $(this).attr('data-cites')).html();
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="site_libs/header-attrs-2.8/header-attrs.js"></script>
  <link href="site_libs/panelset-0.2.6/panelset.css" rel="stylesheet" />
  <script src="site_libs/panelset-0.2.6/panelset.js"></script>
  <script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
  <script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
  <link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
  <script src="site_libs/datatables-binding-0.18/datatables.js"></script>
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
  <script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
  <script src="site_libs/jszip-1.10.20/jszip.min.js"></script>
  <link href="site_libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
  <script src="site_libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
  <link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
  <script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Roboto:300|Nunito+Sans:300|Inconsolata|Source+Sans+Pro|Work+Sans:200,300,400,500');
  /********************************/
  /* controls general properties of site*/
  /********************************/

  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
    background-color: #FEFEFE;
  }

  d-title h1{
  font-size: 2.1em;
  font-family: 'Work Sans';
  font-weight: 300;
  }

  d-article h1 {
    font-size: 1.8em;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 300;
  }

  d-article h2 {
    font-size: 1.65em;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 500;
  }

  d-article h3 {
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
    margin-top: 1em;
  }

  d-article p {
    font-size: 18px;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
  }

  d-byline .authors-affiliations {
    grid-column-end: span 3;
    grid-column-gap: 8px;
  }

  img.orcid-id {
    margin: 0;
  }

  a.name:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  a.affiliation:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  blockquote p {
  	font-size: 1.3em;
  	font-family: 'Lato';
    font-style: normal;
    margin: 0.5em 0 0.5em;
  }

  d-article li {
    margin: 0px 0px 4px;
    font-size: 18px;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
  }

  /********************************/
  /* controls CODE*/
  /********************************/

  /** pre.sourceCode controls code block without lang **/

  /** Controls r code  **/

  pre.sourceCode.r {
    margin-top: 0px;
     color: #333333;
     background-color: #E8E8E8;
     border-radius: 3px;
     font-size: 0.80em;
     font-family: 'Monaco', 'PT Mono', monospace;
     overflow-x: scroll !important;
     border-left: 6px solid #2271B2;
     border-top: 1px solid #B7B7B7;
     border-right: 1px solid #B7B7B7;
     border-bottom: 1px solid #B7B7B7;
     padding: 18px 0px 18px 20px;
   }

   pre.sourceCode.bash {
     margin-top: 0px;
      color: #333333;
      background-color: #E8E8E8;
      border-radius: 3px;
      font-size: 0.80em;
      font-family: 'Monaco', 'PT Mono', monospace;
      overflow-x: scroll !important;
      border-left: 6px solid #B22271;
      border-top: 1px solid #B7B7B7;
      border-right: 1px solid #B7B7B7;
      border-bottom: 1px solid #B7B7B7;
      padding: 18px 0px 18px 20px;
    }
  /** Controls code output **/

  d-article pre {
     margin-top: 0px;
      color: #333333;
      background-color: #FAFAFA;
      font-size: 0.8em;
      border: 1px solid #ccc;
      overflow-x: scroll !important;
      padding:  10px 10px 10px 18px;
      border-radius: 5px;

    }

  /** Controls inline code  **/

  p code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }

  d-article aside code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-family: 'Monaco', 'PT Mono', monospace;
    color: #BF2727;
  }

  /********************************/
  /* controls NAV BAR and HEADERS*/
  /********************************/

  .nav-dropdown-content .nav-dropdown-header {
    text-transform: none;
  }

  .distill-site-header .logo img {
    margin-top: 10px;
    max-height: 30px;
    margin-bottom: 10px;
  }

  .distill-site-nav {
      background-color: #113858;
      font-size: 16px;
      font-weight: 300;
  }

  img {
    margin: 5px;
  }

  /** LIGHTBOX MARKUP **/
  .thumbnail {
    max-width: 100%;
    position: relative;
  }

  /** DETAILS fold down**/

  details {
    background-color: #f6f6f6;
    cursor: pointer;
    color: #2d2d2d;
    margin-bottom:1vh;
    padding: 1em;
    border: none;
    text-align: left;
    outline: none;
  }

  .active, details:hover {
    transition: max-height 0.2s ease-out;
    outline: none;

    /* background-color: #acc2c2; */
  }


  details {
    display: flex;
    justify-content: space-between;
    align-content: center;
    border-radius: 4px;
    outline: none;

  }

  details.active {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    outline: none;

  }

  summary::-webkit-details-marker {
    display: none;
    outline: none;

  }
  summary:focus {
    outline: none;
  }
  summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f0fe";
  }

  details[open] summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f146";
  }

  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"9. Metadata Analysis","description":"Workflow for ...","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title"></a>
<a href="index.html">
<i class="fa fa-home" aria-hidden="true"></i>
</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
A. Processing Workflows
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="dada2.html">1. DADA2</a>
<a href="data-prep.html">2. Data Set Prep</a>
<a href="otu.html">3. OTU Clustering</a>
<hr/>
<span class="nav-dropdown-header">Filtering</span>
<a href="pime.html">4a. PIME Filtering</a>
<a href="perfect.html">4b. PERFect Filtering</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
B. Diversity Workflows
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="taxa.html">5. Taxonomic</a>
<a href="alpha.html">6. Alpha Diversity</a>
<a href="beta.html">7. Beta Diversity</a>
<a href="da.html">8. DA ASVs &amp; OTUs</a>
<a href="metadata.html">9. Metadata</a>
</div>
</div>
<a href="data-availability.html">Data Products</a>
</div>
<div class="nav-right">
<a href="https://github.com/sweltr/high-temp/">
<i class="fa fa-github fa-lg" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>9. Metadata Analysis</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->
<p><p>Workflow for </p></p>
</div>


<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#multivariate-summary">Multivariate Summary</a>
<ul>
<li><a href="#s-rrna">16S rRNA</a></li>
<li><a href="#its">ITS</a></li>
</ul></li>
<li><a href="#multivariate-analysis-overview">Multivariate Analysis Overview</a></li>
<li><a href="#metadata">Metadata</a></li>
<li><a href="#s-rrna-1">16s rRNA</a>
<ul>
<li><a href="#data-formating">Data Formating</a></li>
<li><a href="#choose-data-set">Choose Data Set</a></li>
<li><a href="#normality-tests">Normality Tests</a></li>
<li><a href="#normalize-parameters">Normalize Parameters</a></li>
<li><a href="#normalized-metadata">Normalized Metadata</a></li>
<li><a href="#autocorrelation-tests">Autocorrelation Tests</a></li>
<li><a href="#autocorrelation-plots">Autocorrelation Plots</a></li>
<li><a href="#dissimilarity-correlation-tests">Dissimilarity Correlation Tests</a></li>
<li><a href="#best-subset-of-variables">Best Subset of Variables</a></li>
<li><a href="#distance-based-redundancy">Distance-based Redundancy</a></li>
<li><a href="#capscale-plots">Capscale Plots</a></li>
</ul></li>
<li><a href="#its-1">ITS</a>
<ul>
<li><a href="#data-formating-1">Data Formating</a></li>
<li><a href="#choose-data-set-1">Choose Data Set</a></li>
<li><a href="#normality-tests-1">Normality Tests</a></li>
<li><a href="#normalize-parameters-1">Normalize Parameters</a></li>
<li><a href="#normalized-metadata-1">Normalized Metadata</a></li>
<li><a href="#autocorrelation-tests-1">Autocorrelation Tests</a></li>
<li><a href="#autocorrelation-plots-1">Autocorrelation Plots</a></li>
<li><a href="#dissimilarity-correlation-tests-1">Dissimilarity Correlation Tests</a></li>
<li><a href="#best-subset-of-variables-1">Best Subset of Variables</a></li>
<li><a href="#distance-based-redundancy-1">Distance-based Redundancy</a></li>
<li><a href="#capscale-plots-1">Capscale Plots</a></li>
<li><a href="#source-code">Source Code</a></li>
<li><a href="#data-availability">Data Availability</a></li>
</ul></li>
</ul>
</nav>
</div>
<details>
<summary>
Show setup information.
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">eval =</span> <span class="cn">FALSE</span>, <span class="at">echo =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">119</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#library(conflicted)</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends(vegan, local = TRUE)  </span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends_reverse(vegan, local = TRUE)  </span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq); <span class="fu">packageVersion</span>(<span class="st">&quot;phyloseq&quot;</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(ape, bestNormalize, DT, tidyverse, gdata, mctoolsr, </span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>               ggpubr, ggvegan, ggfortify, vegan,</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">install =</span> <span class="cn">FALSE</span>, <span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">scipen=</span><span class="dv">999</span>)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_current<span class="sc">$</span><span class="fu">get</span>(<span class="fu">c</span>(</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache&quot;</span>,</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.path&quot;</span>,</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.rebuild&quot;</span>,</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;dependson&quot;</span>,</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;autodep&quot;</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>))</span></code></pre></div>
</div>
</details>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="multivariate-summary">Multivariate Summary</h1>
<p>Summary results for of <code>envfit</code> and <code>bioenv</code> tests for edaphic properties, soil functional response, and temperature adaptation for both 16S rRNA and ITS data sets.</p>
<h2 id="s-rrna">16S rRNA</h2>
<p>Metadata parameters removed based on autocorrelation results.</p>
<ul>
<li>edaphic properties: TEB + DON + Na + Al + Ca</li>
<li>soil functional response: micN + micNP + enzCN + enzCP + BP_ase + CE_ase + LP_ase + N_ase + P_ase</li>
<li>temperature adaptation: NUE + PUE + SI</li>
</ul>
<p>Metadata parameters removed from capscale analysis based on Degrees of Freedom.</p>
<ul>
<li>edaphic properties: Mg + Mn</li>
<li>soil functional response: NONE</li>
<li>temperature adaptation: NONE</li>
</ul>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">
<div id="95p2uxf1n8lpjbc2t0r9" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="95p2uxf1n8lpjbc2t0r9">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Results of envfit &amp; bioenv test for the 16S rRNA data set.<\/em>\n<\/caption>","data":[["edaphic","edaphic","edaphic","functional_response","functional_response","functional_response","functional_response","functional_response","functional_response","functional_response","temperature adaptation","temperature adaptation","temperature adaptation","temperature adaptation","temperature adaptation"],["AST","H2O","DOC","AG_ase","BG_ase","S_ase","XY_ase","PX_ase","CO2","enzNP","S_Q10","LP_Q10","Tmin","CUEcp","P_Q10"],[0.828679090556174,0.518932896196791,0.445904146570695,0.443541369086058,0.560388090143001,0.737082621987766,0.519301170320349,0.763994367410604,0.504122883729818,0.624183202101924,0.495643228019936,0.413129988672484,0.445590698633893,null,null],[0.000999000999000999,0.00999000999000999,0.023976023976024,0.025974025974026,0.00699300699300699,0.001998001998002,0.00899100899100899,0.000999000999000999,0.012987012987013,0.003996003996004,0.014985014985015,0.040959040959041,0.02997002997003,null,null],[1,null,null,0.55880778688965,null,0.613853482557918,0.455966824638411,0.611525394072235,null,0.462326352892391,0.439483618153207,0.377176456710113,0.404434042774157,0.325039873858187,0.517723387090826],[0.001,null,null,0.001,null,0.001,0.002,0.001,null,0.006,0.001,0.005,0.005,0.013,0.001]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>metadata_set<\/th>\n      <th>parameter<\/th>\n      <th>r2 (envfit)<\/th>\n      <th>Pr(&gt;r) (envfit)<\/th>\n      <th>r (bioenv)<\/th>\n      <th>pval (bioenv)<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blftip","buttons":["copy","csv","excel"],"pageLength":-1,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"targets":2,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"className":"dt-right","targets":[2,3,4,5]}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-size':'80%'});\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\n}"}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.rowCallback"],"jsHooks":[]}</script>
</div>
<h2 id="its">ITS</h2>
<p>Metadata parameters removed based on autocorrelation results.</p>
<ul>
<li>edaphic properties: TEB + DON + Na + Al + Ca</li>
<li>soil functional response: micN + micNP + enzCN + enzCP + BP_ase + CE_ase + LP_ase + N_ase + P_ase</li>
<li>temperature adaptation: NUE + PUE + P_Q10 + SI</li>
</ul>
<p>Metadata parameters removed from capscale analysis based on Degrees of Freedom.</p>
<ul>
<li>edaphic properties: Mg + Mn + Na + Al + Fe + K</li>
<li>soil functional response: NONE</li>
<li>temperature adaptation: S_Q10</li>
</ul>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">
<div id="w114ycn9zt4kx4ip8165" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="w114ycn9zt4kx4ip8165">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Results of envfit &amp; bioenv test for the ITS data set.<\/em>\n<\/caption>","data":[["edaphic","edaphic","functional_response","functional_response","functional_response","functional_response","functional_response","functional_response","temperature adaptation","temperature adaptation","temperature adaptation"],["AST","DOC","micP","micCP","AG_ase","PX_ase","enzNP","XY_ase","XY_Q10","CUEcp","Tmin"],[0.484886445899775,0.535380586373238,0.692467233357286,0.583125334332718,0.505997322492504,0.500221400157373,0.547105397610172,null,0.616573511458765,0.478583809955221,0.474866778699216],[0.036963036963037,0.027972027972028,0.001998001998002,0.015984015984016,0.036963036963037,0.034965034965035,0.013986013986014,null,0.00999000999000999,0.034965034965035,0.027972027972028],[1,null,null,null,null,0.685359444078673,0.552852211080059,0.505448232350988,0.725775481284095,null,0.616015933846717],[0.001,null,null,null,null,0.001,0.001,0.002,0.001,null,0.001]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>metadata_set<\/th>\n      <th>parameter<\/th>\n      <th>r2 (envfit)<\/th>\n      <th>Pr(&gt;r) (envfit)<\/th>\n      <th>r (bioenv)<\/th>\n      <th>pval (bioenv)<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blftip","buttons":["copy","csv","excel"],"pageLength":-1,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"targets":2,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"className":"dt-right","targets":[2,3,4,5]}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-size':'80%'});\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\n}"}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.rowCallback"],"jsHooks":[]}</script>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="multivariate-analysis-overview">Multivariate Analysis Overview</h1>
<p>In this workflow we compare the environmental metadata with the microbial community data. Part 1 involves the <a href="#s-rrna-1">16S rRNA</a> community data and Part 2 deals with the <a href="#its-1">ITS</a> community data. Each workflow contains the same major steps:</p>
<ol type="1">
<li>Metadata Normality Tests: Shapiro-Wilk Normality Test to test whether each matadata parameter is normally distributed.</li>
<li>Normalize Parameters: R package <a href="https://github.com/petersonR/bestNormalize"><code>bestNormalize</code></a> to find and execute the best normalizing transformation.</li>
<li>Split Metadata parameters into groups: a) Environmental and edaphic properties, b) Microbial functional responses, and c) Temperature adaptation properties.</li>
<li>Autocorrelation Tests: Test all possible pair-wise comparisons, on both normalized and non-normalized data sets, for each group.</li>
<li>Remove autocorrelated parameters from each group.</li>
<li>Dissimilarity Correlation Tests: Use Mantel Tests to see if any on the metadata groups are significantly correlated with the community data.</li>
<li>Best Subset of Variables: Determine which of the metadata parameters from each group are the most strongly correlated with the community data. For this we use the <code>bioenv</code> function from the <code>vegan</code> <span class="citation" data-cites="oksanen2013community">(<a href="#ref-oksanen2013community" role="doc-biblioref">Oksanen et al. 2012</a>)</span> package.</li>
<li>Distance-based Redundancy Analysis: Ordination analysis of samples and metadata vector overlays using <code>capscale</code>, also from the <code>vegan</code> package.</li>
</ol>
<h1 id="metadata">Metadata</h1>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="hgiy1zr42yw7eymtlich" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="hgiy1zr42yw7eymtlich">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Metadata summary for each sample.<\/em>\n<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15"],["P02_D00_010_C0A","P04_D00_010_C0B","P06_D00_010_C0C","P08_D00_010_C0D","P10_D00_010_C0E","P01_D00_010_W3A","P03_D00_010_W3B","P05_D00_010_W3C","P07_D00_010_W3D","P09_D00_010_W3E","P01_D00_010_W8A","P03_D00_010_W8B","P05_D00_010_W8C","P07_D00_010_W8D","P09_D00_010_W8E"],["P02","P04","P06","P08","P10","P01","P03","P05","P07","P09","P01","P03","P05","P07","P09"],["C","C","C","C","C","W","W","W","W","W","W","W","W","W","W"],[0,0,0,0,0,3,3,3,3,3,8,8,8,8,8],["A","B","C","D","E","A","B","C","D","E","A","B","C","D","E"],["W","W","W","W","W","W","W","W","W","W","W","W","W","W","W"],[26,26,26,26,26,29,29,29,29,29,34,34,34,34,34],[26.2,26.29,26.26,25.8,26.09,29.38,29.29,28.53,28.78,28.8,30.66,41.77,36.24,29.9,31.61],[0.3793,0.3832,0.3774,0.4011,0.3702,0.3999,0.4007,0.3545,0.3849,0.3646,0.261,0.243,0.233,0.204,0.359],[0.3367,0.3033,0.3333,0.2433,0.26,0.4267,0.32,0.3333,0.4367,0.2833,0.4267,0.32,0.3333,0.4367,0.2833],[0.43,0.33,0.3,0.27,0.32,0.35,0.37,0.33,0.35,0.36,0.35,0.37,0.33,0.35,0.36],[0.01,0,0,0,0.02,0,0.01,0,0,0.02,0,0.01,0,0,0.02],[36.65,29.23,32.07,25.49,37.06,31.43,27.19,26.14,28.13,39.79,31.43,27.19,26.14,28.13,39.79],[0.01,0.02,0,0,0.01,0.01,0.01,0,0.01,0.04,0.01,0.01,0,0.01,0.04],[0.69,1.18,0.57,1.86,1.38,1.72,0.85,0.67,1.6,1.09,1.72,0.85,0.67,1.6,1.09],[8.16,14.59,21.32,14.93,21.43,9.98,16.51,14.83,18.87,20.64,9.98,16.51,14.83,18.87,20.64],[0.04,0.12,0.06,0.02,0.07,0.04,0.06,0.05,0.07,0.08,0.04,0.06,0.05,0.07,0.08],[0.11,0.14,0.14,0.08,0.13,0.15,0.12,0,0.1,0.19,0.15,0.12,0,0.1,0.19],[45.6,45.1,54.1,42.3,60,43.3,44.7,41.6,48.7,61.7,43.3,44.7,41.6,48.7,61.7],[45.7,45.3,54.2,42.4,60.1,43.3,44.7,41.7,48.8,61.8,43.3,44.7,41.7,48.8,61.8],[6.18,5.73,5.93,6.01,6.28,6.01,5.86,5.87,6.21,6.1,6.01,5.86,5.87,6.21,6.1],[1.42,1.5,0.75,3.67,4.08,1.33,2.83,1.17,1,2.58,0.12,0.2,0.11,0.02,0],[3.92,3.17,1.83,4.33,4.75,3.67,3.33,2.17,2.42,3.5,1.65,1.28,1.18,2.6,0.92],[24.33,19.92,19.67,106.42,37.33,47.17,42.5,11.25,20.67,11.58,11.31,22.71,36.61,12.31,7.84],[28.25,23.08,21.5,110.75,42.08,50.83,45.83,13.42,23.08,15.08,12.96,23.99,37.79,14.91,8.76],[4.81,5.57,2.92,2.04,3.44,8.03,3.11,2.32,4.28,2.03,4.08,2.5,3.91,8.19,6.5],[3.13,3.22,2.27,3.71,4.53,4.67,2.57,3.31,5.07,2.9,2.65,1.06,2.3,6.39,2.76],[7.58,7.95,7.04,5.97,9.45,9.52,8.66,7.3,9.15,8.28,8.55,2.8,4.25,5.62,6.08],[75.86,59.51,56.68,44.89,63.79,118.74,60.57,64.98,73.98,43.84,39.7,33.2,45,44.3,30.8],[8.22,6.32,5.42,5.22,4.91,11.66,6.48,7.03,8.47,5.56,4.2,2.5,4.3,3.7,1.6],[9.44,10.67,11.09,11.57,14.67,11.15,10.12,11.93,9.93,12.21,9.45,13.29,10.46,11.98,19.27],[444.72,440.6,390.66,277.37,334.93,690.11,364.62,284.5,396.42,392.56,469,117,440,676,416],[110.13,86.19,72.61,56.55,72.64,135.48,66.4,48.79,73.73,62.37,84,20,97,147,91],[65.03,81.45,85.7,33.4,64.23,87.98,70.68,52.5,93.13,66.65,55,11,56,94,63],[4.42,5.14,5.26,4.72,4.49,4.61,5.07,5.57,4.72,5.78,5.58,5.85,4.54,4.6,4.57],[8.13,5.39,4.5,11.55,5.54,8.61,6.7,8.33,4.66,7.34,8.48,10.54,7.91,7.18,6.6],[2.37,1.09,0.83,2.45,1.17,1.74,1.31,1.39,0.98,1.19,1.52,1.8,1.74,1.56,1.44],[0.24,0.22,0.21,0.15,0.28,0.26,0.33,0.34,0.32,0.2,0.15,0.02,0.15,0.39,0.37],[4.22,4.57,3.22,3.55,4.62,4.1,4.37,3.01,4.14,3.92,6.97,3.04,9.09,8.61,10.48],[2.54,3.02,2.33,1.77,2.11,3.5,3.05,2.34,3.45,2.26,5.34,2.89,7.28,7.11,6.46],[1.09,1.09,0.75,0.68,0.86,1,1.26,0.76,1.07,0.86,2.07,0.65,1.5,1.53,2.21],[15.07,20.27,14.79,13.02,14.8,20.43,19.21,15.19,23.26,16.9,25.65,18.11,44.04,35.1,35.75],[3.54,3.65,2.57,2.95,3.37,5.63,3.58,2.61,3.7,2.83,4.52,2.91,5.56,8.34,9.43],[3.8,4.92,5.45,3.56,4.54,6.08,5.72,3.2,5.86,5.26,6.7,1.28,5.57,10.68,11.78],[1.01,1.11,0.66,0.71,0.72,1.17,1.36,0.86,1.11,1.01,2.52,0.76,1.75,1.42,1.84],[1.51,1.51,1.83,2.08,1.54,1.77,2.48,1.48,1.72,1.67,2.01,1.49,1.73,1.31,1.93],[1.31,1.5,1.31,1.76,1.37,1.38,1.36,1.35,1.45,1.36,1.75,1.56,1.72,1.39,1.53],[1.34,1.58,1.44,1.74,1.41,1.39,1.42,1.39,1.46,1.56,1.51,1.55,1.42,1.42,1.44],[1.89,2.3,2.36,2,1.81,2.03,1.99,1.88,1.76,1.81,1.87,1.9,1.95,1.96,2.11],[1.22,1.37,1.24,1.48,1.27,1.37,1.27,1.28,1.28,1.35,1.39,1.51,1.48,1.33,1.38],[1.54,1.85,1.6,1.69,1.47,1.46,1.54,1.51,1.36,1.48,1.56,1.75,1.68,1.54,1.46],[1.23,1.39,1.47,1.48,1.33,1.3,1.27,1.3,1.3,1.32,1.65,1.39,1.6,1.51,1.5],[1.59,1.76,1.63,1.83,1.65,1.72,1.72,1.82,1.63,1.84,1.86,1.89,1.88,1.91,1.76],[5.27,4.06,4.28,2.08,5.21,11.78,4.46,4.62,6.19,9.39,4.28,40.47,15.37,8.87,13.6],[38.61,29.51,27.78,21.96,32.22,52.8,32.1,24.11,42.4,30.37,40.69,13.94,32.65,57.13,53.79],[1.81,2.14,1.85,2.07,1.8,1.76,2,1.79,1.96,1.9,1.71,1.77,1.67,1.4,1.79],[94.13,81.25,92.5,92.38,111.88,147.13,119.75,110.75,97,122.63,211.18,225.56,245.65,294.39,339.6],[1.58,1.24,1.53,3.58,1.14,1.22,1.41,1.33,1.15,1.39,1.07,1.15,1.23,1.27,1.27],[0.22,0.25,0.27,0.24,0.22,0.32,0.26,0.27,0.27,0.26,0.23,0.29,0.19,0.26,0.24],[0.21,0.19,0.16,0.22,0.16,0.21,0.18,0.18,0.15,0.21,0.18,0.33,0.23,0.17,0.19],[0.87,0.85,0.83,0.85,0.87,0.77,0.84,0.83,0.83,0.84,0.87,0.81,0.9,0.84,0.85],[0.88,0.9,0.92,0.88,0.92,0.88,0.9,0.9,0.92,0.88,0.9,0.77,0.87,0.91,0.89],[0.14,-2.63,-0.13,0,-2.46,0.07,-0.29,0.81,0.65,-0.24,2.19,0.63,0.94,1.48,0.96],[1.44,1.25,1.42,1.54,1.29,1.51,1.49,1.77,1.61,1.51,2.3,1.72,2.08,1.81,1.82],[0.5024,0.555,0.4624,0.5058,0.5231,0.4614,0.5565,0.4877,0.4941,0.5112,0.6456,0.5302,0.6931,0.5933,0.6515],[0.6554,0.6173,0.5556,0.6037,0.6607,0.5911,0.6412,0.5596,0.5762,0.6064,0.7166,0.4918,0.6412,0.6628,0.7218],[1.3044,1.1123,1.2017,1.1935,1.263,1.2812,1.1522,1.1473,1.1662,1.1862,1.11,0.9277,0.9251,1.1173,1.1079]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>id<\/th>\n      <th>PLOT<\/th>\n      <th>TREAT<\/th>\n      <th>TREAT_T<\/th>\n      <th>PAIR<\/th>\n      <th>SEASON<\/th>\n      <th>ATAP<\/th>\n      <th>AST<\/th>\n      <th>H2O<\/th>\n      <th>N<\/th>\n      <th>P<\/th>\n      <th>Al<\/th>\n      <th>Ca<\/th>\n      <th>Fe<\/th>\n      <th>K<\/th>\n      <th>Mg<\/th>\n      <th>Mn<\/th>\n      <th>Na<\/th>\n      <th>TEB<\/th>\n      <th>ECEC<\/th>\n      <th>pH<\/th>\n      <th>minPO4<\/th>\n      <th>minNH4<\/th>\n      <th>minNO3<\/th>\n      <th>minTIN<\/th>\n      <th>NH4<\/th>\n      <th>NO3<\/th>\n      <th>PO4<\/th>\n      <th>DOC<\/th>\n      <th>DON<\/th>\n      <th>DOCN<\/th>\n      <th>micC<\/th>\n      <th>micN<\/th>\n      <th>micP<\/th>\n      <th>micCN<\/th>\n      <th>micCP<\/th>\n      <th>micNP<\/th>\n      <th>AG_ase<\/th>\n      <th>BG_ase<\/th>\n      <th>BP_ase<\/th>\n      <th>CE_ase<\/th>\n      <th>P_ase<\/th>\n      <th>N_ase<\/th>\n      <th>S_ase<\/th>\n      <th>XY_ase<\/th>\n      <th>AG_Q10<\/th>\n      <th>BG_Q10<\/th>\n      <th>BP_Q10<\/th>\n      <th>CE_Q10<\/th>\n      <th>P_Q10<\/th>\n      <th>N_Q10<\/th>\n      <th>S_Q10<\/th>\n      <th>XY_Q10<\/th>\n      <th>CO2<\/th>\n      <th>LP_ase<\/th>\n      <th>LP_Q10<\/th>\n      <th>PX_ase<\/th>\n      <th>PX_Q10<\/th>\n      <th>CUEcn<\/th>\n      <th>CUEcp<\/th>\n      <th>NUE<\/th>\n      <th>PUE<\/th>\n      <th>Tmin<\/th>\n      <th>SI<\/th>\n      <th>enzCN<\/th>\n      <th>enzCP<\/th>\n      <th>enzNP<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"className":"dt-right","targets":[4,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\nvar value=data[8]; $(this.api().cell(row, 8).node()).css({'font-size':'80%'});\nvar value=data[9]; $(this.api().cell(row, 9).node()).css({'font-size':'80%'});\nvar value=data[10]; $(this.api().cell(row, 10).node()).css({'font-size':'80%'});\nvar value=data[11]; $(this.api().cell(row, 11).node()).css({'font-size':'80%'});\nvar value=data[12]; $(this.api().cell(row, 12).node()).css({'font-size':'80%'});\nvar value=data[13]; $(this.api().cell(row, 13).node()).css({'font-size':'80%'});\nvar value=data[14]; $(this.api().cell(row, 14).node()).css({'font-size':'80%'});\nvar value=data[15]; $(this.api().cell(row, 15).node()).css({'font-size':'80%'});\nvar value=data[16]; $(this.api().cell(row, 16).node()).css({'font-size':'80%'});\nvar value=data[17]; $(this.api().cell(row, 17).node()).css({'font-size':'80%'});\nvar value=data[18]; $(this.api().cell(row, 18).node()).css({'font-size':'80%'});\nvar value=data[19]; $(this.api().cell(row, 19).node()).css({'font-size':'80%'});\nvar value=data[20]; $(this.api().cell(row, 20).node()).css({'font-size':'80%'});\nvar value=data[21]; $(this.api().cell(row, 21).node()).css({'font-size':'80%'});\nvar value=data[22]; $(this.api().cell(row, 22).node()).css({'font-size':'80%'});\nvar value=data[23]; $(this.api().cell(row, 23).node()).css({'font-size':'80%'});\nvar value=data[24]; $(this.api().cell(row, 24).node()).css({'font-size':'80%'});\nvar value=data[25]; $(this.api().cell(row, 25).node()).css({'font-size':'80%'});\nvar value=data[26]; $(this.api().cell(row, 26).node()).css({'font-size':'80%'});\nvar value=data[27]; $(this.api().cell(row, 27).node()).css({'font-size':'80%'});\nvar value=data[28]; $(this.api().cell(row, 28).node()).css({'font-size':'80%'});\nvar value=data[29]; $(this.api().cell(row, 29).node()).css({'font-size':'80%'});\nvar value=data[30]; $(this.api().cell(row, 30).node()).css({'font-size':'80%'});\nvar value=data[31]; $(this.api().cell(row, 31).node()).css({'font-size':'80%'});\nvar value=data[32]; $(this.api().cell(row, 32).node()).css({'font-size':'80%'});\nvar value=data[33]; $(this.api().cell(row, 33).node()).css({'font-size':'80%'});\nvar value=data[34]; $(this.api().cell(row, 34).node()).css({'font-size':'80%'});\nvar value=data[35]; $(this.api().cell(row, 35).node()).css({'font-size':'80%'});\nvar value=data[36]; $(this.api().cell(row, 36).node()).css({'font-size':'80%'});\nvar value=data[37]; $(this.api().cell(row, 37).node()).css({'font-size':'80%'});\nvar value=data[38]; $(this.api().cell(row, 38).node()).css({'font-size':'80%'});\nvar value=data[39]; $(this.api().cell(row, 39).node()).css({'font-size':'80%'});\nvar value=data[40]; $(this.api().cell(row, 40).node()).css({'font-size':'80%'});\nvar value=data[41]; $(this.api().cell(row, 41).node()).css({'font-size':'80%'});\nvar value=data[42]; $(this.api().cell(row, 42).node()).css({'font-size':'80%'});\nvar value=data[43]; $(this.api().cell(row, 43).node()).css({'font-size':'80%'});\nvar value=data[44]; $(this.api().cell(row, 44).node()).css({'font-size':'80%'});\nvar value=data[45]; $(this.api().cell(row, 45).node()).css({'font-size':'80%'});\nvar value=data[46]; $(this.api().cell(row, 46).node()).css({'font-size':'80%'});\nvar value=data[47]; $(this.api().cell(row, 47).node()).css({'font-size':'80%'});\nvar value=data[48]; $(this.api().cell(row, 48).node()).css({'font-size':'80%'});\nvar value=data[49]; $(this.api().cell(row, 49).node()).css({'font-size':'80%'});\nvar value=data[50]; $(this.api().cell(row, 50).node()).css({'font-size':'80%'});\nvar value=data[51]; $(this.api().cell(row, 51).node()).css({'font-size':'80%'});\nvar value=data[52]; $(this.api().cell(row, 52).node()).css({'font-size':'80%'});\nvar value=data[53]; $(this.api().cell(row, 53).node()).css({'font-size':'80%'});\nvar value=data[54]; $(this.api().cell(row, 54).node()).css({'font-size':'80%'});\nvar value=data[55]; $(this.api().cell(row, 55).node()).css({'font-size':'80%'});\nvar value=data[56]; $(this.api().cell(row, 56).node()).css({'font-size':'80%'});\nvar value=data[57]; $(this.api().cell(row, 57).node()).css({'font-size':'80%'});\nvar value=data[58]; $(this.api().cell(row, 58).node()).css({'font-size':'80%'});\nvar value=data[59]; $(this.api().cell(row, 59).node()).css({'font-size':'80%'});\nvar value=data[60]; $(this.api().cell(row, 60).node()).css({'font-size':'80%'});\nvar value=data[61]; $(this.api().cell(row, 61).node()).css({'font-size':'80%'});\nvar value=data[62]; $(this.api().cell(row, 62).node()).css({'font-size':'80%'});\nvar value=data[63]; $(this.api().cell(row, 63).node()).css({'font-size':'80%'});\nvar value=data[64]; $(this.api().cell(row, 64).node()).css({'font-size':'80%'});\nvar value=data[65]; $(this.api().cell(row, 65).node()).css({'font-size':'80%'});\nvar value=data[66]; $(this.api().cell(row, 66).node()).css({'font-size':'80%'});\nvar value=data[67]; $(this.api().cell(row, 67).node()).css({'font-size':'80%'});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
<h1 id="s-rrna-1">16s rRNA</h1>
<h2 id="data-formating">Data Formating</h2>
<ol type="1">
<li>Transform <code>ps</code> Objects</li>
</ol>
<p>The first step is to transform the phyloseq objects.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>samp_ps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ssu18_ps_work&quot;</span>, <span class="st">&quot;ssu18_ps_pime&quot;</span>, <span class="st">&quot;ssu18_ps_perfect&quot;</span>, </span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;ssu18_ps_work_otu&quot;</span>, <span class="st">&quot;ssu18_ps_pime_otu&quot;</span>, <span class="st">&quot;ssu18_ps_perfect_otu&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> samp_ps) {</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(tmp_get, <span class="cf">function</span>(otu) <span class="fl">1e5</span> <span class="sc">*</span> otu<span class="sc">/</span><span class="fu">sum</span>(otu))</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>     tmp_ps<span class="sc">@</span>phy_tree <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">prune_samples</span>(<span class="fu">sample_sums</span>(tmp_ps) <span class="sc">&gt;</span> <span class="dv">0</span>, tmp_ps)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>     tmp_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(tmp_ps), <span class="at">rooted =</span> <span class="cn">TRUE</span>, <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(tmp_ps))</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> tmp_samp <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;TREAT_T&quot;</span> <span class="ot">=</span> <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(tmp_ps, sample_data, tmp_tree)</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>     <span class="fu">sample_data</span>(tmp_ps) <span class="ot">&lt;-</span> tmp_samp</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>))</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_name)</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_ps)</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<ol start="2" type="1">
<li>Format for <code>mctoolsr</code></li>
</ol>
<p>Then a bit of formatting to make the data compatible with <code>mctoolsr</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> samp_ps) {</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>)))</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>     tmp_tax <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(tmp_get))</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>     tmp_tax<span class="sc">$</span>ASV_SEQ <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>     tmp_col_names <span class="ot">&lt;-</span> <span class="fu">colnames</span>(tmp_tax)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>     tmp_tax_merge <span class="ot">&lt;-</span> tmp_tax <span class="sc">%&gt;%</span> tidyr<span class="sc">::</span><span class="fu">unite</span>(taxonomy, </span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>                                               <span class="fu">all_of</span>(tmp_col_names), </span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">sep =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>     tmp_tax_merge <span class="ot">&lt;-</span> tmp_tax_merge <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;#OTU_ID&quot;</span>)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>     tmp_otu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(tmp_get)))</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>     tmp_otu <span class="ot">&lt;-</span> tmp_otu <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;#OTU_ID&quot;</span>)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>     tmp_otu_tax <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_otu, tmp_tax_merge, <span class="at">by =</span> <span class="st">&quot;#OTU_ID&quot;</span>)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>     tmp_samp[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>)] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> tmp_samp <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;#SampleID&quot;</span>)</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>     tmp_metad <span class="ot">&lt;-</span> metad <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;#SampleID&quot;</span> <span class="ot">=</span> <span class="st">&quot;id&quot;</span>)</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>     tmp_metad[,<span class="dv">2</span><span class="sc">:</span><span class="dv">5</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>     tmp_md <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_samp, tmp_metad, <span class="at">by =</span> <span class="st">&quot;#SampleID&quot;</span>)</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>     tmp_otu_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_otu_tax&quot;</span>))</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_otu_name)</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_otu_name, tmp_otu_tax)</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>     tmp_md_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_md&quot;</span>))</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_md_name)</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_md_name, tmp_md)</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>     tmp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&quot;files/metadata/tables/&quot;</span>)</span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>     <span class="fu">write_delim</span>(tmp_otu_tax, <span class="fu">paste</span>(tmp_path, tmp_otu_name, <span class="st">&quot;.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>     <span class="fu">write_delim</span>(tmp_md, <span class="fu">paste</span>(tmp_path, tmp_md_name, <span class="st">&quot;.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> samp_ps) {</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>     tmp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&quot;files/metadata/tables/&quot;</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>     tmp_otu_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_otu_tax&quot;</span>))</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>     tmp_md_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_md&quot;</span>))</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>     tmp_tax_table_fp <span class="ot">&lt;-</span> <span class="fu">paste</span>(tmp_path, tmp_otu_name, <span class="st">&quot;.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>     tmp_map_fp <span class="ot">&lt;-</span> <span class="fu">paste</span>(tmp_path, tmp_md_name, <span class="st">&quot;.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>     tmp_input <span class="ot">&lt;-</span> <span class="fu">load_taxa_table</span>(tmp_tax_table_fp, tmp_map_fp)</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>     tmp_input_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_mc&quot;</span>))</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_input_name)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_input_name, tmp_input)</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;_md&quot;</span>))</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;_otu_tax&quot;</span>))</span></code></pre></div>
</details>
</div>
<h2 id="choose-data-set">Choose Data Set</h2>
<p>At this point in the code we need to choose a data set to use, formatted with <code>mctoolsr</code>. Remember, there are four choices:</p>
<ol type="1">
<li>Complete ASV data set.</li>
<li>PIME filtered ASV data set.</li>
<li>PERfect filtered ASV data set.</li>
<li>Complete OTU data set.</li>
<li>PIME filtered OTU data set.</li>
<li>PERfect filtered OTU data set.</li>
</ol>
<p>This way, if we want to test other data sets we only need to change the name here.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;_mc&quot;</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>ssu18_select_mc <span class="ot">&lt;-</span> ssu18_ps_pime_mc</span></code></pre></div>
</div>
<h2 id="normality-tests">Normality Tests</h2>
<p>Before proceeding, we need to test each parameter in the metadata to see which ones are and are not normally distributed. For that, we use the Shapiro-Wilk Normality Test. Here we only need one of the metadata files.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>temp_md <span class="ot">&lt;-</span> ssu18_select_mc<span class="sc">$</span>map_loaded</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>temp_md[,<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>shap_results <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">colnames</span>(temp_md)) {</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  tmp_shap <span class="ot">&lt;-</span> <span class="fu">shapiro.test</span>(temp_md[[i]])</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  tmp_p <span class="ot">&lt;-</span> <span class="fu">round</span>(tmp_shap<span class="sc">$</span>p.value, <span class="at">digits =</span> <span class="dv">5</span>)</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>  tmp_res <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">isTRUE</span>(tmp_shap<span class="sc">$</span>p.value <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  shap_results <span class="ot">&lt;-</span> <span class="fu">rbind</span>(shap_results, <span class="fu">data.frame</span>(i, tmp_p, tmp_res))</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(shap_results) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;parameter&quot;</span>, <span class="st">&quot;p-value&quot;</span>, <span class="st">&quot;tranform&quot;</span>)</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>shap_results</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">filter</span>(shap_results, tranform <span class="sc">==</span> <span class="st">&quot;TRUE&quot;</span>)</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>md_to_tranform <span class="ot">&lt;-</span> shap_results<span class="sc">$</span>parameter[shap_results<span class="sc">$</span>tranform <span class="sc">==</span> <span class="cn">TRUE</span>]</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;temp_md&quot;</span>))</span></code></pre></div>
</details>
</div>
<details>
<summary>
Show the results of each normality test for metadata parameters
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>   parameter p-value tranform
1        AST 0.00276     TRUE
2        H2O 0.00188     TRUE
3          N 0.05771    FALSE
4          P 0.29892    FALSE
5         Al 0.00028     TRUE
6         Ca 0.03383     TRUE
7         Fe 0.00068     TRUE
8          K 0.13159    FALSE
9         Mg 0.16381    FALSE
10        Mn 0.32078    FALSE
11        Na 0.05849    FALSE
12       TEB 0.00435     TRUE
13      ECEC 0.00433     TRUE
14        pH 0.54818    FALSE
15    minPO4 0.05053    FALSE
16    minNH4 0.70522    FALSE
17    minNO3 0.00050     TRUE
18    minTIN 0.00072     TRUE
19       NH4 0.07406    FALSE
20       NO3 0.48101    FALSE
21       PO4 0.24574    FALSE
22       DOC 0.03038     TRUE
23       DON 0.79687    FALSE
24      DOCN 0.00309     TRUE
25      micC 0.16051    FALSE
26      micN 0.77320    FALSE
27      micP 0.23294    FALSE
28     micCN 0.04004     TRUE
29     micCP 0.66207    FALSE
30     micNP 0.34934    FALSE
31    AG_ase 0.70541    FALSE
32    BG_ase 0.00305     TRUE
33    BP_ase 0.00483     TRUE
34    CE_ase 0.03066     TRUE
35     P_ase 0.00864     TRUE
36     N_ase 0.00206     TRUE
37     S_ase 0.05361    FALSE
38    XY_ase 0.04239     TRUE
39    AG_Q10 0.25180    FALSE
40    BG_Q10 0.01348     TRUE
41    BP_Q10 0.03611     TRUE
42    CE_Q10 0.06190    FALSE
43     P_Q10 0.28886    FALSE
44     N_Q10 0.37946    FALSE
45     S_Q10 0.27576    FALSE
46    XY_Q10 0.27276    FALSE
47       CO2 0.00006     TRUE
48    LP_ase 0.59206    FALSE
49    LP_Q10 0.45921    FALSE
50    PX_ase 0.00822     TRUE
51    PX_Q10 0.00000     TRUE
52     CUEcn 0.91397    FALSE
53     CUEcp 0.00370     TRUE
54       NUE 0.33275    FALSE
55       PUE 0.00025     TRUE
56      Tmin 0.05998    FALSE
57        SI 0.28369    FALSE
58     enzCN 0.09137    FALSE
59     enzCP 0.92171    FALSE
60     enzNP 0.09436    FALSE</code></pre>
</div>
</details>
<p>Looks like we need to transform 25 metadata parameters.</p>
<h2 id="normalize-parameters">Normalize Parameters</h2>
<p>Here we use the R package <a href="https://cran.r-project.org/web/packages/bestNormalize/vignettes/bestNormalize.html"><code>bestNormalize</code></a> to find and execute the best normalizing transformation. The function will test the following normalizing transformations:</p>
<ul>
<li><p><code>arcsinh_x</code> performs an arcsinh transformation.</p></li>
<li><p><code>boxcox</code> Perform a Box-Cox transformation and center/scale a vector to attempt normalization. <code>boxcox</code> estimates the optimal value of lambda for the Box-Cox transformation. The function will return an error if a user attempt to transform nonpositive data.</p></li>
<li><p><code>yeojohnson</code> Perform a Yeo-Johnson Transformation and center/scale a vector to attempt normalization. <code>yeojohnson</code> estimates the optimal value of lambda for the Yeo-Johnson transformation. The Yeo-Johnson is similar to the Box-Cox method, however it allows for the transformation of nonpositive data as well.</p></li>
<li><p><code>orderNorm</code> The Ordered Quantile (ORQ) normalization transformation, <code>orderNorm()</code>, is a rank-based procedure by which the values of a vector are mapped to their percentile, which is then mapped to the same percentile of the normal distribution. Without the presence of ties, this essentially guarantees that the transformation leads to a uniform distribution.</p></li>
<li><p><code>log_x</code> performs a simple log transformation. The parameter a is essentially estimated by the training set by default (estimated as the minimum possible to some extent epsilon), while the base must be specified beforehand. The default base of the log is 10.</p></li>
<li><p><code>sqrt_x</code> performs a simple square-root transformation. The parameter a is essentially estimated by the training set by default (estimated as the minimum possible), while the base must be specified beforehand.</p></li>
<li><p><code>exp_x</code> performs a simple exponential transformation.</p></li>
</ul>
<p>See this GitHub issue (<a href="https://github.com/petersonR/bestNormalize/issues/5">#5</a>) for a description on getting reproducible results. Apparently, you can get different results because the <code>bestNormalize()</code> function uses repeated cross-validation (and doesnt automatically set the seed), so the results will be slightly different each time the function is executed.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">119</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> md_to_tranform) {</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  tmp_md <span class="ot">&lt;-</span> ssu18_select_mc<span class="sc">$</span>map_loaded</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  tmp_best_norm <span class="ot">&lt;-</span> <span class="fu">bestNormalize</span>(tmp_md[[i]], <span class="at">r =</span> <span class="dv">1</span>, <span class="at">k =</span> <span class="dv">5</span>, <span class="at">loo =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_best_norm_test&quot;</span>))</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_best_norm)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(tmp_name)</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<details>
<summary>
Show the chosen transformations
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
## bestNormalize Chosen transformation of AST ## 
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
25.800 26.275 28.800 30.280 41.770 
_____________________________________

## bestNormalize Chosen transformation of H2O ## 
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
   0%   25%   50%   75%  100% 
0.204 0.308 0.370 0.384 0.401 
_____________________________________

## bestNormalize Chosen transformation of Al ## 
Standardized asinh(x) Transformation with 15 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 0.0059997 
 - sd (before standardization) = 0.008280287 
_____________________________________

## bestNormalize Chosen transformation of Ca ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
25.49 27.19 29.23 34.36 39.79 
_____________________________________

## bestNormalize Chosen transformation of Fe ## 
Standardized asinh(x) Transformation with 15 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 0.0119984 
 - sd (before standardization) = 0.01264569 
_____________________________________

## bestNormalize Chosen transformation of TEB ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
41.6 43.3 45.1 51.4 61.7 
_____________________________________

## bestNormalize Chosen transformation of ECEC ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
41.7 43.3 45.3 51.5 61.8 
_____________________________________

## bestNormalize Chosen transformation of minNO3 ## 
Standardized Box Cox Transformation with 15 nonmissing obs.:
 Estimated statistics:
 - lambda = -0.3344538 
 - mean (before standardization) = 1.907818 
 - sd (before standardization) = 0.2388776 
_____________________________________

## bestNormalize Chosen transformation of minTIN ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 14 unique values 
 - Original quantiles:
     0%     25%     50%     75%    100% 
  8.760  14.995  23.080  39.935 110.750 
_____________________________________

## bestNormalize Chosen transformation of DOC ## 
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
     0%     25%     50%     75%    100% 
 30.800  44.070  56.680  64.385 118.740 
_____________________________________

## bestNormalize Chosen transformation of DOCN ## 
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 9.440 10.290 11.150 12.095 19.270 
_____________________________________

## bestNormalize Chosen transformation of micCN ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 14 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
4.420 4.585 4.720 5.415 5.850 
_____________________________________

## bestNormalize Chosen transformation of BG_ase ## 
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 3.010  3.735  4.220  5.795 10.480 
_____________________________________

## bestNormalize Chosen transformation of BP_ase ## 
Standardized Log_b(x + a) Transformation with 15 nonmissing obs.:
 Relevant statistics:
 - a = 0 
 - b = 10 
 - mean (before standardization) = 0.5217757 
 - sd (before standardization) = 0.2001498 
_____________________________________

## bestNormalize Chosen transformation of CE_ase ## 
Standardized Log_b(x + a) Transformation with 15 nonmissing obs.:
 Relevant statistics:
 - a = 0 
 - b = 10 
 - mean (before standardization) = 0.03306978 
 - sd (before standardization) = 0.1652591 
_____________________________________

## bestNormalize Chosen transformation of P_ase ## 
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
13.020 15.130 19.210 24.455 44.040 
_____________________________________

## bestNormalize Chosen transformation of N_ase ## 
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
  0%  25%  50%  75% 100% 
2.57 2.93 3.58 5.04 9.43 
_____________________________________

## bestNormalize Chosen transformation of XY_ase ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 13 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
0.66 0.81 1.11 1.39 2.52 
_____________________________________

## bestNormalize Chosen transformation of BG_Q10 ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 13 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
1.310 1.360 1.390 1.545 1.760 
_____________________________________

## bestNormalize Chosen transformation of BP_Q10 ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 11 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
1.340 1.415 1.440 1.530 1.740 
_____________________________________

## bestNormalize Chosen transformation of CO2 ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 14 unique values 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 2.080  4.370  5.270 10.585 40.470 
_____________________________________

## bestNormalize Chosen transformation of PX_ase ## 
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
     0%     25%     50%     75%    100% 
 81.250  95.565 119.750 218.370 339.600 
_____________________________________

## bestNormalize Chosen transformation of PX_Q10 ## 
Standardized Yeo-Johnson Transformation with 15 nonmissing obs.:
 Estimated statistics:
 - lambda = -4.999946 
 - mean (before standardization) = 0.1968247 
 - sd (before standardization) = 0.001294011 
_____________________________________

## bestNormalize Chosen transformation of CUEcp ## 
Standardized Box Cox Transformation with 15 nonmissing obs.:
 Estimated statistics:
 - lambda = -0.9999576 
 - mean (before standardization) = -4.229896 
 - sd (before standardization) = 0.9115063 
_____________________________________

## bestNormalize Chosen transformation of PUE ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 7 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
0.770 0.880 0.900 0.905 0.920 
_____________________________________</code></pre>
</div>
</details>
<details>
<summary>
Show the complete <code>bestNormalize</code> results
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
## Results of bestNormalize for AST ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 3.4
 - Box-Cox: 1.8
 - Center+scale: 4.2
 - Exp(x): 21.2667
 - Log_b(x+a): 3.4
 - orderNorm (ORQ): 0.2
 - sqrt(x + a): 4.2
 - Yeo-Johnson: 2.8667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
25.800 26.275 28.800 30.280 41.770 
_____________________________________

## Results of bestNormalize for H2O ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 6.8667
 - Box-Cox: 5.8
 - Center+scale: 5.8
 - Exp(x): 5.8
 - Log_b(x+a): 6.0667
 - orderNorm (ORQ): 0.7333
 - sqrt(x + a): 6.8667
 - Yeo-Johnson: 5.8
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
   0%   25%   50%   75%  100% 
0.204 0.308 0.370 0.384 0.401 
_____________________________________

## Results of bestNormalize for Al ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 8.2
 - Center+scale: 8.2
 - Exp(x): 8.2
 - Log_b(x+a): 8.2
 - orderNorm (ORQ): 8.2
 - sqrt(x + a): 8.2
 - Yeo-Johnson: 8.2
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized asinh(x) Transformation with 15 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 0.0059997 
 - sd (before standardization) = 0.008280287 
_____________________________________

## Results of bestNormalize for Ca ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.8
 - Box-Cox: 1.8
 - Center+scale: 3.4
 - Exp(x): 18.0667
 - Log_b(x+a): 1.8
 - orderNorm (ORQ): 0.2
 - sqrt(x + a): 1.8
 - Yeo-Johnson: 1.2667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
25.49 27.19 29.23 34.36 39.79 
_____________________________________

## Results of bestNormalize for Fe ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 6.3333
 - Center+scale: 6.3333
 - Exp(x): 6.3333
 - Log_b(x+a): 6.3333
 - orderNorm (ORQ): 6.3333
 - sqrt(x + a): 6.3333
 - Yeo-Johnson: 6.3333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized asinh(x) Transformation with 15 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 0.0119984 
 - sd (before standardization) = 0.01264569 
_____________________________________

## Results of bestNormalize for TEB ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 2.3333
 - Box-Cox: 1.2667
 - Center+scale: 6.0667
 - Exp(x): 18.0667
 - Log_b(x+a): 2.3333
 - orderNorm (ORQ): 0.4667
 - sqrt(x + a): 5.5333
 - Yeo-Johnson: 1.5333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
41.6 43.3 45.1 51.4 61.7 
_____________________________________

## Results of bestNormalize for ECEC ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.2667
 - Box-Cox: 1.2667
 - Center+scale: 6.0667
 - Exp(x): 18.0667
 - Log_b(x+a): 1.2667
 - orderNorm (ORQ): 0.4667
 - sqrt(x + a): 3.9333
 - Yeo-Johnson: 1.5333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
41.7 43.3 45.3 51.5 61.8 
_____________________________________

## Results of bestNormalize for minNO3 ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.8
 - Box-Cox: 1.2667
 - Center+scale: 3.9333
 - Exp(x): 21.2667
 - Log_b(x+a): 1.8
 - orderNorm (ORQ): 1.2667
 - sqrt(x + a): 1.8
 - Yeo-Johnson: 1.2667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized Box Cox Transformation with 15 nonmissing obs.:
 Estimated statistics:
 - lambda = -0.3344538 
 - mean (before standardization) = 1.907818 
 - sd (before standardization) = 0.2388776 
_____________________________________

## Results of bestNormalize for minTIN ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.8
 - Box-Cox: 1.8
 - Center+scale: 3.9333
 - Exp(x): 21.2667
 - Log_b(x+a): 1.8
 - orderNorm (ORQ): 0.7333
 - sqrt(x + a): 1.8
 - Yeo-Johnson: 1.8
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 14 unique values 
 - Original quantiles:
     0%     25%     50%     75%    100% 
  8.760  14.995  23.080  39.935 110.750 
_____________________________________

## Results of bestNormalize for DOC ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 2.3333
 - Box-Cox: 1.2667
 - Center+scale: 2.8667
 - Exp(x): 21.2667
 - Log_b(x+a): 2.3333
 - orderNorm (ORQ): 0.4667
 - sqrt(x + a): 2.8667
 - Yeo-Johnson: 1.2667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
     0%     25%     50%     75%    100% 
 30.800  44.070  56.680  64.385 118.740 
_____________________________________

## Results of bestNormalize for DOCN ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 2.3333
 - Box-Cox: 0.7333
 - Center+scale: 3.9333
 - Exp(x): 21.2667
 - Log_b(x+a): 2.3333
 - orderNorm (ORQ): 0.2
 - sqrt(x + a): 3.1333
 - Yeo-Johnson: 0.2
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 9.440 10.290 11.150 12.095 19.270 
_____________________________________

## Results of bestNormalize for micCN ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 3.1333
 - Box-Cox: 3.1333
 - Center+scale: 3.1333
 - Exp(x): 5.5333
 - Log_b(x+a): 3.1333
 - orderNorm (ORQ): 0.2
 - sqrt(x + a): 3.1333
 - Yeo-Johnson: 3.1333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 14 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
4.420 4.585 4.720 5.415 5.850 
_____________________________________

## Results of bestNormalize for BG_ase ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 2.8667
 - Box-Cox: 2.3333
 - Center+scale: 5
 - Exp(x): 21.2667
 - Log_b(x+a): 2.8667
 - orderNorm (ORQ): 0.4667
 - sqrt(x + a): 2.8667
 - Yeo-Johnson: 2.3333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 3.010  3.735  4.220  5.795 10.480 
_____________________________________

## Results of bestNormalize for BP_ase ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.5333
 - Box-Cox: 1.5333
 - Center+scale: 3.1333
 - Exp(x): 11.9333
 - Log_b(x+a): 0.7333
 - orderNorm (ORQ): 0.7333
 - sqrt(x + a): 3.1333
 - Yeo-Johnson: 1.5333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized Log_b(x + a) Transformation with 15 nonmissing obs.:
 Relevant statistics:
 - a = 0 
 - b = 10 
 - mean (before standardization) = 0.5217757 
 - sd (before standardization) = 0.2001498 
_____________________________________

## Results of bestNormalize for CE_ase ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1
 - Box-Cox: 0.7333
 - Center+scale: 1.8
 - Exp(x): 4.2
 - Log_b(x+a): 0.4667
 - orderNorm (ORQ): 1
 - sqrt(x + a): 1
 - Yeo-Johnson: 0.4667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized Log_b(x + a) Transformation with 15 nonmissing obs.:
 Relevant statistics:
 - a = 0 
 - b = 10 
 - mean (before standardization) = 0.03306978 
 - sd (before standardization) = 0.1652591 
_____________________________________

## Results of bestNormalize for P_ase ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 2.0667
 - Box-Cox: 1.2667
 - Center+scale: 3.6667
 - Exp(x): 21.2667
 - Log_b(x+a): 2.0667
 - orderNorm (ORQ): 0.2
 - sqrt(x + a): 2.3333
 - Yeo-Johnson: 1.2667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
13.020 15.130 19.210 24.455 44.040 
_____________________________________

## Results of bestNormalize for N_ase ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.2667
 - Box-Cox: 1.8
 - Center+scale: 3.1333
 - Exp(x): 17.8
 - Log_b(x+a): 1.2667
 - orderNorm (ORQ): 0.2
 - sqrt(x + a): 3.1333
 - Yeo-Johnson: 1.2667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
  0%  25%  50%  75% 100% 
2.57 2.93 3.58 5.04 9.43 
_____________________________________

## Results of bestNormalize for XY_ase ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 0.7333
 - Box-Cox: 0.7333
 - Center+scale: 2.8667
 - Exp(x): 6.3333
 - Log_b(x+a): 0.7333
 - orderNorm (ORQ): 0.2
 - sqrt(x + a): 1.8
 - Yeo-Johnson: 0.7333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 13 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
0.66 0.81 1.11 1.39 2.52 
_____________________________________

## Results of bestNormalize for BG_Q10 ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 2.3333
 - Box-Cox: 2.3333
 - Center+scale: 2.3333
 - Exp(x): 6.0667
 - Log_b(x+a): 2.3333
 - orderNorm (ORQ): 0.2
 - sqrt(x + a): 2.3333
 - Yeo-Johnson: 2.3333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 13 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
1.310 1.360 1.390 1.545 1.760 
_____________________________________

## Results of bestNormalize for BP_Q10 ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 2.6
 - Box-Cox: 2.6
 - Center+scale: 2.6
 - Exp(x): 2.6
 - Log_b(x+a): 2.6
 - orderNorm (ORQ): 0.7333
 - sqrt(x + a): 2.6
 - Yeo-Johnson: 2.3333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 11 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
1.340 1.415 1.440 1.530 1.740 
_____________________________________

## Results of bestNormalize for CO2 ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.5333
 - Box-Cox: 2.6
 - Center+scale: 11.9333
 - Exp(x): 21.2667
 - Log_b(x+a): 1.5333
 - orderNorm (ORQ): 1.2667
 - sqrt(x + a): 3.4
 - Yeo-Johnson: 2.0667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 14 unique values 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 2.080  4.370  5.270 10.585 40.470 
_____________________________________

## Results of bestNormalize for PX_ase ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 2.3333
 - Box-Cox: 1.2667
 - Center+scale: 7.6667
 - Exp(x): 21.2667
 - Log_b(x+a): 2.3333
 - orderNorm (ORQ): 0.2
 - sqrt(x + a): 2.8667
 - Yeo-Johnson: 1.2667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
     0%     25%     50%     75%    100% 
 81.250  95.565 119.750 218.370 339.600 
_____________________________________

## Results of bestNormalize for PX_Q10 ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 14.6
 - Box-Cox: 4.2
 - Center+scale: 21.2667
 - Exp(x): 21.2667
 - Log_b(x+a): 11.9333
 - orderNorm (ORQ): 2.0667
 - sqrt(x + a): 21.2667
 - Yeo-Johnson: 1.5333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized Yeo-Johnson Transformation with 15 nonmissing obs.:
 Estimated statistics:
 - lambda = -4.999946 
 - mean (before standardization) = 0.1968247 
 - sd (before standardization) = 0.001294011 
_____________________________________

## Results of bestNormalize for CUEcp ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 3.9333
 - Box-Cox: 1.5333
 - Center+scale: 3.9333
 - Exp(x): 3.9333
 - Log_b(x+a): 2.0667
 - orderNorm (ORQ): 1.5333
 - sqrt(x + a): 2.0667
 - Yeo-Johnson: 2.0667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized Box Cox Transformation with 15 nonmissing obs.:
 Estimated statistics:
 - lambda = -0.9999576 
 - mean (before standardization) = -4.229896 
 - sd (before standardization) = 0.9115063 
_____________________________________

## Results of bestNormalize for PUE ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 8.2
 - Box-Cox: 9.8
 - Center+scale: 8.2
 - Exp(x): 3.9333
 - Log_b(x+a): 9.8
 - orderNorm (ORQ): 2.0667
 - sqrt(x + a): 9.8
 - Yeo-Johnson: 9.8
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 7 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
0.770 0.880 0.900 0.905 0.920 
_____________________________________</code></pre>
</div>
</details>
<p>Great, now we can add the normalized transformed data back to our <code>mctoolsr</code> metadata file.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>ssu18_select_mc_norm <span class="ot">&lt;-</span> ssu18_select_mc</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> md_to_tranform) {</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>      tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_best_norm_test&quot;</span>)))</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>      tmp_new_data <span class="ot">&lt;-</span> tmp_get<span class="sc">$</span>x.t</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>      ssu18_select_mc_norm<span class="sc">$</span>map_loaded[[i]] <span class="ot">&lt;-</span> tmp_new_data</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>}     </span></code></pre></div>
</details>
</div>
<p>And rerun the Shapiro Tests.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>temp_md_norm <span class="ot">&lt;-</span> ssu18_select_mc_norm<span class="sc">$</span>map_loaded</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>temp_md_norm[,<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>shap_results_norm <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">colnames</span>(temp_md_norm)) {</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  tmp_shap <span class="ot">&lt;-</span> <span class="fu">shapiro.test</span>(temp_md_norm[[i]])</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  tmp_p <span class="ot">&lt;-</span> <span class="fu">round</span>(tmp_shap<span class="sc">$</span>p.value, <span class="at">digits =</span> <span class="dv">5</span>)</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>  tmp_res <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">isTRUE</span>(tmp_shap<span class="sc">$</span>p.value <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  shap_results_norm <span class="ot">&lt;-</span> <span class="fu">rbind</span>(shap_results_norm, <span class="fu">data.frame</span>(i, tmp_p, tmp_res))</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(shap_results_norm) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;parameter&quot;</span>, <span class="st">&quot;p-value&quot;</span>, <span class="st">&quot;tranform&quot;</span>)</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>shap_results_norm</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;temp_md_norm&quot;</span>))</span></code></pre></div>
</details>
</div>
<p>And check if there are any parameters that are still significant for the normality test.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>shap_results<span class="sc">$</span>parameter[shap_results_norm<span class="sc">$</span>tranform <span class="sc">==</span> <span class="cn">TRUE</span>]</span></code></pre></div>
<pre><code>[1] &quot;Al&quot; &quot;Fe&quot;</code></pre>
</div>
<p>Ok. Looks like <code>bestNormalize</code> was unable to find a suitable transformation for <code>Al</code> and <code>Fe</code>. This is likely because there is very little variation in these metadata and/or there are too few significant digits.</p>
<h2 id="normalized-metadata">Normalized Metadata</h2>
<p>Finally, here is a new summary table that includes all of the normalized data.</p>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="ffn29i70ljbq7tir2u6u" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="ffn29i70ljbq7tir2u6u">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Metadata summary for each sample.<\/em>\n<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15"],["P10_D00_010_C0E","P02_D00_010_C0A","P04_D00_010_C0B","P06_D00_010_C0C","P08_D00_010_C0D","P01_D00_010_W3A","P03_D00_010_W3B","P05_D00_010_W3C","P07_D00_010_W3D","P09_D00_010_W3E","P01_D00_010_W8A","P03_D00_010_W8B","P05_D00_010_W8C","P07_D00_010_W8D","P09_D00_010_W8E"],["P10","P02","P04","P06","P08","P01","P03","P05","P07","P09","P01","P03","P05","P07","P09"],["C","C","C","C","C","W","W","W","W","W","W","W","W","W","W"],[0,0,0,0,0,3,3,3,3,3,8,8,8,8,8],["E","A","B","C","D","A","B","C","D","E","A","B","C","D","E"],["W","W","W","W","W","W","W","W","W","W","W","W","W","W","W"],[26,26,26,26,26,29,29,29,29,29,34,34,34,34,34],[-1.2815515655446,-0.967421566101701,-0.524400512708041,-0.727913290881644,-1.83391463581591,0.340694827087795,0.167894004788105,-0.340694827087795,-0.167894004788105,0,0.727913290881644,1.83391463581591,1.2815515655446,0.524400512708041,0.967421566101701],[0,0.340694827087795,0.524400512708041,0.167894004788105,1.83391463581591,0.967421566101701,1.2815515655446,-0.524400512708041,0.727913290881644,-0.167894004788105,-0.727913290881644,-0.967421566101701,-1.2815515655446,-1.83391463581591,-0.340694827087795],[0.26,0.3367,0.3033,0.3333,0.2433,0.4267,0.32,0.3333,0.4367,0.2833,0.4267,0.32,0.3333,0.4367,0.2833],[0.32,0.43,0.33,0.3,0.27,0.35,0.37,0.33,0.35,0.36,0.35,0.37,0.33,0.35,0.36],[1.69063798084328,0.483091169498865,-0.72457638344738,-0.72457638344738,-0.72457638344738,-0.72457638344738,0.483091169498865,-0.72457638344738,-0.72457638344738,1.69063798084328,-0.72457638344738,0.483091169498865,-0.72457638344738,-0.72457638344738,1.69063798084328],[0.967421566101701,0.727913290881644,0,0.524400512708041,-1.83391463581591,0.2533471031358,-0.622925723210088,-1.11077161663679,-0.2533471031358,1.50108594604402,0.2533471031358,-0.622925723210088,-1.11077161663679,-0.2533471031358,1.50108594604402],[-0.158043327838275,-0.158043327838275,0.632647390072185,-0.94881310625632,-0.94881310625632,-0.158043327838275,-0.158043327838275,-0.94881310625632,-0.158043327838275,2.21347582882965,-0.158043327838275,-0.158043327838275,-0.94881310625632,-0.158043327838275,2.21347582882965],[1.38,0.69,1.18,0.57,1.86,1.72,0.85,0.67,1.6,1.09,1.72,0.85,0.67,1.6,1.09],[21.43,8.16,14.59,21.32,14.93,9.98,16.51,14.83,18.87,20.64,9.98,16.51,14.83,18.87,20.64],[0.07,0.04,0.12,0.06,0.02,0.04,0.06,0.05,0.07,0.08,0.04,0.06,0.05,0.07,0.08],[0.13,0.11,0.14,0.14,0.08,0.15,0.12,0,0.1,0.19,0.15,0.12,0,0.1,0.19],[0.967421566101701,0.167894004788105,0,0.727913290881644,-0.967421566101701,-0.622925723210088,-0.2533471031358,-1.50108594604402,0.430727299295457,1.50108594604402,-0.622925723210088,-0.2533471031358,-1.50108594604402,0.430727299295457,1.50108594604402],[0.967421566101701,0.167894004788105,0,0.727913290881644,-0.967421566101701,-0.622925723210088,-0.2533471031358,-1.50108594604402,0.430727299295457,1.50108594604402,-0.622925723210088,-0.2533471031358,-1.50108594604402,0.430727299295457,1.50108594604402],[6.28,6.18,5.73,5.93,6.01,6.01,5.86,5.87,6.21,6.1,6.01,5.86,5.87,6.21,6.1],[4.08,1.42,1.5,0.75,3.67,1.33,2.83,1.17,1,2.58,0.12,0.2,0.11,0.02,0],[4.75,3.92,3.17,1.83,4.33,3.67,3.33,2.17,2.42,3.5,1.65,1.28,1.18,2.6,0.92],[0.800075495378716,0.225910256500609,-0.0718237473488426,-0.0913033482304395,1.90256471255424,1.0808166102378,0.958426341448772,-1.04084844177283,-0.0152893922003602,-0.987240048760923,-1.03094652082699,0.125567199190938,0.775699874530057,-0.875578712682559,-1.75603027801819],[0.727913290881644,0.340694827087795,-0.0836517339071291,-0.340694827087795,1.83391463581591,1.2815515655446,0.967421566101701,-0.967421566101701,-0.0836517339071291,-0.524400512708041,-1.2815515655446,0.167894004788105,0.524400512708041,-0.727913290881644,-1.83391463581591],[3.44,4.81,5.57,2.92,2.04,8.03,3.11,2.32,4.28,2.03,4.08,2.5,3.91,8.19,6.5],[4.53,3.13,3.22,2.27,3.71,4.67,2.57,3.31,5.07,2.9,2.65,1.06,2.3,6.39,2.76],[9.45,7.58,7.95,7.04,5.97,9.52,8.66,7.3,9.15,8.28,8.55,2.8,4.25,5.62,6.08],[0.524400512708041,1.2815515655446,0.167894004788105,0,-0.340694827087795,1.83391463581591,0.340694827087795,0.727913290881644,0.967421566101701,-0.727913290881644,-0.967421566101701,-1.2815515655446,-0.167894004788105,-0.524400512708041,-1.83391463581591],[4.91,8.22,6.32,5.42,5.22,11.66,6.48,7.03,8.47,5.56,4.2,2.5,4.3,3.7,1.6],[1.2815515655446,-1.83391463581591,-0.340694827087795,-0.167894004788105,0.167894004788105,0,-0.727913290881644,0.340694827087795,-0.967421566101701,0.727913290881644,-1.2815515655446,0.967421566101701,-0.524400512708041,0.524400512708041,1.83391463581591],[334.93,444.72,440.6,390.66,277.37,690.11,364.62,284.5,396.42,392.56,469,117,440,676,416],[72.64,110.13,86.19,72.61,56.55,135.48,66.4,48.79,73.73,62.37,84,20,97,147,91],[64.23,65.03,81.45,85.7,33.4,87.98,70.68,52.5,93.13,66.65,55,11,56,94,63],[-1.2815515655446,-1.83391463581591,0.340694827087795,0.524400512708041,-0.0836517339071291,-0.340694827087795,0.167894004788105,0.727913290881644,-0.0836517339071291,1.2815515655446,0.967421566101701,1.83391463581591,-0.967421566101701,-0.524400512708041,-0.727913290881644],[5.54,8.13,5.39,4.5,11.55,8.61,6.7,8.33,4.66,7.34,8.48,10.54,7.91,7.18,6.6],[1.17,2.37,1.09,0.83,2.45,1.74,1.31,1.39,0.98,1.19,1.52,1.8,1.74,1.56,1.44],[0.28,0.24,0.22,0.21,0.15,0.26,0.33,0.34,0.32,0.2,0.15,0.02,0.15,0.39,0.37],[0.524400512708041,0,0.340694827087795,-0.967421566101701,-0.727913290881644,-0.340694827087795,0.167894004788105,-1.83391463581591,-0.167894004788105,-0.524400512708041,0.727913290881644,-1.2815515655446,1.2815515655446,0.967421566101701,1.83391463581591],[-0.986727244118644,-0.58427243563632,-0.208687672444211,-0.771521136656039,-1.36798756847004,0.111378057987956,-0.187239258527842,-0.762228416356262,0.0801567010134151,-0.837708961586596,1.02805739308463,-0.304161649328782,1.70050424243836,1.64923375869141,1.44120418990896],[-0.596465321572654,0.0263629704507989,0.0263629704507989,-0.956125913417099,-1.213614603327,-0.200108652559654,0.407243931999038,-0.921317974355604,-0.0223043644559385,-0.596465321572654,1.71186069793441,-1.33218934661468,0.865437830030367,0.917478361853037,1.88384473515684],[-0.967421566101701,-0.727913290881644,0.167894004788105,-1.2815515655446,-1.83391463581591,0.340694827087795,0,-0.524400512708041,0.524400512708041,-0.340694827087795,0.727913290881644,-0.167894004788105,1.83391463581591,0.967421566101701,1.2815515655446],[-0.340694827087795,-0.167894004788105,0.167894004788105,-1.83391463581591,-0.524400512708041,0.967421566101701,0,-1.2815515655446,0.340694827087795,-0.967421566101701,0.524400512708041,-0.727913290881644,0.727913290881644,1.2815515655446,1.83391463581591],[4.54,3.8,4.92,5.45,3.56,6.08,5.72,3.2,5.86,5.26,6.7,1.28,5.57,10.68,11.78],[-0.967421566101701,-0.2533471031358,0.0836517339071291,-1.83391463581591,-1.2815515655446,0.340694827087795,0.524400512708041,-0.524400512708041,0.0836517339071291,-0.2533471031358,1.83391463581591,-0.727913290881644,0.967421566101701,0.727913290881644,1.2815515655446],[1.54,1.51,1.51,1.83,2.08,1.77,2.48,1.48,1.72,1.67,2.01,1.49,1.73,1.31,1.93],[-0.340694827087795,-1.50108594604402,0.340694827087795,-1.50108594604402,1.83391463581591,-0.167894004788105,-0.622925723210088,-0.967421566101701,0.167894004788105,-0.622925723210088,1.2815515655446,0.727913290881644,0.967421566101701,0,0.524400512708041],[-0.727913290881644,-1.83391463581591,1.2815515655446,0.0836517339071291,1.83391463581591,-1.11077161663679,-0.340694827087795,-1.11077161663679,0.340694827087795,0.967421566101701,0.524400512708041,0.727913290881644,-0.340694827087795,-0.340694827087795,0.0836517339071291],[1.81,1.89,2.3,2.36,2,2.03,1.99,1.88,1.76,1.81,1.87,1.9,1.95,1.96,2.11],[1.27,1.22,1.37,1.24,1.48,1.37,1.27,1.28,1.28,1.35,1.39,1.51,1.48,1.33,1.38],[1.47,1.54,1.85,1.6,1.69,1.46,1.54,1.51,1.36,1.48,1.56,1.75,1.68,1.54,1.46],[1.33,1.23,1.39,1.47,1.48,1.3,1.27,1.3,1.3,1.32,1.65,1.39,1.6,1.51,1.5],[1.65,1.59,1.76,1.63,1.83,1.72,1.72,1.82,1.63,1.84,1.86,1.89,1.88,1.91,1.76],[-0.167894004788105,0,-1.2815515655446,-0.841621233572914,-1.83391463581591,0.727913290881644,-0.524400512708041,-0.340694827087795,0.167894004788105,0.524400512708041,-0.841621233572914,1.83391463581591,1.2815515655446,0.340694827087795,0.967421566101701],[32.22,38.61,29.51,27.78,21.96,52.8,32.1,24.11,42.4,30.37,40.69,13.94,32.65,57.13,53.79],[1.8,1.81,2.14,1.85,2.07,1.76,2,1.79,1.96,1.9,1.71,1.77,1.67,1.4,1.79],[-0.167894004788105,-0.727913290881644,-1.83391463581591,-0.967421566101701,-1.2815515655446,0.340694827087795,0,-0.340694827087795,-0.524400512708041,0.167894004788105,0.524400512708041,0.727913290881644,0.967421566101701,1.2815515655446,1.83391463581591],[-0.988344269843095,1.10338097153064,-0.2852713152544,0.964385838976133,2.3788150796016,-0.410973475656995,0.554298233213666,0.20471700316793,-0.90899688317639,0.473407676311783,-1.61136909739316,-0.90899688317639,-0.347276912447769,-0.108887982926752,-0.108887982926752],[0.22,0.22,0.25,0.27,0.24,0.32,0.26,0.27,0.27,0.26,0.23,0.29,0.19,0.26,0.24],[-1.11885251957508,0.51359672617582,-0.0362830420472531,-1.11885251957508,0.751045552546806,0.51359672617582,-0.357045199875853,-0.357045199875853,-1.57593516019186,0.51359672617582,-0.357045199875853,2.41320231511534,0.967847073188575,-0.71554323631409,-0.0362830420472531],[0.87,0.87,0.85,0.83,0.85,0.77,0.84,0.83,0.83,0.84,0.87,0.81,0.9,0.84,0.85],[1.2815515655446,-0.622925723210088,0.2533471031358,1.2815515655446,-0.622925723210088,-0.622925723210088,0.2533471031358,0.2533471031358,1.2815515655446,-0.622925723210088,0.2533471031358,-1.83391463581591,-1.2815515655446,0.727913290881644,-0.167894004788105],[-2.46,0.14,-2.63,-0.13,0,0.07,-0.29,0.81,0.65,-0.24,2.19,0.63,0.94,1.48,0.96],[1.29,1.44,1.25,1.42,1.54,1.51,1.49,1.77,1.61,1.51,2.3,1.72,2.08,1.81,1.82],[0.5231,0.5024,0.555,0.4624,0.5058,0.4614,0.5565,0.4877,0.4941,0.5112,0.6456,0.5302,0.6931,0.5933,0.6515],[0.6607,0.6554,0.6173,0.5556,0.6037,0.5911,0.6412,0.5596,0.5762,0.6064,0.7166,0.4918,0.6412,0.6628,0.7218],[1.263,1.3044,1.1123,1.2017,1.1935,1.2812,1.1522,1.1473,1.1662,1.1862,1.11,0.9277,0.9251,1.1173,1.1079]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>id<\/th>\n      <th>PLOT<\/th>\n      <th>TREAT<\/th>\n      <th>TREAT_T<\/th>\n      <th>PAIR<\/th>\n      <th>SEASON<\/th>\n      <th>ATAP<\/th>\n      <th>AST<\/th>\n      <th>H2O<\/th>\n      <th>N<\/th>\n      <th>P<\/th>\n      <th>Al<\/th>\n      <th>Ca<\/th>\n      <th>Fe<\/th>\n      <th>K<\/th>\n      <th>Mg<\/th>\n      <th>Mn<\/th>\n      <th>Na<\/th>\n      <th>TEB<\/th>\n      <th>ECEC<\/th>\n      <th>pH<\/th>\n      <th>minPO4<\/th>\n      <th>minNH4<\/th>\n      <th>minNO3<\/th>\n      <th>minTIN<\/th>\n      <th>NH4<\/th>\n      <th>NO3<\/th>\n      <th>PO4<\/th>\n      <th>DOC<\/th>\n      <th>DON<\/th>\n      <th>DOCN<\/th>\n      <th>micC<\/th>\n      <th>micN<\/th>\n      <th>micP<\/th>\n      <th>micCN<\/th>\n      <th>micCP<\/th>\n      <th>micNP<\/th>\n      <th>AG_ase<\/th>\n      <th>BG_ase<\/th>\n      <th>BP_ase<\/th>\n      <th>CE_ase<\/th>\n      <th>P_ase<\/th>\n      <th>N_ase<\/th>\n      <th>S_ase<\/th>\n      <th>XY_ase<\/th>\n      <th>AG_Q10<\/th>\n      <th>BG_Q10<\/th>\n      <th>BP_Q10<\/th>\n      <th>CE_Q10<\/th>\n      <th>P_Q10<\/th>\n      <th>N_Q10<\/th>\n      <th>S_Q10<\/th>\n      <th>XY_Q10<\/th>\n      <th>CO2<\/th>\n      <th>LP_ase<\/th>\n      <th>LP_Q10<\/th>\n      <th>PX_ase<\/th>\n      <th>PX_Q10<\/th>\n      <th>CUEcn<\/th>\n      <th>CUEcp<\/th>\n      <th>NUE<\/th>\n      <th>PUE<\/th>\n      <th>Tmin<\/th>\n      <th>SI<\/th>\n      <th>enzCN<\/th>\n      <th>enzCP<\/th>\n      <th>enzNP<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"targets":8,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":9,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":10,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":12,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":13,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":14,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":19,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":20,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":24,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":25,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":29,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":31,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":35,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":39,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":40,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":41,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":42,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":43,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":45,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":47,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":48,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":54,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":57,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":58,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":60,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":62,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":65,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":66,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":67,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"className":"dt-right","targets":[4,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\nvar value=data[8]; $(this.api().cell(row, 8).node()).css({'font-size':'80%'});\nvar value=data[9]; $(this.api().cell(row, 9).node()).css({'font-size':'80%'});\nvar value=data[10]; $(this.api().cell(row, 10).node()).css({'font-size':'80%'});\nvar value=data[11]; $(this.api().cell(row, 11).node()).css({'font-size':'80%'});\nvar value=data[12]; $(this.api().cell(row, 12).node()).css({'font-size':'80%'});\nvar value=data[13]; $(this.api().cell(row, 13).node()).css({'font-size':'80%'});\nvar value=data[14]; $(this.api().cell(row, 14).node()).css({'font-size':'80%'});\nvar value=data[15]; $(this.api().cell(row, 15).node()).css({'font-size':'80%'});\nvar value=data[16]; $(this.api().cell(row, 16).node()).css({'font-size':'80%'});\nvar value=data[17]; $(this.api().cell(row, 17).node()).css({'font-size':'80%'});\nvar value=data[18]; $(this.api().cell(row, 18).node()).css({'font-size':'80%'});\nvar value=data[19]; $(this.api().cell(row, 19).node()).css({'font-size':'80%'});\nvar value=data[20]; $(this.api().cell(row, 20).node()).css({'font-size':'80%'});\nvar value=data[21]; $(this.api().cell(row, 21).node()).css({'font-size':'80%'});\nvar value=data[22]; $(this.api().cell(row, 22).node()).css({'font-size':'80%'});\nvar value=data[23]; $(this.api().cell(row, 23).node()).css({'font-size':'80%'});\nvar value=data[24]; $(this.api().cell(row, 24).node()).css({'font-size':'80%'});\nvar value=data[25]; $(this.api().cell(row, 25).node()).css({'font-size':'80%'});\nvar value=data[26]; $(this.api().cell(row, 26).node()).css({'font-size':'80%'});\nvar value=data[27]; $(this.api().cell(row, 27).node()).css({'font-size':'80%'});\nvar value=data[28]; $(this.api().cell(row, 28).node()).css({'font-size':'80%'});\nvar value=data[29]; $(this.api().cell(row, 29).node()).css({'font-size':'80%'});\nvar value=data[30]; $(this.api().cell(row, 30).node()).css({'font-size':'80%'});\nvar value=data[31]; $(this.api().cell(row, 31).node()).css({'font-size':'80%'});\nvar value=data[32]; $(this.api().cell(row, 32).node()).css({'font-size':'80%'});\nvar value=data[33]; $(this.api().cell(row, 33).node()).css({'font-size':'80%'});\nvar value=data[34]; $(this.api().cell(row, 34).node()).css({'font-size':'80%'});\nvar value=data[35]; $(this.api().cell(row, 35).node()).css({'font-size':'80%'});\nvar value=data[36]; $(this.api().cell(row, 36).node()).css({'font-size':'80%'});\nvar value=data[37]; $(this.api().cell(row, 37).node()).css({'font-size':'80%'});\nvar value=data[38]; $(this.api().cell(row, 38).node()).css({'font-size':'80%'});\nvar value=data[39]; $(this.api().cell(row, 39).node()).css({'font-size':'80%'});\nvar value=data[40]; $(this.api().cell(row, 40).node()).css({'font-size':'80%'});\nvar value=data[41]; $(this.api().cell(row, 41).node()).css({'font-size':'80%'});\nvar value=data[42]; $(this.api().cell(row, 42).node()).css({'font-size':'80%'});\nvar value=data[43]; $(this.api().cell(row, 43).node()).css({'font-size':'80%'});\nvar value=data[44]; $(this.api().cell(row, 44).node()).css({'font-size':'80%'});\nvar value=data[45]; $(this.api().cell(row, 45).node()).css({'font-size':'80%'});\nvar value=data[46]; $(this.api().cell(row, 46).node()).css({'font-size':'80%'});\nvar value=data[47]; $(this.api().cell(row, 47).node()).css({'font-size':'80%'});\nvar value=data[48]; $(this.api().cell(row, 48).node()).css({'font-size':'80%'});\nvar value=data[49]; $(this.api().cell(row, 49).node()).css({'font-size':'80%'});\nvar value=data[50]; $(this.api().cell(row, 50).node()).css({'font-size':'80%'});\nvar value=data[51]; $(this.api().cell(row, 51).node()).css({'font-size':'80%'});\nvar value=data[52]; $(this.api().cell(row, 52).node()).css({'font-size':'80%'});\nvar value=data[53]; $(this.api().cell(row, 53).node()).css({'font-size':'80%'});\nvar value=data[54]; $(this.api().cell(row, 54).node()).css({'font-size':'80%'});\nvar value=data[55]; $(this.api().cell(row, 55).node()).css({'font-size':'80%'});\nvar value=data[56]; $(this.api().cell(row, 56).node()).css({'font-size':'80%'});\nvar value=data[57]; $(this.api().cell(row, 57).node()).css({'font-size':'80%'});\nvar value=data[58]; $(this.api().cell(row, 58).node()).css({'font-size':'80%'});\nvar value=data[59]; $(this.api().cell(row, 59).node()).css({'font-size':'80%'});\nvar value=data[60]; $(this.api().cell(row, 60).node()).css({'font-size':'80%'});\nvar value=data[61]; $(this.api().cell(row, 61).node()).css({'font-size':'80%'});\nvar value=data[62]; $(this.api().cell(row, 62).node()).css({'font-size':'80%'});\nvar value=data[63]; $(this.api().cell(row, 63).node()).css({'font-size':'80%'});\nvar value=data[64]; $(this.api().cell(row, 64).node()).css({'font-size':'80%'});\nvar value=data[65]; $(this.api().cell(row, 65).node()).css({'font-size':'80%'});\nvar value=data[66]; $(this.api().cell(row, 66).node()).css({'font-size':'80%'});\nvar value=data[67]; $(this.api().cell(row, 67).node()).css({'font-size':'80%'});\n}"}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render","options.columnDefs.5.render","options.columnDefs.6.render","options.columnDefs.7.render","options.columnDefs.8.render","options.columnDefs.9.render","options.columnDefs.10.render","options.columnDefs.11.render","options.columnDefs.12.render","options.columnDefs.13.render","options.columnDefs.14.render","options.columnDefs.15.render","options.columnDefs.16.render","options.columnDefs.17.render","options.columnDefs.18.render","options.columnDefs.19.render","options.columnDefs.20.render","options.columnDefs.21.render","options.columnDefs.22.render","options.columnDefs.23.render","options.columnDefs.24.render","options.columnDefs.25.render","options.columnDefs.26.render","options.columnDefs.27.render","options.columnDefs.28.render","options.rowCallback"],"jsHooks":[]}</script>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="autocorrelation-tests">Autocorrelation Tests</h2>
<p>Next, we test the metadata for autocorrelations. Do we do this on the original data or the transformed data? No idea, so lets do both.</p>
<h3 id="split-metadata">Split Metadata</h3>
<p>We need to split the data into different groups.</p>
<ol type="A">
<li>Environmental and edaphic properties</li>
<li>Microbial functional responses</li>
<li>Temperature adaptation properties</li>
</ol>
<p>We first create lists of metadata parameters.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>div <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;PLOT&quot;</span>, <span class="st">&quot;TREAT&quot;</span>, <span class="st">&quot;TREAT_T&quot;</span>, <span class="st">&quot;PAIR&quot;</span>, <span class="st">&quot;Observed&quot;</span>, <span class="st">&quot;Shannon_exp&quot;</span>, </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>         <span class="st">&quot;InvSimpson&quot;</span>, <span class="st">&quot;ATAP&quot;</span>)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>edaphic <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;AST&quot;</span>, <span class="st">&quot;H2O&quot;</span>, <span class="st">&quot;N&quot;</span>, <span class="st">&quot;P&quot;</span>, <span class="st">&quot;Al&quot;</span>, </span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;Ca&quot;</span>, <span class="st">&quot;Fe&quot;</span>, <span class="st">&quot;K&quot;</span>, <span class="st">&quot;Mg&quot;</span>, <span class="st">&quot;Mn&quot;</span>, <span class="st">&quot;Na&quot;</span>, <span class="st">&quot;TEB&quot;</span>, <span class="st">&quot;ECEC&quot;</span>, <span class="st">&quot;pH&quot;</span>, </span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;NH4&quot;</span>, <span class="st">&quot;NO3&quot;</span>, <span class="st">&quot;PO4&quot;</span>, <span class="st">&quot;DOC&quot;</span>, <span class="st">&quot;DON&quot;</span>, <span class="st">&quot;DOCN&quot;</span>)</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>soil_funct <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;micC&quot;</span>, <span class="st">&quot;micN&quot;</span>, <span class="st">&quot;micP&quot;</span>, <span class="st">&quot;micCN&quot;</span>, <span class="st">&quot;micCP&quot;</span>, <span class="st">&quot;micNP&quot;</span>, </span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot;AG_ase&quot;</span>, <span class="st">&quot;BG_ase&quot;</span>, <span class="st">&quot;BP_ase&quot;</span>, <span class="st">&quot;CE_ase&quot;</span>, <span class="st">&quot;P_ase&quot;</span>, <span class="st">&quot;N_ase&quot;</span>, </span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot;S_ase&quot;</span>, <span class="st">&quot;XY_ase&quot;</span>,  <span class="st">&quot;LP_ase&quot;</span>, <span class="st">&quot;PX_ase&quot;</span>, <span class="st">&quot;CO2&quot;</span>, </span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot;enzCN&quot;</span>, <span class="st">&quot;enzCP&quot;</span>, <span class="st">&quot;enzNP&quot;</span>)</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>temp_adapt <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;AG_Q10&quot;</span>, <span class="st">&quot;BG_Q10&quot;</span>, <span class="st">&quot;BP_Q10&quot;</span>, <span class="st">&quot;CE_Q10&quot;</span>, <span class="st">&quot;P_Q10&quot;</span>, <span class="st">&quot;N_Q10&quot;</span>, </span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot;S_Q10&quot;</span>, <span class="st">&quot;XY_Q10&quot;</span>, <span class="st">&quot;LP_Q10&quot;</span>, <span class="st">&quot;PX_Q10&quot;</span>, <span class="st">&quot;CUEcn&quot;</span>, <span class="st">&quot;CUEcp&quot;</span>, </span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot;NUE&quot;</span>,<span class="st">&quot;PUE&quot;</span>, <span class="st">&quot;Tmin&quot;</span>, <span class="st">&quot;SI&quot;</span>)</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>md_groups <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;edaphic&quot;</span>, <span class="st">&quot;soil_funct&quot;</span>, <span class="st">&quot;temp_adapt&quot;</span>)</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a><span class="co"># NoT uSed: minPO4, minNH4, minNO3, minTIN</span></span></code></pre></div>
</details>
</div>
<p>And then use the lists to split the data sets by metadata group. Here, we do this for the original metadata and the metadata after normalization.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>select_md <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ssu18_select_mc&quot;</span>, <span class="st">&quot;ssu18_select_mc_norm&quot;</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> select_md) {</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>   <span class="co">#tmp_get &lt;- get(purrr::map_chr(i, ~ paste0(i, &quot;_mc&quot;)))</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>   tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>   tmp_md_all <span class="ot">&lt;-</span> tmp_get<span class="sc">$</span>map_loaded</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>   tmp_div <span class="ot">&lt;-</span> tmp_md_all <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(div))</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>   tmp_div <span class="ot">&lt;-</span> tmp_div <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="do">## edaphic      </span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>   tmp_sub_edaphic <span class="ot">&lt;-</span> tmp_md_all <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(edaphic))</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>   tmp_sub_edaphic <span class="ot">&lt;-</span> tmp_sub_edaphic <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>   tmp_edaphic <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_div, tmp_sub_edaphic, <span class="at">by =</span> <span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>   tmp_edaphic <span class="ot">&lt;-</span> tmp_edaphic <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a><span class="do">## soil_funct</span></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>   tmp_sub_soil_funct <span class="ot">&lt;-</span> tmp_md_all <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(soil_funct))</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>   tmp_sub_soil_funct <span class="ot">&lt;-</span> tmp_sub_soil_funct <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>   tmp_soil_funct <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_div, tmp_sub_soil_funct, <span class="at">by =</span> <span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>   tmp_soil_funct <span class="ot">&lt;-</span> tmp_soil_funct <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;SampleID&quot;</span>)      </span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a><span class="do">## temp_adapt</span></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>   tmp_sub_temp_adapt <span class="ot">&lt;-</span> tmp_md_all <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(temp_adapt))</span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>   tmp_sub_temp_adapt <span class="ot">&lt;-</span> tmp_sub_temp_adapt <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>   tmp_temp_adapt <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_div, tmp_sub_temp_adapt, <span class="at">by =</span> <span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>   tmp_temp_adapt <span class="ot">&lt;-</span> tmp_temp_adapt <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;SampleID&quot;</span>)      </span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a><span class="do">## combine</span></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>   tmp_list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">data_loaded =</span> ssu18_select_mc<span class="sc">$</span>data_loaded, </span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">map_loaded =</span> ssu18_select_mc<span class="sc">$</span>map_loaded, </span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxonomy_loaded =</span> ssu18_select_mc<span class="sc">$</span>taxonomy_loaded,</span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>                    <span class="at">edaphic =</span> tmp_edaphic, </span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">soil_funct =</span> tmp_soil_funct, </span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>                    <span class="at">temp_adapt =</span> tmp_temp_adapt)</span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a>   tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_split&quot;</span>))</span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a>   <span class="fu">print</span>(tmp_name)</span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a>   <span class="fu">assign</span>(tmp_name, tmp_list)</span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a>   <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<h3 id="generate-autocorrelation-plots">Generate Autocorrelation Plots</h3>
<p>A little housekeeping to get rid of parameters we dont need (e.g., plot number, pair, etc.).</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>edaphic_cor <span class="ot">&lt;-</span> ssu18_select_mc_split<span class="sc">$</span>edaphic</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>edaphic_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>edaphic_norm_cor <span class="ot">&lt;-</span> ssu18_select_mc_norm_split<span class="sc">$</span>edaphic</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>edaphic_norm_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>soil_funct_cor <span class="ot">&lt;-</span> ssu18_select_mc_split<span class="sc">$</span>soil_funct</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>soil_funct_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>soil_funct_norm_cor <span class="ot">&lt;-</span> ssu18_select_mc_norm_split<span class="sc">$</span>soil_funct</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>soil_funct_norm_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>temp_adapt_cor <span class="ot">&lt;-</span> ssu18_select_mc_split<span class="sc">$</span>temp_adapt</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>temp_adapt_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>temp_adapt_norm_cor <span class="ot">&lt;-</span> ssu18_select_mc_norm_split<span class="sc">$</span>temp_adapt</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>temp_adapt_norm_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span></code></pre></div>
</details>
</div>
<p>And finally the code to create the plots.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;_cor$&quot;</span>)) {</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  tmp_cormat <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">cor</span>(tmp_get), <span class="dv">2</span>)</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  tmp_melted_cormat <span class="ot">&lt;-</span> reshape2<span class="sc">::</span><span class="fu">melt</span>(tmp_cormat)</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  tmp_get_lower_tri <span class="ot">&lt;-</span> <span class="cf">function</span>(tmp_cormat){</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>    tmp_cormat[<span class="fu">upper.tri</span>(tmp_cormat)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(tmp_cormat)</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get upper triangle of the correlation matrix</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  tmp_get_upper_tri <span class="ot">&lt;-</span> <span class="cf">function</span>(tmp_cormat){</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>    tmp_cormat[<span class="fu">lower.tri</span>(tmp_cormat)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(tmp_cormat)</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  tmp_upper_tri <span class="ot">&lt;-</span> <span class="fu">tmp_get_upper_tri</span>(tmp_cormat)</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>  tmp_melted_cormat <span class="ot">&lt;-</span> reshape2<span class="sc">::</span><span class="fu">melt</span>(tmp_upper_tri, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="at">data =</span> tmp_melted_cormat, <span class="fu">aes</span>(<span class="at">x =</span> Var1, <span class="at">y =</span> Var2, <span class="at">fill =</span> value)) <span class="sc">+</span> </span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>()</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>  tmp_ggheatmap <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tmp_melted_cormat, <span class="fu">aes</span>(Var2, Var1, <span class="at">fill =</span> value)) <span class="sc">+</span></span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>(<span class="at">color =</span> <span class="st">&quot;white&quot;</span>) <span class="sc">+</span></span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> <span class="st">&quot;blue&quot;</span>, <span class="at">high =</span> <span class="st">&quot;red&quot;</span>, <span class="at">mid =</span> <span class="st">&quot;white&quot;</span>, </span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>                         <span class="at">midpoint =</span> <span class="dv">0</span>, <span class="at">limit =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="at">space =</span> <span class="st">&quot;Lab&quot;</span>, </span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>                         <span class="at">name=</span><span class="st">&quot;Pearson</span><span class="sc">\n</span><span class="st">Correlation&quot;</span>) <span class="sc">+</span></span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span> </span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, </span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">size =</span> <span class="dv">7</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">7</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_fixed</span>() <span class="sc">+</span> </span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="fu">aes</span>(Var2, Var1, <span class="at">label =</span> value), <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="fl">1.75</span>) <span class="sc">+</span></span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-36"><a href="#cb18-36" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-37"><a href="#cb18-37" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-38"><a href="#cb18-38" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">0</span>),</span>
<span id="cb18-39"><a href="#cb18-39" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.6</span>, <span class="fl">0.7</span>),</span>
<span id="cb18-40"><a href="#cb18-40" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.direction =</span> <span class="st">&quot;horizontal&quot;</span>) <span class="sc">+</span></span>
<span id="cb18-41"><a href="#cb18-41" aria-hidden="true" tabindex="-1"></a>      <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_colorbar</span>(<span class="at">barwidth =</span> <span class="dv">7</span>, <span class="at">barheight =</span> <span class="dv">1</span>,</span>
<span id="cb18-42"><a href="#cb18-42" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">title.position =</span> <span class="st">&quot;top&quot;</span>, </span>
<span id="cb18-43"><a href="#cb18-43" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">title.hjust =</span> <span class="fl">0.5</span>))</span>
<span id="cb18-44"><a href="#cb18-44" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_ggheatmap&quot;</span>))</span>
<span id="cb18-45"><a href="#cb18-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_ggheatmap)</span>
<span id="cb18-46"><a href="#cb18-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(tmp_name)</span>
<span id="cb18-47"><a href="#cb18-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb18-48"><a href="#cb18-48" aria-hidden="true" tabindex="-1"></a>}  </span>
<span id="cb18-49"><a href="#cb18-49" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;_ggheatmap&quot;</span>)</span></code></pre></div>
</details>
</div>
<h2 id="autocorrelation-plots">Autocorrelation Plots</h2>
<p><br/></p>
<div class="l-body-outset">
<div class="panelset">
<section id="edaphic-properties" class="panel">
<h4>Edaphic properties</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="metadata_files/figure-html5/unnamed-chunk-24-1.png" width="624" /></p>
</div>
</section>
<section id="edaphic-properties-normalized" class="panel">
<h4>Edaphic properties (normalized)</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="metadata_files/figure-html5/unnamed-chunk-25-1.png" width="624" /></p>
</div>
</section>
<section id="functional-responses" class="panel">
<h4>Functional responses</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="metadata_files/figure-html5/unnamed-chunk-26-1.png" width="624" /></p>
</div>
</section>
<section id="functional-responses-normalized" class="panel">
<h4>Functional responses (normalized)</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="metadata_files/figure-html5/unnamed-chunk-27-1.png" width="624" /></p>
</div>
</section>
<section id="temperature-adaptation" class="panel">
<h4>Temperature adaptation</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="metadata_files/figure-html5/unnamed-chunk-28-1.png" width="624" /></p>
</div>
</section>
<section id="temperature-adaptation-normalized" class="panel">
<h4>Temperature adaptation (normalized)</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="metadata_files/figure-html5/unnamed-chunk-29-1.png" width="624" /></p>
</div>
</section>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-page">

</div>
<p>Now we can remove parameters based on the autocorrelation analysis:</p>
<ol type="A">
<li>Environmental and edaphic properties: TEB, DON, Na, Al, Ca.</li>
<li>Microbial functional responses: micN, micNP, enzCN, enzCP, BP_ase, CE_ase, LP_ase, N_ase, P_ase.</li>
<li>Temperature adaptation properties: NUE, PUE, SI.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>edaphic_remove <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;TEB&quot;</span>, <span class="st">&quot;DON&quot;</span>, <span class="st">&quot;Na&quot;</span>, <span class="st">&quot;Al&quot;</span>, <span class="st">&quot;Ca&quot;</span>)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>soil_funct_remove <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;micN&quot;</span>, <span class="st">&quot;micNP&quot;</span>, <span class="st">&quot;enzCN&quot;</span>, <span class="st">&quot;enzCP&quot;</span>, <span class="st">&quot;BP_ase&quot;</span>, </span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&quot;CE_ase&quot;</span>, <span class="st">&quot;LP_ase&quot;</span>, <span class="st">&quot;N_ase&quot;</span>, <span class="st">&quot;P_ase&quot;</span>)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>temp_adapt_remove <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;NUE&quot;</span>, <span class="st">&quot;PUE&quot;</span>, <span class="st">&quot;SI&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>tmp_df <span class="ot">&lt;-</span> ssu18_select_mc_split</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>edaphic <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>edaphic[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>edaphic) <span class="sc">%in%</span>  edaphic_remove]</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>soil_funct <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>soil_funct[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>soil_funct) <span class="sc">%in%</span>  soil_funct_remove]</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>temp_adapt <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>temp_adapt[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>temp_adapt) <span class="sc">%in%</span>  temp_adapt_remove]</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>ssu18_select_mc_split_no_ac <span class="ot">&lt;-</span> tmp_df</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>tmp_df <span class="ot">&lt;-</span> ssu18_select_mc_norm_split</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>edaphic <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>edaphic[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>edaphic) <span class="sc">%in%</span>  edaphic_remove]</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>soil_funct <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>soil_funct[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>soil_funct) <span class="sc">%in%</span>  soil_funct_remove]</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>temp_adapt <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>temp_adapt[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>temp_adapt) <span class="sc">%in%</span>  temp_adapt_remove]</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>ssu18_select_mc_norm_split_no_ac <span class="ot">&lt;-</span> tmp_df</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span></code></pre></div>
</details>
</div>
<h2 id="dissimilarity-correlation-tests">Dissimilarity Correlation Tests</h2>
<p>Lets see if any on the metadata groups are significantly correlated with the community data. Basically, we create distance matrices for the community data and each metadata group and then run Mantel tests for all comparisons. For the community data we calculate Bray-Curtis distances for the community data and Euclidean distances for the metadata. We use the function <code>mantel.test</code> from the <code>ape</code> package and <code>mantel</code> from the <code>vegan</code> package for the analyses.</p>
<p>In summary, we test both <code>mantel.test</code> and <code>mantel</code> on Bray-Curtis distance community distances against Euclidean distances for each metadata group (<code>edaphic</code>, <code>soil_funct</code>, <code>temp_adapt</code>) <strong>a</strong>) before normalizing and before removing autocorrelated parameters, <strong>b</strong>) before normalizing and after removing autocorrelated parameters, <strong>c</strong>) after normalizing and before removing autocorrelated parameters, and <strong>d</strong>) after normalizing and after removing autocorrelated parameters.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>man_df <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ssu18_select_mc_split&quot;</span>, <span class="st">&quot;ssu18_select_mc_split_no_ac&quot;</span>, </span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;ssu18_select_mc_norm_split&quot;</span>, <span class="st">&quot;ssu18_select_mc_norm_split_no_ac&quot;</span>)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> man_df) {</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  tmp_dm_otu <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(<span class="fu">t</span>(tmp_get<span class="sc">$</span>data_loaded), </span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>                             <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>                             <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a><span class="co"># EDAPHIC</span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>  tmp_dm_md_edaphic <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_get<span class="sc">$</span>edaphic[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_get<span class="sc">$</span>edaphic)], </span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">method =</span><span class="st">&quot;euclidean&quot;</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>  tmp_man1_edaphic <span class="ot">&lt;-</span> <span class="fu">mantel.test</span>(tmp_dm_otu, tmp_dm_md_edaphic, <span class="at">nperm =</span> <span class="dv">999</span>, </span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>                             <span class="at">graph =</span> <span class="cn">FALSE</span>, <span class="at">alternative =</span> <span class="st">&quot;two.sided&quot;</span>)</span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>  tmp_man2_edaphic <span class="ot">&lt;-</span> <span class="fu">mantel</span>(tmp_dm_otu, tmp_dm_md_edaphic, <span class="at">permutations =</span> <span class="dv">999</span>)</span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a><span class="co"># SOIL FUNCT</span></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>  tmp_dm_md_soil_funct <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_get<span class="sc">$</span>soil_funct[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_get<span class="sc">$</span>soil_funct)], </span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>  tmp_man1_soil_funct <span class="ot">&lt;-</span> <span class="fu">mantel.test</span>(tmp_dm_otu, tmp_dm_md_soil_funct, <span class="at">nperm =</span> <span class="dv">999</span>, </span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>                                <span class="at">graph =</span> <span class="cn">FALSE</span>, <span class="at">alternative =</span> <span class="st">&quot;two.sided&quot;</span>)</span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>  tmp_man2_soil_funct <span class="ot">&lt;-</span> <span class="fu">mantel</span>(tmp_dm_otu, tmp_dm_md_soil_funct, <span class="at">permutations =</span> <span class="dv">999</span>)</span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a><span class="co"># TEMP ADAPT</span></span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a>  tmp_dm_md_temp_adapt <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_get<span class="sc">$</span>temp_adapt[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_get<span class="sc">$</span>temp_adapt)], </span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb21-29"><a href="#cb21-29" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb21-30"><a href="#cb21-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-31"><a href="#cb21-31" aria-hidden="true" tabindex="-1"></a>  tmp_man1_temp_adapt <span class="ot">&lt;-</span> <span class="fu">mantel.test</span>(tmp_dm_otu, tmp_dm_md_temp_adapt, <span class="at">nperm =</span> <span class="dv">999</span>, </span>
<span id="cb21-32"><a href="#cb21-32" aria-hidden="true" tabindex="-1"></a>                                <span class="at">graph =</span> <span class="cn">FALSE</span>, <span class="at">alternative =</span> <span class="st">&quot;two.sided&quot;</span>)</span>
<span id="cb21-33"><a href="#cb21-33" aria-hidden="true" tabindex="-1"></a>  tmp_man2_temp_adapt <span class="ot">&lt;-</span> <span class="fu">mantel</span>(tmp_dm_otu, tmp_dm_md_temp_adapt, <span class="at">permutations =</span> <span class="dv">999</span>)</span>
<span id="cb21-34"><a href="#cb21-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-35"><a href="#cb21-35" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_mantel_tests&quot;</span>))</span>
<span id="cb21-36"><a href="#cb21-36" aria-hidden="true" tabindex="-1"></a>  tmp_df <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">edaphic_ape_man =</span> tmp_man1_edaphic, </span>
<span id="cb21-37"><a href="#cb21-37" aria-hidden="true" tabindex="-1"></a>                 <span class="at">edaphic_vegan_man =</span> tmp_man2_edaphic,</span>
<span id="cb21-38"><a href="#cb21-38" aria-hidden="true" tabindex="-1"></a>                 <span class="at">soil_funct_ape_man =</span> tmp_man1_soil_funct, </span>
<span id="cb21-39"><a href="#cb21-39" aria-hidden="true" tabindex="-1"></a>                 <span class="at">soil_funct_vegan_man =</span> tmp_man2_soil_funct,</span>
<span id="cb21-40"><a href="#cb21-40" aria-hidden="true" tabindex="-1"></a>                 <span class="at">temp_adapt_ape_man =</span> tmp_man1_temp_adapt, </span>
<span id="cb21-41"><a href="#cb21-41" aria-hidden="true" tabindex="-1"></a>                 <span class="at">temp_adapt_vegan_man =</span> tmp_man2_temp_adapt)</span>
<span id="cb21-42"><a href="#cb21-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_df)</span>
<span id="cb21-43"><a href="#cb21-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(tmp_name)</span>
<span id="cb21-44"><a href="#cb21-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb21-45"><a href="#cb21-45" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="dissimilarity-correlation-results">Dissimilarity Correlation Results</h3>
<p><br/></p>
<div class="layout-chunk" data-layout="l-page">
<table>
<colgroup>
<col style="width: 10%" />
<col style="width: 10%" />
<col style="width: 10%" />
<col style="width: 12%" />
<col style="width: 13%" />
<col style="width: 14%" />
<col style="width: 13%" />
<col style="width: 14%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">normalized?</th>
<th style="text-align: left;">AC removed?</th>
<th style="text-align: right;">edaphic_ape</th>
<th style="text-align: right;">edaphic_vegan</th>
<th style="text-align: right;">soil_funct_ape</th>
<th style="text-align: right;">soil_funct_vegan</th>
<th style="text-align: right;">temp_adapt_ape</th>
<th style="text-align: right;">temp_adapt_vegan</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">no</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">0.375</td>
<td style="text-align: right;">0.194</td>
<td style="text-align: right;">0.059</td>
<td style="text-align: right;">0.019</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">no</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">0.399</td>
<td style="text-align: right;">0.170</td>
<td style="text-align: right;">0.031</td>
<td style="text-align: right;">0.020</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="odd">
<td style="text-align: left;">yes</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: right;">0.537</td>
<td style="text-align: right;">0.250</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">yes</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">0.007</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: right;">0.541</td>
<td style="text-align: right;">0.288</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.001</td>
</tr>
</tbody>
</table>
</div>
<p>Moving on.</p>
<h2 id="best-subset-of-variables">Best Subset of Variables</h2>
<p>Now we want to know which of the metadata parameters are the most strongly correlated with the community data. For this we use the <code>bioenv</code> function from the <code>vegan</code> package. <code>bioenv</code><em>Best Subset of Environmental Variables with Maximum (Rank) Correlation with Community Dissimilarities</em>finds the best subset of environmental variables, so that the Euclidean distances of scaled environmental variables have the maximum (rank) correlation with community dissimilarities.</p>
<p>Next we will use the metadata set where autocorrelated parameters were removed and the remainder of the parameters were normalized (where applicable based on the Shapiro tests).</p>
<p>We run <code>bioenv</code> against the three groups of metadata parameters. We then run <code>bioenv</code> again, but this time against the individual parameters identified as significantly correlated.</p>
<h3 id="edaphic-properties-1">Edaphic Properties</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>tmp_env <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>edaphic)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>tmp_env[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>edaphic_bioenv <span class="ot">&lt;-</span> <span class="fu">bioenv</span>(<span class="fu">wisconsin</span>(tmp_comm), tmp_env, </span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>, <span class="at">index =</span> <span class="st">&quot;bray&quot;</span>, </span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">upto =</span> <span class="fu">ncol</span>(tmp_env), <span class="at">metric =</span> <span class="st">&quot;euclidean&quot;</span>)</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>bioenv_list <span class="ot">&lt;-</span> edaphic_bioenv<span class="sc">$</span>models[[edaphic_bioenv<span class="sc">$</span>whichbest]]<span class="sc">$</span>best</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>bioenv_best <span class="ot">&lt;-</span> <span class="fu">bioenvdist</span>(edaphic_bioenv, <span class="at">which =</span> <span class="st">&quot;best&quot;</span>)</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> bioenv_list) {</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>       tmp_dp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(edaphic_bioenv<span class="sc">$</span>x)</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>       tmp_md <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_dp[[i]], </span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>       tmp_man <span class="ot">&lt;-</span> <span class="fu">mantel</span>(bioenv_best, tmp_md, </span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>                         <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>)</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>       tmp_md_name <span class="ot">&lt;-</span> <span class="fu">names</span>(tmp_dp)[[i]]</span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>       tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(tmp_md_name, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_bioenv_mantel_test&quot;</span>))</span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_name, tmp_man)</span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;_bioenv_mantel_test&quot;</span>)</span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a>edaphic_bioenv_ind_mantel <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">AST =</span> AST_bioenv_mantel_test)</span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;_bioenv_mantel_test&quot;</span>))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
Call:
bioenv(comm = wisconsin(tmp_comm), env = tmp_env, method = &quot;spearman&quot;,      index = &quot;bray&quot;, upto = ncol(tmp_env), metric = &quot;euclidean&quot;) 

Subset of environmental variables with best correlation to community data.

Correlations:    spearman 
Dissimilarities: bray 
Metric:          euclidean 

Best model has 1 parameters (max. 15 allowed):
AST
with correlation  0.6808861 </code></pre>
</div>
<details>
<summary>
Show the results of individual edaphic metadata Mantel tests
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>$AST

Mantel statistic based on Spearman&#39;s rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = &quot;spearman&quot;,      permutations = 999) 

Mantel statistic r:     1 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.177 0.226 0.271 0.326 
Permutation: free
Number of permutations: 999</code></pre>
</div>
</details>
<p><code>bioenv</code> found the following edaphic properties significantly correlated with the community data: <strong>AST</strong></p>
<h3 id="soil-functional-response">Soil Functional Response</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>tmp_env <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>soil_funct)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>tmp_env[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>soil_funct_bioenv <span class="ot">&lt;-</span> <span class="fu">bioenv</span>(<span class="fu">wisconsin</span>(tmp_comm), tmp_env, </span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>, <span class="at">index =</span> <span class="st">&quot;bray&quot;</span>, </span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">upto =</span> <span class="fu">ncol</span>(tmp_env), <span class="at">metric =</span> <span class="st">&quot;euclidean&quot;</span>)</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>bioenv_list <span class="ot">&lt;-</span> soil_funct_bioenv<span class="sc">$</span>models[[soil_funct_bioenv<span class="sc">$</span>whichbest]]<span class="sc">$</span>best</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>bioenv_best <span class="ot">&lt;-</span> <span class="fu">bioenvdist</span>(soil_funct_bioenv, <span class="at">which =</span> <span class="st">&quot;best&quot;</span>)</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> bioenv_list) {</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>       tmp_dp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(soil_funct_bioenv<span class="sc">$</span>x)</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>       tmp_md <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_dp[[i]], </span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>       tmp_man <span class="ot">&lt;-</span> <span class="fu">mantel</span>(bioenv_best, tmp_md, </span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>                         <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>)</span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>       tmp_md_name <span class="ot">&lt;-</span> <span class="fu">names</span>(tmp_dp)[[i]]</span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>       tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(tmp_md_name, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_bioenv_mantel_test&quot;</span>))</span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_name, tmp_man)</span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;_bioenv_mantel_test&quot;</span>)</span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>soil_funct_bioenv_ind_mantel <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">AG_ase =</span> AG_ase_bioenv_mantel_test, </span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">enzNP =</span> enzNP_bioenv_mantel_test, </span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">S_ase =</span> S_ase_bioenv_mantel_test, </span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">PX_ase =</span> PX_ase_bioenv_mantel_test,</span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">XY_ase =</span> XY_ase_bioenv_mantel_test)</span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;_bioenv_mantel_test&quot;</span>))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
Call:
bioenv(comm = wisconsin(tmp_comm), env = tmp_env, method = &quot;spearman&quot;,      index = &quot;bray&quot;, upto = ncol(tmp_env), metric = &quot;euclidean&quot;) 

Subset of environmental variables with best correlation to community data.

Correlations:    spearman 
Dissimilarities: bray 
Metric:          euclidean 

Best model has 5 parameters (max. 11 allowed):
AG_ase S_ase XY_ase PX_ase enzNP
with correlation  0.7727866 </code></pre>
</div>
<details>
<summary>
Show the results of individual functional response metadata Mantel tests
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>$AG_ase

Mantel statistic based on Spearman&#39;s rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = &quot;spearman&quot;,      permutations = 999) 

Mantel statistic r: 0.5588 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.186 0.235 0.281 0.330 
Permutation: free
Number of permutations: 999


$enzNP

Mantel statistic based on Spearman&#39;s rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = &quot;spearman&quot;,      permutations = 999) 

Mantel statistic r: 0.4623 
      Significance: 0.006 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.219 0.280 0.336 0.413 
Permutation: free
Number of permutations: 999


$S_ase

Mantel statistic based on Spearman&#39;s rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = &quot;spearman&quot;,      permutations = 999) 

Mantel statistic r: 0.6139 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.223 0.305 0.384 0.466 
Permutation: free
Number of permutations: 999


$PX_ase

Mantel statistic based on Spearman&#39;s rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = &quot;spearman&quot;,      permutations = 999) 

Mantel statistic r: 0.6115 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.176 0.222 0.289 0.346 
Permutation: free
Number of permutations: 999


$XY_ase

Mantel statistic based on Spearman&#39;s rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = &quot;spearman&quot;,      permutations = 999) 

Mantel statistic r: 0.456 
      Significance: 0.002 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.189 0.258 0.304 0.347 
Permutation: free
Number of permutations: 999</code></pre>
</div>
</details>
<p><code>bioenv</code> found the following soil functions significantly correlated with the community data: <strong>AG_ase, enzNP, S_ase, PX_ase, XY_ase</strong></p>
<h3 id="temperature-adaptation-1">Temperature Adaptation</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>tmp_env <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>temp_adapt)</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>tmp_env[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioenv <span class="ot">&lt;-</span> <span class="fu">bioenv</span>(<span class="fu">wisconsin</span>(tmp_comm), tmp_env, </span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>, <span class="at">index =</span> <span class="st">&quot;bray&quot;</span>, </span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">upto =</span> <span class="fu">ncol</span>(tmp_env), <span class="at">metric =</span> <span class="st">&quot;euclidean&quot;</span>)</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>bioenv_list <span class="ot">&lt;-</span> temp_adapt_bioenv<span class="sc">$</span>models[[temp_adapt_bioenv<span class="sc">$</span>whichbest]]<span class="sc">$</span>best</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>bioenv_best <span class="ot">&lt;-</span> <span class="fu">bioenvdist</span>(temp_adapt_bioenv, <span class="at">which =</span> <span class="st">&quot;best&quot;</span>)</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> bioenv_list) {</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>       tmp_dp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(temp_adapt_bioenv<span class="sc">$</span>x)</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>       tmp_md <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_dp[[i]], </span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>       tmp_man <span class="ot">&lt;-</span> <span class="fu">mantel</span>(bioenv_best, tmp_md, </span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>                         <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>)</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>       tmp_md_name <span class="ot">&lt;-</span> <span class="fu">names</span>(tmp_dp)[[i]]</span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>       tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(tmp_md_name, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_bioenv_mantel_test&quot;</span>))</span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_name, tmp_man)</span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;_bioenv_mantel_test&quot;</span>)</span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioenv_ind_mantel <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">CUEcp =</span> CUEcp_bioenv_mantel_test, </span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">LP_Q10 =</span> LP_Q10_bioenv_mantel_test, </span>
<span id="cb28-28"><a href="#cb28-28" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">P_Q10 =</span> P_Q10_bioenv_mantel_test, </span>
<span id="cb28-29"><a href="#cb28-29" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">S_Q10 =</span> S_Q10_bioenv_mantel_test, </span>
<span id="cb28-30"><a href="#cb28-30" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">Tmin =</span> Tmin_bioenv_mantel_test)</span>
<span id="cb28-31"><a href="#cb28-31" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;_bioenv_mantel_test&quot;</span>))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
Call:
bioenv(comm = wisconsin(tmp_comm), env = tmp_env, method = &quot;spearman&quot;,      index = &quot;bray&quot;, upto = ncol(tmp_env), metric = &quot;euclidean&quot;) 

Subset of environmental variables with best correlation to community data.

Correlations:    spearman 
Dissimilarities: bray 
Metric:          euclidean 

Best model has 5 parameters (max. 13 allowed):
P_Q10 S_Q10 LP_Q10 CUEcp Tmin
with correlation  0.4237093 </code></pre>
</div>
<details>
<summary>
Show the results of individual temperature adaptation metadata Mantel tests
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>$CUEcp

Mantel statistic based on Spearman&#39;s rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = &quot;spearman&quot;,      permutations = 999) 

Mantel statistic r: 0.325 
      Significance: 0.013 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.188 0.226 0.280 0.331 
Permutation: free
Number of permutations: 999


$LP_Q10

Mantel statistic based on Spearman&#39;s rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = &quot;spearman&quot;,      permutations = 999) 

Mantel statistic r: 0.3772 
      Significance: 0.005 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.203 0.272 0.306 0.351 
Permutation: free
Number of permutations: 999


$P_Q10

Mantel statistic based on Spearman&#39;s rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = &quot;spearman&quot;,      permutations = 999) 

Mantel statistic r: 0.5177 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.158 0.211 0.261 0.309 
Permutation: free
Number of permutations: 999


$S_Q10

Mantel statistic based on Spearman&#39;s rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = &quot;spearman&quot;,      permutations = 999) 

Mantel statistic r: 0.4395 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.159 0.205 0.252 0.301 
Permutation: free
Number of permutations: 999


$Tmin

Mantel statistic based on Spearman&#39;s rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = &quot;spearman&quot;,      permutations = 999) 

Mantel statistic r: 0.4044 
      Significance: 0.005 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.207 0.267 0.305 0.359 
Permutation: free
Number of permutations: 999</code></pre>
</div>
</details>
<p><code>bioenv</code> found the following temperature adaptations significantly correlated with the community data: <strong>CUEcp, LP_Q10, P_Q10, S_Q10, Tmin</strong></p>
<h2 id="distance-based-redundancy">Distance-based Redundancy</h2>
<p>Now we turn our attention to distance-based redundancy analysis (dbRDA), an ordination method similar to Redundancy Analysis (rda) but it allows non-Euclidean dissimilarity indices, such as Manhattan or BrayCurtis distance. For this, we use <code>capscale</code> from the <code>vegan</code> package. <code>capscale</code> is a constrained versions of metric scaling (principal coordinates analysis), which are based on the Euclidean distance but can be used, and are more useful, with other dissimilarity measures. The functions can also perform unconstrained principal coordinates analysis, optionally using extended dissimilarities.</p>
<p>For each of the three metadata subsets, we perform the following steps:</p>
<ol type="1">
<li>Run <code>rankindex</code> to compare metadata and community dissimilarity indices for gradient detection. This will help us select the best dissimilarity metric to use.</li>
<li>Run <code>capscale</code> for distance-based redundancy analysis.</li>
<li>Run <code>envfit</code> to fit environmental parameters onto the ordination. This function basically calculates correlation scores between the metadata parameters and the ordination axes.</li>
<li>Select metadata parameters significant for <code>bioenv</code> (see above) and/or <code>envfit</code> analyses.</li>
<li>Run <code>envfit</code> on ASVs.</li>
<li>Plot the ordination and vector overlays.</li>
</ol>
<h3 id="edaphic-properties-2">Edaphic Properties</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>tmp_md <span class="ot">&lt;-</span> ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>edaphic</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>tmp_md<span class="sc">$</span>TREAT_T <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tmp_md<span class="sc">$</span>TREAT_T)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>edaphic_rank <span class="ot">&lt;-</span> <span class="fu">rankindex</span>(tmp_md[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_md)], tmp_comm, </span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">indices =</span> <span class="fu">c</span>(<span class="st">&quot;euc&quot;</span>, <span class="st">&quot;man&quot;</span>, <span class="st">&quot;gow&quot;</span>,<span class="st">&quot;bra&quot;</span>, <span class="st">&quot;kul&quot;</span>), </span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">stepacross =</span> <span class="cn">FALSE</span>, <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>)</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>      euc       man       gow       bra       kul 
0.3171263 0.4336927 0.1028302 0.4336927 0.4336927 </code></pre>
</div>
<p>Lets run <code>capscale</code> using Bray-Curtis. Note, we have 15 metadata parameters in this group but, for some reason, <code>capscale</code> only works with 13 parameters. This may have to do with degrees of freedom?</p>
<ul>
<li>Starting properties: AST, H2O, N, P, Al, Ca, Fe, K, Mg, Mn, Na, TEB, ECEC, pH, NH4, NO3, PO4, DOC, DON, DOCN</li>
<li>Autocorrelated removed: TEB, DON, Na, Al, Ca</li>
<li>Remove for capscale: Mg, Mn</li>
</ul>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>edaphic_cap <span class="ot">&lt;-</span> <span class="fu">capscale</span>(tmp_comm <span class="sc">~</span> AST <span class="sc">+</span> H2O <span class="sc">+</span> N <span class="sc">+</span> P <span class="sc">+</span> Fe <span class="sc">+</span> K <span class="sc">+</span> ECEC <span class="sc">+</span> </span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>                                   pH <span class="sc">+</span> NH4 <span class="sc">+</span> NO3 <span class="sc">+</span> PO4 <span class="sc">+</span> DOC <span class="sc">+</span> DOCN, </span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>                        tmp_md, <span class="at">dist =</span> <span class="st">&quot;bray&quot;</span>)</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tmp_md)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>Call: capscale(formula = tmp_comm ~ AST + H2O + N + P + Fe + K
+ ECEC + pH + NH4 + NO3 + PO4 + DOC + DOCN, data = tmp_md,
distance = &quot;bray&quot;)

              Inertia Proportion Rank
Total         1.67890    1.00000     
Constrained   1.63037    0.97110   13
Unconstrained 0.04853    0.02890    1
Inertia is squared Bray distance 
Species scores projected from &#39;tmp_comm&#39; 

Eigenvalues for constrained axes:
  CAP1   CAP2   CAP3   CAP4   CAP5   CAP6   CAP7   CAP8   CAP9  CAP10 
0.6287 0.2955 0.1984 0.1104 0.1085 0.0661 0.0536 0.0511 0.0359 0.0277 
 CAP11  CAP12  CAP13 
0.0247 0.0218 0.0080 

Eigenvalues for unconstrained axes:
   MDS1 
0.04853 </code></pre>
</div>
<p>Now we can look at the variance against each principal component.</p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="metadata_files/figure-html5/unnamed-chunk-52-1.png" width="624" /></p>
</div>
<p>And then make some quick and dirty plots. This will also come in handy later when we need to parse out data a better plot visualization. The <code>ggplot</code> function <code>autoplot</code> stores these data in a more accessible way than the raw results from <code>capscale</code></p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">plot</span>(edaphic_cap) </span></code></pre></div>
<p><img src="metadata_files/figure-html5/unnamed-chunk-53-1.png" width="624" /></p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt <span class="ot">&lt;-</span> ggplot2<span class="sc">::</span><span class="fu">autoplot</span>(edaphic_cap, <span class="at">arrows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt</span></code></pre></div>
<p><img src="metadata_files/figure-html5/unnamed-chunk-53-2.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Next, we need to grab capscale scores for the samples and create a data frame of the first two dimensions. We will also need to add some of the sample details to the data frame. For this we use the vegan function <code>scores</code> which gets species or site scores from the ordination.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">&quot;sites&quot;</span>)</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> tmp_samp_scores <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">SampleID =</span> Label)</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md_sub <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>edaphic_plot_data <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_md_sub, tmp_samp_scores, <span class="at">by =</span> <span class="st">&quot;SampleID&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>Now we have a new data frame that contains sample details and capscale values.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>          SampleID PLOT TREAT TREAT_T PAIR        CAP1           CAP2
1  P10_D00_010_C0E  P10     C       0    E -0.51023381  0.16071056348
2  P02_D00_010_C0A  P02     C       0    A -0.48448722 -0.19055366181
3  P04_D00_010_C0B  P04     C       0    B -0.58218619  0.02325887477
4  P06_D00_010_C0C  P06     C       0    C -0.53370906  0.09973976767
5  P08_D00_010_C0D  P08     C       0    D -0.68492943  0.13143857676
6  P01_D00_010_W3A  P01     W       3    A -0.13286443 -0.28661650806
7  P03_D00_010_W3B  P03     W       3    B -0.25415783  0.00002258673
8  P05_D00_010_W3C  P05     W       3    C -0.38829574 -0.08695154559
9  P07_D00_010_W3D  P07     W       3    D -0.01360582 -0.49632449304
10 P09_D00_010_W3E  P09     W       3    E -0.10639798 -0.54219706978
11 P01_D00_010_W8A  P01     W       8    A  0.78689725  0.02214444812
12 P03_D00_010_W8B  P03     W       8    B  0.89574754 -0.92786190328
13 P05_D00_010_W8C  P05     W       8    C  1.07945322 -0.28874425650
14 P07_D00_010_W8D  P07     W       8    D  0.41234456  0.75597843926
15 P09_D00_010_W8E  P09     W       8    E  0.51642495  1.62595618128</code></pre>
</div>
<p>We can then do the same with the metadata vectors. Here though we only need the scores and parameter name.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>edaphic_md_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">&quot;biplot&quot;</span>)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>edaphic_md_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>edaphic_md_scores <span class="ot">&lt;-</span> edaphic_md_scores <span class="sc">%&gt;%</span> </span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;Label&quot;</span>)</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>     parameters        CAP1         CAP2
AST         AST  0.97882379 -0.055074169
H2O         H2O -0.76548607 -0.129779379
N             N  0.38130076 -0.131035479
P             P  0.26536024 -0.132125558
Fe           Fe  0.12138586  0.353093177
K             K -0.02490580  0.151975508
ECEC       ECEC -0.15365335  0.379848234
pH           pH -0.07327989  0.272676813
NH4         NH4  0.19922925  0.516074241
NO3         NO3 -0.27681227  0.288729283
PO4         PO4 -0.60140669 -0.009980058
DOC         DOC -0.59979130 -0.399930429
DOCN       DOCN  0.11210058  0.422975484</code></pre>
</div>
<p>Lets run some quick correlations of metadata with ordination axes to see which parameters are significant. For this we use the vegan function <code>envfit</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> edaphic_plot_data[, <span class="dv">6</span><span class="sc">:</span><span class="dv">7</span>]</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_samp_scores_sub)</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>tmp_param_list <span class="ot">&lt;-</span> edaphic_md_scores<span class="sc">$</span>parameters</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> <span class="fu">subset</span>(tmp_md,  <span class="at">select =</span>  tmp_param_list)</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>envfit_edaphic_md <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, tmp_md_sub,</span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
***VECTORS

         CAP1     CAP2     r2   Pr(&gt;r)    
AST   0.99825 -0.05912 0.8287 0.000999 ***
H2O  -0.98657 -0.16333 0.5189 0.009990 ** 
N     0.94537 -0.32599 0.1403 0.398601    
P     0.89518 -0.44571 0.0759 0.602398    
Fe    0.32411  0.94602 0.1194 0.432567    
K    -0.16554  0.98620 0.0204 0.890110    
ECEC -0.37874  0.92550 0.1444 0.368631    
pH   -0.26344  0.96468 0.0685 0.672328    
NH4   0.35944  0.93317 0.2621 0.152847    
NO3  -0.69392  0.72006 0.1380 0.417582    
PO4  -0.99991 -0.01327 0.3118 0.090909 .  
DOC  -0.83396 -0.55183 0.4459 0.023976 *  
DOCN  0.25463  0.96704 0.1640 0.318681    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_edaphic_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_edaphic_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_edaphic_md<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.4</span>))</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(edaphic_md_signif_hits)</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(edaphic_md_signif_hits)</span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif <span class="ot">&lt;-</span> edaphic_md_scores[edaphic_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> edaphic_md_signif_hits,]</span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif<span class="sc">$</span>parameters</span></code></pre></div>
</details>
</div>
<p><code>envfit</code> found that AST, H2O, DOC were significantly correlated.</p>
<p>Now lets see if the same parameters are significant for the <code>envfit</code> and <code>bioenv</code> analyses.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;Significant parameters from bioenv analysis.&quot;
[1] &quot;AST&quot;
_____________________________________
[1] &quot;Significant parameters from envfit analysis.&quot;
[1] &quot;AST&quot; &quot;H2O&quot; &quot;DOC&quot;
_____________________________________
[1] &quot;Found in bioenv but not envfit.&quot;
character(0)
_____________________________________
[1] &quot;Found in envfit but not bioenv.&quot;
[1] &quot;H2O&quot; &quot;DOC&quot;
_____________________________________
[1] &quot;Found in envfit and bioenv.&quot;
[1] &quot;AST&quot; &quot;H2O&quot; &quot;DOC&quot;</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>new_edaphic_md_signif_hits <span class="ot">&lt;-</span> edaphic_sig_diff</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="co">#new_edaphic_md_signif_hits &lt;- append(edaphic_md_signif_hits, edaphic_sig_diff)</span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_all <span class="ot">&lt;-</span> edaphic_md_scores[edaphic_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> new_edaphic_md_signif_hits,]</span></code></pre></div>
</details>
</div>
<p>Check. Next, we run <code>envfit</code> for the ASVs.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>envfit_edaphic_asv <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, </span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>                             tmp_comm[, <span class="fu">order</span>(<span class="fu">colSums</span>(<span class="sc">-</span>tmp_comm))][, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>edaphic_asv_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">&quot;species&quot;</span>)</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>edaphic_asv_scores <span class="ot">&lt;-</span> edaphic_asv_scores <span class="sc">%&gt;%</span> </span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span></span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;Label&quot;</span>)</span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>edaphic_asv_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
***VECTORS

          CAP1     CAP2     r2   Pr(&gt;r)    
ASV1   0.99502  0.09969 0.6491 0.005994 ** 
ASV2   0.54850 -0.83615 0.8984 0.000999 ***
ASV3   0.79245  0.60994 0.9429 0.000999 ***
ASV4  -0.12958  0.99157 0.2913 0.120879    
ASV5  -0.56363 -0.82603 0.1586 0.362637    
ASV6  -0.10651  0.99431 0.7559 0.000999 ***
ASV8  -0.84329  0.53746 0.4370 0.030969 *  
ASV7  -0.52510 -0.85104 0.0681 0.655345    
ASV9   0.51828 -0.85521 0.2559 0.185814    
ASV10 -0.34456  0.93876 0.8331 0.000999 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_edaphic_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_edaphic_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_edaphic_asv<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.5</span>))</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(edaphic_asv_signif_hits)</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(edaphic_asv_signif_hits)</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif <span class="ot">&lt;-</span> edaphic_asv_scores[edaphic_asv_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> edaphic_asv_signif_hits,]</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>      parameters        CAP1        CAP2 variable_type
ASV3        ASV3  0.48974117  0.51152737           ASV
ASV6        ASV6 -0.02123424  0.24919713           ASV
ASV2        ASV2  0.23670463 -0.49287683           ASV
ASV10      ASV10 -0.02578511  0.09240257           ASV
ASV1        ASV1  1.18949655  0.17867402           ASV</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_all<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">&quot;metadata&quot;</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">&quot;ASV&quot;</span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>edaphic_bioplot_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(edaphic_md_signif_all, edaphic_asv_signif)</span></code></pre></div>
</details>
</div>
<p>The last thing to do is categorize parameters scores and ASV scores into different variable types for plotting.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>edaphic_bioplot_data_md <span class="ot">&lt;-</span> <span class="fu">subset</span>(edaphic_bioplot_data, </span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>                                  edaphic_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">&quot;metadata&quot;</span>)</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>edaphic_bioplot_data_asv <span class="ot">&lt;-</span> <span class="fu">subset</span>(edaphic_bioplot_data, </span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>                                   edaphic_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">&quot;ASV&quot;</span>)</span></code></pre></div>
</details>
</div>
<details>
<summary>
Show the code for the plot
</summary>
<div class="layout-chunk" data-layout="l-body">

</div>
</details>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="soil-functional-response-1">Soil Functional Response</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>tmp_md <span class="ot">&lt;-</span> ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>soil_funct</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>tmp_md<span class="sc">$</span>TREAT_T <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tmp_md<span class="sc">$</span>TREAT_T)</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>soil_funct_rank <span class="ot">&lt;-</span> <span class="fu">rankindex</span>(tmp_md[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_md)], tmp_comm, </span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">indices =</span> <span class="fu">c</span>(<span class="st">&quot;euc&quot;</span>, <span class="st">&quot;man&quot;</span>, <span class="st">&quot;gow&quot;</span>,<span class="st">&quot;bra&quot;</span>, <span class="st">&quot;kul&quot;</span>), </span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">stepacross =</span> <span class="cn">FALSE</span>, <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>)</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>      euc       man       gow       bra       kul 
0.5840452 0.7341800 0.2168775 0.7341800 0.7341800 </code></pre>
</div>
<p>Lets run <code>capscale</code> using Bray-Curtis. Note, we have 11 metadata parameters in this group but, for some reason, <code>capscale</code> only works with 13 parameters. This may have to do with degrees of freedom?</p>
<ul>
<li>Starting properties: micC, micN, micP, micCN, micCP, micNP, AG_ase, BG_ase, BP_ase, CE_ase, P_ase, N_ase, S_ase, XY_ase, LP_ase, PX_ase, CO2, enzCN, enzCP, enzNP</li>
<li>Autocorrelated removed: micN, micNP, enzCN, enzCP, BP_ase, CE_ase, LP_ase, N_ase, P_ase</li>
<li>Remove for capscale: NONE</li>
</ul>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>soil_funct_cap <span class="ot">&lt;-</span> <span class="fu">capscale</span>(tmp_comm <span class="sc">~</span> micC <span class="sc">+</span> micP <span class="sc">+</span> micCN <span class="sc">+</span> micCP <span class="sc">+</span> AG_ase <span class="sc">+</span> BG_ase <span class="sc">+</span> </span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>                                      S_ase <span class="sc">+</span> XY_ase <span class="sc">+</span> PX_ase <span class="sc">+</span> CO2 <span class="sc">+</span> enzNP, </span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>                           tmp_md, <span class="at">dist =</span> <span class="st">&quot;bray&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>Call: capscale(formula = tmp_comm ~ micC + micP + micCN +
micCP + AG_ase + BG_ase + S_ase + XY_ase + PX_ase + CO2 +
enzNP, data = tmp_md, distance = &quot;bray&quot;)

              Inertia Proportion Rank
Total         1.67890    1.00000     
Constrained   1.52672    0.90936   11
Unconstrained 0.15218    0.09064    3
Inertia is squared Bray distance 
Species scores projected from &#39;tmp_comm&#39; 

Eigenvalues for constrained axes:
  CAP1   CAP2   CAP3   CAP4   CAP5   CAP6   CAP7   CAP8   CAP9  CAP10 
0.6202 0.2950 0.1827 0.1164 0.0950 0.0555 0.0491 0.0468 0.0290 0.0246 
 CAP11 
0.0126 

Eigenvalues for unconstrained axes:
   MDS1    MDS2    MDS3 
0.06264 0.05568 0.03386 </code></pre>
</div>
<p>Now we can look at the variance against each principal component.</p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="metadata_files/figure-html5/unnamed-chunk-78-1.png" width="624" /></p>
</div>
<p>And then make some quick and dirty plots. This will also come in handy later when we need to parse out data a better plot visualization. The <code>ggplot</code> function <code>autoplot</code> stores these data in a more accessible way than the raw results from <code>capscale</code></p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">plot</span>(soil_funct_cap) </span></code></pre></div>
<p><img src="metadata_files/figure-html5/unnamed-chunk-79-1.png" width="624" /></p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt <span class="ot">&lt;-</span> <span class="fu">autoplot</span>(soil_funct_cap, <span class="at">arrows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt</span></code></pre></div>
<p><img src="metadata_files/figure-html5/unnamed-chunk-79-2.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Next, we need to grab capscale scores for the samples and create a data frame of the first two dimensions. We will also need to add some of the sample details to the data frame. For this we use the vegan function <code>scores</code> which gets species or site scores from the ordination.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">&quot;sites&quot;</span>)</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> tmp_samp_scores <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">SampleID =</span> Label)</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md_sub <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>soil_funct_plot_data <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_md_sub, tmp_samp_scores, </span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">by =</span> <span class="st">&quot;SampleID&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>Now we have a new data frame that contains sample details and capscale values.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>          SampleID PLOT TREAT TREAT_T PAIR        CAP1        CAP2
1  P10_D00_010_C0E  P10     C       0    E -0.53163678  0.19467717
2  P02_D00_010_C0A  P02     C       0    A -0.49565928 -0.17246578
3  P04_D00_010_C0B  P04     C       0    B -0.58149488  0.04815608
4  P06_D00_010_C0C  P06     C       0    C -0.53700662  0.10581563
5  P08_D00_010_C0D  P08     C       0    D -0.69408551  0.15871504
6  P01_D00_010_W3A  P01     W       3    A -0.13789308 -0.32241637
7  P03_D00_010_W3B  P03     W       3    B -0.23991258 -0.02936566
8  P05_D00_010_W3C  P05     W       3    C -0.37861217 -0.07503400
9  P07_D00_010_W3D  P07     W       3    D -0.01748558 -0.49326034
10 P09_D00_010_W3E  P09     W       3    E -0.10498973 -0.54355896
11 P01_D00_010_W8A  P01     W       8    A  0.78557974  0.01332924
12 P03_D00_010_W8B  P03     W       8    B  0.89342218 -0.94105534
13 P05_D00_010_W8C  P05     W       8    C  1.07904998 -0.30014637
14 P07_D00_010_W8D  P07     W       8    D  0.41449841  0.74331915
15 P09_D00_010_W8E  P09     W       8    E  0.54622591  1.61329051</code></pre>
</div>
<p>We can then do the same with the metadata vectors. Here though we only need the scores and parameter name.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>soil_funct_md_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">&quot;biplot&quot;</span>)</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>soil_funct_md_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>soil_funct_md_scores <span class="ot">&lt;-</span> soil_funct_md_scores <span class="sc">%&gt;%</span> </span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span> </span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;Label&quot;</span>)</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>       parameters        CAP1       CAP2
micC         micC  0.05033428  0.3114850
micP         micP -0.28522874  0.2555622
micCN       micCN  0.18231162 -0.4195484
micCP       micCP  0.23124711 -0.2302962
AG_ase     AG_ase -0.26468219  0.6079572
BG_ase     BG_ase  0.42574674  0.6072615
S_ase       S_ase  0.29987944  0.7961890
XY_ase     XY_ase  0.64821470  0.3027833
PX_ase     PX_ase  0.80304377  0.3296233
CO2           CO2  0.68903357 -0.1628577
enzNP       enzNP -0.77637066  0.1324578</code></pre>
</div>
<p>Lets run some quick correlations of metadata with ordination axes to see which parameters are significant. For this we use the vegan function <code>envfit</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> soil_funct_plot_data[, <span class="dv">6</span><span class="sc">:</span><span class="dv">7</span>]</span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_samp_scores_sub)</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>tmp_param_list <span class="ot">&lt;-</span> soil_funct_md_scores<span class="sc">$</span>parameters</span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> <span class="fu">subset</span>(tmp_md,  <span class="at">select =</span>  tmp_param_list)</span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-7"><a href="#cb62-7" aria-hidden="true" tabindex="-1"></a>envfit_soil_funct_md <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, tmp_md_sub,</span>
<span id="cb62-8"><a href="#cb62-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
***VECTORS

           CAP1     CAP2     r2   Pr(&gt;r)    
micC    0.16467  0.98635 0.1012 0.511489    
micP   -0.74314  0.66913 0.1473 0.385614    
micCN   0.39228 -0.91985 0.2111 0.244755    
micCP   0.70633 -0.70788 0.1070 0.504496    
AG_ase -0.39292  0.91957 0.4435 0.025974 *  
BG_ase  0.57346  0.81923 0.5604 0.006993 ** 
S_ase   0.35512  0.93482 0.7371 0.001998 ** 
XY_ase  0.90321  0.42921 0.5193 0.008991 ** 
PX_ase  0.92246  0.38609 0.7640 0.000999 ***
CO2     0.97421 -0.22565 0.5041 0.012987 *  
enzNP  -0.98661  0.16307 0.6242 0.003996 ** 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_soil_funct_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_soil_funct_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_soil_funct_md<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.4</span>))</span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(soil_funct_md_signif_hits)</span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(soil_funct_md_signif_hits)</span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif <span class="ot">&lt;-</span> soil_funct_md_scores[soil_funct_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a>                                               soil_funct_md_signif_hits,]</span>
<span id="cb64-8"><a href="#cb64-8" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif<span class="sc">$</span>parameters</span></code></pre></div>
</details>
</div>
<p><code>envfit</code> found that AG_ase, BG_ase, S_ase, XY_ase, PX_ase, CO2, enzNP were significantly correlated.</p>
<p>Now lets see if the same parameters are significant for the <code>envfit</code> and <code>bioenv</code> analyses.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;Significant parameters from bioenv analysis.&quot;
[1] &quot;AG_ase&quot; &quot;enzNP&quot;  &quot;S_ase&quot;  &quot;PX_ase&quot; &quot;XY_ase&quot;
_____________________________________
[1] &quot;Significant parameters from envfit analysis.&quot;
[1] &quot;AG_ase&quot; &quot;BG_ase&quot; &quot;S_ase&quot;  &quot;XY_ase&quot; &quot;PX_ase&quot; &quot;CO2&quot;    &quot;enzNP&quot; 
_____________________________________
[1] &quot;Found in bioenv but not envfit.&quot;
character(0)
_____________________________________
[1] &quot;Found in envfit but not bioenv.&quot;
[1] &quot;BG_ase&quot; &quot;CO2&quot;   
_____________________________________
[1] &quot;Found in envfit and bioenv.&quot;
[1] &quot;AG_ase&quot; &quot;BG_ase&quot; &quot;S_ase&quot;  &quot;XY_ase&quot; &quot;PX_ase&quot; &quot;CO2&quot;    &quot;enzNP&quot; </code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co">#new_soil_funct_md_signif_hits &lt;- append(soil_funct_md_signif_hits, soil_funct_sig_diff)</span></span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>new_soil_funct_md_signif_hits <span class="ot">&lt;-</span> soil_funct_sig_diff</span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_all <span class="ot">&lt;-</span> soil_funct_md_scores[soil_funct_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>                                                   new_soil_funct_md_signif_hits,]</span></code></pre></div>
</details>
</div>
<p>Check. Next, we run <code>envfit</code> for the ASVs.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>envfit_soil_funct_asv <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, </span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>                                tmp_comm[, <span class="fu">order</span>(<span class="fu">colSums</span>(<span class="sc">-</span>tmp_comm))][, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>], </span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">&quot;species&quot;</span>)</span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_scores <span class="ot">&lt;-</span> soil_funct_asv_scores <span class="sc">%&gt;%</span> </span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span> </span>
<span id="cb67-7"><a href="#cb67-7" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;Label&quot;</span>)</span>
<span id="cb67-8"><a href="#cb67-8" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
***VECTORS

          CAP1     CAP2     r2   Pr(&gt;r)    
ASV1   0.99572  0.09244 0.6448 0.008991 ** 
ASV2   0.53392 -0.84554 0.9003 0.000999 ***
ASV3   0.80024  0.59968 0.9414 0.000999 ***
ASV4  -0.10124  0.99486 0.2814 0.142857    
ASV5  -0.56790 -0.82310 0.1663 0.313686    
ASV6  -0.08053  0.99675 0.7514 0.000999 ***
ASV8  -0.83447  0.55105 0.4303 0.017982 *  
ASV7  -0.57807 -0.81599 0.0697 0.619381    
ASV9   0.49314 -0.86995 0.2612 0.147852    
ASV10 -0.32480  0.94578 0.8408 0.000999 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_soil_funct_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_soil_funct_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_soil_funct_asv<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.5</span>))</span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(soil_funct_asv_signif_hits)</span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(soil_funct_asv_signif_hits)</span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif <span class="ot">&lt;-</span> soil_funct_asv_scores[soil_funct_asv_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a>                                                 soil_funct_asv_signif_hits,]</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>      parameters        CAP1        CAP2 variable_type
ASV3        ASV3  0.49966533  0.50499834           ASV
ASV6        ASV6 -0.01098408  0.25326180           ASV
ASV2        ASV2  0.22435017 -0.49303800           ASV
ASV10      ASV10 -0.02420323  0.09642322           ASV
ASV1        ASV1  1.17846608  0.16123440           ASV</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_all<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">&quot;metadata&quot;</span></span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">&quot;ASV&quot;</span></span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>soil_funct_bioplot_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(soil_funct_md_signif_all, soil_funct_asv_signif)</span></code></pre></div>
</details>
</div>
<p>The last thing to do is categorize parameters scores and ASV scores into different variable types for plotting.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>soil_funct_bioplot_data_md <span class="ot">&lt;-</span> <span class="fu">subset</span>(soil_funct_bioplot_data, </span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>                                  soil_funct_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">&quot;metadata&quot;</span>)</span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>soil_funct_bioplot_data_asv <span class="ot">&lt;-</span> <span class="fu">subset</span>(soil_funct_bioplot_data, </span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a>                                   soil_funct_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">&quot;ASV&quot;</span>)</span></code></pre></div>
</details>
</div>
<details>
<summary>
Show the code for the plot
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>soil_funct_cap_vals <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(soil_funct_cap<span class="sc">$</span>CCA<span class="sc">$</span>eig[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>soil_funct_cap1 <span class="ot">&lt;-</span> <span class="fu">signif</span>((soil_funct_cap_vals[<span class="dv">1</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a>soil_funct_cap2 <span class="ot">&lt;-</span> <span class="fu">signif</span>((soil_funct_cap_vals[<span class="dv">2</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a>cpa1_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;CAP1&quot;</span>, <span class="st">&quot; (&quot;</span>, soil_funct_cap1, <span class="st">&quot;%)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a>cpa2_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;CAP2&quot;</span>, <span class="st">&quot; (&quot;</span>, soil_funct_cap2, <span class="st">&quot;%)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-8"><a href="#cb73-8" aria-hidden="true" tabindex="-1"></a>swel_col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#2271B2&quot;</span>, <span class="st">&quot;#71B222&quot;</span>, <span class="st">&quot;#B22271&quot;</span>)</span>
<span id="cb73-9"><a href="#cb73-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-10"><a href="#cb73-10" aria-hidden="true" tabindex="-1"></a>soil_funct_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(soil_funct_plot_data) <span class="sc">+</span></span>
<span id="cb73-11"><a href="#cb73-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">shape =</span> TREAT,</span>
<span id="cb73-12"><a href="#cb73-12" aria-hidden="true" tabindex="-1"></a>                           <span class="at">colour =</span> TREAT_T), <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb73-13"><a href="#cb73-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> swel_col) <span class="sc">+</span></span>
<span id="cb73-14"><a href="#cb73-14" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_text(data = soil_funct_plot_data, aes(x = CAP1, y = CAP2, #UNCOMMENT to add sample labels</span></span>
<span id="cb73-15"><a href="#cb73-15" aria-hidden="true" tabindex="-1"></a><span class="co">#                              label = SampleID), size = 3) + </span></span>
<span id="cb73-16"><a href="#cb73-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">xend =</span> CAP1, <span class="at">yend =</span> CAP2),</span>
<span id="cb73-17"><a href="#cb73-17" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> soil_funct_bioplot_data_md, <span class="at">linetype =</span> <span class="st">&quot;solid&quot;</span>,</span>
<span id="cb73-18"><a href="#cb73-18" aria-hidden="true" tabindex="-1"></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">&quot;cm&quot;</span>)), <span class="at">size =</span> <span class="fl">0.4</span>, </span>
<span id="cb73-19"><a href="#cb73-19" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">&quot;#191919&quot;</span>) <span class="sc">+</span></span>
<span id="cb73-20"><a href="#cb73-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> soil_funct_bioplot_data_md, </span>
<span id="cb73-21"><a href="#cb73-21" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">label =</span> parameters), <span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb73-22"><a href="#cb73-22" aria-hidden="true" tabindex="-1"></a>            <span class="at">nudge_x =</span> <span class="fl">0.1</span>, <span class="at">nudge_y =</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb73-23"><a href="#cb73-23" aria-hidden="true" tabindex="-1"></a>  <span class="do">### This code adds ASV vectors, Uncomment to add</span></span>
<span id="cb73-24"><a href="#cb73-24" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_segment(aes(x = 0, y = 0, xend = CAP1, yend = CAP2),</span></span>
<span id="cb73-25"><a href="#cb73-25" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             data = soil_funct_bioplot_data_asv, linetype = &quot;solid&quot;,</span></span>
<span id="cb73-26"><a href="#cb73-26" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             arrow = arrow(length = unit(0.3, &quot;cm&quot;)), size = 0.2,</span></span>
<span id="cb73-27"><a href="#cb73-27" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             color = &quot;#676767&quot;) +</span></span>
<span id="cb73-28"><a href="#cb73-28" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text(data = soil_funct_bioplot_data_asv, </span></span>
<span id="cb73-29"><a href="#cb73-29" aria-hidden="true" tabindex="-1"></a>  <span class="co">#          aes(x = CAP1, y = CAP2, label = parameters), size = 2.5,</span></span>
<span id="cb73-30"><a href="#cb73-30" aria-hidden="true" tabindex="-1"></a>  <span class="co">#          nudge_x = 0.05, nudge_y = 0.05) +</span></span>
<span id="cb73-31"><a href="#cb73-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span> </span>
<span id="cb73-32"><a href="#cb73-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Capscale Analysis&quot;</span>,</span>
<span id="cb73-33"><a href="#cb73-33" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">&quot;Soil Functional Response&quot;</span>, </span>
<span id="cb73-34"><a href="#cb73-34" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> cpa1_lab, </span>
<span id="cb73-35"><a href="#cb73-35" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> cpa2_lab)</span>
<span id="cb73-36"><a href="#cb73-36" aria-hidden="true" tabindex="-1"></a>soil_funct_plot <span class="ot">&lt;-</span> soil_funct_plot <span class="sc">+</span> <span class="fu">coord_fixed</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb73-37"><a href="#cb73-37" aria-hidden="true" tabindex="-1"></a>soil_funct_plot</span>
<span id="cb73-38"><a href="#cb73-38" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">&quot;files/metadata/figures/ssu18_soil_funct_capscale.png&quot;</span>,</span>
<span id="cb73-39"><a href="#cb73-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">16</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">units =</span> <span class="st">&#39;cm&#39;</span>, <span class="at">res =</span> <span class="dv">600</span>, <span class="at">bg =</span> <span class="st">&quot;white&quot;</span>)</span>
<span id="cb73-40"><a href="#cb73-40" aria-hidden="true" tabindex="-1"></a>soil_funct_plot</span>
<span id="cb73-41"><a href="#cb73-41" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb73-42"><a href="#cb73-42" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">&quot;files/metadata/figures/ssu18_soil_funct_capscale.pdf&quot;</span>,</span>
<span id="cb73-43"><a href="#cb73-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">5</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb73-44"><a href="#cb73-44" aria-hidden="true" tabindex="-1"></a>soil_funct_plot</span>
<span id="cb73-45"><a href="#cb73-45" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
</div>
</details>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="temperature-adaptation-2">Temperature Adaptation</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>tmp_md <span class="ot">&lt;-</span> ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>temp_adapt</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>tmp_md<span class="sc">$</span>TREAT_T <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tmp_md<span class="sc">$</span>TREAT_T)</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a>temp_adapt_rank <span class="ot">&lt;-</span> <span class="fu">rankindex</span>(tmp_md[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_md)], tmp_comm, </span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">indices =</span> <span class="fu">c</span>(<span class="st">&quot;euc&quot;</span>, <span class="st">&quot;man&quot;</span>, <span class="st">&quot;gow&quot;</span>,<span class="st">&quot;bra&quot;</span>, <span class="st">&quot;kul&quot;</span>), </span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">stepacross =</span> <span class="cn">FALSE</span>, <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>)</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>      euc       man       gow       bra       kul 
0.2732117 0.3330085 0.3456355 0.3330085 0.3330085 </code></pre>
</div>
<p>Lets run <code>capscale</code> using Bray-Curtis. Note, we have 13 metadata parameters in this group but, for some reason, <code>capscale</code> only works with 13 parameters. This may have to do with degrees of freedom?</p>
<ul>
<li>Starting properties: AG_Q10, BG_Q10, BP_Q10, CE_Q10, P_Q10, N_Q10, S_Q10, XY_Q10, LP_Q10, PX_Q10, CUEcn, CUEcp, NUE, PUE, Tmin, SI</li>
<li>Autocorrelated removed: NUE, PUE, SI</li>
<li>Remove for capscale: NONE</li>
</ul>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>temp_adapt_cap <span class="ot">&lt;-</span> <span class="fu">capscale</span>(tmp_comm <span class="sc">~</span>  AG_Q10 <span class="sc">+</span> BG_Q10 <span class="sc">+</span> BP_Q10 <span class="sc">+</span> CE_Q10 <span class="sc">+</span> </span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>                                       P_Q10 <span class="sc">+</span> N_Q10 <span class="sc">+</span> S_Q10 <span class="sc">+</span> XY_Q10 <span class="sc">+</span> </span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>                                       LP_Q10 <span class="sc">+</span> PX_Q10 <span class="sc">+</span> CUEcn <span class="sc">+</span> CUEcp <span class="sc">+</span> Tmin, </span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a>                             tmp_md, <span class="at">dist =</span> <span class="st">&quot;bray&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>Call: capscale(formula = tmp_comm ~ AG_Q10 + BG_Q10 + BP_Q10 +
CE_Q10 + P_Q10 + N_Q10 + S_Q10 + XY_Q10 + LP_Q10 + PX_Q10 +
CUEcn + CUEcp + Tmin, data = tmp_md, distance = &quot;bray&quot;)

              Inertia Proportion Rank
Total         1.67890    1.00000     
Constrained   1.57681    0.93919   13
Unconstrained 0.10209    0.06081    1
Inertia is squared Bray distance 
Species scores projected from &#39;tmp_comm&#39; 

Eigenvalues for constrained axes:
  CAP1   CAP2   CAP3   CAP4   CAP5   CAP6   CAP7   CAP8   CAP9  CAP10 
0.6298 0.2776 0.1985 0.1087 0.0849 0.0614 0.0528 0.0450 0.0352 0.0273 
 CAP11  CAP12  CAP13 
0.0244 0.0218 0.0095 

Eigenvalues for unconstrained axes:
   MDS1 
0.10209 </code></pre>
</div>
<p>Now we can look at the variance against each principal component.</p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="metadata_files/figure-html5/unnamed-chunk-104-1.png" width="624" /></p>
</div>
<p>And then make some quick and dirty plots. This will also come in handy later when we need to parse out data a better plot visualization. The <code>ggplot</code> function <code>autoplot</code> stores these data in a more accessible way than the raw results from <code>capscale</code></p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">plot</span>(temp_adapt_cap) </span></code></pre></div>
<p><img src="metadata_files/figure-html5/unnamed-chunk-105-1.png" width="624" /></p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt <span class="ot">&lt;-</span> <span class="fu">autoplot</span>(temp_adapt_cap, <span class="at">arrows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt</span></code></pre></div>
<p><img src="metadata_files/figure-html5/unnamed-chunk-105-2.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Next, we need to grab capscale scores for the samples and create a data frame of the first two dimensions. We will also need to add some of the sample details to the data frame. For this we use the vegan function <code>scores</code> which gets species or site scores from the ordination.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">&quot;sites&quot;</span>)</span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> tmp_samp_scores <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">SampleID =</span> Label)</span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md_sub <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb80-7"><a href="#cb80-7" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot_data <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_md_sub, tmp_samp_scores, </span>
<span id="cb80-8"><a href="#cb80-8" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">by =</span> <span class="st">&quot;SampleID&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>Now we have a new data frame that contains sample details and capscale values.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>          SampleID PLOT TREAT TREAT_T PAIR        CAP1        CAP2
1  P10_D00_010_C0E  P10     C       0    E -0.50625375  0.10348626
2  P02_D00_010_C0A  P02     C       0    A -0.48509579 -0.14580527
3  P04_D00_010_C0B  P04     C       0    B -0.57845671  0.03776429
4  P06_D00_010_C0C  P06     C       0    C -0.53336078  0.08936059
5  P08_D00_010_C0D  P08     C       0    D -0.68345909  0.10974979
6  P01_D00_010_W3A  P01     W       3    A -0.13776351 -0.22674265
7  P03_D00_010_W3B  P03     W       3    B -0.25679729  0.01972197
8  P05_D00_010_W3C  P05     W       3    C -0.38722889 -0.11236725
9  P07_D00_010_W3D  P07     W       3    D -0.01518359 -0.46741081
10 P09_D00_010_W3E  P09     W       3    E -0.11194937 -0.54460136
11 P01_D00_010_W8A  P01     W       8    A  0.78329851  0.09997647
12 P03_D00_010_W8B  P03     W       8    B  0.89642936 -1.09522700
13 P05_D00_010_W8C  P05     W       8    C  1.07566246 -0.26764258
14 P07_D00_010_W8D  P07     W       8    D  0.41185806  0.81673870
15 P09_D00_010_W8E  P09     W       8    E  0.52830039  1.58299886</code></pre>
</div>
<p>We can then do the same with the metadata vectors. Here though we only need the scores and parameter name.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">&quot;biplot&quot;</span>)</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_scores <span class="ot">&lt;-</span> temp_adapt_md_scores <span class="sc">%&gt;%</span> </span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span> </span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;Label&quot;</span>)</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>       parameters        CAP1        CAP2
AG_Q10     AG_Q10 -0.07373951  0.23793282
BG_Q10     BG_Q10  0.72604945  0.10579764
BP_Q10     BP_Q10  0.07015779 -0.12413231
CE_Q10     CE_Q10 -0.37489864  0.55022094
P_Q10       P_Q10  0.87190011 -0.25252818
N_Q10       N_Q10  0.09422795 -0.32515404
S_Q10       S_Q10  0.91309230  0.55878516
XY_Q10     XY_Q10  0.92245895 -0.02395029
LP_Q10     LP_Q10 -0.90356227 -0.36479560
PX_Q10     PX_Q10 -0.87475774  0.18121787
CUEcn       CUEcn -0.23872665 -0.39707742
CUEcp       CUEcp  0.57949983 -0.62675335
Tmin         Tmin  0.99158847  0.19050834</code></pre>
</div>
<p>Lets run some quick correlations of metadata with ordination axes to see which parameters are significant. For this we use the vegan function <code>envfit</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> temp_adapt_plot_data[, <span class="dv">6</span><span class="sc">:</span><span class="dv">7</span>]</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_samp_scores_sub)</span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>tmp_param_list <span class="ot">&lt;-</span> temp_adapt_md_scores<span class="sc">$</span>parameters</span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> <span class="fu">subset</span>(tmp_md,  <span class="at">select =</span>  tmp_param_list)</span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a>envfit_temp_adapt_md <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, tmp_md_sub,</span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
***VECTORS

           CAP1     CAP2     r2  Pr(&gt;r)  
AG_Q10 -0.30810  0.95136 0.0260 0.83117  
BG_Q10  0.99031  0.13889 0.2354 0.17782  
BP_Q10  0.50944 -0.86050 0.0086 0.95504  
CE_Q10 -0.58116  0.81379 0.1875 0.25075  
P_Q10   0.96427 -0.26494 0.3590 0.05395 .
N_Q10   0.28973 -0.95711 0.0479 0.76723  
S_Q10   0.86347  0.50440 0.4956 0.01499 *
XY_Q10  0.99973 -0.02334 0.3725 0.04895 *
LP_Q10 -0.93292 -0.36008 0.4131 0.04096 *
PX_Q10 -0.98131  0.19244 0.3483 0.07992 .
CUEcn  -0.53407 -0.84544 0.0908 0.55245  
CUEcp   0.69625 -0.71780 0.3103 0.12288  
Tmin    0.98341  0.18140 0.4456 0.02997 *
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_temp_adapt_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_temp_adapt_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_temp_adapt_md<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.4</span>))</span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(temp_adapt_md_signif_hits)</span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(temp_adapt_md_signif_hits)</span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif <span class="ot">&lt;-</span> temp_adapt_md_scores[temp_adapt_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb86-7"><a href="#cb86-7" aria-hidden="true" tabindex="-1"></a>                                               temp_adapt_md_signif_hits,]</span></code></pre></div>
</details>
</div>
<p><code>envfit</code> found that S_Q10, LP_Q10, Tmin were significantly correlated.</p>
<p>Now lets see if the same parameters are significant for the <code>envfit</code> and <code>bioenv</code> analyses.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;Significant parameters from bioenv analysis.&quot;
[1] &quot;CUEcp&quot;  &quot;LP_Q10&quot; &quot;P_Q10&quot;  &quot;S_Q10&quot;  &quot;Tmin&quot;  
_____________________________________
[1] &quot;Significant parameters from envfit analysis.&quot;
[1] &quot;S_Q10&quot;  &quot;LP_Q10&quot; &quot;Tmin&quot;  
_____________________________________
[1] &quot;Found in bioenv but not envfit.&quot;
[1] &quot;CUEcp&quot; &quot;P_Q10&quot;
_____________________________________
[1] &quot;Found in envfit but not bioenv.&quot;
character(0)
_____________________________________
[1] &quot;Found in envfit and  bioenv.&quot;
[1] &quot;S_Q10&quot;  &quot;LP_Q10&quot; &quot;Tmin&quot;   &quot;CUEcp&quot;  &quot;P_Q10&quot; </code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="co">#new_temp_adapt_md_signif_hits &lt;- base::append(temp_adapt_md_signif_hits, temp_adapt_sig_diff)</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>new_temp_adapt_md_signif_hits <span class="ot">&lt;-</span> temp_adapt_sig_diff</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_all <span class="ot">&lt;-</span> temp_adapt_md_scores[temp_adapt_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>                                                   new_temp_adapt_md_signif_hits,]</span></code></pre></div>
</details>
</div>
<p>Check. Next, we run <code>envfit</code> for the ASVs.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>envfit_temp_adapt_asv <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, </span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>                                tmp_comm[, <span class="fu">order</span>(<span class="fu">colSums</span>(<span class="sc">-</span>tmp_comm))][, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],</span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">&quot;species&quot;</span>)</span>
<span id="cb89-6"><a href="#cb89-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-7"><a href="#cb89-7" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_scores <span class="ot">&lt;-</span> temp_adapt_asv_scores <span class="sc">%&gt;%</span> </span>
<span id="cb89-8"><a href="#cb89-8" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span></span>
<span id="cb89-9"><a href="#cb89-9" aria-hidden="true" tabindex="-1"></a>                         tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;Label&quot;</span>)</span>
<span id="cb89-10"><a href="#cb89-10" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
***VECTORS

          CAP1     CAP2     r2   Pr(&gt;r)    
ASV1   0.98984  0.14217 0.6541 0.009990 ** 
ASV2   0.55070 -0.83470 0.8953 0.000999 ***
ASV3   0.79919  0.60108 0.9513 0.000999 ***
ASV4  -0.12171  0.99257 0.3110 0.118881    
ASV5  -0.67644 -0.73649 0.1181 0.487512    
ASV6  -0.10078  0.99491 0.7028 0.003996 ** 
ASV8  -0.85434  0.51972 0.4279 0.043956 *  
ASV7  -0.57853 -0.81566 0.0602 0.676324    
ASV9   0.53361 -0.84573 0.2425 0.185814    
ASV10 -0.34761  0.93764 0.8031 0.000999 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_temp_adapt_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_temp_adapt_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_temp_adapt_asv<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.5</span>))</span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(temp_adapt_asv_signif_hits)</span>
<span id="cb91-5"><a href="#cb91-5" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(temp_adapt_asv_signif_hits)</span>
<span id="cb91-6"><a href="#cb91-6" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif <span class="ot">&lt;-</span> temp_adapt_asv_scores[temp_adapt_asv_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb91-7"><a href="#cb91-7" aria-hidden="true" tabindex="-1"></a>                                                 temp_adapt_asv_signif_hits,]</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>      parameters        CAP1        CAP2 variable_type
ASV3        ASV3  0.49583686  0.48216614           ASV
ASV6        ASV6 -0.01759546  0.20461984           ASV
ASV2        ASV2  0.23578207 -0.44144325           ASV
ASV10      ASV10 -0.02475900  0.08142465           ASV
ASV1        ASV1  1.18724098  0.33621587           ASV</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_all<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">&quot;metadata&quot;</span></span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">&quot;ASV&quot;</span></span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioplot_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(temp_adapt_md_signif_all, temp_adapt_asv_signif)</span></code></pre></div>
</details>
</div>
<p>The last thing to do is categorize parameters scores and ASV scores into different variable types for plotting.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioplot_data_md <span class="ot">&lt;-</span> <span class="fu">subset</span>(temp_adapt_bioplot_data, </span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>                                  temp_adapt_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">&quot;metadata&quot;</span>)</span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioplot_data_asv <span class="ot">&lt;-</span> <span class="fu">subset</span>(temp_adapt_bioplot_data, </span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a>                                   temp_adapt_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">&quot;ASV&quot;</span>)</span></code></pre></div>
</details>
</div>
<details>
<summary>
Show the code for the plot
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>temp_adapt_cap_vals <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(temp_adapt_cap<span class="sc">$</span>CCA<span class="sc">$</span>eig[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>temp_adapt_cap1 <span class="ot">&lt;-</span> <span class="fu">signif</span>((temp_adapt_cap_vals[<span class="dv">1</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>temp_adapt_cap2 <span class="ot">&lt;-</span> <span class="fu">signif</span>((temp_adapt_cap_vals[<span class="dv">2</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a>cpa1_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;CAP1&quot;</span>, <span class="st">&quot; (&quot;</span>, temp_adapt_cap1, <span class="st">&quot;%)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb95-6"><a href="#cb95-6" aria-hidden="true" tabindex="-1"></a>cpa2_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;CAP2&quot;</span>, <span class="st">&quot; (&quot;</span>, temp_adapt_cap2, <span class="st">&quot;%)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb95-7"><a href="#cb95-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-8"><a href="#cb95-8" aria-hidden="true" tabindex="-1"></a>swel_col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#2271B2&quot;</span>, <span class="st">&quot;#71B222&quot;</span>, <span class="st">&quot;#B22271&quot;</span>)</span>
<span id="cb95-9"><a href="#cb95-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-10"><a href="#cb95-10" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(temp_adapt_plot_data) <span class="sc">+</span></span>
<span id="cb95-11"><a href="#cb95-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">shape =</span> TREAT,</span>
<span id="cb95-12"><a href="#cb95-12" aria-hidden="true" tabindex="-1"></a>                           <span class="at">colour =</span> TREAT_T), <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb95-13"><a href="#cb95-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> swel_col) <span class="sc">+</span></span>
<span id="cb95-14"><a href="#cb95-14" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_text(data = temp_adapt_plot_data, aes(x = CAP1, y = CAP2, #UNCOMMENT to add sample labels</span></span>
<span id="cb95-15"><a href="#cb95-15" aria-hidden="true" tabindex="-1"></a><span class="co">#                              label = SampleID), size = 3) + </span></span>
<span id="cb95-16"><a href="#cb95-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">xend =</span> CAP1, <span class="at">yend =</span> CAP2),</span>
<span id="cb95-17"><a href="#cb95-17" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> temp_adapt_bioplot_data_md, <span class="at">linetype =</span> <span class="st">&quot;solid&quot;</span>,</span>
<span id="cb95-18"><a href="#cb95-18" aria-hidden="true" tabindex="-1"></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">&quot;cm&quot;</span>)), <span class="at">size =</span> <span class="fl">0.4</span>,</span>
<span id="cb95-19"><a href="#cb95-19" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">&quot;#191919&quot;</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb95-20"><a href="#cb95-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> temp_adapt_bioplot_data_md, </span>
<span id="cb95-21"><a href="#cb95-21" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">label =</span> parameters), <span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb95-22"><a href="#cb95-22" aria-hidden="true" tabindex="-1"></a>            <span class="at">nudge_x =</span> <span class="fl">0.1</span>, <span class="at">nudge_y =</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb95-23"><a href="#cb95-23" aria-hidden="true" tabindex="-1"></a>  <span class="do">### This code adds ASV vectors, Uncomment to add</span></span>
<span id="cb95-24"><a href="#cb95-24" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_segment(aes(x = 0, y = 0, xend = CAP1, yend = CAP2),</span></span>
<span id="cb95-25"><a href="#cb95-25" aria-hidden="true" tabindex="-1"></a><span class="co">#               data = temp_adapt_bioplot_data_asv, linetype = &quot;solid&quot;,</span></span>
<span id="cb95-26"><a href="#cb95-26" aria-hidden="true" tabindex="-1"></a><span class="co">#               arrow = arrow(length = unit(0.3, &quot;cm&quot;)), size = 0.2,</span></span>
<span id="cb95-27"><a href="#cb95-27" aria-hidden="true" tabindex="-1"></a><span class="co">#               color = &quot;#676767&quot;) +</span></span>
<span id="cb95-28"><a href="#cb95-28" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_text(data = temp_adapt_bioplot_data_asv, </span></span>
<span id="cb95-29"><a href="#cb95-29" aria-hidden="true" tabindex="-1"></a><span class="co">#            aes(x = CAP1, y = CAP2, label = parameters), size = 2.5,</span></span>
<span id="cb95-30"><a href="#cb95-30" aria-hidden="true" tabindex="-1"></a><span class="co">#            nudge_x = 0.05, nudge_y = 0.05) +</span></span>
<span id="cb95-31"><a href="#cb95-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span> </span>
<span id="cb95-32"><a href="#cb95-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Capscale Analysis&quot;</span>,</span>
<span id="cb95-33"><a href="#cb95-33" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">&quot;Temperature Adaptation&quot;</span>,</span>
<span id="cb95-34"><a href="#cb95-34" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> cpa1_lab, </span>
<span id="cb95-35"><a href="#cb95-35" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> cpa2_lab)</span>
<span id="cb95-36"><a href="#cb95-36" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot <span class="ot">&lt;-</span> temp_adapt_plot <span class="sc">+</span> <span class="fu">coord_fixed</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb95-37"><a href="#cb95-37" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot</span>
<span id="cb95-38"><a href="#cb95-38" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">&quot;files/metadata/figures/ssu18_temp_adapt_capscale.png&quot;</span>,</span>
<span id="cb95-39"><a href="#cb95-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">16</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">units =</span> <span class="st">&#39;cm&#39;</span>, <span class="at">res =</span> <span class="dv">600</span>, <span class="at">bg =</span> <span class="st">&quot;white&quot;</span>)</span>
<span id="cb95-40"><a href="#cb95-40" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot</span>
<span id="cb95-41"><a href="#cb95-41" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb95-42"><a href="#cb95-42" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">&quot;files/metadata/figures/ssu18_temp_adapt_capscale.pdf&quot;</span>,</span>
<span id="cb95-43"><a href="#cb95-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">5</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb95-44"><a href="#cb95-44" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot</span>
<span id="cb95-45"><a href="#cb95-45" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
</div>
</details>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="capscale-plots">Capscale Plots</h2>
<div class="l-body-outset">
<div class="panelset">
<section id="edaphic-properties-3" class="panel">
<h4>Edaphic properties</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="metadata_files/figure-html5/unnamed-chunk-125-1.png" width="624" /></p>
</div>
</section>
<section id="soil-functional-response-2" class="panel">
<h4>Soil Functional Response</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="metadata_files/figure-html5/unnamed-chunk-126-1.png" width="624" /></p>
</div>
</section>
<section id="temperature-adaptation-3" class="panel">
<h4>Temperature Adaptation</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="metadata_files/figure-html5/unnamed-chunk-127-1.png" width="624" /></p>
</div>
</section>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="its-1">ITS</h1>
<h2 id="data-formating-1">Data Formating</h2>
<ol type="1">
<li>Transform <code>ps</code> Objects</li>
</ol>
<p>The first step is to transform the phyloseq objects.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a>samp_ps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;its18_ps_work&quot;</span>, <span class="st">&quot;its18_ps_pime&quot;</span>, <span class="st">&quot;its18_ps_perfect&quot;</span>, </span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;its18_ps_work_otu&quot;</span>, <span class="st">&quot;its18_ps_pime_otu&quot;</span>, <span class="st">&quot;its18_ps_perfect_otu&quot;</span>)</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> samp_ps) {</span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(tmp_get, <span class="cf">function</span>(otu) <span class="fl">1e5</span> <span class="sc">*</span> otu<span class="sc">/</span><span class="fu">sum</span>(otu))</span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a>     tmp_ps<span class="sc">@</span>phy_tree <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">prune_samples</span>(<span class="fu">sample_sums</span>(tmp_ps) <span class="sc">&gt;</span> <span class="dv">0</span>, tmp_ps)</span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> tmp_samp <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;TREAT_T&quot;</span> <span class="ot">=</span> <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb96-10"><a href="#cb96-10" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(tmp_ps, sample_data)</span>
<span id="cb96-11"><a href="#cb96-11" aria-hidden="true" tabindex="-1"></a>     <span class="fu">sample_data</span>(tmp_ps) <span class="ot">&lt;-</span> tmp_samp</span>
<span id="cb96-12"><a href="#cb96-12" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>))</span>
<span id="cb96-13"><a href="#cb96-13" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_name)</span>
<span id="cb96-14"><a href="#cb96-14" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_ps)</span>
<span id="cb96-15"><a href="#cb96-15" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb96-16"><a href="#cb96-16" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<ol start="2" type="1">
<li>Format for <code>mctoolsr</code></li>
</ol>
<p>Then a bit of formatting to make the data compatible with <code>mctoolsr</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> samp_ps) {</span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>)))</span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>     tmp_tax <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(tmp_get))</span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a>     tmp_tax<span class="sc">$</span>ASV_SEQ <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a>     tmp_col_names <span class="ot">&lt;-</span> <span class="fu">colnames</span>(tmp_tax)</span>
<span id="cb97-7"><a href="#cb97-7" aria-hidden="true" tabindex="-1"></a>     tmp_tax_merge <span class="ot">&lt;-</span> tmp_tax <span class="sc">%&gt;%</span> tidyr<span class="sc">::</span><span class="fu">unite</span>(taxonomy, </span>
<span id="cb97-8"><a href="#cb97-8" aria-hidden="true" tabindex="-1"></a>                                               <span class="fu">all_of</span>(tmp_col_names), </span>
<span id="cb97-9"><a href="#cb97-9" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">sep =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb97-10"><a href="#cb97-10" aria-hidden="true" tabindex="-1"></a>     tmp_tax_merge <span class="ot">&lt;-</span> tmp_tax_merge <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;#OTU_ID&quot;</span>)</span>
<span id="cb97-11"><a href="#cb97-11" aria-hidden="true" tabindex="-1"></a>     tmp_otu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(tmp_get)))</span>
<span id="cb97-12"><a href="#cb97-12" aria-hidden="true" tabindex="-1"></a>     tmp_otu <span class="ot">&lt;-</span> tmp_otu <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;#OTU_ID&quot;</span>)</span>
<span id="cb97-13"><a href="#cb97-13" aria-hidden="true" tabindex="-1"></a>     tmp_otu_tax <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_otu, tmp_tax_merge, <span class="at">by =</span> <span class="st">&quot;#OTU_ID&quot;</span>)</span>
<span id="cb97-14"><a href="#cb97-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-15"><a href="#cb97-15" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb97-16"><a href="#cb97-16" aria-hidden="true" tabindex="-1"></a>     tmp_samp[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>)] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb97-17"><a href="#cb97-17" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> tmp_samp <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;#SampleID&quot;</span>)</span>
<span id="cb97-18"><a href="#cb97-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-19"><a href="#cb97-19" aria-hidden="true" tabindex="-1"></a>     tmp_metad <span class="ot">&lt;-</span> metad <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;#SampleID&quot;</span> <span class="ot">=</span> <span class="st">&quot;id&quot;</span>)</span>
<span id="cb97-20"><a href="#cb97-20" aria-hidden="true" tabindex="-1"></a>     tmp_metad[,<span class="dv">2</span><span class="sc">:</span><span class="dv">5</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb97-21"><a href="#cb97-21" aria-hidden="true" tabindex="-1"></a>     tmp_md <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_samp, tmp_metad, <span class="at">by =</span> <span class="st">&quot;#SampleID&quot;</span>)</span>
<span id="cb97-22"><a href="#cb97-22" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb97-23"><a href="#cb97-23" aria-hidden="true" tabindex="-1"></a>     tmp_otu_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_otu_tax&quot;</span>))</span>
<span id="cb97-24"><a href="#cb97-24" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_otu_name)</span>
<span id="cb97-25"><a href="#cb97-25" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_otu_name, tmp_otu_tax)</span>
<span id="cb97-26"><a href="#cb97-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-27"><a href="#cb97-27" aria-hidden="true" tabindex="-1"></a>     tmp_md_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_md&quot;</span>))</span>
<span id="cb97-28"><a href="#cb97-28" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_md_name)</span>
<span id="cb97-29"><a href="#cb97-29" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_md_name, tmp_md)</span>
<span id="cb97-30"><a href="#cb97-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-31"><a href="#cb97-31" aria-hidden="true" tabindex="-1"></a>     tmp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&quot;files/metadata/tables/&quot;</span>)</span>
<span id="cb97-32"><a href="#cb97-32" aria-hidden="true" tabindex="-1"></a>     <span class="fu">write_delim</span>(tmp_otu_tax, <span class="fu">paste</span>(tmp_path, tmp_otu_name, <span class="st">&quot;.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb97-33"><a href="#cb97-33" aria-hidden="true" tabindex="-1"></a>     <span class="fu">write_delim</span>(tmp_md, <span class="fu">paste</span>(tmp_path, tmp_md_name, <span class="st">&quot;.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb97-34"><a href="#cb97-34" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb97-35"><a href="#cb97-35" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> samp_ps) {</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>     tmp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&quot;files/metadata/tables/&quot;</span>)</span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a>     tmp_otu_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_otu_tax&quot;</span>))</span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a>     tmp_md_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_md&quot;</span>))</span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a>     tmp_tax_table_fp <span class="ot">&lt;-</span> <span class="fu">paste</span>(tmp_path, tmp_otu_name, <span class="st">&quot;.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb98-6"><a href="#cb98-6" aria-hidden="true" tabindex="-1"></a>     tmp_map_fp <span class="ot">&lt;-</span> <span class="fu">paste</span>(tmp_path, tmp_md_name, <span class="st">&quot;.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb98-7"><a href="#cb98-7" aria-hidden="true" tabindex="-1"></a>     tmp_input <span class="ot">&lt;-</span> <span class="fu">load_taxa_table</span>(tmp_tax_table_fp, tmp_map_fp)</span>
<span id="cb98-8"><a href="#cb98-8" aria-hidden="true" tabindex="-1"></a>     tmp_input_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_mc&quot;</span>))</span>
<span id="cb98-9"><a href="#cb98-9" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_input_name)</span>
<span id="cb98-10"><a href="#cb98-10" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_input_name, tmp_input)</span>
<span id="cb98-11"><a href="#cb98-11" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb98-12"><a href="#cb98-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb98-13"><a href="#cb98-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;_md&quot;</span>))</span>
<span id="cb98-14"><a href="#cb98-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;_otu_tax&quot;</span>))</span></code></pre></div>
</details>
</div>
<h2 id="choose-data-set-1">Choose Data Set</h2>
<p>At this point in the code we need to choose a data set to use, formatted with <code>mctoolsr</code>. Remember, there are four choices:</p>
<ol type="1">
<li>Complete ASV data set.</li>
<li>PIME filtered ASV data set.</li>
<li>PERfect filtered ASV data set.</li>
<li>Complete OTU data set.</li>
<li>PIME filtered OTU data set.</li>
<li>PERfect filtered OTU data set.</li>
</ol>
<p>This way, if we want to test other data sets we only need to change the name here.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;_mc&quot;</span>)</span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>its18_select_mc <span class="ot">&lt;-</span> its18_ps_pime_mc</span></code></pre></div>
</div>
<h2 id="normality-tests-1">Normality Tests</h2>
<p>Before proceeding, we need to test each parameter in the metadata to see which ones are and are not normally distributed. For that, we use the Shapiro-Wilk Normality Test. Here we only need one of the metadata files.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>temp_md <span class="ot">&lt;-</span> its18_select_mc<span class="sc">$</span>map_loaded</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>temp_md[,<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>shap_results <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">colnames</span>(temp_md)) {</span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a>  tmp_shap <span class="ot">&lt;-</span> <span class="fu">shapiro.test</span>(temp_md[[i]])</span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a>  tmp_p <span class="ot">&lt;-</span> <span class="fu">round</span>(tmp_shap<span class="sc">$</span>p.value, <span class="at">digits =</span> <span class="dv">5</span>)</span>
<span id="cb100-8"><a href="#cb100-8" aria-hidden="true" tabindex="-1"></a>  tmp_res <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">isTRUE</span>(tmp_shap<span class="sc">$</span>p.value <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span>
<span id="cb100-9"><a href="#cb100-9" aria-hidden="true" tabindex="-1"></a>  shap_results <span class="ot">&lt;-</span> <span class="fu">rbind</span>(shap_results, <span class="fu">data.frame</span>(i, tmp_p, tmp_res))</span>
<span id="cb100-10"><a href="#cb100-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb100-11"><a href="#cb100-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb100-12"><a href="#cb100-12" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(shap_results) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;parameter&quot;</span>, <span class="st">&quot;p-value&quot;</span>, <span class="st">&quot;tranform&quot;</span>)</span>
<span id="cb100-13"><a href="#cb100-13" aria-hidden="true" tabindex="-1"></a>shap_results</span>
<span id="cb100-14"><a href="#cb100-14" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">filter</span>(shap_results, tranform <span class="sc">==</span> <span class="st">&quot;TRUE&quot;</span>)</span>
<span id="cb100-15"><a href="#cb100-15" aria-hidden="true" tabindex="-1"></a>md_to_tranform <span class="ot">&lt;-</span> shap_results<span class="sc">$</span>parameter[shap_results<span class="sc">$</span>tranform <span class="sc">==</span> <span class="cn">TRUE</span>]</span>
<span id="cb100-16"><a href="#cb100-16" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;temp_md&quot;</span>))</span></code></pre></div>
</details>
</div>
<details>
<summary>
Show the results of each normality test for metadata parameters
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>   parameter p-value tranform
1        AST 0.00288     TRUE
2        H2O 0.00272     TRUE
3          N 0.09024    FALSE
4          P 0.46522    FALSE
5         Al 0.00063     TRUE
6         Ca 0.11893    FALSE
7         Fe 0.00088     TRUE
8          K 0.18425    FALSE
9         Mg 0.27096    FALSE
10        Mn 0.26036    FALSE
11        Na 0.11161    FALSE
12       TEB 0.01070     TRUE
13      ECEC 0.01083     TRUE
14        pH 0.62168    FALSE
15    minPO4 0.11512    FALSE
16    minNH4 0.66069    FALSE
17    minNO3 0.00112     TRUE
18    minTIN 0.00133     TRUE
19       NH4 0.05982    FALSE
20       NO3 0.74913    FALSE
21       PO4 0.13322    FALSE
22       DOC 0.02215     TRUE
23       DON 0.29704    FALSE
24      DOCN 0.33006    FALSE
25      micC 0.20336    FALSE
26      micN 0.61244    FALSE
27      micP 0.17886    FALSE
28     micCN 0.07687    FALSE
29     micCP 0.66220    FALSE
30     micNP 0.39825    FALSE
31    AG_ase 0.93318    FALSE
32    BG_ase 0.00403     TRUE
33    BP_ase 0.00669     TRUE
34    CE_ase 0.14601    FALSE
35     P_ase 0.00981     TRUE
36     N_ase 0.00733     TRUE
37     S_ase 0.10815    FALSE
38    XY_ase 0.02719     TRUE
39    AG_Q10 0.26659    FALSE
40    BG_Q10 0.01649     TRUE
41    BP_Q10 0.11576    FALSE
42    CE_Q10 0.03868     TRUE
43     P_Q10 0.34975    FALSE
44     N_Q10 0.76801    FALSE
45     S_Q10 0.40364    FALSE
46    XY_Q10 0.18425    FALSE
47       CO2 0.00011     TRUE
48    LP_ase 0.77547    FALSE
49    LP_Q10 0.80669    FALSE
50    PX_ase 0.01754     TRUE
51    PX_Q10 0.00002     TRUE
52     CUEcn 0.96287    FALSE
53     CUEcp 0.01163     TRUE
54       NUE 0.49521    FALSE
55       PUE 0.00081     TRUE
56      Tmin 0.13400    FALSE
57        SI 0.09324    FALSE
58     enzCN 0.13857    FALSE
59     enzCP 0.87590    FALSE
60     enzNP 0.11430    FALSE</code></pre>
</div>
</details>
<p>Looks like we need to transform 21 metadata parameters.</p>
<h2 id="normalize-parameters-1">Normalize Parameters</h2>
<p>Here we use the R package <a href="https://cran.r-project.org/web/packages/bestNormalize/vignettes/bestNormalize.html"><code>bestNormalize</code></a> to find and execute the best normalizing transformation. The function will test the following normalizing transformations:</p>
<ul>
<li><p><code>arcsinh_x</code> performs an arcsinh transformation.</p></li>
<li><p><code>boxcox</code> Perform a Box-Cox transformation and center/scale a vector to attempt normalization. <code>boxcox</code> estimates the optimal value of lambda for the Box-Cox transformation. The function will return an error if a user attempt to transform nonpositive data.</p></li>
<li><p><code>yeojohnson</code> Perform a Yeo-Johnson Transformation and center/scale a vector to attempt normalization. <code>yeojohnson</code> estimates the optimal value of lambda for the Yeo-Johnson transformation. The Yeo-Johnson is similar to the Box-Cox method, however it allows for the transformation of nonpositive data as well.</p></li>
<li><p><code>orderNorm</code> The Ordered Quantile (ORQ) normalization transformation, <code>orderNorm()</code>, is a rank-based procedure by which the values of a vector are mapped to their percentile, which is then mapped to the same percentile of the normal distribution. Without the presence of ties, this essentially guarantees that the transformation leads to a uniform distribution.</p></li>
<li><p><code>log_x</code> performs a simple log transformation. The parameter a is essentially estimated by the training set by default (estimated as the minimum possible to some extent epsilon), while the base must be specified beforehand. The default base of the log is 10.</p></li>
<li><p><code>sqrt_x</code> performs a simple square-root transformation. The parameter a is essentially estimated by the training set by default (estimated as the minimum possible), while the base must be specified beforehand.</p></li>
<li><p><code>exp_x</code> performs a simple exponential transformation.</p></li>
</ul>
<p>See this GitHub issue (<a href="https://github.com/petersonR/bestNormalize/issues/5">#5</a>) for a description on getting reproducible results. Apparently, you can get different results because the <code>bestNormalize()</code> function uses repeated cross-validation (and doesnt automatically set the seed), so the results will be slightly different each time the function is executed.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">119</span>)</span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> md_to_tranform) {</span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a>  tmp_md <span class="ot">&lt;-</span> its18_select_mc<span class="sc">$</span>map_loaded</span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a>  tmp_best_norm <span class="ot">&lt;-</span> <span class="fu">bestNormalize</span>(tmp_md[[i]], <span class="at">r =</span> <span class="dv">1</span>, <span class="at">k =</span> <span class="dv">5</span>, <span class="at">loo =</span> <span class="cn">TRUE</span>)</span>
<span id="cb102-6"><a href="#cb102-6" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_best_norm_test&quot;</span>))</span>
<span id="cb102-7"><a href="#cb102-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_best_norm)</span>
<span id="cb102-8"><a href="#cb102-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(tmp_name)</span>
<span id="cb102-9"><a href="#cb102-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb102-10"><a href="#cb102-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<details>
<summary>
Show the chosen transformations
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
## bestNormalize Chosen transformation of AST ## 
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
   0%   25%   50%   75%  100% 
25.80 26.26 28.80 29.90 41.77 
_____________________________________

## bestNormalize Chosen transformation of H2O ## 
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
   0%   25%   50%   75%  100% 
0.204 0.261 0.377 0.385 0.401 
_____________________________________

## bestNormalize Chosen transformation of Al ## 
Standardized asinh(x) Transformation with 13 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 0.005384372 
 - sd (before standardization) = 0.007762057 
_____________________________________

## bestNormalize Chosen transformation of Fe ## 
Standardized asinh(x) Transformation with 13 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 0.01076821 
 - sd (before standardization) = 0.0103749 
_____________________________________

## bestNormalize Chosen transformation of TEB ## 
orderNorm Transformation with 13 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
41.6 43.3 45.1 48.7 61.7 
_____________________________________

## bestNormalize Chosen transformation of ECEC ## 
orderNorm Transformation with 13 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
41.7 43.3 45.3 48.8 61.8 
_____________________________________

## bestNormalize Chosen transformation of minNO3 ## 
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 11.31  19.67  22.71  37.33 106.42 
_____________________________________

## bestNormalize Chosen transformation of minTIN ## 
orderNorm Transformation with 13 nonmissing obs and ties
 - 12 unique values 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 12.96  21.50  23.99  42.08 110.75 
_____________________________________

## bestNormalize Chosen transformation of DOC ## 
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 33.20  44.30  56.68  63.79 118.74 
_____________________________________

## bestNormalize Chosen transformation of BG_ase ## 
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
  0%  25%  50%  75% 100% 
3.04 3.92 4.22 4.62 9.09 
_____________________________________

## bestNormalize Chosen transformation of BP_ase ## 
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
  0%  25%  50%  75% 100% 
1.77 2.33 3.02 3.50 7.28 
_____________________________________

## bestNormalize Chosen transformation of P_ase ## 
Standardized Box Cox Transformation with 13 nonmissing obs.:
 Estimated statistics:
 - lambda = -0.9999576 
 - mean (before standardization) = 0.9480149 
 - sd (before standardization) = 0.01593697 
_____________________________________

## bestNormalize Chosen transformation of N_ase ## 
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
  0%  25%  50%  75% 100% 
2.57 2.95 3.58 4.52 8.34 
_____________________________________

## bestNormalize Chosen transformation of XY_ase ## 
orderNorm Transformation with 13 nonmissing obs and ties
 - 11 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
0.66 0.76 1.11 1.36 2.52 
_____________________________________

## bestNormalize Chosen transformation of BG_Q10 ## 
orderNorm Transformation with 13 nonmissing obs and ties
 - 11 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
1.31 1.36 1.39 1.56 1.76 
_____________________________________

## bestNormalize Chosen transformation of CE_Q10 ## 
orderNorm Transformation with 13 nonmissing obs and ties
 - 12 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
1.76 1.87 1.95 2.00 2.36 
_____________________________________

## bestNormalize Chosen transformation of CO2 ## 
Standardized asinh(x) Transformation with 13 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 2.626573 
 - sd (before standardization) = 0.7396036 
_____________________________________

## bestNormalize Chosen transformation of PX_ase ## 
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 81.25  94.13 119.75 211.18 294.39 
_____________________________________

## bestNormalize Chosen transformation of PX_Q10 ## 
Standardized Yeo-Johnson Transformation with 13 nonmissing obs.:
 Estimated statistics:
 - lambda = -4.999963 
 - mean (before standardization) = 0.1968145 
 - sd (before standardization) = 0.001394949 
_____________________________________

## bestNormalize Chosen transformation of CUEcp ## 
orderNorm Transformation with 13 nonmissing obs and ties
 - 9 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
0.15 0.17 0.19 0.21 0.33 
_____________________________________

## bestNormalize Chosen transformation of PUE ## 
orderNorm Transformation with 13 nonmissing obs and ties
 - 6 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
0.77 0.88 0.90 0.91 0.92 
_____________________________________</code></pre>
</div>
</details>
<details>
<summary>
Show the complete <code>bestNormalize</code> results
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
## Results of bestNormalize for AST ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 3.2051
 - Box-Cox: 3.2051
 - Center+scale: 4.1282
 - Exp(x): 17.9744
 - Log_b(x+a): 3.2051
 - orderNorm (ORQ): 0.1282
 - sqrt(x + a): 4.1282
 - Yeo-Johnson: 2.2821
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
   0%   25%   50%   75%  100% 
25.80 26.26 28.80 29.90 41.77 
_____________________________________

## Results of bestNormalize for H2O ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 7.2051
 - Box-Cox: 6.2821
 - Center+scale: 8.1282
 - Exp(x): 6.2821
 - Log_b(x+a): 7.2051
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 7.2051
 - Yeo-Johnson: 6.2821
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
   0%   25%   50%   75%  100% 
0.204 0.261 0.377 0.385 0.401 
_____________________________________

## Results of bestNormalize for Al ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 7.5128
 - Center+scale: 7.5128
 - Exp(x): 7.5128
 - Log_b(x+a): 9.359
 - orderNorm (ORQ): 7.5128
 - sqrt(x + a): 7.5128
 - Yeo-Johnson: 7.5128
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized asinh(x) Transformation with 13 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 0.005384372 
 - sd (before standardization) = 0.007762057 
_____________________________________

## Results of bestNormalize for Fe ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 7.2051
 - Center+scale: 7.2051
 - Exp(x): 7.2051
 - Log_b(x+a): 7.2051
 - orderNorm (ORQ): 7.2051
 - sqrt(x + a): 7.2051
 - Yeo-Johnson: 7.2051
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized asinh(x) Transformation with 13 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 0.01076821 
 - sd (before standardization) = 0.0103749 
_____________________________________

## Results of bestNormalize for TEB ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.6667
 - Box-Cox: 1.6667
 - Center+scale: 4.7436
 - Exp(x): 17.9744
 - Log_b(x+a): 1.6667
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 3.2051
 - Yeo-Johnson: 1.359
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
41.6 43.3 45.1 48.7 61.7 
_____________________________________

## Results of bestNormalize for ECEC ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.6667
 - Box-Cox: 1.6667
 - Center+scale: 3.2051
 - Exp(x): 17.9744
 - Log_b(x+a): 1.6667
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 3.2051
 - Yeo-Johnson: 1.359
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
41.7 43.3 45.3 48.8 61.8 
_____________________________________

## Results of bestNormalize for minNO3 ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 3.5128
 - Box-Cox: 3.5128
 - Center+scale: 5.359
 - Exp(x): 17.9744
 - Log_b(x+a): 3.5128
 - orderNorm (ORQ): 0.7436
 - sqrt(x + a): 2.5897
 - Yeo-Johnson: 3.5128
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 11.31  19.67  22.71  37.33 106.42 
_____________________________________

## Results of bestNormalize for minTIN ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.359
 - Box-Cox: 2.2821
 - Center+scale: 4.1282
 - Exp(x): 17.9744
 - Log_b(x+a): 1.359
 - orderNorm (ORQ): 0.7436
 - sqrt(x + a): 2.2821
 - Yeo-Johnson: 2.2821
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and ties
 - 12 unique values 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 12.96  21.50  23.99  42.08 110.75 
_____________________________________

## Results of bestNormalize for DOC ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 2.8974
 - Box-Cox: 2.8974
 - Center+scale: 3.5128
 - Exp(x): 17.9744
 - Log_b(x+a): 2.8974
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 1.9744
 - Yeo-Johnson: 2.8974
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 33.20  44.30  56.68  63.79 118.74 
_____________________________________

## Results of bestNormalize for BG_ase ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 3.8205
 - Box-Cox: 2.2821
 - Center+scale: 4.1282
 - Exp(x): 14.8974
 - Log_b(x+a): 3.8205
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 3.5128
 - Yeo-Johnson: 1.6667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
  0%  25%  50%  75% 100% 
3.04 3.92 4.22 4.62 9.09 
_____________________________________

## Results of bestNormalize for BP_ase ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.6667
 - Box-Cox: 1.6667
 - Center+scale: 4.1282
 - Exp(x): 5.6667
 - Log_b(x+a): 1.6667
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 1.6667
 - Yeo-Johnson: 1.6667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
  0%  25%  50%  75% 100% 
1.77 2.33 3.02 3.50 7.28 
_____________________________________

## Results of bestNormalize for P_ase ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.6667
 - Box-Cox: 0.4359
 - Center+scale: 3.2051
 - Exp(x): 17.9744
 - Log_b(x+a): 1.6667
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 1.9744
 - Yeo-Johnson: 0.4359
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized Box Cox Transformation with 13 nonmissing obs.:
 Estimated statistics:
 - lambda = -0.9999576 
 - mean (before standardization) = 0.9480149 
 - sd (before standardization) = 0.01593697 
_____________________________________

## Results of bestNormalize for N_ase ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.9744
 - Box-Cox: 2.2821
 - Center+scale: 2.8974
 - Exp(x): 17.9744
 - Log_b(x+a): 1.9744
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 2.8974
 - Yeo-Johnson: 0.4359
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
  0%  25%  50%  75% 100% 
2.57 2.95 3.58 4.52 8.34 
_____________________________________

## Results of bestNormalize for XY_ase ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.0513
 - Box-Cox: 0.4359
 - Center+scale: 2.8974
 - Exp(x): 9.0513
 - Log_b(x+a): 1.0513
 - orderNorm (ORQ): 0.1282
 - sqrt(x + a): 1.359
 - Yeo-Johnson: 0.4359
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and ties
 - 11 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
0.66 0.76 1.11 1.36 2.52 
_____________________________________

## Results of bestNormalize for BG_Q10 ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.9744
 - Box-Cox: 1.9744
 - Center+scale: 2.2821
 - Exp(x): 5.359
 - Log_b(x+a): 1.9744
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 1.9744
 - Yeo-Johnson: 1.9744
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and ties
 - 11 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
1.31 1.36 1.39 1.56 1.76 
_____________________________________

## Results of bestNormalize for CE_Q10 ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.6667
 - Box-Cox: 1.6667
 - Center+scale: 1.6667
 - Exp(x): 4.1282
 - Log_b(x+a): 1.6667
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 1.6667
 - Yeo-Johnson: 0.7436
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and ties
 - 12 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
1.76 1.87 1.95 2.00 2.36 
_____________________________________

## Results of bestNormalize for CO2 ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.0513
 - Box-Cox: 3.5128
 - Center+scale: 11.5128
 - Exp(x): 17.9744
 - Log_b(x+a): 1.0513
 - orderNorm (ORQ): 1.0513
 - sqrt(x + a): 2.8974
 - Yeo-Johnson: 1.9744
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized asinh(x) Transformation with 13 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 2.626573 
 - sd (before standardization) = 0.7396036 
_____________________________________

## Results of bestNormalize for PX_ase ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.9744
 - Box-Cox: 2.2821
 - Center+scale: 4.1282
 - Exp(x): 17.9744
 - Log_b(x+a): 1.9744
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 1.9744
 - Yeo-Johnson: 2.2821
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 81.25  94.13 119.75 211.18 294.39 
_____________________________________

## Results of bestNormalize for PX_Q10 ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 14.5897
 - Box-Cox: 3.2051
 - Center+scale: 17.9744
 - Exp(x): 17.9744
 - Log_b(x+a): 11.5128
 - orderNorm (ORQ): 2.2821
 - sqrt(x + a): 17.9744
 - Yeo-Johnson: 1.0513
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized Yeo-Johnson Transformation with 13 nonmissing obs.:
 Estimated statistics:
 - lambda = -4.999963 
 - mean (before standardization) = 0.1968145 
 - sd (before standardization) = 0.001394949 
_____________________________________

## Results of bestNormalize for CUEcp ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 3.5128
 - Box-Cox: 1.359
 - Center+scale: 3.5128
 - Exp(x): 4.4359
 - Log_b(x+a): 1.359
 - orderNorm (ORQ): 0.7436
 - sqrt(x + a): 3.5128
 - Yeo-Johnson: 2.5897
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and ties
 - 9 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
0.15 0.17 0.19 0.21 0.33 
_____________________________________

## Results of bestNormalize for PUE ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 5.9744
 - Box-Cox: 7.2051
 - Center+scale: 5.9744
 - Exp(x): 5.9744
 - Log_b(x+a): 7.2051
 - orderNorm (ORQ): 2.5897
 - sqrt(x + a): 5.9744
 - Yeo-Johnson: 5.9744
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and ties
 - 6 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
0.77 0.88 0.90 0.91 0.92 
_____________________________________</code></pre>
</div>
</details>
<p>Great, now we can add the normalized transformed data back to our <code>mctoolsr</code> metadata file.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a>its18_select_mc_norm <span class="ot">&lt;-</span> its18_select_mc</span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> md_to_tranform) {</span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a>      tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_best_norm_test&quot;</span>)))</span>
<span id="cb105-5"><a href="#cb105-5" aria-hidden="true" tabindex="-1"></a>      tmp_new_data <span class="ot">&lt;-</span> tmp_get<span class="sc">$</span>x.t</span>
<span id="cb105-6"><a href="#cb105-6" aria-hidden="true" tabindex="-1"></a>      its18_select_mc_norm<span class="sc">$</span>map_loaded[[i]] <span class="ot">&lt;-</span> tmp_new_data</span>
<span id="cb105-7"><a href="#cb105-7" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb105-8"><a href="#cb105-8" aria-hidden="true" tabindex="-1"></a>}     </span></code></pre></div>
</details>
</div>
<p>And rerun the Shapiro Tests.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>temp_md_norm <span class="ot">&lt;-</span> its18_select_mc_norm<span class="sc">$</span>map_loaded</span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>temp_md_norm[,<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a>shap_results_norm <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">colnames</span>(temp_md_norm)) {</span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a>  tmp_shap <span class="ot">&lt;-</span> <span class="fu">shapiro.test</span>(temp_md_norm[[i]])</span>
<span id="cb106-7"><a href="#cb106-7" aria-hidden="true" tabindex="-1"></a>  tmp_p <span class="ot">&lt;-</span> <span class="fu">round</span>(tmp_shap<span class="sc">$</span>p.value, <span class="at">digits =</span> <span class="dv">5</span>)</span>
<span id="cb106-8"><a href="#cb106-8" aria-hidden="true" tabindex="-1"></a>  tmp_res <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">isTRUE</span>(tmp_shap<span class="sc">$</span>p.value <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span>
<span id="cb106-9"><a href="#cb106-9" aria-hidden="true" tabindex="-1"></a>  shap_results_norm <span class="ot">&lt;-</span> <span class="fu">rbind</span>(shap_results_norm, <span class="fu">data.frame</span>(i, tmp_p, tmp_res))</span>
<span id="cb106-10"><a href="#cb106-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb106-11"><a href="#cb106-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb106-12"><a href="#cb106-12" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(shap_results_norm) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;parameter&quot;</span>, <span class="st">&quot;p-value&quot;</span>, <span class="st">&quot;tranform&quot;</span>)</span>
<span id="cb106-13"><a href="#cb106-13" aria-hidden="true" tabindex="-1"></a>shap_results_norm</span>
<span id="cb106-14"><a href="#cb106-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;temp_md_norm&quot;</span>))</span></code></pre></div>
</details>
</div>
<p>And check if there are any parameters that are still significant for the normality test.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a>shap_results<span class="sc">$</span>parameter[shap_results_norm<span class="sc">$</span>tranform <span class="sc">==</span> <span class="cn">TRUE</span>]</span></code></pre></div>
<pre><code>[1] &quot;Al&quot; &quot;Fe&quot;</code></pre>
</div>
<p>Ok. Looks like <code>bestNormalize</code> was unable to find a suitable transformation for <code>Al</code> and <code>Fe</code>. This is likely because there is very little variation in these metadata and/or there are too few significant digits.</p>
<h2 id="normalized-metadata-1">Normalized Metadata</h2>
<p>Finally, here is a new summary table that includes all of the normalized data.</p>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="ffn29i70ljbq7tir2u6u" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="ffn29i70ljbq7tir2u6u">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Metadata summary for each sample.<\/em>\n<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13"],["P10_D00_010_C0E","P02_D00_010_C0A","P04_D00_010_C0B","P06_D00_010_C0C","P08_D00_010_C0D","P01_D00_010_W3A","P03_D00_010_W3B","P07_D00_010_W3D","P09_D00_010_W3E","P01_D00_010_W8A","P03_D00_010_W8B","P05_D00_010_W8C","P07_D00_010_W8D"],["P10","P02","P04","P06","P08","P01","P03","P07","P09","P01","P03","P05","P07"],["C","C","C","C","C","W","W","W","W","W","W","W","W"],[0,0,0,0,0,3,3,3,3,8,8,8,8],["E","A","B","C","D","A","B","D","E","A","B","C","D"],["W","W","W","W","W","W","W","W","W","W","W","W","W"],[26,26,26,26,26,29,29,29,29,34,34,34,34],[-1.19837970230692,-0.869423773288886,-0.395725295814487,-0.615141104595974,-1.76882503851871,0.395725295814487,0.194028142423926,-0.194028142423926,0,0.869423773288886,1.76882503851871,1.19837970230692,0.615141104595973],[-0.194028142423926,0.194028142423926,0.395725295814487,0,1.76882503851871,0.869423773288886,1.19837970230692,0.615141104595973,-0.395725295814487,-0.615141104595974,-0.869423773288886,-1.19837970230692,-1.76882503851871],[0.26,0.3367,0.3033,0.3333,0.2433,0.4267,0.32,0.4367,0.2833,0.4267,0.32,0.3333,0.4367],[0.32,0.43,0.33,0.3,0.27,0.35,0.37,0.35,0.36,0.35,0.37,0.33,0.35],[1.88278632503216,0.594618335419179,-0.693678457040233,-0.693678457040233,-0.693678457040233,-0.693678457040233,0.594618335419179,-0.693678457040233,1.88278632503216,-0.693678457040233,0.594618335419179,-0.693678457040233,-0.693678457040233],[37.06,36.65,29.23,32.07,25.49,31.43,27.19,28.13,39.79,31.43,27.19,26.14,28.13],[-0.0740607187919569,-0.0740607187919569,0.889691753751938,-1.03790955614442,-1.03790955614442,-0.0740607187919569,-0.0740607187919569,-0.0740607187919569,2.81652266501696,-0.0740607187919569,-0.0740607187919569,-1.03790955614442,-0.0740607187919569],[1.38,0.69,1.18,0.57,1.86,1.72,0.85,1.6,1.09,1.72,0.85,0.67,1.6],[21.43,8.16,14.59,21.32,14.93,9.98,16.51,18.87,20.64,9.98,16.51,14.83,18.87],[0.07,0.04,0.12,0.06,0.02,0.04,0.06,0.07,0.08,0.04,0.06,0.05,0.07],[0.13,0.11,0.14,0.14,0.08,0.15,0.12,0.1,0.19,0.15,0.12,0,0.1],[1.19837970230692,0.194028142423926,0,0.869423773288886,-1.19837970230692,-0.736315917376129,-0.293381232121193,0.502402223373355,1.76882503851871,-0.736315917376129,-0.293381232121193,-1.76882503851871,0.502402223373355],[1.19837970230692,0.194028142423926,0,0.869423773288886,-1.19837970230692,-0.736315917376129,-0.293381232121193,0.502402223373355,1.76882503851871,-0.736315917376129,-0.293381232121193,-1.76882503851871,0.502402223373355],[6.28,6.18,5.73,5.93,6.01,6.01,5.86,6.21,6.1,6.01,5.86,5.87,6.21],[4.08,1.42,1.5,0.75,3.67,1.33,2.83,1,2.58,0.12,0.2,0.11,0.02],[4.75,3.92,3.17,1.83,4.33,3.67,3.33,2.42,3.5,1.65,1.28,1.18,2.6],[0.615141104595973,0.194028142423926,-0.395725295814487,-0.615141104595974,1.76882503851871,1.19837970230692,0.869423773288886,-0.194028142423926,-1.19837970230692,-1.76882503851871,0,0.395725295814487,-0.869423773288886],[0.615141104595973,0.194028142423926,-0.293381232121193,-0.615141104595974,1.76882503851871,1.19837970230692,0.869423773288886,-0.293381232121193,-0.869423773288886,-1.76882503851871,0,0.395725295814487,-1.19837970230692],[3.44,4.81,5.57,2.92,2.04,8.03,3.11,4.28,2.03,4.08,2.5,3.91,8.19],[4.53,3.13,3.22,2.27,3.71,4.67,2.57,5.07,2.9,2.65,1.06,2.3,6.39],[9.45,7.58,7.95,7.04,5.97,9.52,8.66,9.15,8.28,8.55,2.8,4.25,5.62],[0.615141104595973,1.19837970230692,0.194028142423926,0,-0.395725295814487,1.76882503851871,0.395725295814487,0.869423773288886,-0.869423773288886,-1.19837970230692,-1.76882503851871,-0.194028142423926,-0.615141104595974],[4.91,8.22,6.32,5.42,5.22,11.66,6.48,8.47,5.56,4.2,2.5,4.3,3.7],[14.67,9.44,10.67,11.09,11.57,11.15,10.12,9.93,12.21,9.45,13.29,10.46,11.98],[334.93,444.72,440.6,390.66,277.37,690.11,364.62,396.42,392.56,469,117,440,676],[72.64,110.13,86.19,72.61,56.55,135.48,66.4,73.73,62.37,84,20,97,147],[64.23,65.03,81.45,85.7,33.4,87.98,70.68,93.13,66.65,55,11,56,94],[4.49,4.42,5.14,5.26,4.72,4.61,5.07,4.72,5.78,5.58,5.85,4.54,4.6],[5.54,8.13,5.39,4.5,11.55,8.61,6.7,4.66,7.34,8.48,10.54,7.91,7.18],[1.17,2.37,1.09,0.83,2.45,1.74,1.31,0.98,1.19,1.52,1.8,1.74,1.56],[0.28,0.24,0.22,0.21,0.15,0.26,0.33,0.32,0.2,0.15,0.02,0.15,0.39],[0.615141104595973,0,0.395725295814487,-1.19837970230692,-0.869423773288886,-0.395725295814487,0.194028142423926,-0.194028142423926,-0.615141104595974,0.869423773288886,-1.76882503851871,1.76882503851871,1.19837970230692],[-1.19837970230692,-0.395725295814487,0,-0.615141104595974,-1.76882503851871,0.615141104595973,0.194028142423926,0.395725295814487,-0.869423773288886,0.869423773288886,-0.194028142423926,1.76882503851871,1.19837970230692],[0.86,1.09,1.09,0.75,0.68,1,1.26,1.07,0.86,2.07,0.65,1.5,1.53],[-0.975759179338554,-0.899790783189584,0.168484710247821,-0.978626089603386,-1.55544155005499,0.192731136000962,-0.00234918359593116,0.566462425407177,-0.448873900343119,0.817855915966737,-0.200773603716638,1.83951429899617,1.47656580322338],[-0.395725295814487,-0.194028142423926,0.194028142423926,-1.76882503851871,-0.615141104595974,1.19837970230692,0,0.395725295814487,-1.19837970230692,0.615141104595973,-0.869423773288886,0.869423773288886,1.76882503851871],[4.54,3.8,4.92,5.45,3.56,6.08,5.72,5.86,5.26,6.7,1.28,5.57,10.68],[-0.869423773288886,-0.293381232121193,0.0965586152896391,-1.76882503851871,-1.19837970230692,0.395725295814487,0.615141104595973,0.0965586152896391,-0.293381232121193,1.76882503851871,-0.615141104595974,1.19837970230692,0.869423773288886],[1.54,1.51,1.51,1.83,2.08,1.77,2.48,1.72,1.67,2.01,1.49,1.73,1.31],[-0.395725295814487,-1.42607687227285,0.395725295814487,-1.42607687227285,1.76882503851871,-0.194028142423926,-0.736315917376129,0.194028142423926,-0.736315917376129,1.19837970230692,0.615141104595973,0.869423773288886,0],[1.41,1.34,1.58,1.44,1.74,1.39,1.42,1.46,1.56,1.51,1.55,1.42,1.42],[-1.0200762327862,-0.395725295814487,1.19837970230692,1.76882503851871,0.615141104595973,0.869423773288886,0.395725295814487,-1.76882503851871,-1.0200762327862,-0.615141104595974,-0.194028142423926,0,0.194028142423926],[1.27,1.22,1.37,1.24,1.48,1.37,1.27,1.28,1.35,1.39,1.51,1.48,1.33],[1.47,1.54,1.85,1.6,1.69,1.46,1.54,1.36,1.48,1.56,1.75,1.68,1.54],[1.33,1.23,1.39,1.47,1.48,1.3,1.27,1.3,1.32,1.65,1.39,1.6,1.51],[1.65,1.59,1.76,1.63,1.83,1.72,1.72,1.63,1.84,1.86,1.89,1.88,1.91],[-0.370144648816923,-0.354937162488166,-0.699577270538405,-0.630197430712108,-1.55180710823738,0.723055311317092,-0.575902041401821,-0.140654760149001,0.4178496338138,-0.630197430712108,2.38950700364869,1.08172615698926,0.34127974728707],[32.22,38.61,29.51,27.78,21.96,52.8,32.1,42.4,30.37,40.69,13.94,32.65,57.13],[1.8,1.81,2.14,1.85,2.07,1.76,2,1.96,1.9,1.71,1.77,1.67,1.4],[-0.194028142423926,-0.615141104595974,-1.76882503851871,-0.869423773288886,-1.19837970230692,0.395725295814487,0,-0.395725295814487,0.194028142423926,0.615141104595973,0.869423773288886,1.19837970230692,1.76882503851871],[-0.909976994138396,1.03036446147988,-0.257786697498746,0.901428992117879,2.21348655625386,-0.374391528355771,0.521020931392077,-0.8363720967674,0.44598468640243,-1.48791276896202,-0.8363720967674,-0.315304817113549,-0.0941686280428208],[0.22,0.22,0.25,0.27,0.24,0.32,0.26,0.27,0.26,0.23,0.29,0.19,0.26],[-1.0200762327862,0.395725295814487,0,-1.0200762327862,0.869423773288886,0.395725295814487,-0.293381232121193,-1.76882503851871,0.395725295814487,-0.293381232121193,1.76882503851871,1.19837970230692,-0.615141104595974],[0.87,0.87,0.85,0.83,0.85,0.77,0.84,0.83,0.84,0.87,0.81,0.9,0.84],[1.19837970230692,-0.502402223373355,0.194028142423926,1.19837970230692,-0.502402223373355,-0.502402223373355,0.194028142423926,1.19837970230692,-0.502402223373355,0.194028142423926,-1.76882503851871,-1.19837970230692,0.615141104595973],[-2.46,0.14,-2.63,-0.13,0,0.07,-0.29,0.65,-0.24,2.19,0.63,0.94,1.48],[1.29,1.44,1.25,1.42,1.54,1.51,1.49,1.61,1.51,2.3,1.72,2.08,1.81],[0.5231,0.5024,0.555,0.4624,0.5058,0.4614,0.5565,0.4941,0.5112,0.6456,0.5302,0.6931,0.5933],[0.6607,0.6554,0.6173,0.5556,0.6037,0.5911,0.6412,0.5762,0.6064,0.7166,0.4918,0.6412,0.6628],[1.263,1.3044,1.1123,1.2017,1.1935,1.2812,1.1522,1.1662,1.1862,1.11,0.9277,0.9251,1.1173]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>id<\/th>\n      <th>PLOT<\/th>\n      <th>TREAT<\/th>\n      <th>TREAT_T<\/th>\n      <th>PAIR<\/th>\n      <th>SEASON<\/th>\n      <th>ATAP<\/th>\n      <th>AST<\/th>\n      <th>H2O<\/th>\n      <th>N<\/th>\n      <th>P<\/th>\n      <th>Al<\/th>\n      <th>Ca<\/th>\n      <th>Fe<\/th>\n      <th>K<\/th>\n      <th>Mg<\/th>\n      <th>Mn<\/th>\n      <th>Na<\/th>\n      <th>TEB<\/th>\n      <th>ECEC<\/th>\n      <th>pH<\/th>\n      <th>minPO4<\/th>\n      <th>minNH4<\/th>\n      <th>minNO3<\/th>\n      <th>minTIN<\/th>\n      <th>NH4<\/th>\n      <th>NO3<\/th>\n      <th>PO4<\/th>\n      <th>DOC<\/th>\n      <th>DON<\/th>\n      <th>DOCN<\/th>\n      <th>micC<\/th>\n      <th>micN<\/th>\n      <th>micP<\/th>\n      <th>micCN<\/th>\n      <th>micCP<\/th>\n      <th>micNP<\/th>\n      <th>AG_ase<\/th>\n      <th>BG_ase<\/th>\n      <th>BP_ase<\/th>\n      <th>CE_ase<\/th>\n      <th>P_ase<\/th>\n      <th>N_ase<\/th>\n      <th>S_ase<\/th>\n      <th>XY_ase<\/th>\n      <th>AG_Q10<\/th>\n      <th>BG_Q10<\/th>\n      <th>BP_Q10<\/th>\n      <th>CE_Q10<\/th>\n      <th>P_Q10<\/th>\n      <th>N_Q10<\/th>\n      <th>S_Q10<\/th>\n      <th>XY_Q10<\/th>\n      <th>CO2<\/th>\n      <th>LP_ase<\/th>\n      <th>LP_Q10<\/th>\n      <th>PX_ase<\/th>\n      <th>PX_Q10<\/th>\n      <th>CUEcn<\/th>\n      <th>CUEcp<\/th>\n      <th>NUE<\/th>\n      <th>PUE<\/th>\n      <th>Tmin<\/th>\n      <th>SI<\/th>\n      <th>enzCN<\/th>\n      <th>enzCP<\/th>\n      <th>enzNP<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"targets":8,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":9,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":10,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":12,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":13,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":14,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":19,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":20,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":24,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":25,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":29,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":31,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":35,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":39,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":40,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":41,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":42,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":43,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":45,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":47,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":48,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":54,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":57,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":58,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":60,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":62,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":65,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":66,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"targets":67,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\");\n  }"},{"className":"dt-right","targets":[4,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\nvar value=data[8]; $(this.api().cell(row, 8).node()).css({'font-size':'80%'});\nvar value=data[9]; $(this.api().cell(row, 9).node()).css({'font-size':'80%'});\nvar value=data[10]; $(this.api().cell(row, 10).node()).css({'font-size':'80%'});\nvar value=data[11]; $(this.api().cell(row, 11).node()).css({'font-size':'80%'});\nvar value=data[12]; $(this.api().cell(row, 12).node()).css({'font-size':'80%'});\nvar value=data[13]; $(this.api().cell(row, 13).node()).css({'font-size':'80%'});\nvar value=data[14]; $(this.api().cell(row, 14).node()).css({'font-size':'80%'});\nvar value=data[15]; $(this.api().cell(row, 15).node()).css({'font-size':'80%'});\nvar value=data[16]; $(this.api().cell(row, 16).node()).css({'font-size':'80%'});\nvar value=data[17]; $(this.api().cell(row, 17).node()).css({'font-size':'80%'});\nvar value=data[18]; $(this.api().cell(row, 18).node()).css({'font-size':'80%'});\nvar value=data[19]; $(this.api().cell(row, 19).node()).css({'font-size':'80%'});\nvar value=data[20]; $(this.api().cell(row, 20).node()).css({'font-size':'80%'});\nvar value=data[21]; $(this.api().cell(row, 21).node()).css({'font-size':'80%'});\nvar value=data[22]; $(this.api().cell(row, 22).node()).css({'font-size':'80%'});\nvar value=data[23]; $(this.api().cell(row, 23).node()).css({'font-size':'80%'});\nvar value=data[24]; $(this.api().cell(row, 24).node()).css({'font-size':'80%'});\nvar value=data[25]; $(this.api().cell(row, 25).node()).css({'font-size':'80%'});\nvar value=data[26]; $(this.api().cell(row, 26).node()).css({'font-size':'80%'});\nvar value=data[27]; $(this.api().cell(row, 27).node()).css({'font-size':'80%'});\nvar value=data[28]; $(this.api().cell(row, 28).node()).css({'font-size':'80%'});\nvar value=data[29]; $(this.api().cell(row, 29).node()).css({'font-size':'80%'});\nvar value=data[30]; $(this.api().cell(row, 30).node()).css({'font-size':'80%'});\nvar value=data[31]; $(this.api().cell(row, 31).node()).css({'font-size':'80%'});\nvar value=data[32]; $(this.api().cell(row, 32).node()).css({'font-size':'80%'});\nvar value=data[33]; $(this.api().cell(row, 33).node()).css({'font-size':'80%'});\nvar value=data[34]; $(this.api().cell(row, 34).node()).css({'font-size':'80%'});\nvar value=data[35]; $(this.api().cell(row, 35).node()).css({'font-size':'80%'});\nvar value=data[36]; $(this.api().cell(row, 36).node()).css({'font-size':'80%'});\nvar value=data[37]; $(this.api().cell(row, 37).node()).css({'font-size':'80%'});\nvar value=data[38]; $(this.api().cell(row, 38).node()).css({'font-size':'80%'});\nvar value=data[39]; $(this.api().cell(row, 39).node()).css({'font-size':'80%'});\nvar value=data[40]; $(this.api().cell(row, 40).node()).css({'font-size':'80%'});\nvar value=data[41]; $(this.api().cell(row, 41).node()).css({'font-size':'80%'});\nvar value=data[42]; $(this.api().cell(row, 42).node()).css({'font-size':'80%'});\nvar value=data[43]; $(this.api().cell(row, 43).node()).css({'font-size':'80%'});\nvar value=data[44]; $(this.api().cell(row, 44).node()).css({'font-size':'80%'});\nvar value=data[45]; $(this.api().cell(row, 45).node()).css({'font-size':'80%'});\nvar value=data[46]; $(this.api().cell(row, 46).node()).css({'font-size':'80%'});\nvar value=data[47]; $(this.api().cell(row, 47).node()).css({'font-size':'80%'});\nvar value=data[48]; $(this.api().cell(row, 48).node()).css({'font-size':'80%'});\nvar value=data[49]; $(this.api().cell(row, 49).node()).css({'font-size':'80%'});\nvar value=data[50]; $(this.api().cell(row, 50).node()).css({'font-size':'80%'});\nvar value=data[51]; $(this.api().cell(row, 51).node()).css({'font-size':'80%'});\nvar value=data[52]; $(this.api().cell(row, 52).node()).css({'font-size':'80%'});\nvar value=data[53]; $(this.api().cell(row, 53).node()).css({'font-size':'80%'});\nvar value=data[54]; $(this.api().cell(row, 54).node()).css({'font-size':'80%'});\nvar value=data[55]; $(this.api().cell(row, 55).node()).css({'font-size':'80%'});\nvar value=data[56]; $(this.api().cell(row, 56).node()).css({'font-size':'80%'});\nvar value=data[57]; $(this.api().cell(row, 57).node()).css({'font-size':'80%'});\nvar value=data[58]; $(this.api().cell(row, 58).node()).css({'font-size':'80%'});\nvar value=data[59]; $(this.api().cell(row, 59).node()).css({'font-size':'80%'});\nvar value=data[60]; $(this.api().cell(row, 60).node()).css({'font-size':'80%'});\nvar value=data[61]; $(this.api().cell(row, 61).node()).css({'font-size':'80%'});\nvar value=data[62]; $(this.api().cell(row, 62).node()).css({'font-size':'80%'});\nvar value=data[63]; $(this.api().cell(row, 63).node()).css({'font-size':'80%'});\nvar value=data[64]; $(this.api().cell(row, 64).node()).css({'font-size':'80%'});\nvar value=data[65]; $(this.api().cell(row, 65).node()).css({'font-size':'80%'});\nvar value=data[66]; $(this.api().cell(row, 66).node()).css({'font-size':'80%'});\nvar value=data[67]; $(this.api().cell(row, 67).node()).css({'font-size':'80%'});\n}"}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render","options.columnDefs.5.render","options.columnDefs.6.render","options.columnDefs.7.render","options.columnDefs.8.render","options.columnDefs.9.render","options.columnDefs.10.render","options.columnDefs.11.render","options.columnDefs.12.render","options.columnDefs.13.render","options.columnDefs.14.render","options.columnDefs.15.render","options.columnDefs.16.render","options.columnDefs.17.render","options.columnDefs.18.render","options.columnDefs.19.render","options.columnDefs.20.render","options.columnDefs.21.render","options.columnDefs.22.render","options.columnDefs.23.render","options.columnDefs.24.render","options.columnDefs.25.render","options.columnDefs.26.render","options.columnDefs.27.render","options.columnDefs.28.render","options.rowCallback"],"jsHooks":[]}</script>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="autocorrelation-tests-1">Autocorrelation Tests</h2>
<p>Next, we test the metadata for autocorrelations. Do we do this on the original data or the transformed data? No idea, so lets do both.</p>
<h3 id="split-metadata-1">Split Metadata</h3>
<p>We need to split the data into different groups.</p>
<ol type="A">
<li>Environmental and edaphic properties</li>
<li>Microbial functional responses</li>
<li>Temperature adaptation properties</li>
</ol>
<p>We first create lists of metadata parameters.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a>div <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;PLOT&quot;</span>, <span class="st">&quot;TREAT&quot;</span>, <span class="st">&quot;TREAT_T&quot;</span>, <span class="st">&quot;PAIR&quot;</span>, <span class="st">&quot;Observed&quot;</span>, <span class="st">&quot;Shannon_exp&quot;</span>, </span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a>         <span class="st">&quot;InvSimpson&quot;</span>, <span class="st">&quot;ATAP&quot;</span>)</span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-4"><a href="#cb109-4" aria-hidden="true" tabindex="-1"></a>edaphic <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;AST&quot;</span>, <span class="st">&quot;H2O&quot;</span>, <span class="st">&quot;N&quot;</span>, <span class="st">&quot;P&quot;</span>, <span class="st">&quot;Al&quot;</span>, </span>
<span id="cb109-5"><a href="#cb109-5" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;Ca&quot;</span>, <span class="st">&quot;Fe&quot;</span>, <span class="st">&quot;K&quot;</span>, <span class="st">&quot;Mg&quot;</span>, <span class="st">&quot;Mn&quot;</span>, <span class="st">&quot;Na&quot;</span>, <span class="st">&quot;TEB&quot;</span>, <span class="st">&quot;ECEC&quot;</span>, <span class="st">&quot;pH&quot;</span>, </span>
<span id="cb109-6"><a href="#cb109-6" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;NH4&quot;</span>, <span class="st">&quot;NO3&quot;</span>, <span class="st">&quot;PO4&quot;</span>, <span class="st">&quot;DOC&quot;</span>, <span class="st">&quot;DON&quot;</span>, <span class="st">&quot;DOCN&quot;</span>)</span>
<span id="cb109-7"><a href="#cb109-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-8"><a href="#cb109-8" aria-hidden="true" tabindex="-1"></a>soil_funct <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;micC&quot;</span>, <span class="st">&quot;micN&quot;</span>, <span class="st">&quot;micP&quot;</span>, <span class="st">&quot;micCN&quot;</span>, <span class="st">&quot;micCP&quot;</span>, <span class="st">&quot;micNP&quot;</span>, </span>
<span id="cb109-9"><a href="#cb109-9" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot;AG_ase&quot;</span>, <span class="st">&quot;BG_ase&quot;</span>, <span class="st">&quot;BP_ase&quot;</span>, <span class="st">&quot;CE_ase&quot;</span>, <span class="st">&quot;P_ase&quot;</span>, <span class="st">&quot;N_ase&quot;</span>, </span>
<span id="cb109-10"><a href="#cb109-10" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot;S_ase&quot;</span>, <span class="st">&quot;XY_ase&quot;</span>,  <span class="st">&quot;LP_ase&quot;</span>, <span class="st">&quot;PX_ase&quot;</span>, <span class="st">&quot;CO2&quot;</span>, </span>
<span id="cb109-11"><a href="#cb109-11" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot;enzCN&quot;</span>, <span class="st">&quot;enzCP&quot;</span>, <span class="st">&quot;enzNP&quot;</span>)</span>
<span id="cb109-12"><a href="#cb109-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-13"><a href="#cb109-13" aria-hidden="true" tabindex="-1"></a>temp_adapt <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;AG_Q10&quot;</span>, <span class="st">&quot;BG_Q10&quot;</span>, <span class="st">&quot;BP_Q10&quot;</span>, <span class="st">&quot;CE_Q10&quot;</span>, <span class="st">&quot;P_Q10&quot;</span>, <span class="st">&quot;N_Q10&quot;</span>, </span>
<span id="cb109-14"><a href="#cb109-14" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot;S_Q10&quot;</span>, <span class="st">&quot;XY_Q10&quot;</span>, <span class="st">&quot;LP_Q10&quot;</span>, <span class="st">&quot;PX_Q10&quot;</span>, <span class="st">&quot;CUEcn&quot;</span>, <span class="st">&quot;CUEcp&quot;</span>, </span>
<span id="cb109-15"><a href="#cb109-15" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot;NUE&quot;</span>, <span class="st">&quot;PUE&quot;</span>, <span class="st">&quot;Tmin&quot;</span>, <span class="st">&quot;SI&quot;</span>)</span>
<span id="cb109-16"><a href="#cb109-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-17"><a href="#cb109-17" aria-hidden="true" tabindex="-1"></a>md_groups <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;edaphic&quot;</span>, <span class="st">&quot;soil_funct&quot;</span>, <span class="st">&quot;temp_adapt&quot;</span>)</span>
<span id="cb109-18"><a href="#cb109-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-19"><a href="#cb109-19" aria-hidden="true" tabindex="-1"></a><span class="co"># NoT uSed: minPO4, minNH4, minNO3, minTIN</span></span></code></pre></div>
</details>
</div>
<p>And then use the lists to split the data sets by metadata group. Here, we do this for the original metadata and the metadata after normalization.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a>select_md <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;its18_select_mc&quot;</span>, <span class="st">&quot;its18_select_mc_norm&quot;</span>)</span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> select_md) {</span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a>   <span class="co">#tmp_get &lt;- get(purrr::map_chr(i, ~ paste0(i, &quot;_mc&quot;)))</span></span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a>   tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb110-5"><a href="#cb110-5" aria-hidden="true" tabindex="-1"></a>   tmp_md_all <span class="ot">&lt;-</span> tmp_get<span class="sc">$</span>map_loaded</span>
<span id="cb110-6"><a href="#cb110-6" aria-hidden="true" tabindex="-1"></a>   tmp_div <span class="ot">&lt;-</span> tmp_md_all <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(div))</span>
<span id="cb110-7"><a href="#cb110-7" aria-hidden="true" tabindex="-1"></a>   tmp_div <span class="ot">&lt;-</span> tmp_div <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb110-8"><a href="#cb110-8" aria-hidden="true" tabindex="-1"></a><span class="do">## edaphic      </span></span>
<span id="cb110-9"><a href="#cb110-9" aria-hidden="true" tabindex="-1"></a>   tmp_sub_edaphic <span class="ot">&lt;-</span> tmp_md_all <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(edaphic))</span>
<span id="cb110-10"><a href="#cb110-10" aria-hidden="true" tabindex="-1"></a>   tmp_sub_edaphic <span class="ot">&lt;-</span> tmp_sub_edaphic <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb110-11"><a href="#cb110-11" aria-hidden="true" tabindex="-1"></a>   tmp_edaphic <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_div, tmp_sub_edaphic, <span class="at">by =</span> <span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb110-12"><a href="#cb110-12" aria-hidden="true" tabindex="-1"></a>   tmp_edaphic <span class="ot">&lt;-</span> tmp_edaphic <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb110-13"><a href="#cb110-13" aria-hidden="true" tabindex="-1"></a><span class="do">## soil_funct</span></span>
<span id="cb110-14"><a href="#cb110-14" aria-hidden="true" tabindex="-1"></a>   tmp_sub_soil_funct <span class="ot">&lt;-</span> tmp_md_all <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(soil_funct))</span>
<span id="cb110-15"><a href="#cb110-15" aria-hidden="true" tabindex="-1"></a>   tmp_sub_soil_funct <span class="ot">&lt;-</span> tmp_sub_soil_funct <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb110-16"><a href="#cb110-16" aria-hidden="true" tabindex="-1"></a>   tmp_soil_funct <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_div, tmp_sub_soil_funct, <span class="at">by =</span> <span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb110-17"><a href="#cb110-17" aria-hidden="true" tabindex="-1"></a>   tmp_soil_funct <span class="ot">&lt;-</span> tmp_soil_funct <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;SampleID&quot;</span>)      </span>
<span id="cb110-18"><a href="#cb110-18" aria-hidden="true" tabindex="-1"></a><span class="do">## temp_adapt</span></span>
<span id="cb110-19"><a href="#cb110-19" aria-hidden="true" tabindex="-1"></a>   tmp_sub_temp_adapt <span class="ot">&lt;-</span> tmp_md_all <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(temp_adapt))</span>
<span id="cb110-20"><a href="#cb110-20" aria-hidden="true" tabindex="-1"></a>   tmp_sub_temp_adapt <span class="ot">&lt;-</span> tmp_sub_temp_adapt <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb110-21"><a href="#cb110-21" aria-hidden="true" tabindex="-1"></a>   tmp_temp_adapt <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_div, tmp_sub_temp_adapt, <span class="at">by =</span> <span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb110-22"><a href="#cb110-22" aria-hidden="true" tabindex="-1"></a>   tmp_temp_adapt <span class="ot">&lt;-</span> tmp_temp_adapt <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;SampleID&quot;</span>)      </span>
<span id="cb110-23"><a href="#cb110-23" aria-hidden="true" tabindex="-1"></a><span class="do">## combine</span></span>
<span id="cb110-24"><a href="#cb110-24" aria-hidden="true" tabindex="-1"></a>   tmp_list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">data_loaded =</span> its18_select_mc<span class="sc">$</span>data_loaded, </span>
<span id="cb110-25"><a href="#cb110-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">map_loaded =</span> its18_select_mc<span class="sc">$</span>map_loaded, </span>
<span id="cb110-26"><a href="#cb110-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxonomy_loaded =</span> its18_select_mc<span class="sc">$</span>taxonomy_loaded,</span>
<span id="cb110-27"><a href="#cb110-27" aria-hidden="true" tabindex="-1"></a>                    <span class="at">edaphic =</span> tmp_edaphic, </span>
<span id="cb110-28"><a href="#cb110-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">soil_funct =</span> tmp_soil_funct, </span>
<span id="cb110-29"><a href="#cb110-29" aria-hidden="true" tabindex="-1"></a>                    <span class="at">temp_adapt =</span> tmp_temp_adapt)</span>
<span id="cb110-30"><a href="#cb110-30" aria-hidden="true" tabindex="-1"></a>   tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_split&quot;</span>))</span>
<span id="cb110-31"><a href="#cb110-31" aria-hidden="true" tabindex="-1"></a>   <span class="fu">print</span>(tmp_name)</span>
<span id="cb110-32"><a href="#cb110-32" aria-hidden="true" tabindex="-1"></a>   <span class="fu">assign</span>(tmp_name, tmp_list)</span>
<span id="cb110-33"><a href="#cb110-33" aria-hidden="true" tabindex="-1"></a>   <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb110-34"><a href="#cb110-34" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<h3 id="generate-autocorrelation-plots-1">Generate Autocorrelation Plots</h3>
<p>A little housekeeping to get rid of parameters we dont need (e.g., plot number, pair, etc.).</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>edaphic_cor <span class="ot">&lt;-</span> its18_select_mc_split<span class="sc">$</span>edaphic</span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a>edaphic_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a>edaphic_norm_cor <span class="ot">&lt;-</span> its18_select_mc_norm_split<span class="sc">$</span>edaphic</span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a>edaphic_norm_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-6"><a href="#cb111-6" aria-hidden="true" tabindex="-1"></a>soil_funct_cor <span class="ot">&lt;-</span> its18_select_mc_split<span class="sc">$</span>soil_funct</span>
<span id="cb111-7"><a href="#cb111-7" aria-hidden="true" tabindex="-1"></a>soil_funct_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb111-8"><a href="#cb111-8" aria-hidden="true" tabindex="-1"></a>soil_funct_norm_cor <span class="ot">&lt;-</span> its18_select_mc_norm_split<span class="sc">$</span>soil_funct</span>
<span id="cb111-9"><a href="#cb111-9" aria-hidden="true" tabindex="-1"></a>soil_funct_norm_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb111-10"><a href="#cb111-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-11"><a href="#cb111-11" aria-hidden="true" tabindex="-1"></a>temp_adapt_cor <span class="ot">&lt;-</span> its18_select_mc_split<span class="sc">$</span>temp_adapt</span>
<span id="cb111-12"><a href="#cb111-12" aria-hidden="true" tabindex="-1"></a>temp_adapt_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb111-13"><a href="#cb111-13" aria-hidden="true" tabindex="-1"></a>temp_adapt_norm_cor <span class="ot">&lt;-</span> its18_select_mc_norm_split<span class="sc">$</span>temp_adapt</span>
<span id="cb111-14"><a href="#cb111-14" aria-hidden="true" tabindex="-1"></a>temp_adapt_norm_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span></code></pre></div>
</details>
</div>
<p>And finally the code to create the plots.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;_cor$&quot;</span>)) {</span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a>  tmp_cormat <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">cor</span>(tmp_get), <span class="dv">2</span>)</span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a>  tmp_melted_cormat <span class="ot">&lt;-</span> reshape2<span class="sc">::</span><span class="fu">melt</span>(tmp_cormat)</span>
<span id="cb112-5"><a href="#cb112-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb112-6"><a href="#cb112-6" aria-hidden="true" tabindex="-1"></a>  tmp_get_lower_tri <span class="ot">&lt;-</span> <span class="cf">function</span>(tmp_cormat){</span>
<span id="cb112-7"><a href="#cb112-7" aria-hidden="true" tabindex="-1"></a>    tmp_cormat[<span class="fu">upper.tri</span>(tmp_cormat)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb112-8"><a href="#cb112-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(tmp_cormat)</span>
<span id="cb112-9"><a href="#cb112-9" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb112-10"><a href="#cb112-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get upper triangle of the correlation matrix</span></span>
<span id="cb112-11"><a href="#cb112-11" aria-hidden="true" tabindex="-1"></a>  tmp_get_upper_tri <span class="ot">&lt;-</span> <span class="cf">function</span>(tmp_cormat){</span>
<span id="cb112-12"><a href="#cb112-12" aria-hidden="true" tabindex="-1"></a>    tmp_cormat[<span class="fu">lower.tri</span>(tmp_cormat)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb112-13"><a href="#cb112-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(tmp_cormat)</span>
<span id="cb112-14"><a href="#cb112-14" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb112-15"><a href="#cb112-15" aria-hidden="true" tabindex="-1"></a>  tmp_upper_tri <span class="ot">&lt;-</span> <span class="fu">tmp_get_upper_tri</span>(tmp_cormat)</span>
<span id="cb112-16"><a href="#cb112-16" aria-hidden="true" tabindex="-1"></a>  tmp_melted_cormat <span class="ot">&lt;-</span> reshape2<span class="sc">::</span><span class="fu">melt</span>(tmp_upper_tri, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb112-17"><a href="#cb112-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="at">data =</span> tmp_melted_cormat, <span class="fu">aes</span>(<span class="at">x =</span> Var1, <span class="at">y =</span> Var2, <span class="at">fill =</span> value)) <span class="sc">+</span> </span>
<span id="cb112-18"><a href="#cb112-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>()</span>
<span id="cb112-19"><a href="#cb112-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb112-20"><a href="#cb112-20" aria-hidden="true" tabindex="-1"></a>  tmp_ggheatmap <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tmp_melted_cormat, <span class="fu">aes</span>(Var2, Var1, <span class="at">fill =</span> value)) <span class="sc">+</span></span>
<span id="cb112-21"><a href="#cb112-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>(<span class="at">color =</span> <span class="st">&quot;white&quot;</span>) <span class="sc">+</span></span>
<span id="cb112-22"><a href="#cb112-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> <span class="st">&quot;blue&quot;</span>, <span class="at">high =</span> <span class="st">&quot;red&quot;</span>, <span class="at">mid =</span> <span class="st">&quot;white&quot;</span>, </span>
<span id="cb112-23"><a href="#cb112-23" aria-hidden="true" tabindex="-1"></a>                         <span class="at">midpoint =</span> <span class="dv">0</span>, <span class="at">limit =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="at">space =</span> <span class="st">&quot;Lab&quot;</span>, </span>
<span id="cb112-24"><a href="#cb112-24" aria-hidden="true" tabindex="-1"></a>                         <span class="at">name=</span><span class="st">&quot;Pearson</span><span class="sc">\n</span><span class="st">Correlation&quot;</span>) <span class="sc">+</span></span>
<span id="cb112-25"><a href="#cb112-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span> </span>
<span id="cb112-26"><a href="#cb112-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, </span>
<span id="cb112-27"><a href="#cb112-27" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">size =</span> <span class="dv">7</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb112-28"><a href="#cb112-28" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">7</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb112-29"><a href="#cb112-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_fixed</span>() <span class="sc">+</span> </span>
<span id="cb112-30"><a href="#cb112-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="fu">aes</span>(Var2, Var1, <span class="at">label =</span> value), <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="fl">1.75</span>) <span class="sc">+</span></span>
<span id="cb112-31"><a href="#cb112-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb112-32"><a href="#cb112-32" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb112-33"><a href="#cb112-33" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb112-34"><a href="#cb112-34" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb112-35"><a href="#cb112-35" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb112-36"><a href="#cb112-36" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb112-37"><a href="#cb112-37" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb112-38"><a href="#cb112-38" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">0</span>),</span>
<span id="cb112-39"><a href="#cb112-39" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.6</span>, <span class="fl">0.7</span>),</span>
<span id="cb112-40"><a href="#cb112-40" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.direction =</span> <span class="st">&quot;horizontal&quot;</span>) <span class="sc">+</span></span>
<span id="cb112-41"><a href="#cb112-41" aria-hidden="true" tabindex="-1"></a>      <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_colorbar</span>(<span class="at">barwidth =</span> <span class="dv">7</span>, <span class="at">barheight =</span> <span class="dv">1</span>,</span>
<span id="cb112-42"><a href="#cb112-42" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">title.position =</span> <span class="st">&quot;top&quot;</span>, </span>
<span id="cb112-43"><a href="#cb112-43" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">title.hjust =</span> <span class="fl">0.5</span>))</span>
<span id="cb112-44"><a href="#cb112-44" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_ggheatmap&quot;</span>))</span>
<span id="cb112-45"><a href="#cb112-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_ggheatmap)</span>
<span id="cb112-46"><a href="#cb112-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(tmp_name)</span>
<span id="cb112-47"><a href="#cb112-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb112-48"><a href="#cb112-48" aria-hidden="true" tabindex="-1"></a>}  </span>
<span id="cb112-49"><a href="#cb112-49" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;_ggheatmap&quot;</span>)</span></code></pre></div>
</details>
</div>
<h2 id="autocorrelation-plots-1">Autocorrelation Plots</h2>
<p><br/></p>
<div class="l-body-outset">
<div class="panelset">
<section id="edaphic-properties-4" class="panel">
<h4>Edaphic properties</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="metadata_files/figure-html5/unnamed-chunk-151-1.png" width="624" /></p>
</div>
</section>
<section id="edaphic-properties-normalized-1" class="panel">
<h4>Edaphic properties (normalized)</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="metadata_files/figure-html5/unnamed-chunk-152-1.png" width="624" /></p>
</div>
</section>
<section id="functional-responses-1" class="panel">
<h4>Functional responses</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="metadata_files/figure-html5/unnamed-chunk-153-1.png" width="624" /></p>
</div>
</section>
<section id="functional-responses-normalized-1" class="panel">
<h4>Functional responses (normalized)</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="metadata_files/figure-html5/unnamed-chunk-154-1.png" width="624" /></p>
</div>
</section>
<section id="temperature-adaptation-4" class="panel">
<h4>Temperature adaptation</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="metadata_files/figure-html5/unnamed-chunk-155-1.png" width="624" /></p>
</div>
</section>
<section id="temperature-adaptation-normalized-1" class="panel">
<h4>Temperature adaptation (normalized)</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="metadata_files/figure-html5/unnamed-chunk-156-1.png" width="624" /></p>
</div>
</section>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-page">

</div>
<p>Now we can remove parameters based on the autocorrelation analysis:</p>
<ol type="A">
<li>Environmental and edaphic properties: TEB, DON, Na, Al, Ca.</li>
<li>Microbial functional responses: micN, micNP, enzCN, enzCP, BP_ase, CE_ase, LP_ase, N_ase, P_ase.</li>
<li>Temperature adaptation properties: NUE, PUE, P_Q10, SI.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a>edaphic_remove <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;TEB&quot;</span>, <span class="st">&quot;DON&quot;</span>, <span class="st">&quot;Na&quot;</span>, <span class="st">&quot;Al&quot;</span>, <span class="st">&quot;Ca&quot;</span>)</span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>soil_funct_remove <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;micN&quot;</span>, <span class="st">&quot;micNP&quot;</span>, <span class="st">&quot;enzCN&quot;</span>, <span class="st">&quot;enzCP&quot;</span>, <span class="st">&quot;BP_ase&quot;</span>, </span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&quot;CE_ase&quot;</span>, <span class="st">&quot;LP_ase&quot;</span>, <span class="st">&quot;N_ase&quot;</span>, <span class="st">&quot;P_ase&quot;</span>)</span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a>temp_adapt_remove <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;NUE&quot;</span>, <span class="st">&quot;PUE&quot;</span>, <span class="st">&quot;P_Q10&quot;</span>, <span class="st">&quot;SI&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a>tmp_df <span class="ot">&lt;-</span> its18_select_mc_split</span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>edaphic <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>edaphic[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>edaphic) <span class="sc">%in%</span>  edaphic_remove]</span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>soil_funct <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>soil_funct[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>soil_funct) <span class="sc">%in%</span>  soil_funct_remove]</span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>temp_adapt <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>temp_adapt[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>temp_adapt) <span class="sc">%in%</span>  temp_adapt_remove]</span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>its18_select_mc_split_no_ac <span class="ot">&lt;-</span> tmp_df</span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb114-7"><a href="#cb114-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-8"><a href="#cb114-8" aria-hidden="true" tabindex="-1"></a>tmp_df <span class="ot">&lt;-</span> its18_select_mc_norm_split</span>
<span id="cb114-9"><a href="#cb114-9" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>edaphic <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>edaphic[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>edaphic) <span class="sc">%in%</span>  edaphic_remove]</span>
<span id="cb114-10"><a href="#cb114-10" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>soil_funct <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>soil_funct[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>soil_funct) <span class="sc">%in%</span>  soil_funct_remove]</span>
<span id="cb114-11"><a href="#cb114-11" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>temp_adapt <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>temp_adapt[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>temp_adapt) <span class="sc">%in%</span>  temp_adapt_remove]</span>
<span id="cb114-12"><a href="#cb114-12" aria-hidden="true" tabindex="-1"></a>its18_select_mc_norm_split_no_ac <span class="ot">&lt;-</span> tmp_df</span>
<span id="cb114-13"><a href="#cb114-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span></code></pre></div>
</details>
</div>
<h2 id="dissimilarity-correlation-tests-1">Dissimilarity Correlation Tests</h2>
<p>Lets see if any on the metadata groups are significantly correlated with the community data. Basically, we create distance matrices for the community data and each metadata group and then run Mantel tests for all comparisons. For the community data we calculate Bray-Curtis distances for the community data and Euclidean distances for the metadata. We use the function <code>mantel.test</code> from the <code>ape</code> package and <code>mantel</code> from the <code>vegan</code> package for the analyses.</p>
<p>In summary, we test both <code>mantel.test</code> and <code>mantel</code> on Bray-Curtis distance community distances against Euclidean distances for each metadata group (<code>edaphic</code>, <code>soil_funct</code>, <code>temp_adapt</code>) <strong>a</strong>) before normalizing and before removing autocorrelated parameters, <strong>b</strong>) before normalizing and after removing autocorrelated parameters, <strong>c</strong>) after normalizing and before removing autocorrelated parameters, and <strong>d</strong>) after normalizing and after removing autocorrelated parameters.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a>man_df <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;its18_select_mc_split&quot;</span>, <span class="st">&quot;its18_select_mc_split_no_ac&quot;</span>, </span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;its18_select_mc_norm_split&quot;</span>, <span class="st">&quot;its18_select_mc_norm_split_no_ac&quot;</span>)</span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-4"><a href="#cb115-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> man_df) {</span>
<span id="cb115-5"><a href="#cb115-5" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb115-6"><a href="#cb115-6" aria-hidden="true" tabindex="-1"></a>  tmp_dm_otu <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(<span class="fu">t</span>(tmp_get<span class="sc">$</span>data_loaded), </span>
<span id="cb115-7"><a href="#cb115-7" aria-hidden="true" tabindex="-1"></a>                             <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb115-8"><a href="#cb115-8" aria-hidden="true" tabindex="-1"></a>                             <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb115-9"><a href="#cb115-9" aria-hidden="true" tabindex="-1"></a><span class="co"># EDAPHIC</span></span>
<span id="cb115-10"><a href="#cb115-10" aria-hidden="true" tabindex="-1"></a>  tmp_dm_md_edaphic <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_get<span class="sc">$</span>edaphic[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_get<span class="sc">$</span>edaphic)], </span>
<span id="cb115-11"><a href="#cb115-11" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">method =</span><span class="st">&quot;euclidean&quot;</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb115-12"><a href="#cb115-12" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb115-13"><a href="#cb115-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-14"><a href="#cb115-14" aria-hidden="true" tabindex="-1"></a>  tmp_man1_edaphic <span class="ot">&lt;-</span> <span class="fu">mantel.test</span>(tmp_dm_otu, tmp_dm_md_edaphic, <span class="at">nperm =</span> <span class="dv">999</span>, </span>
<span id="cb115-15"><a href="#cb115-15" aria-hidden="true" tabindex="-1"></a>                             <span class="at">graph =</span> <span class="cn">FALSE</span>, <span class="at">alternative =</span> <span class="st">&quot;two.sided&quot;</span>)</span>
<span id="cb115-16"><a href="#cb115-16" aria-hidden="true" tabindex="-1"></a>  tmp_man2_edaphic <span class="ot">&lt;-</span> <span class="fu">mantel</span>(tmp_dm_otu, tmp_dm_md_edaphic, <span class="at">permutations =</span> <span class="dv">999</span>)</span>
<span id="cb115-17"><a href="#cb115-17" aria-hidden="true" tabindex="-1"></a><span class="co"># SOIL FUNCT</span></span>
<span id="cb115-18"><a href="#cb115-18" aria-hidden="true" tabindex="-1"></a>  tmp_dm_md_soil_funct <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_get<span class="sc">$</span>soil_funct[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_get<span class="sc">$</span>soil_funct)], </span>
<span id="cb115-19"><a href="#cb115-19" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb115-20"><a href="#cb115-20" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb115-21"><a href="#cb115-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-22"><a href="#cb115-22" aria-hidden="true" tabindex="-1"></a>  tmp_man1_soil_funct <span class="ot">&lt;-</span> <span class="fu">mantel.test</span>(tmp_dm_otu, tmp_dm_md_soil_funct, <span class="at">nperm =</span> <span class="dv">999</span>, </span>
<span id="cb115-23"><a href="#cb115-23" aria-hidden="true" tabindex="-1"></a>                                <span class="at">graph =</span> <span class="cn">FALSE</span>, <span class="at">alternative =</span> <span class="st">&quot;two.sided&quot;</span>)</span>
<span id="cb115-24"><a href="#cb115-24" aria-hidden="true" tabindex="-1"></a>  tmp_man2_soil_funct <span class="ot">&lt;-</span> <span class="fu">mantel</span>(tmp_dm_otu, tmp_dm_md_soil_funct, <span class="at">permutations =</span> <span class="dv">999</span>)</span>
<span id="cb115-25"><a href="#cb115-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-26"><a href="#cb115-26" aria-hidden="true" tabindex="-1"></a><span class="co"># TEMP ADAPT</span></span>
<span id="cb115-27"><a href="#cb115-27" aria-hidden="true" tabindex="-1"></a>  tmp_dm_md_temp_adapt <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_get<span class="sc">$</span>temp_adapt[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_get<span class="sc">$</span>temp_adapt)], </span>
<span id="cb115-28"><a href="#cb115-28" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb115-29"><a href="#cb115-29" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb115-30"><a href="#cb115-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-31"><a href="#cb115-31" aria-hidden="true" tabindex="-1"></a>  tmp_man1_temp_adapt <span class="ot">&lt;-</span> <span class="fu">mantel.test</span>(tmp_dm_otu, tmp_dm_md_temp_adapt, <span class="at">nperm =</span> <span class="dv">999</span>, </span>
<span id="cb115-32"><a href="#cb115-32" aria-hidden="true" tabindex="-1"></a>                                <span class="at">graph =</span> <span class="cn">FALSE</span>, <span class="at">alternative =</span> <span class="st">&quot;two.sided&quot;</span>)</span>
<span id="cb115-33"><a href="#cb115-33" aria-hidden="true" tabindex="-1"></a>  tmp_man2_temp_adapt <span class="ot">&lt;-</span> <span class="fu">mantel</span>(tmp_dm_otu, tmp_dm_md_temp_adapt, <span class="at">permutations =</span> <span class="dv">999</span>)</span>
<span id="cb115-34"><a href="#cb115-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-35"><a href="#cb115-35" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_mantel_tests&quot;</span>))</span>
<span id="cb115-36"><a href="#cb115-36" aria-hidden="true" tabindex="-1"></a>  tmp_df <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">edaphic_ape_man =</span> tmp_man1_edaphic, </span>
<span id="cb115-37"><a href="#cb115-37" aria-hidden="true" tabindex="-1"></a>                 <span class="at">edaphic_vegan_man =</span> tmp_man2_edaphic,</span>
<span id="cb115-38"><a href="#cb115-38" aria-hidden="true" tabindex="-1"></a>                 <span class="at">soil_funct_ape_man =</span> tmp_man1_soil_funct, </span>
<span id="cb115-39"><a href="#cb115-39" aria-hidden="true" tabindex="-1"></a>                 <span class="at">soil_funct_vegan_man =</span> tmp_man2_soil_funct,</span>
<span id="cb115-40"><a href="#cb115-40" aria-hidden="true" tabindex="-1"></a>                 <span class="at">temp_adapt_ape_man =</span> tmp_man1_temp_adapt, </span>
<span id="cb115-41"><a href="#cb115-41" aria-hidden="true" tabindex="-1"></a>                 <span class="at">temp_adapt_vegan_man =</span> tmp_man2_temp_adapt)</span>
<span id="cb115-42"><a href="#cb115-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_df)</span>
<span id="cb115-43"><a href="#cb115-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(tmp_name)</span>
<span id="cb115-44"><a href="#cb115-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb115-45"><a href="#cb115-45" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="dissimilarity-correlation-results-1">Dissimilarity Correlation Results</h3>
<p><br/></p>
<div class="layout-chunk" data-layout="l-page">
<table>
<colgroup>
<col style="width: 10%" />
<col style="width: 10%" />
<col style="width: 10%" />
<col style="width: 12%" />
<col style="width: 13%" />
<col style="width: 14%" />
<col style="width: 13%" />
<col style="width: 14%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">normalized?</th>
<th style="text-align: left;">AC removed?</th>
<th style="text-align: right;">edaphic_ape</th>
<th style="text-align: right;">edaphic_vegan</th>
<th style="text-align: right;">soil_funct_ape</th>
<th style="text-align: right;">soil_funct_vegan</th>
<th style="text-align: right;">temp_adapt_ape</th>
<th style="text-align: right;">temp_adapt_vegan</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">no</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">0.119</td>
<td style="text-align: right;">0.062</td>
<td style="text-align: right;">0.097</td>
<td style="text-align: right;">0.047</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.002</td>
</tr>
<tr class="even">
<td style="text-align: left;">no</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">0.081</td>
<td style="text-align: right;">0.037</td>
<td style="text-align: right;">0.095</td>
<td style="text-align: right;">0.037</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="odd">
<td style="text-align: left;">yes</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">0.151</td>
<td style="text-align: right;">0.081</td>
<td style="text-align: right;">0.345</td>
<td style="text-align: right;">0.163</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">yes</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: right;">0.357</td>
<td style="text-align: right;">0.180</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.001</td>
</tr>
</tbody>
</table>
</div>
<p>Moving on.</p>
<h2 id="best-subset-of-variables-1">Best Subset of Variables</h2>
<p>Now we want to know which of the metadata parameters are the most strongly correlated with the community data. For this we use the <code>bioenv</code> function from the <code>vegan</code> package. <code>bioenv</code><em>Best Subset of Environmental Variables with Maximum (Rank) Correlation with Community Dissimilarities</em>finds the best subset of environmental variables, so that the Euclidean distances of scaled environmental variables have the maximum (rank) correlation with community dissimilarities.</p>
<p>Since we know that each of the Mantel tests we ran above are significant, here we will use the metadata set where autocorrelated parameters were removed and the remainder of the parameters were normalized (where applicable based on the Shapiro tests).</p>
<p>We run <code>bioenv</code> against the three groups of metadata parameters. We then run <code>bioenv</code> again, but this time against the individual parameters identified as significantly correlated.</p>
<h3 id="edaphic-properties-5">Edaphic Properties</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(its18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>tmp_env <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(its18_select_mc_norm_split_no_ac<span class="sc">$</span>edaphic)</span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a>tmp_env[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a>edaphic_bioenv <span class="ot">&lt;-</span> <span class="fu">bioenv</span>(<span class="fu">wisconsin</span>(tmp_comm), tmp_env, </span>
<span id="cb116-6"><a href="#cb116-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>, <span class="at">index =</span> <span class="st">&quot;bray&quot;</span>, </span>
<span id="cb116-7"><a href="#cb116-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">upto =</span> <span class="fu">ncol</span>(tmp_env), <span class="at">metric =</span> <span class="st">&quot;euclidean&quot;</span>)</span>
<span id="cb116-8"><a href="#cb116-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-9"><a href="#cb116-9" aria-hidden="true" tabindex="-1"></a>bioenv_list <span class="ot">&lt;-</span> edaphic_bioenv<span class="sc">$</span>models[[edaphic_bioenv<span class="sc">$</span>whichbest]]<span class="sc">$</span>best</span>
<span id="cb116-10"><a href="#cb116-10" aria-hidden="true" tabindex="-1"></a>bioenv_best <span class="ot">&lt;-</span> <span class="fu">bioenvdist</span>(edaphic_bioenv, <span class="at">which =</span> <span class="st">&quot;best&quot;</span>)</span>
<span id="cb116-11"><a href="#cb116-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-12"><a href="#cb116-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> bioenv_list) {</span>
<span id="cb116-13"><a href="#cb116-13" aria-hidden="true" tabindex="-1"></a>       tmp_dp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(edaphic_bioenv<span class="sc">$</span>x)</span>
<span id="cb116-14"><a href="#cb116-14" aria-hidden="true" tabindex="-1"></a>       tmp_md <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_dp[[i]], </span>
<span id="cb116-15"><a href="#cb116-15" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb116-16"><a href="#cb116-16" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb116-17"><a href="#cb116-17" aria-hidden="true" tabindex="-1"></a>       tmp_man <span class="ot">&lt;-</span> <span class="fu">mantel</span>(bioenv_best, tmp_md, </span>
<span id="cb116-18"><a href="#cb116-18" aria-hidden="true" tabindex="-1"></a>                         <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>)</span>
<span id="cb116-19"><a href="#cb116-19" aria-hidden="true" tabindex="-1"></a>       tmp_md_name <span class="ot">&lt;-</span> <span class="fu">names</span>(tmp_dp)[[i]]</span>
<span id="cb116-20"><a href="#cb116-20" aria-hidden="true" tabindex="-1"></a>       tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(tmp_md_name, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_bioenv_mantel_test&quot;</span>))</span>
<span id="cb116-21"><a href="#cb116-21" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_name, tmp_man)</span>
<span id="cb116-22"><a href="#cb116-22" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb116-23"><a href="#cb116-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb116-24"><a href="#cb116-24" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;_bioenv_mantel_test&quot;</span>)</span>
<span id="cb116-25"><a href="#cb116-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-26"><a href="#cb116-26" aria-hidden="true" tabindex="-1"></a>edaphic_bioenv_ind_mantel <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">AST =</span> AST_bioenv_mantel_test)</span>
<span id="cb116-27"><a href="#cb116-27" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;_bioenv_mantel_test&quot;</span>))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
Call:
bioenv(comm = wisconsin(tmp_comm), env = tmp_env, method = &quot;spearman&quot;,      index = &quot;bray&quot;, upto = ncol(tmp_env), metric = &quot;euclidean&quot;) 

Subset of environmental variables with best correlation to community data.

Correlations:    spearman 
Dissimilarities: bray 
Metric:          euclidean 

Best model has 1 parameters (max. 15 allowed):
AST
with correlation  0.65665 </code></pre>
</div>
<details>
<summary>
Show the results of individual edaphic metadata Mantel tests
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>$AST

Mantel statistic based on Spearman&#39;s rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = &quot;spearman&quot;,      permutations = 999) 

Mantel statistic r:     1 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.192 0.242 0.272 0.339 
Permutation: free
Number of permutations: 999</code></pre>
</div>
</details>
<p><code>bioenv</code> found the following edaphic properties significantly correlated with the community data: <strong>AST</strong></p>
<h3 id="soil-functional-response-3">Soil Functional Response</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(its18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a>tmp_env <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(its18_select_mc_norm_split_no_ac<span class="sc">$</span>soil_funct)</span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a>tmp_env[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb119-4"><a href="#cb119-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-5"><a href="#cb119-5" aria-hidden="true" tabindex="-1"></a>soil_funct_bioenv <span class="ot">&lt;-</span> <span class="fu">bioenv</span>(<span class="fu">wisconsin</span>(tmp_comm), tmp_env, </span>
<span id="cb119-6"><a href="#cb119-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>, <span class="at">index =</span> <span class="st">&quot;bray&quot;</span>, </span>
<span id="cb119-7"><a href="#cb119-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">upto =</span> <span class="fu">ncol</span>(tmp_env), <span class="at">metric =</span> <span class="st">&quot;euclidean&quot;</span>)</span>
<span id="cb119-8"><a href="#cb119-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-9"><a href="#cb119-9" aria-hidden="true" tabindex="-1"></a>bioenv_list <span class="ot">&lt;-</span> soil_funct_bioenv<span class="sc">$</span>models[[soil_funct_bioenv<span class="sc">$</span>whichbest]]<span class="sc">$</span>best</span>
<span id="cb119-10"><a href="#cb119-10" aria-hidden="true" tabindex="-1"></a>bioenv_best <span class="ot">&lt;-</span> <span class="fu">bioenvdist</span>(soil_funct_bioenv, <span class="at">which =</span> <span class="st">&quot;best&quot;</span>)</span>
<span id="cb119-11"><a href="#cb119-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-12"><a href="#cb119-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> bioenv_list) {</span>
<span id="cb119-13"><a href="#cb119-13" aria-hidden="true" tabindex="-1"></a>       tmp_dp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(soil_funct_bioenv<span class="sc">$</span>x)</span>
<span id="cb119-14"><a href="#cb119-14" aria-hidden="true" tabindex="-1"></a>       tmp_md <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_dp[[i]], </span>
<span id="cb119-15"><a href="#cb119-15" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb119-16"><a href="#cb119-16" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb119-17"><a href="#cb119-17" aria-hidden="true" tabindex="-1"></a>       tmp_man <span class="ot">&lt;-</span> <span class="fu">mantel</span>(bioenv_best, tmp_md, </span>
<span id="cb119-18"><a href="#cb119-18" aria-hidden="true" tabindex="-1"></a>                         <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>)</span>
<span id="cb119-19"><a href="#cb119-19" aria-hidden="true" tabindex="-1"></a>       tmp_md_name <span class="ot">&lt;-</span> <span class="fu">names</span>(tmp_dp)[[i]]</span>
<span id="cb119-20"><a href="#cb119-20" aria-hidden="true" tabindex="-1"></a>       tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(tmp_md_name, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_bioenv_mantel_test&quot;</span>))</span>
<span id="cb119-21"><a href="#cb119-21" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_name, tmp_man)</span>
<span id="cb119-22"><a href="#cb119-22" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb119-23"><a href="#cb119-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb119-24"><a href="#cb119-24" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;_bioenv_mantel_test&quot;</span>)</span>
<span id="cb119-25"><a href="#cb119-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-26"><a href="#cb119-26" aria-hidden="true" tabindex="-1"></a>soil_funct_bioenv_ind_mantel <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">enzNP =</span> enzNP_bioenv_mantel_test, </span>
<span id="cb119-27"><a href="#cb119-27" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">PX_ase =</span> PX_ase_bioenv_mantel_test,</span>
<span id="cb119-28"><a href="#cb119-28" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">XY_ase =</span> XY_ase_bioenv_mantel_test)</span>
<span id="cb119-29"><a href="#cb119-29" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;_bioenv_mantel_test&quot;</span>))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
Call:
bioenv(comm = wisconsin(tmp_comm), env = tmp_env, method = &quot;spearman&quot;,      index = &quot;bray&quot;, upto = ncol(tmp_env), metric = &quot;euclidean&quot;) 

Subset of environmental variables with best correlation to community data.

Correlations:    spearman 
Dissimilarities: bray 
Metric:          euclidean 

Best model has 3 parameters (max. 11 allowed):
XY_ase PX_ase enzNP
with correlation  0.7491622 </code></pre>
</div>
<details>
<summary>
Show the results of individual functional response metadata Mantel tests
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>$enzNP

Mantel statistic based on Spearman&#39;s rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = &quot;spearman&quot;,      permutations = 999) 

Mantel statistic r: 0.5529 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.202 0.277 0.330 0.417 
Permutation: free
Number of permutations: 999


$PX_ase

Mantel statistic based on Spearman&#39;s rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = &quot;spearman&quot;,      permutations = 999) 

Mantel statistic r: 0.6854 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.209 0.282 0.361 0.412 
Permutation: free
Number of permutations: 999


$XY_ase

Mantel statistic based on Spearman&#39;s rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = &quot;spearman&quot;,      permutations = 999) 

Mantel statistic r: 0.5054 
      Significance: 0.002 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.202 0.261 0.308 0.359 
Permutation: free
Number of permutations: 999</code></pre>
</div>
</details>
<p><code>bioenv</code> found the following soil functions significantly correlated with the community data: <strong>enzNP, PX_ase, XY_ase</strong></p>
<h3 id="temperature-adaptation-5">Temperature Adaptation</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(its18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a>tmp_env <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(its18_select_mc_norm_split_no_ac<span class="sc">$</span>temp_adapt)</span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a>tmp_env[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb122-4"><a href="#cb122-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-5"><a href="#cb122-5" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioenv <span class="ot">&lt;-</span> <span class="fu">bioenv</span>(<span class="fu">wisconsin</span>(tmp_comm), tmp_env, </span>
<span id="cb122-6"><a href="#cb122-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>, <span class="at">index =</span> <span class="st">&quot;bray&quot;</span>, </span>
<span id="cb122-7"><a href="#cb122-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">upto =</span> <span class="fu">ncol</span>(tmp_env), <span class="at">metric =</span> <span class="st">&quot;euclidean&quot;</span>)</span>
<span id="cb122-8"><a href="#cb122-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-9"><a href="#cb122-9" aria-hidden="true" tabindex="-1"></a>bioenv_list <span class="ot">&lt;-</span> temp_adapt_bioenv<span class="sc">$</span>models[[temp_adapt_bioenv<span class="sc">$</span>whichbest]]<span class="sc">$</span>best</span>
<span id="cb122-10"><a href="#cb122-10" aria-hidden="true" tabindex="-1"></a>bioenv_best <span class="ot">&lt;-</span> <span class="fu">bioenvdist</span>(temp_adapt_bioenv, <span class="at">which =</span> <span class="st">&quot;best&quot;</span>)</span>
<span id="cb122-11"><a href="#cb122-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-12"><a href="#cb122-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> bioenv_list) {</span>
<span id="cb122-13"><a href="#cb122-13" aria-hidden="true" tabindex="-1"></a>       tmp_dp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(temp_adapt_bioenv<span class="sc">$</span>x)</span>
<span id="cb122-14"><a href="#cb122-14" aria-hidden="true" tabindex="-1"></a>       tmp_md <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_dp[[i]], </span>
<span id="cb122-15"><a href="#cb122-15" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb122-16"><a href="#cb122-16" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb122-17"><a href="#cb122-17" aria-hidden="true" tabindex="-1"></a>       tmp_man <span class="ot">&lt;-</span> <span class="fu">mantel</span>(bioenv_best, tmp_md, </span>
<span id="cb122-18"><a href="#cb122-18" aria-hidden="true" tabindex="-1"></a>                         <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>)</span>
<span id="cb122-19"><a href="#cb122-19" aria-hidden="true" tabindex="-1"></a>       tmp_md_name <span class="ot">&lt;-</span> <span class="fu">names</span>(tmp_dp)[[i]]</span>
<span id="cb122-20"><a href="#cb122-20" aria-hidden="true" tabindex="-1"></a>       tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(tmp_md_name, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_bioenv_mantel_test&quot;</span>))</span>
<span id="cb122-21"><a href="#cb122-21" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_name, tmp_man)</span>
<span id="cb122-22"><a href="#cb122-22" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb122-23"><a href="#cb122-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb122-24"><a href="#cb122-24" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;_bioenv_mantel_test&quot;</span>)</span>
<span id="cb122-25"><a href="#cb122-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-26"><a href="#cb122-26" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioenv_ind_mantel <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">XY_Q10 =</span> XY_Q10_bioenv_mantel_test,</span>
<span id="cb122-27"><a href="#cb122-27" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">Tmin =</span> Tmin_bioenv_mantel_test)</span>
<span id="cb122-28"><a href="#cb122-28" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;_bioenv_mantel_test&quot;</span>))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
Call:
bioenv(comm = wisconsin(tmp_comm), env = tmp_env, method = &quot;spearman&quot;,      index = &quot;bray&quot;, upto = ncol(tmp_env), metric = &quot;euclidean&quot;) 

Subset of environmental variables with best correlation to community data.

Correlations:    spearman 
Dissimilarities: bray 
Metric:          euclidean 

Best model has 2 parameters (max. 12 allowed):
XY_Q10 Tmin
with correlation  0.5637527 </code></pre>
</div>
<details>
<summary>
Show the results of individual temperature adaptation metadata Mantel tests
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>$XY_Q10

Mantel statistic based on Spearman&#39;s rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = &quot;spearman&quot;,      permutations = 999) 

Mantel statistic r: 0.7258 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.134 0.200 0.241 0.316 
Permutation: free
Number of permutations: 999


$Tmin

Mantel statistic based on Spearman&#39;s rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = &quot;spearman&quot;,      permutations = 999) 

Mantel statistic r: 0.616 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.189 0.245 0.297 0.360 
Permutation: free
Number of permutations: 999</code></pre>
</div>
</details>
<p><code>bioenv</code> found the following temperature adaptations significantly correlated with the community data: <strong>XY_Q10, Tmin</strong></p>
<h2 id="distance-based-redundancy-1">Distance-based Redundancy</h2>
<p>Now we turn our attention to distance-based redundancy analysis (dbRDA), an ordination method similar to Redundancy Analysis (rda) but it allows non-Euclidean dissimilarity indices, such as Manhattan or BrayCurtis distance. For this, we use <code>capscale</code> from the <code>vegan</code> package. <code>capscale</code> is a constrained versions of metric scaling (principal coordinates analysis), which are based on the Euclidean distance but can be used, and are more useful, with other dissimilarity measures. The functions can also perform unconstrained principal coordinates analysis, optionally using extended dissimilarities.</p>
<p>For each of the three metadata subsets, we perform the following steps:</p>
<ol type="1">
<li>Run <code>rankindex</code> to compare metadata and community dissimilarity indices for gradient detection. This will help us select the best dissimilarity metric to use.</li>
<li>Run <code>capscale</code> for distance-based redundancy analysis.</li>
<li>Run <code>envfit</code> to fit environmental parameters onto the ordination. This function basically calculates correlation scores between the metadata parameters and the ordination axes.</li>
<li>Select metadata parameters significant for <code>bioenv</code> (see above) and/or <code>envfit</code> analyses.</li>
<li>Run <code>envfit</code> on ASVs.</li>
<li>Plot the ordination and vector overlays.</li>
</ol>
<h3 id="edaphic-properties-6">Edaphic Properties</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a>tmp_md <span class="ot">&lt;-</span> its18_select_mc_norm_split_no_ac<span class="sc">$</span>edaphic</span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a>tmp_md<span class="sc">$</span>TREAT_T <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tmp_md<span class="sc">$</span>TREAT_T)</span>
<span id="cb125-3"><a href="#cb125-3" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(its18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb125-4"><a href="#cb125-4" aria-hidden="true" tabindex="-1"></a>edaphic_rank <span class="ot">&lt;-</span> <span class="fu">rankindex</span>(tmp_md[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_md)], tmp_comm, </span>
<span id="cb125-5"><a href="#cb125-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">indices =</span> <span class="fu">c</span>(<span class="st">&quot;euc&quot;</span>, <span class="st">&quot;man&quot;</span>, <span class="st">&quot;gow&quot;</span>,<span class="st">&quot;bra&quot;</span>, <span class="st">&quot;kul&quot;</span>), </span>
<span id="cb125-6"><a href="#cb125-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">stepacross =</span> <span class="cn">FALSE</span>, <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>)</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>        euc         man         gow         bra         kul 
 0.15398526  0.29983940 -0.02940098  0.29983940  0.29983940 </code></pre>
</div>
<p>Lets run <code>capscale</code> using Bray-Curtis. Note, we have 15 metadata parameters in this group but, for some reason, <code>capscale</code> only works with 13 parameters. This may have to do with degrees of freedom?</p>
<ul>
<li>Starting properties: AST, H2O, N, P, Al, Ca, Fe, K, Mg, Mn, Na, TEB, ECEC, pH, NH4, NO3, PO4, DOC, DON, DOCN.</li>
<li>Autocorrelated removed: TEB, DON, Na, Al, Ca.</li>
<li>Remove for capscale: Mg, Mn, Na, Al, Fe, K</li>
</ul>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a>edaphic_cap <span class="ot">&lt;-</span> <span class="fu">capscale</span>(tmp_comm <span class="sc">~</span> AST <span class="sc">+</span> H2O <span class="sc">+</span> N <span class="sc">+</span> P <span class="sc">+</span> ECEC <span class="sc">+</span> pH <span class="sc">+</span> </span>
<span id="cb127-2"><a href="#cb127-2" aria-hidden="true" tabindex="-1"></a>                                   NH4 <span class="sc">+</span> NO3 <span class="sc">+</span> PO4 <span class="sc">+</span> DOC <span class="sc">+</span> DOCN, </span>
<span id="cb127-3"><a href="#cb127-3" aria-hidden="true" tabindex="-1"></a>                        tmp_md, <span class="at">dist =</span> <span class="st">&quot;bray&quot;</span>)</span>
<span id="cb127-4"><a href="#cb127-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tmp_md)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>Call: capscale(formula = tmp_comm ~ AST + H2O + N + P + ECEC +
pH + NH4 + NO3 + PO4 + DOC + DOCN, data = tmp_md, distance =
&quot;bray&quot;)

              Inertia Proportion Rank
Total         3.88624    1.00000     
Constrained   3.66503    0.94308   11
Unconstrained 0.22120    0.05692    1
Inertia is squared Bray distance 
Species scores projected from &#39;tmp_comm&#39; 

Eigenvalues for constrained axes:
  CAP1   CAP2   CAP3   CAP4   CAP5   CAP6   CAP7   CAP8   CAP9  CAP10 
0.9655 0.5791 0.4327 0.3671 0.2987 0.2608 0.2301 0.1867 0.1339 0.1135 
 CAP11 
0.0970 

Eigenvalues for unconstrained axes:
  MDS1 
0.2212 </code></pre>
</div>
<p>Now we can look at the variance against each principal component.</p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="metadata_files/figure-html5/unnamed-chunk-179-1.png" width="624" /></p>
</div>
<p>And then make some quick and dirty plots. This will also come in handy later when we need to parse out data a better plot visualization. The <code>ggplot</code> function <code>autoplot</code> stores these data in a more accessible way than the raw results from <code>capscale</code></p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">plot</span>(edaphic_cap) </span></code></pre></div>
<p><img src="metadata_files/figure-html5/unnamed-chunk-180-1.png" width="624" /></p>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt <span class="ot">&lt;-</span> ggplot2<span class="sc">::</span><span class="fu">autoplot</span>(edaphic_cap, <span class="at">arrows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb130-2"><a href="#cb130-2" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt</span></code></pre></div>
<p><img src="metadata_files/figure-html5/unnamed-chunk-180-2.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Next, we need to grab capscale scores for the samples and create a data frame of the first two dimensions. We will also need to add some of the sample details to the data frame. For this we use the vegan function <code>scores</code> which gets species or site scores from the ordination.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">&quot;sites&quot;</span>)</span>
<span id="cb131-2"><a href="#cb131-2" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb131-3"><a href="#cb131-3" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> tmp_samp_scores <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">SampleID =</span> Label)</span>
<span id="cb131-4"><a href="#cb131-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-5"><a href="#cb131-5" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb131-6"><a href="#cb131-6" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md_sub <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb131-7"><a href="#cb131-7" aria-hidden="true" tabindex="-1"></a>edaphic_plot_data <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_md_sub, tmp_samp_scores, <span class="at">by =</span> <span class="st">&quot;SampleID&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>Now we have a new data frame that contains sample details and capscale values.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>          SampleID PLOT TREAT TREAT_T PAIR        CAP1        CAP2
1  P10_D00_010_C0E  P10     C       0    E -0.67150210 -1.07183211
2  P02_D00_010_C0A  P02     C       0    A -0.91911652 -0.41647565
3  P04_D00_010_C0B  P04     C       0    B -0.95654588  0.64087450
4  P06_D00_010_C0C  P06     C       0    C -0.87992057  0.68082267
5  P08_D00_010_C0D  P08     C       0    D -0.07766291 -0.98166349
6  P01_D00_010_W3A  P01     W       3    A -0.79165161 -0.43340347
7  P03_D00_010_W3B  P03     W       3    B  0.24722812  0.81805851
8  P07_D00_010_W3D  P07     W       3    D  0.20151965  1.07639049
9  P09_D00_010_W3E  P09     W       3    E  0.58233467  0.46562279
10 P01_D00_010_W8A  P01     W       8    A  0.92158009 -0.06825914
11 P03_D00_010_W8B  P03     W       8    B  0.94079184 -0.52231363
12 P05_D00_010_W8C  P05     W       8    C  0.87642984 -0.89155825
13 P07_D00_010_W8D  P07     W       8    D  0.52651539  0.70373678</code></pre>
</div>
<p>We can then do the same with the metadata vectors. Here though we only need the scores and parameter name.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a>edaphic_md_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">&quot;biplot&quot;</span>)</span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a>edaphic_md_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb133-3"><a href="#cb133-3" aria-hidden="true" tabindex="-1"></a>edaphic_md_scores <span class="ot">&lt;-</span> edaphic_md_scores <span class="sc">%&gt;%</span> </span>
<span id="cb133-4"><a href="#cb133-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span></span>
<span id="cb133-5"><a href="#cb133-5" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;Label&quot;</span>)</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>     parameters         CAP1         CAP2
AST         AST  0.834527196  0.117196419
H2O         H2O -0.596509128  0.006131247
N             N  0.245428273  0.506082992
P             P  0.097845581  0.205793005
ECEC       ECEC -0.304860202  0.538475609
pH           pH -0.081297235 -0.092037201
NH4         NH4 -0.297079795  0.210528906
NO3         NO3 -0.264254841  0.225782469
PO4         PO4 -0.583698114  0.354036263
DOC         DOC -0.884261492  0.025360275
DOCN       DOCN  0.007258364 -0.462404778</code></pre>
</div>
<p>Lets run some quick correlations of metadata with ordination axes to see which parameters are significant. For this we use the vegan function <code>envfit</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> edaphic_plot_data[, <span class="dv">6</span><span class="sc">:</span><span class="dv">7</span>]</span>
<span id="cb135-2"><a href="#cb135-2" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_samp_scores_sub)</span>
<span id="cb135-3"><a href="#cb135-3" aria-hidden="true" tabindex="-1"></a>tmp_param_list <span class="ot">&lt;-</span> edaphic_md_scores<span class="sc">$</span>parameters</span>
<span id="cb135-4"><a href="#cb135-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-5"><a href="#cb135-5" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> <span class="fu">subset</span>(tmp_md,  <span class="at">select =</span>  tmp_param_list)</span>
<span id="cb135-6"><a href="#cb135-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-7"><a href="#cb135-7" aria-hidden="true" tabindex="-1"></a>envfit_edaphic_md <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, tmp_md_sub,</span>
<span id="cb135-8"><a href="#cb135-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
***VECTORS

         CAP1     CAP2     r2  Pr(&gt;r)  
AST   0.99126  0.13189 0.4849 0.03696 *
H2O  -0.99989  0.01511 0.2434 0.25275  
N     0.43981  0.89809 0.2124 0.27473  
P     0.43273  0.90152 0.0349 0.86014  
ECEC -0.50151  0.86515 0.2595 0.21778  
pH   -0.66815 -0.74403 0.0102 0.94705  
NH4  -0.81970  0.57279 0.0906 0.60140  
NO3  -0.76538  0.64357 0.0824 0.64635  
PO4  -0.85783  0.51393 0.3187 0.15185  
DOC  -0.99945  0.03317 0.5354 0.02797 *
DOCN  0.02108 -0.99978 0.1437 0.47552  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_edaphic_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb137-2"><a href="#cb137-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_edaphic_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb137-3"><a href="#cb137-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_edaphic_md<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.4</span>))</span>
<span id="cb137-4"><a href="#cb137-4" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(edaphic_md_signif_hits)</span>
<span id="cb137-5"><a href="#cb137-5" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(edaphic_md_signif_hits)</span>
<span id="cb137-6"><a href="#cb137-6" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif <span class="ot">&lt;-</span> edaphic_md_scores[edaphic_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> edaphic_md_signif_hits,]</span>
<span id="cb137-7"><a href="#cb137-7" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif<span class="sc">$</span>parameters</span></code></pre></div>
</details>
</div>
<p><code>envfit</code> found that AST, DOC were significantly correlated.</p>
<p>Now lets see if the same parameters are significant for the <code>envfit</code> and <code>bioenv</code> analyses.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;Significant parameters from bioenv analysis.&quot;
[1] &quot;AST&quot;
_____________________________________
[1] &quot;Significant parameters from envfit analysis.&quot;
[1] &quot;AST&quot; &quot;DOC&quot;
_____________________________________
[1] &quot;Found in bioenv but not envfit.&quot;
character(0)
_____________________________________
[1] &quot;Found in envfit but not bioenv.&quot;
[1] &quot;DOC&quot;
_____________________________________
[1] &quot;Found in envfit and bioenv.&quot;
[1] &quot;AST&quot; &quot;DOC&quot;</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a>new_edaphic_md_signif_hits <span class="ot">&lt;-</span> edaphic_sig_diff</span>
<span id="cb139-2"><a href="#cb139-2" aria-hidden="true" tabindex="-1"></a><span class="co">#new_edaphic_md_signif_hits &lt;- append(edaphic_md_signif_hits, edaphic_sig_diff)</span></span>
<span id="cb139-3"><a href="#cb139-3" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_all <span class="ot">&lt;-</span> edaphic_md_scores[edaphic_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> new_edaphic_md_signif_hits,]</span></code></pre></div>
</details>
</div>
<p>Check. Next, we run <code>envfit</code> for the ASVs.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="#cb140-1" aria-hidden="true" tabindex="-1"></a>envfit_edaphic_asv <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, </span>
<span id="cb140-2"><a href="#cb140-2" aria-hidden="true" tabindex="-1"></a>                             tmp_comm[, <span class="fu">order</span>(<span class="fu">colSums</span>(<span class="sc">-</span>tmp_comm))][, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],</span>
<span id="cb140-3"><a href="#cb140-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb140-4"><a href="#cb140-4" aria-hidden="true" tabindex="-1"></a>edaphic_asv_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">&quot;species&quot;</span>)</span>
<span id="cb140-5"><a href="#cb140-5" aria-hidden="true" tabindex="-1"></a>edaphic_asv_scores <span class="ot">&lt;-</span> edaphic_asv_scores <span class="sc">%&gt;%</span> </span>
<span id="cb140-6"><a href="#cb140-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span></span>
<span id="cb140-7"><a href="#cb140-7" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;Label&quot;</span>)</span>
<span id="cb140-8"><a href="#cb140-8" aria-hidden="true" tabindex="-1"></a>edaphic_asv_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
***VECTORS

          CAP1     CAP2     r2   Pr(&gt;r)   
ASV1  -0.61940 -0.78508 0.3539 0.010989 * 
ASV2   0.73098 -0.68240 0.5575 0.003996 **
ASV4  -0.33832 -0.94103 0.4713 0.010989 * 
ASV3  -0.89855  0.43888 0.4897 0.002997 **
ASV15  0.95614  0.29292 0.2269 0.192807   
ASV8  -0.87871 -0.47735 0.1267 0.795205   
ASV9   0.73176  0.68156 0.1410 0.468531   
ASV24  0.45502  0.89048 0.1661 0.459540   
ASV12  0.12678  0.99193 0.7513 0.001998 **
ASV13 -0.42157 -0.90680 0.4286 0.060939 . 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_edaphic_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb142-2"><a href="#cb142-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_edaphic_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb142-3"><a href="#cb142-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_edaphic_asv<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.5</span>))</span>
<span id="cb142-4"><a href="#cb142-4" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(edaphic_asv_signif_hits)</span>
<span id="cb142-5"><a href="#cb142-5" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(edaphic_asv_signif_hits)</span>
<span id="cb142-6"><a href="#cb142-6" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif <span class="ot">&lt;-</span> edaphic_asv_scores[edaphic_asv_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> edaphic_asv_signif_hits,]</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>      parameters       CAP1       CAP2 variable_type
ASV12      ASV12 0.03291551  0.2041479           ASV
ASV2        ASV2 0.48655008 -0.3456690           ASV</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="#cb144-1" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_all<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">&quot;metadata&quot;</span></span>
<span id="cb144-2"><a href="#cb144-2" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">&quot;ASV&quot;</span></span>
<span id="cb144-3"><a href="#cb144-3" aria-hidden="true" tabindex="-1"></a>edaphic_bioplot_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(edaphic_md_signif_all, edaphic_asv_signif)</span></code></pre></div>
</details>
</div>
<p>The last thing to do is categorize parameters scores and ASV scores into different variable types for plotting.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a>edaphic_bioplot_data_md <span class="ot">&lt;-</span> <span class="fu">subset</span>(edaphic_bioplot_data, </span>
<span id="cb145-2"><a href="#cb145-2" aria-hidden="true" tabindex="-1"></a>                                  edaphic_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">&quot;metadata&quot;</span>)</span>
<span id="cb145-3"><a href="#cb145-3" aria-hidden="true" tabindex="-1"></a>edaphic_bioplot_data_asv <span class="ot">&lt;-</span> <span class="fu">subset</span>(edaphic_bioplot_data, </span>
<span id="cb145-4"><a href="#cb145-4" aria-hidden="true" tabindex="-1"></a>                                   edaphic_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">&quot;ASV&quot;</span>)</span></code></pre></div>
</details>
</div>
<details>
<summary>
Show the code for the plot
</summary>
<div class="layout-chunk" data-layout="l-body">

</div>
</details>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="soil-functional-response-4">Soil Functional Response</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="#cb146-1" aria-hidden="true" tabindex="-1"></a>tmp_md <span class="ot">&lt;-</span> its18_select_mc_norm_split_no_ac<span class="sc">$</span>soil_funct</span>
<span id="cb146-2"><a href="#cb146-2" aria-hidden="true" tabindex="-1"></a>tmp_md<span class="sc">$</span>TREAT_T <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tmp_md<span class="sc">$</span>TREAT_T)</span>
<span id="cb146-3"><a href="#cb146-3" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(its18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb146-4"><a href="#cb146-4" aria-hidden="true" tabindex="-1"></a>soil_funct_rank <span class="ot">&lt;-</span> <span class="fu">rankindex</span>(tmp_md[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_md)], tmp_comm, </span>
<span id="cb146-5"><a href="#cb146-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">indices =</span> <span class="fu">c</span>(<span class="st">&quot;euc&quot;</span>, <span class="st">&quot;man&quot;</span>, <span class="st">&quot;gow&quot;</span>, <span class="st">&quot;bra&quot;</span>, <span class="st">&quot;kul&quot;</span>), </span>
<span id="cb146-6"><a href="#cb146-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">stepacross =</span> <span class="cn">FALSE</span>, <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>)</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>        euc         man         gow         bra         kul 
 0.06025620  0.38790324 -0.06200129  0.38790324  0.38790324 </code></pre>
</div>
<p>Lets run <code>capscale</code> using Bray-Curtis Note, we have 11 metadata parameters in this group but, for some reason, <code>capscale</code> only works with 13 parameters. This may have to do with degrees of freedom?</p>
<ul>
<li>Starting properties: micC, micN, micP, micCN, micCP, micNP, AG_ase, BG_ase, BP_ase, CE_ase, P_ase, N_ase, S_ase, XY_ase, LP_ase, PX_ase, CO2, enzCN, enzCP, enzNP</li>
<li>Autocorrelated removed: micN, micNP, enzCN, enzCP, BP_ase, CE_ase, LP_ase, N_ase, P_ase</li>
<li>Remove for capscale: NONE</li>
</ul>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a>soil_funct_cap <span class="ot">&lt;-</span> <span class="fu">capscale</span>(tmp_comm <span class="sc">~</span>  micC <span class="sc">+</span> micP <span class="sc">+</span> micCN <span class="sc">+</span> micCP <span class="sc">+</span> AG_ase <span class="sc">+</span> BG_ase <span class="sc">+</span> </span>
<span id="cb148-2"><a href="#cb148-2" aria-hidden="true" tabindex="-1"></a>                                       S_ase <span class="sc">+</span> XY_ase <span class="sc">+</span> PX_ase <span class="sc">+</span> CO2 <span class="sc">+</span> enzNP, </span>
<span id="cb148-3"><a href="#cb148-3" aria-hidden="true" tabindex="-1"></a>                           tmp_md, <span class="at">dist =</span> <span class="st">&quot;bray&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>Call: capscale(formula = tmp_comm ~ micC + micP + micCN +
micCP + AG_ase + BG_ase + S_ase + XY_ase + PX_ase + CO2 +
enzNP, data = tmp_md, distance = &quot;bray&quot;)

              Inertia Proportion Rank
Total         3.88624    1.00000     
Constrained   3.59902    0.92609   11
Unconstrained 0.28722    0.07391    1
Inertia is squared Bray distance 
Species scores projected from &#39;tmp_comm&#39; 

Eigenvalues for constrained axes:
  CAP1   CAP2   CAP3   CAP4   CAP5   CAP6   CAP7   CAP8   CAP9  CAP10 
0.9666 0.5891 0.3939 0.3377 0.2686 0.2635 0.2306 0.1940 0.1389 0.1143 
 CAP11 
0.1018 

Eigenvalues for unconstrained axes:
   MDS1 
0.28722 </code></pre>
</div>
<p>Now we can look at the variance against each principal component.</p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="metadata_files/figure-html5/unnamed-chunk-205-1.png" width="624" /></p>
</div>
<p>And then make some quick and dirty plots. This will also come in handy later when we need to parse out data a better plot visualization. The <code>ggplot</code> function <code>autoplot</code> stores these data in a more accessible way than the raw results from <code>capscale</code></p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="#cb150-1" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">plot</span>(soil_funct_cap) </span></code></pre></div>
<p><img src="metadata_files/figure-html5/unnamed-chunk-206-1.png" width="624" /></p>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt <span class="ot">&lt;-</span> <span class="fu">autoplot</span>(soil_funct_cap, <span class="at">arrows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb151-2"><a href="#cb151-2" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt</span></code></pre></div>
<p><img src="metadata_files/figure-html5/unnamed-chunk-206-2.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Next, we need to grab capscale scores for the samples and create a data frame of the first two dimensions. We will also need to add some of the sample details to the data frame. For this we use the vegan function <code>scores</code> which gets species or site scores from the ordination.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="#cb152-1" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">&quot;sites&quot;</span>)</span>
<span id="cb152-2"><a href="#cb152-2" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb152-3"><a href="#cb152-3" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> tmp_samp_scores <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">SampleID =</span> Label)</span>
<span id="cb152-4"><a href="#cb152-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-5"><a href="#cb152-5" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb152-6"><a href="#cb152-6" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md_sub <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb152-7"><a href="#cb152-7" aria-hidden="true" tabindex="-1"></a>soil_funct_plot_data <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_md_sub, tmp_samp_scores, </span>
<span id="cb152-8"><a href="#cb152-8" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">by =</span> <span class="st">&quot;SampleID&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>Now we have a new data frame that contains sample details and capscale values.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>          SampleID PLOT TREAT TREAT_T PAIR        CAP1       CAP2
1  P10_D00_010_C0E  P10     C       0    E -0.69489162 -0.6675049
2  P02_D00_010_C0A  P02     C       0    A -0.91653485 -0.6229848
3  P04_D00_010_C0B  P04     C       0    B -0.95221897  0.3986135
4  P06_D00_010_C0C  P06     C       0    C -0.85713972  0.6268677
5  P08_D00_010_C0D  P08     C       0    D -0.09115803 -1.0802252
6  P01_D00_010_W3A  P01     W       3    A -0.80868973 -0.1643386
7  P03_D00_010_W3B  P03     W       3    B  0.27067347  0.9200087
8  P07_D00_010_W3D  P07     W       3    D  0.20427135  1.0264780
9  P09_D00_010_W3E  P09     W       3    E  0.57963547  0.4440199
10 P01_D00_010_W8A  P01     W       8    A  0.91663981 -0.1382072
11 P03_D00_010_W8B  P03     W       8    B  0.92199435 -0.6783194
12 P05_D00_010_W8C  P05     W       8    C  0.86406950 -0.9575454
13 P07_D00_010_W8D  P07     W       8    D  0.56334898  0.8931377</code></pre>
</div>
<p>We can then do the same with the metadata vectors. Here though we only need the scores and parameter name.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="#cb154-1" aria-hidden="true" tabindex="-1"></a>soil_funct_md_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">&quot;biplot&quot;</span>)</span>
<span id="cb154-2"><a href="#cb154-2" aria-hidden="true" tabindex="-1"></a>soil_funct_md_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb154-3"><a href="#cb154-3" aria-hidden="true" tabindex="-1"></a>soil_funct_md_scores <span class="ot">&lt;-</span> soil_funct_md_scores <span class="sc">%&gt;%</span> </span>
<span id="cb154-4"><a href="#cb154-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span> </span>
<span id="cb154-5"><a href="#cb154-5" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;Label&quot;</span>)</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>       parameters       CAP1         CAP2
micC         micC -0.2359175  0.400384343
micP         micP -0.4872005  0.846844390
micCN       micCN  0.5085406  0.215263003
micCP       micCP  0.4299937 -0.786734788
AG_ase     AG_ase -0.3381241  0.763769448
BG_ase     BG_ase  0.2150249  0.021210923
S_ase       S_ase  0.1648172  0.678698244
XY_ase     XY_ase  0.6119362  0.164529372
PX_ase     PX_ase  0.8280069 -0.009182309
CO2           CO2  0.5373302 -0.181241128
enzNP       enzNP -0.8493447  0.170758052</code></pre>
</div>
<p>Lets run some quick correlations of metadata with ordination axes to see which parameters are significant. For this we use the vegan function <code>envfit</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="#cb156-1" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> soil_funct_plot_data[, <span class="dv">6</span><span class="sc">:</span><span class="dv">7</span>]</span>
<span id="cb156-2"><a href="#cb156-2" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_samp_scores_sub)</span>
<span id="cb156-3"><a href="#cb156-3" aria-hidden="true" tabindex="-1"></a>tmp_param_list <span class="ot">&lt;-</span> soil_funct_md_scores<span class="sc">$</span>parameters</span>
<span id="cb156-4"><a href="#cb156-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-5"><a href="#cb156-5" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> <span class="fu">subset</span>(tmp_md,  <span class="at">select =</span>  tmp_param_list)</span>
<span id="cb156-6"><a href="#cb156-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-7"><a href="#cb156-7" aria-hidden="true" tabindex="-1"></a>envfit_soil_funct_md <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, tmp_md_sub,</span>
<span id="cb156-8"><a href="#cb156-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
***VECTORS

           CAP1     CAP2     r2   Pr(&gt;r)   
micC   -0.50911  0.86070 0.1567 0.442557   
micP   -0.50009  0.86597 0.6925 0.001998 **
micCN   0.92125  0.38897 0.2225 0.261738   
micCP   0.48092 -0.87676 0.5831 0.015984 * 
AG_ase -0.40574  0.91399 0.5060 0.036963 * 
BG_ase  0.99507  0.09917 0.0341 0.832168   
S_ase   0.23862  0.97111 0.3542 0.099900 . 
XY_ase  0.96571  0.25962 0.2931 0.170829   
PX_ase  0.99996 -0.00944 0.5002 0.034965 * 
CO2     0.94850 -0.31679 0.2342 0.250749   
enzNP  -0.98088  0.19460 0.5471 0.013986 * 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="#cb158-1" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_soil_funct_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb158-2"><a href="#cb158-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_soil_funct_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb158-3"><a href="#cb158-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_soil_funct_md<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.4</span>))</span>
<span id="cb158-4"><a href="#cb158-4" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(soil_funct_md_signif_hits)</span>
<span id="cb158-5"><a href="#cb158-5" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(soil_funct_md_signif_hits)</span>
<span id="cb158-6"><a href="#cb158-6" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif <span class="ot">&lt;-</span> soil_funct_md_scores[soil_funct_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb158-7"><a href="#cb158-7" aria-hidden="true" tabindex="-1"></a>                                               soil_funct_md_signif_hits,]</span>
<span id="cb158-8"><a href="#cb158-8" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif<span class="sc">$</span>parameters</span></code></pre></div>
</details>
</div>
<p><code>envfit</code> found that micP, micCP, AG_ase, PX_ase, enzNP were significantly correlated.</p>
<p>Now lets see if the same parameters are significant for the <code>envfit</code> and <code>bioenv</code> analyses.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;Significant parameters from bioenv analysis.&quot;
[1] &quot;enzNP&quot;  &quot;PX_ase&quot; &quot;XY_ase&quot;
_____________________________________
[1] &quot;Significant parameters from envfit analysis.&quot;
[1] &quot;micP&quot;   &quot;micCP&quot;  &quot;AG_ase&quot; &quot;PX_ase&quot; &quot;enzNP&quot; 
_____________________________________
[1] &quot;Found in bioenv but not envfit.&quot;
[1] &quot;XY_ase&quot;
_____________________________________
[1] &quot;Found in envfit but not bioenv.&quot;
[1] &quot;micP&quot;   &quot;micCP&quot;  &quot;AG_ase&quot;
_____________________________________
[1] &quot;Found in envfit and bioenv.&quot;
[1] &quot;micP&quot;   &quot;micCP&quot;  &quot;AG_ase&quot; &quot;PX_ase&quot; &quot;enzNP&quot;  &quot;XY_ase&quot;</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="#cb160-1" aria-hidden="true" tabindex="-1"></a><span class="co">#new_soil_funct_md_signif_hits &lt;- append(soil_funct_md_signif_hits, soil_funct_sig_diff)</span></span>
<span id="cb160-2"><a href="#cb160-2" aria-hidden="true" tabindex="-1"></a>new_soil_funct_md_signif_hits <span class="ot">&lt;-</span> soil_funct_sig_diff</span>
<span id="cb160-3"><a href="#cb160-3" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_all <span class="ot">&lt;-</span> soil_funct_md_scores[soil_funct_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb160-4"><a href="#cb160-4" aria-hidden="true" tabindex="-1"></a>                                                   new_soil_funct_md_signif_hits,]</span></code></pre></div>
</details>
</div>
<p>Check. Next, we run <code>envfit</code> for the ASVs.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="#cb161-1" aria-hidden="true" tabindex="-1"></a>envfit_soil_funct_asv <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, </span>
<span id="cb161-2"><a href="#cb161-2" aria-hidden="true" tabindex="-1"></a>                                tmp_comm[, <span class="fu">order</span>(<span class="fu">colSums</span>(<span class="sc">-</span>tmp_comm))][, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>], </span>
<span id="cb161-3"><a href="#cb161-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb161-4"><a href="#cb161-4" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">&quot;species&quot;</span>)</span>
<span id="cb161-5"><a href="#cb161-5" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_scores <span class="ot">&lt;-</span> soil_funct_asv_scores <span class="sc">%&gt;%</span> </span>
<span id="cb161-6"><a href="#cb161-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span> </span>
<span id="cb161-7"><a href="#cb161-7" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;Label&quot;</span>)</span>
<span id="cb161-8"><a href="#cb161-8" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
***VECTORS

          CAP1     CAP2     r2   Pr(&gt;r)    
ASV1  -0.79963 -0.60050 0.2256 0.245754    
ASV2   0.68178 -0.73156 0.6129 0.000999 ***
ASV4  -0.33208 -0.94325 0.5433 0.000999 ***
ASV3  -0.92213  0.38687 0.4516 0.010989 *  
ASV15  0.95847  0.28518 0.2287 0.159840    
ASV8  -0.97867 -0.20544 0.1064 0.930070    
ASV9   0.76380  0.64546 0.1326 0.481518    
ASV24  0.40983  0.91216 0.2438 0.065934 .  
ASV12  0.14788  0.98901 0.7872 0.000999 ***
ASV13 -0.42711 -0.90420 0.4546 0.044955 *  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="#cb163-1" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_soil_funct_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb163-2"><a href="#cb163-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_soil_funct_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb163-3"><a href="#cb163-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_soil_funct_asv<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.5</span>))</span>
<span id="cb163-4"><a href="#cb163-4" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(soil_funct_asv_signif_hits)</span>
<span id="cb163-5"><a href="#cb163-5" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(soil_funct_asv_signif_hits)</span>
<span id="cb163-6"><a href="#cb163-6" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif <span class="ot">&lt;-</span> soil_funct_asv_scores[soil_funct_asv_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb163-7"><a href="#cb163-7" aria-hidden="true" tabindex="-1"></a>                                                 soil_funct_asv_signif_hits,]</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>      parameters        CAP1       CAP2 variable_type
ASV4        ASV4 -0.10677580 -0.2996742           ASV
ASV12      ASV12  0.04088024  0.2662493           ASV
ASV2        ASV2  0.46539405 -0.4834130           ASV</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb165"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb165-1"><a href="#cb165-1" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_all<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">&quot;metadata&quot;</span></span>
<span id="cb165-2"><a href="#cb165-2" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">&quot;ASV&quot;</span></span>
<span id="cb165-3"><a href="#cb165-3" aria-hidden="true" tabindex="-1"></a>soil_funct_bioplot_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(soil_funct_md_signif_all, soil_funct_asv_signif)</span></code></pre></div>
</details>
</div>
<p>The last thing to do is categorize parameters scores and ASV scores into different variable types for plotting.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb166"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb166-1"><a href="#cb166-1" aria-hidden="true" tabindex="-1"></a>soil_funct_bioplot_data_md <span class="ot">&lt;-</span> <span class="fu">subset</span>(soil_funct_bioplot_data, </span>
<span id="cb166-2"><a href="#cb166-2" aria-hidden="true" tabindex="-1"></a>                                  soil_funct_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">&quot;metadata&quot;</span>)</span>
<span id="cb166-3"><a href="#cb166-3" aria-hidden="true" tabindex="-1"></a>soil_funct_bioplot_data_asv <span class="ot">&lt;-</span> <span class="fu">subset</span>(soil_funct_bioplot_data, </span>
<span id="cb166-4"><a href="#cb166-4" aria-hidden="true" tabindex="-1"></a>                                   soil_funct_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">&quot;ASV&quot;</span>)</span></code></pre></div>
</details>
</div>
<details>
<summary>
Show the code for the plot
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="#cb167-1" aria-hidden="true" tabindex="-1"></a>soil_funct_cap_vals <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(soil_funct_cap<span class="sc">$</span>CCA<span class="sc">$</span>eig[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span>
<span id="cb167-2"><a href="#cb167-2" aria-hidden="true" tabindex="-1"></a>soil_funct_cap1 <span class="ot">&lt;-</span> <span class="fu">signif</span>((soil_funct_cap_vals[<span class="dv">1</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb167-3"><a href="#cb167-3" aria-hidden="true" tabindex="-1"></a>soil_funct_cap2 <span class="ot">&lt;-</span> <span class="fu">signif</span>((soil_funct_cap_vals[<span class="dv">2</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb167-4"><a href="#cb167-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb167-5"><a href="#cb167-5" aria-hidden="true" tabindex="-1"></a>cpa1_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;CAP1&quot;</span>, <span class="st">&quot; (&quot;</span>, soil_funct_cap1, <span class="st">&quot;%)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb167-6"><a href="#cb167-6" aria-hidden="true" tabindex="-1"></a>cpa2_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;CAP2&quot;</span>, <span class="st">&quot; (&quot;</span>, soil_funct_cap2, <span class="st">&quot;%)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb167-7"><a href="#cb167-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb167-8"><a href="#cb167-8" aria-hidden="true" tabindex="-1"></a>swel_col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#2271B2&quot;</span>, <span class="st">&quot;#71B222&quot;</span>, <span class="st">&quot;#B22271&quot;</span>)</span>
<span id="cb167-9"><a href="#cb167-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb167-10"><a href="#cb167-10" aria-hidden="true" tabindex="-1"></a>soil_funct_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(soil_funct_plot_data) <span class="sc">+</span></span>
<span id="cb167-11"><a href="#cb167-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">shape =</span> TREAT,</span>
<span id="cb167-12"><a href="#cb167-12" aria-hidden="true" tabindex="-1"></a>                           <span class="at">colour =</span> TREAT_T), <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb167-13"><a href="#cb167-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> swel_col) <span class="sc">+</span></span>
<span id="cb167-14"><a href="#cb167-14" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_text(data = soil_funct_plot_data, aes(x = CAP1, y = CAP2, #UNCOMMENT to add sample labels</span></span>
<span id="cb167-15"><a href="#cb167-15" aria-hidden="true" tabindex="-1"></a><span class="co">#                              label = SampleID), size = 3) + </span></span>
<span id="cb167-16"><a href="#cb167-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">xend =</span> CAP1, <span class="at">yend =</span> CAP2),</span>
<span id="cb167-17"><a href="#cb167-17" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> soil_funct_bioplot_data_md, <span class="at">linetype =</span> <span class="st">&quot;solid&quot;</span>,</span>
<span id="cb167-18"><a href="#cb167-18" aria-hidden="true" tabindex="-1"></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">&quot;cm&quot;</span>)), <span class="at">size =</span> <span class="fl">0.4</span>, </span>
<span id="cb167-19"><a href="#cb167-19" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">&quot;#191919&quot;</span>) <span class="sc">+</span></span>
<span id="cb167-20"><a href="#cb167-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> soil_funct_bioplot_data_md, </span>
<span id="cb167-21"><a href="#cb167-21" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">label =</span> parameters), <span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb167-22"><a href="#cb167-22" aria-hidden="true" tabindex="-1"></a>            <span class="at">nudge_x =</span> <span class="fl">0.1</span>, <span class="at">nudge_y =</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb167-23"><a href="#cb167-23" aria-hidden="true" tabindex="-1"></a>  <span class="do">## USE this code to overlay ASV vestors</span></span>
<span id="cb167-24"><a href="#cb167-24" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_segment(aes(x = 0, y = 0, xend = CAP1, yend = CAP2),</span></span>
<span id="cb167-25"><a href="#cb167-25" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             data = soil_funct_bioplot_data_asv, linetype = &quot;solid&quot;,</span></span>
<span id="cb167-26"><a href="#cb167-26" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             arrow = arrow(length = unit(0.3, &quot;cm&quot;)), size = 0.2,</span></span>
<span id="cb167-27"><a href="#cb167-27" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             color = &quot;#676767&quot;) +</span></span>
<span id="cb167-28"><a href="#cb167-28" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text(data = soil_funct_bioplot_data_asv, </span></span>
<span id="cb167-29"><a href="#cb167-29" aria-hidden="true" tabindex="-1"></a>  <span class="co">#          aes(x = CAP1, y = CAP2, label = parameters), size = 2.5,</span></span>
<span id="cb167-30"><a href="#cb167-30" aria-hidden="true" tabindex="-1"></a>  <span class="co">#          nudge_x = 0.05, nudge_y = 0.05) +</span></span>
<span id="cb167-31"><a href="#cb167-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span> </span>
<span id="cb167-32"><a href="#cb167-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Capscale Analysis&quot;</span>,</span>
<span id="cb167-33"><a href="#cb167-33" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">&quot;Soil Functional Response&quot;</span>, </span>
<span id="cb167-34"><a href="#cb167-34" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> cpa1_lab, </span>
<span id="cb167-35"><a href="#cb167-35" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> cpa2_lab)</span>
<span id="cb167-36"><a href="#cb167-36" aria-hidden="true" tabindex="-1"></a>soil_funct_plot <span class="ot">&lt;-</span> soil_funct_plot <span class="sc">+</span> <span class="fu">coord_fixed</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb167-37"><a href="#cb167-37" aria-hidden="true" tabindex="-1"></a>soil_funct_plot</span>
<span id="cb167-38"><a href="#cb167-38" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">&quot;files/metadata/figures/its18_soil_funct_capscale.png&quot;</span>,</span>
<span id="cb167-39"><a href="#cb167-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">16</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">units =</span> <span class="st">&#39;cm&#39;</span>, <span class="at">res =</span> <span class="dv">600</span>, <span class="at">bg =</span> <span class="st">&quot;white&quot;</span>)</span>
<span id="cb167-40"><a href="#cb167-40" aria-hidden="true" tabindex="-1"></a>soil_funct_plot</span>
<span id="cb167-41"><a href="#cb167-41" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb167-42"><a href="#cb167-42" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">&quot;files/metadata/figures/its18_soil_funct_capscale.pdf&quot;</span>,</span>
<span id="cb167-43"><a href="#cb167-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">5</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb167-44"><a href="#cb167-44" aria-hidden="true" tabindex="-1"></a>soil_funct_plot</span>
<span id="cb167-45"><a href="#cb167-45" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
</div>
</details>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="temperature-adaptation-6">Temperature Adaptation</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb168"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb168-1"><a href="#cb168-1" aria-hidden="true" tabindex="-1"></a>tmp_md <span class="ot">&lt;-</span> its18_select_mc_norm_split_no_ac<span class="sc">$</span>temp_adapt</span>
<span id="cb168-2"><a href="#cb168-2" aria-hidden="true" tabindex="-1"></a>tmp_md<span class="sc">$</span>TREAT_T <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tmp_md<span class="sc">$</span>TREAT_T)</span>
<span id="cb168-3"><a href="#cb168-3" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(its18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb168-4"><a href="#cb168-4" aria-hidden="true" tabindex="-1"></a>temp_adapt_rank <span class="ot">&lt;-</span> <span class="fu">rankindex</span>(tmp_md[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_md)], tmp_comm, </span>
<span id="cb168-5"><a href="#cb168-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">indices =</span> <span class="fu">c</span>(<span class="st">&quot;euc&quot;</span>, <span class="st">&quot;man&quot;</span>, <span class="st">&quot;gow&quot;</span>, <span class="st">&quot;bra&quot;</span>, <span class="st">&quot;kul&quot;</span>), </span>
<span id="cb168-6"><a href="#cb168-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">stepacross =</span> <span class="cn">FALSE</span>, <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>)</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>       euc        man        gow        bra        kul 
0.05132842 0.39465598 0.27457353 0.39465598 0.39465598 </code></pre>
</div>
<p>Lets run <code>capscale</code> using Bray-Curtis. Note, we have 12 metadata parameters in this group but, for some reason, <code>capscale</code> only works with 13 parameters. This may have to do with degrees of freedom?</p>
<ul>
<li>Starting properties: AG_Q10, BG_Q10, BP_Q10, CE_Q10, P_Q10, N_Q10, S_Q10, XY_Q10, LP_Q10, PX_Q10, CUEcn, CUEcp, NUE, PUE, Tmin, SI</li>
<li>Autocorrelated removed: NUE, PUE, P_Q10, SI<br />
</li>
<li>Remove for capscale: S_Q10</li>
</ul>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb170"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb170-1"><a href="#cb170-1" aria-hidden="true" tabindex="-1"></a>temp_adapt_cap <span class="ot">&lt;-</span> <span class="fu">capscale</span>(tmp_comm <span class="sc">~</span>  AG_Q10 <span class="sc">+</span> BG_Q10 <span class="sc">+</span> BP_Q10 <span class="sc">+</span> CE_Q10 <span class="sc">+</span> </span>
<span id="cb170-2"><a href="#cb170-2" aria-hidden="true" tabindex="-1"></a>                                       N_Q10 <span class="sc">+</span> XY_Q10 <span class="sc">+</span> LP_Q10 <span class="sc">+</span> PX_Q10 <span class="sc">+</span> </span>
<span id="cb170-3"><a href="#cb170-3" aria-hidden="true" tabindex="-1"></a>                                       CUEcn <span class="sc">+</span> CUEcp <span class="sc">+</span> Tmin, </span>
<span id="cb170-4"><a href="#cb170-4" aria-hidden="true" tabindex="-1"></a>                           tmp_md, <span class="at">dist =</span> <span class="st">&quot;bray&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>Call: capscale(formula = tmp_comm ~ AG_Q10 + BG_Q10 + BP_Q10 +
CE_Q10 + N_Q10 + XY_Q10 + LP_Q10 + PX_Q10 + CUEcn + CUEcp +
Tmin, data = tmp_md, distance = &quot;bray&quot;)

              Inertia Proportion Rank
Total         3.88624    1.00000     
Constrained   3.64416    0.93771   11
Unconstrained 0.24208    0.06229    1
Inertia is squared Bray distance 
Species scores projected from &#39;tmp_comm&#39; 

Eigenvalues for constrained axes:
  CAP1   CAP2   CAP3   CAP4   CAP5   CAP6   CAP7   CAP8   CAP9  CAP10 
0.9367 0.5976 0.4447 0.3685 0.3136 0.2644 0.2093 0.1488 0.1377 0.1235 
 CAP11 
0.0994 

Eigenvalues for unconstrained axes:
   MDS1 
0.24208 </code></pre>
</div>
<p>Now we can look at the variance against each principal component.</p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="metadata_files/figure-html5/unnamed-chunk-231-1.png" width="624" /></p>
</div>
<p>And then make some quick and dirty plots. This will also come in handy later when we need to parse out data a better plot visualization. The <code>ggplot</code> function <code>autoplot</code> stores these data in a more accessible way than the raw results from <code>capscale</code></p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb172"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb172-1"><a href="#cb172-1" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">plot</span>(temp_adapt_cap) </span></code></pre></div>
<p><img src="metadata_files/figure-html5/unnamed-chunk-232-1.png" width="624" /></p>
<div class="sourceCode" id="cb173"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb173-1"><a href="#cb173-1" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt <span class="ot">&lt;-</span> <span class="fu">autoplot</span>(temp_adapt_cap, <span class="at">arrows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb173-2"><a href="#cb173-2" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt</span></code></pre></div>
<p><img src="metadata_files/figure-html5/unnamed-chunk-232-2.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Next, we need to grab capscale scores for the samples and create a data frame of the first two dimensions. We will also need to add some of the sample details to the data frame. For this we use the vegan function <code>scores</code> which gets species or site scores from the ordination.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="#cb174-1" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">&quot;sites&quot;</span>)</span>
<span id="cb174-2"><a href="#cb174-2" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb174-3"><a href="#cb174-3" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> tmp_samp_scores <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">SampleID =</span> Label)</span>
<span id="cb174-4"><a href="#cb174-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-5"><a href="#cb174-5" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb174-6"><a href="#cb174-6" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md_sub <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb174-7"><a href="#cb174-7" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot_data <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_md_sub, tmp_samp_scores, </span>
<span id="cb174-8"><a href="#cb174-8" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">by =</span> <span class="st">&quot;SampleID&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>Now we have a new data frame that contains sample details and capscale values.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>          SampleID PLOT TREAT TREAT_T PAIR        CAP1       CAP2
1  P10_D00_010_C0E  P10     C       0    E -0.70909404  0.8884734
2  P02_D00_010_C0A  P02     C       0    A -0.88932428  0.4678685
3  P04_D00_010_C0B  P04     C       0    B -0.94720017 -0.5168746
4  P06_D00_010_C0C  P06     C       0    C -0.94117253 -0.7283971
5  P08_D00_010_C0D  P08     C       0    D -0.09697067  1.0013016
6  P01_D00_010_W3A  P01     W       3    A -0.79138504  0.3434287
7  P03_D00_010_W3B  P03     W       3    B  0.28447425 -0.8800769
8  P07_D00_010_W3D  P07     W       3    D  0.18992920 -0.9828048
9  P09_D00_010_W3E  P09     W       3    E  0.57704340 -0.4016378
10 P01_D00_010_W8A  P01     W       8    A  0.98434765  0.1432237
11 P03_D00_010_W8B  P03     W       8    B  0.88625427  0.6330378
12 P05_D00_010_W8C  P05     W       8    C  0.84253827  0.9056297
13 P07_D00_010_W8D  P07     W       8    D  0.61055970 -0.8731722</code></pre>
</div>
<p>We can then do the same with the metadata vectors. Here though we only need the scores and parameter name.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb176-1"><a href="#cb176-1" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">&quot;biplot&quot;</span>)</span>
<span id="cb176-2"><a href="#cb176-2" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb176-3"><a href="#cb176-3" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_scores <span class="ot">&lt;-</span> temp_adapt_md_scores <span class="sc">%&gt;%</span> </span>
<span id="cb176-4"><a href="#cb176-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span> </span>
<span id="cb176-5"><a href="#cb176-5" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;Label&quot;</span>)</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>       parameters         CAP1        CAP2
AG_Q10     AG_Q10  0.138111424 -0.09067317
BG_Q10     BG_Q10  0.529871661  0.44030269
BP_Q10     BP_Q10  0.213674949  0.18420563
CE_Q10     CE_Q10 -0.470374849 -0.07935444
N_Q10       N_Q10  0.002856862  0.28186886
XY_Q10     XY_Q10  0.826073837  0.18643092
LP_Q10     LP_Q10 -0.428296135 -0.07664534
PX_Q10     PX_Q10 -0.361920412  0.05606159
CUEcn       CUEcn -0.146194994 -0.40205479
CUEcp       CUEcp  0.318510258  0.66752667
Tmin         Tmin  0.740618311 -0.05737920</code></pre>
</div>
<p>Lets run some quick correlations of metadata with ordination axes to see which parameters are significant. For this we use the vegan function <code>envfit</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="#cb178-1" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> temp_adapt_plot_data[, <span class="dv">6</span><span class="sc">:</span><span class="dv">7</span>]</span>
<span id="cb178-2"><a href="#cb178-2" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_samp_scores_sub)</span>
<span id="cb178-3"><a href="#cb178-3" aria-hidden="true" tabindex="-1"></a>tmp_param_list <span class="ot">&lt;-</span> temp_adapt_md_scores<span class="sc">$</span>parameters</span>
<span id="cb178-4"><a href="#cb178-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-5"><a href="#cb178-5" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> <span class="fu">subset</span>(tmp_md,  <span class="at">select =</span>  tmp_param_list)</span>
<span id="cb178-6"><a href="#cb178-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-7"><a href="#cb178-7" aria-hidden="true" tabindex="-1"></a>envfit_temp_adapt_md <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, tmp_md_sub,</span>
<span id="cb178-8"><a href="#cb178-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
***VECTORS

           CAP1     CAP2     r2  Pr(&gt;r)   
AG_Q10  0.82904 -0.55918 0.0237 0.89710   
BG_Q10  0.76182  0.64778 0.4109 0.07792 . 
BP_Q10  0.74983  0.66163 0.0689 0.71528   
CE_Q10 -0.98593 -0.16714 0.1956 0.34865   
N_Q10   0.00657  0.99998 0.0700 0.71029   
XY_Q10  0.97498  0.22227 0.6166 0.00999 **
LP_Q10 -0.98418 -0.17719 0.1627 0.44955   
PX_Q10 -0.98713  0.15993 0.1155 0.54246   
CUEcn  -0.33206 -0.94326 0.1605 0.40460   
CUEcp   0.42027  0.90740 0.4786 0.03497 * 
Tmin    0.99659 -0.08248 0.4749 0.02797 * 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb180"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb180-1"><a href="#cb180-1" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_temp_adapt_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb180-2"><a href="#cb180-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_temp_adapt_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb180-3"><a href="#cb180-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_temp_adapt_md<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.4</span>))</span>
<span id="cb180-4"><a href="#cb180-4" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(temp_adapt_md_signif_hits)</span>
<span id="cb180-5"><a href="#cb180-5" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(temp_adapt_md_signif_hits)</span>
<span id="cb180-6"><a href="#cb180-6" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif <span class="ot">&lt;-</span> temp_adapt_md_scores[temp_adapt_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb180-7"><a href="#cb180-7" aria-hidden="true" tabindex="-1"></a>                                               temp_adapt_md_signif_hits,]</span></code></pre></div>
</details>
</div>
<p><code>envfit</code> found that XY_Q10, CUEcp, Tmin were significantly correlated.</p>
<p>Now lets see if the same parameters are significant for the <code>envfit</code> and <code>bioenv</code> analyses.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] &quot;Significant parameters from bioenv analysis.&quot;
[1] &quot;XY_Q10&quot; &quot;Tmin&quot;  
_____________________________________
[1] &quot;Significant parameters from envfit analysis.&quot;
[1] &quot;XY_Q10&quot; &quot;CUEcp&quot;  &quot;Tmin&quot;  
_____________________________________
[1] &quot;Found in bioenv but not envfit.&quot;
character(0)
_____________________________________
[1] &quot;Found in envfit but not bioenv.&quot;
[1] &quot;CUEcp&quot;
_____________________________________
[1] &quot;Found in envfit and  bioenv.&quot;
[1] &quot;XY_Q10&quot; &quot;CUEcp&quot;  &quot;Tmin&quot;  </code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb182"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb182-1"><a href="#cb182-1" aria-hidden="true" tabindex="-1"></a><span class="co">#new_temp_adapt_md_signif_hits &lt;- base::append(temp_adapt_md_signif_hits, temp_adapt_sig_diff)</span></span>
<span id="cb182-2"><a href="#cb182-2" aria-hidden="true" tabindex="-1"></a>new_temp_adapt_md_signif_hits <span class="ot">&lt;-</span> temp_adapt_sig_diff[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb182-3"><a href="#cb182-3" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_all <span class="ot">&lt;-</span> temp_adapt_md_scores[temp_adapt_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb182-4"><a href="#cb182-4" aria-hidden="true" tabindex="-1"></a>                                                   new_temp_adapt_md_signif_hits,]</span></code></pre></div>
</details>
</div>
<p>Check. Next, we run <code>envfit</code> for the ASVs.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="#cb183-1" aria-hidden="true" tabindex="-1"></a>envfit_temp_adapt_asv <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, </span>
<span id="cb183-2"><a href="#cb183-2" aria-hidden="true" tabindex="-1"></a>                                tmp_comm[, <span class="fu">order</span>(<span class="fu">colSums</span>(<span class="sc">-</span>tmp_comm))][, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],</span>
<span id="cb183-3"><a href="#cb183-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb183-4"><a href="#cb183-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-5"><a href="#cb183-5" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">&quot;species&quot;</span>)</span>
<span id="cb183-6"><a href="#cb183-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-7"><a href="#cb183-7" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_scores <span class="ot">&lt;-</span> temp_adapt_asv_scores <span class="sc">%&gt;%</span> </span>
<span id="cb183-8"><a href="#cb183-8" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span></span>
<span id="cb183-9"><a href="#cb183-9" aria-hidden="true" tabindex="-1"></a>                         tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;Label&quot;</span>)</span>
<span id="cb183-10"><a href="#cb183-10" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
***VECTORS

          CAP1     CAP2     r2   Pr(&gt;r)    
ASV1  -0.68969  0.72410 0.2975 0.051948 .  
ASV2   0.68393  0.72955 0.5592 0.003996 ** 
ASV4  -0.34226  0.93961 0.4935 0.005994 ** 
ASV3  -0.89727 -0.44148 0.5062 0.000999 ***
ASV15  0.96528 -0.26123 0.2497 0.122877    
ASV8  -0.90985  0.41495 0.1146 0.923077    
ASV9   0.76209 -0.64748 0.1338 0.474525    
ASV24  0.44129 -0.89736 0.2375 0.095904 .  
ASV12  0.14568 -0.98933 0.7737 0.002997 ** 
ASV13 -0.41856  0.90819 0.4357 0.056943 .  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb185-1"><a href="#cb185-1" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_temp_adapt_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb185-2"><a href="#cb185-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_temp_adapt_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb185-3"><a href="#cb185-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_temp_adapt_asv<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.5</span>))</span>
<span id="cb185-4"><a href="#cb185-4" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(temp_adapt_asv_signif_hits)</span>
<span id="cb185-5"><a href="#cb185-5" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(temp_adapt_asv_signif_hits)</span>
<span id="cb185-6"><a href="#cb185-6" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif <span class="ot">&lt;-</span> temp_adapt_asv_scores[temp_adapt_asv_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb185-7"><a href="#cb185-7" aria-hidden="true" tabindex="-1"></a>                                                 temp_adapt_asv_signif_hits,]</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>      parameters        CAP1       CAP2 variable_type
ASV12      ASV12  0.03533894 -0.2327656           ASV
ASV2        ASV2  0.38413654  0.3977765           ASV
ASV3        ASV3 -0.45977392 -0.1843674           ASV</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="#cb187-1" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_all<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">&quot;metadata&quot;</span></span>
<span id="cb187-2"><a href="#cb187-2" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">&quot;ASV&quot;</span></span>
<span id="cb187-3"><a href="#cb187-3" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioplot_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(temp_adapt_md_signif_all, temp_adapt_asv_signif)</span></code></pre></div>
</details>
</div>
<p>The last thing to do is categorize parameters scores and ASV scores into different variable types for plotting.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="#cb188-1" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioplot_data_md <span class="ot">&lt;-</span> <span class="fu">subset</span>(temp_adapt_bioplot_data, </span>
<span id="cb188-2"><a href="#cb188-2" aria-hidden="true" tabindex="-1"></a>                                  temp_adapt_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">&quot;metadata&quot;</span>)</span>
<span id="cb188-3"><a href="#cb188-3" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioplot_data_asv <span class="ot">&lt;-</span> <span class="fu">subset</span>(temp_adapt_bioplot_data, </span>
<span id="cb188-4"><a href="#cb188-4" aria-hidden="true" tabindex="-1"></a>                                   temp_adapt_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">&quot;ASV&quot;</span>)</span></code></pre></div>
</details>
</div>
<details>
<summary>
Show the code for the plot
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb189"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb189-1"><a href="#cb189-1" aria-hidden="true" tabindex="-1"></a>temp_adapt_cap_vals <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(temp_adapt_cap<span class="sc">$</span>CCA<span class="sc">$</span>eig[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span>
<span id="cb189-2"><a href="#cb189-2" aria-hidden="true" tabindex="-1"></a>temp_adapt_cap1 <span class="ot">&lt;-</span> <span class="fu">signif</span>((temp_adapt_cap_vals[<span class="dv">1</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb189-3"><a href="#cb189-3" aria-hidden="true" tabindex="-1"></a>temp_adapt_cap2 <span class="ot">&lt;-</span> <span class="fu">signif</span>((temp_adapt_cap_vals[<span class="dv">2</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb189-4"><a href="#cb189-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5"><a href="#cb189-5" aria-hidden="true" tabindex="-1"></a>cpa1_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;CAP1&quot;</span>, <span class="st">&quot; (&quot;</span>, temp_adapt_cap1, <span class="st">&quot;%)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb189-6"><a href="#cb189-6" aria-hidden="true" tabindex="-1"></a>cpa2_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;CAP2&quot;</span>, <span class="st">&quot; (&quot;</span>, temp_adapt_cap2, <span class="st">&quot;%)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb189-7"><a href="#cb189-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-8"><a href="#cb189-8" aria-hidden="true" tabindex="-1"></a>swel_col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#2271B2&quot;</span>, <span class="st">&quot;#71B222&quot;</span>, <span class="st">&quot;#B22271&quot;</span>)</span>
<span id="cb189-9"><a href="#cb189-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-10"><a href="#cb189-10" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(temp_adapt_plot_data) <span class="sc">+</span></span>
<span id="cb189-11"><a href="#cb189-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">shape =</span> TREAT,</span>
<span id="cb189-12"><a href="#cb189-12" aria-hidden="true" tabindex="-1"></a>                           <span class="at">colour =</span> TREAT_T), <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb189-13"><a href="#cb189-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> swel_col) <span class="sc">+</span></span>
<span id="cb189-14"><a href="#cb189-14" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_text(data = temp_adapt_plot_data, aes(x = CAP1, y = CAP2, #UNCOMMENT to add sample labels</span></span>
<span id="cb189-15"><a href="#cb189-15" aria-hidden="true" tabindex="-1"></a><span class="co">#                              label = SampleID), size = 3) + </span></span>
<span id="cb189-16"><a href="#cb189-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">xend =</span> CAP1, <span class="at">yend =</span> CAP2),</span>
<span id="cb189-17"><a href="#cb189-17" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> temp_adapt_bioplot_data_md, <span class="at">linetype =</span> <span class="st">&quot;solid&quot;</span>,</span>
<span id="cb189-18"><a href="#cb189-18" aria-hidden="true" tabindex="-1"></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">&quot;cm&quot;</span>)), <span class="at">size =</span> <span class="fl">0.4</span>,</span>
<span id="cb189-19"><a href="#cb189-19" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">&quot;#191919&quot;</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb189-20"><a href="#cb189-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> temp_adapt_bioplot_data_md, </span>
<span id="cb189-21"><a href="#cb189-21" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">label =</span> parameters), <span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb189-22"><a href="#cb189-22" aria-hidden="true" tabindex="-1"></a>            <span class="at">nudge_x =</span> <span class="fl">0.1</span>, <span class="at">nudge_y =</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb189-23"><a href="#cb189-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># uSe to include ASV vectors</span></span>
<span id="cb189-24"><a href="#cb189-24" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_segment(aes(x = 0, y = 0, xend = CAP1, yend = CAP2),</span></span>
<span id="cb189-25"><a href="#cb189-25" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             data = temp_adapt_bioplot_data_asv, linetype = &quot;solid&quot;,</span></span>
<span id="cb189-26"><a href="#cb189-26" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             arrow = arrow(length = unit(0.3, &quot;cm&quot;)), size = 0.2,</span></span>
<span id="cb189-27"><a href="#cb189-27" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             color = &quot;#676767&quot;) +</span></span>
<span id="cb189-28"><a href="#cb189-28" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text(data = temp_adapt_bioplot_data_asv, </span></span>
<span id="cb189-29"><a href="#cb189-29" aria-hidden="true" tabindex="-1"></a>  <span class="co">#          aes(x = CAP1, y = CAP2, label = parameters), size = 2.5,</span></span>
<span id="cb189-30"><a href="#cb189-30" aria-hidden="true" tabindex="-1"></a>  <span class="co">#          nudge_x = 0.05, nudge_y = 0.05) +</span></span>
<span id="cb189-31"><a href="#cb189-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span> </span>
<span id="cb189-32"><a href="#cb189-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Capscale Analysis&quot;</span>,</span>
<span id="cb189-33"><a href="#cb189-33" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">&quot;Temperature Adaptation&quot;</span>,</span>
<span id="cb189-34"><a href="#cb189-34" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> cpa1_lab, </span>
<span id="cb189-35"><a href="#cb189-35" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> cpa2_lab)</span>
<span id="cb189-36"><a href="#cb189-36" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot <span class="ot">&lt;-</span> temp_adapt_plot <span class="sc">+</span> <span class="fu">coord_fixed</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb189-37"><a href="#cb189-37" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot</span>
<span id="cb189-38"><a href="#cb189-38" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">&quot;files/metadata/figures/its18_temp_adapt_capscale.png&quot;</span>,</span>
<span id="cb189-39"><a href="#cb189-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">16</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">units =</span> <span class="st">&#39;cm&#39;</span>, <span class="at">res =</span> <span class="dv">600</span>, <span class="at">bg =</span> <span class="st">&quot;white&quot;</span>)</span>
<span id="cb189-40"><a href="#cb189-40" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot</span>
<span id="cb189-41"><a href="#cb189-41" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb189-42"><a href="#cb189-42" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">&quot;files/metadata/figures/its18_temp_adapt_capscale.pdf&quot;</span>,</span>
<span id="cb189-43"><a href="#cb189-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">5</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb189-44"><a href="#cb189-44" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot</span>
<span id="cb189-45"><a href="#cb189-45" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
</div>
</details>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="capscale-plots-1">Capscale Plots</h2>
<div class="l-body-outset">
<div class="panelset">
<section id="edaphic-properties-7" class="panel">
<h4>Edaphic properties</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="metadata_files/figure-html5/unnamed-chunk-252-1.png" width="624" /></p>
</div>
</section>
<section id="soil-functional-response-5" class="panel">
<h4>Soil Functional Response</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="metadata_files/figure-html5/unnamed-chunk-253-1.png" width="624" /></p>
</div>
</section>
<section id="temperature-adaptation-7" class="panel">
<h4>Temperature Adaptation</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="metadata_files/figure-html5/unnamed-chunk-254-1.png" width="624" /></p>
</div>
</section>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 class="appendix" id="source-code">Source Code</h2>
<p>You can find the source code for this page by <a href="https://github.com/sweltr/high-temp/blob/master/metadata.Rmd">clicking this link</a>.</p>
<h2 class="appendix" id="data-availability">Data Availability</h2>
<p>Raw fastq files available on figshare at XXXXXXXX. Trimmed fastq files (primers removed) available through the ENA under project accession number <a href="">XXXXXXXX</a>. Output files from this workflow available on figshare at <a href="">XXXXXXXX.</a>.</p>
<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-oksanen2013community" class="csl-entry" role="doc-biblioentry">
Oksanen, Jari, F. Guillaume Blanchet, Roeland Kindt, Pierre Legendre, Peter R. Minchin, R. B. OHara, Gavin L. Simpson, Peter Slymos, M. Henry H. Stevens, and Helene Wagner. 2012. <span>Vegan: Community Ecology Package.</span> <em>R Package Version</em> 2 (0). <a href="https://cran.r-project.org/web/packages/vegan/index.html">https://cran.r-project.org/web/packages/vegan/index.html</a>.
</div>
</div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/sweltr/high-temp/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/sweltr/high-temp/">https://github.com/sweltr/high-temp/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<div class="distill-site-nav distill-site-footer">
<p> Copyright 2021 The SWELTR Project.</p>
<p>Site constructed using <a href="https://rstudio.github.io/distill/">Distill for R Markdown</a>.</p>
</div>
<!--/radix_placeholder_navigation_after_body-->


</body>

</html>
