<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-0.9.598">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="description" content="Workflow to assess the relationship between environmental parameters and microbial communities.">
<title>SWELTR - 8. Metadata Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./data-availability.html" rel="next">
<link href="./da.html" rel="prev">
<link href="./assets/favicon.ico" rel="icon">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script><style>
    .quarto-title-block .quarto-title-banner {
      background: #113858;
    }
    </style>
<script src="site_libs/font-awesome-5.13.0/js/script.js"></script><script src="site_libs/core-js-2.5.3/shim.min.js"></script><script src="site_libs/react-17.0.0/react.min.js"></script><script src="site_libs/react-17.0.0/react-dom.min.js"></script><script src="site_libs/reactwidget-1.0.0/react-tools.js"></script><script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><script src="site_libs/reactable-binding-0.2.3.9000/reactable.js"></script><link rel="stylesheet" href="assets/css/styles.css">
</head>
<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg navbar-dark "><div class="navbar-container container-fluid">
      <a class="navbar-brand" href="./index.html">
    <img src="./assets/icon.png" alt=""></a>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item compact">
    <a class="nav-link" href="./index.html"><i class="bi bi-home" role="img">
</i> 
 </a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-a.-processing" role="button" data-bs-toggle="dropdown" aria-expanded="false">A. Processing</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-a.-processing">
<li class="dropdown-header">Workflows</li>
        <li><hr class="dropdown-divider"></li>
        <li>
    <a class="dropdown-item" href="./dada2.html">
 <span class="dropdown-text">1. DADA2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./data-prep.html">
 <span class="dropdown-text">2. Data Set Prep</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./filtering.html">
 <span class="dropdown-text">3. Filtering</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-b.-diversity" role="button" data-bs-toggle="dropdown" aria-expanded="false">B. Diversity</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-b.-diversity">
<li class="dropdown-header">Workflows</li>
        <li><hr class="dropdown-divider"></li>
        <li>
    <a class="dropdown-item" href="./taxa.html">
 <span class="dropdown-text">4. Taxonomic</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./alpha.html">
 <span class="dropdown-text">5. Alpha Diversity</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./beta.html">
 <span class="dropdown-text">6. Beta Diversity</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./da.html">
 <span class="dropdown-text">7. DA ASVs &amp; Taxa</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./metadata.html">
 <span class="dropdown-text">8. Metadata</span></a>
  </li>  
    </ul>
</li>
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item">
    <a class="nav-link active" href="./data-availability.html" aria-current="page">DATA</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./pub.html">PAPER</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./som.html">SOM</a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/sweltr/high-temp/"><i class="bi bi-github" role="img" aria-label="Agua Salud GitHub">
</i> 
 </a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="./about.html"><i class="bi bi-question" role="img">
</i> 
 </a>
  </li>  
</ul>
<div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }"><div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">8. Metadata Analysis</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title d-none d-lg-block">8. Metadata Analysis</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i></button></div></div>
                  <div>
        <div class="description">
          <p>Workflow to assess the relationship between environmental parameters and microbial communities.</p>
        </div>
      </div>
                </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto"><div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">A. Processing</a>
      <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./dada2.html" class="sidebar-item-text sidebar-link">1. DADA2 Workflow</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./data-prep.html" class="sidebar-item-text sidebar-link">2. Data Set Preparation</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./filtering.html" class="sidebar-item-text sidebar-link">3. Filtering</a>
  </div>
</li>
    </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">B. Diversity</a>
      <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./taxa.html" class="sidebar-item-text sidebar-link">4. Taxonomic Diversity</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./alpha.html" class="sidebar-item-text sidebar-link">5. Alpha Diversity Estimates</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./beta.html" class="sidebar-item-text sidebar-link">6. Beta Diversity &amp; Dispersion Estimates</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./da.html" class="sidebar-item-text sidebar-link">7. Differentially Abundant ASVs &amp; Taxa</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./metadata.html" class="sidebar-item-text sidebar-link active">8. Metadata Analysis</a>
  </div>
</li>
    </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">More</a>
      <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./data-availability.html" class="sidebar-item-text sidebar-link">Data Availability</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./som.html" class="sidebar-item-text sidebar-link">Electronic Supplementary Material</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./about.html" class="sidebar-item-text sidebar-link">Website Contruction</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pub.html" class="sidebar-item-text sidebar-link">Code for Publication Figures</a>
  </div>
</li>
    </ul>
</li>
    </ul>
</div>
</nav><!-- margin-sidebar --><div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li>
<a href="#synopsis" id="toc-synopsis" class="nav-link active" data-scroll-target="#synopsis">Synopsis</a>
  <ul class="collapse">
<li><a href="#workflow-input" id="toc-workflow-input" class="nav-link" data-scroll-target="#workflow-input">Workflow Input</a></li>
  <li><a href="#metadata" id="toc-metadata" class="nav-link" data-scroll-target="#metadata">Metadata</a></li>
  </ul>
</li>
  <li>
<a href="#multivariate-summary" id="toc-multivariate-summary" class="nav-link" data-scroll-target="#multivariate-summary">Multivariate Summary</a>
  <ul class="collapse">
<li><a href="#s-rrna" id="toc-s-rrna" class="nav-link" data-scroll-target="#s-rrna">16S rRNA</a></li>
  <li><a href="#its" id="toc-its" class="nav-link" data-scroll-target="#its">ITS</a></li>
  </ul>
</li>
  <li>
<a href="#s-rrna-1" id="toc-s-rrna-1" class="nav-link" data-scroll-target="#s-rrna-1">16s rRNA</a>
  <ul class="collapse">
<li><a href="#data-formating" id="toc-data-formating" class="nav-link" data-scroll-target="#data-formating">Data Formating</a></li>
  <li><a href="#choose-data-set" id="toc-choose-data-set" class="nav-link" data-scroll-target="#choose-data-set">Choose Data Set</a></li>
  <li><a href="#normality-tests" id="toc-normality-tests" class="nav-link" data-scroll-target="#normality-tests">Normality Tests</a></li>
  <li><a href="#normalize-parameters" id="toc-normalize-parameters" class="nav-link" data-scroll-target="#normalize-parameters">Normalize Parameters</a></li>
  <li><a href="#normalized-metadata" id="toc-normalized-metadata" class="nav-link" data-scroll-target="#normalized-metadata">Normalized Metadata</a></li>
  <li><a href="#autocorrelation-tests" id="toc-autocorrelation-tests" class="nav-link" data-scroll-target="#autocorrelation-tests">Autocorrelation Tests</a></li>
  <li><a href="#autocorrelation-plots" id="toc-autocorrelation-plots" class="nav-link" data-scroll-target="#autocorrelation-plots">Autocorrelation Plots</a></li>
  <li><a href="#dissimilarity-correlation-tests" id="toc-dissimilarity-correlation-tests" class="nav-link" data-scroll-target="#dissimilarity-correlation-tests">Dissimilarity Correlation Tests</a></li>
  <li><a href="#best-subset-of-variables" id="toc-best-subset-of-variables" class="nav-link" data-scroll-target="#best-subset-of-variables">Best Subset of Variables</a></li>
  <li><a href="#distance-based-redundancy" id="toc-distance-based-redundancy" class="nav-link" data-scroll-target="#distance-based-redundancy">Distance-based Redundancy</a></li>
  <li><a href="#capscale-plots" id="toc-capscale-plots" class="nav-link" data-scroll-target="#capscale-plots">Capscale Plots</a></li>
  </ul>
</li>
  <li>
<a href="#its-1" id="toc-its-1" class="nav-link" data-scroll-target="#its-1">ITS</a>
  <ul class="collapse">
<li><a href="#data-formating-1" id="toc-data-formating-1" class="nav-link" data-scroll-target="#data-formating-1">Data Formating</a></li>
  <li><a href="#choose-data-set-1" id="toc-choose-data-set-1" class="nav-link" data-scroll-target="#choose-data-set-1">Choose Data Set</a></li>
  <li><a href="#normality-tests-1" id="toc-normality-tests-1" class="nav-link" data-scroll-target="#normality-tests-1">Normality Tests</a></li>
  <li><a href="#normalize-parameters-1" id="toc-normalize-parameters-1" class="nav-link" data-scroll-target="#normalize-parameters-1">Normalize Parameters</a></li>
  <li><a href="#normalized-metadata-1" id="toc-normalized-metadata-1" class="nav-link" data-scroll-target="#normalized-metadata-1">Normalized Metadata</a></li>
  <li><a href="#autocorrelation-tests-1" id="toc-autocorrelation-tests-1" class="nav-link" data-scroll-target="#autocorrelation-tests-1">Autocorrelation Tests</a></li>
  <li><a href="#autocorrelation-plots-1" id="toc-autocorrelation-plots-1" class="nav-link" data-scroll-target="#autocorrelation-plots-1">Autocorrelation Plots</a></li>
  <li><a href="#dissimilarity-correlation-tests-1" id="toc-dissimilarity-correlation-tests-1" class="nav-link" data-scroll-target="#dissimilarity-correlation-tests-1">Dissimilarity Correlation Tests</a></li>
  <li><a href="#best-subset-of-variables-1" id="toc-best-subset-of-variables-1" class="nav-link" data-scroll-target="#best-subset-of-variables-1">Best Subset of Variables</a></li>
  <li><a href="#distance-based-redundancy-1" id="toc-distance-based-redundancy-1" class="nav-link" data-scroll-target="#distance-based-redundancy-1">Distance-based Redundancy</a></li>
  <li><a href="#capscale-plots-1" id="toc-capscale-plots-1" class="nav-link" data-scroll-target="#capscale-plots-1">Capscale Plots</a></li>
  </ul>
</li>
  <li><a href="#workflow-output" id="toc-workflow-output" class="nav-link" data-scroll-target="#workflow-output">Workflow Output</a></li>
  </ul><div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/sweltr/high-temp/edit/main/metadata.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/sweltr/high-temp/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block page-columns page-full" id="quarto-document-content"><details><summary><strong>Click here</strong> for setup information.
</summary><div class="cell">
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>eval <span class="op">=</span> <span class="cn">FALSE</span>, echo <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">119</span><span class="op">)</span>
<span class="co">#library(conflicted)</span>
<span class="co">#pacman::p_depends(vegan, local = TRUE)  </span>
<span class="co">#pacman::p_depends_reverse(vegan, local = TRUE)  </span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/utils/packageDescription.html">packageVersion</a></span><span class="op">(</span><span class="st">"phyloseq"</span><span class="op">)</span>
<span class="fu">pacman</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pacman/man/p_load.html">p_load</a></span><span class="op">(</span><span class="va">ape</span>, <span class="va">bestNormalize</span>, <span class="va">tidyverse</span>, <span class="va">gdata</span>, <span class="va">mctoolsr</span>, 
               <span class="va">ggpubr</span>, <span class="va">ggvegan</span>, <span class="va">ggfortify</span>, <span class="va">vegan</span>, <span class="va">reactable</span>, 
               <span class="va">downloadthis</span>, <span class="va">captioner</span>,
               install <span class="op">=</span> <span class="cn">FALSE</span>, update <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>scipen<span class="op">=</span><span class="fl">999</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html">opts_current</a></span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
  <span class="st">"cache"</span>,
  <span class="st">"cache.path"</span>,
  <span class="st">"cache.rebuild"</span>,
  <span class="st">"dependson"</span>,
  <span class="st">"autodep"</span>
<span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</details><div class="cell">

</div>
<div class="cell">

</div>
<section id="synopsis" class="level1"><h1>Synopsis</h1>
<p>In this workflow we compare the environmental metadata with the microbial community data. Part 1 involves the <a href="#s-rrna-1">16S rRNA</a> community data and Part 2 deals with the <a href="#its-1">ITS</a> community data. Each workflow contains the same major steps:</p>
<ol type="1">
<li>Metadata Normality Tests: Shapiro-Wilk Normality Test to test whether each matadata parameter is normally distributed.</li>
<li>Normalize Parameters: R package <a href="https://github.com/petersonR/bestNormalize"><code>bestNormalize</code></a> to find and execute the best normalizing transformation.</li>
<li>Split Metadata parameters into groups: a) Environmental and edaphic properties, b) Microbial functional responses, and c) Temperature adaptation properties.</li>
<li>Autocorrelation Tests: Test all possible pair-wise comparisons, on both normalized and non-normalized data sets, for each group.</li>
<li>Remove autocorrelated parameters from each group.</li>
<li>Dissimilarity Correlation Tests: Use Mantel Tests to see if any on the metadata groups are significantly correlated with the community data.</li>
<li>Best Subset of Variables: Determine which of the metadata parameters from each group are the most strongly correlated with the community data. For this we use the <code>bioenv</code> function from the <code>vegan</code> <span class="citation" data-cites="oksanen2013community">(<a href="#ref-oksanen2013community" role="doc-biblioref">Oksanen et al. 2012</a>)</span> package.</li>
<li>Distance-based Redundancy Analysis: Ordination analysis of samples and metadata vector overlays using <code>capscale</code>, also from the <code>vegan</code> package.</li>
</ol>
<section id="workflow-input" class="level2"><h2 class="anchored" data-anchor-id="workflow-input">Workflow Input</h2>
<p>Files needed to run this workflow can be downloaded from figshare. You will also need the metadata table, which can be accessed below.</p>
<iframe src="https://widgets.figshare.com/articles/16826779/embed?show_title=1" width="100%" height="251" allowfullscreen="" frameborder="0">
</iframe>
</section><section id="metadata" class="level2"><h2 class="anchored" data-anchor-id="metadata">Metadata</h2>
<p>In this section we provide access to the complete environmental metadata containing 61 parameters across all 15 samples.</p>
<p><br></p>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,U2FtcGxlX0lELFBMT1QsVFJFQVQsVFJFQVRfVCxQQUlSLFNFQVNPTixBVEFQLEFTVCxIMk8sTixQLEFsLENhLEZlLEssTWcsTW4sTmEsVEVCLEVDRUMscEgsbWluUE80LG1pbk5INCxtaW5OTzMsbWluVElOLE5INCxOTzMsUE80LERPQyxET04sRE9DTixtaWNDLG1pY04sbWljUCxtaWNDTixtaWNDUCxtaWNOUCxBR19hc2UsQkdfYXNlLEJQX2FzZSxDRV9hc2UsUF9hc2UsTl9hc2UsU19hc2UsWFlfYXNlLEFHX1ExMCxCR19RMTAsQlBfUTEwLENFX1ExMCxQX1ExMCxOX1ExMCxTX1ExMCxYWV9RMTAsQ08yLExQX2FzZSxMUF9RMTAsUFhfYXNlLFBYX1ExMCxDVUVjbixDVUVjcCxOVUUsUFVFLFRtaW4sU0ksZW56Q04sZW56Q1AsZW56TlAKUDAxX0QwMF8wMTBfVzNBLFAwMSxXLDMsQSxXLDI5LDI5LjM4LDAuMzk5OSwwLjQyNjcsMC4zNSwwLDMxLjQzLDAuMDEsMS43Miw5Ljk4LDAuMDQsMC4xNSw0My4zLDQzLjMsNi4wMSwxLjMzLDMuNjcsNDcuMTcsNTAuODMsOC4wMyw0LjY3LDkuNTIsMTE4Ljc0LDExLjY2LDExLjE1LDY5MC4xMSwxMzUuNDgsODcuOTgsNC42MSw4LjYxLDEuNzQsMC4yNiw0LjEsMy41LDEsMjAuNDMsNS42Myw2LjA4LDEuMTcsMS43NywxLjM4LDEuMzksMi4wMywxLjM3LDEuNDYsMS4zLDEuNzIsMTEuNzgsNTIuOCwxLjc2LDE0Ny4xMywxLjIyLDAuMzIsMC4yMSwwLjc3LDAuODgsMC4wNywxLjUxLDAuNDYxNCwwLjU5MTEsMS4yODEyClAwMV9EMDBfMDEwX1c4QSxQMDEsVyw4LEEsVywzNCwzMC42NiwwLjI2MSwwLjQyNjcsMC4zNSwwLDMxLjQzLDAuMDEsMS43Miw5Ljk4LDAuMDQsMC4xNSw0My4zLDQzLjMsNi4wMSwwLjEyLDEuNjUsMTEuMzEsMTIuOTYsNC4wOCwyLjY1LDguNTUsMzkuNyw0LjIsOS40NSw0NjksODQsNTUsNS41OCw4LjQ4LDEuNTIsMC4xNSw2Ljk3LDUuMzQsMi4wNywyNS42NSw0LjUyLDYuNywyLjUyLDIuMDEsMS43NSwxLjUxLDEuODcsMS4zOSwxLjU2LDEuNjUsMS44Niw0LjI4LDQwLjY5LDEuNzEsMjExLjE4LDEuMDcsMC4yMywwLjE4LDAuODcsMC45LDIuMTksMi4zLDAuNjQ1NiwwLjcxNjYsMS4xMQpQMDJfRDAwXzAxMF9DMEEsUDAyLEMsMCxBLFcsMjYsMjYuMiwwLjM3OTMsMC4zMzY3LDAuNDMsMC4wMSwzNi42NSwwLjAxLDAuNjksOC4xNiwwLjA0LDAuMTEsNDUuNiw0NS43LDYuMTgsMS40MiwzLjkyLDI0LjMzLDI4LjI1LDQuODEsMy4xMyw3LjU4LDc1Ljg2LDguMjIsOS40NCw0NDQuNzIsMTEwLjEzLDY1LjAzLDQuNDIsOC4xMywyLjM3LDAuMjQsNC4yMiwyLjU0LDEuMDksMTUuMDcsMy41NCwzLjgsMS4wMSwxLjUxLDEuMzEsMS4zNCwxLjg5LDEuMjIsMS41NCwxLjIzLDEuNTksNS4yNywzOC42MSwxLjgxLDk0LjEzLDEuNTgsMC4yMiwwLjIxLDAuODcsMC44OCwwLjE0LDEuNDQsMC41MDI0LDAuNjU1NCwxLjMwNDQKUDAzX0QwMF8wMTBfVzNCLFAwMyxXLDMsQixXLDI5LDI5LjI5LDAuNDAwNywwLjMyLDAuMzcsMC4wMSwyNy4xOSwwLjAxLDAuODUsMTYuNTEsMC4wNiwwLjEyLDQ0LjcsNDQuNyw1Ljg2LDIuODMsMy4zMyw0Mi41LDQ1LjgzLDMuMTEsMi41Nyw4LjY2LDYwLjU3LDYuNDgsMTAuMTIsMzY0LjYyLDY2LjQsNzAuNjgsNS4wNyw2LjcsMS4zMSwwLjMzLDQuMzcsMy4wNSwxLjI2LDE5LjIxLDMuNTgsNS43MiwxLjM2LDIuNDgsMS4zNiwxLjQyLDEuOTksMS4yNywxLjU0LDEuMjcsMS43Miw0LjQ2LDMyLjEsMiwxMTkuNzUsMS40MSwwLjI2LDAuMTgsMC44NCwwLjksLTAuMjksMS40OSwwLjU1NjUsMC42NDEyLDEuMTUyMgpQMDNfRDAwXzAxMF9XOEIsUDAzLFcsOCxCLFcsMzQsNDEuNzcsMC4yNDMsMC4zMiwwLjM3LDAuMDEsMjcuMTksMC4wMSwwLjg1LDE2LjUxLDAuMDYsMC4xMiw0NC43LDQ0LjcsNS44NiwwLjIsMS4yOCwyMi43MSwyMy45OSwyLjUsMS4wNiwyLjgsMzMuMiwyLjUsMTMuMjksMTE3LDIwLDExLDUuODUsMTAuNTQsMS44LDAuMDIsMy4wNCwyLjg5LDAuNjUsMTguMTEsMi45MSwxLjI4LDAuNzYsMS40OSwxLjU2LDEuNTUsMS45LDEuNTEsMS43NSwxLjM5LDEuODksNDAuNDcsMTMuOTQsMS43NywyMjUuNTYsMS4xNSwwLjI5LDAuMzMsMC44MSwwLjc3LDAuNjMsMS43MiwwLjUzMDIsMC40OTE4LDAuOTI3NwpQMDRfRDAwXzAxMF9DMEIsUDA0LEMsMCxCLFcsMjYsMjYuMjksMC4zODMyLDAuMzAzMywwLjMzLDAsMjkuMjMsMC4wMiwxLjE4LDE0LjU5LDAuMTIsMC4xNCw0NS4xLDQ1LjMsNS43MywxLjUsMy4xNywxOS45MiwyMy4wOCw1LjU3LDMuMjIsNy45NSw1OS41MSw2LjMyLDEwLjY3LDQ0MC42LDg2LjE5LDgxLjQ1LDUuMTQsNS4zOSwxLjA5LDAuMjIsNC41NywzLjAyLDEuMDksMjAuMjcsMy42NSw0LjkyLDEuMTEsMS41MSwxLjUsMS41OCwyLjMsMS4zNywxLjg1LDEuMzksMS43Niw0LjA2LDI5LjUxLDIuMTQsODEuMjUsMS4yNCwwLjI1LDAuMTksMC44NSwwLjksLTIuNjMsMS4yNSwwLjU1NSwwLjYxNzMsMS4xMTIzClAwNV9EMDBfMDEwX1czQyxQMDUsVywzLEMsVywyOSwyOC41MywwLjM1NDUsMC4zMzMzLDAuMzMsMCwyNi4xNCwwLDAuNjcsMTQuODMsMC4wNSwwLDQxLjYsNDEuNyw1Ljg3LDEuMTcsMi4xNywxMS4yNSwxMy40MiwyLjMyLDMuMzEsNy4zLDY0Ljk4LDcuMDMsMTEuOTMsMjg0LjUsNDguNzksNTIuNSw1LjU3LDguMzMsMS4zOSwwLjM0LDMuMDEsMi4zNCwwLjc2LDE1LjE5LDIuNjEsMy4yLDAuODYsMS40OCwxLjM1LDEuMzksMS44OCwxLjI4LDEuNTEsMS4zLDEuODIsNC42MiwyNC4xMSwxLjc5LDExMC43NSwxLjMzLDAuMjcsMC4xOCwwLjgzLDAuOSwwLjgxLDEuNzcsMC40ODc3LDAuNTU5NiwxLjE0NzMKUDA1X0QwMF8wMTBfVzhDLFAwNSxXLDgsQyxXLDM0LDM2LjI0LDAuMjMzLDAuMzMzMywwLjMzLDAsMjYuMTQsMCwwLjY3LDE0LjgzLDAuMDUsMCw0MS42LDQxLjcsNS44NywwLjExLDEuMTgsMzYuNjEsMzcuNzksMy45MSwyLjMsNC4yNSw0NSw0LjMsMTAuNDYsNDQwLDk3LDU2LDQuNTQsNy45MSwxLjc0LDAuMTUsOS4wOSw3LjI4LDEuNSw0NC4wNCw1LjU2LDUuNTcsMS43NSwxLjczLDEuNzIsMS40MiwxLjk1LDEuNDgsMS42OCwxLjYsMS44OCwxNS4zNywzMi42NSwxLjY3LDI0NS42NSwxLjIzLDAuMTksMC4yMywwLjksMC44NywwLjk0LDIuMDgsMC42OTMxLDAuNjQxMiwwLjkyNTEKUDA2X0QwMF8wMTBfQzBDLFAwNixDLDAsQyxXLDI2LDI2LjI2LDAuMzc3NCwwLjMzMzMsMC4zLDAsMzIuMDcsMCwwLjU3LDIxLjMyLDAuMDYsMC4xNCw1NC4xLDU0LjIsNS45MywwLjc1LDEuODMsMTkuNjcsMjEuNSwyLjkyLDIuMjcsNy4wNCw1Ni42OCw1LjQyLDExLjA5LDM5MC42Niw3Mi42MSw4NS43LDUuMjYsNC41LDAuODMsMC4yMSwzLjIyLDIuMzMsMC43NSwxNC43OSwyLjU3LDUuNDUsMC42NiwxLjgzLDEuMzEsMS40NCwyLjM2LDEuMjQsMS42LDEuNDcsMS42Myw0LjI4LDI3Ljc4LDEuODUsOTIuNSwxLjUzLDAuMjcsMC4xNiwwLjgzLDAuOTIsLTAuMTMsMS40MiwwLjQ2MjQsMC41NTU2LDEuMjAxNwpQMDdfRDAwXzAxMF9XM0QsUDA3LFcsMyxELFcsMjksMjguNzgsMC4zODQ5LDAuNDM2NywwLjM1LDAsMjguMTMsMC4wMSwxLjYsMTguODcsMC4wNywwLjEsNDguNyw0OC44LDYuMjEsMSwyLjQyLDIwLjY3LDIzLjA4LDQuMjgsNS4wNyw5LjE1LDczLjk4LDguNDcsOS45MywzOTYuNDIsNzMuNzMsOTMuMTMsNC43Miw0LjY2LDAuOTgsMC4zMiw0LjE0LDMuNDUsMS4wNywyMy4yNiwzLjcsNS44NiwxLjExLDEuNzIsMS40NSwxLjQ2LDEuNzYsMS4yOCwxLjM2LDEuMywxLjYzLDYuMTksNDIuNCwxLjk2LDk3LDEuMTUsMC4yNywwLjE1LDAuODMsMC45MiwwLjY1LDEuNjEsMC40OTQxLDAuNTc2MiwxLjE2NjIKUDA3X0QwMF8wMTBfVzhELFAwNyxXLDgsRCxXLDM0LDI5LjksMC4yMDQsMC40MzY3LDAuMzUsMCwyOC4xMywwLjAxLDEuNiwxOC44NywwLjA3LDAuMSw0OC43LDQ4LjgsNi4yMSwwLjAyLDIuNiwxMi4zMSwxNC45MSw4LjE5LDYuMzksNS42Miw0NC4zLDMuNywxMS45OCw2NzYsMTQ3LDk0LDQuNiw3LjE4LDEuNTYsMC4zOSw4LjYxLDcuMTEsMS41MywzNS4xLDguMzQsMTAuNjgsMS40MiwxLjMxLDEuMzksMS40MiwxLjk2LDEuMzMsMS41NCwxLjUxLDEuOTEsOC44Nyw1Ny4xMywxLjQsMjk0LjM5LDEuMjcsMC4yNiwwLjE3LDAuODQsMC45MSwxLjQ4LDEuODEsMC41OTMzLDAuNjYyOCwxLjExNzMKUDA4X0QwMF8wMTBfQzBELFAwOCxDLDAsRCxXLDI2LDI1LjgsMC40MDExLDAuMjQzMywwLjI3LDAsMjUuNDksMCwxLjg2LDE0LjkzLDAuMDIsMC4wOCw0Mi4zLDQyLjQsNi4wMSwzLjY3LDQuMzMsMTA2LjQyLDExMC43NSwyLjA0LDMuNzEsNS45Nyw0NC44OSw1LjIyLDExLjU3LDI3Ny4zNyw1Ni41NSwzMy40LDQuNzIsMTEuNTUsMi40NSwwLjE1LDMuNTUsMS43NywwLjY4LDEzLjAyLDIuOTUsMy41NiwwLjcxLDIuMDgsMS43NiwxLjc0LDIsMS40OCwxLjY5LDEuNDgsMS44MywyLjA4LDIxLjk2LDIuMDcsOTIuMzgsMy41OCwwLjI0LDAuMjIsMC44NSwwLjg4LDAsMS41NCwwLjUwNTgsMC42MDM3LDEuMTkzNQpQMDlfRDAwXzAxMF9XM0UsUDA5LFcsMyxFLFcsMjksMjguOCwwLjM2NDYsMC4yODMzLDAuMzYsMC4wMiwzOS43OSwwLjA0LDEuMDksMjAuNjQsMC4wOCwwLjE5LDYxLjcsNjEuOCw2LjEsMi41OCwzLjUsMTEuNTgsMTUuMDgsMi4wMywyLjksOC4yOCw0My44NCw1LjU2LDEyLjIxLDM5Mi41Niw2Mi4zNyw2Ni42NSw1Ljc4LDcuMzQsMS4xOSwwLjIsMy45MiwyLjI2LDAuODYsMTYuOSwyLjgzLDUuMjYsMS4wMSwxLjY3LDEuMzYsMS41NiwxLjgxLDEuMzUsMS40OCwxLjMyLDEuODQsOS4zOSwzMC4zNywxLjksMTIyLjYzLDEuMzksMC4yNiwwLjIxLDAuODQsMC44OCwtMC4yNCwxLjUxLDAuNTExMiwwLjYwNjQsMS4xODYyClAwOV9EMDBfMDEwX1c4RSxQMDksVyw4LEUsVywzNCwzMS42MSwwLjM1OSwwLjI4MzMsMC4zNiwwLjAyLDM5Ljc5LDAuMDQsMS4wOSwyMC42NCwwLjA4LDAuMTksNjEuNyw2MS44LDYuMSwwLDAuOTIsNy44NCw4Ljc2LDYuNSwyLjc2LDYuMDgsMzAuOCwxLjYsMTkuMjcsNDE2LDkxLDYzLDQuNTcsNi42LDEuNDQsMC4zNywxMC40OCw2LjQ2LDIuMjEsMzUuNzUsOS40MywxMS43OCwxLjg0LDEuOTMsMS41MywxLjQ0LDIuMTEsMS4zOCwxLjQ2LDEuNSwxLjc2LDEzLjYsNTMuNzksMS43OSwzMzkuNiwxLjI3LDAuMjQsMC4xOSwwLjg1LDAuODksMC45NiwxLjgyLDAuNjUxNSwwLjcyMTgsMS4xMDc5ClAxMF9EMDBfMDEwX0MwRSxQMTAsQywwLEUsVywyNiwyNi4wOSwwLjM3MDIsMC4yNiwwLjMyLDAuMDIsMzcuMDYsMC4wMSwxLjM4LDIxLjQzLDAuMDcsMC4xMyw2MCw2MC4xLDYuMjgsNC4wOCw0Ljc1LDM3LjMzLDQyLjA4LDMuNDQsNC41Myw5LjQ1LDYzLjc5LDQuOTEsMTQuNjcsMzM0LjkzLDcyLjY0LDY0LjIzLDQuNDksNS41NCwxLjE3LDAuMjgsNC42MiwyLjExLDAuODYsMTQuOCwzLjM3LDQuNTQsMC43MiwxLjU0LDEuMzcsMS40MSwxLjgxLDEuMjcsMS40NywxLjMzLDEuNjUsNS4yMSwzMi4yMiwxLjgsMTExLjg4LDEuMTQsMC4yMiwwLjE2LDAuODcsMC45MiwtMi40NiwxLjI5LDAuNTIzMSwwLjY2MDcsMS4yNjMK').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'metadata.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
</div>
<p><small><strong>(common) Table 1 | Metadata values for each sample.</strong></small></p>
<div class="cell">
<div class="cell-output-display">

<div id="htmlwidget-dc29ecf1654ad0043dad" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-dc29ecf1654ad0043dad">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Sample_ID":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"PLOT":["P01","P01","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09","P10"],"TREAT":["W","W","C","W","W","C","W","W","C","W","W","C","W","W","C"],"TREAT_T":[3,8,0,3,8,0,3,8,0,3,8,0,3,8,0],"PAIR":["A","A","A","B","B","B","C","C","C","D","D","D","E","E","E"],"SEASON":["W","W","W","W","W","W","W","W","W","W","W","W","W","W","W"],"ATAP":[29,34,26,29,34,26,29,34,26,29,34,26,29,34,26],"AST":[29.38,30.66,26.2,29.29,41.77,26.29,28.53,36.24,26.26,28.78,29.9,25.8,28.8,31.61,26.09],"H2O":[0.3999,0.261,0.3793,0.4007,0.243,0.3832,0.3545,0.233,0.3774,0.3849,0.204,0.4011,0.3646,0.359,0.3702],"N":[0.4267,0.4267,0.3367,0.32,0.32,0.3033,0.3333,0.3333,0.3333,0.4367,0.4367,0.2433,0.2833,0.2833,0.26],"P":[0.35,0.35,0.43,0.37,0.37,0.33,0.33,0.33,0.3,0.35,0.35,0.27,0.36,0.36,0.32],"Al":[0,0,0.01,0.01,0.01,0,0,0,0,0,0,0,0.02,0.02,0.02],"Ca":[31.43,31.43,36.65,27.19,27.19,29.23,26.14,26.14,32.07,28.13,28.13,25.49,39.79,39.79,37.06],"Fe":[0.01,0.01,0.01,0.01,0.01,0.02,0,0,0,0.01,0.01,0,0.04,0.04,0.01],"K":[1.72,1.72,0.69,0.85,0.85,1.18,0.67,0.67,0.57,1.6,1.6,1.86,1.09,1.09,1.38],"Mg":[9.98,9.98,8.16,16.51,16.51,14.59,14.83,14.83,21.32,18.87,18.87,14.93,20.64,20.64,21.43],"Mn":[0.04,0.04,0.04,0.06,0.06,0.12,0.05,0.05,0.06,0.07,0.07,0.02,0.08,0.08,0.07],"Na":[0.15,0.15,0.11,0.12,0.12,0.14,0,0,0.14,0.1,0.1,0.08,0.19,0.19,0.13],"TEB":[43.3,43.3,45.6,44.7,44.7,45.1,41.6,41.6,54.1,48.7,48.7,42.3,61.7,61.7,60],"ECEC":[43.3,43.3,45.7,44.7,44.7,45.3,41.7,41.7,54.2,48.8,48.8,42.4,61.8,61.8,60.1],"pH":[6.01,6.01,6.18,5.86,5.86,5.73,5.87,5.87,5.93,6.21,6.21,6.01,6.1,6.1,6.28],"minPO4":[1.33,0.12,1.42,2.83,0.2,1.5,1.17,0.11,0.75,1,0.02,3.67,2.58,0,4.08],"minNH4":[3.67,1.65,3.92,3.33,1.28,3.17,2.17,1.18,1.83,2.42,2.6,4.33,3.5,0.92,4.75],"minNO3":[47.17,11.31,24.33,42.5,22.71,19.92,11.25,36.61,19.67,20.67,12.31,106.42,11.58,7.84,37.33],"minTIN":[50.83,12.96,28.25,45.83,23.99,23.08,13.42,37.79,21.5,23.08,14.91,110.75,15.08,8.76,42.08],"NH4":[8.03,4.08,4.81,3.11,2.5,5.57,2.32,3.91,2.92,4.28,8.19,2.04,2.03,6.5,3.44],"NO3":[4.67,2.65,3.13,2.57,1.06,3.22,3.31,2.3,2.27,5.07,6.39,3.71,2.9,2.76,4.53],"PO4":[9.52,8.55,7.58,8.66,2.8,7.95,7.3,4.25,7.04,9.15,5.62,5.97,8.28,6.08,9.45],"DOC":[118.74,39.7,75.86,60.57,33.2,59.51,64.98,45,56.68,73.98,44.3,44.89,43.84,30.8,63.79],"DON":[11.66,4.2,8.22,6.48,2.5,6.32,7.03,4.3,5.42,8.47,3.7,5.22,5.56,1.6,4.91],"DOCN":[11.15,9.45,9.44,10.12,13.29,10.67,11.93,10.46,11.09,9.93,11.98,11.57,12.21,19.27,14.67],"micC":[690.11,469,444.72,364.62,117,440.6,284.5,440,390.66,396.42,676,277.37,392.56,416,334.93],"micN":[135.48,84,110.13,66.4,20,86.19,48.79,97,72.61,73.73,147,56.55,62.37,91,72.64],"micP":[87.98,55,65.03,70.68,11,81.45,52.5,56,85.7,93.13,94,33.4,66.65,63,64.23],"micCN":[4.61,5.58,4.42,5.07,5.85,5.14,5.57,4.54,5.26,4.72,4.6,4.72,5.78,4.57,4.49],"micCP":[8.61,8.48,8.13,6.7,10.54,5.39,8.33,7.91,4.5,4.66,7.18,11.55,7.34,6.6,5.54],"micNP":[1.74,1.52,2.37,1.31,1.8,1.09,1.39,1.74,0.83,0.98,1.56,2.45,1.19,1.44,1.17],"AG_ase":[0.26,0.15,0.24,0.33,0.02,0.22,0.34,0.15,0.21,0.32,0.39,0.15,0.2,0.37,0.28],"BG_ase":[4.1,6.97,4.22,4.37,3.04,4.57,3.01,9.09,3.22,4.14,8.61,3.55,3.92,10.48,4.62],"BP_ase":[3.5,5.34,2.54,3.05,2.89,3.02,2.34,7.28,2.33,3.45,7.11,1.77,2.26,6.46,2.11],"CE_ase":[1,2.07,1.09,1.26,0.65,1.09,0.76,1.5,0.75,1.07,1.53,0.68,0.86,2.21,0.86],"P_ase":[20.43,25.65,15.07,19.21,18.11,20.27,15.19,44.04,14.79,23.26,35.1,13.02,16.9,35.75,14.8],"N_ase":[5.63,4.52,3.54,3.58,2.91,3.65,2.61,5.56,2.57,3.7,8.34,2.95,2.83,9.43,3.37],"S_ase":[6.08,6.7,3.8,5.72,1.28,4.92,3.2,5.57,5.45,5.86,10.68,3.56,5.26,11.78,4.54],"XY_ase":[1.17,2.52,1.01,1.36,0.76,1.11,0.86,1.75,0.66,1.11,1.42,0.71,1.01,1.84,0.72],"AG_Q10":[1.77,2.01,1.51,2.48,1.49,1.51,1.48,1.73,1.83,1.72,1.31,2.08,1.67,1.93,1.54],"BG_Q10":[1.38,1.75,1.31,1.36,1.56,1.5,1.35,1.72,1.31,1.45,1.39,1.76,1.36,1.53,1.37],"BP_Q10":[1.39,1.51,1.34,1.42,1.55,1.58,1.39,1.42,1.44,1.46,1.42,1.74,1.56,1.44,1.41],"CE_Q10":[2.03,1.87,1.89,1.99,1.9,2.3,1.88,1.95,2.36,1.76,1.96,2,1.81,2.11,1.81],"P_Q10":[1.37,1.39,1.22,1.27,1.51,1.37,1.28,1.48,1.24,1.28,1.33,1.48,1.35,1.38,1.27],"N_Q10":[1.46,1.56,1.54,1.54,1.75,1.85,1.51,1.68,1.6,1.36,1.54,1.69,1.48,1.46,1.47],"S_Q10":[1.3,1.65,1.23,1.27,1.39,1.39,1.3,1.6,1.47,1.3,1.51,1.48,1.32,1.5,1.33],"XY_Q10":[1.72,1.86,1.59,1.72,1.89,1.76,1.82,1.88,1.63,1.63,1.91,1.83,1.84,1.76,1.65],"CO2":[11.78,4.28,5.27,4.46,40.47,4.06,4.62,15.37,4.28,6.19,8.87,2.08,9.39,13.6,5.21],"LP_ase":[52.8,40.69,38.61,32.1,13.94,29.51,24.11,32.65,27.78,42.4,57.13,21.96,30.37,53.79,32.22],"LP_Q10":[1.76,1.71,1.81,2,1.77,2.14,1.79,1.67,1.85,1.96,1.4,2.07,1.9,1.79,1.8],"PX_ase":[147.13,211.18,94.13,119.75,225.56,81.25,110.75,245.65,92.5,97,294.39,92.38,122.63,339.6,111.88],"PX_Q10":[1.22,1.07,1.58,1.41,1.15,1.24,1.33,1.23,1.53,1.15,1.27,3.58,1.39,1.27,1.14],"CUEcn":[0.32,0.23,0.22,0.26,0.29,0.25,0.27,0.19,0.27,0.27,0.26,0.24,0.26,0.24,0.22],"CUEcp":[0.21,0.18,0.21,0.18,0.33,0.19,0.18,0.23,0.16,0.15,0.17,0.22,0.21,0.19,0.16],"NUE":[0.77,0.87,0.87,0.84,0.81,0.85,0.83,0.9,0.83,0.83,0.84,0.85,0.84,0.85,0.87],"PUE":[0.88,0.9,0.88,0.9,0.77,0.9,0.9,0.87,0.92,0.92,0.91,0.88,0.88,0.89,0.92],"Tmin":[0.07,2.19,0.14,-0.29,0.63,-2.63,0.81,0.94,-0.13,0.65,1.48,0,-0.24,0.96,-2.46],"SI":[1.51,2.3,1.44,1.49,1.72,1.25,1.77,2.08,1.42,1.61,1.81,1.54,1.51,1.82,1.29],"enzCN":[0.4614,0.6456,0.5024,0.5565,0.5302,0.555,0.4877,0.6931,0.4624,0.4941,0.5933,0.5058,0.5112,0.6515,0.5231],"enzCP":[0.5911,0.7166,0.6554,0.6412,0.4918,0.6173,0.5596,0.6412,0.5556,0.5762,0.6628,0.6037,0.6064,0.7218,0.6607],"enzNP":[1.2812,1.11,1.3044,1.1522,0.9277,1.1123,1.1473,0.9251,1.2017,1.1662,1.1173,1.1935,1.1862,1.1079,1.263]},"columns":[{"accessor":"Sample_ID","name":"Sample ID","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01_D00_010_W3A","P01_D00_010_W8A","P02_D00_010_C0A","P03_D00_010_W3B","P03_D00_010_W8B","P04_D00_010_C0B","P05_D00_010_W3C","P05_D00_010_W8C","P06_D00_010_C0C","P07_D00_010_W3D","P07_D00_010_W8D","P08_D00_010_C0D","P09_D00_010_W3E","P09_D00_010_W8E","P10_D00_010_C0E"],"header":"Sample ID","align":"left","headerStyle":{"borderRight":"1px solid #eee","fontSize":"0.9em"},"footerStyle":{"fontWeight":"bold"},"minWidth":150,"sticky":"left","style":{"borderRight":"1px solid #eee"}},{"accessor":"PLOT","name":"PLOT","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P01","P01","P02","P03","P03","P04","P05","P05","P06","P07","P07","P08","P09","P09","P10"],"header":"PLOT","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"TREAT","name":"TREAT","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["W","W","C","W","W","C","W","W","C","W","W","C","W","W","C"],"header":"TREAT","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"TREAT_T","name":"TREAT_T","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["3","8","0","3","8","0","3","8","0","3","8","0","3","8","0"],"header":"TREAT T","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"PAIR","name":"PAIR","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["A","A","A","B","B","B","C","C","C","D","D","D","E","E","E"],"header":"PAIR","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"SEASON","name":"SEASON","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["W","W","W","W","W","W","W","W","W","W","W","W","W","W","W"],"header":"SEASON","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"ATAP","name":"ATAP","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["29","34","26","29","34","26","29","34","26","29","34","26","29","34","26"],"header":"ATAP","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"AST","name":"AST","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["29.38","30.66","26.2","29.29","41.77","26.29","28.53","36.24","26.26","28.78","29.9","25.8","28.8","31.61","26.09"],"header":"AST","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"H2O","name":"H2O","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.3999","0.261","0.3793","0.4007","0.243","0.3832","0.3545","0.233","0.3774","0.3849","0.204","0.4011","0.3646","0.359","0.3702"],"header":"H2O","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"N","name":"N","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.4267","0.4267","0.3367","0.32","0.32","0.3033","0.3333","0.3333","0.3333","0.4367","0.4367","0.2433","0.2833","0.2833","0.26"],"header":"N","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"P","name":"P","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.35","0.35","0.43","0.37","0.37","0.33","0.33","0.33","0.3","0.35","0.35","0.27","0.36","0.36","0.32"],"header":"P","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Al","name":"Al","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.0","0.0","0.01","0.01","0.01","0.0","0.0","0.0","0.0","0.0","0.0","0.0","0.02","0.02","0.02"],"header":"Al","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Ca","name":"Ca","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["31.43","31.43","36.65","27.19","27.19","29.23","26.14","26.14","32.07","28.13","28.13","25.49","39.79","39.79","37.06"],"header":"Ca","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Fe","name":"Fe","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.01","0.01","0.01","0.01","0.01","0.02","0.0","0.0","0.0","0.01","0.01","0.0","0.04","0.04","0.01"],"header":"Fe","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"K","name":"K","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.72","1.72","0.69","0.85","0.85","1.18","0.67","0.67","0.57","1.6","1.6","1.86","1.09","1.09","1.38"],"header":"K","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Mg","name":"Mg","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["9.98","9.98","8.16","16.51","16.51","14.59","14.83","14.83","21.32","18.87","18.87","14.93","20.64","20.64","21.43"],"header":"Mg","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Mn","name":"Mn","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.04","0.04","0.04","0.06","0.06","0.12","0.05","0.05","0.06","0.07","0.07","0.02","0.08","0.08","0.07"],"header":"Mn","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Na","name":"Na","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.15","0.15","0.11","0.12","0.12","0.14","0.0","0.0","0.14","0.1","0.1","0.08","0.19","0.19","0.13"],"header":"Na","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"TEB","name":"TEB","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["43.3","43.3","45.6","44.7","44.7","45.1","41.6","41.6","54.1","48.7","48.7","42.3","61.7","61.7","60.0"],"header":"TEB","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"ECEC","name":"ECEC","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["43.3","43.3","45.7","44.7","44.7","45.3","41.7","41.7","54.2","48.8","48.8","42.4","61.8","61.8","60.1"],"header":"ECEC","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"pH","name":"pH","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["6.01","6.01","6.18","5.86","5.86","5.73","5.87","5.87","5.93","6.21","6.21","6.01","6.1","6.1","6.28"],"header":"pH","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"minPO4","name":"minPO4","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.33","0.12","1.42","2.83","0.2","1.5","1.17","0.11","0.75","1.0","0.02","3.67","2.58","0.0","4.08"],"header":"minPO4","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"minNH4","name":"minNH4","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["3.67","1.65","3.92","3.33","1.28","3.17","2.17","1.18","1.83","2.42","2.6","4.33","3.5","0.92","4.75"],"header":"minNH4","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"minNO3","name":"minNO3","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["47.17","11.31","24.33","42.5","22.71","19.92","11.25","36.61","19.67","20.67","12.31","106.42","11.58","7.84","37.33"],"header":"minNO3","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"minTIN","name":"minTIN","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["50.83","12.96","28.25","45.83","23.99","23.08","13.42","37.79","21.5","23.08","14.91","110.75","15.08","8.76","42.08"],"header":"minTIN","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"NH4","name":"NH4","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["8.03","4.08","4.81","3.11","2.5","5.57","2.32","3.91","2.92","4.28","8.19","2.04","2.03","6.5","3.44"],"header":"NH4","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"NO3","name":"NO3","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["4.67","2.65","3.13","2.57","1.06","3.22","3.31","2.3","2.27","5.07","6.39","3.71","2.9","2.76","4.53"],"header":"NO3","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"PO4","name":"PO4","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["9.52","8.55","7.58","8.66","2.8","7.95","7.3","4.25","7.04","9.15","5.62","5.97","8.28","6.08","9.45"],"header":"PO4","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"DOC","name":"DOC","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["118.74","39.7","75.86","60.57","33.2","59.51","64.98","45.0","56.68","73.98","44.3","44.89","43.84","30.8","63.79"],"header":"DOC","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"DON","name":"DON","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["11.66","4.2","8.22","6.48","2.5","6.32","7.03","4.3","5.42","8.47","3.7","5.22","5.56","1.6","4.91"],"header":"DON","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"DOCN","name":"DOCN","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["11.15","9.45","9.44","10.12","13.29","10.67","11.93","10.46","11.09","9.93","11.98","11.57","12.21","19.27","14.67"],"header":"DOCN","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"micC","name":"micC","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["690.11","469.0","444.72","364.62","117.0","440.6","284.5","440.0","390.66","396.42","676.0","277.37","392.56","416.0","334.93"],"header":"micC","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"micN","name":"micN","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["135.48","84.0","110.13","66.4","20.0","86.19","48.79","97.0","72.61","73.73","147.0","56.55","62.37","91.0","72.64"],"header":"micN","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"micP","name":"micP","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["87.98","55.0","65.03","70.68","11.0","81.45","52.5","56.0","85.7","93.13","94.0","33.4","66.65","63.0","64.23"],"header":"micP","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"micCN","name":"micCN","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["4.61","5.58","4.42","5.07","5.85","5.14","5.57","4.54","5.26","4.72","4.6","4.72","5.78","4.57","4.49"],"header":"micCN","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"micCP","name":"micCP","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["8.61","8.48","8.13","6.7","10.54","5.39","8.33","7.91","4.5","4.66","7.18","11.55","7.34","6.6","5.54"],"header":"micCP","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"micNP","name":"micNP","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.74","1.52","2.37","1.31","1.8","1.09","1.39","1.74","0.83","0.98","1.56","2.45","1.19","1.44","1.17"],"header":"micNP","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"AG_ase","name":"AG_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.26","0.15","0.24","0.33","0.02","0.22","0.34","0.15","0.21","0.32","0.39","0.15","0.2","0.37","0.28"],"header":"AG ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"BG_ase","name":"BG_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["4.1","6.97","4.22","4.37","3.04","4.57","3.01","9.09","3.22","4.14","8.61","3.55","3.92","10.48","4.62"],"header":"BG ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"BP_ase","name":"BP_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["3.5","5.34","2.54","3.05","2.89","3.02","2.34","7.28","2.33","3.45","7.11","1.77","2.26","6.46","2.11"],"header":"BP ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"CE_ase","name":"CE_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.0","2.07","1.09","1.26","0.65","1.09","0.76","1.5","0.75","1.07","1.53","0.68","0.86","2.21","0.86"],"header":"CE ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"P_ase","name":"P_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["20.43","25.65","15.07","19.21","18.11","20.27","15.19","44.04","14.79","23.26","35.1","13.02","16.9","35.75","14.8"],"header":"P ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"N_ase","name":"N_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["5.63","4.52","3.54","3.58","2.91","3.65","2.61","5.56","2.57","3.7","8.34","2.95","2.83","9.43","3.37"],"header":"N ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"S_ase","name":"S_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["6.08","6.7","3.8","5.72","1.28","4.92","3.2","5.57","5.45","5.86","10.68","3.56","5.26","11.78","4.54"],"header":"S ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"XY_ase","name":"XY_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.17","2.52","1.01","1.36","0.76","1.11","0.86","1.75","0.66","1.11","1.42","0.71","1.01","1.84","0.72"],"header":"XY ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"AG_Q10","name":"AG_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.77","2.01","1.51","2.48","1.49","1.51","1.48","1.73","1.83","1.72","1.31","2.08","1.67","1.93","1.54"],"header":"AG Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"BG_Q10","name":"BG_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.38","1.75","1.31","1.36","1.56","1.5","1.35","1.72","1.31","1.45","1.39","1.76","1.36","1.53","1.37"],"header":"BG Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"BP_Q10","name":"BP_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.39","1.51","1.34","1.42","1.55","1.58","1.39","1.42","1.44","1.46","1.42","1.74","1.56","1.44","1.41"],"header":"BP Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"CE_Q10","name":"CE_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["2.03","1.87","1.89","1.99","1.9","2.3","1.88","1.95","2.36","1.76","1.96","2.0","1.81","2.11","1.81"],"header":"CE Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"P_Q10","name":"P_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.37","1.39","1.22","1.27","1.51","1.37","1.28","1.48","1.24","1.28","1.33","1.48","1.35","1.38","1.27"],"header":"P Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"N_Q10","name":"N_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.46","1.56","1.54","1.54","1.75","1.85","1.51","1.68","1.6","1.36","1.54","1.69","1.48","1.46","1.47"],"header":"N Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"S_Q10","name":"S_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.3","1.65","1.23","1.27","1.39","1.39","1.3","1.6","1.47","1.3","1.51","1.48","1.32","1.5","1.33"],"header":"S Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"XY_Q10","name":"XY_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.72","1.86","1.59","1.72","1.89","1.76","1.82","1.88","1.63","1.63","1.91","1.83","1.84","1.76","1.65"],"header":"XY Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"CO2","name":"CO2","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["11.78","4.28","5.27","4.46","40.47","4.06","4.62","15.37","4.28","6.19","8.87","2.08","9.39","13.6","5.21"],"header":"CO2","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"LP_ase","name":"LP_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["52.8","40.69","38.61","32.1","13.94","29.51","24.11","32.65","27.78","42.4","57.13","21.96","30.37","53.79","32.22"],"header":"LP ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"LP_Q10","name":"LP_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.76","1.71","1.81","2.0","1.77","2.14","1.79","1.67","1.85","1.96","1.4","2.07","1.9","1.79","1.8"],"header":"LP Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"PX_ase","name":"PX_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["147.13","211.18","94.13","119.75","225.56","81.25","110.75","245.65","92.5","97.0","294.39","92.38","122.63","339.6","111.88"],"header":"PX ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"PX_Q10","name":"PX_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.22","1.07","1.58","1.41","1.15","1.24","1.33","1.23","1.53","1.15","1.27","3.58","1.39","1.27","1.14"],"header":"PX Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"CUEcn","name":"CUEcn","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.32","0.23","0.22","0.26","0.29","0.25","0.27","0.19","0.27","0.27","0.26","0.24","0.26","0.24","0.22"],"header":"CUEcn","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"CUEcp","name":"CUEcp","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.21","0.18","0.21","0.18","0.33","0.19","0.18","0.23","0.16","0.15","0.17","0.22","0.21","0.19","0.16"],"header":"CUEcp","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"NUE","name":"NUE","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.77","0.87","0.87","0.84","0.81","0.85","0.83","0.9","0.83","0.83","0.84","0.85","0.84","0.85","0.87"],"header":"NUE","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"PUE","name":"PUE","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.88","0.9","0.88","0.9","0.77","0.9","0.9","0.87","0.92","0.92","0.91","0.88","0.88","0.89","0.92"],"header":"PUE","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Tmin","name":"Tmin","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.07","2.19","0.14","-0.29","0.63","-2.63","0.81","0.94","-0.13","0.65","1.48","0.0","-0.24","0.96","-2.46"],"header":"Tmin","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"SI","name":"SI","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.51","2.3","1.44","1.49","1.72","1.25","1.77","2.08","1.42","1.61","1.81","1.54","1.51","1.82","1.29"],"header":"SI","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"enzCN","name":"enzCN","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.4614","0.6456","0.5024","0.5565","0.5302","0.555","0.4877","0.6931","0.4624","0.4941","0.5933","0.5058","0.5112","0.6515","0.5231"],"header":"enzCN","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"enzCP","name":"enzCP","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.5911","0.7166","0.6554","0.6412","0.4918","0.6173","0.5596","0.6412","0.5556","0.5762","0.6628","0.6037","0.6064","0.7218","0.6607"],"header":"enzCP","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"enzNP","name":"enzNP","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.2812","1.11","1.3044","1.1522","0.9277","1.1123","1.1473","0.9251","1.2017","1.1662","1.1173","1.1935","1.1862","1.1079","1.263"],"header":"enzNP","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"2fceb59691ab8660564d28367278fbe8"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section></section><section id="multivariate-summary" class="level1"><h1>Multivariate Summary</h1>
<p>Summary results for of <code>envfit</code> and <code>bioenv</code> tests for edaphic properties, soil functional response, and temperature adaptation for both 16S rRNA and ITS data sets.</p>
<section id="s-rrna" class="level2"><h2 class="anchored" data-anchor-id="s-rrna">16S rRNA</h2>
<p>Metadata parameters removed based on autocorrelation results.</p>
<ul>
<li>edaphic properties: TEB + DON + Na + Al + Ca</li>
<li>soil functional response: micN + micNP + enzCN + enzCP + BP<sub>ase</sub> + CE<sub>ase</sub> + LP<sub>ase</sub> + N<sub>ase</sub> + P<sub>ase</sub>
</li>
<li>temperature adaptation: NUE + PUE + SI</li>
</ul>
<p>Metadata parameters removed from capscale analysis based on Degrees of Freedom.</p>
<ul>
<li>edaphic properties: Mg + Mn</li>
<li>soil functional response: NONE</li>
<li>temperature adaptation: NONE</li>
</ul>
<p><br></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,bWV0YWRhdGFfc2V0LHBhcmFtZXRlcixyMi4uZW52Zml0LixQci4uci4uLmVudmZpdC4sci4uYmlvZW52LixwdmFsLi5iaW9lbnYuCmVkYXBoaWMsQVNULDAuODI4Njc5MDkwNTU2MTc0LDkuOTkwMDA5OTkwMDA5OTllLTQsMSwwLjAwMQplZGFwaGljLEgyTywwLjUxODkzMjg5NjE5Njc5MSwwLjAwOTk5MDAwOTk5MDAwOTk5LE5BLE5BCmVkYXBoaWMsRE9DLDAuNDQ1OTA0MTQ2NTcwNjk1LDAuMDIzOTc2MDIzOTc2MDI0LE5BLE5BCmZ1bmN0aW9uYWwgcmVzcG9uc2UsQUdfYXNlLDAuNDQzNTQxMzY5MDg2MDU4LDAuMDI1OTc0MDI1OTc0MDI2LDAuNTU4ODA3Nzg2ODg5NjUsMC4wMDEKZnVuY3Rpb25hbCByZXNwb25zZSxCR19hc2UsMC41NjAzODgwOTAxNDMwMDEsMC4wMDY5OTMwMDY5OTMwMDY5OSxOQSxOQQpmdW5jdGlvbmFsIHJlc3BvbnNlLFNfYXNlLDAuNzM3MDgyNjIxOTg3NzY2LDAuMDAxOTk4MDAxOTk4MDAyLDAuNjEzODUzNDgyNTU3OTE4LDAuMDAxCmZ1bmN0aW9uYWwgcmVzcG9uc2UsWFlfYXNlLDAuNTE5MzAxMTcwMzIwMzQ5LDAuMDA4OTkxMDA4OTkxMDA4OTksMC40NTU5NjY4MjQ2Mzg0MTEsMC4wMDIKZnVuY3Rpb25hbCByZXNwb25zZSxQWF9hc2UsMC43NjM5OTQzNjc0MTA2MDQsOS45OTAwMDk5OTAwMDk5OWUtNCwwLjYxMTUyNTM5NDA3MjIzNSwwLjAwMQpmdW5jdGlvbmFsIHJlc3BvbnNlLENPMiwwLjUwNDEyMjg4MzcyOTgxOCwwLjAxMjk4NzAxMjk4NzAxMyxOQSxOQQpmdW5jdGlvbmFsIHJlc3BvbnNlLGVuek5QLDAuNjI0MTgzMjAyMTAxOTI0LDAuMDAzOTk2MDAzOTk2MDA0LDAuNDYyMzI2MzUyODkyMzkxLDAuMDA2CnRlbXBlcmF0dXJlIGFkYXB0YXRpb24sU19RMTAsMC40OTU2NDMyMjgwMTk5MzYsMC4wMTQ5ODUwMTQ5ODUwMTUsMC40Mzk0ODM2MTgxNTMyMDcsMC4wMDEKdGVtcGVyYXR1cmUgYWRhcHRhdGlvbixMUF9RMTAsMC40MTMxMjk5ODg2NzI0ODQsMC4wNDA5NTkwNDA5NTkwNDEsMC4zNzcxNzY0NTY3MTAxMTMsMC4wMDUKdGVtcGVyYXR1cmUgYWRhcHRhdGlvbixUbWluLDAuNDQ1NTkwNjk4NjMzODkzLDAuMDI5OTcwMDI5OTcwMDMsMC40MDQ0MzQwNDI3NzQxNTcsMC4wMDUKdGVtcGVyYXR1cmUgYWRhcHRhdGlvbixDVUVjcCxOQSxOQSwwLjMyNTAzOTg3Mzg1ODE4NywwLjAxMwp0ZW1wZXJhdHVyZSBhZGFwdGF0aW9uLFBfUTEwLE5BLE5BLDAuNTE3NzIzMzg3MDkwODI2LDAuMDAxCg==').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'ssu_multivariate_summary.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
</div>
<div class="cell">

</div>
<p><small><strong>(16S rRNA) Table 1 | Significant results of envfit &amp; bioenv tests for the 16S rRNA data set.</strong></small></p>
<div class="cell">
<div class="cell-output-display">

<div id="htmlwidget-8265b90b3a32bfcc4437" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-8265b90b3a32bfcc4437">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"metadata_set":["edaphic","edaphic","edaphic","functional_response","functional_response","functional_response","functional_response","functional_response","functional_response","functional_response","temperature adaptation","temperature adaptation","temperature adaptation","temperature adaptation","temperature adaptation"],"parameter":["AST","H2O","DOC","AG_ase","BG_ase","S_ase","XY_ase","PX_ase","CO2","enzNP","S_Q10","LP_Q10","Tmin","CUEcp","P_Q10"],"envfit_r2":["0.829","0.519","0.446","0.444","0.56","0.737","0.519","0.764","0.504","0.624","0.496","0.413","0.446","",""],"envfit_pval":["0.001","0.01","0.024","0.026","0.007","0.002","0.009","0.001","0.013","0.004","0.015","0.041","0.03","",""],"bioenv_r":["1","","","0.559","","0.614","0.456","0.612","","0.462","0.439","0.377","0.404","0.325","0.518"],"bioenv_pval":["0.001","","","0.001","","0.001","0.002","0.001","","0.006","0.001","0.005","0.005","0.013","0.001"]},"columns":[{"accessor":"metadata_set","name":"Metadata group","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["edaphic","edaphic","edaphic","functional_response","functional_response","functional_response","functional_response","functional_response","functional_response","functional_response","temperature adaptation","temperature adaptation","temperature adaptation","temperature adaptation","temperature adaptation"],"header":"Metadata group","minWidth":90,"align":"left","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"5px solid #eee"},"headerStyle":{"borderRight":"5px solid #eee"}},{"accessor":"parameter","name":"parameter","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["AST","H2O","DOC","AG_ase","BG_ase","S_ase","XY_ase","PX_ase","CO2","enzNP","S_Q10","LP_Q10","Tmin","CUEcp","P_Q10"],"header":"parameter","minWidth":50,"align":"center","footerStyle":{"fontWeight":"bold"},"style":{"borderRight":"5px solid #eee"},"headerStyle":{"borderRight":"5px solid #eee"}},{"accessor":"envfit_r2","name":"r2","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["0.829","0.519","0.446","0.444","0.56","0.737","0.519","0.764","0.504","0.624","0.496","0.413","0.446","",""],"header":"r2","minWidth":50,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"envfit_pval","name":"P(>r)","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["0.001","0.01","0.024","0.026","0.007","0.002","0.009","0.001","0.013","0.004","0.015","0.041","0.03","",""],"header":"P(>r)","minWidth":50,"align":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"}],"headerStyle":{"borderRight":"5px solid #eee"}},{"accessor":"bioenv_r","name":"r","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["1","","","0.559","","0.614","0.456","0.612","","0.462","0.439","0.377","0.404","0.325","0.518"],"header":"r","minWidth":40,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"bioenv_pval","name":"p-value","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["0.001","","","0.001","","0.001","0.002","0.001","","0.006","0.001","0.005","0.005","0.013","0.001"],"header":"p-value","minWidth":50,"align":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}]}],"columnGroups":[{"name":"envfit","headerStyle":{"fontSize":"1.2em"},"columns":["envfit_r2","envfit_pval"]},{"name":"bioenv","headerStyle":{"fontSize":"1.2em"},"columns":["bioenv_r","bioenv_pval"]}],"pivotBy":["metadata_set"],"defaultPageSize":15,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"defaultExpanded":true,"highlight":true,"bordered":true,"striped":true,"compact":true,"nowrap":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"f8921c85bc20a3af01c8aab7f90684d9"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section><section id="its" class="level2"><h2 class="anchored" data-anchor-id="its">ITS</h2>
<p>Metadata parameters removed based on autocorrelation results.</p>
<ul>
<li>edaphic properties: TEB + DON + Na + Al + Ca</li>
<li>soil functional response: micN + micNP + enzCN + enzCP + BP<sub>ase</sub> + CE<sub>ase</sub> + LP<sub>ase</sub> + N<sub>ase</sub> + P<sub>ase</sub>
</li>
<li>temperature adaptation: NUE + PUE + P<sub>Q10</sub> + SI</li>
</ul>
<p>Metadata parameters removed from capscale analysis based on Degrees of Freedom.</p>
<ul>
<li>edaphic properties: Mg + Mn + Na + Al + Fe + K</li>
<li>soil functional response: NONE</li>
<li>temperature adaptation: S<sub>Q10</sub>
</li>
</ul>
<p><br></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,bWV0YWRhdGFfc2V0LHBhcmFtZXRlcixyMi4uZW52Zml0LixQci4uci4uLmVudmZpdC4sci4uYmlvZW52LixwdmFsLi5iaW9lbnYuCmVkYXBoaWMsQVNULDAuNDg0ODg2NDQ1ODk5Nzc1LDAuMDM2OTYzMDM2OTYzMDM3LDEsMC4wMDEKZWRhcGhpYyxET0MsMC41MzUzODA1ODYzNzMyMzgsMC4wMjc5NzIwMjc5NzIwMjgsTkEsTkEKZnVuY3Rpb25hbCByZXNwb25zZSxtaWNQLDAuNjkyNDY3MjMzMzU3Mjg2LDAuMDAxOTk4MDAxOTk4MDAyLE5BLE5BCmZ1bmN0aW9uYWwgcmVzcG9uc2UsbWljQ1AsMC41ODMxMjUzMzQzMzI3MTgsMC4wMTU5ODQwMTU5ODQwMTYsTkEsTkEKZnVuY3Rpb25hbCByZXNwb25zZSxBR19hc2UsMC41MDU5OTczMjI0OTI1MDQsMC4wMzY5NjMwMzY5NjMwMzcsTkEsTkEKZnVuY3Rpb25hbCByZXNwb25zZSxQWF9hc2UsMC41MDAyMjE0MDAxNTczNzMsMC4wMzQ5NjUwMzQ5NjUwMzUsMC42ODUzNTk0NDQwNzg2NzMsMC4wMDEKZnVuY3Rpb25hbCByZXNwb25zZSxlbnpOUCwwLjU0NzEwNTM5NzYxMDE3MiwwLjAxMzk4NjAxMzk4NjAxNCwwLjU1Mjg1MjIxMTA4MDA1OSwwLjAwMQpmdW5jdGlvbmFsIHJlc3BvbnNlLFhZX2FzZSxOQSxOQSwwLjUwNTQ0ODIzMjM1MDk4OCwwLjAwMgp0ZW1wZXJhdHVyZSBhZGFwdGF0aW9uLFhZX1ExMCwwLjYxNjU3MzUxMTQ1ODc2NSwwLjAwOTk5MDAwOTk5MDAwOTk5LDAuNzI1Nzc1NDgxMjg0MDk1LDAuMDAxCnRlbXBlcmF0dXJlIGFkYXB0YXRpb24sQ1VFY3AsMC40Nzg1ODM4MDk5NTUyMjEsMC4wMzQ5NjUwMzQ5NjUwMzUsTkEsTkEKdGVtcGVyYXR1cmUgYWRhcHRhdGlvbixUbWluLDAuNDc0ODY2Nzc4Njk5MjE2LDAuMDI3OTcyMDI3OTcyMDI4LDAuNjE2MDE1OTMzODQ2NzE3LDAuMDAxCg==').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'its_multivariate_summary.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
</div>
<div class="cell">

</div>
<p><small><strong>(ITS) Table 1 | Significant results of envfit &amp; bioenv tests for the ITS data set.</strong></small></p>
<div class="cell">
<div class="cell-output-display">

<div id="htmlwidget-bf6d657f1ee5f79ebe40" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-bf6d657f1ee5f79ebe40">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"metadata_set":["edaphic","edaphic","functional_response","functional_response","functional_response","functional_response","functional_response","functional_response","temperature adaptation","temperature adaptation","temperature adaptation"],"parameter":["AST","DOC","micP","micCP","AG_ase","PX_ase","enzNP","XY_ase","XY_Q10","CUEcp","Tmin"],"envfit_r2":["0.485","0.535","0.692","0.583","0.506","0.5","0.547","","0.617","0.479","0.475"],"envfit_pval":["0.037","0.028","0.002","0.016","0.037","0.035","0.014","","0.01","0.035","0.028"],"bioenv_r":["1","","","","","0.685","0.553","0.505","0.726","","0.616"],"bioenv_pval":["0.001","","","","","0.001","0.001","0.002","0.001","","0.001"]},"columns":[{"accessor":"metadata_set","name":"Metadata group","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["edaphic","edaphic","functional_response","functional_response","functional_response","functional_response","functional_response","functional_response","temperature adaptation","temperature adaptation","temperature adaptation"],"header":"Metadata group","minWidth":90,"align":"left","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"5px solid #eee"},"headerStyle":{"borderRight":"5px solid #eee"}},{"accessor":"parameter","name":"parameter","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["AST","DOC","micP","micCP","AG_ase","PX_ase","enzNP","XY_ase","XY_Q10","CUEcp","Tmin"],"header":"parameter","minWidth":50,"align":"center","footerStyle":{"fontWeight":"bold"},"style":{"borderRight":"5px solid #eee"},"headerStyle":{"borderRight":"5px solid #eee"}},{"accessor":"envfit_r2","name":"r2","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["0.485","0.535","0.692","0.583","0.506","0.5","0.547","","0.617","0.479","0.475"],"header":"r2","minWidth":50,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"envfit_pval","name":"P(>r)","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["0.037","0.028","0.002","0.016","0.037","0.035","0.014","","0.01","0.035","0.028"],"header":"P(>r)","minWidth":50,"align":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"},{"color":"#B22271","fontWeight":"bold","borderRight":"5px solid #eee"}],"headerStyle":{"borderRight":"5px solid #eee"}},{"accessor":"bioenv_r","name":"r","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["1","","","","","0.685","0.553","0.505","0.726","","0.616"],"header":"r","minWidth":40,"align":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"bioenv_pval","name":"p-value","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["0.001","","","","","0.001","0.001","0.002","0.001","","0.001"],"header":"p-value","minWidth":50,"align":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}]}],"columnGroups":[{"name":"envfit","headerStyle":{"fontSize":"1.2em"},"columns":["envfit_r2","envfit_pval"]},{"name":"bioenv","headerStyle":{"fontSize":"1.2em"},"columns":["bioenv_r","bioenv_pval"]}],"pivotBy":["metadata_set"],"defaultPageSize":15,"showPageSizeOptions":true,"pageSizeOptions":[5,10,11],"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"defaultExpanded":true,"highlight":true,"bordered":true,"striped":true,"compact":true,"nowrap":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"5ae27f58dea759ea70d2fa87b9f7647c"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="cell">

</div>
</section></section><section id="s-rrna-1" class="level1 page-columns page-full"><h1>16s rRNA</h1>
<section id="data-formating" class="level2"><h2 class="anchored" data-anchor-id="data-formating">Data Formating</h2>
<ol type="1">
<li>Transform <code>ps</code> Objects</li>
</ol>
<p>The first step is to transform the phyloseq objects.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">samp_ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ssu18_ps_work"</span>, <span class="st">"ssu18_ps_pime"</span>, <span class="st">"ssu18_ps_perfect"</span>, 
             <span class="st">"ssu18_ps_work_otu"</span>, <span class="st">"ssu18_ps_pime_otu"</span>, <span class="st">"ssu18_ps_perfect_otu"</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">samp_ps</span><span class="op">)</span> <span class="op">{</span>
     <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">get</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span>
     <span class="va">tmp_ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transformcounts.html">transform_sample_counts</a></span><span class="op">(</span><span class="va">tmp_get</span>, <span class="kw">function</span><span class="op">(</span><span class="va">otu</span><span class="op">)</span> <span class="fl">1e5</span> <span class="op">*</span> <span class="va">otu</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">otu</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_ps</span><span class="op">@</span><span class="va">phy_tree</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
     <span class="va">tmp_ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_samples-methods.html">prune_samples</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_sums.html">sample_sums</a></span><span class="op">(</span><span class="va">tmp_ps</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="va">tmp_ps</span><span class="op">)</span>
     <span class="va">tmp_tree</span> <span class="op">&lt;-</span> <span class="fu">rtree</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/ntaxa-methods.html">ntaxa</a></span><span class="op">(</span><span class="va">tmp_ps</span><span class="op">)</span>, rooted <span class="op">=</span> <span class="cn">TRUE</span>, tip.label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_names-methods.html">taxa_names</a></span><span class="op">(</span><span class="va">tmp_ps</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_samp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_samp</span> <span class="op">&lt;-</span> <span class="va">tmp_samp</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span><span class="st">"TREAT_T"</span> <span class="op">=</span> <span class="st">"TEMP"</span><span class="op">)</span>
     <span class="va">tmp_ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/merge_phyloseq.html">merge_phyloseq</a></span><span class="op">(</span><span class="va">tmp_ps</span>, <span class="va">sample_data</span>, <span class="va">tmp_tree</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">tmp_ps</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">tmp_samp</span>
     <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_prop"</span><span class="op">)</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_name</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_ps</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<ol start="2" type="1">
<li>Format for <code>mctoolsr</code>
</li>
</ol>
<p>Then a bit of formatting to make the data compatible with <code>mctoolsr</code>.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">samp_ps</span><span class="op">)</span> <span class="op">{</span>
     <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_prop"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_tax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_tax</span><span class="op">$</span><span class="va">ASV_SEQ</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
     
     <span class="va">tmp_col_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">tmp_tax</span><span class="op">)</span>
     <span class="va">tmp_tax_merge</span> <span class="op">&lt;-</span> <span class="va">tmp_tax</span> <span class="op">%&gt;%</span> <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/unite.html">unite</a></span><span class="op">(</span><span class="va">taxonomy</span>, 
                                               <span class="fu">all_of</span><span class="op">(</span><span class="va">tmp_col_names</span><span class="op">)</span>, 
                                               sep <span class="op">=</span> <span class="st">";"</span><span class="op">)</span>
     <span class="va">tmp_tax_merge</span> <span class="op">&lt;-</span> <span class="va">tmp_tax_merge</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"#OTU_ID"</span><span class="op">)</span>
     <span class="va">tmp_otu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_otu</span> <span class="op">&lt;-</span> <span class="va">tmp_otu</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"#OTU_ID"</span><span class="op">)</span>
     <span class="va">tmp_otu_tax</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_otu</span>, <span class="va">tmp_tax_merge</span>, by <span class="op">=</span> <span class="st">"#OTU_ID"</span><span class="op">)</span>

     <span class="va">tmp_samp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_samp</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
     <span class="va">tmp_samp</span> <span class="op">&lt;-</span> <span class="va">tmp_samp</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"#SampleID"</span><span class="op">)</span>

     <span class="va">tmp_metad</span> <span class="op">&lt;-</span> <span class="va">metad</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span><span class="st">"#SampleID"</span> <span class="op">=</span> <span class="st">"id"</span><span class="op">)</span>
     <span class="va">tmp_metad</span><span class="op">[</span>,<span class="fl">2</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
     <span class="va">tmp_md</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_samp</span>, <span class="va">tmp_metad</span>, by <span class="op">=</span> <span class="st">"#SampleID"</span><span class="op">)</span>
     
     <span class="va">tmp_otu_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_otu_tax"</span><span class="op">)</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_otu_name</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_otu_name</span>, <span class="va">tmp_otu_tax</span><span class="op">)</span>

     <span class="va">tmp_md_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_md"</span><span class="op">)</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_md_name</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_md_name</span>, <span class="va">tmp_md</span><span class="op">)</span>

     <span class="va">tmp_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="st">"files/metadata/tables/"</span><span class="op">)</span>
     <span class="fu">write_delim</span><span class="op">(</span><span class="va">tmp_otu_tax</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">tmp_path</span>, <span class="va">tmp_otu_name</span>, <span class="st">".txt"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>, delim <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span>
     <span class="fu">write_delim</span><span class="op">(</span><span class="va">tmp_md</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">tmp_path</span>, <span class="va">tmp_md_name</span>, <span class="st">".txt"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>, delim <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">samp_ps</span><span class="op">)</span> <span class="op">{</span>
     <span class="va">tmp_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="st">"files/metadata/tables/"</span><span class="op">)</span>
     <span class="va">tmp_otu_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_otu_tax"</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_md_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_md"</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_tax_table_fp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">tmp_path</span>, <span class="va">tmp_otu_name</span>, <span class="st">".txt"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
     <span class="va">tmp_map_fp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">tmp_path</span>, <span class="va">tmp_md_name</span>, <span class="st">".txt"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
     <span class="va">tmp_input</span> <span class="op">&lt;-</span> <span class="fu">load_taxa_table</span><span class="op">(</span><span class="va">tmp_tax_table_fp</span>, <span class="va">tmp_map_fp</span><span class="op">)</span>
     <span class="va">tmp_input_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_mc"</span><span class="op">)</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_input_name</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_input_name</span>, <span class="va">tmp_input</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_md"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_otu_tax"</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="choose-data-set" class="level2"><h2 class="anchored" data-anchor-id="choose-data-set">Choose Data Set</h2>
<p>At this point in the code we need to choose a data set to use, formatted with <code>mctoolsr</code>. Remember, there are four choices:</p>
<ol type="1">
<li>FULL, unfiltered data set.</li>
<li>Arbitrary filtered data set.</li>
<li>PERfect filtered data set.</li>
<li>PIME filtered data set.</li>
</ol>
<p>This way, if we want to test other data sets we only need to change the name here.</p>
<div class="cell">
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/ls.html">objects</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_mc"</span><span class="op">)</span>
<span class="va">ssu18_select_mc</span> <span class="op">&lt;-</span> <span class="va">ssu18_ps_pime_mc</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="normality-tests" class="level2"><h2 class="anchored" data-anchor-id="normality-tests">Normality Tests</h2>
<p>Before proceeding, we need to test each parameter in the metadata to see which ones are and are not normally distributed. For that, we use the Shapiro-Wilk Normality Test. Here we only need one of the metadata files.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">temp_md</span> <span class="op">&lt;-</span> <span class="va">ssu18_select_mc</span><span class="op">$</span><span class="va">map_loaded</span>
<span class="va">temp_md</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">9</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">shap_results</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">temp_md</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_shap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/shapiro.test.html">shapiro.test</a></span><span class="op">(</span><span class="va">temp_md</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
  <span class="va">tmp_p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">tmp_shap</span><span class="op">$</span><span class="va">p.value</span>, digits <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>
  <span class="va">tmp_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/eval.html">eval</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Logic.html">isTRUE</a></span><span class="op">(</span><span class="va">tmp_shap</span><span class="op">$</span><span class="va">p.value</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span>
  <span class="va">shap_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">shap_results</span>, <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">i</span>, <span class="va">tmp_p</span>, <span class="va">tmp_res</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">shap_results</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"parameter"</span>, <span class="st">"p-value"</span>, <span class="st">"tranform"</span><span class="op">)</span>
<span class="va">shap_results</span>
<span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">shap_results</span>, <span class="va">tranform</span> <span class="op">==</span> <span class="st">"TRUE"</span><span class="op">)</span>
<span class="va">md_to_tranform</span> <span class="op">&lt;-</span> <span class="va">shap_results</span><span class="op">$</span><span class="va">parameter</span><span class="op">[</span><span class="va">shap_results</span><span class="op">$</span><span class="va">tranform</span> <span class="op">==</span> <span class="cn">TRUE</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"temp_md"</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<details><summary><strong>Show the results</strong> of each normality test for metadata parameters
</summary><p></p>
<p><small><strong>(16S rRNA) Table 2 | Results of the Shapiro-Wilk Normality Tests. P-values in red are significance (p-value &lt; 0.05) meaning the parameter needs to be normalized.</strong></small></p>
<div class="cell">
<div class="cell-output-display">

<div id="htmlwidget-18069ed4e250861e54bb" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-18069ed4e250861e54bb">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"parameter":["AST","H2O","N","P","Al","Ca","Fe","K","Mg","Mn","Na","TEB","ECEC","pH","minPO4","minNH4","minNO3","minTIN","NH4","NO3","PO4","DOC","DON","DOCN","micC","micN","micP","micCN","micCP","micNP","AG_ase","BG_ase","BP_ase","CE_ase","P_ase","N_ase","S_ase","XY_ase","AG_Q10","BG_Q10","BP_Q10","CE_Q10","P_Q10","N_Q10","S_Q10","XY_Q10","CO2","LP_ase","LP_Q10","PX_ase","PX_Q10","CUEcn","CUEcp","NUE","PUE","Tmin","SI","enzCN","enzCP","enzNP"],"p-value":[0.00276,0.00188,0.05771,0.29892,0.00028,0.03383,0.00068,0.13159,0.16381,0.32078,0.05849,0.00435,0.00433,0.54818,0.05053,0.70522,0.0005,0.00072,0.07406,0.48101,0.24574,0.03038,0.79687,0.00309,0.16051,0.7732,0.23294,0.04004,0.66207,0.34934,0.70541,0.00305,0.00483,0.03066,0.00864,0.00206,0.05361,0.04239,0.2518,0.01348,0.03611,0.0619,0.28886,0.37946,0.27576,0.27276,6e-05,0.59206,0.45921,0.00822,0,0.91397,0.0037,0.33275,0.00025,0.05998,0.28369,0.09137,0.92171,0.09436],"tranform":[true,true,false,false,true,true,true,false,false,false,false,true,true,false,false,false,true,true,false,false,false,true,false,true,false,false,false,true,false,false,false,true,true,true,true,true,false,true,false,true,true,false,false,false,false,false,true,false,false,true,true,false,true,false,true,false,false,false,false,false]},"columns":[{"accessor":"parameter","name":"Parameter","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["AST","H2O","N","P","Al","Ca","Fe","K","Mg","Mn","Na","TEB","ECEC","pH","minPO4","minNH4","minNO3","minTIN","NH4","NO3","PO4","DOC","DON","DOCN","micC","micN","micP","micCN","micCP","micNP","AG_ase","BG_ase","BP_ase","CE_ase","P_ase","N_ase","S_ase","XY_ase","AG_Q10","BG_Q10","BP_Q10","CE_Q10","P_Q10","N_Q10","S_Q10","XY_Q10","CO2","LP_ase","LP_Q10","PX_ase","PX_Q10","CUEcn","CUEcp","NUE","PUE","Tmin","SI","enzCN","enzCP","enzNP"],"header":"Parameter","align":"left","headerStyle":{"borderRight":"1px solid #eee","fontSize":"0.9em"},"footerStyle":{"fontWeight":"bold"},"minWidth":150,"sticky":"left","style":{"borderRight":"1px solid #eee"}},{"accessor":"p-value","name":"p-value","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.00276","0.00188","0.05771","0.29892","0.00028","0.03383","0.00068","0.13159","0.16381","0.32078","0.05849","0.00435","0.00433","0.54818","0.05053","0.70522","0.0005","0.00072","0.07406","0.48101","0.24574","0.03038","0.79687","0.00309","0.16051","0.7732","0.23294","0.04004","0.66207","0.34934","0.70541","0.00305","0.00483","0.03066","0.00864","0.00206","0.05361","0.04239","0.2518","0.01348","0.03611","0.0619","0.28886","0.37946","0.27576","0.27276","0.00006","0.59206","0.45921","0.00822","0.0","0.91397","0.0037","0.33275","0.00025","0.05998","0.28369","0.09137","0.92171","0.09436"],"header":"p-value","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"},"style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"tranform","name":"tranform","type":"logical","sortable":true,"resizable":true,"filterable":true,"cell":["TRUE","TRUE","FALSE","FALSE","TRUE","TRUE","TRUE","FALSE","FALSE","FALSE","FALSE","TRUE","TRUE","FALSE","FALSE","FALSE","TRUE","TRUE","FALSE","FALSE","FALSE","TRUE","FALSE","TRUE","FALSE","FALSE","FALSE","TRUE","FALSE","FALSE","FALSE","TRUE","TRUE","TRUE","TRUE","TRUE","FALSE","TRUE","FALSE","TRUE","TRUE","FALSE","FALSE","FALSE","FALSE","FALSE","TRUE","FALSE","FALSE","TRUE","TRUE","FALSE","TRUE","FALSE","TRUE","FALSE","FALSE","FALSE","FALSE","FALSE"],"header":"tranform","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":60,"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"c00f9cada74cd3856858bafcb7cb61a5"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</details><p>Looks like we need to transform 25 metadata parameters.</p>
</section><section id="normalize-parameters" class="level2"><h2 class="anchored" data-anchor-id="normalize-parameters">Normalize Parameters</h2>
<p>Here we use the R package <a href="https://cran.r-project.org/web/packages/bestNormalize/vignettes/bestNormalize.html"><code>bestNormalize</code></a> to find and execute the best normalizing transformation. The function will test the following normalizing transformations:</p>
<ul>
<li><p><code>arcsinh_x</code> performs an arcsinh transformation.</p></li>
<li><p><code>boxcox</code> Perform a Box-Cox transformation and center/scale a vector to attempt normalization. <code>boxcox</code> estimates the optimal value of lambda for the Box-Cox transformation. The function will return an error if a user attempt to transform nonpositive data.</p></li>
<li><p><code>yeojohnson</code> Perform a Yeo-Johnson Transformation and center/scale a vector to attempt normalization. <code>yeojohnson</code> estimates the optimal value of lambda for the Yeo-Johnson transformation. The Yeo-Johnson is similar to the Box-Cox method, however it allows for the transformation of nonpositive data as well.</p></li>
<li><p><code>orderNorm</code> The Ordered Quantile (ORQ) normalization transformation, <code>orderNorm()</code>, is a rank-based procedure by which the values of a vector are mapped to their percentile, which is then mapped to the same percentile of the normal distribution. Without the presence of ties, this essentially guarantees that the transformation leads to a uniform distribution.</p></li>
<li><p><code>log_x</code> performs a simple log transformation. The parameter a is essentially estimated by the training set by default (estimated as the minimum possible to some extent epsilon), while the base must be specified beforehand. The default base of the log is 10.</p></li>
<li><p><code>sqrt_x</code> performs a simple square-root transformation. The parameter a is essentially estimated by the training set by default (estimated as the minimum possible), while the base must be specified beforehand.</p></li>
<li><p><code>exp_x</code> performs a simple exponential transformation.</p></li>
</ul>
<p>See this GitHub issue (<a href="https://github.com/petersonR/bestNormalize/issues/5">#5</a>) for a description on getting reproducible results. Apparently, you can get different results because the <code>bestNormalize()</code> function uses repeated cross-validation (and doesnt automatically set the seed), so the results will be slightly different each time the function is executed.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">119</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">md_to_tranform</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_md</span> <span class="op">&lt;-</span> <span class="va">ssu18_select_mc</span><span class="op">$</span><span class="va">map_loaded</span>
  <span class="va">tmp_best_norm</span> <span class="op">&lt;-</span> <span class="fu">bestNormalize</span><span class="op">(</span><span class="va">tmp_md</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, r <span class="op">=</span> <span class="fl">1</span>, k <span class="op">=</span> <span class="fl">5</span>, loo <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
  <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_best_norm_test"</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_best_norm</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_name</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<details><summary>
Show the chosen transformations
</summary><div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
## bestNormalize Chosen transformation of AST ## 
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
25.800 26.275 28.800 30.280 41.770 
_____________________________________

## bestNormalize Chosen transformation of H2O ## 
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
   0%   25%   50%   75%  100% 
0.204 0.308 0.370 0.384 0.401 
_____________________________________

## bestNormalize Chosen transformation of Al ## 
Standardized asinh(x) Transformation with 15 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 0.0059997 
 - sd (before standardization) = 0.008280287 
_____________________________________

## bestNormalize Chosen transformation of Ca ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
25.49 27.19 29.23 34.36 39.79 
_____________________________________

## bestNormalize Chosen transformation of Fe ## 
Standardized asinh(x) Transformation with 15 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 0.0119984 
 - sd (before standardization) = 0.01264569 
_____________________________________

## bestNormalize Chosen transformation of TEB ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
41.6 43.3 45.1 51.4 61.7 
_____________________________________

## bestNormalize Chosen transformation of ECEC ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
41.7 43.3 45.3 51.5 61.8 
_____________________________________

## bestNormalize Chosen transformation of minNO3 ## 
Standardized Box Cox Transformation with 15 nonmissing obs.:
 Estimated statistics:
 - lambda = -0.3344538 
 - mean (before standardization) = 1.907818 
 - sd (before standardization) = 0.2388776 
_____________________________________

## bestNormalize Chosen transformation of minTIN ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 14 unique values 
 - Original quantiles:
     0%     25%     50%     75%    100% 
  8.760  14.995  23.080  39.935 110.750 
_____________________________________

## bestNormalize Chosen transformation of DOC ## 
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
     0%     25%     50%     75%    100% 
 30.800  44.070  56.680  64.385 118.740 
_____________________________________

## bestNormalize Chosen transformation of DOCN ## 
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 9.440 10.290 11.150 12.095 19.270 
_____________________________________

## bestNormalize Chosen transformation of micCN ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 14 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
4.420 4.585 4.720 5.415 5.850 
_____________________________________

## bestNormalize Chosen transformation of BG_ase ## 
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 3.010  3.735  4.220  5.795 10.480 
_____________________________________

## bestNormalize Chosen transformation of BP_ase ## 
Standardized Log_b(x + a) Transformation with 15 nonmissing obs.:
 Relevant statistics:
 - a = 0 
 - b = 10 
 - mean (before standardization) = 0.5217757 
 - sd (before standardization) = 0.2001498 
_____________________________________

## bestNormalize Chosen transformation of CE_ase ## 
Standardized Log_b(x + a) Transformation with 15 nonmissing obs.:
 Relevant statistics:
 - a = 0 
 - b = 10 
 - mean (before standardization) = 0.03306978 
 - sd (before standardization) = 0.1652591 
_____________________________________

## bestNormalize Chosen transformation of P_ase ## 
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
13.020 15.130 19.210 24.455 44.040 
_____________________________________

## bestNormalize Chosen transformation of N_ase ## 
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
  0%  25%  50%  75% 100% 
2.57 2.93 3.58 5.04 9.43 
_____________________________________

## bestNormalize Chosen transformation of XY_ase ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 13 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
0.66 0.81 1.11 1.39 2.52 
_____________________________________

## bestNormalize Chosen transformation of BG_Q10 ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 13 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
1.310 1.360 1.390 1.545 1.760 
_____________________________________

## bestNormalize Chosen transformation of BP_Q10 ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 11 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
1.340 1.415 1.440 1.530 1.740 
_____________________________________

## bestNormalize Chosen transformation of CO2 ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 14 unique values 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 2.080  4.370  5.270 10.585 40.470 
_____________________________________

## bestNormalize Chosen transformation of PX_ase ## 
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
     0%     25%     50%     75%    100% 
 81.250  95.565 119.750 218.370 339.600 
_____________________________________

## bestNormalize Chosen transformation of PX_Q10 ## 
Standardized Yeo-Johnson Transformation with 15 nonmissing obs.:
 Estimated statistics:
 - lambda = -4.999946 
 - mean (before standardization) = 0.1968247 
 - sd (before standardization) = 0.001294011 
_____________________________________

## bestNormalize Chosen transformation of CUEcp ## 
Standardized Box Cox Transformation with 15 nonmissing obs.:
 Estimated statistics:
 - lambda = -0.9999576 
 - mean (before standardization) = -4.229896 
 - sd (before standardization) = 0.9115063 
_____________________________________

## bestNormalize Chosen transformation of PUE ## 
orderNorm Transformation with 15 nonmissing obs and ties
 - 7 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
0.770 0.880 0.900 0.905 0.920 
_____________________________________</code></pre>
</div>
</div>
</details><details><summary>
Show the complete <code>bestNormalize</code> results
</summary><div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
## Results of bestNormalize for AST ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 3.4
 - Box-Cox: 1.8
 - Center+scale: 4.2
 - Exp(x): 21.2667
 - Log_b(x+a): 3.4
 - orderNorm (ORQ): 0.2
 - sqrt(x + a): 4.2
 - Yeo-Johnson: 2.8667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
25.800 26.275 28.800 30.280 41.770 
_____________________________________

## Results of bestNormalize for H2O ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 6.8667
 - Box-Cox: 5.8
 - Center+scale: 5.8
 - Exp(x): 5.8
 - Log_b(x+a): 6.0667
 - orderNorm (ORQ): 0.7333
 - sqrt(x + a): 6.8667
 - Yeo-Johnson: 5.8
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
   0%   25%   50%   75%  100% 
0.204 0.308 0.370 0.384 0.401 
_____________________________________

## Results of bestNormalize for Al ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 8.2
 - Center+scale: 8.2
 - Exp(x): 8.2
 - Log_b(x+a): 8.2
 - orderNorm (ORQ): 8.2
 - sqrt(x + a): 8.2
 - Yeo-Johnson: 8.2
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized asinh(x) Transformation with 15 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 0.0059997 
 - sd (before standardization) = 0.008280287 
_____________________________________

## Results of bestNormalize for Ca ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.8
 - Box-Cox: 1.8
 - Center+scale: 3.4
 - Exp(x): 18.0667
 - Log_b(x+a): 1.8
 - orderNorm (ORQ): 0.2
 - sqrt(x + a): 1.8
 - Yeo-Johnson: 1.2667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
25.49 27.19 29.23 34.36 39.79 
_____________________________________

## Results of bestNormalize for Fe ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 6.3333
 - Center+scale: 6.3333
 - Exp(x): 6.3333
 - Log_b(x+a): 6.3333
 - orderNorm (ORQ): 6.3333
 - sqrt(x + a): 6.3333
 - Yeo-Johnson: 6.3333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized asinh(x) Transformation with 15 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 0.0119984 
 - sd (before standardization) = 0.01264569 
_____________________________________

## Results of bestNormalize for TEB ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 2.3333
 - Box-Cox: 1.2667
 - Center+scale: 6.0667
 - Exp(x): 18.0667
 - Log_b(x+a): 2.3333
 - orderNorm (ORQ): 0.4667
 - sqrt(x + a): 5.5333
 - Yeo-Johnson: 1.5333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
41.6 43.3 45.1 51.4 61.7 
_____________________________________

## Results of bestNormalize for ECEC ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.2667
 - Box-Cox: 1.2667
 - Center+scale: 6.0667
 - Exp(x): 18.0667
 - Log_b(x+a): 1.2667
 - orderNorm (ORQ): 0.4667
 - sqrt(x + a): 3.9333
 - Yeo-Johnson: 1.5333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
41.7 43.3 45.3 51.5 61.8 
_____________________________________

## Results of bestNormalize for minNO3 ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.8
 - Box-Cox: 1.2667
 - Center+scale: 3.9333
 - Exp(x): 21.2667
 - Log_b(x+a): 1.8
 - orderNorm (ORQ): 1.2667
 - sqrt(x + a): 1.8
 - Yeo-Johnson: 1.2667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized Box Cox Transformation with 15 nonmissing obs.:
 Estimated statistics:
 - lambda = -0.3344538 
 - mean (before standardization) = 1.907818 
 - sd (before standardization) = 0.2388776 
_____________________________________

## Results of bestNormalize for minTIN ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.8
 - Box-Cox: 1.8
 - Center+scale: 3.9333
 - Exp(x): 21.2667
 - Log_b(x+a): 1.8
 - orderNorm (ORQ): 0.7333
 - sqrt(x + a): 1.8
 - Yeo-Johnson: 1.8
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 14 unique values 
 - Original quantiles:
     0%     25%     50%     75%    100% 
  8.760  14.995  23.080  39.935 110.750 
_____________________________________

## Results of bestNormalize for DOC ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 2.3333
 - Box-Cox: 1.2667
 - Center+scale: 2.8667
 - Exp(x): 21.2667
 - Log_b(x+a): 2.3333
 - orderNorm (ORQ): 0.4667
 - sqrt(x + a): 2.8667
 - Yeo-Johnson: 1.2667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
     0%     25%     50%     75%    100% 
 30.800  44.070  56.680  64.385 118.740 
_____________________________________

## Results of bestNormalize for DOCN ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 2.3333
 - Box-Cox: 0.7333
 - Center+scale: 3.9333
 - Exp(x): 21.2667
 - Log_b(x+a): 2.3333
 - orderNorm (ORQ): 0.2
 - sqrt(x + a): 3.1333
 - Yeo-Johnson: 0.2
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 9.440 10.290 11.150 12.095 19.270 
_____________________________________

## Results of bestNormalize for micCN ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 3.1333
 - Box-Cox: 3.1333
 - Center+scale: 3.1333
 - Exp(x): 5.5333
 - Log_b(x+a): 3.1333
 - orderNorm (ORQ): 0.2
 - sqrt(x + a): 3.1333
 - Yeo-Johnson: 3.1333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 14 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
4.420 4.585 4.720 5.415 5.850 
_____________________________________

## Results of bestNormalize for BG_ase ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 2.8667
 - Box-Cox: 2.3333
 - Center+scale: 5
 - Exp(x): 21.2667
 - Log_b(x+a): 2.8667
 - orderNorm (ORQ): 0.4667
 - sqrt(x + a): 2.8667
 - Yeo-Johnson: 2.3333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 3.010  3.735  4.220  5.795 10.480 
_____________________________________

## Results of bestNormalize for BP_ase ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.5333
 - Box-Cox: 1.5333
 - Center+scale: 3.1333
 - Exp(x): 11.9333
 - Log_b(x+a): 0.7333
 - orderNorm (ORQ): 0.7333
 - sqrt(x + a): 3.1333
 - Yeo-Johnson: 1.5333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized Log_b(x + a) Transformation with 15 nonmissing obs.:
 Relevant statistics:
 - a = 0 
 - b = 10 
 - mean (before standardization) = 0.5217757 
 - sd (before standardization) = 0.2001498 
_____________________________________

## Results of bestNormalize for CE_ase ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1
 - Box-Cox: 0.7333
 - Center+scale: 1.8
 - Exp(x): 4.2
 - Log_b(x+a): 0.4667
 - orderNorm (ORQ): 1
 - sqrt(x + a): 1
 - Yeo-Johnson: 0.4667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized Log_b(x + a) Transformation with 15 nonmissing obs.:
 Relevant statistics:
 - a = 0 
 - b = 10 
 - mean (before standardization) = 0.03306978 
 - sd (before standardization) = 0.1652591 
_____________________________________

## Results of bestNormalize for P_ase ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 2.0667
 - Box-Cox: 1.2667
 - Center+scale: 3.6667
 - Exp(x): 21.2667
 - Log_b(x+a): 2.0667
 - orderNorm (ORQ): 0.2
 - sqrt(x + a): 2.3333
 - Yeo-Johnson: 1.2667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
13.020 15.130 19.210 24.455 44.040 
_____________________________________

## Results of bestNormalize for N_ase ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.2667
 - Box-Cox: 1.8
 - Center+scale: 3.1333
 - Exp(x): 17.8
 - Log_b(x+a): 1.2667
 - orderNorm (ORQ): 0.2
 - sqrt(x + a): 3.1333
 - Yeo-Johnson: 1.2667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
  0%  25%  50%  75% 100% 
2.57 2.93 3.58 5.04 9.43 
_____________________________________

## Results of bestNormalize for XY_ase ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 0.7333
 - Box-Cox: 0.7333
 - Center+scale: 2.8667
 - Exp(x): 6.3333
 - Log_b(x+a): 0.7333
 - orderNorm (ORQ): 0.2
 - sqrt(x + a): 1.8
 - Yeo-Johnson: 0.7333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 13 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
0.66 0.81 1.11 1.39 2.52 
_____________________________________

## Results of bestNormalize for BG_Q10 ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 2.3333
 - Box-Cox: 2.3333
 - Center+scale: 2.3333
 - Exp(x): 6.0667
 - Log_b(x+a): 2.3333
 - orderNorm (ORQ): 0.2
 - sqrt(x + a): 2.3333
 - Yeo-Johnson: 2.3333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 13 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
1.310 1.360 1.390 1.545 1.760 
_____________________________________

## Results of bestNormalize for BP_Q10 ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 2.6
 - Box-Cox: 2.6
 - Center+scale: 2.6
 - Exp(x): 2.6
 - Log_b(x+a): 2.6
 - orderNorm (ORQ): 0.7333
 - sqrt(x + a): 2.6
 - Yeo-Johnson: 2.3333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 11 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
1.340 1.415 1.440 1.530 1.740 
_____________________________________

## Results of bestNormalize for CO2 ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.5333
 - Box-Cox: 2.6
 - Center+scale: 11.9333
 - Exp(x): 21.2667
 - Log_b(x+a): 1.5333
 - orderNorm (ORQ): 1.2667
 - sqrt(x + a): 3.4
 - Yeo-Johnson: 2.0667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 14 unique values 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 2.080  4.370  5.270 10.585 40.470 
_____________________________________

## Results of bestNormalize for PX_ase ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 2.3333
 - Box-Cox: 1.2667
 - Center+scale: 7.6667
 - Exp(x): 21.2667
 - Log_b(x+a): 2.3333
 - orderNorm (ORQ): 0.2
 - sqrt(x + a): 2.8667
 - Yeo-Johnson: 1.2667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and no ties 
 - Original quantiles:
     0%     25%     50%     75%    100% 
 81.250  95.565 119.750 218.370 339.600 
_____________________________________

## Results of bestNormalize for PX_Q10 ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 14.6
 - Box-Cox: 4.2
 - Center+scale: 21.2667
 - Exp(x): 21.2667
 - Log_b(x+a): 11.9333
 - orderNorm (ORQ): 2.0667
 - sqrt(x + a): 21.2667
 - Yeo-Johnson: 1.5333
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized Yeo-Johnson Transformation with 15 nonmissing obs.:
 Estimated statistics:
 - lambda = -4.999946 
 - mean (before standardization) = 0.1968247 
 - sd (before standardization) = 0.001294011 
_____________________________________

## Results of bestNormalize for CUEcp ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 3.9333
 - Box-Cox: 1.5333
 - Center+scale: 3.9333
 - Exp(x): 3.9333
 - Log_b(x+a): 2.0667
 - orderNorm (ORQ): 1.5333
 - sqrt(x + a): 2.0667
 - Yeo-Johnson: 2.0667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized Box Cox Transformation with 15 nonmissing obs.:
 Estimated statistics:
 - lambda = -0.9999576 
 - mean (before standardization) = -4.229896 
 - sd (before standardization) = 0.9115063 
_____________________________________

## Results of bestNormalize for PUE ## 
Best Normalizing transformation with 15 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 8.2
 - Box-Cox: 9.8
 - Center+scale: 8.2
 - Exp(x): 3.9333
 - Log_b(x+a): 9.8
 - orderNorm (ORQ): 2.0667
 - sqrt(x + a): 9.8
 - Yeo-Johnson: 9.8
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 15 nonmissing obs and ties
 - 7 unique values 
 - Original quantiles:
   0%   25%   50%   75%  100% 
0.770 0.880 0.900 0.905 0.920 
_____________________________________</code></pre>
</div>
</div>
</details><p>Great, now we can add the normalized transformed data back to our <code>mctoolsr</code> metadata file.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ssu18_select_mc_norm</span> <span class="op">&lt;-</span> <span class="va">ssu18_select_mc</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">md_to_tranform</span><span class="op">)</span> <span class="op">{</span>
      <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">i</span>, <span class="st">"_best_norm_test"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
      <span class="va">tmp_new_data</span> <span class="op">&lt;-</span> <span class="va">tmp_get</span><span class="op">$</span><span class="va">x.t</span>
      <span class="va">ssu18_select_mc_norm</span><span class="op">$</span><span class="va">map_loaded</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">tmp_new_data</span>
      <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>     </code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>And rerun the Shapiro Tests.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">temp_md_norm</span> <span class="op">&lt;-</span> <span class="va">ssu18_select_mc_norm</span><span class="op">$</span><span class="va">map_loaded</span>
<span class="va">temp_md_norm</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">9</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">shap_results_norm</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">temp_md_norm</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_shap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/shapiro.test.html">shapiro.test</a></span><span class="op">(</span><span class="va">temp_md_norm</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
  <span class="va">tmp_p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">tmp_shap</span><span class="op">$</span><span class="va">p.value</span>, digits <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>
  <span class="va">tmp_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/eval.html">eval</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Logic.html">isTRUE</a></span><span class="op">(</span><span class="va">tmp_shap</span><span class="op">$</span><span class="va">p.value</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span>
  <span class="va">shap_results_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">shap_results_norm</span>, <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">i</span>, <span class="va">tmp_p</span>, <span class="va">tmp_res</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">shap_results_norm</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"parameter"</span>, <span class="st">"p-value"</span>, <span class="st">"tranform"</span><span class="op">)</span>
<span class="va">shap_results_norm</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"temp_md_norm"</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>And check if there are any parameters that are still significant for the normality test.</p>
<div class="cell">
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">shap_results</span><span class="op">$</span><span class="va">parameter</span><span class="op">[</span><span class="va">shap_results_norm</span><span class="op">$</span><span class="va">tranform</span> <span class="op">==</span> <span class="cn">TRUE</span><span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Al" "Fe"</code></pre>
</div>
</div>
<p>Ok. Looks like <code>bestNormalize</code> was unable to find a suitable transformation for <code>Al</code> and <code>Fe</code>. This is likely because there is very little variation in these metadata and/or there are too few significant digits.</p>
</section><section id="normalized-metadata" class="level2"><h2 class="anchored" data-anchor-id="normalized-metadata">Normalized Metadata</h2>
<p>Finally, here is a new summary table that includes all of the normalized data.</p>
<p></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,Sample_ID,ATAP,AST,H2O,N,P,Al,Ca,Fe,K,Mg,Mn,Na,TEB,ECEC,pH,minPO4,minNH4,minNO3,minTIN,NH4,NO3,PO4,DOC,DON,DOCN,micC,micN,micP,micCN,micCP,micNP,AG_ase,BG_ase,BP_ase,CE_ase,P_ase,N_ase,S_ase,XY_ase,AG_Q10,BG_Q10,BP_Q10,CE_Q10,P_Q10,N_Q10,S_Q10,XY_Q10,CO2,LP_ase,LP_Q10,PX_ase,PX_Q10,CUEcn,CUEcp,NUE,PUE,Tmin,SI,enzCN,enzCP,enzNP
P10_D00_010_C0E,26,-1.2815515655446008,0,0.26,0.32,1.6906379808432759,0.9674215661017014,-0.1580433278382755,1.38,21.43,0.07,0.13,0.9674215661017014,0.9674215661017014,6.28,4.08,4.75,0.8000754953787156,0.7279132908816445,3.44,4.53,9.45,0.5244005127080407,4.91,1.2815515655446008,334.93,72.64,64.23,-1.2815515655446008,5.54,1.17,0.28,0.5244005127080407,-0.9867272441186439,-0.596465321572654,-0.9674215661017014,-0.34069482708779547,4.54,-0.9674215661017014,1.54,-0.34069482708779547,-0.727913290881644,1.81,1.27,1.47,1.33,1.65,-0.1678940047881054,32.22,1.8,-0.1678940047881054,-0.9883442698430949,0.22,-1.118852519575083,0.87,1.2815515655446008,-2.46,1.29,0.5231,0.6607,1.263
P02_D00_010_C0A,26,-0.9674215661017014,0.34069482708779536,0.3367,0.43,0.4830911694988652,0.7279132908816445,-0.1580433278382755,0.69,8.16,0.04,0.11,0.1678940047881054,0.1678940047881054,6.18,1.42,3.92,0.22591025650060867,0.34069482708779536,4.81,3.13,7.58,1.2815515655446008,8.22,-1.8339146358159142,444.72,110.13,65.03,-1.8339146358159142,8.13,2.37,0.24,0,-0.5842724356363201,0.026362970450798936,-0.727913290881644,-0.1678940047881054,3.8,-0.2533471031357998,1.51,-1.5010859460440247,-1.8339146358159142,1.89,1.22,1.54,1.23,1.59,0,38.61,1.81,-0.727913290881644,1.103380971530638,0.22,0.5135967261758202,0.87,-0.6229257232100878,0.14,1.44,0.5024,0.6554,1.3044
P04_D00_010_C0B,26,-0.5244005127080407,0.5244005127080407,0.3033,0.33,-0.7245763834473804,0,0.6326473900721853,1.18,14.59,0.12,0.14,0,0,5.73,1.5,3.17,-0.07182374734884264,-0.08365173390712907,5.57,3.22,7.95,0.1678940047881054,6.32,-0.34069482708779547,440.6,86.19,81.45,0.34069482708779536,5.39,1.09,0.22,0.34069482708779536,-0.2086876724442108,0.026362970450798936,0.1678940047881054,0.1678940047881054,4.92,0.08365173390712907,1.51,0.34069482708779536,1.2815515655446008,2.3,1.37,1.85,1.39,1.76,-1.2815515655446008,29.51,2.14,-1.8339146358159142,-0.2852713152543998,0.25,-0.03628304204725309,0.85,0.2533471031357998,-2.63,1.25,0.555,0.6173,1.1123
P06_D00_010_C0C,26,-0.727913290881644,0.1678940047881054,0.3333,0.3,-0.7245763834473804,0.5244005127080407,-0.9488131062563203,0.57,21.32,0.06,0.14,0.7279132908816445,0.7279132908816445,5.93,0.75,1.83,-0.09130334823043947,-0.34069482708779547,2.92,2.27,7.04,0,5.42,-0.1678940047881054,390.66,72.61,85.7,0.5244005127080407,4.5,0.83,0.21,-0.9674215661017014,-0.7715211366560389,-0.9561259134170994,-1.2815515655446008,-1.8339146358159142,5.45,-1.8339146358159142,1.83,-1.5010859460440247,0.08365173390712907,2.36,1.24,1.6,1.47,1.63,-0.8416212335729142,27.78,1.85,-0.9674215661017014,0.9643858389761334,0.27,-1.118852519575083,0.83,1.2815515655446008,-0.13,1.42,0.4624,0.5556,1.2017
P08_D00_010_C0D,26,-1.8339146358159142,1.8339146358159142,0.2433,0.27,-0.7245763834473804,-1.8339146358159142,-0.9488131062563203,1.86,14.93,0.02,0.08,-0.9674215661017014,-0.9674215661017014,6.01,3.67,4.33,1.9025647125542418,1.8339146358159142,2.04,3.71,5.97,-0.34069482708779547,5.22,0.1678940047881054,277.37,56.55,33.4,-0.08365173390712907,11.55,2.45,0.15,-0.727913290881644,-1.3679875684700413,-1.2136146033270037,-1.8339146358159142,-0.5244005127080407,3.56,-1.2815515655446008,2.08,1.8339146358159142,1.8339146358159142,2,1.48,1.69,1.48,1.83,-1.8339146358159142,21.96,2.07,-1.2815515655446008,2.3788150796016017,0.24,0.7510455525468056,0.85,-0.6229257232100878,0,1.54,0.5058,0.6037,1.1935
P01_D00_010_W3A,29,0.34069482708779536,0.9674215661017014,0.4267,0.35,-0.7245763834473804,0.2533471031357998,-0.1580433278382755,1.72,9.98,0.04,0.15,-0.6229257232100878,-0.6229257232100878,6.01,1.33,3.67,1.0808166102377958,1.2815515655446008,8.03,4.67,9.52,1.8339146358159142,11.66,0,690.11,135.48,87.98,-0.34069482708779547,8.61,1.74,0.26,-0.34069482708779547,0.11137805798795644,-0.20010865255965368,0.34069482708779536,0.9674215661017014,6.08,0.34069482708779536,1.77,-0.1678940047881054,-1.1107716166367856,2.03,1.37,1.46,1.3,1.72,0.7279132908816445,52.8,1.76,0.34069482708779536,-0.41097347565699544,0.32,0.5135967261758202,0.77,-0.6229257232100878,0.07,1.51,0.4614,0.5911,1.2812
P03_D00_010_W3B,29,0.1678940047881054,1.2815515655446008,0.32,0.37,0.4830911694988652,-0.6229257232100878,-0.1580433278382755,0.85,16.51,0.06,0.12,-0.2533471031357998,-0.2533471031357998,5.86,2.83,3.33,0.9584263414487719,0.9674215661017014,3.11,2.57,8.66,0.34069482708779536,6.48,-0.727913290881644,364.62,66.4,70.68,0.1678940047881054,6.7,1.31,0.33,0.1678940047881054,-0.18723925852784215,0.4072439319990379,0,0,5.72,0.5244005127080407,2.48,-0.6229257232100878,-0.34069482708779547,1.99,1.27,1.54,1.27,1.72,-0.5244005127080407,32.1,2,0,0.5542982332136657,0.26,-0.35704519987585326,0.84,0.2533471031357998,-0.29,1.49,0.5565,0.6412,1.1522
P05_D00_010_W3C,29,-0.34069482708779547,-0.5244005127080407,0.3333,0.33,-0.7245763834473804,-1.1107716166367856,-0.9488131062563203,0.67,14.83,0.05,0,-1.5010859460440247,-1.5010859460440247,5.87,1.17,2.17,-1.0408484417728274,-0.9674215661017014,2.32,3.31,7.3,0.7279132908816445,7.03,0.34069482708779536,284.5,48.79,52.5,0.7279132908816445,8.33,1.39,0.34,-1.8339146358159142,-0.7622284163562623,-0.9213179743556038,-0.5244005127080407,-1.2815515655446008,3.2,-0.5244005127080407,1.48,-0.9674215661017014,-1.1107716166367856,1.88,1.28,1.51,1.3,1.82,-0.34069482708779547,24.11,1.79,-0.34069482708779547,0.20471700316792996,0.27,-0.35704519987585326,0.83,0.2533471031357998,0.81,1.77,0.4877,0.5596,1.1473
P07_D00_010_W3D,29,-0.1678940047881054,0.7279132908816445,0.4367,0.35,-0.7245763834473804,-0.2533471031357998,-0.1580433278382755,1.6,18.87,0.07,0.1,0.43072729929545733,0.43072729929545733,6.21,1,2.42,-0.01528939220036025,-0.08365173390712907,4.28,5.07,9.15,0.9674215661017014,8.47,-0.9674215661017014,396.42,73.73,93.13,-0.08365173390712907,4.66,0.98,0.32,-0.1678940047881054,0.08015670101341514,-0.022304364455938485,0.5244005127080407,0.34069482708779536,5.86,0.08365173390712907,1.72,0.1678940047881054,0.34069482708779536,1.76,1.28,1.36,1.3,1.63,0.1678940047881054,42.4,1.96,-0.5244005127080407,-0.9089968831763905,0.27,-1.5759351601918585,0.83,1.2815515655446008,0.65,1.61,0.4941,0.5762,1.1662
P09_D00_010_W3E,29,0,-0.1678940047881054,0.2833,0.36,1.6906379808432759,1.5010859460440247,2.2134758288296497,1.09,20.64,0.08,0.19,1.5010859460440247,1.5010859460440247,6.1,2.58,3.5,-0.9872400487609231,-0.5244005127080407,2.03,2.9,8.28,-0.727913290881644,5.56,0.7279132908816445,392.56,62.37,66.65,1.2815515655446008,7.34,1.19,0.2,-0.5244005127080407,-0.8377089615865957,-0.596465321572654,-0.34069482708779547,-0.9674215661017014,5.26,-0.2533471031357998,1.67,-0.6229257232100878,0.9674215661017014,1.81,1.35,1.48,1.32,1.84,0.5244005127080407,30.37,1.9,0.1678940047881054,0.4734076763117828,0.26,0.5135967261758202,0.84,-0.6229257232100878,-0.24,1.51,0.5112,0.6064,1.1862
P01_D00_010_W8A,34,0.7279132908816445,-0.727913290881644,0.4267,0.35,-0.7245763834473804,0.2533471031357998,-0.1580433278382755,1.72,9.98,0.04,0.15,-0.6229257232100878,-0.6229257232100878,6.01,0.12,1.65,-1.0309465208269855,-1.2815515655446008,4.08,2.65,8.55,-0.9674215661017014,4.2,-1.2815515655446008,469,84,55,0.9674215661017014,8.48,1.52,0.15,0.7279132908816445,1.0280573930846275,1.7118606979344075,0.7279132908816445,0.5244005127080407,6.7,1.8339146358159142,2.01,1.2815515655446008,0.5244005127080407,1.87,1.39,1.56,1.65,1.86,-0.8416212335729142,40.69,1.71,0.5244005127080407,-1.6113690973931636,0.23,-0.35704519987585326,0.87,0.2533471031357998,2.19,2.3,0.6456,0.7166,1.11
P03_D00_010_W8B,34,1.8339146358159142,-0.9674215661017014,0.32,0.37,0.4830911694988652,-0.6229257232100878,-0.1580433278382755,0.85,16.51,0.06,0.12,-0.2533471031357998,-0.2533471031357998,5.86,0.2,1.28,0.12556719919093765,0.1678940047881054,2.5,1.06,2.8,-1.2815515655446008,2.5,0.9674215661017014,117,20,11,1.8339146358159142,10.54,1.8,0.02,-1.2815515655446008,-0.304161649328782,-1.3321893466146788,-0.1678940047881054,-0.727913290881644,1.28,-0.727913290881644,1.49,0.7279132908816445,0.7279132908816445,1.9,1.51,1.75,1.39,1.89,1.8339146358159142,13.94,1.77,0.7279132908816445,-0.9089968831763905,0.29,2.41320231511534,0.81,-1.8339146358159142,0.63,1.72,0.5302,0.4918,0.9277
P05_D00_010_W8C,34,1.2815515655446008,-1.2815515655446008,0.3333,0.33,-0.7245763834473804,-1.1107716166367856,-0.9488131062563203,0.67,14.83,0.05,0,-1.5010859460440247,-1.5010859460440247,5.87,0.11,1.18,0.775699874530057,0.5244005127080407,3.91,2.3,4.25,-0.1678940047881054,4.3,-0.5244005127080407,440,97,56,-0.9674215661017014,7.91,1.74,0.15,1.2815515655446008,1.7005042424383616,0.8654378300303668,1.8339146358159142,0.7279132908816445,5.57,0.9674215661017014,1.73,0.9674215661017014,-0.34069482708779547,1.95,1.48,1.68,1.6,1.88,1.2815515655446008,32.65,1.67,0.9674215661017014,-0.3472769124477694,0.19,0.9678470731885745,0.9,-1.2815515655446008,0.94,2.08,0.6931,0.6412,0.9251
P07_D00_010_W8D,34,0.5244005127080407,-1.8339146358159142,0.4367,0.35,-0.7245763834473804,-0.2533471031357998,-0.1580433278382755,1.6,18.87,0.07,0.1,0.43072729929545733,0.43072729929545733,6.21,0.02,2.6,-0.8755787126825595,-0.727913290881644,8.19,6.39,5.62,-0.5244005127080407,3.7,0.5244005127080407,676,147,94,-0.5244005127080407,7.18,1.56,0.39,0.9674215661017014,1.6492337586914145,0.9174783618530372,0.9674215661017014,1.2815515655446008,10.68,0.7279132908816445,1.31,0,-0.34069482708779547,1.96,1.33,1.54,1.51,1.91,0.34069482708779536,57.13,1.4,1.2815515655446008,-0.1088879829267522,0.26,-0.7155432363140904,0.84,0.7279132908816445,1.48,1.81,0.5933,0.6628,1.1173
P09_D00_010_W8E,34,0.9674215661017014,-0.34069482708779547,0.2833,0.36,1.6906379808432759,1.5010859460440247,2.2134758288296497,1.09,20.64,0.08,0.19,1.5010859460440247,1.5010859460440247,6.1,0,0.92,-1.7560302780181911,-1.8339146358159142,6.5,2.76,6.08,-1.8339146358159142,1.6,1.8339146358159142,416,91,63,-0.727913290881644,6.6,1.44,0.37,1.8339146358159142,1.4412041899089638,1.8838447351568386,1.2815515655446008,1.8339146358159142,11.78,1.2815515655446008,1.93,0.5244005127080407,0.08365173390712907,2.11,1.38,1.46,1.5,1.76,0.9674215661017014,53.79,1.79,1.8339146358159142,-0.1088879829267522,0.24,-0.03628304204725309,0.85,-0.1678940047881054,0.96,1.82,0.6515,0.7218,1.1079
').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'normalized_metadata.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
</div>
<p><small><strong>(16S rRNA) Table 3 | Results of bestNormalize function applied to each parameter.</strong></small></p>
<div class="cell">
<div class="cell-output-display">

<div id="htmlwidget-2ae193cc8a21bcc729d6" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-2ae193cc8a21bcc729d6">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Sample_ID":["P10_D00_010_C0E","P02_D00_010_C0A","P04_D00_010_C0B","P06_D00_010_C0C","P08_D00_010_C0D","P01_D00_010_W3A","P03_D00_010_W3B","P05_D00_010_W3C","P07_D00_010_W3D","P09_D00_010_W3E","P01_D00_010_W8A","P03_D00_010_W8B","P05_D00_010_W8C","P07_D00_010_W8D","P09_D00_010_W8E"],"ATAP":[26,26,26,26,26,29,29,29,29,29,34,34,34,34,34],"AST":[-1.282,-0.967,-0.524,-0.728,-1.834,0.341,0.168,-0.341,-0.168,0,0.728,1.834,1.282,0.524,0.967],"H2O":[0,0.341,0.524,0.168,1.834,0.967,1.282,-0.524,0.728,-0.168,-0.728,-0.967,-1.282,-1.834,-0.341],"N":[0.26,0.337,0.303,0.333,0.243,0.427,0.32,0.333,0.437,0.283,0.427,0.32,0.333,0.437,0.283],"P":[0.32,0.43,0.33,0.3,0.27,0.35,0.37,0.33,0.35,0.36,0.35,0.37,0.33,0.35,0.36],"Al":[1.691,0.483,-0.725,-0.725,-0.725,-0.725,0.483,-0.725,-0.725,1.691,-0.725,0.483,-0.725,-0.725,1.691],"Ca":[0.967,0.728,0,0.524,-1.834,0.253,-0.623,-1.111,-0.253,1.501,0.253,-0.623,-1.111,-0.253,1.501],"Fe":[-0.158,-0.158,0.633,-0.949,-0.949,-0.158,-0.158,-0.949,-0.158,2.213,-0.158,-0.158,-0.949,-0.158,2.213],"K":[1.38,0.69,1.18,0.57,1.86,1.72,0.85,0.67,1.6,1.09,1.72,0.85,0.67,1.6,1.09],"Mg":[21.43,8.16,14.59,21.32,14.93,9.98,16.51,14.83,18.87,20.64,9.98,16.51,14.83,18.87,20.64],"Mn":[0.07,0.04,0.12,0.06,0.02,0.04,0.06,0.05,0.07,0.08,0.04,0.06,0.05,0.07,0.08],"Na":[0.13,0.11,0.14,0.14,0.08,0.15,0.12,0,0.1,0.19,0.15,0.12,0,0.1,0.19],"TEB":[0.967,0.168,0,0.728,-0.967,-0.623,-0.253,-1.501,0.431,1.501,-0.623,-0.253,-1.501,0.431,1.501],"ECEC":[0.967,0.168,0,0.728,-0.967,-0.623,-0.253,-1.501,0.431,1.501,-0.623,-0.253,-1.501,0.431,1.501],"pH":[6.28,6.18,5.73,5.93,6.01,6.01,5.86,5.87,6.21,6.1,6.01,5.86,5.87,6.21,6.1],"minPO4":[4.08,1.42,1.5,0.75,3.67,1.33,2.83,1.17,1,2.58,0.12,0.2,0.11,0.02,0],"minNH4":[4.75,3.92,3.17,1.83,4.33,3.67,3.33,2.17,2.42,3.5,1.65,1.28,1.18,2.6,0.92],"minNO3":[0.8,0.226,-0.072,-0.091,1.903,1.081,0.958,-1.041,-0.015,-0.987,-1.031,0.126,0.776,-0.876,-1.756],"minTIN":[0.728,0.341,-0.084,-0.341,1.834,1.282,0.967,-0.967,-0.084,-0.524,-1.282,0.168,0.524,-0.728,-1.834],"NH4":[3.44,4.81,5.57,2.92,2.04,8.03,3.11,2.32,4.28,2.03,4.08,2.5,3.91,8.19,6.5],"NO3":[4.53,3.13,3.22,2.27,3.71,4.67,2.57,3.31,5.07,2.9,2.65,1.06,2.3,6.39,2.76],"PO4":[9.45,7.58,7.95,7.04,5.97,9.52,8.66,7.3,9.15,8.28,8.55,2.8,4.25,5.62,6.08],"DOC":[0.524,1.282,0.168,0,-0.341,1.834,0.341,0.728,0.967,-0.728,-0.967,-1.282,-0.168,-0.524,-1.834],"DON":[4.91,8.22,6.32,5.42,5.22,11.66,6.48,7.03,8.47,5.56,4.2,2.5,4.3,3.7,1.6],"DOCN":[1.282,-1.834,-0.341,-0.168,0.168,0,-0.728,0.341,-0.967,0.728,-1.282,0.967,-0.524,0.524,1.834],"micC":[334.93,444.72,440.6,390.66,277.37,690.11,364.62,284.5,396.42,392.56,469,117,440,676,416],"micN":[72.64,110.13,86.19,72.61,56.55,135.48,66.4,48.79,73.73,62.37,84,20,97,147,91],"micP":[64.23,65.03,81.45,85.7,33.4,87.98,70.68,52.5,93.13,66.65,55,11,56,94,63],"micCN":[-1.282,-1.834,0.341,0.524,-0.084,-0.341,0.168,0.728,-0.084,1.282,0.967,1.834,-0.967,-0.524,-0.728],"micCP":[5.54,8.13,5.39,4.5,11.55,8.61,6.7,8.33,4.66,7.34,8.48,10.54,7.91,7.18,6.6],"micNP":[1.17,2.37,1.09,0.83,2.45,1.74,1.31,1.39,0.98,1.19,1.52,1.8,1.74,1.56,1.44],"AG_ase":[0.28,0.24,0.22,0.21,0.15,0.26,0.33,0.34,0.32,0.2,0.15,0.02,0.15,0.39,0.37],"BG_ase":[0.524,0,0.341,-0.967,-0.728,-0.341,0.168,-1.834,-0.168,-0.524,0.728,-1.282,1.282,0.967,1.834],"BP_ase":[-0.987,-0.584,-0.209,-0.772,-1.368,0.111,-0.187,-0.762,0.08,-0.838,1.028,-0.304,1.701,1.649,1.441],"CE_ase":[-0.596,0.026,0.026,-0.956,-1.214,-0.2,0.407,-0.921,-0.022,-0.596,1.712,-1.332,0.865,0.917,1.884],"P_ase":[-0.967,-0.728,0.168,-1.282,-1.834,0.341,0,-0.524,0.524,-0.341,0.728,-0.168,1.834,0.967,1.282],"N_ase":[-0.341,-0.168,0.168,-1.834,-0.524,0.967,0,-1.282,0.341,-0.967,0.524,-0.728,0.728,1.282,1.834],"S_ase":[4.54,3.8,4.92,5.45,3.56,6.08,5.72,3.2,5.86,5.26,6.7,1.28,5.57,10.68,11.78],"XY_ase":[-0.967,-0.253,0.084,-1.834,-1.282,0.341,0.524,-0.524,0.084,-0.253,1.834,-0.728,0.967,0.728,1.282],"AG_Q10":[1.54,1.51,1.51,1.83,2.08,1.77,2.48,1.48,1.72,1.67,2.01,1.49,1.73,1.31,1.93],"BG_Q10":[-0.341,-1.501,0.341,-1.501,1.834,-0.168,-0.623,-0.967,0.168,-0.623,1.282,0.728,0.967,0,0.524],"BP_Q10":[-0.728,-1.834,1.282,0.084,1.834,-1.111,-0.341,-1.111,0.341,0.967,0.524,0.728,-0.341,-0.341,0.084],"CE_Q10":[1.81,1.89,2.3,2.36,2,2.03,1.99,1.88,1.76,1.81,1.87,1.9,1.95,1.96,2.11],"P_Q10":[1.27,1.22,1.37,1.24,1.48,1.37,1.27,1.28,1.28,1.35,1.39,1.51,1.48,1.33,1.38],"N_Q10":[1.47,1.54,1.85,1.6,1.69,1.46,1.54,1.51,1.36,1.48,1.56,1.75,1.68,1.54,1.46],"S_Q10":[1.33,1.23,1.39,1.47,1.48,1.3,1.27,1.3,1.3,1.32,1.65,1.39,1.6,1.51,1.5],"XY_Q10":[1.65,1.59,1.76,1.63,1.83,1.72,1.72,1.82,1.63,1.84,1.86,1.89,1.88,1.91,1.76],"CO2":[-0.168,0,-1.282,-0.842,-1.834,0.728,-0.524,-0.341,0.168,0.524,-0.842,1.834,1.282,0.341,0.967],"LP_ase":[32.22,38.61,29.51,27.78,21.96,52.8,32.1,24.11,42.4,30.37,40.69,13.94,32.65,57.13,53.79],"LP_Q10":[1.8,1.81,2.14,1.85,2.07,1.76,2,1.79,1.96,1.9,1.71,1.77,1.67,1.4,1.79],"PX_ase":[-0.168,-0.728,-1.834,-0.967,-1.282,0.341,0,-0.341,-0.524,0.168,0.524,0.728,0.967,1.282,1.834],"PX_Q10":[-0.988,1.103,-0.285,0.964,2.379,-0.411,0.554,0.205,-0.909,0.473,-1.611,-0.909,-0.347,-0.109,-0.109],"CUEcn":[0.22,0.22,0.25,0.27,0.24,0.32,0.26,0.27,0.27,0.26,0.23,0.29,0.19,0.26,0.24],"CUEcp":[-1.119,0.514,-0.036,-1.119,0.751,0.514,-0.357,-0.357,-1.576,0.514,-0.357,2.413,0.968,-0.716,-0.036],"NUE":[0.87,0.87,0.85,0.83,0.85,0.77,0.84,0.83,0.83,0.84,0.87,0.81,0.9,0.84,0.85],"PUE":[1.282,-0.623,0.253,1.282,-0.623,-0.623,0.253,0.253,1.282,-0.623,0.253,-1.834,-1.282,0.728,-0.168],"Tmin":[-2.46,0.14,-2.63,-0.13,0,0.07,-0.29,0.81,0.65,-0.24,2.19,0.63,0.94,1.48,0.96],"SI":[1.29,1.44,1.25,1.42,1.54,1.51,1.49,1.77,1.61,1.51,2.3,1.72,2.08,1.81,1.82],"enzCN":[0.523,0.502,0.555,0.462,0.506,0.461,0.556,0.488,0.494,0.511,0.646,0.53,0.693,0.593,0.651],"enzCP":[0.661,0.655,0.617,0.556,0.604,0.591,0.641,0.56,0.576,0.606,0.717,0.492,0.641,0.663,0.722],"enzNP":[1.263,1.304,1.112,1.202,1.194,1.281,1.152,1.147,1.166,1.186,1.11,0.928,0.925,1.117,1.108]},"columns":[{"accessor":"Sample_ID","name":"Sample ID","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P10_D00_010_C0E","P02_D00_010_C0A","P04_D00_010_C0B","P06_D00_010_C0C","P08_D00_010_C0D","P01_D00_010_W3A","P03_D00_010_W3B","P05_D00_010_W3C","P07_D00_010_W3D","P09_D00_010_W3E","P01_D00_010_W8A","P03_D00_010_W8B","P05_D00_010_W8C","P07_D00_010_W8D","P09_D00_010_W8E"],"header":"Sample ID","align":"left","headerStyle":{"borderRight":"1px solid #eee","fontSize":"0.9em"},"footerStyle":{"fontWeight":"bold"},"minWidth":150,"sticky":"left","style":{"borderRight":"1px solid #eee"}},{"accessor":"ATAP","name":"ATAP","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["26","26","26","26","26","29","29","29","29","29","34","34","34","34","34"],"header":"ATAP","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"AST","name":"AST","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-1.282","-0.967","-0.524","-0.728","-1.834","0.341","0.168","-0.341","-0.168","0.0","0.728","1.834","1.282","0.524","0.967"],"header":"AST","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"H2O","name":"H2O","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.0","0.341","0.524","0.168","1.834","0.967","1.282","-0.524","0.728","-0.168","-0.728","-0.967","-1.282","-1.834","-0.341"],"header":"H2O","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"N","name":"N","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.26","0.337","0.303","0.333","0.243","0.427","0.32","0.333","0.437","0.283","0.427","0.32","0.333","0.437","0.283"],"header":"N","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"P","name":"P","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.32","0.43","0.33","0.3","0.27","0.35","0.37","0.33","0.35","0.36","0.35","0.37","0.33","0.35","0.36"],"header":"P","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Al","name":"Al","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.691","0.483","-0.725","-0.725","-0.725","-0.725","0.483","-0.725","-0.725","1.691","-0.725","0.483","-0.725","-0.725","1.691"],"header":"Al","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Ca","name":"Ca","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.967","0.728","0.0","0.524","-1.834","0.253","-0.623","-1.111","-0.253","1.501","0.253","-0.623","-1.111","-0.253","1.501"],"header":"Ca","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Fe","name":"Fe","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.158","-0.158","0.633","-0.949","-0.949","-0.158","-0.158","-0.949","-0.158","2.213","-0.158","-0.158","-0.949","-0.158","2.213"],"header":"Fe","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"K","name":"K","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.38","0.69","1.18","0.57","1.86","1.72","0.85","0.67","1.6","1.09","1.72","0.85","0.67","1.6","1.09"],"header":"K","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Mg","name":"Mg","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["21.43","8.16","14.59","21.32","14.93","9.98","16.51","14.83","18.87","20.64","9.98","16.51","14.83","18.87","20.64"],"header":"Mg","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Mn","name":"Mn","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.07","0.04","0.12","0.06","0.02","0.04","0.06","0.05","0.07","0.08","0.04","0.06","0.05","0.07","0.08"],"header":"Mn","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Na","name":"Na","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.13","0.11","0.14","0.14","0.08","0.15","0.12","0.0","0.1","0.19","0.15","0.12","0.0","0.1","0.19"],"header":"Na","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"TEB","name":"TEB","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.967","0.168","0.0","0.728","-0.967","-0.623","-0.253","-1.501","0.431","1.501","-0.623","-0.253","-1.501","0.431","1.501"],"header":"TEB","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"ECEC","name":"ECEC","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.967","0.168","0.0","0.728","-0.967","-0.623","-0.253","-1.501","0.431","1.501","-0.623","-0.253","-1.501","0.431","1.501"],"header":"ECEC","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"pH","name":"pH","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["6.28","6.18","5.73","5.93","6.01","6.01","5.86","5.87","6.21","6.1","6.01","5.86","5.87","6.21","6.1"],"header":"pH","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"minPO4","name":"minPO4","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["4.08","1.42","1.5","0.75","3.67","1.33","2.83","1.17","1.0","2.58","0.12","0.2","0.11","0.02","0.0"],"header":"minPO4","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"minNH4","name":"minNH4","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["4.75","3.92","3.17","1.83","4.33","3.67","3.33","2.17","2.42","3.5","1.65","1.28","1.18","2.6","0.92"],"header":"minNH4","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"minNO3","name":"minNO3","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.8","0.226","-0.072","-0.091","1.903","1.081","0.958","-1.041","-0.015","-0.987","-1.031","0.126","0.776","-0.876","-1.756"],"header":"minNO3","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"minTIN","name":"minTIN","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.728","0.341","-0.084","-0.341","1.834","1.282","0.967","-0.967","-0.084","-0.524","-1.282","0.168","0.524","-0.728","-1.834"],"header":"minTIN","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"NH4","name":"NH4","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["3.44","4.81","5.57","2.92","2.04","8.03","3.11","2.32","4.28","2.03","4.08","2.5","3.91","8.19","6.5"],"header":"NH4","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"NO3","name":"NO3","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["4.53","3.13","3.22","2.27","3.71","4.67","2.57","3.31","5.07","2.9","2.65","1.06","2.3","6.39","2.76"],"header":"NO3","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"PO4","name":"PO4","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["9.45","7.58","7.95","7.04","5.97","9.52","8.66","7.3","9.15","8.28","8.55","2.8","4.25","5.62","6.08"],"header":"PO4","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"DOC","name":"DOC","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.524","1.282","0.168","0.0","-0.341","1.834","0.341","0.728","0.967","-0.728","-0.967","-1.282","-0.168","-0.524","-1.834"],"header":"DOC","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"DON","name":"DON","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["4.91","8.22","6.32","5.42","5.22","11.66","6.48","7.03","8.47","5.56","4.2","2.5","4.3","3.7","1.6"],"header":"DON","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"DOCN","name":"DOCN","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.282","-1.834","-0.341","-0.168","0.168","0.0","-0.728","0.341","-0.967","0.728","-1.282","0.967","-0.524","0.524","1.834"],"header":"DOCN","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"micC","name":"micC","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["334.93","444.72","440.6","390.66","277.37","690.11","364.62","284.5","396.42","392.56","469.0","117.0","440.0","676.0","416.0"],"header":"micC","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"micN","name":"micN","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["72.64","110.13","86.19","72.61","56.55","135.48","66.4","48.79","73.73","62.37","84.0","20.0","97.0","147.0","91.0"],"header":"micN","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"micP","name":"micP","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["64.23","65.03","81.45","85.7","33.4","87.98","70.68","52.5","93.13","66.65","55.0","11.0","56.0","94.0","63.0"],"header":"micP","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"micCN","name":"micCN","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-1.282","-1.834","0.341","0.524","-0.084","-0.341","0.168","0.728","-0.084","1.282","0.967","1.834","-0.967","-0.524","-0.728"],"header":"micCN","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"micCP","name":"micCP","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["5.54","8.13","5.39","4.5","11.55","8.61","6.7","8.33","4.66","7.34","8.48","10.54","7.91","7.18","6.6"],"header":"micCP","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"micNP","name":"micNP","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.17","2.37","1.09","0.83","2.45","1.74","1.31","1.39","0.98","1.19","1.52","1.8","1.74","1.56","1.44"],"header":"micNP","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"AG_ase","name":"AG_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.28","0.24","0.22","0.21","0.15","0.26","0.33","0.34","0.32","0.2","0.15","0.02","0.15","0.39","0.37"],"header":"AG ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"BG_ase","name":"BG_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.524","0.0","0.341","-0.967","-0.728","-0.341","0.168","-1.834","-0.168","-0.524","0.728","-1.282","1.282","0.967","1.834"],"header":"BG ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"BP_ase","name":"BP_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.987","-0.584","-0.209","-0.772","-1.368","0.111","-0.187","-0.762","0.08","-0.838","1.028","-0.304","1.701","1.649","1.441"],"header":"BP ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"CE_ase","name":"CE_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.596","0.026","0.026","-0.956","-1.214","-0.2","0.407","-0.921","-0.022","-0.596","1.712","-1.332","0.865","0.917","1.884"],"header":"CE ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"P_ase","name":"P_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.967","-0.728","0.168","-1.282","-1.834","0.341","0.0","-0.524","0.524","-0.341","0.728","-0.168","1.834","0.967","1.282"],"header":"P ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"N_ase","name":"N_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.341","-0.168","0.168","-1.834","-0.524","0.967","0.0","-1.282","0.341","-0.967","0.524","-0.728","0.728","1.282","1.834"],"header":"N ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"S_ase","name":"S_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["4.54","3.8","4.92","5.45","3.56","6.08","5.72","3.2","5.86","5.26","6.7","1.28","5.57","10.68","11.78"],"header":"S ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"XY_ase","name":"XY_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.967","-0.253","0.084","-1.834","-1.282","0.341","0.524","-0.524","0.084","-0.253","1.834","-0.728","0.967","0.728","1.282"],"header":"XY ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"AG_Q10","name":"AG_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.54","1.51","1.51","1.83","2.08","1.77","2.48","1.48","1.72","1.67","2.01","1.49","1.73","1.31","1.93"],"header":"AG Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"BG_Q10","name":"BG_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.341","-1.501","0.341","-1.501","1.834","-0.168","-0.623","-0.967","0.168","-0.623","1.282","0.728","0.967","0.0","0.524"],"header":"BG Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"BP_Q10","name":"BP_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.728","-1.834","1.282","0.084","1.834","-1.111","-0.341","-1.111","0.341","0.967","0.524","0.728","-0.341","-0.341","0.084"],"header":"BP Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"CE_Q10","name":"CE_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.81","1.89","2.3","2.36","2.0","2.03","1.99","1.88","1.76","1.81","1.87","1.9","1.95","1.96","2.11"],"header":"CE Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"P_Q10","name":"P_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.27","1.22","1.37","1.24","1.48","1.37","1.27","1.28","1.28","1.35","1.39","1.51","1.48","1.33","1.38"],"header":"P Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"N_Q10","name":"N_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.47","1.54","1.85","1.6","1.69","1.46","1.54","1.51","1.36","1.48","1.56","1.75","1.68","1.54","1.46"],"header":"N Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"S_Q10","name":"S_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.33","1.23","1.39","1.47","1.48","1.3","1.27","1.3","1.3","1.32","1.65","1.39","1.6","1.51","1.5"],"header":"S Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"XY_Q10","name":"XY_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.65","1.59","1.76","1.63","1.83","1.72","1.72","1.82","1.63","1.84","1.86","1.89","1.88","1.91","1.76"],"header":"XY Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"CO2","name":"CO2","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.168","0.0","-1.282","-0.842","-1.834","0.728","-0.524","-0.341","0.168","0.524","-0.842","1.834","1.282","0.341","0.967"],"header":"CO2","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"LP_ase","name":"LP_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["32.22","38.61","29.51","27.78","21.96","52.8","32.1","24.11","42.4","30.37","40.69","13.94","32.65","57.13","53.79"],"header":"LP ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"LP_Q10","name":"LP_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.8","1.81","2.14","1.85","2.07","1.76","2.0","1.79","1.96","1.9","1.71","1.77","1.67","1.4","1.79"],"header":"LP Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"PX_ase","name":"PX_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.168","-0.728","-1.834","-0.967","-1.282","0.341","0.0","-0.341","-0.524","0.168","0.524","0.728","0.967","1.282","1.834"],"header":"PX ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"PX_Q10","name":"PX_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.988","1.103","-0.285","0.964","2.379","-0.411","0.554","0.205","-0.909","0.473","-1.611","-0.909","-0.347","-0.109","-0.109"],"header":"PX Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"CUEcn","name":"CUEcn","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.22","0.22","0.25","0.27","0.24","0.32","0.26","0.27","0.27","0.26","0.23","0.29","0.19","0.26","0.24"],"header":"CUEcn","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"CUEcp","name":"CUEcp","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-1.119","0.514","-0.036","-1.119","0.751","0.514","-0.357","-0.357","-1.576","0.514","-0.357","2.413","0.968","-0.716","-0.036"],"header":"CUEcp","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"NUE","name":"NUE","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.87","0.87","0.85","0.83","0.85","0.77","0.84","0.83","0.83","0.84","0.87","0.81","0.9","0.84","0.85"],"header":"NUE","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"PUE","name":"PUE","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.282","-0.623","0.253","1.282","-0.623","-0.623","0.253","0.253","1.282","-0.623","0.253","-1.834","-1.282","0.728","-0.168"],"header":"PUE","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Tmin","name":"Tmin","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-2.46","0.14","-2.63","-0.13","0.0","0.07","-0.29","0.81","0.65","-0.24","2.19","0.63","0.94","1.48","0.96"],"header":"Tmin","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"SI","name":"SI","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.29","1.44","1.25","1.42","1.54","1.51","1.49","1.77","1.61","1.51","2.3","1.72","2.08","1.81","1.82"],"header":"SI","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"enzCN","name":"enzCN","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.523","0.502","0.555","0.462","0.506","0.461","0.556","0.488","0.494","0.511","0.646","0.53","0.693","0.593","0.651"],"header":"enzCN","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"enzCP","name":"enzCP","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.661","0.655","0.617","0.556","0.604","0.591","0.641","0.56","0.576","0.606","0.717","0.492","0.641","0.663","0.722"],"header":"enzCP","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"enzNP","name":"enzNP","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.263","1.304","1.112","1.202","1.194","1.281","1.152","1.147","1.166","1.186","1.11","0.928","0.925","1.117","1.108"],"header":"enzNP","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,15],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"bb6f883cc66c59a8e1422935cd2a09dd"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="cell">

</div>
</section><section id="autocorrelation-tests" class="level2"><h2 class="anchored" data-anchor-id="autocorrelation-tests">Autocorrelation Tests</h2>
<p>Next, we test the metadata for autocorrelations. Do we do this on the original data or the transformed data? No idea, so lets do both.</p>
<section id="split-metadata" class="level3"><h3 class="anchored" data-anchor-id="split-metadata">Split Metadata</h3>
<p>We need to split the data into different groups.</p>
<ol type="A">
<li>Environmental and edaphic properties</li>
<li>Microbial functional responses</li>
<li>Temperature adaptation properties</li>
</ol>
<p>We first create lists of metadata parameters.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">div</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"PLOT"</span>, <span class="st">"TREAT"</span>, <span class="st">"TREAT_T"</span>, <span class="st">"PAIR"</span>, <span class="st">"Observed"</span>, <span class="st">"Shannon_exp"</span>, 
         <span class="st">"InvSimpson"</span>, <span class="st">"ATAP"</span><span class="op">)</span>

<span class="va">edaphic</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"AST"</span>, <span class="st">"H2O"</span>, <span class="st">"N"</span>, <span class="st">"P"</span>, <span class="st">"Al"</span>, 
             <span class="st">"Ca"</span>, <span class="st">"Fe"</span>, <span class="st">"K"</span>, <span class="st">"Mg"</span>, <span class="st">"Mn"</span>, <span class="st">"Na"</span>, <span class="st">"TEB"</span>, <span class="st">"ECEC"</span>, <span class="st">"pH"</span>, 
             <span class="st">"NH4"</span>, <span class="st">"NO3"</span>, <span class="st">"PO4"</span>, <span class="st">"DOC"</span>, <span class="st">"DON"</span>, <span class="st">"DOCN"</span><span class="op">)</span>

<span class="va">soil_funct</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"micC"</span>, <span class="st">"micN"</span>, <span class="st">"micP"</span>, <span class="st">"micCN"</span>, <span class="st">"micCP"</span>, <span class="st">"micNP"</span>, 
                <span class="st">"AG_ase"</span>, <span class="st">"BG_ase"</span>, <span class="st">"BP_ase"</span>, <span class="st">"CE_ase"</span>, <span class="st">"P_ase"</span>, <span class="st">"N_ase"</span>, 
                <span class="st">"S_ase"</span>, <span class="st">"XY_ase"</span>,  <span class="st">"LP_ase"</span>, <span class="st">"PX_ase"</span>, <span class="st">"CO2"</span>, 
                <span class="st">"enzCN"</span>, <span class="st">"enzCP"</span>, <span class="st">"enzNP"</span><span class="op">)</span>

<span class="va">temp_adapt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"AG_Q10"</span>, <span class="st">"BG_Q10"</span>, <span class="st">"BP_Q10"</span>, <span class="st">"CE_Q10"</span>, <span class="st">"P_Q10"</span>, <span class="st">"N_Q10"</span>, 
                <span class="st">"S_Q10"</span>, <span class="st">"XY_Q10"</span>, <span class="st">"LP_Q10"</span>, <span class="st">"PX_Q10"</span>, <span class="st">"CUEcn"</span>, <span class="st">"CUEcp"</span>, 
                <span class="st">"NUE"</span>,<span class="st">"PUE"</span>, <span class="st">"Tmin"</span>, <span class="st">"SI"</span><span class="op">)</span>

<span class="va">md_groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"edaphic"</span>, <span class="st">"soil_funct"</span>, <span class="st">"temp_adapt"</span><span class="op">)</span>

<span class="co"># NoT uSed: minPO4, minNH4, minNO3, minTIN</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>And then use the lists to split the data sets by metadata group. Here, we do this for the original metadata and the metadata after normalization.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">select_md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ssu18_select_mc"</span>, <span class="st">"ssu18_select_mc_norm"</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">select_md</span><span class="op">)</span> <span class="op">{</span>
   <span class="co">#tmp_get &lt;- get(purrr::map_chr(i, ~ paste0(i, "_mc")))</span>
   <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">get</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span>
   <span class="va">tmp_md_all</span> <span class="op">&lt;-</span> <span class="va">tmp_get</span><span class="op">$</span><span class="va">map_loaded</span>
   <span class="va">tmp_div</span> <span class="op">&lt;-</span> <span class="va">tmp_md_all</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="fu">all_of</span><span class="op">(</span><span class="va">div</span><span class="op">)</span><span class="op">)</span>
   <span class="va">tmp_div</span> <span class="op">&lt;-</span> <span class="va">tmp_div</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>
<span class="co">## edaphic      </span>
   <span class="va">tmp_sub_edaphic</span> <span class="op">&lt;-</span> <span class="va">tmp_md_all</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="fu">all_of</span><span class="op">(</span><span class="va">edaphic</span><span class="op">)</span><span class="op">)</span>
   <span class="va">tmp_sub_edaphic</span> <span class="op">&lt;-</span> <span class="va">tmp_sub_edaphic</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>
   <span class="va">tmp_edaphic</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_div</span>, <span class="va">tmp_sub_edaphic</span>, by <span class="op">=</span> <span class="st">"SampleID"</span><span class="op">)</span>
   <span class="va">tmp_edaphic</span> <span class="op">&lt;-</span> <span class="va">tmp_edaphic</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>
<span class="co">## soil_funct</span>
   <span class="va">tmp_sub_soil_funct</span> <span class="op">&lt;-</span> <span class="va">tmp_md_all</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="fu">all_of</span><span class="op">(</span><span class="va">soil_funct</span><span class="op">)</span><span class="op">)</span>
   <span class="va">tmp_sub_soil_funct</span> <span class="op">&lt;-</span> <span class="va">tmp_sub_soil_funct</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>
   <span class="va">tmp_soil_funct</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_div</span>, <span class="va">tmp_sub_soil_funct</span>, by <span class="op">=</span> <span class="st">"SampleID"</span><span class="op">)</span>
   <span class="va">tmp_soil_funct</span> <span class="op">&lt;-</span> <span class="va">tmp_soil_funct</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>      
<span class="co">## temp_adapt</span>
   <span class="va">tmp_sub_temp_adapt</span> <span class="op">&lt;-</span> <span class="va">tmp_md_all</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="fu">all_of</span><span class="op">(</span><span class="va">temp_adapt</span><span class="op">)</span><span class="op">)</span>
   <span class="va">tmp_sub_temp_adapt</span> <span class="op">&lt;-</span> <span class="va">tmp_sub_temp_adapt</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>
   <span class="va">tmp_temp_adapt</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_div</span>, <span class="va">tmp_sub_temp_adapt</span>, by <span class="op">=</span> <span class="st">"SampleID"</span><span class="op">)</span>
   <span class="va">tmp_temp_adapt</span> <span class="op">&lt;-</span> <span class="va">tmp_temp_adapt</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>      
<span class="co">## combine</span>
   <span class="va">tmp_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>data_loaded <span class="op">=</span> <span class="va">ssu18_select_mc</span><span class="op">$</span><span class="va">data_loaded</span>, 
                    map_loaded <span class="op">=</span> <span class="va">ssu18_select_mc</span><span class="op">$</span><span class="va">map_loaded</span>, 
                    taxonomy_loaded <span class="op">=</span> <span class="va">ssu18_select_mc</span><span class="op">$</span><span class="va">taxonomy_loaded</span>,
                    edaphic <span class="op">=</span> <span class="va">tmp_edaphic</span>, 
                    soil_funct <span class="op">=</span> <span class="va">tmp_soil_funct</span>, 
                    temp_adapt <span class="op">=</span> <span class="va">tmp_temp_adapt</span><span class="op">)</span>
   <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_split"</span><span class="op">)</span><span class="op">)</span>
   <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_name</span><span class="op">)</span>
   <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_list</span><span class="op">)</span>
   <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="generate-autocorrelation-plots" class="level3"><h3 class="anchored" data-anchor-id="generate-autocorrelation-plots">Generate Autocorrelation Plots</h3>
<p>A little housekeeping to get rid of parameters we dont need (e.g., plot number, pair, etc.).</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">edaphic_cor</span> <span class="op">&lt;-</span> <span class="va">ssu18_select_mc_split</span><span class="op">$</span><span class="va">edaphic</span>
<span class="va">edaphic_cor</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">edaphic_norm_cor</span> <span class="op">&lt;-</span> <span class="va">ssu18_select_mc_norm_split</span><span class="op">$</span><span class="va">edaphic</span>
<span class="va">edaphic_norm_cor</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>

<span class="va">soil_funct_cor</span> <span class="op">&lt;-</span> <span class="va">ssu18_select_mc_split</span><span class="op">$</span><span class="va">soil_funct</span>
<span class="va">soil_funct_cor</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">soil_funct_norm_cor</span> <span class="op">&lt;-</span> <span class="va">ssu18_select_mc_norm_split</span><span class="op">$</span><span class="va">soil_funct</span>
<span class="va">soil_funct_norm_cor</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>

<span class="va">temp_adapt_cor</span> <span class="op">&lt;-</span> <span class="va">ssu18_select_mc_split</span><span class="op">$</span><span class="va">temp_adapt</span>
<span class="va">temp_adapt_cor</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">temp_adapt_norm_cor</span> <span class="op">&lt;-</span> <span class="va">ssu18_select_mc_norm_split</span><span class="op">$</span><span class="va">temp_adapt</span>
<span class="va">temp_adapt_norm_cor</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>And finally the code to create the plots.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">objects</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_cor$"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">get</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span>
  <span class="va">tmp_cormat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span>
  <span class="va">tmp_melted_cormat</span> <span class="op">&lt;-</span> <span class="fu">reshape2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html">melt</a></span><span class="op">(</span><span class="va">tmp_cormat</span><span class="op">)</span>
  
  <span class="va">tmp_get_lower_tri</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tmp_cormat</span><span class="op">)</span><span class="op">{</span>
    <span class="va">tmp_cormat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lower.tri.html">upper.tri</a></span><span class="op">(</span><span class="va">tmp_cormat</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">tmp_cormat</span><span class="op">)</span>
    <span class="op">}</span>
  <span class="co"># Get upper triangle of the correlation matrix</span>
  <span class="va">tmp_get_upper_tri</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tmp_cormat</span><span class="op">)</span><span class="op">{</span>
    <span class="va">tmp_cormat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lower.tri.html">lower.tri</a></span><span class="op">(</span><span class="va">tmp_cormat</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">tmp_cormat</span><span class="op">)</span>
    <span class="op">}</span>
  <span class="va">tmp_upper_tri</span> <span class="op">&lt;-</span> <span class="fu">tmp_get_upper_tri</span><span class="op">(</span><span class="va">tmp_cormat</span><span class="op">)</span>
  <span class="va">tmp_melted_cormat</span> <span class="op">&lt;-</span> <span class="fu">reshape2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html">melt</a></span><span class="op">(</span><span class="va">tmp_upper_tri</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
  <span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp_melted_cormat</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Var1</span>, y <span class="op">=</span> <span class="va">Var2</span>, fill <span class="op">=</span> <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu">geom_tile</span><span class="op">(</span><span class="op">)</span>
  
  <span class="va">tmp_ggheatmap</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp_melted_cormat</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">Var2</span>, <span class="va">Var1</span>, fill <span class="op">=</span> <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">scale_fill_gradient2</span><span class="op">(</span>low <span class="op">=</span> <span class="st">"blue"</span>, high <span class="op">=</span> <span class="st">"red"</span>, mid <span class="op">=</span> <span class="st">"white"</span>, 
                         midpoint <span class="op">=</span> <span class="fl">0</span>, limit <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>,<span class="fl">1</span><span class="op">)</span>, space <span class="op">=</span> <span class="st">"Lab"</span>, 
                         name<span class="op">=</span><span class="st">"Pearson\nCorrelation"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, vjust <span class="op">=</span> <span class="fl">1</span>, 
                                     size <span class="op">=</span> <span class="fl">7</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,
          axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>vjust <span class="op">=</span> <span class="fl">1</span>, size <span class="op">=</span> <span class="fl">7</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu">coord_fixed</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu">geom_text</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">Var2</span>, <span class="va">Var1</span>, label <span class="op">=</span> <span class="va">value</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">1.75</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">theme</span><span class="op">(</span>
      axis.title.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
      axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
      panel.grid.major <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
      panel.border <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
      panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
      axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
      legend.justification <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span>,
      legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.6</span>, <span class="fl">0.7</span><span class="op">)</span>,
      legend.direction <span class="op">=</span> <span class="st">"horizontal"</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu">guides</span><span class="op">(</span>fill <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>barwidth <span class="op">=</span> <span class="fl">7</span>, barheight <span class="op">=</span> <span class="fl">1</span>,
                                   title.position <span class="op">=</span> <span class="st">"top"</span>, 
                                   title.hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_ggheatmap"</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_ggheatmap</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_name</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>  
<span class="fu"><a href="https://rdrr.io/r/base/ls.html">objects</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_ggheatmap"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section></section><section id="autocorrelation-plots" class="level2 page-columns page-full"><h2 class="anchored" data-anchor-id="autocorrelation-plots">Autocorrelation Plots</h2>
<p><br></p>
<div class="column-body">
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true">Edaphic properties</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false">Edaphic properties (normalized)</a></li>
</ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 1 | Results of autocorrelation analysis for non-normalized Environmental and edaphic parameters.</strong></small></p>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 2 | Results of autocorrelation analysis for normalized Environmental and edaphic parameters.</strong></small></p>
</div>
</div>
</div>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-2-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-1" role="tab" aria-controls="tabset-2-1" aria-selected="true">Functional responses</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-2" role="tab" aria-controls="tabset-2-2" aria-selected="false">Functional responses (normalized)</a></li>
</ul>
<div class="tab-content">
<div id="tabset-2-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-2-1-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-38-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 3 | Results of autocorrelation analysis for non-normalized Functional response parameters.</strong></small></p>
</div>
<div id="tabset-2-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-2-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-39-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 4 | Results of autocorrelation analysis for normalized Functional response parameters.</strong></small></p>
</div>
</div>
</div>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-3-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-1" role="tab" aria-controls="tabset-3-1" aria-selected="true">Temperature adaptation</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-3-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-2" role="tab" aria-controls="tabset-3-2" aria-selected="false">Temperature adaptation (normalized)</a></li>
</ul>
<div class="tab-content">
<div id="tabset-3-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-3-1-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-40-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 5 | Results of autocorrelation analysis for non-normalized Temperature adaptation parameters.</strong></small></p>
</div>
<div id="tabset-3-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-3-2-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-41-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 6 | Results of autocorrelation analysis for normalized Temperature adaptation parameters.</strong></small></p>
</div>
</div>
</div>
</div>
<div class="cell">

</div>
<p>Now we can remove parameters based on the autocorrelation analysis:</p>
<ol type="A">
<li>Environmental and edaphic properties: TEB, DON, Na, Al, Ca.</li>
<li>Microbial functional responses: micN, micNP, enzCN, enzCP, BP<sub>ase</sub>, CE<sub>ase</sub>, LP<sub>ase</sub>, N<sub>ase</sub>, P<sub>ase</sub>.</li>
<li>Temperature adaptation properties: NUE, PUE, SI.</li>
</ol>
<div class="cell">

</div>
<div class="cell">
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">edaphic_remove</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TEB"</span>, <span class="st">"DON"</span>, <span class="st">"Na"</span>, <span class="st">"Al"</span>, <span class="st">"Ca"</span><span class="op">)</span>
<span class="va">soil_funct_remove</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"micN"</span>, <span class="st">"micNP"</span>, <span class="st">"enzCN"</span>, <span class="st">"enzCP"</span>, <span class="st">"BP_ase"</span>, 
                       <span class="st">"CE_ase"</span>, <span class="st">"LP_ase"</span>, <span class="st">"N_ase"</span>, <span class="st">"P_ase"</span><span class="op">)</span>
<span class="va">temp_adapt_remove</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NUE"</span>, <span class="st">"PUE"</span>, <span class="st">"SI"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_df</span> <span class="op">&lt;-</span> <span class="va">ssu18_select_mc_split</span>
<span class="va">tmp_df</span><span class="op">$</span><span class="va">edaphic</span> <span class="op">&lt;-</span> <span class="va">tmp_df</span><span class="op">$</span><span class="va">edaphic</span><span class="op">[</span>, <span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_df</span><span class="op">$</span><span class="va">edaphic</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span>  <span class="va">edaphic_remove</span><span class="op">]</span>
<span class="va">tmp_df</span><span class="op">$</span><span class="va">soil_funct</span> <span class="op">&lt;-</span> <span class="va">tmp_df</span><span class="op">$</span><span class="va">soil_funct</span><span class="op">[</span>, <span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_df</span><span class="op">$</span><span class="va">soil_funct</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span>  <span class="va">soil_funct_remove</span><span class="op">]</span>
<span class="va">tmp_df</span><span class="op">$</span><span class="va">temp_adapt</span> <span class="op">&lt;-</span> <span class="va">tmp_df</span><span class="op">$</span><span class="va">temp_adapt</span><span class="op">[</span>, <span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_df</span><span class="op">$</span><span class="va">temp_adapt</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span>  <span class="va">temp_adapt_remove</span><span class="op">]</span>
<span class="va">ssu18_select_mc_split_no_ac</span> <span class="op">&lt;-</span> <span class="va">tmp_df</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>

<span class="va">tmp_df</span> <span class="op">&lt;-</span> <span class="va">ssu18_select_mc_norm_split</span>
<span class="va">tmp_df</span><span class="op">$</span><span class="va">edaphic</span> <span class="op">&lt;-</span> <span class="va">tmp_df</span><span class="op">$</span><span class="va">edaphic</span><span class="op">[</span>, <span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_df</span><span class="op">$</span><span class="va">edaphic</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span>  <span class="va">edaphic_remove</span><span class="op">]</span>
<span class="va">tmp_df</span><span class="op">$</span><span class="va">soil_funct</span> <span class="op">&lt;-</span> <span class="va">tmp_df</span><span class="op">$</span><span class="va">soil_funct</span><span class="op">[</span>, <span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_df</span><span class="op">$</span><span class="va">soil_funct</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span>  <span class="va">soil_funct_remove</span><span class="op">]</span>
<span class="va">tmp_df</span><span class="op">$</span><span class="va">temp_adapt</span> <span class="op">&lt;-</span> <span class="va">tmp_df</span><span class="op">$</span><span class="va">temp_adapt</span><span class="op">[</span>, <span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_df</span><span class="op">$</span><span class="va">temp_adapt</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span>  <span class="va">temp_adapt_remove</span><span class="op">]</span>
<span class="va">ssu18_select_mc_norm_split_no_ac</span> <span class="op">&lt;-</span> <span class="va">tmp_df</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="dissimilarity-correlation-tests" class="level2"><h2 class="anchored" data-anchor-id="dissimilarity-correlation-tests">Dissimilarity Correlation Tests</h2>
<p>Lets see if any on the metadata groups are significantly correlated with the community data. Basically, we create distance matrices for the community data and each metadata group and then run Mantel tests for all comparisons. For the community data we calculate Bray-Curtis distances for the community data and Euclidean distances for the metadata. We use the function <code>mantel.test</code> from the <code>ape</code> package and <code>mantel</code> from the <code>vegan</code> package for the analyses.</p>
<p>In summary, we test both <code>mantel.test</code> and <code>mantel</code> on Bray-Curtis distance community distances against Euclidean distances for each metadata group (<code>edaphic</code>, <code>soil_funct</code>, <code>temp_adapt</code>) <strong>a</strong>) before normalizing and before removing autocorrelated parameters, <strong>b</strong>) before normalizing and after removing autocorrelated parameters, <strong>c</strong>) after normalizing and before removing autocorrelated parameters, and <strong>d</strong>) after normalizing and after removing autocorrelated parameters.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">man_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ssu18_select_mc_split"</span>, <span class="st">"ssu18_select_mc_split_no_ac"</span>, 
            <span class="st">"ssu18_select_mc_norm_split"</span>, <span class="st">"ssu18_select_mc_norm_split_no_ac"</span><span class="op">)</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">man_df</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">get</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span>
  <span class="va">tmp_dm_otu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">vegdist</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">$</span><span class="va">data_loaded</span><span class="op">)</span>, 
                             method <span class="op">=</span> <span class="st">"bray"</span>, binary <span class="op">=</span> <span class="cn">FALSE</span>, 
                             diag <span class="op">=</span> <span class="cn">TRUE</span>, upper <span class="op">=</span> <span class="cn">TRUE</span>, na.rm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>
<span class="co"># EDAPHIC</span>
  <span class="va">tmp_dm_md_edaphic</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">vegdist</span><span class="op">(</span><span class="va">tmp_get</span><span class="op">$</span><span class="va">edaphic</span><span class="op">[</span>, <span class="fl">8</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">$</span><span class="va">edaphic</span><span class="op">)</span><span class="op">]</span>, 
                                    method <span class="op">=</span><span class="st">"euclidean"</span>, binary <span class="op">=</span> <span class="cn">FALSE</span>, 
                                    diag <span class="op">=</span> <span class="cn">TRUE</span>, upper <span class="op">=</span> <span class="cn">TRUE</span>, na.rm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>

  <span class="va">tmp_man1_edaphic</span> <span class="op">&lt;-</span> <span class="fu">mantel.test</span><span class="op">(</span><span class="va">tmp_dm_otu</span>, <span class="va">tmp_dm_md_edaphic</span>, nperm <span class="op">=</span> <span class="fl">999</span>, 
                             graph <span class="op">=</span> <span class="cn">FALSE</span>, alternative <span class="op">=</span> <span class="st">"two.sided"</span><span class="op">)</span>
  <span class="va">tmp_man2_edaphic</span> <span class="op">&lt;-</span> <span class="fu">mantel</span><span class="op">(</span><span class="va">tmp_dm_otu</span>, <span class="va">tmp_dm_md_edaphic</span>, permutations <span class="op">=</span> <span class="fl">999</span><span class="op">)</span>
<span class="co"># SOIL FUNCT</span>
  <span class="va">tmp_dm_md_soil_funct</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">vegdist</span><span class="op">(</span><span class="va">tmp_get</span><span class="op">$</span><span class="va">soil_funct</span><span class="op">[</span>, <span class="fl">8</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">$</span><span class="va">soil_funct</span><span class="op">)</span><span class="op">]</span>, 
                                       method <span class="op">=</span> <span class="st">"euclidean"</span>, binary <span class="op">=</span> <span class="cn">FALSE</span>, 
                                       diag <span class="op">=</span> <span class="cn">TRUE</span>, upper <span class="op">=</span> <span class="cn">TRUE</span>, na.rm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>

  <span class="va">tmp_man1_soil_funct</span> <span class="op">&lt;-</span> <span class="fu">mantel.test</span><span class="op">(</span><span class="va">tmp_dm_otu</span>, <span class="va">tmp_dm_md_soil_funct</span>, nperm <span class="op">=</span> <span class="fl">999</span>, 
                                graph <span class="op">=</span> <span class="cn">FALSE</span>, alternative <span class="op">=</span> <span class="st">"two.sided"</span><span class="op">)</span>
  <span class="va">tmp_man2_soil_funct</span> <span class="op">&lt;-</span> <span class="fu">mantel</span><span class="op">(</span><span class="va">tmp_dm_otu</span>, <span class="va">tmp_dm_md_soil_funct</span>, permutations <span class="op">=</span> <span class="fl">999</span><span class="op">)</span>

<span class="co"># TEMP ADAPT</span>
  <span class="va">tmp_dm_md_temp_adapt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">vegdist</span><span class="op">(</span><span class="va">tmp_get</span><span class="op">$</span><span class="va">temp_adapt</span><span class="op">[</span>, <span class="fl">8</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">$</span><span class="va">temp_adapt</span><span class="op">)</span><span class="op">]</span>, 
                                       method <span class="op">=</span> <span class="st">"euclidean"</span>, binary <span class="op">=</span> <span class="cn">FALSE</span>, 
                                       diag <span class="op">=</span> <span class="cn">TRUE</span>, upper <span class="op">=</span> <span class="cn">TRUE</span>, na.rm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>

  <span class="va">tmp_man1_temp_adapt</span> <span class="op">&lt;-</span> <span class="fu">mantel.test</span><span class="op">(</span><span class="va">tmp_dm_otu</span>, <span class="va">tmp_dm_md_temp_adapt</span>, nperm <span class="op">=</span> <span class="fl">999</span>, 
                                graph <span class="op">=</span> <span class="cn">FALSE</span>, alternative <span class="op">=</span> <span class="st">"two.sided"</span><span class="op">)</span>
  <span class="va">tmp_man2_temp_adapt</span> <span class="op">&lt;-</span> <span class="fu">mantel</span><span class="op">(</span><span class="va">tmp_dm_otu</span>, <span class="va">tmp_dm_md_temp_adapt</span>, permutations <span class="op">=</span> <span class="fl">999</span><span class="op">)</span>

  <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_mantel_tests"</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>edaphic_ape_man <span class="op">=</span> <span class="va">tmp_man1_edaphic</span>, 
                 edaphic_vegan_man <span class="op">=</span> <span class="va">tmp_man2_edaphic</span>,
                 soil_funct_ape_man <span class="op">=</span> <span class="va">tmp_man1_soil_funct</span>, 
                 soil_funct_vegan_man <span class="op">=</span> <span class="va">tmp_man2_soil_funct</span>,
                 temp_adapt_ape_man <span class="op">=</span> <span class="va">tmp_man1_temp_adapt</span>, 
                 temp_adapt_vegan_man <span class="op">=</span> <span class="va">tmp_man2_temp_adapt</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_df</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_name</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">

</div>
<section id="dissimilarity-correlation-results" class="level3"><h3 class="anchored" data-anchor-id="dissimilarity-correlation-results">Dissimilarity Correlation Results</h3>
<p><br></p>
<p><small><strong>(16S rRNA) Table 4 | Summary of Dissimilarity Correlation Tests using <code>mantel.test</code> from the ape package and <code>mantel</code> from the vegan. P-values in red indicate significance (p-value &lt; 0.05)</strong></small></p>
<div class="cell">
<div class="cell-output-display">

<div id="htmlwidget-f4461daff4b1e7125a7f" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f4461daff4b1e7125a7f">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"normalized?":["no","no","yes","yes"],"AC removed?":["no","yes","no","yes"],"edaphic_ape":[0.375,0.399,0.003,0.007],"edaphic_vegan":[0.194,0.17,0.002,0.002],"soil_funct_ape":[0.059,0.031,0.537,0.541],"soil_funct_vegan":[0.019,0.02,0.25,0.288],"temp_adapt_ape":[0.001,0.001,0.001,0.001],"temp_adapt_vegan":[0.001,0.001,0.001,0.001]},"columns":[{"accessor":"normalized?","name":"normalized?","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["no","no","yes","yes"],"header":"normalized?","minWidth":70,"align":"left","style":{"borderRight":"5px solid #eee"},"headerStyle":{"borderRight":"5px solid #eee"}},{"accessor":"AC removed?","name":"AC removed?","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["no","yes","no","yes"],"header":"AC removed?","minWidth":70,"align":"left","style":{"borderRight":"5px solid #eee"},"headerStyle":{"borderRight":"5px solid #eee"}},{"accessor":"edaphic_ape","name":"ape","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.375","0.399","0.003","0.007"],"header":"ape","minWidth":50,"align":"center","style":[{"color":"#2271B2"},{"color":"#2271B2"},{"color":"#B22271"},{"color":"#B22271"}]},{"accessor":"edaphic_vegan","name":"vegan","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.194","0.170","0.002","0.002"],"header":"vegan","minWidth":50,"align":"center","style":[{"color":"#2271B2"},{"color":"#2271B2"},{"color":"#B22271"},{"color":"#B22271"}]},{"accessor":"soil_funct_ape","name":"ape","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.059","0.031","0.537","0.541"],"header":"ape","minWidth":50,"align":"center","style":[{"color":"#2271B2"},{"color":"#B22271"},{"color":"#2271B2"},{"color":"#2271B2"}]},{"accessor":"soil_funct_vegan","name":"vegan","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.019","0.020","0.250","0.288"],"header":"vegan","minWidth":50,"align":"center","style":[{"color":"#B22271"},{"color":"#B22271"},{"color":"#2271B2"},{"color":"#2271B2"}]},{"accessor":"temp_adapt_ape","name":"ape","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.001","0.001","0.001","0.001"],"header":"ape","minWidth":50,"align":"center","style":[{"color":"#B22271"},{"color":"#B22271"},{"color":"#B22271"},{"color":"#B22271"}]},{"accessor":"temp_adapt_vegan","name":"vegan","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.001","0.001","0.001","0.001"],"header":"vegan","minWidth":50,"align":"center","style":[{"color":"#B22271"},{"color":"#B22271"},{"color":"#B22271"},{"color":"#B22271"}]}],"columnGroups":[{"name":"Environmental & edaphic","headerStyle":{"fontSize":"1.2em"},"columns":["edaphic_ape","edaphic_vegan"]},{"name":"Functional responses","headerStyle":{"fontSize":"1.2em"},"columns":["soil_funct_ape","soil_funct_vegan"]},{"name":"Temperature adaptation","headerStyle":{"fontSize":"1.2em"},"columns":["temp_adapt_ape","temp_adapt_vegan"]}],"defaultPageSize":15,"showPageSizeOptions":true,"pageSizeOptions":[5,10,4],"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"nowrap":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"47bcd92150c399e91c716485760465a6"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>Moving on.</p>
</section></section><section id="best-subset-of-variables" class="level2"><h2 class="anchored" data-anchor-id="best-subset-of-variables">Best Subset of Variables</h2>
<p>Now we want to know which of the metadata parameters are the most strongly correlated with the community data. For this we use the <code>bioenv</code> function from the <code>vegan</code> package. <code>bioenv</code><em>Best Subset of Environmental Variables with Maximum (Rank) Correlation with Community Dissimilarities</em>finds the best subset of environmental variables, so that the Euclidean distances of scaled environmental variables have the maximum (rank) correlation with community dissimilarities.</p>
<p>Next we will use the metadata set where autocorrelated parameters were removed and the remainder of the parameters were normalized (where applicable based on the Shapiro tests).</p>
<p>We run <code>bioenv</code> against the three groups of metadata parameters. We then run <code>bioenv</code> again, but this time against the individual parameters identified as significantly correlated.</p>
<section id="edaphic-properties-1" class="level3"><h3 class="anchored" data-anchor-id="edaphic-properties-1">Edaphic Properties</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_comm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="va">ssu18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">data_loaded</span><span class="op">)</span><span class="op">)</span>
<span class="va">tmp_env</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">ssu18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">edaphic</span><span class="op">)</span>
<span class="va">tmp_env</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>

<span class="va">edaphic_bioenv</span> <span class="op">&lt;-</span> <span class="fu">bioenv</span><span class="op">(</span><span class="fu">wisconsin</span><span class="op">(</span><span class="va">tmp_comm</span><span class="op">)</span>, <span class="va">tmp_env</span>, 
                         method <span class="op">=</span> <span class="st">"spearman"</span>, index <span class="op">=</span> <span class="st">"bray"</span>, 
                         upto <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_env</span><span class="op">)</span>, metric <span class="op">=</span> <span class="st">"euclidean"</span><span class="op">)</span>

<span class="va">bioenv_list</span> <span class="op">&lt;-</span> <span class="va">edaphic_bioenv</span><span class="op">$</span><span class="va">models</span><span class="op">[[</span><span class="va">edaphic_bioenv</span><span class="op">$</span><span class="va">whichbest</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">best</span>
<span class="va">bioenv_best</span> <span class="op">&lt;-</span> <span class="fu">bioenvdist</span><span class="op">(</span><span class="va">edaphic_bioenv</span>, which <span class="op">=</span> <span class="st">"best"</span><span class="op">)</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">bioenv_list</span><span class="op">)</span> <span class="op">{</span>
       <span class="va">tmp_dp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">edaphic_bioenv</span><span class="op">$</span><span class="va">x</span><span class="op">)</span>
       <span class="va">tmp_md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">vegdist</span><span class="op">(</span><span class="va">tmp_dp</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, 
                                   method <span class="op">=</span> <span class="st">"euclidean"</span>, binary <span class="op">=</span> <span class="cn">FALSE</span>, 
                                   diag <span class="op">=</span> <span class="cn">TRUE</span>, upper <span class="op">=</span> <span class="cn">TRUE</span>, na.rm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>
       <span class="va">tmp_man</span> <span class="op">&lt;-</span> <span class="fu">mantel</span><span class="op">(</span><span class="va">bioenv_best</span>, <span class="va">tmp_md</span>, 
                         permutations <span class="op">=</span> <span class="fl">999</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span>
       <span class="va">tmp_md_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_dp</span><span class="op">)</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>
       <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">tmp_md_name</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_bioenv_mantel_test"</span><span class="op">)</span><span class="op">)</span>
       <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_man</span><span class="op">)</span>
       <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/ls.html">objects</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_bioenv_mantel_test"</span><span class="op">)</span>
<span class="va">edaphic_bioenv_ind_mantel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>AST <span class="op">=</span> <span class="va">AST_bioenv_mantel_test</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_bioenv_mantel_test"</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
bioenv(comm = wisconsin(tmp_comm), env = tmp_env, method = "spearman",      index = "bray", upto = ncol(tmp_env), metric = "euclidean") 

Subset of environmental variables with best correlation to community data.

Correlations:    spearman 
Dissimilarities: bray 
Metric:          euclidean 

Best model has 1 parameters (max. 15 allowed):
AST
with correlation  0.6808861 </code></pre>
</div>
</div>
<details><summary>
Show the results of individual edaphic metadata Mantel tests
</summary><div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>$AST

Mantel statistic based on Spearman's rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = "spearman",      permutations = 999) 

Mantel statistic r:     1 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.177 0.226 0.271 0.326 
Permutation: free
Number of permutations: 999</code></pre>
</div>
</div>
</details><p><code>bioenv</code> found the following edaphic properties significantly correlated with the community data: <strong>AST</strong></p>
</section><section id="soil-functional-response" class="level3"><h3 class="anchored" data-anchor-id="soil-functional-response">Soil Functional Response</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_comm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="va">ssu18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">data_loaded</span><span class="op">)</span><span class="op">)</span>
<span class="va">tmp_env</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">ssu18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">soil_funct</span><span class="op">)</span>
<span class="va">tmp_env</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>

<span class="va">soil_funct_bioenv</span> <span class="op">&lt;-</span> <span class="fu">bioenv</span><span class="op">(</span><span class="fu">wisconsin</span><span class="op">(</span><span class="va">tmp_comm</span><span class="op">)</span>, <span class="va">tmp_env</span>, 
                         method <span class="op">=</span> <span class="st">"spearman"</span>, index <span class="op">=</span> <span class="st">"bray"</span>, 
                         upto <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_env</span><span class="op">)</span>, metric <span class="op">=</span> <span class="st">"euclidean"</span><span class="op">)</span>

<span class="va">bioenv_list</span> <span class="op">&lt;-</span> <span class="va">soil_funct_bioenv</span><span class="op">$</span><span class="va">models</span><span class="op">[[</span><span class="va">soil_funct_bioenv</span><span class="op">$</span><span class="va">whichbest</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">best</span>
<span class="va">bioenv_best</span> <span class="op">&lt;-</span> <span class="fu">bioenvdist</span><span class="op">(</span><span class="va">soil_funct_bioenv</span>, which <span class="op">=</span> <span class="st">"best"</span><span class="op">)</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">bioenv_list</span><span class="op">)</span> <span class="op">{</span>
       <span class="va">tmp_dp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">soil_funct_bioenv</span><span class="op">$</span><span class="va">x</span><span class="op">)</span>
       <span class="va">tmp_md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">vegdist</span><span class="op">(</span><span class="va">tmp_dp</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, 
                                   method <span class="op">=</span> <span class="st">"euclidean"</span>, binary <span class="op">=</span> <span class="cn">FALSE</span>, 
                                   diag <span class="op">=</span> <span class="cn">TRUE</span>, upper <span class="op">=</span> <span class="cn">TRUE</span>, na.rm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>
       <span class="va">tmp_man</span> <span class="op">&lt;-</span> <span class="fu">mantel</span><span class="op">(</span><span class="va">bioenv_best</span>, <span class="va">tmp_md</span>, 
                         permutations <span class="op">=</span> <span class="fl">999</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span>
       <span class="va">tmp_md_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_dp</span><span class="op">)</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>
       <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">tmp_md_name</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_bioenv_mantel_test"</span><span class="op">)</span><span class="op">)</span>
       <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_man</span><span class="op">)</span>
       <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/ls.html">objects</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_bioenv_mantel_test"</span><span class="op">)</span>

<span class="va">soil_funct_bioenv_ind_mantel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>AG_ase <span class="op">=</span> <span class="va">AG_ase_bioenv_mantel_test</span>, 
                                     enzNP <span class="op">=</span> <span class="va">enzNP_bioenv_mantel_test</span>, 
                                     S_ase <span class="op">=</span> <span class="va">S_ase_bioenv_mantel_test</span>, 
                                     PX_ase <span class="op">=</span> <span class="va">PX_ase_bioenv_mantel_test</span>,
                                     XY_ase <span class="op">=</span> <span class="va">XY_ase_bioenv_mantel_test</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_bioenv_mantel_test"</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
bioenv(comm = wisconsin(tmp_comm), env = tmp_env, method = "spearman",      index = "bray", upto = ncol(tmp_env), metric = "euclidean") 

Subset of environmental variables with best correlation to community data.

Correlations:    spearman 
Dissimilarities: bray 
Metric:          euclidean 

Best model has 5 parameters (max. 11 allowed):
AG_ase S_ase XY_ase PX_ase enzNP
with correlation  0.7727866 </code></pre>
</div>
</div>
<details><summary>
Show the results of individual functional response metadata Mantel tests
</summary><div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>$AG_ase

Mantel statistic based on Spearman's rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = "spearman",      permutations = 999) 

Mantel statistic r: 0.5588 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.186 0.235 0.281 0.330 
Permutation: free
Number of permutations: 999


$enzNP

Mantel statistic based on Spearman's rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = "spearman",      permutations = 999) 

Mantel statistic r: 0.4623 
      Significance: 0.006 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.219 0.280 0.336 0.413 
Permutation: free
Number of permutations: 999


$S_ase

Mantel statistic based on Spearman's rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = "spearman",      permutations = 999) 

Mantel statistic r: 0.6139 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.223 0.305 0.384 0.466 
Permutation: free
Number of permutations: 999


$PX_ase

Mantel statistic based on Spearman's rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = "spearman",      permutations = 999) 

Mantel statistic r: 0.6115 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.176 0.222 0.289 0.346 
Permutation: free
Number of permutations: 999


$XY_ase

Mantel statistic based on Spearman's rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = "spearman",      permutations = 999) 

Mantel statistic r: 0.456 
      Significance: 0.002 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.189 0.258 0.304 0.347 
Permutation: free
Number of permutations: 999</code></pre>
</div>
</div>
</details><p><code>bioenv</code> found the following soil functions significantly correlated with the community data: <strong>AG_ase, enzNP, S_ase, PX_ase, XY_ase</strong></p>
</section><section id="temperature-adaptation-1" class="level3"><h3 class="anchored" data-anchor-id="temperature-adaptation-1">Temperature Adaptation</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_comm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="va">ssu18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">data_loaded</span><span class="op">)</span><span class="op">)</span>
<span class="va">tmp_env</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">ssu18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">temp_adapt</span><span class="op">)</span>
<span class="va">tmp_env</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>

<span class="va">temp_adapt_bioenv</span> <span class="op">&lt;-</span> <span class="fu">bioenv</span><span class="op">(</span><span class="fu">wisconsin</span><span class="op">(</span><span class="va">tmp_comm</span><span class="op">)</span>, <span class="va">tmp_env</span>, 
                         method <span class="op">=</span> <span class="st">"spearman"</span>, index <span class="op">=</span> <span class="st">"bray"</span>, 
                         upto <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_env</span><span class="op">)</span>, metric <span class="op">=</span> <span class="st">"euclidean"</span><span class="op">)</span>

<span class="va">bioenv_list</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_bioenv</span><span class="op">$</span><span class="va">models</span><span class="op">[[</span><span class="va">temp_adapt_bioenv</span><span class="op">$</span><span class="va">whichbest</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">best</span>
<span class="va">bioenv_best</span> <span class="op">&lt;-</span> <span class="fu">bioenvdist</span><span class="op">(</span><span class="va">temp_adapt_bioenv</span>, which <span class="op">=</span> <span class="st">"best"</span><span class="op">)</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">bioenv_list</span><span class="op">)</span> <span class="op">{</span>
       <span class="va">tmp_dp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">temp_adapt_bioenv</span><span class="op">$</span><span class="va">x</span><span class="op">)</span>
       <span class="va">tmp_md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">vegdist</span><span class="op">(</span><span class="va">tmp_dp</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, 
                                   method <span class="op">=</span> <span class="st">"euclidean"</span>, binary <span class="op">=</span> <span class="cn">FALSE</span>, 
                                   diag <span class="op">=</span> <span class="cn">TRUE</span>, upper <span class="op">=</span> <span class="cn">TRUE</span>, na.rm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>
       <span class="va">tmp_man</span> <span class="op">&lt;-</span> <span class="fu">mantel</span><span class="op">(</span><span class="va">bioenv_best</span>, <span class="va">tmp_md</span>, 
                         permutations <span class="op">=</span> <span class="fl">999</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span>
       <span class="va">tmp_md_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_dp</span><span class="op">)</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>
       <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">tmp_md_name</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_bioenv_mantel_test"</span><span class="op">)</span><span class="op">)</span>
       <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_man</span><span class="op">)</span>
       <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/ls.html">objects</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_bioenv_mantel_test"</span><span class="op">)</span>

<span class="va">temp_adapt_bioenv_ind_mantel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>CUEcp <span class="op">=</span> <span class="va">CUEcp_bioenv_mantel_test</span>, 
                                     LP_Q10 <span class="op">=</span> <span class="va">LP_Q10_bioenv_mantel_test</span>, 
                                     P_Q10 <span class="op">=</span> <span class="va">P_Q10_bioenv_mantel_test</span>, 
                                     S_Q10 <span class="op">=</span> <span class="va">S_Q10_bioenv_mantel_test</span>, 
                                     Tmin <span class="op">=</span> <span class="va">Tmin_bioenv_mantel_test</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_bioenv_mantel_test"</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
bioenv(comm = wisconsin(tmp_comm), env = tmp_env, method = "spearman",      index = "bray", upto = ncol(tmp_env), metric = "euclidean") 

Subset of environmental variables with best correlation to community data.

Correlations:    spearman 
Dissimilarities: bray 
Metric:          euclidean 

Best model has 5 parameters (max. 13 allowed):
P_Q10 S_Q10 LP_Q10 CUEcp Tmin
with correlation  0.4237093 </code></pre>
</div>
</div>
<details><summary>
Show the results of individual temperature adaptation metadata Mantel tests
</summary><div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>$CUEcp

Mantel statistic based on Spearman's rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = "spearman",      permutations = 999) 

Mantel statistic r: 0.325 
      Significance: 0.013 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.188 0.226 0.280 0.331 
Permutation: free
Number of permutations: 999


$LP_Q10

Mantel statistic based on Spearman's rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = "spearman",      permutations = 999) 

Mantel statistic r: 0.3772 
      Significance: 0.005 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.203 0.272 0.306 0.351 
Permutation: free
Number of permutations: 999


$P_Q10

Mantel statistic based on Spearman's rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = "spearman",      permutations = 999) 

Mantel statistic r: 0.5177 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.158 0.211 0.261 0.309 
Permutation: free
Number of permutations: 999


$S_Q10

Mantel statistic based on Spearman's rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = "spearman",      permutations = 999) 

Mantel statistic r: 0.4395 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.159 0.205 0.252 0.301 
Permutation: free
Number of permutations: 999


$Tmin

Mantel statistic based on Spearman's rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = "spearman",      permutations = 999) 

Mantel statistic r: 0.4044 
      Significance: 0.005 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.207 0.267 0.305 0.359 
Permutation: free
Number of permutations: 999</code></pre>
</div>
</div>
</details><p><code>bioenv</code> found the following temperature adaptations significantly correlated with the community data: <strong>CUEcp, LP_Q10, P_Q10, S_Q10, Tmin</strong></p>
</section></section><section id="distance-based-redundancy" class="level2"><h2 class="anchored" data-anchor-id="distance-based-redundancy">Distance-based Redundancy</h2>
<p>Now we turn our attention to distance-based redundancy analysis (dbRDA), an ordination method similar to Redundancy Analysis (rda) but it allows non-Euclidean dissimilarity indices, such as Manhattan or BrayCurtis distance. For this, we use <code>capscale</code> from the <code>vegan</code> package. <code>capscale</code> is a constrained versions of metric scaling (principal coordinates analysis), which are based on the Euclidean distance but can be used, and are more useful, with other dissimilarity measures. The functions can also perform unconstrained principal coordinates analysis, optionally using extended dissimilarities.</p>
<p>For each of the three metadata subsets, we perform the following steps:</p>
<ol type="1">
<li>Run <code>rankindex</code> to compare metadata and community dissimilarity indices for gradient detection. This will help us select the best dissimilarity metric to use.</li>
<li>Run <code>capscale</code> for distance-based redundancy analysis.</li>
<li>Run <code>envfit</code> to fit environmental parameters onto the ordination. This function basically calculates correlation scores between the metadata parameters and the ordination axes.</li>
<li>Select metadata parameters significant for <code>bioenv</code> (see above) and/or <code>envfit</code> analyses.</li>
<li>Run <code>envfit</code> on ASVs.</li>
<li>Plot the ordination and vector overlays.</li>
</ol>
<section id="edaphic-properties-2" class="level3"><h3 class="anchored" data-anchor-id="edaphic-properties-2">Edaphic Properties</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_md</span> <span class="op">&lt;-</span> <span class="va">ssu18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">edaphic</span>
<span class="va">tmp_md</span><span class="op">$</span><span class="va">TREAT_T</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">tmp_md</span><span class="op">$</span><span class="va">TREAT_T</span><span class="op">)</span>
<span class="va">tmp_comm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="va">ssu18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">data_loaded</span><span class="op">)</span><span class="op">)</span>
<span class="va">edaphic_rank</span> <span class="op">&lt;-</span> <span class="fu">rankindex</span><span class="op">(</span><span class="va">tmp_md</span><span class="op">[</span>, <span class="fl">8</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_md</span><span class="op">)</span><span class="op">]</span>, <span class="va">tmp_comm</span>, 
          indices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"euc"</span>, <span class="st">"man"</span>, <span class="st">"gow"</span>,<span class="st">"bra"</span>, <span class="st">"kul"</span><span class="op">)</span>, 
          stepacross <span class="op">=</span> <span class="cn">FALSE</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>      euc       man       gow       bra       kul 
0.3171263 0.4336927 0.1028302 0.4336927 0.4336927 </code></pre>
</div>
</div>
<p>Lets run <code>capscale</code> using Bray-Curtis. Note, we have 15 metadata parameters in this group but, for some reason, <code>capscale</code> only works with 13 parameters. This may have to do with degrees of freedom?</p>
<ul>
<li>Starting properties: AST, H2O, N, P, Al, Ca, Fe, K, Mg, Mn, Na, TEB, ECEC, pH, NH4, NO3, PO4, DOC, DON, DOCN</li>
<li>Autocorrelated removed: TEB, DON, Na, Al, Ca</li>
<li>Remove for capscale: Mg, Mn</li>
</ul>
<div class="cell">

</div>
<div class="cell">
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">edaphic_cap</span> <span class="op">&lt;-</span> <span class="fu">capscale</span><span class="op">(</span><span class="va">tmp_comm</span> <span class="op">~</span> <span class="va">AST</span> <span class="op">+</span> <span class="va">H2O</span> <span class="op">+</span> <span class="va">N</span> <span class="op">+</span> <span class="va">P</span> <span class="op">+</span> <span class="va">Fe</span> <span class="op">+</span> <span class="va">K</span> <span class="op">+</span> <span class="va">ECEC</span> <span class="op">+</span> 
                                   <span class="va">pH</span> <span class="op">+</span> <span class="va">NH4</span> <span class="op">+</span> <span class="va">NO3</span> <span class="op">+</span> <span class="va">PO4</span> <span class="op">+</span> <span class="va">DOC</span> <span class="op">+</span> <span class="va">DOCN</span>, 
                        <span class="va">tmp_md</span>, dist <span class="op">=</span> <span class="st">"bray"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">tmp_md</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Call: capscale(formula = tmp_comm ~ AST + H2O + N + P + Fe + K + ECEC +
pH + NH4 + NO3 + PO4 + DOC + DOCN, data = tmp_md, distance = "bray")

              Inertia Proportion Rank
Total         1.67890    1.00000     
Constrained   1.63037    0.97110   13
Unconstrained 0.04853    0.02890    1
Inertia is squared Bray distance 
Species scores projected from 'tmp_comm' 

Eigenvalues for constrained axes:
  CAP1   CAP2   CAP3   CAP4   CAP5   CAP6   CAP7   CAP8   CAP9  CAP10  CAP11 
0.6287 0.2955 0.1984 0.1104 0.1085 0.0661 0.0536 0.0511 0.0359 0.0277 0.0247 
 CAP12  CAP13 
0.0218 0.0080 

Eigenvalues for unconstrained axes:
   MDS1 
0.04853 </code></pre>
</div>
</div>
<p>Now we can look at the variance against each principal component.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-63-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>And then make some quick and dirty plots. This will also come in handy later when we need to parse out data a better plot visualization. The <code>ggplot</code> function <code>autoplot</code> stores these data in a more accessible way than the raw results from <code>capscale</code></p>
<div class="cell">
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span><span class="op">(</span><span class="va">edaphic_cap</span><span class="op">)</span> </code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-64-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_auto_plt</span> <span class="op">&lt;-</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html">autoplot</a></span><span class="op">(</span><span class="va">edaphic_cap</span>, arrows <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">tmp_auto_plt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-64-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">

</div>
<p>Next, we need to grab capscale scores for the samples and create a data frame of the first two dimensions. We will also need to add some of the sample details to the data frame. For this we use the vegan function <code>scores</code> which gets species or site scores from the ordination.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb36"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_samp_scores</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tmp_auto_plt</span><span class="op">$</span><span class="va">plot_env</span><span class="op">$</span><span class="va">obj</span>, <span class="va">Score</span> <span class="op">==</span> <span class="st">"sites"</span><span class="op">)</span>
<span class="va">tmp_samp_scores</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">tmp_samp_scores</span> <span class="op">&lt;-</span> <span class="va">tmp_samp_scores</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>SampleID <span class="op">=</span> <span class="va">Label</span><span class="op">)</span>

<span class="va">tmp_md_sub</span> <span class="op">&lt;-</span> <span class="va">tmp_md</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>
<span class="va">tmp_md_sub</span> <span class="op">&lt;-</span> <span class="va">tmp_md_sub</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>
<span class="va">edaphic_plot_data</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_md_sub</span>, <span class="va">tmp_samp_scores</span>, by <span class="op">=</span> <span class="st">"SampleID"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Now we have a new data frame that contains sample details and capscale values.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>          SampleID PLOT TREAT TREAT_T PAIR        CAP1           CAP2
1  P10_D00_010_C0E  P10     C       0    E -0.51023381  0.16071056348
2  P02_D00_010_C0A  P02     C       0    A -0.48448722 -0.19055366181
3  P04_D00_010_C0B  P04     C       0    B -0.58218619  0.02325887477
4  P06_D00_010_C0C  P06     C       0    C -0.53370906  0.09973976767
5  P08_D00_010_C0D  P08     C       0    D -0.68492943  0.13143857676
6  P01_D00_010_W3A  P01     W       3    A -0.13286443 -0.28661650806
7  P03_D00_010_W3B  P03     W       3    B -0.25415783  0.00002258673
8  P05_D00_010_W3C  P05     W       3    C -0.38829574 -0.08695154559
9  P07_D00_010_W3D  P07     W       3    D -0.01360582 -0.49632449304
10 P09_D00_010_W3E  P09     W       3    E -0.10639798 -0.54219706978
11 P01_D00_010_W8A  P01     W       8    A  0.78689725  0.02214444812
12 P03_D00_010_W8B  P03     W       8    B  0.89574754 -0.92786190328
13 P05_D00_010_W8C  P05     W       8    C  1.07945322 -0.28874425650
14 P07_D00_010_W8D  P07     W       8    D  0.41234456  0.75597843926
15 P09_D00_010_W8E  P09     W       8    E  0.51642495  1.62595618128</code></pre>
</div>
</div>
<p>We can then do the same with the metadata vectors. Here though we only need the scores and parameter name.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb38"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">edaphic_md_scores</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tmp_auto_plt</span><span class="op">$</span><span class="va">plot_env</span><span class="op">$</span><span class="va">obj</span>, <span class="va">Score</span> <span class="op">==</span> <span class="st">"biplot"</span><span class="op">)</span>
<span class="va">edaphic_md_scores</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">edaphic_md_scores</span> <span class="op">&lt;-</span> <span class="va">edaphic_md_scores</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>parameters <span class="op">=</span> <span class="va">Label</span>, .before <span class="op">=</span> <span class="va">CAP1</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"Label"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>     parameters        CAP1         CAP2
AST         AST  0.97882379 -0.055074169
H2O         H2O -0.76548607 -0.129779379
N             N  0.38130076 -0.131035479
P             P  0.26536024 -0.132125558
Fe           Fe  0.12138586  0.353093177
K             K -0.02490580  0.151975508
ECEC       ECEC -0.15365335  0.379848234
pH           pH -0.07327989  0.272676813
NH4         NH4  0.19922925  0.516074241
NO3         NO3 -0.27681227  0.288729283
PO4         PO4 -0.60140669 -0.009980058
DOC         DOC -0.59979130 -0.399930429
DOCN       DOCN  0.11210058  0.422975484</code></pre>
</div>
</div>
<p>Lets run some quick correlations of metadata with ordination axes to see which parameters are significant. For this we use the vegan function <code>envfit</code>.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb40"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_samp_scores_sub</span> <span class="op">&lt;-</span> <span class="va">edaphic_plot_data</span><span class="op">[</span>, <span class="fl">6</span><span class="op">:</span><span class="fl">7</span><span class="op">]</span>
<span class="va">tmp_samp_scores_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">tmp_samp_scores_sub</span><span class="op">)</span>
<span class="va">tmp_param_list</span> <span class="op">&lt;-</span> <span class="va">edaphic_md_scores</span><span class="op">$</span><span class="va">parameters</span>

<span class="va">tmp_md_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">tmp_md</span>,  select <span class="op">=</span>  <span class="va">tmp_param_list</span><span class="op">)</span>

<span class="va">envfit_edaphic_md</span> <span class="op">&lt;-</span> <span class="fu">envfit</span><span class="op">(</span><span class="va">tmp_samp_scores_sub</span>, <span class="va">tmp_md_sub</span>,
                 perm <span class="op">=</span> <span class="fl">1000</span>, choices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
***VECTORS

         CAP1     CAP2     r2   Pr(&gt;r)    
AST   0.99825 -0.05912 0.8287 0.000999 ***
H2O  -0.98657 -0.16333 0.5189 0.009990 ** 
N     0.94537 -0.32599 0.1403 0.398601    
P     0.89518 -0.44571 0.0759 0.602398    
Fe    0.32411  0.94602 0.1194 0.432567    
K    -0.16554  0.98620 0.0204 0.890110    
ECEC -0.37874  0.92550 0.1444 0.368631    
pH   -0.26344  0.96468 0.0685 0.672328    
NH4   0.35944  0.93317 0.2621 0.152847    
NO3  -0.69392  0.72006 0.1380 0.417582    
PO4  -0.99991 -0.01327 0.3118 0.090909 .  
DOC  -0.83396 -0.55183 0.4459 0.023976 *  
DOCN  0.25463  0.96704 0.1640 0.318681    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb42"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">edaphic_md_signif_hits</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">envfit_edaphic_md</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span>, 
                                  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">envfit_edaphic_md</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span> 
                                  <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">envfit_edaphic_md</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">r</span> <span class="op">&gt;</span> <span class="fl">0.4</span><span class="op">)</span><span class="op">)</span>
<span class="va">edaphic_md_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">edaphic_md_signif_hits</span><span class="op">)</span>
<span class="va">edaphic_md_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">edaphic_md_signif_hits</span><span class="op">)</span>
<span class="va">edaphic_md_signif</span> <span class="op">&lt;-</span> <span class="va">edaphic_md_scores</span><span class="op">[</span><span class="va">edaphic_md_scores</span><span class="op">$</span><span class="va">parameters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">edaphic_md_signif_hits</span>,<span class="op">]</span>
<span class="va">edaphic_md_signif</span><span class="op">$</span><span class="va">parameters</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><code>envfit</code> found that AST, H2O, DOC were significantly correlated.</p>
<p>Now lets see if the same parameters are significant for the <code>envfit</code> and <code>bioenv</code> analyses.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Significant parameters from bioenv analysis."
[1] "AST"
_____________________________________
[1] "Significant parameters from envfit analysis."
[1] "AST" "H2O" "DOC"
_____________________________________
[1] "Found in bioenv but not envfit."
character(0)
_____________________________________
[1] "Found in envfit but not bioenv."
[1] "H2O" "DOC"
_____________________________________
[1] "Found in envfit and bioenv."
[1] "AST" "H2O" "DOC"</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb44"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">new_edaphic_md_signif_hits</span> <span class="op">&lt;-</span> <span class="va">edaphic_sig_diff</span>
<span class="co">#new_edaphic_md_signif_hits &lt;- append(edaphic_md_signif_hits, edaphic_sig_diff)</span>
<span class="va">edaphic_md_signif_all</span> <span class="op">&lt;-</span> <span class="va">edaphic_md_scores</span><span class="op">[</span><span class="va">edaphic_md_scores</span><span class="op">$</span><span class="va">parameters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">new_edaphic_md_signif_hits</span>,<span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Check. Next, we run <code>envfit</code> for the ASVs.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb45"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">envfit_edaphic_asv</span> <span class="op">&lt;-</span> <span class="fu">envfit</span><span class="op">(</span><span class="va">tmp_samp_scores_sub</span>, 
                             <span class="va">tmp_comm</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span><span class="op">(</span><span class="op">-</span><span class="va">tmp_comm</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>,
                             perm <span class="op">=</span> <span class="fl">1000</span>, choices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="va">edaphic_asv_scores</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tmp_auto_plt</span><span class="op">$</span><span class="va">plot_env</span><span class="op">$</span><span class="va">obj</span>, <span class="va">Score</span> <span class="op">==</span> <span class="st">"species"</span><span class="op">)</span>
<span class="va">edaphic_asv_scores</span> <span class="op">&lt;-</span> <span class="va">edaphic_asv_scores</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>parameters <span class="op">=</span> <span class="va">Label</span>, .before <span class="op">=</span> <span class="va">CAP1</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"Label"</span><span class="op">)</span>
<span class="va">edaphic_asv_scores</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
***VECTORS

          CAP1     CAP2     r2   Pr(&gt;r)    
ASV1   0.99502  0.09969 0.6491 0.005994 ** 
ASV2   0.54850 -0.83615 0.8984 0.000999 ***
ASV3   0.79245  0.60994 0.9429 0.000999 ***
ASV4  -0.12958  0.99157 0.2913 0.120879    
ASV5  -0.56363 -0.82603 0.1586 0.362637    
ASV6  -0.10651  0.99431 0.7559 0.000999 ***
ASV8  -0.84329  0.53746 0.4370 0.030969 *  
ASV7  -0.52510 -0.85104 0.0681 0.655345    
ASV9   0.51828 -0.85521 0.2559 0.185814    
ASV10 -0.34456  0.93876 0.8331 0.000999 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb47"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">edaphic_asv_signif_hits</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">envfit_edaphic_asv</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span>, 
                                  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">envfit_edaphic_asv</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span> 
                                  <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">envfit_edaphic_asv</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">r</span> <span class="op">&gt;</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>
<span class="va">edaphic_asv_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">edaphic_asv_signif_hits</span><span class="op">)</span>
<span class="va">edaphic_asv_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">edaphic_asv_signif_hits</span><span class="op">)</span>
<span class="va">edaphic_asv_signif</span> <span class="op">&lt;-</span> <span class="va">edaphic_asv_scores</span><span class="op">[</span><span class="va">edaphic_asv_scores</span><span class="op">$</span><span class="va">parameters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">edaphic_asv_signif_hits</span>,<span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>      parameters        CAP1        CAP2 variable_type
ASV3        ASV3  0.48974117  0.51152737           ASV
ASV6        ASV6 -0.02123424  0.24919713           ASV
ASV2        ASV2  0.23670463 -0.49287683           ASV
ASV10      ASV10 -0.02578511  0.09240257           ASV
ASV1        ASV1  1.18949655  0.17867402           ASV</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb49"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">edaphic_md_signif_all</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">&lt;-</span> <span class="st">"metadata"</span>
<span class="va">edaphic_asv_signif</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">&lt;-</span> <span class="st">"ASV"</span>
<span class="va">edaphic_bioplot_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">edaphic_md_signif_all</span>, <span class="va">edaphic_asv_signif</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The last thing to do is categorize parameters scores and ASV scores into different variable types for plotting.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb50"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">edaphic_bioplot_data_md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">edaphic_bioplot_data</span>, 
                                  <span class="va">edaphic_bioplot_data</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">==</span> <span class="st">"metadata"</span><span class="op">)</span>
<span class="va">edaphic_bioplot_data_asv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">edaphic_bioplot_data</span>, 
                                   <span class="va">edaphic_bioplot_data</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">==</span> <span class="st">"ASV"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<details><summary>
Show the code for the plot
</summary><div class="cell">

</div>
</details><div class="cell">

</div>
</section><section id="soil-functional-response-1" class="level3"><h3 class="anchored" data-anchor-id="soil-functional-response-1">Soil Functional Response</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb51"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_md</span> <span class="op">&lt;-</span> <span class="va">ssu18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">soil_funct</span>
<span class="va">tmp_md</span><span class="op">$</span><span class="va">TREAT_T</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">tmp_md</span><span class="op">$</span><span class="va">TREAT_T</span><span class="op">)</span>
<span class="va">tmp_comm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="va">ssu18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">data_loaded</span><span class="op">)</span><span class="op">)</span>
<span class="va">soil_funct_rank</span> <span class="op">&lt;-</span> <span class="fu">rankindex</span><span class="op">(</span><span class="va">tmp_md</span><span class="op">[</span>, <span class="fl">8</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_md</span><span class="op">)</span><span class="op">]</span>, <span class="va">tmp_comm</span>, 
          indices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"euc"</span>, <span class="st">"man"</span>, <span class="st">"gow"</span>,<span class="st">"bra"</span>, <span class="st">"kul"</span><span class="op">)</span>, 
          stepacross <span class="op">=</span> <span class="cn">FALSE</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>      euc       man       gow       bra       kul 
0.5840452 0.7341800 0.2168775 0.7341800 0.7341800 </code></pre>
</div>
</div>
<p>Lets run <code>capscale</code> using Bray-Curtis. Note, we have 11 metadata parameters in this group but, for some reason, <code>capscale</code> only works with 13 parameters. This may have to do with degrees of freedom?</p>
<ul>
<li>Starting properties: micC, micN, micP, micCN, micCP, micNP, AG_ase, BG_ase, BP_ase, CE_ase, P_ase, N_ase, S_ase, XY_ase, LP_ase, PX_ase, CO2, enzCN, enzCP, enzNP</li>
<li>Autocorrelated removed: micN, micNP, enzCN, enzCP, BP_ase, CE_ase, LP_ase, N_ase, P_ase</li>
<li>Remove for capscale: NONE</li>
</ul>
<div class="cell">

</div>
<div class="cell">
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">soil_funct_cap</span> <span class="op">&lt;-</span> <span class="fu">capscale</span><span class="op">(</span><span class="va">tmp_comm</span> <span class="op">~</span> <span class="va">micC</span> <span class="op">+</span> <span class="va">micP</span> <span class="op">+</span> <span class="va">micCN</span> <span class="op">+</span> <span class="va">micCP</span> <span class="op">+</span> <span class="va">AG_ase</span> <span class="op">+</span> <span class="va">BG_ase</span> <span class="op">+</span> 
                                      <span class="va">S_ase</span> <span class="op">+</span> <span class="va">XY_ase</span> <span class="op">+</span> <span class="va">PX_ase</span> <span class="op">+</span> <span class="va">CO2</span> <span class="op">+</span> <span class="va">enzNP</span>, 
                           <span class="va">tmp_md</span>, dist <span class="op">=</span> <span class="st">"bray"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Call: capscale(formula = tmp_comm ~ micC + micP + micCN + micCP +
AG_ase + BG_ase + S_ase + XY_ase + PX_ase + CO2 + enzNP, data = tmp_md,
distance = "bray")

              Inertia Proportion Rank
Total         1.67890    1.00000     
Constrained   1.52672    0.90936   11
Unconstrained 0.15218    0.09064    3
Inertia is squared Bray distance 
Species scores projected from 'tmp_comm' 

Eigenvalues for constrained axes:
  CAP1   CAP2   CAP3   CAP4   CAP5   CAP6   CAP7   CAP8   CAP9  CAP10  CAP11 
0.6202 0.2950 0.1827 0.1164 0.0950 0.0555 0.0491 0.0468 0.0290 0.0246 0.0126 

Eigenvalues for unconstrained axes:
   MDS1    MDS2    MDS3 
0.06264 0.05568 0.03386 </code></pre>
</div>
</div>
<p>Now we can look at the variance against each principal component.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-88-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>And then make some quick and dirty plots. This will also come in handy later when we need to parse out data a better plot visualization. The <code>ggplot</code> function <code>autoplot</code> stores these data in a more accessible way than the raw results from <code>capscale</code></p>
<div class="cell">
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span><span class="op">(</span><span class="va">soil_funct_cap</span><span class="op">)</span> </code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-89-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_auto_plt</span> <span class="op">&lt;-</span> <span class="fu">autoplot</span><span class="op">(</span><span class="va">soil_funct_cap</span>, arrows <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">tmp_auto_plt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-89-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">

</div>
<p>Next, we need to grab capscale scores for the samples and create a data frame of the first two dimensions. We will also need to add some of the sample details to the data frame. For this we use the vegan function <code>scores</code> which gets species or site scores from the ordination.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb57"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_samp_scores</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tmp_auto_plt</span><span class="op">$</span><span class="va">plot_env</span><span class="op">$</span><span class="va">obj</span>, <span class="va">Score</span> <span class="op">==</span> <span class="st">"sites"</span><span class="op">)</span>
<span class="va">tmp_samp_scores</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">tmp_samp_scores</span> <span class="op">&lt;-</span> <span class="va">tmp_samp_scores</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>SampleID <span class="op">=</span> <span class="va">Label</span><span class="op">)</span>

<span class="va">tmp_md_sub</span> <span class="op">&lt;-</span> <span class="va">tmp_md</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>
<span class="va">tmp_md_sub</span> <span class="op">&lt;-</span> <span class="va">tmp_md_sub</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>
<span class="va">soil_funct_plot_data</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_md_sub</span>, <span class="va">tmp_samp_scores</span>, 
                                         by <span class="op">=</span> <span class="st">"SampleID"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Now we have a new data frame that contains sample details and capscale values.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>          SampleID PLOT TREAT TREAT_T PAIR        CAP1        CAP2
1  P10_D00_010_C0E  P10     C       0    E -0.53163678  0.19467717
2  P02_D00_010_C0A  P02     C       0    A -0.49565928 -0.17246578
3  P04_D00_010_C0B  P04     C       0    B -0.58149488  0.04815608
4  P06_D00_010_C0C  P06     C       0    C -0.53700662  0.10581563
5  P08_D00_010_C0D  P08     C       0    D -0.69408551  0.15871504
6  P01_D00_010_W3A  P01     W       3    A -0.13789308 -0.32241637
7  P03_D00_010_W3B  P03     W       3    B -0.23991258 -0.02936566
8  P05_D00_010_W3C  P05     W       3    C -0.37861217 -0.07503400
9  P07_D00_010_W3D  P07     W       3    D -0.01748558 -0.49326034
10 P09_D00_010_W3E  P09     W       3    E -0.10498973 -0.54355896
11 P01_D00_010_W8A  P01     W       8    A  0.78557974  0.01332924
12 P03_D00_010_W8B  P03     W       8    B  0.89342218 -0.94105534
13 P05_D00_010_W8C  P05     W       8    C  1.07904998 -0.30014637
14 P07_D00_010_W8D  P07     W       8    D  0.41449841  0.74331915
15 P09_D00_010_W8E  P09     W       8    E  0.54622591  1.61329051</code></pre>
</div>
</div>
<p>We can then do the same with the metadata vectors. Here though we only need the scores and parameter name.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb59"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">soil_funct_md_scores</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tmp_auto_plt</span><span class="op">$</span><span class="va">plot_env</span><span class="op">$</span><span class="va">obj</span>, <span class="va">Score</span> <span class="op">==</span> <span class="st">"biplot"</span><span class="op">)</span>
<span class="va">soil_funct_md_scores</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">soil_funct_md_scores</span> <span class="op">&lt;-</span> <span class="va">soil_funct_md_scores</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>parameters <span class="op">=</span> <span class="va">Label</span>, .before <span class="op">=</span> <span class="va">CAP1</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"Label"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>       parameters        CAP1       CAP2
micC         micC  0.05033428  0.3114850
micP         micP -0.28522874  0.2555622
micCN       micCN  0.18231162 -0.4195484
micCP       micCP  0.23124711 -0.2302962
AG_ase     AG_ase -0.26468219  0.6079572
BG_ase     BG_ase  0.42574674  0.6072615
S_ase       S_ase  0.29987944  0.7961890
XY_ase     XY_ase  0.64821470  0.3027833
PX_ase     PX_ase  0.80304377  0.3296233
CO2           CO2  0.68903357 -0.1628577
enzNP       enzNP -0.77637066  0.1324578</code></pre>
</div>
</div>
<p>Lets run some quick correlations of metadata with ordination axes to see which parameters are significant. For this we use the vegan function <code>envfit</code>.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb61"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_samp_scores_sub</span> <span class="op">&lt;-</span> <span class="va">soil_funct_plot_data</span><span class="op">[</span>, <span class="fl">6</span><span class="op">:</span><span class="fl">7</span><span class="op">]</span>
<span class="va">tmp_samp_scores_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">tmp_samp_scores_sub</span><span class="op">)</span>
<span class="va">tmp_param_list</span> <span class="op">&lt;-</span> <span class="va">soil_funct_md_scores</span><span class="op">$</span><span class="va">parameters</span>

<span class="va">tmp_md_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">tmp_md</span>,  select <span class="op">=</span>  <span class="va">tmp_param_list</span><span class="op">)</span>

<span class="va">envfit_soil_funct_md</span> <span class="op">&lt;-</span> <span class="fu">envfit</span><span class="op">(</span><span class="va">tmp_samp_scores_sub</span>, <span class="va">tmp_md_sub</span>,
                 perm <span class="op">=</span> <span class="fl">1000</span>, choices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
***VECTORS

           CAP1     CAP2     r2   Pr(&gt;r)    
micC    0.16467  0.98635 0.1012 0.511489    
micP   -0.74314  0.66913 0.1473 0.385614    
micCN   0.39228 -0.91985 0.2111 0.244755    
micCP   0.70633 -0.70788 0.1070 0.504496    
AG_ase -0.39292  0.91957 0.4435 0.025974 *  
BG_ase  0.57346  0.81923 0.5604 0.006993 ** 
S_ase   0.35512  0.93482 0.7371 0.001998 ** 
XY_ase  0.90321  0.42921 0.5193 0.008991 ** 
PX_ase  0.92246  0.38609 0.7640 0.000999 ***
CO2     0.97421 -0.22565 0.5041 0.012987 *  
enzNP  -0.98661  0.16307 0.6242 0.003996 ** 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb63"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">soil_funct_md_signif_hits</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">envfit_soil_funct_md</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span>, 
                                  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">envfit_soil_funct_md</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span> 
                                  <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">envfit_soil_funct_md</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">r</span> <span class="op">&gt;</span> <span class="fl">0.4</span><span class="op">)</span><span class="op">)</span>
<span class="va">soil_funct_md_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">soil_funct_md_signif_hits</span><span class="op">)</span>
<span class="va">soil_funct_md_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">soil_funct_md_signif_hits</span><span class="op">)</span>
<span class="va">soil_funct_md_signif</span> <span class="op">&lt;-</span> <span class="va">soil_funct_md_scores</span><span class="op">[</span><span class="va">soil_funct_md_scores</span><span class="op">$</span><span class="va">parameters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> 
                                               <span class="va">soil_funct_md_signif_hits</span>,<span class="op">]</span>
<span class="va">soil_funct_md_signif</span><span class="op">$</span><span class="va">parameters</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><code>envfit</code> found that AG_ase, BG_ase, S_ase, XY_ase, PX_ase, CO2, enzNP were significantly correlated.</p>
<p>Now lets see if the same parameters are significant for the <code>envfit</code> and <code>bioenv</code> analyses.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Significant parameters from bioenv analysis."
[1] "AG_ase" "enzNP"  "S_ase"  "PX_ase" "XY_ase"
_____________________________________
[1] "Significant parameters from envfit analysis."
[1] "AG_ase" "BG_ase" "S_ase"  "XY_ase" "PX_ase" "CO2"    "enzNP" 
_____________________________________
[1] "Found in bioenv but not envfit."
character(0)
_____________________________________
[1] "Found in envfit but not bioenv."
[1] "BG_ase" "CO2"   
_____________________________________
[1] "Found in envfit and bioenv."
[1] "AG_ase" "BG_ase" "S_ase"  "XY_ase" "PX_ase" "CO2"    "enzNP" </code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb65"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="co">#new_soil_funct_md_signif_hits &lt;- append(soil_funct_md_signif_hits, soil_funct_sig_diff)</span>
<span class="va">new_soil_funct_md_signif_hits</span> <span class="op">&lt;-</span> <span class="va">soil_funct_sig_diff</span>
<span class="va">soil_funct_md_signif_all</span> <span class="op">&lt;-</span> <span class="va">soil_funct_md_scores</span><span class="op">[</span><span class="va">soil_funct_md_scores</span><span class="op">$</span><span class="va">parameters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> 
                                                   <span class="va">new_soil_funct_md_signif_hits</span>,<span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Check. Next, we run <code>envfit</code> for the ASVs.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb66"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">envfit_soil_funct_asv</span> <span class="op">&lt;-</span> <span class="fu">envfit</span><span class="op">(</span><span class="va">tmp_samp_scores_sub</span>, 
                                <span class="va">tmp_comm</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span><span class="op">(</span><span class="op">-</span><span class="va">tmp_comm</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>, 
                                perm <span class="op">=</span> <span class="fl">1000</span>, choices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="va">soil_funct_asv_scores</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tmp_auto_plt</span><span class="op">$</span><span class="va">plot_env</span><span class="op">$</span><span class="va">obj</span>, <span class="va">Score</span> <span class="op">==</span> <span class="st">"species"</span><span class="op">)</span>
<span class="va">soil_funct_asv_scores</span> <span class="op">&lt;-</span> <span class="va">soil_funct_asv_scores</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>parameters <span class="op">=</span> <span class="va">Label</span>, .before <span class="op">=</span> <span class="va">CAP1</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"Label"</span><span class="op">)</span>
<span class="va">soil_funct_asv_scores</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
***VECTORS

          CAP1     CAP2     r2   Pr(&gt;r)    
ASV1   0.99572  0.09244 0.6448 0.008991 ** 
ASV2   0.53392 -0.84554 0.9003 0.000999 ***
ASV3   0.80024  0.59968 0.9414 0.000999 ***
ASV4  -0.10124  0.99486 0.2814 0.142857    
ASV5  -0.56790 -0.82310 0.1663 0.313686    
ASV6  -0.08053  0.99675 0.7514 0.000999 ***
ASV8  -0.83447  0.55105 0.4303 0.017982 *  
ASV7  -0.57807 -0.81599 0.0697 0.619381    
ASV9   0.49314 -0.86995 0.2612 0.147852    
ASV10 -0.32480  0.94578 0.8408 0.000999 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb68"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">soil_funct_asv_signif_hits</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">envfit_soil_funct_asv</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span>, 
                                  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">envfit_soil_funct_asv</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span> 
                                  <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">envfit_soil_funct_asv</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">r</span> <span class="op">&gt;</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>
<span class="va">soil_funct_asv_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">soil_funct_asv_signif_hits</span><span class="op">)</span>
<span class="va">soil_funct_asv_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">soil_funct_asv_signif_hits</span><span class="op">)</span>
<span class="va">soil_funct_asv_signif</span> <span class="op">&lt;-</span> <span class="va">soil_funct_asv_scores</span><span class="op">[</span><span class="va">soil_funct_asv_scores</span><span class="op">$</span><span class="va">parameters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> 
                                                 <span class="va">soil_funct_asv_signif_hits</span>,<span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>      parameters        CAP1        CAP2 variable_type
ASV3        ASV3  0.49966533  0.50499834           ASV
ASV6        ASV6 -0.01098408  0.25326180           ASV
ASV2        ASV2  0.22435017 -0.49303800           ASV
ASV10      ASV10 -0.02420323  0.09642322           ASV
ASV1        ASV1  1.17846608  0.16123440           ASV</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb70"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">soil_funct_md_signif_all</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">&lt;-</span> <span class="st">"metadata"</span>
<span class="va">soil_funct_asv_signif</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">&lt;-</span> <span class="st">"ASV"</span>
<span class="va">soil_funct_bioplot_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">soil_funct_md_signif_all</span>, <span class="va">soil_funct_asv_signif</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The last thing to do is categorize parameters scores and ASV scores into different variable types for plotting.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb71"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">soil_funct_bioplot_data_md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">soil_funct_bioplot_data</span>, 
                                  <span class="va">soil_funct_bioplot_data</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">==</span> <span class="st">"metadata"</span><span class="op">)</span>
<span class="va">soil_funct_bioplot_data_asv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">soil_funct_bioplot_data</span>, 
                                   <span class="va">soil_funct_bioplot_data</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">==</span> <span class="st">"ASV"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<details><summary>
Show the code for the plot
</summary><div class="cell">
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">soil_funct_cap_vals</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">soil_funct_cap</span><span class="op">$</span><span class="va">CCA</span><span class="op">$</span><span class="va">eig</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>
<span class="va">soil_funct_cap1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span><span class="op">(</span><span class="va">soil_funct_cap_vals</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span>, digits<span class="op">=</span><span class="fl">3</span><span class="op">)</span>
<span class="va">soil_funct_cap2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span><span class="op">(</span><span class="va">soil_funct_cap_vals</span><span class="op">[</span><span class="fl">2</span>,<span class="op">]</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span>, digits<span class="op">=</span><span class="fl">3</span><span class="op">)</span>

<span class="va">cpa1_lab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"CAP1"</span>, <span class="st">" ("</span>, <span class="va">soil_funct_cap1</span>, <span class="st">"%)"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
<span class="va">cpa2_lab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"CAP2"</span>, <span class="st">" ("</span>, <span class="va">soil_funct_cap2</span>, <span class="st">"%)"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>

<span class="va">swel_col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#2271B2"</span>, <span class="st">"#71B222"</span>, <span class="st">"#B22271"</span><span class="op">)</span>

<span class="va">soil_funct_plot</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">soil_funct_plot_data</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">CAP1</span>, y <span class="op">=</span> <span class="va">CAP2</span>, shape <span class="op">=</span> <span class="va">TREAT</span>,
                           colour <span class="op">=</span> <span class="va">TREAT_T</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_colour_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">swel_col</span><span class="op">)</span> <span class="op">+</span>
<span class="co">#  geom_text(data = soil_funct_plot_data, aes(x = CAP1, y = CAP2, #UNCOMMENT to add sample labels</span>
<span class="co">#                              label = SampleID), size = 3) + </span>
  <span class="fu">geom_segment</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="fl">0</span>, y <span class="op">=</span> <span class="fl">0</span>, xend <span class="op">=</span> <span class="va">CAP1</span>, yend <span class="op">=</span> <span class="va">CAP2</span><span class="op">)</span>,
               data <span class="op">=</span> <span class="va">soil_funct_bioplot_data_md</span>, linetype <span class="op">=</span> <span class="st">"solid"</span>,
               arrow <span class="op">=</span> <span class="fu">arrow</span><span class="op">(</span>length <span class="op">=</span> <span class="fu">unit</span><span class="op">(</span><span class="fl">0.3</span>, <span class="st">"cm"</span><span class="op">)</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">0.4</span>, 
               color <span class="op">=</span> <span class="st">"#191919"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_text</span><span class="op">(</span>data <span class="op">=</span> <span class="va">soil_funct_bioplot_data_md</span>, 
            <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">CAP1</span>, y <span class="op">=</span> <span class="va">CAP2</span>, label <span class="op">=</span> <span class="va">parameters</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">3</span>,
            nudge_x <span class="op">=</span> <span class="fl">0.1</span>, nudge_y <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">+</span>
  <span class="co">### This code adds ASV vectors, Uncomment to add</span>
  <span class="co">#geom_segment(aes(x = 0, y = 0, xend = CAP1, yend = CAP2),</span>
  <span class="co">#             data = soil_funct_bioplot_data_asv, linetype = "solid",</span>
  <span class="co">#             arrow = arrow(length = unit(0.3, "cm")), size = 0.2,</span>
  <span class="co">#             color = "#676767") +</span>
  <span class="co">#geom_text(data = soil_funct_bioplot_data_asv, </span>
  <span class="co">#          aes(x = CAP1, y = CAP2, label = parameters), size = 2.5,</span>
  <span class="co">#          nudge_x = 0.05, nudge_y = 0.05) +</span>
  <span class="fu">theme_classic</span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Capscale Analysis"</span>,
       subtitle <span class="op">=</span> <span class="st">"Soil Functional Response"</span>, 
       x <span class="op">=</span> <span class="va">cpa1_lab</span>, 
       y <span class="op">=</span> <span class="va">cpa2_lab</span><span class="op">)</span>
<span class="va">soil_funct_plot</span> <span class="op">&lt;-</span> <span class="va">soil_funct_plot</span> <span class="op">+</span> <span class="fu">coord_fixed</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
<span class="va">soil_funct_plot</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/png.html">png</a></span><span class="op">(</span><span class="st">"files/metadata/figures/ssu18_soil_funct_capscale.png"</span>,
    height <span class="op">=</span> <span class="fl">16</span>, width <span class="op">=</span> <span class="fl">20</span>, units <span class="op">=</span> <span class="st">'cm'</span>, res <span class="op">=</span> <span class="fl">600</span>, bg <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span>
<span class="va">soil_funct_plot</span>
<span class="fu"><a href="https://rdrr.io/r/base/invisible.html">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span><span class="op">(</span><span class="st">"files/metadata/figures/ssu18_soil_funct_capscale.pdf"</span>,
    height <span class="op">=</span> <span class="fl">5</span>, width <span class="op">=</span> <span class="fl">6</span><span class="op">)</span>
<span class="va">soil_funct_plot</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span><span class="op">(</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</details><div class="cell">

</div>
</section><section id="temperature-adaptation-2" class="level3"><h3 class="anchored" data-anchor-id="temperature-adaptation-2">Temperature Adaptation</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb73"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_md</span> <span class="op">&lt;-</span> <span class="va">ssu18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">temp_adapt</span>
<span class="va">tmp_md</span><span class="op">$</span><span class="va">TREAT_T</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">tmp_md</span><span class="op">$</span><span class="va">TREAT_T</span><span class="op">)</span>
<span class="va">tmp_comm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="va">ssu18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">data_loaded</span><span class="op">)</span><span class="op">)</span>
<span class="va">temp_adapt_rank</span> <span class="op">&lt;-</span> <span class="fu">rankindex</span><span class="op">(</span><span class="va">tmp_md</span><span class="op">[</span>, <span class="fl">8</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_md</span><span class="op">)</span><span class="op">]</span>, <span class="va">tmp_comm</span>, 
          indices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"euc"</span>, <span class="st">"man"</span>, <span class="st">"gow"</span>,<span class="st">"bra"</span>, <span class="st">"kul"</span><span class="op">)</span>, 
          stepacross <span class="op">=</span> <span class="cn">FALSE</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>      euc       man       gow       bra       kul 
0.2732117 0.3330085 0.3456355 0.3330085 0.3330085 </code></pre>
</div>
</div>
<p>Lets run <code>capscale</code> using Bray-Curtis. Note, we have 13 metadata parameters in this group but, for some reason, <code>capscale</code> only works with 13 parameters. This may have to do with degrees of freedom?</p>
<ul>
<li>Starting properties: AG_Q10, BG_Q10, BP_Q10, CE_Q10, P_Q10, N_Q10, S_Q10, XY_Q10, LP_Q10, PX_Q10, CUEcn, CUEcp, NUE, PUE, Tmin, SI</li>
<li>Autocorrelated removed: NUE, PUE, SI</li>
<li>Remove for capscale: NONE</li>
</ul>
<div class="cell">

</div>
<div class="cell">
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">temp_adapt_cap</span> <span class="op">&lt;-</span> <span class="fu">capscale</span><span class="op">(</span><span class="va">tmp_comm</span> <span class="op">~</span>  <span class="va">AG_Q10</span> <span class="op">+</span> <span class="va">BG_Q10</span> <span class="op">+</span> <span class="va">BP_Q10</span> <span class="op">+</span> <span class="va">CE_Q10</span> <span class="op">+</span> 
                                       <span class="va">P_Q10</span> <span class="op">+</span> <span class="va">N_Q10</span> <span class="op">+</span> <span class="va">S_Q10</span> <span class="op">+</span> <span class="va">XY_Q10</span> <span class="op">+</span> 
                                       <span class="va">LP_Q10</span> <span class="op">+</span> <span class="va">PX_Q10</span> <span class="op">+</span> <span class="va">CUEcn</span> <span class="op">+</span> <span class="va">CUEcp</span> <span class="op">+</span> <span class="va">Tmin</span>, 
                             <span class="va">tmp_md</span>, dist <span class="op">=</span> <span class="st">"bray"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Call: capscale(formula = tmp_comm ~ AG_Q10 + BG_Q10 + BP_Q10 + CE_Q10 +
P_Q10 + N_Q10 + S_Q10 + XY_Q10 + LP_Q10 + PX_Q10 + CUEcn + CUEcp +
Tmin, data = tmp_md, distance = "bray")

              Inertia Proportion Rank
Total         1.67890    1.00000     
Constrained   1.57681    0.93919   13
Unconstrained 0.10209    0.06081    1
Inertia is squared Bray distance 
Species scores projected from 'tmp_comm' 

Eigenvalues for constrained axes:
  CAP1   CAP2   CAP3   CAP4   CAP5   CAP6   CAP7   CAP8   CAP9  CAP10  CAP11 
0.6298 0.2776 0.1985 0.1087 0.0849 0.0614 0.0528 0.0450 0.0352 0.0273 0.0244 
 CAP12  CAP13 
0.0218 0.0095 

Eigenvalues for unconstrained axes:
   MDS1 
0.10209 </code></pre>
</div>
</div>
<p>Now we can look at the variance against each principal component.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-113-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>And then make some quick and dirty plots. This will also come in handy later when we need to parse out data a better plot visualization. The <code>ggplot</code> function <code>autoplot</code> stores these data in a more accessible way than the raw results from <code>capscale</code></p>
<div class="cell">
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span><span class="op">(</span><span class="va">temp_adapt_cap</span><span class="op">)</span> </code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-114-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_auto_plt</span> <span class="op">&lt;-</span> <span class="fu">autoplot</span><span class="op">(</span><span class="va">temp_adapt_cap</span>, arrows <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">tmp_auto_plt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-114-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">

</div>
<p>Next, we need to grab capscale scores for the samples and create a data frame of the first two dimensions. We will also need to add some of the sample details to the data frame. For this we use the vegan function <code>scores</code> which gets species or site scores from the ordination.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb79"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_samp_scores</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tmp_auto_plt</span><span class="op">$</span><span class="va">plot_env</span><span class="op">$</span><span class="va">obj</span>, <span class="va">Score</span> <span class="op">==</span> <span class="st">"sites"</span><span class="op">)</span>
<span class="va">tmp_samp_scores</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">tmp_samp_scores</span> <span class="op">&lt;-</span> <span class="va">tmp_samp_scores</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>SampleID <span class="op">=</span> <span class="va">Label</span><span class="op">)</span>

<span class="va">tmp_md_sub</span> <span class="op">&lt;-</span> <span class="va">tmp_md</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>
<span class="va">tmp_md_sub</span> <span class="op">&lt;-</span> <span class="va">tmp_md_sub</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>
<span class="va">temp_adapt_plot_data</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_md_sub</span>, <span class="va">tmp_samp_scores</span>, 
                                         by <span class="op">=</span> <span class="st">"SampleID"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Now we have a new data frame that contains sample details and capscale values.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>          SampleID PLOT TREAT TREAT_T PAIR        CAP1        CAP2
1  P10_D00_010_C0E  P10     C       0    E -0.50625375  0.10348626
2  P02_D00_010_C0A  P02     C       0    A -0.48509579 -0.14580527
3  P04_D00_010_C0B  P04     C       0    B -0.57845671  0.03776429
4  P06_D00_010_C0C  P06     C       0    C -0.53336078  0.08936059
5  P08_D00_010_C0D  P08     C       0    D -0.68345909  0.10974979
6  P01_D00_010_W3A  P01     W       3    A -0.13776351 -0.22674265
7  P03_D00_010_W3B  P03     W       3    B -0.25679729  0.01972197
8  P05_D00_010_W3C  P05     W       3    C -0.38722889 -0.11236725
9  P07_D00_010_W3D  P07     W       3    D -0.01518359 -0.46741081
10 P09_D00_010_W3E  P09     W       3    E -0.11194937 -0.54460136
11 P01_D00_010_W8A  P01     W       8    A  0.78329851  0.09997647
12 P03_D00_010_W8B  P03     W       8    B  0.89642936 -1.09522700
13 P05_D00_010_W8C  P05     W       8    C  1.07566246 -0.26764258
14 P07_D00_010_W8D  P07     W       8    D  0.41185806  0.81673870
15 P09_D00_010_W8E  P09     W       8    E  0.52830039  1.58299886</code></pre>
</div>
</div>
<p>We can then do the same with the metadata vectors. Here though we only need the scores and parameter name.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb81"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">temp_adapt_md_scores</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tmp_auto_plt</span><span class="op">$</span><span class="va">plot_env</span><span class="op">$</span><span class="va">obj</span>, <span class="va">Score</span> <span class="op">==</span> <span class="st">"biplot"</span><span class="op">)</span>
<span class="va">temp_adapt_md_scores</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">temp_adapt_md_scores</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_md_scores</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>parameters <span class="op">=</span> <span class="va">Label</span>, .before <span class="op">=</span> <span class="va">CAP1</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"Label"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>       parameters        CAP1        CAP2
AG_Q10     AG_Q10 -0.07373951  0.23793282
BG_Q10     BG_Q10  0.72604945  0.10579764
BP_Q10     BP_Q10  0.07015779 -0.12413231
CE_Q10     CE_Q10 -0.37489864  0.55022094
P_Q10       P_Q10  0.87190011 -0.25252818
N_Q10       N_Q10  0.09422795 -0.32515404
S_Q10       S_Q10  0.91309230  0.55878516
XY_Q10     XY_Q10  0.92245895 -0.02395029
LP_Q10     LP_Q10 -0.90356227 -0.36479560
PX_Q10     PX_Q10 -0.87475774  0.18121787
CUEcn       CUEcn -0.23872665 -0.39707742
CUEcp       CUEcp  0.57949983 -0.62675335
Tmin         Tmin  0.99158847  0.19050834</code></pre>
</div>
</div>
<p>Lets run some quick correlations of metadata with ordination axes to see which parameters are significant. For this we use the vegan function <code>envfit</code>.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb83"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_samp_scores_sub</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_plot_data</span><span class="op">[</span>, <span class="fl">6</span><span class="op">:</span><span class="fl">7</span><span class="op">]</span>
<span class="va">tmp_samp_scores_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">tmp_samp_scores_sub</span><span class="op">)</span>
<span class="va">tmp_param_list</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_md_scores</span><span class="op">$</span><span class="va">parameters</span>

<span class="va">tmp_md_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">tmp_md</span>,  select <span class="op">=</span>  <span class="va">tmp_param_list</span><span class="op">)</span>

<span class="va">envfit_temp_adapt_md</span> <span class="op">&lt;-</span> <span class="fu">envfit</span><span class="op">(</span><span class="va">tmp_samp_scores_sub</span>, <span class="va">tmp_md_sub</span>,
                 perm <span class="op">=</span> <span class="fl">1000</span>, choices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
***VECTORS

           CAP1     CAP2     r2  Pr(&gt;r)  
AG_Q10 -0.30810  0.95136 0.0260 0.83117  
BG_Q10  0.99031  0.13889 0.2354 0.17782  
BP_Q10  0.50944 -0.86050 0.0086 0.95504  
CE_Q10 -0.58116  0.81379 0.1875 0.25075  
P_Q10   0.96427 -0.26494 0.3590 0.05395 .
N_Q10   0.28973 -0.95711 0.0479 0.76723  
S_Q10   0.86347  0.50440 0.4956 0.01499 *
XY_Q10  0.99973 -0.02334 0.3725 0.04895 *
LP_Q10 -0.93292 -0.36008 0.4131 0.04096 *
PX_Q10 -0.98131  0.19244 0.3483 0.07992 .
CUEcn  -0.53407 -0.84544 0.0908 0.55245  
CUEcp   0.69625 -0.71780 0.3103 0.12288  
Tmin    0.98341  0.18140 0.4456 0.02997 *
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb85"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">temp_adapt_md_signif_hits</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">envfit_temp_adapt_md</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span>, 
                                  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">envfit_temp_adapt_md</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span> 
                                  <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">envfit_temp_adapt_md</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">r</span> <span class="op">&gt;</span> <span class="fl">0.4</span><span class="op">)</span><span class="op">)</span>
<span class="va">temp_adapt_md_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">temp_adapt_md_signif_hits</span><span class="op">)</span>
<span class="va">temp_adapt_md_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">temp_adapt_md_signif_hits</span><span class="op">)</span>
<span class="va">temp_adapt_md_signif</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_md_scores</span><span class="op">[</span><span class="va">temp_adapt_md_scores</span><span class="op">$</span><span class="va">parameters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> 
                                               <span class="va">temp_adapt_md_signif_hits</span>,<span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><code>envfit</code> found that S_Q10, LP_Q10, Tmin were significantly correlated.</p>
<p>Now lets see if the same parameters are significant for the <code>envfit</code> and <code>bioenv</code> analyses.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Significant parameters from bioenv analysis."
[1] "CUEcp"  "LP_Q10" "P_Q10"  "S_Q10"  "Tmin"  
_____________________________________
[1] "Significant parameters from envfit analysis."
[1] "S_Q10"  "LP_Q10" "Tmin"  
_____________________________________
[1] "Found in bioenv but not envfit."
[1] "CUEcp" "P_Q10"
_____________________________________
[1] "Found in envfit but not bioenv."
character(0)
_____________________________________
[1] "Found in envfit and  bioenv."
[1] "S_Q10"  "LP_Q10" "Tmin"   "CUEcp"  "P_Q10" </code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb87"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="co">#new_temp_adapt_md_signif_hits &lt;- base::append(temp_adapt_md_signif_hits, temp_adapt_sig_diff)</span>
<span class="va">new_temp_adapt_md_signif_hits</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_sig_diff</span>
<span class="va">temp_adapt_md_signif_all</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_md_scores</span><span class="op">[</span><span class="va">temp_adapt_md_scores</span><span class="op">$</span><span class="va">parameters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> 
                                                   <span class="va">new_temp_adapt_md_signif_hits</span>,<span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Check. Next, we run <code>envfit</code> for the ASVs.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb88"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">envfit_temp_adapt_asv</span> <span class="op">&lt;-</span> <span class="fu">envfit</span><span class="op">(</span><span class="va">tmp_samp_scores_sub</span>, 
                                <span class="va">tmp_comm</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span><span class="op">(</span><span class="op">-</span><span class="va">tmp_comm</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>,
                                perm <span class="op">=</span> <span class="fl">1000</span>, choices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>

<span class="va">temp_adapt_asv_scores</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tmp_auto_plt</span><span class="op">$</span><span class="va">plot_env</span><span class="op">$</span><span class="va">obj</span>, <span class="va">Score</span> <span class="op">==</span> <span class="st">"species"</span><span class="op">)</span>

<span class="va">temp_adapt_asv_scores</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_asv_scores</span> <span class="op">%&gt;%</span> 
                         <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>parameters <span class="op">=</span> <span class="va">Label</span>, .before <span class="op">=</span> <span class="va">CAP1</span><span class="op">)</span> <span class="op">%&gt;%</span>
                         <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"Label"</span><span class="op">)</span>
<span class="va">temp_adapt_asv_scores</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
***VECTORS

          CAP1     CAP2     r2   Pr(&gt;r)    
ASV1   0.98984  0.14217 0.6541 0.009990 ** 
ASV2   0.55070 -0.83470 0.8953 0.000999 ***
ASV3   0.79919  0.60108 0.9513 0.000999 ***
ASV4  -0.12171  0.99257 0.3110 0.118881    
ASV5  -0.67644 -0.73649 0.1181 0.487512    
ASV6  -0.10078  0.99491 0.7028 0.003996 ** 
ASV8  -0.85434  0.51972 0.4279 0.043956 *  
ASV7  -0.57853 -0.81566 0.0602 0.676324    
ASV9   0.53361 -0.84573 0.2425 0.185814    
ASV10 -0.34761  0.93764 0.8031 0.000999 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb90"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">temp_adapt_asv_signif_hits</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">envfit_temp_adapt_asv</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span>, 
                                  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">envfit_temp_adapt_asv</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span> 
                                  <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">envfit_temp_adapt_asv</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">r</span> <span class="op">&gt;</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>
<span class="va">temp_adapt_asv_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">temp_adapt_asv_signif_hits</span><span class="op">)</span>
<span class="va">temp_adapt_asv_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">temp_adapt_asv_signif_hits</span><span class="op">)</span>
<span class="va">temp_adapt_asv_signif</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_asv_scores</span><span class="op">[</span><span class="va">temp_adapt_asv_scores</span><span class="op">$</span><span class="va">parameters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> 
                                                 <span class="va">temp_adapt_asv_signif_hits</span>,<span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>      parameters        CAP1        CAP2 variable_type
ASV3        ASV3  0.49583686  0.48216614           ASV
ASV6        ASV6 -0.01759546  0.20461984           ASV
ASV2        ASV2  0.23578207 -0.44144325           ASV
ASV10      ASV10 -0.02475900  0.08142465           ASV
ASV1        ASV1  1.18724098  0.33621587           ASV</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb92"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">temp_adapt_md_signif_all</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">&lt;-</span> <span class="st">"metadata"</span>
<span class="va">temp_adapt_asv_signif</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">&lt;-</span> <span class="st">"ASV"</span>
<span class="va">temp_adapt_bioplot_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">temp_adapt_md_signif_all</span>, <span class="va">temp_adapt_asv_signif</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The last thing to do is categorize parameters scores and ASV scores into different variable types for plotting.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb93"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">temp_adapt_bioplot_data_md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">temp_adapt_bioplot_data</span>, 
                                  <span class="va">temp_adapt_bioplot_data</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">==</span> <span class="st">"metadata"</span><span class="op">)</span>
<span class="va">temp_adapt_bioplot_data_asv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">temp_adapt_bioplot_data</span>, 
                                   <span class="va">temp_adapt_bioplot_data</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">==</span> <span class="st">"ASV"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<details><summary>
Show the code for the plot
</summary><div class="cell">
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">temp_adapt_cap_vals</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">temp_adapt_cap</span><span class="op">$</span><span class="va">CCA</span><span class="op">$</span><span class="va">eig</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>
<span class="va">temp_adapt_cap1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span><span class="op">(</span><span class="va">temp_adapt_cap_vals</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span>, digits<span class="op">=</span><span class="fl">3</span><span class="op">)</span>
<span class="va">temp_adapt_cap2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span><span class="op">(</span><span class="va">temp_adapt_cap_vals</span><span class="op">[</span><span class="fl">2</span>,<span class="op">]</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span>, digits<span class="op">=</span><span class="fl">3</span><span class="op">)</span>

<span class="va">cpa1_lab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"CAP1"</span>, <span class="st">" ("</span>, <span class="va">temp_adapt_cap1</span>, <span class="st">"%)"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
<span class="va">cpa2_lab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"CAP2"</span>, <span class="st">" ("</span>, <span class="va">temp_adapt_cap2</span>, <span class="st">"%)"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>

<span class="va">swel_col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#2271B2"</span>, <span class="st">"#71B222"</span>, <span class="st">"#B22271"</span><span class="op">)</span>

<span class="va">temp_adapt_plot</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">temp_adapt_plot_data</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">CAP1</span>, y <span class="op">=</span> <span class="va">CAP2</span>, shape <span class="op">=</span> <span class="va">TREAT</span>,
                           colour <span class="op">=</span> <span class="va">TREAT_T</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_colour_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">swel_col</span><span class="op">)</span> <span class="op">+</span>
<span class="co">#  geom_text(data = temp_adapt_plot_data, aes(x = CAP1, y = CAP2, #UNCOMMENT to add sample labels</span>
<span class="co">#                              label = SampleID), size = 3) + </span>
  <span class="fu">geom_segment</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="fl">0</span>, y <span class="op">=</span> <span class="fl">0</span>, xend <span class="op">=</span> <span class="va">CAP1</span>, yend <span class="op">=</span> <span class="va">CAP2</span><span class="op">)</span>,
               data <span class="op">=</span> <span class="va">temp_adapt_bioplot_data_md</span>, linetype <span class="op">=</span> <span class="st">"solid"</span>,
               arrow <span class="op">=</span> <span class="fu">arrow</span><span class="op">(</span>length <span class="op">=</span> <span class="fu">unit</span><span class="op">(</span><span class="fl">0.3</span>, <span class="st">"cm"</span><span class="op">)</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">0.4</span>,
               color <span class="op">=</span> <span class="st">"#191919"</span>, inherit.aes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_text</span><span class="op">(</span>data <span class="op">=</span> <span class="va">temp_adapt_bioplot_data_md</span>, 
            <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">CAP1</span>, y <span class="op">=</span> <span class="va">CAP2</span>, label <span class="op">=</span> <span class="va">parameters</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">3</span>,
            nudge_x <span class="op">=</span> <span class="fl">0.1</span>, nudge_y <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">+</span>
  <span class="co">### This code adds ASV vectors, Uncomment to add</span>
<span class="co">#  geom_segment(aes(x = 0, y = 0, xend = CAP1, yend = CAP2),</span>
<span class="co">#               data = temp_adapt_bioplot_data_asv, linetype = "solid",</span>
<span class="co">#               arrow = arrow(length = unit(0.3, "cm")), size = 0.2,</span>
<span class="co">#               color = "#676767") +</span>
<span class="co">#  geom_text(data = temp_adapt_bioplot_data_asv, </span>
<span class="co">#            aes(x = CAP1, y = CAP2, label = parameters), size = 2.5,</span>
<span class="co">#            nudge_x = 0.05, nudge_y = 0.05) +</span>
  <span class="fu">theme_classic</span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Capscale Analysis"</span>,
       subtitle <span class="op">=</span> <span class="st">"Temperature Adaptation"</span>,
       x <span class="op">=</span> <span class="va">cpa1_lab</span>, 
       y <span class="op">=</span> <span class="va">cpa2_lab</span><span class="op">)</span>
<span class="va">temp_adapt_plot</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_plot</span> <span class="op">+</span> <span class="fu">coord_fixed</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
<span class="va">temp_adapt_plot</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/png.html">png</a></span><span class="op">(</span><span class="st">"files/metadata/figures/ssu18_temp_adapt_capscale.png"</span>,
    height <span class="op">=</span> <span class="fl">16</span>, width <span class="op">=</span> <span class="fl">20</span>, units <span class="op">=</span> <span class="st">'cm'</span>, res <span class="op">=</span> <span class="fl">600</span>, bg <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span>
<span class="va">temp_adapt_plot</span>
<span class="fu"><a href="https://rdrr.io/r/base/invisible.html">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span><span class="op">(</span><span class="st">"files/metadata/figures/ssu18_temp_adapt_capscale.pdf"</span>,
    height <span class="op">=</span> <span class="fl">5</span>, width <span class="op">=</span> <span class="fl">6</span><span class="op">)</span>
<span class="va">temp_adapt_plot</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span><span class="op">(</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</details><div class="cell">

</div>
</section></section><section id="capscale-plots" class="level2 page-columns page-full"><h2 class="anchored" data-anchor-id="capscale-plots">Capscale Plots</h2>
<div class="column-body">
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-4-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-4-1" role="tab" aria-controls="tabset-4-1" aria-selected="true">Edaphic properties</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-4-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-4-2" role="tab" aria-controls="tabset-4-2" aria-selected="false">Soil Functional Response</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-4-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-4-3" role="tab" aria-controls="tabset-4-3" aria-selected="false">Temperature Adaptation</a></li>
</ul>
<div class="tab-content">
<div id="tabset-4-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-4-1-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-133-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 7 | Distance-based Redundancy Analysis (db-RDA) of PIME filtered data based on Bray-Curtis dissimilarity showing the relationships between community composition change versus edaphic properties.</strong></small></p>
</div>
<div id="tabset-4-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-4-2-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-134-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 8 | Distance-based Redundancy Analysis (db-RDA) of PIME filtered data based on Bray-Curtis dissimilarity showing the relationships between community composition change versus microbial functional response.</strong></small></p>
</div>
<div id="tabset-4-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-4-3-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-135-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 9 | Distance-based Redundancy Analysis (db-RDA) of PIME filtered data based on Bray-Curtis dissimilarity showing the relationships between community composition change versus temperature adaptation.</strong></small></p>
</div>
</div>
</div>
</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
</section></section><section id="its-1" class="level1 page-columns page-full"><h1>ITS</h1>
<section id="data-formating-1" class="level2"><h2 class="anchored" data-anchor-id="data-formating-1">Data Formating</h2>
<ol type="1">
<li>Transform <code>ps</code> Objects</li>
</ol>
<p>The first step is to transform the phyloseq objects.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb95"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">samp_ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"its18_ps_work"</span>, <span class="st">"its18_ps_pime"</span>, <span class="st">"its18_ps_perfect"</span>, 
             <span class="st">"its18_ps_work_otu"</span>, <span class="st">"its18_ps_pime_otu"</span>, <span class="st">"its18_ps_perfect_otu"</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">samp_ps</span><span class="op">)</span> <span class="op">{</span>
     <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">get</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span>
     <span class="va">tmp_ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transformcounts.html">transform_sample_counts</a></span><span class="op">(</span><span class="va">tmp_get</span>, <span class="kw">function</span><span class="op">(</span><span class="va">otu</span><span class="op">)</span> <span class="fl">1e5</span> <span class="op">*</span> <span class="va">otu</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">otu</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_ps</span><span class="op">@</span><span class="va">phy_tree</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
     <span class="va">tmp_ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_samples-methods.html">prune_samples</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_sums.html">sample_sums</a></span><span class="op">(</span><span class="va">tmp_ps</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="va">tmp_ps</span><span class="op">)</span>
     <span class="va">tmp_samp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_samp</span> <span class="op">&lt;-</span> <span class="va">tmp_samp</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span><span class="st">"TREAT_T"</span> <span class="op">=</span> <span class="st">"TEMP"</span><span class="op">)</span>
     <span class="va">tmp_ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/merge_phyloseq.html">merge_phyloseq</a></span><span class="op">(</span><span class="va">tmp_ps</span>, <span class="va">sample_data</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">tmp_ps</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">tmp_samp</span>
     <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_prop"</span><span class="op">)</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_name</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_ps</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<ol start="2" type="1">
<li>Format for <code>mctoolsr</code>
</li>
</ol>
<p>Then a bit of formatting to make the data compatible with <code>mctoolsr</code>.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb96"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">samp_ps</span><span class="op">)</span> <span class="op">{</span>
     <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_prop"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_tax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_tax</span><span class="op">$</span><span class="va">ASV_SEQ</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
     
     <span class="va">tmp_col_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">tmp_tax</span><span class="op">)</span>
     <span class="va">tmp_tax_merge</span> <span class="op">&lt;-</span> <span class="va">tmp_tax</span> <span class="op">%&gt;%</span> <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/unite.html">unite</a></span><span class="op">(</span><span class="va">taxonomy</span>, 
                                               <span class="fu">all_of</span><span class="op">(</span><span class="va">tmp_col_names</span><span class="op">)</span>, 
                                               sep <span class="op">=</span> <span class="st">";"</span><span class="op">)</span>
     <span class="va">tmp_tax_merge</span> <span class="op">&lt;-</span> <span class="va">tmp_tax_merge</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"#OTU_ID"</span><span class="op">)</span>
     <span class="va">tmp_otu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_otu</span> <span class="op">&lt;-</span> <span class="va">tmp_otu</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"#OTU_ID"</span><span class="op">)</span>
     <span class="va">tmp_otu_tax</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_otu</span>, <span class="va">tmp_tax_merge</span>, by <span class="op">=</span> <span class="st">"#OTU_ID"</span><span class="op">)</span>

     <span class="va">tmp_samp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_samp</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
     <span class="va">tmp_samp</span> <span class="op">&lt;-</span> <span class="va">tmp_samp</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"#SampleID"</span><span class="op">)</span>

     <span class="va">tmp_metad</span> <span class="op">&lt;-</span> <span class="va">metad</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span><span class="st">"#SampleID"</span> <span class="op">=</span> <span class="st">"id"</span><span class="op">)</span>
     <span class="va">tmp_metad</span><span class="op">[</span>,<span class="fl">2</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
     <span class="va">tmp_md</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_samp</span>, <span class="va">tmp_metad</span>, by <span class="op">=</span> <span class="st">"#SampleID"</span><span class="op">)</span>
     
     <span class="va">tmp_otu_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_otu_tax"</span><span class="op">)</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_otu_name</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_otu_name</span>, <span class="va">tmp_otu_tax</span><span class="op">)</span>

     <span class="va">tmp_md_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_md"</span><span class="op">)</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_md_name</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_md_name</span>, <span class="va">tmp_md</span><span class="op">)</span>

     <span class="va">tmp_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="st">"files/metadata/tables/"</span><span class="op">)</span>
     <span class="fu">write_delim</span><span class="op">(</span><span class="va">tmp_otu_tax</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">tmp_path</span>, <span class="va">tmp_otu_name</span>, <span class="st">".txt"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>, delim <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span>
     <span class="fu">write_delim</span><span class="op">(</span><span class="va">tmp_md</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">tmp_path</span>, <span class="va">tmp_md_name</span>, <span class="st">".txt"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>, delim <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb97"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">samp_ps</span><span class="op">)</span> <span class="op">{</span>
     <span class="va">tmp_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="st">"files/metadata/tables/"</span><span class="op">)</span>
     <span class="va">tmp_otu_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_otu_tax"</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_md_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_md"</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_tax_table_fp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">tmp_path</span>, <span class="va">tmp_otu_name</span>, <span class="st">".txt"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
     <span class="va">tmp_map_fp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">tmp_path</span>, <span class="va">tmp_md_name</span>, <span class="st">".txt"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
     <span class="va">tmp_input</span> <span class="op">&lt;-</span> <span class="fu">load_taxa_table</span><span class="op">(</span><span class="va">tmp_tax_table_fp</span>, <span class="va">tmp_map_fp</span><span class="op">)</span>
     <span class="va">tmp_input_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_mc"</span><span class="op">)</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_input_name</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_input_name</span>, <span class="va">tmp_input</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_md"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_otu_tax"</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="choose-data-set-1" class="level2"><h2 class="anchored" data-anchor-id="choose-data-set-1">Choose Data Set</h2>
<p>At this point in the code we need to choose a data set to use, formatted with <code>mctoolsr</code>. Remember, there are four choices:</p>
<ol type="1">
<li>Complete ASV data set.</li>
<li>Arbitrary filtered ASV data set.</li>
<li>PERfect filtered ASV data set.</li>
<li>PIME filtered ASV data set.</li>
</ol>
<p>This way, if we want to test other data sets we only need to change the name here.</p>
<div class="cell">
<div class="sourceCode" id="cb98"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/ls.html">objects</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_mc"</span><span class="op">)</span>
<span class="va">its18_select_mc</span> <span class="op">&lt;-</span> <span class="va">its18_ps_pime_mc</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="normality-tests-1" class="level2"><h2 class="anchored" data-anchor-id="normality-tests-1">Normality Tests</h2>
<p>Before proceeding, we need to test each parameter in the metadata to see which ones are and are not normally distributed. For that, we use the Shapiro-Wilk Normality Test. Here we only need one of the metadata files.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb99"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">temp_md</span> <span class="op">&lt;-</span> <span class="va">its18_select_mc</span><span class="op">$</span><span class="va">map_loaded</span>
<span class="va">temp_md</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">9</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">shap_results</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">temp_md</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_shap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/shapiro.test.html">shapiro.test</a></span><span class="op">(</span><span class="va">temp_md</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
  <span class="va">tmp_p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">tmp_shap</span><span class="op">$</span><span class="va">p.value</span>, digits <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>
  <span class="va">tmp_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/eval.html">eval</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Logic.html">isTRUE</a></span><span class="op">(</span><span class="va">tmp_shap</span><span class="op">$</span><span class="va">p.value</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span>
  <span class="va">shap_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">shap_results</span>, <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">i</span>, <span class="va">tmp_p</span>, <span class="va">tmp_res</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">shap_results</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"parameter"</span>, <span class="st">"p-value"</span>, <span class="st">"tranform"</span><span class="op">)</span>
<span class="va">shap_results</span>
<span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">shap_results</span>, <span class="va">tranform</span> <span class="op">==</span> <span class="st">"TRUE"</span><span class="op">)</span>
<span class="va">md_to_tranform</span> <span class="op">&lt;-</span> <span class="va">shap_results</span><span class="op">$</span><span class="va">parameter</span><span class="op">[</span><span class="va">shap_results</span><span class="op">$</span><span class="va">tranform</span> <span class="op">==</span> <span class="cn">TRUE</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"temp_md"</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<details><summary><strong>Show the results</strong> of each normality test for metadata parameters
</summary><p></p>
<p><small><strong>(ITS) Table 2 | Results of the Shapiro-Wilk Normality Tests. P-values in red are significance (p-value &lt; 0.05) meaning the parameter needs to be normalized.</strong></small></p>
<div class="cell">
<div class="cell-output-display">

<div id="htmlwidget-692cbfe2fd0f28261429" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-692cbfe2fd0f28261429">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"parameter":["AST","H2O","N","P","Al","Ca","Fe","K","Mg","Mn","Na","TEB","ECEC","pH","minPO4","minNH4","minNO3","minTIN","NH4","NO3","PO4","DOC","DON","DOCN","micC","micN","micP","micCN","micCP","micNP","AG_ase","BG_ase","BP_ase","CE_ase","P_ase","N_ase","S_ase","XY_ase","AG_Q10","BG_Q10","BP_Q10","CE_Q10","P_Q10","N_Q10","S_Q10","XY_Q10","CO2","LP_ase","LP_Q10","PX_ase","PX_Q10","CUEcn","CUEcp","NUE","PUE","Tmin","SI","enzCN","enzCP","enzNP"],"p-value":[0.00288,0.00272,0.09024,0.46522,0.00063,0.11893,0.00088,0.18425,0.27096,0.26036,0.11161,0.0107,0.01083,0.62168,0.11512,0.66069,0.00112,0.00133,0.05982,0.74913,0.13322,0.02215,0.29704,0.33006,0.20336,0.61244,0.17886,0.07687,0.6622,0.39825,0.93318,0.00403,0.00669,0.14601,0.00981,0.00733,0.10815,0.02719,0.26659,0.01649,0.11576,0.03868,0.34975,0.76801,0.40364,0.18425,0.00011,0.77547,0.80669,0.01754,2e-05,0.96287,0.01163,0.49521,0.00081,0.134,0.09324,0.13857,0.8759,0.1143],"tranform":[true,true,false,false,true,false,true,false,false,false,false,true,true,false,false,false,true,true,false,false,false,true,false,false,false,false,false,false,false,false,false,true,true,false,true,true,false,true,false,true,false,true,false,false,false,false,true,false,false,true,true,false,true,false,true,false,false,false,false,false]},"columns":[{"accessor":"parameter","name":"Parameter","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["AST","H2O","N","P","Al","Ca","Fe","K","Mg","Mn","Na","TEB","ECEC","pH","minPO4","minNH4","minNO3","minTIN","NH4","NO3","PO4","DOC","DON","DOCN","micC","micN","micP","micCN","micCP","micNP","AG_ase","BG_ase","BP_ase","CE_ase","P_ase","N_ase","S_ase","XY_ase","AG_Q10","BG_Q10","BP_Q10","CE_Q10","P_Q10","N_Q10","S_Q10","XY_Q10","CO2","LP_ase","LP_Q10","PX_ase","PX_Q10","CUEcn","CUEcp","NUE","PUE","Tmin","SI","enzCN","enzCP","enzNP"],"header":"Parameter","align":"left","headerStyle":{"borderRight":"1px solid #eee","fontSize":"0.9em"},"footerStyle":{"fontWeight":"bold"},"minWidth":150,"sticky":"left","style":{"borderRight":"1px solid #eee"}},{"accessor":"p-value","name":"p-value","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.00288","0.00272","0.09024","0.46522","0.00063","0.11893","0.00088","0.18425","0.27096","0.26036","0.11161","0.0107","0.01083","0.62168","0.11512","0.66069","0.00112","0.00133","0.05982","0.74913","0.13322","0.02215","0.29704","0.33006","0.20336","0.61244","0.17886","0.07687","0.6622","0.39825","0.93318","0.00403","0.00669","0.14601","0.00981","0.00733","0.10815","0.02719","0.26659","0.01649","0.11576","0.03868","0.34975","0.76801","0.40364","0.18425","0.00011","0.77547","0.80669","0.01754","0.00002","0.96287","0.01163","0.49521","0.00081","0.134","0.09324","0.13857","0.8759","0.1143"],"header":"p-value","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"},"style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"tranform","name":"tranform","type":"logical","sortable":true,"resizable":true,"filterable":true,"cell":["TRUE","TRUE","FALSE","FALSE","TRUE","FALSE","TRUE","FALSE","FALSE","FALSE","FALSE","TRUE","TRUE","FALSE","FALSE","FALSE","TRUE","TRUE","FALSE","FALSE","FALSE","TRUE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","TRUE","TRUE","FALSE","TRUE","TRUE","FALSE","TRUE","FALSE","TRUE","FALSE","TRUE","FALSE","FALSE","FALSE","FALSE","TRUE","FALSE","FALSE","TRUE","TRUE","FALSE","TRUE","FALSE","TRUE","FALSE","FALSE","FALSE","FALSE","FALSE"],"header":"tranform","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":60,"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"bc26461385648ff42e5213a4925456ab"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</details><p>Looks like we need to transform 21 metadata parameters.</p>
</section><section id="normalize-parameters-1" class="level2"><h2 class="anchored" data-anchor-id="normalize-parameters-1">Normalize Parameters</h2>
<p>Here we use the R package <a href="https://cran.r-project.org/web/packages/bestNormalize/vignettes/bestNormalize.html"><code>bestNormalize</code></a> to find and execute the best normalizing transformation. The function will test the following normalizing transformations:</p>
<ul>
<li><p><code>arcsinh_x</code> performs an arcsinh transformation.</p></li>
<li><p><code>boxcox</code> Perform a Box-Cox transformation and center/scale a vector to attempt normalization. <code>boxcox</code> estimates the optimal value of lambda for the Box-Cox transformation. The function will return an error if a user attempt to transform nonpositive data.</p></li>
<li><p><code>yeojohnson</code> Perform a Yeo-Johnson Transformation and center/scale a vector to attempt normalization. <code>yeojohnson</code> estimates the optimal value of lambda for the Yeo-Johnson transformation. The Yeo-Johnson is similar to the Box-Cox method, however it allows for the transformation of nonpositive data as well.</p></li>
<li><p><code>orderNorm</code> The Ordered Quantile (ORQ) normalization transformation, <code>orderNorm()</code>, is a rank-based procedure by which the values of a vector are mapped to their percentile, which is then mapped to the same percentile of the normal distribution. Without the presence of ties, this essentially guarantees that the transformation leads to a uniform distribution.</p></li>
<li><p><code>log_x</code> performs a simple log transformation. The parameter a is essentially estimated by the training set by default (estimated as the minimum possible to some extent epsilon), while the base must be specified beforehand. The default base of the log is 10.</p></li>
<li><p><code>sqrt_x</code> performs a simple square-root transformation. The parameter a is essentially estimated by the training set by default (estimated as the minimum possible), while the base must be specified beforehand.</p></li>
<li><p><code>exp_x</code> performs a simple exponential transformation.</p></li>
</ul>
<p>See this GitHub issue (<a href="https://github.com/petersonR/bestNormalize/issues/5">#5</a>) for a description on getting reproducible results. Apparently, you can get different results because the <code>bestNormalize()</code> function uses repeated cross-validation (and doesnt automatically set the seed), so the results will be slightly different each time the function is executed.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb100"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">119</span><span class="op">)</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">md_to_tranform</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_md</span> <span class="op">&lt;-</span> <span class="va">its18_select_mc</span><span class="op">$</span><span class="va">map_loaded</span>
  <span class="va">tmp_best_norm</span> <span class="op">&lt;-</span> <span class="fu">bestNormalize</span><span class="op">(</span><span class="va">tmp_md</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, r <span class="op">=</span> <span class="fl">1</span>, k <span class="op">=</span> <span class="fl">5</span>, loo <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
  <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_best_norm_test"</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_best_norm</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_name</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<details><summary>
Show the chosen transformations
</summary><div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
## bestNormalize Chosen transformation of AST ## 
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
   0%   25%   50%   75%  100% 
25.80 26.26 28.80 29.90 41.77 
_____________________________________

## bestNormalize Chosen transformation of H2O ## 
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
   0%   25%   50%   75%  100% 
0.204 0.261 0.377 0.385 0.401 
_____________________________________

## bestNormalize Chosen transformation of Al ## 
Standardized asinh(x) Transformation with 13 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 0.005384372 
 - sd (before standardization) = 0.007762057 
_____________________________________

## bestNormalize Chosen transformation of Fe ## 
Standardized asinh(x) Transformation with 13 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 0.01076821 
 - sd (before standardization) = 0.0103749 
_____________________________________

## bestNormalize Chosen transformation of TEB ## 
orderNorm Transformation with 13 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
41.6 43.3 45.1 48.7 61.7 
_____________________________________

## bestNormalize Chosen transformation of ECEC ## 
orderNorm Transformation with 13 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
41.7 43.3 45.3 48.8 61.8 
_____________________________________

## bestNormalize Chosen transformation of minNO3 ## 
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 11.31  19.67  22.71  37.33 106.42 
_____________________________________

## bestNormalize Chosen transformation of minTIN ## 
orderNorm Transformation with 13 nonmissing obs and ties
 - 12 unique values 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 12.96  21.50  23.99  42.08 110.75 
_____________________________________

## bestNormalize Chosen transformation of DOC ## 
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 33.20  44.30  56.68  63.79 118.74 
_____________________________________

## bestNormalize Chosen transformation of BG_ase ## 
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
  0%  25%  50%  75% 100% 
3.04 3.92 4.22 4.62 9.09 
_____________________________________

## bestNormalize Chosen transformation of BP_ase ## 
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
  0%  25%  50%  75% 100% 
1.77 2.33 3.02 3.50 7.28 
_____________________________________

## bestNormalize Chosen transformation of P_ase ## 
Standardized Box Cox Transformation with 13 nonmissing obs.:
 Estimated statistics:
 - lambda = -0.9999576 
 - mean (before standardization) = 0.9480149 
 - sd (before standardization) = 0.01593697 
_____________________________________

## bestNormalize Chosen transformation of N_ase ## 
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
  0%  25%  50%  75% 100% 
2.57 2.95 3.58 4.52 8.34 
_____________________________________

## bestNormalize Chosen transformation of XY_ase ## 
orderNorm Transformation with 13 nonmissing obs and ties
 - 11 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
0.66 0.76 1.11 1.36 2.52 
_____________________________________

## bestNormalize Chosen transformation of BG_Q10 ## 
orderNorm Transformation with 13 nonmissing obs and ties
 - 11 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
1.31 1.36 1.39 1.56 1.76 
_____________________________________

## bestNormalize Chosen transformation of CE_Q10 ## 
orderNorm Transformation with 13 nonmissing obs and ties
 - 12 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
1.76 1.87 1.95 2.00 2.36 
_____________________________________

## bestNormalize Chosen transformation of CO2 ## 
Standardized asinh(x) Transformation with 13 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 2.626573 
 - sd (before standardization) = 0.7396036 
_____________________________________

## bestNormalize Chosen transformation of PX_ase ## 
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 81.25  94.13 119.75 211.18 294.39 
_____________________________________

## bestNormalize Chosen transformation of PX_Q10 ## 
Standardized Yeo-Johnson Transformation with 13 nonmissing obs.:
 Estimated statistics:
 - lambda = -4.999963 
 - mean (before standardization) = 0.1968145 
 - sd (before standardization) = 0.001394949 
_____________________________________

## bestNormalize Chosen transformation of CUEcp ## 
orderNorm Transformation with 13 nonmissing obs and ties
 - 9 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
0.15 0.17 0.19 0.21 0.33 
_____________________________________

## bestNormalize Chosen transformation of PUE ## 
orderNorm Transformation with 13 nonmissing obs and ties
 - 6 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
0.77 0.88 0.90 0.91 0.92 
_____________________________________</code></pre>
</div>
</div>
</details><details><summary>
Show the complete <code>bestNormalize</code> results
</summary><div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
## Results of bestNormalize for AST ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 3.2051
 - Box-Cox: 3.2051
 - Center+scale: 4.1282
 - Exp(x): 17.9744
 - Log_b(x+a): 3.2051
 - orderNorm (ORQ): 0.1282
 - sqrt(x + a): 4.1282
 - Yeo-Johnson: 2.2821
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
   0%   25%   50%   75%  100% 
25.80 26.26 28.80 29.90 41.77 
_____________________________________

## Results of bestNormalize for H2O ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 7.2051
 - Box-Cox: 6.2821
 - Center+scale: 8.1282
 - Exp(x): 6.2821
 - Log_b(x+a): 7.2051
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 7.2051
 - Yeo-Johnson: 6.2821
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
   0%   25%   50%   75%  100% 
0.204 0.261 0.377 0.385 0.401 
_____________________________________

## Results of bestNormalize for Al ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 7.5128
 - Center+scale: 7.5128
 - Exp(x): 7.5128
 - Log_b(x+a): 9.359
 - orderNorm (ORQ): 7.5128
 - sqrt(x + a): 7.5128
 - Yeo-Johnson: 7.5128
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized asinh(x) Transformation with 13 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 0.005384372 
 - sd (before standardization) = 0.007762057 
_____________________________________

## Results of bestNormalize for Fe ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 7.2051
 - Center+scale: 7.2051
 - Exp(x): 7.2051
 - Log_b(x+a): 7.2051
 - orderNorm (ORQ): 7.2051
 - sqrt(x + a): 7.2051
 - Yeo-Johnson: 7.2051
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized asinh(x) Transformation with 13 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 0.01076821 
 - sd (before standardization) = 0.0103749 
_____________________________________

## Results of bestNormalize for TEB ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.6667
 - Box-Cox: 1.6667
 - Center+scale: 4.7436
 - Exp(x): 17.9744
 - Log_b(x+a): 1.6667
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 3.2051
 - Yeo-Johnson: 1.359
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
41.6 43.3 45.1 48.7 61.7 
_____________________________________

## Results of bestNormalize for ECEC ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.6667
 - Box-Cox: 1.6667
 - Center+scale: 3.2051
 - Exp(x): 17.9744
 - Log_b(x+a): 1.6667
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 3.2051
 - Yeo-Johnson: 1.359
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and ties
 - 10 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
41.7 43.3 45.3 48.8 61.8 
_____________________________________

## Results of bestNormalize for minNO3 ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 3.5128
 - Box-Cox: 3.5128
 - Center+scale: 5.359
 - Exp(x): 17.9744
 - Log_b(x+a): 3.5128
 - orderNorm (ORQ): 0.7436
 - sqrt(x + a): 2.5897
 - Yeo-Johnson: 3.5128
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 11.31  19.67  22.71  37.33 106.42 
_____________________________________

## Results of bestNormalize for minTIN ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.359
 - Box-Cox: 2.2821
 - Center+scale: 4.1282
 - Exp(x): 17.9744
 - Log_b(x+a): 1.359
 - orderNorm (ORQ): 0.7436
 - sqrt(x + a): 2.2821
 - Yeo-Johnson: 2.2821
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and ties
 - 12 unique values 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 12.96  21.50  23.99  42.08 110.75 
_____________________________________

## Results of bestNormalize for DOC ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 2.8974
 - Box-Cox: 2.8974
 - Center+scale: 3.5128
 - Exp(x): 17.9744
 - Log_b(x+a): 2.8974
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 1.9744
 - Yeo-Johnson: 2.8974
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 33.20  44.30  56.68  63.79 118.74 
_____________________________________

## Results of bestNormalize for BG_ase ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 3.8205
 - Box-Cox: 2.2821
 - Center+scale: 4.1282
 - Exp(x): 14.8974
 - Log_b(x+a): 3.8205
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 3.5128
 - Yeo-Johnson: 1.6667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
  0%  25%  50%  75% 100% 
3.04 3.92 4.22 4.62 9.09 
_____________________________________

## Results of bestNormalize for BP_ase ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.6667
 - Box-Cox: 1.6667
 - Center+scale: 4.1282
 - Exp(x): 5.6667
 - Log_b(x+a): 1.6667
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 1.6667
 - Yeo-Johnson: 1.6667
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
  0%  25%  50%  75% 100% 
1.77 2.33 3.02 3.50 7.28 
_____________________________________

## Results of bestNormalize for P_ase ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.6667
 - Box-Cox: 0.4359
 - Center+scale: 3.2051
 - Exp(x): 17.9744
 - Log_b(x+a): 1.6667
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 1.9744
 - Yeo-Johnson: 0.4359
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized Box Cox Transformation with 13 nonmissing obs.:
 Estimated statistics:
 - lambda = -0.9999576 
 - mean (before standardization) = 0.9480149 
 - sd (before standardization) = 0.01593697 
_____________________________________

## Results of bestNormalize for N_ase ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.9744
 - Box-Cox: 2.2821
 - Center+scale: 2.8974
 - Exp(x): 17.9744
 - Log_b(x+a): 1.9744
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 2.8974
 - Yeo-Johnson: 0.4359
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
  0%  25%  50%  75% 100% 
2.57 2.95 3.58 4.52 8.34 
_____________________________________

## Results of bestNormalize for XY_ase ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.0513
 - Box-Cox: 0.4359
 - Center+scale: 2.8974
 - Exp(x): 9.0513
 - Log_b(x+a): 1.0513
 - orderNorm (ORQ): 0.1282
 - sqrt(x + a): 1.359
 - Yeo-Johnson: 0.4359
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and ties
 - 11 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
0.66 0.76 1.11 1.36 2.52 
_____________________________________

## Results of bestNormalize for BG_Q10 ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.9744
 - Box-Cox: 1.9744
 - Center+scale: 2.2821
 - Exp(x): 5.359
 - Log_b(x+a): 1.9744
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 1.9744
 - Yeo-Johnson: 1.9744
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and ties
 - 11 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
1.31 1.36 1.39 1.56 1.76 
_____________________________________

## Results of bestNormalize for CE_Q10 ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.6667
 - Box-Cox: 1.6667
 - Center+scale: 1.6667
 - Exp(x): 4.1282
 - Log_b(x+a): 1.6667
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 1.6667
 - Yeo-Johnson: 0.7436
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and ties
 - 12 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
1.76 1.87 1.95 2.00 2.36 
_____________________________________

## Results of bestNormalize for CO2 ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.0513
 - Box-Cox: 3.5128
 - Center+scale: 11.5128
 - Exp(x): 17.9744
 - Log_b(x+a): 1.0513
 - orderNorm (ORQ): 1.0513
 - sqrt(x + a): 2.8974
 - Yeo-Johnson: 1.9744
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized asinh(x) Transformation with 13 nonmissing obs.:
 Relevant statistics:
 - mean (before standardization) = 2.626573 
 - sd (before standardization) = 0.7396036 
_____________________________________

## Results of bestNormalize for PX_ase ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 1.9744
 - Box-Cox: 2.2821
 - Center+scale: 4.1282
 - Exp(x): 17.9744
 - Log_b(x+a): 1.9744
 - orderNorm (ORQ): 0.4359
 - sqrt(x + a): 1.9744
 - Yeo-Johnson: 2.2821
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and no ties 
 - Original quantiles:
    0%    25%    50%    75%   100% 
 81.25  94.13 119.75 211.18 294.39 
_____________________________________

## Results of bestNormalize for PX_Q10 ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 14.5897
 - Box-Cox: 3.2051
 - Center+scale: 17.9744
 - Exp(x): 17.9744
 - Log_b(x+a): 11.5128
 - orderNorm (ORQ): 2.2821
 - sqrt(x + a): 17.9744
 - Yeo-Johnson: 1.0513
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
Standardized Yeo-Johnson Transformation with 13 nonmissing obs.:
 Estimated statistics:
 - lambda = -4.999963 
 - mean (before standardization) = 0.1968145 
 - sd (before standardization) = 0.001394949 
_____________________________________

## Results of bestNormalize for CUEcp ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 3.5128
 - Box-Cox: 1.359
 - Center+scale: 3.5128
 - Exp(x): 4.4359
 - Log_b(x+a): 1.359
 - orderNorm (ORQ): 0.7436
 - sqrt(x + a): 3.5128
 - Yeo-Johnson: 2.5897
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and ties
 - 9 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
0.15 0.17 0.19 0.21 0.33 
_____________________________________

## Results of bestNormalize for PUE ## 
Best Normalizing transformation with 13 Observations
 Estimated Normality Statistics (Pearson P / df, lower =&gt; more normal):
 - arcsinh(x): 5.9744
 - Box-Cox: 7.2051
 - Center+scale: 5.9744
 - Exp(x): 5.9744
 - Log_b(x+a): 7.2051
 - orderNorm (ORQ): 2.5897
 - sqrt(x + a): 5.9744
 - Yeo-Johnson: 5.9744
Estimation method: Out-of-sample via leave-one-out CV
 
Based off these, bestNormalize chose:
orderNorm Transformation with 13 nonmissing obs and ties
 - 6 unique values 
 - Original quantiles:
  0%  25%  50%  75% 100% 
0.77 0.88 0.90 0.91 0.92 
_____________________________________</code></pre>
</div>
</div>
</details><p>Great, now we can add the normalized transformed data back to our <code>mctoolsr</code> metadata file.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb103"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">its18_select_mc_norm</span> <span class="op">&lt;-</span> <span class="va">its18_select_mc</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">md_to_tranform</span><span class="op">)</span> <span class="op">{</span>
      <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">i</span>, <span class="st">"_best_norm_test"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
      <span class="va">tmp_new_data</span> <span class="op">&lt;-</span> <span class="va">tmp_get</span><span class="op">$</span><span class="va">x.t</span>
      <span class="va">its18_select_mc_norm</span><span class="op">$</span><span class="va">map_loaded</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">tmp_new_data</span>
      <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>     </code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>And rerun the Shapiro Tests.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb104"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">temp_md_norm</span> <span class="op">&lt;-</span> <span class="va">its18_select_mc_norm</span><span class="op">$</span><span class="va">map_loaded</span>
<span class="va">temp_md_norm</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">9</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">shap_results_norm</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">temp_md_norm</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_shap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/shapiro.test.html">shapiro.test</a></span><span class="op">(</span><span class="va">temp_md_norm</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
  <span class="va">tmp_p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">tmp_shap</span><span class="op">$</span><span class="va">p.value</span>, digits <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>
  <span class="va">tmp_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/eval.html">eval</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Logic.html">isTRUE</a></span><span class="op">(</span><span class="va">tmp_shap</span><span class="op">$</span><span class="va">p.value</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span>
  <span class="va">shap_results_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">shap_results_norm</span>, <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">i</span>, <span class="va">tmp_p</span>, <span class="va">tmp_res</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">shap_results_norm</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"parameter"</span>, <span class="st">"p-value"</span>, <span class="st">"tranform"</span><span class="op">)</span>
<span class="va">shap_results_norm</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"temp_md_norm"</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>And check if there are any parameters that are still significant for the normality test.</p>
<div class="cell">
<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">shap_results</span><span class="op">$</span><span class="va">parameter</span><span class="op">[</span><span class="va">shap_results_norm</span><span class="op">$</span><span class="va">tranform</span> <span class="op">==</span> <span class="cn">TRUE</span><span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Al" "Fe"</code></pre>
</div>
</div>
<p>Ok. Looks like <code>bestNormalize</code> was unable to find a suitable transformation for <code>Al</code> and <code>Fe</code>. This is likely because there is very little variation in these metadata and/or there are too few significant digits.</p>
</section><section id="normalized-metadata-1" class="level2"><h2 class="anchored" data-anchor-id="normalized-metadata-1">Normalized Metadata</h2>
<p>Finally, here is a new summary table that includes all of the normalized data.</p>
<p></p>
<div class="cell">
<div class="cell-output-display">
<a onclick="fetch('data:text/csv;base64,Sample_ID,ATAP,AST,H2O,N,P,Al,Ca,Fe,K,Mg,Mn,Na,TEB,ECEC,pH,minPO4,minNH4,minNO3,minTIN,NH4,NO3,PO4,DOC,DON,DOCN,micC,micN,micP,micCN,micCP,micNP,AG_ase,BG_ase,BP_ase,CE_ase,P_ase,N_ase,S_ase,XY_ase,AG_Q10,BG_Q10,BP_Q10,CE_Q10,P_Q10,N_Q10,S_Q10,XY_Q10,CO2,LP_ase,LP_Q10,PX_ase,PX_Q10,CUEcn,CUEcp,NUE,PUE,Tmin,SI,enzCN,enzCP,enzNP
P10_D00_010_C0E,26,-1.1983797023069245,-0.1940281424239263,0.26,0.32,1.8827863250321621,37.06,-0.0740607187919569,1.38,21.43,0.07,0.13,1.1983797023069245,1.1983797023069245,6.28,4.08,4.75,0.6151411045959734,0.6151411045959734,3.44,4.53,9.45,0.6151411045959734,4.91,14.67,334.93,72.64,64.23,4.49,5.54,1.17,0.28,0.6151411045959734,-1.1983797023069245,0.86,-0.9757591793385537,-0.3957252958144873,4.54,-0.869423773288886,1.54,-0.3957252958144873,1.41,-1.0200762327862016,1.27,1.47,1.33,1.65,-0.3701446488169234,32.22,1.8,-0.1940281424239263,-0.9099769941383964,0.22,-1.0200762327862016,0.87,1.1983797023069245,-2.46,1.29,0.5231,0.6607,1.263
P02_D00_010_C0A,26,-0.869423773288886,0.1940281424239262,0.3367,0.43,0.5946183354191794,36.65,-0.0740607187919569,0.69,8.16,0.04,0.11,0.1940281424239262,0.1940281424239262,6.18,1.42,3.92,0.1940281424239262,0.1940281424239262,4.81,3.13,7.58,1.1983797023069245,8.22,9.44,444.72,110.13,65.03,4.42,8.13,2.37,0.24,0,-0.3957252958144873,1.09,-0.899790783189584,-0.1940281424239263,3.8,-0.29338123212119327,1.51,-1.4260768722728472,1.34,-0.3957252958144873,1.22,1.54,1.23,1.59,-0.35493716248816587,38.61,1.81,-0.6151411045959736,1.0303644614798795,0.22,0.3957252958144873,0.87,-0.5024022233733554,0.14,1.44,0.5024,0.6554,1.3044
P04_D00_010_C0B,26,-0.3957252958144873,0.3957252958144873,0.3033,0.33,-0.6936784570402328,29.23,0.8896917537519375,1.18,14.59,0.12,0.14,0,0,5.73,1.5,3.17,-0.3957252958144873,-0.29338123212119327,5.57,3.22,7.95,0.1940281424239262,6.32,10.67,440.6,86.19,81.45,5.14,5.39,1.09,0.22,0.3957252958144873,0,1.09,0.16848471024782147,0.1940281424239262,4.92,0.09655861528963908,1.51,0.3957252958144873,1.58,1.1983797023069245,1.37,1.85,1.39,1.76,-0.6995772705384055,29.51,2.14,-1.7688250385187059,-0.2577866974987461,0.25,0,0.85,0.1940281424239262,-2.63,1.25,0.555,0.6173,1.1123
P06_D00_010_C0C,26,-0.6151411045959736,0,0.3333,0.3,-0.6936784570402328,32.07,-1.0379095561444154,0.57,21.32,0.06,0.14,0.869423773288886,0.869423773288886,5.93,0.75,1.83,-0.6151411045959736,-0.6151411045959736,2.92,2.27,7.04,0,5.42,11.09,390.66,72.61,85.7,5.26,4.5,0.83,0.21,-1.1983797023069245,-0.6151411045959736,0.75,-0.9786260896033859,-1.7688250385187059,5.45,-1.7688250385187059,1.83,-1.4260768722728472,1.44,1.7688250385187059,1.24,1.6,1.47,1.63,-0.6301974307121084,27.78,1.85,-0.869423773288886,0.9014289921178789,0.27,-1.0200762327862016,0.83,1.1983797023069245,-0.13,1.42,0.4624,0.5556,1.2017
P08_D00_010_C0D,26,-1.7688250385187059,1.7688250385187059,0.2433,0.27,-0.6936784570402328,25.49,-1.0379095561444154,1.86,14.93,0.02,0.08,-1.1983797023069245,-1.1983797023069245,6.01,3.67,4.33,1.7688250385187059,1.7688250385187059,2.04,3.71,5.97,-0.3957252958144873,5.22,11.57,277.37,56.55,33.4,4.72,11.55,2.45,0.15,-0.869423773288886,-1.7688250385187059,0.68,-1.5554415500549932,-0.6151411045959736,3.56,-1.1983797023069245,2.08,1.7688250385187059,1.74,0.6151411045959734,1.48,1.69,1.48,1.83,-1.5518071082373759,21.96,2.07,-1.1983797023069245,2.213486556253857,0.24,0.869423773288886,0.85,-0.5024022233733554,0,1.54,0.5058,0.6037,1.1935
P01_D00_010_W3A,29,0.3957252958144873,0.869423773288886,0.4267,0.35,-0.6936784570402328,31.43,-0.0740607187919569,1.72,9.98,0.04,0.15,-0.7363159173761294,-0.7363159173761294,6.01,1.33,3.67,1.1983797023069245,1.1983797023069245,8.03,4.67,9.52,1.7688250385187059,11.66,11.15,690.11,135.48,87.98,4.61,8.61,1.74,0.26,-0.3957252958144873,0.6151411045959734,1,0.19273113600096195,1.1983797023069245,6.08,0.3957252958144873,1.77,-0.1940281424239263,1.39,0.869423773288886,1.37,1.46,1.3,1.72,0.7230553113170921,52.8,1.76,0.3957252958144873,-0.3743915283557706,0.32,0.3957252958144873,0.77,-0.5024022233733554,0.07,1.51,0.4614,0.5911,1.2812
P03_D00_010_W3B,29,0.1940281424239262,1.1983797023069245,0.32,0.37,0.5946183354191794,27.19,-0.0740607187919569,0.85,16.51,0.06,0.12,-0.29338123212119327,-0.29338123212119327,5.86,2.83,3.33,0.869423773288886,0.869423773288886,3.11,2.57,8.66,0.3957252958144873,6.48,10.12,364.62,66.4,70.68,5.07,6.7,1.31,0.33,0.1940281424239262,0.1940281424239262,1.26,-0.0023491835959311615,0,5.72,0.6151411045959734,2.48,-0.7363159173761294,1.42,0.3957252958144873,1.27,1.54,1.27,1.72,-0.575902041401821,32.1,2,0,0.5210209313920766,0.26,-0.29338123212119327,0.84,0.1940281424239262,-0.29,1.49,0.5565,0.6412,1.1522
P07_D00_010_W3D,29,-0.1940281424239263,0.6151411045959734,0.4367,0.35,-0.6936784570402328,28.13,-0.0740607187919569,1.6,18.87,0.07,0.1,0.5024022233733554,0.5024022233733554,6.21,1,2.42,-0.1940281424239263,-0.29338123212119327,4.28,5.07,9.15,0.869423773288886,8.47,9.93,396.42,73.73,93.13,4.72,4.66,0.98,0.32,-0.1940281424239263,0.3957252958144873,1.07,0.5664624254071774,0.3957252958144873,5.86,0.09655861528963908,1.72,0.1940281424239262,1.46,-1.7688250385187059,1.28,1.36,1.3,1.63,-0.140654760149001,42.4,1.96,-0.3957252958144873,-0.8363720967673999,0.27,-1.7688250385187059,0.83,1.1983797023069245,0.65,1.61,0.4941,0.5762,1.1662
P09_D00_010_W3E,29,0,-0.3957252958144873,0.2833,0.36,1.8827863250321621,39.79,2.816522665016964,1.09,20.64,0.08,0.19,1.7688250385187059,1.7688250385187059,6.1,2.58,3.5,-1.1983797023069245,-0.869423773288886,2.03,2.9,8.28,-0.869423773288886,5.56,12.21,392.56,62.37,66.65,5.78,7.34,1.19,0.2,-0.6151411045959736,-0.869423773288886,0.86,-0.4488739003431186,-1.1983797023069245,5.26,-0.29338123212119327,1.67,-0.7363159173761294,1.56,-1.0200762327862016,1.35,1.48,1.32,1.84,0.41784963381380025,30.37,1.9,0.1940281424239262,0.44598468640242955,0.26,0.3957252958144873,0.84,-0.5024022233733554,-0.24,1.51,0.5112,0.6064,1.1862
P01_D00_010_W8A,34,0.869423773288886,-0.6151411045959736,0.4267,0.35,-0.6936784570402328,31.43,-0.0740607187919569,1.72,9.98,0.04,0.15,-0.7363159173761294,-0.7363159173761294,6.01,0.12,1.65,-1.7688250385187059,-1.7688250385187059,4.08,2.65,8.55,-1.1983797023069245,4.2,9.45,469,84,55,5.58,8.48,1.52,0.15,0.869423773288886,0.869423773288886,2.07,0.8178559159667375,0.6151411045959734,6.7,1.7688250385187059,2.01,1.1983797023069245,1.51,-0.6151411045959736,1.39,1.56,1.65,1.86,-0.6301974307121084,40.69,1.71,0.6151411045959734,-1.4879127689620189,0.23,-0.29338123212119327,0.87,0.1940281424239262,2.19,2.3,0.6456,0.7166,1.11
P03_D00_010_W8B,34,1.7688250385187059,-0.869423773288886,0.32,0.37,0.5946183354191794,27.19,-0.0740607187919569,0.85,16.51,0.06,0.12,-0.29338123212119327,-0.29338123212119327,5.86,0.2,1.28,0,0,2.5,1.06,2.8,-1.7688250385187059,2.5,13.29,117,20,11,5.85,10.54,1.8,0.02,-1.7688250385187059,-0.1940281424239263,0.65,-0.2007736037166385,-0.869423773288886,1.28,-0.6151411045959736,1.49,0.6151411045959734,1.55,-0.1940281424239263,1.51,1.75,1.39,1.89,2.389507003648687,13.94,1.77,0.869423773288886,-0.8363720967673999,0.29,1.7688250385187059,0.81,-1.7688250385187059,0.63,1.72,0.5302,0.4918,0.9277
P05_D00_010_W8C,34,1.1983797023069245,-1.1983797023069245,0.3333,0.33,-0.6936784570402328,26.14,-1.0379095561444154,0.67,14.83,0.05,0,-1.7688250385187059,-1.7688250385187059,5.87,0.11,1.18,0.3957252958144873,0.3957252958144873,3.91,2.3,4.25,-0.1940281424239263,4.3,10.46,440,97,56,4.54,7.91,1.74,0.15,1.7688250385187059,1.7688250385187059,1.5,1.8395142989961664,0.869423773288886,5.57,1.1983797023069245,1.73,0.869423773288886,1.42,0,1.48,1.68,1.6,1.88,1.0817261569892598,32.65,1.67,1.1983797023069245,-0.3153048171135492,0.19,1.1983797023069245,0.9,-1.1983797023069245,0.94,2.08,0.6931,0.6412,0.9251
P07_D00_010_W8D,34,0.6151411045959734,-1.7688250385187059,0.4367,0.35,-0.6936784570402328,28.13,-0.0740607187919569,1.6,18.87,0.07,0.1,0.5024022233733554,0.5024022233733554,6.21,0.02,2.6,-0.869423773288886,-1.1983797023069245,8.19,6.39,5.62,-0.6151411045959736,3.7,11.98,676,147,94,4.6,7.18,1.56,0.39,1.1983797023069245,1.1983797023069245,1.53,1.4765658032233824,1.7688250385187059,10.68,0.869423773288886,1.31,0,1.42,0.1940281424239262,1.33,1.54,1.51,1.91,0.34127974728706956,57.13,1.4,1.7688250385187059,-0.09416862804282078,0.26,-0.6151411045959736,0.84,0.6151411045959734,1.48,1.81,0.5933,0.6628,1.1173
').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'normalized_metadata.csv'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
</div>
<p><small><strong>(ITS) Table 3 | Results of bestNormalize function applied to each parameter.</strong></small></p>
<div class="cell">
<div class="cell-output-display">

<div id="htmlwidget-44a11444c376f1ec0e85" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-44a11444c376f1ec0e85">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Sample_ID":["P10_D00_010_C0E","P02_D00_010_C0A","P04_D00_010_C0B","P06_D00_010_C0C","P08_D00_010_C0D","P01_D00_010_W3A","P03_D00_010_W3B","P07_D00_010_W3D","P09_D00_010_W3E","P01_D00_010_W8A","P03_D00_010_W8B","P05_D00_010_W8C","P07_D00_010_W8D"],"ATAP":[26,26,26,26,26,29,29,29,29,34,34,34,34],"AST":[-1.198,-0.869,-0.396,-0.615,-1.769,0.396,0.194,-0.194,0,0.869,1.769,1.198,0.615],"H2O":[-0.194,0.194,0.396,0,1.769,0.869,1.198,0.615,-0.396,-0.615,-0.869,-1.198,-1.769],"N":[0.26,0.337,0.303,0.333,0.243,0.427,0.32,0.437,0.283,0.427,0.32,0.333,0.437],"P":[0.32,0.43,0.33,0.3,0.27,0.35,0.37,0.35,0.36,0.35,0.37,0.33,0.35],"Al":[1.883,0.595,-0.694,-0.694,-0.694,-0.694,0.595,-0.694,1.883,-0.694,0.595,-0.694,-0.694],"Ca":[37.06,36.65,29.23,32.07,25.49,31.43,27.19,28.13,39.79,31.43,27.19,26.14,28.13],"Fe":[-0.074,-0.074,0.89,-1.038,-1.038,-0.074,-0.074,-0.074,2.817,-0.074,-0.074,-1.038,-0.074],"K":[1.38,0.69,1.18,0.57,1.86,1.72,0.85,1.6,1.09,1.72,0.85,0.67,1.6],"Mg":[21.43,8.16,14.59,21.32,14.93,9.98,16.51,18.87,20.64,9.98,16.51,14.83,18.87],"Mn":[0.07,0.04,0.12,0.06,0.02,0.04,0.06,0.07,0.08,0.04,0.06,0.05,0.07],"Na":[0.13,0.11,0.14,0.14,0.08,0.15,0.12,0.1,0.19,0.15,0.12,0,0.1],"TEB":[1.198,0.194,0,0.869,-1.198,-0.736,-0.293,0.502,1.769,-0.736,-0.293,-1.769,0.502],"ECEC":[1.198,0.194,0,0.869,-1.198,-0.736,-0.293,0.502,1.769,-0.736,-0.293,-1.769,0.502],"pH":[6.28,6.18,5.73,5.93,6.01,6.01,5.86,6.21,6.1,6.01,5.86,5.87,6.21],"minPO4":[4.08,1.42,1.5,0.75,3.67,1.33,2.83,1,2.58,0.12,0.2,0.11,0.02],"minNH4":[4.75,3.92,3.17,1.83,4.33,3.67,3.33,2.42,3.5,1.65,1.28,1.18,2.6],"minNO3":[0.615,0.194,-0.396,-0.615,1.769,1.198,0.869,-0.194,-1.198,-1.769,0,0.396,-0.869],"minTIN":[0.615,0.194,-0.293,-0.615,1.769,1.198,0.869,-0.293,-0.869,-1.769,0,0.396,-1.198],"NH4":[3.44,4.81,5.57,2.92,2.04,8.03,3.11,4.28,2.03,4.08,2.5,3.91,8.19],"NO3":[4.53,3.13,3.22,2.27,3.71,4.67,2.57,5.07,2.9,2.65,1.06,2.3,6.39],"PO4":[9.45,7.58,7.95,7.04,5.97,9.52,8.66,9.15,8.28,8.55,2.8,4.25,5.62],"DOC":[0.615,1.198,0.194,0,-0.396,1.769,0.396,0.869,-0.869,-1.198,-1.769,-0.194,-0.615],"DON":[4.91,8.22,6.32,5.42,5.22,11.66,6.48,8.47,5.56,4.2,2.5,4.3,3.7],"DOCN":[14.67,9.44,10.67,11.09,11.57,11.15,10.12,9.93,12.21,9.45,13.29,10.46,11.98],"micC":[334.93,444.72,440.6,390.66,277.37,690.11,364.62,396.42,392.56,469,117,440,676],"micN":[72.64,110.13,86.19,72.61,56.55,135.48,66.4,73.73,62.37,84,20,97,147],"micP":[64.23,65.03,81.45,85.7,33.4,87.98,70.68,93.13,66.65,55,11,56,94],"micCN":[4.49,4.42,5.14,5.26,4.72,4.61,5.07,4.72,5.78,5.58,5.85,4.54,4.6],"micCP":[5.54,8.13,5.39,4.5,11.55,8.61,6.7,4.66,7.34,8.48,10.54,7.91,7.18],"micNP":[1.17,2.37,1.09,0.83,2.45,1.74,1.31,0.98,1.19,1.52,1.8,1.74,1.56],"AG_ase":[0.28,0.24,0.22,0.21,0.15,0.26,0.33,0.32,0.2,0.15,0.02,0.15,0.39],"BG_ase":[0.615,0,0.396,-1.198,-0.869,-0.396,0.194,-0.194,-0.615,0.869,-1.769,1.769,1.198],"BP_ase":[-1.198,-0.396,0,-0.615,-1.769,0.615,0.194,0.396,-0.869,0.869,-0.194,1.769,1.198],"CE_ase":[0.86,1.09,1.09,0.75,0.68,1,1.26,1.07,0.86,2.07,0.65,1.5,1.53],"P_ase":[-0.976,-0.9,0.168,-0.979,-1.555,0.193,-0.002,0.566,-0.449,0.818,-0.201,1.84,1.477],"N_ase":[-0.396,-0.194,0.194,-1.769,-0.615,1.198,0,0.396,-1.198,0.615,-0.869,0.869,1.769],"S_ase":[4.54,3.8,4.92,5.45,3.56,6.08,5.72,5.86,5.26,6.7,1.28,5.57,10.68],"XY_ase":[-0.869,-0.293,0.097,-1.769,-1.198,0.396,0.615,0.097,-0.293,1.769,-0.615,1.198,0.869],"AG_Q10":[1.54,1.51,1.51,1.83,2.08,1.77,2.48,1.72,1.67,2.01,1.49,1.73,1.31],"BG_Q10":[-0.396,-1.426,0.396,-1.426,1.769,-0.194,-0.736,0.194,-0.736,1.198,0.615,0.869,0],"BP_Q10":[1.41,1.34,1.58,1.44,1.74,1.39,1.42,1.46,1.56,1.51,1.55,1.42,1.42],"CE_Q10":[-1.0200762327862,-0.395725295814487,1.19837970230692,1.76882503851871,0.615141104595973,0.869423773288886,0.395725295814487,-1.76882503851871,-1.0200762327862,-0.615141104595974,-0.194028142423926,0,0.194028142423926],"P_Q10":[1.27,1.22,1.37,1.24,1.48,1.37,1.27,1.28,1.35,1.39,1.51,1.48,1.33],"N_Q10":[1.47,1.54,1.85,1.6,1.69,1.46,1.54,1.36,1.48,1.56,1.75,1.68,1.54],"S_Q10":[1.33,1.23,1.39,1.47,1.48,1.3,1.27,1.3,1.32,1.65,1.39,1.6,1.51],"XY_Q10":[1.65,1.59,1.76,1.63,1.83,1.72,1.72,1.63,1.84,1.86,1.89,1.88,1.91],"CO2":[-0.37,-0.355,-0.7,-0.63,-1.552,0.723,-0.576,-0.141,0.418,-0.63,2.39,1.082,0.341],"LP_ase":[32.22,38.61,29.51,27.78,21.96,52.8,32.1,42.4,30.37,40.69,13.94,32.65,57.13],"LP_Q10":[1.8,1.81,2.14,1.85,2.07,1.76,2,1.96,1.9,1.71,1.77,1.67,1.4],"PX_ase":[-0.194,-0.615,-1.769,-0.869,-1.198,0.396,0,-0.396,0.194,0.615,0.869,1.198,1.769],"PX_Q10":[-0.91,1.03,-0.258,0.901,2.213,-0.374,0.521,-0.836,0.446,-1.488,-0.836,-0.315,-0.094],"CUEcn":[0.22,0.22,0.25,0.27,0.24,0.32,0.26,0.27,0.26,0.23,0.29,0.19,0.26],"CUEcp":[-1.02,0.396,0,-1.02,0.869,0.396,-0.293,-1.769,0.396,-0.293,1.769,1.198,-0.615],"NUE":[0.87,0.87,0.85,0.83,0.85,0.77,0.84,0.83,0.84,0.87,0.81,0.9,0.84],"PUE":[1.198,-0.502,0.194,1.198,-0.502,-0.502,0.194,1.198,-0.502,0.194,-1.769,-1.198,0.615],"Tmin":[-2.46,0.14,-2.63,-0.13,0,0.07,-0.29,0.65,-0.24,2.19,0.63,0.94,1.48],"SI":[1.29,1.44,1.25,1.42,1.54,1.51,1.49,1.61,1.51,2.3,1.72,2.08,1.81],"enzCN":[0.523,0.502,0.555,0.462,0.506,0.461,0.556,0.494,0.511,0.646,0.53,0.693,0.593],"enzCP":[0.661,0.655,0.617,0.556,0.604,0.591,0.641,0.576,0.606,0.717,0.492,0.641,0.663],"enzNP":[1.263,1.304,1.112,1.202,1.194,1.281,1.152,1.166,1.186,1.11,0.928,0.925,1.117]},"columns":[{"accessor":"Sample_ID","name":"Sample ID","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["P10_D00_010_C0E","P02_D00_010_C0A","P04_D00_010_C0B","P06_D00_010_C0C","P08_D00_010_C0D","P01_D00_010_W3A","P03_D00_010_W3B","P07_D00_010_W3D","P09_D00_010_W3E","P01_D00_010_W8A","P03_D00_010_W8B","P05_D00_010_W8C","P07_D00_010_W8D"],"header":"Sample ID","align":"left","headerStyle":{"borderRight":"1px solid #eee","fontSize":"0.9em"},"footerStyle":{"fontWeight":"bold"},"minWidth":150,"sticky":"left","style":{"borderRight":"1px solid #eee"}},{"accessor":"ATAP","name":"ATAP","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["26","26","26","26","26","29","29","29","29","34","34","34","34"],"header":"ATAP","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"AST","name":"AST","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-1.198","-0.869","-0.396","-0.615","-1.769","0.396","0.194","-0.194","0.0","0.869","1.769","1.198","0.615"],"header":"AST","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"H2O","name":"H2O","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.194","0.194","0.396","0.0","1.769","0.869","1.198","0.615","-0.396","-0.615","-0.869","-1.198","-1.769"],"header":"H2O","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"N","name":"N","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.26","0.337","0.303","0.333","0.243","0.427","0.32","0.437","0.283","0.427","0.32","0.333","0.437"],"header":"N","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"P","name":"P","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.32","0.43","0.33","0.3","0.27","0.35","0.37","0.35","0.36","0.35","0.37","0.33","0.35"],"header":"P","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Al","name":"Al","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.883","0.595","-0.694","-0.694","-0.694","-0.694","0.595","-0.694","1.883","-0.694","0.595","-0.694","-0.694"],"header":"Al","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Ca","name":"Ca","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["37.06","36.65","29.23","32.07","25.49","31.43","27.19","28.13","39.79","31.43","27.19","26.14","28.13"],"header":"Ca","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Fe","name":"Fe","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.074","-0.074","0.89","-1.038","-1.038","-0.074","-0.074","-0.074","2.817","-0.074","-0.074","-1.038","-0.074"],"header":"Fe","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"K","name":"K","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.38","0.69","1.18","0.57","1.86","1.72","0.85","1.6","1.09","1.72","0.85","0.67","1.6"],"header":"K","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Mg","name":"Mg","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["21.43","8.16","14.59","21.32","14.93","9.98","16.51","18.87","20.64","9.98","16.51","14.83","18.87"],"header":"Mg","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Mn","name":"Mn","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.07","0.04","0.12","0.06","0.02","0.04","0.06","0.07","0.08","0.04","0.06","0.05","0.07"],"header":"Mn","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Na","name":"Na","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.13","0.11","0.14","0.14","0.08","0.15","0.12","0.1","0.19","0.15","0.12","0.0","0.1"],"header":"Na","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"TEB","name":"TEB","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.198","0.194","0.0","0.869","-1.198","-0.736","-0.293","0.502","1.769","-0.736","-0.293","-1.769","0.502"],"header":"TEB","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"ECEC","name":"ECEC","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.198","0.194","0.0","0.869","-1.198","-0.736","-0.293","0.502","1.769","-0.736","-0.293","-1.769","0.502"],"header":"ECEC","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"pH","name":"pH","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["6.28","6.18","5.73","5.93","6.01","6.01","5.86","6.21","6.1","6.01","5.86","5.87","6.21"],"header":"pH","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"minPO4","name":"minPO4","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["4.08","1.42","1.5","0.75","3.67","1.33","2.83","1.0","2.58","0.12","0.2","0.11","0.02"],"header":"minPO4","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"minNH4","name":"minNH4","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["4.75","3.92","3.17","1.83","4.33","3.67","3.33","2.42","3.5","1.65","1.28","1.18","2.6"],"header":"minNH4","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"minNO3","name":"minNO3","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.615","0.194","-0.396","-0.615","1.769","1.198","0.869","-0.194","-1.198","-1.769","0.0","0.396","-0.869"],"header":"minNO3","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"minTIN","name":"minTIN","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.615","0.194","-0.293","-0.615","1.769","1.198","0.869","-0.293","-0.869","-1.769","0.0","0.396","-1.198"],"header":"minTIN","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"NH4","name":"NH4","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["3.44","4.81","5.57","2.92","2.04","8.03","3.11","4.28","2.03","4.08","2.5","3.91","8.19"],"header":"NH4","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"NO3","name":"NO3","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["4.53","3.13","3.22","2.27","3.71","4.67","2.57","5.07","2.9","2.65","1.06","2.3","6.39"],"header":"NO3","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"PO4","name":"PO4","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["9.45","7.58","7.95","7.04","5.97","9.52","8.66","9.15","8.28","8.55","2.8","4.25","5.62"],"header":"PO4","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"DOC","name":"DOC","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.615","1.198","0.194","0.0","-0.396","1.769","0.396","0.869","-0.869","-1.198","-1.769","-0.194","-0.615"],"header":"DOC","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"DON","name":"DON","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["4.91","8.22","6.32","5.42","5.22","11.66","6.48","8.47","5.56","4.2","2.5","4.3","3.7"],"header":"DON","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"DOCN","name":"DOCN","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["14.67","9.44","10.67","11.09","11.57","11.15","10.12","9.93","12.21","9.45","13.29","10.46","11.98"],"header":"DOCN","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"micC","name":"micC","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["334.93","444.72","440.6","390.66","277.37","690.11","364.62","396.42","392.56","469.0","117.0","440.0","676.0"],"header":"micC","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"micN","name":"micN","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["72.64","110.13","86.19","72.61","56.55","135.48","66.4","73.73","62.37","84.0","20.0","97.0","147.0"],"header":"micN","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"micP","name":"micP","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["64.23","65.03","81.45","85.7","33.4","87.98","70.68","93.13","66.65","55.0","11.0","56.0","94.0"],"header":"micP","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"micCN","name":"micCN","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["4.49","4.42","5.14","5.26","4.72","4.61","5.07","4.72","5.78","5.58","5.85","4.54","4.6"],"header":"micCN","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"micCP","name":"micCP","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["5.54","8.13","5.39","4.5","11.55","8.61","6.7","4.66","7.34","8.48","10.54","7.91","7.18"],"header":"micCP","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"micNP","name":"micNP","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.17","2.37","1.09","0.83","2.45","1.74","1.31","0.98","1.19","1.52","1.8","1.74","1.56"],"header":"micNP","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"AG_ase","name":"AG_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.28","0.24","0.22","0.21","0.15","0.26","0.33","0.32","0.2","0.15","0.02","0.15","0.39"],"header":"AG ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"BG_ase","name":"BG_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.615","0.0","0.396","-1.198","-0.869","-0.396","0.194","-0.194","-0.615","0.869","-1.769","1.769","1.198"],"header":"BG ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"BP_ase","name":"BP_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-1.198","-0.396","0.0","-0.615","-1.769","0.615","0.194","0.396","-0.869","0.869","-0.194","1.769","1.198"],"header":"BP ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"CE_ase","name":"CE_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.86","1.09","1.09","0.75","0.68","1.0","1.26","1.07","0.86","2.07","0.65","1.5","1.53"],"header":"CE ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"P_ase","name":"P_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.976","-0.9","0.168","-0.979","-1.555","0.193","-0.002","0.566","-0.449","0.818","-0.201","1.84","1.477"],"header":"P ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"N_ase","name":"N_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.396","-0.194","0.194","-1.769","-0.615","1.198","0.0","0.396","-1.198","0.615","-0.869","0.869","1.769"],"header":"N ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"S_ase","name":"S_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["4.54","3.8","4.92","5.45","3.56","6.08","5.72","5.86","5.26","6.7","1.28","5.57","10.68"],"header":"S ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"XY_ase","name":"XY_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.869","-0.293","0.097","-1.769","-1.198","0.396","0.615","0.097","-0.293","1.769","-0.615","1.198","0.869"],"header":"XY ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"AG_Q10","name":"AG_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.54","1.51","1.51","1.83","2.08","1.77","2.48","1.72","1.67","2.01","1.49","1.73","1.31"],"header":"AG Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"BG_Q10","name":"BG_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.396","-1.426","0.396","-1.426","1.769","-0.194","-0.736","0.194","-0.736","1.198","0.615","0.869","0.0"],"header":"BG Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"BP_Q10","name":"BP_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.41","1.34","1.58","1.44","1.74","1.39","1.42","1.46","1.56","1.51","1.55","1.42","1.42"],"header":"BP Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"CE_Q10","name":"CE_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-1.020076","-0.3957253","1.19838","1.768825","0.6151411","0.8694238","0.3957253","-1.768825","-1.020076","-0.6151411","-0.1940281","0.0","0.1940281"],"header":"CE Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"P_Q10","name":"P_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.27","1.22","1.37","1.24","1.48","1.37","1.27","1.28","1.35","1.39","1.51","1.48","1.33"],"header":"P Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"N_Q10","name":"N_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.47","1.54","1.85","1.6","1.69","1.46","1.54","1.36","1.48","1.56","1.75","1.68","1.54"],"header":"N Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"S_Q10","name":"S_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.33","1.23","1.39","1.47","1.48","1.3","1.27","1.3","1.32","1.65","1.39","1.6","1.51"],"header":"S Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"XY_Q10","name":"XY_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.65","1.59","1.76","1.63","1.83","1.72","1.72","1.63","1.84","1.86","1.89","1.88","1.91"],"header":"XY Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"CO2","name":"CO2","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.37","-0.355","-0.7","-0.63","-1.552","0.723","-0.576","-0.141","0.418","-0.63","2.39","1.082","0.341"],"header":"CO2","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"LP_ase","name":"LP_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["32.22","38.61","29.51","27.78","21.96","52.8","32.1","42.4","30.37","40.69","13.94","32.65","57.13"],"header":"LP ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"LP_Q10","name":"LP_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.8","1.81","2.14","1.85","2.07","1.76","2.0","1.96","1.9","1.71","1.77","1.67","1.4"],"header":"LP Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"PX_ase","name":"PX_ase","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.194","-0.615","-1.769","-0.869","-1.198","0.396","0.0","-0.396","0.194","0.615","0.869","1.198","1.769"],"header":"PX ase","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"PX_Q10","name":"PX_Q10","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-0.91","1.03","-0.258","0.901","2.213","-0.374","0.521","-0.836","0.446","-1.488","-0.836","-0.315","-0.094"],"header":"PX Q10","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"CUEcn","name":"CUEcn","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.22","0.22","0.25","0.27","0.24","0.32","0.26","0.27","0.26","0.23","0.29","0.19","0.26"],"header":"CUEcn","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"CUEcp","name":"CUEcp","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-1.02","0.396","0.0","-1.02","0.869","0.396","-0.293","-1.769","0.396","-0.293","1.769","1.198","-0.615"],"header":"CUEcp","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"NUE","name":"NUE","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.87","0.87","0.85","0.83","0.85","0.77","0.84","0.83","0.84","0.87","0.81","0.9","0.84"],"header":"NUE","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"PUE","name":"PUE","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.198","-0.502","0.194","1.198","-0.502","-0.502","0.194","1.198","-0.502","0.194","-1.769","-1.198","0.615"],"header":"PUE","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"Tmin","name":"Tmin","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["-2.46","0.14","-2.63","-0.13","0.0","0.07","-0.29","0.65","-0.24","2.19","0.63","0.94","1.48"],"header":"Tmin","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"SI","name":"SI","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.29","1.44","1.25","1.42","1.54","1.51","1.49","1.61","1.51","2.3","1.72","2.08","1.81"],"header":"SI","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"enzCN","name":"enzCN","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.523","0.502","0.555","0.462","0.506","0.461","0.556","0.494","0.511","0.646","0.53","0.693","0.593"],"header":"enzCN","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"enzCP","name":"enzCP","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["0.661","0.655","0.617","0.556","0.604","0.591","0.641","0.576","0.606","0.717","0.492","0.641","0.663"],"header":"enzCP","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}},{"accessor":"enzNP","name":"enzNP","type":"numeric","sortable":true,"resizable":true,"filterable":true,"cell":["1.263","1.304","1.112","1.202","1.194","1.281","1.152","1.166","1.186","1.11","0.928","0.925","1.117"],"header":"enzNP","align":"center","headerStyle":{"fontSize":"0.8em"},"footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,13],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"8a4d641dcb3a670abee2f295258383fc"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="cell">

</div>
</section><section id="autocorrelation-tests-1" class="level2"><h2 class="anchored" data-anchor-id="autocorrelation-tests-1">Autocorrelation Tests</h2>
<p>Next, we test the metadata for autocorrelations. Do we do this on the original data or the transformed data? No idea, so lets do both.</p>
<section id="split-metadata-1" class="level3"><h3 class="anchored" data-anchor-id="split-metadata-1">Split Metadata</h3>
<p>We need to split the data into different groups.</p>
<ol type="A">
<li>Environmental and edaphic properties</li>
<li>Microbial functional responses</li>
<li>Temperature adaptation properties</li>
</ol>
<p>We first create lists of metadata parameters.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb107"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">div</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"PLOT"</span>, <span class="st">"TREAT"</span>, <span class="st">"TREAT_T"</span>, <span class="st">"PAIR"</span>, <span class="st">"Observed"</span>, <span class="st">"Shannon_exp"</span>, 
         <span class="st">"InvSimpson"</span>, <span class="st">"ATAP"</span><span class="op">)</span>

<span class="va">edaphic</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"AST"</span>, <span class="st">"H2O"</span>, <span class="st">"N"</span>, <span class="st">"P"</span>, <span class="st">"Al"</span>, 
             <span class="st">"Ca"</span>, <span class="st">"Fe"</span>, <span class="st">"K"</span>, <span class="st">"Mg"</span>, <span class="st">"Mn"</span>, <span class="st">"Na"</span>, <span class="st">"TEB"</span>, <span class="st">"ECEC"</span>, <span class="st">"pH"</span>, 
             <span class="st">"NH4"</span>, <span class="st">"NO3"</span>, <span class="st">"PO4"</span>, <span class="st">"DOC"</span>, <span class="st">"DON"</span>, <span class="st">"DOCN"</span><span class="op">)</span>

<span class="va">soil_funct</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"micC"</span>, <span class="st">"micN"</span>, <span class="st">"micP"</span>, <span class="st">"micCN"</span>, <span class="st">"micCP"</span>, <span class="st">"micNP"</span>, 
                <span class="st">"AG_ase"</span>, <span class="st">"BG_ase"</span>, <span class="st">"BP_ase"</span>, <span class="st">"CE_ase"</span>, <span class="st">"P_ase"</span>, <span class="st">"N_ase"</span>, 
                <span class="st">"S_ase"</span>, <span class="st">"XY_ase"</span>,  <span class="st">"LP_ase"</span>, <span class="st">"PX_ase"</span>, <span class="st">"CO2"</span>, 
                <span class="st">"enzCN"</span>, <span class="st">"enzCP"</span>, <span class="st">"enzNP"</span><span class="op">)</span>

<span class="va">temp_adapt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"AG_Q10"</span>, <span class="st">"BG_Q10"</span>, <span class="st">"BP_Q10"</span>, <span class="st">"CE_Q10"</span>, <span class="st">"P_Q10"</span>, <span class="st">"N_Q10"</span>, 
                <span class="st">"S_Q10"</span>, <span class="st">"XY_Q10"</span>, <span class="st">"LP_Q10"</span>, <span class="st">"PX_Q10"</span>, <span class="st">"CUEcn"</span>, <span class="st">"CUEcp"</span>, 
                <span class="st">"NUE"</span>, <span class="st">"PUE"</span>, <span class="st">"Tmin"</span>, <span class="st">"SI"</span><span class="op">)</span>

<span class="va">md_groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"edaphic"</span>, <span class="st">"soil_funct"</span>, <span class="st">"temp_adapt"</span><span class="op">)</span>

<span class="co"># NoT uSed: minPO4, minNH4, minNO3, minTIN</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>And then use the lists to split the data sets by metadata group. Here, we do this for the original metadata and the metadata after normalization.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb108"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">select_md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"its18_select_mc"</span>, <span class="st">"its18_select_mc_norm"</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">select_md</span><span class="op">)</span> <span class="op">{</span>
   <span class="co">#tmp_get &lt;- get(purrr::map_chr(i, ~ paste0(i, "_mc")))</span>
   <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">get</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span>
   <span class="va">tmp_md_all</span> <span class="op">&lt;-</span> <span class="va">tmp_get</span><span class="op">$</span><span class="va">map_loaded</span>
   <span class="va">tmp_div</span> <span class="op">&lt;-</span> <span class="va">tmp_md_all</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="fu">all_of</span><span class="op">(</span><span class="va">div</span><span class="op">)</span><span class="op">)</span>
   <span class="va">tmp_div</span> <span class="op">&lt;-</span> <span class="va">tmp_div</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>
<span class="co">## edaphic      </span>
   <span class="va">tmp_sub_edaphic</span> <span class="op">&lt;-</span> <span class="va">tmp_md_all</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="fu">all_of</span><span class="op">(</span><span class="va">edaphic</span><span class="op">)</span><span class="op">)</span>
   <span class="va">tmp_sub_edaphic</span> <span class="op">&lt;-</span> <span class="va">tmp_sub_edaphic</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>
   <span class="va">tmp_edaphic</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_div</span>, <span class="va">tmp_sub_edaphic</span>, by <span class="op">=</span> <span class="st">"SampleID"</span><span class="op">)</span>
   <span class="va">tmp_edaphic</span> <span class="op">&lt;-</span> <span class="va">tmp_edaphic</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>
<span class="co">## soil_funct</span>
   <span class="va">tmp_sub_soil_funct</span> <span class="op">&lt;-</span> <span class="va">tmp_md_all</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="fu">all_of</span><span class="op">(</span><span class="va">soil_funct</span><span class="op">)</span><span class="op">)</span>
   <span class="va">tmp_sub_soil_funct</span> <span class="op">&lt;-</span> <span class="va">tmp_sub_soil_funct</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>
   <span class="va">tmp_soil_funct</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_div</span>, <span class="va">tmp_sub_soil_funct</span>, by <span class="op">=</span> <span class="st">"SampleID"</span><span class="op">)</span>
   <span class="va">tmp_soil_funct</span> <span class="op">&lt;-</span> <span class="va">tmp_soil_funct</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>      
<span class="co">## temp_adapt</span>
   <span class="va">tmp_sub_temp_adapt</span> <span class="op">&lt;-</span> <span class="va">tmp_md_all</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="fu">all_of</span><span class="op">(</span><span class="va">temp_adapt</span><span class="op">)</span><span class="op">)</span>
   <span class="va">tmp_sub_temp_adapt</span> <span class="op">&lt;-</span> <span class="va">tmp_sub_temp_adapt</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>
   <span class="va">tmp_temp_adapt</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_div</span>, <span class="va">tmp_sub_temp_adapt</span>, by <span class="op">=</span> <span class="st">"SampleID"</span><span class="op">)</span>
   <span class="va">tmp_temp_adapt</span> <span class="op">&lt;-</span> <span class="va">tmp_temp_adapt</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>      
<span class="co">## combine</span>
   <span class="va">tmp_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>data_loaded <span class="op">=</span> <span class="va">its18_select_mc</span><span class="op">$</span><span class="va">data_loaded</span>, 
                    map_loaded <span class="op">=</span> <span class="va">its18_select_mc</span><span class="op">$</span><span class="va">map_loaded</span>, 
                    taxonomy_loaded <span class="op">=</span> <span class="va">its18_select_mc</span><span class="op">$</span><span class="va">taxonomy_loaded</span>,
                    edaphic <span class="op">=</span> <span class="va">tmp_edaphic</span>, 
                    soil_funct <span class="op">=</span> <span class="va">tmp_soil_funct</span>, 
                    temp_adapt <span class="op">=</span> <span class="va">tmp_temp_adapt</span><span class="op">)</span>
   <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_split"</span><span class="op">)</span><span class="op">)</span>
   <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_name</span><span class="op">)</span>
   <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_list</span><span class="op">)</span>
   <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="generate-autocorrelation-plots-1" class="level3"><h3 class="anchored" data-anchor-id="generate-autocorrelation-plots-1">Generate Autocorrelation Plots</h3>
<p>A little housekeeping to get rid of parameters we dont need (e.g., plot number, pair, etc.).</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb109"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">edaphic_cor</span> <span class="op">&lt;-</span> <span class="va">its18_select_mc_split</span><span class="op">$</span><span class="va">edaphic</span>
<span class="va">edaphic_cor</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">edaphic_norm_cor</span> <span class="op">&lt;-</span> <span class="va">its18_select_mc_norm_split</span><span class="op">$</span><span class="va">edaphic</span>
<span class="va">edaphic_norm_cor</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>

<span class="va">soil_funct_cor</span> <span class="op">&lt;-</span> <span class="va">its18_select_mc_split</span><span class="op">$</span><span class="va">soil_funct</span>
<span class="va">soil_funct_cor</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">soil_funct_norm_cor</span> <span class="op">&lt;-</span> <span class="va">its18_select_mc_norm_split</span><span class="op">$</span><span class="va">soil_funct</span>
<span class="va">soil_funct_norm_cor</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>

<span class="va">temp_adapt_cor</span> <span class="op">&lt;-</span> <span class="va">its18_select_mc_split</span><span class="op">$</span><span class="va">temp_adapt</span>
<span class="va">temp_adapt_cor</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">temp_adapt_norm_cor</span> <span class="op">&lt;-</span> <span class="va">its18_select_mc_norm_split</span><span class="op">$</span><span class="va">temp_adapt</span>
<span class="va">temp_adapt_norm_cor</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>And finally the code to create the plots.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb110"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">objects</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_cor$"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">get</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span>
  <span class="va">tmp_cormat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span>
  <span class="va">tmp_melted_cormat</span> <span class="op">&lt;-</span> <span class="fu">reshape2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html">melt</a></span><span class="op">(</span><span class="va">tmp_cormat</span><span class="op">)</span>
  
  <span class="va">tmp_get_lower_tri</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tmp_cormat</span><span class="op">)</span><span class="op">{</span>
    <span class="va">tmp_cormat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lower.tri.html">upper.tri</a></span><span class="op">(</span><span class="va">tmp_cormat</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">tmp_cormat</span><span class="op">)</span>
    <span class="op">}</span>
  <span class="co"># Get upper triangle of the correlation matrix</span>
  <span class="va">tmp_get_upper_tri</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tmp_cormat</span><span class="op">)</span><span class="op">{</span>
    <span class="va">tmp_cormat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lower.tri.html">lower.tri</a></span><span class="op">(</span><span class="va">tmp_cormat</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">tmp_cormat</span><span class="op">)</span>
    <span class="op">}</span>
  <span class="va">tmp_upper_tri</span> <span class="op">&lt;-</span> <span class="fu">tmp_get_upper_tri</span><span class="op">(</span><span class="va">tmp_cormat</span><span class="op">)</span>
  <span class="va">tmp_melted_cormat</span> <span class="op">&lt;-</span> <span class="fu">reshape2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html">melt</a></span><span class="op">(</span><span class="va">tmp_upper_tri</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
  <span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp_melted_cormat</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Var1</span>, y <span class="op">=</span> <span class="va">Var2</span>, fill <span class="op">=</span> <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu">geom_tile</span><span class="op">(</span><span class="op">)</span>
  
  <span class="va">tmp_ggheatmap</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp_melted_cormat</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">Var2</span>, <span class="va">Var1</span>, fill <span class="op">=</span> <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">scale_fill_gradient2</span><span class="op">(</span>low <span class="op">=</span> <span class="st">"blue"</span>, high <span class="op">=</span> <span class="st">"red"</span>, mid <span class="op">=</span> <span class="st">"white"</span>, 
                         midpoint <span class="op">=</span> <span class="fl">0</span>, limit <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>,<span class="fl">1</span><span class="op">)</span>, space <span class="op">=</span> <span class="st">"Lab"</span>, 
                         name<span class="op">=</span><span class="st">"Pearson\nCorrelation"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, vjust <span class="op">=</span> <span class="fl">1</span>, 
                                     size <span class="op">=</span> <span class="fl">7</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,
          axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>vjust <span class="op">=</span> <span class="fl">1</span>, size <span class="op">=</span> <span class="fl">7</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu">coord_fixed</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu">geom_text</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">Var2</span>, <span class="va">Var1</span>, label <span class="op">=</span> <span class="va">value</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">1.75</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">theme</span><span class="op">(</span>
      axis.title.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
      axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
      panel.grid.major <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
      panel.border <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
      panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
      axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
      legend.justification <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span>,
      legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.6</span>, <span class="fl">0.7</span><span class="op">)</span>,
      legend.direction <span class="op">=</span> <span class="st">"horizontal"</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu">guides</span><span class="op">(</span>fill <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>barwidth <span class="op">=</span> <span class="fl">7</span>, barheight <span class="op">=</span> <span class="fl">1</span>,
                                   title.position <span class="op">=</span> <span class="st">"top"</span>, 
                                   title.hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_ggheatmap"</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_ggheatmap</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_name</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>  
<span class="fu"><a href="https://rdrr.io/r/base/ls.html">objects</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_ggheatmap"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section></section><section id="autocorrelation-plots-1" class="level2 page-columns page-full"><h2 class="anchored" data-anchor-id="autocorrelation-plots-1">Autocorrelation Plots</h2>
<p><br></p>
<div class="column-body">
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-5-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-5-1" role="tab" aria-controls="tabset-5-1" aria-selected="true">Edaphic properties</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-5-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-5-2" role="tab" aria-controls="tabset-5-2" aria-selected="false">Edaphic properties (normalized)</a></li>
</ul>
<div class="tab-content">
<div id="tabset-5-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-5-1-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-162-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><small><strong>(ITS) Figure 1 | Results of autocorrelation analysis for non-normalized Environmental and edaphic parameters.</strong></small></p>
</div>
<div id="tabset-5-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-5-2-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-163-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><small><strong>(ITS) Figure 2 | Results of autocorrelation analysis for normalized Environmental and edaphic parameters.</strong></small></p>
</div>
</div>
</div>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-6-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-6-1" role="tab" aria-controls="tabset-6-1" aria-selected="true">Functional responses</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-6-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-6-2" role="tab" aria-controls="tabset-6-2" aria-selected="false">Functional responses (normalized)</a></li>
</ul>
<div class="tab-content">
<div id="tabset-6-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-6-1-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-164-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><small><strong>(ITS) Figure 3 | Results of autocorrelation analysis for non-normalized Functional response parameters.</strong></small></p>
</div>
<div id="tabset-6-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-6-2-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-165-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><small><strong>(ITS) Figure 4 | Results of autocorrelation analysis for normalized Functional response parameters.</strong></small></p>
</div>
</div>
</div>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-7-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-7-1" role="tab" aria-controls="tabset-7-1" aria-selected="true">Temperature adaptation</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-7-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-7-2" role="tab" aria-controls="tabset-7-2" aria-selected="false">Temperature adaptation (normalized)</a></li>
</ul>
<div class="tab-content">
<div id="tabset-7-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-7-1-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-166-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><small><strong>(ITS) Figure 5 | Results of autocorrelation analysis for non-normalized Temperature adaptation parameters.</strong></small></p>
</div>
<div id="tabset-7-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-7-2-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-167-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><small><strong>(ITS) Figure 6 | Results of autocorrelation analysis for normalized Temperature adaptation parameters.</strong></small></p>
</div>
</div>
</div>
</div>
<div class="cell">

</div>
<p>Now we can remove parameters based on the autocorrelation analysis:</p>
<ol type="A">
<li>Environmental and edaphic properties: TEB, DON, Na, Al, Ca.</li>
<li>Microbial functional responses: micN, micNP, enzCN, enzCP, BP_ase, CE_ase, LP_ase, N_ase, P_ase.</li>
<li>Temperature adaptation properties: NUE, PUE, P_Q10, SI.</li>
</ol>
<div class="cell">

</div>
<div class="cell">
<div class="sourceCode" id="cb111"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">edaphic_remove</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TEB"</span>, <span class="st">"DON"</span>, <span class="st">"Na"</span>, <span class="st">"Al"</span>, <span class="st">"Ca"</span><span class="op">)</span>
<span class="va">soil_funct_remove</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"micN"</span>, <span class="st">"micNP"</span>, <span class="st">"enzCN"</span>, <span class="st">"enzCP"</span>, <span class="st">"BP_ase"</span>, 
                       <span class="st">"CE_ase"</span>, <span class="st">"LP_ase"</span>, <span class="st">"N_ase"</span>, <span class="st">"P_ase"</span><span class="op">)</span>
<span class="va">temp_adapt_remove</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NUE"</span>, <span class="st">"PUE"</span>, <span class="st">"P_Q10"</span>, <span class="st">"SI"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb112"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_df</span> <span class="op">&lt;-</span> <span class="va">its18_select_mc_split</span>
<span class="va">tmp_df</span><span class="op">$</span><span class="va">edaphic</span> <span class="op">&lt;-</span> <span class="va">tmp_df</span><span class="op">$</span><span class="va">edaphic</span><span class="op">[</span>, <span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_df</span><span class="op">$</span><span class="va">edaphic</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span>  <span class="va">edaphic_remove</span><span class="op">]</span>
<span class="va">tmp_df</span><span class="op">$</span><span class="va">soil_funct</span> <span class="op">&lt;-</span> <span class="va">tmp_df</span><span class="op">$</span><span class="va">soil_funct</span><span class="op">[</span>, <span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_df</span><span class="op">$</span><span class="va">soil_funct</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span>  <span class="va">soil_funct_remove</span><span class="op">]</span>
<span class="va">tmp_df</span><span class="op">$</span><span class="va">temp_adapt</span> <span class="op">&lt;-</span> <span class="va">tmp_df</span><span class="op">$</span><span class="va">temp_adapt</span><span class="op">[</span>, <span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_df</span><span class="op">$</span><span class="va">temp_adapt</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span>  <span class="va">temp_adapt_remove</span><span class="op">]</span>
<span class="va">its18_select_mc_split_no_ac</span> <span class="op">&lt;-</span> <span class="va">tmp_df</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>

<span class="va">tmp_df</span> <span class="op">&lt;-</span> <span class="va">its18_select_mc_norm_split</span>
<span class="va">tmp_df</span><span class="op">$</span><span class="va">edaphic</span> <span class="op">&lt;-</span> <span class="va">tmp_df</span><span class="op">$</span><span class="va">edaphic</span><span class="op">[</span>, <span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_df</span><span class="op">$</span><span class="va">edaphic</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span>  <span class="va">edaphic_remove</span><span class="op">]</span>
<span class="va">tmp_df</span><span class="op">$</span><span class="va">soil_funct</span> <span class="op">&lt;-</span> <span class="va">tmp_df</span><span class="op">$</span><span class="va">soil_funct</span><span class="op">[</span>, <span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_df</span><span class="op">$</span><span class="va">soil_funct</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span>  <span class="va">soil_funct_remove</span><span class="op">]</span>
<span class="va">tmp_df</span><span class="op">$</span><span class="va">temp_adapt</span> <span class="op">&lt;-</span> <span class="va">tmp_df</span><span class="op">$</span><span class="va">temp_adapt</span><span class="op">[</span>, <span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_df</span><span class="op">$</span><span class="va">temp_adapt</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span>  <span class="va">temp_adapt_remove</span><span class="op">]</span>
<span class="va">its18_select_mc_norm_split_no_ac</span> <span class="op">&lt;-</span> <span class="va">tmp_df</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="dissimilarity-correlation-tests-1" class="level2"><h2 class="anchored" data-anchor-id="dissimilarity-correlation-tests-1">Dissimilarity Correlation Tests</h2>
<p>Lets see if any on the metadata groups are significantly correlated with the community data. Basically, we create distance matrices for the community data and each metadata group and then run Mantel tests for all comparisons. For the community data we calculate Bray-Curtis distances for the community data and Euclidean distances for the metadata. We use the function <code>mantel.test</code> from the <code>ape</code> package and <code>mantel</code> from the <code>vegan</code> package for the analyses.</p>
<p>In summary, we test both <code>mantel.test</code> and <code>mantel</code> on Bray-Curtis distance community distances against Euclidean distances for each metadata group (<code>edaphic</code>, <code>soil_funct</code>, <code>temp_adapt</code>) <strong>a</strong>) before normalizing and before removing autocorrelated parameters, <strong>b</strong>) before normalizing and after removing autocorrelated parameters, <strong>c</strong>) after normalizing and before removing autocorrelated parameters, and <strong>d</strong>) after normalizing and after removing autocorrelated parameters.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb113"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">man_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"its18_select_mc_split"</span>, <span class="st">"its18_select_mc_split_no_ac"</span>, 
            <span class="st">"its18_select_mc_norm_split"</span>, <span class="st">"its18_select_mc_norm_split_no_ac"</span><span class="op">)</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">man_df</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">get</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span>
  <span class="va">tmp_dm_otu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">vegdist</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">$</span><span class="va">data_loaded</span><span class="op">)</span>, 
                             method <span class="op">=</span> <span class="st">"bray"</span>, binary <span class="op">=</span> <span class="cn">FALSE</span>, 
                             diag <span class="op">=</span> <span class="cn">TRUE</span>, upper <span class="op">=</span> <span class="cn">TRUE</span>, na.rm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>
<span class="co"># EDAPHIC</span>
  <span class="va">tmp_dm_md_edaphic</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">vegdist</span><span class="op">(</span><span class="va">tmp_get</span><span class="op">$</span><span class="va">edaphic</span><span class="op">[</span>, <span class="fl">8</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">$</span><span class="va">edaphic</span><span class="op">)</span><span class="op">]</span>, 
                                    method <span class="op">=</span><span class="st">"euclidean"</span>, binary <span class="op">=</span> <span class="cn">FALSE</span>, 
                                    diag <span class="op">=</span> <span class="cn">TRUE</span>, upper <span class="op">=</span> <span class="cn">TRUE</span>, na.rm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>

  <span class="va">tmp_man1_edaphic</span> <span class="op">&lt;-</span> <span class="fu">mantel.test</span><span class="op">(</span><span class="va">tmp_dm_otu</span>, <span class="va">tmp_dm_md_edaphic</span>, nperm <span class="op">=</span> <span class="fl">999</span>, 
                             graph <span class="op">=</span> <span class="cn">FALSE</span>, alternative <span class="op">=</span> <span class="st">"two.sided"</span><span class="op">)</span>
  <span class="va">tmp_man2_edaphic</span> <span class="op">&lt;-</span> <span class="fu">mantel</span><span class="op">(</span><span class="va">tmp_dm_otu</span>, <span class="va">tmp_dm_md_edaphic</span>, permutations <span class="op">=</span> <span class="fl">999</span><span class="op">)</span>
<span class="co"># SOIL FUNCT</span>
  <span class="va">tmp_dm_md_soil_funct</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">vegdist</span><span class="op">(</span><span class="va">tmp_get</span><span class="op">$</span><span class="va">soil_funct</span><span class="op">[</span>, <span class="fl">8</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">$</span><span class="va">soil_funct</span><span class="op">)</span><span class="op">]</span>, 
                                       method <span class="op">=</span> <span class="st">"euclidean"</span>, binary <span class="op">=</span> <span class="cn">FALSE</span>, 
                                       diag <span class="op">=</span> <span class="cn">TRUE</span>, upper <span class="op">=</span> <span class="cn">TRUE</span>, na.rm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>

  <span class="va">tmp_man1_soil_funct</span> <span class="op">&lt;-</span> <span class="fu">mantel.test</span><span class="op">(</span><span class="va">tmp_dm_otu</span>, <span class="va">tmp_dm_md_soil_funct</span>, nperm <span class="op">=</span> <span class="fl">999</span>, 
                                graph <span class="op">=</span> <span class="cn">FALSE</span>, alternative <span class="op">=</span> <span class="st">"two.sided"</span><span class="op">)</span>
  <span class="va">tmp_man2_soil_funct</span> <span class="op">&lt;-</span> <span class="fu">mantel</span><span class="op">(</span><span class="va">tmp_dm_otu</span>, <span class="va">tmp_dm_md_soil_funct</span>, permutations <span class="op">=</span> <span class="fl">999</span><span class="op">)</span>

<span class="co"># TEMP ADAPT</span>
  <span class="va">tmp_dm_md_temp_adapt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">vegdist</span><span class="op">(</span><span class="va">tmp_get</span><span class="op">$</span><span class="va">temp_adapt</span><span class="op">[</span>, <span class="fl">8</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">$</span><span class="va">temp_adapt</span><span class="op">)</span><span class="op">]</span>, 
                                       method <span class="op">=</span> <span class="st">"euclidean"</span>, binary <span class="op">=</span> <span class="cn">FALSE</span>, 
                                       diag <span class="op">=</span> <span class="cn">TRUE</span>, upper <span class="op">=</span> <span class="cn">TRUE</span>, na.rm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>

  <span class="va">tmp_man1_temp_adapt</span> <span class="op">&lt;-</span> <span class="fu">mantel.test</span><span class="op">(</span><span class="va">tmp_dm_otu</span>, <span class="va">tmp_dm_md_temp_adapt</span>, nperm <span class="op">=</span> <span class="fl">999</span>, 
                                graph <span class="op">=</span> <span class="cn">FALSE</span>, alternative <span class="op">=</span> <span class="st">"two.sided"</span><span class="op">)</span>
  <span class="va">tmp_man2_temp_adapt</span> <span class="op">&lt;-</span> <span class="fu">mantel</span><span class="op">(</span><span class="va">tmp_dm_otu</span>, <span class="va">tmp_dm_md_temp_adapt</span>, permutations <span class="op">=</span> <span class="fl">999</span><span class="op">)</span>

  <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_mantel_tests"</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>edaphic_ape_man <span class="op">=</span> <span class="va">tmp_man1_edaphic</span>, 
                 edaphic_vegan_man <span class="op">=</span> <span class="va">tmp_man2_edaphic</span>,
                 soil_funct_ape_man <span class="op">=</span> <span class="va">tmp_man1_soil_funct</span>, 
                 soil_funct_vegan_man <span class="op">=</span> <span class="va">tmp_man2_soil_funct</span>,
                 temp_adapt_ape_man <span class="op">=</span> <span class="va">tmp_man1_temp_adapt</span>, 
                 temp_adapt_vegan_man <span class="op">=</span> <span class="va">tmp_man2_temp_adapt</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_df</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_name</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">

</div>
<section id="dissimilarity-correlation-results-1" class="level3"><h3 class="anchored" data-anchor-id="dissimilarity-correlation-results-1">Dissimilarity Correlation Results</h3>
<p><small><strong>(ITS) Table 4 | Summary of Dissimilarity Correlation Tests using <code>mantel.test</code> from the ape package and <code>mantel</code> from the vegan. P-values in red indicate significance (p-value &lt; 0.05)</strong></small></p>
<div class="cell">
<div class="cell-output-display">

<div id="htmlwidget-ae88d6572272c23fc065" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ae88d6572272c23fc065">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"normalized?":["no","no","yes","yes"],"AC removed?":["no","yes","no","yes"],"edaphic_ape":[0.119,0.081,0.151,0.003],"edaphic_vegan":[0.062,0.037,0.081,0.003],"soil_funct_ape":[0.097,0.095,0.345,0.357],"soil_funct_vegan":[0.047,0.037,0.163,0.18],"temp_adapt_ape":[0.001,0.001,0.001,0.001],"temp_adapt_vegan":[0.002,0.001,0.001,0.001]},"columns":[{"accessor":"normalized?","name":"normalized?","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["no","no","yes","yes"],"header":"normalized?","minWidth":70,"align":"left","style":{"borderRight":"5px solid #eee"},"headerStyle":{"borderRight":"5px solid #eee"}},{"accessor":"AC removed?","name":"AC removed?","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["no","yes","no","yes"],"header":"AC removed?","minWidth":70,"align":"left","style":{"borderRight":"5px solid #eee"},"headerStyle":{"borderRight":"5px solid #eee"}},{"accessor":"edaphic_ape","name":"ape","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.119","0.081","0.151","0.003"],"header":"ape","minWidth":50,"align":"center","style":[{"color":"#2271B2"},{"color":"#2271B2"},{"color":"#2271B2"},{"color":"#B22271"}]},{"accessor":"edaphic_vegan","name":"vegan","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.062","0.037","0.081","0.003"],"header":"vegan","minWidth":50,"align":"center","style":[{"color":"#2271B2"},{"color":"#B22271"},{"color":"#2271B2"},{"color":"#B22271"}]},{"accessor":"soil_funct_ape","name":"ape","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.097","0.095","0.345","0.357"],"header":"ape","minWidth":50,"align":"center","style":[{"color":"#2271B2"},{"color":"#2271B2"},{"color":"#2271B2"},{"color":"#2271B2"}]},{"accessor":"soil_funct_vegan","name":"vegan","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.047","0.037","0.163","0.180"],"header":"vegan","minWidth":50,"align":"center","style":[{"color":"#B22271"},{"color":"#B22271"},{"color":"#2271B2"},{"color":"#2271B2"}]},{"accessor":"temp_adapt_ape","name":"ape","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.001","0.001","0.001","0.001"],"header":"ape","minWidth":50,"align":"center","style":[{"color":"#B22271"},{"color":"#B22271"},{"color":"#B22271"},{"color":"#B22271"}]},{"accessor":"temp_adapt_vegan","name":"vegan","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.002","0.001","0.001","0.001"],"header":"vegan","minWidth":50,"align":"center","style":[{"color":"#B22271"},{"color":"#B22271"},{"color":"#B22271"},{"color":"#B22271"}]}],"columnGroups":[{"name":"Environmental & edaphic","headerStyle":{"fontSize":"1.2em"},"columns":["edaphic_ape","edaphic_vegan"]},{"name":"Functional responses","headerStyle":{"fontSize":"1.2em"},"columns":["soil_funct_ape","soil_funct_vegan"]},{"name":"Temperature adaptation","headerStyle":{"fontSize":"1.2em"},"columns":["temp_adapt_ape","temp_adapt_vegan"]}],"defaultPageSize":15,"showPageSizeOptions":true,"pageSizeOptions":[5,10,4],"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"nowrap":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"210b202e7958afc66573d023a0503fe5"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>Moving on.</p>
</section></section><section id="best-subset-of-variables-1" class="level2"><h2 class="anchored" data-anchor-id="best-subset-of-variables-1">Best Subset of Variables</h2>
<p>Now we want to know which of the metadata parameters are the most strongly correlated with the community data. For this we use the <code>bioenv</code> function from the <code>vegan</code> package. <code>bioenv</code><em>Best Subset of Environmental Variables with Maximum (Rank) Correlation with Community Dissimilarities</em>finds the best subset of environmental variables, so that the Euclidean distances of scaled environmental variables have the maximum (rank) correlation with community dissimilarities.</p>
<p>Since we know that each of the Mantel tests we ran above are significant, here we will use the metadata set where autocorrelated parameters were removed and the remainder of the parameters were normalized (where applicable based on the Shapiro tests).</p>
<p>We run <code>bioenv</code> against the three groups of metadata parameters. We then run <code>bioenv</code> again, but this time against the individual parameters identified as significantly correlated.</p>
<section id="edaphic-properties-5" class="level3"><h3 class="anchored" data-anchor-id="edaphic-properties-5">Edaphic Properties</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb114"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_comm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="va">its18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">data_loaded</span><span class="op">)</span><span class="op">)</span>
<span class="va">tmp_env</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">its18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">edaphic</span><span class="op">)</span>
<span class="va">tmp_env</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>

<span class="va">edaphic_bioenv</span> <span class="op">&lt;-</span> <span class="fu">bioenv</span><span class="op">(</span><span class="fu">wisconsin</span><span class="op">(</span><span class="va">tmp_comm</span><span class="op">)</span>, <span class="va">tmp_env</span>, 
                         method <span class="op">=</span> <span class="st">"spearman"</span>, index <span class="op">=</span> <span class="st">"bray"</span>, 
                         upto <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_env</span><span class="op">)</span>, metric <span class="op">=</span> <span class="st">"euclidean"</span><span class="op">)</span>

<span class="va">bioenv_list</span> <span class="op">&lt;-</span> <span class="va">edaphic_bioenv</span><span class="op">$</span><span class="va">models</span><span class="op">[[</span><span class="va">edaphic_bioenv</span><span class="op">$</span><span class="va">whichbest</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">best</span>
<span class="va">bioenv_best</span> <span class="op">&lt;-</span> <span class="fu">bioenvdist</span><span class="op">(</span><span class="va">edaphic_bioenv</span>, which <span class="op">=</span> <span class="st">"best"</span><span class="op">)</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">bioenv_list</span><span class="op">)</span> <span class="op">{</span>
       <span class="va">tmp_dp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">edaphic_bioenv</span><span class="op">$</span><span class="va">x</span><span class="op">)</span>
       <span class="va">tmp_md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">vegdist</span><span class="op">(</span><span class="va">tmp_dp</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, 
                                   method <span class="op">=</span> <span class="st">"euclidean"</span>, binary <span class="op">=</span> <span class="cn">FALSE</span>, 
                                   diag <span class="op">=</span> <span class="cn">TRUE</span>, upper <span class="op">=</span> <span class="cn">TRUE</span>, na.rm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>
       <span class="va">tmp_man</span> <span class="op">&lt;-</span> <span class="fu">mantel</span><span class="op">(</span><span class="va">bioenv_best</span>, <span class="va">tmp_md</span>, 
                         permutations <span class="op">=</span> <span class="fl">999</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span>
       <span class="va">tmp_md_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_dp</span><span class="op">)</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>
       <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">tmp_md_name</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_bioenv_mantel_test"</span><span class="op">)</span><span class="op">)</span>
       <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_man</span><span class="op">)</span>
       <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/ls.html">objects</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_bioenv_mantel_test"</span><span class="op">)</span>

<span class="va">edaphic_bioenv_ind_mantel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>AST <span class="op">=</span> <span class="va">AST_bioenv_mantel_test</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_bioenv_mantel_test"</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
bioenv(comm = wisconsin(tmp_comm), env = tmp_env, method = "spearman",      index = "bray", upto = ncol(tmp_env), metric = "euclidean") 

Subset of environmental variables with best correlation to community data.

Correlations:    spearman 
Dissimilarities: bray 
Metric:          euclidean 

Best model has 1 parameters (max. 15 allowed):
AST
with correlation  0.65665 </code></pre>
</div>
</div>
<details><summary>
Show the results of individual edaphic metadata Mantel tests
</summary><div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>$AST

Mantel statistic based on Spearman's rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = "spearman",      permutations = 999) 

Mantel statistic r:     1 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.192 0.242 0.272 0.339 
Permutation: free
Number of permutations: 999</code></pre>
</div>
</div>
</details><p><code>bioenv</code> found the following edaphic properties significantly correlated with the community data: <strong>AST</strong></p>
</section><section id="soil-functional-response-3" class="level3"><h3 class="anchored" data-anchor-id="soil-functional-response-3">Soil Functional Response</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb117"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_comm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="va">its18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">data_loaded</span><span class="op">)</span><span class="op">)</span>
<span class="va">tmp_env</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">its18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">soil_funct</span><span class="op">)</span>
<span class="va">tmp_env</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>

<span class="va">soil_funct_bioenv</span> <span class="op">&lt;-</span> <span class="fu">bioenv</span><span class="op">(</span><span class="fu">wisconsin</span><span class="op">(</span><span class="va">tmp_comm</span><span class="op">)</span>, <span class="va">tmp_env</span>, 
                         method <span class="op">=</span> <span class="st">"spearman"</span>, index <span class="op">=</span> <span class="st">"bray"</span>, 
                         upto <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_env</span><span class="op">)</span>, metric <span class="op">=</span> <span class="st">"euclidean"</span><span class="op">)</span>

<span class="va">bioenv_list</span> <span class="op">&lt;-</span> <span class="va">soil_funct_bioenv</span><span class="op">$</span><span class="va">models</span><span class="op">[[</span><span class="va">soil_funct_bioenv</span><span class="op">$</span><span class="va">whichbest</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">best</span>
<span class="va">bioenv_best</span> <span class="op">&lt;-</span> <span class="fu">bioenvdist</span><span class="op">(</span><span class="va">soil_funct_bioenv</span>, which <span class="op">=</span> <span class="st">"best"</span><span class="op">)</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">bioenv_list</span><span class="op">)</span> <span class="op">{</span>
       <span class="va">tmp_dp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">soil_funct_bioenv</span><span class="op">$</span><span class="va">x</span><span class="op">)</span>
       <span class="va">tmp_md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">vegdist</span><span class="op">(</span><span class="va">tmp_dp</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, 
                                   method <span class="op">=</span> <span class="st">"euclidean"</span>, binary <span class="op">=</span> <span class="cn">FALSE</span>, 
                                   diag <span class="op">=</span> <span class="cn">TRUE</span>, upper <span class="op">=</span> <span class="cn">TRUE</span>, na.rm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>
       <span class="va">tmp_man</span> <span class="op">&lt;-</span> <span class="fu">mantel</span><span class="op">(</span><span class="va">bioenv_best</span>, <span class="va">tmp_md</span>, 
                         permutations <span class="op">=</span> <span class="fl">999</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span>
       <span class="va">tmp_md_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_dp</span><span class="op">)</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>
       <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">tmp_md_name</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_bioenv_mantel_test"</span><span class="op">)</span><span class="op">)</span>
       <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_man</span><span class="op">)</span>
       <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/ls.html">objects</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_bioenv_mantel_test"</span><span class="op">)</span>

<span class="va">soil_funct_bioenv_ind_mantel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>enzNP <span class="op">=</span> <span class="va">enzNP_bioenv_mantel_test</span>, 
                                     PX_ase <span class="op">=</span> <span class="va">PX_ase_bioenv_mantel_test</span>,
                                     XY_ase <span class="op">=</span> <span class="va">XY_ase_bioenv_mantel_test</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_bioenv_mantel_test"</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
bioenv(comm = wisconsin(tmp_comm), env = tmp_env, method = "spearman",      index = "bray", upto = ncol(tmp_env), metric = "euclidean") 

Subset of environmental variables with best correlation to community data.

Correlations:    spearman 
Dissimilarities: bray 
Metric:          euclidean 

Best model has 3 parameters (max. 11 allowed):
XY_ase PX_ase enzNP
with correlation  0.7491622 </code></pre>
</div>
</div>
<details><summary>
Show the results of individual functional response metadata Mantel tests
</summary><div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>$enzNP

Mantel statistic based on Spearman's rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = "spearman",      permutations = 999) 

Mantel statistic r: 0.5529 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.202 0.277 0.330 0.417 
Permutation: free
Number of permutations: 999


$PX_ase

Mantel statistic based on Spearman's rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = "spearman",      permutations = 999) 

Mantel statistic r: 0.6854 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.209 0.282 0.361 0.412 
Permutation: free
Number of permutations: 999


$XY_ase

Mantel statistic based on Spearman's rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = "spearman",      permutations = 999) 

Mantel statistic r: 0.5054 
      Significance: 0.002 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.202 0.261 0.308 0.359 
Permutation: free
Number of permutations: 999</code></pre>
</div>
</div>
</details><p><code>bioenv</code> found the following soil functions significantly correlated with the community data: <strong>enzNP, PX_ase, XY_ase</strong></p>
</section><section id="temperature-adaptation-5" class="level3"><h3 class="anchored" data-anchor-id="temperature-adaptation-5">Temperature Adaptation</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb120"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_comm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="va">its18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">data_loaded</span><span class="op">)</span><span class="op">)</span>
<span class="va">tmp_env</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">its18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">temp_adapt</span><span class="op">)</span>
<span class="va">tmp_env</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>

<span class="va">temp_adapt_bioenv</span> <span class="op">&lt;-</span> <span class="fu">bioenv</span><span class="op">(</span><span class="fu">wisconsin</span><span class="op">(</span><span class="va">tmp_comm</span><span class="op">)</span>, <span class="va">tmp_env</span>, 
                         method <span class="op">=</span> <span class="st">"spearman"</span>, index <span class="op">=</span> <span class="st">"bray"</span>, 
                         upto <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_env</span><span class="op">)</span>, metric <span class="op">=</span> <span class="st">"euclidean"</span><span class="op">)</span>

<span class="va">bioenv_list</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_bioenv</span><span class="op">$</span><span class="va">models</span><span class="op">[[</span><span class="va">temp_adapt_bioenv</span><span class="op">$</span><span class="va">whichbest</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">best</span>
<span class="va">bioenv_best</span> <span class="op">&lt;-</span> <span class="fu">bioenvdist</span><span class="op">(</span><span class="va">temp_adapt_bioenv</span>, which <span class="op">=</span> <span class="st">"best"</span><span class="op">)</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">bioenv_list</span><span class="op">)</span> <span class="op">{</span>
       <span class="va">tmp_dp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">temp_adapt_bioenv</span><span class="op">$</span><span class="va">x</span><span class="op">)</span>
       <span class="va">tmp_md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">vegdist</span><span class="op">(</span><span class="va">tmp_dp</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, 
                                   method <span class="op">=</span> <span class="st">"euclidean"</span>, binary <span class="op">=</span> <span class="cn">FALSE</span>, 
                                   diag <span class="op">=</span> <span class="cn">TRUE</span>, upper <span class="op">=</span> <span class="cn">TRUE</span>, na.rm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>
       <span class="va">tmp_man</span> <span class="op">&lt;-</span> <span class="fu">mantel</span><span class="op">(</span><span class="va">bioenv_best</span>, <span class="va">tmp_md</span>, 
                         permutations <span class="op">=</span> <span class="fl">999</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span>
       <span class="va">tmp_md_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_dp</span><span class="op">)</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>
       <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">tmp_md_name</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_bioenv_mantel_test"</span><span class="op">)</span><span class="op">)</span>
       <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_man</span><span class="op">)</span>
       <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/ls.html">objects</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_bioenv_mantel_test"</span><span class="op">)</span>

<span class="va">temp_adapt_bioenv_ind_mantel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>XY_Q10 <span class="op">=</span> <span class="va">XY_Q10_bioenv_mantel_test</span>,
                                     Tmin <span class="op">=</span> <span class="va">Tmin_bioenv_mantel_test</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_bioenv_mantel_test"</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
bioenv(comm = wisconsin(tmp_comm), env = tmp_env, method = "spearman",      index = "bray", upto = ncol(tmp_env), metric = "euclidean") 

Subset of environmental variables with best correlation to community data.

Correlations:    spearman 
Dissimilarities: bray 
Metric:          euclidean 

Best model has 2 parameters (max. 12 allowed):
XY_Q10 Tmin
with correlation  0.5637527 </code></pre>
</div>
</div>
<details><summary>
Show the results of individual temperature adaptation metadata Mantel tests
</summary><div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>$XY_Q10

Mantel statistic based on Spearman's rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = "spearman",      permutations = 999) 

Mantel statistic r: 0.7258 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.134 0.200 0.241 0.316 
Permutation: free
Number of permutations: 999


$Tmin

Mantel statistic based on Spearman's rank correlation rho 

Call:
mantel(xdis = bioenv_best, ydis = tmp_md, method = "spearman",      permutations = 999) 

Mantel statistic r: 0.616 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.189 0.245 0.297 0.360 
Permutation: free
Number of permutations: 999</code></pre>
</div>
</div>
</details><p><code>bioenv</code> found the following temperature adaptations significantly correlated with the community data: <strong>XY_Q10, Tmin</strong></p>
</section></section><section id="distance-based-redundancy-1" class="level2"><h2 class="anchored" data-anchor-id="distance-based-redundancy-1">Distance-based Redundancy</h2>
<p>Now we turn our attention to distance-based redundancy analysis (dbRDA), an ordination method similar to Redundancy Analysis (rda) but it allows non-Euclidean dissimilarity indices, such as Manhattan or BrayCurtis distance. For this, we use <code>capscale</code> from the <code>vegan</code> package. <code>capscale</code> is a constrained versions of metric scaling (principal coordinates analysis), which are based on the Euclidean distance but can be used, and are more useful, with other dissimilarity measures. The functions can also perform unconstrained principal coordinates analysis, optionally using extended dissimilarities.</p>
<p>For each of the three metadata subsets, we perform the following steps:</p>
<ol type="1">
<li>Run <code>rankindex</code> to compare metadata and community dissimilarity indices for gradient detection. This will help us select the best dissimilarity metric to use.</li>
<li>Run <code>capscale</code> for distance-based redundancy analysis.</li>
<li>Run <code>envfit</code> to fit environmental parameters onto the ordination. This function basically calculates correlation scores between the metadata parameters and the ordination axes.</li>
<li>Select metadata parameters significant for <code>bioenv</code> (see above) and/or <code>envfit</code> analyses.</li>
<li>Run <code>envfit</code> on ASVs.</li>
<li>Plot the ordination and vector overlays.</li>
</ol>
<section id="edaphic-properties-6" class="level3"><h3 class="anchored" data-anchor-id="edaphic-properties-6">Edaphic Properties</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb123"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_md</span> <span class="op">&lt;-</span> <span class="va">its18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">edaphic</span>
<span class="va">tmp_md</span><span class="op">$</span><span class="va">TREAT_T</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">tmp_md</span><span class="op">$</span><span class="va">TREAT_T</span><span class="op">)</span>
<span class="va">tmp_comm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="va">its18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">data_loaded</span><span class="op">)</span><span class="op">)</span>
<span class="va">edaphic_rank</span> <span class="op">&lt;-</span> <span class="fu">rankindex</span><span class="op">(</span><span class="va">tmp_md</span><span class="op">[</span>, <span class="fl">8</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_md</span><span class="op">)</span><span class="op">]</span>, <span class="va">tmp_comm</span>, 
          indices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"euc"</span>, <span class="st">"man"</span>, <span class="st">"gow"</span>,<span class="st">"bra"</span>, <span class="st">"kul"</span><span class="op">)</span>, 
          stepacross <span class="op">=</span> <span class="cn">FALSE</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>        euc         man         gow         bra         kul 
 0.15398526  0.29983940 -0.02940098  0.29983940  0.29983940 </code></pre>
</div>
</div>
<p>Lets run <code>capscale</code> using Bray-Curtis. Note, we have 15 metadata parameters in this group but, for some reason, <code>capscale</code> only works with 13 parameters. This may have to do with degrees of freedom?</p>
<ul>
<li>Starting properties: AST, H2O, N, P, Al, Ca, Fe, K, Mg, Mn, Na, TEB, ECEC, pH, NH4, NO3, PO4, DOC, DON, DOCN.</li>
<li>Autocorrelated removed: TEB, DON, Na, Al, Ca.</li>
<li>Remove for capscale: Mg, Mn, Na, Al, Fe, K</li>
</ul>
<div class="cell">

</div>
<div class="cell">
<div class="sourceCode" id="cb125"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">edaphic_cap</span> <span class="op">&lt;-</span> <span class="fu">capscale</span><span class="op">(</span><span class="va">tmp_comm</span> <span class="op">~</span> <span class="va">AST</span> <span class="op">+</span> <span class="va">H2O</span> <span class="op">+</span> <span class="va">N</span> <span class="op">+</span> <span class="va">P</span> <span class="op">+</span> <span class="va">ECEC</span> <span class="op">+</span> <span class="va">pH</span> <span class="op">+</span> 
                                   <span class="va">NH4</span> <span class="op">+</span> <span class="va">NO3</span> <span class="op">+</span> <span class="va">PO4</span> <span class="op">+</span> <span class="va">DOC</span> <span class="op">+</span> <span class="va">DOCN</span>, 
                        <span class="va">tmp_md</span>, dist <span class="op">=</span> <span class="st">"bray"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">tmp_md</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Call: capscale(formula = tmp_comm ~ AST + H2O + N + P + ECEC + pH + NH4
+ NO3 + PO4 + DOC + DOCN, data = tmp_md, distance = "bray")

              Inertia Proportion Rank
Total         3.88624    1.00000     
Constrained   3.66503    0.94308   11
Unconstrained 0.22120    0.05692    1
Inertia is squared Bray distance 
Species scores projected from 'tmp_comm' 

Eigenvalues for constrained axes:
  CAP1   CAP2   CAP3   CAP4   CAP5   CAP6   CAP7   CAP8   CAP9  CAP10  CAP11 
0.9655 0.5791 0.4327 0.3671 0.2987 0.2608 0.2301 0.1867 0.1339 0.1135 0.0970 

Eigenvalues for unconstrained axes:
  MDS1 
0.2212 </code></pre>
</div>
</div>
<p>Now we can look at the variance against each principal component.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-189-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>And then make some quick and dirty plots. This will also come in handy later when we need to parse out data a better plot visualization. The <code>ggplot</code> function <code>autoplot</code> stores these data in a more accessible way than the raw results from <code>capscale</code></p>
<div class="cell">
<div class="sourceCode" id="cb127"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span><span class="op">(</span><span class="va">edaphic_cap</span><span class="op">)</span> </code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-190-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode" id="cb128"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_auto_plt</span> <span class="op">&lt;-</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html">autoplot</a></span><span class="op">(</span><span class="va">edaphic_cap</span>, arrows <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">tmp_auto_plt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-190-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">

</div>
<p>Next, we need to grab capscale scores for the samples and create a data frame of the first two dimensions. We will also need to add some of the sample details to the data frame. For this we use the vegan function <code>scores</code> which gets species or site scores from the ordination.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb129"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_samp_scores</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tmp_auto_plt</span><span class="op">$</span><span class="va">plot_env</span><span class="op">$</span><span class="va">obj</span>, <span class="va">Score</span> <span class="op">==</span> <span class="st">"sites"</span><span class="op">)</span>
<span class="va">tmp_samp_scores</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">tmp_samp_scores</span> <span class="op">&lt;-</span> <span class="va">tmp_samp_scores</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>SampleID <span class="op">=</span> <span class="va">Label</span><span class="op">)</span>

<span class="va">tmp_md_sub</span> <span class="op">&lt;-</span> <span class="va">tmp_md</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>
<span class="va">tmp_md_sub</span> <span class="op">&lt;-</span> <span class="va">tmp_md_sub</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>
<span class="va">edaphic_plot_data</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_md_sub</span>, <span class="va">tmp_samp_scores</span>, by <span class="op">=</span> <span class="st">"SampleID"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Now we have a new data frame that contains sample details and capscale values.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>          SampleID PLOT TREAT TREAT_T PAIR        CAP1        CAP2
1  P10_D00_010_C0E  P10     C       0    E -0.67150210 -1.07183211
2  P02_D00_010_C0A  P02     C       0    A -0.91911652 -0.41647565
3  P04_D00_010_C0B  P04     C       0    B -0.95654588  0.64087450
4  P06_D00_010_C0C  P06     C       0    C -0.87992057  0.68082267
5  P08_D00_010_C0D  P08     C       0    D -0.07766291 -0.98166349
6  P01_D00_010_W3A  P01     W       3    A -0.79165161 -0.43340347
7  P03_D00_010_W3B  P03     W       3    B  0.24722812  0.81805851
8  P07_D00_010_W3D  P07     W       3    D  0.20151965  1.07639049
9  P09_D00_010_W3E  P09     W       3    E  0.58233467  0.46562279
10 P01_D00_010_W8A  P01     W       8    A  0.92158009 -0.06825914
11 P03_D00_010_W8B  P03     W       8    B  0.94079184 -0.52231363
12 P05_D00_010_W8C  P05     W       8    C  0.87642984 -0.89155825
13 P07_D00_010_W8D  P07     W       8    D  0.52651539  0.70373678</code></pre>
</div>
</div>
<p>We can then do the same with the metadata vectors. Here though we only need the scores and parameter name.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb131"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">edaphic_md_scores</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tmp_auto_plt</span><span class="op">$</span><span class="va">plot_env</span><span class="op">$</span><span class="va">obj</span>, <span class="va">Score</span> <span class="op">==</span> <span class="st">"biplot"</span><span class="op">)</span>
<span class="va">edaphic_md_scores</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">edaphic_md_scores</span> <span class="op">&lt;-</span> <span class="va">edaphic_md_scores</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>parameters <span class="op">=</span> <span class="va">Label</span>, .before <span class="op">=</span> <span class="va">CAP1</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"Label"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>     parameters         CAP1         CAP2
AST         AST  0.834527196  0.117196419
H2O         H2O -0.596509128  0.006131247
N             N  0.245428273  0.506082992
P             P  0.097845581  0.205793005
ECEC       ECEC -0.304860202  0.538475609
pH           pH -0.081297235 -0.092037201
NH4         NH4 -0.297079795  0.210528906
NO3         NO3 -0.264254841  0.225782469
PO4         PO4 -0.583698114  0.354036263
DOC         DOC -0.884261492  0.025360275
DOCN       DOCN  0.007258364 -0.462404778</code></pre>
</div>
</div>
<p>Lets run some quick correlations of metadata with ordination axes to see which parameters are significant. For this we use the vegan function <code>envfit</code>.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb133"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_samp_scores_sub</span> <span class="op">&lt;-</span> <span class="va">edaphic_plot_data</span><span class="op">[</span>, <span class="fl">6</span><span class="op">:</span><span class="fl">7</span><span class="op">]</span>
<span class="va">tmp_samp_scores_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">tmp_samp_scores_sub</span><span class="op">)</span>
<span class="va">tmp_param_list</span> <span class="op">&lt;-</span> <span class="va">edaphic_md_scores</span><span class="op">$</span><span class="va">parameters</span>

<span class="va">tmp_md_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">tmp_md</span>,  select <span class="op">=</span>  <span class="va">tmp_param_list</span><span class="op">)</span>

<span class="va">envfit_edaphic_md</span> <span class="op">&lt;-</span> <span class="fu">envfit</span><span class="op">(</span><span class="va">tmp_samp_scores_sub</span>, <span class="va">tmp_md_sub</span>,
                 perm <span class="op">=</span> <span class="fl">1000</span>, choices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
***VECTORS

         CAP1     CAP2     r2  Pr(&gt;r)  
AST   0.99126  0.13189 0.4849 0.03696 *
H2O  -0.99989  0.01511 0.2434 0.25275  
N     0.43981  0.89809 0.2124 0.27473  
P     0.43273  0.90152 0.0349 0.86014  
ECEC -0.50151  0.86515 0.2595 0.21778  
pH   -0.66815 -0.74403 0.0102 0.94705  
NH4  -0.81970  0.57279 0.0906 0.60140  
NO3  -0.76538  0.64357 0.0824 0.64635  
PO4  -0.85783  0.51393 0.3187 0.15185  
DOC  -0.99945  0.03317 0.5354 0.02797 *
DOCN  0.02108 -0.99978 0.1437 0.47552  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb135"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">edaphic_md_signif_hits</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">envfit_edaphic_md</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span>, 
                                  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">envfit_edaphic_md</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span> 
                                  <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">envfit_edaphic_md</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">r</span> <span class="op">&gt;</span> <span class="fl">0.4</span><span class="op">)</span><span class="op">)</span>
<span class="va">edaphic_md_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">edaphic_md_signif_hits</span><span class="op">)</span>
<span class="va">edaphic_md_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">edaphic_md_signif_hits</span><span class="op">)</span>
<span class="va">edaphic_md_signif</span> <span class="op">&lt;-</span> <span class="va">edaphic_md_scores</span><span class="op">[</span><span class="va">edaphic_md_scores</span><span class="op">$</span><span class="va">parameters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">edaphic_md_signif_hits</span>,<span class="op">]</span>
<span class="va">edaphic_md_signif</span><span class="op">$</span><span class="va">parameters</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><code>envfit</code> found that AST, DOC were significantly correlated.</p>
<p>Now lets see if the same parameters are significant for the <code>envfit</code> and <code>bioenv</code> analyses.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Significant parameters from bioenv analysis."
[1] "AST"
_____________________________________
[1] "Significant parameters from envfit analysis."
[1] "AST" "DOC"
_____________________________________
[1] "Found in bioenv but not envfit."
character(0)
_____________________________________
[1] "Found in envfit but not bioenv."
[1] "DOC"
_____________________________________
[1] "Found in envfit and bioenv."
[1] "AST" "DOC"</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb137"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">new_edaphic_md_signif_hits</span> <span class="op">&lt;-</span> <span class="va">edaphic_sig_diff</span>
<span class="co">#new_edaphic_md_signif_hits &lt;- append(edaphic_md_signif_hits, edaphic_sig_diff)</span>
<span class="va">edaphic_md_signif_all</span> <span class="op">&lt;-</span> <span class="va">edaphic_md_scores</span><span class="op">[</span><span class="va">edaphic_md_scores</span><span class="op">$</span><span class="va">parameters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">new_edaphic_md_signif_hits</span>,<span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Check. Next, we run <code>envfit</code> for the ASVs.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb138"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">envfit_edaphic_asv</span> <span class="op">&lt;-</span> <span class="fu">envfit</span><span class="op">(</span><span class="va">tmp_samp_scores_sub</span>, 
                             <span class="va">tmp_comm</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span><span class="op">(</span><span class="op">-</span><span class="va">tmp_comm</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>,
                             perm <span class="op">=</span> <span class="fl">1000</span>, choices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="va">edaphic_asv_scores</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tmp_auto_plt</span><span class="op">$</span><span class="va">plot_env</span><span class="op">$</span><span class="va">obj</span>, <span class="va">Score</span> <span class="op">==</span> <span class="st">"species"</span><span class="op">)</span>
<span class="va">edaphic_asv_scores</span> <span class="op">&lt;-</span> <span class="va">edaphic_asv_scores</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>parameters <span class="op">=</span> <span class="va">Label</span>, .before <span class="op">=</span> <span class="va">CAP1</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"Label"</span><span class="op">)</span>
<span class="va">edaphic_asv_scores</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
***VECTORS

          CAP1     CAP2     r2   Pr(&gt;r)   
ASV1  -0.61940 -0.78508 0.3539 0.010989 * 
ASV2   0.73098 -0.68240 0.5575 0.003996 **
ASV4  -0.33832 -0.94103 0.4713 0.010989 * 
ASV3  -0.89855  0.43888 0.4897 0.002997 **
ASV15  0.95614  0.29292 0.2269 0.192807   
ASV8  -0.87871 -0.47735 0.1267 0.795205   
ASV9   0.73176  0.68156 0.1410 0.468531   
ASV24  0.45502  0.89048 0.1661 0.459540   
ASV12  0.12678  0.99193 0.7513 0.001998 **
ASV13 -0.42157 -0.90680 0.4286 0.060939 . 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb140"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">edaphic_asv_signif_hits</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">envfit_edaphic_asv</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span>, 
                                  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">envfit_edaphic_asv</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span> 
                                  <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">envfit_edaphic_asv</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">r</span> <span class="op">&gt;</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>
<span class="va">edaphic_asv_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">edaphic_asv_signif_hits</span><span class="op">)</span>
<span class="va">edaphic_asv_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">edaphic_asv_signif_hits</span><span class="op">)</span>
<span class="va">edaphic_asv_signif</span> <span class="op">&lt;-</span> <span class="va">edaphic_asv_scores</span><span class="op">[</span><span class="va">edaphic_asv_scores</span><span class="op">$</span><span class="va">parameters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">edaphic_asv_signif_hits</span>,<span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>      parameters       CAP1       CAP2 variable_type
ASV12      ASV12 0.03291551  0.2041479           ASV
ASV2        ASV2 0.48655008 -0.3456690           ASV</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb142"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">edaphic_md_signif_all</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">&lt;-</span> <span class="st">"metadata"</span>
<span class="va">edaphic_asv_signif</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">&lt;-</span> <span class="st">"ASV"</span>
<span class="va">edaphic_bioplot_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">edaphic_md_signif_all</span>, <span class="va">edaphic_asv_signif</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The last thing to do is categorize parameters scores and ASV scores into different variable types for plotting.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb143"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">edaphic_bioplot_data_md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">edaphic_bioplot_data</span>, 
                                  <span class="va">edaphic_bioplot_data</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">==</span> <span class="st">"metadata"</span><span class="op">)</span>
<span class="va">edaphic_bioplot_data_asv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">edaphic_bioplot_data</span>, 
                                   <span class="va">edaphic_bioplot_data</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">==</span> <span class="st">"ASV"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<details><summary>
Show the code for the plot
</summary><div class="cell">

</div>
</details><div class="cell">

</div>
</section><section id="soil-functional-response-4" class="level3"><h3 class="anchored" data-anchor-id="soil-functional-response-4">Soil Functional Response</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb144"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_md</span> <span class="op">&lt;-</span> <span class="va">its18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">soil_funct</span>
<span class="va">tmp_md</span><span class="op">$</span><span class="va">TREAT_T</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">tmp_md</span><span class="op">$</span><span class="va">TREAT_T</span><span class="op">)</span>
<span class="va">tmp_comm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="va">its18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">data_loaded</span><span class="op">)</span><span class="op">)</span>
<span class="va">soil_funct_rank</span> <span class="op">&lt;-</span> <span class="fu">rankindex</span><span class="op">(</span><span class="va">tmp_md</span><span class="op">[</span>, <span class="fl">8</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_md</span><span class="op">)</span><span class="op">]</span>, <span class="va">tmp_comm</span>, 
          indices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"euc"</span>, <span class="st">"man"</span>, <span class="st">"gow"</span>, <span class="st">"bra"</span>, <span class="st">"kul"</span><span class="op">)</span>, 
          stepacross <span class="op">=</span> <span class="cn">FALSE</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>        euc         man         gow         bra         kul 
 0.06025620  0.38790324 -0.06200129  0.38790324  0.38790324 </code></pre>
</div>
</div>
<p>Lets run <code>capscale</code> using Bray-Curtis Note, we have 11 metadata parameters in this group but, for some reason, <code>capscale</code> only works with 13 parameters. This may have to do with degrees of freedom?</p>
<ul>
<li>Starting properties: micC, micN, micP, micCN, micCP, micNP, AG_ase, BG_ase, BP_ase, CE_ase, P_ase, N_ase, S_ase, XY_ase, LP_ase, PX_ase, CO2, enzCN, enzCP, enzNP</li>
<li>Autocorrelated removed: micN, micNP, enzCN, enzCP, BP_ase, CE_ase, LP_ase, N_ase, P_ase</li>
<li>Remove for capscale: NONE</li>
</ul>
<div class="cell">

</div>
<div class="cell">
<div class="sourceCode" id="cb146"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">soil_funct_cap</span> <span class="op">&lt;-</span> <span class="fu">capscale</span><span class="op">(</span><span class="va">tmp_comm</span> <span class="op">~</span>  <span class="va">micC</span> <span class="op">+</span> <span class="va">micP</span> <span class="op">+</span> <span class="va">micCN</span> <span class="op">+</span> <span class="va">micCP</span> <span class="op">+</span> <span class="va">AG_ase</span> <span class="op">+</span> <span class="va">BG_ase</span> <span class="op">+</span> 
                                       <span class="va">S_ase</span> <span class="op">+</span> <span class="va">XY_ase</span> <span class="op">+</span> <span class="va">PX_ase</span> <span class="op">+</span> <span class="va">CO2</span> <span class="op">+</span> <span class="va">enzNP</span>, 
                           <span class="va">tmp_md</span>, dist <span class="op">=</span> <span class="st">"bray"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Call: capscale(formula = tmp_comm ~ micC + micP + micCN + micCP +
AG_ase + BG_ase + S_ase + XY_ase + PX_ase + CO2 + enzNP, data = tmp_md,
distance = "bray")

              Inertia Proportion Rank
Total         3.88624    1.00000     
Constrained   3.59902    0.92609   11
Unconstrained 0.28722    0.07391    1
Inertia is squared Bray distance 
Species scores projected from 'tmp_comm' 

Eigenvalues for constrained axes:
  CAP1   CAP2   CAP3   CAP4   CAP5   CAP6   CAP7   CAP8   CAP9  CAP10  CAP11 
0.9666 0.5891 0.3939 0.3377 0.2686 0.2635 0.2306 0.1940 0.1389 0.1143 0.1018 

Eigenvalues for unconstrained axes:
   MDS1 
0.28722 </code></pre>
</div>
</div>
<p>Now we can look at the variance against each principal component.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-214-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>And then make some quick and dirty plots. This will also come in handy later when we need to parse out data a better plot visualization. The <code>ggplot</code> function <code>autoplot</code> stores these data in a more accessible way than the raw results from <code>capscale</code></p>
<div class="cell">
<div class="sourceCode" id="cb148"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span><span class="op">(</span><span class="va">soil_funct_cap</span><span class="op">)</span> </code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-215-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode" id="cb149"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_auto_plt</span> <span class="op">&lt;-</span> <span class="fu">autoplot</span><span class="op">(</span><span class="va">soil_funct_cap</span>, arrows <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">tmp_auto_plt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-215-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">

</div>
<p>Next, we need to grab capscale scores for the samples and create a data frame of the first two dimensions. We will also need to add some of the sample details to the data frame. For this we use the vegan function <code>scores</code> which gets species or site scores from the ordination.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb150"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_samp_scores</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tmp_auto_plt</span><span class="op">$</span><span class="va">plot_env</span><span class="op">$</span><span class="va">obj</span>, <span class="va">Score</span> <span class="op">==</span> <span class="st">"sites"</span><span class="op">)</span>
<span class="va">tmp_samp_scores</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">tmp_samp_scores</span> <span class="op">&lt;-</span> <span class="va">tmp_samp_scores</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>SampleID <span class="op">=</span> <span class="va">Label</span><span class="op">)</span>

<span class="va">tmp_md_sub</span> <span class="op">&lt;-</span> <span class="va">tmp_md</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>
<span class="va">tmp_md_sub</span> <span class="op">&lt;-</span> <span class="va">tmp_md_sub</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>
<span class="va">soil_funct_plot_data</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_md_sub</span>, <span class="va">tmp_samp_scores</span>, 
                                         by <span class="op">=</span> <span class="st">"SampleID"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Now we have a new data frame that contains sample details and capscale values.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>          SampleID PLOT TREAT TREAT_T PAIR        CAP1       CAP2
1  P10_D00_010_C0E  P10     C       0    E -0.69489162 -0.6675049
2  P02_D00_010_C0A  P02     C       0    A -0.91653485 -0.6229848
3  P04_D00_010_C0B  P04     C       0    B -0.95221897  0.3986135
4  P06_D00_010_C0C  P06     C       0    C -0.85713972  0.6268677
5  P08_D00_010_C0D  P08     C       0    D -0.09115803 -1.0802252
6  P01_D00_010_W3A  P01     W       3    A -0.80868973 -0.1643386
7  P03_D00_010_W3B  P03     W       3    B  0.27067347  0.9200087
8  P07_D00_010_W3D  P07     W       3    D  0.20427135  1.0264780
9  P09_D00_010_W3E  P09     W       3    E  0.57963547  0.4440199
10 P01_D00_010_W8A  P01     W       8    A  0.91663981 -0.1382072
11 P03_D00_010_W8B  P03     W       8    B  0.92199435 -0.6783194
12 P05_D00_010_W8C  P05     W       8    C  0.86406950 -0.9575454
13 P07_D00_010_W8D  P07     W       8    D  0.56334898  0.8931377</code></pre>
</div>
</div>
<p>We can then do the same with the metadata vectors. Here though we only need the scores and parameter name.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb152"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">soil_funct_md_scores</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tmp_auto_plt</span><span class="op">$</span><span class="va">plot_env</span><span class="op">$</span><span class="va">obj</span>, <span class="va">Score</span> <span class="op">==</span> <span class="st">"biplot"</span><span class="op">)</span>
<span class="va">soil_funct_md_scores</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">soil_funct_md_scores</span> <span class="op">&lt;-</span> <span class="va">soil_funct_md_scores</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>parameters <span class="op">=</span> <span class="va">Label</span>, .before <span class="op">=</span> <span class="va">CAP1</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"Label"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>       parameters       CAP1         CAP2
micC         micC -0.2359175  0.400384343
micP         micP -0.4872005  0.846844390
micCN       micCN  0.5085406  0.215263003
micCP       micCP  0.4299937 -0.786734788
AG_ase     AG_ase -0.3381241  0.763769448
BG_ase     BG_ase  0.2150249  0.021210923
S_ase       S_ase  0.1648172  0.678698244
XY_ase     XY_ase  0.6119362  0.164529372
PX_ase     PX_ase  0.8280069 -0.009182309
CO2           CO2  0.5373302 -0.181241128
enzNP       enzNP -0.8493447  0.170758052</code></pre>
</div>
</div>
<p>Lets run some quick correlations of metadata with ordination axes to see which parameters are significant. For this we use the vegan function <code>envfit</code>.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb154"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_samp_scores_sub</span> <span class="op">&lt;-</span> <span class="va">soil_funct_plot_data</span><span class="op">[</span>, <span class="fl">6</span><span class="op">:</span><span class="fl">7</span><span class="op">]</span>
<span class="va">tmp_samp_scores_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">tmp_samp_scores_sub</span><span class="op">)</span>
<span class="va">tmp_param_list</span> <span class="op">&lt;-</span> <span class="va">soil_funct_md_scores</span><span class="op">$</span><span class="va">parameters</span>

<span class="va">tmp_md_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">tmp_md</span>,  select <span class="op">=</span>  <span class="va">tmp_param_list</span><span class="op">)</span>

<span class="va">envfit_soil_funct_md</span> <span class="op">&lt;-</span> <span class="fu">envfit</span><span class="op">(</span><span class="va">tmp_samp_scores_sub</span>, <span class="va">tmp_md_sub</span>,
                 perm <span class="op">=</span> <span class="fl">1000</span>, choices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
***VECTORS

           CAP1     CAP2     r2   Pr(&gt;r)   
micC   -0.50911  0.86070 0.1567 0.442557   
micP   -0.50009  0.86597 0.6925 0.001998 **
micCN   0.92125  0.38897 0.2225 0.261738   
micCP   0.48092 -0.87676 0.5831 0.015984 * 
AG_ase -0.40574  0.91399 0.5060 0.036963 * 
BG_ase  0.99507  0.09917 0.0341 0.832168   
S_ase   0.23862  0.97111 0.3542 0.099900 . 
XY_ase  0.96571  0.25962 0.2931 0.170829   
PX_ase  0.99996 -0.00944 0.5002 0.034965 * 
CO2     0.94850 -0.31679 0.2342 0.250749   
enzNP  -0.98088  0.19460 0.5471 0.013986 * 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb156"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">soil_funct_md_signif_hits</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">envfit_soil_funct_md</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span>, 
                                  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">envfit_soil_funct_md</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span> 
                                  <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">envfit_soil_funct_md</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">r</span> <span class="op">&gt;</span> <span class="fl">0.4</span><span class="op">)</span><span class="op">)</span>
<span class="va">soil_funct_md_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">soil_funct_md_signif_hits</span><span class="op">)</span>
<span class="va">soil_funct_md_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">soil_funct_md_signif_hits</span><span class="op">)</span>
<span class="va">soil_funct_md_signif</span> <span class="op">&lt;-</span> <span class="va">soil_funct_md_scores</span><span class="op">[</span><span class="va">soil_funct_md_scores</span><span class="op">$</span><span class="va">parameters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> 
                                               <span class="va">soil_funct_md_signif_hits</span>,<span class="op">]</span>
<span class="va">soil_funct_md_signif</span><span class="op">$</span><span class="va">parameters</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><code>envfit</code> found that micP, micCP, AG_ase, PX_ase, enzNP were significantly correlated.</p>
<p>Now lets see if the same parameters are significant for the <code>envfit</code> and <code>bioenv</code> analyses.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Significant parameters from bioenv analysis."
[1] "enzNP"  "PX_ase" "XY_ase"
_____________________________________
[1] "Significant parameters from envfit analysis."
[1] "micP"   "micCP"  "AG_ase" "PX_ase" "enzNP" 
_____________________________________
[1] "Found in bioenv but not envfit."
[1] "XY_ase"
_____________________________________
[1] "Found in envfit but not bioenv."
[1] "micP"   "micCP"  "AG_ase"
_____________________________________
[1] "Found in envfit and bioenv."
[1] "micP"   "micCP"  "AG_ase" "PX_ase" "enzNP"  "XY_ase"</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb158"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="co">#new_soil_funct_md_signif_hits &lt;- append(soil_funct_md_signif_hits, soil_funct_sig_diff)</span>
<span class="va">new_soil_funct_md_signif_hits</span> <span class="op">&lt;-</span> <span class="va">soil_funct_sig_diff</span>
<span class="va">soil_funct_md_signif_all</span> <span class="op">&lt;-</span> <span class="va">soil_funct_md_scores</span><span class="op">[</span><span class="va">soil_funct_md_scores</span><span class="op">$</span><span class="va">parameters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> 
                                                   <span class="va">new_soil_funct_md_signif_hits</span>,<span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Check. Next, we run <code>envfit</code> for the ASVs.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb159"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">envfit_soil_funct_asv</span> <span class="op">&lt;-</span> <span class="fu">envfit</span><span class="op">(</span><span class="va">tmp_samp_scores_sub</span>, 
                                <span class="va">tmp_comm</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span><span class="op">(</span><span class="op">-</span><span class="va">tmp_comm</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>, 
                                perm <span class="op">=</span> <span class="fl">1000</span>, choices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="va">soil_funct_asv_scores</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tmp_auto_plt</span><span class="op">$</span><span class="va">plot_env</span><span class="op">$</span><span class="va">obj</span>, <span class="va">Score</span> <span class="op">==</span> <span class="st">"species"</span><span class="op">)</span>
<span class="va">soil_funct_asv_scores</span> <span class="op">&lt;-</span> <span class="va">soil_funct_asv_scores</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>parameters <span class="op">=</span> <span class="va">Label</span>, .before <span class="op">=</span> <span class="va">CAP1</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"Label"</span><span class="op">)</span>
<span class="va">soil_funct_asv_scores</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
***VECTORS

          CAP1     CAP2     r2   Pr(&gt;r)    
ASV1  -0.79963 -0.60050 0.2256 0.245754    
ASV2   0.68178 -0.73156 0.6129 0.000999 ***
ASV4  -0.33208 -0.94325 0.5433 0.000999 ***
ASV3  -0.92213  0.38687 0.4516 0.010989 *  
ASV15  0.95847  0.28518 0.2287 0.159840    
ASV8  -0.97867 -0.20544 0.1064 0.930070    
ASV9   0.76380  0.64546 0.1326 0.481518    
ASV24  0.40983  0.91216 0.2438 0.065934 .  
ASV12  0.14788  0.98901 0.7872 0.000999 ***
ASV13 -0.42711 -0.90420 0.4546 0.044955 *  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb161"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">soil_funct_asv_signif_hits</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">envfit_soil_funct_asv</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span>, 
                                  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">envfit_soil_funct_asv</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span> 
                                  <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">envfit_soil_funct_asv</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">r</span> <span class="op">&gt;</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>
<span class="va">soil_funct_asv_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">soil_funct_asv_signif_hits</span><span class="op">)</span>
<span class="va">soil_funct_asv_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">soil_funct_asv_signif_hits</span><span class="op">)</span>
<span class="va">soil_funct_asv_signif</span> <span class="op">&lt;-</span> <span class="va">soil_funct_asv_scores</span><span class="op">[</span><span class="va">soil_funct_asv_scores</span><span class="op">$</span><span class="va">parameters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> 
                                                 <span class="va">soil_funct_asv_signif_hits</span>,<span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>      parameters        CAP1       CAP2 variable_type
ASV4        ASV4 -0.10677580 -0.2996742           ASV
ASV12      ASV12  0.04088024  0.2662493           ASV
ASV2        ASV2  0.46539405 -0.4834130           ASV</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb163"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">soil_funct_md_signif_all</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">&lt;-</span> <span class="st">"metadata"</span>
<span class="va">soil_funct_asv_signif</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">&lt;-</span> <span class="st">"ASV"</span>
<span class="va">soil_funct_bioplot_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">soil_funct_md_signif_all</span>, <span class="va">soil_funct_asv_signif</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The last thing to do is categorize parameters scores and ASV scores into different variable types for plotting.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb164"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">soil_funct_bioplot_data_md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">soil_funct_bioplot_data</span>, 
                                  <span class="va">soil_funct_bioplot_data</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">==</span> <span class="st">"metadata"</span><span class="op">)</span>
<span class="va">soil_funct_bioplot_data_asv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">soil_funct_bioplot_data</span>, 
                                   <span class="va">soil_funct_bioplot_data</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">==</span> <span class="st">"ASV"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<details><summary>
Show the code for the plot
</summary><div class="cell">
<div class="sourceCode" id="cb165"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">soil_funct_cap_vals</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">soil_funct_cap</span><span class="op">$</span><span class="va">CCA</span><span class="op">$</span><span class="va">eig</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>
<span class="va">soil_funct_cap1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span><span class="op">(</span><span class="va">soil_funct_cap_vals</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span>, digits<span class="op">=</span><span class="fl">3</span><span class="op">)</span>
<span class="va">soil_funct_cap2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span><span class="op">(</span><span class="va">soil_funct_cap_vals</span><span class="op">[</span><span class="fl">2</span>,<span class="op">]</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span>, digits<span class="op">=</span><span class="fl">3</span><span class="op">)</span>

<span class="va">cpa1_lab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"CAP1"</span>, <span class="st">" ("</span>, <span class="va">soil_funct_cap1</span>, <span class="st">"%)"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
<span class="va">cpa2_lab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"CAP2"</span>, <span class="st">" ("</span>, <span class="va">soil_funct_cap2</span>, <span class="st">"%)"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>

<span class="va">swel_col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#2271B2"</span>, <span class="st">"#71B222"</span>, <span class="st">"#B22271"</span><span class="op">)</span>

<span class="va">soil_funct_plot</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">soil_funct_plot_data</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">CAP1</span>, y <span class="op">=</span> <span class="va">CAP2</span>, shape <span class="op">=</span> <span class="va">TREAT</span>,
                           colour <span class="op">=</span> <span class="va">TREAT_T</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_colour_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">swel_col</span><span class="op">)</span> <span class="op">+</span>
<span class="co">#  geom_text(data = soil_funct_plot_data, aes(x = CAP1, y = CAP2, #UNCOMMENT to add sample labels</span>
<span class="co">#                              label = SampleID), size = 3) + </span>
  <span class="fu">geom_segment</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="fl">0</span>, y <span class="op">=</span> <span class="fl">0</span>, xend <span class="op">=</span> <span class="va">CAP1</span>, yend <span class="op">=</span> <span class="va">CAP2</span><span class="op">)</span>,
               data <span class="op">=</span> <span class="va">soil_funct_bioplot_data_md</span>, linetype <span class="op">=</span> <span class="st">"solid"</span>,
               arrow <span class="op">=</span> <span class="fu">arrow</span><span class="op">(</span>length <span class="op">=</span> <span class="fu">unit</span><span class="op">(</span><span class="fl">0.3</span>, <span class="st">"cm"</span><span class="op">)</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">0.4</span>, 
               color <span class="op">=</span> <span class="st">"#191919"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_text</span><span class="op">(</span>data <span class="op">=</span> <span class="va">soil_funct_bioplot_data_md</span>, 
            <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">CAP1</span>, y <span class="op">=</span> <span class="va">CAP2</span>, label <span class="op">=</span> <span class="va">parameters</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">3</span>,
            nudge_x <span class="op">=</span> <span class="fl">0.1</span>, nudge_y <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">+</span>
  <span class="co">## USE this code to overlay ASV vestors</span>
  <span class="co">#geom_segment(aes(x = 0, y = 0, xend = CAP1, yend = CAP2),</span>
  <span class="co">#             data = soil_funct_bioplot_data_asv, linetype = "solid",</span>
  <span class="co">#             arrow = arrow(length = unit(0.3, "cm")), size = 0.2,</span>
  <span class="co">#             color = "#676767") +</span>
  <span class="co">#geom_text(data = soil_funct_bioplot_data_asv, </span>
  <span class="co">#          aes(x = CAP1, y = CAP2, label = parameters), size = 2.5,</span>
  <span class="co">#          nudge_x = 0.05, nudge_y = 0.05) +</span>
  <span class="fu">theme_classic</span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Capscale Analysis"</span>,
       subtitle <span class="op">=</span> <span class="st">"Soil Functional Response"</span>, 
       x <span class="op">=</span> <span class="va">cpa1_lab</span>, 
       y <span class="op">=</span> <span class="va">cpa2_lab</span><span class="op">)</span>
<span class="va">soil_funct_plot</span> <span class="op">&lt;-</span> <span class="va">soil_funct_plot</span> <span class="op">+</span> <span class="fu">coord_fixed</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
<span class="va">soil_funct_plot</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/png.html">png</a></span><span class="op">(</span><span class="st">"files/metadata/figures/its18_soil_funct_capscale.png"</span>,
    height <span class="op">=</span> <span class="fl">16</span>, width <span class="op">=</span> <span class="fl">20</span>, units <span class="op">=</span> <span class="st">'cm'</span>, res <span class="op">=</span> <span class="fl">600</span>, bg <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span>
<span class="va">soil_funct_plot</span>
<span class="fu"><a href="https://rdrr.io/r/base/invisible.html">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span><span class="op">(</span><span class="st">"files/metadata/figures/its18_soil_funct_capscale.pdf"</span>,
    height <span class="op">=</span> <span class="fl">5</span>, width <span class="op">=</span> <span class="fl">6</span><span class="op">)</span>
<span class="va">soil_funct_plot</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span><span class="op">(</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</details><div class="cell">

</div>
</section><section id="temperature-adaptation-6" class="level3"><h3 class="anchored" data-anchor-id="temperature-adaptation-6">Temperature Adaptation</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb166"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_md</span> <span class="op">&lt;-</span> <span class="va">its18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">temp_adapt</span>
<span class="va">tmp_md</span><span class="op">$</span><span class="va">TREAT_T</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">tmp_md</span><span class="op">$</span><span class="va">TREAT_T</span><span class="op">)</span>
<span class="va">tmp_comm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="va">its18_select_mc_norm_split_no_ac</span><span class="op">$</span><span class="va">data_loaded</span><span class="op">)</span><span class="op">)</span>
<span class="va">temp_adapt_rank</span> <span class="op">&lt;-</span> <span class="fu">rankindex</span><span class="op">(</span><span class="va">tmp_md</span><span class="op">[</span>, <span class="fl">8</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_md</span><span class="op">)</span><span class="op">]</span>, <span class="va">tmp_comm</span>, 
          indices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"euc"</span>, <span class="st">"man"</span>, <span class="st">"gow"</span>, <span class="st">"bra"</span>, <span class="st">"kul"</span><span class="op">)</span>, 
          stepacross <span class="op">=</span> <span class="cn">FALSE</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>       euc        man        gow        bra        kul 
0.05132842 0.39465598 0.27457353 0.39465598 0.39465598 </code></pre>
</div>
</div>
<p>Lets run <code>capscale</code> using Bray-Curtis. Note, we have 12 metadata parameters in this group but, for some reason, <code>capscale</code> only works with 13 parameters. This may have to do with degrees of freedom?</p>
<ul>
<li>Starting properties: AG_Q10, BG_Q10, BP_Q10, CE_Q10, P_Q10, N_Q10, S_Q10, XY_Q10, LP_Q10, PX_Q10, CUEcn, CUEcp, NUE, PUE, Tmin, SI</li>
<li>Autocorrelated removed: NUE, PUE, P_Q10, SI<br>
</li>
<li>Remove for capscale: S_Q10</li>
</ul>
<div class="cell">

</div>
<div class="cell">
<div class="sourceCode" id="cb168"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">temp_adapt_cap</span> <span class="op">&lt;-</span> <span class="fu">capscale</span><span class="op">(</span><span class="va">tmp_comm</span> <span class="op">~</span>  <span class="va">AG_Q10</span> <span class="op">+</span> <span class="va">BG_Q10</span> <span class="op">+</span> <span class="va">BP_Q10</span> <span class="op">+</span> <span class="va">CE_Q10</span> <span class="op">+</span> 
                                       <span class="va">N_Q10</span> <span class="op">+</span> <span class="va">XY_Q10</span> <span class="op">+</span> <span class="va">LP_Q10</span> <span class="op">+</span> <span class="va">PX_Q10</span> <span class="op">+</span> 
                                       <span class="va">CUEcn</span> <span class="op">+</span> <span class="va">CUEcp</span> <span class="op">+</span> <span class="va">Tmin</span>, 
                           <span class="va">tmp_md</span>, dist <span class="op">=</span> <span class="st">"bray"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Call: capscale(formula = tmp_comm ~ AG_Q10 + BG_Q10 + BP_Q10 + CE_Q10 +
N_Q10 + XY_Q10 + LP_Q10 + PX_Q10 + CUEcn + CUEcp + Tmin, data = tmp_md,
distance = "bray")

              Inertia Proportion Rank
Total         3.88624    1.00000     
Constrained   3.64416    0.93771   11
Unconstrained 0.24208    0.06229    1
Inertia is squared Bray distance 
Species scores projected from 'tmp_comm' 

Eigenvalues for constrained axes:
  CAP1   CAP2   CAP3   CAP4   CAP5   CAP6   CAP7   CAP8   CAP9  CAP10  CAP11 
0.9367 0.5976 0.4447 0.3685 0.3136 0.2644 0.2093 0.1488 0.1377 0.1235 0.0994 

Eigenvalues for unconstrained axes:
   MDS1 
0.24208 </code></pre>
</div>
</div>
<p>Now we can look at the variance against each principal component.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-239-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>And then make some quick and dirty plots. This will also come in handy later when we need to parse out data a better plot visualization. The <code>ggplot</code> function <code>autoplot</code> stores these data in a more accessible way than the raw results from <code>capscale</code></p>
<div class="cell">
<div class="sourceCode" id="cb170"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span><span class="op">(</span><span class="va">temp_adapt_cap</span><span class="op">)</span> </code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-240-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode" id="cb171"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_auto_plt</span> <span class="op">&lt;-</span> <span class="fu">autoplot</span><span class="op">(</span><span class="va">temp_adapt_cap</span>, arrows <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">tmp_auto_plt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-240-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">

</div>
<p>Next, we need to grab capscale scores for the samples and create a data frame of the first two dimensions. We will also need to add some of the sample details to the data frame. For this we use the vegan function <code>scores</code> which gets species or site scores from the ordination.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb172"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_samp_scores</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tmp_auto_plt</span><span class="op">$</span><span class="va">plot_env</span><span class="op">$</span><span class="va">obj</span>, <span class="va">Score</span> <span class="op">==</span> <span class="st">"sites"</span><span class="op">)</span>
<span class="va">tmp_samp_scores</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">tmp_samp_scores</span> <span class="op">&lt;-</span> <span class="va">tmp_samp_scores</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>SampleID <span class="op">=</span> <span class="va">Label</span><span class="op">)</span>

<span class="va">tmp_md_sub</span> <span class="op">&lt;-</span> <span class="va">tmp_md</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>
<span class="va">tmp_md_sub</span> <span class="op">&lt;-</span> <span class="va">tmp_md_sub</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span>
<span class="va">temp_adapt_plot_data</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_md_sub</span>, <span class="va">tmp_samp_scores</span>, 
                                         by <span class="op">=</span> <span class="st">"SampleID"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Now we have a new data frame that contains sample details and capscale values.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>          SampleID PLOT TREAT TREAT_T PAIR        CAP1       CAP2
1  P10_D00_010_C0E  P10     C       0    E -0.70909404  0.8884734
2  P02_D00_010_C0A  P02     C       0    A -0.88932428  0.4678685
3  P04_D00_010_C0B  P04     C       0    B -0.94720017 -0.5168746
4  P06_D00_010_C0C  P06     C       0    C -0.94117253 -0.7283971
5  P08_D00_010_C0D  P08     C       0    D -0.09697067  1.0013016
6  P01_D00_010_W3A  P01     W       3    A -0.79138504  0.3434287
7  P03_D00_010_W3B  P03     W       3    B  0.28447425 -0.8800769
8  P07_D00_010_W3D  P07     W       3    D  0.18992920 -0.9828048
9  P09_D00_010_W3E  P09     W       3    E  0.57704340 -0.4016378
10 P01_D00_010_W8A  P01     W       8    A  0.98434765  0.1432237
11 P03_D00_010_W8B  P03     W       8    B  0.88625427  0.6330378
12 P05_D00_010_W8C  P05     W       8    C  0.84253827  0.9056297
13 P07_D00_010_W8D  P07     W       8    D  0.61055970 -0.8731722</code></pre>
</div>
</div>
<p>We can then do the same with the metadata vectors. Here though we only need the scores and parameter name.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb174"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">temp_adapt_md_scores</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tmp_auto_plt</span><span class="op">$</span><span class="va">plot_env</span><span class="op">$</span><span class="va">obj</span>, <span class="va">Score</span> <span class="op">==</span> <span class="st">"biplot"</span><span class="op">)</span>
<span class="va">temp_adapt_md_scores</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">temp_adapt_md_scores</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_md_scores</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>parameters <span class="op">=</span> <span class="va">Label</span>, .before <span class="op">=</span> <span class="va">CAP1</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"Label"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>       parameters         CAP1        CAP2
AG_Q10     AG_Q10  0.138111424 -0.09067317
BG_Q10     BG_Q10  0.529871661  0.44030269
BP_Q10     BP_Q10  0.213674949  0.18420563
CE_Q10     CE_Q10 -0.470374849 -0.07935444
N_Q10       N_Q10  0.002856862  0.28186886
XY_Q10     XY_Q10  0.826073837  0.18643092
LP_Q10     LP_Q10 -0.428296135 -0.07664534
PX_Q10     PX_Q10 -0.361920412  0.05606159
CUEcn       CUEcn -0.146194994 -0.40205479
CUEcp       CUEcp  0.318510258  0.66752667
Tmin         Tmin  0.740618311 -0.05737920</code></pre>
</div>
</div>
<p>Lets run some quick correlations of metadata with ordination axes to see which parameters are significant. For this we use the vegan function <code>envfit</code>.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb176"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_samp_scores_sub</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_plot_data</span><span class="op">[</span>, <span class="fl">6</span><span class="op">:</span><span class="fl">7</span><span class="op">]</span>
<span class="va">tmp_samp_scores_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">tmp_samp_scores_sub</span><span class="op">)</span>
<span class="va">tmp_param_list</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_md_scores</span><span class="op">$</span><span class="va">parameters</span>

<span class="va">tmp_md_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">tmp_md</span>,  select <span class="op">=</span>  <span class="va">tmp_param_list</span><span class="op">)</span>

<span class="va">envfit_temp_adapt_md</span> <span class="op">&lt;-</span> <span class="fu">envfit</span><span class="op">(</span><span class="va">tmp_samp_scores_sub</span>, <span class="va">tmp_md_sub</span>,
                 perm <span class="op">=</span> <span class="fl">1000</span>, choices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
***VECTORS

           CAP1     CAP2     r2  Pr(&gt;r)   
AG_Q10  0.82904 -0.55918 0.0237 0.89710   
BG_Q10  0.76182  0.64778 0.4109 0.07792 . 
BP_Q10  0.74983  0.66163 0.0689 0.71528   
CE_Q10 -0.98593 -0.16714 0.1956 0.34865   
N_Q10   0.00657  0.99998 0.0700 0.71029   
XY_Q10  0.97498  0.22227 0.6166 0.00999 **
LP_Q10 -0.98418 -0.17719 0.1627 0.44955   
PX_Q10 -0.98713  0.15993 0.1155 0.54246   
CUEcn  -0.33206 -0.94326 0.1605 0.40460   
CUEcp   0.42027  0.90740 0.4786 0.03497 * 
Tmin    0.99659 -0.08248 0.4749 0.02797 * 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb178"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">temp_adapt_md_signif_hits</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">envfit_temp_adapt_md</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span>, 
                                  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">envfit_temp_adapt_md</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span> 
                                  <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">envfit_temp_adapt_md</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">r</span> <span class="op">&gt;</span> <span class="fl">0.4</span><span class="op">)</span><span class="op">)</span>
<span class="va">temp_adapt_md_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">temp_adapt_md_signif_hits</span><span class="op">)</span>
<span class="va">temp_adapt_md_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">temp_adapt_md_signif_hits</span><span class="op">)</span>
<span class="va">temp_adapt_md_signif</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_md_scores</span><span class="op">[</span><span class="va">temp_adapt_md_scores</span><span class="op">$</span><span class="va">parameters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> 
                                               <span class="va">temp_adapt_md_signif_hits</span>,<span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><code>envfit</code> found that XY_Q10, CUEcp, Tmin were significantly correlated.</p>
<p>Now lets see if the same parameters are significant for the <code>envfit</code> and <code>bioenv</code> analyses.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Significant parameters from bioenv analysis."
[1] "XY_Q10" "Tmin"  
_____________________________________
[1] "Significant parameters from envfit analysis."
[1] "XY_Q10" "CUEcp"  "Tmin"  
_____________________________________
[1] "Found in bioenv but not envfit."
character(0)
_____________________________________
[1] "Found in envfit but not bioenv."
[1] "CUEcp"
_____________________________________
[1] "Found in envfit and  bioenv."
[1] "XY_Q10" "CUEcp"  "Tmin"  </code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb180"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="co">#new_temp_adapt_md_signif_hits &lt;- base::append(temp_adapt_md_signif_hits, temp_adapt_sig_diff)</span>
<span class="va">new_temp_adapt_md_signif_hits</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_sig_diff</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>
<span class="va">temp_adapt_md_signif_all</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_md_scores</span><span class="op">[</span><span class="va">temp_adapt_md_scores</span><span class="op">$</span><span class="va">parameters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> 
                                                   <span class="va">new_temp_adapt_md_signif_hits</span>,<span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Check. Next, we run <code>envfit</code> for the ASVs.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb181"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">envfit_temp_adapt_asv</span> <span class="op">&lt;-</span> <span class="fu">envfit</span><span class="op">(</span><span class="va">tmp_samp_scores_sub</span>, 
                                <span class="va">tmp_comm</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span><span class="op">(</span><span class="op">-</span><span class="va">tmp_comm</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>,
                                perm <span class="op">=</span> <span class="fl">1000</span>, choices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>

<span class="va">temp_adapt_asv_scores</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tmp_auto_plt</span><span class="op">$</span><span class="va">plot_env</span><span class="op">$</span><span class="va">obj</span>, <span class="va">Score</span> <span class="op">==</span> <span class="st">"species"</span><span class="op">)</span>

<span class="va">temp_adapt_asv_scores</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_asv_scores</span> <span class="op">%&gt;%</span> 
                         <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>parameters <span class="op">=</span> <span class="va">Label</span>, .before <span class="op">=</span> <span class="va">CAP1</span><span class="op">)</span> <span class="op">%&gt;%</span>
                         <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"Label"</span><span class="op">)</span>
<span class="va">temp_adapt_asv_scores</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
***VECTORS

          CAP1     CAP2     r2   Pr(&gt;r)    
ASV1  -0.68969  0.72410 0.2975 0.051948 .  
ASV2   0.68393  0.72955 0.5592 0.003996 ** 
ASV4  -0.34226  0.93961 0.4935 0.005994 ** 
ASV3  -0.89727 -0.44148 0.5062 0.000999 ***
ASV15  0.96528 -0.26123 0.2497 0.122877    
ASV8  -0.90985  0.41495 0.1146 0.923077    
ASV9   0.76209 -0.64748 0.1338 0.474525    
ASV24  0.44129 -0.89736 0.2375 0.095904 .  
ASV12  0.14568 -0.98933 0.7737 0.002997 ** 
ASV13 -0.41856  0.90819 0.4357 0.056943 .  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
Permutation: free
Number of permutations: 1000</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb183"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">temp_adapt_asv_signif_hits</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">envfit_temp_adapt_asv</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span>, 
                                  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">envfit_temp_adapt_asv</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">pvals</span> 
                                  <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">envfit_temp_adapt_asv</span><span class="op">$</span><span class="va">vectors</span><span class="op">$</span><span class="va">r</span> <span class="op">&gt;</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>
<span class="va">temp_adapt_asv_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">temp_adapt_asv_signif_hits</span><span class="op">)</span>
<span class="va">temp_adapt_asv_signif_hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">temp_adapt_asv_signif_hits</span><span class="op">)</span>
<span class="va">temp_adapt_asv_signif</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_asv_scores</span><span class="op">[</span><span class="va">temp_adapt_asv_scores</span><span class="op">$</span><span class="va">parameters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> 
                                                 <span class="va">temp_adapt_asv_signif_hits</span>,<span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>      parameters        CAP1       CAP2 variable_type
ASV12      ASV12  0.03533894 -0.2327656           ASV
ASV2        ASV2  0.38413654  0.3977765           ASV
ASV3        ASV3 -0.45977392 -0.1843674           ASV</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb185"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">temp_adapt_md_signif_all</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">&lt;-</span> <span class="st">"metadata"</span>
<span class="va">temp_adapt_asv_signif</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">&lt;-</span> <span class="st">"ASV"</span>
<span class="va">temp_adapt_bioplot_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">temp_adapt_md_signif_all</span>, <span class="va">temp_adapt_asv_signif</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The last thing to do is categorize parameters scores and ASV scores into different variable types for plotting.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb186"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">temp_adapt_bioplot_data_md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">temp_adapt_bioplot_data</span>, 
                                  <span class="va">temp_adapt_bioplot_data</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">==</span> <span class="st">"metadata"</span><span class="op">)</span>
<span class="va">temp_adapt_bioplot_data_asv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">temp_adapt_bioplot_data</span>, 
                                   <span class="va">temp_adapt_bioplot_data</span><span class="op">$</span><span class="va">variable_type</span> <span class="op">==</span> <span class="st">"ASV"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<details><summary>
Show the code for the plot
</summary><div class="cell">
<div class="sourceCode" id="cb187"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">temp_adapt_cap_vals</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">temp_adapt_cap</span><span class="op">$</span><span class="va">CCA</span><span class="op">$</span><span class="va">eig</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>
<span class="va">temp_adapt_cap1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span><span class="op">(</span><span class="va">temp_adapt_cap_vals</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span>, digits<span class="op">=</span><span class="fl">3</span><span class="op">)</span>
<span class="va">temp_adapt_cap2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span><span class="op">(</span><span class="va">temp_adapt_cap_vals</span><span class="op">[</span><span class="fl">2</span>,<span class="op">]</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span>, digits<span class="op">=</span><span class="fl">3</span><span class="op">)</span>

<span class="va">cpa1_lab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"CAP1"</span>, <span class="st">" ("</span>, <span class="va">temp_adapt_cap1</span>, <span class="st">"%)"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
<span class="va">cpa2_lab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"CAP2"</span>, <span class="st">" ("</span>, <span class="va">temp_adapt_cap2</span>, <span class="st">"%)"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>

<span class="va">swel_col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#2271B2"</span>, <span class="st">"#71B222"</span>, <span class="st">"#B22271"</span><span class="op">)</span>

<span class="va">temp_adapt_plot</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">temp_adapt_plot_data</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">CAP1</span>, y <span class="op">=</span> <span class="va">CAP2</span>, shape <span class="op">=</span> <span class="va">TREAT</span>,
                           colour <span class="op">=</span> <span class="va">TREAT_T</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_colour_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">swel_col</span><span class="op">)</span> <span class="op">+</span>
<span class="co">#  geom_text(data = temp_adapt_plot_data, aes(x = CAP1, y = CAP2, #UNCOMMENT to add sample labels</span>
<span class="co">#                              label = SampleID), size = 3) + </span>
  <span class="fu">geom_segment</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="fl">0</span>, y <span class="op">=</span> <span class="fl">0</span>, xend <span class="op">=</span> <span class="va">CAP1</span>, yend <span class="op">=</span> <span class="va">CAP2</span><span class="op">)</span>,
               data <span class="op">=</span> <span class="va">temp_adapt_bioplot_data_md</span>, linetype <span class="op">=</span> <span class="st">"solid"</span>,
               arrow <span class="op">=</span> <span class="fu">arrow</span><span class="op">(</span>length <span class="op">=</span> <span class="fu">unit</span><span class="op">(</span><span class="fl">0.3</span>, <span class="st">"cm"</span><span class="op">)</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">0.4</span>,
               color <span class="op">=</span> <span class="st">"#191919"</span>, inherit.aes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_text</span><span class="op">(</span>data <span class="op">=</span> <span class="va">temp_adapt_bioplot_data_md</span>, 
            <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">CAP1</span>, y <span class="op">=</span> <span class="va">CAP2</span>, label <span class="op">=</span> <span class="va">parameters</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">3</span>,
            nudge_x <span class="op">=</span> <span class="fl">0.1</span>, nudge_y <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">+</span>
  <span class="co"># uSe to include ASV vectors</span>
  <span class="co">#geom_segment(aes(x = 0, y = 0, xend = CAP1, yend = CAP2),</span>
  <span class="co">#             data = temp_adapt_bioplot_data_asv, linetype = "solid",</span>
  <span class="co">#             arrow = arrow(length = unit(0.3, "cm")), size = 0.2,</span>
  <span class="co">#             color = "#676767") +</span>
  <span class="co">#geom_text(data = temp_adapt_bioplot_data_asv, </span>
  <span class="co">#          aes(x = CAP1, y = CAP2, label = parameters), size = 2.5,</span>
  <span class="co">#          nudge_x = 0.05, nudge_y = 0.05) +</span>
  <span class="fu">theme_classic</span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Capscale Analysis"</span>,
       subtitle <span class="op">=</span> <span class="st">"Temperature Adaptation"</span>,
       x <span class="op">=</span> <span class="va">cpa1_lab</span>, 
       y <span class="op">=</span> <span class="va">cpa2_lab</span><span class="op">)</span>
<span class="va">temp_adapt_plot</span> <span class="op">&lt;-</span> <span class="va">temp_adapt_plot</span> <span class="op">+</span> <span class="fu">coord_fixed</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
<span class="va">temp_adapt_plot</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/png.html">png</a></span><span class="op">(</span><span class="st">"files/metadata/figures/its18_temp_adapt_capscale.png"</span>,
    height <span class="op">=</span> <span class="fl">16</span>, width <span class="op">=</span> <span class="fl">20</span>, units <span class="op">=</span> <span class="st">'cm'</span>, res <span class="op">=</span> <span class="fl">600</span>, bg <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span>
<span class="va">temp_adapt_plot</span>
<span class="fu"><a href="https://rdrr.io/r/base/invisible.html">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span><span class="op">(</span><span class="st">"files/metadata/figures/its18_temp_adapt_capscale.pdf"</span>,
    height <span class="op">=</span> <span class="fl">5</span>, width <span class="op">=</span> <span class="fl">6</span><span class="op">)</span>
<span class="va">temp_adapt_plot</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span><span class="op">(</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</details><div class="cell">

</div>
</section></section><section id="capscale-plots-1" class="level2 page-columns page-full"><h2 class="anchored" data-anchor-id="capscale-plots-1">Capscale Plots</h2>
<div class="column-body">
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-8-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-8-1" role="tab" aria-controls="tabset-8-1" aria-selected="true">Edaphic properties</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-8-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-8-2" role="tab" aria-controls="tabset-8-2" aria-selected="false">Soil Functional Response</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-8-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-8-3" role="tab" aria-controls="tabset-8-3" aria-selected="false">Temperature Adaptation</a></li>
</ul>
<div class="tab-content">
<div id="tabset-8-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-8-1-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-259-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><small><strong>(ITS) Figure 7 | Distance-based Redundancy Analysis (db-RDA) of PIME filtered data based on Bray-Curtis dissimilarity showing the relationships between community composition change versus edaphic properties.</strong></small></p>
</div>
<div id="tabset-8-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-8-2-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-260-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><small><strong>(ITS) Figure 8 | Distance-based Redundancy Analysis (db-RDA) of PIME filtered data based on Bray-Curtis dissimilarity showing the relationships between community composition change versus microbial functional response.</strong></small></p>
</div>
<div id="tabset-8-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-8-3-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="metadata_files/figure-html/unnamed-chunk-261-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><small><strong>(ITS) Figure 9 | Distance-based Redundancy Analysis (db-RDA) of PIME filtered data based on Bray-Curtis dissimilarity showing the relationships between community composition change versus temperature adaptation.</strong></small></p>
</div>
</div>
</div>
</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
</section></section><section id="workflow-output" class="level1"><h1>Workflow Output</h1>
<p>Data products generated in this workflow can be downloaded from figshare.</p>
<iframe src="https://widgets.figshare.com/articles/16828294/embed?show_title=1" width="100%" height="251" allowfullscreen="" frameborder="0">
</iframe>
<p></p>
<p><a href="da.html" class="btnnav button_round" style="float: left;">Previous workflow: <br> 7. Differentially Abundant ASVs &amp; Taxa</a><br><br></p>
<div class="cell">

</div>
</section><div id="quarto-appendix" class="default"><section id="source-code" class="level4 appendix"><h2 class="quarto-appendix-heading">Source Code</h2><div class="quarto-appendix-contents">
<p>The source code for this page can be accessed on GitHub <svg aria-hidden="true" role="img" viewbox="0 0 496 512" style="height:1em;width:0.97em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:currentColor;overflow:visible;position:relative;"><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"></path></svg> by <a href="https://github.com/sweltr/high-temp/blob/main/metadata.qmd">clicking this link</a>.</p>
</div></section><section id="data-availability" class="level4 appendix"><h2 class="quarto-appendix-heading">Data Availability</h2><div class="quarto-appendix-contents">
<p>Data generated in this workflow and the Rdata need to run the workflow can be accessed on figshare at <a href="https://doi.org/10.25573/data.16828294">10.25573/data.16828294</a>.</p>
</div></section><section id="last-updated-on" class="level4 appendix"><h2 class="quarto-appendix-heading">Last updated on</h2><div class="quarto-appendix-contents">
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "2022-06-29 07:22:52 EST"</code></pre>
</div>
</div>


<!-- -->


</div></section><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-oksanen2013community" class="csl-entry" role="doc-biblioentry">
Oksanen, Jari, F. Guillaume Blanchet, Roeland Kindt, Pierre Legendre, Peter R. Minchin, R. B. OHara, Gavin L. Simpson, Peter Slymos, M. Henry H. Stevens, and Helene Wagner. 2012. <span>Vegan: Community Ecology Package.</span> <em>R Package Version</em> 2 (0). <a href="https://cran.r-project.org/web/packages/vegan/index.html">https://cran.r-project.org/web/packages/vegan/index.html</a>.
</div>
</div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./da.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">7. Differentially Abundant ASVs &amp; Taxa</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./data-availability.html" class="pagination-link">
        <span class="nav-page-text">Data Availability</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb189" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb189-1"><a href="#cb189-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb189-2"><a href="#cb189-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "8. Metadata Analysis"</span></span>
<span id="cb189-3"><a href="#cb189-3" aria-hidden="true" tabindex="-1"></a><span class="an">description:</span><span class="co"> |</span></span>
<span id="cb189-4"><a href="#cb189-4" aria-hidden="true" tabindex="-1"></a><span class="co">  Workflow to assess the relationship between environmental parameters and microbial communities. </span></span>
<span id="cb189-5"><a href="#cb189-5" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb189-6"><a href="#cb189-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-7"><a href="#cb189-7" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb189-8"><a href="#cb189-8" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;&lt;strong&gt;</span>Click here<span class="kw">&lt;/strong&gt;</span> for setup information.<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb189-9"><a href="#cb189-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-12"><a href="#cb189-12" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-13"><a href="#cb189-13" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb189-14"><a href="#cb189-14" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb189-15"><a href="#cb189-15" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">eval =</span> <span class="cn">FALSE</span>, <span class="at">echo =</span> <span class="cn">TRUE</span>)</span>
<span id="cb189-16"><a href="#cb189-16" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">119</span>)</span>
<span id="cb189-17"><a href="#cb189-17" aria-hidden="true" tabindex="-1"></a><span class="co">#library(conflicted)</span></span>
<span id="cb189-18"><a href="#cb189-18" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends(vegan, local = TRUE)  </span></span>
<span id="cb189-19"><a href="#cb189-19" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends_reverse(vegan, local = TRUE)  </span></span>
<span id="cb189-20"><a href="#cb189-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq); <span class="fu">packageVersion</span>(<span class="st">"phyloseq"</span>)</span>
<span id="cb189-21"><a href="#cb189-21" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(ape, bestNormalize, tidyverse, gdata, mctoolsr, </span>
<span id="cb189-22"><a href="#cb189-22" aria-hidden="true" tabindex="-1"></a>               ggpubr, ggvegan, ggfortify, vegan, reactable, </span>
<span id="cb189-23"><a href="#cb189-23" aria-hidden="true" tabindex="-1"></a>               downloadthis, captioner,</span>
<span id="cb189-24"><a href="#cb189-24" aria-hidden="true" tabindex="-1"></a>               <span class="at">install =</span> <span class="cn">FALSE</span>, <span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb189-25"><a href="#cb189-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-26"><a href="#cb189-26" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">scipen=</span><span class="dv">999</span>)</span>
<span id="cb189-27"><a href="#cb189-27" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_current<span class="sc">$</span><span class="fu">get</span>(<span class="fu">c</span>(</span>
<span id="cb189-28"><a href="#cb189-28" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cache"</span>,</span>
<span id="cb189-29"><a href="#cb189-29" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cache.path"</span>,</span>
<span id="cb189-30"><a href="#cb189-30" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cache.rebuild"</span>,</span>
<span id="cb189-31"><a href="#cb189-31" aria-hidden="true" tabindex="-1"></a>  <span class="st">"dependson"</span>,</span>
<span id="cb189-32"><a href="#cb189-32" aria-hidden="true" tabindex="-1"></a>  <span class="st">"autodep"</span></span>
<span id="cb189-33"><a href="#cb189-33" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb189-34"><a href="#cb189-34" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-35"><a href="#cb189-35" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb189-36"><a href="#cb189-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-39"><a href="#cb189-39" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-40"><a href="#cb189-40" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb189-41"><a href="#cb189-41" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-42"><a href="#cb189-42" aria-hidden="true" tabindex="-1"></a><span class="do">## Load for summary #3</span></span>
<span id="cb189-43"><a href="#cb189-43" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb189-44"><a href="#cb189-44" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"page_build/metadata_summary.rdata"</span>)</span>
<span id="cb189-45"><a href="#cb189-45" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-46"><a href="#cb189-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-49"><a href="#cb189-49" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-50"><a href="#cb189-50" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-51"><a href="#cb189-51" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-52"><a href="#cb189-52" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb189-53"><a href="#cb189-53" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the caption(s) with captioner</span></span>
<span id="cb189-54"><a href="#cb189-54" aria-hidden="true" tabindex="-1"></a>caption_tab_ssu <span class="ot">&lt;-</span> <span class="fu">captioner</span>(<span class="at">prefix =</span> <span class="st">"(16S rRNA) Table"</span>, <span class="at">suffix =</span> <span class="st">" |"</span>, <span class="at">style =</span> <span class="st">"b"</span>)</span>
<span id="cb189-55"><a href="#cb189-55" aria-hidden="true" tabindex="-1"></a>caption_fig_ssu <span class="ot">&lt;-</span> <span class="fu">captioner</span>(<span class="at">prefix =</span> <span class="st">"(16S rRNA) Figure"</span>, <span class="at">suffix =</span> <span class="st">" |"</span>, <span class="at">style =</span> <span class="st">"b"</span>)</span>
<span id="cb189-56"><a href="#cb189-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-57"><a href="#cb189-57" aria-hidden="true" tabindex="-1"></a>caption_tab_its <span class="ot">&lt;-</span> <span class="fu">captioner</span>(<span class="at">prefix =</span> <span class="st">"(ITS) Table"</span>, <span class="at">suffix =</span> <span class="st">" |"</span>, <span class="at">style =</span> <span class="st">"b"</span>)</span>
<span id="cb189-58"><a href="#cb189-58" aria-hidden="true" tabindex="-1"></a>caption_fig_its <span class="ot">&lt;-</span> <span class="fu">captioner</span>(<span class="at">prefix =</span> <span class="st">"(ITS) Figure"</span>, <span class="at">suffix =</span> <span class="st">" |"</span>, <span class="at">style =</span> <span class="st">"b"</span>)</span>
<span id="cb189-59"><a href="#cb189-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-60"><a href="#cb189-60" aria-hidden="true" tabindex="-1"></a>caption_tab <span class="ot">&lt;-</span> <span class="fu">captioner</span>(<span class="at">prefix =</span> <span class="st">"(common) Table"</span>, <span class="at">suffix =</span> <span class="st">" |"</span>, <span class="at">style =</span> <span class="st">"b"</span>)</span>
<span id="cb189-61"><a href="#cb189-61" aria-hidden="true" tabindex="-1"></a>caption_fig <span class="ot">&lt;-</span> <span class="fu">captioner</span>(<span class="at">prefix =</span> <span class="st">"(common) Figure"</span>, <span class="at">suffix =</span> <span class="st">" |"</span>, <span class="at">style =</span> <span class="st">"b"</span>)</span>
<span id="cb189-62"><a href="#cb189-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-63"><a href="#cb189-63" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a function for referring to the tables in text</span></span>
<span id="cb189-64"><a href="#cb189-64" aria-hidden="true" tabindex="-1"></a>ref <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">str_extract</span>(x, <span class="st">"[^|]*"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb189-65"><a href="#cb189-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">trimws</span>(<span class="at">which =</span> <span class="st">"right"</span>, <span class="at">whitespace =</span> <span class="st">"[ ]"</span>)</span>
<span id="cb189-66"><a href="#cb189-66" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-67"><a href="#cb189-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-70"><a href="#cb189-70" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-71"><a href="#cb189-71" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-72"><a href="#cb189-72" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-73"><a href="#cb189-73" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"assets/captions/captions_metadata.R"</span>)</span>
<span id="cb189-74"><a href="#cb189-74" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-75"><a href="#cb189-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-76"><a href="#cb189-76" aria-hidden="true" tabindex="-1"></a><span class="fu"># Synopsis</span></span>
<span id="cb189-77"><a href="#cb189-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-78"><a href="#cb189-78" aria-hidden="true" tabindex="-1"></a>In this workflow we compare the environmental metadata with the microbial community data. Part 1 involves the <span class="co">[</span><span class="ot">16S rRNA</span><span class="co">](#s-rrna-1)</span> community data and Part 2 deals with the <span class="co">[</span><span class="ot">ITS</span><span class="co">](#its-1)</span> community data. Each workflow contains the same major steps:</span>
<span id="cb189-79"><a href="#cb189-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-80"><a href="#cb189-80" aria-hidden="true" tabindex="-1"></a>1) Metadata Normality Tests: Shapiro-Wilk Normality Test to test whether each matadata parameter is normally distributed.</span>
<span id="cb189-81"><a href="#cb189-81" aria-hidden="true" tabindex="-1"></a>2) Normalize Parameters: R package <span class="co">[</span><span class="ot">`bestNormalize`</span><span class="co">](https://github.com/petersonR/bestNormalize)</span> to find and execute the best normalizing transformation. </span>
<span id="cb189-82"><a href="#cb189-82" aria-hidden="true" tabindex="-1"></a>3) Split Metadata parameters into groups: a) Environmental and edaphic properties, b) Microbial functional responses, and c) Temperature adaptation properties.</span>
<span id="cb189-83"><a href="#cb189-83" aria-hidden="true" tabindex="-1"></a>4) Autocorrelation Tests: Test all possible pair-wise comparisons, on both normalized and non-normalized data sets, for each group.</span>
<span id="cb189-84"><a href="#cb189-84" aria-hidden="true" tabindex="-1"></a>5) Remove autocorrelated parameters from each group.</span>
<span id="cb189-85"><a href="#cb189-85" aria-hidden="true" tabindex="-1"></a>6) Dissimilarity Correlation Tests: Use Mantel Tests to see if any on the metadata groups are significantly correlated with the community data.</span>
<span id="cb189-86"><a href="#cb189-86" aria-hidden="true" tabindex="-1"></a>7) Best Subset of Variables: Determine which of the metadata parameters from each group are the most strongly correlated with the community data. For this we use the <span class="in">`bioenv`</span> function from the <span class="in">`vegan`</span> <span class="co">[</span><span class="ot">@oksanen2013community</span><span class="co">]</span> package.</span>
<span id="cb189-87"><a href="#cb189-87" aria-hidden="true" tabindex="-1"></a>8) Distance-based Redundancy Analysis: Ordination analysis of samples and metadata vector overlays using <span class="in">`capscale`</span>, also from the <span class="in">`vegan`</span> package.  </span>
<span id="cb189-88"><a href="#cb189-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-89"><a href="#cb189-89" aria-hidden="true" tabindex="-1"></a><span class="fu">## Workflow Input</span></span>
<span id="cb189-90"><a href="#cb189-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-91"><a href="#cb189-91" aria-hidden="true" tabindex="-1"></a>Files needed to run this workflow can be downloaded from figshare. You will also need the metadata table, which can be accessed below. </span>
<span id="cb189-92"><a href="#cb189-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-93"><a href="#cb189-93" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;iframe</span> <span class="er">src</span><span class="ot">=</span><span class="st">"https://widgets.figshare.com/articles/16826779/embed?show_title=1"</span> <span class="er">width</span><span class="ot">=</span><span class="st">"100%"</span> <span class="er">height</span><span class="ot">=</span><span class="st">"251"</span> <span class="er">allowfullscreen</span> <span class="er">frameborder</span><span class="ot">=</span><span class="st">"0"</span><span class="kw">&gt;&lt;/iframe&gt;</span></span>
<span id="cb189-94"><a href="#cb189-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-95"><a href="#cb189-95" aria-hidden="true" tabindex="-1"></a><span class="fu">## Metadata  </span></span>
<span id="cb189-96"><a href="#cb189-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-97"><a href="#cb189-97" aria-hidden="true" tabindex="-1"></a>In this section we provide access to the complete environmental metadata containing 61 parameters across all 15 samples. </span>
<span id="cb189-98"><a href="#cb189-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-99"><a href="#cb189-99" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br/&gt;</span></span>
<span id="cb189-100"><a href="#cb189-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-103"><a href="#cb189-103" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-104"><a href="#cb189-104" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-105"><a href="#cb189-105" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-106"><a href="#cb189-106" aria-hidden="true" tabindex="-1"></a>metad <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"files/metadata/tables/metadata.txt"</span>,</span>
<span id="cb189-107"><a href="#cb189-107" aria-hidden="true" tabindex="-1"></a>                        <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb189-108"><a href="#cb189-108" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-109"><a href="#cb189-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-112"><a href="#cb189-112" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-113"><a href="#cb189-113" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-114"><a href="#cb189-114" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-115"><a href="#cb189-115" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> metad</span>
<span id="cb189-116"><a href="#cb189-116" aria-hidden="true" tabindex="-1"></a>seq_table <span class="sc">%&gt;%</span></span>
<span id="cb189-117"><a href="#cb189-117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb189-118"><a href="#cb189-118" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"metadata"</span>,</span>
<span id="cb189-119"><a href="#cb189-119" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb189-120"><a href="#cb189-120" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb189-121"><a href="#cb189-121" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb189-122"><a href="#cb189-122" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb189-123"><a href="#cb189-123" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-124"><a href="#cb189-124" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb189-125"><a href="#cb189-125" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-126"><a href="#cb189-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-127"><a href="#cb189-127" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab("metadata_results")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-128"><a href="#cb189-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-131"><a href="#cb189-131" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-132"><a href="#cb189-132" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-133"><a href="#cb189-133" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-134"><a href="#cb189-134" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb189-135"><a href="#cb189-135" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(<span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>),</span>
<span id="cb189-136"><a href="#cb189-136" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb189-137"><a href="#cb189-137" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">1</span>),</span>
<span id="cb189-138"><a href="#cb189-138" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">TRUE</span>, <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-139"><a href="#cb189-139" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb189-140"><a href="#cb189-140" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb189-141"><a href="#cb189-141" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb189-142"><a href="#cb189-142" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Sample ID"</span>, </span>
<span id="cb189-143"><a href="#cb189-143" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb189-144"><a href="#cb189-144" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>),</span>
<span id="cb189-145"><a href="#cb189-145" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>, <span class="at">fontSize =</span> <span class="st">"0.9em"</span>), </span>
<span id="cb189-146"><a href="#cb189-146" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb189-147"><a href="#cb189-147" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">150</span>), </span>
<span id="cb189-148"><a href="#cb189-148" aria-hidden="true" tabindex="-1"></a>    <span class="at">raw_reads =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values)))</span>
<span id="cb189-149"><a href="#cb189-149" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb189-150"><a href="#cb189-150" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">5</span>, </span>
<span id="cb189-151"><a href="#cb189-151" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb189-152"><a href="#cb189-152" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb189-153"><a href="#cb189-153" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb189-154"><a href="#cb189-154" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-155"><a href="#cb189-155" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb189-156"><a href="#cb189-156" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(seq_table)</span>
<span id="cb189-157"><a href="#cb189-157" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-158"><a href="#cb189-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-159"><a href="#cb189-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-160"><a href="#cb189-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-161"><a href="#cb189-161" aria-hidden="true" tabindex="-1"></a><span class="fu"># Multivariate Summary</span></span>
<span id="cb189-162"><a href="#cb189-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-163"><a href="#cb189-163" aria-hidden="true" tabindex="-1"></a>Summary results for of <span class="in">`envfit`</span> and <span class="in">`bioenv`</span> tests for edaphic properties, soil functional response, and temperature adaptation for both 16S rRNA and ITS data sets.</span>
<span id="cb189-164"><a href="#cb189-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-165"><a href="#cb189-165" aria-hidden="true" tabindex="-1"></a><span class="fu">## 16S rRNA</span></span>
<span id="cb189-166"><a href="#cb189-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-167"><a href="#cb189-167" aria-hidden="true" tabindex="-1"></a>Metadata parameters removed based on autocorrelation results.</span>
<span id="cb189-168"><a href="#cb189-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-169"><a href="#cb189-169" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>edaphic properties: TEB + DON + Na + Al + Ca</span>
<span id="cb189-170"><a href="#cb189-170" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>soil functional response: micN + micNP + enzCN + enzCP + BP~ase~ + CE~ase~ + LP~ase~ + N~ase~ + P~ase~</span>
<span id="cb189-171"><a href="#cb189-171" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>temperature adaptation: NUE + PUE + SI</span>
<span id="cb189-172"><a href="#cb189-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-173"><a href="#cb189-173" aria-hidden="true" tabindex="-1"></a>Metadata parameters removed from capscale analysis based on Degrees of Freedom.</span>
<span id="cb189-174"><a href="#cb189-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-175"><a href="#cb189-175" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>edaphic properties: Mg + Mn</span>
<span id="cb189-176"><a href="#cb189-176" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>soil functional response: NONE</span>
<span id="cb189-177"><a href="#cb189-177" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>temperature adaptation: NONE</span>
<span id="cb189-178"><a href="#cb189-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-179"><a href="#cb189-179" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br/&gt;</span></span>
<span id="cb189-180"><a href="#cb189-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-183"><a href="#cb189-183" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-184"><a href="#cb189-184" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-185"><a href="#cb189-185" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-186"><a href="#cb189-186" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"files/metadata/tables/ssu18_sig_envfit_bioenv_summary.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb189-187"><a href="#cb189-187" aria-hidden="true" tabindex="-1"></a>seq_table <span class="sc">%&gt;%</span></span>
<span id="cb189-188"><a href="#cb189-188" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb189-189"><a href="#cb189-189" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"ssu_multivariate_summary"</span>,</span>
<span id="cb189-190"><a href="#cb189-190" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb189-191"><a href="#cb189-191" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb189-192"><a href="#cb189-192" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb189-193"><a href="#cb189-193" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb189-194"><a href="#cb189-194" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-195"><a href="#cb189-195" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb189-196"><a href="#cb189-196" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-197"><a href="#cb189-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-200"><a href="#cb189-200" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-201"><a href="#cb189-201" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-202"><a href="#cb189-202" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-203"><a href="#cb189-203" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> ssu18_sig_summary <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"envfit_r2"</span> <span class="ot">=</span> <span class="dv">3</span>, </span>
<span id="cb189-204"><a href="#cb189-204" aria-hidden="true" tabindex="-1"></a>                                                         <span class="st">"envfit_pval"</span> <span class="ot">=</span> <span class="dv">4</span>, </span>
<span id="cb189-205"><a href="#cb189-205" aria-hidden="true" tabindex="-1"></a>                                                         <span class="st">"bioenv_r"</span> <span class="ot">=</span> <span class="dv">5</span>, </span>
<span id="cb189-206"><a href="#cb189-206" aria-hidden="true" tabindex="-1"></a>                                                         <span class="st">"bioenv_pval"</span> <span class="ot">=</span> <span class="dv">6</span>)</span>
<span id="cb189-207"><a href="#cb189-207" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> seq_table <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="fu">across</span>(<span class="at">.cols =</span> <span class="fu">c</span>(<span class="dv">3</span><span class="sc">:</span><span class="dv">6</span>), round, <span class="at">digits =</span> <span class="dv">3</span>)) </span>
<span id="cb189-208"><a href="#cb189-208" aria-hidden="true" tabindex="-1"></a>seq_table[<span class="fu">is.na</span>(seq_table)] <span class="ot">&lt;-</span> <span class="st">""</span></span>
<span id="cb189-209"><a href="#cb189-209" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-210"><a href="#cb189-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-211"><a href="#cb189-211" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab_ssu("ssu_multi_summ")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-212"><a href="#cb189-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-215"><a href="#cb189-215" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-216"><a href="#cb189-216" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-217"><a href="#cb189-217" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-218"><a href="#cb189-218" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table, <span class="at">groupBy =</span> <span class="st">"metadata_set"</span>, <span class="at">defaultExpanded =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-219"><a href="#cb189-219" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb189-220"><a href="#cb189-220" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb189-221"><a href="#cb189-221" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">3</span>),</span>
<span id="cb189-222"><a href="#cb189-222" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">FALSE</span>, <span class="at">sortable =</span> <span class="cn">FALSE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-223"><a href="#cb189-223" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>), <span class="at">minWidth =</span> <span class="dv">50</span>), </span>
<span id="cb189-224"><a href="#cb189-224" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb189-225"><a href="#cb189-225" aria-hidden="true" tabindex="-1"></a>    <span class="at">metadata_set =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Metadata group"</span>, </span>
<span id="cb189-226"><a href="#cb189-226" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb189-227"><a href="#cb189-227" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>),</span>
<span id="cb189-228"><a href="#cb189-228" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>), </span>
<span id="cb189-229"><a href="#cb189-229" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb189-230"><a href="#cb189-230" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">90</span>),</span>
<span id="cb189-231"><a href="#cb189-231" aria-hidden="true" tabindex="-1"></a>    <span class="at">parameter =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"parameter"</span>, </span>
<span id="cb189-232"><a href="#cb189-232" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>),</span>
<span id="cb189-233"><a href="#cb189-233" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>)),</span>
<span id="cb189-234"><a href="#cb189-234" aria-hidden="true" tabindex="-1"></a>    <span class="at">envfit_r2 =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"r2"</span>),</span>
<span id="cb189-235"><a href="#cb189-235" aria-hidden="true" tabindex="-1"></a>    <span class="at">envfit_pval =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"P(&gt;r)"</span>, </span>
<span id="cb189-236"><a href="#cb189-236" aria-hidden="true" tabindex="-1"></a>                                <span class="at">style =</span> <span class="cf">function</span>(value) {</span>
<span id="cb189-237"><a href="#cb189-237" aria-hidden="true" tabindex="-1"></a>                                         <span class="cf">if</span> (value <span class="sc">&gt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb189-238"><a href="#cb189-238" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#2271B2"</span> <span class="co">#green</span></span>
<span id="cb189-239"><a href="#cb189-239" aria-hidden="true" tabindex="-1"></a>                                         } <span class="cf">else</span> <span class="cf">if</span> (value <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb189-240"><a href="#cb189-240" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#B22271"</span> <span class="co">#RED "#", "#"</span></span>
<span id="cb189-241"><a href="#cb189-241" aria-hidden="true" tabindex="-1"></a>                                         } <span class="cf">else</span> {</span>
<span id="cb189-242"><a href="#cb189-242" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#777"</span></span>
<span id="cb189-243"><a href="#cb189-243" aria-hidden="true" tabindex="-1"></a>                                         }</span>
<span id="cb189-244"><a href="#cb189-244" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">list</span>(<span class="at">color =</span> color, <span class="at">fontWeight =</span> <span class="st">"bold"</span>, </span>
<span id="cb189-245"><a href="#cb189-245" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>)}, </span>
<span id="cb189-246"><a href="#cb189-246" aria-hidden="true" tabindex="-1"></a>                         <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>)),</span>
<span id="cb189-247"><a href="#cb189-247" aria-hidden="true" tabindex="-1"></a>    <span class="at">bioenv_r =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"r"</span>, <span class="at">minWidth =</span> <span class="dv">40</span>),</span>
<span id="cb189-248"><a href="#cb189-248" aria-hidden="true" tabindex="-1"></a>    <span class="at">bioenv_pval =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"p-value"</span>, </span>
<span id="cb189-249"><a href="#cb189-249" aria-hidden="true" tabindex="-1"></a>                                <span class="at">style =</span> <span class="cf">function</span>(value) {</span>
<span id="cb189-250"><a href="#cb189-250" aria-hidden="true" tabindex="-1"></a>                                         <span class="cf">if</span> (value <span class="sc">&gt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb189-251"><a href="#cb189-251" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#2271B2"</span> <span class="co">#green</span></span>
<span id="cb189-252"><a href="#cb189-252" aria-hidden="true" tabindex="-1"></a>                                         } <span class="cf">else</span> <span class="cf">if</span> (value <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb189-253"><a href="#cb189-253" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#B22271"</span> <span class="co">#RED "#", "#"</span></span>
<span id="cb189-254"><a href="#cb189-254" aria-hidden="true" tabindex="-1"></a>                                         } <span class="cf">else</span> {</span>
<span id="cb189-255"><a href="#cb189-255" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#777"</span></span>
<span id="cb189-256"><a href="#cb189-256" aria-hidden="true" tabindex="-1"></a>                                         }</span>
<span id="cb189-257"><a href="#cb189-257" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">list</span>(<span class="at">color =</span> color, <span class="at">fontWeight =</span> <span class="st">"bold"</span>)})), </span>
<span id="cb189-258"><a href="#cb189-258" aria-hidden="true" tabindex="-1"></a>    <span class="at">columnGroups =</span> <span class="fu">list</span>(</span>
<span id="cb189-259"><a href="#cb189-259" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colGroup</span>(<span class="at">name =</span> <span class="st">"envfit"</span>, </span>
<span id="cb189-260"><a href="#cb189-260" aria-hidden="true" tabindex="-1"></a>             <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"envfit_r2"</span>, <span class="st">"envfit_pval"</span>), <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"1.2em"</span>)),</span>
<span id="cb189-261"><a href="#cb189-261" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colGroup</span>(<span class="at">name =</span> <span class="st">"bioenv"</span>, </span>
<span id="cb189-262"><a href="#cb189-262" aria-hidden="true" tabindex="-1"></a>             <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"bioenv_r"</span>, <span class="st">"bioenv_pval"</span>), <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"1.2em"</span>))),</span>
<span id="cb189-263"><a href="#cb189-263" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">FALSE</span>, <span class="at">defaultPageSize =</span> <span class="dv">15</span>, <span class="at">showPagination =</span> <span class="cn">FALSE</span>,</span>
<span id="cb189-264"><a href="#cb189-264" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb189-265"><a href="#cb189-265" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb189-266"><a href="#cb189-266" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">TRUE</span>, </span>
<span id="cb189-267"><a href="#cb189-267" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">FALSE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-268"><a href="#cb189-268" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb189-269"><a href="#cb189-269" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(seq_table)</span>
<span id="cb189-270"><a href="#cb189-270" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-271"><a href="#cb189-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-272"><a href="#cb189-272" aria-hidden="true" tabindex="-1"></a><span class="fu">## ITS</span></span>
<span id="cb189-273"><a href="#cb189-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-274"><a href="#cb189-274" aria-hidden="true" tabindex="-1"></a>Metadata parameters removed based on autocorrelation results.</span>
<span id="cb189-275"><a href="#cb189-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-276"><a href="#cb189-276" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>edaphic properties: TEB + DON + Na + Al + Ca</span>
<span id="cb189-277"><a href="#cb189-277" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>soil functional response: micN + micNP + enzCN + enzCP + BP~ase~ + CE~ase~ + LP~ase~ + N~ase~ + P~ase~</span>
<span id="cb189-278"><a href="#cb189-278" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>temperature adaptation: NUE + PUE + P~Q10~ + SI</span>
<span id="cb189-279"><a href="#cb189-279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-280"><a href="#cb189-280" aria-hidden="true" tabindex="-1"></a>Metadata parameters removed from capscale analysis based on Degrees of Freedom.</span>
<span id="cb189-281"><a href="#cb189-281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-282"><a href="#cb189-282" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>edaphic properties: Mg + Mn + Na + Al + Fe + K</span>
<span id="cb189-283"><a href="#cb189-283" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>soil functional response: NONE</span>
<span id="cb189-284"><a href="#cb189-284" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>temperature adaptation: S~Q10~</span>
<span id="cb189-285"><a href="#cb189-285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-286"><a href="#cb189-286" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br/&gt;</span></span>
<span id="cb189-287"><a href="#cb189-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-290"><a href="#cb189-290" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-291"><a href="#cb189-291" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-292"><a href="#cb189-292" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-293"><a href="#cb189-293" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"files/metadata/tables/its18_sig_envfit_bioenv_summary.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb189-294"><a href="#cb189-294" aria-hidden="true" tabindex="-1"></a>seq_table <span class="sc">%&gt;%</span></span>
<span id="cb189-295"><a href="#cb189-295" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb189-296"><a href="#cb189-296" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"its_multivariate_summary"</span>,</span>
<span id="cb189-297"><a href="#cb189-297" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb189-298"><a href="#cb189-298" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb189-299"><a href="#cb189-299" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb189-300"><a href="#cb189-300" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb189-301"><a href="#cb189-301" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-302"><a href="#cb189-302" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb189-303"><a href="#cb189-303" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-304"><a href="#cb189-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-307"><a href="#cb189-307" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-308"><a href="#cb189-308" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-309"><a href="#cb189-309" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-310"><a href="#cb189-310" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> its18_sig_summary <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"envfit_r2"</span> <span class="ot">=</span> <span class="dv">3</span>, </span>
<span id="cb189-311"><a href="#cb189-311" aria-hidden="true" tabindex="-1"></a>                                                         <span class="st">"envfit_pval"</span> <span class="ot">=</span> <span class="dv">4</span>, </span>
<span id="cb189-312"><a href="#cb189-312" aria-hidden="true" tabindex="-1"></a>                                                         <span class="st">"bioenv_r"</span> <span class="ot">=</span> <span class="dv">5</span>, </span>
<span id="cb189-313"><a href="#cb189-313" aria-hidden="true" tabindex="-1"></a>                                                         <span class="st">"bioenv_pval"</span> <span class="ot">=</span> <span class="dv">6</span>)</span>
<span id="cb189-314"><a href="#cb189-314" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> seq_table <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="fu">across</span>(<span class="at">.cols =</span> <span class="fu">c</span>(<span class="dv">3</span><span class="sc">:</span><span class="dv">6</span>), round, <span class="at">digits =</span> <span class="dv">3</span>)) </span>
<span id="cb189-315"><a href="#cb189-315" aria-hidden="true" tabindex="-1"></a>seq_table[<span class="fu">is.na</span>(seq_table)] <span class="ot">&lt;-</span> <span class="st">""</span></span>
<span id="cb189-316"><a href="#cb189-316" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-317"><a href="#cb189-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-318"><a href="#cb189-318" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab_its("its_multi_summ")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-319"><a href="#cb189-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-322"><a href="#cb189-322" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-323"><a href="#cb189-323" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-324"><a href="#cb189-324" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-325"><a href="#cb189-325" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table, <span class="at">groupBy =</span> <span class="st">"metadata_set"</span>, <span class="at">defaultExpanded =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-326"><a href="#cb189-326" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb189-327"><a href="#cb189-327" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb189-328"><a href="#cb189-328" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">3</span>),</span>
<span id="cb189-329"><a href="#cb189-329" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">FALSE</span>, <span class="at">sortable =</span> <span class="cn">FALSE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-330"><a href="#cb189-330" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>), <span class="at">minWidth =</span> <span class="dv">50</span>), </span>
<span id="cb189-331"><a href="#cb189-331" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb189-332"><a href="#cb189-332" aria-hidden="true" tabindex="-1"></a>    <span class="at">metadata_set =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Metadata group"</span>, </span>
<span id="cb189-333"><a href="#cb189-333" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb189-334"><a href="#cb189-334" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>),</span>
<span id="cb189-335"><a href="#cb189-335" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>), </span>
<span id="cb189-336"><a href="#cb189-336" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb189-337"><a href="#cb189-337" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">90</span>),</span>
<span id="cb189-338"><a href="#cb189-338" aria-hidden="true" tabindex="-1"></a>    <span class="at">parameter =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"parameter"</span>, </span>
<span id="cb189-339"><a href="#cb189-339" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>),</span>
<span id="cb189-340"><a href="#cb189-340" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>)),</span>
<span id="cb189-341"><a href="#cb189-341" aria-hidden="true" tabindex="-1"></a>    <span class="at">envfit_r2 =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"r2"</span>),</span>
<span id="cb189-342"><a href="#cb189-342" aria-hidden="true" tabindex="-1"></a>    <span class="at">envfit_pval =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"P(&gt;r)"</span>, </span>
<span id="cb189-343"><a href="#cb189-343" aria-hidden="true" tabindex="-1"></a>                                <span class="at">style =</span> <span class="cf">function</span>(value) {</span>
<span id="cb189-344"><a href="#cb189-344" aria-hidden="true" tabindex="-1"></a>                                         <span class="cf">if</span> (value <span class="sc">&gt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb189-345"><a href="#cb189-345" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#2271B2"</span> <span class="co">#green</span></span>
<span id="cb189-346"><a href="#cb189-346" aria-hidden="true" tabindex="-1"></a>                                         } <span class="cf">else</span> <span class="cf">if</span> (value <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb189-347"><a href="#cb189-347" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#B22271"</span> <span class="co">#RED "#", "#"</span></span>
<span id="cb189-348"><a href="#cb189-348" aria-hidden="true" tabindex="-1"></a>                                         } <span class="cf">else</span> {</span>
<span id="cb189-349"><a href="#cb189-349" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#777"</span></span>
<span id="cb189-350"><a href="#cb189-350" aria-hidden="true" tabindex="-1"></a>                                         }</span>
<span id="cb189-351"><a href="#cb189-351" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">list</span>(<span class="at">color =</span> color, <span class="at">fontWeight =</span> <span class="st">"bold"</span>, </span>
<span id="cb189-352"><a href="#cb189-352" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>)}, </span>
<span id="cb189-353"><a href="#cb189-353" aria-hidden="true" tabindex="-1"></a>                         <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>)),</span>
<span id="cb189-354"><a href="#cb189-354" aria-hidden="true" tabindex="-1"></a>    <span class="at">bioenv_r =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"r"</span>, <span class="at">minWidth =</span> <span class="dv">40</span>),</span>
<span id="cb189-355"><a href="#cb189-355" aria-hidden="true" tabindex="-1"></a>    <span class="at">bioenv_pval =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"p-value"</span>, </span>
<span id="cb189-356"><a href="#cb189-356" aria-hidden="true" tabindex="-1"></a>                                <span class="at">style =</span> <span class="cf">function</span>(value) {</span>
<span id="cb189-357"><a href="#cb189-357" aria-hidden="true" tabindex="-1"></a>                                         <span class="cf">if</span> (value <span class="sc">&gt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb189-358"><a href="#cb189-358" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#2271B2"</span> <span class="co">#green</span></span>
<span id="cb189-359"><a href="#cb189-359" aria-hidden="true" tabindex="-1"></a>                                         } <span class="cf">else</span> <span class="cf">if</span> (value <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb189-360"><a href="#cb189-360" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#B22271"</span> <span class="co">#RED "#", "#"</span></span>
<span id="cb189-361"><a href="#cb189-361" aria-hidden="true" tabindex="-1"></a>                                         } <span class="cf">else</span> {</span>
<span id="cb189-362"><a href="#cb189-362" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#777"</span></span>
<span id="cb189-363"><a href="#cb189-363" aria-hidden="true" tabindex="-1"></a>                                         }</span>
<span id="cb189-364"><a href="#cb189-364" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">list</span>(<span class="at">color =</span> color, <span class="at">fontWeight =</span> <span class="st">"bold"</span>)})), </span>
<span id="cb189-365"><a href="#cb189-365" aria-hidden="true" tabindex="-1"></a>    <span class="at">columnGroups =</span> <span class="fu">list</span>(</span>
<span id="cb189-366"><a href="#cb189-366" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colGroup</span>(<span class="at">name =</span> <span class="st">"envfit"</span>, </span>
<span id="cb189-367"><a href="#cb189-367" aria-hidden="true" tabindex="-1"></a>             <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"envfit_r2"</span>, <span class="st">"envfit_pval"</span>), <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"1.2em"</span>)),</span>
<span id="cb189-368"><a href="#cb189-368" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colGroup</span>(<span class="at">name =</span> <span class="st">"bioenv"</span>, </span>
<span id="cb189-369"><a href="#cb189-369" aria-hidden="true" tabindex="-1"></a>             <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"bioenv_r"</span>, <span class="st">"bioenv_pval"</span>), <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"1.2em"</span>))),</span>
<span id="cb189-370"><a href="#cb189-370" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">FALSE</span>, <span class="at">defaultPageSize =</span> <span class="dv">15</span>, <span class="at">showPagination =</span> <span class="cn">FALSE</span>,</span>
<span id="cb189-371"><a href="#cb189-371" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb189-372"><a href="#cb189-372" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb189-373"><a href="#cb189-373" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">TRUE</span>, </span>
<span id="cb189-374"><a href="#cb189-374" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">FALSE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-375"><a href="#cb189-375" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb189-376"><a href="#cb189-376" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(seq_table)</span>
<span id="cb189-377"><a href="#cb189-377" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-378"><a href="#cb189-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-381"><a href="#cb189-381" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-382"><a href="#cb189-382" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb189-383"><a href="#cb189-383" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-384"><a href="#cb189-384" aria-hidden="true" tabindex="-1"></a><span class="do">## Load to build page only #2</span></span>
<span id="cb189-385"><a href="#cb189-385" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"page_build/metadata_ssu18_wf.rdata"</span>)</span>
<span id="cb189-386"><a href="#cb189-386" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-387"><a href="#cb189-387" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-390"><a href="#cb189-390" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-391"><a href="#cb189-391" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-392"><a href="#cb189-392" aria-hidden="true" tabindex="-1"></a>swel_col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#2271B2"</span>, <span class="st">"#71B222"</span>, <span class="st">"#B22271"</span>)</span>
<span id="cb189-393"><a href="#cb189-393" aria-hidden="true" tabindex="-1"></a>tmp_1 <span class="ot">&lt;-</span> ssu18_ps_work_mc<span class="sc">$</span>data_loaded</span>
<span id="cb189-394"><a href="#cb189-394" aria-hidden="true" tabindex="-1"></a>tmp_2 <span class="ot">&lt;-</span> ssu18_ps_pime_mc<span class="sc">$</span>data_loaded</span>
<span id="cb189-395"><a href="#cb189-395" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-396"><a href="#cb189-396" aria-hidden="true" tabindex="-1"></a>tmp_1 <span class="ot">&lt;-</span> tmp_1 <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"ID"</span>)</span>
<span id="cb189-397"><a href="#cb189-397" aria-hidden="true" tabindex="-1"></a>tmp_2 <span class="ot">&lt;-</span> tmp_2 <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"ID"</span>)</span>
<span id="cb189-398"><a href="#cb189-398" aria-hidden="true" tabindex="-1"></a>tmp_3 <span class="ot">&lt;-</span> <span class="fu">left_join</span>(tmp_2, tmp_1, <span class="at">by =</span> <span class="st">"ID"</span>)</span>
<span id="cb189-399"><a href="#cb189-399" aria-hidden="true" tabindex="-1"></a>tmp_3 <span class="ot">&lt;-</span> tmp_3[ , <span class="fu">order</span>(<span class="fu">names</span>(tmp_3))]</span>
<span id="cb189-400"><a href="#cb189-400" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(tmp_3, <span class="st">"tmp_3.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb189-401"><a href="#cb189-401" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-402"><a href="#cb189-402" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-403"><a href="#cb189-403" aria-hidden="true" tabindex="-1"></a><span class="fu"># 16s rRNA</span></span>
<span id="cb189-404"><a href="#cb189-404" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-405"><a href="#cb189-405" aria-hidden="true" tabindex="-1"></a><span class="fu">## Data Formating </span></span>
<span id="cb189-406"><a href="#cb189-406" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-409"><a href="#cb189-409" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-410"><a href="#cb189-410" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb189-411"><a href="#cb189-411" aria-hidden="true" tabindex="-1"></a><span class="do">## Initial Load for  ANALYSIS #1</span></span>
<span id="cb189-412"><a href="#cb189-412" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">119</span>)</span>
<span id="cb189-413"><a href="#cb189-413" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"files/alpha/rdata/ssu18_ps_work.rds"</span>)</span>
<span id="cb189-414"><a href="#cb189-414" aria-hidden="true" tabindex="-1"></a>ssu18_ps_filt <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"files/alpha/rdata/ssu18_ps_filt.rds"</span>)</span>
<span id="cb189-415"><a href="#cb189-415" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"files/alpha/rdata/ssu18_ps_perfect.rds"</span>)</span>
<span id="cb189-416"><a href="#cb189-416" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"files/alpha/rdata/ssu18_ps_pime.rds"</span>)</span>
<span id="cb189-417"><a href="#cb189-417" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-418"><a href="#cb189-418" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-419"><a href="#cb189-419" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>Transform <span class="in">`ps`</span> Objects</span>
<span id="cb189-420"><a href="#cb189-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-421"><a href="#cb189-421" aria-hidden="true" tabindex="-1"></a>The first step is to transform the phyloseq objects. </span>
<span id="cb189-422"><a href="#cb189-422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-425"><a href="#cb189-425" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-426"><a href="#cb189-426" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-427"><a href="#cb189-427" aria-hidden="true" tabindex="-1"></a>samp_ps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ssu18_ps_work"</span>, <span class="st">"ssu18_ps_pime"</span>, <span class="st">"ssu18_ps_perfect"</span>, </span>
<span id="cb189-428"><a href="#cb189-428" aria-hidden="true" tabindex="-1"></a>             <span class="st">"ssu18_ps_work_otu"</span>, <span class="st">"ssu18_ps_pime_otu"</span>, <span class="st">"ssu18_ps_perfect_otu"</span>)</span>
<span id="cb189-429"><a href="#cb189-429" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> samp_ps) {</span>
<span id="cb189-430"><a href="#cb189-430" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb189-431"><a href="#cb189-431" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(tmp_get, <span class="cf">function</span>(otu) <span class="fl">1e5</span> <span class="sc">*</span> otu<span class="sc">/</span><span class="fu">sum</span>(otu))</span>
<span id="cb189-432"><a href="#cb189-432" aria-hidden="true" tabindex="-1"></a>     tmp_ps<span class="sc">@</span>phy_tree <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-433"><a href="#cb189-433" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">prune_samples</span>(<span class="fu">sample_sums</span>(tmp_ps) <span class="sc">&gt;</span> <span class="dv">0</span>, tmp_ps)</span>
<span id="cb189-434"><a href="#cb189-434" aria-hidden="true" tabindex="-1"></a>     tmp_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(tmp_ps), <span class="at">rooted =</span> <span class="cn">TRUE</span>, <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(tmp_ps))</span>
<span id="cb189-435"><a href="#cb189-435" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb189-436"><a href="#cb189-436" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> tmp_samp <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"TREAT_T"</span> <span class="ot">=</span> <span class="st">"TEMP"</span>)</span>
<span id="cb189-437"><a href="#cb189-437" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(tmp_ps, sample_data, tmp_tree)</span>
<span id="cb189-438"><a href="#cb189-438" aria-hidden="true" tabindex="-1"></a>     <span class="fu">sample_data</span>(tmp_ps) <span class="ot">&lt;-</span> tmp_samp</span>
<span id="cb189-439"><a href="#cb189-439" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_prop"</span>))</span>
<span id="cb189-440"><a href="#cb189-440" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_name)</span>
<span id="cb189-441"><a href="#cb189-441" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_ps)</span>
<span id="cb189-442"><a href="#cb189-442" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-443"><a href="#cb189-443" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-444"><a href="#cb189-444" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-445"><a href="#cb189-445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-446"><a href="#cb189-446" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>Format for <span class="in">`mctoolsr`</span></span>
<span id="cb189-447"><a href="#cb189-447" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-448"><a href="#cb189-448" aria-hidden="true" tabindex="-1"></a>Then a bit of formatting to make the data compatible with <span class="in">`mctoolsr`</span>. </span>
<span id="cb189-449"><a href="#cb189-449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-452"><a href="#cb189-452" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-453"><a href="#cb189-453" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-454"><a href="#cb189-454" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> samp_ps) {</span>
<span id="cb189-455"><a href="#cb189-455" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_prop"</span>)))</span>
<span id="cb189-456"><a href="#cb189-456" aria-hidden="true" tabindex="-1"></a>     tmp_tax <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(tmp_get))</span>
<span id="cb189-457"><a href="#cb189-457" aria-hidden="true" tabindex="-1"></a>     tmp_tax<span class="sc">$</span>ASV_SEQ <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-458"><a href="#cb189-458" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb189-459"><a href="#cb189-459" aria-hidden="true" tabindex="-1"></a>     tmp_col_names <span class="ot">&lt;-</span> <span class="fu">colnames</span>(tmp_tax)</span>
<span id="cb189-460"><a href="#cb189-460" aria-hidden="true" tabindex="-1"></a>     tmp_tax_merge <span class="ot">&lt;-</span> tmp_tax <span class="sc">%&gt;%</span> tidyr<span class="sc">::</span><span class="fu">unite</span>(taxonomy, </span>
<span id="cb189-461"><a href="#cb189-461" aria-hidden="true" tabindex="-1"></a>                                               <span class="fu">all_of</span>(tmp_col_names), </span>
<span id="cb189-462"><a href="#cb189-462" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">sep =</span> <span class="st">";"</span>)</span>
<span id="cb189-463"><a href="#cb189-463" aria-hidden="true" tabindex="-1"></a>     tmp_tax_merge <span class="ot">&lt;-</span> tmp_tax_merge <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"#OTU_ID"</span>)</span>
<span id="cb189-464"><a href="#cb189-464" aria-hidden="true" tabindex="-1"></a>     tmp_otu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(tmp_get)))</span>
<span id="cb189-465"><a href="#cb189-465" aria-hidden="true" tabindex="-1"></a>     tmp_otu <span class="ot">&lt;-</span> tmp_otu <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"#OTU_ID"</span>)</span>
<span id="cb189-466"><a href="#cb189-466" aria-hidden="true" tabindex="-1"></a>     tmp_otu_tax <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_otu, tmp_tax_merge, <span class="at">by =</span> <span class="st">"#OTU_ID"</span>)</span>
<span id="cb189-467"><a href="#cb189-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-468"><a href="#cb189-468" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb189-469"><a href="#cb189-469" aria-hidden="true" tabindex="-1"></a>     tmp_samp[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>)] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-470"><a href="#cb189-470" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> tmp_samp <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"#SampleID"</span>)</span>
<span id="cb189-471"><a href="#cb189-471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-472"><a href="#cb189-472" aria-hidden="true" tabindex="-1"></a>     tmp_metad <span class="ot">&lt;-</span> metad <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"#SampleID"</span> <span class="ot">=</span> <span class="st">"id"</span>)</span>
<span id="cb189-473"><a href="#cb189-473" aria-hidden="true" tabindex="-1"></a>     tmp_metad[,<span class="dv">2</span><span class="sc">:</span><span class="dv">5</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-474"><a href="#cb189-474" aria-hidden="true" tabindex="-1"></a>     tmp_md <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_samp, tmp_metad, <span class="at">by =</span> <span class="st">"#SampleID"</span>)</span>
<span id="cb189-475"><a href="#cb189-475" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb189-476"><a href="#cb189-476" aria-hidden="true" tabindex="-1"></a>     tmp_otu_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_otu_tax"</span>))</span>
<span id="cb189-477"><a href="#cb189-477" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_otu_name)</span>
<span id="cb189-478"><a href="#cb189-478" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_otu_name, tmp_otu_tax)</span>
<span id="cb189-479"><a href="#cb189-479" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-480"><a href="#cb189-480" aria-hidden="true" tabindex="-1"></a>     tmp_md_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_md"</span>))</span>
<span id="cb189-481"><a href="#cb189-481" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_md_name)</span>
<span id="cb189-482"><a href="#cb189-482" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_md_name, tmp_md)</span>
<span id="cb189-483"><a href="#cb189-483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-484"><a href="#cb189-484" aria-hidden="true" tabindex="-1"></a>     tmp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"files/metadata/tables/"</span>)</span>
<span id="cb189-485"><a href="#cb189-485" aria-hidden="true" tabindex="-1"></a>     <span class="fu">write_delim</span>(tmp_otu_tax, <span class="fu">paste</span>(tmp_path, tmp_otu_name, <span class="st">".txt"</span>, <span class="at">sep =</span> <span class="st">""</span>), <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb189-486"><a href="#cb189-486" aria-hidden="true" tabindex="-1"></a>     <span class="fu">write_delim</span>(tmp_md, <span class="fu">paste</span>(tmp_path, tmp_md_name, <span class="st">".txt"</span>, <span class="at">sep =</span> <span class="st">""</span>), <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb189-487"><a href="#cb189-487" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-488"><a href="#cb189-488" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-489"><a href="#cb189-489" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-490"><a href="#cb189-490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-493"><a href="#cb189-493" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-494"><a href="#cb189-494" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-495"><a href="#cb189-495" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> samp_ps) {</span>
<span id="cb189-496"><a href="#cb189-496" aria-hidden="true" tabindex="-1"></a>     tmp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"files/metadata/tables/"</span>)</span>
<span id="cb189-497"><a href="#cb189-497" aria-hidden="true" tabindex="-1"></a>     tmp_otu_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_otu_tax"</span>))</span>
<span id="cb189-498"><a href="#cb189-498" aria-hidden="true" tabindex="-1"></a>     tmp_md_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_md"</span>))</span>
<span id="cb189-499"><a href="#cb189-499" aria-hidden="true" tabindex="-1"></a>     tmp_tax_table_fp <span class="ot">&lt;-</span> <span class="fu">paste</span>(tmp_path, tmp_otu_name, <span class="st">".txt"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb189-500"><a href="#cb189-500" aria-hidden="true" tabindex="-1"></a>     tmp_map_fp <span class="ot">&lt;-</span> <span class="fu">paste</span>(tmp_path, tmp_md_name, <span class="st">".txt"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb189-501"><a href="#cb189-501" aria-hidden="true" tabindex="-1"></a>     tmp_input <span class="ot">&lt;-</span> <span class="fu">load_taxa_table</span>(tmp_tax_table_fp, tmp_map_fp)</span>
<span id="cb189-502"><a href="#cb189-502" aria-hidden="true" tabindex="-1"></a>     tmp_input_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_mc"</span>))</span>
<span id="cb189-503"><a href="#cb189-503" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_input_name)</span>
<span id="cb189-504"><a href="#cb189-504" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_input_name, tmp_input)</span>
<span id="cb189-505"><a href="#cb189-505" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-506"><a href="#cb189-506" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-507"><a href="#cb189-507" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"_md"</span>))</span>
<span id="cb189-508"><a href="#cb189-508" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"_otu_tax"</span>))</span>
<span id="cb189-509"><a href="#cb189-509" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-510"><a href="#cb189-510" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-511"><a href="#cb189-511" aria-hidden="true" tabindex="-1"></a><span class="fu">## Choose Data Set</span></span>
<span id="cb189-512"><a href="#cb189-512" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-513"><a href="#cb189-513" aria-hidden="true" tabindex="-1"></a>At this point in the code we need to choose a data set to use, formatted with <span class="in">`mctoolsr`</span>. Remember, there are four choices:</span>
<span id="cb189-514"><a href="#cb189-514" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-515"><a href="#cb189-515" aria-hidden="true" tabindex="-1"></a>1) FULL, unfiltered data set.</span>
<span id="cb189-516"><a href="#cb189-516" aria-hidden="true" tabindex="-1"></a>2) Arbitrary filtered data set.</span>
<span id="cb189-517"><a href="#cb189-517" aria-hidden="true" tabindex="-1"></a>3) PERfect filtered data set.</span>
<span id="cb189-518"><a href="#cb189-518" aria-hidden="true" tabindex="-1"></a>4) PIME filtered data set.</span>
<span id="cb189-519"><a href="#cb189-519" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-520"><a href="#cb189-520" aria-hidden="true" tabindex="-1"></a>This way, if we want to test other data sets we only need to change the name here.</span>
<span id="cb189-521"><a href="#cb189-521" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-524"><a href="#cb189-524" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-525"><a href="#cb189-525" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">"_mc"</span>)</span>
<span id="cb189-526"><a href="#cb189-526" aria-hidden="true" tabindex="-1"></a>ssu18_select_mc <span class="ot">&lt;-</span> ssu18_ps_pime_mc</span>
<span id="cb189-527"><a href="#cb189-527" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-528"><a href="#cb189-528" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-529"><a href="#cb189-529" aria-hidden="true" tabindex="-1"></a><span class="fu">## Normality Tests</span></span>
<span id="cb189-530"><a href="#cb189-530" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-531"><a href="#cb189-531" aria-hidden="true" tabindex="-1"></a>Before proceeding, we need to test each parameter in the metadata to see which ones are and are not normally distributed. For that, we use the Shapiro-Wilk Normality Test. Here we only need one of the metadata files.</span>
<span id="cb189-532"><a href="#cb189-532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-535"><a href="#cb189-535" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-536"><a href="#cb189-536" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-537"><a href="#cb189-537" aria-hidden="true" tabindex="-1"></a>temp_md <span class="ot">&lt;-</span> ssu18_select_mc<span class="sc">$</span>map_loaded</span>
<span id="cb189-538"><a href="#cb189-538" aria-hidden="true" tabindex="-1"></a>temp_md[,<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-539"><a href="#cb189-539" aria-hidden="true" tabindex="-1"></a>shap_results <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-540"><a href="#cb189-540" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-541"><a href="#cb189-541" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">colnames</span>(temp_md)) {</span>
<span id="cb189-542"><a href="#cb189-542" aria-hidden="true" tabindex="-1"></a>  tmp_shap <span class="ot">&lt;-</span> <span class="fu">shapiro.test</span>(temp_md[[i]])</span>
<span id="cb189-543"><a href="#cb189-543" aria-hidden="true" tabindex="-1"></a>  tmp_p <span class="ot">&lt;-</span> <span class="fu">round</span>(tmp_shap<span class="sc">$</span>p.value, <span class="at">digits =</span> <span class="dv">5</span>)</span>
<span id="cb189-544"><a href="#cb189-544" aria-hidden="true" tabindex="-1"></a>  tmp_res <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">isTRUE</span>(tmp_shap<span class="sc">$</span>p.value <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span>
<span id="cb189-545"><a href="#cb189-545" aria-hidden="true" tabindex="-1"></a>  shap_results <span class="ot">&lt;-</span> <span class="fu">rbind</span>(shap_results, <span class="fu">data.frame</span>(i, tmp_p, tmp_res))</span>
<span id="cb189-546"><a href="#cb189-546" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-547"><a href="#cb189-547" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-548"><a href="#cb189-548" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(shap_results) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"parameter"</span>, <span class="st">"p-value"</span>, <span class="st">"tranform"</span>)</span>
<span id="cb189-549"><a href="#cb189-549" aria-hidden="true" tabindex="-1"></a>shap_results</span>
<span id="cb189-550"><a href="#cb189-550" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">filter</span>(shap_results, tranform <span class="sc">==</span> <span class="st">"TRUE"</span>)</span>
<span id="cb189-551"><a href="#cb189-551" aria-hidden="true" tabindex="-1"></a>md_to_tranform <span class="ot">&lt;-</span> shap_results<span class="sc">$</span>parameter[shap_results<span class="sc">$</span>tranform <span class="sc">==</span> <span class="cn">TRUE</span>]</span>
<span id="cb189-552"><a href="#cb189-552" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"temp_md"</span>))</span>
<span id="cb189-553"><a href="#cb189-553" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-554"><a href="#cb189-554" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb189-555"><a href="#cb189-555" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>**Show the results** of each normality test for metadata parameters<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb189-556"><a href="#cb189-556" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-557"><a href="#cb189-557" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb189-558"><a href="#cb189-558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-559"><a href="#cb189-559" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab_ssu("ssu_normality_tests")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-560"><a href="#cb189-560" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-563"><a href="#cb189-563" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-564"><a href="#cb189-564" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-565"><a href="#cb189-565" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-566"><a href="#cb189-566" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> shap_results</span>
<span id="cb189-567"><a href="#cb189-567" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb189-568"><a href="#cb189-568" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(<span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>),</span>
<span id="cb189-569"><a href="#cb189-569" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb189-570"><a href="#cb189-570" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">1</span>),</span>
<span id="cb189-571"><a href="#cb189-571" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">TRUE</span>, <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-572"><a href="#cb189-572" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb189-573"><a href="#cb189-573" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb189-574"><a href="#cb189-574" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb189-575"><a href="#cb189-575" aria-hidden="true" tabindex="-1"></a>    <span class="at">parameter =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Parameter"</span>, </span>
<span id="cb189-576"><a href="#cb189-576" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb189-577"><a href="#cb189-577" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>),</span>
<span id="cb189-578"><a href="#cb189-578" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>, <span class="at">fontSize =</span> <span class="st">"0.9em"</span>), </span>
<span id="cb189-579"><a href="#cb189-579" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb189-580"><a href="#cb189-580" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">150</span>), </span>
<span id="cb189-581"><a href="#cb189-581" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">p-value</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"p-value"</span>, <span class="at">style =</span> <span class="cf">function</span>(value) {</span>
<span id="cb189-582"><a href="#cb189-582" aria-hidden="true" tabindex="-1"></a>                                         <span class="cf">if</span> (value <span class="sc">&gt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb189-583"><a href="#cb189-583" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#2271B2"</span> <span class="co">#green</span></span>
<span id="cb189-584"><a href="#cb189-584" aria-hidden="true" tabindex="-1"></a>                                         } <span class="cf">else</span> <span class="cf">if</span> (value <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb189-585"><a href="#cb189-585" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#B22271"</span> <span class="co">#RED "#", "#"</span></span>
<span id="cb189-586"><a href="#cb189-586" aria-hidden="true" tabindex="-1"></a>                                         } <span class="cf">else</span> {</span>
<span id="cb189-587"><a href="#cb189-587" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#777"</span></span>
<span id="cb189-588"><a href="#cb189-588" aria-hidden="true" tabindex="-1"></a>                                         }</span>
<span id="cb189-589"><a href="#cb189-589" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">list</span>(<span class="at">color =</span> color, <span class="at">fontWeight =</span> <span class="st">"bold"</span>)})</span>
<span id="cb189-590"><a href="#cb189-590" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb189-591"><a href="#cb189-591" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">60</span>, </span>
<span id="cb189-592"><a href="#cb189-592" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb189-593"><a href="#cb189-593" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">FALSE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb189-594"><a href="#cb189-594" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb189-595"><a href="#cb189-595" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-596"><a href="#cb189-596" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb189-597"><a href="#cb189-597" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(seq_table)</span>
<span id="cb189-598"><a href="#cb189-598" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-599"><a href="#cb189-599" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb189-600"><a href="#cb189-600" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-601"><a href="#cb189-601" aria-hidden="true" tabindex="-1"></a>Looks like we need to transform <span class="in">`r length(md_to_tranform)`</span> metadata parameters.</span>
<span id="cb189-602"><a href="#cb189-602" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-603"><a href="#cb189-603" aria-hidden="true" tabindex="-1"></a><span class="fu">## Normalize Parameters</span></span>
<span id="cb189-604"><a href="#cb189-604" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-605"><a href="#cb189-605" aria-hidden="true" tabindex="-1"></a>Here we use the R package <span class="co">[</span><span class="ot">`bestNormalize`</span><span class="co">](https://cran.r-project.org/web/packages/bestNormalize/vignettes/bestNormalize.html)</span> to find and execute the best normalizing transformation. The function will test the following normalizing transformations:</span>
<span id="cb189-606"><a href="#cb189-606" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-607"><a href="#cb189-607" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="in">`arcsinh_x`</span> performs an arcsinh transformation.</span>
<span id="cb189-608"><a href="#cb189-608" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-609"><a href="#cb189-609" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="in">`boxcox`</span> Perform a Box-Cox transformation and center/scale a vector to attempt normalization. <span class="in">`boxcox`</span> estimates the optimal value of lambda for the Box-Cox transformation. The function will return an error if a user attempt to transform nonpositive data.</span>
<span id="cb189-610"><a href="#cb189-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-611"><a href="#cb189-611" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="in">`yeojohnson`</span> Perform a Yeo-Johnson Transformation and center/scale a vector to attempt normalization. <span class="in">`yeojohnson`</span> estimates the optimal value of lambda for the Yeo-Johnson transformation. The Yeo-Johnson is similar to the Box-Cox method, however it allows for the transformation of nonpositive data as well. </span>
<span id="cb189-612"><a href="#cb189-612" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-613"><a href="#cb189-613" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="in">`orderNorm`</span> The Ordered Quantile (ORQ) normalization transformation, <span class="in">`orderNorm()`</span>, is a rank-based procedure by which the values of a vector are mapped to their percentile, which is then mapped to the same percentile of the normal distribution. Without the presence of ties, this essentially guarantees that the transformation leads to a uniform distribution.</span>
<span id="cb189-614"><a href="#cb189-614" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-615"><a href="#cb189-615" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="in">`log_x`</span> performs a simple log transformation. The parameter a is essentially estimated by the training set by default (estimated as the minimum possible to some extent epsilon), while the base must be specified beforehand. The default base of the log is 10.</span>
<span id="cb189-616"><a href="#cb189-616" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-617"><a href="#cb189-617" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="in">`sqrt_x`</span> performs a simple square-root transformation. The parameter a is essentially estimated by the training set by default (estimated as the minimum possible), while the base must be specified beforehand.</span>
<span id="cb189-618"><a href="#cb189-618" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-619"><a href="#cb189-619" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`exp_x`</span> performs a simple exponential transformation.</span>
<span id="cb189-620"><a href="#cb189-620" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-621"><a href="#cb189-621" aria-hidden="true" tabindex="-1"></a>See this GitHub issue (<span class="co">[</span><span class="ot">#5</span><span class="co">](https://github.com/petersonR/bestNormalize/issues/5)</span>) for a description on getting reproducible results. Apparently, you can get different results because the <span class="in">`bestNormalize()`</span> function uses repeated cross-validation (and doesn't automatically set the seed), so the results will be slightly different each time the function is executed.</span>
<span id="cb189-622"><a href="#cb189-622" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-625"><a href="#cb189-625" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-626"><a href="#cb189-626" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-627"><a href="#cb189-627" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">119</span>)</span>
<span id="cb189-628"><a href="#cb189-628" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> md_to_tranform) {</span>
<span id="cb189-629"><a href="#cb189-629" aria-hidden="true" tabindex="-1"></a>  tmp_md <span class="ot">&lt;-</span> ssu18_select_mc<span class="sc">$</span>map_loaded</span>
<span id="cb189-630"><a href="#cb189-630" aria-hidden="true" tabindex="-1"></a>  tmp_best_norm <span class="ot">&lt;-</span> <span class="fu">bestNormalize</span>(tmp_md[[i]], <span class="at">r =</span> <span class="dv">1</span>, <span class="at">k =</span> <span class="dv">5</span>, <span class="at">loo =</span> <span class="cn">TRUE</span>)</span>
<span id="cb189-631"><a href="#cb189-631" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_best_norm_test"</span>))</span>
<span id="cb189-632"><a href="#cb189-632" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_best_norm)</span>
<span id="cb189-633"><a href="#cb189-633" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(tmp_name)</span>
<span id="cb189-634"><a href="#cb189-634" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-635"><a href="#cb189-635" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-636"><a href="#cb189-636" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-637"><a href="#cb189-637" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-638"><a href="#cb189-638" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb189-639"><a href="#cb189-639" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show the chosen transformations<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb189-640"><a href="#cb189-640" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-643"><a href="#cb189-643" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-644"><a href="#cb189-644" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-645"><a href="#cb189-645" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-646"><a href="#cb189-646" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> md_to_tranform) {</span>
<span id="cb189-647"><a href="#cb189-647" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_best_norm_test"</span>)))</span>
<span id="cb189-648"><a href="#cb189-648" aria-hidden="true" tabindex="-1"></a>  tmp_print <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"bestNormalize Chosen transformation of"</span>, i)</span>
<span id="cb189-649"><a href="#cb189-649" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-650"><a href="#cb189-650" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"##"</span>, tmp_print, <span class="st">"##"</span>, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-651"><a href="#cb189-651" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(tmp_get<span class="sc">$</span>chosen_transform)</span>
<span id="cb189-652"><a href="#cb189-652" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-653"><a href="#cb189-653" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-654"><a href="#cb189-654" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-655"><a href="#cb189-655" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-656"><a href="#cb189-656" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-657"><a href="#cb189-657" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb189-658"><a href="#cb189-658" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-659"><a href="#cb189-659" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb189-660"><a href="#cb189-660" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show the complete <span class="in">`bestNormalize`</span> results<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb189-661"><a href="#cb189-661" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-664"><a href="#cb189-664" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-665"><a href="#cb189-665" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-666"><a href="#cb189-666" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-667"><a href="#cb189-667" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> md_to_tranform) {</span>
<span id="cb189-668"><a href="#cb189-668" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_best_norm_test"</span>)))</span>
<span id="cb189-669"><a href="#cb189-669" aria-hidden="true" tabindex="-1"></a>  tmp_print <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Results of bestNormalize for"</span>, i)</span>
<span id="cb189-670"><a href="#cb189-670" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-671"><a href="#cb189-671" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"##"</span>, tmp_print, <span class="st">"##"</span>, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-672"><a href="#cb189-672" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(tmp_get)</span>
<span id="cb189-673"><a href="#cb189-673" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-674"><a href="#cb189-674" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-675"><a href="#cb189-675" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-676"><a href="#cb189-676" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-677"><a href="#cb189-677" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-678"><a href="#cb189-678" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb189-679"><a href="#cb189-679" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-680"><a href="#cb189-680" aria-hidden="true" tabindex="-1"></a>Great, now we can add the normalized transformed data back to our <span class="in">`mctoolsr`</span> metadata file. </span>
<span id="cb189-681"><a href="#cb189-681" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-684"><a href="#cb189-684" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-685"><a href="#cb189-685" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-686"><a href="#cb189-686" aria-hidden="true" tabindex="-1"></a>ssu18_select_mc_norm <span class="ot">&lt;-</span> ssu18_select_mc</span>
<span id="cb189-687"><a href="#cb189-687" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-688"><a href="#cb189-688" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> md_to_tranform) {</span>
<span id="cb189-689"><a href="#cb189-689" aria-hidden="true" tabindex="-1"></a>      tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">"_best_norm_test"</span>)))</span>
<span id="cb189-690"><a href="#cb189-690" aria-hidden="true" tabindex="-1"></a>      tmp_new_data <span class="ot">&lt;-</span> tmp_get<span class="sc">$</span>x.t</span>
<span id="cb189-691"><a href="#cb189-691" aria-hidden="true" tabindex="-1"></a>      ssu18_select_mc_norm<span class="sc">$</span>map_loaded[[i]] <span class="ot">&lt;-</span> tmp_new_data</span>
<span id="cb189-692"><a href="#cb189-692" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-693"><a href="#cb189-693" aria-hidden="true" tabindex="-1"></a>}     </span>
<span id="cb189-694"><a href="#cb189-694" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-695"><a href="#cb189-695" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-696"><a href="#cb189-696" aria-hidden="true" tabindex="-1"></a>And rerun the Shapiro Tests. </span>
<span id="cb189-697"><a href="#cb189-697" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-700"><a href="#cb189-700" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-701"><a href="#cb189-701" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-702"><a href="#cb189-702" aria-hidden="true" tabindex="-1"></a>temp_md_norm <span class="ot">&lt;-</span> ssu18_select_mc_norm<span class="sc">$</span>map_loaded</span>
<span id="cb189-703"><a href="#cb189-703" aria-hidden="true" tabindex="-1"></a>temp_md_norm[,<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-704"><a href="#cb189-704" aria-hidden="true" tabindex="-1"></a>shap_results_norm <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-705"><a href="#cb189-705" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-706"><a href="#cb189-706" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">colnames</span>(temp_md_norm)) {</span>
<span id="cb189-707"><a href="#cb189-707" aria-hidden="true" tabindex="-1"></a>  tmp_shap <span class="ot">&lt;-</span> <span class="fu">shapiro.test</span>(temp_md_norm[[i]])</span>
<span id="cb189-708"><a href="#cb189-708" aria-hidden="true" tabindex="-1"></a>  tmp_p <span class="ot">&lt;-</span> <span class="fu">round</span>(tmp_shap<span class="sc">$</span>p.value, <span class="at">digits =</span> <span class="dv">5</span>)</span>
<span id="cb189-709"><a href="#cb189-709" aria-hidden="true" tabindex="-1"></a>  tmp_res <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">isTRUE</span>(tmp_shap<span class="sc">$</span>p.value <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span>
<span id="cb189-710"><a href="#cb189-710" aria-hidden="true" tabindex="-1"></a>  shap_results_norm <span class="ot">&lt;-</span> <span class="fu">rbind</span>(shap_results_norm, <span class="fu">data.frame</span>(i, tmp_p, tmp_res))</span>
<span id="cb189-711"><a href="#cb189-711" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-712"><a href="#cb189-712" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-713"><a href="#cb189-713" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(shap_results_norm) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"parameter"</span>, <span class="st">"p-value"</span>, <span class="st">"tranform"</span>)</span>
<span id="cb189-714"><a href="#cb189-714" aria-hidden="true" tabindex="-1"></a>shap_results_norm</span>
<span id="cb189-715"><a href="#cb189-715" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"temp_md_norm"</span>))</span>
<span id="cb189-716"><a href="#cb189-716" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-717"><a href="#cb189-717" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-718"><a href="#cb189-718" aria-hidden="true" tabindex="-1"></a>And check if there are any parameters that are still significant for the normality test.</span>
<span id="cb189-719"><a href="#cb189-719" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-722"><a href="#cb189-722" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-723"><a href="#cb189-723" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-724"><a href="#cb189-724" aria-hidden="true" tabindex="-1"></a>shap_results<span class="sc">$</span>parameter[shap_results_norm<span class="sc">$</span>tranform <span class="sc">==</span> <span class="cn">TRUE</span>]</span>
<span id="cb189-725"><a href="#cb189-725" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-726"><a href="#cb189-726" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-727"><a href="#cb189-727" aria-hidden="true" tabindex="-1"></a>Ok. Looks like <span class="in">`bestNormalize`</span> was unable to find a suitable transformation for <span class="in">`Al`</span> and <span class="in">`Fe`</span>. This is likely because there is very little variation in these metadata and/or there are too few significant digits.</span>
<span id="cb189-728"><a href="#cb189-728" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-729"><a href="#cb189-729" aria-hidden="true" tabindex="-1"></a><span class="fu">## Normalized Metadata</span></span>
<span id="cb189-730"><a href="#cb189-730" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-731"><a href="#cb189-731" aria-hidden="true" tabindex="-1"></a>Finally, here is a new summary table that includes all of the normalized data.</span>
<span id="cb189-732"><a href="#cb189-732" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-733"><a href="#cb189-733" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb189-734"><a href="#cb189-734" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-737"><a href="#cb189-737" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-738"><a href="#cb189-738" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-739"><a href="#cb189-739" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-740"><a href="#cb189-740" aria-hidden="true" tabindex="-1"></a>metad_norm <span class="ot">&lt;-</span> ssu18_select_mc_norm<span class="sc">$</span>map_loaded</span>
<span id="cb189-741"><a href="#cb189-741" aria-hidden="true" tabindex="-1"></a>metad_norm <span class="ot">&lt;-</span> metad_norm <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"Sample_ID"</span>)</span>
<span id="cb189-742"><a href="#cb189-742" aria-hidden="true" tabindex="-1"></a>metad_norm <span class="ot">&lt;-</span> metad_norm[<span class="fu">order</span>(<span class="fu">match</span>(metad_norm<span class="sc">$</span>Sample_ID, metad<span class="sc">$</span>Sample_ID)), ]</span>
<span id="cb189-743"><a href="#cb189-743" aria-hidden="true" tabindex="-1"></a>metad_norm[,<span class="dv">2</span><span class="sc">:</span><span class="dv">9</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-744"><a href="#cb189-744" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> metad_norm</span>
<span id="cb189-745"><a href="#cb189-745" aria-hidden="true" tabindex="-1"></a>seq_table <span class="sc">%&gt;%</span></span>
<span id="cb189-746"><a href="#cb189-746" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb189-747"><a href="#cb189-747" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"normalized_metadata"</span>,</span>
<span id="cb189-748"><a href="#cb189-748" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb189-749"><a href="#cb189-749" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb189-750"><a href="#cb189-750" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb189-751"><a href="#cb189-751" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb189-752"><a href="#cb189-752" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-753"><a href="#cb189-753" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb189-754"><a href="#cb189-754" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-755"><a href="#cb189-755" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-756"><a href="#cb189-756" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab_ssu("ssu_normalized_metadata")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-757"><a href="#cb189-757" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-760"><a href="#cb189-760" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-761"><a href="#cb189-761" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-762"><a href="#cb189-762" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-763"><a href="#cb189-763" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> seq_table <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="fu">across</span>(<span class="at">.cols =</span> <span class="fu">c</span>(<span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">7</span><span class="sc">:</span><span class="dv">9</span>, <span class="dv">14</span><span class="sc">:</span><span class="dv">15</span>,<span class="dv">19</span><span class="sc">:</span><span class="dv">20</span>, </span>
<span id="cb189-764"><a href="#cb189-764" aria-hidden="true" tabindex="-1"></a>                                                            <span class="dv">24</span>,<span class="dv">26</span>,<span class="dv">30</span>,<span class="dv">34</span><span class="sc">:</span><span class="dv">38</span>,<span class="dv">40</span>, <span class="dv">42</span><span class="sc">:</span><span class="dv">43</span>, </span>
<span id="cb189-765"><a href="#cb189-765" aria-hidden="true" tabindex="-1"></a>                                                            <span class="dv">49</span>, <span class="dv">52</span><span class="sc">:</span><span class="dv">53</span>,<span class="dv">55</span>,<span class="dv">57</span>,<span class="dv">60</span><span class="sc">:</span><span class="dv">62</span>), </span>
<span id="cb189-766"><a href="#cb189-766" aria-hidden="true" tabindex="-1"></a>                                                  round, <span class="at">digits =</span> <span class="dv">3</span>)) </span>
<span id="cb189-767"><a href="#cb189-767" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb189-768"><a href="#cb189-768" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(<span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>),</span>
<span id="cb189-769"><a href="#cb189-769" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb189-770"><a href="#cb189-770" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">1</span>),</span>
<span id="cb189-771"><a href="#cb189-771" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">TRUE</span>, <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-772"><a href="#cb189-772" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb189-773"><a href="#cb189-773" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb189-774"><a href="#cb189-774" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb189-775"><a href="#cb189-775" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Sample ID"</span>, </span>
<span id="cb189-776"><a href="#cb189-776" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb189-777"><a href="#cb189-777" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>),</span>
<span id="cb189-778"><a href="#cb189-778" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>, <span class="at">fontSize =</span> <span class="st">"0.9em"</span>), </span>
<span id="cb189-779"><a href="#cb189-779" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb189-780"><a href="#cb189-780" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">150</span>), </span>
<span id="cb189-781"><a href="#cb189-781" aria-hidden="true" tabindex="-1"></a>    <span class="at">raw_reads =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values)))</span>
<span id="cb189-782"><a href="#cb189-782" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb189-783"><a href="#cb189-783" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">5</span>, </span>
<span id="cb189-784"><a href="#cb189-784" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb189-785"><a href="#cb189-785" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb189-786"><a href="#cb189-786" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb189-787"><a href="#cb189-787" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-788"><a href="#cb189-788" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb189-789"><a href="#cb189-789" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(seq_table)</span>
<span id="cb189-790"><a href="#cb189-790" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-791"><a href="#cb189-791" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-794"><a href="#cb189-794" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-795"><a href="#cb189-795" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-796"><a href="#cb189-796" aria-hidden="true" tabindex="-1"></a><span class="co"># Code to run individual transformations for each metadata and generate histograms</span></span>
<span id="cb189-797"><a href="#cb189-797" aria-hidden="true" tabindex="-1"></a><span class="do">##for (i in md_to_tranform) {</span></span>
<span id="cb189-798"><a href="#cb189-798" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_data &lt;- ssu18_select_mc$map_loaded</span></span>
<span id="cb189-799"><a href="#cb189-799" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_no_transform_obj &lt;- no_transform(tmp_data[[i]])</span></span>
<span id="cb189-800"><a href="#cb189-800" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_arcsinh_obj &lt;- arcsinh_x(tmp_data[[i]])</span></span>
<span id="cb189-801"><a href="#cb189-801" aria-hidden="true" tabindex="-1"></a><span class="do">##     #tmp_boxcox_obj &lt;- boxcox(tmp_data[[i]])</span></span>
<span id="cb189-802"><a href="#cb189-802" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_yeojohnson_obj &lt;- yeojohnson(tmp_data[[i]])</span></span>
<span id="cb189-803"><a href="#cb189-803" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_orderNorm_obj &lt;- orderNorm(tmp_data[[i]])</span></span>
<span id="cb189-804"><a href="#cb189-804" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_log_obj &lt;- log_x(tmp_data[[i]])</span></span>
<span id="cb189-805"><a href="#cb189-805" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_sqrt_x_obj &lt;- sqrt_x(tmp_data[[i]])</span></span>
<span id="cb189-806"><a href="#cb189-806" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_exp_x_obj &lt;- exp_x(tmp_data[[i]])</span></span>
<span id="cb189-807"><a href="#cb189-807" aria-hidden="true" tabindex="-1"></a><span class="do">##     </span></span>
<span id="cb189-808"><a href="#cb189-808" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_no_transform &lt;- hist(tmp_no_transform_obj$x.t, main = i, breaks = 6, xlab = "no_transform")</span></span>
<span id="cb189-809"><a href="#cb189-809" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_name &lt;- purrr::map_chr(i, ~ paste0(., "_hist_plot_no_transform"))</span></span>
<span id="cb189-810"><a href="#cb189-810" aria-hidden="true" tabindex="-1"></a><span class="do">##     assign(tmp_plot_name, tmp_plot_no_transform)</span></span>
<span id="cb189-811"><a href="#cb189-811" aria-hidden="true" tabindex="-1"></a><span class="do">##     print(tmp_plot_name)</span></span>
<span id="cb189-812"><a href="#cb189-812" aria-hidden="true" tabindex="-1"></a><span class="do">##     </span></span>
<span id="cb189-813"><a href="#cb189-813" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_arcsinh &lt;- hist(tmp_arcsinh_obj$x.t, main = i, breaks = 6, xlab = "arcsinh")</span></span>
<span id="cb189-814"><a href="#cb189-814" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_name &lt;- purrr::map_chr(i, ~ paste0(., "_hist_plot_arcsinh"))</span></span>
<span id="cb189-815"><a href="#cb189-815" aria-hidden="true" tabindex="-1"></a><span class="do">##     assign(tmp_plot_name, tmp_plot_arcsinh)</span></span>
<span id="cb189-816"><a href="#cb189-816" aria-hidden="true" tabindex="-1"></a><span class="do">##     print(tmp_plot_name)</span></span>
<span id="cb189-817"><a href="#cb189-817" aria-hidden="true" tabindex="-1"></a><span class="do">##     # cannot run because of Al and Fe</span></span>
<span id="cb189-818"><a href="#cb189-818" aria-hidden="true" tabindex="-1"></a><span class="do">##     #tmp_plot_boxcox &lt;- hist(tmp_boxcox_obj$x.t, main = i, breaks = 6, xlab = "boxcox")</span></span>
<span id="cb189-819"><a href="#cb189-819" aria-hidden="true" tabindex="-1"></a><span class="do">##     #tmp_plot_name &lt;- purrr::map_chr(i, ~ paste0(., "_hist_plot_boxcox"))</span></span>
<span id="cb189-820"><a href="#cb189-820" aria-hidden="true" tabindex="-1"></a><span class="do">##     #assign(tmp_plot_name, tmp_plot_boxcox)</span></span>
<span id="cb189-821"><a href="#cb189-821" aria-hidden="true" tabindex="-1"></a><span class="do">##     #print(tmp_plot_name)</span></span>
<span id="cb189-822"><a href="#cb189-822" aria-hidden="true" tabindex="-1"></a><span class="do">##     </span></span>
<span id="cb189-823"><a href="#cb189-823" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_yeojohnson &lt;- hist(tmp_yeojohnson_obj$x.t, main = i, breaks = 6, xlab = "yeojohnson")</span></span>
<span id="cb189-824"><a href="#cb189-824" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_name &lt;- purrr::map_chr(i, ~ paste0(., "_hist_plot_yeojohnson"))</span></span>
<span id="cb189-825"><a href="#cb189-825" aria-hidden="true" tabindex="-1"></a><span class="do">##     assign(tmp_plot_name, tmp_plot_yeojohnson)</span></span>
<span id="cb189-826"><a href="#cb189-826" aria-hidden="true" tabindex="-1"></a><span class="do">##     print(tmp_plot_name)</span></span>
<span id="cb189-827"><a href="#cb189-827" aria-hidden="true" tabindex="-1"></a><span class="do">##     </span></span>
<span id="cb189-828"><a href="#cb189-828" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_orderNorm &lt;- hist(tmp_orderNorm_obj$x.t, main = i, breaks = 6, xlab = "orderNorm")</span></span>
<span id="cb189-829"><a href="#cb189-829" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_name &lt;- purrr::map_chr(i, ~ paste0(., "_hist_plot_orderNorm"))</span></span>
<span id="cb189-830"><a href="#cb189-830" aria-hidden="true" tabindex="-1"></a><span class="do">##     assign(tmp_plot_name, tmp_plot_orderNorm)</span></span>
<span id="cb189-831"><a href="#cb189-831" aria-hidden="true" tabindex="-1"></a><span class="do">##     print(tmp_plot_name)</span></span>
<span id="cb189-832"><a href="#cb189-832" aria-hidden="true" tabindex="-1"></a><span class="do">##     </span></span>
<span id="cb189-833"><a href="#cb189-833" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_log &lt;- hist(tmp_log_obj$x.t, main = i, breaks = 6, xlab = "log")</span></span>
<span id="cb189-834"><a href="#cb189-834" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_name &lt;- purrr::map_chr(i, ~ paste0(., "_hist_plot_log"))</span></span>
<span id="cb189-835"><a href="#cb189-835" aria-hidden="true" tabindex="-1"></a><span class="do">##     assign(tmp_plot_name, tmp_plot_log)</span></span>
<span id="cb189-836"><a href="#cb189-836" aria-hidden="true" tabindex="-1"></a><span class="do">##     print(tmp_plot_name)</span></span>
<span id="cb189-837"><a href="#cb189-837" aria-hidden="true" tabindex="-1"></a><span class="do">##     </span></span>
<span id="cb189-838"><a href="#cb189-838" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_sqrt &lt;- hist(tmp_sqrt_x_obj$x.t, main = i, breaks = 6, xlab = "sqrt")</span></span>
<span id="cb189-839"><a href="#cb189-839" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_name &lt;- purrr::map_chr(i, ~ paste0(., "_hist_plot_sqrt"))</span></span>
<span id="cb189-840"><a href="#cb189-840" aria-hidden="true" tabindex="-1"></a><span class="do">##     assign(tmp_plot_name, tmp_plot_sqrt)</span></span>
<span id="cb189-841"><a href="#cb189-841" aria-hidden="true" tabindex="-1"></a><span class="do">##     print(tmp_plot_name)</span></span>
<span id="cb189-842"><a href="#cb189-842" aria-hidden="true" tabindex="-1"></a><span class="do">##     </span></span>
<span id="cb189-843"><a href="#cb189-843" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_exp &lt;- hist(tmp_exp_x_obj$x.t, main = i, breaks = 6, xlab = "exp")</span></span>
<span id="cb189-844"><a href="#cb189-844" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_name &lt;- purrr::map_chr(i, ~ paste0(., "_hist_plot_exp"))</span></span>
<span id="cb189-845"><a href="#cb189-845" aria-hidden="true" tabindex="-1"></a><span class="do">##     assign(tmp_plot_name, tmp_plot_exp)</span></span>
<span id="cb189-846"><a href="#cb189-846" aria-hidden="true" tabindex="-1"></a><span class="do">##     print(tmp_plot_name)</span></span>
<span id="cb189-847"><a href="#cb189-847" aria-hidden="true" tabindex="-1"></a><span class="do">##</span></span>
<span id="cb189-848"><a href="#cb189-848" aria-hidden="true" tabindex="-1"></a><span class="do">##     rm(list = ls(pattern = "tmp_"))</span></span>
<span id="cb189-849"><a href="#cb189-849" aria-hidden="true" tabindex="-1"></a><span class="do">##}</span></span>
<span id="cb189-850"><a href="#cb189-850" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-851"><a href="#cb189-851" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-852"><a href="#cb189-852" aria-hidden="true" tabindex="-1"></a><span class="fu">## Autocorrelation Tests</span></span>
<span id="cb189-853"><a href="#cb189-853" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-854"><a href="#cb189-854" aria-hidden="true" tabindex="-1"></a>Next, we  test the metadata for autocorrelations. Do we do this on the original data or the transformed data? No idea, so let's do both.</span>
<span id="cb189-855"><a href="#cb189-855" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-856"><a href="#cb189-856" aria-hidden="true" tabindex="-1"></a><span class="fu">### Split Metadata</span></span>
<span id="cb189-857"><a href="#cb189-857" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-858"><a href="#cb189-858" aria-hidden="true" tabindex="-1"></a>We need to split the data into different groups.</span>
<span id="cb189-859"><a href="#cb189-859" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-860"><a href="#cb189-860" aria-hidden="true" tabindex="-1"></a>A) Environmental and edaphic properties</span>
<span id="cb189-861"><a href="#cb189-861" aria-hidden="true" tabindex="-1"></a>B) Microbial functional responses</span>
<span id="cb189-862"><a href="#cb189-862" aria-hidden="true" tabindex="-1"></a>C) Temperature adaptation properties</span>
<span id="cb189-863"><a href="#cb189-863" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-864"><a href="#cb189-864" aria-hidden="true" tabindex="-1"></a>We first create lists of metadata parameters. </span>
<span id="cb189-865"><a href="#cb189-865" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-868"><a href="#cb189-868" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-869"><a href="#cb189-869" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-870"><a href="#cb189-870" aria-hidden="true" tabindex="-1"></a>div <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"PLOT"</span>, <span class="st">"TREAT"</span>, <span class="st">"TREAT_T"</span>, <span class="st">"PAIR"</span>, <span class="st">"Observed"</span>, <span class="st">"Shannon_exp"</span>, </span>
<span id="cb189-871"><a href="#cb189-871" aria-hidden="true" tabindex="-1"></a>         <span class="st">"InvSimpson"</span>, <span class="st">"ATAP"</span>)</span>
<span id="cb189-872"><a href="#cb189-872" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-873"><a href="#cb189-873" aria-hidden="true" tabindex="-1"></a>edaphic <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"AST"</span>, <span class="st">"H2O"</span>, <span class="st">"N"</span>, <span class="st">"P"</span>, <span class="st">"Al"</span>, </span>
<span id="cb189-874"><a href="#cb189-874" aria-hidden="true" tabindex="-1"></a>             <span class="st">"Ca"</span>, <span class="st">"Fe"</span>, <span class="st">"K"</span>, <span class="st">"Mg"</span>, <span class="st">"Mn"</span>, <span class="st">"Na"</span>, <span class="st">"TEB"</span>, <span class="st">"ECEC"</span>, <span class="st">"pH"</span>, </span>
<span id="cb189-875"><a href="#cb189-875" aria-hidden="true" tabindex="-1"></a>             <span class="st">"NH4"</span>, <span class="st">"NO3"</span>, <span class="st">"PO4"</span>, <span class="st">"DOC"</span>, <span class="st">"DON"</span>, <span class="st">"DOCN"</span>)</span>
<span id="cb189-876"><a href="#cb189-876" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-877"><a href="#cb189-877" aria-hidden="true" tabindex="-1"></a>soil_funct <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"micC"</span>, <span class="st">"micN"</span>, <span class="st">"micP"</span>, <span class="st">"micCN"</span>, <span class="st">"micCP"</span>, <span class="st">"micNP"</span>, </span>
<span id="cb189-878"><a href="#cb189-878" aria-hidden="true" tabindex="-1"></a>                <span class="st">"AG_ase"</span>, <span class="st">"BG_ase"</span>, <span class="st">"BP_ase"</span>, <span class="st">"CE_ase"</span>, <span class="st">"P_ase"</span>, <span class="st">"N_ase"</span>, </span>
<span id="cb189-879"><a href="#cb189-879" aria-hidden="true" tabindex="-1"></a>                <span class="st">"S_ase"</span>, <span class="st">"XY_ase"</span>,  <span class="st">"LP_ase"</span>, <span class="st">"PX_ase"</span>, <span class="st">"CO2"</span>, </span>
<span id="cb189-880"><a href="#cb189-880" aria-hidden="true" tabindex="-1"></a>                <span class="st">"enzCN"</span>, <span class="st">"enzCP"</span>, <span class="st">"enzNP"</span>)</span>
<span id="cb189-881"><a href="#cb189-881" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-882"><a href="#cb189-882" aria-hidden="true" tabindex="-1"></a>temp_adapt <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"AG_Q10"</span>, <span class="st">"BG_Q10"</span>, <span class="st">"BP_Q10"</span>, <span class="st">"CE_Q10"</span>, <span class="st">"P_Q10"</span>, <span class="st">"N_Q10"</span>, </span>
<span id="cb189-883"><a href="#cb189-883" aria-hidden="true" tabindex="-1"></a>                <span class="st">"S_Q10"</span>, <span class="st">"XY_Q10"</span>, <span class="st">"LP_Q10"</span>, <span class="st">"PX_Q10"</span>, <span class="st">"CUEcn"</span>, <span class="st">"CUEcp"</span>, </span>
<span id="cb189-884"><a href="#cb189-884" aria-hidden="true" tabindex="-1"></a>                <span class="st">"NUE"</span>,<span class="st">"PUE"</span>, <span class="st">"Tmin"</span>, <span class="st">"SI"</span>)</span>
<span id="cb189-885"><a href="#cb189-885" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-886"><a href="#cb189-886" aria-hidden="true" tabindex="-1"></a>md_groups <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"edaphic"</span>, <span class="st">"soil_funct"</span>, <span class="st">"temp_adapt"</span>)</span>
<span id="cb189-887"><a href="#cb189-887" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-888"><a href="#cb189-888" aria-hidden="true" tabindex="-1"></a><span class="co"># NoT uSed: minPO4, minNH4, minNO3, minTIN</span></span>
<span id="cb189-889"><a href="#cb189-889" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-890"><a href="#cb189-890" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-891"><a href="#cb189-891" aria-hidden="true" tabindex="-1"></a>And then use the lists to split the data sets by metadata  group. Here, we do this for the original metadata and the metadata after normalization.</span>
<span id="cb189-892"><a href="#cb189-892" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-895"><a href="#cb189-895" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-896"><a href="#cb189-896" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-897"><a href="#cb189-897" aria-hidden="true" tabindex="-1"></a>select_md <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ssu18_select_mc"</span>, <span class="st">"ssu18_select_mc_norm"</span>)</span>
<span id="cb189-898"><a href="#cb189-898" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> select_md) {</span>
<span id="cb189-899"><a href="#cb189-899" aria-hidden="true" tabindex="-1"></a>   <span class="co">#tmp_get &lt;- get(purrr::map_chr(i, ~ paste0(i, "_mc")))</span></span>
<span id="cb189-900"><a href="#cb189-900" aria-hidden="true" tabindex="-1"></a>   tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb189-901"><a href="#cb189-901" aria-hidden="true" tabindex="-1"></a>   tmp_md_all <span class="ot">&lt;-</span> tmp_get<span class="sc">$</span>map_loaded</span>
<span id="cb189-902"><a href="#cb189-902" aria-hidden="true" tabindex="-1"></a>   tmp_div <span class="ot">&lt;-</span> tmp_md_all <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(div))</span>
<span id="cb189-903"><a href="#cb189-903" aria-hidden="true" tabindex="-1"></a>   tmp_div <span class="ot">&lt;-</span> tmp_div <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb189-904"><a href="#cb189-904" aria-hidden="true" tabindex="-1"></a><span class="do">## edaphic      </span></span>
<span id="cb189-905"><a href="#cb189-905" aria-hidden="true" tabindex="-1"></a>   tmp_sub_edaphic <span class="ot">&lt;-</span> tmp_md_all <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(edaphic))</span>
<span id="cb189-906"><a href="#cb189-906" aria-hidden="true" tabindex="-1"></a>   tmp_sub_edaphic <span class="ot">&lt;-</span> tmp_sub_edaphic <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb189-907"><a href="#cb189-907" aria-hidden="true" tabindex="-1"></a>   tmp_edaphic <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_div, tmp_sub_edaphic, <span class="at">by =</span> <span class="st">"SampleID"</span>)</span>
<span id="cb189-908"><a href="#cb189-908" aria-hidden="true" tabindex="-1"></a>   tmp_edaphic <span class="ot">&lt;-</span> tmp_edaphic <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb189-909"><a href="#cb189-909" aria-hidden="true" tabindex="-1"></a><span class="do">## soil_funct</span></span>
<span id="cb189-910"><a href="#cb189-910" aria-hidden="true" tabindex="-1"></a>   tmp_sub_soil_funct <span class="ot">&lt;-</span> tmp_md_all <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(soil_funct))</span>
<span id="cb189-911"><a href="#cb189-911" aria-hidden="true" tabindex="-1"></a>   tmp_sub_soil_funct <span class="ot">&lt;-</span> tmp_sub_soil_funct <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb189-912"><a href="#cb189-912" aria-hidden="true" tabindex="-1"></a>   tmp_soil_funct <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_div, tmp_sub_soil_funct, <span class="at">by =</span> <span class="st">"SampleID"</span>)</span>
<span id="cb189-913"><a href="#cb189-913" aria-hidden="true" tabindex="-1"></a>   tmp_soil_funct <span class="ot">&lt;-</span> tmp_soil_funct <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"SampleID"</span>)      </span>
<span id="cb189-914"><a href="#cb189-914" aria-hidden="true" tabindex="-1"></a><span class="do">## temp_adapt</span></span>
<span id="cb189-915"><a href="#cb189-915" aria-hidden="true" tabindex="-1"></a>   tmp_sub_temp_adapt <span class="ot">&lt;-</span> tmp_md_all <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(temp_adapt))</span>
<span id="cb189-916"><a href="#cb189-916" aria-hidden="true" tabindex="-1"></a>   tmp_sub_temp_adapt <span class="ot">&lt;-</span> tmp_sub_temp_adapt <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb189-917"><a href="#cb189-917" aria-hidden="true" tabindex="-1"></a>   tmp_temp_adapt <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_div, tmp_sub_temp_adapt, <span class="at">by =</span> <span class="st">"SampleID"</span>)</span>
<span id="cb189-918"><a href="#cb189-918" aria-hidden="true" tabindex="-1"></a>   tmp_temp_adapt <span class="ot">&lt;-</span> tmp_temp_adapt <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"SampleID"</span>)      </span>
<span id="cb189-919"><a href="#cb189-919" aria-hidden="true" tabindex="-1"></a><span class="do">## combine</span></span>
<span id="cb189-920"><a href="#cb189-920" aria-hidden="true" tabindex="-1"></a>   tmp_list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">data_loaded =</span> ssu18_select_mc<span class="sc">$</span>data_loaded, </span>
<span id="cb189-921"><a href="#cb189-921" aria-hidden="true" tabindex="-1"></a>                    <span class="at">map_loaded =</span> ssu18_select_mc<span class="sc">$</span>map_loaded, </span>
<span id="cb189-922"><a href="#cb189-922" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxonomy_loaded =</span> ssu18_select_mc<span class="sc">$</span>taxonomy_loaded,</span>
<span id="cb189-923"><a href="#cb189-923" aria-hidden="true" tabindex="-1"></a>                    <span class="at">edaphic =</span> tmp_edaphic, </span>
<span id="cb189-924"><a href="#cb189-924" aria-hidden="true" tabindex="-1"></a>                    <span class="at">soil_funct =</span> tmp_soil_funct, </span>
<span id="cb189-925"><a href="#cb189-925" aria-hidden="true" tabindex="-1"></a>                    <span class="at">temp_adapt =</span> tmp_temp_adapt)</span>
<span id="cb189-926"><a href="#cb189-926" aria-hidden="true" tabindex="-1"></a>   tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_split"</span>))</span>
<span id="cb189-927"><a href="#cb189-927" aria-hidden="true" tabindex="-1"></a>   <span class="fu">print</span>(tmp_name)</span>
<span id="cb189-928"><a href="#cb189-928" aria-hidden="true" tabindex="-1"></a>   <span class="fu">assign</span>(tmp_name, tmp_list)</span>
<span id="cb189-929"><a href="#cb189-929" aria-hidden="true" tabindex="-1"></a>   <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-930"><a href="#cb189-930" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-931"><a href="#cb189-931" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-932"><a href="#cb189-932" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-933"><a href="#cb189-933" aria-hidden="true" tabindex="-1"></a><span class="fu">### Generate Autocorrelation Plots</span></span>
<span id="cb189-934"><a href="#cb189-934" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-935"><a href="#cb189-935" aria-hidden="true" tabindex="-1"></a>A little housekeeping to get rid of parameters we don't need (e.g., plot number, pair, etc.).</span>
<span id="cb189-936"><a href="#cb189-936" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-939"><a href="#cb189-939" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-940"><a href="#cb189-940" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-941"><a href="#cb189-941" aria-hidden="true" tabindex="-1"></a>edaphic_cor <span class="ot">&lt;-</span> ssu18_select_mc_split<span class="sc">$</span>edaphic</span>
<span id="cb189-942"><a href="#cb189-942" aria-hidden="true" tabindex="-1"></a>edaphic_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-943"><a href="#cb189-943" aria-hidden="true" tabindex="-1"></a>edaphic_norm_cor <span class="ot">&lt;-</span> ssu18_select_mc_norm_split<span class="sc">$</span>edaphic</span>
<span id="cb189-944"><a href="#cb189-944" aria-hidden="true" tabindex="-1"></a>edaphic_norm_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-945"><a href="#cb189-945" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-946"><a href="#cb189-946" aria-hidden="true" tabindex="-1"></a>soil_funct_cor <span class="ot">&lt;-</span> ssu18_select_mc_split<span class="sc">$</span>soil_funct</span>
<span id="cb189-947"><a href="#cb189-947" aria-hidden="true" tabindex="-1"></a>soil_funct_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-948"><a href="#cb189-948" aria-hidden="true" tabindex="-1"></a>soil_funct_norm_cor <span class="ot">&lt;-</span> ssu18_select_mc_norm_split<span class="sc">$</span>soil_funct</span>
<span id="cb189-949"><a href="#cb189-949" aria-hidden="true" tabindex="-1"></a>soil_funct_norm_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-950"><a href="#cb189-950" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-951"><a href="#cb189-951" aria-hidden="true" tabindex="-1"></a>temp_adapt_cor <span class="ot">&lt;-</span> ssu18_select_mc_split<span class="sc">$</span>temp_adapt</span>
<span id="cb189-952"><a href="#cb189-952" aria-hidden="true" tabindex="-1"></a>temp_adapt_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-953"><a href="#cb189-953" aria-hidden="true" tabindex="-1"></a>temp_adapt_norm_cor <span class="ot">&lt;-</span> ssu18_select_mc_norm_split<span class="sc">$</span>temp_adapt</span>
<span id="cb189-954"><a href="#cb189-954" aria-hidden="true" tabindex="-1"></a>temp_adapt_norm_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-955"><a href="#cb189-955" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-956"><a href="#cb189-956" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-957"><a href="#cb189-957" aria-hidden="true" tabindex="-1"></a>And finally the code to create the plots. </span>
<span id="cb189-958"><a href="#cb189-958" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-961"><a href="#cb189-961" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-962"><a href="#cb189-962" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-963"><a href="#cb189-963" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">"_cor$"</span>)) {</span>
<span id="cb189-964"><a href="#cb189-964" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb189-965"><a href="#cb189-965" aria-hidden="true" tabindex="-1"></a>  tmp_cormat <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">cor</span>(tmp_get), <span class="dv">2</span>)</span>
<span id="cb189-966"><a href="#cb189-966" aria-hidden="true" tabindex="-1"></a>  tmp_melted_cormat <span class="ot">&lt;-</span> reshape2<span class="sc">::</span><span class="fu">melt</span>(tmp_cormat)</span>
<span id="cb189-967"><a href="#cb189-967" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb189-968"><a href="#cb189-968" aria-hidden="true" tabindex="-1"></a>  tmp_get_lower_tri <span class="ot">&lt;-</span> <span class="cf">function</span>(tmp_cormat){</span>
<span id="cb189-969"><a href="#cb189-969" aria-hidden="true" tabindex="-1"></a>    tmp_cormat[<span class="fu">upper.tri</span>(tmp_cormat)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb189-970"><a href="#cb189-970" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(tmp_cormat)</span>
<span id="cb189-971"><a href="#cb189-971" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb189-972"><a href="#cb189-972" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get upper triangle of the correlation matrix</span></span>
<span id="cb189-973"><a href="#cb189-973" aria-hidden="true" tabindex="-1"></a>  tmp_get_upper_tri <span class="ot">&lt;-</span> <span class="cf">function</span>(tmp_cormat){</span>
<span id="cb189-974"><a href="#cb189-974" aria-hidden="true" tabindex="-1"></a>    tmp_cormat[<span class="fu">lower.tri</span>(tmp_cormat)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb189-975"><a href="#cb189-975" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(tmp_cormat)</span>
<span id="cb189-976"><a href="#cb189-976" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb189-977"><a href="#cb189-977" aria-hidden="true" tabindex="-1"></a>  tmp_upper_tri <span class="ot">&lt;-</span> <span class="fu">tmp_get_upper_tri</span>(tmp_cormat)</span>
<span id="cb189-978"><a href="#cb189-978" aria-hidden="true" tabindex="-1"></a>  tmp_melted_cormat <span class="ot">&lt;-</span> reshape2<span class="sc">::</span><span class="fu">melt</span>(tmp_upper_tri, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb189-979"><a href="#cb189-979" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="at">data =</span> tmp_melted_cormat, <span class="fu">aes</span>(<span class="at">x =</span> Var1, <span class="at">y =</span> Var2, <span class="at">fill =</span> value)) <span class="sc">+</span> </span>
<span id="cb189-980"><a href="#cb189-980" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>()</span>
<span id="cb189-981"><a href="#cb189-981" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb189-982"><a href="#cb189-982" aria-hidden="true" tabindex="-1"></a>  tmp_ggheatmap <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tmp_melted_cormat, <span class="fu">aes</span>(Var2, Var1, <span class="at">fill =</span> value)) <span class="sc">+</span></span>
<span id="cb189-983"><a href="#cb189-983" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>(<span class="at">color =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb189-984"><a href="#cb189-984" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> <span class="st">"blue"</span>, <span class="at">high =</span> <span class="st">"red"</span>, <span class="at">mid =</span> <span class="st">"white"</span>, </span>
<span id="cb189-985"><a href="#cb189-985" aria-hidden="true" tabindex="-1"></a>                         <span class="at">midpoint =</span> <span class="dv">0</span>, <span class="at">limit =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="at">space =</span> <span class="st">"Lab"</span>, </span>
<span id="cb189-986"><a href="#cb189-986" aria-hidden="true" tabindex="-1"></a>                         <span class="at">name=</span><span class="st">"Pearson</span><span class="sc">\n</span><span class="st">Correlation"</span>) <span class="sc">+</span></span>
<span id="cb189-987"><a href="#cb189-987" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span> </span>
<span id="cb189-988"><a href="#cb189-988" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, </span>
<span id="cb189-989"><a href="#cb189-989" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">size =</span> <span class="dv">7</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb189-990"><a href="#cb189-990" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">7</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb189-991"><a href="#cb189-991" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_fixed</span>() <span class="sc">+</span> </span>
<span id="cb189-992"><a href="#cb189-992" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="fu">aes</span>(Var2, Var1, <span class="at">label =</span> value), <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="fl">1.75</span>) <span class="sc">+</span></span>
<span id="cb189-993"><a href="#cb189-993" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb189-994"><a href="#cb189-994" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb189-995"><a href="#cb189-995" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb189-996"><a href="#cb189-996" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb189-997"><a href="#cb189-997" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb189-998"><a href="#cb189-998" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb189-999"><a href="#cb189-999" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb189-1000"><a href="#cb189-1000" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">0</span>),</span>
<span id="cb189-1001"><a href="#cb189-1001" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.6</span>, <span class="fl">0.7</span>),</span>
<span id="cb189-1002"><a href="#cb189-1002" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.direction =</span> <span class="st">"horizontal"</span>) <span class="sc">+</span></span>
<span id="cb189-1003"><a href="#cb189-1003" aria-hidden="true" tabindex="-1"></a>      <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_colorbar</span>(<span class="at">barwidth =</span> <span class="dv">7</span>, <span class="at">barheight =</span> <span class="dv">1</span>,</span>
<span id="cb189-1004"><a href="#cb189-1004" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">title.position =</span> <span class="st">"top"</span>, </span>
<span id="cb189-1005"><a href="#cb189-1005" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">title.hjust =</span> <span class="fl">0.5</span>))</span>
<span id="cb189-1006"><a href="#cb189-1006" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_ggheatmap"</span>))</span>
<span id="cb189-1007"><a href="#cb189-1007" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_ggheatmap)</span>
<span id="cb189-1008"><a href="#cb189-1008" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(tmp_name)</span>
<span id="cb189-1009"><a href="#cb189-1009" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-1010"><a href="#cb189-1010" aria-hidden="true" tabindex="-1"></a>}  </span>
<span id="cb189-1011"><a href="#cb189-1011" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">"_ggheatmap"</span>)</span>
<span id="cb189-1012"><a href="#cb189-1012" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1013"><a href="#cb189-1013" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1014"><a href="#cb189-1014" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1015"><a href="#cb189-1015" aria-hidden="true" tabindex="-1"></a><span class="fu">## Autocorrelation Plots</span></span>
<span id="cb189-1016"><a href="#cb189-1016" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br/&gt;</span></span>
<span id="cb189-1017"><a href="#cb189-1017" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1018"><a href="#cb189-1018" aria-hidden="true" tabindex="-1"></a>::: {.column-body}</span>
<span id="cb189-1019"><a href="#cb189-1019" aria-hidden="true" tabindex="-1"></a>::: {.panel-tabset}</span>
<span id="cb189-1020"><a href="#cb189-1020" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1021"><a href="#cb189-1021" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Edaphic properties</span></span>
<span id="cb189-1022"><a href="#cb189-1022" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1025"><a href="#cb189-1025" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1026"><a href="#cb189-1026" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1027"><a href="#cb189-1027" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1028"><a href="#cb189-1028" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb189-1029"><a href="#cb189-1029" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4.5</span></span>
<span id="cb189-1030"><a href="#cb189-1030" aria-hidden="true" tabindex="-1"></a>edaphic_cor_ggheatmap</span>
<span id="cb189-1031"><a href="#cb189-1031" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1032"><a href="#cb189-1032" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig_ssu("ssu_autocor_edaph_plots")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-1033"><a href="#cb189-1033" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1034"><a href="#cb189-1034" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Edaphic properties (normalized)</span></span>
<span id="cb189-1035"><a href="#cb189-1035" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1038"><a href="#cb189-1038" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1039"><a href="#cb189-1039" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1040"><a href="#cb189-1040" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1041"><a href="#cb189-1041" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb189-1042"><a href="#cb189-1042" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4.5</span></span>
<span id="cb189-1043"><a href="#cb189-1043" aria-hidden="true" tabindex="-1"></a>edaphic_norm_cor_ggheatmap</span>
<span id="cb189-1044"><a href="#cb189-1044" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1045"><a href="#cb189-1045" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig_ssu("ssu_autocor_edaph_norm_plots")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-1046"><a href="#cb189-1046" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb189-1047"><a href="#cb189-1047" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1048"><a href="#cb189-1048" aria-hidden="true" tabindex="-1"></a>::: {.panel-tabset}</span>
<span id="cb189-1049"><a href="#cb189-1049" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1050"><a href="#cb189-1050" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Functional responses</span></span>
<span id="cb189-1051"><a href="#cb189-1051" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1054"><a href="#cb189-1054" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1055"><a href="#cb189-1055" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1056"><a href="#cb189-1056" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1057"><a href="#cb189-1057" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb189-1058"><a href="#cb189-1058" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4.5</span></span>
<span id="cb189-1059"><a href="#cb189-1059" aria-hidden="true" tabindex="-1"></a>soil_funct_cor_ggheatmap</span>
<span id="cb189-1060"><a href="#cb189-1060" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1061"><a href="#cb189-1061" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig_ssu("ssu_autocor_funct_plots")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-1062"><a href="#cb189-1062" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1063"><a href="#cb189-1063" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Functional responses (normalized)</span></span>
<span id="cb189-1064"><a href="#cb189-1064" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1067"><a href="#cb189-1067" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1068"><a href="#cb189-1068" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1069"><a href="#cb189-1069" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1070"><a href="#cb189-1070" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb189-1071"><a href="#cb189-1071" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4.5</span></span>
<span id="cb189-1072"><a href="#cb189-1072" aria-hidden="true" tabindex="-1"></a>soil_funct_norm_cor_ggheatmap</span>
<span id="cb189-1073"><a href="#cb189-1073" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1074"><a href="#cb189-1074" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig_ssu("ssu_autocor_funct_norm_plots")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-1075"><a href="#cb189-1075" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb189-1076"><a href="#cb189-1076" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1077"><a href="#cb189-1077" aria-hidden="true" tabindex="-1"></a>::: {.panel-tabset}</span>
<span id="cb189-1078"><a href="#cb189-1078" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1079"><a href="#cb189-1079" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Temperature adaptation </span></span>
<span id="cb189-1080"><a href="#cb189-1080" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1083"><a href="#cb189-1083" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1084"><a href="#cb189-1084" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1085"><a href="#cb189-1085" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1086"><a href="#cb189-1086" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb189-1087"><a href="#cb189-1087" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4.5</span></span>
<span id="cb189-1088"><a href="#cb189-1088" aria-hidden="true" tabindex="-1"></a>temp_adapt_cor_ggheatmap</span>
<span id="cb189-1089"><a href="#cb189-1089" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1090"><a href="#cb189-1090" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig_ssu("ssu_autocor_temp_plots")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-1091"><a href="#cb189-1091" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1092"><a href="#cb189-1092" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Temperature adaptation (normalized)</span></span>
<span id="cb189-1093"><a href="#cb189-1093" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1096"><a href="#cb189-1096" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1097"><a href="#cb189-1097" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1098"><a href="#cb189-1098" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1099"><a href="#cb189-1099" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb189-1100"><a href="#cb189-1100" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4.5</span></span>
<span id="cb189-1101"><a href="#cb189-1101" aria-hidden="true" tabindex="-1"></a>temp_adapt_norm_cor_ggheatmap</span>
<span id="cb189-1102"><a href="#cb189-1102" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1103"><a href="#cb189-1103" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig_ssu("ssu_autocor_temp_norm_plots")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-1104"><a href="#cb189-1104" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb189-1105"><a href="#cb189-1105" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb189-1106"><a href="#cb189-1106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1109"><a href="#cb189-1109" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1110"><a href="#cb189-1110" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1111"><a href="#cb189-1111" aria-hidden="true" tabindex="-1"></a>auto_cor_figs <span class="ot">&lt;-</span> <span class="fu">ggarrange</span>(</span>
<span id="cb189-1112"><a href="#cb189-1112" aria-hidden="true" tabindex="-1"></a>  edaphic_cor_ggheatmap, edaphic_norm_cor_ggheatmap,</span>
<span id="cb189-1113"><a href="#cb189-1113" aria-hidden="true" tabindex="-1"></a>  soil_funct_cor_ggheatmap, soil_funct_norm_cor_ggheatmap,</span>
<span id="cb189-1114"><a href="#cb189-1114" aria-hidden="true" tabindex="-1"></a>  temp_adapt_cor_ggheatmap, temp_adapt_norm_cor_ggheatmap,</span>
<span id="cb189-1115"><a href="#cb189-1115" aria-hidden="true" tabindex="-1"></a>  <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">3</span>, <span class="at">common.legend =</span> <span class="cn">FALSE</span>)</span>
<span id="cb189-1116"><a href="#cb189-1116" aria-hidden="true" tabindex="-1"></a>auto_cor_figs</span>
<span id="cb189-1117"><a href="#cb189-1117" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb189-1118"><a href="#cb189-1118" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">"files/metadata/figures/ssu18_auto_cor_figs.png"</span>,</span>
<span id="cb189-1119"><a href="#cb189-1119" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">32</span>, <span class="at">width =</span> <span class="dv">24</span>, <span class="at">units =</span> <span class="st">'cm'</span>, <span class="at">res =</span> <span class="dv">600</span>, <span class="at">bg =</span> <span class="st">"white"</span>)</span>
<span id="cb189-1120"><a href="#cb189-1120" aria-hidden="true" tabindex="-1"></a>auto_cor_figs</span>
<span id="cb189-1121"><a href="#cb189-1121" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb189-1122"><a href="#cb189-1122" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"files/metadata/figures/ssu18_auto_cor_figs.pdf"</span>,</span>
<span id="cb189-1123"><a href="#cb189-1123" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">10</span>, <span class="at">width =</span> <span class="dv">12</span>)</span>
<span id="cb189-1124"><a href="#cb189-1124" aria-hidden="true" tabindex="-1"></a>auto_cor_figs</span>
<span id="cb189-1125"><a href="#cb189-1125" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb189-1126"><a href="#cb189-1126" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1127"><a href="#cb189-1127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1128"><a href="#cb189-1128" aria-hidden="true" tabindex="-1"></a>Now we can remove parameters based on the autocorrelation analysis:</span>
<span id="cb189-1129"><a href="#cb189-1129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1130"><a href="#cb189-1130" aria-hidden="true" tabindex="-1"></a>A) Environmental and edaphic properties: TEB, DON, Na, Al, Ca.</span>
<span id="cb189-1131"><a href="#cb189-1131" aria-hidden="true" tabindex="-1"></a>B) Microbial functional responses: micN, micNP, enzCN, enzCP, BP~ase~, CE~ase~, LP~ase~, N~ase~, P~ase~.</span>
<span id="cb189-1132"><a href="#cb189-1132" aria-hidden="true" tabindex="-1"></a>C) Temperature adaptation properties: NUE, PUE, SI.</span>
<span id="cb189-1133"><a href="#cb189-1133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1136"><a href="#cb189-1136" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1137"><a href="#cb189-1137" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1138"><a href="#cb189-1138" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb189-1139"><a href="#cb189-1139" aria-hidden="true" tabindex="-1"></a><span class="co"># FOR </span><span class="al">TESTING</span></span>
<span id="cb189-1140"><a href="#cb189-1140" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">"_cor_ggheatmap"</span>)</span>
<span id="cb189-1141"><a href="#cb189-1141" aria-hidden="true" tabindex="-1"></a>edaphic_norm_cor_ggheatmap</span>
<span id="cb189-1142"><a href="#cb189-1142" aria-hidden="true" tabindex="-1"></a>soil_funct_norm_cor_ggheatmap</span>
<span id="cb189-1143"><a href="#cb189-1143" aria-hidden="true" tabindex="-1"></a>temp_adapt_norm_cor_ggheatmap</span>
<span id="cb189-1144"><a href="#cb189-1144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1145"><a href="#cb189-1145" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(edaphic_norm_cor)</span>
<span id="cb189-1146"><a href="#cb189-1146" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(soil_funct_norm_cor)</span>
<span id="cb189-1147"><a href="#cb189-1147" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(temp_adapt_norm_cor)</span>
<span id="cb189-1148"><a href="#cb189-1148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1149"><a href="#cb189-1149" aria-hidden="true" tabindex="-1"></a>tmp_res <span class="ot">&lt;-</span> temp_adapt_norm_cor_ggheatmap<span class="sc">$</span>data</span>
<span id="cb189-1150"><a href="#cb189-1150" aria-hidden="true" tabindex="-1"></a>tmp_res <span class="sc">%&gt;%</span> <span class="fu">filter</span>(value <span class="sc">&gt;</span> <span class="fl">0.7</span> <span class="sc">|</span> value <span class="sc">&lt;</span> <span class="sc">-</span><span class="fl">0.7</span>) <span class="sc">%&gt;%</span> <span class="fu">filter_</span>(<span class="st">"Var1 != Var2"</span>)</span>
<span id="cb189-1151"><a href="#cb189-1151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1152"><a href="#cb189-1152" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_res %&gt;% filter(Var1 == "DOCN" | Var2 == "DOCN") %&gt;% filter_("Var1 != Var2")</span></span>
<span id="cb189-1153"><a href="#cb189-1153" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(tmp_res <span class="sc">%&gt;%</span> <span class="fu">filter</span>(value <span class="sc">&gt;</span> <span class="fl">0.7</span> <span class="sc">|</span> value <span class="sc">&lt;</span> <span class="sc">-</span><span class="fl">0.7</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb189-1154"><a href="#cb189-1154" aria-hidden="true" tabindex="-1"></a>              <span class="fu">filter_</span>(<span class="st">"Var1 != Var2"</span>), </span>
<span id="cb189-1155"><a href="#cb189-1155" aria-hidden="true" tabindex="-1"></a>            <span class="st">"soil_funct_norm7.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, </span>
<span id="cb189-1156"><a href="#cb189-1156" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb189-1157"><a href="#cb189-1157" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1158"><a href="#cb189-1158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1159"><a href="#cb189-1159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1162"><a href="#cb189-1162" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1163"><a href="#cb189-1163" aria-hidden="true" tabindex="-1"></a>edaphic_remove <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"TEB"</span>, <span class="st">"DON"</span>, <span class="st">"Na"</span>, <span class="st">"Al"</span>, <span class="st">"Ca"</span>)</span>
<span id="cb189-1164"><a href="#cb189-1164" aria-hidden="true" tabindex="-1"></a>soil_funct_remove <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"micN"</span>, <span class="st">"micNP"</span>, <span class="st">"enzCN"</span>, <span class="st">"enzCP"</span>, <span class="st">"BP_ase"</span>, </span>
<span id="cb189-1165"><a href="#cb189-1165" aria-hidden="true" tabindex="-1"></a>                       <span class="st">"CE_ase"</span>, <span class="st">"LP_ase"</span>, <span class="st">"N_ase"</span>, <span class="st">"P_ase"</span>)</span>
<span id="cb189-1166"><a href="#cb189-1166" aria-hidden="true" tabindex="-1"></a>temp_adapt_remove <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"NUE"</span>, <span class="st">"PUE"</span>, <span class="st">"SI"</span>)</span>
<span id="cb189-1167"><a href="#cb189-1167" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1168"><a href="#cb189-1168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1171"><a href="#cb189-1171" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1172"><a href="#cb189-1172" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-1173"><a href="#cb189-1173" aria-hidden="true" tabindex="-1"></a>tmp_df <span class="ot">&lt;-</span> ssu18_select_mc_split</span>
<span id="cb189-1174"><a href="#cb189-1174" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>edaphic <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>edaphic[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>edaphic) <span class="sc">%in%</span>  edaphic_remove]</span>
<span id="cb189-1175"><a href="#cb189-1175" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>soil_funct <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>soil_funct[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>soil_funct) <span class="sc">%in%</span>  soil_funct_remove]</span>
<span id="cb189-1176"><a href="#cb189-1176" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>temp_adapt <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>temp_adapt[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>temp_adapt) <span class="sc">%in%</span>  temp_adapt_remove]</span>
<span id="cb189-1177"><a href="#cb189-1177" aria-hidden="true" tabindex="-1"></a>ssu18_select_mc_split_no_ac <span class="ot">&lt;-</span> tmp_df</span>
<span id="cb189-1178"><a href="#cb189-1178" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-1179"><a href="#cb189-1179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1180"><a href="#cb189-1180" aria-hidden="true" tabindex="-1"></a>tmp_df <span class="ot">&lt;-</span> ssu18_select_mc_norm_split</span>
<span id="cb189-1181"><a href="#cb189-1181" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>edaphic <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>edaphic[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>edaphic) <span class="sc">%in%</span>  edaphic_remove]</span>
<span id="cb189-1182"><a href="#cb189-1182" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>soil_funct <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>soil_funct[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>soil_funct) <span class="sc">%in%</span>  soil_funct_remove]</span>
<span id="cb189-1183"><a href="#cb189-1183" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>temp_adapt <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>temp_adapt[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>temp_adapt) <span class="sc">%in%</span>  temp_adapt_remove]</span>
<span id="cb189-1184"><a href="#cb189-1184" aria-hidden="true" tabindex="-1"></a>ssu18_select_mc_norm_split_no_ac <span class="ot">&lt;-</span> tmp_df</span>
<span id="cb189-1185"><a href="#cb189-1185" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-1186"><a href="#cb189-1186" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1187"><a href="#cb189-1187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1188"><a href="#cb189-1188" aria-hidden="true" tabindex="-1"></a><span class="fu">## Dissimilarity Correlation Tests</span></span>
<span id="cb189-1189"><a href="#cb189-1189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1190"><a href="#cb189-1190" aria-hidden="true" tabindex="-1"></a>Let's see if any on the metadata groups are significantly correlated with the community data. Basically, we create distance matrices for the community data and each metadata group and then run Mantel tests for all comparisons. For the community data we calculate Bray-Curtis distances for the community data and  Euclidean distances for the metadata. We use the function <span class="in">`mantel.test`</span> from the <span class="in">`ape`</span> package and <span class="in">`mantel`</span> from the <span class="in">`vegan`</span> package for the analyses. </span>
<span id="cb189-1191"><a href="#cb189-1191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1192"><a href="#cb189-1192" aria-hidden="true" tabindex="-1"></a>In summary, we test both <span class="in">`mantel.test`</span> and <span class="in">`mantel`</span> on  Bray-Curtis distance community distances against Euclidean distances for each metadata group (<span class="in">`edaphic`</span>, <span class="in">`soil_funct`</span>, <span class="in">`temp_adapt`</span>)  **a**) before normalizing and before removing autocorrelated parameters, **b**) before normalizing and after removing autocorrelated parameters, **c**) after normalizing and before removing autocorrelated parameters, and **d**) after normalizing and after removing autocorrelated parameters.</span>
<span id="cb189-1193"><a href="#cb189-1193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1196"><a href="#cb189-1196" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1197"><a href="#cb189-1197" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-1198"><a href="#cb189-1198" aria-hidden="true" tabindex="-1"></a>man_df <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ssu18_select_mc_split"</span>, <span class="st">"ssu18_select_mc_split_no_ac"</span>, </span>
<span id="cb189-1199"><a href="#cb189-1199" aria-hidden="true" tabindex="-1"></a>            <span class="st">"ssu18_select_mc_norm_split"</span>, <span class="st">"ssu18_select_mc_norm_split_no_ac"</span>)</span>
<span id="cb189-1200"><a href="#cb189-1200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1201"><a href="#cb189-1201" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> man_df) {</span>
<span id="cb189-1202"><a href="#cb189-1202" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb189-1203"><a href="#cb189-1203" aria-hidden="true" tabindex="-1"></a>  tmp_dm_otu <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(<span class="fu">t</span>(tmp_get<span class="sc">$</span>data_loaded), </span>
<span id="cb189-1204"><a href="#cb189-1204" aria-hidden="true" tabindex="-1"></a>                             <span class="at">method =</span> <span class="st">"bray"</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb189-1205"><a href="#cb189-1205" aria-hidden="true" tabindex="-1"></a>                             <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb189-1206"><a href="#cb189-1206" aria-hidden="true" tabindex="-1"></a><span class="co"># EDAPHIC</span></span>
<span id="cb189-1207"><a href="#cb189-1207" aria-hidden="true" tabindex="-1"></a>  tmp_dm_md_edaphic <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_get<span class="sc">$</span>edaphic[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_get<span class="sc">$</span>edaphic)], </span>
<span id="cb189-1208"><a href="#cb189-1208" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">method =</span><span class="st">"euclidean"</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb189-1209"><a href="#cb189-1209" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb189-1210"><a href="#cb189-1210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1211"><a href="#cb189-1211" aria-hidden="true" tabindex="-1"></a>  tmp_man1_edaphic <span class="ot">&lt;-</span> <span class="fu">mantel.test</span>(tmp_dm_otu, tmp_dm_md_edaphic, <span class="at">nperm =</span> <span class="dv">999</span>, </span>
<span id="cb189-1212"><a href="#cb189-1212" aria-hidden="true" tabindex="-1"></a>                             <span class="at">graph =</span> <span class="cn">FALSE</span>, <span class="at">alternative =</span> <span class="st">"two.sided"</span>)</span>
<span id="cb189-1213"><a href="#cb189-1213" aria-hidden="true" tabindex="-1"></a>  tmp_man2_edaphic <span class="ot">&lt;-</span> <span class="fu">mantel</span>(tmp_dm_otu, tmp_dm_md_edaphic, <span class="at">permutations =</span> <span class="dv">999</span>)</span>
<span id="cb189-1214"><a href="#cb189-1214" aria-hidden="true" tabindex="-1"></a><span class="co"># SOIL FUNCT</span></span>
<span id="cb189-1215"><a href="#cb189-1215" aria-hidden="true" tabindex="-1"></a>  tmp_dm_md_soil_funct <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_get<span class="sc">$</span>soil_funct[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_get<span class="sc">$</span>soil_funct)], </span>
<span id="cb189-1216"><a href="#cb189-1216" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">method =</span> <span class="st">"euclidean"</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb189-1217"><a href="#cb189-1217" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb189-1218"><a href="#cb189-1218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1219"><a href="#cb189-1219" aria-hidden="true" tabindex="-1"></a>  tmp_man1_soil_funct <span class="ot">&lt;-</span> <span class="fu">mantel.test</span>(tmp_dm_otu, tmp_dm_md_soil_funct, <span class="at">nperm =</span> <span class="dv">999</span>, </span>
<span id="cb189-1220"><a href="#cb189-1220" aria-hidden="true" tabindex="-1"></a>                                <span class="at">graph =</span> <span class="cn">FALSE</span>, <span class="at">alternative =</span> <span class="st">"two.sided"</span>)</span>
<span id="cb189-1221"><a href="#cb189-1221" aria-hidden="true" tabindex="-1"></a>  tmp_man2_soil_funct <span class="ot">&lt;-</span> <span class="fu">mantel</span>(tmp_dm_otu, tmp_dm_md_soil_funct, <span class="at">permutations =</span> <span class="dv">999</span>)</span>
<span id="cb189-1222"><a href="#cb189-1222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1223"><a href="#cb189-1223" aria-hidden="true" tabindex="-1"></a><span class="co"># TEMP ADAPT</span></span>
<span id="cb189-1224"><a href="#cb189-1224" aria-hidden="true" tabindex="-1"></a>  tmp_dm_md_temp_adapt <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_get<span class="sc">$</span>temp_adapt[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_get<span class="sc">$</span>temp_adapt)], </span>
<span id="cb189-1225"><a href="#cb189-1225" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">method =</span> <span class="st">"euclidean"</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb189-1226"><a href="#cb189-1226" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb189-1227"><a href="#cb189-1227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1228"><a href="#cb189-1228" aria-hidden="true" tabindex="-1"></a>  tmp_man1_temp_adapt <span class="ot">&lt;-</span> <span class="fu">mantel.test</span>(tmp_dm_otu, tmp_dm_md_temp_adapt, <span class="at">nperm =</span> <span class="dv">999</span>, </span>
<span id="cb189-1229"><a href="#cb189-1229" aria-hidden="true" tabindex="-1"></a>                                <span class="at">graph =</span> <span class="cn">FALSE</span>, <span class="at">alternative =</span> <span class="st">"two.sided"</span>)</span>
<span id="cb189-1230"><a href="#cb189-1230" aria-hidden="true" tabindex="-1"></a>  tmp_man2_temp_adapt <span class="ot">&lt;-</span> <span class="fu">mantel</span>(tmp_dm_otu, tmp_dm_md_temp_adapt, <span class="at">permutations =</span> <span class="dv">999</span>)</span>
<span id="cb189-1231"><a href="#cb189-1231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1232"><a href="#cb189-1232" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_mantel_tests"</span>))</span>
<span id="cb189-1233"><a href="#cb189-1233" aria-hidden="true" tabindex="-1"></a>  tmp_df <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">edaphic_ape_man =</span> tmp_man1_edaphic, </span>
<span id="cb189-1234"><a href="#cb189-1234" aria-hidden="true" tabindex="-1"></a>                 <span class="at">edaphic_vegan_man =</span> tmp_man2_edaphic,</span>
<span id="cb189-1235"><a href="#cb189-1235" aria-hidden="true" tabindex="-1"></a>                 <span class="at">soil_funct_ape_man =</span> tmp_man1_soil_funct, </span>
<span id="cb189-1236"><a href="#cb189-1236" aria-hidden="true" tabindex="-1"></a>                 <span class="at">soil_funct_vegan_man =</span> tmp_man2_soil_funct,</span>
<span id="cb189-1237"><a href="#cb189-1237" aria-hidden="true" tabindex="-1"></a>                 <span class="at">temp_adapt_ape_man =</span> tmp_man1_temp_adapt, </span>
<span id="cb189-1238"><a href="#cb189-1238" aria-hidden="true" tabindex="-1"></a>                 <span class="at">temp_adapt_vegan_man =</span> tmp_man2_temp_adapt)</span>
<span id="cb189-1239"><a href="#cb189-1239" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_df)</span>
<span id="cb189-1240"><a href="#cb189-1240" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(tmp_name)</span>
<span id="cb189-1241"><a href="#cb189-1241" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-1242"><a href="#cb189-1242" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-1243"><a href="#cb189-1243" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1244"><a href="#cb189-1244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1247"><a href="#cb189-1247" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1248"><a href="#cb189-1248" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1249"><a href="#cb189-1249" aria-hidden="true" tabindex="-1"></a>tmp_objects <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ssu18_select_mc_split_mantel_tests"</span>, </span>
<span id="cb189-1250"><a href="#cb189-1250" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"ssu18_select_mc_split_no_ac_mantel_tests"</span>,</span>
<span id="cb189-1251"><a href="#cb189-1251" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"ssu18_select_mc_norm_split_mantel_tests"</span>,</span>
<span id="cb189-1252"><a href="#cb189-1252" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"ssu18_select_mc_norm_split_no_ac_mantel_tests"</span>)</span>
<span id="cb189-1253"><a href="#cb189-1253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1254"><a href="#cb189-1254" aria-hidden="true" tabindex="-1"></a>tmp_norm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">c</span>(<span class="st">"no"</span>, <span class="st">"no"</span>, <span class="st">"yes"</span>, <span class="st">"yes"</span>))</span>
<span id="cb189-1255"><a href="#cb189-1255" aria-hidden="true" tabindex="-1"></a>tmp_ac <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">c</span>(<span class="st">"no"</span>, <span class="st">"yes"</span>, <span class="st">"no"</span>, <span class="st">"yes"</span>))</span>
<span id="cb189-1256"><a href="#cb189-1256" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_mantel_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(tmp_norm, tmp_ac) <span class="sc">%&gt;%</span></span>
<span id="cb189-1257"><a href="#cb189-1257" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"normalized?"</span> <span class="ot">=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb189-1258"><a href="#cb189-1258" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"AC removed?"</span> <span class="ot">=</span> <span class="dv">2</span>)</span>
<span id="cb189-1259"><a href="#cb189-1259" aria-hidden="true" tabindex="-1"></a>tmp_pvalue <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">c</span>(<span class="fu">get</span>(tmp_objects[<span class="dv">1</span>])<span class="sc">$</span>edaphic_ape_man<span class="sc">$</span>p,</span>
<span id="cb189-1260"><a href="#cb189-1260" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">2</span>])<span class="sc">$</span>edaphic_ape_man<span class="sc">$</span>p,</span>
<span id="cb189-1261"><a href="#cb189-1261" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">3</span>])<span class="sc">$</span>edaphic_ape_man<span class="sc">$</span>p,</span>
<span id="cb189-1262"><a href="#cb189-1262" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">4</span>])<span class="sc">$</span>edaphic_ape_man<span class="sc">$</span>p))</span>
<span id="cb189-1263"><a href="#cb189-1263" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_mantel_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(ssu18_ps_pime_mantel_summary, tmp_pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb189-1264"><a href="#cb189-1264" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"edaphic_ape"</span> <span class="ot">=</span> <span class="dv">3</span>)</span>
<span id="cb189-1265"><a href="#cb189-1265" aria-hidden="true" tabindex="-1"></a>tmp_pvalue <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">c</span>(<span class="fu">get</span>(tmp_objects[<span class="dv">1</span>])<span class="sc">$</span>edaphic_vegan_man<span class="sc">$</span>signif,</span>
<span id="cb189-1266"><a href="#cb189-1266" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">2</span>])<span class="sc">$</span>edaphic_vegan_man<span class="sc">$</span>signif,</span>
<span id="cb189-1267"><a href="#cb189-1267" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">3</span>])<span class="sc">$</span>edaphic_vegan_man<span class="sc">$</span>signif,</span>
<span id="cb189-1268"><a href="#cb189-1268" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">4</span>])<span class="sc">$</span>edaphic_vegan_man<span class="sc">$</span>signif))</span>
<span id="cb189-1269"><a href="#cb189-1269" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_mantel_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(ssu18_ps_pime_mantel_summary, tmp_pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb189-1270"><a href="#cb189-1270" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"edaphic_vegan"</span> <span class="ot">=</span> <span class="dv">4</span>)</span>
<span id="cb189-1271"><a href="#cb189-1271" aria-hidden="true" tabindex="-1"></a>tmp_pvalue <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">c</span>(<span class="fu">get</span>(tmp_objects[<span class="dv">1</span>])<span class="sc">$</span>soil_funct_ape_man<span class="sc">$</span>p,</span>
<span id="cb189-1272"><a href="#cb189-1272" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">2</span>])<span class="sc">$</span>soil_funct_ape_man<span class="sc">$</span>p,</span>
<span id="cb189-1273"><a href="#cb189-1273" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">3</span>])<span class="sc">$</span>soil_funct_ape_man<span class="sc">$</span>p,</span>
<span id="cb189-1274"><a href="#cb189-1274" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">4</span>])<span class="sc">$</span>soil_funct_ape_man<span class="sc">$</span>p))</span>
<span id="cb189-1275"><a href="#cb189-1275" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_mantel_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(ssu18_ps_pime_mantel_summary, tmp_pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb189-1276"><a href="#cb189-1276" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"soil_funct_ape"</span> <span class="ot">=</span> <span class="dv">5</span>)</span>
<span id="cb189-1277"><a href="#cb189-1277" aria-hidden="true" tabindex="-1"></a>tmp_pvalue <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">c</span>(<span class="fu">get</span>(tmp_objects[<span class="dv">1</span>])<span class="sc">$</span>soil_funct_vegan_man<span class="sc">$</span>signif,</span>
<span id="cb189-1278"><a href="#cb189-1278" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">2</span>])<span class="sc">$</span>soil_funct_vegan_man<span class="sc">$</span>signif,</span>
<span id="cb189-1279"><a href="#cb189-1279" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">3</span>])<span class="sc">$</span>soil_funct_vegan_man<span class="sc">$</span>signif,</span>
<span id="cb189-1280"><a href="#cb189-1280" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">4</span>])<span class="sc">$</span>soil_funct_vegan_man<span class="sc">$</span>signif))</span>
<span id="cb189-1281"><a href="#cb189-1281" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_mantel_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(ssu18_ps_pime_mantel_summary, tmp_pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb189-1282"><a href="#cb189-1282" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"soil_funct_vegan"</span> <span class="ot">=</span> <span class="dv">6</span>)</span>
<span id="cb189-1283"><a href="#cb189-1283" aria-hidden="true" tabindex="-1"></a>tmp_pvalue <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">c</span>(<span class="fu">get</span>(tmp_objects[<span class="dv">1</span>])<span class="sc">$</span>temp_adapt_ape_man<span class="sc">$</span>p,</span>
<span id="cb189-1284"><a href="#cb189-1284" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">2</span>])<span class="sc">$</span>temp_adapt_ape_man<span class="sc">$</span>p,</span>
<span id="cb189-1285"><a href="#cb189-1285" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">3</span>])<span class="sc">$</span>temp_adapt_ape_man<span class="sc">$</span>p,</span>
<span id="cb189-1286"><a href="#cb189-1286" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">4</span>])<span class="sc">$</span>temp_adapt_ape_man<span class="sc">$</span>p))</span>
<span id="cb189-1287"><a href="#cb189-1287" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_mantel_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(ssu18_ps_pime_mantel_summary, tmp_pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb189-1288"><a href="#cb189-1288" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"temp_adapt_ape"</span> <span class="ot">=</span> <span class="dv">7</span>)</span>
<span id="cb189-1289"><a href="#cb189-1289" aria-hidden="true" tabindex="-1"></a>tmp_pvalue <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">c</span>(<span class="fu">get</span>(tmp_objects[<span class="dv">1</span>])<span class="sc">$</span>temp_adapt_vegan_man<span class="sc">$</span>signif,</span>
<span id="cb189-1290"><a href="#cb189-1290" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">2</span>])<span class="sc">$</span>temp_adapt_vegan_man<span class="sc">$</span>signif,</span>
<span id="cb189-1291"><a href="#cb189-1291" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">3</span>])<span class="sc">$</span>temp_adapt_vegan_man<span class="sc">$</span>signif,</span>
<span id="cb189-1292"><a href="#cb189-1292" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">4</span>])<span class="sc">$</span>temp_adapt_vegan_man<span class="sc">$</span>signif))</span>
<span id="cb189-1293"><a href="#cb189-1293" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_mantel_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(ssu18_ps_pime_mantel_summary, tmp_pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb189-1294"><a href="#cb189-1294" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"temp_adapt_vegan"</span> <span class="ot">=</span> <span class="dv">8</span>)</span>
<span id="cb189-1295"><a href="#cb189-1295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1296"><a href="#cb189-1296" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-1297"><a href="#cb189-1297" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1298"><a href="#cb189-1298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1299"><a href="#cb189-1299" aria-hidden="true" tabindex="-1"></a><span class="fu">### Dissimilarity Correlation Results</span></span>
<span id="cb189-1300"><a href="#cb189-1300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1301"><a href="#cb189-1301" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br/&gt;</span></span>
<span id="cb189-1302"><a href="#cb189-1302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1303"><a href="#cb189-1303" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab_ssu("ssu_mantel_summ")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-1304"><a href="#cb189-1304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1305"><a href="#cb189-1305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1308"><a href="#cb189-1308" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1309"><a href="#cb189-1309" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1310"><a href="#cb189-1310" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1311"><a href="#cb189-1311" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> ssu18_ps_pime_mantel_summary</span>
<span id="cb189-1312"><a href="#cb189-1312" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1313"><a href="#cb189-1313" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb189-1314"><a href="#cb189-1314" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb189-1315"><a href="#cb189-1315" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb189-1316"><a href="#cb189-1316" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">3</span>),</span>
<span id="cb189-1317"><a href="#cb189-1317" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">FALSE</span>, <span class="at">sortable =</span> <span class="cn">FALSE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>, <span class="at">minWidth =</span> <span class="dv">50</span>,</span>
<span id="cb189-1318"><a href="#cb189-1318" aria-hidden="true" tabindex="-1"></a>    <span class="at">style =</span> <span class="cf">function</span>(value) {</span>
<span id="cb189-1319"><a href="#cb189-1319" aria-hidden="true" tabindex="-1"></a>                                         <span class="cf">if</span> (value <span class="sc">&gt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb189-1320"><a href="#cb189-1320" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#2271B2"</span> <span class="co">#green</span></span>
<span id="cb189-1321"><a href="#cb189-1321" aria-hidden="true" tabindex="-1"></a>                                         } <span class="cf">else</span> <span class="cf">if</span> (value <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb189-1322"><a href="#cb189-1322" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#B22271"</span> <span class="co">#RED "#", "#"</span></span>
<span id="cb189-1323"><a href="#cb189-1323" aria-hidden="true" tabindex="-1"></a>                                         } <span class="cf">else</span> {</span>
<span id="cb189-1324"><a href="#cb189-1324" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#777"</span></span>
<span id="cb189-1325"><a href="#cb189-1325" aria-hidden="true" tabindex="-1"></a>                                         }</span>
<span id="cb189-1326"><a href="#cb189-1326" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">list</span>(<span class="at">color =</span> color)}</span>
<span id="cb189-1327"><a href="#cb189-1327" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb189-1328"><a href="#cb189-1328" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb189-1329"><a href="#cb189-1329" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb189-1330"><a href="#cb189-1330" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">normalized?</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"normalized?"</span>, </span>
<span id="cb189-1331"><a href="#cb189-1331" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>),</span>
<span id="cb189-1332"><a href="#cb189-1332" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>), </span>
<span id="cb189-1333"><a href="#cb189-1333" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb189-1334"><a href="#cb189-1334" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">70</span></span>
<span id="cb189-1335"><a href="#cb189-1335" aria-hidden="true" tabindex="-1"></a>                       ),</span>
<span id="cb189-1336"><a href="#cb189-1336" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">AC removed?</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"AC removed?"</span>, </span>
<span id="cb189-1337"><a href="#cb189-1337" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb189-1338"><a href="#cb189-1338" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">70</span>,</span>
<span id="cb189-1339"><a href="#cb189-1339" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>), </span>
<span id="cb189-1340"><a href="#cb189-1340" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>)</span>
<span id="cb189-1341"><a href="#cb189-1341" aria-hidden="true" tabindex="-1"></a>                       ),</span>
<span id="cb189-1342"><a href="#cb189-1342" aria-hidden="true" tabindex="-1"></a>    <span class="at">edaphic_ape =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"ape"</span>, </span>
<span id="cb189-1343"><a href="#cb189-1343" aria-hidden="true" tabindex="-1"></a>                                ),</span>
<span id="cb189-1344"><a href="#cb189-1344" aria-hidden="true" tabindex="-1"></a>    <span class="at">edaphic_vegan =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"vegan"</span></span>
<span id="cb189-1345"><a href="#cb189-1345" aria-hidden="true" tabindex="-1"></a>                           ),</span>
<span id="cb189-1346"><a href="#cb189-1346" aria-hidden="true" tabindex="-1"></a>    <span class="at">soil_funct_ape =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"ape"</span></span>
<span id="cb189-1347"><a href="#cb189-1347" aria-hidden="true" tabindex="-1"></a>                       ),</span>
<span id="cb189-1348"><a href="#cb189-1348" aria-hidden="true" tabindex="-1"></a>    <span class="at">soil_funct_vegan =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"vegan"</span></span>
<span id="cb189-1349"><a href="#cb189-1349" aria-hidden="true" tabindex="-1"></a>                              ),</span>
<span id="cb189-1350"><a href="#cb189-1350" aria-hidden="true" tabindex="-1"></a>    <span class="at">temp_adapt_ape =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"ape"</span></span>
<span id="cb189-1351"><a href="#cb189-1351" aria-hidden="true" tabindex="-1"></a>                       ),</span>
<span id="cb189-1352"><a href="#cb189-1352" aria-hidden="true" tabindex="-1"></a>    <span class="at">temp_adapt_vegan =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"vegan"</span></span>
<span id="cb189-1353"><a href="#cb189-1353" aria-hidden="true" tabindex="-1"></a>                              )</span>
<span id="cb189-1354"><a href="#cb189-1354" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb189-1355"><a href="#cb189-1355" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb189-1356"><a href="#cb189-1356" aria-hidden="true" tabindex="-1"></a>    <span class="at">columnGroups =</span> <span class="fu">list</span>(</span>
<span id="cb189-1357"><a href="#cb189-1357" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colGroup</span>(<span class="at">name =</span> <span class="st">"Environmental &amp; edaphic"</span>, </span>
<span id="cb189-1358"><a href="#cb189-1358" aria-hidden="true" tabindex="-1"></a>             <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"edaphic_ape"</span>, <span class="st">"edaphic_vegan"</span>), </span>
<span id="cb189-1359"><a href="#cb189-1359" aria-hidden="true" tabindex="-1"></a>             <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"1.2em"</span></span>
<span id="cb189-1360"><a href="#cb189-1360" aria-hidden="true" tabindex="-1"></a>                                ),</span>
<span id="cb189-1361"><a href="#cb189-1361" aria-hidden="true" tabindex="-1"></a>             ),</span>
<span id="cb189-1362"><a href="#cb189-1362" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colGroup</span>(<span class="at">name =</span> <span class="st">"Functional responses"</span>, </span>
<span id="cb189-1363"><a href="#cb189-1363" aria-hidden="true" tabindex="-1"></a>             <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"soil_funct_ape"</span>, <span class="st">"soil_funct_vegan"</span>), </span>
<span id="cb189-1364"><a href="#cb189-1364" aria-hidden="true" tabindex="-1"></a>             <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"1.2em"</span></span>
<span id="cb189-1365"><a href="#cb189-1365" aria-hidden="true" tabindex="-1"></a>                                )</span>
<span id="cb189-1366"><a href="#cb189-1366" aria-hidden="true" tabindex="-1"></a>             ),</span>
<span id="cb189-1367"><a href="#cb189-1367" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colGroup</span>(<span class="at">name =</span> <span class="st">"Temperature adaptation"</span>, </span>
<span id="cb189-1368"><a href="#cb189-1368" aria-hidden="true" tabindex="-1"></a>             <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"temp_adapt_ape"</span>, <span class="st">"temp_adapt_vegan"</span>), </span>
<span id="cb189-1369"><a href="#cb189-1369" aria-hidden="true" tabindex="-1"></a>             <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"1.2em"</span>)</span>
<span id="cb189-1370"><a href="#cb189-1370" aria-hidden="true" tabindex="-1"></a>             )</span>
<span id="cb189-1371"><a href="#cb189-1371" aria-hidden="true" tabindex="-1"></a>      ),</span>
<span id="cb189-1372"><a href="#cb189-1372" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">FALSE</span>, <span class="at">defaultPageSize =</span> <span class="dv">15</span>, <span class="at">showPagination =</span> <span class="cn">FALSE</span>,</span>
<span id="cb189-1373"><a href="#cb189-1373" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb189-1374"><a href="#cb189-1374" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb189-1375"><a href="#cb189-1375" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">TRUE</span>, </span>
<span id="cb189-1376"><a href="#cb189-1376" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">FALSE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-1377"><a href="#cb189-1377" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb189-1378"><a href="#cb189-1378" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(seq_table)</span>
<span id="cb189-1379"><a href="#cb189-1379" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1380"><a href="#cb189-1380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1381"><a href="#cb189-1381" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1382"><a href="#cb189-1382" aria-hidden="true" tabindex="-1"></a>Moving on.</span>
<span id="cb189-1383"><a href="#cb189-1383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1384"><a href="#cb189-1384" aria-hidden="true" tabindex="-1"></a><span class="fu">## Best Subset of Variables</span></span>
<span id="cb189-1385"><a href="#cb189-1385" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1386"><a href="#cb189-1386" aria-hidden="true" tabindex="-1"></a>Now we want to know which of the metadata parameters are the most strongly correlated with the community data. For this we use the <span class="in">`bioenv`</span> function from the <span class="in">`vegan`</span> package. <span class="in">`bioenv`</span>---*Best Subset of Environmental Variables with Maximum (Rank) Correlation with Community Dissimilarities*---finds the best subset of environmental variables, so that the Euclidean distances of scaled environmental variables have the maximum (rank) correlation with community dissimilarities. </span>
<span id="cb189-1387"><a href="#cb189-1387" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1388"><a href="#cb189-1388" aria-hidden="true" tabindex="-1"></a>Next we will use the metadata set where autocorrelated parameters were removed and the remainder of the parameters were normalized (where applicable based on the Shapiro tests). </span>
<span id="cb189-1389"><a href="#cb189-1389" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1390"><a href="#cb189-1390" aria-hidden="true" tabindex="-1"></a>We run <span class="in">`bioenv`</span> against the three groups of metadata parameters. We then run <span class="in">`bioenv`</span> again, but this time against the individual parameters identified as significantly correlated.</span>
<span id="cb189-1391"><a href="#cb189-1391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1392"><a href="#cb189-1392" aria-hidden="true" tabindex="-1"></a><span class="fu">### Edaphic Properties</span></span>
<span id="cb189-1393"><a href="#cb189-1393" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1396"><a href="#cb189-1396" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1397"><a href="#cb189-1397" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-1398"><a href="#cb189-1398" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb189-1399"><a href="#cb189-1399" aria-hidden="true" tabindex="-1"></a>tmp_env <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>edaphic)</span>
<span id="cb189-1400"><a href="#cb189-1400" aria-hidden="true" tabindex="-1"></a>tmp_env[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-1401"><a href="#cb189-1401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1402"><a href="#cb189-1402" aria-hidden="true" tabindex="-1"></a>edaphic_bioenv <span class="ot">&lt;-</span> <span class="fu">bioenv</span>(<span class="fu">wisconsin</span>(tmp_comm), tmp_env, </span>
<span id="cb189-1403"><a href="#cb189-1403" aria-hidden="true" tabindex="-1"></a>                         <span class="at">method =</span> <span class="st">"spearman"</span>, <span class="at">index =</span> <span class="st">"bray"</span>, </span>
<span id="cb189-1404"><a href="#cb189-1404" aria-hidden="true" tabindex="-1"></a>                         <span class="at">upto =</span> <span class="fu">ncol</span>(tmp_env), <span class="at">metric =</span> <span class="st">"euclidean"</span>)</span>
<span id="cb189-1405"><a href="#cb189-1405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1406"><a href="#cb189-1406" aria-hidden="true" tabindex="-1"></a>bioenv_list <span class="ot">&lt;-</span> edaphic_bioenv<span class="sc">$</span>models[[edaphic_bioenv<span class="sc">$</span>whichbest]]<span class="sc">$</span>best</span>
<span id="cb189-1407"><a href="#cb189-1407" aria-hidden="true" tabindex="-1"></a>bioenv_best <span class="ot">&lt;-</span> <span class="fu">bioenvdist</span>(edaphic_bioenv, <span class="at">which =</span> <span class="st">"best"</span>)</span>
<span id="cb189-1408"><a href="#cb189-1408" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1409"><a href="#cb189-1409" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> bioenv_list) {</span>
<span id="cb189-1410"><a href="#cb189-1410" aria-hidden="true" tabindex="-1"></a>       tmp_dp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(edaphic_bioenv<span class="sc">$</span>x)</span>
<span id="cb189-1411"><a href="#cb189-1411" aria-hidden="true" tabindex="-1"></a>       tmp_md <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_dp[[i]], </span>
<span id="cb189-1412"><a href="#cb189-1412" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">"euclidean"</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb189-1413"><a href="#cb189-1413" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb189-1414"><a href="#cb189-1414" aria-hidden="true" tabindex="-1"></a>       tmp_man <span class="ot">&lt;-</span> <span class="fu">mantel</span>(bioenv_best, tmp_md, </span>
<span id="cb189-1415"><a href="#cb189-1415" aria-hidden="true" tabindex="-1"></a>                         <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb189-1416"><a href="#cb189-1416" aria-hidden="true" tabindex="-1"></a>       tmp_md_name <span class="ot">&lt;-</span> <span class="fu">names</span>(tmp_dp)[[i]]</span>
<span id="cb189-1417"><a href="#cb189-1417" aria-hidden="true" tabindex="-1"></a>       tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(tmp_md_name, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_bioenv_mantel_test"</span>))</span>
<span id="cb189-1418"><a href="#cb189-1418" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_name, tmp_man)</span>
<span id="cb189-1419"><a href="#cb189-1419" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-1420"><a href="#cb189-1420" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-1421"><a href="#cb189-1421" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">"_bioenv_mantel_test"</span>)</span>
<span id="cb189-1422"><a href="#cb189-1422" aria-hidden="true" tabindex="-1"></a>edaphic_bioenv_ind_mantel <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">AST =</span> AST_bioenv_mantel_test)</span>
<span id="cb189-1423"><a href="#cb189-1423" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"_bioenv_mantel_test"</span>))</span>
<span id="cb189-1424"><a href="#cb189-1424" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1425"><a href="#cb189-1425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1428"><a href="#cb189-1428" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1429"><a href="#cb189-1429" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1430"><a href="#cb189-1430" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1431"><a href="#cb189-1431" aria-hidden="true" tabindex="-1"></a>edaphic_bioenv</span>
<span id="cb189-1432"><a href="#cb189-1432" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1433"><a href="#cb189-1433" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1434"><a href="#cb189-1434" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb189-1435"><a href="#cb189-1435" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show the results of individual edaphic metadata Mantel tests<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb189-1436"><a href="#cb189-1436" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1439"><a href="#cb189-1439" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1440"><a href="#cb189-1440" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1441"><a href="#cb189-1441" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1442"><a href="#cb189-1442" aria-hidden="true" tabindex="-1"></a>edaphic_bioenv_ind_mantel</span>
<span id="cb189-1443"><a href="#cb189-1443" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1444"><a href="#cb189-1444" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb189-1445"><a href="#cb189-1445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1446"><a href="#cb189-1446" aria-hidden="true" tabindex="-1"></a><span class="in">`bioenv`</span> found the following edaphic properties significantly correlated with the community data: </span>
<span id="cb189-1447"><a href="#cb189-1447" aria-hidden="true" tabindex="-1"></a>**`r row.names(summary(edaphic_bioenv_ind_mantel))`**</span>
<span id="cb189-1448"><a href="#cb189-1448" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1449"><a href="#cb189-1449" aria-hidden="true" tabindex="-1"></a><span class="fu">### Soil Functional Response</span></span>
<span id="cb189-1450"><a href="#cb189-1450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1453"><a href="#cb189-1453" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1454"><a href="#cb189-1454" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-1455"><a href="#cb189-1455" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb189-1456"><a href="#cb189-1456" aria-hidden="true" tabindex="-1"></a>tmp_env <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>soil_funct)</span>
<span id="cb189-1457"><a href="#cb189-1457" aria-hidden="true" tabindex="-1"></a>tmp_env[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-1458"><a href="#cb189-1458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1459"><a href="#cb189-1459" aria-hidden="true" tabindex="-1"></a>soil_funct_bioenv <span class="ot">&lt;-</span> <span class="fu">bioenv</span>(<span class="fu">wisconsin</span>(tmp_comm), tmp_env, </span>
<span id="cb189-1460"><a href="#cb189-1460" aria-hidden="true" tabindex="-1"></a>                         <span class="at">method =</span> <span class="st">"spearman"</span>, <span class="at">index =</span> <span class="st">"bray"</span>, </span>
<span id="cb189-1461"><a href="#cb189-1461" aria-hidden="true" tabindex="-1"></a>                         <span class="at">upto =</span> <span class="fu">ncol</span>(tmp_env), <span class="at">metric =</span> <span class="st">"euclidean"</span>)</span>
<span id="cb189-1462"><a href="#cb189-1462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1463"><a href="#cb189-1463" aria-hidden="true" tabindex="-1"></a>bioenv_list <span class="ot">&lt;-</span> soil_funct_bioenv<span class="sc">$</span>models[[soil_funct_bioenv<span class="sc">$</span>whichbest]]<span class="sc">$</span>best</span>
<span id="cb189-1464"><a href="#cb189-1464" aria-hidden="true" tabindex="-1"></a>bioenv_best <span class="ot">&lt;-</span> <span class="fu">bioenvdist</span>(soil_funct_bioenv, <span class="at">which =</span> <span class="st">"best"</span>)</span>
<span id="cb189-1465"><a href="#cb189-1465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1466"><a href="#cb189-1466" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> bioenv_list) {</span>
<span id="cb189-1467"><a href="#cb189-1467" aria-hidden="true" tabindex="-1"></a>       tmp_dp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(soil_funct_bioenv<span class="sc">$</span>x)</span>
<span id="cb189-1468"><a href="#cb189-1468" aria-hidden="true" tabindex="-1"></a>       tmp_md <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_dp[[i]], </span>
<span id="cb189-1469"><a href="#cb189-1469" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">"euclidean"</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb189-1470"><a href="#cb189-1470" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb189-1471"><a href="#cb189-1471" aria-hidden="true" tabindex="-1"></a>       tmp_man <span class="ot">&lt;-</span> <span class="fu">mantel</span>(bioenv_best, tmp_md, </span>
<span id="cb189-1472"><a href="#cb189-1472" aria-hidden="true" tabindex="-1"></a>                         <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb189-1473"><a href="#cb189-1473" aria-hidden="true" tabindex="-1"></a>       tmp_md_name <span class="ot">&lt;-</span> <span class="fu">names</span>(tmp_dp)[[i]]</span>
<span id="cb189-1474"><a href="#cb189-1474" aria-hidden="true" tabindex="-1"></a>       tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(tmp_md_name, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_bioenv_mantel_test"</span>))</span>
<span id="cb189-1475"><a href="#cb189-1475" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_name, tmp_man)</span>
<span id="cb189-1476"><a href="#cb189-1476" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-1477"><a href="#cb189-1477" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-1478"><a href="#cb189-1478" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">"_bioenv_mantel_test"</span>)</span>
<span id="cb189-1479"><a href="#cb189-1479" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1480"><a href="#cb189-1480" aria-hidden="true" tabindex="-1"></a>soil_funct_bioenv_ind_mantel <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">AG_ase =</span> AG_ase_bioenv_mantel_test, </span>
<span id="cb189-1481"><a href="#cb189-1481" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">enzNP =</span> enzNP_bioenv_mantel_test, </span>
<span id="cb189-1482"><a href="#cb189-1482" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">S_ase =</span> S_ase_bioenv_mantel_test, </span>
<span id="cb189-1483"><a href="#cb189-1483" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">PX_ase =</span> PX_ase_bioenv_mantel_test,</span>
<span id="cb189-1484"><a href="#cb189-1484" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">XY_ase =</span> XY_ase_bioenv_mantel_test)</span>
<span id="cb189-1485"><a href="#cb189-1485" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"_bioenv_mantel_test"</span>))</span>
<span id="cb189-1486"><a href="#cb189-1486" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1487"><a href="#cb189-1487" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1490"><a href="#cb189-1490" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1491"><a href="#cb189-1491" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1492"><a href="#cb189-1492" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1493"><a href="#cb189-1493" aria-hidden="true" tabindex="-1"></a>soil_funct_bioenv</span>
<span id="cb189-1494"><a href="#cb189-1494" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1495"><a href="#cb189-1495" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1496"><a href="#cb189-1496" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb189-1497"><a href="#cb189-1497" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show the results of individual functional response metadata Mantel tests<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb189-1498"><a href="#cb189-1498" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1501"><a href="#cb189-1501" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1502"><a href="#cb189-1502" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1503"><a href="#cb189-1503" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1504"><a href="#cb189-1504" aria-hidden="true" tabindex="-1"></a>soil_funct_bioenv_ind_mantel</span>
<span id="cb189-1505"><a href="#cb189-1505" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1506"><a href="#cb189-1506" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb189-1507"><a href="#cb189-1507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1508"><a href="#cb189-1508" aria-hidden="true" tabindex="-1"></a><span class="in">`bioenv`</span> found the following soil functions  significantly correlated with the community data: </span>
<span id="cb189-1509"><a href="#cb189-1509" aria-hidden="true" tabindex="-1"></a>**`r row.names(summary(soil_funct_bioenv_ind_mantel))`**</span>
<span id="cb189-1510"><a href="#cb189-1510" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1511"><a href="#cb189-1511" aria-hidden="true" tabindex="-1"></a><span class="fu">### Temperature Adaptation</span></span>
<span id="cb189-1512"><a href="#cb189-1512" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1515"><a href="#cb189-1515" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1516"><a href="#cb189-1516" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-1517"><a href="#cb189-1517" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb189-1518"><a href="#cb189-1518" aria-hidden="true" tabindex="-1"></a>tmp_env <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>temp_adapt)</span>
<span id="cb189-1519"><a href="#cb189-1519" aria-hidden="true" tabindex="-1"></a>tmp_env[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-1520"><a href="#cb189-1520" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1521"><a href="#cb189-1521" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioenv <span class="ot">&lt;-</span> <span class="fu">bioenv</span>(<span class="fu">wisconsin</span>(tmp_comm), tmp_env, </span>
<span id="cb189-1522"><a href="#cb189-1522" aria-hidden="true" tabindex="-1"></a>                         <span class="at">method =</span> <span class="st">"spearman"</span>, <span class="at">index =</span> <span class="st">"bray"</span>, </span>
<span id="cb189-1523"><a href="#cb189-1523" aria-hidden="true" tabindex="-1"></a>                         <span class="at">upto =</span> <span class="fu">ncol</span>(tmp_env), <span class="at">metric =</span> <span class="st">"euclidean"</span>)</span>
<span id="cb189-1524"><a href="#cb189-1524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1525"><a href="#cb189-1525" aria-hidden="true" tabindex="-1"></a>bioenv_list <span class="ot">&lt;-</span> temp_adapt_bioenv<span class="sc">$</span>models[[temp_adapt_bioenv<span class="sc">$</span>whichbest]]<span class="sc">$</span>best</span>
<span id="cb189-1526"><a href="#cb189-1526" aria-hidden="true" tabindex="-1"></a>bioenv_best <span class="ot">&lt;-</span> <span class="fu">bioenvdist</span>(temp_adapt_bioenv, <span class="at">which =</span> <span class="st">"best"</span>)</span>
<span id="cb189-1527"><a href="#cb189-1527" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1528"><a href="#cb189-1528" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> bioenv_list) {</span>
<span id="cb189-1529"><a href="#cb189-1529" aria-hidden="true" tabindex="-1"></a>       tmp_dp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(temp_adapt_bioenv<span class="sc">$</span>x)</span>
<span id="cb189-1530"><a href="#cb189-1530" aria-hidden="true" tabindex="-1"></a>       tmp_md <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_dp[[i]], </span>
<span id="cb189-1531"><a href="#cb189-1531" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">"euclidean"</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb189-1532"><a href="#cb189-1532" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb189-1533"><a href="#cb189-1533" aria-hidden="true" tabindex="-1"></a>       tmp_man <span class="ot">&lt;-</span> <span class="fu">mantel</span>(bioenv_best, tmp_md, </span>
<span id="cb189-1534"><a href="#cb189-1534" aria-hidden="true" tabindex="-1"></a>                         <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb189-1535"><a href="#cb189-1535" aria-hidden="true" tabindex="-1"></a>       tmp_md_name <span class="ot">&lt;-</span> <span class="fu">names</span>(tmp_dp)[[i]]</span>
<span id="cb189-1536"><a href="#cb189-1536" aria-hidden="true" tabindex="-1"></a>       tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(tmp_md_name, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_bioenv_mantel_test"</span>))</span>
<span id="cb189-1537"><a href="#cb189-1537" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_name, tmp_man)</span>
<span id="cb189-1538"><a href="#cb189-1538" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-1539"><a href="#cb189-1539" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-1540"><a href="#cb189-1540" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">"_bioenv_mantel_test"</span>)</span>
<span id="cb189-1541"><a href="#cb189-1541" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1542"><a href="#cb189-1542" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioenv_ind_mantel <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">CUEcp =</span> CUEcp_bioenv_mantel_test, </span>
<span id="cb189-1543"><a href="#cb189-1543" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">LP_Q10 =</span> LP_Q10_bioenv_mantel_test, </span>
<span id="cb189-1544"><a href="#cb189-1544" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">P_Q10 =</span> P_Q10_bioenv_mantel_test, </span>
<span id="cb189-1545"><a href="#cb189-1545" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">S_Q10 =</span> S_Q10_bioenv_mantel_test, </span>
<span id="cb189-1546"><a href="#cb189-1546" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">Tmin =</span> Tmin_bioenv_mantel_test)</span>
<span id="cb189-1547"><a href="#cb189-1547" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"_bioenv_mantel_test"</span>))</span>
<span id="cb189-1548"><a href="#cb189-1548" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1549"><a href="#cb189-1549" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1552"><a href="#cb189-1552" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1553"><a href="#cb189-1553" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1554"><a href="#cb189-1554" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1555"><a href="#cb189-1555" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioenv</span>
<span id="cb189-1556"><a href="#cb189-1556" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1557"><a href="#cb189-1557" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1558"><a href="#cb189-1558" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb189-1559"><a href="#cb189-1559" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show the results of individual temperature adaptation metadata Mantel tests<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb189-1560"><a href="#cb189-1560" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1563"><a href="#cb189-1563" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1564"><a href="#cb189-1564" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1565"><a href="#cb189-1565" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1566"><a href="#cb189-1566" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioenv_ind_mantel</span>
<span id="cb189-1567"><a href="#cb189-1567" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1568"><a href="#cb189-1568" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb189-1569"><a href="#cb189-1569" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1570"><a href="#cb189-1570" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1571"><a href="#cb189-1571" aria-hidden="true" tabindex="-1"></a><span class="in">`bioenv`</span> found the following temperature adaptations significantly correlated with the community data: </span>
<span id="cb189-1572"><a href="#cb189-1572" aria-hidden="true" tabindex="-1"></a>**`r row.names(summary(temp_adapt_bioenv_ind_mantel))`**</span>
<span id="cb189-1573"><a href="#cb189-1573" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1574"><a href="#cb189-1574" aria-hidden="true" tabindex="-1"></a><span class="fu">## Distance-based Redundancy</span></span>
<span id="cb189-1575"><a href="#cb189-1575" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1576"><a href="#cb189-1576" aria-hidden="true" tabindex="-1"></a>Now we turn our attention to distance-based redundancy analysis (dbRDA), an ordination method similar to Redundancy Analysis (rda) but it allows non-Euclidean dissimilarity indices, such as Manhattan or BrayCurtis distance. For this, we use <span class="in">`capscale`</span> from the <span class="in">`vegan`</span> package. <span class="in">`capscale`</span>  is a constrained versions of metric scaling (principal coordinates analysis), which are based on the Euclidean distance but can be used, and are more useful, with other dissimilarity measures. The functions can also perform unconstrained principal coordinates analysis, optionally using extended dissimilarities.</span>
<span id="cb189-1577"><a href="#cb189-1577" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1578"><a href="#cb189-1578" aria-hidden="true" tabindex="-1"></a>For each of the three metadata subsets, we perform the following steps:</span>
<span id="cb189-1579"><a href="#cb189-1579" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1580"><a href="#cb189-1580" aria-hidden="true" tabindex="-1"></a>1) Run <span class="in">`rankindex`</span> to compare metadata and community dissimilarity indices for gradient detection. This will help us select the best dissimilarity metric to use.</span>
<span id="cb189-1581"><a href="#cb189-1581" aria-hidden="true" tabindex="-1"></a>2) Run <span class="in">`capscale`</span> for distance-based redundancy analysis.</span>
<span id="cb189-1582"><a href="#cb189-1582" aria-hidden="true" tabindex="-1"></a>3) Run <span class="in">`envfit`</span> to fit environmental parameters onto the ordination. This function basically calculates correlation scores between the metadata parameters and the ordination axes. </span>
<span id="cb189-1583"><a href="#cb189-1583" aria-hidden="true" tabindex="-1"></a>4) Select metadata parameters significant for <span class="in">`bioenv`</span> (see above) and/or <span class="in">`envfit`</span> analyses.</span>
<span id="cb189-1584"><a href="#cb189-1584" aria-hidden="true" tabindex="-1"></a>5) Run <span class="in">`envfit`</span> on ASVs.</span>
<span id="cb189-1585"><a href="#cb189-1585" aria-hidden="true" tabindex="-1"></a>6) Plot the ordination and vector overlays. </span>
<span id="cb189-1586"><a href="#cb189-1586" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1587"><a href="#cb189-1587" aria-hidden="true" tabindex="-1"></a><span class="fu">### Edaphic Properties</span></span>
<span id="cb189-1588"><a href="#cb189-1588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1591"><a href="#cb189-1591" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1592"><a href="#cb189-1592" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-1593"><a href="#cb189-1593" aria-hidden="true" tabindex="-1"></a>tmp_md <span class="ot">&lt;-</span> ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>edaphic</span>
<span id="cb189-1594"><a href="#cb189-1594" aria-hidden="true" tabindex="-1"></a>tmp_md<span class="sc">$</span>TREAT_T <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tmp_md<span class="sc">$</span>TREAT_T)</span>
<span id="cb189-1595"><a href="#cb189-1595" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb189-1596"><a href="#cb189-1596" aria-hidden="true" tabindex="-1"></a>edaphic_rank <span class="ot">&lt;-</span> <span class="fu">rankindex</span>(tmp_md[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_md)], tmp_comm, </span>
<span id="cb189-1597"><a href="#cb189-1597" aria-hidden="true" tabindex="-1"></a>          <span class="at">indices =</span> <span class="fu">c</span>(<span class="st">"euc"</span>, <span class="st">"man"</span>, <span class="st">"gow"</span>,<span class="st">"bra"</span>, <span class="st">"kul"</span>), </span>
<span id="cb189-1598"><a href="#cb189-1598" aria-hidden="true" tabindex="-1"></a>          <span class="at">stepacross =</span> <span class="cn">FALSE</span>, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb189-1599"><a href="#cb189-1599" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1600"><a href="#cb189-1600" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1603"><a href="#cb189-1603" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1604"><a href="#cb189-1604" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1605"><a href="#cb189-1605" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1606"><a href="#cb189-1606" aria-hidden="true" tabindex="-1"></a>edaphic_rank</span>
<span id="cb189-1607"><a href="#cb189-1607" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1608"><a href="#cb189-1608" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1609"><a href="#cb189-1609" aria-hidden="true" tabindex="-1"></a>Let's run <span class="in">`capscale`</span> using Bray-Curtis. Note, we have <span class="in">`r ncol(ssu18_select_mc_norm_split_no_ac$edaphic) - 8`</span> metadata parameters in this group but, for some reason, <span class="in">`capscale`</span> only works with 13 parameters. This may have to do with degrees of freedom? </span>
<span id="cb189-1610"><a href="#cb189-1610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1611"><a href="#cb189-1611" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Starting properties: AST, H2O, N, P, Al, Ca, Fe, K, Mg, Mn, Na, TEB, ECEC, pH, NH4, NO3, PO4, DOC, DON, DOCN</span>
<span id="cb189-1612"><a href="#cb189-1612" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Autocorrelated removed: TEB, DON, Na, Al, Ca</span>
<span id="cb189-1613"><a href="#cb189-1613" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Remove for capscale: Mg, Mn</span>
<span id="cb189-1614"><a href="#cb189-1614" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1617"><a href="#cb189-1617" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1618"><a href="#cb189-1618" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1619"><a href="#cb189-1619" aria-hidden="true" tabindex="-1"></a><span class="co"># All edaphic: AST, H2O, N, P, Al, Ca, Fe, K, Mg, Mn, Na, TEB, ECEC, pH, NH4, NO3, PO4, DOC, DON, DOCN</span></span>
<span id="cb189-1620"><a href="#cb189-1620" aria-hidden="true" tabindex="-1"></a><span class="co"># REMOVED edaphic: TEB, DON, Na, Al, Ca</span></span>
<span id="cb189-1621"><a href="#cb189-1621" aria-hidden="true" tabindex="-1"></a><span class="co"># Remaining edaphic: AST, H2O, N, P, Fe, K, Mg, Mn, ECEC, pH, NH4, NO3, PO4, DOC, DOCN</span></span>
<span id="cb189-1622"><a href="#cb189-1622" aria-hidden="true" tabindex="-1"></a><span class="co"># 15 total, only works with 13</span></span>
<span id="cb189-1623"><a href="#cb189-1623" aria-hidden="true" tabindex="-1"></a><span class="co"># Removed  Mg + Mn</span></span>
<span id="cb189-1624"><a href="#cb189-1624" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TEST</span><span class="co"> with , metaMDSdist = TRUE AND/OR , metaMDS = TRUE</span></span>
<span id="cb189-1625"><a href="#cb189-1625" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1626"><a href="#cb189-1626" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1629"><a href="#cb189-1629" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1630"><a href="#cb189-1630" aria-hidden="true" tabindex="-1"></a>edaphic_cap <span class="ot">&lt;-</span> <span class="fu">capscale</span>(tmp_comm <span class="sc">~</span> AST <span class="sc">+</span> H2O <span class="sc">+</span> N <span class="sc">+</span> P <span class="sc">+</span> Fe <span class="sc">+</span> K <span class="sc">+</span> ECEC <span class="sc">+</span> </span>
<span id="cb189-1631"><a href="#cb189-1631" aria-hidden="true" tabindex="-1"></a>                                   pH <span class="sc">+</span> NH4 <span class="sc">+</span> NO3 <span class="sc">+</span> PO4 <span class="sc">+</span> DOC <span class="sc">+</span> DOCN, </span>
<span id="cb189-1632"><a href="#cb189-1632" aria-hidden="true" tabindex="-1"></a>                        tmp_md, <span class="at">dist =</span> <span class="st">"bray"</span>)</span>
<span id="cb189-1633"><a href="#cb189-1633" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tmp_md)</span>
<span id="cb189-1634"><a href="#cb189-1634" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1635"><a href="#cb189-1635" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1638"><a href="#cb189-1638" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1639"><a href="#cb189-1639" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1640"><a href="#cb189-1640" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1641"><a href="#cb189-1641" aria-hidden="true" tabindex="-1"></a>edaphic_cap</span>
<span id="cb189-1642"><a href="#cb189-1642" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1643"><a href="#cb189-1643" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1644"><a href="#cb189-1644" aria-hidden="true" tabindex="-1"></a>Now we can look at the variance against each principal component. </span>
<span id="cb189-1645"><a href="#cb189-1645" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1648"><a href="#cb189-1648" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1649"><a href="#cb189-1649" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1650"><a href="#cb189-1650" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1651"><a href="#cb189-1651" aria-hidden="true" tabindex="-1"></a>stats<span class="sc">::</span><span class="fu">screeplot</span>(edaphic_cap)</span>
<span id="cb189-1652"><a href="#cb189-1652" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1653"><a href="#cb189-1653" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1654"><a href="#cb189-1654" aria-hidden="true" tabindex="-1"></a>And then make some quick and dirty plots. This will also come in handy later when we need to parse out data a better plot visualization. The <span class="in">`ggplot`</span> function <span class="in">`autoplot`</span> stores these data in a more accessible way than the raw results from <span class="in">`capscale`</span></span>
<span id="cb189-1655"><a href="#cb189-1655" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1658"><a href="#cb189-1658" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1659"><a href="#cb189-1659" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1660"><a href="#cb189-1660" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">plot</span>(edaphic_cap) </span>
<span id="cb189-1661"><a href="#cb189-1661" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt <span class="ot">&lt;-</span> ggplot2<span class="sc">::</span><span class="fu">autoplot</span>(edaphic_cap, <span class="at">arrows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb189-1662"><a href="#cb189-1662" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt</span>
<span id="cb189-1663"><a href="#cb189-1663" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1664"><a href="#cb189-1664" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1667"><a href="#cb189-1667" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1668"><a href="#cb189-1668" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1669"><a href="#cb189-1669" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(edaphic_cap) <span class="co"># overall test of the significant of the analysis</span></span>
<span id="cb189-1670"><a href="#cb189-1670" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(edaphic_cap, <span class="at">by =</span> <span class="st">"axis"</span>, <span class="at">perm.max =</span> <span class="dv">500</span>) <span class="co"># test axes for significance</span></span>
<span id="cb189-1671"><a href="#cb189-1671" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(edaphic_cap, <span class="at">by =</span> <span class="st">"terms"</span>, <span class="at">permu =</span> <span class="dv">500</span>) <span class="co"># test for sign. environ. variables</span></span>
<span id="cb189-1672"><a href="#cb189-1672" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1673"><a href="#cb189-1673" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1674"><a href="#cb189-1674" aria-hidden="true" tabindex="-1"></a>Next, we need to grab capscale scores for the samples and create a data frame of the first two dimensions. We will also need to add some of the sample details to the data frame. For this we use the vegan function <span class="in">`scores`</span> which gets species or site scores from the ordination.</span>
<span id="cb189-1675"><a href="#cb189-1675" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1678"><a href="#cb189-1678" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1679"><a href="#cb189-1679" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-1680"><a href="#cb189-1680" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">"sites"</span>)</span>
<span id="cb189-1681"><a href="#cb189-1681" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-1682"><a href="#cb189-1682" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> tmp_samp_scores <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">SampleID =</span> Label)</span>
<span id="cb189-1683"><a href="#cb189-1683" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1684"><a href="#cb189-1684" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb189-1685"><a href="#cb189-1685" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md_sub <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb189-1686"><a href="#cb189-1686" aria-hidden="true" tabindex="-1"></a>edaphic_plot_data <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_md_sub, tmp_samp_scores, <span class="at">by =</span> <span class="st">"SampleID"</span>)</span>
<span id="cb189-1687"><a href="#cb189-1687" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1688"><a href="#cb189-1688" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1689"><a href="#cb189-1689" aria-hidden="true" tabindex="-1"></a>Now we have a new data frame that contains sample details and capscale values. </span>
<span id="cb189-1690"><a href="#cb189-1690" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1693"><a href="#cb189-1693" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1694"><a href="#cb189-1694" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1695"><a href="#cb189-1695" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1696"><a href="#cb189-1696" aria-hidden="true" tabindex="-1"></a>edaphic_plot_data</span>
<span id="cb189-1697"><a href="#cb189-1697" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1698"><a href="#cb189-1698" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1699"><a href="#cb189-1699" aria-hidden="true" tabindex="-1"></a>We can then do the same with the metadata vectors. Here though we only need the scores and parameter name. </span>
<span id="cb189-1700"><a href="#cb189-1700" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1703"><a href="#cb189-1703" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1704"><a href="#cb189-1704" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-1705"><a href="#cb189-1705" aria-hidden="true" tabindex="-1"></a>edaphic_md_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">"biplot"</span>)</span>
<span id="cb189-1706"><a href="#cb189-1706" aria-hidden="true" tabindex="-1"></a>edaphic_md_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-1707"><a href="#cb189-1707" aria-hidden="true" tabindex="-1"></a>edaphic_md_scores <span class="ot">&lt;-</span> edaphic_md_scores <span class="sc">%&gt;%</span> </span>
<span id="cb189-1708"><a href="#cb189-1708" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span></span>
<span id="cb189-1709"><a href="#cb189-1709" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"Label"</span>)</span>
<span id="cb189-1710"><a href="#cb189-1710" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1711"><a href="#cb189-1711" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1712"><a href="#cb189-1712" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1715"><a href="#cb189-1715" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1716"><a href="#cb189-1716" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1717"><a href="#cb189-1717" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1718"><a href="#cb189-1718" aria-hidden="true" tabindex="-1"></a>edaphic_md_scores</span>
<span id="cb189-1719"><a href="#cb189-1719" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1720"><a href="#cb189-1720" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1721"><a href="#cb189-1721" aria-hidden="true" tabindex="-1"></a>Let's run some quick correlations of metadata with ordination axes to see which parameters are significant. For this we use the vegan function <span class="in">`envfit`</span>.</span>
<span id="cb189-1722"><a href="#cb189-1722" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1725"><a href="#cb189-1725" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1726"><a href="#cb189-1726" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-1727"><a href="#cb189-1727" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> edaphic_plot_data[, <span class="dv">6</span><span class="sc">:</span><span class="dv">7</span>]</span>
<span id="cb189-1728"><a href="#cb189-1728" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_samp_scores_sub)</span>
<span id="cb189-1729"><a href="#cb189-1729" aria-hidden="true" tabindex="-1"></a>tmp_param_list <span class="ot">&lt;-</span> edaphic_md_scores<span class="sc">$</span>parameters</span>
<span id="cb189-1730"><a href="#cb189-1730" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1731"><a href="#cb189-1731" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> <span class="fu">subset</span>(tmp_md,  <span class="at">select =</span>  tmp_param_list)</span>
<span id="cb189-1732"><a href="#cb189-1732" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1733"><a href="#cb189-1733" aria-hidden="true" tabindex="-1"></a>envfit_edaphic_md <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, tmp_md_sub,</span>
<span id="cb189-1734"><a href="#cb189-1734" aria-hidden="true" tabindex="-1"></a>                 <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb189-1735"><a href="#cb189-1735" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1736"><a href="#cb189-1736" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1739"><a href="#cb189-1739" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1740"><a href="#cb189-1740" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1741"><a href="#cb189-1741" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1742"><a href="#cb189-1742" aria-hidden="true" tabindex="-1"></a>envfit_edaphic_md</span>
<span id="cb189-1743"><a href="#cb189-1743" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1744"><a href="#cb189-1744" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1747"><a href="#cb189-1747" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1748"><a href="#cb189-1748" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-1749"><a href="#cb189-1749" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_edaphic_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb189-1750"><a href="#cb189-1750" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_edaphic_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb189-1751"><a href="#cb189-1751" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_edaphic_md<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.4</span>))</span>
<span id="cb189-1752"><a href="#cb189-1752" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(edaphic_md_signif_hits)</span>
<span id="cb189-1753"><a href="#cb189-1753" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(edaphic_md_signif_hits)</span>
<span id="cb189-1754"><a href="#cb189-1754" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif <span class="ot">&lt;-</span> edaphic_md_scores[edaphic_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> edaphic_md_signif_hits,]</span>
<span id="cb189-1755"><a href="#cb189-1755" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif<span class="sc">$</span>parameters</span>
<span id="cb189-1756"><a href="#cb189-1756" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1757"><a href="#cb189-1757" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1758"><a href="#cb189-1758" aria-hidden="true" tabindex="-1"></a><span class="in">`envfit`</span> found that <span class="in">`r edaphic_md_signif$parameters`</span> were significantly correlated. </span>
<span id="cb189-1759"><a href="#cb189-1759" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1760"><a href="#cb189-1760" aria-hidden="true" tabindex="-1"></a>Now let's see if the same parameters are significant for the <span class="in">`envfit`</span> and <span class="in">`bioenv`</span> analyses.</span>
<span id="cb189-1761"><a href="#cb189-1761" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1764"><a href="#cb189-1764" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1765"><a href="#cb189-1765" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1766"><a href="#cb189-1766" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1767"><a href="#cb189-1767" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hold</span></span>
<span id="cb189-1768"><a href="#cb189-1768" aria-hidden="true" tabindex="-1"></a><span class="co">#| comment: ''</span></span>
<span id="cb189-1769"><a href="#cb189-1769" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Significant parameters from bioenv analysis."</span>)</span>
<span id="cb189-1770"><a href="#cb189-1770" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(<span class="fu">summary</span>(edaphic_bioenv_ind_mantel))</span>
<span id="cb189-1771"><a href="#cb189-1771" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-1772"><a href="#cb189-1772" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-1773"><a href="#cb189-1773" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Significant parameters from envfit analysis."</span>)</span>
<span id="cb189-1774"><a href="#cb189-1774" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif<span class="sc">$</span>parameters</span>
<span id="cb189-1775"><a href="#cb189-1775" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-1776"><a href="#cb189-1776" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-1777"><a href="#cb189-1777" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Found in bioenv but not envfit."</span>)</span>
<span id="cb189-1778"><a href="#cb189-1778" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">setdiff</span>(<span class="fu">row.names</span>(<span class="fu">summary</span>(edaphic_bioenv_ind_mantel)), </span>
<span id="cb189-1779"><a href="#cb189-1779" aria-hidden="true" tabindex="-1"></a>                       edaphic_md_signif<span class="sc">$</span>parameters)</span>
<span id="cb189-1780"><a href="#cb189-1780" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-1781"><a href="#cb189-1781" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-1782"><a href="#cb189-1782" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Found in envfit but not bioenv."</span>)</span>
<span id="cb189-1783"><a href="#cb189-1783" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">setdiff</span>(edaphic_md_signif<span class="sc">$</span>parameters, </span>
<span id="cb189-1784"><a href="#cb189-1784" aria-hidden="true" tabindex="-1"></a>              <span class="fu">row.names</span>(<span class="fu">summary</span>(edaphic_bioenv_ind_mantel)))</span>
<span id="cb189-1785"><a href="#cb189-1785" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-1786"><a href="#cb189-1786" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-1787"><a href="#cb189-1787" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Found in envfit and bioenv."</span>)</span>
<span id="cb189-1788"><a href="#cb189-1788" aria-hidden="true" tabindex="-1"></a>edaphic_sig_diff <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">union</span>(edaphic_md_signif<span class="sc">$</span>parameters, </span>
<span id="cb189-1789"><a href="#cb189-1789" aria-hidden="true" tabindex="-1"></a>              <span class="fu">row.names</span>(<span class="fu">summary</span>(edaphic_bioenv_ind_mantel)))</span>
<span id="cb189-1790"><a href="#cb189-1790" aria-hidden="true" tabindex="-1"></a>edaphic_sig_diff</span>
<span id="cb189-1791"><a href="#cb189-1791" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1792"><a href="#cb189-1792" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1795"><a href="#cb189-1795" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1796"><a href="#cb189-1796" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-1797"><a href="#cb189-1797" aria-hidden="true" tabindex="-1"></a>new_edaphic_md_signif_hits <span class="ot">&lt;-</span> edaphic_sig_diff</span>
<span id="cb189-1798"><a href="#cb189-1798" aria-hidden="true" tabindex="-1"></a><span class="co">#new_edaphic_md_signif_hits &lt;- append(edaphic_md_signif_hits, edaphic_sig_diff)</span></span>
<span id="cb189-1799"><a href="#cb189-1799" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_all <span class="ot">&lt;-</span> edaphic_md_scores[edaphic_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> new_edaphic_md_signif_hits,]</span>
<span id="cb189-1800"><a href="#cb189-1800" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1801"><a href="#cb189-1801" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1802"><a href="#cb189-1802" aria-hidden="true" tabindex="-1"></a>Check. Next, we run <span class="in">`envfit`</span> for the ASVs.</span>
<span id="cb189-1803"><a href="#cb189-1803" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1806"><a href="#cb189-1806" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1807"><a href="#cb189-1807" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-1808"><a href="#cb189-1808" aria-hidden="true" tabindex="-1"></a>envfit_edaphic_asv <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, </span>
<span id="cb189-1809"><a href="#cb189-1809" aria-hidden="true" tabindex="-1"></a>                             tmp_comm[, <span class="fu">order</span>(<span class="fu">colSums</span>(<span class="sc">-</span>tmp_comm))][, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],</span>
<span id="cb189-1810"><a href="#cb189-1810" aria-hidden="true" tabindex="-1"></a>                             <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb189-1811"><a href="#cb189-1811" aria-hidden="true" tabindex="-1"></a>edaphic_asv_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">"species"</span>)</span>
<span id="cb189-1812"><a href="#cb189-1812" aria-hidden="true" tabindex="-1"></a>edaphic_asv_scores <span class="ot">&lt;-</span> edaphic_asv_scores <span class="sc">%&gt;%</span> </span>
<span id="cb189-1813"><a href="#cb189-1813" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span></span>
<span id="cb189-1814"><a href="#cb189-1814" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"Label"</span>)</span>
<span id="cb189-1815"><a href="#cb189-1815" aria-hidden="true" tabindex="-1"></a>edaphic_asv_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-1816"><a href="#cb189-1816" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1817"><a href="#cb189-1817" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1820"><a href="#cb189-1820" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1821"><a href="#cb189-1821" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1822"><a href="#cb189-1822" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1823"><a href="#cb189-1823" aria-hidden="true" tabindex="-1"></a>envfit_edaphic_asv</span>
<span id="cb189-1824"><a href="#cb189-1824" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1825"><a href="#cb189-1825" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1828"><a href="#cb189-1828" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1829"><a href="#cb189-1829" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-1830"><a href="#cb189-1830" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_edaphic_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb189-1831"><a href="#cb189-1831" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_edaphic_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb189-1832"><a href="#cb189-1832" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_edaphic_asv<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.5</span>))</span>
<span id="cb189-1833"><a href="#cb189-1833" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(edaphic_asv_signif_hits)</span>
<span id="cb189-1834"><a href="#cb189-1834" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(edaphic_asv_signif_hits)</span>
<span id="cb189-1835"><a href="#cb189-1835" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif <span class="ot">&lt;-</span> edaphic_asv_scores[edaphic_asv_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> edaphic_asv_signif_hits,]</span>
<span id="cb189-1836"><a href="#cb189-1836" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1837"><a href="#cb189-1837" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1840"><a href="#cb189-1840" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1841"><a href="#cb189-1841" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1842"><a href="#cb189-1842" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1843"><a href="#cb189-1843" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif</span>
<span id="cb189-1844"><a href="#cb189-1844" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1845"><a href="#cb189-1845" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb189-1846"><a href="#cb189-1846" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1849"><a href="#cb189-1849" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1850"><a href="#cb189-1850" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-1851"><a href="#cb189-1851" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_all<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">"metadata"</span></span>
<span id="cb189-1852"><a href="#cb189-1852" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">"ASV"</span></span>
<span id="cb189-1853"><a href="#cb189-1853" aria-hidden="true" tabindex="-1"></a>edaphic_bioplot_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(edaphic_md_signif_all, edaphic_asv_signif)</span>
<span id="cb189-1854"><a href="#cb189-1854" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1855"><a href="#cb189-1855" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1856"><a href="#cb189-1856" aria-hidden="true" tabindex="-1"></a>The last thing to do is categorize parameters scores and ASV scores into different variable types for plotting.</span>
<span id="cb189-1857"><a href="#cb189-1857" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1860"><a href="#cb189-1860" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1861"><a href="#cb189-1861" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-1862"><a href="#cb189-1862" aria-hidden="true" tabindex="-1"></a>edaphic_bioplot_data_md <span class="ot">&lt;-</span> <span class="fu">subset</span>(edaphic_bioplot_data, </span>
<span id="cb189-1863"><a href="#cb189-1863" aria-hidden="true" tabindex="-1"></a>                                  edaphic_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">"metadata"</span>)</span>
<span id="cb189-1864"><a href="#cb189-1864" aria-hidden="true" tabindex="-1"></a>edaphic_bioplot_data_asv <span class="ot">&lt;-</span> <span class="fu">subset</span>(edaphic_bioplot_data, </span>
<span id="cb189-1865"><a href="#cb189-1865" aria-hidden="true" tabindex="-1"></a>                                   edaphic_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">"ASV"</span>)</span>
<span id="cb189-1866"><a href="#cb189-1866" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1867"><a href="#cb189-1867" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1868"><a href="#cb189-1868" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb189-1869"><a href="#cb189-1869" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show the code for the plot<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb189-1870"><a href="#cb189-1870" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1873"><a href="#cb189-1873" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1874"><a href="#cb189-1874" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1875"><a href="#cb189-1875" aria-hidden="true" tabindex="-1"></a>edaphic_cap_vals <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(edaphic_cap<span class="sc">$</span>CCA<span class="sc">$</span>eig[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span>
<span id="cb189-1876"><a href="#cb189-1876" aria-hidden="true" tabindex="-1"></a>edaphic_cap1 <span class="ot">&lt;-</span> <span class="fu">signif</span>((edaphic_cap_vals[<span class="dv">1</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb189-1877"><a href="#cb189-1877" aria-hidden="true" tabindex="-1"></a>edaphic_cap2 <span class="ot">&lt;-</span> <span class="fu">signif</span>((edaphic_cap_vals[<span class="dv">2</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb189-1878"><a href="#cb189-1878" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1879"><a href="#cb189-1879" aria-hidden="true" tabindex="-1"></a>cpa1_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"CAP1"</span>, <span class="st">" ("</span>, edaphic_cap1, <span class="st">"%)"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb189-1880"><a href="#cb189-1880" aria-hidden="true" tabindex="-1"></a>cpa2_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"CAP2"</span>, <span class="st">" ("</span>, edaphic_cap2, <span class="st">"%)"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb189-1881"><a href="#cb189-1881" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1882"><a href="#cb189-1882" aria-hidden="true" tabindex="-1"></a>swel_col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#2271B2"</span>, <span class="st">"#71B222"</span>, <span class="st">"#B22271"</span>)</span>
<span id="cb189-1883"><a href="#cb189-1883" aria-hidden="true" tabindex="-1"></a>edaphic_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(edaphic_plot_data) <span class="sc">+</span></span>
<span id="cb189-1884"><a href="#cb189-1884" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">shape =</span> TREAT,</span>
<span id="cb189-1885"><a href="#cb189-1885" aria-hidden="true" tabindex="-1"></a>                           <span class="at">colour =</span> TREAT_T), <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb189-1886"><a href="#cb189-1886" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> swel_col) <span class="sc">+</span></span>
<span id="cb189-1887"><a href="#cb189-1887" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_text(data = edaphic_plot_data, aes(x = CAP1, y = CAP2, #UNCOMMENT to add sample labels</span></span>
<span id="cb189-1888"><a href="#cb189-1888" aria-hidden="true" tabindex="-1"></a><span class="co">#                              label = SampleID), size = 3) + </span></span>
<span id="cb189-1889"><a href="#cb189-1889" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">xend =</span> CAP1, <span class="at">yend =</span> CAP2),</span>
<span id="cb189-1890"><a href="#cb189-1890" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> edaphic_bioplot_data_md, <span class="at">linetype =</span> <span class="st">"solid"</span>,</span>
<span id="cb189-1891"><a href="#cb189-1891" aria-hidden="true" tabindex="-1"></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">"cm"</span>)), <span class="at">size =</span> <span class="fl">0.4</span>, </span>
<span id="cb189-1892"><a href="#cb189-1892" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">"#191919"</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb189-1893"><a href="#cb189-1893" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> edaphic_bioplot_data_md, </span>
<span id="cb189-1894"><a href="#cb189-1894" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">label =</span> parameters), <span class="at">size =</span> <span class="dv">3</span>, </span>
<span id="cb189-1895"><a href="#cb189-1895" aria-hidden="true" tabindex="-1"></a>            <span class="at">nudge_x =</span> <span class="fl">0.1</span>, <span class="at">nudge_y =</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb189-1896"><a href="#cb189-1896" aria-hidden="true" tabindex="-1"></a>  <span class="do">### This code adds ASV vectors, Uncomment to add</span></span>
<span id="cb189-1897"><a href="#cb189-1897" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_segment(aes(x = 0, y = 0, xend = CAP1, yend = CAP2),</span></span>
<span id="cb189-1898"><a href="#cb189-1898" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             data = edaphic_bioplot_data_asv, linetype = "solid",</span></span>
<span id="cb189-1899"><a href="#cb189-1899" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             arrow = arrow(length = unit(0.3, "cm")),  size = 0.2, </span></span>
<span id="cb189-1900"><a href="#cb189-1900" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             color = "#676767") +</span></span>
<span id="cb189-1901"><a href="#cb189-1901" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text(data = edaphic_bioplot_data_asv, </span></span>
<span id="cb189-1902"><a href="#cb189-1902" aria-hidden="true" tabindex="-1"></a>  <span class="co">#          aes(x = CAP1, y = CAP2, label = parameters), size = 2.5,</span></span>
<span id="cb189-1903"><a href="#cb189-1903" aria-hidden="true" tabindex="-1"></a>  <span class="co">#          nudge_x = 0.1, nudge_y = 0.05) +</span></span>
<span id="cb189-1904"><a href="#cb189-1904" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span> </span>
<span id="cb189-1905"><a href="#cb189-1905" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Capscale Analysis"</span>,</span>
<span id="cb189-1906"><a href="#cb189-1906" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">"Edaphic properties"</span>, </span>
<span id="cb189-1907"><a href="#cb189-1907" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> cpa1_lab, </span>
<span id="cb189-1908"><a href="#cb189-1908" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> cpa2_lab)</span>
<span id="cb189-1909"><a href="#cb189-1909" aria-hidden="true" tabindex="-1"></a>edaphic_plot <span class="ot">&lt;-</span> edaphic_plot <span class="sc">+</span> <span class="fu">coord_fixed</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb189-1910"><a href="#cb189-1910" aria-hidden="true" tabindex="-1"></a>edaphic_plot</span>
<span id="cb189-1911"><a href="#cb189-1911" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">"files/metadata/figures/ssu18_edaphic_capscale.png"</span>,</span>
<span id="cb189-1912"><a href="#cb189-1912" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">16</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">units =</span> <span class="st">'cm'</span>, <span class="at">res =</span> <span class="dv">600</span>, <span class="at">bg =</span> <span class="st">"white"</span>)</span>
<span id="cb189-1913"><a href="#cb189-1913" aria-hidden="true" tabindex="-1"></a>edaphic_plot</span>
<span id="cb189-1914"><a href="#cb189-1914" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb189-1915"><a href="#cb189-1915" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"files/metadata/figures/ssu18_edaphic_capscale.pdf"</span>,</span>
<span id="cb189-1916"><a href="#cb189-1916" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">5</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb189-1917"><a href="#cb189-1917" aria-hidden="true" tabindex="-1"></a>edaphic_plot</span>
<span id="cb189-1918"><a href="#cb189-1918" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb189-1919"><a href="#cb189-1919" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1920"><a href="#cb189-1920" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb189-1921"><a href="#cb189-1921" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1924"><a href="#cb189-1924" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1925"><a href="#cb189-1925" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1926"><a href="#cb189-1926" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-1927"><a href="#cb189-1927" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1928"><a href="#cb189-1928" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1929"><a href="#cb189-1929" aria-hidden="true" tabindex="-1"></a><span class="fu">### Soil Functional Response</span></span>
<span id="cb189-1930"><a href="#cb189-1930" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1933"><a href="#cb189-1933" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1934"><a href="#cb189-1934" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-1935"><a href="#cb189-1935" aria-hidden="true" tabindex="-1"></a>tmp_md <span class="ot">&lt;-</span> ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>soil_funct</span>
<span id="cb189-1936"><a href="#cb189-1936" aria-hidden="true" tabindex="-1"></a>tmp_md<span class="sc">$</span>TREAT_T <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tmp_md<span class="sc">$</span>TREAT_T)</span>
<span id="cb189-1937"><a href="#cb189-1937" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb189-1938"><a href="#cb189-1938" aria-hidden="true" tabindex="-1"></a>soil_funct_rank <span class="ot">&lt;-</span> <span class="fu">rankindex</span>(tmp_md[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_md)], tmp_comm, </span>
<span id="cb189-1939"><a href="#cb189-1939" aria-hidden="true" tabindex="-1"></a>          <span class="at">indices =</span> <span class="fu">c</span>(<span class="st">"euc"</span>, <span class="st">"man"</span>, <span class="st">"gow"</span>,<span class="st">"bra"</span>, <span class="st">"kul"</span>), </span>
<span id="cb189-1940"><a href="#cb189-1940" aria-hidden="true" tabindex="-1"></a>          <span class="at">stepacross =</span> <span class="cn">FALSE</span>, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb189-1941"><a href="#cb189-1941" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1942"><a href="#cb189-1942" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1945"><a href="#cb189-1945" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1946"><a href="#cb189-1946" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1947"><a href="#cb189-1947" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1948"><a href="#cb189-1948" aria-hidden="true" tabindex="-1"></a>soil_funct_rank</span>
<span id="cb189-1949"><a href="#cb189-1949" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1950"><a href="#cb189-1950" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1951"><a href="#cb189-1951" aria-hidden="true" tabindex="-1"></a>Let's run <span class="in">`capscale`</span> using Bray-Curtis. Note, we have <span class="in">`r ncol(ssu18_select_mc_norm_split_no_ac$soil_funct) - 8`</span> metadata parameters in this group but, for some reason, <span class="in">`capscale`</span> only works with 13 parameters. This may have to do with degrees of freedom? </span>
<span id="cb189-1952"><a href="#cb189-1952" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1953"><a href="#cb189-1953" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Starting properties: micC, micN, micP, micCN, micCP, micNP, AG_ase, BG_ase, BP_ase, CE_ase, P_ase, N_ase, S_ase, XY_ase,  LP_ase, PX_ase, CO2, enzCN, enzCP, enzNP</span>
<span id="cb189-1954"><a href="#cb189-1954" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Autocorrelated removed: micN, micNP, enzCN, enzCP, BP_ase, CE_ase, LP_ase, N_ase, P_ase</span>
<span id="cb189-1955"><a href="#cb189-1955" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Remove for capscale: NONE</span>
<span id="cb189-1956"><a href="#cb189-1956" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1959"><a href="#cb189-1959" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1960"><a href="#cb189-1960" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1961"><a href="#cb189-1961" aria-hidden="true" tabindex="-1"></a><span class="co"># All soil prop: micC, micN, micP, micCN, micCP, micNP, AG_ase, BG_ase, BP_ase, CE_ase, P_ase, N_ase, S_ase, XY_ase,  LP_ase, PX_ase, CO2, enzCN, enzCP, enzNP</span></span>
<span id="cb189-1962"><a href="#cb189-1962" aria-hidden="true" tabindex="-1"></a><span class="co"># REMOVED soil prop: micN, micNP, enzCN, enzCP, BP_ase, CE_ase, LP_ase, N_ase, P_ase</span></span>
<span id="cb189-1963"><a href="#cb189-1963" aria-hidden="true" tabindex="-1"></a><span class="co"># Remaining soil prop: micC, micP, micCN, micCP, AG_ase, BG_ase, S_ase, XY_ase, PX_ase, CO2, enzNP</span></span>
<span id="cb189-1964"><a href="#cb189-1964" aria-hidden="true" tabindex="-1"></a><span class="co"># 11 total, only works with 13</span></span>
<span id="cb189-1965"><a href="#cb189-1965" aria-hidden="true" tabindex="-1"></a><span class="co"># capscale Removed  NONE</span></span>
<span id="cb189-1966"><a href="#cb189-1966" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1967"><a href="#cb189-1967" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1968"><a href="#cb189-1968" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1971"><a href="#cb189-1971" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1972"><a href="#cb189-1972" aria-hidden="true" tabindex="-1"></a>soil_funct_cap <span class="ot">&lt;-</span> <span class="fu">capscale</span>(tmp_comm <span class="sc">~</span> micC <span class="sc">+</span> micP <span class="sc">+</span> micCN <span class="sc">+</span> micCP <span class="sc">+</span> AG_ase <span class="sc">+</span> BG_ase <span class="sc">+</span> </span>
<span id="cb189-1973"><a href="#cb189-1973" aria-hidden="true" tabindex="-1"></a>                                      S_ase <span class="sc">+</span> XY_ase <span class="sc">+</span> PX_ase <span class="sc">+</span> CO2 <span class="sc">+</span> enzNP, </span>
<span id="cb189-1974"><a href="#cb189-1974" aria-hidden="true" tabindex="-1"></a>                           tmp_md, <span class="at">dist =</span> <span class="st">"bray"</span>)</span>
<span id="cb189-1975"><a href="#cb189-1975" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1976"><a href="#cb189-1976" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1979"><a href="#cb189-1979" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1980"><a href="#cb189-1980" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1981"><a href="#cb189-1981" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1982"><a href="#cb189-1982" aria-hidden="true" tabindex="-1"></a>soil_funct_cap</span>
<span id="cb189-1983"><a href="#cb189-1983" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1984"><a href="#cb189-1984" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1985"><a href="#cb189-1985" aria-hidden="true" tabindex="-1"></a>Now we can look at the variance against each principal component. </span>
<span id="cb189-1986"><a href="#cb189-1986" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1989"><a href="#cb189-1989" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-1990"><a href="#cb189-1990" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-1991"><a href="#cb189-1991" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-1992"><a href="#cb189-1992" aria-hidden="true" tabindex="-1"></a><span class="fu">screeplot</span>(soil_funct_cap)</span>
<span id="cb189-1993"><a href="#cb189-1993" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-1994"><a href="#cb189-1994" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1995"><a href="#cb189-1995" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1996"><a href="#cb189-1996" aria-hidden="true" tabindex="-1"></a>And then make some quick and dirty plots. This will also come in handy later when we need to parse out data a better plot visualization. The <span class="in">`ggplot`</span> function <span class="in">`autoplot`</span> stores these data in a more accessible way than the raw results from <span class="in">`capscale`</span></span>
<span id="cb189-1997"><a href="#cb189-1997" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2000"><a href="#cb189-2000" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2001"><a href="#cb189-2001" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2002"><a href="#cb189-2002" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">plot</span>(soil_funct_cap) </span>
<span id="cb189-2003"><a href="#cb189-2003" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt <span class="ot">&lt;-</span> <span class="fu">autoplot</span>(soil_funct_cap, <span class="at">arrows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb189-2004"><a href="#cb189-2004" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt</span>
<span id="cb189-2005"><a href="#cb189-2005" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2006"><a href="#cb189-2006" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2009"><a href="#cb189-2009" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2010"><a href="#cb189-2010" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2011"><a href="#cb189-2011" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(soil_funct_cap) <span class="co"># overall test of the significant of the analysis</span></span>
<span id="cb189-2012"><a href="#cb189-2012" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(soil_funct_cap, <span class="at">by =</span> <span class="st">"axis"</span>, <span class="at">perm.max=</span><span class="dv">500</span>) <span class="co"># test axes for significance</span></span>
<span id="cb189-2013"><a href="#cb189-2013" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(soil_funct_cap, <span class="at">by =</span> <span class="st">"terms"</span>, <span class="at">permu=</span><span class="dv">500</span>) <span class="co"># test for sign. environ. variables</span></span>
<span id="cb189-2014"><a href="#cb189-2014" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2015"><a href="#cb189-2015" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2016"><a href="#cb189-2016" aria-hidden="true" tabindex="-1"></a>Next, we need to grab capscale scores for the samples and create a data frame of the first two dimensions. We will also need to add some of the sample details to the data frame. For this we use the vegan function <span class="in">`scores`</span> which gets species or site scores from the ordination.</span>
<span id="cb189-2017"><a href="#cb189-2017" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2020"><a href="#cb189-2020" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2021"><a href="#cb189-2021" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2022"><a href="#cb189-2022" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">"sites"</span>)</span>
<span id="cb189-2023"><a href="#cb189-2023" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-2024"><a href="#cb189-2024" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> tmp_samp_scores <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">SampleID =</span> Label)</span>
<span id="cb189-2025"><a href="#cb189-2025" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2026"><a href="#cb189-2026" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb189-2027"><a href="#cb189-2027" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md_sub <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb189-2028"><a href="#cb189-2028" aria-hidden="true" tabindex="-1"></a>soil_funct_plot_data <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_md_sub, tmp_samp_scores, </span>
<span id="cb189-2029"><a href="#cb189-2029" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">by =</span> <span class="st">"SampleID"</span>)</span>
<span id="cb189-2030"><a href="#cb189-2030" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2031"><a href="#cb189-2031" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2032"><a href="#cb189-2032" aria-hidden="true" tabindex="-1"></a>Now we have a new data frame that contains sample details and capscale values. </span>
<span id="cb189-2033"><a href="#cb189-2033" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2036"><a href="#cb189-2036" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2037"><a href="#cb189-2037" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2038"><a href="#cb189-2038" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2039"><a href="#cb189-2039" aria-hidden="true" tabindex="-1"></a>soil_funct_plot_data</span>
<span id="cb189-2040"><a href="#cb189-2040" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2041"><a href="#cb189-2041" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2042"><a href="#cb189-2042" aria-hidden="true" tabindex="-1"></a>We can then do the same with the metadata vectors. Here though we only need the scores and parameter name. </span>
<span id="cb189-2043"><a href="#cb189-2043" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2046"><a href="#cb189-2046" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2047"><a href="#cb189-2047" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2048"><a href="#cb189-2048" aria-hidden="true" tabindex="-1"></a>soil_funct_md_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">"biplot"</span>)</span>
<span id="cb189-2049"><a href="#cb189-2049" aria-hidden="true" tabindex="-1"></a>soil_funct_md_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-2050"><a href="#cb189-2050" aria-hidden="true" tabindex="-1"></a>soil_funct_md_scores <span class="ot">&lt;-</span> soil_funct_md_scores <span class="sc">%&gt;%</span> </span>
<span id="cb189-2051"><a href="#cb189-2051" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span> </span>
<span id="cb189-2052"><a href="#cb189-2052" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"Label"</span>)</span>
<span id="cb189-2053"><a href="#cb189-2053" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2054"><a href="#cb189-2054" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2055"><a href="#cb189-2055" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2058"><a href="#cb189-2058" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2059"><a href="#cb189-2059" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2060"><a href="#cb189-2060" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2061"><a href="#cb189-2061" aria-hidden="true" tabindex="-1"></a>soil_funct_md_scores</span>
<span id="cb189-2062"><a href="#cb189-2062" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2063"><a href="#cb189-2063" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2064"><a href="#cb189-2064" aria-hidden="true" tabindex="-1"></a>Let's run some quick correlations of metadata with ordination axes to see which parameters are significant. For this we use the vegan function <span class="in">`envfit`</span>.</span>
<span id="cb189-2065"><a href="#cb189-2065" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2068"><a href="#cb189-2068" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2069"><a href="#cb189-2069" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2070"><a href="#cb189-2070" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> soil_funct_plot_data[, <span class="dv">6</span><span class="sc">:</span><span class="dv">7</span>]</span>
<span id="cb189-2071"><a href="#cb189-2071" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_samp_scores_sub)</span>
<span id="cb189-2072"><a href="#cb189-2072" aria-hidden="true" tabindex="-1"></a>tmp_param_list <span class="ot">&lt;-</span> soil_funct_md_scores<span class="sc">$</span>parameters</span>
<span id="cb189-2073"><a href="#cb189-2073" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2074"><a href="#cb189-2074" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> <span class="fu">subset</span>(tmp_md,  <span class="at">select =</span>  tmp_param_list)</span>
<span id="cb189-2075"><a href="#cb189-2075" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2076"><a href="#cb189-2076" aria-hidden="true" tabindex="-1"></a>envfit_soil_funct_md <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, tmp_md_sub,</span>
<span id="cb189-2077"><a href="#cb189-2077" aria-hidden="true" tabindex="-1"></a>                 <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb189-2078"><a href="#cb189-2078" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2079"><a href="#cb189-2079" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2082"><a href="#cb189-2082" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2083"><a href="#cb189-2083" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2084"><a href="#cb189-2084" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2085"><a href="#cb189-2085" aria-hidden="true" tabindex="-1"></a>envfit_soil_funct_md</span>
<span id="cb189-2086"><a href="#cb189-2086" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2087"><a href="#cb189-2087" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2090"><a href="#cb189-2090" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2091"><a href="#cb189-2091" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2092"><a href="#cb189-2092" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_soil_funct_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb189-2093"><a href="#cb189-2093" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_soil_funct_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb189-2094"><a href="#cb189-2094" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_soil_funct_md<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.4</span>))</span>
<span id="cb189-2095"><a href="#cb189-2095" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(soil_funct_md_signif_hits)</span>
<span id="cb189-2096"><a href="#cb189-2096" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(soil_funct_md_signif_hits)</span>
<span id="cb189-2097"><a href="#cb189-2097" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif <span class="ot">&lt;-</span> soil_funct_md_scores[soil_funct_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb189-2098"><a href="#cb189-2098" aria-hidden="true" tabindex="-1"></a>                                               soil_funct_md_signif_hits,]</span>
<span id="cb189-2099"><a href="#cb189-2099" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif<span class="sc">$</span>parameters</span>
<span id="cb189-2100"><a href="#cb189-2100" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2101"><a href="#cb189-2101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2102"><a href="#cb189-2102" aria-hidden="true" tabindex="-1"></a><span class="in">`envfit`</span> found that <span class="in">`r soil_funct_md_signif$parameters`</span> were significantly correlated. </span>
<span id="cb189-2103"><a href="#cb189-2103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2104"><a href="#cb189-2104" aria-hidden="true" tabindex="-1"></a>Now let's see if the same parameters are significant for the <span class="in">`envfit`</span> and <span class="in">`bioenv`</span> analyses.</span>
<span id="cb189-2105"><a href="#cb189-2105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2108"><a href="#cb189-2108" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2109"><a href="#cb189-2109" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2110"><a href="#cb189-2110" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2111"><a href="#cb189-2111" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hold</span></span>
<span id="cb189-2112"><a href="#cb189-2112" aria-hidden="true" tabindex="-1"></a><span class="co">#| comment: '' </span></span>
<span id="cb189-2113"><a href="#cb189-2113" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Significant parameters from bioenv analysis."</span>)</span>
<span id="cb189-2114"><a href="#cb189-2114" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(<span class="fu">summary</span>(soil_funct_bioenv_ind_mantel))</span>
<span id="cb189-2115"><a href="#cb189-2115" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-2116"><a href="#cb189-2116" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-2117"><a href="#cb189-2117" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Significant parameters from envfit analysis."</span>)</span>
<span id="cb189-2118"><a href="#cb189-2118" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif<span class="sc">$</span>parameters</span>
<span id="cb189-2119"><a href="#cb189-2119" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-2120"><a href="#cb189-2120" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-2121"><a href="#cb189-2121" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Found in bioenv but not envfit."</span>)</span>
<span id="cb189-2122"><a href="#cb189-2122" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">setdiff</span>(<span class="fu">row.names</span>(<span class="fu">summary</span>(soil_funct_bioenv_ind_mantel)), </span>
<span id="cb189-2123"><a href="#cb189-2123" aria-hidden="true" tabindex="-1"></a>                       soil_funct_md_signif<span class="sc">$</span>parameters)</span>
<span id="cb189-2124"><a href="#cb189-2124" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-2125"><a href="#cb189-2125" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-2126"><a href="#cb189-2126" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Found in envfit but not bioenv."</span>)</span>
<span id="cb189-2127"><a href="#cb189-2127" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">setdiff</span>(soil_funct_md_signif<span class="sc">$</span>parameters, </span>
<span id="cb189-2128"><a href="#cb189-2128" aria-hidden="true" tabindex="-1"></a>              <span class="fu">row.names</span>(<span class="fu">summary</span>(soil_funct_bioenv_ind_mantel)))</span>
<span id="cb189-2129"><a href="#cb189-2129" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-2130"><a href="#cb189-2130" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-2131"><a href="#cb189-2131" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Found in envfit and bioenv."</span>)</span>
<span id="cb189-2132"><a href="#cb189-2132" aria-hidden="true" tabindex="-1"></a>soil_funct_sig_diff <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">union</span>(soil_funct_md_signif<span class="sc">$</span>parameters, </span>
<span id="cb189-2133"><a href="#cb189-2133" aria-hidden="true" tabindex="-1"></a>                                   <span class="fu">row.names</span>(<span class="fu">summary</span>(soil_funct_bioenv_ind_mantel)))</span>
<span id="cb189-2134"><a href="#cb189-2134" aria-hidden="true" tabindex="-1"></a>soil_funct_sig_diff</span>
<span id="cb189-2135"><a href="#cb189-2135" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2136"><a href="#cb189-2136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2137"><a href="#cb189-2137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2140"><a href="#cb189-2140" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2141"><a href="#cb189-2141" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2142"><a href="#cb189-2142" aria-hidden="true" tabindex="-1"></a><span class="co">#new_soil_funct_md_signif_hits &lt;- append(soil_funct_md_signif_hits, soil_funct_sig_diff)</span></span>
<span id="cb189-2143"><a href="#cb189-2143" aria-hidden="true" tabindex="-1"></a>new_soil_funct_md_signif_hits <span class="ot">&lt;-</span> soil_funct_sig_diff</span>
<span id="cb189-2144"><a href="#cb189-2144" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_all <span class="ot">&lt;-</span> soil_funct_md_scores[soil_funct_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb189-2145"><a href="#cb189-2145" aria-hidden="true" tabindex="-1"></a>                                                   new_soil_funct_md_signif_hits,]</span>
<span id="cb189-2146"><a href="#cb189-2146" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2147"><a href="#cb189-2147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2148"><a href="#cb189-2148" aria-hidden="true" tabindex="-1"></a>Check. Next, we run <span class="in">`envfit`</span> for the ASVs.</span>
<span id="cb189-2149"><a href="#cb189-2149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2152"><a href="#cb189-2152" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2153"><a href="#cb189-2153" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2154"><a href="#cb189-2154" aria-hidden="true" tabindex="-1"></a>envfit_soil_funct_asv <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, </span>
<span id="cb189-2155"><a href="#cb189-2155" aria-hidden="true" tabindex="-1"></a>                                tmp_comm[, <span class="fu">order</span>(<span class="fu">colSums</span>(<span class="sc">-</span>tmp_comm))][, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>], </span>
<span id="cb189-2156"><a href="#cb189-2156" aria-hidden="true" tabindex="-1"></a>                                <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb189-2157"><a href="#cb189-2157" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">"species"</span>)</span>
<span id="cb189-2158"><a href="#cb189-2158" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_scores <span class="ot">&lt;-</span> soil_funct_asv_scores <span class="sc">%&gt;%</span> </span>
<span id="cb189-2159"><a href="#cb189-2159" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span> </span>
<span id="cb189-2160"><a href="#cb189-2160" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"Label"</span>)</span>
<span id="cb189-2161"><a href="#cb189-2161" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-2162"><a href="#cb189-2162" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2163"><a href="#cb189-2163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2166"><a href="#cb189-2166" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2167"><a href="#cb189-2167" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2168"><a href="#cb189-2168" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2169"><a href="#cb189-2169" aria-hidden="true" tabindex="-1"></a>envfit_soil_funct_asv</span>
<span id="cb189-2170"><a href="#cb189-2170" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2171"><a href="#cb189-2171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2174"><a href="#cb189-2174" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2175"><a href="#cb189-2175" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2176"><a href="#cb189-2176" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_soil_funct_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb189-2177"><a href="#cb189-2177" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_soil_funct_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb189-2178"><a href="#cb189-2178" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_soil_funct_asv<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.5</span>))</span>
<span id="cb189-2179"><a href="#cb189-2179" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(soil_funct_asv_signif_hits)</span>
<span id="cb189-2180"><a href="#cb189-2180" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(soil_funct_asv_signif_hits)</span>
<span id="cb189-2181"><a href="#cb189-2181" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif <span class="ot">&lt;-</span> soil_funct_asv_scores[soil_funct_asv_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb189-2182"><a href="#cb189-2182" aria-hidden="true" tabindex="-1"></a>                                                 soil_funct_asv_signif_hits,]</span>
<span id="cb189-2183"><a href="#cb189-2183" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2184"><a href="#cb189-2184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2187"><a href="#cb189-2187" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2188"><a href="#cb189-2188" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2189"><a href="#cb189-2189" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2190"><a href="#cb189-2190" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif</span>
<span id="cb189-2191"><a href="#cb189-2191" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2192"><a href="#cb189-2192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2195"><a href="#cb189-2195" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2196"><a href="#cb189-2196" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2197"><a href="#cb189-2197" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_all<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">"metadata"</span></span>
<span id="cb189-2198"><a href="#cb189-2198" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">"ASV"</span></span>
<span id="cb189-2199"><a href="#cb189-2199" aria-hidden="true" tabindex="-1"></a>soil_funct_bioplot_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(soil_funct_md_signif_all, soil_funct_asv_signif)</span>
<span id="cb189-2200"><a href="#cb189-2200" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2201"><a href="#cb189-2201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2202"><a href="#cb189-2202" aria-hidden="true" tabindex="-1"></a>The last thing to do is categorize parameters scores and ASV scores into different variable types for plotting.</span>
<span id="cb189-2203"><a href="#cb189-2203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2206"><a href="#cb189-2206" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2207"><a href="#cb189-2207" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2208"><a href="#cb189-2208" aria-hidden="true" tabindex="-1"></a>soil_funct_bioplot_data_md <span class="ot">&lt;-</span> <span class="fu">subset</span>(soil_funct_bioplot_data, </span>
<span id="cb189-2209"><a href="#cb189-2209" aria-hidden="true" tabindex="-1"></a>                                  soil_funct_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">"metadata"</span>)</span>
<span id="cb189-2210"><a href="#cb189-2210" aria-hidden="true" tabindex="-1"></a>soil_funct_bioplot_data_asv <span class="ot">&lt;-</span> <span class="fu">subset</span>(soil_funct_bioplot_data, </span>
<span id="cb189-2211"><a href="#cb189-2211" aria-hidden="true" tabindex="-1"></a>                                   soil_funct_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">"ASV"</span>)</span>
<span id="cb189-2212"><a href="#cb189-2212" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2213"><a href="#cb189-2213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2214"><a href="#cb189-2214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2215"><a href="#cb189-2215" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb189-2216"><a href="#cb189-2216" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show the code for the plot<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb189-2217"><a href="#cb189-2217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2220"><a href="#cb189-2220" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2221"><a href="#cb189-2221" aria-hidden="true" tabindex="-1"></a>soil_funct_cap_vals <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(soil_funct_cap<span class="sc">$</span>CCA<span class="sc">$</span>eig[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span>
<span id="cb189-2222"><a href="#cb189-2222" aria-hidden="true" tabindex="-1"></a>soil_funct_cap1 <span class="ot">&lt;-</span> <span class="fu">signif</span>((soil_funct_cap_vals[<span class="dv">1</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb189-2223"><a href="#cb189-2223" aria-hidden="true" tabindex="-1"></a>soil_funct_cap2 <span class="ot">&lt;-</span> <span class="fu">signif</span>((soil_funct_cap_vals[<span class="dv">2</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb189-2224"><a href="#cb189-2224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2225"><a href="#cb189-2225" aria-hidden="true" tabindex="-1"></a>cpa1_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"CAP1"</span>, <span class="st">" ("</span>, soil_funct_cap1, <span class="st">"%)"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb189-2226"><a href="#cb189-2226" aria-hidden="true" tabindex="-1"></a>cpa2_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"CAP2"</span>, <span class="st">" ("</span>, soil_funct_cap2, <span class="st">"%)"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb189-2227"><a href="#cb189-2227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2228"><a href="#cb189-2228" aria-hidden="true" tabindex="-1"></a>swel_col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#2271B2"</span>, <span class="st">"#71B222"</span>, <span class="st">"#B22271"</span>)</span>
<span id="cb189-2229"><a href="#cb189-2229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2230"><a href="#cb189-2230" aria-hidden="true" tabindex="-1"></a>soil_funct_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(soil_funct_plot_data) <span class="sc">+</span></span>
<span id="cb189-2231"><a href="#cb189-2231" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">shape =</span> TREAT,</span>
<span id="cb189-2232"><a href="#cb189-2232" aria-hidden="true" tabindex="-1"></a>                           <span class="at">colour =</span> TREAT_T), <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb189-2233"><a href="#cb189-2233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> swel_col) <span class="sc">+</span></span>
<span id="cb189-2234"><a href="#cb189-2234" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_text(data = soil_funct_plot_data, aes(x = CAP1, y = CAP2, #UNCOMMENT to add sample labels</span></span>
<span id="cb189-2235"><a href="#cb189-2235" aria-hidden="true" tabindex="-1"></a><span class="co">#                              label = SampleID), size = 3) + </span></span>
<span id="cb189-2236"><a href="#cb189-2236" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">xend =</span> CAP1, <span class="at">yend =</span> CAP2),</span>
<span id="cb189-2237"><a href="#cb189-2237" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> soil_funct_bioplot_data_md, <span class="at">linetype =</span> <span class="st">"solid"</span>,</span>
<span id="cb189-2238"><a href="#cb189-2238" aria-hidden="true" tabindex="-1"></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">"cm"</span>)), <span class="at">size =</span> <span class="fl">0.4</span>, </span>
<span id="cb189-2239"><a href="#cb189-2239" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">"#191919"</span>) <span class="sc">+</span></span>
<span id="cb189-2240"><a href="#cb189-2240" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> soil_funct_bioplot_data_md, </span>
<span id="cb189-2241"><a href="#cb189-2241" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">label =</span> parameters), <span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb189-2242"><a href="#cb189-2242" aria-hidden="true" tabindex="-1"></a>            <span class="at">nudge_x =</span> <span class="fl">0.1</span>, <span class="at">nudge_y =</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb189-2243"><a href="#cb189-2243" aria-hidden="true" tabindex="-1"></a>  <span class="do">### This code adds ASV vectors, Uncomment to add</span></span>
<span id="cb189-2244"><a href="#cb189-2244" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_segment(aes(x = 0, y = 0, xend = CAP1, yend = CAP2),</span></span>
<span id="cb189-2245"><a href="#cb189-2245" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             data = soil_funct_bioplot_data_asv, linetype = "solid",</span></span>
<span id="cb189-2246"><a href="#cb189-2246" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             arrow = arrow(length = unit(0.3, "cm")), size = 0.2,</span></span>
<span id="cb189-2247"><a href="#cb189-2247" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             color = "#676767") +</span></span>
<span id="cb189-2248"><a href="#cb189-2248" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text(data = soil_funct_bioplot_data_asv, </span></span>
<span id="cb189-2249"><a href="#cb189-2249" aria-hidden="true" tabindex="-1"></a>  <span class="co">#          aes(x = CAP1, y = CAP2, label = parameters), size = 2.5,</span></span>
<span id="cb189-2250"><a href="#cb189-2250" aria-hidden="true" tabindex="-1"></a>  <span class="co">#          nudge_x = 0.05, nudge_y = 0.05) +</span></span>
<span id="cb189-2251"><a href="#cb189-2251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span> </span>
<span id="cb189-2252"><a href="#cb189-2252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Capscale Analysis"</span>,</span>
<span id="cb189-2253"><a href="#cb189-2253" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">"Soil Functional Response"</span>, </span>
<span id="cb189-2254"><a href="#cb189-2254" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> cpa1_lab, </span>
<span id="cb189-2255"><a href="#cb189-2255" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> cpa2_lab)</span>
<span id="cb189-2256"><a href="#cb189-2256" aria-hidden="true" tabindex="-1"></a>soil_funct_plot <span class="ot">&lt;-</span> soil_funct_plot <span class="sc">+</span> <span class="fu">coord_fixed</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb189-2257"><a href="#cb189-2257" aria-hidden="true" tabindex="-1"></a>soil_funct_plot</span>
<span id="cb189-2258"><a href="#cb189-2258" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">"files/metadata/figures/ssu18_soil_funct_capscale.png"</span>,</span>
<span id="cb189-2259"><a href="#cb189-2259" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">16</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">units =</span> <span class="st">'cm'</span>, <span class="at">res =</span> <span class="dv">600</span>, <span class="at">bg =</span> <span class="st">"white"</span>)</span>
<span id="cb189-2260"><a href="#cb189-2260" aria-hidden="true" tabindex="-1"></a>soil_funct_plot</span>
<span id="cb189-2261"><a href="#cb189-2261" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb189-2262"><a href="#cb189-2262" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"files/metadata/figures/ssu18_soil_funct_capscale.pdf"</span>,</span>
<span id="cb189-2263"><a href="#cb189-2263" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">5</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb189-2264"><a href="#cb189-2264" aria-hidden="true" tabindex="-1"></a>soil_funct_plot</span>
<span id="cb189-2265"><a href="#cb189-2265" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb189-2266"><a href="#cb189-2266" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2267"><a href="#cb189-2267" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb189-2268"><a href="#cb189-2268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2271"><a href="#cb189-2271" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2272"><a href="#cb189-2272" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2273"><a href="#cb189-2273" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-2274"><a href="#cb189-2274" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2275"><a href="#cb189-2275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2276"><a href="#cb189-2276" aria-hidden="true" tabindex="-1"></a><span class="fu">### Temperature Adaptation</span></span>
<span id="cb189-2277"><a href="#cb189-2277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2280"><a href="#cb189-2280" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2281"><a href="#cb189-2281" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2282"><a href="#cb189-2282" aria-hidden="true" tabindex="-1"></a>tmp_md <span class="ot">&lt;-</span> ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>temp_adapt</span>
<span id="cb189-2283"><a href="#cb189-2283" aria-hidden="true" tabindex="-1"></a>tmp_md<span class="sc">$</span>TREAT_T <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tmp_md<span class="sc">$</span>TREAT_T)</span>
<span id="cb189-2284"><a href="#cb189-2284" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(ssu18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb189-2285"><a href="#cb189-2285" aria-hidden="true" tabindex="-1"></a>temp_adapt_rank <span class="ot">&lt;-</span> <span class="fu">rankindex</span>(tmp_md[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_md)], tmp_comm, </span>
<span id="cb189-2286"><a href="#cb189-2286" aria-hidden="true" tabindex="-1"></a>          <span class="at">indices =</span> <span class="fu">c</span>(<span class="st">"euc"</span>, <span class="st">"man"</span>, <span class="st">"gow"</span>,<span class="st">"bra"</span>, <span class="st">"kul"</span>), </span>
<span id="cb189-2287"><a href="#cb189-2287" aria-hidden="true" tabindex="-1"></a>          <span class="at">stepacross =</span> <span class="cn">FALSE</span>, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb189-2288"><a href="#cb189-2288" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2289"><a href="#cb189-2289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2292"><a href="#cb189-2292" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2293"><a href="#cb189-2293" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2294"><a href="#cb189-2294" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2295"><a href="#cb189-2295" aria-hidden="true" tabindex="-1"></a>temp_adapt_rank</span>
<span id="cb189-2296"><a href="#cb189-2296" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2297"><a href="#cb189-2297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2298"><a href="#cb189-2298" aria-hidden="true" tabindex="-1"></a>Let's run <span class="in">`capscale`</span> using Bray-Curtis. Note, we have <span class="in">`r ncol(ssu18_select_mc_norm_split_no_ac$temp_adapt) - 8`</span> metadata parameters in this group but, for some reason, <span class="in">`capscale`</span> only works with 13 parameters. This may have to do with degrees of freedom? </span>
<span id="cb189-2299"><a href="#cb189-2299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2300"><a href="#cb189-2300" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Starting properties: AG_Q10, BG_Q10, BP_Q10, CE_Q10, P_Q10, N_Q10, S_Q10, XY_Q10, LP_Q10, PX_Q10, CUEcn, CUEcp, NUE, PUE, Tmin, SI</span>
<span id="cb189-2301"><a href="#cb189-2301" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Autocorrelated removed: NUE, PUE, SI</span>
<span id="cb189-2302"><a href="#cb189-2302" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Remove for capscale: NONE</span>
<span id="cb189-2303"><a href="#cb189-2303" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2306"><a href="#cb189-2306" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2307"><a href="#cb189-2307" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2308"><a href="#cb189-2308" aria-hidden="true" tabindex="-1"></a><span class="co"># All temp_adapt: AG_Q10, BG_Q10, BP_Q10, CE_Q10, P_Q10, N_Q10, S_Q10, XY_Q10, LP_Q10, PX_Q10, CUEcn, CUEcp, NUE, PUE, Tmin, SI</span></span>
<span id="cb189-2309"><a href="#cb189-2309" aria-hidden="true" tabindex="-1"></a><span class="co"># REMOVED temp_adapt: NUE, PUE, SI</span></span>
<span id="cb189-2310"><a href="#cb189-2310" aria-hidden="true" tabindex="-1"></a><span class="co"># Remaining temp_adapt: AG_Q10, BG_Q10, BP_Q10, CE_Q10, P_Q10, N_Q10, S_Q10, XY_Q10, LP_Q10, PX_Q10, CUEcn, CUEcp, Tmin</span></span>
<span id="cb189-2311"><a href="#cb189-2311" aria-hidden="true" tabindex="-1"></a><span class="co"># 14 total, only works with 13</span></span>
<span id="cb189-2312"><a href="#cb189-2312" aria-hidden="true" tabindex="-1"></a><span class="co"># Removed:  NONE</span></span>
<span id="cb189-2313"><a href="#cb189-2313" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2314"><a href="#cb189-2314" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2317"><a href="#cb189-2317" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2318"><a href="#cb189-2318" aria-hidden="true" tabindex="-1"></a>temp_adapt_cap <span class="ot">&lt;-</span> <span class="fu">capscale</span>(tmp_comm <span class="sc">~</span>  AG_Q10 <span class="sc">+</span> BG_Q10 <span class="sc">+</span> BP_Q10 <span class="sc">+</span> CE_Q10 <span class="sc">+</span> </span>
<span id="cb189-2319"><a href="#cb189-2319" aria-hidden="true" tabindex="-1"></a>                                       P_Q10 <span class="sc">+</span> N_Q10 <span class="sc">+</span> S_Q10 <span class="sc">+</span> XY_Q10 <span class="sc">+</span> </span>
<span id="cb189-2320"><a href="#cb189-2320" aria-hidden="true" tabindex="-1"></a>                                       LP_Q10 <span class="sc">+</span> PX_Q10 <span class="sc">+</span> CUEcn <span class="sc">+</span> CUEcp <span class="sc">+</span> Tmin, </span>
<span id="cb189-2321"><a href="#cb189-2321" aria-hidden="true" tabindex="-1"></a>                             tmp_md, <span class="at">dist =</span> <span class="st">"bray"</span>)</span>
<span id="cb189-2322"><a href="#cb189-2322" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2323"><a href="#cb189-2323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2326"><a href="#cb189-2326" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2327"><a href="#cb189-2327" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2328"><a href="#cb189-2328" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2329"><a href="#cb189-2329" aria-hidden="true" tabindex="-1"></a>temp_adapt_cap</span>
<span id="cb189-2330"><a href="#cb189-2330" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2331"><a href="#cb189-2331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2332"><a href="#cb189-2332" aria-hidden="true" tabindex="-1"></a>Now we can look at the variance against each principal component. </span>
<span id="cb189-2333"><a href="#cb189-2333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2336"><a href="#cb189-2336" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2337"><a href="#cb189-2337" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2338"><a href="#cb189-2338" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2339"><a href="#cb189-2339" aria-hidden="true" tabindex="-1"></a><span class="fu">screeplot</span>(temp_adapt_cap)</span>
<span id="cb189-2340"><a href="#cb189-2340" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2341"><a href="#cb189-2341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2342"><a href="#cb189-2342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2343"><a href="#cb189-2343" aria-hidden="true" tabindex="-1"></a>And then make some quick and dirty plots. This will also come in handy later when we need to parse out data a better plot visualization. The <span class="in">`ggplot`</span> function <span class="in">`autoplot`</span> stores these data in a more accessible way than the raw results from <span class="in">`capscale`</span></span>
<span id="cb189-2344"><a href="#cb189-2344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2347"><a href="#cb189-2347" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2348"><a href="#cb189-2348" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2349"><a href="#cb189-2349" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">plot</span>(temp_adapt_cap) </span>
<span id="cb189-2350"><a href="#cb189-2350" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt <span class="ot">&lt;-</span> <span class="fu">autoplot</span>(temp_adapt_cap, <span class="at">arrows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb189-2351"><a href="#cb189-2351" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt</span>
<span id="cb189-2352"><a href="#cb189-2352" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2353"><a href="#cb189-2353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2356"><a href="#cb189-2356" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2357"><a href="#cb189-2357" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2358"><a href="#cb189-2358" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(temp_adapt_cap) <span class="co"># overall test of the significant of the analysis</span></span>
<span id="cb189-2359"><a href="#cb189-2359" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(temp_adapt_cap, <span class="at">by =</span> <span class="st">"axis"</span>, <span class="at">perm.max =</span> <span class="dv">500</span>) <span class="co"># test axes for significance</span></span>
<span id="cb189-2360"><a href="#cb189-2360" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(temp_adapt_cap, <span class="at">by =</span> <span class="st">"terms"</span>, <span class="at">permu =</span> <span class="dv">500</span>) <span class="co"># test for sign. environ. variables</span></span>
<span id="cb189-2361"><a href="#cb189-2361" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2362"><a href="#cb189-2362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2363"><a href="#cb189-2363" aria-hidden="true" tabindex="-1"></a>Next, we need to grab capscale scores for the samples and create a data frame of the first two dimensions. We will also need to add some of the sample details to the data frame. For this we use the vegan function <span class="in">`scores`</span> which gets species or site scores from the ordination.</span>
<span id="cb189-2364"><a href="#cb189-2364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2367"><a href="#cb189-2367" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2368"><a href="#cb189-2368" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2369"><a href="#cb189-2369" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">"sites"</span>)</span>
<span id="cb189-2370"><a href="#cb189-2370" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-2371"><a href="#cb189-2371" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> tmp_samp_scores <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">SampleID =</span> Label)</span>
<span id="cb189-2372"><a href="#cb189-2372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2373"><a href="#cb189-2373" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb189-2374"><a href="#cb189-2374" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md_sub <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb189-2375"><a href="#cb189-2375" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot_data <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_md_sub, tmp_samp_scores, </span>
<span id="cb189-2376"><a href="#cb189-2376" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">by =</span> <span class="st">"SampleID"</span>)</span>
<span id="cb189-2377"><a href="#cb189-2377" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2378"><a href="#cb189-2378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2379"><a href="#cb189-2379" aria-hidden="true" tabindex="-1"></a>Now we have a new data frame that contains sample details and capscale values. </span>
<span id="cb189-2380"><a href="#cb189-2380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2383"><a href="#cb189-2383" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2384"><a href="#cb189-2384" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2385"><a href="#cb189-2385" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2386"><a href="#cb189-2386" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot_data</span>
<span id="cb189-2387"><a href="#cb189-2387" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2388"><a href="#cb189-2388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2389"><a href="#cb189-2389" aria-hidden="true" tabindex="-1"></a>We can then do the same with the metadata vectors. Here though we only need the scores and parameter name. </span>
<span id="cb189-2390"><a href="#cb189-2390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2393"><a href="#cb189-2393" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2394"><a href="#cb189-2394" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2395"><a href="#cb189-2395" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">"biplot"</span>)</span>
<span id="cb189-2396"><a href="#cb189-2396" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-2397"><a href="#cb189-2397" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_scores <span class="ot">&lt;-</span> temp_adapt_md_scores <span class="sc">%&gt;%</span> </span>
<span id="cb189-2398"><a href="#cb189-2398" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span> </span>
<span id="cb189-2399"><a href="#cb189-2399" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"Label"</span>)</span>
<span id="cb189-2400"><a href="#cb189-2400" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2401"><a href="#cb189-2401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2402"><a href="#cb189-2402" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2405"><a href="#cb189-2405" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2406"><a href="#cb189-2406" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2407"><a href="#cb189-2407" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2408"><a href="#cb189-2408" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_scores</span>
<span id="cb189-2409"><a href="#cb189-2409" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2410"><a href="#cb189-2410" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2411"><a href="#cb189-2411" aria-hidden="true" tabindex="-1"></a>Let's run some quick correlations of metadata with ordination axes to see which parameters are significant. For this we use the vegan function <span class="in">`envfit`</span>.</span>
<span id="cb189-2412"><a href="#cb189-2412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2415"><a href="#cb189-2415" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2416"><a href="#cb189-2416" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2417"><a href="#cb189-2417" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> temp_adapt_plot_data[, <span class="dv">6</span><span class="sc">:</span><span class="dv">7</span>]</span>
<span id="cb189-2418"><a href="#cb189-2418" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_samp_scores_sub)</span>
<span id="cb189-2419"><a href="#cb189-2419" aria-hidden="true" tabindex="-1"></a>tmp_param_list <span class="ot">&lt;-</span> temp_adapt_md_scores<span class="sc">$</span>parameters</span>
<span id="cb189-2420"><a href="#cb189-2420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2421"><a href="#cb189-2421" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> <span class="fu">subset</span>(tmp_md,  <span class="at">select =</span>  tmp_param_list)</span>
<span id="cb189-2422"><a href="#cb189-2422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2423"><a href="#cb189-2423" aria-hidden="true" tabindex="-1"></a>envfit_temp_adapt_md <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, tmp_md_sub,</span>
<span id="cb189-2424"><a href="#cb189-2424" aria-hidden="true" tabindex="-1"></a>                 <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb189-2425"><a href="#cb189-2425" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2426"><a href="#cb189-2426" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2429"><a href="#cb189-2429" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2430"><a href="#cb189-2430" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2431"><a href="#cb189-2431" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2432"><a href="#cb189-2432" aria-hidden="true" tabindex="-1"></a>envfit_temp_adapt_md</span>
<span id="cb189-2433"><a href="#cb189-2433" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2434"><a href="#cb189-2434" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2437"><a href="#cb189-2437" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2438"><a href="#cb189-2438" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2439"><a href="#cb189-2439" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_temp_adapt_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb189-2440"><a href="#cb189-2440" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_temp_adapt_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb189-2441"><a href="#cb189-2441" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_temp_adapt_md<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.4</span>))</span>
<span id="cb189-2442"><a href="#cb189-2442" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(temp_adapt_md_signif_hits)</span>
<span id="cb189-2443"><a href="#cb189-2443" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(temp_adapt_md_signif_hits)</span>
<span id="cb189-2444"><a href="#cb189-2444" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif <span class="ot">&lt;-</span> temp_adapt_md_scores[temp_adapt_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb189-2445"><a href="#cb189-2445" aria-hidden="true" tabindex="-1"></a>                                               temp_adapt_md_signif_hits,]</span>
<span id="cb189-2446"><a href="#cb189-2446" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2447"><a href="#cb189-2447" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2448"><a href="#cb189-2448" aria-hidden="true" tabindex="-1"></a><span class="in">`envfit`</span> found that <span class="in">`r temp_adapt_md_signif$parameters`</span> were significantly correlated. </span>
<span id="cb189-2449"><a href="#cb189-2449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2450"><a href="#cb189-2450" aria-hidden="true" tabindex="-1"></a>Now let's see if the same parameters are significant for the <span class="in">`envfit`</span> and <span class="in">`bioenv`</span> analyses.</span>
<span id="cb189-2451"><a href="#cb189-2451" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2454"><a href="#cb189-2454" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2455"><a href="#cb189-2455" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2456"><a href="#cb189-2456" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2457"><a href="#cb189-2457" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hold</span></span>
<span id="cb189-2458"><a href="#cb189-2458" aria-hidden="true" tabindex="-1"></a><span class="co">#| comment: '' </span></span>
<span id="cb189-2459"><a href="#cb189-2459" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Significant parameters from bioenv analysis."</span>)</span>
<span id="cb189-2460"><a href="#cb189-2460" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(<span class="fu">summary</span>(temp_adapt_bioenv_ind_mantel))</span>
<span id="cb189-2461"><a href="#cb189-2461" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-2462"><a href="#cb189-2462" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-2463"><a href="#cb189-2463" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Significant parameters from envfit analysis."</span>)</span>
<span id="cb189-2464"><a href="#cb189-2464" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif<span class="sc">$</span>parameters</span>
<span id="cb189-2465"><a href="#cb189-2465" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-2466"><a href="#cb189-2466" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-2467"><a href="#cb189-2467" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Found in bioenv but not envfit."</span>)</span>
<span id="cb189-2468"><a href="#cb189-2468" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">setdiff</span>(<span class="fu">row.names</span>(<span class="fu">summary</span>(temp_adapt_bioenv_ind_mantel)), </span>
<span id="cb189-2469"><a href="#cb189-2469" aria-hidden="true" tabindex="-1"></a>                       temp_adapt_md_signif<span class="sc">$</span>parameters)</span>
<span id="cb189-2470"><a href="#cb189-2470" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-2471"><a href="#cb189-2471" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-2472"><a href="#cb189-2472" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Found in envfit but not bioenv."</span>)</span>
<span id="cb189-2473"><a href="#cb189-2473" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">setdiff</span>(temp_adapt_md_signif<span class="sc">$</span>parameters, </span>
<span id="cb189-2474"><a href="#cb189-2474" aria-hidden="true" tabindex="-1"></a>              <span class="fu">row.names</span>(<span class="fu">summary</span>(temp_adapt_bioenv_ind_mantel)))</span>
<span id="cb189-2475"><a href="#cb189-2475" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-2476"><a href="#cb189-2476" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-2477"><a href="#cb189-2477" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Found in envfit and  bioenv."</span>)</span>
<span id="cb189-2478"><a href="#cb189-2478" aria-hidden="true" tabindex="-1"></a>temp_adapt_sig_diff <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">union</span>(temp_adapt_md_signif<span class="sc">$</span>parameters, </span>
<span id="cb189-2479"><a href="#cb189-2479" aria-hidden="true" tabindex="-1"></a>                                   <span class="fu">row.names</span>(<span class="fu">summary</span>(temp_adapt_bioenv_ind_mantel)))</span>
<span id="cb189-2480"><a href="#cb189-2480" aria-hidden="true" tabindex="-1"></a>temp_adapt_sig_diff</span>
<span id="cb189-2481"><a href="#cb189-2481" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2482"><a href="#cb189-2482" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2485"><a href="#cb189-2485" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2486"><a href="#cb189-2486" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2487"><a href="#cb189-2487" aria-hidden="true" tabindex="-1"></a><span class="co">#new_temp_adapt_md_signif_hits &lt;- base::append(temp_adapt_md_signif_hits, temp_adapt_sig_diff)</span></span>
<span id="cb189-2488"><a href="#cb189-2488" aria-hidden="true" tabindex="-1"></a>new_temp_adapt_md_signif_hits <span class="ot">&lt;-</span> temp_adapt_sig_diff</span>
<span id="cb189-2489"><a href="#cb189-2489" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_all <span class="ot">&lt;-</span> temp_adapt_md_scores[temp_adapt_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb189-2490"><a href="#cb189-2490" aria-hidden="true" tabindex="-1"></a>                                                   new_temp_adapt_md_signif_hits,]</span>
<span id="cb189-2491"><a href="#cb189-2491" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2492"><a href="#cb189-2492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2493"><a href="#cb189-2493" aria-hidden="true" tabindex="-1"></a>Check. Next, we run <span class="in">`envfit`</span> for the ASVs.</span>
<span id="cb189-2494"><a href="#cb189-2494" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2497"><a href="#cb189-2497" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2498"><a href="#cb189-2498" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2499"><a href="#cb189-2499" aria-hidden="true" tabindex="-1"></a>envfit_temp_adapt_asv <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, </span>
<span id="cb189-2500"><a href="#cb189-2500" aria-hidden="true" tabindex="-1"></a>                                tmp_comm[, <span class="fu">order</span>(<span class="fu">colSums</span>(<span class="sc">-</span>tmp_comm))][, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],</span>
<span id="cb189-2501"><a href="#cb189-2501" aria-hidden="true" tabindex="-1"></a>                                <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb189-2502"><a href="#cb189-2502" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2503"><a href="#cb189-2503" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">"species"</span>)</span>
<span id="cb189-2504"><a href="#cb189-2504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2505"><a href="#cb189-2505" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_scores <span class="ot">&lt;-</span> temp_adapt_asv_scores <span class="sc">%&gt;%</span> </span>
<span id="cb189-2506"><a href="#cb189-2506" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span></span>
<span id="cb189-2507"><a href="#cb189-2507" aria-hidden="true" tabindex="-1"></a>                         tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"Label"</span>)</span>
<span id="cb189-2508"><a href="#cb189-2508" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-2509"><a href="#cb189-2509" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2510"><a href="#cb189-2510" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2513"><a href="#cb189-2513" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2514"><a href="#cb189-2514" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2515"><a href="#cb189-2515" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2516"><a href="#cb189-2516" aria-hidden="true" tabindex="-1"></a>envfit_temp_adapt_asv</span>
<span id="cb189-2517"><a href="#cb189-2517" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2518"><a href="#cb189-2518" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2521"><a href="#cb189-2521" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2522"><a href="#cb189-2522" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2523"><a href="#cb189-2523" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_temp_adapt_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb189-2524"><a href="#cb189-2524" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_temp_adapt_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb189-2525"><a href="#cb189-2525" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_temp_adapt_asv<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.5</span>))</span>
<span id="cb189-2526"><a href="#cb189-2526" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(temp_adapt_asv_signif_hits)</span>
<span id="cb189-2527"><a href="#cb189-2527" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(temp_adapt_asv_signif_hits)</span>
<span id="cb189-2528"><a href="#cb189-2528" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif <span class="ot">&lt;-</span> temp_adapt_asv_scores[temp_adapt_asv_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb189-2529"><a href="#cb189-2529" aria-hidden="true" tabindex="-1"></a>                                                 temp_adapt_asv_signif_hits,]</span>
<span id="cb189-2530"><a href="#cb189-2530" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2531"><a href="#cb189-2531" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2534"><a href="#cb189-2534" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2535"><a href="#cb189-2535" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2536"><a href="#cb189-2536" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2537"><a href="#cb189-2537" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif</span>
<span id="cb189-2538"><a href="#cb189-2538" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2539"><a href="#cb189-2539" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb189-2540"><a href="#cb189-2540" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2543"><a href="#cb189-2543" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2544"><a href="#cb189-2544" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2545"><a href="#cb189-2545" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_all<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">"metadata"</span></span>
<span id="cb189-2546"><a href="#cb189-2546" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">"ASV"</span></span>
<span id="cb189-2547"><a href="#cb189-2547" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioplot_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(temp_adapt_md_signif_all, temp_adapt_asv_signif)</span>
<span id="cb189-2548"><a href="#cb189-2548" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2549"><a href="#cb189-2549" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2550"><a href="#cb189-2550" aria-hidden="true" tabindex="-1"></a>The last thing to do is categorize parameters scores and ASV scores into different variable types for plotting.</span>
<span id="cb189-2551"><a href="#cb189-2551" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2554"><a href="#cb189-2554" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2555"><a href="#cb189-2555" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2556"><a href="#cb189-2556" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioplot_data_md <span class="ot">&lt;-</span> <span class="fu">subset</span>(temp_adapt_bioplot_data, </span>
<span id="cb189-2557"><a href="#cb189-2557" aria-hidden="true" tabindex="-1"></a>                                  temp_adapt_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">"metadata"</span>)</span>
<span id="cb189-2558"><a href="#cb189-2558" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioplot_data_asv <span class="ot">&lt;-</span> <span class="fu">subset</span>(temp_adapt_bioplot_data, </span>
<span id="cb189-2559"><a href="#cb189-2559" aria-hidden="true" tabindex="-1"></a>                                   temp_adapt_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">"ASV"</span>)</span>
<span id="cb189-2560"><a href="#cb189-2560" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2561"><a href="#cb189-2561" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2562"><a href="#cb189-2562" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb189-2563"><a href="#cb189-2563" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show the code for the plot<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb189-2564"><a href="#cb189-2564" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2567"><a href="#cb189-2567" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2568"><a href="#cb189-2568" aria-hidden="true" tabindex="-1"></a>temp_adapt_cap_vals <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(temp_adapt_cap<span class="sc">$</span>CCA<span class="sc">$</span>eig[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span>
<span id="cb189-2569"><a href="#cb189-2569" aria-hidden="true" tabindex="-1"></a>temp_adapt_cap1 <span class="ot">&lt;-</span> <span class="fu">signif</span>((temp_adapt_cap_vals[<span class="dv">1</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb189-2570"><a href="#cb189-2570" aria-hidden="true" tabindex="-1"></a>temp_adapt_cap2 <span class="ot">&lt;-</span> <span class="fu">signif</span>((temp_adapt_cap_vals[<span class="dv">2</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb189-2571"><a href="#cb189-2571" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2572"><a href="#cb189-2572" aria-hidden="true" tabindex="-1"></a>cpa1_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"CAP1"</span>, <span class="st">" ("</span>, temp_adapt_cap1, <span class="st">"%)"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb189-2573"><a href="#cb189-2573" aria-hidden="true" tabindex="-1"></a>cpa2_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"CAP2"</span>, <span class="st">" ("</span>, temp_adapt_cap2, <span class="st">"%)"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb189-2574"><a href="#cb189-2574" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2575"><a href="#cb189-2575" aria-hidden="true" tabindex="-1"></a>swel_col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#2271B2"</span>, <span class="st">"#71B222"</span>, <span class="st">"#B22271"</span>)</span>
<span id="cb189-2576"><a href="#cb189-2576" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2577"><a href="#cb189-2577" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(temp_adapt_plot_data) <span class="sc">+</span></span>
<span id="cb189-2578"><a href="#cb189-2578" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">shape =</span> TREAT,</span>
<span id="cb189-2579"><a href="#cb189-2579" aria-hidden="true" tabindex="-1"></a>                           <span class="at">colour =</span> TREAT_T), <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb189-2580"><a href="#cb189-2580" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> swel_col) <span class="sc">+</span></span>
<span id="cb189-2581"><a href="#cb189-2581" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_text(data = temp_adapt_plot_data, aes(x = CAP1, y = CAP2, #UNCOMMENT to add sample labels</span></span>
<span id="cb189-2582"><a href="#cb189-2582" aria-hidden="true" tabindex="-1"></a><span class="co">#                              label = SampleID), size = 3) + </span></span>
<span id="cb189-2583"><a href="#cb189-2583" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">xend =</span> CAP1, <span class="at">yend =</span> CAP2),</span>
<span id="cb189-2584"><a href="#cb189-2584" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> temp_adapt_bioplot_data_md, <span class="at">linetype =</span> <span class="st">"solid"</span>,</span>
<span id="cb189-2585"><a href="#cb189-2585" aria-hidden="true" tabindex="-1"></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">"cm"</span>)), <span class="at">size =</span> <span class="fl">0.4</span>,</span>
<span id="cb189-2586"><a href="#cb189-2586" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">"#191919"</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb189-2587"><a href="#cb189-2587" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> temp_adapt_bioplot_data_md, </span>
<span id="cb189-2588"><a href="#cb189-2588" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">label =</span> parameters), <span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb189-2589"><a href="#cb189-2589" aria-hidden="true" tabindex="-1"></a>            <span class="at">nudge_x =</span> <span class="fl">0.1</span>, <span class="at">nudge_y =</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb189-2590"><a href="#cb189-2590" aria-hidden="true" tabindex="-1"></a>  <span class="do">### This code adds ASV vectors, Uncomment to add</span></span>
<span id="cb189-2591"><a href="#cb189-2591" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_segment(aes(x = 0, y = 0, xend = CAP1, yend = CAP2),</span></span>
<span id="cb189-2592"><a href="#cb189-2592" aria-hidden="true" tabindex="-1"></a><span class="co">#               data = temp_adapt_bioplot_data_asv, linetype = "solid",</span></span>
<span id="cb189-2593"><a href="#cb189-2593" aria-hidden="true" tabindex="-1"></a><span class="co">#               arrow = arrow(length = unit(0.3, "cm")), size = 0.2,</span></span>
<span id="cb189-2594"><a href="#cb189-2594" aria-hidden="true" tabindex="-1"></a><span class="co">#               color = "#676767") +</span></span>
<span id="cb189-2595"><a href="#cb189-2595" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_text(data = temp_adapt_bioplot_data_asv, </span></span>
<span id="cb189-2596"><a href="#cb189-2596" aria-hidden="true" tabindex="-1"></a><span class="co">#            aes(x = CAP1, y = CAP2, label = parameters), size = 2.5,</span></span>
<span id="cb189-2597"><a href="#cb189-2597" aria-hidden="true" tabindex="-1"></a><span class="co">#            nudge_x = 0.05, nudge_y = 0.05) +</span></span>
<span id="cb189-2598"><a href="#cb189-2598" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span> </span>
<span id="cb189-2599"><a href="#cb189-2599" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Capscale Analysis"</span>,</span>
<span id="cb189-2600"><a href="#cb189-2600" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">"Temperature Adaptation"</span>,</span>
<span id="cb189-2601"><a href="#cb189-2601" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> cpa1_lab, </span>
<span id="cb189-2602"><a href="#cb189-2602" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> cpa2_lab)</span>
<span id="cb189-2603"><a href="#cb189-2603" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot <span class="ot">&lt;-</span> temp_adapt_plot <span class="sc">+</span> <span class="fu">coord_fixed</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb189-2604"><a href="#cb189-2604" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot</span>
<span id="cb189-2605"><a href="#cb189-2605" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">"files/metadata/figures/ssu18_temp_adapt_capscale.png"</span>,</span>
<span id="cb189-2606"><a href="#cb189-2606" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">16</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">units =</span> <span class="st">'cm'</span>, <span class="at">res =</span> <span class="dv">600</span>, <span class="at">bg =</span> <span class="st">"white"</span>)</span>
<span id="cb189-2607"><a href="#cb189-2607" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot</span>
<span id="cb189-2608"><a href="#cb189-2608" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb189-2609"><a href="#cb189-2609" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"files/metadata/figures/ssu18_temp_adapt_capscale.pdf"</span>,</span>
<span id="cb189-2610"><a href="#cb189-2610" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">5</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb189-2611"><a href="#cb189-2611" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot</span>
<span id="cb189-2612"><a href="#cb189-2612" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb189-2613"><a href="#cb189-2613" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2614"><a href="#cb189-2614" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb189-2615"><a href="#cb189-2615" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2616"><a href="#cb189-2616" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2619"><a href="#cb189-2619" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2620"><a href="#cb189-2620" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2621"><a href="#cb189-2621" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-2622"><a href="#cb189-2622" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2623"><a href="#cb189-2623" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2624"><a href="#cb189-2624" aria-hidden="true" tabindex="-1"></a><span class="fu">## Capscale Plots</span></span>
<span id="cb189-2625"><a href="#cb189-2625" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2626"><a href="#cb189-2626" aria-hidden="true" tabindex="-1"></a>::: {.column-body}</span>
<span id="cb189-2627"><a href="#cb189-2627" aria-hidden="true" tabindex="-1"></a>::: {.panel-tabset}</span>
<span id="cb189-2628"><a href="#cb189-2628" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2629"><a href="#cb189-2629" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Edaphic properties</span></span>
<span id="cb189-2630"><a href="#cb189-2630" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2633"><a href="#cb189-2633" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2634"><a href="#cb189-2634" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2635"><a href="#cb189-2635" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2636"><a href="#cb189-2636" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb189-2637"><a href="#cb189-2637" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4.5</span></span>
<span id="cb189-2638"><a href="#cb189-2638" aria-hidden="true" tabindex="-1"></a>edaphic_plot</span>
<span id="cb189-2639"><a href="#cb189-2639" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2640"><a href="#cb189-2640" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig_ssu("ssu_cap_edaph_plots")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-2641"><a href="#cb189-2641" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2642"><a href="#cb189-2642" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Soil Functional Response</span></span>
<span id="cb189-2643"><a href="#cb189-2643" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2646"><a href="#cb189-2646" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2647"><a href="#cb189-2647" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2648"><a href="#cb189-2648" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2649"><a href="#cb189-2649" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb189-2650"><a href="#cb189-2650" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4.5</span></span>
<span id="cb189-2651"><a href="#cb189-2651" aria-hidden="true" tabindex="-1"></a>soil_funct_plot</span>
<span id="cb189-2652"><a href="#cb189-2652" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2653"><a href="#cb189-2653" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig_ssu("ssu_cap_funct_plots")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-2654"><a href="#cb189-2654" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2655"><a href="#cb189-2655" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Temperature Adaptation</span></span>
<span id="cb189-2656"><a href="#cb189-2656" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2659"><a href="#cb189-2659" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2660"><a href="#cb189-2660" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2661"><a href="#cb189-2661" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2662"><a href="#cb189-2662" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb189-2663"><a href="#cb189-2663" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4.5</span></span>
<span id="cb189-2664"><a href="#cb189-2664" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot</span>
<span id="cb189-2665"><a href="#cb189-2665" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2666"><a href="#cb189-2666" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig_ssu("ssu_cap_temp_plots")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-2667"><a href="#cb189-2667" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb189-2668"><a href="#cb189-2668" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb189-2669"><a href="#cb189-2669" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2672"><a href="#cb189-2672" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2673"><a href="#cb189-2673" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2674"><a href="#cb189-2674" aria-hidden="true" tabindex="-1"></a>ssu18_capscale_plots <span class="ot">&lt;-</span> <span class="fu">ggarrange</span>(</span>
<span id="cb189-2675"><a href="#cb189-2675" aria-hidden="true" tabindex="-1"></a>  edaphic_plot,</span>
<span id="cb189-2676"><a href="#cb189-2676" aria-hidden="true" tabindex="-1"></a>  soil_funct_plot,</span>
<span id="cb189-2677"><a href="#cb189-2677" aria-hidden="true" tabindex="-1"></a>  temp_adapt_plot,</span>
<span id="cb189-2678"><a href="#cb189-2678" aria-hidden="true" tabindex="-1"></a>  <span class="at">ncol =</span> <span class="dv">3</span>, <span class="at">nrow =</span> <span class="dv">1</span>, <span class="at">common.legend =</span> <span class="cn">TRUE</span>, <span class="at">legend =</span> <span class="st">"bottom"</span>)</span>
<span id="cb189-2679"><a href="#cb189-2679" aria-hidden="true" tabindex="-1"></a>ssu18_capscale_plots</span>
<span id="cb189-2680"><a href="#cb189-2680" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb189-2681"><a href="#cb189-2681" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">"files/metadata/figures/ssu18_capscale_plots.png"</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">width =</span> <span class="dv">36</span>,</span>
<span id="cb189-2682"><a href="#cb189-2682" aria-hidden="true" tabindex="-1"></a>    <span class="at">units =</span> <span class="st">'cm'</span>, <span class="at">res =</span> <span class="dv">600</span>, <span class="at">bg =</span> <span class="st">"white"</span>)</span>
<span id="cb189-2683"><a href="#cb189-2683" aria-hidden="true" tabindex="-1"></a>ssu18_capscale_plots</span>
<span id="cb189-2684"><a href="#cb189-2684" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb189-2685"><a href="#cb189-2685" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"files/metadata/figures/ssu18_capscale_plots.pdf"</span>, <span class="at">height =</span> <span class="dv">6</span>, <span class="at">width =</span> <span class="dv">18</span>)</span>
<span id="cb189-2686"><a href="#cb189-2686" aria-hidden="true" tabindex="-1"></a>ssu18_capscale_plots</span>
<span id="cb189-2687"><a href="#cb189-2687" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb189-2688"><a href="#cb189-2688" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2689"><a href="#cb189-2689" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2690"><a href="#cb189-2690" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2693"><a href="#cb189-2693" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2694"><a href="#cb189-2694" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2695"><a href="#cb189-2695" aria-hidden="true" tabindex="-1"></a><span class="do">###########################################</span></span>
<span id="cb189-2696"><a href="#cb189-2696" aria-hidden="true" tabindex="-1"></a><span class="do">######## CODE FOR SUMMARY DATA ############</span></span>
<span id="cb189-2697"><a href="#cb189-2697" aria-hidden="true" tabindex="-1"></a><span class="do">###########################################</span></span>
<span id="cb189-2698"><a href="#cb189-2698" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2699"><a href="#cb189-2699" aria-hidden="true" tabindex="-1"></a><span class="do">## Edaphic</span></span>
<span id="cb189-2700"><a href="#cb189-2700" aria-hidden="true" tabindex="-1"></a>bioenv_set <span class="ot">&lt;-</span> edaphic_bioenv_ind_mantel</span>
<span id="cb189-2701"><a href="#cb189-2701" aria-hidden="true" tabindex="-1"></a>envfit_set <span class="ot">&lt;-</span> envfit_edaphic_md</span>
<span id="cb189-2702"><a href="#cb189-2702" aria-hidden="true" tabindex="-1"></a>res_name <span class="ot">&lt;-</span> <span class="st">"ssu18_edaphic_sig_summary"</span></span>
<span id="cb189-2703"><a href="#cb189-2703" aria-hidden="true" tabindex="-1"></a>metadata_set <span class="ot">&lt;-</span> <span class="st">"edaphic"</span></span>
<span id="cb189-2704"><a href="#cb189-2704" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2705"><a href="#cb189-2705" aria-hidden="true" tabindex="-1"></a><span class="do">## BIOENV</span></span>
<span id="cb189-2706"><a href="#cb189-2706" aria-hidden="true" tabindex="-1"></a>bioenv_tab <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-2707"><a href="#cb189-2707" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">names</span>(bioenv_set)) {</span>
<span id="cb189-2708"><a href="#cb189-2708" aria-hidden="true" tabindex="-1"></a>  tmp_r <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(bioenv_set[[i]]<span class="sc">$</span>statistic)</span>
<span id="cb189-2709"><a href="#cb189-2709" aria-hidden="true" tabindex="-1"></a>  <span class="fu">row.names</span>(tmp_r) <span class="ot">&lt;-</span> i</span>
<span id="cb189-2710"><a href="#cb189-2710" aria-hidden="true" tabindex="-1"></a>  tmp_p <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(bioenv_set[[i]]<span class="sc">$</span>signif)</span>
<span id="cb189-2711"><a href="#cb189-2711" aria-hidden="true" tabindex="-1"></a>  <span class="fu">row.names</span>(tmp_p) <span class="ot">&lt;-</span> i</span>
<span id="cb189-2712"><a href="#cb189-2712" aria-hidden="true" tabindex="-1"></a>  bioenv_tab <span class="ot">&lt;-</span> <span class="fu">rbind</span>(bioenv_tab, <span class="fu">data.frame</span>(tmp_r, tmp_p))</span>
<span id="cb189-2713"><a href="#cb189-2713" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-2714"><a href="#cb189-2714" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-2715"><a href="#cb189-2715" aria-hidden="true" tabindex="-1"></a>bioenv_tab <span class="ot">&lt;-</span> bioenv_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"r (bioenv)"</span> <span class="ot">=</span> <span class="dv">1</span>, <span class="st">"pval (bioenv)"</span> <span class="ot">=</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb189-2716"><a href="#cb189-2716" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"parameter"</span>)</span>
<span id="cb189-2717"><a href="#cb189-2717" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2718"><a href="#cb189-2718" aria-hidden="true" tabindex="-1"></a><span class="do">## ENVFIT</span></span>
<span id="cb189-2719"><a href="#cb189-2719" aria-hidden="true" tabindex="-1"></a>tmp_r <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(envfit_set<span class="sc">$</span>vectors<span class="sc">$</span>r)</span>
<span id="cb189-2720"><a href="#cb189-2720" aria-hidden="true" tabindex="-1"></a>tmp_r <span class="ot">&lt;-</span> tmp_r <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"parameter"</span>)</span>
<span id="cb189-2721"><a href="#cb189-2721" aria-hidden="true" tabindex="-1"></a>tmp_p <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(envfit_set<span class="sc">$</span>vectors<span class="sc">$</span>pvals)</span>
<span id="cb189-2722"><a href="#cb189-2722" aria-hidden="true" tabindex="-1"></a>tmp_p <span class="ot">&lt;-</span> tmp_p <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"parameter"</span>)</span>
<span id="cb189-2723"><a href="#cb189-2723" aria-hidden="true" tabindex="-1"></a>tmp_rp <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(tmp_r, tmp_p, <span class="at">by =</span> <span class="st">"parameter"</span>)</span>
<span id="cb189-2724"><a href="#cb189-2724" aria-hidden="true" tabindex="-1"></a>tmp_rp <span class="ot">&lt;-</span> tmp_rp <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"r2 (envfit)"</span> <span class="ot">=</span> <span class="dv">2</span>, <span class="st">"Pr(&gt;r) (envfit)"</span> <span class="ot">=</span> <span class="dv">3</span>)</span>
<span id="cb189-2725"><a href="#cb189-2725" aria-hidden="true" tabindex="-1"></a>tmp_rp <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(tmp_rp, <span class="fu">c</span>(tmp_rp<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;r)</span><span class="st">`</span> <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> tmp_rp<span class="sc">$</span>r2 <span class="sc">&gt;</span> <span class="fl">0.4</span>))</span>
<span id="cb189-2726"><a href="#cb189-2726" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2727"><a href="#cb189-2727" aria-hidden="true" tabindex="-1"></a>tmp_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(tmp_rp, bioenv_tab, <span class="at">by =</span> <span class="st">"parameter"</span>) </span>
<span id="cb189-2728"><a href="#cb189-2728" aria-hidden="true" tabindex="-1"></a>tmp_summary<span class="sc">$</span>metadata_set <span class="ot">&lt;-</span> metadata_set</span>
<span id="cb189-2729"><a href="#cb189-2729" aria-hidden="true" tabindex="-1"></a>tmp_summary <span class="ot">&lt;-</span> tmp_summary[, <span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)]</span>
<span id="cb189-2730"><a href="#cb189-2730" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(res_name, tmp_summary)</span>
<span id="cb189-2731"><a href="#cb189-2731" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-2732"><a href="#cb189-2732" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2733"><a href="#cb189-2733" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2736"><a href="#cb189-2736" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2737"><a href="#cb189-2737" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2738"><a href="#cb189-2738" aria-hidden="true" tabindex="-1"></a><span class="do">## SOIL_FUNCT</span></span>
<span id="cb189-2739"><a href="#cb189-2739" aria-hidden="true" tabindex="-1"></a>bioenv_set <span class="ot">&lt;-</span> soil_funct_bioenv_ind_mantel</span>
<span id="cb189-2740"><a href="#cb189-2740" aria-hidden="true" tabindex="-1"></a>envfit_set <span class="ot">&lt;-</span> envfit_soil_funct_md</span>
<span id="cb189-2741"><a href="#cb189-2741" aria-hidden="true" tabindex="-1"></a>res_name <span class="ot">&lt;-</span> <span class="st">"ssu18_soil_funct_sig_summary"</span></span>
<span id="cb189-2742"><a href="#cb189-2742" aria-hidden="true" tabindex="-1"></a>metadata_set <span class="ot">&lt;-</span> <span class="st">"functional_response"</span></span>
<span id="cb189-2743"><a href="#cb189-2743" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2744"><a href="#cb189-2744" aria-hidden="true" tabindex="-1"></a><span class="do">## BIOENV</span></span>
<span id="cb189-2745"><a href="#cb189-2745" aria-hidden="true" tabindex="-1"></a>bioenv_tab <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-2746"><a href="#cb189-2746" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">names</span>(bioenv_set)) {</span>
<span id="cb189-2747"><a href="#cb189-2747" aria-hidden="true" tabindex="-1"></a>  tmp_r <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(bioenv_set[[i]]<span class="sc">$</span>statistic)</span>
<span id="cb189-2748"><a href="#cb189-2748" aria-hidden="true" tabindex="-1"></a>  <span class="fu">row.names</span>(tmp_r) <span class="ot">&lt;-</span> i</span>
<span id="cb189-2749"><a href="#cb189-2749" aria-hidden="true" tabindex="-1"></a>  tmp_p <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(bioenv_set[[i]]<span class="sc">$</span>signif)</span>
<span id="cb189-2750"><a href="#cb189-2750" aria-hidden="true" tabindex="-1"></a>  <span class="fu">row.names</span>(tmp_p) <span class="ot">&lt;-</span> i</span>
<span id="cb189-2751"><a href="#cb189-2751" aria-hidden="true" tabindex="-1"></a>  bioenv_tab <span class="ot">&lt;-</span> <span class="fu">rbind</span>(bioenv_tab, <span class="fu">data.frame</span>(tmp_r, tmp_p))</span>
<span id="cb189-2752"><a href="#cb189-2752" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-2753"><a href="#cb189-2753" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-2754"><a href="#cb189-2754" aria-hidden="true" tabindex="-1"></a>bioenv_tab <span class="ot">&lt;-</span> bioenv_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"r (bioenv)"</span> <span class="ot">=</span> <span class="dv">1</span>, <span class="st">"pval (bioenv)"</span> <span class="ot">=</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb189-2755"><a href="#cb189-2755" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"parameter"</span>)</span>
<span id="cb189-2756"><a href="#cb189-2756" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2757"><a href="#cb189-2757" aria-hidden="true" tabindex="-1"></a><span class="do">## ENVFIT</span></span>
<span id="cb189-2758"><a href="#cb189-2758" aria-hidden="true" tabindex="-1"></a>tmp_r <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(envfit_set<span class="sc">$</span>vectors<span class="sc">$</span>r)</span>
<span id="cb189-2759"><a href="#cb189-2759" aria-hidden="true" tabindex="-1"></a>tmp_r <span class="ot">&lt;-</span> tmp_r <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"parameter"</span>)</span>
<span id="cb189-2760"><a href="#cb189-2760" aria-hidden="true" tabindex="-1"></a>tmp_p <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(envfit_set<span class="sc">$</span>vectors<span class="sc">$</span>pvals)</span>
<span id="cb189-2761"><a href="#cb189-2761" aria-hidden="true" tabindex="-1"></a>tmp_p <span class="ot">&lt;-</span> tmp_p <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"parameter"</span>)</span>
<span id="cb189-2762"><a href="#cb189-2762" aria-hidden="true" tabindex="-1"></a>tmp_rp <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(tmp_r, tmp_p, <span class="at">by =</span> <span class="st">"parameter"</span>)</span>
<span id="cb189-2763"><a href="#cb189-2763" aria-hidden="true" tabindex="-1"></a>tmp_rp <span class="ot">&lt;-</span> tmp_rp <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"r2 (envfit)"</span> <span class="ot">=</span> <span class="dv">2</span>, <span class="st">"Pr(&gt;r) (envfit)"</span> <span class="ot">=</span> <span class="dv">3</span>)</span>
<span id="cb189-2764"><a href="#cb189-2764" aria-hidden="true" tabindex="-1"></a>tmp_rp <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(tmp_rp, <span class="fu">c</span>(tmp_rp<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;r)</span><span class="st">`</span> <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> tmp_rp<span class="sc">$</span>r2 <span class="sc">&gt;</span> <span class="fl">0.4</span>))</span>
<span id="cb189-2765"><a href="#cb189-2765" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2766"><a href="#cb189-2766" aria-hidden="true" tabindex="-1"></a>tmp_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(tmp_rp, bioenv_tab, <span class="at">by =</span> <span class="st">"parameter"</span>) </span>
<span id="cb189-2767"><a href="#cb189-2767" aria-hidden="true" tabindex="-1"></a>tmp_summary<span class="sc">$</span>metadata_set <span class="ot">&lt;-</span> metadata_set</span>
<span id="cb189-2768"><a href="#cb189-2768" aria-hidden="true" tabindex="-1"></a>tmp_summary <span class="ot">&lt;-</span> tmp_summary[, <span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)]</span>
<span id="cb189-2769"><a href="#cb189-2769" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(res_name, tmp_summary)</span>
<span id="cb189-2770"><a href="#cb189-2770" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-2771"><a href="#cb189-2771" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2772"><a href="#cb189-2772" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2775"><a href="#cb189-2775" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2776"><a href="#cb189-2776" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2777"><a href="#cb189-2777" aria-hidden="true" tabindex="-1"></a><span class="do">## TEMP</span></span>
<span id="cb189-2778"><a href="#cb189-2778" aria-hidden="true" tabindex="-1"></a>bioenv_set <span class="ot">&lt;-</span> temp_adapt_bioenv_ind_mantel</span>
<span id="cb189-2779"><a href="#cb189-2779" aria-hidden="true" tabindex="-1"></a>envfit_set <span class="ot">&lt;-</span> envfit_temp_adapt_md</span>
<span id="cb189-2780"><a href="#cb189-2780" aria-hidden="true" tabindex="-1"></a>res_name <span class="ot">&lt;-</span> <span class="st">"ssu18_temp_adapt_sig_summary"</span></span>
<span id="cb189-2781"><a href="#cb189-2781" aria-hidden="true" tabindex="-1"></a>metadata_set <span class="ot">&lt;-</span> <span class="st">"temperature adaptation"</span></span>
<span id="cb189-2782"><a href="#cb189-2782" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2783"><a href="#cb189-2783" aria-hidden="true" tabindex="-1"></a><span class="do">## BIOENV</span></span>
<span id="cb189-2784"><a href="#cb189-2784" aria-hidden="true" tabindex="-1"></a>bioenv_tab <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-2785"><a href="#cb189-2785" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">names</span>(bioenv_set)) {</span>
<span id="cb189-2786"><a href="#cb189-2786" aria-hidden="true" tabindex="-1"></a>  tmp_r <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(bioenv_set[[i]]<span class="sc">$</span>statistic)</span>
<span id="cb189-2787"><a href="#cb189-2787" aria-hidden="true" tabindex="-1"></a>  <span class="fu">row.names</span>(tmp_r) <span class="ot">&lt;-</span> i</span>
<span id="cb189-2788"><a href="#cb189-2788" aria-hidden="true" tabindex="-1"></a>  tmp_p <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(bioenv_set[[i]]<span class="sc">$</span>signif)</span>
<span id="cb189-2789"><a href="#cb189-2789" aria-hidden="true" tabindex="-1"></a>  <span class="fu">row.names</span>(tmp_p) <span class="ot">&lt;-</span> i</span>
<span id="cb189-2790"><a href="#cb189-2790" aria-hidden="true" tabindex="-1"></a>  bioenv_tab <span class="ot">&lt;-</span> <span class="fu">rbind</span>(bioenv_tab, <span class="fu">data.frame</span>(tmp_r, tmp_p))</span>
<span id="cb189-2791"><a href="#cb189-2791" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-2792"><a href="#cb189-2792" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-2793"><a href="#cb189-2793" aria-hidden="true" tabindex="-1"></a>bioenv_tab <span class="ot">&lt;-</span> bioenv_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"r (bioenv)"</span> <span class="ot">=</span> <span class="dv">1</span>, <span class="st">"pval (bioenv)"</span> <span class="ot">=</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb189-2794"><a href="#cb189-2794" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"parameter"</span>)</span>
<span id="cb189-2795"><a href="#cb189-2795" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2796"><a href="#cb189-2796" aria-hidden="true" tabindex="-1"></a><span class="do">## ENVFIT</span></span>
<span id="cb189-2797"><a href="#cb189-2797" aria-hidden="true" tabindex="-1"></a>tmp_r <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(envfit_set<span class="sc">$</span>vectors<span class="sc">$</span>r)</span>
<span id="cb189-2798"><a href="#cb189-2798" aria-hidden="true" tabindex="-1"></a>tmp_r <span class="ot">&lt;-</span> tmp_r <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"parameter"</span>)</span>
<span id="cb189-2799"><a href="#cb189-2799" aria-hidden="true" tabindex="-1"></a>tmp_p <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(envfit_set<span class="sc">$</span>vectors<span class="sc">$</span>pvals)</span>
<span id="cb189-2800"><a href="#cb189-2800" aria-hidden="true" tabindex="-1"></a>tmp_p <span class="ot">&lt;-</span> tmp_p <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"parameter"</span>)</span>
<span id="cb189-2801"><a href="#cb189-2801" aria-hidden="true" tabindex="-1"></a>tmp_rp <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(tmp_r, tmp_p, <span class="at">by =</span> <span class="st">"parameter"</span>)</span>
<span id="cb189-2802"><a href="#cb189-2802" aria-hidden="true" tabindex="-1"></a>tmp_rp <span class="ot">&lt;-</span> tmp_rp <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"r2 (envfit)"</span> <span class="ot">=</span> <span class="dv">2</span>, <span class="st">"Pr(&gt;r) (envfit)"</span> <span class="ot">=</span> <span class="dv">3</span>)</span>
<span id="cb189-2803"><a href="#cb189-2803" aria-hidden="true" tabindex="-1"></a>tmp_rp <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(tmp_rp, <span class="fu">c</span>(tmp_rp<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;r)</span><span class="st">`</span> <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> tmp_rp<span class="sc">$</span>r2 <span class="sc">&gt;</span> <span class="fl">0.4</span>))</span>
<span id="cb189-2804"><a href="#cb189-2804" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2805"><a href="#cb189-2805" aria-hidden="true" tabindex="-1"></a>tmp_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(tmp_rp, bioenv_tab, <span class="at">by =</span> <span class="st">"parameter"</span>) </span>
<span id="cb189-2806"><a href="#cb189-2806" aria-hidden="true" tabindex="-1"></a>tmp_summary<span class="sc">$</span>metadata_set <span class="ot">&lt;-</span> metadata_set</span>
<span id="cb189-2807"><a href="#cb189-2807" aria-hidden="true" tabindex="-1"></a>tmp_summary <span class="ot">&lt;-</span> tmp_summary[, <span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)]</span>
<span id="cb189-2808"><a href="#cb189-2808" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(res_name, tmp_summary)</span>
<span id="cb189-2809"><a href="#cb189-2809" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-2810"><a href="#cb189-2810" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(bioenv_set, envfit_set, res_name, metadata_set, bioenv_tab)</span>
<span id="cb189-2811"><a href="#cb189-2811" aria-hidden="true" tabindex="-1"></a>ssu18_sig_summary <span class="ot">&lt;-</span> <span class="fu">rbind</span>(ssu18_edaphic_sig_summary, ssu18_soil_funct_sig_summary, ssu18_temp_adapt_sig_summary)</span>
<span id="cb189-2812"><a href="#cb189-2812" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(ssu18_edaphic_sig_summary, ssu18_soil_funct_sig_summary, ssu18_temp_adapt_sig_summary)</span>
<span id="cb189-2813"><a href="#cb189-2813" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2814"><a href="#cb189-2814" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2815"><a href="#cb189-2815" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2818"><a href="#cb189-2818" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2819"><a href="#cb189-2819" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2820"><a href="#cb189-2820" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="st">"page_build/metadata_ssu18_wf.rdata"</span>)</span>
<span id="cb189-2821"><a href="#cb189-2821" aria-hidden="true" tabindex="-1"></a>gdata<span class="sc">::</span><span class="fu">keep</span>(metad, ssu18_sig_summary, <span class="at">sure =</span> <span class="cn">TRUE</span>)</span>
<span id="cb189-2822"><a href="#cb189-2822" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span>
<span id="cb189-2823"><a href="#cb189-2823" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2824"><a href="#cb189-2824" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2825"><a href="#cb189-2825" aria-hidden="true" tabindex="-1"></a><span class="fu"># ITS</span></span>
<span id="cb189-2826"><a href="#cb189-2826" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2829"><a href="#cb189-2829" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2830"><a href="#cb189-2830" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb189-2831"><a href="#cb189-2831" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-2832"><a href="#cb189-2832" aria-hidden="true" tabindex="-1"></a><span class="do">## Load to build page only #2</span></span>
<span id="cb189-2833"><a href="#cb189-2833" aria-hidden="true" tabindex="-1"></a><span class="co">#remove(list = ls())</span></span>
<span id="cb189-2834"><a href="#cb189-2834" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"page_build/metadata_its18_wf.rdata"</span>)</span>
<span id="cb189-2835"><a href="#cb189-2835" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2836"><a href="#cb189-2836" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2837"><a href="#cb189-2837" aria-hidden="true" tabindex="-1"></a><span class="fu">## Data Formating </span></span>
<span id="cb189-2838"><a href="#cb189-2838" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2841"><a href="#cb189-2841" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2842"><a href="#cb189-2842" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb189-2843"><a href="#cb189-2843" aria-hidden="true" tabindex="-1"></a><span class="do">## Initial Load for  ANALYSIS #1</span></span>
<span id="cb189-2844"><a href="#cb189-2844" aria-hidden="true" tabindex="-1"></a><span class="co">#remove(list = ls())</span></span>
<span id="cb189-2845"><a href="#cb189-2845" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">119</span>)</span>
<span id="cb189-2846"><a href="#cb189-2846" aria-hidden="true" tabindex="-1"></a>its18_ps_work <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"files/alpha/rdata/its18_ps_work.rds"</span>)</span>
<span id="cb189-2847"><a href="#cb189-2847" aria-hidden="true" tabindex="-1"></a>its18_ps_filt <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"files/alpha/rdata/its18_ps_filt.rds"</span>)</span>
<span id="cb189-2848"><a href="#cb189-2848" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"files/alpha/rdata/its18_ps_perfect.rds"</span>)</span>
<span id="cb189-2849"><a href="#cb189-2849" aria-hidden="true" tabindex="-1"></a>its18_ps_pime <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"files/alpha/rdata/its18_ps_pime.rds"</span>)</span>
<span id="cb189-2850"><a href="#cb189-2850" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2851"><a href="#cb189-2851" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2852"><a href="#cb189-2852" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>Transform <span class="in">`ps`</span> Objects</span>
<span id="cb189-2853"><a href="#cb189-2853" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2854"><a href="#cb189-2854" aria-hidden="true" tabindex="-1"></a>The first step is to transform the phyloseq objects. </span>
<span id="cb189-2855"><a href="#cb189-2855" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2856"><a href="#cb189-2856" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2859"><a href="#cb189-2859" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2860"><a href="#cb189-2860" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2861"><a href="#cb189-2861" aria-hidden="true" tabindex="-1"></a>samp_ps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"its18_ps_work"</span>, <span class="st">"its18_ps_pime"</span>, <span class="st">"its18_ps_perfect"</span>, </span>
<span id="cb189-2862"><a href="#cb189-2862" aria-hidden="true" tabindex="-1"></a>             <span class="st">"its18_ps_work_otu"</span>, <span class="st">"its18_ps_pime_otu"</span>, <span class="st">"its18_ps_perfect_otu"</span>)</span>
<span id="cb189-2863"><a href="#cb189-2863" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> samp_ps) {</span>
<span id="cb189-2864"><a href="#cb189-2864" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb189-2865"><a href="#cb189-2865" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(tmp_get, <span class="cf">function</span>(otu) <span class="fl">1e5</span> <span class="sc">*</span> otu<span class="sc">/</span><span class="fu">sum</span>(otu))</span>
<span id="cb189-2866"><a href="#cb189-2866" aria-hidden="true" tabindex="-1"></a>     tmp_ps<span class="sc">@</span>phy_tree <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-2867"><a href="#cb189-2867" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">prune_samples</span>(<span class="fu">sample_sums</span>(tmp_ps) <span class="sc">&gt;</span> <span class="dv">0</span>, tmp_ps)</span>
<span id="cb189-2868"><a href="#cb189-2868" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb189-2869"><a href="#cb189-2869" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> tmp_samp <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"TREAT_T"</span> <span class="ot">=</span> <span class="st">"TEMP"</span>)</span>
<span id="cb189-2870"><a href="#cb189-2870" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(tmp_ps, sample_data)</span>
<span id="cb189-2871"><a href="#cb189-2871" aria-hidden="true" tabindex="-1"></a>     <span class="fu">sample_data</span>(tmp_ps) <span class="ot">&lt;-</span> tmp_samp</span>
<span id="cb189-2872"><a href="#cb189-2872" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_prop"</span>))</span>
<span id="cb189-2873"><a href="#cb189-2873" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_name)</span>
<span id="cb189-2874"><a href="#cb189-2874" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_ps)</span>
<span id="cb189-2875"><a href="#cb189-2875" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-2876"><a href="#cb189-2876" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-2877"><a href="#cb189-2877" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2878"><a href="#cb189-2878" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2879"><a href="#cb189-2879" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>Format for <span class="in">`mctoolsr`</span></span>
<span id="cb189-2880"><a href="#cb189-2880" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2881"><a href="#cb189-2881" aria-hidden="true" tabindex="-1"></a>Then a bit of formatting to make the data compatible with <span class="in">`mctoolsr`</span>. </span>
<span id="cb189-2882"><a href="#cb189-2882" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2885"><a href="#cb189-2885" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2886"><a href="#cb189-2886" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2887"><a href="#cb189-2887" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> samp_ps) {</span>
<span id="cb189-2888"><a href="#cb189-2888" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_prop"</span>)))</span>
<span id="cb189-2889"><a href="#cb189-2889" aria-hidden="true" tabindex="-1"></a>     tmp_tax <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(tmp_get))</span>
<span id="cb189-2890"><a href="#cb189-2890" aria-hidden="true" tabindex="-1"></a>     tmp_tax<span class="sc">$</span>ASV_SEQ <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-2891"><a href="#cb189-2891" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb189-2892"><a href="#cb189-2892" aria-hidden="true" tabindex="-1"></a>     tmp_col_names <span class="ot">&lt;-</span> <span class="fu">colnames</span>(tmp_tax)</span>
<span id="cb189-2893"><a href="#cb189-2893" aria-hidden="true" tabindex="-1"></a>     tmp_tax_merge <span class="ot">&lt;-</span> tmp_tax <span class="sc">%&gt;%</span> tidyr<span class="sc">::</span><span class="fu">unite</span>(taxonomy, </span>
<span id="cb189-2894"><a href="#cb189-2894" aria-hidden="true" tabindex="-1"></a>                                               <span class="fu">all_of</span>(tmp_col_names), </span>
<span id="cb189-2895"><a href="#cb189-2895" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">sep =</span> <span class="st">";"</span>)</span>
<span id="cb189-2896"><a href="#cb189-2896" aria-hidden="true" tabindex="-1"></a>     tmp_tax_merge <span class="ot">&lt;-</span> tmp_tax_merge <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"#OTU_ID"</span>)</span>
<span id="cb189-2897"><a href="#cb189-2897" aria-hidden="true" tabindex="-1"></a>     tmp_otu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(tmp_get)))</span>
<span id="cb189-2898"><a href="#cb189-2898" aria-hidden="true" tabindex="-1"></a>     tmp_otu <span class="ot">&lt;-</span> tmp_otu <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"#OTU_ID"</span>)</span>
<span id="cb189-2899"><a href="#cb189-2899" aria-hidden="true" tabindex="-1"></a>     tmp_otu_tax <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_otu, tmp_tax_merge, <span class="at">by =</span> <span class="st">"#OTU_ID"</span>)</span>
<span id="cb189-2900"><a href="#cb189-2900" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2901"><a href="#cb189-2901" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb189-2902"><a href="#cb189-2902" aria-hidden="true" tabindex="-1"></a>     tmp_samp[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>)] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-2903"><a href="#cb189-2903" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> tmp_samp <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"#SampleID"</span>)</span>
<span id="cb189-2904"><a href="#cb189-2904" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2905"><a href="#cb189-2905" aria-hidden="true" tabindex="-1"></a>     tmp_metad <span class="ot">&lt;-</span> metad <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"#SampleID"</span> <span class="ot">=</span> <span class="st">"id"</span>)</span>
<span id="cb189-2906"><a href="#cb189-2906" aria-hidden="true" tabindex="-1"></a>     tmp_metad[,<span class="dv">2</span><span class="sc">:</span><span class="dv">5</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-2907"><a href="#cb189-2907" aria-hidden="true" tabindex="-1"></a>     tmp_md <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_samp, tmp_metad, <span class="at">by =</span> <span class="st">"#SampleID"</span>)</span>
<span id="cb189-2908"><a href="#cb189-2908" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb189-2909"><a href="#cb189-2909" aria-hidden="true" tabindex="-1"></a>     tmp_otu_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_otu_tax"</span>))</span>
<span id="cb189-2910"><a href="#cb189-2910" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_otu_name)</span>
<span id="cb189-2911"><a href="#cb189-2911" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_otu_name, tmp_otu_tax)</span>
<span id="cb189-2912"><a href="#cb189-2912" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2913"><a href="#cb189-2913" aria-hidden="true" tabindex="-1"></a>     tmp_md_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_md"</span>))</span>
<span id="cb189-2914"><a href="#cb189-2914" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_md_name)</span>
<span id="cb189-2915"><a href="#cb189-2915" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_md_name, tmp_md)</span>
<span id="cb189-2916"><a href="#cb189-2916" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2917"><a href="#cb189-2917" aria-hidden="true" tabindex="-1"></a>     tmp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"files/metadata/tables/"</span>)</span>
<span id="cb189-2918"><a href="#cb189-2918" aria-hidden="true" tabindex="-1"></a>     <span class="fu">write_delim</span>(tmp_otu_tax, <span class="fu">paste</span>(tmp_path, tmp_otu_name, <span class="st">".txt"</span>, <span class="at">sep =</span> <span class="st">""</span>), <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb189-2919"><a href="#cb189-2919" aria-hidden="true" tabindex="-1"></a>     <span class="fu">write_delim</span>(tmp_md, <span class="fu">paste</span>(tmp_path, tmp_md_name, <span class="st">".txt"</span>, <span class="at">sep =</span> <span class="st">""</span>), <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb189-2920"><a href="#cb189-2920" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-2921"><a href="#cb189-2921" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-2922"><a href="#cb189-2922" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2923"><a href="#cb189-2923" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2926"><a href="#cb189-2926" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2927"><a href="#cb189-2927" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2928"><a href="#cb189-2928" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> samp_ps) {</span>
<span id="cb189-2929"><a href="#cb189-2929" aria-hidden="true" tabindex="-1"></a>     tmp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"files/metadata/tables/"</span>)</span>
<span id="cb189-2930"><a href="#cb189-2930" aria-hidden="true" tabindex="-1"></a>     tmp_otu_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_otu_tax"</span>))</span>
<span id="cb189-2931"><a href="#cb189-2931" aria-hidden="true" tabindex="-1"></a>     tmp_md_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_md"</span>))</span>
<span id="cb189-2932"><a href="#cb189-2932" aria-hidden="true" tabindex="-1"></a>     tmp_tax_table_fp <span class="ot">&lt;-</span> <span class="fu">paste</span>(tmp_path, tmp_otu_name, <span class="st">".txt"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb189-2933"><a href="#cb189-2933" aria-hidden="true" tabindex="-1"></a>     tmp_map_fp <span class="ot">&lt;-</span> <span class="fu">paste</span>(tmp_path, tmp_md_name, <span class="st">".txt"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb189-2934"><a href="#cb189-2934" aria-hidden="true" tabindex="-1"></a>     tmp_input <span class="ot">&lt;-</span> <span class="fu">load_taxa_table</span>(tmp_tax_table_fp, tmp_map_fp)</span>
<span id="cb189-2935"><a href="#cb189-2935" aria-hidden="true" tabindex="-1"></a>     tmp_input_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_mc"</span>))</span>
<span id="cb189-2936"><a href="#cb189-2936" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_input_name)</span>
<span id="cb189-2937"><a href="#cb189-2937" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_input_name, tmp_input)</span>
<span id="cb189-2938"><a href="#cb189-2938" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-2939"><a href="#cb189-2939" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-2940"><a href="#cb189-2940" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"_md"</span>))</span>
<span id="cb189-2941"><a href="#cb189-2941" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"_otu_tax"</span>))</span>
<span id="cb189-2942"><a href="#cb189-2942" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2943"><a href="#cb189-2943" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2944"><a href="#cb189-2944" aria-hidden="true" tabindex="-1"></a><span class="fu">## Choose Data Set</span></span>
<span id="cb189-2945"><a href="#cb189-2945" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2946"><a href="#cb189-2946" aria-hidden="true" tabindex="-1"></a>At this point in the code we need to choose a data set to use, formatted with <span class="in">`mctoolsr`</span>. Remember, there are four choices:</span>
<span id="cb189-2947"><a href="#cb189-2947" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2948"><a href="#cb189-2948" aria-hidden="true" tabindex="-1"></a>1) Complete ASV data set.</span>
<span id="cb189-2949"><a href="#cb189-2949" aria-hidden="true" tabindex="-1"></a>2) Arbitrary filtered ASV data set.</span>
<span id="cb189-2950"><a href="#cb189-2950" aria-hidden="true" tabindex="-1"></a>3) PERfect filtered ASV data set.</span>
<span id="cb189-2951"><a href="#cb189-2951" aria-hidden="true" tabindex="-1"></a>4) PIME filtered ASV data set.</span>
<span id="cb189-2952"><a href="#cb189-2952" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2953"><a href="#cb189-2953" aria-hidden="true" tabindex="-1"></a>This way, if we want to test other data sets we only need to change the name here.</span>
<span id="cb189-2954"><a href="#cb189-2954" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2957"><a href="#cb189-2957" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2958"><a href="#cb189-2958" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">"_mc"</span>)</span>
<span id="cb189-2959"><a href="#cb189-2959" aria-hidden="true" tabindex="-1"></a>its18_select_mc <span class="ot">&lt;-</span> its18_ps_pime_mc</span>
<span id="cb189-2960"><a href="#cb189-2960" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2961"><a href="#cb189-2961" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2962"><a href="#cb189-2962" aria-hidden="true" tabindex="-1"></a><span class="fu">## Normality Tests</span></span>
<span id="cb189-2963"><a href="#cb189-2963" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2964"><a href="#cb189-2964" aria-hidden="true" tabindex="-1"></a>Before proceeding, we need to test each parameter in the metadata to see which ones are and are not normally distributed. For that, we use the Shapiro-Wilk Normality Test. Here we only need one of the metadata files.</span>
<span id="cb189-2965"><a href="#cb189-2965" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2968"><a href="#cb189-2968" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2969"><a href="#cb189-2969" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-2970"><a href="#cb189-2970" aria-hidden="true" tabindex="-1"></a>temp_md <span class="ot">&lt;-</span> its18_select_mc<span class="sc">$</span>map_loaded</span>
<span id="cb189-2971"><a href="#cb189-2971" aria-hidden="true" tabindex="-1"></a>temp_md[,<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-2972"><a href="#cb189-2972" aria-hidden="true" tabindex="-1"></a>shap_results <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-2973"><a href="#cb189-2973" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2974"><a href="#cb189-2974" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">colnames</span>(temp_md)) {</span>
<span id="cb189-2975"><a href="#cb189-2975" aria-hidden="true" tabindex="-1"></a>  tmp_shap <span class="ot">&lt;-</span> <span class="fu">shapiro.test</span>(temp_md[[i]])</span>
<span id="cb189-2976"><a href="#cb189-2976" aria-hidden="true" tabindex="-1"></a>  tmp_p <span class="ot">&lt;-</span> <span class="fu">round</span>(tmp_shap<span class="sc">$</span>p.value, <span class="at">digits =</span> <span class="dv">5</span>)</span>
<span id="cb189-2977"><a href="#cb189-2977" aria-hidden="true" tabindex="-1"></a>  tmp_res <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">isTRUE</span>(tmp_shap<span class="sc">$</span>p.value <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span>
<span id="cb189-2978"><a href="#cb189-2978" aria-hidden="true" tabindex="-1"></a>  shap_results <span class="ot">&lt;-</span> <span class="fu">rbind</span>(shap_results, <span class="fu">data.frame</span>(i, tmp_p, tmp_res))</span>
<span id="cb189-2979"><a href="#cb189-2979" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-2980"><a href="#cb189-2980" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-2981"><a href="#cb189-2981" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(shap_results) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"parameter"</span>, <span class="st">"p-value"</span>, <span class="st">"tranform"</span>)</span>
<span id="cb189-2982"><a href="#cb189-2982" aria-hidden="true" tabindex="-1"></a>shap_results</span>
<span id="cb189-2983"><a href="#cb189-2983" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">filter</span>(shap_results, tranform <span class="sc">==</span> <span class="st">"TRUE"</span>)</span>
<span id="cb189-2984"><a href="#cb189-2984" aria-hidden="true" tabindex="-1"></a>md_to_tranform <span class="ot">&lt;-</span> shap_results<span class="sc">$</span>parameter[shap_results<span class="sc">$</span>tranform <span class="sc">==</span> <span class="cn">TRUE</span>]</span>
<span id="cb189-2985"><a href="#cb189-2985" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"temp_md"</span>))</span>
<span id="cb189-2986"><a href="#cb189-2986" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-2987"><a href="#cb189-2987" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2988"><a href="#cb189-2988" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb189-2989"><a href="#cb189-2989" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>**Show the results** of each normality test for metadata parameters<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb189-2990"><a href="#cb189-2990" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2991"><a href="#cb189-2991" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb189-2992"><a href="#cb189-2992" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2993"><a href="#cb189-2993" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab_its("its_normality_tests")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-2994"><a href="#cb189-2994" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-2997"><a href="#cb189-2997" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-2998"><a href="#cb189-2998" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-2999"><a href="#cb189-2999" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-3000"><a href="#cb189-3000" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> shap_results</span>
<span id="cb189-3001"><a href="#cb189-3001" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb189-3002"><a href="#cb189-3002" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(<span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>),</span>
<span id="cb189-3003"><a href="#cb189-3003" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb189-3004"><a href="#cb189-3004" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">1</span>),</span>
<span id="cb189-3005"><a href="#cb189-3005" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">TRUE</span>, <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-3006"><a href="#cb189-3006" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb189-3007"><a href="#cb189-3007" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb189-3008"><a href="#cb189-3008" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb189-3009"><a href="#cb189-3009" aria-hidden="true" tabindex="-1"></a>    <span class="at">parameter =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Parameter"</span>, </span>
<span id="cb189-3010"><a href="#cb189-3010" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb189-3011"><a href="#cb189-3011" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>),</span>
<span id="cb189-3012"><a href="#cb189-3012" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>, <span class="at">fontSize =</span> <span class="st">"0.9em"</span>), </span>
<span id="cb189-3013"><a href="#cb189-3013" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb189-3014"><a href="#cb189-3014" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">150</span>), </span>
<span id="cb189-3015"><a href="#cb189-3015" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">p-value</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"p-value"</span>, <span class="at">style =</span> <span class="cf">function</span>(value) {</span>
<span id="cb189-3016"><a href="#cb189-3016" aria-hidden="true" tabindex="-1"></a>                                         <span class="cf">if</span> (value <span class="sc">&gt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb189-3017"><a href="#cb189-3017" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#2271B2"</span> <span class="co">#green</span></span>
<span id="cb189-3018"><a href="#cb189-3018" aria-hidden="true" tabindex="-1"></a>                                         } <span class="cf">else</span> <span class="cf">if</span> (value <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb189-3019"><a href="#cb189-3019" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#B22271"</span> <span class="co">#RED "#", "#"</span></span>
<span id="cb189-3020"><a href="#cb189-3020" aria-hidden="true" tabindex="-1"></a>                                         } <span class="cf">else</span> {</span>
<span id="cb189-3021"><a href="#cb189-3021" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#777"</span></span>
<span id="cb189-3022"><a href="#cb189-3022" aria-hidden="true" tabindex="-1"></a>                                         }</span>
<span id="cb189-3023"><a href="#cb189-3023" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">list</span>(<span class="at">color =</span> color, <span class="at">fontWeight =</span> <span class="st">"bold"</span>)})</span>
<span id="cb189-3024"><a href="#cb189-3024" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb189-3025"><a href="#cb189-3025" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">60</span>, </span>
<span id="cb189-3026"><a href="#cb189-3026" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb189-3027"><a href="#cb189-3027" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">FALSE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb189-3028"><a href="#cb189-3028" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb189-3029"><a href="#cb189-3029" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-3030"><a href="#cb189-3030" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb189-3031"><a href="#cb189-3031" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(seq_table)</span>
<span id="cb189-3032"><a href="#cb189-3032" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3033"><a href="#cb189-3033" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb189-3034"><a href="#cb189-3034" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3035"><a href="#cb189-3035" aria-hidden="true" tabindex="-1"></a>Looks like we need to transform <span class="in">`r length(md_to_tranform)`</span> metadata parameters.</span>
<span id="cb189-3036"><a href="#cb189-3036" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3037"><a href="#cb189-3037" aria-hidden="true" tabindex="-1"></a><span class="fu">## Normalize Parameters</span></span>
<span id="cb189-3038"><a href="#cb189-3038" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3039"><a href="#cb189-3039" aria-hidden="true" tabindex="-1"></a>Here we use the R package <span class="co">[</span><span class="ot">`bestNormalize`</span><span class="co">](https://cran.r-project.org/web/packages/bestNormalize/vignettes/bestNormalize.html)</span> to find and execute the best normalizing transformation. The function will test the following normalizing transformations:</span>
<span id="cb189-3040"><a href="#cb189-3040" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3041"><a href="#cb189-3041" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="in">`arcsinh_x`</span> performs an arcsinh transformation.</span>
<span id="cb189-3042"><a href="#cb189-3042" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3043"><a href="#cb189-3043" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="in">`boxcox`</span> Perform a Box-Cox transformation and center/scale a vector to attempt normalization. <span class="in">`boxcox`</span> estimates the optimal value of lambda for the Box-Cox transformation. The function will return an error if a user attempt to transform nonpositive data.</span>
<span id="cb189-3044"><a href="#cb189-3044" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3045"><a href="#cb189-3045" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="in">`yeojohnson`</span> Perform a Yeo-Johnson Transformation and center/scale a vector to attempt normalization. <span class="in">`yeojohnson`</span> estimates the optimal value of lambda for the Yeo-Johnson transformation. The Yeo-Johnson is similar to the Box-Cox method, however it allows for the transformation of nonpositive data as well. </span>
<span id="cb189-3046"><a href="#cb189-3046" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3047"><a href="#cb189-3047" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="in">`orderNorm`</span> The Ordered Quantile (ORQ) normalization transformation, <span class="in">`orderNorm()`</span>, is a rank-based procedure by which the values of a vector are mapped to their percentile, which is then mapped to the same percentile of the normal distribution. Without the presence of ties, this essentially guarantees that the transformation leads to a uniform distribution.</span>
<span id="cb189-3048"><a href="#cb189-3048" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3049"><a href="#cb189-3049" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="in">`log_x`</span> performs a simple log transformation. The parameter a is essentially estimated by the training set by default (estimated as the minimum possible to some extent epsilon), while the base must be specified beforehand. The default base of the log is 10.</span>
<span id="cb189-3050"><a href="#cb189-3050" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3051"><a href="#cb189-3051" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="in">`sqrt_x`</span> performs a simple square-root transformation. The parameter a is essentially estimated by the training set by default (estimated as the minimum possible), while the base must be specified beforehand.</span>
<span id="cb189-3052"><a href="#cb189-3052" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3053"><a href="#cb189-3053" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`exp_x`</span> performs a simple exponential transformation.</span>
<span id="cb189-3054"><a href="#cb189-3054" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3055"><a href="#cb189-3055" aria-hidden="true" tabindex="-1"></a>See this GitHub issue (<span class="co">[</span><span class="ot">#5</span><span class="co">](https://github.com/petersonR/bestNormalize/issues/5)</span>) for a description on getting reproducible results. Apparently, you can get different results because the <span class="in">`bestNormalize()`</span> function uses repeated cross-validation (and doesn't automatically set the seed), so the results will be slightly different each time the function is executed.</span>
<span id="cb189-3056"><a href="#cb189-3056" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3059"><a href="#cb189-3059" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3060"><a href="#cb189-3060" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-3061"><a href="#cb189-3061" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">119</span>)</span>
<span id="cb189-3062"><a href="#cb189-3062" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3063"><a href="#cb189-3063" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> md_to_tranform) {</span>
<span id="cb189-3064"><a href="#cb189-3064" aria-hidden="true" tabindex="-1"></a>  tmp_md <span class="ot">&lt;-</span> its18_select_mc<span class="sc">$</span>map_loaded</span>
<span id="cb189-3065"><a href="#cb189-3065" aria-hidden="true" tabindex="-1"></a>  tmp_best_norm <span class="ot">&lt;-</span> <span class="fu">bestNormalize</span>(tmp_md[[i]], <span class="at">r =</span> <span class="dv">1</span>, <span class="at">k =</span> <span class="dv">5</span>, <span class="at">loo =</span> <span class="cn">TRUE</span>)</span>
<span id="cb189-3066"><a href="#cb189-3066" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_best_norm_test"</span>))</span>
<span id="cb189-3067"><a href="#cb189-3067" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_best_norm)</span>
<span id="cb189-3068"><a href="#cb189-3068" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(tmp_name)</span>
<span id="cb189-3069"><a href="#cb189-3069" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-3070"><a href="#cb189-3070" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-3071"><a href="#cb189-3071" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3072"><a href="#cb189-3072" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3073"><a href="#cb189-3073" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb189-3074"><a href="#cb189-3074" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show the chosen transformations<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb189-3075"><a href="#cb189-3075" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3078"><a href="#cb189-3078" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3079"><a href="#cb189-3079" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3080"><a href="#cb189-3080" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-3081"><a href="#cb189-3081" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> md_to_tranform) {</span>
<span id="cb189-3082"><a href="#cb189-3082" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_best_norm_test"</span>)))</span>
<span id="cb189-3083"><a href="#cb189-3083" aria-hidden="true" tabindex="-1"></a>  tmp_print <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"bestNormalize Chosen transformation of"</span>, i)</span>
<span id="cb189-3084"><a href="#cb189-3084" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-3085"><a href="#cb189-3085" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"##"</span>, tmp_print, <span class="st">"##"</span>, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-3086"><a href="#cb189-3086" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(tmp_get<span class="sc">$</span>chosen_transform)</span>
<span id="cb189-3087"><a href="#cb189-3087" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-3088"><a href="#cb189-3088" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-3089"><a href="#cb189-3089" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-3090"><a href="#cb189-3090" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-3091"><a href="#cb189-3091" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3092"><a href="#cb189-3092" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb189-3093"><a href="#cb189-3093" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3094"><a href="#cb189-3094" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb189-3095"><a href="#cb189-3095" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show  the complete <span class="in">`bestNormalize`</span> results<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb189-3096"><a href="#cb189-3096" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3099"><a href="#cb189-3099" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3100"><a href="#cb189-3100" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3101"><a href="#cb189-3101" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-3102"><a href="#cb189-3102" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> md_to_tranform) {</span>
<span id="cb189-3103"><a href="#cb189-3103" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_best_norm_test"</span>)))</span>
<span id="cb189-3104"><a href="#cb189-3104" aria-hidden="true" tabindex="-1"></a>  tmp_print <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Results of bestNormalize for"</span>, i)</span>
<span id="cb189-3105"><a href="#cb189-3105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-3106"><a href="#cb189-3106" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"##"</span>, tmp_print, <span class="st">"##"</span>, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-3107"><a href="#cb189-3107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(tmp_get)</span>
<span id="cb189-3108"><a href="#cb189-3108" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-3109"><a href="#cb189-3109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-3110"><a href="#cb189-3110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-3111"><a href="#cb189-3111" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-3112"><a href="#cb189-3112" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3113"><a href="#cb189-3113" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb189-3114"><a href="#cb189-3114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3115"><a href="#cb189-3115" aria-hidden="true" tabindex="-1"></a>Great, now we can add the normalized transformed data back to our <span class="in">`mctoolsr`</span> metadata file. </span>
<span id="cb189-3116"><a href="#cb189-3116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3119"><a href="#cb189-3119" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3120"><a href="#cb189-3120" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-3121"><a href="#cb189-3121" aria-hidden="true" tabindex="-1"></a>its18_select_mc_norm <span class="ot">&lt;-</span> its18_select_mc</span>
<span id="cb189-3122"><a href="#cb189-3122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3123"><a href="#cb189-3123" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> md_to_tranform) {</span>
<span id="cb189-3124"><a href="#cb189-3124" aria-hidden="true" tabindex="-1"></a>      tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">"_best_norm_test"</span>)))</span>
<span id="cb189-3125"><a href="#cb189-3125" aria-hidden="true" tabindex="-1"></a>      tmp_new_data <span class="ot">&lt;-</span> tmp_get<span class="sc">$</span>x.t</span>
<span id="cb189-3126"><a href="#cb189-3126" aria-hidden="true" tabindex="-1"></a>      its18_select_mc_norm<span class="sc">$</span>map_loaded[[i]] <span class="ot">&lt;-</span> tmp_new_data</span>
<span id="cb189-3127"><a href="#cb189-3127" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-3128"><a href="#cb189-3128" aria-hidden="true" tabindex="-1"></a>}     </span>
<span id="cb189-3129"><a href="#cb189-3129" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3130"><a href="#cb189-3130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3131"><a href="#cb189-3131" aria-hidden="true" tabindex="-1"></a>And rerun the Shapiro Tests. </span>
<span id="cb189-3132"><a href="#cb189-3132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3135"><a href="#cb189-3135" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3136"><a href="#cb189-3136" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-3137"><a href="#cb189-3137" aria-hidden="true" tabindex="-1"></a>temp_md_norm <span class="ot">&lt;-</span> its18_select_mc_norm<span class="sc">$</span>map_loaded</span>
<span id="cb189-3138"><a href="#cb189-3138" aria-hidden="true" tabindex="-1"></a>temp_md_norm[,<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-3139"><a href="#cb189-3139" aria-hidden="true" tabindex="-1"></a>shap_results_norm <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-3140"><a href="#cb189-3140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3141"><a href="#cb189-3141" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">colnames</span>(temp_md_norm)) {</span>
<span id="cb189-3142"><a href="#cb189-3142" aria-hidden="true" tabindex="-1"></a>  tmp_shap <span class="ot">&lt;-</span> <span class="fu">shapiro.test</span>(temp_md_norm[[i]])</span>
<span id="cb189-3143"><a href="#cb189-3143" aria-hidden="true" tabindex="-1"></a>  tmp_p <span class="ot">&lt;-</span> <span class="fu">round</span>(tmp_shap<span class="sc">$</span>p.value, <span class="at">digits =</span> <span class="dv">5</span>)</span>
<span id="cb189-3144"><a href="#cb189-3144" aria-hidden="true" tabindex="-1"></a>  tmp_res <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">isTRUE</span>(tmp_shap<span class="sc">$</span>p.value <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span>
<span id="cb189-3145"><a href="#cb189-3145" aria-hidden="true" tabindex="-1"></a>  shap_results_norm <span class="ot">&lt;-</span> <span class="fu">rbind</span>(shap_results_norm, <span class="fu">data.frame</span>(i, tmp_p, tmp_res))</span>
<span id="cb189-3146"><a href="#cb189-3146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-3147"><a href="#cb189-3147" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-3148"><a href="#cb189-3148" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(shap_results_norm) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"parameter"</span>, <span class="st">"p-value"</span>, <span class="st">"tranform"</span>)</span>
<span id="cb189-3149"><a href="#cb189-3149" aria-hidden="true" tabindex="-1"></a>shap_results_norm</span>
<span id="cb189-3150"><a href="#cb189-3150" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"temp_md_norm"</span>))</span>
<span id="cb189-3151"><a href="#cb189-3151" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3152"><a href="#cb189-3152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3153"><a href="#cb189-3153" aria-hidden="true" tabindex="-1"></a>And check if there are any parameters that are still significant for the normality test.</span>
<span id="cb189-3154"><a href="#cb189-3154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3157"><a href="#cb189-3157" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3158"><a href="#cb189-3158" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-3159"><a href="#cb189-3159" aria-hidden="true" tabindex="-1"></a>shap_results<span class="sc">$</span>parameter[shap_results_norm<span class="sc">$</span>tranform <span class="sc">==</span> <span class="cn">TRUE</span>]</span>
<span id="cb189-3160"><a href="#cb189-3160" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3161"><a href="#cb189-3161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3162"><a href="#cb189-3162" aria-hidden="true" tabindex="-1"></a>Ok. Looks like <span class="in">`bestNormalize`</span> was unable to find a suitable transformation for <span class="in">`Al`</span> and <span class="in">`Fe`</span>. This is likely because there is very little variation in these metadata and/or there are too few significant digits.</span>
<span id="cb189-3163"><a href="#cb189-3163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3164"><a href="#cb189-3164" aria-hidden="true" tabindex="-1"></a><span class="fu">## Normalized Metadata</span></span>
<span id="cb189-3165"><a href="#cb189-3165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3166"><a href="#cb189-3166" aria-hidden="true" tabindex="-1"></a>Finally, here is a new summary table that includes all of the normalized data.</span>
<span id="cb189-3167"><a href="#cb189-3167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3168"><a href="#cb189-3168" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb189-3169"><a href="#cb189-3169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3172"><a href="#cb189-3172" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3173"><a href="#cb189-3173" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3174"><a href="#cb189-3174" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-3175"><a href="#cb189-3175" aria-hidden="true" tabindex="-1"></a>metad_norm <span class="ot">&lt;-</span> its18_select_mc_norm<span class="sc">$</span>map_loaded</span>
<span id="cb189-3176"><a href="#cb189-3176" aria-hidden="true" tabindex="-1"></a>metad_norm <span class="ot">&lt;-</span> metad_norm <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"Sample_ID"</span>)</span>
<span id="cb189-3177"><a href="#cb189-3177" aria-hidden="true" tabindex="-1"></a>metad_norm <span class="ot">&lt;-</span> metad_norm[<span class="fu">order</span>(<span class="fu">match</span>(metad_norm<span class="sc">$</span>Sample_ID, metad<span class="sc">$</span>Sample_ID)), ]</span>
<span id="cb189-3178"><a href="#cb189-3178" aria-hidden="true" tabindex="-1"></a>metad_norm[,<span class="dv">2</span><span class="sc">:</span><span class="dv">9</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-3179"><a href="#cb189-3179" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> metad_norm</span>
<span id="cb189-3180"><a href="#cb189-3180" aria-hidden="true" tabindex="-1"></a>seq_table <span class="sc">%&gt;%</span></span>
<span id="cb189-3181"><a href="#cb189-3181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_this</span>(</span>
<span id="cb189-3182"><a href="#cb189-3182" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"normalized_metadata"</span>,</span>
<span id="cb189-3183"><a href="#cb189-3183" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".csv"</span>,</span>
<span id="cb189-3184"><a href="#cb189-3184" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download data as csv file"</span>,</span>
<span id="cb189-3185"><a href="#cb189-3185" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"default"</span>,</span>
<span id="cb189-3186"><a href="#cb189-3186" aria-hidden="true" tabindex="-1"></a>    <span class="at">csv2 =</span> <span class="cn">FALSE</span>,</span>
<span id="cb189-3187"><a href="#cb189-3187" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-3188"><a href="#cb189-3188" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span>)</span>
<span id="cb189-3189"><a href="#cb189-3189" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3190"><a href="#cb189-3190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3191"><a href="#cb189-3191" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab_its("its_normalized_metadata")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-3192"><a href="#cb189-3192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3195"><a href="#cb189-3195" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3196"><a href="#cb189-3196" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3197"><a href="#cb189-3197" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-3198"><a href="#cb189-3198" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> seq_table <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="fu">across</span>(<span class="at">.cols =</span> <span class="fu">c</span>(<span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">7</span><span class="sc">:</span><span class="dv">9</span>, <span class="dv">14</span><span class="sc">:</span><span class="dv">15</span>,<span class="dv">19</span><span class="sc">:</span><span class="dv">20</span>, </span>
<span id="cb189-3199"><a href="#cb189-3199" aria-hidden="true" tabindex="-1"></a>                                                            <span class="dv">24</span>,<span class="dv">26</span>,<span class="dv">30</span>,<span class="dv">34</span><span class="sc">:</span><span class="dv">38</span>,<span class="dv">40</span>, <span class="dv">42</span><span class="sc">:</span><span class="dv">43</span>, </span>
<span id="cb189-3200"><a href="#cb189-3200" aria-hidden="true" tabindex="-1"></a>                                                            <span class="dv">49</span>, <span class="dv">52</span><span class="sc">:</span><span class="dv">53</span>,<span class="dv">55</span>,<span class="dv">57</span>,<span class="dv">60</span><span class="sc">:</span><span class="dv">62</span>), </span>
<span id="cb189-3201"><a href="#cb189-3201" aria-hidden="true" tabindex="-1"></a>                                                  round, <span class="at">digits =</span> <span class="dv">3</span>)) </span>
<span id="cb189-3202"><a href="#cb189-3202" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb189-3203"><a href="#cb189-3203" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(<span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>),</span>
<span id="cb189-3204"><a href="#cb189-3204" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb189-3205"><a href="#cb189-3205" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">1</span>),</span>
<span id="cb189-3206"><a href="#cb189-3206" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">TRUE</span>, <span class="at">sortable =</span> <span class="cn">TRUE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-3207"><a href="#cb189-3207" aria-hidden="true" tabindex="-1"></a>    <span class="at">footerStyle =</span> <span class="fu">list</span>(<span class="at">fontWeight =</span> <span class="st">"bold"</span>)</span>
<span id="cb189-3208"><a href="#cb189-3208" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb189-3209"><a href="#cb189-3209" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb189-3210"><a href="#cb189-3210" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Sample ID"</span>, </span>
<span id="cb189-3211"><a href="#cb189-3211" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sticky =</span> <span class="st">"left"</span>, </span>
<span id="cb189-3212"><a href="#cb189-3212" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>),</span>
<span id="cb189-3213"><a href="#cb189-3213" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"1px solid #eee"</span>, <span class="at">fontSize =</span> <span class="st">"0.9em"</span>), </span>
<span id="cb189-3214"><a href="#cb189-3214" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb189-3215"><a href="#cb189-3215" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">150</span>), </span>
<span id="cb189-3216"><a href="#cb189-3216" aria-hidden="true" tabindex="-1"></a>    <span class="at">raw_reads =</span> <span class="fu">colDef</span>(<span class="at">footer =</span> <span class="cf">function</span>(values) <span class="fu">sprintf</span>(<span class="st">"%.0f"</span>, <span class="fu">sum</span>(values)))</span>
<span id="cb189-3217"><a href="#cb189-3217" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb189-3218"><a href="#cb189-3218" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">TRUE</span>, <span class="at">defaultPageSize =</span> <span class="dv">5</span>, </span>
<span id="cb189-3219"><a href="#cb189-3219" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb189-3220"><a href="#cb189-3220" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb189-3221"><a href="#cb189-3221" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">FALSE</span>, </span>
<span id="cb189-3222"><a href="#cb189-3222" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">TRUE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-3223"><a href="#cb189-3223" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb189-3224"><a href="#cb189-3224" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(seq_table)</span>
<span id="cb189-3225"><a href="#cb189-3225" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3226"><a href="#cb189-3226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3227"><a href="#cb189-3227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3230"><a href="#cb189-3230" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3231"><a href="#cb189-3231" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3232"><a href="#cb189-3232" aria-hidden="true" tabindex="-1"></a><span class="co"># Code to run individual transformations for each metadata and generate histograms</span></span>
<span id="cb189-3233"><a href="#cb189-3233" aria-hidden="true" tabindex="-1"></a><span class="do">##for (i in md_to_tranform) {</span></span>
<span id="cb189-3234"><a href="#cb189-3234" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_data &lt;- its18_select_mc$map_loaded</span></span>
<span id="cb189-3235"><a href="#cb189-3235" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_no_transform_obj &lt;- no_transform(tmp_data[[i]])</span></span>
<span id="cb189-3236"><a href="#cb189-3236" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_arcsinh_obj &lt;- arcsinh_x(tmp_data[[i]])</span></span>
<span id="cb189-3237"><a href="#cb189-3237" aria-hidden="true" tabindex="-1"></a><span class="do">##     #tmp_boxcox_obj &lt;- boxcox(tmp_data[[i]])</span></span>
<span id="cb189-3238"><a href="#cb189-3238" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_yeojohnson_obj &lt;- yeojohnson(tmp_data[[i]])</span></span>
<span id="cb189-3239"><a href="#cb189-3239" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_orderNorm_obj &lt;- orderNorm(tmp_data[[i]])</span></span>
<span id="cb189-3240"><a href="#cb189-3240" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_log_obj &lt;- log_x(tmp_data[[i]])</span></span>
<span id="cb189-3241"><a href="#cb189-3241" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_sqrt_x_obj &lt;- sqrt_x(tmp_data[[i]])</span></span>
<span id="cb189-3242"><a href="#cb189-3242" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_exp_x_obj &lt;- exp_x(tmp_data[[i]])</span></span>
<span id="cb189-3243"><a href="#cb189-3243" aria-hidden="true" tabindex="-1"></a><span class="do">##     </span></span>
<span id="cb189-3244"><a href="#cb189-3244" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_no_transform &lt;- hist(tmp_no_transform_obj$x.t, main = i, breaks = 6, xlab = "no_transform")</span></span>
<span id="cb189-3245"><a href="#cb189-3245" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_name &lt;- purrr::map_chr(i, ~ paste0(., "_hist_plot_no_transform"))</span></span>
<span id="cb189-3246"><a href="#cb189-3246" aria-hidden="true" tabindex="-1"></a><span class="do">##     assign(tmp_plot_name, tmp_plot_no_transform)</span></span>
<span id="cb189-3247"><a href="#cb189-3247" aria-hidden="true" tabindex="-1"></a><span class="do">##     print(tmp_plot_name)</span></span>
<span id="cb189-3248"><a href="#cb189-3248" aria-hidden="true" tabindex="-1"></a><span class="do">##     </span></span>
<span id="cb189-3249"><a href="#cb189-3249" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_arcsinh &lt;- hist(tmp_arcsinh_obj$x.t, main = i, breaks = 6, xlab = "arcsinh")</span></span>
<span id="cb189-3250"><a href="#cb189-3250" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_name &lt;- purrr::map_chr(i, ~ paste0(., "_hist_plot_arcsinh"))</span></span>
<span id="cb189-3251"><a href="#cb189-3251" aria-hidden="true" tabindex="-1"></a><span class="do">##     assign(tmp_plot_name, tmp_plot_arcsinh)</span></span>
<span id="cb189-3252"><a href="#cb189-3252" aria-hidden="true" tabindex="-1"></a><span class="do">##     print(tmp_plot_name)</span></span>
<span id="cb189-3253"><a href="#cb189-3253" aria-hidden="true" tabindex="-1"></a><span class="do">##     # cannot run because of Al and Fe</span></span>
<span id="cb189-3254"><a href="#cb189-3254" aria-hidden="true" tabindex="-1"></a><span class="do">##     #tmp_plot_boxcox &lt;- hist(tmp_boxcox_obj$x.t, main = i, breaks = 6, xlab = "boxcox")</span></span>
<span id="cb189-3255"><a href="#cb189-3255" aria-hidden="true" tabindex="-1"></a><span class="do">##     #tmp_plot_name &lt;- purrr::map_chr(i, ~ paste0(., "_hist_plot_boxcox"))</span></span>
<span id="cb189-3256"><a href="#cb189-3256" aria-hidden="true" tabindex="-1"></a><span class="do">##     #assign(tmp_plot_name, tmp_plot_boxcox)</span></span>
<span id="cb189-3257"><a href="#cb189-3257" aria-hidden="true" tabindex="-1"></a><span class="do">##     #print(tmp_plot_name)</span></span>
<span id="cb189-3258"><a href="#cb189-3258" aria-hidden="true" tabindex="-1"></a><span class="do">##     </span></span>
<span id="cb189-3259"><a href="#cb189-3259" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_yeojohnson &lt;- hist(tmp_yeojohnson_obj$x.t, main = i, breaks = 6, xlab = "yeojohnson")</span></span>
<span id="cb189-3260"><a href="#cb189-3260" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_name &lt;- purrr::map_chr(i, ~ paste0(., "_hist_plot_yeojohnson"))</span></span>
<span id="cb189-3261"><a href="#cb189-3261" aria-hidden="true" tabindex="-1"></a><span class="do">##     assign(tmp_plot_name, tmp_plot_yeojohnson)</span></span>
<span id="cb189-3262"><a href="#cb189-3262" aria-hidden="true" tabindex="-1"></a><span class="do">##     print(tmp_plot_name)</span></span>
<span id="cb189-3263"><a href="#cb189-3263" aria-hidden="true" tabindex="-1"></a><span class="do">##     </span></span>
<span id="cb189-3264"><a href="#cb189-3264" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_orderNorm &lt;- hist(tmp_orderNorm_obj$x.t, main = i, breaks = 6, xlab = "orderNorm")</span></span>
<span id="cb189-3265"><a href="#cb189-3265" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_name &lt;- purrr::map_chr(i, ~ paste0(., "_hist_plot_orderNorm"))</span></span>
<span id="cb189-3266"><a href="#cb189-3266" aria-hidden="true" tabindex="-1"></a><span class="do">##     assign(tmp_plot_name, tmp_plot_orderNorm)</span></span>
<span id="cb189-3267"><a href="#cb189-3267" aria-hidden="true" tabindex="-1"></a><span class="do">##     print(tmp_plot_name)</span></span>
<span id="cb189-3268"><a href="#cb189-3268" aria-hidden="true" tabindex="-1"></a><span class="do">##     </span></span>
<span id="cb189-3269"><a href="#cb189-3269" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_log &lt;- hist(tmp_log_obj$x.t, main = i, breaks = 6, xlab = "log")</span></span>
<span id="cb189-3270"><a href="#cb189-3270" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_name &lt;- purrr::map_chr(i, ~ paste0(., "_hist_plot_log"))</span></span>
<span id="cb189-3271"><a href="#cb189-3271" aria-hidden="true" tabindex="-1"></a><span class="do">##     assign(tmp_plot_name, tmp_plot_log)</span></span>
<span id="cb189-3272"><a href="#cb189-3272" aria-hidden="true" tabindex="-1"></a><span class="do">##     print(tmp_plot_name)</span></span>
<span id="cb189-3273"><a href="#cb189-3273" aria-hidden="true" tabindex="-1"></a><span class="do">##     </span></span>
<span id="cb189-3274"><a href="#cb189-3274" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_sqrt &lt;- hist(tmp_sqrt_x_obj$x.t, main = i, breaks = 6, xlab = "sqrt")</span></span>
<span id="cb189-3275"><a href="#cb189-3275" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_name &lt;- purrr::map_chr(i, ~ paste0(., "_hist_plot_sqrt"))</span></span>
<span id="cb189-3276"><a href="#cb189-3276" aria-hidden="true" tabindex="-1"></a><span class="do">##     assign(tmp_plot_name, tmp_plot_sqrt)</span></span>
<span id="cb189-3277"><a href="#cb189-3277" aria-hidden="true" tabindex="-1"></a><span class="do">##     print(tmp_plot_name)</span></span>
<span id="cb189-3278"><a href="#cb189-3278" aria-hidden="true" tabindex="-1"></a><span class="do">##     </span></span>
<span id="cb189-3279"><a href="#cb189-3279" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_exp &lt;- hist(tmp_exp_x_obj$x.t, main = i, breaks = 6, xlab = "exp")</span></span>
<span id="cb189-3280"><a href="#cb189-3280" aria-hidden="true" tabindex="-1"></a><span class="do">##     tmp_plot_name &lt;- purrr::map_chr(i, ~ paste0(., "_hist_plot_exp"))</span></span>
<span id="cb189-3281"><a href="#cb189-3281" aria-hidden="true" tabindex="-1"></a><span class="do">##     assign(tmp_plot_name, tmp_plot_exp)</span></span>
<span id="cb189-3282"><a href="#cb189-3282" aria-hidden="true" tabindex="-1"></a><span class="do">##     print(tmp_plot_name)</span></span>
<span id="cb189-3283"><a href="#cb189-3283" aria-hidden="true" tabindex="-1"></a><span class="do">##</span></span>
<span id="cb189-3284"><a href="#cb189-3284" aria-hidden="true" tabindex="-1"></a><span class="do">##     rm(list = ls(pattern = "tmp_"))</span></span>
<span id="cb189-3285"><a href="#cb189-3285" aria-hidden="true" tabindex="-1"></a><span class="do">##}</span></span>
<span id="cb189-3286"><a href="#cb189-3286" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3287"><a href="#cb189-3287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3288"><a href="#cb189-3288" aria-hidden="true" tabindex="-1"></a><span class="fu">## Autocorrelation Tests</span></span>
<span id="cb189-3289"><a href="#cb189-3289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3290"><a href="#cb189-3290" aria-hidden="true" tabindex="-1"></a>Next, we  test the metadata for autocorrelations. Do we do this on the original data or the transformed data? No idea, so let's do both.</span>
<span id="cb189-3291"><a href="#cb189-3291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3292"><a href="#cb189-3292" aria-hidden="true" tabindex="-1"></a><span class="fu">### Split Metadata</span></span>
<span id="cb189-3293"><a href="#cb189-3293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3294"><a href="#cb189-3294" aria-hidden="true" tabindex="-1"></a>We need to split the data into different groups.</span>
<span id="cb189-3295"><a href="#cb189-3295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3296"><a href="#cb189-3296" aria-hidden="true" tabindex="-1"></a>A) Environmental and edaphic properties</span>
<span id="cb189-3297"><a href="#cb189-3297" aria-hidden="true" tabindex="-1"></a>B) Microbial functional responses</span>
<span id="cb189-3298"><a href="#cb189-3298" aria-hidden="true" tabindex="-1"></a>C) Temperature adaptation properties</span>
<span id="cb189-3299"><a href="#cb189-3299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3300"><a href="#cb189-3300" aria-hidden="true" tabindex="-1"></a>We first create lists of metadata parameters. </span>
<span id="cb189-3301"><a href="#cb189-3301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3304"><a href="#cb189-3304" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3305"><a href="#cb189-3305" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-3306"><a href="#cb189-3306" aria-hidden="true" tabindex="-1"></a>div <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"PLOT"</span>, <span class="st">"TREAT"</span>, <span class="st">"TREAT_T"</span>, <span class="st">"PAIR"</span>, <span class="st">"Observed"</span>, <span class="st">"Shannon_exp"</span>, </span>
<span id="cb189-3307"><a href="#cb189-3307" aria-hidden="true" tabindex="-1"></a>         <span class="st">"InvSimpson"</span>, <span class="st">"ATAP"</span>)</span>
<span id="cb189-3308"><a href="#cb189-3308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3309"><a href="#cb189-3309" aria-hidden="true" tabindex="-1"></a>edaphic <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"AST"</span>, <span class="st">"H2O"</span>, <span class="st">"N"</span>, <span class="st">"P"</span>, <span class="st">"Al"</span>, </span>
<span id="cb189-3310"><a href="#cb189-3310" aria-hidden="true" tabindex="-1"></a>             <span class="st">"Ca"</span>, <span class="st">"Fe"</span>, <span class="st">"K"</span>, <span class="st">"Mg"</span>, <span class="st">"Mn"</span>, <span class="st">"Na"</span>, <span class="st">"TEB"</span>, <span class="st">"ECEC"</span>, <span class="st">"pH"</span>, </span>
<span id="cb189-3311"><a href="#cb189-3311" aria-hidden="true" tabindex="-1"></a>             <span class="st">"NH4"</span>, <span class="st">"NO3"</span>, <span class="st">"PO4"</span>, <span class="st">"DOC"</span>, <span class="st">"DON"</span>, <span class="st">"DOCN"</span>)</span>
<span id="cb189-3312"><a href="#cb189-3312" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3313"><a href="#cb189-3313" aria-hidden="true" tabindex="-1"></a>soil_funct <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"micC"</span>, <span class="st">"micN"</span>, <span class="st">"micP"</span>, <span class="st">"micCN"</span>, <span class="st">"micCP"</span>, <span class="st">"micNP"</span>, </span>
<span id="cb189-3314"><a href="#cb189-3314" aria-hidden="true" tabindex="-1"></a>                <span class="st">"AG_ase"</span>, <span class="st">"BG_ase"</span>, <span class="st">"BP_ase"</span>, <span class="st">"CE_ase"</span>, <span class="st">"P_ase"</span>, <span class="st">"N_ase"</span>, </span>
<span id="cb189-3315"><a href="#cb189-3315" aria-hidden="true" tabindex="-1"></a>                <span class="st">"S_ase"</span>, <span class="st">"XY_ase"</span>,  <span class="st">"LP_ase"</span>, <span class="st">"PX_ase"</span>, <span class="st">"CO2"</span>, </span>
<span id="cb189-3316"><a href="#cb189-3316" aria-hidden="true" tabindex="-1"></a>                <span class="st">"enzCN"</span>, <span class="st">"enzCP"</span>, <span class="st">"enzNP"</span>)</span>
<span id="cb189-3317"><a href="#cb189-3317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3318"><a href="#cb189-3318" aria-hidden="true" tabindex="-1"></a>temp_adapt <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"AG_Q10"</span>, <span class="st">"BG_Q10"</span>, <span class="st">"BP_Q10"</span>, <span class="st">"CE_Q10"</span>, <span class="st">"P_Q10"</span>, <span class="st">"N_Q10"</span>, </span>
<span id="cb189-3319"><a href="#cb189-3319" aria-hidden="true" tabindex="-1"></a>                <span class="st">"S_Q10"</span>, <span class="st">"XY_Q10"</span>, <span class="st">"LP_Q10"</span>, <span class="st">"PX_Q10"</span>, <span class="st">"CUEcn"</span>, <span class="st">"CUEcp"</span>, </span>
<span id="cb189-3320"><a href="#cb189-3320" aria-hidden="true" tabindex="-1"></a>                <span class="st">"NUE"</span>, <span class="st">"PUE"</span>, <span class="st">"Tmin"</span>, <span class="st">"SI"</span>)</span>
<span id="cb189-3321"><a href="#cb189-3321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3322"><a href="#cb189-3322" aria-hidden="true" tabindex="-1"></a>md_groups <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"edaphic"</span>, <span class="st">"soil_funct"</span>, <span class="st">"temp_adapt"</span>)</span>
<span id="cb189-3323"><a href="#cb189-3323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3324"><a href="#cb189-3324" aria-hidden="true" tabindex="-1"></a><span class="co"># NoT uSed: minPO4, minNH4, minNO3, minTIN</span></span>
<span id="cb189-3325"><a href="#cb189-3325" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3326"><a href="#cb189-3326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3327"><a href="#cb189-3327" aria-hidden="true" tabindex="-1"></a>And then use the lists to split the data sets by metadata  group. Here, we do this for the original metadata and the metadata after normalization.</span>
<span id="cb189-3328"><a href="#cb189-3328" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3331"><a href="#cb189-3331" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3332"><a href="#cb189-3332" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-3333"><a href="#cb189-3333" aria-hidden="true" tabindex="-1"></a>select_md <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"its18_select_mc"</span>, <span class="st">"its18_select_mc_norm"</span>)</span>
<span id="cb189-3334"><a href="#cb189-3334" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> select_md) {</span>
<span id="cb189-3335"><a href="#cb189-3335" aria-hidden="true" tabindex="-1"></a>   <span class="co">#tmp_get &lt;- get(purrr::map_chr(i, ~ paste0(i, "_mc")))</span></span>
<span id="cb189-3336"><a href="#cb189-3336" aria-hidden="true" tabindex="-1"></a>   tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb189-3337"><a href="#cb189-3337" aria-hidden="true" tabindex="-1"></a>   tmp_md_all <span class="ot">&lt;-</span> tmp_get<span class="sc">$</span>map_loaded</span>
<span id="cb189-3338"><a href="#cb189-3338" aria-hidden="true" tabindex="-1"></a>   tmp_div <span class="ot">&lt;-</span> tmp_md_all <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(div))</span>
<span id="cb189-3339"><a href="#cb189-3339" aria-hidden="true" tabindex="-1"></a>   tmp_div <span class="ot">&lt;-</span> tmp_div <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb189-3340"><a href="#cb189-3340" aria-hidden="true" tabindex="-1"></a><span class="do">## edaphic      </span></span>
<span id="cb189-3341"><a href="#cb189-3341" aria-hidden="true" tabindex="-1"></a>   tmp_sub_edaphic <span class="ot">&lt;-</span> tmp_md_all <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(edaphic))</span>
<span id="cb189-3342"><a href="#cb189-3342" aria-hidden="true" tabindex="-1"></a>   tmp_sub_edaphic <span class="ot">&lt;-</span> tmp_sub_edaphic <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb189-3343"><a href="#cb189-3343" aria-hidden="true" tabindex="-1"></a>   tmp_edaphic <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_div, tmp_sub_edaphic, <span class="at">by =</span> <span class="st">"SampleID"</span>)</span>
<span id="cb189-3344"><a href="#cb189-3344" aria-hidden="true" tabindex="-1"></a>   tmp_edaphic <span class="ot">&lt;-</span> tmp_edaphic <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb189-3345"><a href="#cb189-3345" aria-hidden="true" tabindex="-1"></a><span class="do">## soil_funct</span></span>
<span id="cb189-3346"><a href="#cb189-3346" aria-hidden="true" tabindex="-1"></a>   tmp_sub_soil_funct <span class="ot">&lt;-</span> tmp_md_all <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(soil_funct))</span>
<span id="cb189-3347"><a href="#cb189-3347" aria-hidden="true" tabindex="-1"></a>   tmp_sub_soil_funct <span class="ot">&lt;-</span> tmp_sub_soil_funct <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb189-3348"><a href="#cb189-3348" aria-hidden="true" tabindex="-1"></a>   tmp_soil_funct <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_div, tmp_sub_soil_funct, <span class="at">by =</span> <span class="st">"SampleID"</span>)</span>
<span id="cb189-3349"><a href="#cb189-3349" aria-hidden="true" tabindex="-1"></a>   tmp_soil_funct <span class="ot">&lt;-</span> tmp_soil_funct <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"SampleID"</span>)      </span>
<span id="cb189-3350"><a href="#cb189-3350" aria-hidden="true" tabindex="-1"></a><span class="do">## temp_adapt</span></span>
<span id="cb189-3351"><a href="#cb189-3351" aria-hidden="true" tabindex="-1"></a>   tmp_sub_temp_adapt <span class="ot">&lt;-</span> tmp_md_all <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(temp_adapt))</span>
<span id="cb189-3352"><a href="#cb189-3352" aria-hidden="true" tabindex="-1"></a>   tmp_sub_temp_adapt <span class="ot">&lt;-</span> tmp_sub_temp_adapt <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb189-3353"><a href="#cb189-3353" aria-hidden="true" tabindex="-1"></a>   tmp_temp_adapt <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_div, tmp_sub_temp_adapt, <span class="at">by =</span> <span class="st">"SampleID"</span>)</span>
<span id="cb189-3354"><a href="#cb189-3354" aria-hidden="true" tabindex="-1"></a>   tmp_temp_adapt <span class="ot">&lt;-</span> tmp_temp_adapt <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"SampleID"</span>)      </span>
<span id="cb189-3355"><a href="#cb189-3355" aria-hidden="true" tabindex="-1"></a><span class="do">## combine</span></span>
<span id="cb189-3356"><a href="#cb189-3356" aria-hidden="true" tabindex="-1"></a>   tmp_list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">data_loaded =</span> its18_select_mc<span class="sc">$</span>data_loaded, </span>
<span id="cb189-3357"><a href="#cb189-3357" aria-hidden="true" tabindex="-1"></a>                    <span class="at">map_loaded =</span> its18_select_mc<span class="sc">$</span>map_loaded, </span>
<span id="cb189-3358"><a href="#cb189-3358" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxonomy_loaded =</span> its18_select_mc<span class="sc">$</span>taxonomy_loaded,</span>
<span id="cb189-3359"><a href="#cb189-3359" aria-hidden="true" tabindex="-1"></a>                    <span class="at">edaphic =</span> tmp_edaphic, </span>
<span id="cb189-3360"><a href="#cb189-3360" aria-hidden="true" tabindex="-1"></a>                    <span class="at">soil_funct =</span> tmp_soil_funct, </span>
<span id="cb189-3361"><a href="#cb189-3361" aria-hidden="true" tabindex="-1"></a>                    <span class="at">temp_adapt =</span> tmp_temp_adapt)</span>
<span id="cb189-3362"><a href="#cb189-3362" aria-hidden="true" tabindex="-1"></a>   tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_split"</span>))</span>
<span id="cb189-3363"><a href="#cb189-3363" aria-hidden="true" tabindex="-1"></a>   <span class="fu">print</span>(tmp_name)</span>
<span id="cb189-3364"><a href="#cb189-3364" aria-hidden="true" tabindex="-1"></a>   <span class="fu">assign</span>(tmp_name, tmp_list)</span>
<span id="cb189-3365"><a href="#cb189-3365" aria-hidden="true" tabindex="-1"></a>   <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-3366"><a href="#cb189-3366" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-3367"><a href="#cb189-3367" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3368"><a href="#cb189-3368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3369"><a href="#cb189-3369" aria-hidden="true" tabindex="-1"></a><span class="fu">### Generate Autocorrelation Plots</span></span>
<span id="cb189-3370"><a href="#cb189-3370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3371"><a href="#cb189-3371" aria-hidden="true" tabindex="-1"></a>A little housekeeping to get rid of parameters we don't need (e.g., plot number, pair, etc.).</span>
<span id="cb189-3372"><a href="#cb189-3372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3375"><a href="#cb189-3375" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3376"><a href="#cb189-3376" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-3377"><a href="#cb189-3377" aria-hidden="true" tabindex="-1"></a>edaphic_cor <span class="ot">&lt;-</span> its18_select_mc_split<span class="sc">$</span>edaphic</span>
<span id="cb189-3378"><a href="#cb189-3378" aria-hidden="true" tabindex="-1"></a>edaphic_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-3379"><a href="#cb189-3379" aria-hidden="true" tabindex="-1"></a>edaphic_norm_cor <span class="ot">&lt;-</span> its18_select_mc_norm_split<span class="sc">$</span>edaphic</span>
<span id="cb189-3380"><a href="#cb189-3380" aria-hidden="true" tabindex="-1"></a>edaphic_norm_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-3381"><a href="#cb189-3381" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3382"><a href="#cb189-3382" aria-hidden="true" tabindex="-1"></a>soil_funct_cor <span class="ot">&lt;-</span> its18_select_mc_split<span class="sc">$</span>soil_funct</span>
<span id="cb189-3383"><a href="#cb189-3383" aria-hidden="true" tabindex="-1"></a>soil_funct_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-3384"><a href="#cb189-3384" aria-hidden="true" tabindex="-1"></a>soil_funct_norm_cor <span class="ot">&lt;-</span> its18_select_mc_norm_split<span class="sc">$</span>soil_funct</span>
<span id="cb189-3385"><a href="#cb189-3385" aria-hidden="true" tabindex="-1"></a>soil_funct_norm_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-3386"><a href="#cb189-3386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3387"><a href="#cb189-3387" aria-hidden="true" tabindex="-1"></a>temp_adapt_cor <span class="ot">&lt;-</span> its18_select_mc_split<span class="sc">$</span>temp_adapt</span>
<span id="cb189-3388"><a href="#cb189-3388" aria-hidden="true" tabindex="-1"></a>temp_adapt_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-3389"><a href="#cb189-3389" aria-hidden="true" tabindex="-1"></a>temp_adapt_norm_cor <span class="ot">&lt;-</span> its18_select_mc_norm_split<span class="sc">$</span>temp_adapt</span>
<span id="cb189-3390"><a href="#cb189-3390" aria-hidden="true" tabindex="-1"></a>temp_adapt_norm_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-3391"><a href="#cb189-3391" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3392"><a href="#cb189-3392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3393"><a href="#cb189-3393" aria-hidden="true" tabindex="-1"></a>And finally the code to create the plots. </span>
<span id="cb189-3394"><a href="#cb189-3394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3397"><a href="#cb189-3397" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3398"><a href="#cb189-3398" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-3399"><a href="#cb189-3399" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">"_cor$"</span>)) {</span>
<span id="cb189-3400"><a href="#cb189-3400" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb189-3401"><a href="#cb189-3401" aria-hidden="true" tabindex="-1"></a>  tmp_cormat <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">cor</span>(tmp_get), <span class="dv">2</span>)</span>
<span id="cb189-3402"><a href="#cb189-3402" aria-hidden="true" tabindex="-1"></a>  tmp_melted_cormat <span class="ot">&lt;-</span> reshape2<span class="sc">::</span><span class="fu">melt</span>(tmp_cormat)</span>
<span id="cb189-3403"><a href="#cb189-3403" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb189-3404"><a href="#cb189-3404" aria-hidden="true" tabindex="-1"></a>  tmp_get_lower_tri <span class="ot">&lt;-</span> <span class="cf">function</span>(tmp_cormat){</span>
<span id="cb189-3405"><a href="#cb189-3405" aria-hidden="true" tabindex="-1"></a>    tmp_cormat[<span class="fu">upper.tri</span>(tmp_cormat)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb189-3406"><a href="#cb189-3406" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(tmp_cormat)</span>
<span id="cb189-3407"><a href="#cb189-3407" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb189-3408"><a href="#cb189-3408" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get upper triangle of the correlation matrix</span></span>
<span id="cb189-3409"><a href="#cb189-3409" aria-hidden="true" tabindex="-1"></a>  tmp_get_upper_tri <span class="ot">&lt;-</span> <span class="cf">function</span>(tmp_cormat){</span>
<span id="cb189-3410"><a href="#cb189-3410" aria-hidden="true" tabindex="-1"></a>    tmp_cormat[<span class="fu">lower.tri</span>(tmp_cormat)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb189-3411"><a href="#cb189-3411" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(tmp_cormat)</span>
<span id="cb189-3412"><a href="#cb189-3412" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb189-3413"><a href="#cb189-3413" aria-hidden="true" tabindex="-1"></a>  tmp_upper_tri <span class="ot">&lt;-</span> <span class="fu">tmp_get_upper_tri</span>(tmp_cormat)</span>
<span id="cb189-3414"><a href="#cb189-3414" aria-hidden="true" tabindex="-1"></a>  tmp_melted_cormat <span class="ot">&lt;-</span> reshape2<span class="sc">::</span><span class="fu">melt</span>(tmp_upper_tri, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb189-3415"><a href="#cb189-3415" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="at">data =</span> tmp_melted_cormat, <span class="fu">aes</span>(<span class="at">x =</span> Var1, <span class="at">y =</span> Var2, <span class="at">fill =</span> value)) <span class="sc">+</span> </span>
<span id="cb189-3416"><a href="#cb189-3416" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>()</span>
<span id="cb189-3417"><a href="#cb189-3417" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb189-3418"><a href="#cb189-3418" aria-hidden="true" tabindex="-1"></a>  tmp_ggheatmap <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tmp_melted_cormat, <span class="fu">aes</span>(Var2, Var1, <span class="at">fill =</span> value)) <span class="sc">+</span></span>
<span id="cb189-3419"><a href="#cb189-3419" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>(<span class="at">color =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb189-3420"><a href="#cb189-3420" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> <span class="st">"blue"</span>, <span class="at">high =</span> <span class="st">"red"</span>, <span class="at">mid =</span> <span class="st">"white"</span>, </span>
<span id="cb189-3421"><a href="#cb189-3421" aria-hidden="true" tabindex="-1"></a>                         <span class="at">midpoint =</span> <span class="dv">0</span>, <span class="at">limit =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="at">space =</span> <span class="st">"Lab"</span>, </span>
<span id="cb189-3422"><a href="#cb189-3422" aria-hidden="true" tabindex="-1"></a>                         <span class="at">name=</span><span class="st">"Pearson</span><span class="sc">\n</span><span class="st">Correlation"</span>) <span class="sc">+</span></span>
<span id="cb189-3423"><a href="#cb189-3423" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span> </span>
<span id="cb189-3424"><a href="#cb189-3424" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, </span>
<span id="cb189-3425"><a href="#cb189-3425" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">size =</span> <span class="dv">7</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb189-3426"><a href="#cb189-3426" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">7</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb189-3427"><a href="#cb189-3427" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_fixed</span>() <span class="sc">+</span> </span>
<span id="cb189-3428"><a href="#cb189-3428" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="fu">aes</span>(Var2, Var1, <span class="at">label =</span> value), <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="fl">1.75</span>) <span class="sc">+</span></span>
<span id="cb189-3429"><a href="#cb189-3429" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb189-3430"><a href="#cb189-3430" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb189-3431"><a href="#cb189-3431" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb189-3432"><a href="#cb189-3432" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb189-3433"><a href="#cb189-3433" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb189-3434"><a href="#cb189-3434" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb189-3435"><a href="#cb189-3435" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb189-3436"><a href="#cb189-3436" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">0</span>),</span>
<span id="cb189-3437"><a href="#cb189-3437" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.6</span>, <span class="fl">0.7</span>),</span>
<span id="cb189-3438"><a href="#cb189-3438" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.direction =</span> <span class="st">"horizontal"</span>) <span class="sc">+</span></span>
<span id="cb189-3439"><a href="#cb189-3439" aria-hidden="true" tabindex="-1"></a>      <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_colorbar</span>(<span class="at">barwidth =</span> <span class="dv">7</span>, <span class="at">barheight =</span> <span class="dv">1</span>,</span>
<span id="cb189-3440"><a href="#cb189-3440" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">title.position =</span> <span class="st">"top"</span>, </span>
<span id="cb189-3441"><a href="#cb189-3441" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">title.hjust =</span> <span class="fl">0.5</span>))</span>
<span id="cb189-3442"><a href="#cb189-3442" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_ggheatmap"</span>))</span>
<span id="cb189-3443"><a href="#cb189-3443" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_ggheatmap)</span>
<span id="cb189-3444"><a href="#cb189-3444" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(tmp_name)</span>
<span id="cb189-3445"><a href="#cb189-3445" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-3446"><a href="#cb189-3446" aria-hidden="true" tabindex="-1"></a>}  </span>
<span id="cb189-3447"><a href="#cb189-3447" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">"_ggheatmap"</span>)</span>
<span id="cb189-3448"><a href="#cb189-3448" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3449"><a href="#cb189-3449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3450"><a href="#cb189-3450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3451"><a href="#cb189-3451" aria-hidden="true" tabindex="-1"></a><span class="fu">## Autocorrelation Plots</span></span>
<span id="cb189-3452"><a href="#cb189-3452" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br/&gt;</span></span>
<span id="cb189-3453"><a href="#cb189-3453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3454"><a href="#cb189-3454" aria-hidden="true" tabindex="-1"></a>::: {.column-body}</span>
<span id="cb189-3455"><a href="#cb189-3455" aria-hidden="true" tabindex="-1"></a>::: {.panel-tabset}</span>
<span id="cb189-3456"><a href="#cb189-3456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3457"><a href="#cb189-3457" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Edaphic properties</span></span>
<span id="cb189-3458"><a href="#cb189-3458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3461"><a href="#cb189-3461" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3462"><a href="#cb189-3462" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3463"><a href="#cb189-3463" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-3464"><a href="#cb189-3464" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb189-3465"><a href="#cb189-3465" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4.5</span></span>
<span id="cb189-3466"><a href="#cb189-3466" aria-hidden="true" tabindex="-1"></a>edaphic_cor_ggheatmap</span>
<span id="cb189-3467"><a href="#cb189-3467" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3468"><a href="#cb189-3468" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig_its("its_autocor_edaph_plots")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-3469"><a href="#cb189-3469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3470"><a href="#cb189-3470" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Edaphic properties (normalized)</span></span>
<span id="cb189-3471"><a href="#cb189-3471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3474"><a href="#cb189-3474" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3475"><a href="#cb189-3475" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3476"><a href="#cb189-3476" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-3477"><a href="#cb189-3477" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb189-3478"><a href="#cb189-3478" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4.5</span></span>
<span id="cb189-3479"><a href="#cb189-3479" aria-hidden="true" tabindex="-1"></a>edaphic_norm_cor_ggheatmap</span>
<span id="cb189-3480"><a href="#cb189-3480" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3481"><a href="#cb189-3481" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig_its("its_autocor_edaph_norm_plots")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-3482"><a href="#cb189-3482" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb189-3483"><a href="#cb189-3483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3484"><a href="#cb189-3484" aria-hidden="true" tabindex="-1"></a>::: {.panel-tabset}</span>
<span id="cb189-3485"><a href="#cb189-3485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3486"><a href="#cb189-3486" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Functional responses</span></span>
<span id="cb189-3487"><a href="#cb189-3487" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3490"><a href="#cb189-3490" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3491"><a href="#cb189-3491" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3492"><a href="#cb189-3492" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-3493"><a href="#cb189-3493" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb189-3494"><a href="#cb189-3494" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4.5</span></span>
<span id="cb189-3495"><a href="#cb189-3495" aria-hidden="true" tabindex="-1"></a>soil_funct_cor_ggheatmap</span>
<span id="cb189-3496"><a href="#cb189-3496" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3497"><a href="#cb189-3497" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig_its("its_autocor_funct_plots")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-3498"><a href="#cb189-3498" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3499"><a href="#cb189-3499" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3500"><a href="#cb189-3500" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Functional responses (normalized)</span></span>
<span id="cb189-3501"><a href="#cb189-3501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3504"><a href="#cb189-3504" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3505"><a href="#cb189-3505" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3506"><a href="#cb189-3506" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-3507"><a href="#cb189-3507" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb189-3508"><a href="#cb189-3508" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4.5</span></span>
<span id="cb189-3509"><a href="#cb189-3509" aria-hidden="true" tabindex="-1"></a>soil_funct_norm_cor_ggheatmap</span>
<span id="cb189-3510"><a href="#cb189-3510" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3511"><a href="#cb189-3511" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig_its("its_autocor_funct_norm_plots")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-3512"><a href="#cb189-3512" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3513"><a href="#cb189-3513" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb189-3514"><a href="#cb189-3514" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3515"><a href="#cb189-3515" aria-hidden="true" tabindex="-1"></a>::: {.panel-tabset}</span>
<span id="cb189-3516"><a href="#cb189-3516" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3517"><a href="#cb189-3517" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Temperature adaptation </span></span>
<span id="cb189-3518"><a href="#cb189-3518" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3521"><a href="#cb189-3521" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3522"><a href="#cb189-3522" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3523"><a href="#cb189-3523" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-3524"><a href="#cb189-3524" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb189-3525"><a href="#cb189-3525" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4.5</span></span>
<span id="cb189-3526"><a href="#cb189-3526" aria-hidden="true" tabindex="-1"></a>temp_adapt_cor_ggheatmap</span>
<span id="cb189-3527"><a href="#cb189-3527" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3528"><a href="#cb189-3528" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig_its("its_autocor_temp_plots")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-3529"><a href="#cb189-3529" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3530"><a href="#cb189-3530" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Temperature adaptation (normalized)</span></span>
<span id="cb189-3531"><a href="#cb189-3531" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3534"><a href="#cb189-3534" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3535"><a href="#cb189-3535" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3536"><a href="#cb189-3536" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-3537"><a href="#cb189-3537" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb189-3538"><a href="#cb189-3538" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4.5</span></span>
<span id="cb189-3539"><a href="#cb189-3539" aria-hidden="true" tabindex="-1"></a>temp_adapt_norm_cor_ggheatmap</span>
<span id="cb189-3540"><a href="#cb189-3540" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3541"><a href="#cb189-3541" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig_its("its_autocor_temp_norm_plots")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-3542"><a href="#cb189-3542" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb189-3543"><a href="#cb189-3543" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb189-3544"><a href="#cb189-3544" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3547"><a href="#cb189-3547" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3548"><a href="#cb189-3548" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3549"><a href="#cb189-3549" aria-hidden="true" tabindex="-1"></a>auto_cor_figs <span class="ot">&lt;-</span> <span class="fu">ggarrange</span>(</span>
<span id="cb189-3550"><a href="#cb189-3550" aria-hidden="true" tabindex="-1"></a>  edaphic_cor_ggheatmap, edaphic_norm_cor_ggheatmap,</span>
<span id="cb189-3551"><a href="#cb189-3551" aria-hidden="true" tabindex="-1"></a>  soil_funct_cor_ggheatmap, soil_funct_norm_cor_ggheatmap,</span>
<span id="cb189-3552"><a href="#cb189-3552" aria-hidden="true" tabindex="-1"></a>  temp_adapt_cor_ggheatmap, temp_adapt_norm_cor_ggheatmap,</span>
<span id="cb189-3553"><a href="#cb189-3553" aria-hidden="true" tabindex="-1"></a>  <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">3</span>, <span class="at">common.legend =</span> <span class="cn">FALSE</span>)</span>
<span id="cb189-3554"><a href="#cb189-3554" aria-hidden="true" tabindex="-1"></a>auto_cor_figs</span>
<span id="cb189-3555"><a href="#cb189-3555" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb189-3556"><a href="#cb189-3556" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">"files/metadata/figures/its18_auto_cor_figs.png"</span>,</span>
<span id="cb189-3557"><a href="#cb189-3557" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">32</span>, <span class="at">width =</span> <span class="dv">24</span>, <span class="at">units =</span> <span class="st">'cm'</span>, <span class="at">res =</span> <span class="dv">600</span>, <span class="at">bg =</span> <span class="st">"white"</span>)</span>
<span id="cb189-3558"><a href="#cb189-3558" aria-hidden="true" tabindex="-1"></a>auto_cor_figs</span>
<span id="cb189-3559"><a href="#cb189-3559" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb189-3560"><a href="#cb189-3560" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"files/metadata/figures/its18_auto_cor_figs.pdf"</span>,</span>
<span id="cb189-3561"><a href="#cb189-3561" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">10</span>, <span class="at">width =</span> <span class="dv">12</span>)</span>
<span id="cb189-3562"><a href="#cb189-3562" aria-hidden="true" tabindex="-1"></a>auto_cor_figs</span>
<span id="cb189-3563"><a href="#cb189-3563" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb189-3564"><a href="#cb189-3564" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3565"><a href="#cb189-3565" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3566"><a href="#cb189-3566" aria-hidden="true" tabindex="-1"></a>Now we can remove parameters based on the autocorrelation analysis:</span>
<span id="cb189-3567"><a href="#cb189-3567" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3568"><a href="#cb189-3568" aria-hidden="true" tabindex="-1"></a>A) Environmental and edaphic properties: TEB, DON, Na, Al, Ca.</span>
<span id="cb189-3569"><a href="#cb189-3569" aria-hidden="true" tabindex="-1"></a>B) Microbial functional responses: micN, micNP, enzCN, enzCP, BP_ase, CE_ase, LP_ase, N_ase, P_ase.</span>
<span id="cb189-3570"><a href="#cb189-3570" aria-hidden="true" tabindex="-1"></a>C) Temperature adaptation properties: NUE, PUE, P_Q10, SI.</span>
<span id="cb189-3571"><a href="#cb189-3571" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3574"><a href="#cb189-3574" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3575"><a href="#cb189-3575" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3576"><a href="#cb189-3576" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb189-3577"><a href="#cb189-3577" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">"_cor_ggheatmap"</span>)</span>
<span id="cb189-3578"><a href="#cb189-3578" aria-hidden="true" tabindex="-1"></a>tmp_res <span class="ot">&lt;-</span> soil_funct_norm_cor_ggheatmap<span class="sc">$</span>data</span>
<span id="cb189-3579"><a href="#cb189-3579" aria-hidden="true" tabindex="-1"></a>value_choice <span class="ot">&lt;-</span> <span class="fl">0.7</span></span>
<span id="cb189-3580"><a href="#cb189-3580" aria-hidden="true" tabindex="-1"></a>tmp_res <span class="sc">%&gt;%</span> <span class="fu">filter</span>(value <span class="sc">&gt;</span> value_choice <span class="sc">|</span> value <span class="sc">&lt;</span> <span class="sc">-</span>value_choice) <span class="sc">%&gt;%</span> <span class="fu">filter_</span>(<span class="st">"Var1 != Var2"</span>)</span>
<span id="cb189-3581"><a href="#cb189-3581" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3582"><a href="#cb189-3582" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_res %&gt;% filter(Var1 == "DOCN" | Var2 == "DOCN") %&gt;% filter_("Var1 != Var2")</span></span>
<span id="cb189-3583"><a href="#cb189-3583" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(tmp_res <span class="sc">%&gt;%</span> <span class="fu">filter</span>(value <span class="sc">&gt;</span> value_choice <span class="sc">|</span> value <span class="sc">&lt;</span> <span class="sc">-</span>value_choice) <span class="sc">%&gt;%</span> </span>
<span id="cb189-3584"><a href="#cb189-3584" aria-hidden="true" tabindex="-1"></a>              <span class="fu">filter_</span>(<span class="st">"Var1 != Var2"</span>), </span>
<span id="cb189-3585"><a href="#cb189-3585" aria-hidden="true" tabindex="-1"></a>            <span class="st">"soil_funct_norm.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, </span>
<span id="cb189-3586"><a href="#cb189-3586" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb189-3587"><a href="#cb189-3587" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3588"><a href="#cb189-3588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3591"><a href="#cb189-3591" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3592"><a href="#cb189-3592" aria-hidden="true" tabindex="-1"></a>edaphic_remove <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"TEB"</span>, <span class="st">"DON"</span>, <span class="st">"Na"</span>, <span class="st">"Al"</span>, <span class="st">"Ca"</span>)</span>
<span id="cb189-3593"><a href="#cb189-3593" aria-hidden="true" tabindex="-1"></a>soil_funct_remove <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"micN"</span>, <span class="st">"micNP"</span>, <span class="st">"enzCN"</span>, <span class="st">"enzCP"</span>, <span class="st">"BP_ase"</span>, </span>
<span id="cb189-3594"><a href="#cb189-3594" aria-hidden="true" tabindex="-1"></a>                       <span class="st">"CE_ase"</span>, <span class="st">"LP_ase"</span>, <span class="st">"N_ase"</span>, <span class="st">"P_ase"</span>)</span>
<span id="cb189-3595"><a href="#cb189-3595" aria-hidden="true" tabindex="-1"></a>temp_adapt_remove <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"NUE"</span>, <span class="st">"PUE"</span>, <span class="st">"P_Q10"</span>, <span class="st">"SI"</span>)</span>
<span id="cb189-3596"><a href="#cb189-3596" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3597"><a href="#cb189-3597" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3600"><a href="#cb189-3600" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3601"><a href="#cb189-3601" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-3602"><a href="#cb189-3602" aria-hidden="true" tabindex="-1"></a>tmp_df <span class="ot">&lt;-</span> its18_select_mc_split</span>
<span id="cb189-3603"><a href="#cb189-3603" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>edaphic <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>edaphic[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>edaphic) <span class="sc">%in%</span>  edaphic_remove]</span>
<span id="cb189-3604"><a href="#cb189-3604" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>soil_funct <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>soil_funct[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>soil_funct) <span class="sc">%in%</span>  soil_funct_remove]</span>
<span id="cb189-3605"><a href="#cb189-3605" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>temp_adapt <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>temp_adapt[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>temp_adapt) <span class="sc">%in%</span>  temp_adapt_remove]</span>
<span id="cb189-3606"><a href="#cb189-3606" aria-hidden="true" tabindex="-1"></a>its18_select_mc_split_no_ac <span class="ot">&lt;-</span> tmp_df</span>
<span id="cb189-3607"><a href="#cb189-3607" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-3608"><a href="#cb189-3608" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3609"><a href="#cb189-3609" aria-hidden="true" tabindex="-1"></a>tmp_df <span class="ot">&lt;-</span> its18_select_mc_norm_split</span>
<span id="cb189-3610"><a href="#cb189-3610" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>edaphic <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>edaphic[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>edaphic) <span class="sc">%in%</span>  edaphic_remove]</span>
<span id="cb189-3611"><a href="#cb189-3611" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>soil_funct <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>soil_funct[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>soil_funct) <span class="sc">%in%</span>  soil_funct_remove]</span>
<span id="cb189-3612"><a href="#cb189-3612" aria-hidden="true" tabindex="-1"></a>tmp_df<span class="sc">$</span>temp_adapt <span class="ot">&lt;-</span> tmp_df<span class="sc">$</span>temp_adapt[, <span class="sc">!</span> <span class="fu">names</span>(tmp_df<span class="sc">$</span>temp_adapt) <span class="sc">%in%</span>  temp_adapt_remove]</span>
<span id="cb189-3613"><a href="#cb189-3613" aria-hidden="true" tabindex="-1"></a>its18_select_mc_norm_split_no_ac <span class="ot">&lt;-</span> tmp_df</span>
<span id="cb189-3614"><a href="#cb189-3614" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-3615"><a href="#cb189-3615" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3616"><a href="#cb189-3616" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3617"><a href="#cb189-3617" aria-hidden="true" tabindex="-1"></a><span class="fu">## Dissimilarity Correlation Tests</span></span>
<span id="cb189-3618"><a href="#cb189-3618" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3619"><a href="#cb189-3619" aria-hidden="true" tabindex="-1"></a>Let's see if any on the metadata groups are significantly correlated with the community data. Basically, we create distance matrices for the community data and each metadata group and then run Mantel tests for all comparisons. For the community data we calculate Bray-Curtis distances for the community data and  Euclidean distances for the metadata. We use the function <span class="in">`mantel.test`</span> from the <span class="in">`ape`</span> package and <span class="in">`mantel`</span> from the <span class="in">`vegan`</span> package for the analyses. </span>
<span id="cb189-3620"><a href="#cb189-3620" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3621"><a href="#cb189-3621" aria-hidden="true" tabindex="-1"></a>In summary, we test both <span class="in">`mantel.test`</span> and <span class="in">`mantel`</span> on  Bray-Curtis distance community distances against Euclidean distances for each metadata group (<span class="in">`edaphic`</span>, <span class="in">`soil_funct`</span>, <span class="in">`temp_adapt`</span>)  **a**) before normalizing and before removing autocorrelated parameters, **b**) before normalizing and after removing autocorrelated parameters, **c**) after normalizing and before removing autocorrelated parameters, and **d**) after normalizing and after removing autocorrelated parameters.</span>
<span id="cb189-3622"><a href="#cb189-3622" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3625"><a href="#cb189-3625" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3626"><a href="#cb189-3626" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-3627"><a href="#cb189-3627" aria-hidden="true" tabindex="-1"></a>man_df <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"its18_select_mc_split"</span>, <span class="st">"its18_select_mc_split_no_ac"</span>, </span>
<span id="cb189-3628"><a href="#cb189-3628" aria-hidden="true" tabindex="-1"></a>            <span class="st">"its18_select_mc_norm_split"</span>, <span class="st">"its18_select_mc_norm_split_no_ac"</span>)</span>
<span id="cb189-3629"><a href="#cb189-3629" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3630"><a href="#cb189-3630" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> man_df) {</span>
<span id="cb189-3631"><a href="#cb189-3631" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb189-3632"><a href="#cb189-3632" aria-hidden="true" tabindex="-1"></a>  tmp_dm_otu <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(<span class="fu">t</span>(tmp_get<span class="sc">$</span>data_loaded), </span>
<span id="cb189-3633"><a href="#cb189-3633" aria-hidden="true" tabindex="-1"></a>                             <span class="at">method =</span> <span class="st">"bray"</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb189-3634"><a href="#cb189-3634" aria-hidden="true" tabindex="-1"></a>                             <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb189-3635"><a href="#cb189-3635" aria-hidden="true" tabindex="-1"></a><span class="co"># EDAPHIC</span></span>
<span id="cb189-3636"><a href="#cb189-3636" aria-hidden="true" tabindex="-1"></a>  tmp_dm_md_edaphic <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_get<span class="sc">$</span>edaphic[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_get<span class="sc">$</span>edaphic)], </span>
<span id="cb189-3637"><a href="#cb189-3637" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">method =</span><span class="st">"euclidean"</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb189-3638"><a href="#cb189-3638" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb189-3639"><a href="#cb189-3639" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3640"><a href="#cb189-3640" aria-hidden="true" tabindex="-1"></a>  tmp_man1_edaphic <span class="ot">&lt;-</span> <span class="fu">mantel.test</span>(tmp_dm_otu, tmp_dm_md_edaphic, <span class="at">nperm =</span> <span class="dv">999</span>, </span>
<span id="cb189-3641"><a href="#cb189-3641" aria-hidden="true" tabindex="-1"></a>                             <span class="at">graph =</span> <span class="cn">FALSE</span>, <span class="at">alternative =</span> <span class="st">"two.sided"</span>)</span>
<span id="cb189-3642"><a href="#cb189-3642" aria-hidden="true" tabindex="-1"></a>  tmp_man2_edaphic <span class="ot">&lt;-</span> <span class="fu">mantel</span>(tmp_dm_otu, tmp_dm_md_edaphic, <span class="at">permutations =</span> <span class="dv">999</span>)</span>
<span id="cb189-3643"><a href="#cb189-3643" aria-hidden="true" tabindex="-1"></a><span class="co"># SOIL FUNCT</span></span>
<span id="cb189-3644"><a href="#cb189-3644" aria-hidden="true" tabindex="-1"></a>  tmp_dm_md_soil_funct <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_get<span class="sc">$</span>soil_funct[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_get<span class="sc">$</span>soil_funct)], </span>
<span id="cb189-3645"><a href="#cb189-3645" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">method =</span> <span class="st">"euclidean"</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb189-3646"><a href="#cb189-3646" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb189-3647"><a href="#cb189-3647" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3648"><a href="#cb189-3648" aria-hidden="true" tabindex="-1"></a>  tmp_man1_soil_funct <span class="ot">&lt;-</span> <span class="fu">mantel.test</span>(tmp_dm_otu, tmp_dm_md_soil_funct, <span class="at">nperm =</span> <span class="dv">999</span>, </span>
<span id="cb189-3649"><a href="#cb189-3649" aria-hidden="true" tabindex="-1"></a>                                <span class="at">graph =</span> <span class="cn">FALSE</span>, <span class="at">alternative =</span> <span class="st">"two.sided"</span>)</span>
<span id="cb189-3650"><a href="#cb189-3650" aria-hidden="true" tabindex="-1"></a>  tmp_man2_soil_funct <span class="ot">&lt;-</span> <span class="fu">mantel</span>(tmp_dm_otu, tmp_dm_md_soil_funct, <span class="at">permutations =</span> <span class="dv">999</span>)</span>
<span id="cb189-3651"><a href="#cb189-3651" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3652"><a href="#cb189-3652" aria-hidden="true" tabindex="-1"></a><span class="co"># TEMP ADAPT</span></span>
<span id="cb189-3653"><a href="#cb189-3653" aria-hidden="true" tabindex="-1"></a>  tmp_dm_md_temp_adapt <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_get<span class="sc">$</span>temp_adapt[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_get<span class="sc">$</span>temp_adapt)], </span>
<span id="cb189-3654"><a href="#cb189-3654" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">method =</span> <span class="st">"euclidean"</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb189-3655"><a href="#cb189-3655" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb189-3656"><a href="#cb189-3656" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3657"><a href="#cb189-3657" aria-hidden="true" tabindex="-1"></a>  tmp_man1_temp_adapt <span class="ot">&lt;-</span> <span class="fu">mantel.test</span>(tmp_dm_otu, tmp_dm_md_temp_adapt, <span class="at">nperm =</span> <span class="dv">999</span>, </span>
<span id="cb189-3658"><a href="#cb189-3658" aria-hidden="true" tabindex="-1"></a>                                <span class="at">graph =</span> <span class="cn">FALSE</span>, <span class="at">alternative =</span> <span class="st">"two.sided"</span>)</span>
<span id="cb189-3659"><a href="#cb189-3659" aria-hidden="true" tabindex="-1"></a>  tmp_man2_temp_adapt <span class="ot">&lt;-</span> <span class="fu">mantel</span>(tmp_dm_otu, tmp_dm_md_temp_adapt, <span class="at">permutations =</span> <span class="dv">999</span>)</span>
<span id="cb189-3660"><a href="#cb189-3660" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3661"><a href="#cb189-3661" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_mantel_tests"</span>))</span>
<span id="cb189-3662"><a href="#cb189-3662" aria-hidden="true" tabindex="-1"></a>  tmp_df <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">edaphic_ape_man =</span> tmp_man1_edaphic, </span>
<span id="cb189-3663"><a href="#cb189-3663" aria-hidden="true" tabindex="-1"></a>                 <span class="at">edaphic_vegan_man =</span> tmp_man2_edaphic,</span>
<span id="cb189-3664"><a href="#cb189-3664" aria-hidden="true" tabindex="-1"></a>                 <span class="at">soil_funct_ape_man =</span> tmp_man1_soil_funct, </span>
<span id="cb189-3665"><a href="#cb189-3665" aria-hidden="true" tabindex="-1"></a>                 <span class="at">soil_funct_vegan_man =</span> tmp_man2_soil_funct,</span>
<span id="cb189-3666"><a href="#cb189-3666" aria-hidden="true" tabindex="-1"></a>                 <span class="at">temp_adapt_ape_man =</span> tmp_man1_temp_adapt, </span>
<span id="cb189-3667"><a href="#cb189-3667" aria-hidden="true" tabindex="-1"></a>                 <span class="at">temp_adapt_vegan_man =</span> tmp_man2_temp_adapt)</span>
<span id="cb189-3668"><a href="#cb189-3668" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_df)</span>
<span id="cb189-3669"><a href="#cb189-3669" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(tmp_name)</span>
<span id="cb189-3670"><a href="#cb189-3670" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-3671"><a href="#cb189-3671" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-3672"><a href="#cb189-3672" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3673"><a href="#cb189-3673" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3676"><a href="#cb189-3676" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3677"><a href="#cb189-3677" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3678"><a href="#cb189-3678" aria-hidden="true" tabindex="-1"></a>tmp_objects <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"its18_select_mc_split_mantel_tests"</span>, </span>
<span id="cb189-3679"><a href="#cb189-3679" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"its18_select_mc_split_no_ac_mantel_tests"</span>,</span>
<span id="cb189-3680"><a href="#cb189-3680" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"its18_select_mc_norm_split_mantel_tests"</span>,</span>
<span id="cb189-3681"><a href="#cb189-3681" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"its18_select_mc_norm_split_no_ac_mantel_tests"</span>)</span>
<span id="cb189-3682"><a href="#cb189-3682" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3683"><a href="#cb189-3683" aria-hidden="true" tabindex="-1"></a>tmp_norm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">c</span>(<span class="st">"no"</span>, <span class="st">"no"</span>, <span class="st">"yes"</span>, <span class="st">"yes"</span>))</span>
<span id="cb189-3684"><a href="#cb189-3684" aria-hidden="true" tabindex="-1"></a>tmp_ac <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">c</span>(<span class="st">"no"</span>, <span class="st">"yes"</span>, <span class="st">"no"</span>, <span class="st">"yes"</span>))</span>
<span id="cb189-3685"><a href="#cb189-3685" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_mantel_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(tmp_norm, tmp_ac) <span class="sc">%&gt;%</span></span>
<span id="cb189-3686"><a href="#cb189-3686" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"normalized?"</span> <span class="ot">=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb189-3687"><a href="#cb189-3687" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"AC removed?"</span> <span class="ot">=</span> <span class="dv">2</span>)</span>
<span id="cb189-3688"><a href="#cb189-3688" aria-hidden="true" tabindex="-1"></a>tmp_pvalue <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">c</span>(<span class="fu">get</span>(tmp_objects[<span class="dv">1</span>])<span class="sc">$</span>edaphic_ape_man<span class="sc">$</span>p,</span>
<span id="cb189-3689"><a href="#cb189-3689" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">2</span>])<span class="sc">$</span>edaphic_ape_man<span class="sc">$</span>p,</span>
<span id="cb189-3690"><a href="#cb189-3690" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">3</span>])<span class="sc">$</span>edaphic_ape_man<span class="sc">$</span>p,</span>
<span id="cb189-3691"><a href="#cb189-3691" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">4</span>])<span class="sc">$</span>edaphic_ape_man<span class="sc">$</span>p))</span>
<span id="cb189-3692"><a href="#cb189-3692" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_mantel_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(its18_ps_pime_mantel_summary, tmp_pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb189-3693"><a href="#cb189-3693" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"edaphic_ape"</span> <span class="ot">=</span> <span class="dv">3</span>)</span>
<span id="cb189-3694"><a href="#cb189-3694" aria-hidden="true" tabindex="-1"></a>tmp_pvalue <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">c</span>(<span class="fu">get</span>(tmp_objects[<span class="dv">1</span>])<span class="sc">$</span>edaphic_vegan_man<span class="sc">$</span>signif,</span>
<span id="cb189-3695"><a href="#cb189-3695" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">2</span>])<span class="sc">$</span>edaphic_vegan_man<span class="sc">$</span>signif,</span>
<span id="cb189-3696"><a href="#cb189-3696" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">3</span>])<span class="sc">$</span>edaphic_vegan_man<span class="sc">$</span>signif,</span>
<span id="cb189-3697"><a href="#cb189-3697" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">4</span>])<span class="sc">$</span>edaphic_vegan_man<span class="sc">$</span>signif))</span>
<span id="cb189-3698"><a href="#cb189-3698" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_mantel_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(its18_ps_pime_mantel_summary, tmp_pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb189-3699"><a href="#cb189-3699" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"edaphic_vegan"</span> <span class="ot">=</span> <span class="dv">4</span>)</span>
<span id="cb189-3700"><a href="#cb189-3700" aria-hidden="true" tabindex="-1"></a>tmp_pvalue <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">c</span>(<span class="fu">get</span>(tmp_objects[<span class="dv">1</span>])<span class="sc">$</span>soil_funct_ape_man<span class="sc">$</span>p,</span>
<span id="cb189-3701"><a href="#cb189-3701" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">2</span>])<span class="sc">$</span>soil_funct_ape_man<span class="sc">$</span>p,</span>
<span id="cb189-3702"><a href="#cb189-3702" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">3</span>])<span class="sc">$</span>soil_funct_ape_man<span class="sc">$</span>p,</span>
<span id="cb189-3703"><a href="#cb189-3703" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">4</span>])<span class="sc">$</span>soil_funct_ape_man<span class="sc">$</span>p))</span>
<span id="cb189-3704"><a href="#cb189-3704" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_mantel_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(its18_ps_pime_mantel_summary, tmp_pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb189-3705"><a href="#cb189-3705" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"soil_funct_ape"</span> <span class="ot">=</span> <span class="dv">5</span>)</span>
<span id="cb189-3706"><a href="#cb189-3706" aria-hidden="true" tabindex="-1"></a>tmp_pvalue <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">c</span>(<span class="fu">get</span>(tmp_objects[<span class="dv">1</span>])<span class="sc">$</span>soil_funct_vegan_man<span class="sc">$</span>signif,</span>
<span id="cb189-3707"><a href="#cb189-3707" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">2</span>])<span class="sc">$</span>soil_funct_vegan_man<span class="sc">$</span>signif,</span>
<span id="cb189-3708"><a href="#cb189-3708" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">3</span>])<span class="sc">$</span>soil_funct_vegan_man<span class="sc">$</span>signif,</span>
<span id="cb189-3709"><a href="#cb189-3709" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">4</span>])<span class="sc">$</span>soil_funct_vegan_man<span class="sc">$</span>signif))</span>
<span id="cb189-3710"><a href="#cb189-3710" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_mantel_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(its18_ps_pime_mantel_summary, tmp_pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb189-3711"><a href="#cb189-3711" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"soil_funct_vegan"</span> <span class="ot">=</span> <span class="dv">6</span>)</span>
<span id="cb189-3712"><a href="#cb189-3712" aria-hidden="true" tabindex="-1"></a>tmp_pvalue <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">c</span>(<span class="fu">get</span>(tmp_objects[<span class="dv">1</span>])<span class="sc">$</span>temp_adapt_ape_man<span class="sc">$</span>p,</span>
<span id="cb189-3713"><a href="#cb189-3713" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">2</span>])<span class="sc">$</span>temp_adapt_ape_man<span class="sc">$</span>p,</span>
<span id="cb189-3714"><a href="#cb189-3714" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">3</span>])<span class="sc">$</span>temp_adapt_ape_man<span class="sc">$</span>p,</span>
<span id="cb189-3715"><a href="#cb189-3715" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">4</span>])<span class="sc">$</span>temp_adapt_ape_man<span class="sc">$</span>p))</span>
<span id="cb189-3716"><a href="#cb189-3716" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_mantel_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(its18_ps_pime_mantel_summary, tmp_pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb189-3717"><a href="#cb189-3717" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"temp_adapt_ape"</span> <span class="ot">=</span> <span class="dv">7</span>)</span>
<span id="cb189-3718"><a href="#cb189-3718" aria-hidden="true" tabindex="-1"></a>tmp_pvalue <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">c</span>(<span class="fu">get</span>(tmp_objects[<span class="dv">1</span>])<span class="sc">$</span>temp_adapt_vegan_man<span class="sc">$</span>signif,</span>
<span id="cb189-3719"><a href="#cb189-3719" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">2</span>])<span class="sc">$</span>temp_adapt_vegan_man<span class="sc">$</span>signif,</span>
<span id="cb189-3720"><a href="#cb189-3720" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">3</span>])<span class="sc">$</span>temp_adapt_vegan_man<span class="sc">$</span>signif,</span>
<span id="cb189-3721"><a href="#cb189-3721" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">get</span>(tmp_objects[<span class="dv">4</span>])<span class="sc">$</span>temp_adapt_vegan_man<span class="sc">$</span>signif))</span>
<span id="cb189-3722"><a href="#cb189-3722" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_mantel_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(its18_ps_pime_mantel_summary, tmp_pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb189-3723"><a href="#cb189-3723" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"temp_adapt_vegan"</span> <span class="ot">=</span> <span class="dv">8</span>)</span>
<span id="cb189-3724"><a href="#cb189-3724" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3725"><a href="#cb189-3725" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-3726"><a href="#cb189-3726" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3727"><a href="#cb189-3727" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3728"><a href="#cb189-3728" aria-hidden="true" tabindex="-1"></a><span class="fu">### Dissimilarity Correlation Results</span></span>
<span id="cb189-3729"><a href="#cb189-3729" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3730"><a href="#cb189-3730" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_tab_its("its_mantel_summ")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-3731"><a href="#cb189-3731" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3732"><a href="#cb189-3732" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3735"><a href="#cb189-3735" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3736"><a href="#cb189-3736" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3737"><a href="#cb189-3737" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-3738"><a href="#cb189-3738" aria-hidden="true" tabindex="-1"></a>seq_table <span class="ot">&lt;-</span> its18_ps_pime_mantel_summary</span>
<span id="cb189-3739"><a href="#cb189-3739" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3740"><a href="#cb189-3740" aria-hidden="true" tabindex="-1"></a><span class="fu">reactable</span>(seq_table,</span>
<span id="cb189-3741"><a href="#cb189-3741" aria-hidden="true" tabindex="-1"></a>  <span class="at">defaultColDef =</span> <span class="fu">colDef</span>(</span>
<span id="cb189-3742"><a href="#cb189-3742" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cf">function</span>(value) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, value, <span class="at">fixed =</span> <span class="cn">TRUE</span>),</span>
<span id="cb189-3743"><a href="#cb189-3743" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">format</span>(value, <span class="at">nsmall =</span> <span class="dv">3</span>),</span>
<span id="cb189-3744"><a href="#cb189-3744" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">"center"</span>, <span class="at">filterable =</span> <span class="cn">FALSE</span>, <span class="at">sortable =</span> <span class="cn">FALSE</span>, <span class="at">resizable =</span> <span class="cn">TRUE</span>, <span class="at">minWidth =</span> <span class="dv">50</span>,</span>
<span id="cb189-3745"><a href="#cb189-3745" aria-hidden="true" tabindex="-1"></a>    <span class="at">style =</span> <span class="cf">function</span>(value) {</span>
<span id="cb189-3746"><a href="#cb189-3746" aria-hidden="true" tabindex="-1"></a>                                         <span class="cf">if</span> (value <span class="sc">&gt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb189-3747"><a href="#cb189-3747" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#2271B2"</span> <span class="co">#green</span></span>
<span id="cb189-3748"><a href="#cb189-3748" aria-hidden="true" tabindex="-1"></a>                                         } <span class="cf">else</span> <span class="cf">if</span> (value <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb189-3749"><a href="#cb189-3749" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#B22271"</span> <span class="co">#RED "#", "#"</span></span>
<span id="cb189-3750"><a href="#cb189-3750" aria-hidden="true" tabindex="-1"></a>                                         } <span class="cf">else</span> {</span>
<span id="cb189-3751"><a href="#cb189-3751" aria-hidden="true" tabindex="-1"></a>                                           color <span class="ot">&lt;-</span> <span class="st">"#777"</span></span>
<span id="cb189-3752"><a href="#cb189-3752" aria-hidden="true" tabindex="-1"></a>                                         }</span>
<span id="cb189-3753"><a href="#cb189-3753" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">list</span>(<span class="at">color =</span> color)}</span>
<span id="cb189-3754"><a href="#cb189-3754" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb189-3755"><a href="#cb189-3755" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb189-3756"><a href="#cb189-3756" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb189-3757"><a href="#cb189-3757" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">normalized?</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"normalized?"</span>, </span>
<span id="cb189-3758"><a href="#cb189-3758" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>),</span>
<span id="cb189-3759"><a href="#cb189-3759" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>), </span>
<span id="cb189-3760"><a href="#cb189-3760" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb189-3761"><a href="#cb189-3761" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">70</span></span>
<span id="cb189-3762"><a href="#cb189-3762" aria-hidden="true" tabindex="-1"></a>                       ),</span>
<span id="cb189-3763"><a href="#cb189-3763" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">AC removed?</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"AC removed?"</span>, </span>
<span id="cb189-3764"><a href="#cb189-3764" aria-hidden="true" tabindex="-1"></a>                       <span class="at">align =</span> <span class="st">"left"</span>,</span>
<span id="cb189-3765"><a href="#cb189-3765" aria-hidden="true" tabindex="-1"></a>                       <span class="at">minWidth =</span> <span class="dv">70</span>,</span>
<span id="cb189-3766"><a href="#cb189-3766" aria-hidden="true" tabindex="-1"></a>                       <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>), </span>
<span id="cb189-3767"><a href="#cb189-3767" aria-hidden="true" tabindex="-1"></a>                       <span class="at">style =</span> <span class="fu">list</span>(<span class="at">borderRight =</span> <span class="st">"5px solid #eee"</span>)</span>
<span id="cb189-3768"><a href="#cb189-3768" aria-hidden="true" tabindex="-1"></a>                       ),</span>
<span id="cb189-3769"><a href="#cb189-3769" aria-hidden="true" tabindex="-1"></a>    <span class="at">edaphic_ape =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"ape"</span>, </span>
<span id="cb189-3770"><a href="#cb189-3770" aria-hidden="true" tabindex="-1"></a>                                ),</span>
<span id="cb189-3771"><a href="#cb189-3771" aria-hidden="true" tabindex="-1"></a>    <span class="at">edaphic_vegan =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"vegan"</span></span>
<span id="cb189-3772"><a href="#cb189-3772" aria-hidden="true" tabindex="-1"></a>                           ),</span>
<span id="cb189-3773"><a href="#cb189-3773" aria-hidden="true" tabindex="-1"></a>    <span class="at">soil_funct_ape =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"ape"</span></span>
<span id="cb189-3774"><a href="#cb189-3774" aria-hidden="true" tabindex="-1"></a>                       ),</span>
<span id="cb189-3775"><a href="#cb189-3775" aria-hidden="true" tabindex="-1"></a>    <span class="at">soil_funct_vegan =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"vegan"</span></span>
<span id="cb189-3776"><a href="#cb189-3776" aria-hidden="true" tabindex="-1"></a>                              ),</span>
<span id="cb189-3777"><a href="#cb189-3777" aria-hidden="true" tabindex="-1"></a>    <span class="at">temp_adapt_ape =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"ape"</span></span>
<span id="cb189-3778"><a href="#cb189-3778" aria-hidden="true" tabindex="-1"></a>                       ),</span>
<span id="cb189-3779"><a href="#cb189-3779" aria-hidden="true" tabindex="-1"></a>    <span class="at">temp_adapt_vegan =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"vegan"</span></span>
<span id="cb189-3780"><a href="#cb189-3780" aria-hidden="true" tabindex="-1"></a>                              )</span>
<span id="cb189-3781"><a href="#cb189-3781" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb189-3782"><a href="#cb189-3782" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb189-3783"><a href="#cb189-3783" aria-hidden="true" tabindex="-1"></a>    <span class="at">columnGroups =</span> <span class="fu">list</span>(</span>
<span id="cb189-3784"><a href="#cb189-3784" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colGroup</span>(<span class="at">name =</span> <span class="st">"Environmental &amp; edaphic"</span>, </span>
<span id="cb189-3785"><a href="#cb189-3785" aria-hidden="true" tabindex="-1"></a>             <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"edaphic_ape"</span>, <span class="st">"edaphic_vegan"</span>), </span>
<span id="cb189-3786"><a href="#cb189-3786" aria-hidden="true" tabindex="-1"></a>             <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"1.2em"</span></span>
<span id="cb189-3787"><a href="#cb189-3787" aria-hidden="true" tabindex="-1"></a>                                ),</span>
<span id="cb189-3788"><a href="#cb189-3788" aria-hidden="true" tabindex="-1"></a>             ),</span>
<span id="cb189-3789"><a href="#cb189-3789" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colGroup</span>(<span class="at">name =</span> <span class="st">"Functional responses"</span>, </span>
<span id="cb189-3790"><a href="#cb189-3790" aria-hidden="true" tabindex="-1"></a>             <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"soil_funct_ape"</span>, <span class="st">"soil_funct_vegan"</span>), </span>
<span id="cb189-3791"><a href="#cb189-3791" aria-hidden="true" tabindex="-1"></a>             <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"1.2em"</span></span>
<span id="cb189-3792"><a href="#cb189-3792" aria-hidden="true" tabindex="-1"></a>                                )</span>
<span id="cb189-3793"><a href="#cb189-3793" aria-hidden="true" tabindex="-1"></a>             ),</span>
<span id="cb189-3794"><a href="#cb189-3794" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colGroup</span>(<span class="at">name =</span> <span class="st">"Temperature adaptation"</span>, </span>
<span id="cb189-3795"><a href="#cb189-3795" aria-hidden="true" tabindex="-1"></a>             <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"temp_adapt_ape"</span>, <span class="st">"temp_adapt_vegan"</span>), </span>
<span id="cb189-3796"><a href="#cb189-3796" aria-hidden="true" tabindex="-1"></a>             <span class="at">headerStyle =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"1.2em"</span>)</span>
<span id="cb189-3797"><a href="#cb189-3797" aria-hidden="true" tabindex="-1"></a>             )</span>
<span id="cb189-3798"><a href="#cb189-3798" aria-hidden="true" tabindex="-1"></a>      ),</span>
<span id="cb189-3799"><a href="#cb189-3799" aria-hidden="true" tabindex="-1"></a>  <span class="at">searchable =</span> <span class="cn">FALSE</span>, <span class="at">defaultPageSize =</span> <span class="dv">15</span>, <span class="at">showPagination =</span> <span class="cn">FALSE</span>,</span>
<span id="cb189-3800"><a href="#cb189-3800" aria-hidden="true" tabindex="-1"></a>  <span class="at">pageSizeOptions =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fu">nrow</span>(seq_table)), </span>
<span id="cb189-3801"><a href="#cb189-3801" aria-hidden="true" tabindex="-1"></a>  <span class="at">showPageSizeOptions =</span> <span class="cn">TRUE</span>, <span class="at">highlight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb189-3802"><a href="#cb189-3802" aria-hidden="true" tabindex="-1"></a>  <span class="at">bordered =</span> <span class="cn">TRUE</span>, <span class="at">striped =</span> <span class="cn">TRUE</span>, <span class="at">compact =</span> <span class="cn">TRUE</span>, </span>
<span id="cb189-3803"><a href="#cb189-3803" aria-hidden="true" tabindex="-1"></a>  <span class="at">wrap =</span> <span class="cn">FALSE</span>, <span class="at">showSortable =</span> <span class="cn">FALSE</span>, <span class="at">fullWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb189-3804"><a href="#cb189-3804" aria-hidden="true" tabindex="-1"></a>  <span class="at">theme =</span> <span class="fu">reactableTheme</span>(<span class="at">style =</span> <span class="fu">list</span>(<span class="at">fontSize =</span> <span class="st">"0.8em"</span>)))</span>
<span id="cb189-3805"><a href="#cb189-3805" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(seq_table)</span>
<span id="cb189-3806"><a href="#cb189-3806" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3807"><a href="#cb189-3807" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3808"><a href="#cb189-3808" aria-hidden="true" tabindex="-1"></a>Moving on.</span>
<span id="cb189-3809"><a href="#cb189-3809" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3810"><a href="#cb189-3810" aria-hidden="true" tabindex="-1"></a><span class="fu">## Best Subset of Variables</span></span>
<span id="cb189-3811"><a href="#cb189-3811" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3812"><a href="#cb189-3812" aria-hidden="true" tabindex="-1"></a>Now we want to know which of the metadata parameters are the most strongly correlated with the community data. For this we use the <span class="in">`bioenv`</span> function from the <span class="in">`vegan`</span> package. <span class="in">`bioenv`</span>---*Best Subset of Environmental Variables with Maximum (Rank) Correlation with Community Dissimilarities*---finds the best subset of environmental variables, so that the Euclidean distances of scaled environmental variables have the maximum (rank) correlation with community dissimilarities. </span>
<span id="cb189-3813"><a href="#cb189-3813" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3814"><a href="#cb189-3814" aria-hidden="true" tabindex="-1"></a>Since we know that each of the Mantel tests we ran above are significant, here we will use the metadata set where autocorrelated parameters were removed and the remainder of the parameters were normalized (where applicable based on the Shapiro tests). </span>
<span id="cb189-3815"><a href="#cb189-3815" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3816"><a href="#cb189-3816" aria-hidden="true" tabindex="-1"></a>We run <span class="in">`bioenv`</span> against the three groups of metadata parameters. We then run <span class="in">`bioenv`</span> again, but this time against the individual parameters identified as significantly correlated.</span>
<span id="cb189-3817"><a href="#cb189-3817" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3818"><a href="#cb189-3818" aria-hidden="true" tabindex="-1"></a><span class="fu">### Edaphic Properties</span></span>
<span id="cb189-3819"><a href="#cb189-3819" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3822"><a href="#cb189-3822" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3823"><a href="#cb189-3823" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-3824"><a href="#cb189-3824" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(its18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb189-3825"><a href="#cb189-3825" aria-hidden="true" tabindex="-1"></a>tmp_env <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(its18_select_mc_norm_split_no_ac<span class="sc">$</span>edaphic)</span>
<span id="cb189-3826"><a href="#cb189-3826" aria-hidden="true" tabindex="-1"></a>tmp_env[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-3827"><a href="#cb189-3827" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3828"><a href="#cb189-3828" aria-hidden="true" tabindex="-1"></a>edaphic_bioenv <span class="ot">&lt;-</span> <span class="fu">bioenv</span>(<span class="fu">wisconsin</span>(tmp_comm), tmp_env, </span>
<span id="cb189-3829"><a href="#cb189-3829" aria-hidden="true" tabindex="-1"></a>                         <span class="at">method =</span> <span class="st">"spearman"</span>, <span class="at">index =</span> <span class="st">"bray"</span>, </span>
<span id="cb189-3830"><a href="#cb189-3830" aria-hidden="true" tabindex="-1"></a>                         <span class="at">upto =</span> <span class="fu">ncol</span>(tmp_env), <span class="at">metric =</span> <span class="st">"euclidean"</span>)</span>
<span id="cb189-3831"><a href="#cb189-3831" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3832"><a href="#cb189-3832" aria-hidden="true" tabindex="-1"></a>bioenv_list <span class="ot">&lt;-</span> edaphic_bioenv<span class="sc">$</span>models[[edaphic_bioenv<span class="sc">$</span>whichbest]]<span class="sc">$</span>best</span>
<span id="cb189-3833"><a href="#cb189-3833" aria-hidden="true" tabindex="-1"></a>bioenv_best <span class="ot">&lt;-</span> <span class="fu">bioenvdist</span>(edaphic_bioenv, <span class="at">which =</span> <span class="st">"best"</span>)</span>
<span id="cb189-3834"><a href="#cb189-3834" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3835"><a href="#cb189-3835" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> bioenv_list) {</span>
<span id="cb189-3836"><a href="#cb189-3836" aria-hidden="true" tabindex="-1"></a>       tmp_dp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(edaphic_bioenv<span class="sc">$</span>x)</span>
<span id="cb189-3837"><a href="#cb189-3837" aria-hidden="true" tabindex="-1"></a>       tmp_md <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_dp[[i]], </span>
<span id="cb189-3838"><a href="#cb189-3838" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">"euclidean"</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb189-3839"><a href="#cb189-3839" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb189-3840"><a href="#cb189-3840" aria-hidden="true" tabindex="-1"></a>       tmp_man <span class="ot">&lt;-</span> <span class="fu">mantel</span>(bioenv_best, tmp_md, </span>
<span id="cb189-3841"><a href="#cb189-3841" aria-hidden="true" tabindex="-1"></a>                         <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb189-3842"><a href="#cb189-3842" aria-hidden="true" tabindex="-1"></a>       tmp_md_name <span class="ot">&lt;-</span> <span class="fu">names</span>(tmp_dp)[[i]]</span>
<span id="cb189-3843"><a href="#cb189-3843" aria-hidden="true" tabindex="-1"></a>       tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(tmp_md_name, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_bioenv_mantel_test"</span>))</span>
<span id="cb189-3844"><a href="#cb189-3844" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_name, tmp_man)</span>
<span id="cb189-3845"><a href="#cb189-3845" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-3846"><a href="#cb189-3846" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-3847"><a href="#cb189-3847" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">"_bioenv_mantel_test"</span>)</span>
<span id="cb189-3848"><a href="#cb189-3848" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3849"><a href="#cb189-3849" aria-hidden="true" tabindex="-1"></a>edaphic_bioenv_ind_mantel <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">AST =</span> AST_bioenv_mantel_test)</span>
<span id="cb189-3850"><a href="#cb189-3850" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"_bioenv_mantel_test"</span>))</span>
<span id="cb189-3851"><a href="#cb189-3851" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3852"><a href="#cb189-3852" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3855"><a href="#cb189-3855" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3856"><a href="#cb189-3856" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3857"><a href="#cb189-3857" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-3858"><a href="#cb189-3858" aria-hidden="true" tabindex="-1"></a>edaphic_bioenv</span>
<span id="cb189-3859"><a href="#cb189-3859" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3860"><a href="#cb189-3860" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3861"><a href="#cb189-3861" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb189-3862"><a href="#cb189-3862" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show the results of individual edaphic metadata Mantel tests<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb189-3863"><a href="#cb189-3863" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3866"><a href="#cb189-3866" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3867"><a href="#cb189-3867" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3868"><a href="#cb189-3868" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-3869"><a href="#cb189-3869" aria-hidden="true" tabindex="-1"></a>edaphic_bioenv_ind_mantel</span>
<span id="cb189-3870"><a href="#cb189-3870" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3871"><a href="#cb189-3871" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb189-3872"><a href="#cb189-3872" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3873"><a href="#cb189-3873" aria-hidden="true" tabindex="-1"></a><span class="in">`bioenv`</span> found the following edaphic properties significantly correlated with the community data: </span>
<span id="cb189-3874"><a href="#cb189-3874" aria-hidden="true" tabindex="-1"></a>**`r row.names(summary(edaphic_bioenv_ind_mantel))`**</span>
<span id="cb189-3875"><a href="#cb189-3875" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3876"><a href="#cb189-3876" aria-hidden="true" tabindex="-1"></a><span class="fu">### Soil Functional Response</span></span>
<span id="cb189-3877"><a href="#cb189-3877" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3880"><a href="#cb189-3880" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3881"><a href="#cb189-3881" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-3882"><a href="#cb189-3882" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(its18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb189-3883"><a href="#cb189-3883" aria-hidden="true" tabindex="-1"></a>tmp_env <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(its18_select_mc_norm_split_no_ac<span class="sc">$</span>soil_funct)</span>
<span id="cb189-3884"><a href="#cb189-3884" aria-hidden="true" tabindex="-1"></a>tmp_env[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-3885"><a href="#cb189-3885" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3886"><a href="#cb189-3886" aria-hidden="true" tabindex="-1"></a>soil_funct_bioenv <span class="ot">&lt;-</span> <span class="fu">bioenv</span>(<span class="fu">wisconsin</span>(tmp_comm), tmp_env, </span>
<span id="cb189-3887"><a href="#cb189-3887" aria-hidden="true" tabindex="-1"></a>                         <span class="at">method =</span> <span class="st">"spearman"</span>, <span class="at">index =</span> <span class="st">"bray"</span>, </span>
<span id="cb189-3888"><a href="#cb189-3888" aria-hidden="true" tabindex="-1"></a>                         <span class="at">upto =</span> <span class="fu">ncol</span>(tmp_env), <span class="at">metric =</span> <span class="st">"euclidean"</span>)</span>
<span id="cb189-3889"><a href="#cb189-3889" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3890"><a href="#cb189-3890" aria-hidden="true" tabindex="-1"></a>bioenv_list <span class="ot">&lt;-</span> soil_funct_bioenv<span class="sc">$</span>models[[soil_funct_bioenv<span class="sc">$</span>whichbest]]<span class="sc">$</span>best</span>
<span id="cb189-3891"><a href="#cb189-3891" aria-hidden="true" tabindex="-1"></a>bioenv_best <span class="ot">&lt;-</span> <span class="fu">bioenvdist</span>(soil_funct_bioenv, <span class="at">which =</span> <span class="st">"best"</span>)</span>
<span id="cb189-3892"><a href="#cb189-3892" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3893"><a href="#cb189-3893" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> bioenv_list) {</span>
<span id="cb189-3894"><a href="#cb189-3894" aria-hidden="true" tabindex="-1"></a>       tmp_dp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(soil_funct_bioenv<span class="sc">$</span>x)</span>
<span id="cb189-3895"><a href="#cb189-3895" aria-hidden="true" tabindex="-1"></a>       tmp_md <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_dp[[i]], </span>
<span id="cb189-3896"><a href="#cb189-3896" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">"euclidean"</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb189-3897"><a href="#cb189-3897" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb189-3898"><a href="#cb189-3898" aria-hidden="true" tabindex="-1"></a>       tmp_man <span class="ot">&lt;-</span> <span class="fu">mantel</span>(bioenv_best, tmp_md, </span>
<span id="cb189-3899"><a href="#cb189-3899" aria-hidden="true" tabindex="-1"></a>                         <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb189-3900"><a href="#cb189-3900" aria-hidden="true" tabindex="-1"></a>       tmp_md_name <span class="ot">&lt;-</span> <span class="fu">names</span>(tmp_dp)[[i]]</span>
<span id="cb189-3901"><a href="#cb189-3901" aria-hidden="true" tabindex="-1"></a>       tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(tmp_md_name, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_bioenv_mantel_test"</span>))</span>
<span id="cb189-3902"><a href="#cb189-3902" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_name, tmp_man)</span>
<span id="cb189-3903"><a href="#cb189-3903" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-3904"><a href="#cb189-3904" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-3905"><a href="#cb189-3905" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">"_bioenv_mantel_test"</span>)</span>
<span id="cb189-3906"><a href="#cb189-3906" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3907"><a href="#cb189-3907" aria-hidden="true" tabindex="-1"></a>soil_funct_bioenv_ind_mantel <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">enzNP =</span> enzNP_bioenv_mantel_test, </span>
<span id="cb189-3908"><a href="#cb189-3908" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">PX_ase =</span> PX_ase_bioenv_mantel_test,</span>
<span id="cb189-3909"><a href="#cb189-3909" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">XY_ase =</span> XY_ase_bioenv_mantel_test)</span>
<span id="cb189-3910"><a href="#cb189-3910" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"_bioenv_mantel_test"</span>))</span>
<span id="cb189-3911"><a href="#cb189-3911" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3912"><a href="#cb189-3912" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3915"><a href="#cb189-3915" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3916"><a href="#cb189-3916" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3917"><a href="#cb189-3917" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-3918"><a href="#cb189-3918" aria-hidden="true" tabindex="-1"></a>soil_funct_bioenv</span>
<span id="cb189-3919"><a href="#cb189-3919" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3920"><a href="#cb189-3920" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3921"><a href="#cb189-3921" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb189-3922"><a href="#cb189-3922" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show the results of individual functional response metadata Mantel tests<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb189-3923"><a href="#cb189-3923" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3926"><a href="#cb189-3926" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3927"><a href="#cb189-3927" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3928"><a href="#cb189-3928" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-3929"><a href="#cb189-3929" aria-hidden="true" tabindex="-1"></a>soil_funct_bioenv_ind_mantel</span>
<span id="cb189-3930"><a href="#cb189-3930" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3931"><a href="#cb189-3931" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb189-3932"><a href="#cb189-3932" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3933"><a href="#cb189-3933" aria-hidden="true" tabindex="-1"></a><span class="in">`bioenv`</span> found the following soil functions  significantly correlated with the community data: </span>
<span id="cb189-3934"><a href="#cb189-3934" aria-hidden="true" tabindex="-1"></a>**`r row.names(summary(soil_funct_bioenv_ind_mantel))`**</span>
<span id="cb189-3935"><a href="#cb189-3935" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3936"><a href="#cb189-3936" aria-hidden="true" tabindex="-1"></a><span class="fu">### Temperature Adaptation</span></span>
<span id="cb189-3937"><a href="#cb189-3937" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3940"><a href="#cb189-3940" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3941"><a href="#cb189-3941" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-3942"><a href="#cb189-3942" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(its18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb189-3943"><a href="#cb189-3943" aria-hidden="true" tabindex="-1"></a>tmp_env <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(its18_select_mc_norm_split_no_ac<span class="sc">$</span>temp_adapt)</span>
<span id="cb189-3944"><a href="#cb189-3944" aria-hidden="true" tabindex="-1"></a>tmp_env[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-3945"><a href="#cb189-3945" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3946"><a href="#cb189-3946" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioenv <span class="ot">&lt;-</span> <span class="fu">bioenv</span>(<span class="fu">wisconsin</span>(tmp_comm), tmp_env, </span>
<span id="cb189-3947"><a href="#cb189-3947" aria-hidden="true" tabindex="-1"></a>                         <span class="at">method =</span> <span class="st">"spearman"</span>, <span class="at">index =</span> <span class="st">"bray"</span>, </span>
<span id="cb189-3948"><a href="#cb189-3948" aria-hidden="true" tabindex="-1"></a>                         <span class="at">upto =</span> <span class="fu">ncol</span>(tmp_env), <span class="at">metric =</span> <span class="st">"euclidean"</span>)</span>
<span id="cb189-3949"><a href="#cb189-3949" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3950"><a href="#cb189-3950" aria-hidden="true" tabindex="-1"></a>bioenv_list <span class="ot">&lt;-</span> temp_adapt_bioenv<span class="sc">$</span>models[[temp_adapt_bioenv<span class="sc">$</span>whichbest]]<span class="sc">$</span>best</span>
<span id="cb189-3951"><a href="#cb189-3951" aria-hidden="true" tabindex="-1"></a>bioenv_best <span class="ot">&lt;-</span> <span class="fu">bioenvdist</span>(temp_adapt_bioenv, <span class="at">which =</span> <span class="st">"best"</span>)</span>
<span id="cb189-3952"><a href="#cb189-3952" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3953"><a href="#cb189-3953" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> bioenv_list) {</span>
<span id="cb189-3954"><a href="#cb189-3954" aria-hidden="true" tabindex="-1"></a>       tmp_dp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(temp_adapt_bioenv<span class="sc">$</span>x)</span>
<span id="cb189-3955"><a href="#cb189-3955" aria-hidden="true" tabindex="-1"></a>       tmp_md <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">vegdist</span>(tmp_dp[[i]], </span>
<span id="cb189-3956"><a href="#cb189-3956" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">"euclidean"</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>, </span>
<span id="cb189-3957"><a href="#cb189-3957" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>))</span>
<span id="cb189-3958"><a href="#cb189-3958" aria-hidden="true" tabindex="-1"></a>       tmp_man <span class="ot">&lt;-</span> <span class="fu">mantel</span>(bioenv_best, tmp_md, </span>
<span id="cb189-3959"><a href="#cb189-3959" aria-hidden="true" tabindex="-1"></a>                         <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb189-3960"><a href="#cb189-3960" aria-hidden="true" tabindex="-1"></a>       tmp_md_name <span class="ot">&lt;-</span> <span class="fu">names</span>(tmp_dp)[[i]]</span>
<span id="cb189-3961"><a href="#cb189-3961" aria-hidden="true" tabindex="-1"></a>       tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(tmp_md_name, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_bioenv_mantel_test"</span>))</span>
<span id="cb189-3962"><a href="#cb189-3962" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_name, tmp_man)</span>
<span id="cb189-3963"><a href="#cb189-3963" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-3964"><a href="#cb189-3964" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-3965"><a href="#cb189-3965" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">"_bioenv_mantel_test"</span>)</span>
<span id="cb189-3966"><a href="#cb189-3966" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3967"><a href="#cb189-3967" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioenv_ind_mantel <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">XY_Q10 =</span> XY_Q10_bioenv_mantel_test,</span>
<span id="cb189-3968"><a href="#cb189-3968" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">Tmin =</span> Tmin_bioenv_mantel_test)</span>
<span id="cb189-3969"><a href="#cb189-3969" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"_bioenv_mantel_test"</span>))</span>
<span id="cb189-3970"><a href="#cb189-3970" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3971"><a href="#cb189-3971" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3974"><a href="#cb189-3974" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3975"><a href="#cb189-3975" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3976"><a href="#cb189-3976" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-3977"><a href="#cb189-3977" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioenv</span>
<span id="cb189-3978"><a href="#cb189-3978" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3979"><a href="#cb189-3979" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3980"><a href="#cb189-3980" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb189-3981"><a href="#cb189-3981" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show the results of individual temperature adaptation metadata Mantel tests<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb189-3982"><a href="#cb189-3982" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3985"><a href="#cb189-3985" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-3986"><a href="#cb189-3986" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-3987"><a href="#cb189-3987" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-3988"><a href="#cb189-3988" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioenv_ind_mantel</span>
<span id="cb189-3989"><a href="#cb189-3989" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-3990"><a href="#cb189-3990" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb189-3991"><a href="#cb189-3991" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3992"><a href="#cb189-3992" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3993"><a href="#cb189-3993" aria-hidden="true" tabindex="-1"></a><span class="in">`bioenv`</span> found the following temperature adaptations significantly correlated with the community data: </span>
<span id="cb189-3994"><a href="#cb189-3994" aria-hidden="true" tabindex="-1"></a>**`r row.names(summary(temp_adapt_bioenv_ind_mantel))`**</span>
<span id="cb189-3995"><a href="#cb189-3995" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3996"><a href="#cb189-3996" aria-hidden="true" tabindex="-1"></a><span class="fu">## Distance-based Redundancy</span></span>
<span id="cb189-3997"><a href="#cb189-3997" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-3998"><a href="#cb189-3998" aria-hidden="true" tabindex="-1"></a>Now we turn our attention to distance-based redundancy analysis (dbRDA), an ordination method similar to Redundancy Analysis (rda) but it allows non-Euclidean dissimilarity indices, such as Manhattan or BrayCurtis distance. For this, we use <span class="in">`capscale`</span> from the <span class="in">`vegan`</span> package. <span class="in">`capscale`</span>  is a constrained versions of metric scaling (principal coordinates analysis), which are based on the Euclidean distance but can be used, and are more useful, with other dissimilarity measures. The functions can also perform unconstrained principal coordinates analysis, optionally using extended dissimilarities.</span>
<span id="cb189-3999"><a href="#cb189-3999" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4000"><a href="#cb189-4000" aria-hidden="true" tabindex="-1"></a>For each of the three metadata subsets, we perform the following steps:</span>
<span id="cb189-4001"><a href="#cb189-4001" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4002"><a href="#cb189-4002" aria-hidden="true" tabindex="-1"></a>1) Run <span class="in">`rankindex`</span> to compare metadata and community dissimilarity indices for gradient detection. This will help us select the best dissimilarity metric to use.</span>
<span id="cb189-4003"><a href="#cb189-4003" aria-hidden="true" tabindex="-1"></a>2) Run <span class="in">`capscale`</span> for distance-based redundancy analysis.</span>
<span id="cb189-4004"><a href="#cb189-4004" aria-hidden="true" tabindex="-1"></a>3) Run <span class="in">`envfit`</span> to fit environmental parameters onto the ordination. This function basically calculates correlation scores between the metadata parameters and the ordination axes. </span>
<span id="cb189-4005"><a href="#cb189-4005" aria-hidden="true" tabindex="-1"></a>4) Select metadata parameters significant for <span class="in">`bioenv`</span> (see above) and/or <span class="in">`envfit`</span> analyses.</span>
<span id="cb189-4006"><a href="#cb189-4006" aria-hidden="true" tabindex="-1"></a>5) Run <span class="in">`envfit`</span> on ASVs.</span>
<span id="cb189-4007"><a href="#cb189-4007" aria-hidden="true" tabindex="-1"></a>6) Plot the ordination and vector overlays. </span>
<span id="cb189-4008"><a href="#cb189-4008" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4009"><a href="#cb189-4009" aria-hidden="true" tabindex="-1"></a><span class="fu">### Edaphic Properties</span></span>
<span id="cb189-4010"><a href="#cb189-4010" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4013"><a href="#cb189-4013" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4014"><a href="#cb189-4014" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4015"><a href="#cb189-4015" aria-hidden="true" tabindex="-1"></a>tmp_md <span class="ot">&lt;-</span> its18_select_mc_norm_split_no_ac<span class="sc">$</span>edaphic</span>
<span id="cb189-4016"><a href="#cb189-4016" aria-hidden="true" tabindex="-1"></a>tmp_md<span class="sc">$</span>TREAT_T <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tmp_md<span class="sc">$</span>TREAT_T)</span>
<span id="cb189-4017"><a href="#cb189-4017" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(its18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb189-4018"><a href="#cb189-4018" aria-hidden="true" tabindex="-1"></a>edaphic_rank <span class="ot">&lt;-</span> <span class="fu">rankindex</span>(tmp_md[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_md)], tmp_comm, </span>
<span id="cb189-4019"><a href="#cb189-4019" aria-hidden="true" tabindex="-1"></a>          <span class="at">indices =</span> <span class="fu">c</span>(<span class="st">"euc"</span>, <span class="st">"man"</span>, <span class="st">"gow"</span>,<span class="st">"bra"</span>, <span class="st">"kul"</span>), </span>
<span id="cb189-4020"><a href="#cb189-4020" aria-hidden="true" tabindex="-1"></a>          <span class="at">stepacross =</span> <span class="cn">FALSE</span>, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb189-4021"><a href="#cb189-4021" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4022"><a href="#cb189-4022" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4025"><a href="#cb189-4025" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4026"><a href="#cb189-4026" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4027"><a href="#cb189-4027" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4028"><a href="#cb189-4028" aria-hidden="true" tabindex="-1"></a>edaphic_rank</span>
<span id="cb189-4029"><a href="#cb189-4029" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4030"><a href="#cb189-4030" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4031"><a href="#cb189-4031" aria-hidden="true" tabindex="-1"></a>Let's run <span class="in">`capscale`</span> using Bray-Curtis. Note, we have <span class="in">`r ncol(its18_select_mc_norm_split_no_ac$edaphic) - 8`</span> metadata parameters in this group but, for some reason, <span class="in">`capscale`</span> only works with 13 parameters. This may have to do with degrees of freedom? </span>
<span id="cb189-4032"><a href="#cb189-4032" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4033"><a href="#cb189-4033" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Starting properties: AST, H2O, N, P, Al, Ca, Fe, K, Mg, Mn, Na, TEB, ECEC, pH, NH4, NO3, PO4, DOC, DON, DOCN.</span>
<span id="cb189-4034"><a href="#cb189-4034" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Autocorrelated removed: TEB, DON, Na, Al, Ca.</span>
<span id="cb189-4035"><a href="#cb189-4035" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Remove for capscale: Mg, Mn, Na, Al, Fe, K</span>
<span id="cb189-4036"><a href="#cb189-4036" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4039"><a href="#cb189-4039" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4040"><a href="#cb189-4040" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4041"><a href="#cb189-4041" aria-hidden="true" tabindex="-1"></a><span class="co"># All edaphic: AST, H2O, N, P, Al, Ca, Fe, K, Mg, Mn, Na, TEB, ECEC, pH, NH4, NO3, PO4, DOC, DON, DOCN</span></span>
<span id="cb189-4042"><a href="#cb189-4042" aria-hidden="true" tabindex="-1"></a><span class="co"># REMOVED edaphic: TEB, DON, Na, Al, Ca</span></span>
<span id="cb189-4043"><a href="#cb189-4043" aria-hidden="true" tabindex="-1"></a><span class="co"># Remaining edaphic: AST, H2O, N, P, Fe, K, Mg, Mn, ECEC, pH, NH4, NO3, PO4, DOC, DOCN</span></span>
<span id="cb189-4044"><a href="#cb189-4044" aria-hidden="true" tabindex="-1"></a><span class="co"># only works with 11</span></span>
<span id="cb189-4045"><a href="#cb189-4045" aria-hidden="true" tabindex="-1"></a><span class="co"># Removed  Mg + Mn + Na + Al + Fe + K</span></span>
<span id="cb189-4046"><a href="#cb189-4046" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TEST</span><span class="co"> with , metaMDSdist = TRUE AND/OR , metaMDS = TRUE</span></span>
<span id="cb189-4047"><a href="#cb189-4047" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4048"><a href="#cb189-4048" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4051"><a href="#cb189-4051" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4052"><a href="#cb189-4052" aria-hidden="true" tabindex="-1"></a>edaphic_cap <span class="ot">&lt;-</span> <span class="fu">capscale</span>(tmp_comm <span class="sc">~</span> AST <span class="sc">+</span> H2O <span class="sc">+</span> N <span class="sc">+</span> P <span class="sc">+</span> ECEC <span class="sc">+</span> pH <span class="sc">+</span> </span>
<span id="cb189-4053"><a href="#cb189-4053" aria-hidden="true" tabindex="-1"></a>                                   NH4 <span class="sc">+</span> NO3 <span class="sc">+</span> PO4 <span class="sc">+</span> DOC <span class="sc">+</span> DOCN, </span>
<span id="cb189-4054"><a href="#cb189-4054" aria-hidden="true" tabindex="-1"></a>                        tmp_md, <span class="at">dist =</span> <span class="st">"bray"</span>)</span>
<span id="cb189-4055"><a href="#cb189-4055" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tmp_md)</span>
<span id="cb189-4056"><a href="#cb189-4056" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4057"><a href="#cb189-4057" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4060"><a href="#cb189-4060" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4061"><a href="#cb189-4061" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4062"><a href="#cb189-4062" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4063"><a href="#cb189-4063" aria-hidden="true" tabindex="-1"></a>edaphic_cap</span>
<span id="cb189-4064"><a href="#cb189-4064" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4065"><a href="#cb189-4065" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4066"><a href="#cb189-4066" aria-hidden="true" tabindex="-1"></a>Now we can look at the variance against each principal component. </span>
<span id="cb189-4067"><a href="#cb189-4067" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4070"><a href="#cb189-4070" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4071"><a href="#cb189-4071" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4072"><a href="#cb189-4072" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4073"><a href="#cb189-4073" aria-hidden="true" tabindex="-1"></a>stats<span class="sc">::</span><span class="fu">screeplot</span>(edaphic_cap)</span>
<span id="cb189-4074"><a href="#cb189-4074" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4075"><a href="#cb189-4075" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4076"><a href="#cb189-4076" aria-hidden="true" tabindex="-1"></a>And then make some quick and dirty plots. This will also come in handy later when we need to parse out data a better plot visualization. The <span class="in">`ggplot`</span> function <span class="in">`autoplot`</span> stores these data in a more accessible way than the raw results from <span class="in">`capscale`</span></span>
<span id="cb189-4077"><a href="#cb189-4077" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4080"><a href="#cb189-4080" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4081"><a href="#cb189-4081" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4082"><a href="#cb189-4082" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">plot</span>(edaphic_cap) </span>
<span id="cb189-4083"><a href="#cb189-4083" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt <span class="ot">&lt;-</span> ggplot2<span class="sc">::</span><span class="fu">autoplot</span>(edaphic_cap, <span class="at">arrows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb189-4084"><a href="#cb189-4084" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt</span>
<span id="cb189-4085"><a href="#cb189-4085" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4086"><a href="#cb189-4086" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4089"><a href="#cb189-4089" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4090"><a href="#cb189-4090" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4091"><a href="#cb189-4091" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(edaphic_cap) <span class="co"># overall test of the significant of the analysis</span></span>
<span id="cb189-4092"><a href="#cb189-4092" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(edaphic_cap, <span class="at">by =</span> <span class="st">"axis"</span>, <span class="at">perm.max =</span> <span class="dv">500</span>) <span class="co"># test axes for significance</span></span>
<span id="cb189-4093"><a href="#cb189-4093" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(edaphic_cap, <span class="at">by =</span> <span class="st">"terms"</span>, <span class="at">permu =</span> <span class="dv">500</span>) <span class="co"># test for sign. environ. variables</span></span>
<span id="cb189-4094"><a href="#cb189-4094" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4095"><a href="#cb189-4095" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4096"><a href="#cb189-4096" aria-hidden="true" tabindex="-1"></a>Next, we need to grab capscale scores for the samples and create a data frame of the first two dimensions. We will also need to add some of the sample details to the data frame. For this we use the vegan function <span class="in">`scores`</span> which gets species or site scores from the ordination.</span>
<span id="cb189-4097"><a href="#cb189-4097" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4100"><a href="#cb189-4100" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4101"><a href="#cb189-4101" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4102"><a href="#cb189-4102" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">"sites"</span>)</span>
<span id="cb189-4103"><a href="#cb189-4103" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-4104"><a href="#cb189-4104" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> tmp_samp_scores <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">SampleID =</span> Label)</span>
<span id="cb189-4105"><a href="#cb189-4105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4106"><a href="#cb189-4106" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb189-4107"><a href="#cb189-4107" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md_sub <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb189-4108"><a href="#cb189-4108" aria-hidden="true" tabindex="-1"></a>edaphic_plot_data <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_md_sub, tmp_samp_scores, <span class="at">by =</span> <span class="st">"SampleID"</span>)</span>
<span id="cb189-4109"><a href="#cb189-4109" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4110"><a href="#cb189-4110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4111"><a href="#cb189-4111" aria-hidden="true" tabindex="-1"></a>Now we have a new data frame that contains sample details and capscale values. </span>
<span id="cb189-4112"><a href="#cb189-4112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4115"><a href="#cb189-4115" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4116"><a href="#cb189-4116" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4117"><a href="#cb189-4117" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4118"><a href="#cb189-4118" aria-hidden="true" tabindex="-1"></a>edaphic_plot_data</span>
<span id="cb189-4119"><a href="#cb189-4119" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4120"><a href="#cb189-4120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4121"><a href="#cb189-4121" aria-hidden="true" tabindex="-1"></a>We can then do the same with the metadata vectors. Here though we only need the scores and parameter name. </span>
<span id="cb189-4122"><a href="#cb189-4122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4125"><a href="#cb189-4125" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4126"><a href="#cb189-4126" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4127"><a href="#cb189-4127" aria-hidden="true" tabindex="-1"></a>edaphic_md_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">"biplot"</span>)</span>
<span id="cb189-4128"><a href="#cb189-4128" aria-hidden="true" tabindex="-1"></a>edaphic_md_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-4129"><a href="#cb189-4129" aria-hidden="true" tabindex="-1"></a>edaphic_md_scores <span class="ot">&lt;-</span> edaphic_md_scores <span class="sc">%&gt;%</span> </span>
<span id="cb189-4130"><a href="#cb189-4130" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span></span>
<span id="cb189-4131"><a href="#cb189-4131" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"Label"</span>)</span>
<span id="cb189-4132"><a href="#cb189-4132" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4133"><a href="#cb189-4133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4134"><a href="#cb189-4134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4137"><a href="#cb189-4137" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4138"><a href="#cb189-4138" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4139"><a href="#cb189-4139" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4140"><a href="#cb189-4140" aria-hidden="true" tabindex="-1"></a>edaphic_md_scores</span>
<span id="cb189-4141"><a href="#cb189-4141" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4142"><a href="#cb189-4142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4143"><a href="#cb189-4143" aria-hidden="true" tabindex="-1"></a>Let's run some quick correlations of metadata with ordination axes to see which parameters are significant. For this we use the vegan function <span class="in">`envfit`</span>.</span>
<span id="cb189-4144"><a href="#cb189-4144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4147"><a href="#cb189-4147" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4148"><a href="#cb189-4148" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4149"><a href="#cb189-4149" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> edaphic_plot_data[, <span class="dv">6</span><span class="sc">:</span><span class="dv">7</span>]</span>
<span id="cb189-4150"><a href="#cb189-4150" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_samp_scores_sub)</span>
<span id="cb189-4151"><a href="#cb189-4151" aria-hidden="true" tabindex="-1"></a>tmp_param_list <span class="ot">&lt;-</span> edaphic_md_scores<span class="sc">$</span>parameters</span>
<span id="cb189-4152"><a href="#cb189-4152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4153"><a href="#cb189-4153" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> <span class="fu">subset</span>(tmp_md,  <span class="at">select =</span>  tmp_param_list)</span>
<span id="cb189-4154"><a href="#cb189-4154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4155"><a href="#cb189-4155" aria-hidden="true" tabindex="-1"></a>envfit_edaphic_md <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, tmp_md_sub,</span>
<span id="cb189-4156"><a href="#cb189-4156" aria-hidden="true" tabindex="-1"></a>                 <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb189-4157"><a href="#cb189-4157" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4158"><a href="#cb189-4158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4161"><a href="#cb189-4161" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4162"><a href="#cb189-4162" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4163"><a href="#cb189-4163" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4164"><a href="#cb189-4164" aria-hidden="true" tabindex="-1"></a>envfit_edaphic_md</span>
<span id="cb189-4165"><a href="#cb189-4165" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4166"><a href="#cb189-4166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4169"><a href="#cb189-4169" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4170"><a href="#cb189-4170" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4171"><a href="#cb189-4171" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_edaphic_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb189-4172"><a href="#cb189-4172" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_edaphic_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb189-4173"><a href="#cb189-4173" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_edaphic_md<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.4</span>))</span>
<span id="cb189-4174"><a href="#cb189-4174" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(edaphic_md_signif_hits)</span>
<span id="cb189-4175"><a href="#cb189-4175" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(edaphic_md_signif_hits)</span>
<span id="cb189-4176"><a href="#cb189-4176" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif <span class="ot">&lt;-</span> edaphic_md_scores[edaphic_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> edaphic_md_signif_hits,]</span>
<span id="cb189-4177"><a href="#cb189-4177" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif<span class="sc">$</span>parameters</span>
<span id="cb189-4178"><a href="#cb189-4178" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4179"><a href="#cb189-4179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4180"><a href="#cb189-4180" aria-hidden="true" tabindex="-1"></a><span class="in">`envfit`</span> found that <span class="in">`r edaphic_md_signif$parameters`</span> were significantly correlated. </span>
<span id="cb189-4181"><a href="#cb189-4181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4182"><a href="#cb189-4182" aria-hidden="true" tabindex="-1"></a>Now let's see if the same parameters are significant for the <span class="in">`envfit`</span> and <span class="in">`bioenv`</span> analyses.</span>
<span id="cb189-4183"><a href="#cb189-4183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4186"><a href="#cb189-4186" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4187"><a href="#cb189-4187" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4188"><a href="#cb189-4188" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4189"><a href="#cb189-4189" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hold</span></span>
<span id="cb189-4190"><a href="#cb189-4190" aria-hidden="true" tabindex="-1"></a><span class="co">#| comment: ''</span></span>
<span id="cb189-4191"><a href="#cb189-4191" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Significant parameters from bioenv analysis."</span>)</span>
<span id="cb189-4192"><a href="#cb189-4192" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(<span class="fu">summary</span>(edaphic_bioenv_ind_mantel))</span>
<span id="cb189-4193"><a href="#cb189-4193" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-4194"><a href="#cb189-4194" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-4195"><a href="#cb189-4195" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Significant parameters from envfit analysis."</span>)</span>
<span id="cb189-4196"><a href="#cb189-4196" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif<span class="sc">$</span>parameters</span>
<span id="cb189-4197"><a href="#cb189-4197" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-4198"><a href="#cb189-4198" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-4199"><a href="#cb189-4199" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Found in bioenv but not envfit."</span>)</span>
<span id="cb189-4200"><a href="#cb189-4200" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">setdiff</span>(<span class="fu">row.names</span>(<span class="fu">summary</span>(edaphic_bioenv_ind_mantel)), </span>
<span id="cb189-4201"><a href="#cb189-4201" aria-hidden="true" tabindex="-1"></a>                       edaphic_md_signif<span class="sc">$</span>parameters)</span>
<span id="cb189-4202"><a href="#cb189-4202" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-4203"><a href="#cb189-4203" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-4204"><a href="#cb189-4204" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Found in envfit but not bioenv."</span>)</span>
<span id="cb189-4205"><a href="#cb189-4205" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">setdiff</span>(edaphic_md_signif<span class="sc">$</span>parameters, </span>
<span id="cb189-4206"><a href="#cb189-4206" aria-hidden="true" tabindex="-1"></a>              <span class="fu">row.names</span>(<span class="fu">summary</span>(edaphic_bioenv_ind_mantel)))</span>
<span id="cb189-4207"><a href="#cb189-4207" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-4208"><a href="#cb189-4208" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-4209"><a href="#cb189-4209" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Found in envfit and bioenv."</span>)</span>
<span id="cb189-4210"><a href="#cb189-4210" aria-hidden="true" tabindex="-1"></a>edaphic_sig_diff <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">union</span>(edaphic_md_signif<span class="sc">$</span>parameters, </span>
<span id="cb189-4211"><a href="#cb189-4211" aria-hidden="true" tabindex="-1"></a>              <span class="fu">row.names</span>(<span class="fu">summary</span>(edaphic_bioenv_ind_mantel)))</span>
<span id="cb189-4212"><a href="#cb189-4212" aria-hidden="true" tabindex="-1"></a>edaphic_sig_diff</span>
<span id="cb189-4213"><a href="#cb189-4213" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4214"><a href="#cb189-4214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4217"><a href="#cb189-4217" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4218"><a href="#cb189-4218" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4219"><a href="#cb189-4219" aria-hidden="true" tabindex="-1"></a>new_edaphic_md_signif_hits <span class="ot">&lt;-</span> edaphic_sig_diff</span>
<span id="cb189-4220"><a href="#cb189-4220" aria-hidden="true" tabindex="-1"></a><span class="co">#new_edaphic_md_signif_hits &lt;- append(edaphic_md_signif_hits, edaphic_sig_diff)</span></span>
<span id="cb189-4221"><a href="#cb189-4221" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_all <span class="ot">&lt;-</span> edaphic_md_scores[edaphic_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> new_edaphic_md_signif_hits,]</span>
<span id="cb189-4222"><a href="#cb189-4222" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4223"><a href="#cb189-4223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4224"><a href="#cb189-4224" aria-hidden="true" tabindex="-1"></a>Check. Next, we run <span class="in">`envfit`</span> for the ASVs.</span>
<span id="cb189-4225"><a href="#cb189-4225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4228"><a href="#cb189-4228" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4229"><a href="#cb189-4229" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4230"><a href="#cb189-4230" aria-hidden="true" tabindex="-1"></a>envfit_edaphic_asv <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, </span>
<span id="cb189-4231"><a href="#cb189-4231" aria-hidden="true" tabindex="-1"></a>                             tmp_comm[, <span class="fu">order</span>(<span class="fu">colSums</span>(<span class="sc">-</span>tmp_comm))][, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],</span>
<span id="cb189-4232"><a href="#cb189-4232" aria-hidden="true" tabindex="-1"></a>                             <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb189-4233"><a href="#cb189-4233" aria-hidden="true" tabindex="-1"></a>edaphic_asv_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">"species"</span>)</span>
<span id="cb189-4234"><a href="#cb189-4234" aria-hidden="true" tabindex="-1"></a>edaphic_asv_scores <span class="ot">&lt;-</span> edaphic_asv_scores <span class="sc">%&gt;%</span> </span>
<span id="cb189-4235"><a href="#cb189-4235" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span></span>
<span id="cb189-4236"><a href="#cb189-4236" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"Label"</span>)</span>
<span id="cb189-4237"><a href="#cb189-4237" aria-hidden="true" tabindex="-1"></a>edaphic_asv_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-4238"><a href="#cb189-4238" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4239"><a href="#cb189-4239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4242"><a href="#cb189-4242" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4243"><a href="#cb189-4243" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4244"><a href="#cb189-4244" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4245"><a href="#cb189-4245" aria-hidden="true" tabindex="-1"></a>envfit_edaphic_asv</span>
<span id="cb189-4246"><a href="#cb189-4246" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4247"><a href="#cb189-4247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4250"><a href="#cb189-4250" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4251"><a href="#cb189-4251" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4252"><a href="#cb189-4252" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_edaphic_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb189-4253"><a href="#cb189-4253" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_edaphic_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb189-4254"><a href="#cb189-4254" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_edaphic_asv<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.5</span>))</span>
<span id="cb189-4255"><a href="#cb189-4255" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(edaphic_asv_signif_hits)</span>
<span id="cb189-4256"><a href="#cb189-4256" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(edaphic_asv_signif_hits)</span>
<span id="cb189-4257"><a href="#cb189-4257" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif <span class="ot">&lt;-</span> edaphic_asv_scores[edaphic_asv_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> edaphic_asv_signif_hits,]</span>
<span id="cb189-4258"><a href="#cb189-4258" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4259"><a href="#cb189-4259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4262"><a href="#cb189-4262" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4263"><a href="#cb189-4263" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4264"><a href="#cb189-4264" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4265"><a href="#cb189-4265" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif</span>
<span id="cb189-4266"><a href="#cb189-4266" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4267"><a href="#cb189-4267" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb189-4268"><a href="#cb189-4268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4271"><a href="#cb189-4271" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4272"><a href="#cb189-4272" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4273"><a href="#cb189-4273" aria-hidden="true" tabindex="-1"></a>edaphic_md_signif_all<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">"metadata"</span></span>
<span id="cb189-4274"><a href="#cb189-4274" aria-hidden="true" tabindex="-1"></a>edaphic_asv_signif<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">"ASV"</span></span>
<span id="cb189-4275"><a href="#cb189-4275" aria-hidden="true" tabindex="-1"></a>edaphic_bioplot_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(edaphic_md_signif_all, edaphic_asv_signif)</span>
<span id="cb189-4276"><a href="#cb189-4276" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4277"><a href="#cb189-4277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4278"><a href="#cb189-4278" aria-hidden="true" tabindex="-1"></a>The last thing to do is categorize parameters scores and ASV scores into different variable types for plotting.</span>
<span id="cb189-4279"><a href="#cb189-4279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4282"><a href="#cb189-4282" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4283"><a href="#cb189-4283" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4284"><a href="#cb189-4284" aria-hidden="true" tabindex="-1"></a>edaphic_bioplot_data_md <span class="ot">&lt;-</span> <span class="fu">subset</span>(edaphic_bioplot_data, </span>
<span id="cb189-4285"><a href="#cb189-4285" aria-hidden="true" tabindex="-1"></a>                                  edaphic_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">"metadata"</span>)</span>
<span id="cb189-4286"><a href="#cb189-4286" aria-hidden="true" tabindex="-1"></a>edaphic_bioplot_data_asv <span class="ot">&lt;-</span> <span class="fu">subset</span>(edaphic_bioplot_data, </span>
<span id="cb189-4287"><a href="#cb189-4287" aria-hidden="true" tabindex="-1"></a>                                   edaphic_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">"ASV"</span>)</span>
<span id="cb189-4288"><a href="#cb189-4288" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4289"><a href="#cb189-4289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4290"><a href="#cb189-4290" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb189-4291"><a href="#cb189-4291" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show  the code for the plot<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb189-4292"><a href="#cb189-4292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4295"><a href="#cb189-4295" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4296"><a href="#cb189-4296" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4297"><a href="#cb189-4297" aria-hidden="true" tabindex="-1"></a>edaphic_cap_vals <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(edaphic_cap<span class="sc">$</span>CCA<span class="sc">$</span>eig[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span>
<span id="cb189-4298"><a href="#cb189-4298" aria-hidden="true" tabindex="-1"></a>edaphic_cap1 <span class="ot">&lt;-</span> <span class="fu">signif</span>((edaphic_cap_vals[<span class="dv">1</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb189-4299"><a href="#cb189-4299" aria-hidden="true" tabindex="-1"></a>edaphic_cap2 <span class="ot">&lt;-</span> <span class="fu">signif</span>((edaphic_cap_vals[<span class="dv">2</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb189-4300"><a href="#cb189-4300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4301"><a href="#cb189-4301" aria-hidden="true" tabindex="-1"></a>cpa1_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"CAP1"</span>, <span class="st">" ("</span>, edaphic_cap1, <span class="st">"%)"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb189-4302"><a href="#cb189-4302" aria-hidden="true" tabindex="-1"></a>cpa2_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"CAP2"</span>, <span class="st">" ("</span>, edaphic_cap2, <span class="st">"%)"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb189-4303"><a href="#cb189-4303" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4304"><a href="#cb189-4304" aria-hidden="true" tabindex="-1"></a>swel_col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#2271B2"</span>, <span class="st">"#71B222"</span>, <span class="st">"#B22271"</span>)</span>
<span id="cb189-4305"><a href="#cb189-4305" aria-hidden="true" tabindex="-1"></a>edaphic_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(edaphic_plot_data) <span class="sc">+</span></span>
<span id="cb189-4306"><a href="#cb189-4306" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">shape =</span> TREAT,</span>
<span id="cb189-4307"><a href="#cb189-4307" aria-hidden="true" tabindex="-1"></a>                           <span class="at">colour =</span> TREAT_T), <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb189-4308"><a href="#cb189-4308" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> swel_col) <span class="sc">+</span></span>
<span id="cb189-4309"><a href="#cb189-4309" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_text(data = edaphic_plot_data, aes(x = CAP1, y = CAP2, #UNCOMMENT to add sample labels</span></span>
<span id="cb189-4310"><a href="#cb189-4310" aria-hidden="true" tabindex="-1"></a><span class="co">#                              label = SampleID), size = 3) + </span></span>
<span id="cb189-4311"><a href="#cb189-4311" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">xend =</span> CAP1, <span class="at">yend =</span> CAP2),</span>
<span id="cb189-4312"><a href="#cb189-4312" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> edaphic_bioplot_data_md, <span class="at">linetype =</span> <span class="st">"solid"</span>,</span>
<span id="cb189-4313"><a href="#cb189-4313" aria-hidden="true" tabindex="-1"></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">"cm"</span>)), <span class="at">size =</span> <span class="fl">0.4</span>, </span>
<span id="cb189-4314"><a href="#cb189-4314" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">"#191919"</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb189-4315"><a href="#cb189-4315" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> edaphic_bioplot_data_md, </span>
<span id="cb189-4316"><a href="#cb189-4316" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">label =</span> parameters), <span class="at">size =</span> <span class="dv">3</span>, </span>
<span id="cb189-4317"><a href="#cb189-4317" aria-hidden="true" tabindex="-1"></a>            <span class="at">nudge_x =</span> <span class="fl">0.1</span>, <span class="at">nudge_y =</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb189-4318"><a href="#cb189-4318" aria-hidden="true" tabindex="-1"></a>  <span class="do">## USe this code to overlay ASV vectors</span></span>
<span id="cb189-4319"><a href="#cb189-4319" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_segment(aes(x = 0, y = 0, xend = CAP1, yend = CAP2),</span></span>
<span id="cb189-4320"><a href="#cb189-4320" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             data = edaphic_bioplot_data_asv, linetype = "solid",</span></span>
<span id="cb189-4321"><a href="#cb189-4321" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             arrow = arrow(length = unit(0.3, "cm")),  size = 0.2, </span></span>
<span id="cb189-4322"><a href="#cb189-4322" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             color = "#676767") +</span></span>
<span id="cb189-4323"><a href="#cb189-4323" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text(data = edaphic_bioplot_data_asv, </span></span>
<span id="cb189-4324"><a href="#cb189-4324" aria-hidden="true" tabindex="-1"></a>  <span class="co">#          aes(x = CAP1, y = CAP2, label = parameters), size = 2.5,</span></span>
<span id="cb189-4325"><a href="#cb189-4325" aria-hidden="true" tabindex="-1"></a>  <span class="co">#          nudge_x = 0.1, nudge_y = 0.05) +</span></span>
<span id="cb189-4326"><a href="#cb189-4326" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span> </span>
<span id="cb189-4327"><a href="#cb189-4327" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Capscale Analysis"</span>,</span>
<span id="cb189-4328"><a href="#cb189-4328" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">"Edaphic properties"</span>, </span>
<span id="cb189-4329"><a href="#cb189-4329" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> cpa1_lab, </span>
<span id="cb189-4330"><a href="#cb189-4330" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> cpa2_lab)</span>
<span id="cb189-4331"><a href="#cb189-4331" aria-hidden="true" tabindex="-1"></a>edaphic_plot <span class="ot">&lt;-</span> edaphic_plot <span class="sc">+</span> <span class="fu">coord_fixed</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb189-4332"><a href="#cb189-4332" aria-hidden="true" tabindex="-1"></a>edaphic_plot</span>
<span id="cb189-4333"><a href="#cb189-4333" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">"files/metadata/figures/its18_edaphic_capscale.png"</span>,</span>
<span id="cb189-4334"><a href="#cb189-4334" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">16</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">units =</span> <span class="st">'cm'</span>, <span class="at">res =</span> <span class="dv">600</span>, <span class="at">bg =</span> <span class="st">"white"</span>)</span>
<span id="cb189-4335"><a href="#cb189-4335" aria-hidden="true" tabindex="-1"></a>edaphic_plot</span>
<span id="cb189-4336"><a href="#cb189-4336" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb189-4337"><a href="#cb189-4337" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"files/metadata/figures/its18_edaphic_capscale.pdf"</span>,</span>
<span id="cb189-4338"><a href="#cb189-4338" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">5</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb189-4339"><a href="#cb189-4339" aria-hidden="true" tabindex="-1"></a>edaphic_plot</span>
<span id="cb189-4340"><a href="#cb189-4340" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb189-4341"><a href="#cb189-4341" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4342"><a href="#cb189-4342" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb189-4343"><a href="#cb189-4343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4344"><a href="#cb189-4344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4347"><a href="#cb189-4347" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4348"><a href="#cb189-4348" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4349"><a href="#cb189-4349" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-4350"><a href="#cb189-4350" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4351"><a href="#cb189-4351" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4352"><a href="#cb189-4352" aria-hidden="true" tabindex="-1"></a><span class="fu">### Soil Functional Response</span></span>
<span id="cb189-4353"><a href="#cb189-4353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4356"><a href="#cb189-4356" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4357"><a href="#cb189-4357" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4358"><a href="#cb189-4358" aria-hidden="true" tabindex="-1"></a>tmp_md <span class="ot">&lt;-</span> its18_select_mc_norm_split_no_ac<span class="sc">$</span>soil_funct</span>
<span id="cb189-4359"><a href="#cb189-4359" aria-hidden="true" tabindex="-1"></a>tmp_md<span class="sc">$</span>TREAT_T <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tmp_md<span class="sc">$</span>TREAT_T)</span>
<span id="cb189-4360"><a href="#cb189-4360" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(its18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb189-4361"><a href="#cb189-4361" aria-hidden="true" tabindex="-1"></a>soil_funct_rank <span class="ot">&lt;-</span> <span class="fu">rankindex</span>(tmp_md[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_md)], tmp_comm, </span>
<span id="cb189-4362"><a href="#cb189-4362" aria-hidden="true" tabindex="-1"></a>          <span class="at">indices =</span> <span class="fu">c</span>(<span class="st">"euc"</span>, <span class="st">"man"</span>, <span class="st">"gow"</span>, <span class="st">"bra"</span>, <span class="st">"kul"</span>), </span>
<span id="cb189-4363"><a href="#cb189-4363" aria-hidden="true" tabindex="-1"></a>          <span class="at">stepacross =</span> <span class="cn">FALSE</span>, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb189-4364"><a href="#cb189-4364" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4365"><a href="#cb189-4365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4368"><a href="#cb189-4368" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4369"><a href="#cb189-4369" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4370"><a href="#cb189-4370" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4371"><a href="#cb189-4371" aria-hidden="true" tabindex="-1"></a>soil_funct_rank</span>
<span id="cb189-4372"><a href="#cb189-4372" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4373"><a href="#cb189-4373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4374"><a href="#cb189-4374" aria-hidden="true" tabindex="-1"></a>Let's run <span class="in">`capscale`</span> using Bray-Curtis Note, we have <span class="in">`r ncol(its18_select_mc_norm_split_no_ac$soil_funct) - 8`</span> metadata parameters in this group but, for some reason, <span class="in">`capscale`</span> only works with 13 parameters. This may have to do with degrees of freedom? </span>
<span id="cb189-4375"><a href="#cb189-4375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4376"><a href="#cb189-4376" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Starting properties: micC, micN, micP, micCN, micCP, micNP, AG_ase, BG_ase, BP_ase, CE_ase, P_ase, N_ase, S_ase, XY_ase,  LP_ase, PX_ase, CO2, enzCN, enzCP, enzNP</span>
<span id="cb189-4377"><a href="#cb189-4377" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Autocorrelated removed: micN, micNP, enzCN, enzCP, BP_ase, CE_ase, LP_ase, N_ase, P_ase</span>
<span id="cb189-4378"><a href="#cb189-4378" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Remove for capscale: NONE</span>
<span id="cb189-4379"><a href="#cb189-4379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4382"><a href="#cb189-4382" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4383"><a href="#cb189-4383" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4384"><a href="#cb189-4384" aria-hidden="true" tabindex="-1"></a><span class="co"># All soil prop: micC, micN, micP, micCN, micCP, micNP, AG_ase, BG_ase, BP_ase, CE_ase, P_ase, N_ase, S_ase, XY_ase,  LP_ase, PX_ase, CO2, enzCN, enzCP, enzNP</span></span>
<span id="cb189-4385"><a href="#cb189-4385" aria-hidden="true" tabindex="-1"></a><span class="co"># REMOVED soil prop: micN, micNP, enzCN, enzCP, BP_ase, CE_ase, LP_ase, N_ase, P_ase</span></span>
<span id="cb189-4386"><a href="#cb189-4386" aria-hidden="true" tabindex="-1"></a><span class="co"># Remaining soil prop: micC, micP, micCN, micCP, AG_ase, BG_ase, S_ase, XY_ase, PX_ase, CO2, enzNP</span></span>
<span id="cb189-4387"><a href="#cb189-4387" aria-hidden="true" tabindex="-1"></a><span class="co"># only works with 11</span></span>
<span id="cb189-4388"><a href="#cb189-4388" aria-hidden="true" tabindex="-1"></a><span class="co"># capscale Removed  NONE</span></span>
<span id="cb189-4389"><a href="#cb189-4389" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4390"><a href="#cb189-4390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4391"><a href="#cb189-4391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4394"><a href="#cb189-4394" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4395"><a href="#cb189-4395" aria-hidden="true" tabindex="-1"></a>soil_funct_cap <span class="ot">&lt;-</span> <span class="fu">capscale</span>(tmp_comm <span class="sc">~</span>  micC <span class="sc">+</span> micP <span class="sc">+</span> micCN <span class="sc">+</span> micCP <span class="sc">+</span> AG_ase <span class="sc">+</span> BG_ase <span class="sc">+</span> </span>
<span id="cb189-4396"><a href="#cb189-4396" aria-hidden="true" tabindex="-1"></a>                                       S_ase <span class="sc">+</span> XY_ase <span class="sc">+</span> PX_ase <span class="sc">+</span> CO2 <span class="sc">+</span> enzNP, </span>
<span id="cb189-4397"><a href="#cb189-4397" aria-hidden="true" tabindex="-1"></a>                           tmp_md, <span class="at">dist =</span> <span class="st">"bray"</span>)</span>
<span id="cb189-4398"><a href="#cb189-4398" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4399"><a href="#cb189-4399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4402"><a href="#cb189-4402" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4403"><a href="#cb189-4403" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4404"><a href="#cb189-4404" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4405"><a href="#cb189-4405" aria-hidden="true" tabindex="-1"></a>soil_funct_cap</span>
<span id="cb189-4406"><a href="#cb189-4406" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4407"><a href="#cb189-4407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4408"><a href="#cb189-4408" aria-hidden="true" tabindex="-1"></a>Now we can look at the variance against each principal component. </span>
<span id="cb189-4409"><a href="#cb189-4409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4412"><a href="#cb189-4412" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4413"><a href="#cb189-4413" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4414"><a href="#cb189-4414" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4415"><a href="#cb189-4415" aria-hidden="true" tabindex="-1"></a><span class="fu">screeplot</span>(soil_funct_cap)</span>
<span id="cb189-4416"><a href="#cb189-4416" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4417"><a href="#cb189-4417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4418"><a href="#cb189-4418" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4419"><a href="#cb189-4419" aria-hidden="true" tabindex="-1"></a>And then make some quick and dirty plots. This will also come in handy later when we need to parse out data a better plot visualization. The <span class="in">`ggplot`</span> function <span class="in">`autoplot`</span> stores these data in a more accessible way than the raw results from <span class="in">`capscale`</span></span>
<span id="cb189-4420"><a href="#cb189-4420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4423"><a href="#cb189-4423" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4424"><a href="#cb189-4424" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4425"><a href="#cb189-4425" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">plot</span>(soil_funct_cap) </span>
<span id="cb189-4426"><a href="#cb189-4426" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt <span class="ot">&lt;-</span> <span class="fu">autoplot</span>(soil_funct_cap, <span class="at">arrows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb189-4427"><a href="#cb189-4427" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt</span>
<span id="cb189-4428"><a href="#cb189-4428" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4429"><a href="#cb189-4429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4432"><a href="#cb189-4432" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4433"><a href="#cb189-4433" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4434"><a href="#cb189-4434" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(soil_funct_cap) <span class="co"># overall test of the significant of the analysis</span></span>
<span id="cb189-4435"><a href="#cb189-4435" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(soil_funct_cap, <span class="at">by =</span> <span class="st">"axis"</span>, <span class="at">perm.max=</span><span class="dv">500</span>) <span class="co"># test axes for significance</span></span>
<span id="cb189-4436"><a href="#cb189-4436" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(soil_funct_cap, <span class="at">by =</span> <span class="st">"terms"</span>, <span class="at">permu=</span><span class="dv">500</span>) <span class="co"># test for sign. environ. variables</span></span>
<span id="cb189-4437"><a href="#cb189-4437" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4438"><a href="#cb189-4438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4439"><a href="#cb189-4439" aria-hidden="true" tabindex="-1"></a>Next, we need to grab capscale scores for the samples and create a data frame of the first two dimensions. We will also need to add some of the sample details to the data frame. For this we use the vegan function <span class="in">`scores`</span> which gets species or site scores from the ordination.</span>
<span id="cb189-4440"><a href="#cb189-4440" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4443"><a href="#cb189-4443" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4444"><a href="#cb189-4444" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4445"><a href="#cb189-4445" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">"sites"</span>)</span>
<span id="cb189-4446"><a href="#cb189-4446" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-4447"><a href="#cb189-4447" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> tmp_samp_scores <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">SampleID =</span> Label)</span>
<span id="cb189-4448"><a href="#cb189-4448" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4449"><a href="#cb189-4449" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb189-4450"><a href="#cb189-4450" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md_sub <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb189-4451"><a href="#cb189-4451" aria-hidden="true" tabindex="-1"></a>soil_funct_plot_data <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_md_sub, tmp_samp_scores, </span>
<span id="cb189-4452"><a href="#cb189-4452" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">by =</span> <span class="st">"SampleID"</span>)</span>
<span id="cb189-4453"><a href="#cb189-4453" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4454"><a href="#cb189-4454" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4455"><a href="#cb189-4455" aria-hidden="true" tabindex="-1"></a>Now we have a new data frame that contains sample details and capscale values. </span>
<span id="cb189-4456"><a href="#cb189-4456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4459"><a href="#cb189-4459" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4460"><a href="#cb189-4460" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4461"><a href="#cb189-4461" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4462"><a href="#cb189-4462" aria-hidden="true" tabindex="-1"></a>soil_funct_plot_data</span>
<span id="cb189-4463"><a href="#cb189-4463" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4464"><a href="#cb189-4464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4465"><a href="#cb189-4465" aria-hidden="true" tabindex="-1"></a>We can then do the same with the metadata vectors. Here though we only need the scores and parameter name. </span>
<span id="cb189-4466"><a href="#cb189-4466" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4469"><a href="#cb189-4469" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4470"><a href="#cb189-4470" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4471"><a href="#cb189-4471" aria-hidden="true" tabindex="-1"></a>soil_funct_md_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">"biplot"</span>)</span>
<span id="cb189-4472"><a href="#cb189-4472" aria-hidden="true" tabindex="-1"></a>soil_funct_md_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-4473"><a href="#cb189-4473" aria-hidden="true" tabindex="-1"></a>soil_funct_md_scores <span class="ot">&lt;-</span> soil_funct_md_scores <span class="sc">%&gt;%</span> </span>
<span id="cb189-4474"><a href="#cb189-4474" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span> </span>
<span id="cb189-4475"><a href="#cb189-4475" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"Label"</span>)</span>
<span id="cb189-4476"><a href="#cb189-4476" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4477"><a href="#cb189-4477" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4478"><a href="#cb189-4478" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4481"><a href="#cb189-4481" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4482"><a href="#cb189-4482" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4483"><a href="#cb189-4483" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4484"><a href="#cb189-4484" aria-hidden="true" tabindex="-1"></a>soil_funct_md_scores</span>
<span id="cb189-4485"><a href="#cb189-4485" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4486"><a href="#cb189-4486" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4487"><a href="#cb189-4487" aria-hidden="true" tabindex="-1"></a>Let's run some quick correlations of metadata with ordination axes to see which parameters are significant. For this we use the vegan function <span class="in">`envfit`</span>.</span>
<span id="cb189-4488"><a href="#cb189-4488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4491"><a href="#cb189-4491" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4492"><a href="#cb189-4492" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4493"><a href="#cb189-4493" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> soil_funct_plot_data[, <span class="dv">6</span><span class="sc">:</span><span class="dv">7</span>]</span>
<span id="cb189-4494"><a href="#cb189-4494" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_samp_scores_sub)</span>
<span id="cb189-4495"><a href="#cb189-4495" aria-hidden="true" tabindex="-1"></a>tmp_param_list <span class="ot">&lt;-</span> soil_funct_md_scores<span class="sc">$</span>parameters</span>
<span id="cb189-4496"><a href="#cb189-4496" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4497"><a href="#cb189-4497" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> <span class="fu">subset</span>(tmp_md,  <span class="at">select =</span>  tmp_param_list)</span>
<span id="cb189-4498"><a href="#cb189-4498" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4499"><a href="#cb189-4499" aria-hidden="true" tabindex="-1"></a>envfit_soil_funct_md <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, tmp_md_sub,</span>
<span id="cb189-4500"><a href="#cb189-4500" aria-hidden="true" tabindex="-1"></a>                 <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb189-4501"><a href="#cb189-4501" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4502"><a href="#cb189-4502" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4505"><a href="#cb189-4505" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4506"><a href="#cb189-4506" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4507"><a href="#cb189-4507" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4508"><a href="#cb189-4508" aria-hidden="true" tabindex="-1"></a>envfit_soil_funct_md</span>
<span id="cb189-4509"><a href="#cb189-4509" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4510"><a href="#cb189-4510" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4513"><a href="#cb189-4513" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4514"><a href="#cb189-4514" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4515"><a href="#cb189-4515" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_soil_funct_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb189-4516"><a href="#cb189-4516" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_soil_funct_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb189-4517"><a href="#cb189-4517" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_soil_funct_md<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.4</span>))</span>
<span id="cb189-4518"><a href="#cb189-4518" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(soil_funct_md_signif_hits)</span>
<span id="cb189-4519"><a href="#cb189-4519" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(soil_funct_md_signif_hits)</span>
<span id="cb189-4520"><a href="#cb189-4520" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif <span class="ot">&lt;-</span> soil_funct_md_scores[soil_funct_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb189-4521"><a href="#cb189-4521" aria-hidden="true" tabindex="-1"></a>                                               soil_funct_md_signif_hits,]</span>
<span id="cb189-4522"><a href="#cb189-4522" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif<span class="sc">$</span>parameters</span>
<span id="cb189-4523"><a href="#cb189-4523" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4524"><a href="#cb189-4524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4525"><a href="#cb189-4525" aria-hidden="true" tabindex="-1"></a><span class="in">`envfit`</span> found that <span class="in">`r soil_funct_md_signif$parameters`</span> were significantly correlated. </span>
<span id="cb189-4526"><a href="#cb189-4526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4527"><a href="#cb189-4527" aria-hidden="true" tabindex="-1"></a>Now let's see if the same parameters are significant for the <span class="in">`envfit`</span> and <span class="in">`bioenv`</span> analyses.</span>
<span id="cb189-4528"><a href="#cb189-4528" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4531"><a href="#cb189-4531" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4532"><a href="#cb189-4532" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4533"><a href="#cb189-4533" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4534"><a href="#cb189-4534" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hold</span></span>
<span id="cb189-4535"><a href="#cb189-4535" aria-hidden="true" tabindex="-1"></a><span class="co">#| comment: '' </span></span>
<span id="cb189-4536"><a href="#cb189-4536" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Significant parameters from bioenv analysis."</span>)</span>
<span id="cb189-4537"><a href="#cb189-4537" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(<span class="fu">summary</span>(soil_funct_bioenv_ind_mantel))</span>
<span id="cb189-4538"><a href="#cb189-4538" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-4539"><a href="#cb189-4539" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-4540"><a href="#cb189-4540" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Significant parameters from envfit analysis."</span>)</span>
<span id="cb189-4541"><a href="#cb189-4541" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif<span class="sc">$</span>parameters</span>
<span id="cb189-4542"><a href="#cb189-4542" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-4543"><a href="#cb189-4543" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-4544"><a href="#cb189-4544" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Found in bioenv but not envfit."</span>)</span>
<span id="cb189-4545"><a href="#cb189-4545" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">setdiff</span>(<span class="fu">row.names</span>(<span class="fu">summary</span>(soil_funct_bioenv_ind_mantel)), </span>
<span id="cb189-4546"><a href="#cb189-4546" aria-hidden="true" tabindex="-1"></a>                       soil_funct_md_signif<span class="sc">$</span>parameters)</span>
<span id="cb189-4547"><a href="#cb189-4547" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-4548"><a href="#cb189-4548" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-4549"><a href="#cb189-4549" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Found in envfit but not bioenv."</span>)</span>
<span id="cb189-4550"><a href="#cb189-4550" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">setdiff</span>(soil_funct_md_signif<span class="sc">$</span>parameters, </span>
<span id="cb189-4551"><a href="#cb189-4551" aria-hidden="true" tabindex="-1"></a>              <span class="fu">row.names</span>(<span class="fu">summary</span>(soil_funct_bioenv_ind_mantel)))</span>
<span id="cb189-4552"><a href="#cb189-4552" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-4553"><a href="#cb189-4553" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-4554"><a href="#cb189-4554" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Found in envfit and bioenv."</span>)</span>
<span id="cb189-4555"><a href="#cb189-4555" aria-hidden="true" tabindex="-1"></a>soil_funct_sig_diff <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">union</span>(soil_funct_md_signif<span class="sc">$</span>parameters, </span>
<span id="cb189-4556"><a href="#cb189-4556" aria-hidden="true" tabindex="-1"></a>                                   <span class="fu">row.names</span>(<span class="fu">summary</span>(soil_funct_bioenv_ind_mantel)))</span>
<span id="cb189-4557"><a href="#cb189-4557" aria-hidden="true" tabindex="-1"></a>soil_funct_sig_diff</span>
<span id="cb189-4558"><a href="#cb189-4558" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4559"><a href="#cb189-4559" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4560"><a href="#cb189-4560" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4563"><a href="#cb189-4563" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4564"><a href="#cb189-4564" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4565"><a href="#cb189-4565" aria-hidden="true" tabindex="-1"></a><span class="co">#new_soil_funct_md_signif_hits &lt;- append(soil_funct_md_signif_hits, soil_funct_sig_diff)</span></span>
<span id="cb189-4566"><a href="#cb189-4566" aria-hidden="true" tabindex="-1"></a>new_soil_funct_md_signif_hits <span class="ot">&lt;-</span> soil_funct_sig_diff</span>
<span id="cb189-4567"><a href="#cb189-4567" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_all <span class="ot">&lt;-</span> soil_funct_md_scores[soil_funct_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb189-4568"><a href="#cb189-4568" aria-hidden="true" tabindex="-1"></a>                                                   new_soil_funct_md_signif_hits,]</span>
<span id="cb189-4569"><a href="#cb189-4569" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4570"><a href="#cb189-4570" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4571"><a href="#cb189-4571" aria-hidden="true" tabindex="-1"></a>Check. Next, we run <span class="in">`envfit`</span> for the ASVs.</span>
<span id="cb189-4572"><a href="#cb189-4572" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4575"><a href="#cb189-4575" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4576"><a href="#cb189-4576" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4577"><a href="#cb189-4577" aria-hidden="true" tabindex="-1"></a>envfit_soil_funct_asv <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, </span>
<span id="cb189-4578"><a href="#cb189-4578" aria-hidden="true" tabindex="-1"></a>                                tmp_comm[, <span class="fu">order</span>(<span class="fu">colSums</span>(<span class="sc">-</span>tmp_comm))][, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>], </span>
<span id="cb189-4579"><a href="#cb189-4579" aria-hidden="true" tabindex="-1"></a>                                <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb189-4580"><a href="#cb189-4580" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">"species"</span>)</span>
<span id="cb189-4581"><a href="#cb189-4581" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_scores <span class="ot">&lt;-</span> soil_funct_asv_scores <span class="sc">%&gt;%</span> </span>
<span id="cb189-4582"><a href="#cb189-4582" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span> </span>
<span id="cb189-4583"><a href="#cb189-4583" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"Label"</span>)</span>
<span id="cb189-4584"><a href="#cb189-4584" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-4585"><a href="#cb189-4585" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4586"><a href="#cb189-4586" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4589"><a href="#cb189-4589" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4590"><a href="#cb189-4590" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4591"><a href="#cb189-4591" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4592"><a href="#cb189-4592" aria-hidden="true" tabindex="-1"></a>envfit_soil_funct_asv</span>
<span id="cb189-4593"><a href="#cb189-4593" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4594"><a href="#cb189-4594" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4597"><a href="#cb189-4597" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4598"><a href="#cb189-4598" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4599"><a href="#cb189-4599" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_soil_funct_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb189-4600"><a href="#cb189-4600" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_soil_funct_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb189-4601"><a href="#cb189-4601" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_soil_funct_asv<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.5</span>))</span>
<span id="cb189-4602"><a href="#cb189-4602" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(soil_funct_asv_signif_hits)</span>
<span id="cb189-4603"><a href="#cb189-4603" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(soil_funct_asv_signif_hits)</span>
<span id="cb189-4604"><a href="#cb189-4604" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif <span class="ot">&lt;-</span> soil_funct_asv_scores[soil_funct_asv_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb189-4605"><a href="#cb189-4605" aria-hidden="true" tabindex="-1"></a>                                                 soil_funct_asv_signif_hits,]</span>
<span id="cb189-4606"><a href="#cb189-4606" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4607"><a href="#cb189-4607" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4610"><a href="#cb189-4610" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4611"><a href="#cb189-4611" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4612"><a href="#cb189-4612" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4613"><a href="#cb189-4613" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif</span>
<span id="cb189-4614"><a href="#cb189-4614" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4615"><a href="#cb189-4615" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4618"><a href="#cb189-4618" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4619"><a href="#cb189-4619" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4620"><a href="#cb189-4620" aria-hidden="true" tabindex="-1"></a>soil_funct_md_signif_all<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">"metadata"</span></span>
<span id="cb189-4621"><a href="#cb189-4621" aria-hidden="true" tabindex="-1"></a>soil_funct_asv_signif<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">"ASV"</span></span>
<span id="cb189-4622"><a href="#cb189-4622" aria-hidden="true" tabindex="-1"></a>soil_funct_bioplot_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(soil_funct_md_signif_all, soil_funct_asv_signif)</span>
<span id="cb189-4623"><a href="#cb189-4623" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4624"><a href="#cb189-4624" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4625"><a href="#cb189-4625" aria-hidden="true" tabindex="-1"></a>The last thing to do is categorize parameters scores and ASV scores into different variable types for plotting.</span>
<span id="cb189-4626"><a href="#cb189-4626" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4629"><a href="#cb189-4629" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4630"><a href="#cb189-4630" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4631"><a href="#cb189-4631" aria-hidden="true" tabindex="-1"></a>soil_funct_bioplot_data_md <span class="ot">&lt;-</span> <span class="fu">subset</span>(soil_funct_bioplot_data, </span>
<span id="cb189-4632"><a href="#cb189-4632" aria-hidden="true" tabindex="-1"></a>                                  soil_funct_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">"metadata"</span>)</span>
<span id="cb189-4633"><a href="#cb189-4633" aria-hidden="true" tabindex="-1"></a>soil_funct_bioplot_data_asv <span class="ot">&lt;-</span> <span class="fu">subset</span>(soil_funct_bioplot_data, </span>
<span id="cb189-4634"><a href="#cb189-4634" aria-hidden="true" tabindex="-1"></a>                                   soil_funct_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">"ASV"</span>)</span>
<span id="cb189-4635"><a href="#cb189-4635" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4636"><a href="#cb189-4636" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4637"><a href="#cb189-4637" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4638"><a href="#cb189-4638" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb189-4639"><a href="#cb189-4639" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show  the code for the plot<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb189-4640"><a href="#cb189-4640" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4643"><a href="#cb189-4643" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4644"><a href="#cb189-4644" aria-hidden="true" tabindex="-1"></a>soil_funct_cap_vals <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(soil_funct_cap<span class="sc">$</span>CCA<span class="sc">$</span>eig[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span>
<span id="cb189-4645"><a href="#cb189-4645" aria-hidden="true" tabindex="-1"></a>soil_funct_cap1 <span class="ot">&lt;-</span> <span class="fu">signif</span>((soil_funct_cap_vals[<span class="dv">1</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb189-4646"><a href="#cb189-4646" aria-hidden="true" tabindex="-1"></a>soil_funct_cap2 <span class="ot">&lt;-</span> <span class="fu">signif</span>((soil_funct_cap_vals[<span class="dv">2</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb189-4647"><a href="#cb189-4647" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4648"><a href="#cb189-4648" aria-hidden="true" tabindex="-1"></a>cpa1_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"CAP1"</span>, <span class="st">" ("</span>, soil_funct_cap1, <span class="st">"%)"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb189-4649"><a href="#cb189-4649" aria-hidden="true" tabindex="-1"></a>cpa2_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"CAP2"</span>, <span class="st">" ("</span>, soil_funct_cap2, <span class="st">"%)"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb189-4650"><a href="#cb189-4650" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4651"><a href="#cb189-4651" aria-hidden="true" tabindex="-1"></a>swel_col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#2271B2"</span>, <span class="st">"#71B222"</span>, <span class="st">"#B22271"</span>)</span>
<span id="cb189-4652"><a href="#cb189-4652" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4653"><a href="#cb189-4653" aria-hidden="true" tabindex="-1"></a>soil_funct_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(soil_funct_plot_data) <span class="sc">+</span></span>
<span id="cb189-4654"><a href="#cb189-4654" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">shape =</span> TREAT,</span>
<span id="cb189-4655"><a href="#cb189-4655" aria-hidden="true" tabindex="-1"></a>                           <span class="at">colour =</span> TREAT_T), <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb189-4656"><a href="#cb189-4656" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> swel_col) <span class="sc">+</span></span>
<span id="cb189-4657"><a href="#cb189-4657" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_text(data = soil_funct_plot_data, aes(x = CAP1, y = CAP2, #UNCOMMENT to add sample labels</span></span>
<span id="cb189-4658"><a href="#cb189-4658" aria-hidden="true" tabindex="-1"></a><span class="co">#                              label = SampleID), size = 3) + </span></span>
<span id="cb189-4659"><a href="#cb189-4659" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">xend =</span> CAP1, <span class="at">yend =</span> CAP2),</span>
<span id="cb189-4660"><a href="#cb189-4660" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> soil_funct_bioplot_data_md, <span class="at">linetype =</span> <span class="st">"solid"</span>,</span>
<span id="cb189-4661"><a href="#cb189-4661" aria-hidden="true" tabindex="-1"></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">"cm"</span>)), <span class="at">size =</span> <span class="fl">0.4</span>, </span>
<span id="cb189-4662"><a href="#cb189-4662" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">"#191919"</span>) <span class="sc">+</span></span>
<span id="cb189-4663"><a href="#cb189-4663" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> soil_funct_bioplot_data_md, </span>
<span id="cb189-4664"><a href="#cb189-4664" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">label =</span> parameters), <span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb189-4665"><a href="#cb189-4665" aria-hidden="true" tabindex="-1"></a>            <span class="at">nudge_x =</span> <span class="fl">0.1</span>, <span class="at">nudge_y =</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb189-4666"><a href="#cb189-4666" aria-hidden="true" tabindex="-1"></a>  <span class="do">## USE this code to overlay ASV vestors</span></span>
<span id="cb189-4667"><a href="#cb189-4667" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_segment(aes(x = 0, y = 0, xend = CAP1, yend = CAP2),</span></span>
<span id="cb189-4668"><a href="#cb189-4668" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             data = soil_funct_bioplot_data_asv, linetype = "solid",</span></span>
<span id="cb189-4669"><a href="#cb189-4669" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             arrow = arrow(length = unit(0.3, "cm")), size = 0.2,</span></span>
<span id="cb189-4670"><a href="#cb189-4670" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             color = "#676767") +</span></span>
<span id="cb189-4671"><a href="#cb189-4671" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text(data = soil_funct_bioplot_data_asv, </span></span>
<span id="cb189-4672"><a href="#cb189-4672" aria-hidden="true" tabindex="-1"></a>  <span class="co">#          aes(x = CAP1, y = CAP2, label = parameters), size = 2.5,</span></span>
<span id="cb189-4673"><a href="#cb189-4673" aria-hidden="true" tabindex="-1"></a>  <span class="co">#          nudge_x = 0.05, nudge_y = 0.05) +</span></span>
<span id="cb189-4674"><a href="#cb189-4674" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span> </span>
<span id="cb189-4675"><a href="#cb189-4675" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Capscale Analysis"</span>,</span>
<span id="cb189-4676"><a href="#cb189-4676" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">"Soil Functional Response"</span>, </span>
<span id="cb189-4677"><a href="#cb189-4677" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> cpa1_lab, </span>
<span id="cb189-4678"><a href="#cb189-4678" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> cpa2_lab)</span>
<span id="cb189-4679"><a href="#cb189-4679" aria-hidden="true" tabindex="-1"></a>soil_funct_plot <span class="ot">&lt;-</span> soil_funct_plot <span class="sc">+</span> <span class="fu">coord_fixed</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb189-4680"><a href="#cb189-4680" aria-hidden="true" tabindex="-1"></a>soil_funct_plot</span>
<span id="cb189-4681"><a href="#cb189-4681" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">"files/metadata/figures/its18_soil_funct_capscale.png"</span>,</span>
<span id="cb189-4682"><a href="#cb189-4682" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">16</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">units =</span> <span class="st">'cm'</span>, <span class="at">res =</span> <span class="dv">600</span>, <span class="at">bg =</span> <span class="st">"white"</span>)</span>
<span id="cb189-4683"><a href="#cb189-4683" aria-hidden="true" tabindex="-1"></a>soil_funct_plot</span>
<span id="cb189-4684"><a href="#cb189-4684" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb189-4685"><a href="#cb189-4685" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"files/metadata/figures/its18_soil_funct_capscale.pdf"</span>,</span>
<span id="cb189-4686"><a href="#cb189-4686" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">5</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb189-4687"><a href="#cb189-4687" aria-hidden="true" tabindex="-1"></a>soil_funct_plot</span>
<span id="cb189-4688"><a href="#cb189-4688" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb189-4689"><a href="#cb189-4689" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4690"><a href="#cb189-4690" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb189-4691"><a href="#cb189-4691" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4694"><a href="#cb189-4694" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4695"><a href="#cb189-4695" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4696"><a href="#cb189-4696" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-4697"><a href="#cb189-4697" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4698"><a href="#cb189-4698" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4699"><a href="#cb189-4699" aria-hidden="true" tabindex="-1"></a><span class="fu">### Temperature Adaptation</span></span>
<span id="cb189-4700"><a href="#cb189-4700" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4703"><a href="#cb189-4703" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4704"><a href="#cb189-4704" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4705"><a href="#cb189-4705" aria-hidden="true" tabindex="-1"></a>tmp_md <span class="ot">&lt;-</span> its18_select_mc_norm_split_no_ac<span class="sc">$</span>temp_adapt</span>
<span id="cb189-4706"><a href="#cb189-4706" aria-hidden="true" tabindex="-1"></a>tmp_md<span class="sc">$</span>TREAT_T <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tmp_md<span class="sc">$</span>TREAT_T)</span>
<span id="cb189-4707"><a href="#cb189-4707" aria-hidden="true" tabindex="-1"></a>tmp_comm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(its18_select_mc_norm_split_no_ac<span class="sc">$</span>data_loaded))</span>
<span id="cb189-4708"><a href="#cb189-4708" aria-hidden="true" tabindex="-1"></a>temp_adapt_rank <span class="ot">&lt;-</span> <span class="fu">rankindex</span>(tmp_md[, <span class="dv">8</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_md)], tmp_comm, </span>
<span id="cb189-4709"><a href="#cb189-4709" aria-hidden="true" tabindex="-1"></a>          <span class="at">indices =</span> <span class="fu">c</span>(<span class="st">"euc"</span>, <span class="st">"man"</span>, <span class="st">"gow"</span>, <span class="st">"bra"</span>, <span class="st">"kul"</span>), </span>
<span id="cb189-4710"><a href="#cb189-4710" aria-hidden="true" tabindex="-1"></a>          <span class="at">stepacross =</span> <span class="cn">FALSE</span>, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb189-4711"><a href="#cb189-4711" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4712"><a href="#cb189-4712" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4715"><a href="#cb189-4715" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4716"><a href="#cb189-4716" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4717"><a href="#cb189-4717" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4718"><a href="#cb189-4718" aria-hidden="true" tabindex="-1"></a>temp_adapt_rank</span>
<span id="cb189-4719"><a href="#cb189-4719" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4720"><a href="#cb189-4720" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4721"><a href="#cb189-4721" aria-hidden="true" tabindex="-1"></a>Let's run <span class="in">`capscale`</span> using Bray-Curtis. Note, we have <span class="in">`r ncol(its18_select_mc_norm_split_no_ac$temp_adapt) - 8`</span> metadata parameters in this group but, for some reason, <span class="in">`capscale`</span> only works with 13 parameters. This may have to do with degrees of freedom? </span>
<span id="cb189-4722"><a href="#cb189-4722" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4723"><a href="#cb189-4723" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Starting properties: AG_Q10, BG_Q10, BP_Q10, CE_Q10, P_Q10, N_Q10, S_Q10, XY_Q10, LP_Q10, PX_Q10, CUEcn, CUEcp, NUE, PUE, Tmin, SI</span>
<span id="cb189-4724"><a href="#cb189-4724" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Autocorrelated removed: NUE, PUE, P_Q10, SI                                                                        </span>
<span id="cb189-4725"><a href="#cb189-4725" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Remove for capscale: S_Q10</span>
<span id="cb189-4726"><a href="#cb189-4726" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4729"><a href="#cb189-4729" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4730"><a href="#cb189-4730" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4731"><a href="#cb189-4731" aria-hidden="true" tabindex="-1"></a><span class="co"># All temp_adapt: AG_Q10, BG_Q10, BP_Q10, CE_Q10, P_Q10, N_Q10, S_Q10, XY_Q10, LP_Q10, PX_Q10, CUEcn, CUEcp, NUE, PUE, Tmin, SI</span></span>
<span id="cb189-4732"><a href="#cb189-4732" aria-hidden="true" tabindex="-1"></a><span class="co"># REMOVED edaphic: NUE, PUE, P_Q10, SI                                                                        </span></span>
<span id="cb189-4733"><a href="#cb189-4733" aria-hidden="true" tabindex="-1"></a><span class="co"># Remaining edaphic: AG_Q10, BG_Q10, BP_Q10, CE_Q10, N_Q10, S_Q10, XY_Q10, LP_Q10, PX_Q10, CUEcn, CUEcp, Tmin</span></span>
<span id="cb189-4734"><a href="#cb189-4734" aria-hidden="true" tabindex="-1"></a><span class="co"># only works with 11</span></span>
<span id="cb189-4735"><a href="#cb189-4735" aria-hidden="true" tabindex="-1"></a><span class="co"># Removed:  S_Q10</span></span>
<span id="cb189-4736"><a href="#cb189-4736" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4737"><a href="#cb189-4737" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4740"><a href="#cb189-4740" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4741"><a href="#cb189-4741" aria-hidden="true" tabindex="-1"></a>temp_adapt_cap <span class="ot">&lt;-</span> <span class="fu">capscale</span>(tmp_comm <span class="sc">~</span>  AG_Q10 <span class="sc">+</span> BG_Q10 <span class="sc">+</span> BP_Q10 <span class="sc">+</span> CE_Q10 <span class="sc">+</span> </span>
<span id="cb189-4742"><a href="#cb189-4742" aria-hidden="true" tabindex="-1"></a>                                       N_Q10 <span class="sc">+</span> XY_Q10 <span class="sc">+</span> LP_Q10 <span class="sc">+</span> PX_Q10 <span class="sc">+</span> </span>
<span id="cb189-4743"><a href="#cb189-4743" aria-hidden="true" tabindex="-1"></a>                                       CUEcn <span class="sc">+</span> CUEcp <span class="sc">+</span> Tmin, </span>
<span id="cb189-4744"><a href="#cb189-4744" aria-hidden="true" tabindex="-1"></a>                           tmp_md, <span class="at">dist =</span> <span class="st">"bray"</span>)</span>
<span id="cb189-4745"><a href="#cb189-4745" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4746"><a href="#cb189-4746" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4749"><a href="#cb189-4749" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4750"><a href="#cb189-4750" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4751"><a href="#cb189-4751" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4752"><a href="#cb189-4752" aria-hidden="true" tabindex="-1"></a>temp_adapt_cap</span>
<span id="cb189-4753"><a href="#cb189-4753" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4754"><a href="#cb189-4754" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4755"><a href="#cb189-4755" aria-hidden="true" tabindex="-1"></a>Now we can look at the variance against each principal component. </span>
<span id="cb189-4756"><a href="#cb189-4756" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4759"><a href="#cb189-4759" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4760"><a href="#cb189-4760" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4761"><a href="#cb189-4761" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4762"><a href="#cb189-4762" aria-hidden="true" tabindex="-1"></a><span class="fu">screeplot</span>(temp_adapt_cap)</span>
<span id="cb189-4763"><a href="#cb189-4763" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4764"><a href="#cb189-4764" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4765"><a href="#cb189-4765" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4766"><a href="#cb189-4766" aria-hidden="true" tabindex="-1"></a>And then make some quick and dirty plots. This will also come in handy later when we need to parse out data a better plot visualization. The <span class="in">`ggplot`</span> function <span class="in">`autoplot`</span> stores these data in a more accessible way than the raw results from <span class="in">`capscale`</span></span>
<span id="cb189-4767"><a href="#cb189-4767" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4770"><a href="#cb189-4770" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4771"><a href="#cb189-4771" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4772"><a href="#cb189-4772" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">plot</span>(temp_adapt_cap) </span>
<span id="cb189-4773"><a href="#cb189-4773" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt <span class="ot">&lt;-</span> <span class="fu">autoplot</span>(temp_adapt_cap, <span class="at">arrows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb189-4774"><a href="#cb189-4774" aria-hidden="true" tabindex="-1"></a>tmp_auto_plt</span>
<span id="cb189-4775"><a href="#cb189-4775" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4776"><a href="#cb189-4776" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4779"><a href="#cb189-4779" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4780"><a href="#cb189-4780" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4781"><a href="#cb189-4781" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(temp_adapt_cap) <span class="co"># overall test of the significant of the analysis</span></span>
<span id="cb189-4782"><a href="#cb189-4782" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(temp_adapt_cap, <span class="at">by =</span> <span class="st">"axis"</span>, <span class="at">perm.max =</span> <span class="dv">500</span>) <span class="co"># test axes for significance</span></span>
<span id="cb189-4783"><a href="#cb189-4783" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(temp_adapt_cap, <span class="at">by =</span> <span class="st">"terms"</span>, <span class="at">permu =</span> <span class="dv">500</span>) <span class="co"># test for sign. environ. variables</span></span>
<span id="cb189-4784"><a href="#cb189-4784" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4785"><a href="#cb189-4785" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4786"><a href="#cb189-4786" aria-hidden="true" tabindex="-1"></a>Next, we need to grab capscale scores for the samples and create a data frame of the first two dimensions. We will also need to add some of the sample details to the data frame. For this we use the vegan function <span class="in">`scores`</span> which gets species or site scores from the ordination.</span>
<span id="cb189-4787"><a href="#cb189-4787" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4790"><a href="#cb189-4790" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4791"><a href="#cb189-4791" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4792"><a href="#cb189-4792" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">"sites"</span>)</span>
<span id="cb189-4793"><a href="#cb189-4793" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-4794"><a href="#cb189-4794" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores <span class="ot">&lt;-</span> tmp_samp_scores <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">SampleID =</span> Label)</span>
<span id="cb189-4795"><a href="#cb189-4795" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4796"><a href="#cb189-4796" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb189-4797"><a href="#cb189-4797" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> tmp_md_sub <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb189-4798"><a href="#cb189-4798" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot_data <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_md_sub, tmp_samp_scores, </span>
<span id="cb189-4799"><a href="#cb189-4799" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">by =</span> <span class="st">"SampleID"</span>)</span>
<span id="cb189-4800"><a href="#cb189-4800" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4801"><a href="#cb189-4801" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4802"><a href="#cb189-4802" aria-hidden="true" tabindex="-1"></a>Now we have a new data frame that contains sample details and capscale values. </span>
<span id="cb189-4803"><a href="#cb189-4803" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4806"><a href="#cb189-4806" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4807"><a href="#cb189-4807" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4808"><a href="#cb189-4808" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4809"><a href="#cb189-4809" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot_data</span>
<span id="cb189-4810"><a href="#cb189-4810" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4811"><a href="#cb189-4811" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4812"><a href="#cb189-4812" aria-hidden="true" tabindex="-1"></a>We can then do the same with the metadata vectors. Here though we only need the scores and parameter name. </span>
<span id="cb189-4813"><a href="#cb189-4813" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4816"><a href="#cb189-4816" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4817"><a href="#cb189-4817" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4818"><a href="#cb189-4818" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">"biplot"</span>)</span>
<span id="cb189-4819"><a href="#cb189-4819" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-4820"><a href="#cb189-4820" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_scores <span class="ot">&lt;-</span> temp_adapt_md_scores <span class="sc">%&gt;%</span> </span>
<span id="cb189-4821"><a href="#cb189-4821" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span> </span>
<span id="cb189-4822"><a href="#cb189-4822" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"Label"</span>)</span>
<span id="cb189-4823"><a href="#cb189-4823" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4824"><a href="#cb189-4824" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4825"><a href="#cb189-4825" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4828"><a href="#cb189-4828" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4829"><a href="#cb189-4829" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4830"><a href="#cb189-4830" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4831"><a href="#cb189-4831" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_scores</span>
<span id="cb189-4832"><a href="#cb189-4832" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4833"><a href="#cb189-4833" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4834"><a href="#cb189-4834" aria-hidden="true" tabindex="-1"></a>Let's run some quick correlations of metadata with ordination axes to see which parameters are significant. For this we use the vegan function <span class="in">`envfit`</span>.</span>
<span id="cb189-4835"><a href="#cb189-4835" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4838"><a href="#cb189-4838" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4839"><a href="#cb189-4839" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4840"><a href="#cb189-4840" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> temp_adapt_plot_data[, <span class="dv">6</span><span class="sc">:</span><span class="dv">7</span>]</span>
<span id="cb189-4841"><a href="#cb189-4841" aria-hidden="true" tabindex="-1"></a>tmp_samp_scores_sub <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_samp_scores_sub)</span>
<span id="cb189-4842"><a href="#cb189-4842" aria-hidden="true" tabindex="-1"></a>tmp_param_list <span class="ot">&lt;-</span> temp_adapt_md_scores<span class="sc">$</span>parameters</span>
<span id="cb189-4843"><a href="#cb189-4843" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4844"><a href="#cb189-4844" aria-hidden="true" tabindex="-1"></a>tmp_md_sub <span class="ot">&lt;-</span> <span class="fu">subset</span>(tmp_md,  <span class="at">select =</span>  tmp_param_list)</span>
<span id="cb189-4845"><a href="#cb189-4845" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4846"><a href="#cb189-4846" aria-hidden="true" tabindex="-1"></a>envfit_temp_adapt_md <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, tmp_md_sub,</span>
<span id="cb189-4847"><a href="#cb189-4847" aria-hidden="true" tabindex="-1"></a>                 <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb189-4848"><a href="#cb189-4848" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4849"><a href="#cb189-4849" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4852"><a href="#cb189-4852" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4853"><a href="#cb189-4853" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4854"><a href="#cb189-4854" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4855"><a href="#cb189-4855" aria-hidden="true" tabindex="-1"></a>envfit_temp_adapt_md</span>
<span id="cb189-4856"><a href="#cb189-4856" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4857"><a href="#cb189-4857" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4860"><a href="#cb189-4860" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4861"><a href="#cb189-4861" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4862"><a href="#cb189-4862" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_temp_adapt_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb189-4863"><a href="#cb189-4863" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_temp_adapt_md<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb189-4864"><a href="#cb189-4864" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_temp_adapt_md<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.4</span>))</span>
<span id="cb189-4865"><a href="#cb189-4865" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(temp_adapt_md_signif_hits)</span>
<span id="cb189-4866"><a href="#cb189-4866" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(temp_adapt_md_signif_hits)</span>
<span id="cb189-4867"><a href="#cb189-4867" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif <span class="ot">&lt;-</span> temp_adapt_md_scores[temp_adapt_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb189-4868"><a href="#cb189-4868" aria-hidden="true" tabindex="-1"></a>                                               temp_adapt_md_signif_hits,]</span>
<span id="cb189-4869"><a href="#cb189-4869" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4870"><a href="#cb189-4870" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4871"><a href="#cb189-4871" aria-hidden="true" tabindex="-1"></a><span class="in">`envfit`</span> found that <span class="in">`r temp_adapt_md_signif$parameters`</span> were significantly correlated. </span>
<span id="cb189-4872"><a href="#cb189-4872" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4873"><a href="#cb189-4873" aria-hidden="true" tabindex="-1"></a>Now let's see if the same parameters are significant for the <span class="in">`envfit`</span> and <span class="in">`bioenv`</span> analyses.</span>
<span id="cb189-4874"><a href="#cb189-4874" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4877"><a href="#cb189-4877" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4878"><a href="#cb189-4878" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4879"><a href="#cb189-4879" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4880"><a href="#cb189-4880" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hold</span></span>
<span id="cb189-4881"><a href="#cb189-4881" aria-hidden="true" tabindex="-1"></a><span class="co">#| comment: '' </span></span>
<span id="cb189-4882"><a href="#cb189-4882" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Significant parameters from bioenv analysis."</span>)</span>
<span id="cb189-4883"><a href="#cb189-4883" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(<span class="fu">summary</span>(temp_adapt_bioenv_ind_mantel))</span>
<span id="cb189-4884"><a href="#cb189-4884" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-4885"><a href="#cb189-4885" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-4886"><a href="#cb189-4886" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Significant parameters from envfit analysis."</span>)</span>
<span id="cb189-4887"><a href="#cb189-4887" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif<span class="sc">$</span>parameters</span>
<span id="cb189-4888"><a href="#cb189-4888" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-4889"><a href="#cb189-4889" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-4890"><a href="#cb189-4890" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Found in bioenv but not envfit."</span>)</span>
<span id="cb189-4891"><a href="#cb189-4891" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">setdiff</span>(<span class="fu">row.names</span>(<span class="fu">summary</span>(temp_adapt_bioenv_ind_mantel)), </span>
<span id="cb189-4892"><a href="#cb189-4892" aria-hidden="true" tabindex="-1"></a>                       temp_adapt_md_signif<span class="sc">$</span>parameters)</span>
<span id="cb189-4893"><a href="#cb189-4893" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-4894"><a href="#cb189-4894" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-4895"><a href="#cb189-4895" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Found in envfit but not bioenv."</span>)</span>
<span id="cb189-4896"><a href="#cb189-4896" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">setdiff</span>(temp_adapt_md_signif<span class="sc">$</span>parameters, </span>
<span id="cb189-4897"><a href="#cb189-4897" aria-hidden="true" tabindex="-1"></a>              <span class="fu">row.names</span>(<span class="fu">summary</span>(temp_adapt_bioenv_ind_mantel)))</span>
<span id="cb189-4898"><a href="#cb189-4898" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"_____________________________________"</span>)</span>
<span id="cb189-4899"><a href="#cb189-4899" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb189-4900"><a href="#cb189-4900" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Found in envfit and  bioenv."</span>)</span>
<span id="cb189-4901"><a href="#cb189-4901" aria-hidden="true" tabindex="-1"></a>temp_adapt_sig_diff <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">union</span>(temp_adapt_md_signif<span class="sc">$</span>parameters, </span>
<span id="cb189-4902"><a href="#cb189-4902" aria-hidden="true" tabindex="-1"></a>                                   <span class="fu">row.names</span>(<span class="fu">summary</span>(temp_adapt_bioenv_ind_mantel)))</span>
<span id="cb189-4903"><a href="#cb189-4903" aria-hidden="true" tabindex="-1"></a>temp_adapt_sig_diff</span>
<span id="cb189-4904"><a href="#cb189-4904" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4905"><a href="#cb189-4905" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4908"><a href="#cb189-4908" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4909"><a href="#cb189-4909" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4910"><a href="#cb189-4910" aria-hidden="true" tabindex="-1"></a><span class="co">#new_temp_adapt_md_signif_hits &lt;- base::append(temp_adapt_md_signif_hits, temp_adapt_sig_diff)</span></span>
<span id="cb189-4911"><a href="#cb189-4911" aria-hidden="true" tabindex="-1"></a>new_temp_adapt_md_signif_hits <span class="ot">&lt;-</span> temp_adapt_sig_diff[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb189-4912"><a href="#cb189-4912" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_all <span class="ot">&lt;-</span> temp_adapt_md_scores[temp_adapt_md_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb189-4913"><a href="#cb189-4913" aria-hidden="true" tabindex="-1"></a>                                                   new_temp_adapt_md_signif_hits,]</span>
<span id="cb189-4914"><a href="#cb189-4914" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4915"><a href="#cb189-4915" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4916"><a href="#cb189-4916" aria-hidden="true" tabindex="-1"></a>Check. Next, we run <span class="in">`envfit`</span> for the ASVs.</span>
<span id="cb189-4917"><a href="#cb189-4917" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4920"><a href="#cb189-4920" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4921"><a href="#cb189-4921" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4922"><a href="#cb189-4922" aria-hidden="true" tabindex="-1"></a>envfit_temp_adapt_asv <span class="ot">&lt;-</span> <span class="fu">envfit</span>(tmp_samp_scores_sub, </span>
<span id="cb189-4923"><a href="#cb189-4923" aria-hidden="true" tabindex="-1"></a>                                tmp_comm[, <span class="fu">order</span>(<span class="fu">colSums</span>(<span class="sc">-</span>tmp_comm))][, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],</span>
<span id="cb189-4924"><a href="#cb189-4924" aria-hidden="true" tabindex="-1"></a>                                <span class="at">perm =</span> <span class="dv">1000</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb189-4925"><a href="#cb189-4925" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4926"><a href="#cb189-4926" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tmp_auto_plt<span class="sc">$</span>plot_env<span class="sc">$</span>obj, Score <span class="sc">==</span> <span class="st">"species"</span>)</span>
<span id="cb189-4927"><a href="#cb189-4927" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4928"><a href="#cb189-4928" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_scores <span class="ot">&lt;-</span> temp_adapt_asv_scores <span class="sc">%&gt;%</span> </span>
<span id="cb189-4929"><a href="#cb189-4929" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">parameters =</span> Label, <span class="at">.before =</span> CAP1) <span class="sc">%&gt;%</span></span>
<span id="cb189-4930"><a href="#cb189-4930" aria-hidden="true" tabindex="-1"></a>                         tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"Label"</span>)</span>
<span id="cb189-4931"><a href="#cb189-4931" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_scores[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-4932"><a href="#cb189-4932" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4933"><a href="#cb189-4933" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4936"><a href="#cb189-4936" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4937"><a href="#cb189-4937" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4938"><a href="#cb189-4938" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4939"><a href="#cb189-4939" aria-hidden="true" tabindex="-1"></a>envfit_temp_adapt_asv</span>
<span id="cb189-4940"><a href="#cb189-4940" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4941"><a href="#cb189-4941" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4944"><a href="#cb189-4944" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4945"><a href="#cb189-4945" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4946"><a href="#cb189-4946" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif_hits <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(envfit_temp_adapt_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals, </span>
<span id="cb189-4947"><a href="#cb189-4947" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(envfit_temp_adapt_asv<span class="sc">$</span>vectors<span class="sc">$</span>pvals </span>
<span id="cb189-4948"><a href="#cb189-4948" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> envfit_temp_adapt_asv<span class="sc">$</span>vectors<span class="sc">$</span>r <span class="sc">&gt;</span> <span class="fl">0.5</span>))</span>
<span id="cb189-4949"><a href="#cb189-4949" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(temp_adapt_asv_signif_hits)</span>
<span id="cb189-4950"><a href="#cb189-4950" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif_hits <span class="ot">&lt;-</span> <span class="fu">rownames</span>(temp_adapt_asv_signif_hits)</span>
<span id="cb189-4951"><a href="#cb189-4951" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif <span class="ot">&lt;-</span> temp_adapt_asv_scores[temp_adapt_asv_scores<span class="sc">$</span>parameters <span class="sc">%in%</span> </span>
<span id="cb189-4952"><a href="#cb189-4952" aria-hidden="true" tabindex="-1"></a>                                                 temp_adapt_asv_signif_hits,]</span>
<span id="cb189-4953"><a href="#cb189-4953" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4954"><a href="#cb189-4954" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4957"><a href="#cb189-4957" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4958"><a href="#cb189-4958" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-4959"><a href="#cb189-4959" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-4960"><a href="#cb189-4960" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif</span>
<span id="cb189-4961"><a href="#cb189-4961" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4962"><a href="#cb189-4962" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb189-4963"><a href="#cb189-4963" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4966"><a href="#cb189-4966" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4967"><a href="#cb189-4967" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4968"><a href="#cb189-4968" aria-hidden="true" tabindex="-1"></a>temp_adapt_md_signif_all<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">"metadata"</span></span>
<span id="cb189-4969"><a href="#cb189-4969" aria-hidden="true" tabindex="-1"></a>temp_adapt_asv_signif<span class="sc">$</span>variable_type <span class="ot">&lt;-</span> <span class="st">"ASV"</span></span>
<span id="cb189-4970"><a href="#cb189-4970" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioplot_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(temp_adapt_md_signif_all, temp_adapt_asv_signif)</span>
<span id="cb189-4971"><a href="#cb189-4971" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4972"><a href="#cb189-4972" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4973"><a href="#cb189-4973" aria-hidden="true" tabindex="-1"></a>The last thing to do is categorize parameters scores and ASV scores into different variable types for plotting.</span>
<span id="cb189-4974"><a href="#cb189-4974" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4977"><a href="#cb189-4977" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4978"><a href="#cb189-4978" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb189-4979"><a href="#cb189-4979" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioplot_data_md <span class="ot">&lt;-</span> <span class="fu">subset</span>(temp_adapt_bioplot_data, </span>
<span id="cb189-4980"><a href="#cb189-4980" aria-hidden="true" tabindex="-1"></a>                                  temp_adapt_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">"metadata"</span>)</span>
<span id="cb189-4981"><a href="#cb189-4981" aria-hidden="true" tabindex="-1"></a>temp_adapt_bioplot_data_asv <span class="ot">&lt;-</span> <span class="fu">subset</span>(temp_adapt_bioplot_data, </span>
<span id="cb189-4982"><a href="#cb189-4982" aria-hidden="true" tabindex="-1"></a>                                   temp_adapt_bioplot_data<span class="sc">$</span>variable_type <span class="sc">==</span> <span class="st">"ASV"</span>)</span>
<span id="cb189-4983"><a href="#cb189-4983" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-4984"><a href="#cb189-4984" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4985"><a href="#cb189-4985" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb189-4986"><a href="#cb189-4986" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>Show  the code for the plot<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb189-4987"><a href="#cb189-4987" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4990"><a href="#cb189-4990" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-4991"><a href="#cb189-4991" aria-hidden="true" tabindex="-1"></a>temp_adapt_cap_vals <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(temp_adapt_cap<span class="sc">$</span>CCA<span class="sc">$</span>eig[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span>
<span id="cb189-4992"><a href="#cb189-4992" aria-hidden="true" tabindex="-1"></a>temp_adapt_cap1 <span class="ot">&lt;-</span> <span class="fu">signif</span>((temp_adapt_cap_vals[<span class="dv">1</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb189-4993"><a href="#cb189-4993" aria-hidden="true" tabindex="-1"></a>temp_adapt_cap2 <span class="ot">&lt;-</span> <span class="fu">signif</span>((temp_adapt_cap_vals[<span class="dv">2</span>,] <span class="sc">*</span> <span class="dv">100</span>), <span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb189-4994"><a href="#cb189-4994" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4995"><a href="#cb189-4995" aria-hidden="true" tabindex="-1"></a>cpa1_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"CAP1"</span>, <span class="st">" ("</span>, temp_adapt_cap1, <span class="st">"%)"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb189-4996"><a href="#cb189-4996" aria-hidden="true" tabindex="-1"></a>cpa2_lab <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"CAP2"</span>, <span class="st">" ("</span>, temp_adapt_cap2, <span class="st">"%)"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb189-4997"><a href="#cb189-4997" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4998"><a href="#cb189-4998" aria-hidden="true" tabindex="-1"></a>swel_col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#2271B2"</span>, <span class="st">"#71B222"</span>, <span class="st">"#B22271"</span>)</span>
<span id="cb189-4999"><a href="#cb189-4999" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5000"><a href="#cb189-5000" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(temp_adapt_plot_data) <span class="sc">+</span></span>
<span id="cb189-5001"><a href="#cb189-5001" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">shape =</span> TREAT,</span>
<span id="cb189-5002"><a href="#cb189-5002" aria-hidden="true" tabindex="-1"></a>                           <span class="at">colour =</span> TREAT_T), <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb189-5003"><a href="#cb189-5003" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> swel_col) <span class="sc">+</span></span>
<span id="cb189-5004"><a href="#cb189-5004" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_text(data = temp_adapt_plot_data, aes(x = CAP1, y = CAP2, #UNCOMMENT to add sample labels</span></span>
<span id="cb189-5005"><a href="#cb189-5005" aria-hidden="true" tabindex="-1"></a><span class="co">#                              label = SampleID), size = 3) + </span></span>
<span id="cb189-5006"><a href="#cb189-5006" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">xend =</span> CAP1, <span class="at">yend =</span> CAP2),</span>
<span id="cb189-5007"><a href="#cb189-5007" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> temp_adapt_bioplot_data_md, <span class="at">linetype =</span> <span class="st">"solid"</span>,</span>
<span id="cb189-5008"><a href="#cb189-5008" aria-hidden="true" tabindex="-1"></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">"cm"</span>)), <span class="at">size =</span> <span class="fl">0.4</span>,</span>
<span id="cb189-5009"><a href="#cb189-5009" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">"#191919"</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb189-5010"><a href="#cb189-5010" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> temp_adapt_bioplot_data_md, </span>
<span id="cb189-5011"><a href="#cb189-5011" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> CAP1, <span class="at">y =</span> CAP2, <span class="at">label =</span> parameters), <span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb189-5012"><a href="#cb189-5012" aria-hidden="true" tabindex="-1"></a>            <span class="at">nudge_x =</span> <span class="fl">0.1</span>, <span class="at">nudge_y =</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb189-5013"><a href="#cb189-5013" aria-hidden="true" tabindex="-1"></a>  <span class="co"># uSe to include ASV vectors</span></span>
<span id="cb189-5014"><a href="#cb189-5014" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_segment(aes(x = 0, y = 0, xend = CAP1, yend = CAP2),</span></span>
<span id="cb189-5015"><a href="#cb189-5015" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             data = temp_adapt_bioplot_data_asv, linetype = "solid",</span></span>
<span id="cb189-5016"><a href="#cb189-5016" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             arrow = arrow(length = unit(0.3, "cm")), size = 0.2,</span></span>
<span id="cb189-5017"><a href="#cb189-5017" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             color = "#676767") +</span></span>
<span id="cb189-5018"><a href="#cb189-5018" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text(data = temp_adapt_bioplot_data_asv, </span></span>
<span id="cb189-5019"><a href="#cb189-5019" aria-hidden="true" tabindex="-1"></a>  <span class="co">#          aes(x = CAP1, y = CAP2, label = parameters), size = 2.5,</span></span>
<span id="cb189-5020"><a href="#cb189-5020" aria-hidden="true" tabindex="-1"></a>  <span class="co">#          nudge_x = 0.05, nudge_y = 0.05) +</span></span>
<span id="cb189-5021"><a href="#cb189-5021" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span> </span>
<span id="cb189-5022"><a href="#cb189-5022" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Capscale Analysis"</span>,</span>
<span id="cb189-5023"><a href="#cb189-5023" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">"Temperature Adaptation"</span>,</span>
<span id="cb189-5024"><a href="#cb189-5024" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> cpa1_lab, </span>
<span id="cb189-5025"><a href="#cb189-5025" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> cpa2_lab)</span>
<span id="cb189-5026"><a href="#cb189-5026" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot <span class="ot">&lt;-</span> temp_adapt_plot <span class="sc">+</span> <span class="fu">coord_fixed</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb189-5027"><a href="#cb189-5027" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot</span>
<span id="cb189-5028"><a href="#cb189-5028" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">"files/metadata/figures/its18_temp_adapt_capscale.png"</span>,</span>
<span id="cb189-5029"><a href="#cb189-5029" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">16</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">units =</span> <span class="st">'cm'</span>, <span class="at">res =</span> <span class="dv">600</span>, <span class="at">bg =</span> <span class="st">"white"</span>)</span>
<span id="cb189-5030"><a href="#cb189-5030" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot</span>
<span id="cb189-5031"><a href="#cb189-5031" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb189-5032"><a href="#cb189-5032" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"files/metadata/figures/its18_temp_adapt_capscale.pdf"</span>,</span>
<span id="cb189-5033"><a href="#cb189-5033" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">5</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb189-5034"><a href="#cb189-5034" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot</span>
<span id="cb189-5035"><a href="#cb189-5035" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb189-5036"><a href="#cb189-5036" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-5037"><a href="#cb189-5037" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb189-5038"><a href="#cb189-5038" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5041"><a href="#cb189-5041" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-5042"><a href="#cb189-5042" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-5043"><a href="#cb189-5043" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-5044"><a href="#cb189-5044" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-5045"><a href="#cb189-5045" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5046"><a href="#cb189-5046" aria-hidden="true" tabindex="-1"></a><span class="fu">## Capscale Plots</span></span>
<span id="cb189-5047"><a href="#cb189-5047" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5048"><a href="#cb189-5048" aria-hidden="true" tabindex="-1"></a>::: {.column-body}</span>
<span id="cb189-5049"><a href="#cb189-5049" aria-hidden="true" tabindex="-1"></a>::: {.panel-tabset}</span>
<span id="cb189-5050"><a href="#cb189-5050" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Edaphic properties</span></span>
<span id="cb189-5051"><a href="#cb189-5051" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5054"><a href="#cb189-5054" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-5055"><a href="#cb189-5055" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-5056"><a href="#cb189-5056" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-5057"><a href="#cb189-5057" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb189-5058"><a href="#cb189-5058" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4.5</span></span>
<span id="cb189-5059"><a href="#cb189-5059" aria-hidden="true" tabindex="-1"></a>edaphic_plot</span>
<span id="cb189-5060"><a href="#cb189-5060" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-5061"><a href="#cb189-5061" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig_its("its_cap_edaph_plots")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-5062"><a href="#cb189-5062" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5063"><a href="#cb189-5063" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Soil Functional Response</span></span>
<span id="cb189-5064"><a href="#cb189-5064" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5067"><a href="#cb189-5067" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-5068"><a href="#cb189-5068" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-5069"><a href="#cb189-5069" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-5070"><a href="#cb189-5070" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb189-5071"><a href="#cb189-5071" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4.5</span></span>
<span id="cb189-5072"><a href="#cb189-5072" aria-hidden="true" tabindex="-1"></a>soil_funct_plot</span>
<span id="cb189-5073"><a href="#cb189-5073" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-5074"><a href="#cb189-5074" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig_its("its_cap_funct_plots")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-5075"><a href="#cb189-5075" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5076"><a href="#cb189-5076" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Temperature Adaptation</span></span>
<span id="cb189-5077"><a href="#cb189-5077" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5080"><a href="#cb189-5080" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-5081"><a href="#cb189-5081" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-5082"><a href="#cb189-5082" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-5083"><a href="#cb189-5083" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb189-5084"><a href="#cb189-5084" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4.5</span></span>
<span id="cb189-5085"><a href="#cb189-5085" aria-hidden="true" tabindex="-1"></a>temp_adapt_plot</span>
<span id="cb189-5086"><a href="#cb189-5086" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-5087"><a href="#cb189-5087" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig_its("its_cap_temp_plots")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb189-5088"><a href="#cb189-5088" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb189-5089"><a href="#cb189-5089" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb189-5090"><a href="#cb189-5090" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5093"><a href="#cb189-5093" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-5094"><a href="#cb189-5094" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-5095"><a href="#cb189-5095" aria-hidden="true" tabindex="-1"></a>its18_capscale_plots <span class="ot">&lt;-</span> <span class="fu">ggarrange</span>(</span>
<span id="cb189-5096"><a href="#cb189-5096" aria-hidden="true" tabindex="-1"></a>  edaphic_plot,</span>
<span id="cb189-5097"><a href="#cb189-5097" aria-hidden="true" tabindex="-1"></a>  soil_funct_plot,</span>
<span id="cb189-5098"><a href="#cb189-5098" aria-hidden="true" tabindex="-1"></a>  temp_adapt_plot,</span>
<span id="cb189-5099"><a href="#cb189-5099" aria-hidden="true" tabindex="-1"></a>  <span class="at">ncol =</span> <span class="dv">3</span>, <span class="at">nrow =</span> <span class="dv">1</span>, <span class="at">common.legend =</span> <span class="cn">TRUE</span>, <span class="at">legend =</span> <span class="st">"bottom"</span>)</span>
<span id="cb189-5100"><a href="#cb189-5100" aria-hidden="true" tabindex="-1"></a>its18_capscale_plots</span>
<span id="cb189-5101"><a href="#cb189-5101" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb189-5102"><a href="#cb189-5102" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">"files/metadata/figures/its18_capscale_plots.png"</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">width =</span> <span class="dv">36</span>,</span>
<span id="cb189-5103"><a href="#cb189-5103" aria-hidden="true" tabindex="-1"></a>    <span class="at">units =</span> <span class="st">'cm'</span>, <span class="at">res =</span> <span class="dv">600</span>, <span class="at">bg =</span> <span class="st">"white"</span>)</span>
<span id="cb189-5104"><a href="#cb189-5104" aria-hidden="true" tabindex="-1"></a>its18_capscale_plots</span>
<span id="cb189-5105"><a href="#cb189-5105" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb189-5106"><a href="#cb189-5106" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"files/metadata/figures/its18_capscale_plots.pdf"</span>, <span class="at">height =</span> <span class="dv">6</span>, <span class="at">width =</span> <span class="dv">18</span>)</span>
<span id="cb189-5107"><a href="#cb189-5107" aria-hidden="true" tabindex="-1"></a>its18_capscale_plots</span>
<span id="cb189-5108"><a href="#cb189-5108" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb189-5109"><a href="#cb189-5109" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-5110"><a href="#cb189-5110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5111"><a href="#cb189-5111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5114"><a href="#cb189-5114" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-5115"><a href="#cb189-5115" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-5116"><a href="#cb189-5116" aria-hidden="true" tabindex="-1"></a><span class="do">###########################################</span></span>
<span id="cb189-5117"><a href="#cb189-5117" aria-hidden="true" tabindex="-1"></a><span class="do">######## CODE FOR SUMMARY DATA ############</span></span>
<span id="cb189-5118"><a href="#cb189-5118" aria-hidden="true" tabindex="-1"></a><span class="do">###########################################</span></span>
<span id="cb189-5119"><a href="#cb189-5119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5120"><a href="#cb189-5120" aria-hidden="true" tabindex="-1"></a><span class="do">## Edaphic</span></span>
<span id="cb189-5121"><a href="#cb189-5121" aria-hidden="true" tabindex="-1"></a>bioenv_set <span class="ot">&lt;-</span> edaphic_bioenv_ind_mantel</span>
<span id="cb189-5122"><a href="#cb189-5122" aria-hidden="true" tabindex="-1"></a>envfit_set <span class="ot">&lt;-</span> envfit_edaphic_md</span>
<span id="cb189-5123"><a href="#cb189-5123" aria-hidden="true" tabindex="-1"></a>res_name <span class="ot">&lt;-</span> <span class="st">"its18_edaphic_sig_summary"</span></span>
<span id="cb189-5124"><a href="#cb189-5124" aria-hidden="true" tabindex="-1"></a>metadata_set <span class="ot">&lt;-</span> <span class="st">"edaphic"</span></span>
<span id="cb189-5125"><a href="#cb189-5125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5126"><a href="#cb189-5126" aria-hidden="true" tabindex="-1"></a><span class="do">## BIOENV</span></span>
<span id="cb189-5127"><a href="#cb189-5127" aria-hidden="true" tabindex="-1"></a>bioenv_tab <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-5128"><a href="#cb189-5128" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">names</span>(bioenv_set)) {</span>
<span id="cb189-5129"><a href="#cb189-5129" aria-hidden="true" tabindex="-1"></a>  tmp_r <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(bioenv_set[[i]]<span class="sc">$</span>statistic)</span>
<span id="cb189-5130"><a href="#cb189-5130" aria-hidden="true" tabindex="-1"></a>  <span class="fu">row.names</span>(tmp_r) <span class="ot">&lt;-</span> i</span>
<span id="cb189-5131"><a href="#cb189-5131" aria-hidden="true" tabindex="-1"></a>  tmp_p <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(bioenv_set[[i]]<span class="sc">$</span>signif)</span>
<span id="cb189-5132"><a href="#cb189-5132" aria-hidden="true" tabindex="-1"></a>  <span class="fu">row.names</span>(tmp_p) <span class="ot">&lt;-</span> i</span>
<span id="cb189-5133"><a href="#cb189-5133" aria-hidden="true" tabindex="-1"></a>  bioenv_tab <span class="ot">&lt;-</span> <span class="fu">rbind</span>(bioenv_tab, <span class="fu">data.frame</span>(tmp_r, tmp_p))</span>
<span id="cb189-5134"><a href="#cb189-5134" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-5135"><a href="#cb189-5135" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-5136"><a href="#cb189-5136" aria-hidden="true" tabindex="-1"></a>bioenv_tab <span class="ot">&lt;-</span> bioenv_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"r (bioenv)"</span> <span class="ot">=</span> <span class="dv">1</span>, <span class="st">"pval (bioenv)"</span> <span class="ot">=</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb189-5137"><a href="#cb189-5137" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"parameter"</span>)</span>
<span id="cb189-5138"><a href="#cb189-5138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5139"><a href="#cb189-5139" aria-hidden="true" tabindex="-1"></a><span class="do">## ENVFIT</span></span>
<span id="cb189-5140"><a href="#cb189-5140" aria-hidden="true" tabindex="-1"></a>tmp_r <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(envfit_set<span class="sc">$</span>vectors<span class="sc">$</span>r)</span>
<span id="cb189-5141"><a href="#cb189-5141" aria-hidden="true" tabindex="-1"></a>tmp_r <span class="ot">&lt;-</span> tmp_r <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"parameter"</span>)</span>
<span id="cb189-5142"><a href="#cb189-5142" aria-hidden="true" tabindex="-1"></a>tmp_p <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(envfit_set<span class="sc">$</span>vectors<span class="sc">$</span>pvals)</span>
<span id="cb189-5143"><a href="#cb189-5143" aria-hidden="true" tabindex="-1"></a>tmp_p <span class="ot">&lt;-</span> tmp_p <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"parameter"</span>)</span>
<span id="cb189-5144"><a href="#cb189-5144" aria-hidden="true" tabindex="-1"></a>tmp_rp <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(tmp_r, tmp_p, <span class="at">by =</span> <span class="st">"parameter"</span>)</span>
<span id="cb189-5145"><a href="#cb189-5145" aria-hidden="true" tabindex="-1"></a>tmp_rp <span class="ot">&lt;-</span> tmp_rp <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"r2 (envfit)"</span> <span class="ot">=</span> <span class="dv">2</span>, <span class="st">"Pr(&gt;r) (envfit)"</span> <span class="ot">=</span> <span class="dv">3</span>)</span>
<span id="cb189-5146"><a href="#cb189-5146" aria-hidden="true" tabindex="-1"></a>tmp_rp <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(tmp_rp, <span class="fu">c</span>(tmp_rp<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;r)</span><span class="st">`</span> <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> tmp_rp<span class="sc">$</span>r2 <span class="sc">&gt;</span> <span class="fl">0.4</span>))</span>
<span id="cb189-5147"><a href="#cb189-5147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5148"><a href="#cb189-5148" aria-hidden="true" tabindex="-1"></a>tmp_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(tmp_rp, bioenv_tab, <span class="at">by =</span> <span class="st">"parameter"</span>) </span>
<span id="cb189-5149"><a href="#cb189-5149" aria-hidden="true" tabindex="-1"></a>tmp_summary<span class="sc">$</span>metadata_set <span class="ot">&lt;-</span> metadata_set</span>
<span id="cb189-5150"><a href="#cb189-5150" aria-hidden="true" tabindex="-1"></a>tmp_summary <span class="ot">&lt;-</span> tmp_summary[, <span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)]</span>
<span id="cb189-5151"><a href="#cb189-5151" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(res_name, tmp_summary)</span>
<span id="cb189-5152"><a href="#cb189-5152" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-5153"><a href="#cb189-5153" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-5154"><a href="#cb189-5154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5157"><a href="#cb189-5157" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-5158"><a href="#cb189-5158" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-5159"><a href="#cb189-5159" aria-hidden="true" tabindex="-1"></a><span class="do">## SOIL_FUNCT</span></span>
<span id="cb189-5160"><a href="#cb189-5160" aria-hidden="true" tabindex="-1"></a>bioenv_set <span class="ot">&lt;-</span> soil_funct_bioenv_ind_mantel</span>
<span id="cb189-5161"><a href="#cb189-5161" aria-hidden="true" tabindex="-1"></a>envfit_set <span class="ot">&lt;-</span> envfit_soil_funct_md</span>
<span id="cb189-5162"><a href="#cb189-5162" aria-hidden="true" tabindex="-1"></a>res_name <span class="ot">&lt;-</span> <span class="st">"its18_soil_funct_sig_summary"</span></span>
<span id="cb189-5163"><a href="#cb189-5163" aria-hidden="true" tabindex="-1"></a>metadata_set <span class="ot">&lt;-</span> <span class="st">"functional_response"</span></span>
<span id="cb189-5164"><a href="#cb189-5164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5165"><a href="#cb189-5165" aria-hidden="true" tabindex="-1"></a><span class="do">## BIOENV</span></span>
<span id="cb189-5166"><a href="#cb189-5166" aria-hidden="true" tabindex="-1"></a>bioenv_tab <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-5167"><a href="#cb189-5167" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">names</span>(bioenv_set)) {</span>
<span id="cb189-5168"><a href="#cb189-5168" aria-hidden="true" tabindex="-1"></a>  tmp_r <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(bioenv_set[[i]]<span class="sc">$</span>statistic)</span>
<span id="cb189-5169"><a href="#cb189-5169" aria-hidden="true" tabindex="-1"></a>  <span class="fu">row.names</span>(tmp_r) <span class="ot">&lt;-</span> i</span>
<span id="cb189-5170"><a href="#cb189-5170" aria-hidden="true" tabindex="-1"></a>  tmp_p <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(bioenv_set[[i]]<span class="sc">$</span>signif)</span>
<span id="cb189-5171"><a href="#cb189-5171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">row.names</span>(tmp_p) <span class="ot">&lt;-</span> i</span>
<span id="cb189-5172"><a href="#cb189-5172" aria-hidden="true" tabindex="-1"></a>  bioenv_tab <span class="ot">&lt;-</span> <span class="fu">rbind</span>(bioenv_tab, <span class="fu">data.frame</span>(tmp_r, tmp_p))</span>
<span id="cb189-5173"><a href="#cb189-5173" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-5174"><a href="#cb189-5174" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-5175"><a href="#cb189-5175" aria-hidden="true" tabindex="-1"></a>bioenv_tab <span class="ot">&lt;-</span> bioenv_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"r (bioenv)"</span> <span class="ot">=</span> <span class="dv">1</span>, <span class="st">"pval (bioenv)"</span> <span class="ot">=</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb189-5176"><a href="#cb189-5176" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"parameter"</span>)</span>
<span id="cb189-5177"><a href="#cb189-5177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5178"><a href="#cb189-5178" aria-hidden="true" tabindex="-1"></a><span class="do">## ENVFIT</span></span>
<span id="cb189-5179"><a href="#cb189-5179" aria-hidden="true" tabindex="-1"></a>tmp_r <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(envfit_set<span class="sc">$</span>vectors<span class="sc">$</span>r)</span>
<span id="cb189-5180"><a href="#cb189-5180" aria-hidden="true" tabindex="-1"></a>tmp_r <span class="ot">&lt;-</span> tmp_r <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"parameter"</span>)</span>
<span id="cb189-5181"><a href="#cb189-5181" aria-hidden="true" tabindex="-1"></a>tmp_p <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(envfit_set<span class="sc">$</span>vectors<span class="sc">$</span>pvals)</span>
<span id="cb189-5182"><a href="#cb189-5182" aria-hidden="true" tabindex="-1"></a>tmp_p <span class="ot">&lt;-</span> tmp_p <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"parameter"</span>)</span>
<span id="cb189-5183"><a href="#cb189-5183" aria-hidden="true" tabindex="-1"></a>tmp_rp <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(tmp_r, tmp_p, <span class="at">by =</span> <span class="st">"parameter"</span>)</span>
<span id="cb189-5184"><a href="#cb189-5184" aria-hidden="true" tabindex="-1"></a>tmp_rp <span class="ot">&lt;-</span> tmp_rp <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"r2 (envfit)"</span> <span class="ot">=</span> <span class="dv">2</span>, <span class="st">"Pr(&gt;r) (envfit)"</span> <span class="ot">=</span> <span class="dv">3</span>)</span>
<span id="cb189-5185"><a href="#cb189-5185" aria-hidden="true" tabindex="-1"></a>tmp_rp <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(tmp_rp, <span class="fu">c</span>(tmp_rp<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;r)</span><span class="st">`</span> <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> tmp_rp<span class="sc">$</span>r2 <span class="sc">&gt;</span> <span class="fl">0.4</span>))</span>
<span id="cb189-5186"><a href="#cb189-5186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5187"><a href="#cb189-5187" aria-hidden="true" tabindex="-1"></a>tmp_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(tmp_rp, bioenv_tab, <span class="at">by =</span> <span class="st">"parameter"</span>) </span>
<span id="cb189-5188"><a href="#cb189-5188" aria-hidden="true" tabindex="-1"></a>tmp_summary<span class="sc">$</span>metadata_set <span class="ot">&lt;-</span> metadata_set</span>
<span id="cb189-5189"><a href="#cb189-5189" aria-hidden="true" tabindex="-1"></a>tmp_summary <span class="ot">&lt;-</span> tmp_summary[, <span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)]</span>
<span id="cb189-5190"><a href="#cb189-5190" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(res_name, tmp_summary)</span>
<span id="cb189-5191"><a href="#cb189-5191" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-5192"><a href="#cb189-5192" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-5193"><a href="#cb189-5193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5196"><a href="#cb189-5196" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-5197"><a href="#cb189-5197" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-5198"><a href="#cb189-5198" aria-hidden="true" tabindex="-1"></a><span class="do">## TEMP</span></span>
<span id="cb189-5199"><a href="#cb189-5199" aria-hidden="true" tabindex="-1"></a>bioenv_set <span class="ot">&lt;-</span> temp_adapt_bioenv_ind_mantel</span>
<span id="cb189-5200"><a href="#cb189-5200" aria-hidden="true" tabindex="-1"></a>envfit_set <span class="ot">&lt;-</span> envfit_temp_adapt_md</span>
<span id="cb189-5201"><a href="#cb189-5201" aria-hidden="true" tabindex="-1"></a>res_name <span class="ot">&lt;-</span> <span class="st">"its18_temp_adapt_sig_summary"</span></span>
<span id="cb189-5202"><a href="#cb189-5202" aria-hidden="true" tabindex="-1"></a>metadata_set <span class="ot">&lt;-</span> <span class="st">"temperature adaptation"</span></span>
<span id="cb189-5203"><a href="#cb189-5203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5204"><a href="#cb189-5204" aria-hidden="true" tabindex="-1"></a><span class="do">## BIOENV</span></span>
<span id="cb189-5205"><a href="#cb189-5205" aria-hidden="true" tabindex="-1"></a>bioenv_tab <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-5206"><a href="#cb189-5206" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">names</span>(bioenv_set)) {</span>
<span id="cb189-5207"><a href="#cb189-5207" aria-hidden="true" tabindex="-1"></a>  tmp_r <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(bioenv_set[[i]]<span class="sc">$</span>statistic)</span>
<span id="cb189-5208"><a href="#cb189-5208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">row.names</span>(tmp_r) <span class="ot">&lt;-</span> i</span>
<span id="cb189-5209"><a href="#cb189-5209" aria-hidden="true" tabindex="-1"></a>  tmp_p <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(bioenv_set[[i]]<span class="sc">$</span>signif)</span>
<span id="cb189-5210"><a href="#cb189-5210" aria-hidden="true" tabindex="-1"></a>  <span class="fu">row.names</span>(tmp_p) <span class="ot">&lt;-</span> i</span>
<span id="cb189-5211"><a href="#cb189-5211" aria-hidden="true" tabindex="-1"></a>  bioenv_tab <span class="ot">&lt;-</span> <span class="fu">rbind</span>(bioenv_tab, <span class="fu">data.frame</span>(tmp_r, tmp_p))</span>
<span id="cb189-5212"><a href="#cb189-5212" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-5213"><a href="#cb189-5213" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-5214"><a href="#cb189-5214" aria-hidden="true" tabindex="-1"></a>bioenv_tab <span class="ot">&lt;-</span> bioenv_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"r (bioenv)"</span> <span class="ot">=</span> <span class="dv">1</span>, <span class="st">"pval (bioenv)"</span> <span class="ot">=</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb189-5215"><a href="#cb189-5215" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"parameter"</span>)</span>
<span id="cb189-5216"><a href="#cb189-5216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5217"><a href="#cb189-5217" aria-hidden="true" tabindex="-1"></a><span class="do">## ENVFIT</span></span>
<span id="cb189-5218"><a href="#cb189-5218" aria-hidden="true" tabindex="-1"></a>tmp_r <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(envfit_set<span class="sc">$</span>vectors<span class="sc">$</span>r)</span>
<span id="cb189-5219"><a href="#cb189-5219" aria-hidden="true" tabindex="-1"></a>tmp_r <span class="ot">&lt;-</span> tmp_r <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"parameter"</span>)</span>
<span id="cb189-5220"><a href="#cb189-5220" aria-hidden="true" tabindex="-1"></a>tmp_p <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(envfit_set<span class="sc">$</span>vectors<span class="sc">$</span>pvals)</span>
<span id="cb189-5221"><a href="#cb189-5221" aria-hidden="true" tabindex="-1"></a>tmp_p <span class="ot">&lt;-</span> tmp_p <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"parameter"</span>)</span>
<span id="cb189-5222"><a href="#cb189-5222" aria-hidden="true" tabindex="-1"></a>tmp_rp <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(tmp_r, tmp_p, <span class="at">by =</span> <span class="st">"parameter"</span>)</span>
<span id="cb189-5223"><a href="#cb189-5223" aria-hidden="true" tabindex="-1"></a>tmp_rp <span class="ot">&lt;-</span> tmp_rp <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"r2 (envfit)"</span> <span class="ot">=</span> <span class="dv">2</span>, <span class="st">"Pr(&gt;r) (envfit)"</span> <span class="ot">=</span> <span class="dv">3</span>)</span>
<span id="cb189-5224"><a href="#cb189-5224" aria-hidden="true" tabindex="-1"></a>tmp_rp <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">subset</span>(tmp_rp, <span class="fu">c</span>(tmp_rp<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;r)</span><span class="st">`</span> <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> tmp_rp<span class="sc">$</span>r2 <span class="sc">&gt;</span> <span class="fl">0.4</span>))</span>
<span id="cb189-5225"><a href="#cb189-5225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5226"><a href="#cb189-5226" aria-hidden="true" tabindex="-1"></a>tmp_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(tmp_rp, bioenv_tab, <span class="at">by =</span> <span class="st">"parameter"</span>) </span>
<span id="cb189-5227"><a href="#cb189-5227" aria-hidden="true" tabindex="-1"></a>tmp_summary<span class="sc">$</span>metadata_set <span class="ot">&lt;-</span> metadata_set</span>
<span id="cb189-5228"><a href="#cb189-5228" aria-hidden="true" tabindex="-1"></a>tmp_summary <span class="ot">&lt;-</span> tmp_summary[, <span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)]</span>
<span id="cb189-5229"><a href="#cb189-5229" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(res_name, tmp_summary)</span>
<span id="cb189-5230"><a href="#cb189-5230" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb189-5231"><a href="#cb189-5231" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(bioenv_set, envfit_set, res_name, metadata_set, bioenv_tab)</span>
<span id="cb189-5232"><a href="#cb189-5232" aria-hidden="true" tabindex="-1"></a>its18_sig_summary <span class="ot">&lt;-</span> <span class="fu">rbind</span>(its18_edaphic_sig_summary, its18_soil_funct_sig_summary, its18_temp_adapt_sig_summary)</span>
<span id="cb189-5233"><a href="#cb189-5233" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(its18_edaphic_sig_summary, its18_soil_funct_sig_summary, its18_temp_adapt_sig_summary)</span>
<span id="cb189-5234"><a href="#cb189-5234" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-5235"><a href="#cb189-5235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5238"><a href="#cb189-5238" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-5239"><a href="#cb189-5239" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-5240"><a href="#cb189-5240" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="st">"page_build/metadata_its18_wf.rdata"</span>)</span>
<span id="cb189-5241"><a href="#cb189-5241" aria-hidden="true" tabindex="-1"></a>gdata<span class="sc">::</span><span class="fu">keep</span>(ssu18_sig_summary, its18_sig_summary, <span class="at">sure =</span> <span class="cn">TRUE</span>)</span>
<span id="cb189-5242"><a href="#cb189-5242" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="st">"page_build/metadata_summary.rdata"</span>)</span>
<span id="cb189-5243"><a href="#cb189-5243" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-5244"><a href="#cb189-5244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5247"><a href="#cb189-5247" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-5248"><a href="#cb189-5248" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb189-5249"><a href="#cb189-5249" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-5250"><a href="#cb189-5250" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb189-5251"><a href="#cb189-5251" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-5252"><a href="#cb189-5252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5253"><a href="#cb189-5253" aria-hidden="true" tabindex="-1"></a><span class="fu"># Workflow Output </span></span>
<span id="cb189-5254"><a href="#cb189-5254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5255"><a href="#cb189-5255" aria-hidden="true" tabindex="-1"></a>Data products generated in this workflow can be downloaded from figshare.</span>
<span id="cb189-5256"><a href="#cb189-5256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5257"><a href="#cb189-5257" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;iframe</span> <span class="er">src</span><span class="ot">=</span><span class="st">"https://widgets.figshare.com/articles/16828294/embed?show_title=1"</span> <span class="er">width</span><span class="ot">=</span><span class="st">"100%"</span> <span class="er">height</span><span class="ot">=</span><span class="st">"251"</span> <span class="er">allowfullscreen</span> <span class="er">frameborder</span><span class="ot">=</span><span class="st">"0"</span><span class="kw">&gt;&lt;/iframe&gt;</span></span>
<span id="cb189-5258"><a href="#cb189-5258" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb189-5259"><a href="#cb189-5259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5260"><a href="#cb189-5260" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"da.html"</span> <span class="er">class</span><span class="ot">=</span><span class="st">"btnnav button_round"</span> <span class="er">style</span><span class="ot">=</span><span class="st">"float: left;"</span><span class="kw">&gt;</span>Previous workflow:</span>
<span id="cb189-5261"><a href="#cb189-5261" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br&gt;</span> 7. Differentially Abundant ASVs &amp; Taxa<span class="kw">&lt;/a&gt;&lt;br&gt;&lt;br&gt;</span></span>
<span id="cb189-5262"><a href="#cb189-5262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5265"><a href="#cb189-5265" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-5266"><a href="#cb189-5266" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false </span></span>
<span id="cb189-5267"><a href="#cb189-5267" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb189-5268"><a href="#cb189-5268" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-5269"><a href="#cb189-5269" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-5270"><a href="#cb189-5270" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb189-5271"><a href="#cb189-5271" aria-hidden="true" tabindex="-1"></a><span class="do">### COmmon formatting scripts</span></span>
<span id="cb189-5272"><a href="#cb189-5272" aria-hidden="true" tabindex="-1"></a><span class="do">### </span><span class="al">NOTE</span><span class="do">: captioner.R must be read BEFORE captions_XYZ.R</span></span>
<span id="cb189-5273"><a href="#cb189-5273" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">"assets"</span>, <span class="st">"functions.R"</span>))</span>
<span id="cb189-5274"><a href="#cb189-5274" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb189-5275"><a href="#cb189-5275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5276"><a href="#cb189-5276" aria-hidden="true" tabindex="-1"></a><span class="fu">####  Source Code {.appendix}</span></span>
<span id="cb189-5277"><a href="#cb189-5277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5278"><a href="#cb189-5278" aria-hidden="true" tabindex="-1"></a>The source code for this page can be accessed on GitHub <span class="in">`r fa(name = "github")`</span> by <span class="co">[</span><span class="ot">clicking this link</span><span class="co">]</span>(<span class="in">`r source_code()`</span>). </span>
<span id="cb189-5279"><a href="#cb189-5279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5280"><a href="#cb189-5280" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Data Availability {.appendix}</span></span>
<span id="cb189-5281"><a href="#cb189-5281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5282"><a href="#cb189-5282" aria-hidden="true" tabindex="-1"></a>Data generated in this workflow and the Rdata need to run the workflow can be accessed on figshare at <span class="co">[</span><span class="ot">10.25573/data.16828294</span><span class="co">](https://doi.org/10.25573/data.16828294)</span>.</span>
<span id="cb189-5283"><a href="#cb189-5283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5284"><a href="#cb189-5284" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Last updated on {.appendix}</span></span>
<span id="cb189-5285"><a href="#cb189-5285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-5288"><a href="#cb189-5288" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb189-5289"><a href="#cb189-5289" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb189-5290"><a href="#cb189-5290" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb189-5291"><a href="#cb189-5291" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb189-5292"><a href="#cb189-5292" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
      <div class="nav-footer-center"> Copyright 2022 <a href="https://metacrobe.github.io/">metacrobe</a> &amp; The SWELTR Project. Site constructed using <a href="https://quarto.org/">Quarto</a>.</div>
  </div>
</footer>


</body></html>