---
title: "microeco"
description: |
  An R package for data mining in microbial community ecology
author:
#  - name: Jarrod J Scott
#    url: https://example.com/norajones
#    affiliation: Spacely Sprockets
#    affiliation_nrl: https://example.com/spacelysprokets
bibliography: assets/cite.bib
output:
    distill::distill_article:
      css: assets/styles.css
      toc: true
      toc_depth: 3
---

<details markdown="1">
<summary>Click here for setup information.</summary>


```{r setup}
knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
set.seed(119)
#library(conflicted)
#pacman::p_depends(phangorn, local = TRUE)  
#pacman::p_depends_reverse(phangorn, local = TRUE)  

library(phyloseq); packageVersion("phyloseq")
library(Biostrings); packageVersion("Biostrings")
pacman::p_load(tidyverse, metacoder, hilldiv, patchwork, ampvis2, plyr,
               agricolae, labdsv, naniar, codefolder, pairwiseAdonis, 
               microbiome, seqRFLP, DT, microeco, file2meco, 
               GUniFrac, ggalluvial, ggdendro, tidytree, ggiraph,
               pheatmap, SpiecEasi, Tax4Fun, WGCNA, microbiomeMarker,
               install = FALSE, update = FALSE)
options(scipen=999)
knitr::opts_current$get(c(
  "cache",
  "cache.path",
  "cache.rebuild",
  "dependson",
  "autodep"
))
```
</details>

<br/>

```{r, eval=TRUE, echo=FALSE}
load("page_build/mmarker_wf.rdata")
```

```{r, eval=FALSE, echo=FALSE}
#TEMP LOAD ONLY REMOVE WHEN WORKFLOW FINISHED
remove(list = ls())
load("page_build/da_wf.rdata")
gdata::keep("ssu18_ps_work_rf_all", 
            "ssu18_ps_work_rf_asv", 
            "ssu18_ps_filt_rf_all", 
            "ssu18_ps_filt_rf_asv", 
            "ssu18_ps_perfect_rf_all", 
            "ssu18_ps_perfect_rf_asv", 
            "ssu18_ps_pime_rf_all", 
            "ssu18_ps_pime_rf_asv", 
            "its18_ps_work_rf_all", 
            "its18_ps_work_rf_asv", 
            "its18_ps_filt_rf_all", 
            "its18_ps_filt_rf_asv", 
            "its18_ps_perfect_rf_all", 
            "its18_ps_perfect_rf_asv", 
            "its18_ps_pime_rf_all", 
            "its18_ps_pime_rf_asv", 
            sure = TRUE)
objects()
```

## 16s rRNA

```{r, code_folding=TRUE}
## FIX ps object
ssu_ps <- ssu18_ps_perfect_rf_all
tmp_tax1 <- data.frame(tax_table(ssu_ps))
#tmp_tax1$ASV_SEQ <- NULL
tmp_rn <- row.names(tmp_tax1)
tmp_tax <- data.frame(lapply(tmp_tax1, function(x) {gsub("\\(|)", "", x)}))
row.names(tmp_tax) <- tmp_rn
identical(row.names(tmp_tax), row.names(tmp_tax1))
#dplyr::filter(tmp_tax, Phylum == "Acidobacteriota")
ps_tax_new <- as.matrix(tmp_tax)
tmp_ps <- phyloseq(otu_table(ssu_ps),
                     phy_tree(ssu_ps),
                     tax_table(ps_tax_new),
                     sample_data(ssu_ps))
ssu_ps <- tmp_ps
phyloseq::rank_names(ssu_ps)
data.frame(tax_table(ssu_ps))
```

```{r, code_folding=TRUE}
ssu_group_anova <-  run_test_multiple_groups(ssu_ps, group = "TEMP", 
                                             taxa_rank = "all", 
                                             method = "anova")
ssu_group_anova@marker_table
marker_table(ssu_group_anova)
```

```{r, echo=FALSE}
## Loop through different norms
#norm: 
#default = TSS
#CSS = NONE = RLE = TMM
#meth <- c("none", "rarefy", "TSS", "CLR")
#for (i in meth) {
#  tmp_pht <- run_posthoc_test(ssu_ps, group = "TEMP", method = "tukey", norm = i, transform = "log10")
#  tmp_name <- purrr::map_chr(i, ~ paste0(., "_results"))
#  assign(tmp_name, tmp_pht)
#  rm(list = ls(pattern = "tmp_"))
#}
## Filter results
#filter(data.frame(CLR_results@result), group_name == "k__Bacteria|p__Actinobacteriota|c__Thermoleophilia|o__Gaiellales")
#filter(data.frame(CSS_results@result), group_name == "k__Bacteria|p__Actinobacteriota|c__Thermoleophilia|o__Gaiellales")
#filter(data.frame(none_results@result), group_name == "k__Bacteria|p__Actinobacteriota|c__Thermoleophilia|o__Gaiellales")
#filter(data.frame(rarefy_results@result), group_name == "k__Bacteria|p__Actinobacteriota|c__Thermoleophilia|o__Gaiellales")
#filter(data.frame(RLE_results@result), group_name == "k__Bacteria|p__Actinobacteriota|c__Thermoleophilia|o__Gaiellales")
#filter(data.frame(TMM_results@result), group_name == "k__Bacteria|p__Actinobacteriota|c__Thermoleophilia|o__Gaiellales")
```

```{r, code_folding=TRUE}
ssu_default_pht <- run_posthoc_test(ssu_ps, group = "TEMP", method = "tukey", transform = "log10")
```

```{r, code_folding=TRUE}
filter(data.frame(ssu_default_pht@result), group_name == "k__Bacteria|p__Actinobacteriota|c__Thermoleophilia|o__Gaiellales")
plot_postHocTest(ssu_default_pht, feature = "k__Bacteria|p__Actinobacteriota|c__Thermoleophilia|o__Gaiellales") & theme_bw()
```


```{r, code_folding=TRUE}
ssu_pht <- ssu_default_pht
ssu18_pht_filt <- filter(data.frame(ssu_pht@result), pvalue <= "0.05")[!grepl("ASV", filter(data.frame(ssu_pht@result), pvalue <= "0.05")$group_name),]
ssu18_pht_filt <- ssu18_pht_filt[!grepl("[a-z]__$", ssu18_pht_filt$group_name),]
ssu18_pht_filt <- unique(ssu18_pht_filt$group_name)
length(ssu18_pht_filt)
save.image("page_build/mmarker_wf.rdata")
```


There **`r length(ssu18_pht_filt)`**

<details markdown="1">
<summary>**Click here** to see a list of significantly different lineage markers.</summary>

```{r, eval=TRUE, echo=FALSE}
ssu18_pht_filt
```
</details>

```{r, code_folding=TRUE}
source("hack_code/plot_postHocTest_jjs.R")
environment(plot_postHocTest_jjs) <- asNamespace('microbiomeMarker')
```

```{r, code_folding=TRUE}
ssu_select <- c(
"k__Bacteria|p__Acidobacteriota|c__Acidobacteriae|o__Subgroup_2", 
"k__Bacteria|p__Bacteroidota|c__Bacteroidia|o__Chitinophagales|f__Saprospiraceae", 
"k__Bacteria|p__Bacteroidota|c__Bacteroidia|o__Cytophagales", 
"k__Bacteria|p__Bacteroidota|c__Bacteroidia|o__Flavobacteriales", 
"k__Bacteria|p__Bacteroidota|c__Bacteroidia|o__Sphingobacteriales", 
"k__Bacteria|p__Myxococcota|c__Polyangia|o__mle1-27", 
"k__Bacteria|p__Proteobacteria|c__Gammaproteobacteria|o__Burkholderiales|f__Comamonadaceae|g__Rubrivivax", 
"k__Bacteria|p__Actinobacteriota|c__Acidimicrobiia|o__Microtrichales", 
"k__Bacteria|p__Actinobacteriota|c__Thermoleophilia|o__Gaiellales", 
"k__Bacteria|p__Firmicutes|c__Bacilli|o__Bacillales", 
"k__Bacteria|p__Myxococcota|c__Myxococcia|o__Myxococcales|f__Myxococcaceae|g__Corallococcus", 
"k__Bacteria|p__Proteobacteria|c__Gammaproteobacteria|o__Burkholderiales|f__Burkholderiaceae|g__Ralstonia" 
)
```

```{r, echo=FALSE}
swel_col <- c("#2271B2", "#71B222", "#B22271")
for (i in ssu_select) {
  tmp_select_feat <- i
  tmp_plot <- plot_postHocTest_jjs(ssu_pht, feature = tmp_select_feat) & theme_bw() 
  tmp_plot <- tmp_plot + geom_boxplot(fill = swel_col) + 
    scale_colour_manual(values = c("#191919", "#191919", "#191919")) + 
    geom_point(size = 2, show.legend = FALSE) + 
    ylab("Relative abundance (% total reads)")
  tmp_name <- purrr::map_chr(i, ~ paste0(., "_tax_plot"))
  #tmp_name <- purrr::map_chr(gsub(pattern = "^.*[a-z]\\|", x = i, replacement = ""), ~ paste0(., "_tax_plot"))
  assign(tmp_name, tmp_plot)
  ggplot2::ggsave(tmp_plot, file = paste0("files/da/figures/", i,".png", sep = ""), height = 7, width = 9, units = 'cm', dpi = 600, bg = "white")
  ggplot2::ggsave(tmp_plot, file = paste0("files/da/figures/", i,".pdf", sep = ""), height = 5, width = 6.5)
  rm(list = ls(pattern = "tmp_"))
} 

```

```{r, code_folding=TRUE}
ssu_title <- c(
"Subgroup_2 (Acidobacteriota)", 
"Saprospiraceae (Bacteroidota)", 
"Cytophagales (Bacteroidota)", 
"Flavobacteriales (Bacteroidota)", 
"Sphingobacteriales (Bacteroidota)", 
"mle1-27 (Myxococcota)", 
"Rubrivivax (Proteobacteria)", 
"Microtrichales (Actinobacteriota)", 
"Gaiellales (Actinobacteriota)", 
"Bacillales (Firmicutes)", 
"Corallococcus (Myxococcota)", 
"Ralstonia (Proteobacteria)"
)
```

```{r, echo=FALSE}
tmp_1 <- get(purrr::map_chr(ssu_select[1], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(ssu_title[1])
tmp_2 <- get(purrr::map_chr(ssu_select[2], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(ssu_title[2])
tmp_3 <- get(purrr::map_chr(ssu_select[3], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(ssu_title[3])
   
tmp_4 <- get(purrr::map_chr(ssu_select[4], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(ssu_title[4])
tmp_5 <- get(purrr::map_chr(ssu_select[5], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(ssu_title[5])
tmp_6 <- get(purrr::map_chr(ssu_select[6], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(ssu_title[6])
  
tmp_7 <- get(purrr::map_chr(ssu_select[7], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(ssu_title[7])
tmp_8 <- get(purrr::map_chr(ssu_select[8], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(ssu_title[8])
tmp_9 <- get(purrr::map_chr(ssu_select[9], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(ssu_title[9])

tmp_10 <- get(purrr::map_chr(ssu_select[10], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(ssu_title[10])
tmp_11 <- get(purrr::map_chr(ssu_select[11], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(ssu_title[11])
tmp_12 <- get(purrr::map_chr(ssu_select[12], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(ssu_title[12])

ssu_taxa_combo_plot <- ((tmp_1 + tmp_2 + tmp_3) / (tmp_4 + tmp_5 + tmp_6) / (tmp_7 + tmp_8 + tmp_9) / (tmp_10 + tmp_11 + tmp_12))
ssu_taxa_combo_plot
```

```{r, echo=FALSE}
ggplot2::ggsave(ssu_taxa_combo_plot, path = "files/da/figures/combo_plots", filename =  "ssu_taxa_marker.png", height = 32, width = 36,
    units = 'cm', dpi = 600, bg = "white")
ggplot2::ggsave(ssu_taxa_combo_plot, path = "files/da/figures/combo_plots", filename = "ssu_taxa_marker.pdf", height = 12, width = 12)
```

```{r, echo=FALSE}
system("cp files/da/figures/combo_plots/ssu_taxa_marker.png include/da/ssu_taxa_marker.png")
```

```{r, echo=FALSE, eval=TRUE, warning=FALSE, layout='l-body-outset'}
knitr::include_graphics("include/da/ssu_taxa_marker.png")
```
<small>`r caption_fig_ssu("ssu_taxa_marker_plot")`</small>

## ITS

```{r, code_folding=TRUE}
## FIX ps object
ps <- its18_ps_perfect_rf_all
tmp_tax1 <- data.frame(tax_table(ps))
tmp_tax1$ASV_SEQ <- NULL
tmp_rn <- row.names(tmp_tax1)
tmp_tax <- data.frame(lapply(tmp_tax1, function(x) {gsub("\\(|)", "", x)}))
row.names(tmp_tax) <- tmp_rn
identical(row.names(tmp_tax), row.names(tmp_tax1))
#dplyr::filter(tmp_tax, Phylum == "Acidobacteriota")
ps_tax_new <- as.matrix(tmp_tax)
tmp_ps <- phyloseq(otu_table(ps),
                     tax_table(ps_tax_new),
                     sample_data(ps))
its_ps <- tmp_ps
phyloseq::rank_names(its_ps)
data.frame(tax_table(its_ps))
```


```{r, code_folding=TRUE}
its_group_anova <-  run_test_multiple_groups(its_ps, group = "TEMP", taxa_rank = "all", method = "anova")
its_group_anova@marker_table
marker_table(its_group_anova)
```

```{r, code_folding=TRUE}
its_default_pht <- run_posthoc_test(its_ps, group = "TEMP", method = "tukey", transform = "log10")
```


```{r}
#filter(data.frame(its_default_pht@result), group_name == "k__Bacteria|p__Actinobacteriota|c__Thermoleophilia|o__Gaiellales")
#plot_postHocTest(its_default_pht, feature = "k__Bacteria|p__Actinobacteriota|c__Thermoleophilia|o__Gaiellales") & theme_bw()
```


```{r, code_folding=TRUE}
its_pht <- its_default_pht
its18_pht_filt <- filter(data.frame(its_pht@result), pvalue <= "0.05")[!grepl("ASV", filter(data.frame(its_pht@result), pvalue <= "0.05")$group_name),]
its18_pht_filt <- its18_pht_filt[!grepl("[a-z]__$", its18_pht_filt$group_name),]
its18_pht_filt <- unique(its18_pht_filt$group_name)
length(its18_pht_filt)
```


There **`r length(its18_pht_filt)`**

<details markdown="1">
<summary>**Click here** to see a list of significantly different lineage markers.</summary>

```{r, eval=TRUE, echo=FALSE}
its18_pht_filt
```
</details>

```{r, code_folding=TRUE}
source("hack_code/plot_postHocTest_jjs.R")
environment(plot_postHocTest_jjs) <- asNamespace('microbiomeMarker')
```

```{r, code_folding=TRUE}
its_select <- c(
"k__Fungi|p__Ascomycota|c__Sordariomycetes|o__Xylariales|f__Microdochiaceae", 
"k__Fungi|p__Basidiomycota|c__Agaricomycetes|o__Agaricales|f__Entolomataceae", 
"k__Fungi|p__Basidiomycota|c__Agaricomycetes|o__Agaricales|f__Clavariaceae", 
"k__Fungi|p__Basidiomycota|c__Agaricomycetes|o__Agaricales", 
"k__Fungi|p__Basidiomycota|c__Microbotryomycetes|o__Sporidiobolales", 
"k__Fungi|p__Rozellomycota|c__Rozellomycotina_cls_Incertae_sedis", 
"k__Fungi|p__Ascomycota|c__Eurotiomycetes|o__Eurotiales|f__Trichocomaceae|g__Talaromyces", 
"k__Fungi|p__Ascomycota|c__Pezizomycetes|o__Pezizales|f__Pyronemataceae", 
"k__Fungi|p__Ascomycota|c__Sordariomycetes|o__Hypocreales|f__Nectriaceae|g__Fusarium", 
"k__Fungi|p__Ascomycota|c__Saccharomycetes|o__Saccharomycetales|f__Metschnikowiaceae", 
"k__Fungi|p__Glomeromycota|c__Glomeromycetes|o__Glomerales", 
"k__Fungi|p__Mortierellomycota|c__Mortierellomycetes|o__Mortierellales"
)
```

```{r, echo=FALSE}
swel_col <- c("#2271B2", "#71B222", "#B22271")
for (i in its_select) {
  tmp_select_feat <- i
  tmp_plot <- plot_postHocTest_jjs(its_pht, feature = tmp_select_feat) & theme_bw() 
  tmp_plot <- tmp_plot + geom_boxplot(fill = swel_col) + 
    scale_colour_manual(values = c("#191919", "#191919", "#191919")) + 
    geom_point(size = 2, show.legend = FALSE) + 
    ylab("Relative abundance (% total reads)")
  tmp_name <- purrr::map_chr(i, ~ paste0(., "_tax_plot"))
  #tmp_name <- purrr::map_chr(gsub(pattern = "^.*[a-z]\\|", x = i, replacement = ""), ~ paste0(., "_tax_plot"))
  assign(tmp_name, tmp_plot)
  ggplot2::ggsave(tmp_plot, file = paste0("files/da/figures/", i,".png", sep = ""), height = 7, width = 9, units = 'cm', dpi = 600, bg = "white")
  ggplot2::ggsave(tmp_plot, file = paste0("files/da/figures/", i,".pdf", sep = ""), height = 5, width = 6.5)
  rm(list = ls(pattern = "tmp_"))
} 
```


```{r, code_folding=TRUE}
its_title <- c(
"Microdochiaceae (Ascomycota)", 
"Entolomataceae (Basidiomycota)", 
"Clavariaceae (Basidiomycota)", 
"Agaricales (Basidiomycota)", 
"Sporidiobolales (Basidiomycota)", 
"Rozellomycotina (Rozellomycota)", 
"Talaromyces (Ascomycota)", 
"Pyronemataceae (Ascomycota)", 
"Fusarium (Ascomycota)", 
"Metschnikowiaceae (Ascomycota)", 
"Glomerales (Glomeromycota)", 
"Mortierellales (Mortierellomycota)"
)
```

```{r, echo=FALSE}
tmp_1 <- get(purrr::map_chr(its_select[1], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(its_title[1])
tmp_2 <- get(purrr::map_chr(its_select[2], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(its_title[2])
tmp_3 <- get(purrr::map_chr(its_select[3], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(its_title[3])
   
tmp_4 <- get(purrr::map_chr(its_select[4], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(its_title[4])
tmp_5 <- get(purrr::map_chr(its_select[5], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(its_title[5])
tmp_6 <- get(purrr::map_chr(its_select[6], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(its_title[6])
  
tmp_7 <- get(purrr::map_chr(its_select[7], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(its_title[7])
tmp_8 <- get(purrr::map_chr(its_select[8], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(its_title[8])
tmp_9 <- get(purrr::map_chr(its_select[9], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(its_title[9])

tmp_10 <- get(purrr::map_chr(its_select[10], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(its_title[10])
tmp_11 <- get(purrr::map_chr(its_select[11], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(its_title[11])
tmp_12 <- get(purrr::map_chr(its_select[12], ~ paste0(., "_tax_plot"))) + 
  geom_point(show.legend = FALSE) + ggtitle(its_title[12])

its_taxa_combo_plot <- ((tmp_1 + tmp_2 + tmp_3) / (tmp_4 + tmp_5 + tmp_6) / (tmp_7 + tmp_8 + tmp_9) / (tmp_10 + tmp_11 + tmp_12))
its_taxa_combo_plot
```

```{r, echo=FALSE}
ggplot2::ggsave(its_taxa_combo_plot, path = "files/da/figures/combo_plots", filename =  "its_taxa_marker.png", height = 32, width = 36,
    units = 'cm', dpi = 600, bg = "white")
ggplot2::ggsave(its_taxa_combo_plot, path = "files/da/figures/combo_plots", filename = "its_taxa_marker.pdf", height = 12, width = 12)
```

```{r, echo=FALSE}
system("cp files/da/figures/combo_plots/its_taxa_marker.png include/da/its_taxa_marker.png")
```

```{r, echo=FALSE, eval=TRUE, warning=FALSE, layout='l-body-outset'}
knitr::include_graphics("include/da/ssu_taxa_marker.png")
```
<small>`r caption_fig_its("its_taxa_marker_plot")`</small>

```{r, echo=FALSE}
save.image("page_build/mmarker_wf.rdata")
```
