<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */
</style>

  <!--radix_placeholder_meta_tags-->
  <title>sweltr: 4. Taxonomic Diversity</title>

  <meta property="description" itemprop="description" content="Reproducible workflow for assessing dominant taxonomic groups in the 16S rRNA and ITS data sets."/>

  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>


  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="sweltr: 4. Taxonomic Diversity"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="Reproducible workflow for assessing dominant taxonomic groups in the 16S rRNA and ITS data sets."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="sweltr"/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="sweltr: 4. Taxonomic Diversity"/>
  <meta property="twitter:description" content="Reproducible workflow for assessing dominant taxonomic groups in the 16S rRNA and ITS data sets."/>

  <!--/radix_placeholder_meta_tags-->
  
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","bibliography","output"]}},"value":[{"type":"character","attributes":{},"value":["4. Taxonomic Diversity"]},{"type":"character","attributes":{},"value":["Reproducible workflow for assessing dominant taxonomic groups in the 16S rRNA and ITS data sets.\n"]},{"type":"character","attributes":{},"value":["assets/cite.bib"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["distill::distill_article"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["css","toc","toc_depth"]}},"value":[{"type":"character","attributes":{},"value":["assets/styles.css"]},{"type":"logical","attributes":{},"value":[true]},{"type":"integer","attributes":{},"value":[3]}]}]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }

  </style>

  <script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <script src="site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
  <script src="site_libs/fuse-6.4.1/fuse.min.js"></script>

  <script type="application/javascript">

  function getMeta(metaName) {
    var metas = document.getElementsByTagName('meta');
    for (let i = 0; i < metas.length; i++) {
      if (metas[i].getAttribute('name') === metaName) {
        return metas[i].getAttribute('content');
      }
    }
    return '';
  }

  function offsetURL(url) {
    var offset = getMeta('distill:offset');
    return offset ? offset + '/' + url : url;
  }

  function createFuseIndex() {

    // create fuse index
    var options = {
      keys: [
        { name: 'title', weight: 20 },
        { name: 'categories', weight: 15 },
        { name: 'description', weight: 10 },
        { name: 'contents', weight: 5 },
      ],
      ignoreLocation: true,
      threshold: 0
    };
    var fuse = new window.Fuse([], options);

    // fetch the main search.json
    return fetch(offsetURL('search.json'))
      .then(function(response) {
        if (response.status == 200) {
          return response.json().then(function(json) {
            // index main articles
            json.articles.forEach(function(article) {
              fuse.add(article);
            });
            // download collections and index their articles
            return Promise.all(json.collections.map(function(collection) {
              return fetch(offsetURL(collection)).then(function(response) {
                if (response.status === 200) {
                  return response.json().then(function(articles) {
                    articles.forEach(function(article) {
                      fuse.add(article);
                    });
                  })
                } else {
                  return Promise.reject(
                    new Error('Unexpected status from search index request: ' +
                              response.status)
                  );
                }
              });
            })).then(function() {
              return fuse;
            });
          });

        } else {
          return Promise.reject(
            new Error('Unexpected status from search index request: ' +
                        response.status)
          );
        }
      });
  }

  window.document.addEventListener("DOMContentLoaded", function (event) {

    // get search element (bail if we don't have one)
    var searchEl = window.document.getElementById('distill-search');
    if (!searchEl)
      return;

    createFuseIndex()
      .then(function(fuse) {

        // make search box visible
        searchEl.classList.remove('hidden');

        // initialize autocomplete
        var options = {
          autoselect: true,
          hint: false,
          minLength: 2,
        };
        window.autocomplete(searchEl, options, [{
          source: function(query, callback) {
            const searchOptions = {
              isCaseSensitive: false,
              shouldSort: true,
              minMatchCharLength: 2,
              limit: 10,
            };
            var results = fuse.search(query, searchOptions);
            callback(results
              .map(function(result) { return result.item; })
            );
          },
          templates: {
            suggestion: function(suggestion) {
              var img = suggestion.preview && Object.keys(suggestion.preview).length > 0
                ? `<img src="${offsetURL(suggestion.preview)}"</img>`
                : '';
              var html = `
                <div class="search-item">
                  <h3>${suggestion.title}</h3>
                  <div class="search-item-description">
                    ${suggestion.description || ''}
                  </div>
                  <div class="search-item-preview">
                    ${img}
                  </div>
                </div>
              `;
              return html;
            }
          }
        }]).on('autocomplete:selected', function(event, suggestion) {
          window.location.href = offsetURL(suggestion.path);
        });
        // remove inline display style on autocompleter (we want to
        // manage responsive display via css)
        $('.algolia-autocomplete').css("display", "");
      })
      .catch(function(error) {
        console.log(error);
      });

  });

  </script>

  <style type="text/css">

  .nav-search {
    font-size: x-small;
  }

  /* Algolioa Autocomplete */

  .algolia-autocomplete {
    display: inline-block;
    margin-left: 10px;
    vertical-align: sub;
    background-color: white;
    color: black;
    padding: 6px;
    padding-top: 8px;
    padding-bottom: 0;
    border-radius: 6px;
    border: 1px #0F2E3D solid;
    width: 180px;
  }


  @media screen and (max-width: 768px) {
    .distill-site-nav .algolia-autocomplete {
      display: none;
      visibility: hidden;
    }
    .distill-site-nav.responsive .algolia-autocomplete {
      display: inline-block;
      visibility: visible;
    }
    .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
      margin-left: 0;
      width: 400px;
      max-height: 400px;
    }
  }

  .algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
    width: 90%;
    outline: none;
    border: none;
  }

  .algolia-autocomplete .aa-hint {
    color: #999;
  }
  .algolia-autocomplete .aa-dropdown-menu {
    width: 550px;
    max-height: 70vh;
    overflow-x: visible;
    overflow-y: scroll;
    padding: 5px;
    margin-top: 3px;
    margin-left: -150px;
    background-color: #fff;
    border-radius: 5px;
    border: 1px solid #999;
    border-top: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
    cursor: pointer;
    padding: 5px 4px;
    border-bottom: 1px solid #eee;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
    border-bottom: none;
    margin-bottom: 2px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
    overflow: hidden;
    font-size: 0.8em;
    line-height: 1.4em;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
    font-size: 1rem;
    margin-block-start: 0;
    margin-block-end: 5px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
    display: inline-block;
    overflow: hidden;
    height: 2.8em;
    width: 80%;
    margin-right: 4%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
    display: inline-block;
    width: 15%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
    height: 3em;
    width: auto;
    display: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
    display: initial;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
    background-color: #eee;
  }
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
    font-weight: bold;
    font-style: normal;
  }

  </style>


  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure img {
    width: 100%;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refHtml = $('#ref-' + $(this).attr('data-cites')).html();
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
  <link href="site_libs/panelset-0.2.6/panelset.css" rel="stylesheet" />
  <script src="site_libs/panelset-0.2.6/panelset.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Inconsolata|Source+Sans+Pro|Work+Sans:200,300,400,500|Karla:400,400i,700,700i');
  /********************************/
  /* controls general properties of site*/
  /********************************/

  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    background-color: #FEFEFE;
  }

  d-title h1{
  font-size: 2.1em;
  font-family: 'Work Sans';
  font-weight: 300;
  }

  d-article h1 {
    font-size: 1.8em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 300;
  }

  d-article h2 {
    font-size: 1.65em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 500;
  }

  d-article h3 {
    margin-top: 1em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
  }

  d-byline .authors-affiliations {
    grid-column-end: span 3;
    grid-column-gap: 8px;
  }

  img.orcid-id {
    margin: 0;
  }

  a.name:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  a.affiliation:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  blockquote p {
  	font-size: 1em;
  	font-family: 'Work Sans';
    font-style: normal;
    margin: 0.5em 0 0.5em;
  }

  d-article li {
    margin: 0px 0px 4px;
    font-size: 18px;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
  }

  /********************************/
  /* controls CODE*/
  /********************************/

  /** pre.sourceCode controls code block without lang **/

  /** Controls r code  **/

  pre.sourceCode.r {
    margin-top: 0px;
     color: #333333;
     background-color: #E8E8E8;
     border-radius: 3px;
     font-size: 0.80em;
     font-family: 'Monaco', 'PT Mono', monospace;
     overflow-x: scroll !important;
     border-left: 6px solid #2271B2;
     border-top: 1px solid #B7B7B7;
     border-right: 1px solid #B7B7B7;
     border-bottom: 1px solid #B7B7B7;
     padding: 18px 0px 18px 20px;
   }

   pre.sourceCode.bash {
     margin-top: 0px;
      color: #333333;
      background-color: #E8E8E8;
      border-radius: 3px;
      font-size: 0.80em;
      font-family: 'Monaco', 'PT Mono', monospace;
      overflow-x: scroll !important;
      border-left: 6px solid #B22271;
      border-top: 1px solid #B7B7B7;
      border-right: 1px solid #B7B7B7;
      border-bottom: 1px solid #B7B7B7;
      padding: 18px 0px 18px 20px;
    }
  /** Controls code output **/

  d-article pre {
     margin-top: 0px;
      color: #333333;
      background-color: #FAFAFA;
      font-size: 0.8em;
      border: 1px solid #ccc;
      overflow-x: scroll !important;
      padding:  10px 10px 10px 18px;
      border-radius: 5px;

    }

  /** Controls inline code  **/

  p code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }

  li code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }

  d-article aside code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-family: 'Monaco', 'PT Mono', monospace;
    color: #BF2727;
  }

  /********************************/
  /* controls NAV BAR and HEADERS*/
  /********************************/

  .nav-dropdown-content .nav-dropdown-header {
    text-transform: none;
    color: #686868;
  }
  .nav-dropdown-content a {
    font-family: 'Work Sans'
  }
  .nav-dropdown .nav-dropbtn {
    color: #FEFEFE;
    font-weight: 400;
  }

  .distill-site-header .logo img {
    margin-top: 10px;
    max-height: 30px;
    margin-bottom: 10px;
  }

  .distill-site-nav {
      background-color: #113858;
      font-size: 16px;
      font-weight: 300;
  }

  img {
    margin: 5px;
  }

  /** LIGHTBOX MARKUP **/
  .thumbnail {
    max-width: 100%;
    position: relative;
  }

  /** DETAILS fold down**/

  details {
    background-color: #f6f6f6;
    cursor: pointer;
    color: #2d2d2d;
    margin-bottom:1vh;
    padding: 1em;
    border: none;
    text-align: left;
    outline: none;
  }

  .active, details:hover {
    transition: max-height 0.2s ease-out;
    outline: none;

    /* background-color: #acc2c2; */
  }


  details {
    display: flex;
    justify-content: space-between;
    align-content: center;
    border-radius: 4px;
    outline: none;

  }

  details.active {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    outline: none;

  }

  summary::-webkit-details-marker {
    display: none;
    outline: none;

  }
  summary:focus {
    outline: none;
  }
  summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f0fe";
  }

  details[open] summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f146";
  }

  /********************************/
  /* controls downloadthis  */
  /********************************/

  button.btn.btn-default {
    font-size: 0.8em;
    font-family: "PT Sans";
    border-color: #B22271;
    border-width: medium;
    margin-bottom: 15px;
  }

  /********************************/
  /* controls reactable  */
  /********************************/

  div.rt-text-content {
    font-size: 1.2em;
  }

  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"4. Taxonomic Diversity","description":"Reproducible workflow for assessing dominant taxonomic groups in the 16S rRNA and ITS data sets.","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title"></a>
<a href="index.html">
<i class="fa fa-home" aria-hidden="true"></i>
</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
A. Processing Workflows
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="dada2.html">1. DADA2</a>
<a href="data-prep.html">2. Data Set Prep</a>
<a href="filtering.html">3. Filtering</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
B. Diversity Workflows
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="taxa.html">4. Taxonomic</a>
<a href="alpha.html">5. Alpha Diversity</a>
<a href="beta.html">6. Beta Diversity</a>
<a href="da.html">7. DA ASVs &amp; Taxa</a>
<a href="metadata.html">8. Metadata</a>
</div>
</div>
<input id="distill-search" class="nav-search hidden" type="text" placeholder="Search..."/>
</div>
<div class="nav-right">
<a href="data-availability.html">Data Products</a>
<a href="som.html">SOM</a>
<a href="https://github.com/sweltr/high-temp/">
<i class="fa fa-github fa-lg" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>4. Taxonomic Diversity</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->
<p><p>Reproducible workflow for assessing dominant taxonomic groups in the 16S rRNA and ITS data sets.</p></p>
</div>


<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#synopsis">Synopsis</a></li>
<li><a href="#taxonomic-assessment">Taxonomic Assessment</a>
<ul>
<li><a href="#s-rrna">16s rRNA</a></li>
<li><a href="#its">ITS</a></li>
</ul></li>
<li><a href="#major-bacterial-taxa-by-family">Major Bacterial taxa by Family</a>
<ul>
<li><a href="#source-code">Source Code</a></li>
</ul></li>
</ul>
</nav>
</div>
<details>
<summary>
<strong>Click here</strong> for setup information.
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">echo =</span> <span class="cn">TRUE</span>, <span class="at">eval =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">119</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#library(conflicted)</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends(phangorn, local = TRUE)  </span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends_reverse(phangorn, local = TRUE)  </span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq); <span class="fu">packageVersion</span>(<span class="st">&quot;phyloseq&quot;</span>)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings); <span class="fu">packageVersion</span>(<span class="st">&quot;Biostrings&quot;</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(tidyverse, metacoder, hilldiv, patchwork, codefolder,</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>               agricolae, labdsv, naniar, codefolder, pairwiseAdonis, </span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>               microbiome, seqRFLP, DT, microeco, file2meco, cowplot,</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>               GUniFrac, ggalluvial, ggdendro, tidytree, igraph, reactable,</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>               pheatmap, SpiecEasi, Tax4Fun, WGCNA, microbiomeMarker,</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>               captioner, downloadthis,</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>               <span class="at">install =</span> <span class="cn">FALSE</span>, <span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">scipen=</span><span class="dv">999</span>)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_current<span class="sc">$</span><span class="fu">get</span>(<span class="fu">c</span>(</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache&quot;</span>,</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.path&quot;</span>,</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.rebuild&quot;</span>,</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;dependson&quot;</span>,</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;autodep&quot;</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>))</span></code></pre></div>
</div>
</details>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="synopsis">Synopsis</h1>
<p>This workflow contains taxonomic diversity assessments for the 16S rRNA and ITS data sets. In order to run the workflow, you either need to first run the <a href="dada2.html">DADA2 Workflow</a>, then the <a href="data-prep.html">Data Preparation workflow</a>, and finally the <a href="filtering.html">Filtering workflow</a>.</p>
<p>In this workflow, we use the <a href="https://github.com/ChiLiubio/microeco">microeco</a> to look at the taxonomic distribution of microbial communities.</p>
<h1 id="taxonomic-assessment">Taxonomic Assessment</h1>
<h2 id="s-rrna">16s rRNA</h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Here we compare the taxonomic breakdown of the Full (unfiltered), Arbitrary filtered, PERfect filtered, and PIME filtered data sets, split by temperature treatment.</p>
<p>Before we plot the data, we want to separate out the Proteobacteria classes so we can plot these along with other phyla. To accomplish this we perform the following steps:</p>
<ol type="1">
<li>Get all Class-level Proteobacteria names</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>ssu18_data_sets <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ssu18_ps_work&quot;</span>,  <span class="st">&quot;ssu18_ps_filt&quot;</span>,</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>                     <span class="st">&quot;ssu18_ps_perfect&quot;</span>, <span class="st">&quot;ssu18_ps_pime&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_proteo&quot;</span>))</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>     tmp_df <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(tmp_get, Phylum <span class="sc">==</span> <span class="st">&quot;Proteobacteria&quot;</span>)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_df)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_name)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>     tmp_get_taxa <span class="ot">&lt;-</span> <span class="fu">get_taxa_unique</span>(tmp_df,</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">taxonomic.rank =</span> <span class="fu">rank_names</span>(tmp_df)[<span class="dv">3</span>],</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">errorIfNULL=</span><span class="cn">TRUE</span>)</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_get_taxa)</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;_proteo&quot;</span>))</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<ol start="2" type="1">
<li>Replace Phylum Proteobacteria with the Class name.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(j, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_proteo_clean&quot;</span>))</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(j)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  tmp_clean <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(tmp_get))</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>   <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(tmp_clean)){</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>       <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">&quot;Proteobacteria&quot;</span> <span class="sc">&amp;</span> tmp_clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">&quot;Alphaproteobacteria&quot;</span>){</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>           phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;Alphaproteobacteria&quot;</span>)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>           tmp_clean[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>   }   <span class="cf">else</span> <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">&quot;Proteobacteria&quot;</span> <span class="sc">&amp;</span> tmp_clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">&quot;Gammaproteobacteria&quot;</span>){</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>           phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;Gammaproteobacteria&quot;</span>)</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>           tmp_clean[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>   }   <span class="cf">else</span> <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">&quot;Proteobacteria&quot;</span> <span class="sc">&amp;</span> tmp_clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">&quot;Zetaproteobacteria&quot;</span>){</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>              phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;Zetaproteobacteria&quot;</span>)</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>           tmp_clean[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>   }   <span class="cf">else</span> <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">&quot;Proteobacteria&quot;</span> <span class="sc">&amp;</span> tmp_clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">&quot;p_Proteobacteria&quot;</span>){</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>           phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;p_Proteobacteria&quot;</span>)</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>           tmp_clean[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>       }</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tax_table</span>(tmp_get) <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_clean)</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rank_names</span>(tmp_get)</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_get)</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">c</span>(tmp_name, tmp_get))</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(<span class="fu">get_taxa_unique</span>(tmp_get,</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>                               <span class="at">taxonomic.rank =</span> <span class="fu">rank_names</span>(tmp_get)[<span class="dv">2</span>],</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>                               <span class="at">errorIfNULL=</span><span class="cn">TRUE</span>)))</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>  tmp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&quot;files/taxa/rdata/&quot;</span>)</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(tmp_get, <span class="fu">paste</span>(tmp_path, j, <span class="st">&quot;_clean.rds&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(class, order, phylum)</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern=</span><span class="st">&quot;_proteo_clean&quot;</span>)</span></code></pre></div>
</details>
</div>
<ol start="3" type="1">
<li>In order to use <code>microeco</code>, we need to add the rank designation as a prefix to each taxa. For example, <code>Actinobacteriota</code> is changed to <code>p__Actinobacteriota</code>.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_proteo_clean&quot;</span>)))</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#tmp_get &lt;- get(i)</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="co">#tmp_path &lt;- file.path(&quot;files/alpha/rdata/&quot;)</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="co">#tmp_read &lt;- readRDS(paste(tmp_path, i, &quot;.rds&quot;, sep = &quot;&quot;))</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  tmp_sam_data <span class="ot">&lt;-</span> <span class="fu">sample_data</span>(tmp_get)</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(tmp_get))</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Phylum <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;p_Proteobacteria&quot;</span>, <span class="st">&quot;Proteobacteria&quot;</span>, tmp_tax_data<span class="sc">$</span>Phylum)</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  <span class="co">#tmp_tax_data[,c(1:6)]</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>ASV_ID <span class="ot">&lt;-</span> <span class="cn">NULL</span> <span class="co"># Some have, some do not</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>ASV_SEQ <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data[] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">lapply</span>(tmp_tax_data, gsub, </span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">pattern =</span> <span class="st">&quot;^[k | p | c | o | f]_.*&quot;</span>, </span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">replacement =</span> <span class="st">&quot;&quot;</span>, <span class="at">fixed =</span> <span class="cn">FALSE</span>))</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Kingdom <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;k__&quot;</span>, tmp_tax_data<span class="sc">$</span>Kingdom, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Phylum <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;p__&quot;</span>, tmp_tax_data<span class="sc">$</span>Phylum, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Class <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;c__&quot;</span>, tmp_tax_data<span class="sc">$</span>Class, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Order <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;o__&quot;</span>, tmp_tax_data<span class="sc">$</span>Order, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Family <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;f__&quot;</span>, tmp_tax_data<span class="sc">$</span>Family, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Genus <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;g__&quot;</span>, tmp_tax_data<span class="sc">$</span>Genus, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_tax_data)</span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>  tmp_ps <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(<span class="fu">otu_table</span>(tmp_get),</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">phy_tree</span>(tmp_get),</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">tax_table</span>(tmp_tax_data),</span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>                     tmp_sam_data)</span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(i, tmp_ps)</span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;_proteo_clean&quot;</span>))</span></code></pre></div>
</details>
</div>
<ol start="4" type="1">
<li>Next, we need to covert each <code>phyloseq object</code> to a <code>microtable class</code>. The microtable class is the basic data structure for the <code>microeco</code> package and designed to store basic information from all the downstream analyses (e.g, alpha diversity, beta diversity, etc.). We use the <a href="https://github.com/ChiLiubio/file2meco">file2meco</a> to read the phyloseq object and convert into a microtable object. We can add <code>_me</code> as a suffix to each object to distiguish it from its phyloseq counterpart.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>    tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>    tmp_otu_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(tmp_get)))</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>    tmp_sample_info <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>    tmp_taxonomy_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(tmp_get))</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>    tmp_phylo_tree <span class="ot">&lt;-</span> <span class="fu">phy_tree</span>(tmp_get)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>    tmp_taxonomy_table <span class="sc">%&lt;&gt;%</span> tidy_taxonomy</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>    tmp_dataset <span class="ot">&lt;-</span> microtable<span class="sc">$</span><span class="fu">new</span>(<span class="at">sample_table =</span> tmp_sample_info, </span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">otu_table =</span> tmp_otu_table, </span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">tax_table =</span> tmp_taxonomy_table, </span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">phylo_tree =</span> tmp_phylo_tree)</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>    tmp_dataset<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(tmp_dataset)</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>    tmp_dataset<span class="sc">$</span>tax_table <span class="sc">%&lt;&gt;%</span> base<span class="sc">::</span><span class="fu">subset</span>(Kingdom <span class="sc">==</span> <span class="st">&quot;k__Archaea&quot;</span> <span class="sc">|</span> Kingdom <span class="sc">==</span> <span class="st">&quot;k__Bacteria&quot;</span>)</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(tmp_dataset)</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>    tmp_dataset<span class="sc">$</span><span class="fu">filter_pollution</span>(<span class="at">taxa =</span> <span class="fu">c</span>(<span class="st">&quot;mitochondria&quot;</span>, <span class="st">&quot;chloroplast&quot;</span>))</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(tmp_dataset)</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>    tmp_dataset<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(tmp_dataset)</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>    tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_me&quot;</span>)) </span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">assign</span>(tmp_name, tmp_dataset)</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>}  </span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span></code></pre></div>
</details>
</div>
<p>Here is an example of what a new microeco object looks like when you call it.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_me</span></code></pre></div>
<pre><code>microtable class:
sample_table have 15 rows and 6 columns
otu_table have 20173 rows and 15 columns
tax_table have 20173 rows and 6 columns
phylo_tree have 20173 tips
Taxa abundance: calculated for Kingdom,Phylum,Class,Order,Family,Genus </code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<ol start="5" type="1">
<li>Now, we calculate the taxa abundance at each taxonomic rank using <code>cal_abund()</code>. This function return a list called <code>taxa_abund</code> containing several data frame of the abundance information at each taxonomic rank. The list is stored in the microtable object automatically. Its worth noting that the <code>cal_abund()</code> function can be used to solve some complex cases, such as supporting both the relative and absolute abundance calculation and selecting the partial taxonomic columns. More information can be found in the description of the <a href="https://github.com/ChiLiubio/file2meco">file2meco package</a>. In the same loop we also create a <code>trans_abund</code> class, which is used to transform taxonomic abundance data for plotting.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  tmp_me <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_me&quot;</span>)))</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  tmp_me<span class="sc">$</span><span class="fu">cal_abund</span>()</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  tmp_me_abund <span class="ot">&lt;-</span> trans_abund<span class="sc">$</span><span class="fu">new</span>(<span class="at">dataset =</span> tmp_me, <span class="at">taxrank =</span> <span class="st">&quot;Phylum&quot;</span>, <span class="at">ntaxa =</span> <span class="dv">12</span>)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  tmp_me_abund<span class="sc">$</span>abund_data<span class="sc">$</span>Abundance <span class="ot">&lt;-</span> tmp_me_abund<span class="sc">$</span>abund_data<span class="sc">$</span>Abundance <span class="sc">/</span> <span class="dv">100</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  tmp_me_abund_gr <span class="ot">&lt;-</span> trans_abund<span class="sc">$</span><span class="fu">new</span>(<span class="at">dataset =</span> tmp_me, <span class="at">taxrank =</span> <span class="st">&quot;Phylum&quot;</span>, <span class="at">ntaxa =</span> <span class="dv">12</span>, </span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">groupmean =</span> <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  tmp_me_abund_gr<span class="sc">$</span>abund_data<span class="sc">$</span>Abundance <span class="ot">&lt;-</span> tmp_me_abund_gr<span class="sc">$</span>abund_data<span class="sc">$</span>Abundance <span class="sc">/</span> <span class="dv">100</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_me_abund&quot;</span>)) </span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_me_abund)  </span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  tmp_name_gr <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_me_abund_group&quot;</span>)) </span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name_gr, tmp_me_abund_gr)  </span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span></code></pre></div>
</details>
</div>
<pre><code>The result is stored in object$taxa_abund ...</code></pre>
<ol start="6" type="1">
<li>I prefer to specify the order of taxa in these kinds of plots. We can look the top <code>ntaxa</code> (defined above) by accessing the <code>use_taxanames</code> character vector of each microtable object.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Hum. Similar but not the same. This means we need to define the order for each separately. Once we do that, we will override the <code>use_taxanames</code> character vectors with the reordered vectors.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_tax_ord <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Alphaproteobacteria&quot;</span>, <span class="st">&quot;Gammaproteobacteria&quot;</span>, <span class="st">&quot;Acidobacteriota&quot;</span>, </span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>                           <span class="st">&quot;Actinobacteriota&quot;</span>, <span class="st">&quot;Bacteroidota&quot;</span>, <span class="st">&quot;Firmicutes&quot;</span>, <span class="st">&quot;Myxococcota&quot;</span>, </span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>                           <span class="st">&quot;Verrucomicrobiota&quot;</span>, <span class="st">&quot;Chloroflexi&quot;</span>, <span class="st">&quot;Planctomycetota&quot;</span>, </span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>                           <span class="st">&quot;Methylomirabilota&quot;</span>, <span class="st">&quot;Crenarchaeota&quot;</span>)</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>ssu18_ps_filt_tax_ord <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Alphaproteobacteria&quot;</span>, <span class="st">&quot;Gammaproteobacteria&quot;</span>, <span class="st">&quot;Acidobacteriota&quot;</span>, </span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>                           <span class="st">&quot;Actinobacteriota&quot;</span>, <span class="st">&quot;Bacteroidota&quot;</span>, <span class="st">&quot;Firmicutes&quot;</span>, <span class="st">&quot;Myxococcota&quot;</span>, </span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>                           <span class="st">&quot;Verrucomicrobiota&quot;</span>, <span class="st">&quot;RCP2-54&quot;</span>, <span class="st">&quot;Planctomycetota&quot;</span>, </span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>                           <span class="st">&quot;Methylomirabilota&quot;</span>, <span class="st">&quot;Crenarchaeota&quot;</span>)</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_tax_ord <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Alphaproteobacteria&quot;</span>, <span class="st">&quot;Gammaproteobacteria&quot;</span>, <span class="st">&quot;Acidobacteriota&quot;</span>, </span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;Actinobacteriota&quot;</span>, <span class="st">&quot;Bacteroidota&quot;</span>, <span class="st">&quot;Firmicutes&quot;</span>, <span class="st">&quot;Myxococcota&quot;</span>, </span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;Verrucomicrobiota&quot;</span>, <span class="st">&quot;RCP2-54&quot;</span>, <span class="st">&quot;Planctomycetota&quot;</span>, </span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;Methylomirabilota&quot;</span>, <span class="st">&quot;Crenarchaeota&quot;</span>)</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_tax_ord <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Alphaproteobacteria&quot;</span>, <span class="st">&quot;Gammaproteobacteria&quot;</span>, <span class="st">&quot;Acidobacteriota&quot;</span>, </span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>                           <span class="st">&quot;Actinobacteriota&quot;</span>, <span class="st">&quot;Bacteroidota&quot;</span>, <span class="st">&quot;Firmicutes&quot;</span>, <span class="st">&quot;Myxococcota&quot;</span>, </span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>                           <span class="st">&quot;Verrucomicrobiota&quot;</span>, <span class="st">&quot;RCP2-54&quot;</span>, <span class="st">&quot;Nitrospirota&quot;</span>, </span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>                           <span class="st">&quot;Methylomirabilota&quot;</span>, <span class="st">&quot;Crenarchaeota&quot;</span>)</span></code></pre></div>
</details>
</div>
<ol start="7" type="1">
<li>And one more little step before plotting. Here we <strong>a</strong>) specify a custom color palette and <strong>b</strong>) specify the sample order.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>top_level <span class="ot">&lt;-</span> <span class="st">&quot;Phylum&quot;</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>ssu18_colvec.tax <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#00463C&quot;</span>,<span class="st">&quot;#FFD5FD&quot;</span>,<span class="st">&quot;#00A51C&quot;</span>,<span class="st">&quot;#C80B2A&quot;</span>,<span class="st">&quot;#00C7F9&quot;</span>,<span class="st">&quot;#FFA035&quot;</span>,</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>                      <span class="st">&quot;#ED0DFD&quot;</span>,<span class="st">&quot;#0063E5&quot;</span>,<span class="st">&quot;#5FFFDE&quot;</span>,<span class="st">&quot;#C00B6F&quot;</span>,<span class="st">&quot;#00A090&quot;</span>,<span class="st">&quot;#FF95BA&quot;</span>)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>ssu18_samp_order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;P02_D00_010_C0A&quot;</span>, <span class="st">&quot;P04_D00_010_C0B&quot;</span>, <span class="st">&quot;P06_D00_010_C0C&quot;</span>, <span class="st">&quot;P08_D00_010_C0D&quot;</span>, <span class="st">&quot;P10_D00_010_C0E&quot;</span>, </span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;P01_D00_010_W3A&quot;</span>, <span class="st">&quot;P03_D00_010_W3B&quot;</span>, <span class="st">&quot;P05_D00_010_W3C&quot;</span>, <span class="st">&quot;P07_D00_010_W3D&quot;</span>, <span class="st">&quot;P09_D00_010_W3E&quot;</span>, </span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;P01_D00_010_W8A&quot;</span>, <span class="st">&quot;P03_D00_010_W8B&quot;</span>, <span class="st">&quot;P05_D00_010_W8C&quot;</span>, <span class="st">&quot;P07_D00_010_W8D&quot;</span>, <span class="st">&quot;P09_D00_010_W8E&quot;</span>)</span></code></pre></div>
</details>
</div>
<ol start="8" type="1">
<li>Now we can generate plots (in a loop) for each faceted data set.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  tmp_abund <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_me_abund&quot;</span>)))</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  tmp_tax_order <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_tax_ord&quot;</span>)))</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  tmp_abund<span class="sc">$</span>use_taxanames <span class="ot">&lt;-</span> tmp_tax_order</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>  tmp_facet_plot <span class="ot">&lt;-</span> tmp_abund<span class="sc">$</span><span class="fu">plot_bar</span>(<span class="at">use_colors =</span> ssu18_colvec.tax, </span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">others_color =</span> <span class="st">&quot;#323232&quot;</span>, </span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">facet =</span> <span class="st">&quot;TEMP&quot;</span>, </span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_keep =</span> <span class="cn">TRUE</span>, </span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_type_hor =</span> <span class="cn">FALSE</span>, </span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">legend_text_italic =</span> <span class="cn">FALSE</span>, </span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_size =</span> <span class="dv">6</span>, </span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">facet_color =</span> <span class="st">&quot;#cccccc&quot;</span>, </span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">order_x =</span> ssu18_samp_order)</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>  tmp_facet_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_facet_plot&quot;</span>)) </span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_facet_name, tmp_facet_plot)  </span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<p>Then add a little formatting to the faceted plots.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>set_to_plot <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ssu18_ps_work_facet_plot&quot;</span>, <span class="st">&quot;ssu18_ps_filt_facet_plot&quot;</span>, <span class="st">&quot;ssu18_ps_perfect_facet_plot&quot;</span>, <span class="st">&quot;ssu18_ps_pime_facet_plot&quot;</span>)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span>  set_to_plot) {</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>    tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>    tmp_get <span class="ot">&lt;-</span> tmp_get <span class="sc">+</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> top_level, <span class="at">reverse =</span> <span class="cn">FALSE</span>,</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>                               <span class="at">keywidth =</span> <span class="fl">0.7</span>, <span class="at">keyheight =</span> <span class="fl">0.7</span>)) <span class="sc">+</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">&quot;Sample&quot;</span>) <span class="sc">+</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>),</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>, <span class="at">angle =</span> <span class="dv">90</span>),</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">0</span>),</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>      <span class="fu">scale_y_continuous</span>()</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(i, tmp_get)</span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>)) </span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<p>And now plots for the groupmeans sets. We can use the same taxa order since that should not have changed.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>set_to_plot <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ssu18_ps_work_group_plot&quot;</span>, <span class="st">&quot;ssu18_ps_filt_group_plot&quot;</span>, <span class="st">&quot;ssu18_ps_perfect_group_plot&quot;</span>, <span class="st">&quot;ssu18_ps_pime_group_plot&quot;</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  tmp_abund <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_me_abund_group&quot;</span>)))</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  tmp_tax_order <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_tax_ord&quot;</span>)))</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  tmp_abund<span class="sc">$</span>use_taxanames <span class="ot">&lt;-</span> tmp_tax_order</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  tmp_group_plot <span class="ot">&lt;-</span> tmp_abund<span class="sc">$</span><span class="fu">plot_bar</span>(<span class="at">use_colors =</span> ssu18_colvec.tax, </span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">others_color =</span> <span class="st">&quot;#323232&quot;</span>, </span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_keep =</span> <span class="cn">TRUE</span>, </span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_type_hor =</span> <span class="cn">TRUE</span>, </span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">legend_text_italic =</span> <span class="cn">FALSE</span>, </span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_size =</span> <span class="dv">10</span>, </span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">facet_color =</span> <span class="st">&quot;#cccccc&quot;</span>)</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  tmp_group_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_group_plot&quot;</span>)) </span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_group_name, tmp_group_plot)  </span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<p>Lets also add a little formatting to the groupmean plots.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span>  set_to_plot) {</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> tmp_get <span class="sc">+</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>       <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>       <span class="fu">ylab</span>(<span class="st">&quot;Relative Abundance (% total reads)&quot;</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">&quot;Temperature&quot;</span>) <span class="sc">+</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>       <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>             <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>             <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>             <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>             <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)) <span class="sc">+</span> </span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>       <span class="fu">scale_y_continuous</span>()</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(i, tmp_get)</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<ol start="10" type="1">
<li>Finally we use the <code>patchwork</code> package to combine the two plots and customize the look.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="do">## single index that acts as an index for referencing elements (variables) in a list</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="do">##  solution modified from this SO answerhttps://stackoverflow.com/a/54451460</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>var_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">var1 =</span> ssu18_data_sets,</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">var2 =</span> <span class="fu">c</span>(<span class="st">&quot;FULL&quot;</span>, <span class="st">&quot;Arbitrary filtered&quot;</span>, <span class="st">&quot;PERfect filtered&quot;</span>, <span class="st">&quot;PIME filtered&quot;</span>)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span>  <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(var_list<span class="sc">$</span>var1)) {</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>     tmp_plot_final_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(var_list<span class="sc">$</span>var1[j], <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_&quot;</span>, top_level, <span class="st">&quot;_plot_final&quot;</span>))</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>     tmp_set_type <span class="ot">&lt;-</span> var_list<span class="sc">$</span>var2[j]</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>     tmp_p_plot <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(var_list<span class="sc">$</span>var1[j], <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_group_plot&quot;</span>)))</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>     tmp_m_plot <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(var_list<span class="sc">$</span>var1[j], <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_facet_plot&quot;</span>)))</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>     tmp_plot_final <span class="ot">&lt;-</span> tmp_p_plot <span class="sc">+</span> tmp_m_plot</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>     tmp_plot_final <span class="ot">&lt;-</span> tmp_plot_final <span class="sc">+</span></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>         <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">&quot;a&quot;</span>, </span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>                         <span class="at">title =</span> <span class="fu">paste</span>(<span class="st">&quot;Top&quot;</span>, top_level, <span class="st">&quot;of&quot;</span>, tmp_set_type, <span class="st">&quot;data set&quot;</span>), </span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>                         <span class="co">#subtitle = &#39;Top taxa of non-filtered data&#39;,</span></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>                         <span class="at">caption =</span> <span class="st">&quot;a) grouped by temperature.,</span></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a><span class="st">                         b) All samples, faceted by temperature.&quot;</span>) <span class="sc">+</span> </span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>       <span class="fu">plot_layout</span>(<span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)) <span class="sc">&amp;</span></span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>       <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">13</span>),</span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>             <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>), </span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>             <span class="at">plot.tag =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>), </span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>             <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>), </span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>             <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>)) </span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_plot_final_name, tmp_plot_final)</span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><br/></p>
<blockquote>
<p>It is important to note that while most of the dominant taxa are the same across the FULL, Arbitrary, PIME, and PERfect data sets, there a few key difference:</p>
</blockquote>
<ol type="1">
<li><p>Planctomycetota was not among the dominant taxa in the PIME data set. Nitrospirota was substituted.</p></li>
<li><p>Chloroflexi was not among the dominant taxa in either the Arbitrary, PERfect, or PIME data sets. RCP2-54 was substituted in both cases.</p></li>
</ol>
<div class="l-body-outset">
<div class="panelset">
<section id="full-unfiltered" class="panel">
<h4>FULL (unfiltered)</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/taxa/ssu18_ps_work_tax_div_bar_plots.png" width="1240" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 1 | Dominant taxonomic groups from the FULL (unfiltered) 16S rRNA data set. A) Grouped by temperature and B) all samples faceted by temperature treatment.</strong></small></p>
</section>
<section id="arbitrary-filtered" class="panel">
<h4>Arbitrary filtered</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/taxa/ssu18_ps_filt_tax_div_bar_plots.png" width="1240" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 2 | Dominant taxonomic groups from the Arbitrary filtered 16S rRNA data set. A) Grouped by temperature and B) all samples faceted by temperature treatment.</strong></small></p>
</section>
<section id="perfect-filtered" class="panel">
<h4>PERFECT filtered</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/taxa/ssu18_ps_perfect_tax_div_bar_plots.png" width="1240" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 3 | Dominant taxonomic groups from the PERfect filtered 16S rRNA data set. A) Grouped by temperature and B) all samples faceted by temperature treatment.</strong></small></p>
</section>
<section id="pime-filtered" class="panel">
<h4>PIME filtered</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/taxa/ssu18_ps_pime_tax_div_bar_plots.png" width="1240" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 4 | Dominant taxonomic groups from the PIME filtered 16S rRNA data set. A) Grouped by temperature and B) all samples faceted by temperature treatment.</strong></small></p>
</section>
</div>
</div>
<h2 id="its">ITS</h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Here we compare the taxonomic breakdown of the FULL (unfiltered), PIME filtered, and PERfect filtered data sets, split by temperature treatment for the ITS data. The workflow is basically the same as the 16S rRNA data. Click the box below to see the entire workflow.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<details>
<summary>
<strong>Detailed workflow</strong> for ITS taxonomic analysis.
</summary>
<ol type="1">
<li>Choose the <strong>number</strong> of taxa to display and the taxonomic <strong>level</strong>. Aggregate the rest into Other.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>top_hits <span class="ot">&lt;-</span> <span class="dv">14</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>top_level <span class="ot">&lt;-</span> <span class="st">&quot;Order&quot;</span></span></code></pre></div>
</div>
<p>As above, in order to use <code>microeco</code>, we need to add the rank designation as a prefix to each taxa. For example, <code>Basidiomycota</code> is changed to <code>p__Basidiomycota</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its18_data_sets) {</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  tmp_sam_data <span class="ot">&lt;-</span> <span class="fu">sample_data</span>(tmp_get)</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(tmp_get))</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>ASV_ID <span class="ot">&lt;-</span> <span class="cn">NULL</span> <span class="co"># Some have, some do not</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>ASV_SEQ <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="co">#  tmp_tax_data[] &lt;- data.frame(lapply(tmp_tax_data, gsub, </span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="co">#                                      pattern = &quot;^[k | p | c | o | f]_.*&quot;, </span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a><span class="co">#                                      replacement = &quot;&quot;, fixed = FALSE))</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Kingdom <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;k__&quot;</span>, tmp_tax_data<span class="sc">$</span>Kingdom, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Phylum <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;p__&quot;</span>, tmp_tax_data<span class="sc">$</span>Phylum, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Class <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;c__&quot;</span>, tmp_tax_data<span class="sc">$</span>Class, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Order <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;o__&quot;</span>, tmp_tax_data<span class="sc">$</span>Order, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Family <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;f__&quot;</span>, tmp_tax_data<span class="sc">$</span>Family, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Genus <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;g__&quot;</span>, tmp_tax_data<span class="sc">$</span>Genus, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_tax_data)</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>  tmp_ps <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(<span class="fu">otu_table</span>(tmp_get),</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">tax_table</span>(tmp_tax_data),</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>                     tmp_sam_data)</span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(i, tmp_ps)</span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="fu">tax_table</span>(its18_ps_pime))</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its18_data_sets) {</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>    tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>    tmp_otu_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(tmp_get)))</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>    tmp_sample_info <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>    tmp_taxonomy_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(tmp_get))</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>    <span class="co">#tmp_taxonomy_table %&lt;&gt;% tidy_taxonomy</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>    tmp_dataset <span class="ot">&lt;-</span> microtable<span class="sc">$</span><span class="fu">new</span>(<span class="at">sample_table =</span> tmp_sample_info, </span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">otu_table =</span> tmp_otu_table, </span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">tax_table =</span> tmp_taxonomy_table)</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>    tmp_dataset<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(tmp_dataset)</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>    tmp_dataset<span class="sc">$</span>tax_table <span class="sc">%&lt;&gt;%</span> base<span class="sc">::</span><span class="fu">subset</span>(Kingdom <span class="sc">==</span> <span class="st">&quot;k__Fungi&quot;</span>)</span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(tmp_dataset)</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>    tmp_dataset<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(tmp_dataset)</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>    tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_me&quot;</span>)) </span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">assign</span>(tmp_name, tmp_dataset)</span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>}  </span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span></code></pre></div>
</div>
<p>Here is an example of what a new microeco object looks like when you call it.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>its18_ps_work_me</span></code></pre></div>
<pre><code>microtable class:
sample_table have 13 rows and 6 columns
otu_table have 3355 rows and 13 columns
tax_table have 3355 rows and 6 columns
Taxa abundance: calculated for Kingdom,Phylum,Class,Order,Family,Genus </code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<ol start="5" type="1">
<li>Now, we calculate the taxa abundance at each taxonomic rank using <code>cal_abund()</code>. This function return a list called <code>taxa_abund</code> containing several data frame of the abundance information at each taxonomic rank. The list is stored in the microtable object automatically. Its worth noting that the <code>cal_abund()</code> function can be used to solve some complex cases, such as supporting both the relative and absolute abundance calculation and selecting the partial taxonomic columns. More information can be found in the description of the <a href="https://github.com/ChiLiubio/file2meco">file2meco package</a>. In the same loop we also create a <code>trans_abund</code> class, which is used to transform taxonomic abundance data for plotting.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its18_data_sets) {</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  tmp_me <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_me&quot;</span>)))</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  tmp_me<span class="sc">$</span><span class="fu">cal_abund</span>()</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  tmp_me_abund <span class="ot">&lt;-</span> trans_abund<span class="sc">$</span><span class="fu">new</span>(<span class="at">dataset =</span> tmp_me, <span class="at">taxrank =</span> top_level, <span class="at">ntaxa =</span> top_hits)</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>  tmp_me_abund<span class="sc">$</span>abund_data<span class="sc">$</span>Abundance <span class="ot">&lt;-</span> tmp_me_abund<span class="sc">$</span>abund_data<span class="sc">$</span>Abundance <span class="sc">/</span> <span class="dv">100</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  tmp_me_abund_gr <span class="ot">&lt;-</span> trans_abund<span class="sc">$</span><span class="fu">new</span>(<span class="at">dataset =</span> tmp_me, <span class="at">taxrank =</span> top_level, <span class="at">ntaxa =</span> top_hits, </span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">groupmean =</span> <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>  tmp_me_abund_gr<span class="sc">$</span>abund_data<span class="sc">$</span>Abundance <span class="ot">&lt;-</span> tmp_me_abund_gr<span class="sc">$</span>abund_data<span class="sc">$</span>Abundance <span class="sc">/</span> <span class="dv">100</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_me_abund&quot;</span>)) </span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_me_abund)  </span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>  tmp_name_gr <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_me_abund_group&quot;</span>)) </span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name_gr, tmp_me_abund_gr)  </span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<pre><code>The result is stored in object$taxa_abund ...</code></pre>
<ol start="6" type="1">
<li>I prefer to specify the order of taxa in these kinds of plots. We can look the top <code>ntaxa</code> (defined above) by accessing the <code>use_taxanames</code> character vector of each microtable object.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Hum. Similar but not the same. This means we need to define the order for each separately. Once we do that, we will override the <code>use_taxanames</code> character vectors with the reordered vectors.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>its18_ps_work_tax_ord <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;k_Fungi&quot;</span>,  <span class="st">&quot;p_Ascomycota&quot;</span>,  <span class="st">&quot;c_Agaricomycetes&quot;</span>, <span class="st">&quot;Agaricales&quot;</span>,  </span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Archaeorhizomycetales&quot;</span>,  <span class="st">&quot;Capnodiales&quot;</span>,  <span class="st">&quot;Eurotiales&quot;</span>,  <span class="st">&quot;Geastrales&quot;</span>,  </span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Glomerales&quot;</span>,  <span class="st">&quot;Helotiales&quot;</span>, <span class="st">&quot;Hypocreales&quot;</span>,  <span class="st">&quot;Saccharomycetales&quot;</span>,  </span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Trichosporonales&quot;</span>,  <span class="st">&quot;Xylariales&quot;</span>))   </span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>its18_ps_filt_tax_ord <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;k_Fungi&quot;</span>,  <span class="st">&quot;p_Ascomycota&quot;</span>,  <span class="st">&quot;Polyporales&quot;</span>, <span class="st">&quot;Agaricales&quot;</span>,  </span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Archaeorhizomycetales&quot;</span>,  <span class="st">&quot;Capnodiales&quot;</span>,  <span class="st">&quot;Eurotiales&quot;</span>,  <span class="st">&quot;Geastrales&quot;</span>,  </span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Glomerales&quot;</span>,  <span class="st">&quot;Mortierellales&quot;</span>, <span class="st">&quot;Hypocreales&quot;</span>,  <span class="st">&quot;Saccharomycetales&quot;</span>,  </span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Trichosporonales&quot;</span>,  <span class="st">&quot;Xylariales&quot;</span>))   </span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect_tax_ord <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;k_Fungi&quot;</span>,  <span class="st">&quot;p_Ascomycota&quot;</span>,  <span class="st">&quot;c_Agaricomycetes&quot;</span>, <span class="st">&quot;Agaricales&quot;</span>,  </span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Archaeorhizomycetales&quot;</span>,  <span class="st">&quot;Capnodiales&quot;</span>,  <span class="st">&quot;Eurotiales&quot;</span>,  <span class="st">&quot;Geastrales&quot;</span>,  </span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Glomerales&quot;</span>,  <span class="st">&quot;Helotiales&quot;</span>,   <span class="st">&quot;Hypocreales&quot;</span>,  <span class="st">&quot;Saccharomycetales&quot;</span>, </span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Trichosporonales&quot;</span>,  <span class="st">&quot;Pezizales&quot;</span>))   </span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_tax_ord <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(</span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;k_Fungi&quot;</span>,  <span class="st">&quot;p_Ascomycota&quot;</span>,  <span class="st">&quot;Polyporales&quot;</span>,  <span class="st">&quot;Agaricales&quot;</span>,  </span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Archaeorhizomycetales&quot;</span>,  <span class="st">&quot;Capnodiales&quot;</span>,  <span class="st">&quot;Eurotiales&quot;</span>,  <span class="st">&quot;Geastrales&quot;</span>, </span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Glomerales&quot;</span>,  <span class="st">&quot;Mortierellales&quot;</span>,  <span class="st">&quot;Hypocreales&quot;</span>,  <span class="st">&quot;Saccharomycetales&quot;</span>,  </span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Trichosporonales&quot;</span>,  <span class="st">&quot;Pezizales&quot;</span>))   </span></code></pre></div>
</details>
</div>
<ol start="7" type="1">
<li>And one more little step before plotting. Here we <strong>a</strong>) specify a custom color palette and <strong>b</strong>) specify the sample order.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>its18_colvec.tax <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&quot;#323232&quot;</span>, <span class="st">&quot;#004949&quot;</span>, <span class="st">&quot;#924900&quot;</span>, <span class="st">&quot;#490092&quot;</span>, </span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>                      <span class="st">&quot;#6db6ff&quot;</span>, <span class="st">&quot;#920000&quot;</span>, <span class="st">&quot;#ffb6db&quot;</span>, <span class="st">&quot;#24ff24&quot;</span>, </span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>                      <span class="st">&quot;#006ddb&quot;</span>, <span class="st">&quot;#db6d00&quot;</span>, <span class="st">&quot;#b66dff&quot;</span>, <span class="st">&quot;#ffff6d&quot;</span>, </span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>                      <span class="st">&quot;#009292&quot;</span>, <span class="st">&quot;#b6dbff&quot;</span>, <span class="st">&quot;#ff6db6&quot;</span>))</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>its18_samp_order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;P02_D00_010_C0A&quot;</span>, <span class="st">&quot;P04_D00_010_C0B&quot;</span>, <span class="st">&quot;P06_D00_010_C0C&quot;</span>, <span class="st">&quot;P08_D00_010_C0D&quot;</span>, <span class="st">&quot;P10_D00_010_C0E&quot;</span>, </span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;P01_D00_010_W3A&quot;</span>, <span class="st">&quot;P03_D00_010_W3B&quot;</span>, <span class="st">&quot;P07_D00_010_W3D&quot;</span>, <span class="st">&quot;P09_D00_010_W3E&quot;</span>, </span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;P01_D00_010_W8A&quot;</span>, <span class="st">&quot;P03_D00_010_W8B&quot;</span>, <span class="st">&quot;P05_D00_010_W8C&quot;</span>, <span class="st">&quot;P07_D00_010_W8D&quot;</span>)</span></code></pre></div>
</div>
<ol start="8" type="1">
<li>Now we can generate plots (in a loop) for each faceted data set.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its18_data_sets) {</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  tmp_abund <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_me_abund&quot;</span>)))</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  tmp_tax_order <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_tax_ord&quot;</span>)))</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  tmp_abund<span class="sc">$</span>use_taxanames <span class="ot">&lt;-</span> tmp_tax_order</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  tmp_facet_plot <span class="ot">&lt;-</span> tmp_abund<span class="sc">$</span><span class="fu">plot_bar</span>(<span class="at">use_colors =</span> its18_colvec.tax, </span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">others_color =</span> <span class="st">&quot;#323232&quot;</span>, </span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">facet =</span> <span class="st">&quot;TEMP&quot;</span>, </span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_keep =</span> <span class="cn">TRUE</span>, </span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_type_hor =</span> <span class="cn">FALSE</span>, </span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">legend_text_italic =</span> <span class="cn">FALSE</span>, </span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_size =</span> <span class="dv">6</span>, </span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">facet_color =</span> <span class="st">&quot;#cccccc&quot;</span>, </span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">order_x =</span> its18_samp_order)</span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>  tmp_facet_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_facet_plot&quot;</span>)) </span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_facet_name, tmp_facet_plot)  </span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<p>Then add a little formatting to the faceted plots.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>set_to_plot <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;its18_ps_work_facet_plot&quot;</span>, <span class="st">&quot;its18_ps_filt_facet_plot&quot;</span>, <span class="st">&quot;its18_ps_perfect_facet_plot&quot;</span>, <span class="st">&quot;its18_ps_pime_facet_plot&quot;</span>)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span>  set_to_plot) {</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>    tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>    tmp_get <span class="ot">&lt;-</span> tmp_get <span class="sc">+</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> top_level, <span class="at">reverse =</span> <span class="cn">FALSE</span>,</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>                               <span class="at">keywidth =</span> <span class="fl">0.7</span>, <span class="at">keyheight =</span> <span class="fl">0.7</span>)) <span class="sc">+</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">&quot;Sample&quot;</span>) <span class="sc">+</span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>),</span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>, <span class="at">angle =</span> <span class="dv">90</span>),</span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">0</span>),</span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a>      <span class="fu">scale_y_continuous</span>()</span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(i, tmp_get)</span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>)) </span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<p>And now plots for the groupmeans sets. We can use the same taxa order since that should not have changed.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>set_to_plot <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;its18_ps_work_group_plot&quot;</span>, <span class="st">&quot;its18_ps_filt_group_plot&quot;</span>, <span class="st">&quot;its18_ps_perfect_group_plot&quot;</span>, <span class="st">&quot;its18_ps_pime_group_plot&quot;</span>)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its18_data_sets) {</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  tmp_abund <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_me_abund_group&quot;</span>)))</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>  tmp_tax_order <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_tax_ord&quot;</span>)))</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>  tmp_abund<span class="sc">$</span>use_taxanames <span class="ot">&lt;-</span> tmp_tax_order</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>  tmp_group_plot <span class="ot">&lt;-</span> tmp_abund<span class="sc">$</span><span class="fu">plot_bar</span>(<span class="at">use_colors =</span> its18_colvec.tax, </span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">others_color =</span> <span class="st">&quot;#323232&quot;</span>, </span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_keep =</span> <span class="cn">TRUE</span>, </span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_type_hor =</span> <span class="cn">TRUE</span>, </span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">legend_text_italic =</span> <span class="cn">FALSE</span>, </span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_size =</span> <span class="dv">10</span>, </span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">facet_color =</span> <span class="st">&quot;#cccccc&quot;</span>)</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>  tmp_group_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_group_plot&quot;</span>)) </span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_group_name, tmp_group_plot)  </span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<p>Lets also add a little formatting to the groupmean plots.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span>  set_to_plot) {</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> tmp_get <span class="sc">+</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>       <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>       <span class="fu">ylab</span>(<span class="st">&quot;Relative Abundance (% total reads)&quot;</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">&quot;Temperature&quot;</span>) <span class="sc">+</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>       <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>             <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>             <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>             <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>             <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)) <span class="sc">+</span> </span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>       <span class="fu">scale_y_continuous</span>()</span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(i, tmp_get)</span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<ol start="10" type="1">
<li>Finally we use the <code>patchwork</code> package to combine the two plots and customize the look.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="do">## single index that acts as an index for referencing elements (variables) in a list</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="do">##  solution modified from this SO answerhttps://stackoverflow.com/a/54451460</span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>var_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">var1 =</span> its18_data_sets,</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">var2 =</span> <span class="fu">c</span>(<span class="st">&quot;FULL&quot;</span>, <span class="st">&quot;Arbitrary filtered&quot;</span>, <span class="st">&quot;PERfect filtered&quot;</span>, <span class="st">&quot;PIME filtered&quot;</span>)</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span>  <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(var_list<span class="sc">$</span>var1)) {</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>     tmp_plot_final_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(var_list<span class="sc">$</span>var1[j], <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_&quot;</span>, top_level, <span class="st">&quot;_plot_final&quot;</span>))</span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>     tmp_set_type <span class="ot">&lt;-</span> var_list<span class="sc">$</span>var2[j]</span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>     tmp_p_plot <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(var_list<span class="sc">$</span>var1[j], <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_group_plot&quot;</span>)))</span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>     tmp_m_plot <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(var_list<span class="sc">$</span>var1[j], <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_facet_plot&quot;</span>)))</span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>     tmp_plot_final <span class="ot">&lt;-</span> tmp_p_plot <span class="sc">+</span> tmp_m_plot</span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>     tmp_plot_final <span class="ot">&lt;-</span> tmp_plot_final <span class="sc">+</span></span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>         <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">&quot;a&quot;</span>, </span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>                         <span class="at">title =</span> <span class="fu">paste</span>(<span class="st">&quot;Top&quot;</span>, top_level, <span class="st">&quot;of&quot;</span>, tmp_set_type, <span class="st">&quot;data set&quot;</span>), </span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>                         <span class="co">#subtitle = &#39;Top taxa of non-filtered data&#39;,</span></span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>                         <span class="at">caption =</span> <span class="st">&quot;a) grouped by temperature.,</span></span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a><span class="st">                         b) All samples, faceted by temperature.&quot;</span>) <span class="sc">+</span> </span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>       <span class="fu">plot_layout</span>(<span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)) <span class="sc">&amp;</span></span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a>       <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">13</span>),</span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>             <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>), </span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a>             <span class="at">plot.tag =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>), </span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a>             <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>), </span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a>             <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>)) </span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_plot_final_name, tmp_plot_final)</span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
</details>
<p><br/></p>
<blockquote>
<p>It is important to note that while most of the dominant taxa are the same across the FULL, Arbitrary, PIME, and PERfect data sets, there a few key difference:</p>
</blockquote>
<ol type="1">
<li><p>c_Agaricomycetes &amp; Helotiales were not among the dominant taxa in the Arbitrary or PIME filtered data sets. Polyporales &amp; Mortierellales were substituted, respectively.</p></li>
<li><p>Xylariales was not among the dominant taxa in either the PIME or PERfect data sets. Pezizales was substituted in both cases.</p></li>
</ol>
<div class="l-body-outset">
<div class="panelset">
<section id="full-unfiltered-1" class="panel">
<h4>FULL (unfiltered)</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/taxa/its18_ps_work_tax_div_bar_plots.png" width="1240" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 5 | Dominant taxonomic groups from the FULL (unfiltered) ITS data set. A) Grouped by temperature and B) all samples faceted by temperature treatment.</strong></small></p>
</section>
<section id="arbitrary-filtered-1" class="panel">
<h4>Arbitrary filtered</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/taxa/its18_ps_filt_tax_div_bar_plots.png" width="1240" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 6 | Dominant taxonomic groups from the Arbitrary filtered ITS data set. A) Grouped by temperature and B) all samples faceted by temperature treatment.</strong></small></p>
</section>
<section id="perfect-filtered-1" class="panel">
<h4>PERFECT filtered</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/taxa/its18_ps_perfect_tax_div_bar_plots.png" width="1240" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 7 | Dominant taxonomic groups from the PERfect filtered ITS data set. A) Grouped by temperature and B) all samples faceted by temperature treatment.</strong></small></p>
</section>
<section id="pime-filtered-1" class="panel">
<h4>PIME filtered</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="include/taxa/its18_ps_pime_tax_div_bar_plots.png" width="1240" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 8 | Dominant taxonomic groups from the PIME filtered ITS data set. A) Grouped by temperature and B) all samples faceted by temperature treatment.</strong></small></p>
</section>
</div>
</div>
<h1 id="major-bacterial-taxa-by-family">Major Bacterial taxa by Family</h1>
<p>We also look at the relative abundance of groups within dominant Phyla. <strong>Please note</strong> these analyses are for the FULL data set only, before any type of filtering. For this analysis, we use the phyloseq object created in the Data Preparation wokflow section <a href="data-prep.html#rename-na-taxonomic-ranks">Rename NA taxonomic ranks</a> where unclassified ranks were renamed by the next highest rank. See step #1 below for an example of the taxa table structure.</p>
<aside>
This code <strong><em>only</em></strong> works for multiple taxa from a <em>single</em> phyloseq object.
</aside>
<details>
<summary>
<strong>Detailed workflow</strong> for assessing Bacterial families.
</summary>
<ol type="1">
<li>Subset Phyla of interest.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;files/data-prep/rdata/ssu18_ps_work.rds&quot;</span>)</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ssu18_ps_work)))[,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body-outset">
<pre><code>          Kingdom          Phylum               Class
ASV9128  Bacteria Acidobacteriota         Subgroup_22
ASV7870  Bacteria Planctomycetota               OM190
ASV8537  Bacteria     Chloroflexi        Anaerolineae
ASV15357 Bacteria  Proteobacteria Gammaproteobacteria
ASV19533 Bacteria      k_Bacteria          k_Bacteria
ASV18180 Bacteria Planctomycetota      Planctomycetes
                      Order              Family                 Genus
ASV9128       c_Subgroup_22       c_Subgroup_22         c_Subgroup_22
ASV7870             c_OM190             c_OM190               c_OM190
ASV8537             SBR1031                 A4b                 f_A4b
ASV15357 Diplorickettsiales Diplorickettsiaceae f_Diplorickettsiaceae
ASV19533         k_Bacteria          k_Bacteria            k_Bacteria
ASV18180         Gemmatales         Gemmataceae         f_Gemmataceae</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>ssu18_data_sets <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ssu18_ps_work&quot;</span>)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_proteo&quot;</span>))</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>     tmp_df <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(tmp_get, Phylum <span class="sc">==</span> <span class="st">&quot;Proteobacteria&quot;</span>)</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_df)</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_name)</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>     tmp_get_taxa <span class="ot">&lt;-</span> <span class="fu">get_taxa_unique</span>(tmp_df,</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">taxonomic.rank =</span> <span class="fu">rank_names</span>(tmp_df)[<span class="dv">3</span>],</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">errorIfNULL=</span><span class="cn">TRUE</span>)</span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_get_taxa)</span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;_proteo&quot;</span>))</span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<ol start="2" type="1">
<li>Replace Phylum Proteobacteria with the Class name.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(j, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_proteo_clean&quot;</span>))</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(j)</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>  tmp_clean <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(tmp_get))</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>   <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(tmp_clean)){</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>       <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">&quot;Proteobacteria&quot;</span> <span class="sc">&amp;</span> tmp_clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">&quot;Alphaproteobacteria&quot;</span>){</span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>           phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;Alphaproteobacteria&quot;</span>)</span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>           tmp_clean[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>   }   <span class="cf">else</span> <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">&quot;Proteobacteria&quot;</span> <span class="sc">&amp;</span> tmp_clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">&quot;Gammaproteobacteria&quot;</span>){</span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a>           phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;Gammaproteobacteria&quot;</span>)</span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a>           tmp_clean[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>   }   <span class="cf">else</span> <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">&quot;Proteobacteria&quot;</span> <span class="sc">&amp;</span> tmp_clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">&quot;Zetaproteobacteria&quot;</span>){</span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a>              phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;Zetaproteobacteria&quot;</span>)</span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a>           tmp_clean[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a>   }   <span class="cf">else</span> <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">&quot;Proteobacteria&quot;</span> <span class="sc">&amp;</span> tmp_clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">&quot;p_Proteobacteria&quot;</span>){</span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a>           phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">&quot;p_Proteobacteria&quot;</span>)</span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a>           tmp_clean[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a>       }</span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tax_table</span>(tmp_get) <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_clean)</span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rank_names</span>(tmp_get)</span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_get)</span>
<span id="cb34-24"><a href="#cb34-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">c</span>(tmp_name, tmp_get))</span>
<span id="cb34-25"><a href="#cb34-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(<span class="fu">get_taxa_unique</span>(tmp_get,</span>
<span id="cb34-26"><a href="#cb34-26" aria-hidden="true" tabindex="-1"></a>                               <span class="at">taxonomic.rank =</span> <span class="fu">rank_names</span>(tmp_get)[<span class="dv">2</span>],</span>
<span id="cb34-27"><a href="#cb34-27" aria-hidden="true" tabindex="-1"></a>                               <span class="at">errorIfNULL=</span><span class="cn">TRUE</span>)))</span>
<span id="cb34-28"><a href="#cb34-28" aria-hidden="true" tabindex="-1"></a>  tmp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&quot;files/taxa/rdata/&quot;</span>)</span>
<span id="cb34-29"><a href="#cb34-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(tmp_get, <span class="fu">paste</span>(tmp_path, j, <span class="st">&quot;_clean.rds&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb34-30"><a href="#cb34-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb34-31"><a href="#cb34-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-32"><a href="#cb34-32" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(class, order, phylum)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>set_to_plot <span class="ot">&lt;-</span> <span class="st">&quot;ssu18_ps_work&quot;</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>tax_group <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Alphaproteobacteria&quot;</span>, <span class="st">&quot;Gammaproteobacteria&quot;</span>, </span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;Acidobacteriota&quot;</span>, <span class="st">&quot;Actinobacteriota&quot;</span>, </span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;Bacteroidota&quot;</span>, <span class="st">&quot;Firmicutes&quot;</span>, </span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;Myxococcota&quot;</span>, <span class="st">&quot;Verrucomicrobiota&quot;</span>)</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> set_to_plot) {</span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (j <span class="cf">in</span> tax_group) {</span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_proteo_clean&quot;</span>)))</span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>  tmp_sub <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(tmp_get, Phylum <span class="sc">==</span> j)</span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_&quot;</span>, j))</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_sub)</span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<ol start="3" type="1">
<li>Choose the <strong>number</strong> of taxa to display and the taxonomic <strong>level</strong>. Aggregate the rest into Other.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>top_hits <span class="ot">&lt;-</span> <span class="dv">12</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>top_level <span class="ot">&lt;-</span> <span class="st">&quot;Family&quot;</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, i)))</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>  tmp_otu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(tmp_get)))</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>  tmp_otu[] <span class="ot">&lt;-</span> <span class="fu">lapply</span>(tmp_otu, as.numeric)</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>  tmp_otu <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_otu)</span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>  tmp_tax <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">data.frame</span>(<span class="fu">tax_table</span>(tmp_get)))</span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>  tmp_samples <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>  tmp_clean_df <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(<span class="fu">otu_table</span>(tmp_otu, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>),</span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">tax_table</span>(tmp_tax),</span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">sample_data</span>(tmp_samples))</span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>  tmp_agg_df <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">aggregate_top_taxa</span>(tmp_clean_df,</span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>                                                    <span class="at">top =</span> top_hits,</span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>                                                    <span class="at">level =</span> top_level)</span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a>  tmp_agg_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, i, <span class="st">&quot;_agg&quot;</span>))</span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_agg_name, tmp_agg_df)</span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;_sep_agg&quot;</span>))</span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group){</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>  tmp_data <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, i, <span class="st">&quot;_agg&quot;</span>))</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(tmp_data)</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>  tmp_list <span class="ot">&lt;-</span> <span class="fu">get_taxa_unique</span>(tmp_get, <span class="at">taxonomic.rank =</span> <span class="fu">rank_names</span>(tmp_get)[<span class="dv">2</span>],</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">errorIfNULL =</span> <span class="cn">TRUE</span>)</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(tmp_data, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_order&quot;</span>))</span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_list)</span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, i, <span class="st">&quot;_agg_order&quot;</span>)))</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;#########&quot;</span>, i, <span class="st">&quot;########&quot;</span>, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>  tmp_print <span class="ot">&lt;-</span> <span class="fu">c</span>(tmp_get)</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(tmp_print, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;####################################################&quot;</span>)</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(i, j)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>tmp_order <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&quot;Other&quot;</span>, <span class="st">&quot;c_Alphaproteobacteria&quot;</span>, <span class="st">&quot;o_Elsterales&quot;</span>, <span class="st">&quot;o_Azospirillales&quot;</span>, <span class="st">&quot;Rhizobiales_Incertae_Sedis&quot;</span>, <span class="st">&quot;Xanthobacteraceae&quot;</span>, <span class="st">&quot;Sphingomonadaceae&quot;</span>, <span class="st">&quot;Rhizobiaceae&quot;</span>, <span class="st">&quot;Micropepsaceae&quot;</span>, <span class="st">&quot;Methyloligellaceae&quot;</span>, <span class="st">&quot;KF-JG30-B3&quot;</span>, <span class="st">&quot;Hyphomicrobiaceae&quot;</span>, <span class="st">&quot;Dongiaceae&quot;</span>))</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(<span class="fu">paste</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, <span class="st">&quot;Alphaproteobacteria&quot;</span>, <span class="st">&quot;_agg_order&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), tmp_order) </span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>tmp_order <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&quot;Other&quot;</span>, <span class="st">&quot;c_Gammaproteobacteria&quot;</span>, <span class="st">&quot;o_PLTA13&quot;</span>, <span class="st">&quot;o_CCD24&quot;</span>, <span class="st">&quot;Unknown_Family&quot;</span>, <span class="st">&quot;Xanthomonadaceae&quot;</span>, <span class="st">&quot;TRA3-20&quot;</span>, <span class="st">&quot;Steroidobacteraceae&quot;</span>, <span class="st">&quot;SC-I-84&quot;</span>, <span class="st">&quot;Nitrosomonadaceae&quot;</span>, <span class="st">&quot;Comamonadaceae&quot;</span>, <span class="st">&quot;Burkholderiaceae&quot;</span>, <span class="st">&quot;B1-7BS&quot;</span>))</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(<span class="fu">paste</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, <span class="st">&quot;Gammaproteobacteria&quot;</span>, <span class="st">&quot;_agg_order&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), tmp_order) </span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>tmp_order <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&quot;Other&quot;</span>, <span class="st">&quot;c_Subgroup_22&quot;</span>, <span class="st">&quot;c_Subgroup_25&quot;</span>, <span class="st">&quot;c_Subgroup_5&quot;</span>, <span class="st">&quot;o_Vicinamibacterales&quot;</span>, <span class="st">&quot;o_11-24&quot;</span>, <span class="st">&quot;o_Subgroup_17&quot;</span>, <span class="st">&quot;o_Subgroup_2&quot;</span>, <span class="st">&quot;o_Subgroup_7&quot;</span>, <span class="st">&quot;o_Acidobacteriales&quot;</span>, <span class="st">&quot;Vicinamibacteraceae&quot;</span>, <span class="st">&quot;Solibacteraceae&quot;</span>, <span class="st">&quot;Pyrinomonadaceae&quot;</span>))</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(<span class="fu">paste</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, <span class="st">&quot;Acidobacteriota&quot;</span>, <span class="st">&quot;_agg_order&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), tmp_order) </span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>tmp_order <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&quot;Other&quot;</span>, <span class="st">&quot;c_MB-A2-108&quot;</span>, <span class="st">&quot;o_IMCC26256&quot;</span>, <span class="st">&quot;o_Gaiellales&quot;</span>, <span class="st">&quot;o_Frankiales&quot;</span>, <span class="st">&quot;Streptomycetaceae&quot;</span>, <span class="st">&quot;Solirubrobacteraceae&quot;</span>, <span class="st">&quot;Nocardioidaceae&quot;</span>, <span class="st">&quot;Mycobacteriaceae&quot;</span>, <span class="st">&quot;Micromonosporaceae&quot;</span>, <span class="st">&quot;Gaiellaceae&quot;</span>, <span class="st">&quot;Acidothermaceae&quot;</span>, <span class="st">&quot;67-14&quot;</span>))</span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(<span class="fu">paste</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, <span class="st">&quot;Actinobacteriota&quot;</span>, <span class="st">&quot;_agg_order&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), tmp_order) </span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a>tmp_order <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&quot;Other&quot;</span>, <span class="st">&quot;c_SJA-28&quot;</span>, <span class="st">&quot;o_Chitinophagales&quot;</span>, <span class="st">&quot;Sphingobacteriaceae&quot;</span>, <span class="st">&quot;Saprospiraceae&quot;</span>, <span class="st">&quot;Microscillaceae&quot;</span>, <span class="st">&quot;Hymenobacteraceae&quot;</span>, <span class="st">&quot;Flavobacteriaceae&quot;</span>, <span class="st">&quot;Cytophagaceae&quot;</span>, <span class="st">&quot;Chitinophagaceae&quot;</span>, <span class="st">&quot;BSV26&quot;</span>, <span class="st">&quot;env.OPS_17&quot;</span>, <span class="st">&quot;AKYH767&quot;</span>))</span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(<span class="fu">paste</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, <span class="st">&quot;Bacteroidota&quot;</span>, <span class="st">&quot;_agg_order&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), tmp_order) </span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a>tmp_order <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&quot;Other&quot;</span>, <span class="st">&quot;p_Firmicutes&quot;</span>, <span class="st">&quot;c_Bacilli&quot;</span>, <span class="st">&quot;o_Bacillales&quot;</span>, <span class="st">&quot;Thermoactinomycetaceae&quot;</span>, <span class="st">&quot;type_III&quot;</span>, <span class="st">&quot;Paenibacillaceae&quot;</span>, <span class="st">&quot;Lachnospiraceae&quot;</span>, <span class="st">&quot;Hungateiclostridiaceae&quot;</span>, <span class="st">&quot;Clostridiaceae&quot;</span>, <span class="st">&quot;Bacillaceae&quot;</span>, <span class="st">&quot;Alicyclobacillaceae&quot;</span>, <span class="st">&quot;Planococcaceae&quot;</span>))</span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(<span class="fu">paste</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, <span class="st">&quot;Firmicutes&quot;</span>, <span class="st">&quot;_agg_order&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), tmp_order) </span>
<span id="cb39-18"><a href="#cb39-18" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span>
<span id="cb39-19"><a href="#cb39-19" aria-hidden="true" tabindex="-1"></a>tmp_order <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&quot;Other&quot;</span>, <span class="st">&quot;p_Myxococcota&quot;</span>, <span class="st">&quot;c_bacteriap25&quot;</span>, <span class="st">&quot;c_Polyangia&quot;</span>, <span class="st">&quot;o_mle1-27&quot;</span>, <span class="st">&quot;o_MSB-4B10&quot;</span>, <span class="st">&quot;Sandaracinaceae&quot;</span>, <span class="st">&quot;Polyangiaceae&quot;</span>, <span class="st">&quot;Phaselicystidaceae&quot;</span>, <span class="st">&quot;Myxococcaceae&quot;</span>, <span class="st">&quot;Haliangiaceae&quot;</span>, <span class="st">&quot;BIrii41&quot;</span>, <span class="st">&quot;Anaeromyxobacteraceae&quot;</span>))</span>
<span id="cb39-20"><a href="#cb39-20" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(<span class="fu">paste</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, <span class="st">&quot;Myxococcota&quot;</span>, <span class="st">&quot;_agg_order&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), tmp_order) </span>
<span id="cb39-21"><a href="#cb39-21" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span>
<span id="cb39-22"><a href="#cb39-22" aria-hidden="true" tabindex="-1"></a>tmp_order <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&quot;Other&quot;</span>, <span class="st">&quot;c_Verrucomicrobiae&quot;</span>, <span class="st">&quot;o_Chlamydiales&quot;</span>, <span class="st">&quot;o_LD1-PA32&quot;</span>, <span class="st">&quot;o_S-BQ2-57_soil_group&quot;</span>, <span class="st">&quot;Xiphinematobacteraceae&quot;</span>, <span class="st">&quot;Simkaniaceae&quot;</span>, <span class="st">&quot;Pedosphaeraceae&quot;</span>, <span class="st">&quot;Parachlamydiaceae&quot;</span>, <span class="st">&quot;Opitutaceae&quot;</span>, <span class="st">&quot;Omnitrophaceae&quot;</span>, <span class="st">&quot;cvE6&quot;</span>, <span class="st">&quot;Chthoniobacteraceae&quot;</span>))</span>
<span id="cb39-23"><a href="#cb39-23" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(<span class="fu">paste</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, <span class="st">&quot;Verrucomicrobiota&quot;</span>, <span class="st">&quot;_agg_order&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), tmp_order) </span>
<span id="cb39-24"><a href="#cb39-24" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span></code></pre></div>
</div>
<ol start="4" type="1">
<li>Now, transform the data to relative abundance.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>  tmp_agg <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, i, <span class="st">&quot;_agg&quot;</span>))</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  tmp_order <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(tmp_agg, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_order&quot;</span>))</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>  tmp_get_agg <span class="ot">&lt;-</span> <span class="fu">get</span>(tmp_agg)</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>  tmp_get_order <span class="ot">&lt;-</span> <span class="fu">get</span>(tmp_order)</span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>  tmp_df <span class="ot">&lt;-</span> tmp_get_agg <span class="sc">%&gt;%</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>            <span class="fu">transform_sample_counts</span>(<span class="cf">function</span>(x) {x<span class="sc">/</span><span class="fu">sum</span>(x)} ) <span class="sc">%&gt;%</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>            <span class="fu">psmelt</span>()</span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>  tmp_df[[top_level]] <span class="ot">&lt;-</span> gdata<span class="sc">::</span><span class="fu">reorder.factor</span>(tmp_df[[top_level]],</span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">new.order =</span> <span class="fu">rev</span>(tmp_get_order))</span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>  tmp_df <span class="ot">&lt;-</span> tmp_df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">get</span>(top_level))</span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(tmp_agg, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_tax&quot;</span>))</span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_df)</span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a>  <span class="co">#print(c(i, tmp_name, tmp_agg))</span></span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, i, <span class="st">&quot;_agg_tax&quot;</span>)))</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>  tmp_levels <span class="ot">&lt;-</span> <span class="fu">levels</span>(tmp_get[[top_level]])</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">c</span>(i, tmp_levels))</span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<ol start="5" type="1">
<li>Plot the data for a single phyloseq object. Here you use an aggregated tax file.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>      tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, i, <span class="st">&quot;_agg_tax&quot;</span>)))</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>      tmp_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(tmp_get, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(TEMP), </span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">y =</span> Abundance, <span class="at">fill =</span> <span class="fu">get</span>(top_level))) <span class="sc">+</span></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>          <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="at">position =</span> <span class="st">&quot;fill&quot;</span>) <span class="sc">+</span></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>          <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> ssu18_colvec.tax) <span class="sc">+</span></span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>          <span class="co">#scale_x_discrete(&quot;Temperature&quot;, expand = waiver(), position = &quot;bottom&quot;, drop = FALSE) +</span></span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a>          <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>          <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> top_level)) <span class="sc">+</span></span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a>          <span class="co">#guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +</span></span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a>          <span class="fu">ylab</span>(<span class="st">&quot;Relative Abundance (% total reads)&quot;</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">&quot;Temperature&quot;</span>) <span class="sc">+</span></span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a>          <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a>                <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb42-14"><a href="#cb42-14" aria-hidden="true" tabindex="-1"></a>                <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb42-15"><a href="#cb42-15" aria-hidden="true" tabindex="-1"></a>                <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb42-16"><a href="#cb42-16" aria-hidden="true" tabindex="-1"></a>                <span class="co">#axis.text = element_text(size = 6),</span></span>
<span id="cb42-17"><a href="#cb42-17" aria-hidden="true" tabindex="-1"></a>                <span class="co">#axis.title = element_text(size = 8),</span></span>
<span id="cb42-18"><a href="#cb42-18" aria-hidden="true" tabindex="-1"></a>                <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb42-19"><a href="#cb42-19" aria-hidden="true" tabindex="-1"></a>      tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, ., <span class="st">&quot;_plot&quot;</span>))</span>
<span id="cb42-20"><a href="#cb42-20" aria-hidden="true" tabindex="-1"></a>      <span class="fu">assign</span>(tmp_name, tmp_plot)</span>
<span id="cb42-21"><a href="#cb42-21" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb42-22"><a href="#cb42-22" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<ol start="6" type="1">
<li>Plot the data for multiple taxa. Here again you use an aggregated tax file. This code can be used to generate plots for multiple data frames by adding the base phyloseq names to the <code>ssu18_split_df</code> variable. This code will also facet the plots by a metadata variable. If you do not want to facet remove the line beginning with <code>facet_grid</code>.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>  tmp_level_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, .)))</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>  tmp_level <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_level_get))</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>  tmp_level <span class="ot">&lt;-</span> tmp_level[<span class="fu">order</span>(tmp_level<span class="sc">$</span>TEMP), ]</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>  tmp_level <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(tmp_level<span class="sc">$</span>SamName)</span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>  tmp_agg_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, .,  <span class="st">&quot;_agg_tax&quot;</span>))</span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(tmp_agg_name)</span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>  tmp_df <span class="ot">&lt;-</span> reshape<span class="sc">::</span><span class="fu">melt</span>(tmp_get, <span class="at">id.vars =</span> <span class="fu">c</span>(<span class="st">&quot;Sample&quot;</span>, <span class="st">&quot;TEMP&quot;</span>, <span class="st">&quot;Abundance&quot;</span>, <span class="st">&quot;Family&quot;</span>))</span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>  tmp_plot_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, ., <span class="st">&quot;_plot_melt&quot;</span>))</span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>  tmp_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(tmp_df,</span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">x =</span> Sample,</span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">y =</span> Abundance, <span class="at">fill =</span> <span class="fu">get</span>(top_level))) <span class="sc">+</span></span>
<span id="cb43-15"><a href="#cb43-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(. <span class="sc">~</span>TEMP, <span class="at">scale =</span> <span class="st">&quot;free_x&quot;</span>, <span class="at">space =</span> <span class="st">&quot;free_x&quot;</span>)<span class="sc">+</span></span>
<span id="cb43-16"><a href="#cb43-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="at">position =</span> <span class="st">&quot;fill&quot;</span>)  <span class="sc">+</span></span>
<span id="cb43-17"><a href="#cb43-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> ssu18_colvec.tax) <span class="sc">+</span></span>
<span id="cb43-18"><a href="#cb43-18" aria-hidden="true" tabindex="-1"></a>    <span class="co">#scale_x_discrete(&quot;Treatment&quot;, expand = waiver(),</span></span>
<span id="cb43-19"><a href="#cb43-19" aria-hidden="true" tabindex="-1"></a>    <span class="co">#                 position = &quot;bottom&quot;, drop = FALSE, limits = tmp_level) +</span></span>
<span id="cb43-20"><a href="#cb43-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb43-21"><a href="#cb43-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> top_level, <span class="at">reverse =</span> <span class="cn">FALSE</span>,</span>
<span id="cb43-22"><a href="#cb43-22" aria-hidden="true" tabindex="-1"></a>                               <span class="at">keywidth =</span> <span class="fl">0.7</span>, <span class="at">keyheight =</span> <span class="fl">0.7</span>)) <span class="sc">+</span></span>
<span id="cb43-23"><a href="#cb43-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb43-24"><a href="#cb43-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb43-25"><a href="#cb43-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb43-26"><a href="#cb43-26" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb43-27"><a href="#cb43-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb43-28"><a href="#cb43-28" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb43-29"><a href="#cb43-29" aria-hidden="true" tabindex="-1"></a>        <span class="co">#legend.text = element_text(size = 6),</span></span>
<span id="cb43-30"><a href="#cb43-30" aria-hidden="true" tabindex="-1"></a>        <span class="co">#legend.title = element_text(size = 8),</span></span>
<span id="cb43-31"><a href="#cb43-31" aria-hidden="true" tabindex="-1"></a>        <span class="co">#legend.key.size = unit(1.5, &quot;cm&quot;),</span></span>
<span id="cb43-32"><a href="#cb43-32" aria-hidden="true" tabindex="-1"></a>        <span class="co">#axis.text.y = element_text(size = 6),</span></span>
<span id="cb43-33"><a href="#cb43-33" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>)</span>
<span id="cb43-34"><a href="#cb43-34" aria-hidden="true" tabindex="-1"></a>        <span class="co">#strip.text = element_text(size = 8, angle = 0),</span></span>
<span id="cb43-35"><a href="#cb43-35" aria-hidden="true" tabindex="-1"></a>        <span class="co">#axis.title = element_text(size = 8)</span></span>
<span id="cb43-36"><a href="#cb43-36" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">+</span> </span>
<span id="cb43-37"><a href="#cb43-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="cn">NULL</span>)</span>
<span id="cb43-38"><a href="#cb43-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb43-39"><a href="#cb43-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_plot_name, tmp_plot)</span>
<span id="cb43-40"><a href="#cb43-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb43-41"><a href="#cb43-41" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<ol start="7" type="1">
<li>Finally we use the <code>patchwork</code> package to combine the two plots and customize the look.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>  tmp_plot_main <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, ., <span class="st">&quot;_plot&quot;</span>)))</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  tmp_plot_melt <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, ., <span class="st">&quot;_plot_melt&quot;</span>)))</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>  tmp_final <span class="ot">&lt;-</span> tmp_plot_main <span class="sc">+</span> tmp_plot_melt</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>  tmp_final <span class="ot">&lt;-</span> tmp_final <span class="sc">+</span> </span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">&#39;a&#39;</span>, </span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">title =</span> i) <span class="sc">+</span> </span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#subtitle = &#39;Top taxa of non-filtered data&#39;,</span></span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#caption = &#39;A) grouped by temperature.,</span></span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#B) All samples, faceted by temperature.&#39;) + </span></span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_layout</span>(<span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)) <span class="sc">&amp;</span></span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>),</span>
<span id="cb44-13"><a href="#cb44-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">1</span>),</span>
<span id="cb44-14"><a href="#cb44-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.tag =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>),</span>
<span id="cb44-15"><a href="#cb44-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>),</span>
<span id="cb44-16"><a href="#cb44-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>),</span>
<span id="cb44-17"><a href="#cb44-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">0</span>),</span>
<span id="cb44-18"><a href="#cb44-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>),</span>
<span id="cb44-19"><a href="#cb44-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>),</span>
<span id="cb44-20"><a href="#cb44-20" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb44-21"><a href="#cb44-21" aria-hidden="true" tabindex="-1"></a>        <span class="co">#legend.position = &quot;right&quot;)</span></span>
<span id="cb44-22"><a href="#cb44-22" aria-hidden="true" tabindex="-1"></a>        <span class="co">#legend.position = &quot;right&quot;,</span></span>
<span id="cb44-23"><a href="#cb44-23" aria-hidden="true" tabindex="-1"></a>        <span class="co">#legend.title = element_text(size = rel(1)),</span></span>
<span id="cb44-24"><a href="#cb44-24" aria-hidden="true" tabindex="-1"></a>        <span class="co">#legend.text = element_text(size = rel(1)))</span></span>
<span id="cb44-25"><a href="#cb44-25" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(set_to_plot, <span class="st">&quot;_&quot;</span>, ., <span class="st">&quot;_final_plot&quot;</span>))</span>
<span id="cb44-26"><a href="#cb44-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_final)</span>
<span id="cb44-27"><a href="#cb44-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb44-28"><a href="#cb44-28" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>taxa_to_plot <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Acidobacteriota&quot;</span>, <span class="st">&quot;Actinobacteriota&quot;</span>, <span class="st">&quot;Alphaproteobacteria&quot;</span>, </span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>                  <span class="st">&quot;Bacteroidota&quot;</span>, <span class="st">&quot;Firmicutes&quot;</span>, <span class="st">&quot;Gammaproteobacteria&quot;</span>, <span class="st">&quot;Myxococcota&quot;</span>, </span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>                  <span class="st">&quot;Verrucomicrobiota&quot;</span>)</span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> taxa_to_plot) {</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">&quot;ssu18_ps_work_&quot;</span>, i, <span class="st">&quot;_final_plot&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)))</span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_get, </span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">&quot;files/taxa/figures/&quot;</span>, i, <span class="st">&quot;_tax_div_bar_plots.png&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), </span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">height =</span> <span class="dv">14</span>, <span class="at">width =</span> <span class="dv">21</span>, <span class="at">units =</span> <span class="st">&#39;cm&#39;</span>, <span class="at">bg =</span> <span class="st">&quot;white&quot;</span>)</span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_get, </span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">&quot;files/taxa/figures/&quot;</span>, i, <span class="st">&quot;_tax_div_bar_plots.pdf&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), </span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">height =</span> <span class="dv">14</span>, <span class="at">width =</span> <span class="dv">21</span>, <span class="at">units =</span> <span class="st">&#39;cm&#39;</span>, <span class="at">bg =</span> <span class="st">&quot;white&quot;</span>)</span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
</details>
<p><br/></p>
<div class="l-body-outset">
<div class="panelset">
<section id="alphaproteobacteria" class="panel">
<h4>Alphaproteobacteria</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="taxa_files/figure-html5/ggg-1.png" width="624" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 9 | Alphaproteobacteria family plots.</strong></small></p>
</section>
<section id="gammaproteobacteria" class="panel">
<h4>Gammaproteobacteria</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="taxa_files/figure-html5/unnamed-chunk-68-1.png" width="624" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 10 | Gammaproteobacteria family plots.</strong></small></p>
</section>
<section id="acidobacteriota" class="panel">
<h4>Acidobacteriota</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="taxa_files/figure-html5/unnamed-chunk-69-1.png" width="624" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 11 | Acidobacteriota family plots.</strong></small></p>
</section>
<section id="actinobacteriota" class="panel">
<h4>Actinobacteriota</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="taxa_files/figure-html5/unnamed-chunk-70-1.png" width="624" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 12 | Actinobacteriota family plots.</strong></small></p>
</section>
<section id="bacteroidota" class="panel">
<h4>Bacteroidota</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="taxa_files/figure-html5/unnamed-chunk-71-1.png" width="624" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 13 | Bacteroidota family plots.</strong></small></p>
</section>
<section id="firmicutes" class="panel">
<h4>Firmicutes</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="taxa_files/figure-html5/unnamed-chunk-72-1.png" width="624" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 14 | Firmicutes family plots.</strong></small></p>
</section>
<section id="myxococcota" class="panel">
<h4>Myxococcota</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="taxa_files/figure-html5/unnamed-chunk-73-1.png" width="624" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 15 | Myxococcota family plots.</strong></small></p>
</section>
<section id="verrucomicrobiota" class="panel">
<h4>Verrucomicrobiota</h4>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="taxa_files/figure-html5/unnamed-chunk-74-1.png" width="624" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 16 | Verrucomicrobiota family plots.</strong></small></p>
</section>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 class="appendix" id="source-code">Source Code</h2>
<p>The source code for this page can be accessed on GitHub by <a href="https://github.com/sweltr/high-temp/blob/master/taxa.Rmd">clicking this link</a>.</p>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/sweltr/high-temp/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/sweltr/high-temp/">https://github.com/sweltr/high-temp/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<div class="distill-site-nav distill-site-footer">
<p> Copyright 2021 The SWELTR Project.</p>
<p>Site constructed using <a href="https://rstudio.github.io/distill/">Distill for R Markdown</a>.</p>
</div>
<!--/radix_placeholder_navigation_after_body-->


</body>

</html>
