<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-0.9.598">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="description" content="Reproducible workflow for assessing dominant taxonomic groups in the 16S rRNA and ITS data sets.">
<title>SWELTR - 4. Taxonomic Diversity</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./alpha.html" rel="next">
<link href="./filtering.html" rel="prev">
<link href="./assets/favicon.ico" rel="icon">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script><style>
    .quarto-title-block .quarto-title-banner {
      background: #113858;
    }
    </style>
<link rel="stylesheet" href="assets/css/styles.css">
</head>
<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg navbar-dark "><div class="navbar-container container-fluid">
      <a class="navbar-brand" href="./index.html">
    <img src="./assets/icon.png" alt=""></a>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item compact">
    <a class="nav-link" href="./index.html"><i class="bi bi-home" role="img">
</i> 
 </a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-a.-processing" role="button" data-bs-toggle="dropdown" aria-expanded="false">A. Processing</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-a.-processing">
<li class="dropdown-header">Workflows</li>
        <li><hr class="dropdown-divider"></li>
        <li>
    <a class="dropdown-item" href="./dada2.html">
 <span class="dropdown-text">1. DADA2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./data-prep.html">
 <span class="dropdown-text">2. Data Set Prep</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./filtering.html">
 <span class="dropdown-text">3. Filtering</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-b.-diversity" role="button" data-bs-toggle="dropdown" aria-expanded="false">B. Diversity</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-b.-diversity">
<li class="dropdown-header">Workflows</li>
        <li><hr class="dropdown-divider"></li>
        <li>
    <a class="dropdown-item" href="./taxa.html">
 <span class="dropdown-text">4. Taxonomic</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./alpha.html">
 <span class="dropdown-text">5. Alpha Diversity</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./beta.html">
 <span class="dropdown-text">6. Beta Diversity</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./da.html">
 <span class="dropdown-text">7. DA ASVs &amp; Taxa</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./metadata.html">
 <span class="dropdown-text">8. Metadata</span></a>
  </li>  
    </ul>
</li>
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item">
    <a class="nav-link active" href="./data-availability.html" aria-current="page">DATA</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./pub.html">PAPER</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./som.html">SOM</a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/sweltr/high-temp/"><i class="bi bi-github" role="img" aria-label="Agua Salud GitHub">
</i> 
 </a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="./about.html"><i class="bi bi-question" role="img">
</i> 
 </a>
  </li>  
</ul>
<div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }"><div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">4. Taxonomic Diversity</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title d-none d-lg-block">4. Taxonomic Diversity</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i></button></div></div>
                  <div>
        <div class="description">
          <p>Reproducible workflow for assessing dominant taxonomic groups in the 16S rRNA and ITS data sets.</p>
        </div>
      </div>
                </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto"><div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">A. Processing</a>
      <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./dada2.html" class="sidebar-item-text sidebar-link">1. DADA2 Workflow</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./data-prep.html" class="sidebar-item-text sidebar-link">2. Data Set Preparation</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./filtering.html" class="sidebar-item-text sidebar-link">3. Filtering</a>
  </div>
</li>
    </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">B. Diversity</a>
      <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./taxa.html" class="sidebar-item-text sidebar-link active">4. Taxonomic Diversity</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./alpha.html" class="sidebar-item-text sidebar-link">5. Alpha Diversity Estimates</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./beta.html" class="sidebar-item-text sidebar-link">6. Beta Diversity &amp; Dispersion Estimates</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./da.html" class="sidebar-item-text sidebar-link">7. Differentially Abundant ASVs &amp; Taxa</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./metadata.html" class="sidebar-item-text sidebar-link">8. Metadata Analysis</a>
  </div>
</li>
    </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">More</a>
      <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./data-availability.html" class="sidebar-item-text sidebar-link">Data Availability</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./som.html" class="sidebar-item-text sidebar-link">Electronic Supplementary Material</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./about.html" class="sidebar-item-text sidebar-link">Website Contruction</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pub.html" class="sidebar-item-text sidebar-link">Code for Publication Figures</a>
  </div>
</li>
    </ul>
</li>
    </ul>
</div>
</nav><!-- margin-sidebar --><div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li>
<a href="#synopsis" id="toc-synopsis" class="nav-link active" data-scroll-target="#synopsis">Synopsis</a>
  <ul class="collapse">
<li><a href="#workflow-input" id="toc-workflow-input" class="nav-link" data-scroll-target="#workflow-input">Workflow Input</a></li>
  </ul>
</li>
  <li>
<a href="#taxonomic-assessment" id="toc-taxonomic-assessment" class="nav-link" data-scroll-target="#taxonomic-assessment">Taxonomic Assessment</a>
  <ul class="collapse">
<li><a href="#s-rrna" id="toc-s-rrna" class="nav-link" data-scroll-target="#s-rrna">16s rRNA</a></li>
  <li><a href="#its" id="toc-its" class="nav-link" data-scroll-target="#its">ITS</a></li>
  </ul>
</li>
  <li><a href="#major-bacterial-taxa-by-family" id="toc-major-bacterial-taxa-by-family" class="nav-link" data-scroll-target="#major-bacterial-taxa-by-family">Major Bacterial taxa by Family</a></li>
  <li><a href="#workflow-output" id="toc-workflow-output" class="nav-link" data-scroll-target="#workflow-output">Workflow Output</a></li>
  </ul><div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/sweltr/high-temp/edit/main/taxa.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/sweltr/high-temp/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block page-columns page-full" id="quarto-document-content"><details><summary><strong>Click here</strong> for setup information.
</summary><div class="cell">
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>echo <span class="op">=</span> <span class="cn">TRUE</span>, eval <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">119</span><span class="op">)</span>
<span class="co">#library(conflicted)</span>
<span class="co">#pacman::p_depends(phangorn, local = TRUE)  </span>
<span class="co">#pacman::p_depends_reverse(phangorn, local = TRUE)  </span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/utils/packageDescription.html">packageVersion</a></span><span class="op">(</span><span class="st">"phyloseq"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings">Biostrings</a></span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/utils/packageDescription.html">packageVersion</a></span><span class="op">(</span><span class="st">"Biostrings"</span><span class="op">)</span>
<span class="fu">pacman</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pacman/man/p_load.html">p_load</a></span><span class="op">(</span><span class="va">tidyverse</span>, <span class="va">patchwork</span>,
               <span class="va">agricolae</span>, <span class="va">labdsv</span>, <span class="va">naniar</span>, 
               <span class="va">microbiome</span>, <span class="va">microeco</span>, <span class="va">file2meco</span>, <span class="va">cowplot</span>,
               <span class="va">GUniFrac</span>, <span class="va">ggalluvial</span>, <span class="va">ggdendro</span>, <span class="va">igraph</span>, <span class="va">reactable</span>,
               <span class="va">pheatmap</span>, <span class="va">microbiomeMarker</span>,
               <span class="va">captioner</span>, <span class="va">downloadthis</span>,
               install <span class="op">=</span> <span class="cn">FALSE</span>, update <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>scipen<span class="op">=</span><span class="fl">999</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html">opts_current</a></span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span>
  <span class="st">"cache"</span>,
  <span class="st">"cache.path"</span>,
  <span class="st">"cache.rebuild"</span>,
  <span class="st">"dependson"</span>,
  <span class="st">"autodep"</span>
<span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</details><div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<section id="synopsis" class="level1"><h1>Synopsis</h1>
<p>This workflow contains taxonomic diversity assessments for the 16S rRNA and ITS data sets. In order to run the workflow, you either need to first run the <a href="dada2.html">DADA2 Workflow</a>, then the <a href="data-prep.html">Data Preparation workflow</a>, and finally the <a href="filtering.html">Filtering workflow</a> or download the files linked below.</p>
<section id="workflow-input" class="level2"><h2 class="anchored" data-anchor-id="workflow-input">Workflow Input</h2>
<p>All files needed to run this workflow can be downloaded from figshare.</p>
<iframe src="https://widgets.figshare.com/articles/14701440/embed?show_title=1" width="100%" height="351" allowfullscreen="" frameborder="0">
</iframe>
<p>In this workflow, we use the <a href="https://github.com/ChiLiubio/microeco">microeco</a> to look at the taxonomic distribution of microbial communities.</p>
</section></section><section id="taxonomic-assessment" class="level1 page-columns page-full"><h1>Taxonomic Assessment</h1>
<section id="s-rrna" class="level2 page-columns page-full"><h2 class="anchored" data-anchor-id="s-rrna">16s rRNA</h2>
<div class="cell">

</div>
<div class="cell">

</div>
<p>Here we compare the taxonomic breakdown of the Full (unfiltered), Arbitrary filtered, PERfect filtered, and PIME filtered data sets, split by temperature treatment.</p>
<p>Before we plot the data, we want to separate out the Proteobacteria classes so we can plot these along with other phyla. To accomplish this we perform the following steps:</p>
<ol type="1">
<li>Get all Class-level Proteobacteria names</li>
</ol>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ssu18_data_sets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"ssu18_ps_work"</span>,  <span class="st">"ssu18_ps_filt"</span>,
                     <span class="st">"ssu18_ps_perfect"</span>, <span class="st">"ssu18_ps_pime"</span><span class="op">)</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">ssu18_data_sets</span><span class="op">)</span> <span class="op">{</span>
     <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_proteo"</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span>
     <span class="va">tmp_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_taxa-methods.html">subset_taxa</a></span><span class="op">(</span><span class="va">tmp_get</span>, <span class="va">Phylum</span> <span class="op">==</span> <span class="st">"Proteobacteria"</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_df</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_name</span><span class="op">)</span>
     <span class="va">tmp_get_taxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/get_taxa_unique.html">get_taxa_unique</a></span><span class="op">(</span><span class="va">tmp_df</span>,
                                     taxonomic.rank <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/rank_names.html">rank_names</a></span><span class="op">(</span><span class="va">tmp_df</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span>,
                                     errorIfNULL<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_get_taxa</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_proteo"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<ol start="2" type="1">
<li>Replace Phylum Proteobacteria with the Class name.</li>
</ol>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="va">ssu18_data_sets</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">j</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_proteo_clean"</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="va">j</span><span class="op">)</span>
  <span class="va">tmp_clean</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span>

   <span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html">nrow</a></span><span class="op">(</span><span class="va">tmp_clean</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
       <span class="kw">if</span> <span class="op">(</span><span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Proteobacteria"</span> <span class="op">&amp;</span> <span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">3</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Alphaproteobacteria"</span><span class="op">)</span><span class="op">{</span>
           <span class="va">phylum</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"Alphaproteobacteria"</span><span class="op">)</span>
           <span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">phylum</span>
   <span class="op">}</span>   <span class="kw">else</span> <span class="kw">if</span> <span class="op">(</span><span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Proteobacteria"</span> <span class="op">&amp;</span> <span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">3</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Gammaproteobacteria"</span><span class="op">)</span><span class="op">{</span>
           <span class="va">phylum</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"Gammaproteobacteria"</span><span class="op">)</span>
           <span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">phylum</span>
   <span class="op">}</span>   <span class="kw">else</span> <span class="kw">if</span> <span class="op">(</span><span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Proteobacteria"</span> <span class="op">&amp;</span> <span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">3</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Zetaproteobacteria"</span><span class="op">)</span><span class="op">{</span>
              <span class="va">phylum</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"Zetaproteobacteria"</span><span class="op">)</span>
           <span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">phylum</span>
   <span class="op">}</span>   <span class="kw">else</span> <span class="kw">if</span> <span class="op">(</span><span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Proteobacteria"</span> <span class="op">&amp;</span> <span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">3</span><span class="op">]</span> <span class="op">==</span> <span class="st">"p_Proteobacteria"</span><span class="op">)</span><span class="op">{</span>
           <span class="va">phylum</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"p_Proteobacteria"</span><span class="op">)</span>
           <span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">phylum</span>
       <span class="op">}</span>
     <span class="op">}</span>
  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">tmp_clean</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/rank_names.html">rank_names</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_get</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/get_taxa_unique.html">get_taxa_unique</a></span><span class="op">(</span><span class="va">tmp_get</span>,
                               taxonomic.rank <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/rank_names.html">rank_names</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,
                               errorIfNULL<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="st">"files/taxa/rdata/"</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">tmp_get</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="va">tmp_path</span>, <span class="va">j</span>, <span class="st">"_clean.rds"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">class</span>, <span class="va">order</span>, <span class="va">phylum</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/ls.html">objects</a></span><span class="op">(</span>pattern<span class="op">=</span><span class="st">"_proteo_clean"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<ol start="3" type="1">
<li>In order to use <code>microeco</code>, we need to add the rank designation as a prefix to each taxa. For example, <code>Actinobacteriota</code> is changed to <code>p__Actinobacteriota</code>.</li>
</ol>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">ssu18_data_sets</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_proteo_clean"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="co">#tmp_get &lt;- get(i)</span>
  <span class="co">#tmp_path &lt;- file.path("files/alpha/rdata/")</span>
  <span class="co">#tmp_read &lt;- readRDS(paste(tmp_path, i, ".rds", sep = ""))</span>
  <span class="va">tmp_sam_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span>
  <span class="va">tmp_tax_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Phylum</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"p_Proteobacteria"</span>, <span class="st">"Proteobacteria"</span>, <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Phylum</span><span class="op">)</span>

  <span class="co">#tmp_tax_data[,c(1:6)]</span>
  <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">ASV_ID</span> <span class="op">&lt;-</span> <span class="cn">NULL</span> <span class="co"># Some have, some do not</span>
  <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">ASV_SEQ</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
  
  <span class="va">tmp_tax_data</span><span class="op">[</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">lapply</a></span><span class="op">(</span><span class="va">tmp_tax_data</span>, <span class="va">gsub</span>, 
                                      pattern <span class="op">=</span> <span class="st">"^[k | p | c | o | f]_.*"</span>, 
                                      replacement <span class="op">=</span> <span class="st">""</span>, fixed <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Kingdom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="st">"k__"</span>, <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Kingdom</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
  <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Phylum</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="st">"p__"</span>, <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Phylum</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
  <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Class</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="st">"c__"</span>, <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Class</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
  <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="st">"o__"</span>, <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Order</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
  <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Family</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="st">"f__"</span>, <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Family</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
  <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Genus</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="st">"g__"</span>, <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Genus</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
  <span class="va">tmp_tax_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">tmp_tax_data</span><span class="op">)</span>
  
  <span class="va">tmp_ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phyloseq.html">phyloseq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span>,
                     <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phy_tree-methods.html">phy_tree</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span>,
                     <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">tmp_tax_data</span><span class="op">)</span>,
                     <span class="va">tmp_sam_data</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">i</span>, <span class="va">tmp_ps</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_proteo_clean"</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<ol start="4" type="1">
<li>Next, we need to covert each <code>phyloseq object</code> to a <code>microtable class</code>. The microtable class is the basic data structure for the <code>microeco</code> package and designed to store basic information from all the downstream analyses (e.g, alpha diversity, beta diversity, etc.). We use the <a href="https://github.com/ChiLiubio/file2meco">file2meco</a> to read the phyloseq object and convert into a microtable object. We can add <code>_me</code> as a suffix to each object to distiguish it from its’ phyloseq counterpart.</li>
</ol>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">ssu18_data_sets</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span>
    <span class="va">tmp_otu_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
    <span class="va">tmp_sample_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span>
    <span class="va">tmp_taxonomy_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span>
    <span class="va">tmp_phylo_tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phy_tree-methods.html">phy_tree</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span>
    <span class="va">tmp_taxonomy_table</span> <span class="op">%&lt;&gt;%</span> <span class="va">tidy_taxonomy</span>
    <span class="va">tmp_dataset</span> <span class="op">&lt;-</span> <span class="va">microtable</span><span class="op">$</span><span class="fu">new</span><span class="op">(</span>sample_table <span class="op">=</span> <span class="va">tmp_sample_info</span>, 
                                  otu_table <span class="op">=</span> <span class="va">tmp_otu_table</span>, 
                                  tax_table <span class="op">=</span> <span class="va">tmp_taxonomy_table</span>, 
                                  phylo_tree <span class="op">=</span> <span class="va">tmp_phylo_tree</span><span class="op">)</span>
    <span class="va">tmp_dataset</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_dataset</span><span class="op">)</span>
    <span class="va">tmp_dataset</span><span class="op">$</span><span class="va">tax_table</span> <span class="op">%&lt;&gt;%</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">Kingdom</span> <span class="op">==</span> <span class="st">"k__Archaea"</span> <span class="op">|</span> <span class="va">Kingdom</span> <span class="op">==</span> <span class="st">"k__Bacteria"</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_dataset</span><span class="op">)</span>
    <span class="va">tmp_dataset</span><span class="op">$</span><span class="fu">filter_pollution</span><span class="op">(</span>taxa <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"mitochondria"</span>, <span class="st">"chloroplast"</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_dataset</span><span class="op">)</span>
    <span class="va">tmp_dataset</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_dataset</span><span class="op">)</span>
    <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_me"</span><span class="op">)</span><span class="op">)</span> 
    <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_dataset</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>  
<span class="fu"><a href="https://rdrr.io/r/base/ls.html">objects</a></span><span class="op">(</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Here is an example of what a new microeco object looks like when you call it.</p>
<div class="cell">
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ssu18_ps_work_me</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>microtable class:
sample_table have 15 rows and 6 columns
otu_table have 20173 rows and 15 columns
tax_table have 20173 rows and 6 columns
phylo_tree have 20173 tips
Taxa abundance: calculated for Kingdom,Phylum,Class,Order,Family,Genus </code></pre>
</div>
</div>
<div class="cell">

</div>
<ol start="5" type="1">
<li>Now, we calculate the taxa abundance at each taxonomic rank using <code>cal_abund()</code>. This function return a list called <code>taxa_abund</code> containing several data frame of the abundance information at each taxonomic rank. The list is stored in the microtable object automatically. It’s worth noting that the <code>cal_abund()</code> function can be used to solve some complex cases, such as supporting both the relative and absolute abundance calculation and selecting the partial taxonomic columns. More information can be found in the description of the <a href="https://github.com/ChiLiubio/file2meco">file2meco package</a>. In the same loop we also create a <code>trans_abund</code> class, which is used to transform taxonomic abundance data for plotting.</li>
</ol>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">ssu18_data_sets</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_me</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_me"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_me</span><span class="op">$</span><span class="fu">cal_abund</span><span class="op">(</span><span class="op">)</span>
  <span class="va">tmp_me_abund</span> <span class="op">&lt;-</span> <span class="va">trans_abund</span><span class="op">$</span><span class="fu">new</span><span class="op">(</span>dataset <span class="op">=</span> <span class="va">tmp_me</span>, taxrank <span class="op">=</span> <span class="st">"Phylum"</span>, ntaxa <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>
  <span class="va">tmp_me_abund</span><span class="op">$</span><span class="va">abund_data</span><span class="op">$</span><span class="va">Abundance</span> <span class="op">&lt;-</span> <span class="va">tmp_me_abund</span><span class="op">$</span><span class="va">abund_data</span><span class="op">$</span><span class="va">Abundance</span> <span class="op">/</span> <span class="fl">100</span>
  <span class="va">tmp_me_abund_gr</span> <span class="op">&lt;-</span> <span class="va">trans_abund</span><span class="op">$</span><span class="fu">new</span><span class="op">(</span>dataset <span class="op">=</span> <span class="va">tmp_me</span>, taxrank <span class="op">=</span> <span class="st">"Phylum"</span>, ntaxa <span class="op">=</span> <span class="fl">12</span>, 
                                  groupmean <span class="op">=</span> <span class="st">"TEMP"</span><span class="op">)</span>
  <span class="va">tmp_me_abund_gr</span><span class="op">$</span><span class="va">abund_data</span><span class="op">$</span><span class="va">Abundance</span> <span class="op">&lt;-</span> <span class="va">tmp_me_abund_gr</span><span class="op">$</span><span class="va">abund_data</span><span class="op">$</span><span class="va">Abundance</span> <span class="op">/</span> <span class="fl">100</span>
  <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_me_abund"</span><span class="op">)</span><span class="op">)</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_me_abund</span><span class="op">)</span>  
  <span class="va">tmp_name_gr</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_me_abund_group"</span><span class="op">)</span><span class="op">)</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name_gr</span>, <span class="va">tmp_me_abund_gr</span><span class="op">)</span>  
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/ls.html">objects</a></span><span class="op">(</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<pre><code>The result is stored in object$taxa_abund ...</code></pre>
<ol start="6" type="1">
<li>I prefer to specify the order of taxa in these kinds of plots. We can look the top <code>ntaxa</code> (defined above) by accessing the <code>use_taxanames</code> character vector of each microtable object.</li>
</ol>
<div class="cell">

</div>
<p>Hum. Similar but not the same. This means we need to define the order for each separately. Once we do that, we will override the <code>use_taxanames</code> character vectors with the reordered vectors.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ssu18_ps_work_tax_ord</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"Alphaproteobacteria"</span>, <span class="st">"Gammaproteobacteria"</span>, <span class="st">"Acidobacteriota"</span>, 
                           <span class="st">"Actinobacteriota"</span>, <span class="st">"Bacteroidota"</span>, <span class="st">"Firmicutes"</span>, <span class="st">"Myxococcota"</span>, 
                           <span class="st">"Verrucomicrobiota"</span>, <span class="st">"Chloroflexi"</span>, <span class="st">"Planctomycetota"</span>, 
                           <span class="st">"Methylomirabilota"</span>, <span class="st">"Crenarchaeota"</span><span class="op">)</span>
<span class="va">ssu18_ps_filt_tax_ord</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"Alphaproteobacteria"</span>, <span class="st">"Gammaproteobacteria"</span>, <span class="st">"Acidobacteriota"</span>, 
                           <span class="st">"Actinobacteriota"</span>, <span class="st">"Bacteroidota"</span>, <span class="st">"Firmicutes"</span>, <span class="st">"Myxococcota"</span>, 
                           <span class="st">"Verrucomicrobiota"</span>, <span class="st">"RCP2-54"</span>, <span class="st">"Planctomycetota"</span>, 
                           <span class="st">"Methylomirabilota"</span>, <span class="st">"Crenarchaeota"</span><span class="op">)</span>
<span class="va">ssu18_ps_perfect_tax_ord</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"Alphaproteobacteria"</span>, <span class="st">"Gammaproteobacteria"</span>, <span class="st">"Acidobacteriota"</span>, 
                              <span class="st">"Actinobacteriota"</span>, <span class="st">"Bacteroidota"</span>, <span class="st">"Firmicutes"</span>, <span class="st">"Myxococcota"</span>, 
                              <span class="st">"Verrucomicrobiota"</span>, <span class="st">"RCP2-54"</span>, <span class="st">"Planctomycetota"</span>, 
                              <span class="st">"Methylomirabilota"</span>, <span class="st">"Crenarchaeota"</span><span class="op">)</span>
<span class="va">ssu18_ps_pime_tax_ord</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"Alphaproteobacteria"</span>, <span class="st">"Gammaproteobacteria"</span>, <span class="st">"Acidobacteriota"</span>, 
                           <span class="st">"Actinobacteriota"</span>, <span class="st">"Bacteroidota"</span>, <span class="st">"Firmicutes"</span>, <span class="st">"Myxococcota"</span>, 
                           <span class="st">"Verrucomicrobiota"</span>, <span class="st">"RCP2-54"</span>, <span class="st">"Nitrospirota"</span>, 
                           <span class="st">"Methylomirabilota"</span>, <span class="st">"Crenarchaeota"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<ol start="7" type="1">
<li>And one more little step before plotting. Here we <strong>a</strong>) specify a custom color palette and <strong>b</strong>) specify the sample order.</li>
</ol>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">top_level</span> <span class="op">&lt;-</span> <span class="st">"Phylum"</span>
<span class="va">ssu18_colvec.tax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"#00463C"</span>,<span class="st">"#FFD5FD"</span>,<span class="st">"#00A51C"</span>,<span class="st">"#C80B2A"</span>,<span class="st">"#00C7F9"</span>,<span class="st">"#FFA035"</span>,
                      <span class="st">"#ED0DFD"</span>,<span class="st">"#0063E5"</span>,<span class="st">"#5FFFDE"</span>,<span class="st">"#C00B6F"</span>,<span class="st">"#00A090"</span>,<span class="st">"#FF95BA"</span><span class="op">)</span>
<span class="va">ssu18_samp_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"P02_D00_010_C0A"</span>, <span class="st">"P04_D00_010_C0B"</span>, <span class="st">"P06_D00_010_C0C"</span>, <span class="st">"P08_D00_010_C0D"</span>, <span class="st">"P10_D00_010_C0E"</span>, 
                    <span class="st">"P01_D00_010_W3A"</span>, <span class="st">"P03_D00_010_W3B"</span>, <span class="st">"P05_D00_010_W3C"</span>, <span class="st">"P07_D00_010_W3D"</span>, <span class="st">"P09_D00_010_W3E"</span>, 
                    <span class="st">"P01_D00_010_W8A"</span>, <span class="st">"P03_D00_010_W8B"</span>, <span class="st">"P05_D00_010_W8C"</span>, <span class="st">"P07_D00_010_W8D"</span>, <span class="st">"P09_D00_010_W8E"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<ol start="8" type="1">
<li>Now we can generate plots (in a loop) for each faceted data set.</li>
</ol>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">ssu18_data_sets</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_abund</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_me_abund"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_tax_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_tax_ord"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_abund</span><span class="op">$</span><span class="va">use_taxanames</span> <span class="op">&lt;-</span> <span class="va">tmp_tax_order</span>
  <span class="va">tmp_facet_plot</span> <span class="op">&lt;-</span> <span class="va">tmp_abund</span><span class="op">$</span><span class="fu">plot_bar</span><span class="op">(</span>use_colors <span class="op">=</span> <span class="va">ssu18_colvec.tax</span>, 
                                       others_color <span class="op">=</span> <span class="st">"#323232"</span>, 
                                       facet <span class="op">=</span> <span class="st">"TEMP"</span>, 
                                       xtext_keep <span class="op">=</span> <span class="cn">TRUE</span>, 
                                       xtext_type_hor <span class="op">=</span> <span class="cn">FALSE</span>, 
                                       legend_text_italic <span class="op">=</span> <span class="cn">FALSE</span>, 
                                       xtext_size <span class="op">=</span> <span class="fl">6</span>, 
                                       facet_color <span class="op">=</span> <span class="st">"#cccccc"</span>, 
                                       order_x <span class="op">=</span> <span class="va">ssu18_samp_order</span><span class="op">)</span>
  <span class="va">tmp_facet_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_facet_plot"</span><span class="op">)</span><span class="op">)</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_facet_name</span>, <span class="va">tmp_facet_plot</span><span class="op">)</span>  
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Then add a little formatting to the faceted plots.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">set_to_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"ssu18_ps_work_facet_plot"</span>, <span class="st">"ssu18_ps_filt_facet_plot"</span>, <span class="st">"ssu18_ps_perfect_facet_plot"</span>, <span class="st">"ssu18_ps_pime_facet_plot"</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span>  <span class="va">set_to_plot</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span>
    <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="va">tmp_get</span> <span class="op">+</span>
    <span class="fu">theme_cowplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">guides</span><span class="op">(</span>fill <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>title <span class="op">=</span> <span class="va">top_level</span>, reverse <span class="op">=</span> <span class="cn">FALSE</span>,
                               keywidth <span class="op">=</span> <span class="fl">0.7</span>, keyheight <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">ylab</span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span> <span class="fu">xlab</span><span class="op">(</span><span class="st">"Sample"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">theme</span><span class="op">(</span>panel.grid.major <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>, panel.grid.minor <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        panel.background <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"transparent"</span>, colour <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>,
        plot.background <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"transparent"</span>, colour <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>,
        panel.border <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>fill <span class="op">=</span> <span class="cn">NA</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">7</span><span class="op">)</span>,
        legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
        legend.position <span class="op">=</span> <span class="st">"right"</span>,
        axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span>,
        axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">6</span>, angle <span class="op">=</span> <span class="fl">90</span><span class="op">)</span>,
        strip.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span>, angle <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,
        axis.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">i</span>, <span class="va">tmp_get</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span> 
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>And now plots for the groupmeans sets. We can use the same taxa order since that should not have changed.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">set_to_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"ssu18_ps_work_group_plot"</span>, <span class="st">"ssu18_ps_filt_group_plot"</span>, <span class="st">"ssu18_ps_perfect_group_plot"</span>, <span class="st">"ssu18_ps_pime_group_plot"</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">ssu18_data_sets</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_abund</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_me_abund_group"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_tax_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_tax_ord"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_abund</span><span class="op">$</span><span class="va">use_taxanames</span> <span class="op">&lt;-</span> <span class="va">tmp_tax_order</span>
  <span class="va">tmp_group_plot</span> <span class="op">&lt;-</span> <span class="va">tmp_abund</span><span class="op">$</span><span class="fu">plot_bar</span><span class="op">(</span>use_colors <span class="op">=</span> <span class="va">ssu18_colvec.tax</span>, 
                                       others_color <span class="op">=</span> <span class="st">"#323232"</span>, 
                                       xtext_keep <span class="op">=</span> <span class="cn">TRUE</span>, 
                                       xtext_type_hor <span class="op">=</span> <span class="cn">TRUE</span>, 
                                       legend_text_italic <span class="op">=</span> <span class="cn">FALSE</span>, 
                                       xtext_size <span class="op">=</span> <span class="fl">10</span>, 
                                       facet_color <span class="op">=</span> <span class="st">"#cccccc"</span><span class="op">)</span>
  <span class="va">tmp_group_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_group_plot"</span><span class="op">)</span><span class="op">)</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_group_name</span>, <span class="va">tmp_group_plot</span><span class="op">)</span>  
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Let’s also add a little formatting to the groupmean plots.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span>  <span class="va">set_to_plot</span><span class="op">)</span> <span class="op">{</span>
     <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span>
     <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="va">tmp_get</span> <span class="op">+</span>
       <span class="fu">theme_cowplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
       <span class="fu">ylab</span><span class="op">(</span><span class="st">"Relative Abundance (% total reads)"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">xlab</span><span class="op">(</span><span class="st">"Temperature"</span><span class="op">)</span> <span class="op">+</span>
       <span class="fu">theme</span><span class="op">(</span>panel.grid.major <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>, panel.grid.minor <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
             panel.background <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"transparent"</span>, colour <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>,
             plot.background <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"transparent"</span>, colour <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>,
             panel.border <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>fill <span class="op">=</span> <span class="cn">NA</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
             legend.position <span class="op">=</span> <span class="st">"none"</span>,
             axis.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span>,
             axis.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
       <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="op">)</span>

     <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">i</span>, <span class="va">tmp_get</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<ol start="10" type="1">
<li>Finally we use the <code>patchwork</code> package to combine the two plots and customize the look.</li>
</ol>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="co">## single index that acts as an index for referencing elements (variables) in a list</span>
<span class="co">##  solution modified from this SO answerhttps://stackoverflow.com/a/54451460</span>
<span class="va">var_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
  var1 <span class="op">=</span> <span class="va">ssu18_data_sets</span>,
  var2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"FULL"</span>, <span class="st">"Arbitrary filtered"</span>, <span class="st">"PERfect filtered"</span>, <span class="st">"PIME filtered"</span><span class="op">)</span>
<span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span>  <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">var_list</span><span class="op">$</span><span class="va">var1</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
     <span class="va">tmp_plot_final_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">var_list</span><span class="op">$</span><span class="va">var1</span><span class="op">[</span><span class="va">j</span><span class="op">]</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_"</span>, <span class="va">top_level</span>, <span class="st">"_plot_final"</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_set_type</span> <span class="op">&lt;-</span> <span class="va">var_list</span><span class="op">$</span><span class="va">var2</span><span class="op">[</span><span class="va">j</span><span class="op">]</span>
     <span class="va">tmp_p_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">var_list</span><span class="op">$</span><span class="va">var1</span><span class="op">[</span><span class="va">j</span><span class="op">]</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_group_plot"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_m_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">var_list</span><span class="op">$</span><span class="va">var1</span><span class="op">[</span><span class="va">j</span><span class="op">]</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_facet_plot"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_plot_final</span> <span class="op">&lt;-</span> <span class="va">tmp_p_plot</span> <span class="op">+</span> <span class="va">tmp_m_plot</span>
     <span class="va">tmp_plot_final</span> <span class="op">&lt;-</span> <span class="va">tmp_plot_final</span> <span class="op">+</span>
         <span class="fu">plot_annotation</span><span class="op">(</span>tag_levels <span class="op">=</span> <span class="st">"a"</span>, 
                         title <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="st">"Top"</span>, <span class="va">top_level</span>, <span class="st">"of"</span>, <span class="va">tmp_set_type</span>, <span class="st">"data set"</span><span class="op">)</span>, 
                         <span class="co">#subtitle = 'Top taxa of non-filtered data',</span>
                         caption <span class="op">=</span> <span class="st">"a) grouped by temperature.,
                         b) All samples, faceted by temperature."</span><span class="op">)</span> <span class="op">+</span> 
       <span class="fu">plot_layout</span><span class="op">(</span>widths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">&amp;</span>
       <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">13</span><span class="op">)</span>,
             plot.subtitle <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>, 
             plot.tag <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>, 
             axis.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>, 
             axis.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span><span class="op">)</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_plot_final_name</span>, <span class="va">tmp_plot_final</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">

</div>
<div class="cell">

</div>
<p><br></p>
<blockquote class="blockquote">
<p>It is important to note that while most of the dominant taxa are the same across the FULL, Arbitrary, PIME, and PERfect data sets, there a few key difference:</p>
</blockquote>
<ol type="1">
<li><p>Planctomycetota was not among the dominant taxa in the PIME data set. Nitrospirota was substituted.</p></li>
<li><p>Chloroflexi was not among the dominant taxa in either the Arbitrary, PERfect, or PIME data sets. RCP2-54 was substituted in both cases.</p></li>
</ol>
<div class="column-body-outset">
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true">FULL (unfiltered)</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false">Arbitrary filtered</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-3" role="tab" aria-controls="tabset-1-3" aria-selected="false">PERFECT filtered</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-4" role="tab" aria-controls="tabset-1-4" aria-selected="false">PIME filtered</a></li>
</ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="include/taxa/ssu18_ps_work_tax_div_bar_plots.png" class="img-fluid" width="1240"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 1 | Dominant taxonomic groups from the FULL (unfiltered) 16S rRNA data set. A) Grouped by temperature and B) all samples faceted by temperature treatment.</strong></small></p>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="include/taxa/ssu18_ps_filt_tax_div_bar_plots.png" class="img-fluid" width="1240"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 2 | Dominant taxonomic groups from the Arbitrary filtered 16S rRNA data set. A) Grouped by temperature and B) all samples faceted by temperature treatment.</strong></small></p>
</div>
<div id="tabset-1-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-3-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="include/taxa/ssu18_ps_perfect_tax_div_bar_plots.png" class="img-fluid" width="1240"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 3 | Dominant taxonomic groups from the PERfect filtered 16S rRNA data set. A) Grouped by temperature and B) all samples faceted by temperature treatment.</strong></small></p>
</div>
<div id="tabset-1-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-4-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="include/taxa/ssu18_ps_pime_tax_div_bar_plots.png" class="img-fluid" width="1240"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 4 | Dominant taxonomic groups from the PIME filtered 16S rRNA data set. A) Grouped by temperature and B) all samples faceted by temperature treatment.</strong></small></p>
</div>
</div>
</div>
</div>
</section><section id="its" class="level2 page-columns page-full"><h2 class="anchored" data-anchor-id="its">ITS</h2>
<div class="cell">

</div>
<div class="cell">

</div>
<p>Here we compare the taxonomic breakdown of the FULL (unfiltered), PIME filtered, and PERfect filtered data sets, split by temperature treatment for the ITS data. The workflow is basically the same as the 16S rRNA data. Click the box below to see the entire workflow.</p>
<div class="cell">

</div>
<details><summary><strong>Detailed workflow</strong> for ITS taxonomic analysis.
</summary><ol type="1">
<li>Choose the <strong>number</strong> of taxa to display and the taxonomic <strong>level</strong>. Aggregate the rest into “Other”.</li>
</ol>
<div class="cell">
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">top_hits</span> <span class="op">&lt;-</span> <span class="fl">14</span>
<span class="va">top_level</span> <span class="op">&lt;-</span> <span class="st">"Order"</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>As above, in order to use <code>microeco</code>, we need to add the rank designation as a prefix to each taxa. For example, <code>Basidiomycota</code> is changed to <code>p__Basidiomycota</code>.</p>
<div class="cell">
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">its18_data_sets</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span>
  <span class="va">tmp_sam_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span>
  <span class="va">tmp_tax_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">ASV_ID</span> <span class="op">&lt;-</span> <span class="cn">NULL</span> <span class="co"># Some have, some do not</span>
  <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">ASV_SEQ</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
  
<span class="co">#  tmp_tax_data[] &lt;- data.frame(lapply(tmp_tax_data, gsub, </span>
<span class="co">#                                      pattern = "^[k | p | c | o | f]_.*", </span>
<span class="co">#                                      replacement = "", fixed = FALSE))</span>
  <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Kingdom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="st">"k__"</span>, <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Kingdom</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
  <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Phylum</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="st">"p__"</span>, <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Phylum</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
  <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Class</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="st">"c__"</span>, <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Class</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
  <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="st">"o__"</span>, <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Order</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
  <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Family</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="st">"f__"</span>, <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Family</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
  <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Genus</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="st">"g__"</span>, <span class="va">tmp_tax_data</span><span class="op">$</span><span class="va">Genus</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
  <span class="va">tmp_tax_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">tmp_tax_data</span><span class="op">)</span>
  
  <span class="va">tmp_ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phyloseq.html">phyloseq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span>,
                     <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">tmp_tax_data</span><span class="op">)</span>,
                     <span class="va">tmp_sam_data</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">i</span>, <span class="va">tmp_ps</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">its18_ps_pime</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">its18_data_sets</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span>
    <span class="va">tmp_otu_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
    <span class="va">tmp_sample_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span>
    <span class="va">tmp_taxonomy_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span>
    <span class="co">#tmp_taxonomy_table %&lt;&gt;% tidy_taxonomy</span>
    <span class="va">tmp_dataset</span> <span class="op">&lt;-</span> <span class="va">microtable</span><span class="op">$</span><span class="fu">new</span><span class="op">(</span>sample_table <span class="op">=</span> <span class="va">tmp_sample_info</span>, 
                                  otu_table <span class="op">=</span> <span class="va">tmp_otu_table</span>, 
                                  tax_table <span class="op">=</span> <span class="va">tmp_taxonomy_table</span><span class="op">)</span>
    <span class="va">tmp_dataset</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_dataset</span><span class="op">)</span>
    <span class="va">tmp_dataset</span><span class="op">$</span><span class="va">tax_table</span> <span class="op">%&lt;&gt;%</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">Kingdom</span> <span class="op">==</span> <span class="st">"k__Fungi"</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_dataset</span><span class="op">)</span>
    <span class="va">tmp_dataset</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_dataset</span><span class="op">)</span>
    <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_me"</span><span class="op">)</span><span class="op">)</span> 
    <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_dataset</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>  
<span class="fu"><a href="https://rdrr.io/r/base/ls.html">objects</a></span><span class="op">(</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Here is an example of what a new microeco object looks like when you call it.</p>
<div class="cell">
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">its18_ps_work_me</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>microtable class:
sample_table have 13 rows and 6 columns
otu_table have 3355 rows and 13 columns
tax_table have 3355 rows and 6 columns
Taxa abundance: calculated for Kingdom,Phylum,Class,Order,Family,Genus </code></pre>
</div>
</div>
<div class="cell">

</div>
<ol start="5" type="1">
<li>Now, we calculate the taxa abundance at each taxonomic rank using <code>cal_abund()</code>. This function return a list called <code>taxa_abund</code> containing several data frame of the abundance information at each taxonomic rank. The list is stored in the microtable object automatically. It’s worth noting that the <code>cal_abund()</code> function can be used to solve some complex cases, such as supporting both the relative and absolute abundance calculation and selecting the partial taxonomic columns. More information can be found in the description of the <a href="https://github.com/ChiLiubio/file2meco">file2meco package</a>. In the same loop we also create a <code>trans_abund</code> class, which is used to transform taxonomic abundance data for plotting.</li>
</ol>
<div class="cell">
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">its18_data_sets</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_me</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_me"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_me</span><span class="op">$</span><span class="fu">cal_abund</span><span class="op">(</span><span class="op">)</span>
  <span class="va">tmp_me_abund</span> <span class="op">&lt;-</span> <span class="va">trans_abund</span><span class="op">$</span><span class="fu">new</span><span class="op">(</span>dataset <span class="op">=</span> <span class="va">tmp_me</span>, taxrank <span class="op">=</span> <span class="va">top_level</span>, ntaxa <span class="op">=</span> <span class="va">top_hits</span><span class="op">)</span>
  <span class="va">tmp_me_abund</span><span class="op">$</span><span class="va">abund_data</span><span class="op">$</span><span class="va">Abundance</span> <span class="op">&lt;-</span> <span class="va">tmp_me_abund</span><span class="op">$</span><span class="va">abund_data</span><span class="op">$</span><span class="va">Abundance</span> <span class="op">/</span> <span class="fl">100</span>
  <span class="va">tmp_me_abund_gr</span> <span class="op">&lt;-</span> <span class="va">trans_abund</span><span class="op">$</span><span class="fu">new</span><span class="op">(</span>dataset <span class="op">=</span> <span class="va">tmp_me</span>, taxrank <span class="op">=</span> <span class="va">top_level</span>, ntaxa <span class="op">=</span> <span class="va">top_hits</span>, 
                                  groupmean <span class="op">=</span> <span class="st">"TEMP"</span><span class="op">)</span>
  <span class="va">tmp_me_abund_gr</span><span class="op">$</span><span class="va">abund_data</span><span class="op">$</span><span class="va">Abundance</span> <span class="op">&lt;-</span> <span class="va">tmp_me_abund_gr</span><span class="op">$</span><span class="va">abund_data</span><span class="op">$</span><span class="va">Abundance</span> <span class="op">/</span> <span class="fl">100</span>
  <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_me_abund"</span><span class="op">)</span><span class="op">)</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_me_abund</span><span class="op">)</span>  
  <span class="va">tmp_name_gr</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_me_abund_group"</span><span class="op">)</span><span class="op">)</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name_gr</span>, <span class="va">tmp_me_abund_gr</span><span class="op">)</span>  
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>The result is stored in object$taxa_abund ...</code></pre>
<ol start="6" type="1">
<li>I prefer to specify the order of taxa in these kinds of plots. We can look the top <code>ntaxa</code> (defined above) by accessing the <code>use_taxanames</code> character vector of each microtable object.</li>
</ol>
<div class="cell">

</div>
<p>Hum. Similar but not the same. This means we need to define the order for each separately. Once we do that, we will override the <code>use_taxanames</code> character vectors with the reordered vectors.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">its18_ps_work_tax_ord</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span>
  <span class="st">"k_Fungi"</span>,  <span class="st">"p_Ascomycota"</span>,  <span class="st">"c_Agaricomycetes"</span>, <span class="st">"Agaricales"</span>,  
  <span class="st">"Archaeorhizomycetales"</span>,  <span class="st">"Capnodiales"</span>,  <span class="st">"Eurotiales"</span>,  <span class="st">"Geastrales"</span>,  
  <span class="st">"Glomerales"</span>,  <span class="st">"Helotiales"</span>, <span class="st">"Hypocreales"</span>,  <span class="st">"Saccharomycetales"</span>,  
  <span class="st">"Trichosporonales"</span>,  <span class="st">"Xylariales"</span><span class="op">)</span><span class="op">)</span>   

<span class="va">its18_ps_filt_tax_ord</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span>
  <span class="st">"k_Fungi"</span>,  <span class="st">"p_Ascomycota"</span>,  <span class="st">"Polyporales"</span>, <span class="st">"Agaricales"</span>,  
  <span class="st">"Archaeorhizomycetales"</span>,  <span class="st">"Capnodiales"</span>,  <span class="st">"Eurotiales"</span>,  <span class="st">"Geastrales"</span>,  
  <span class="st">"Glomerales"</span>,  <span class="st">"Mortierellales"</span>, <span class="st">"Hypocreales"</span>,  <span class="st">"Saccharomycetales"</span>,  
  <span class="st">"Trichosporonales"</span>,  <span class="st">"Xylariales"</span><span class="op">)</span><span class="op">)</span>   

<span class="va">its18_ps_perfect_tax_ord</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span>
  <span class="st">"k_Fungi"</span>,  <span class="st">"p_Ascomycota"</span>,  <span class="st">"c_Agaricomycetes"</span>, <span class="st">"Agaricales"</span>,  
  <span class="st">"Archaeorhizomycetales"</span>,  <span class="st">"Capnodiales"</span>,  <span class="st">"Eurotiales"</span>,  <span class="st">"Geastrales"</span>,  
  <span class="st">"Glomerales"</span>,  <span class="st">"Helotiales"</span>,   <span class="st">"Hypocreales"</span>,  <span class="st">"Saccharomycetales"</span>, 
  <span class="st">"Trichosporonales"</span>,  <span class="st">"Pezizales"</span><span class="op">)</span><span class="op">)</span>   
<span class="va">its18_ps_pime_tax_ord</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span>
  <span class="st">"k_Fungi"</span>,  <span class="st">"p_Ascomycota"</span>,  <span class="st">"Polyporales"</span>,  <span class="st">"Agaricales"</span>,  
  <span class="st">"Archaeorhizomycetales"</span>,  <span class="st">"Capnodiales"</span>,  <span class="st">"Eurotiales"</span>,  <span class="st">"Geastrales"</span>, 
  <span class="st">"Glomerales"</span>,  <span class="st">"Mortierellales"</span>,  <span class="st">"Hypocreales"</span>,  <span class="st">"Saccharomycetales"</span>,  
  <span class="st">"Trichosporonales"</span>,  <span class="st">"Pezizales"</span><span class="op">)</span><span class="op">)</span>   </code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<ol start="7" type="1">
<li>And one more little step before plotting. Here we <strong>a</strong>) specify a custom color palette and <strong>b</strong>) specify the sample order.</li>
</ol>
<div class="cell">
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">its18_colvec.tax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"#323232"</span>, <span class="st">"#004949"</span>, <span class="st">"#924900"</span>, <span class="st">"#490092"</span>, 
                      <span class="st">"#6db6ff"</span>, <span class="st">"#920000"</span>, <span class="st">"#ffb6db"</span>, <span class="st">"#24ff24"</span>, 
                      <span class="st">"#006ddb"</span>, <span class="st">"#db6d00"</span>, <span class="st">"#b66dff"</span>, <span class="st">"#ffff6d"</span>, 
                      <span class="st">"#009292"</span>, <span class="st">"#b6dbff"</span>, <span class="st">"#ff6db6"</span><span class="op">)</span><span class="op">)</span>
<span class="va">its18_samp_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"P02_D00_010_C0A"</span>, <span class="st">"P04_D00_010_C0B"</span>, <span class="st">"P06_D00_010_C0C"</span>, <span class="st">"P08_D00_010_C0D"</span>, <span class="st">"P10_D00_010_C0E"</span>, 
                    <span class="st">"P01_D00_010_W3A"</span>, <span class="st">"P03_D00_010_W3B"</span>, <span class="st">"P07_D00_010_W3D"</span>, <span class="st">"P09_D00_010_W3E"</span>, 
                    <span class="st">"P01_D00_010_W8A"</span>, <span class="st">"P03_D00_010_W8B"</span>, <span class="st">"P05_D00_010_W8C"</span>, <span class="st">"P07_D00_010_W8D"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="8" type="1">
<li>Now we can generate plots (in a loop) for each faceted data set.</li>
</ol>
<div class="cell">
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">its18_data_sets</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_abund</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_me_abund"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_tax_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_tax_ord"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_abund</span><span class="op">$</span><span class="va">use_taxanames</span> <span class="op">&lt;-</span> <span class="va">tmp_tax_order</span>
  <span class="va">tmp_facet_plot</span> <span class="op">&lt;-</span> <span class="va">tmp_abund</span><span class="op">$</span><span class="fu">plot_bar</span><span class="op">(</span>use_colors <span class="op">=</span> <span class="va">its18_colvec.tax</span>, 
                                       others_color <span class="op">=</span> <span class="st">"#323232"</span>, 
                                       facet <span class="op">=</span> <span class="st">"TEMP"</span>, 
                                       xtext_keep <span class="op">=</span> <span class="cn">TRUE</span>, 
                                       xtext_type_hor <span class="op">=</span> <span class="cn">FALSE</span>, 
                                       legend_text_italic <span class="op">=</span> <span class="cn">FALSE</span>, 
                                       xtext_size <span class="op">=</span> <span class="fl">6</span>, 
                                       facet_color <span class="op">=</span> <span class="st">"#cccccc"</span>, 
                                       order_x <span class="op">=</span> <span class="va">its18_samp_order</span><span class="op">)</span>
  <span class="va">tmp_facet_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_facet_plot"</span><span class="op">)</span><span class="op">)</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_facet_name</span>, <span class="va">tmp_facet_plot</span><span class="op">)</span>  
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Then add a little formatting to the faceted plots.</p>
<div class="cell">
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">set_to_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"its18_ps_work_facet_plot"</span>, <span class="st">"its18_ps_filt_facet_plot"</span>, <span class="st">"its18_ps_perfect_facet_plot"</span>, <span class="st">"its18_ps_pime_facet_plot"</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span>  <span class="va">set_to_plot</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span>
    <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="va">tmp_get</span> <span class="op">+</span>
    <span class="fu">theme_cowplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">guides</span><span class="op">(</span>fill <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>title <span class="op">=</span> <span class="va">top_level</span>, reverse <span class="op">=</span> <span class="cn">FALSE</span>,
                               keywidth <span class="op">=</span> <span class="fl">0.7</span>, keyheight <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">ylab</span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span> <span class="fu">xlab</span><span class="op">(</span><span class="st">"Sample"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">theme</span><span class="op">(</span>panel.grid.major <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>, panel.grid.minor <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        panel.background <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"transparent"</span>, colour <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>,
        plot.background <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"transparent"</span>, colour <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>,
        panel.border <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>fill <span class="op">=</span> <span class="cn">NA</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">7</span><span class="op">)</span>,
        legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
        legend.position <span class="op">=</span> <span class="st">"right"</span>,
        axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span>,
        axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">6</span>, angle <span class="op">=</span> <span class="fl">90</span><span class="op">)</span>,
        strip.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span>, angle <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,
        axis.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">i</span>, <span class="va">tmp_get</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span> 
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>And now plots for the groupmeans sets. We can use the same taxa order since that should not have changed.</p>
<div class="cell">
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">set_to_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"its18_ps_work_group_plot"</span>, <span class="st">"its18_ps_filt_group_plot"</span>, <span class="st">"its18_ps_perfect_group_plot"</span>, <span class="st">"its18_ps_pime_group_plot"</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">its18_data_sets</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_abund</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_me_abund_group"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_tax_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_tax_ord"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_abund</span><span class="op">$</span><span class="va">use_taxanames</span> <span class="op">&lt;-</span> <span class="va">tmp_tax_order</span>
  <span class="va">tmp_group_plot</span> <span class="op">&lt;-</span> <span class="va">tmp_abund</span><span class="op">$</span><span class="fu">plot_bar</span><span class="op">(</span>use_colors <span class="op">=</span> <span class="va">its18_colvec.tax</span>, 
                                       others_color <span class="op">=</span> <span class="st">"#323232"</span>, 
                                       xtext_keep <span class="op">=</span> <span class="cn">TRUE</span>, 
                                       xtext_type_hor <span class="op">=</span> <span class="cn">TRUE</span>, 
                                       legend_text_italic <span class="op">=</span> <span class="cn">FALSE</span>, 
                                       xtext_size <span class="op">=</span> <span class="fl">10</span>, 
                                       facet_color <span class="op">=</span> <span class="st">"#cccccc"</span><span class="op">)</span>
  <span class="va">tmp_group_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_group_plot"</span><span class="op">)</span><span class="op">)</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_group_name</span>, <span class="va">tmp_group_plot</span><span class="op">)</span>  
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Let’s also add a little formatting to the groupmean plots.</p>
<div class="cell">
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span>  <span class="va">set_to_plot</span><span class="op">)</span> <span class="op">{</span>
     <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span>
     <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="va">tmp_get</span> <span class="op">+</span>
       <span class="fu">theme_cowplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
       <span class="fu">ylab</span><span class="op">(</span><span class="st">"Relative Abundance (% total reads)"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">xlab</span><span class="op">(</span><span class="st">"Temperature"</span><span class="op">)</span> <span class="op">+</span>
       <span class="fu">theme</span><span class="op">(</span>panel.grid.major <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>, panel.grid.minor <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
             panel.background <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"transparent"</span>, colour <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>,
             plot.background <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"transparent"</span>, colour <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>,
             panel.border <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>fill <span class="op">=</span> <span class="cn">NA</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
             legend.position <span class="op">=</span> <span class="st">"none"</span>,
             axis.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span>,
             axis.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
       <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="op">)</span>

     <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">i</span>, <span class="va">tmp_get</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="10" type="1">
<li>Finally we use the <code>patchwork</code> package to combine the two plots and customize the look.</li>
</ol>
<div class="cell">
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="co">## single index that acts as an index for referencing elements (variables) in a list</span>
<span class="co">##  solution modified from this SO answerhttps://stackoverflow.com/a/54451460</span>
<span class="va">var_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
  var1 <span class="op">=</span> <span class="va">its18_data_sets</span>,
  var2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"FULL"</span>, <span class="st">"Arbitrary filtered"</span>, <span class="st">"PERfect filtered"</span>, <span class="st">"PIME filtered"</span><span class="op">)</span>
<span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span>  <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">var_list</span><span class="op">$</span><span class="va">var1</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
     <span class="va">tmp_plot_final_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">var_list</span><span class="op">$</span><span class="va">var1</span><span class="op">[</span><span class="va">j</span><span class="op">]</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_"</span>, <span class="va">top_level</span>, <span class="st">"_plot_final"</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_set_type</span> <span class="op">&lt;-</span> <span class="va">var_list</span><span class="op">$</span><span class="va">var2</span><span class="op">[</span><span class="va">j</span><span class="op">]</span>
     <span class="va">tmp_p_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">var_list</span><span class="op">$</span><span class="va">var1</span><span class="op">[</span><span class="va">j</span><span class="op">]</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_group_plot"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_m_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">var_list</span><span class="op">$</span><span class="va">var1</span><span class="op">[</span><span class="va">j</span><span class="op">]</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_facet_plot"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_plot_final</span> <span class="op">&lt;-</span> <span class="va">tmp_p_plot</span> <span class="op">+</span> <span class="va">tmp_m_plot</span>
     <span class="va">tmp_plot_final</span> <span class="op">&lt;-</span> <span class="va">tmp_plot_final</span> <span class="op">+</span>
         <span class="fu">plot_annotation</span><span class="op">(</span>tag_levels <span class="op">=</span> <span class="st">"a"</span>, 
                         title <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="st">"Top"</span>, <span class="va">top_level</span>, <span class="st">"of"</span>, <span class="va">tmp_set_type</span>, <span class="st">"data set"</span><span class="op">)</span>, 
                         <span class="co">#subtitle = 'Top taxa of non-filtered data',</span>
                         caption <span class="op">=</span> <span class="st">"a) grouped by temperature.,
                         b) All samples, faceted by temperature."</span><span class="op">)</span> <span class="op">+</span> 
       <span class="fu">plot_layout</span><span class="op">(</span>widths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">&amp;</span>
       <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">13</span><span class="op">)</span>,
             plot.subtitle <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>, 
             plot.tag <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>, 
             axis.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>, 
             axis.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span><span class="op">)</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_plot_final_name</span>, <span class="va">tmp_plot_final</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">

</div>
<div class="cell">

</div>
</details><p><br></p>
<blockquote class="blockquote">
<p>It is important to note that while most of the dominant taxa are the same across the FULL, Arbitrary, PIME, and PERfect data sets, there a few key difference:</p>
</blockquote>
<ol type="1">
<li><p>c_Agaricomycetes &amp; Helotiales were not among the dominant taxa in the Arbitrary or PIME filtered data sets. Polyporales &amp; Mortierellales were substituted, respectively.</p></li>
<li><p>Xylariales was not among the dominant taxa in either the PIME or PERfect data sets. Pezizales was substituted in both cases.</p></li>
</ol>
<div class="column-body-outset">
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-2-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-1" role="tab" aria-controls="tabset-2-1" aria-selected="true">FULL (unfiltered)</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-2" role="tab" aria-controls="tabset-2-2" aria-selected="false">Arbitrary filtered</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-3" role="tab" aria-controls="tabset-2-3" aria-selected="false">PERFECT filtered</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-4" role="tab" aria-controls="tabset-2-4" aria-selected="false">PIME filtered</a></li>
</ul>
<div class="tab-content">
<div id="tabset-2-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-2-1-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="include/taxa/its18_ps_work_tax_div_bar_plots.png" class="img-fluid" width="1240"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 5 | Dominant taxonomic groups from the FULL (unfiltered) ITS data set. A) Grouped by temperature and B) all samples faceted by temperature treatment.</strong></small></p>
</div>
<div id="tabset-2-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-2-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="include/taxa/its18_ps_filt_tax_div_bar_plots.png" class="img-fluid" width="1240"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 6 | Dominant taxonomic groups from the Arbitrary filtered ITS data set. A) Grouped by temperature and B) all samples faceted by temperature treatment.</strong></small></p>
</div>
<div id="tabset-2-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-3-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="include/taxa/its18_ps_perfect_tax_div_bar_plots.png" class="img-fluid" width="1240"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 7 | Dominant taxonomic groups from the PERfect filtered ITS data set. A) Grouped by temperature and B) all samples faceted by temperature treatment.</strong></small></p>
</div>
<div id="tabset-2-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-4-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="include/taxa/its18_ps_pime_tax_div_bar_plots.png" class="img-fluid" width="1240"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 8 | Dominant taxonomic groups from the PIME filtered ITS data set. A) Grouped by temperature and B) all samples faceted by temperature treatment.</strong></small></p>
</div>
</div>
</div>
</div>
</section></section><section id="major-bacterial-taxa-by-family" class="level1 page-columns page-full"><h1>Major Bacterial taxa by Family</h1>
<p>We also look at the relative abundance of groups within dominant Phyla. <strong>Please note</strong> these analyses are for the FULL data set only, before any type of filtering. For this analysis, we use the phyloseq object created in the Data Preparation wokflow section <a href="data-prep.html#rename-na-taxonomic-ranks">Rename NA taxonomic ranks</a> where unclassified ranks were renamed by the next highest rank. See step #1 below for an example of the taxa table structure.</p>
<div class="callout-note callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>This code <strong><em>only</em></strong> works for multiple taxa from a <em>single</em> phyloseq object.</p>
</div>
</div>
<details><summary><strong>Detailed workflow</strong> for assessing Bacterial families.
</summary><ol type="1">
<li>Subset Phyla of interest.</li>
</ol>
<div class="cell">
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ssu18_ps_work</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"files/data-prep/rdata/ssu18_ps_work.rds"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ssu18_ps_work</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>          Kingdom          Phylum               Class              Order
ASV9128  Bacteria Acidobacteriota         Subgroup_22      c_Subgroup_22
ASV7870  Bacteria Planctomycetota               OM190            c_OM190
ASV8537  Bacteria     Chloroflexi        Anaerolineae            SBR1031
ASV15357 Bacteria  Proteobacteria Gammaproteobacteria Diplorickettsiales
ASV19533 Bacteria      k_Bacteria          k_Bacteria         k_Bacteria
ASV18180 Bacteria Planctomycetota      Planctomycetes         Gemmatales
                      Family                 Genus
ASV9128        c_Subgroup_22         c_Subgroup_22
ASV7870              c_OM190               c_OM190
ASV8537                  A4b                 f_A4b
ASV15357 Diplorickettsiaceae f_Diplorickettsiaceae
ASV19533          k_Bacteria            k_Bacteria
ASV18180         Gemmataceae         f_Gemmataceae</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">ssu18_data_sets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"ssu18_ps_work"</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">ssu18_data_sets</span><span class="op">)</span> <span class="op">{</span>
     <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_proteo"</span><span class="op">)</span><span class="op">)</span>
     <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span>
     <span class="va">tmp_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_taxa-methods.html">subset_taxa</a></span><span class="op">(</span><span class="va">tmp_get</span>, <span class="va">Phylum</span> <span class="op">==</span> <span class="st">"Proteobacteria"</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_df</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_name</span><span class="op">)</span>
     <span class="va">tmp_get_taxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/get_taxa_unique.html">get_taxa_unique</a></span><span class="op">(</span><span class="va">tmp_df</span>,
                                     taxonomic.rank <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/rank_names.html">rank_names</a></span><span class="op">(</span><span class="va">tmp_df</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span>,
                                     errorIfNULL<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">tmp_get_taxa</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_proteo"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="2" type="1">
<li>Replace Phylum Proteobacteria with the Class name.</li>
</ol>
<div class="cell">
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="va">ssu18_data_sets</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">j</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_proteo_clean"</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="va">j</span><span class="op">)</span>
  <span class="va">tmp_clean</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span>

   <span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html">nrow</a></span><span class="op">(</span><span class="va">tmp_clean</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
       <span class="kw">if</span> <span class="op">(</span><span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Proteobacteria"</span> <span class="op">&amp;</span> <span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">3</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Alphaproteobacteria"</span><span class="op">)</span><span class="op">{</span>
           <span class="va">phylum</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"Alphaproteobacteria"</span><span class="op">)</span>
           <span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">phylum</span>
   <span class="op">}</span>   <span class="kw">else</span> <span class="kw">if</span> <span class="op">(</span><span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Proteobacteria"</span> <span class="op">&amp;</span> <span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">3</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Gammaproteobacteria"</span><span class="op">)</span><span class="op">{</span>
           <span class="va">phylum</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"Gammaproteobacteria"</span><span class="op">)</span>
           <span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">phylum</span>
   <span class="op">}</span>   <span class="kw">else</span> <span class="kw">if</span> <span class="op">(</span><span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Proteobacteria"</span> <span class="op">&amp;</span> <span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">3</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Zetaproteobacteria"</span><span class="op">)</span><span class="op">{</span>
              <span class="va">phylum</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"Zetaproteobacteria"</span><span class="op">)</span>
           <span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">phylum</span>
   <span class="op">}</span>   <span class="kw">else</span> <span class="kw">if</span> <span class="op">(</span><span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Proteobacteria"</span> <span class="op">&amp;</span> <span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>,<span class="fl">3</span><span class="op">]</span> <span class="op">==</span> <span class="st">"p_Proteobacteria"</span><span class="op">)</span><span class="op">{</span>
           <span class="va">phylum</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"p_Proteobacteria"</span><span class="op">)</span>
           <span class="va">tmp_clean</span><span class="op">[</span><span class="va">i</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">phylum</span>
       <span class="op">}</span>
     <span class="op">}</span>
  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">tmp_clean</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/rank_names.html">rank_names</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_get</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/get_taxa_unique.html">get_taxa_unique</a></span><span class="op">(</span><span class="va">tmp_get</span>,
                               taxonomic.rank <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/rank_names.html">rank_names</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,
                               errorIfNULL<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="st">"files/taxa/rdata/"</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">tmp_get</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="va">tmp_path</span>, <span class="va">j</span>, <span class="st">"_clean.rds"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">class</span>, <span class="va">order</span>, <span class="va">phylum</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">set_to_plot</span> <span class="op">&lt;-</span> <span class="st">"ssu18_ps_work"</span>
<span class="va">tax_group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"Alphaproteobacteria"</span>, <span class="st">"Gammaproteobacteria"</span>, 
               <span class="st">"Acidobacteriota"</span>, <span class="st">"Actinobacteriota"</span>, 
               <span class="st">"Bacteroidota"</span>, <span class="st">"Firmicutes"</span>, 
               <span class="st">"Myxococcota"</span>, <span class="st">"Verrucomicrobiota"</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">set_to_plot</span><span class="op">)</span> <span class="op">{</span>
  <span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="va">tax_group</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_proteo_clean"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_taxa-methods.html">subset_taxa</a></span><span class="op">(</span><span class="va">tmp_get</span>, <span class="va">Phylum</span> <span class="op">==</span> <span class="va">j</span><span class="op">)</span>
  <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_"</span>, <span class="va">j</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_sub</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
  <span class="op">}</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">

</div>
<ol start="3" type="1">
<li>Choose the <strong>number</strong> of taxa to display and the taxonomic <strong>level</strong>. Aggregate the rest into “Other”.</li>
</ol>
<div class="cell">
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">top_hits</span> <span class="op">&lt;-</span> <span class="fl">12</span>
<span class="va">top_level</span> <span class="op">&lt;-</span> <span class="st">"Family"</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">tax_group</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="va">i</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_otu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_otu</span><span class="op">[</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">lapply</a></span><span class="op">(</span><span class="va">tmp_otu</span>, <span class="va">as.numeric</span><span class="op">)</span>
  <span class="va">tmp_otu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">tmp_otu</span><span class="op">)</span>
  <span class="va">tmp_tax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_clean_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/merge_phyloseq.html">merge_phyloseq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">tmp_otu</span>, taxa_are_rows <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,
                          <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">tmp_tax</span><span class="op">)</span>,
                          <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">tmp_samples</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_agg_df</span> <span class="op">&lt;-</span> <span class="fu">microbiome</span><span class="fu">::</span><span class="fu">aggregate_top_taxa</span><span class="op">(</span><span class="va">tmp_clean_df</span>,
                                                    top <span class="op">=</span> <span class="va">top_hits</span>,
                                                    level <span class="op">=</span> <span class="va">top_level</span><span class="op">)</span>
  <span class="va">tmp_agg_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="va">i</span>, <span class="st">"_agg"</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_agg_name</span>, <span class="va">tmp_agg_df</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"_sep_agg"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">tax_group</span><span class="op">)</span><span class="op">{</span>
  <span class="va">tmp_data</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="va">i</span>, <span class="st">"_agg"</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="va">tmp_data</span><span class="op">)</span>
  <span class="va">tmp_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/get_taxa_unique.html">get_taxa_unique</a></span><span class="op">(</span><span class="va">tmp_get</span>, taxonomic.rank <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/rank_names.html">rank_names</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,
                        errorIfNULL <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
  <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">tmp_data</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_order"</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_list</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">tax_group</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="va">i</span>, <span class="st">"_agg_order"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\n"</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"#########"</span>, <span class="va">i</span>, <span class="st">"########"</span>, <span class="st">"\n"</span><span class="op">)</span>
  <span class="va">tmp_print</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="va">tmp_print</span>, <span class="st">"\n"</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"####################################################"</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\n"</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">i</span>, <span class="va">j</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">tmp_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"Other"</span>, <span class="st">"c_Alphaproteobacteria"</span>, <span class="st">"o_Elsterales"</span>, <span class="st">"o_Azospirillales"</span>, <span class="st">"Rhizobiales_Incertae_Sedis"</span>, <span class="st">"Xanthobacteraceae"</span>, <span class="st">"Sphingomonadaceae"</span>, <span class="st">"Rhizobiaceae"</span>, <span class="st">"Micropepsaceae"</span>, <span class="st">"Methyloligellaceae"</span>, <span class="st">"KF-JG30-B3"</span>, <span class="st">"Hyphomicrobiaceae"</span>, <span class="st">"Dongiaceae"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="st">"Alphaproteobacteria"</span>, <span class="st">"_agg_order"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>, <span class="va">tmp_order</span><span class="op">)</span> 
<span class="co">###################</span>
<span class="va">tmp_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"Other"</span>, <span class="st">"c_Gammaproteobacteria"</span>, <span class="st">"o_PLTA13"</span>, <span class="st">"o_CCD24"</span>, <span class="st">"Unknown_Family"</span>, <span class="st">"Xanthomonadaceae"</span>, <span class="st">"TRA3-20"</span>, <span class="st">"Steroidobacteraceae"</span>, <span class="st">"SC-I-84"</span>, <span class="st">"Nitrosomonadaceae"</span>, <span class="st">"Comamonadaceae"</span>, <span class="st">"Burkholderiaceae"</span>, <span class="st">"B1-7BS"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="st">"Gammaproteobacteria"</span>, <span class="st">"_agg_order"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>, <span class="va">tmp_order</span><span class="op">)</span> 
<span class="co">###################</span>
<span class="va">tmp_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"Other"</span>, <span class="st">"c_Subgroup_22"</span>, <span class="st">"c_Subgroup_25"</span>, <span class="st">"c_Subgroup_5"</span>, <span class="st">"o_Vicinamibacterales"</span>, <span class="st">"o_11-24"</span>, <span class="st">"o_Subgroup_17"</span>, <span class="st">"o_Subgroup_2"</span>, <span class="st">"o_Subgroup_7"</span>, <span class="st">"o_Acidobacteriales"</span>, <span class="st">"Vicinamibacteraceae"</span>, <span class="st">"Solibacteraceae"</span>, <span class="st">"Pyrinomonadaceae"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="st">"Acidobacteriota"</span>, <span class="st">"_agg_order"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>, <span class="va">tmp_order</span><span class="op">)</span> 
<span class="co">###################</span>
<span class="va">tmp_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"Other"</span>, <span class="st">"c_MB-A2-108"</span>, <span class="st">"o_IMCC26256"</span>, <span class="st">"o_Gaiellales"</span>, <span class="st">"o_Frankiales"</span>, <span class="st">"Streptomycetaceae"</span>, <span class="st">"Solirubrobacteraceae"</span>, <span class="st">"Nocardioidaceae"</span>, <span class="st">"Mycobacteriaceae"</span>, <span class="st">"Micromonosporaceae"</span>, <span class="st">"Gaiellaceae"</span>, <span class="st">"Acidothermaceae"</span>, <span class="st">"67-14"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="st">"Actinobacteriota"</span>, <span class="st">"_agg_order"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>, <span class="va">tmp_order</span><span class="op">)</span> 
<span class="co">###################</span>
<span class="va">tmp_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"Other"</span>, <span class="st">"c_SJA-28"</span>, <span class="st">"o_Chitinophagales"</span>, <span class="st">"Sphingobacteriaceae"</span>, <span class="st">"Saprospiraceae"</span>, <span class="st">"Microscillaceae"</span>, <span class="st">"Hymenobacteraceae"</span>, <span class="st">"Flavobacteriaceae"</span>, <span class="st">"Cytophagaceae"</span>, <span class="st">"Chitinophagaceae"</span>, <span class="st">"BSV26"</span>, <span class="st">"env.OPS_17"</span>, <span class="st">"AKYH767"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="st">"Bacteroidota"</span>, <span class="st">"_agg_order"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>, <span class="va">tmp_order</span><span class="op">)</span> 
<span class="co">###################</span>
<span class="va">tmp_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"Other"</span>, <span class="st">"p_Firmicutes"</span>, <span class="st">"c_Bacilli"</span>, <span class="st">"o_Bacillales"</span>, <span class="st">"Thermoactinomycetaceae"</span>, <span class="st">"type_III"</span>, <span class="st">"Paenibacillaceae"</span>, <span class="st">"Lachnospiraceae"</span>, <span class="st">"Hungateiclostridiaceae"</span>, <span class="st">"Clostridiaceae"</span>, <span class="st">"Bacillaceae"</span>, <span class="st">"Alicyclobacillaceae"</span>, <span class="st">"Planococcaceae"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="st">"Firmicutes"</span>, <span class="st">"_agg_order"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>, <span class="va">tmp_order</span><span class="op">)</span> 
<span class="co">###################</span>
<span class="va">tmp_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"Other"</span>, <span class="st">"p_Myxococcota"</span>, <span class="st">"c_bacteriap25"</span>, <span class="st">"c_Polyangia"</span>, <span class="st">"o_mle1-27"</span>, <span class="st">"o_MSB-4B10"</span>, <span class="st">"Sandaracinaceae"</span>, <span class="st">"Polyangiaceae"</span>, <span class="st">"Phaselicystidaceae"</span>, <span class="st">"Myxococcaceae"</span>, <span class="st">"Haliangiaceae"</span>, <span class="st">"BIrii41"</span>, <span class="st">"Anaeromyxobacteraceae"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="st">"Myxococcota"</span>, <span class="st">"_agg_order"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>, <span class="va">tmp_order</span><span class="op">)</span> 
<span class="co">###################</span>
<span class="va">tmp_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"Other"</span>, <span class="st">"c_Verrucomicrobiae"</span>, <span class="st">"o_Chlamydiales"</span>, <span class="st">"o_LD1-PA32"</span>, <span class="st">"o_S-BQ2-57_soil_group"</span>, <span class="st">"Xiphinematobacteraceae"</span>, <span class="st">"Simkaniaceae"</span>, <span class="st">"Pedosphaeraceae"</span>, <span class="st">"Parachlamydiaceae"</span>, <span class="st">"Opitutaceae"</span>, <span class="st">"Omnitrophaceae"</span>, <span class="st">"cvE6"</span>, <span class="st">"Chthoniobacteraceae"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html">paste</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="st">"Verrucomicrobiota"</span>, <span class="st">"_agg_order"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>, <span class="va">tmp_order</span><span class="op">)</span> 
<span class="co">###################</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="4" type="1">
<li>Now, transform the data to relative abundance.</li>
</ol>
<div class="cell">
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">tax_group</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_agg</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="va">i</span>, <span class="st">"_agg"</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_order</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">tmp_agg</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_order"</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_get_agg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="va">tmp_agg</span><span class="op">)</span>
  <span class="va">tmp_get_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="va">tmp_order</span><span class="op">)</span>
  <span class="va">tmp_df</span> <span class="op">&lt;-</span> <span class="va">tmp_get_agg</span> <span class="op">%&gt;%</span>
            <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transformcounts.html">transform_sample_counts</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span><span class="va">x</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">}</span> <span class="op">)</span> <span class="op">%&gt;%</span>
            <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/psmelt.html">psmelt</a></span><span class="op">(</span><span class="op">)</span>
  <span class="va">tmp_df</span><span class="op">[[</span><span class="va">top_level</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">gdata</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gdata/man/reorder.html">reorder.factor</a></span><span class="op">(</span><span class="va">tmp_df</span><span class="op">[[</span><span class="va">top_level</span><span class="op">]</span><span class="op">]</span>,
                                                  new.order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">rev</a></span><span class="op">(</span><span class="va">tmp_get_order</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_df</span> <span class="op">&lt;-</span> <span class="va">tmp_df</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="va">top_level</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">tmp_agg</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"_tax"</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_df</span><span class="op">)</span>
  <span class="co">#print(c(i, tmp_name, tmp_agg))</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">tax_group</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="va">i</span>, <span class="st">"_agg_tax"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_levels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Factor-class.html">levels</a></span><span class="op">(</span><span class="va">tmp_get</span><span class="op">[[</span><span class="va">top_level</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="va">i</span>, <span class="va">tmp_levels</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="5" type="1">
<li>Plot the data for a single phyloseq object. Here you use an aggregated tax file.</li>
</ol>
<div class="cell">

</div>
<div class="cell">
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">tax_group</span><span class="op">)</span> <span class="op">{</span>
      <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="va">i</span>, <span class="st">"_agg_tax"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
      <span class="va">tmp_plot</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">tmp_get</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">TEMP</span><span class="op">)</span>, 
                                        y <span class="op">=</span> <span class="va">Abundance</span>, fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="va">top_level</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu">geom_bar</span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"fill"</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu">scale_fill_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">ssu18_colvec.tax</span><span class="op">)</span> <span class="op">+</span>
          <span class="co">#scale_x_discrete("Temperature", expand = waiver(), position = "bottom", drop = FALSE) +</span>
          <span class="fu">theme_cowplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu">guides</span><span class="op">(</span>fill <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>title <span class="op">=</span> <span class="va">top_level</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
          <span class="co">#guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +</span>
          <span class="fu">ylab</span><span class="op">(</span><span class="st">"Relative Abundance (% total reads)"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">xlab</span><span class="op">(</span><span class="st">"Temperature"</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu">theme</span><span class="op">(</span>panel.grid.major <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>, panel.grid.minor <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
                panel.background <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"transparent"</span>, colour <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>,
                plot.background <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"transparent"</span>, colour <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>,
                panel.border <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>fill <span class="op">=</span> <span class="cn">NA</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
                <span class="co">#axis.text = element_text(size = 6),</span>
                <span class="co">#axis.title = element_text(size = 8),</span>
                legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span>
      <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="va">.</span>, <span class="st">"_plot"</span><span class="op">)</span><span class="op">)</span>
      <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_plot</span><span class="op">)</span>
      <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="6" type="1">
<li>Plot the data for multiple taxa. Here again you use an aggregated tax file. This code can be used to generate plots for multiple data frames by adding the base phyloseq names to the <code>ssu18_split_df</code> variable. This code will also facet the plots by a metadata variable. If you do not want to facet remove the line beginning with <code>facet_grid</code>.</li>
</ol>
<div class="cell">
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">tax_group</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_level_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="va">.</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_level</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">tmp_level_get</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_level</span> <span class="op">&lt;-</span> <span class="va">tmp_level</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html">order</a></span><span class="op">(</span><span class="va">tmp_level</span><span class="op">$</span><span class="va">TEMP</span><span class="op">)</span>, <span class="op">]</span>
  <span class="va">tmp_level</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.vector.html">as.vector</a></span><span class="op">(</span><span class="va">tmp_level</span><span class="op">$</span><span class="va">SamName</span><span class="op">)</span>

  <span class="va">tmp_agg_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="va">.</span>,  <span class="st">"_agg_tax"</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="va">tmp_agg_name</span><span class="op">)</span>
  <span class="va">tmp_df</span> <span class="op">&lt;-</span> <span class="fu">reshape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape/man/melt-24.html">melt</a></span><span class="op">(</span><span class="va">tmp_get</span>, id.vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"Sample"</span>, <span class="st">"TEMP"</span>, <span class="st">"Abundance"</span>, <span class="st">"Family"</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_plot_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="va">.</span>, <span class="st">"_plot_melt"</span><span class="op">)</span><span class="op">)</span>

  <span class="va">tmp_plot</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">tmp_df</span>,
                <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Sample</span>,
                    y <span class="op">=</span> <span class="va">Abundance</span>, fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="va">top_level</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">facet_grid</span><span class="op">(</span><span class="va">.</span> <span class="op">~</span><span class="va">TEMP</span>, scale <span class="op">=</span> <span class="st">"free_x"</span>, space <span class="op">=</span> <span class="st">"free_x"</span><span class="op">)</span><span class="op">+</span>
    <span class="fu">geom_bar</span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"fill"</span><span class="op">)</span>  <span class="op">+</span>
    <span class="fu">scale_fill_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">ssu18_colvec.tax</span><span class="op">)</span> <span class="op">+</span>
    <span class="co">#scale_x_discrete("Treatment", expand = waiver(),</span>
    <span class="co">#                 position = "bottom", drop = FALSE, limits = tmp_level) +</span>
    <span class="fu">theme_cowplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">guides</span><span class="op">(</span>fill <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>title <span class="op">=</span> <span class="va">top_level</span>, reverse <span class="op">=</span> <span class="cn">FALSE</span>,
                               keywidth <span class="op">=</span> <span class="fl">0.7</span>, keyheight <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">ylab</span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">theme</span><span class="op">(</span>panel.grid.major <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>, panel.grid.minor <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        panel.background <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"transparent"</span>, colour <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>,
        plot.background <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"transparent"</span>, colour <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>,
        panel.border <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>fill <span class="op">=</span> <span class="cn">NA</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
        legend.position <span class="op">=</span> <span class="st">"right"</span>,
        <span class="co">#legend.text = element_text(size = 6),</span>
        <span class="co">#legend.title = element_text(size = 8),</span>
        <span class="co">#legend.key.size = unit(1.5, "cm"),</span>
        <span class="co">#axis.text.y = element_text(size = 6),</span>
        axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span><span class="op">)</span>
        <span class="co">#strip.text = element_text(size = 8, angle = 0),</span>
        <span class="co">#axis.title = element_text(size = 8)</span>
        <span class="op">)</span> <span class="op">+</span> 
    <span class="fu">ylab</span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span>
  
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_plot_name</span>, <span class="va">tmp_plot</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="7" type="1">
<li>Finally we use the <code>patchwork</code> package to combine the two plots and customize the look.</li>
</ol>
<div class="cell">
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">tax_group</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_plot_main</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="va">.</span>, <span class="st">"_plot"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_plot_melt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="va">.</span>, <span class="st">"_plot_melt"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tmp_final</span> <span class="op">&lt;-</span> <span class="va">tmp_plot_main</span> <span class="op">+</span> <span class="va">tmp_plot_melt</span>
  <span class="va">tmp_final</span> <span class="op">&lt;-</span> <span class="va">tmp_final</span> <span class="op">+</span> 
    <span class="fu">plot_annotation</span><span class="op">(</span>tag_levels <span class="op">=</span> <span class="st">'a'</span>, 
                    title <span class="op">=</span> <span class="va">i</span><span class="op">)</span> <span class="op">+</span> 
                    <span class="co">#subtitle = 'Top taxa of non-filtered data',</span>
                    <span class="co">#caption = 'A) grouped by temperature.,</span>
                    <span class="co">#B) All samples, faceted by temperature.') + </span>
    <span class="fu">plot_layout</span><span class="op">(</span>widths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">&amp;</span>
    <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">9</span><span class="op">)</span>,
        plot.subtitle <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,
        plot.tag <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span>,
        axis.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">7</span><span class="op">)</span>,
        axis.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span>,
        strip.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span>, angle <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">7</span><span class="op">)</span>,
        legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">9</span><span class="op">)</span>,
        <span class="op">)</span>
        <span class="co">#legend.position = "right")</span>
        <span class="co">#legend.position = "right",</span>
        <span class="co">#legend.title = element_text(size = rel(1)),</span>
        <span class="co">#legend.text = element_text(size = rel(1)))</span>
  <span class="va">tmp_name</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">set_to_plot</span>, <span class="st">"_"</span>, <span class="va">.</span>, <span class="st">"_final_plot"</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="va">tmp_name</span>, <span class="va">tmp_final</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"tmp_"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">taxa_to_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"Acidobacteriota"</span>, <span class="st">"Actinobacteriota"</span>, <span class="st">"Alphaproteobacteria"</span>, 
                  <span class="st">"Bacteroidota"</span>, <span class="st">"Firmicutes"</span>, <span class="st">"Gammaproteobacteria"</span>, <span class="st">"Myxococcota"</span>, 
                  <span class="st">"Verrucomicrobiota"</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">taxa_to_plot</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tmp_get</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html">get</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"ssu18_ps_work_"</span>, <span class="va">i</span>, <span class="st">"_final_plot"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="va">tmp_get</span>, 
                file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"files/taxa/figures/"</span>, <span class="va">i</span>, <span class="st">"_tax_div_bar_plots.png"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>, 
                height <span class="op">=</span> <span class="fl">14</span>, width <span class="op">=</span> <span class="fl">21</span>, units <span class="op">=</span> <span class="st">'cm'</span>, bg <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="va">tmp_get</span>, 
                file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"files/taxa/figures/"</span>, <span class="va">i</span>, <span class="st">"_tax_div_bar_plots.pdf"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>, 
                height <span class="op">=</span> <span class="fl">14</span>, width <span class="op">=</span> <span class="fl">21</span>, units <span class="op">=</span> <span class="st">'cm'</span>, bg <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span>
<span class="op">}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</details><p><br></p>
<div class="column-body-outset">
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-3-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-1" role="tab" aria-controls="tabset-3-1" aria-selected="true">Alphaproteobacteria</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-3-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-2" role="tab" aria-controls="tabset-3-2" aria-selected="false">Gammaproteobacteria</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-3-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-3" role="tab" aria-controls="tabset-3-3" aria-selected="false">Acidobacteriota</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-3-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-4" role="tab" aria-controls="tabset-3-4" aria-selected="false">Actinobacteriota</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-3-5-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-5" role="tab" aria-controls="tabset-3-5" aria-selected="false">Bacteroidota</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-3-6-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-6" role="tab" aria-controls="tabset-3-6" aria-selected="false">Firmicutes</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-3-7-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-7" role="tab" aria-controls="tabset-3-7" aria-selected="false">Myxococcota</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-3-8-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-8" role="tab" aria-controls="tabset-3-8" aria-selected="false">Verrucomicrobiota</a></li>
</ul>
<div class="tab-content">
<div id="tabset-3-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-3-1-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="include/taxa/Alphaproteobacteria_tax_div_bar_plots.png" class="img-fluid" width="1240"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 9 | Alphaproteobacteria family plots.</strong></small></p>
</div>
<div id="tabset-3-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-3-2-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="include/taxa/Gammaproteobacteria_tax_div_bar_plots.png" class="img-fluid" width="1240"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 10 | Gammaproteobacteria family plots.</strong></small></p>
</div>
<div id="tabset-3-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-3-3-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="include/taxa/Acidobacteriota_tax_div_bar_plots.png" class="img-fluid" width="1240"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 11 | Acidobacteriota family plots.</strong></small></p>
</div>
<div id="tabset-3-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-3-4-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="include/taxa/Actinobacteriota_tax_div_bar_plots.png" class="img-fluid" width="1240"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 12 | Actinobacteriota family plots.</strong></small></p>
</div>
<div id="tabset-3-5" class="tab-pane" role="tabpanel" aria-labelledby="tabset-3-5-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="include/taxa/Bacteroidota_tax_div_bar_plots.png" class="img-fluid" width="1240"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 13 | Bacteroidota family plots.</strong></small></p>
</div>
<div id="tabset-3-6" class="tab-pane" role="tabpanel" aria-labelledby="tabset-3-6-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="include/taxa/Firmicutes_tax_div_bar_plots.png" class="img-fluid" width="1240"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 14 | Firmicutes family plots.</strong></small></p>
</div>
<div id="tabset-3-7" class="tab-pane" role="tabpanel" aria-labelledby="tabset-3-7-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="include/taxa/Myxococcota_tax_div_bar_plots.png" class="img-fluid" width="1240"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 15 | Myxococcota family plots.</strong></small></p>
</div>
<div id="tabset-3-8" class="tab-pane" role="tabpanel" aria-labelledby="tabset-3-8-tab">
<div class="cell">
<div class="cell-output-display">
<p><img src="include/taxa/Verrucomicrobiota_tax_div_bar_plots.png" class="img-fluid" width="1240"></p>
</div>
</div>
<p><small><strong>(16S rRNA) Figure 16 | Verrucomicrobiota family plots.</strong></small></p>
</div>
</div>
</div>
</div>
<div class="cell">

</div>
</section><section id="workflow-output" class="level1"><h1>Workflow Output</h1>
<p>Data products generated in this workflow can be downloaded from figshare.</p>
<iframe src="https://widgets.figshare.com/articles/16826737/embed?show_title=1" width="100%" height="221" allowfullscreen="" frameborder="0">
</iframe>
<p></p>
<p><a href="alpha.html" class="btnnav button_round" style="float: right;">Next workflow: <br> 5. Alpha Diversity Estimates</a> <a href="filtering.html" class="btnnav button_round" style="float: left;">Previous workflow: <br> 3. Filtering</a><br><br></p>
<div class="cell">

</div>
</section><div id="quarto-appendix" class="default"><section id="source-code" class="level4 appendix"><h2 class="quarto-appendix-heading">Source Code</h2><div class="quarto-appendix-contents">
<p>The source code for this page can be accessed on GitHub <svg aria-hidden="true" role="img" viewbox="0 0 496 512" style="height:1em;width:0.97em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:currentColor;overflow:visible;position:relative;"><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"></path></svg> by <a href="https://github.com/sweltr/high-temp/blob/main/taxa.qmd">clicking this link</a>.</p>
</div></section><section id="data-availability" class="level4 appendix"><h2 class="quarto-appendix-heading">Data Availability</h2><div class="quarto-appendix-contents">
<p>Data generated in this workflow and the Rdata need to run the workflow can be accessed on figshare at <a href="https://doi.org/10.25573/data.16826737">10.25573/data.16826737</a>.</p>
</div></section><section id="last-updated-on" class="level4 appendix"><h2 class="quarto-appendix-heading">Last updated on</h2><div class="quarto-appendix-contents">
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "2022-06-29 07:21:21 EST"</code></pre>
</div>
</div>


<!-- -->

</div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./filtering.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">3. Filtering</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./alpha.html" class="pagination-link">
        <span class="nav-page-text">5. Alpha Diversity Estimates</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb47" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "4. Taxonomic Diversity"</span></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a><span class="an">description:</span><span class="co"> |</span></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a><span class="co">  Reproducible workflow for assessing dominant taxonomic groups in the 16S rRNA and ITS data sets.</span></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true</span></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-depth: 3</span></span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;&lt;strong&gt;</span>Click here<span class="kw">&lt;/strong&gt;</span> for setup information.<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup, message=FALSE, results = 'hide'}</span></span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">echo =</span> <span class="cn">TRUE</span>, <span class="at">eval =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">119</span>)</span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a><span class="co">#library(conflicted)</span></span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends(phangorn, local = TRUE)  </span></span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends_reverse(phangorn, local = TRUE)  </span></span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-21"><a href="#cb47-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq); <span class="fu">packageVersion</span>(<span class="st">"phyloseq"</span>)</span>
<span id="cb47-22"><a href="#cb47-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings); <span class="fu">packageVersion</span>(<span class="st">"Biostrings"</span>)</span>
<span id="cb47-23"><a href="#cb47-23" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(tidyverse, patchwork,</span>
<span id="cb47-24"><a href="#cb47-24" aria-hidden="true" tabindex="-1"></a>               agricolae, labdsv, naniar, </span>
<span id="cb47-25"><a href="#cb47-25" aria-hidden="true" tabindex="-1"></a>               microbiome, microeco, file2meco, cowplot,</span>
<span id="cb47-26"><a href="#cb47-26" aria-hidden="true" tabindex="-1"></a>               GUniFrac, ggalluvial, ggdendro, igraph, reactable,</span>
<span id="cb47-27"><a href="#cb47-27" aria-hidden="true" tabindex="-1"></a>               pheatmap, microbiomeMarker,</span>
<span id="cb47-28"><a href="#cb47-28" aria-hidden="true" tabindex="-1"></a>               captioner, downloadthis,</span>
<span id="cb47-29"><a href="#cb47-29" aria-hidden="true" tabindex="-1"></a>               <span class="at">install =</span> <span class="cn">FALSE</span>, <span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-30"><a href="#cb47-30" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">scipen=</span><span class="dv">999</span>)</span>
<span id="cb47-31"><a href="#cb47-31" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_current<span class="sc">$</span><span class="fu">get</span>(<span class="fu">c</span>(</span>
<span id="cb47-32"><a href="#cb47-32" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cache"</span>,</span>
<span id="cb47-33"><a href="#cb47-33" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cache.path"</span>,</span>
<span id="cb47-34"><a href="#cb47-34" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cache.rebuild"</span>,</span>
<span id="cb47-35"><a href="#cb47-35" aria-hidden="true" tabindex="-1"></a>  <span class="st">"dependson"</span>,</span>
<span id="cb47-36"><a href="#cb47-36" aria-hidden="true" tabindex="-1"></a>  <span class="st">"autodep"</span></span>
<span id="cb47-37"><a href="#cb47-37" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb47-38"><a href="#cb47-38" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-39"><a href="#cb47-39" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb47-40"><a href="#cb47-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-43"><a href="#cb47-43" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-44"><a href="#cb47-44" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-45"><a href="#cb47-45" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-46"><a href="#cb47-46" aria-hidden="true" tabindex="-1"></a><span class="co">#TEMP LOAD ONLY REMOVE WHEN WORKFLOW FINISHED</span></span>
<span id="cb47-47"><a href="#cb47-47" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb47-48"><a href="#cb47-48" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"page_build/taxa_wf.rdata"</span>)</span>
<span id="cb47-49"><a href="#cb47-49" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-50"><a href="#cb47-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-53"><a href="#cb47-53" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-54"><a href="#cb47-54" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-55"><a href="#cb47-55" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-56"><a href="#cb47-56" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb47-57"><a href="#cb47-57" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the caption(s) with captioner</span></span>
<span id="cb47-58"><a href="#cb47-58" aria-hidden="true" tabindex="-1"></a>caption_tab <span class="ot">&lt;-</span> <span class="fu">captioner</span>(<span class="at">prefix =</span> <span class="st">"(16S rRNA) Table"</span>, <span class="at">suffix =</span> <span class="st">" |"</span>, <span class="at">style =</span> <span class="st">"b"</span>)</span>
<span id="cb47-59"><a href="#cb47-59" aria-hidden="true" tabindex="-1"></a>caption_fig <span class="ot">&lt;-</span> <span class="fu">captioner</span>(<span class="at">prefix =</span> <span class="st">"(16S rRNA) Figure"</span>, <span class="at">suffix =</span> <span class="st">" |"</span>, <span class="at">style =</span> <span class="st">"b"</span>)</span>
<span id="cb47-60"><a href="#cb47-60" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a function for referring to the tables in text</span></span>
<span id="cb47-61"><a href="#cb47-61" aria-hidden="true" tabindex="-1"></a>ref <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">str_extract</span>(x, <span class="st">"[^|]*"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb47-62"><a href="#cb47-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">trimws</span>(<span class="at">which =</span> <span class="st">"right"</span>, <span class="at">whitespace =</span> <span class="st">"[ ]"</span>)</span>
<span id="cb47-63"><a href="#cb47-63" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-64"><a href="#cb47-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-67"><a href="#cb47-67" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-68"><a href="#cb47-68" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-69"><a href="#cb47-69" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-70"><a href="#cb47-70" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"assets/captions/captions_taxa.R"</span>)</span>
<span id="cb47-71"><a href="#cb47-71" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-72"><a href="#cb47-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-73"><a href="#cb47-73" aria-hidden="true" tabindex="-1"></a><span class="fu"># Synopsis</span></span>
<span id="cb47-74"><a href="#cb47-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-75"><a href="#cb47-75" aria-hidden="true" tabindex="-1"></a>This workflow contains taxonomic diversity assessments for the 16S rRNA and ITS data sets. In order to run the workflow, you either need to first run the  <span class="co">[</span><span class="ot">DADA2 Workflow</span><span class="co">](dada2.html)</span>, then the <span class="co">[</span><span class="ot">Data Preparation workflow</span><span class="co">](data-prep.html)</span>, and finally the <span class="co">[</span><span class="ot">Filtering workflow</span><span class="co">](filtering.html)</span> or download the files linked below. </span>
<span id="cb47-76"><a href="#cb47-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-77"><a href="#cb47-77" aria-hidden="true" tabindex="-1"></a><span class="fu">## Workflow Input</span></span>
<span id="cb47-78"><a href="#cb47-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-79"><a href="#cb47-79" aria-hidden="true" tabindex="-1"></a>All files needed to run this workflow can be downloaded from figshare. </span>
<span id="cb47-80"><a href="#cb47-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-81"><a href="#cb47-81" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;iframe</span> <span class="er">src</span><span class="ot">=</span><span class="st">"https://widgets.figshare.com/articles/14701440/embed?show_title=1"</span> <span class="er">width</span><span class="ot">=</span><span class="st">"100%"</span> <span class="er">height</span><span class="ot">=</span><span class="st">"351"</span> <span class="er">allowfullscreen</span> <span class="er">frameborder</span><span class="ot">=</span><span class="st">"0"</span><span class="kw">&gt;&lt;/iframe&gt;</span></span>
<span id="cb47-82"><a href="#cb47-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-83"><a href="#cb47-83" aria-hidden="true" tabindex="-1"></a>In this workflow, we use the <span class="co">[</span><span class="ot">microeco</span><span class="co">](https://github.com/ChiLiubio/microeco)</span> to look at the taxonomic distribution of microbial communities.</span>
<span id="cb47-84"><a href="#cb47-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-85"><a href="#cb47-85" aria-hidden="true" tabindex="-1"></a><span class="fu"># Taxonomic Assessment</span></span>
<span id="cb47-86"><a href="#cb47-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-87"><a href="#cb47-87" aria-hidden="true" tabindex="-1"></a><span class="fu">## 16s rRNA</span></span>
<span id="cb47-88"><a href="#cb47-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-91"><a href="#cb47-91" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-92"><a href="#cb47-92" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-93"><a href="#cb47-93" aria-hidden="true" tabindex="-1"></a><span class="do">## Initial Load for  ANALYSIS #1</span></span>
<span id="cb47-94"><a href="#cb47-94" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb47-95"><a href="#cb47-95" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">119</span>)</span>
<span id="cb47-96"><a href="#cb47-96" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"files/data-prep/rdata/ssu18_ps_work.rds"</span>)</span>
<span id="cb47-97"><a href="#cb47-97" aria-hidden="true" tabindex="-1"></a>ssu18_ps_filt <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"files/filtering/arbitrary/rdata/ssu18_ps_filt.rds"</span>)</span>
<span id="cb47-98"><a href="#cb47-98" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"files/filtering/perfect/rdata/ssu18_ps_perfect.rds"</span>)</span>
<span id="cb47-99"><a href="#cb47-99" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"files/filtering/pime/rdata/ssu18_ps_pime.rds"</span>)</span>
<span id="cb47-100"><a href="#cb47-100" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span>
<span id="cb47-101"><a href="#cb47-101" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-102"><a href="#cb47-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-105"><a href="#cb47-105" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-106"><a href="#cb47-106" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-107"><a href="#cb47-107" aria-hidden="true" tabindex="-1"></a>swel_col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#2271B2"</span>, <span class="st">"#71B222"</span>, <span class="st">"#B22271"</span>)</span>
<span id="cb47-108"><a href="#cb47-108" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-109"><a href="#cb47-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-110"><a href="#cb47-110" aria-hidden="true" tabindex="-1"></a>Here we compare the taxonomic breakdown of the Full (unfiltered), Arbitrary filtered, PERfect filtered, and PIME filtered  data sets, split by temperature treatment.</span>
<span id="cb47-111"><a href="#cb47-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-112"><a href="#cb47-112" aria-hidden="true" tabindex="-1"></a>Before we plot the data, we want to separate out the Proteobacteria classes so we can plot these along with other phyla. To accomplish this we perform the following steps:</span>
<span id="cb47-113"><a href="#cb47-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-114"><a href="#cb47-114" aria-hidden="true" tabindex="-1"></a>1) Get all Class-level Proteobacteria names</span>
<span id="cb47-115"><a href="#cb47-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-118"><a href="#cb47-118" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-119"><a href="#cb47-119" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb47-120"><a href="#cb47-120" aria-hidden="true" tabindex="-1"></a>ssu18_data_sets <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ssu18_ps_work"</span>,  <span class="st">"ssu18_ps_filt"</span>,</span>
<span id="cb47-121"><a href="#cb47-121" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"ssu18_ps_perfect"</span>, <span class="st">"ssu18_ps_pime"</span>)</span>
<span id="cb47-122"><a href="#cb47-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-123"><a href="#cb47-123" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb47-124"><a href="#cb47-124" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_proteo"</span>))</span>
<span id="cb47-125"><a href="#cb47-125" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb47-126"><a href="#cb47-126" aria-hidden="true" tabindex="-1"></a>     tmp_df <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(tmp_get, Phylum <span class="sc">==</span> <span class="st">"Proteobacteria"</span>)</span>
<span id="cb47-127"><a href="#cb47-127" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_df)</span>
<span id="cb47-128"><a href="#cb47-128" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_name)</span>
<span id="cb47-129"><a href="#cb47-129" aria-hidden="true" tabindex="-1"></a>     tmp_get_taxa <span class="ot">&lt;-</span> <span class="fu">get_taxa_unique</span>(tmp_df,</span>
<span id="cb47-130"><a href="#cb47-130" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">taxonomic.rank =</span> <span class="fu">rank_names</span>(tmp_df)[<span class="dv">3</span>],</span>
<span id="cb47-131"><a href="#cb47-131" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">errorIfNULL=</span><span class="cn">TRUE</span>)</span>
<span id="cb47-132"><a href="#cb47-132" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_get_taxa)</span>
<span id="cb47-133"><a href="#cb47-133" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-134"><a href="#cb47-134" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"_proteo"</span>))</span>
<span id="cb47-135"><a href="#cb47-135" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-136"><a href="#cb47-136" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-137"><a href="#cb47-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-138"><a href="#cb47-138" aria-hidden="true" tabindex="-1"></a>2) Replace Phylum Proteobacteria with the Class name.</span>
<span id="cb47-139"><a href="#cb47-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-142"><a href="#cb47-142" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-143"><a href="#cb47-143" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb47-144"><a href="#cb47-144" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb47-145"><a href="#cb47-145" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(j, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_proteo_clean"</span>))</span>
<span id="cb47-146"><a href="#cb47-146" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(j)</span>
<span id="cb47-147"><a href="#cb47-147" aria-hidden="true" tabindex="-1"></a>  tmp_clean <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(tmp_get))</span>
<span id="cb47-148"><a href="#cb47-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-149"><a href="#cb47-149" aria-hidden="true" tabindex="-1"></a>   <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(tmp_clean)){</span>
<span id="cb47-150"><a href="#cb47-150" aria-hidden="true" tabindex="-1"></a>       <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">"Proteobacteria"</span> <span class="sc">&amp;</span> tmp_clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">"Alphaproteobacteria"</span>){</span>
<span id="cb47-151"><a href="#cb47-151" aria-hidden="true" tabindex="-1"></a>           phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">"Alphaproteobacteria"</span>)</span>
<span id="cb47-152"><a href="#cb47-152" aria-hidden="true" tabindex="-1"></a>           tmp_clean[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb47-153"><a href="#cb47-153" aria-hidden="true" tabindex="-1"></a>   }   <span class="cf">else</span> <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">"Proteobacteria"</span> <span class="sc">&amp;</span> tmp_clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">"Gammaproteobacteria"</span>){</span>
<span id="cb47-154"><a href="#cb47-154" aria-hidden="true" tabindex="-1"></a>           phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">"Gammaproteobacteria"</span>)</span>
<span id="cb47-155"><a href="#cb47-155" aria-hidden="true" tabindex="-1"></a>           tmp_clean[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb47-156"><a href="#cb47-156" aria-hidden="true" tabindex="-1"></a>   }   <span class="cf">else</span> <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">"Proteobacteria"</span> <span class="sc">&amp;</span> tmp_clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">"Zetaproteobacteria"</span>){</span>
<span id="cb47-157"><a href="#cb47-157" aria-hidden="true" tabindex="-1"></a>              phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">"Zetaproteobacteria"</span>)</span>
<span id="cb47-158"><a href="#cb47-158" aria-hidden="true" tabindex="-1"></a>           tmp_clean[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb47-159"><a href="#cb47-159" aria-hidden="true" tabindex="-1"></a>   }   <span class="cf">else</span> <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">"Proteobacteria"</span> <span class="sc">&amp;</span> tmp_clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">"p_Proteobacteria"</span>){</span>
<span id="cb47-160"><a href="#cb47-160" aria-hidden="true" tabindex="-1"></a>           phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">"p_Proteobacteria"</span>)</span>
<span id="cb47-161"><a href="#cb47-161" aria-hidden="true" tabindex="-1"></a>           tmp_clean[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb47-162"><a href="#cb47-162" aria-hidden="true" tabindex="-1"></a>       }</span>
<span id="cb47-163"><a href="#cb47-163" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb47-164"><a href="#cb47-164" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tax_table</span>(tmp_get) <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_clean)</span>
<span id="cb47-165"><a href="#cb47-165" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rank_names</span>(tmp_get)</span>
<span id="cb47-166"><a href="#cb47-166" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_get)</span>
<span id="cb47-167"><a href="#cb47-167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">c</span>(tmp_name, tmp_get))</span>
<span id="cb47-168"><a href="#cb47-168" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(<span class="fu">get_taxa_unique</span>(tmp_get,</span>
<span id="cb47-169"><a href="#cb47-169" aria-hidden="true" tabindex="-1"></a>                               <span class="at">taxonomic.rank =</span> <span class="fu">rank_names</span>(tmp_get)[<span class="dv">2</span>],</span>
<span id="cb47-170"><a href="#cb47-170" aria-hidden="true" tabindex="-1"></a>                               <span class="at">errorIfNULL=</span><span class="cn">TRUE</span>)))</span>
<span id="cb47-171"><a href="#cb47-171" aria-hidden="true" tabindex="-1"></a>  tmp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"files/taxa/rdata/"</span>)</span>
<span id="cb47-172"><a href="#cb47-172" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(tmp_get, <span class="fu">paste</span>(tmp_path, j, <span class="st">"_clean.rds"</span>, <span class="at">sep =</span> <span class="st">""</span>))</span>
<span id="cb47-173"><a href="#cb47-173" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-174"><a href="#cb47-174" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-175"><a href="#cb47-175" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(class, order, phylum)</span>
<span id="cb47-176"><a href="#cb47-176" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern=</span><span class="st">"_proteo_clean"</span>)</span>
<span id="cb47-177"><a href="#cb47-177" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-178"><a href="#cb47-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-179"><a href="#cb47-179" aria-hidden="true" tabindex="-1"></a>3) In order to use <span class="in">`microeco`</span>, we need to add the rank designation as a prefix to each taxa. For example, <span class="in">`Actinobacteriota`</span> is changed to <span class="in">`p__Actinobacteriota`</span>. </span>
<span id="cb47-180"><a href="#cb47-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-183"><a href="#cb47-183" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-184"><a href="#cb47-184" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb47-185"><a href="#cb47-185" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb47-186"><a href="#cb47-186" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_proteo_clean"</span>)))</span>
<span id="cb47-187"><a href="#cb47-187" aria-hidden="true" tabindex="-1"></a>  <span class="co">#tmp_get &lt;- get(i)</span></span>
<span id="cb47-188"><a href="#cb47-188" aria-hidden="true" tabindex="-1"></a>  <span class="co">#tmp_path &lt;- file.path("files/alpha/rdata/")</span></span>
<span id="cb47-189"><a href="#cb47-189" aria-hidden="true" tabindex="-1"></a>  <span class="co">#tmp_read &lt;- readRDS(paste(tmp_path, i, ".rds", sep = ""))</span></span>
<span id="cb47-190"><a href="#cb47-190" aria-hidden="true" tabindex="-1"></a>  tmp_sam_data <span class="ot">&lt;-</span> <span class="fu">sample_data</span>(tmp_get)</span>
<span id="cb47-191"><a href="#cb47-191" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(tmp_get))</span>
<span id="cb47-192"><a href="#cb47-192" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Phylum <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"p_Proteobacteria"</span>, <span class="st">"Proteobacteria"</span>, tmp_tax_data<span class="sc">$</span>Phylum)</span>
<span id="cb47-193"><a href="#cb47-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-194"><a href="#cb47-194" aria-hidden="true" tabindex="-1"></a>  <span class="co">#tmp_tax_data[,c(1:6)]</span></span>
<span id="cb47-195"><a href="#cb47-195" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>ASV_ID <span class="ot">&lt;-</span> <span class="cn">NULL</span> <span class="co"># Some have, some do not</span></span>
<span id="cb47-196"><a href="#cb47-196" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>ASV_SEQ <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb47-197"><a href="#cb47-197" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb47-198"><a href="#cb47-198" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data[] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">lapply</span>(tmp_tax_data, gsub, </span>
<span id="cb47-199"><a href="#cb47-199" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">pattern =</span> <span class="st">"^[k | p | c | o | f]_.*"</span>, </span>
<span id="cb47-200"><a href="#cb47-200" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">replacement =</span> <span class="st">""</span>, <span class="at">fixed =</span> <span class="cn">FALSE</span>))</span>
<span id="cb47-201"><a href="#cb47-201" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Kingdom <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"k__"</span>, tmp_tax_data<span class="sc">$</span>Kingdom, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb47-202"><a href="#cb47-202" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Phylum <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"p__"</span>, tmp_tax_data<span class="sc">$</span>Phylum, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb47-203"><a href="#cb47-203" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Class <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"c__"</span>, tmp_tax_data<span class="sc">$</span>Class, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb47-204"><a href="#cb47-204" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Order <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"o__"</span>, tmp_tax_data<span class="sc">$</span>Order, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb47-205"><a href="#cb47-205" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Family <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"f__"</span>, tmp_tax_data<span class="sc">$</span>Family, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb47-206"><a href="#cb47-206" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Genus <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"g__"</span>, tmp_tax_data<span class="sc">$</span>Genus, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb47-207"><a href="#cb47-207" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_tax_data)</span>
<span id="cb47-208"><a href="#cb47-208" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb47-209"><a href="#cb47-209" aria-hidden="true" tabindex="-1"></a>  tmp_ps <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(<span class="fu">otu_table</span>(tmp_get),</span>
<span id="cb47-210"><a href="#cb47-210" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">phy_tree</span>(tmp_get),</span>
<span id="cb47-211"><a href="#cb47-211" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">tax_table</span>(tmp_tax_data),</span>
<span id="cb47-212"><a href="#cb47-212" aria-hidden="true" tabindex="-1"></a>                     tmp_sam_data)</span>
<span id="cb47-213"><a href="#cb47-213" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(i, tmp_ps)</span>
<span id="cb47-214"><a href="#cb47-214" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-215"><a href="#cb47-215" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-216"><a href="#cb47-216" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"_proteo_clean"</span>))</span>
<span id="cb47-217"><a href="#cb47-217" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-218"><a href="#cb47-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-219"><a href="#cb47-219" aria-hidden="true" tabindex="-1"></a>4) Next, we need to covert each <span class="in">`phyloseq object`</span> to a <span class="in">`microtable class`</span>.  The microtable class is the basic data structure for the <span class="in">`microeco`</span> package and designed to store basic information from all the downstream analyses (e.g, alpha diversity, beta diversity, etc.). We use the <span class="co">[</span><span class="ot">file2meco</span><span class="co">](https://github.com/ChiLiubio/file2meco)</span> to read the phyloseq object and convert into a microtable object. We can add <span class="in">`_me`</span> as a suffix to each object to distiguish it from its' phyloseq counterpart. </span>
<span id="cb47-220"><a href="#cb47-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-223"><a href="#cb47-223" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-224"><a href="#cb47-224" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb47-225"><a href="#cb47-225" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb47-226"><a href="#cb47-226" aria-hidden="true" tabindex="-1"></a>    tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb47-227"><a href="#cb47-227" aria-hidden="true" tabindex="-1"></a>    tmp_otu_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(tmp_get)))</span>
<span id="cb47-228"><a href="#cb47-228" aria-hidden="true" tabindex="-1"></a>    tmp_sample_info <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb47-229"><a href="#cb47-229" aria-hidden="true" tabindex="-1"></a>    tmp_taxonomy_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(tmp_get))</span>
<span id="cb47-230"><a href="#cb47-230" aria-hidden="true" tabindex="-1"></a>    tmp_phylo_tree <span class="ot">&lt;-</span> <span class="fu">phy_tree</span>(tmp_get)</span>
<span id="cb47-231"><a href="#cb47-231" aria-hidden="true" tabindex="-1"></a>    tmp_taxonomy_table <span class="sc">%&lt;&gt;%</span> tidy_taxonomy</span>
<span id="cb47-232"><a href="#cb47-232" aria-hidden="true" tabindex="-1"></a>    tmp_dataset <span class="ot">&lt;-</span> microtable<span class="sc">$</span><span class="fu">new</span>(<span class="at">sample_table =</span> tmp_sample_info, </span>
<span id="cb47-233"><a href="#cb47-233" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">otu_table =</span> tmp_otu_table, </span>
<span id="cb47-234"><a href="#cb47-234" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">tax_table =</span> tmp_taxonomy_table, </span>
<span id="cb47-235"><a href="#cb47-235" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">phylo_tree =</span> tmp_phylo_tree)</span>
<span id="cb47-236"><a href="#cb47-236" aria-hidden="true" tabindex="-1"></a>    tmp_dataset<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb47-237"><a href="#cb47-237" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(tmp_dataset)</span>
<span id="cb47-238"><a href="#cb47-238" aria-hidden="true" tabindex="-1"></a>    tmp_dataset<span class="sc">$</span>tax_table <span class="sc">%&lt;&gt;%</span> base<span class="sc">::</span><span class="fu">subset</span>(Kingdom <span class="sc">==</span> <span class="st">"k__Archaea"</span> <span class="sc">|</span> Kingdom <span class="sc">==</span> <span class="st">"k__Bacteria"</span>)</span>
<span id="cb47-239"><a href="#cb47-239" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(tmp_dataset)</span>
<span id="cb47-240"><a href="#cb47-240" aria-hidden="true" tabindex="-1"></a>    tmp_dataset<span class="sc">$</span><span class="fu">filter_pollution</span>(<span class="at">taxa =</span> <span class="fu">c</span>(<span class="st">"mitochondria"</span>, <span class="st">"chloroplast"</span>))</span>
<span id="cb47-241"><a href="#cb47-241" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(tmp_dataset)</span>
<span id="cb47-242"><a href="#cb47-242" aria-hidden="true" tabindex="-1"></a>    tmp_dataset<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb47-243"><a href="#cb47-243" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(tmp_dataset)</span>
<span id="cb47-244"><a href="#cb47-244" aria-hidden="true" tabindex="-1"></a>    tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_me"</span>)) </span>
<span id="cb47-245"><a href="#cb47-245" aria-hidden="true" tabindex="-1"></a>    <span class="fu">assign</span>(tmp_name, tmp_dataset)</span>
<span id="cb47-246"><a href="#cb47-246" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-247"><a href="#cb47-247" aria-hidden="true" tabindex="-1"></a>}  </span>
<span id="cb47-248"><a href="#cb47-248" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span>
<span id="cb47-249"><a href="#cb47-249" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-250"><a href="#cb47-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-251"><a href="#cb47-251" aria-hidden="true" tabindex="-1"></a>Here is an example of what a new microeco object looks like when you call it.</span>
<span id="cb47-252"><a href="#cb47-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-255"><a href="#cb47-255" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-256"><a href="#cb47-256" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb47-257"><a href="#cb47-257" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-258"><a href="#cb47-258" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_me</span>
<span id="cb47-259"><a href="#cb47-259" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-260"><a href="#cb47-260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-263"><a href="#cb47-263" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-264"><a href="#cb47-264" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-265"><a href="#cb47-265" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu18_ps_work_me, <span class="st">"files/taxa/rdata/ssu18_ps_work_me.rds"</span>)</span>
<span id="cb47-266"><a href="#cb47-266" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu18_ps_filt_me, <span class="st">"files/taxa/rdata/ssu18_ps_filt_me.rds"</span>)</span>
<span id="cb47-267"><a href="#cb47-267" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu18_ps_perfect_me, <span class="st">"files/taxa/rdata/ssu18_ps_perfect_me.rds"</span>)</span>
<span id="cb47-268"><a href="#cb47-268" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu18_ps_pime_me, <span class="st">"files/taxa/rdata/ssu18_ps_pime_me.rds"</span>)</span>
<span id="cb47-269"><a href="#cb47-269" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-270"><a href="#cb47-270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-271"><a href="#cb47-271" aria-hidden="true" tabindex="-1"></a>5) Now, we calculate the taxa abundance at each taxonomic rank using <span class="in">`cal_abund()`</span>. This function return a list called <span class="in">`taxa_abund`</span> containing several data frame of the abundance information at each taxonomic rank. The list is stored in the microtable object automatically. It’s worth noting that the <span class="in">`cal_abund()`</span> function can be used to solve some complex cases, such as supporting both the relative and absolute abundance calculation and selecting the partial taxonomic columns. More information can be found in the description of the <span class="co">[</span><span class="ot">file2meco package</span><span class="co">](https://github.com/ChiLiubio/file2meco)</span>. In the same loop we also create a <span class="in">`trans_abund`</span> class, which is used to transform taxonomic abundance data for plotting.</span>
<span id="cb47-272"><a href="#cb47-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-275"><a href="#cb47-275" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-276"><a href="#cb47-276" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb47-277"><a href="#cb47-277" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb47-278"><a href="#cb47-278" aria-hidden="true" tabindex="-1"></a>  tmp_me <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_me"</span>)))</span>
<span id="cb47-279"><a href="#cb47-279" aria-hidden="true" tabindex="-1"></a>  tmp_me<span class="sc">$</span><span class="fu">cal_abund</span>()</span>
<span id="cb47-280"><a href="#cb47-280" aria-hidden="true" tabindex="-1"></a>  tmp_me_abund <span class="ot">&lt;-</span> trans_abund<span class="sc">$</span><span class="fu">new</span>(<span class="at">dataset =</span> tmp_me, <span class="at">taxrank =</span> <span class="st">"Phylum"</span>, <span class="at">ntaxa =</span> <span class="dv">12</span>)</span>
<span id="cb47-281"><a href="#cb47-281" aria-hidden="true" tabindex="-1"></a>  tmp_me_abund<span class="sc">$</span>abund_data<span class="sc">$</span>Abundance <span class="ot">&lt;-</span> tmp_me_abund<span class="sc">$</span>abund_data<span class="sc">$</span>Abundance <span class="sc">/</span> <span class="dv">100</span></span>
<span id="cb47-282"><a href="#cb47-282" aria-hidden="true" tabindex="-1"></a>  tmp_me_abund_gr <span class="ot">&lt;-</span> trans_abund<span class="sc">$</span><span class="fu">new</span>(<span class="at">dataset =</span> tmp_me, <span class="at">taxrank =</span> <span class="st">"Phylum"</span>, <span class="at">ntaxa =</span> <span class="dv">12</span>, </span>
<span id="cb47-283"><a href="#cb47-283" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">groupmean =</span> <span class="st">"TEMP"</span>)</span>
<span id="cb47-284"><a href="#cb47-284" aria-hidden="true" tabindex="-1"></a>  tmp_me_abund_gr<span class="sc">$</span>abund_data<span class="sc">$</span>Abundance <span class="ot">&lt;-</span> tmp_me_abund_gr<span class="sc">$</span>abund_data<span class="sc">$</span>Abundance <span class="sc">/</span> <span class="dv">100</span></span>
<span id="cb47-285"><a href="#cb47-285" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_me_abund"</span>)) </span>
<span id="cb47-286"><a href="#cb47-286" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_me_abund)  </span>
<span id="cb47-287"><a href="#cb47-287" aria-hidden="true" tabindex="-1"></a>  tmp_name_gr <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_me_abund_group"</span>)) </span>
<span id="cb47-288"><a href="#cb47-288" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name_gr, tmp_me_abund_gr)  </span>
<span id="cb47-289"><a href="#cb47-289" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-290"><a href="#cb47-290" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-291"><a href="#cb47-291" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span>
<span id="cb47-292"><a href="#cb47-292" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-293"><a href="#cb47-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-294"><a href="#cb47-294" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-295"><a href="#cb47-295" aria-hidden="true" tabindex="-1"></a><span class="in">The result is stored in object$taxa_abund ...</span></span>
<span id="cb47-296"><a href="#cb47-296" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-297"><a href="#cb47-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-298"><a href="#cb47-298" aria-hidden="true" tabindex="-1"></a>6) I prefer to specify the order of taxa in these kinds of plots. We can look the top <span class="in">`ntaxa`</span> (defined above) by accessing the <span class="in">`use_taxanames`</span> character vector of each microtable object.</span>
<span id="cb47-299"><a href="#cb47-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-302"><a href="#cb47-302" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-303"><a href="#cb47-303" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-304"><a href="#cb47-304" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"FULL Data set (ssu18_ps_work_me)"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-305"><a href="#cb47-305" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_me_abund<span class="sc">$</span>use_taxanames</span>
<span id="cb47-306"><a href="#cb47-306" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"--------------------------------------------"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-307"><a href="#cb47-307" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Arbitrary filtered Data set (ssu18_ps_filt_me)"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-308"><a href="#cb47-308" aria-hidden="true" tabindex="-1"></a>ssu18_ps_filt_me_abund<span class="sc">$</span>use_taxanames</span>
<span id="cb47-309"><a href="#cb47-309" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"--------------------------------------------"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-310"><a href="#cb47-310" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"PERfect filtered Data set (ssu18_ps_perfect)"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-311"><a href="#cb47-311" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_me_abund<span class="sc">$</span>use_taxanames</span>
<span id="cb47-312"><a href="#cb47-312" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"--------------------------------------------"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-313"><a href="#cb47-313" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"PIME filtered Data set (ssu18_ps_pime_me)"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-314"><a href="#cb47-314" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_me_abund<span class="sc">$</span>use_taxanames</span>
<span id="cb47-315"><a href="#cb47-315" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"--------------------------------------------"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-316"><a href="#cb47-316" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-317"><a href="#cb47-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-318"><a href="#cb47-318" aria-hidden="true" tabindex="-1"></a>Hum. Similar but not the same. This means we need to define the order for each separately. Once we do that, we will override the <span class="in">`use_taxanames`</span> character vectors with the reordered vectors.</span>
<span id="cb47-319"><a href="#cb47-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-322"><a href="#cb47-322" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-323"><a href="#cb47-323" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb47-324"><a href="#cb47-324" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work_tax_ord <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Alphaproteobacteria"</span>, <span class="st">"Gammaproteobacteria"</span>, <span class="st">"Acidobacteriota"</span>, </span>
<span id="cb47-325"><a href="#cb47-325" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Actinobacteriota"</span>, <span class="st">"Bacteroidota"</span>, <span class="st">"Firmicutes"</span>, <span class="st">"Myxococcota"</span>, </span>
<span id="cb47-326"><a href="#cb47-326" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Verrucomicrobiota"</span>, <span class="st">"Chloroflexi"</span>, <span class="st">"Planctomycetota"</span>, </span>
<span id="cb47-327"><a href="#cb47-327" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Methylomirabilota"</span>, <span class="st">"Crenarchaeota"</span>)</span>
<span id="cb47-328"><a href="#cb47-328" aria-hidden="true" tabindex="-1"></a>ssu18_ps_filt_tax_ord <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Alphaproteobacteria"</span>, <span class="st">"Gammaproteobacteria"</span>, <span class="st">"Acidobacteriota"</span>, </span>
<span id="cb47-329"><a href="#cb47-329" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Actinobacteriota"</span>, <span class="st">"Bacteroidota"</span>, <span class="st">"Firmicutes"</span>, <span class="st">"Myxococcota"</span>, </span>
<span id="cb47-330"><a href="#cb47-330" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Verrucomicrobiota"</span>, <span class="st">"RCP2-54"</span>, <span class="st">"Planctomycetota"</span>, </span>
<span id="cb47-331"><a href="#cb47-331" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Methylomirabilota"</span>, <span class="st">"Crenarchaeota"</span>)</span>
<span id="cb47-332"><a href="#cb47-332" aria-hidden="true" tabindex="-1"></a>ssu18_ps_perfect_tax_ord <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Alphaproteobacteria"</span>, <span class="st">"Gammaproteobacteria"</span>, <span class="st">"Acidobacteriota"</span>, </span>
<span id="cb47-333"><a href="#cb47-333" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"Actinobacteriota"</span>, <span class="st">"Bacteroidota"</span>, <span class="st">"Firmicutes"</span>, <span class="st">"Myxococcota"</span>, </span>
<span id="cb47-334"><a href="#cb47-334" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"Verrucomicrobiota"</span>, <span class="st">"RCP2-54"</span>, <span class="st">"Planctomycetota"</span>, </span>
<span id="cb47-335"><a href="#cb47-335" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"Methylomirabilota"</span>, <span class="st">"Crenarchaeota"</span>)</span>
<span id="cb47-336"><a href="#cb47-336" aria-hidden="true" tabindex="-1"></a>ssu18_ps_pime_tax_ord <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Alphaproteobacteria"</span>, <span class="st">"Gammaproteobacteria"</span>, <span class="st">"Acidobacteriota"</span>, </span>
<span id="cb47-337"><a href="#cb47-337" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Actinobacteriota"</span>, <span class="st">"Bacteroidota"</span>, <span class="st">"Firmicutes"</span>, <span class="st">"Myxococcota"</span>, </span>
<span id="cb47-338"><a href="#cb47-338" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Verrucomicrobiota"</span>, <span class="st">"RCP2-54"</span>, <span class="st">"Nitrospirota"</span>, </span>
<span id="cb47-339"><a href="#cb47-339" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Methylomirabilota"</span>, <span class="st">"Crenarchaeota"</span>)</span>
<span id="cb47-340"><a href="#cb47-340" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-341"><a href="#cb47-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-342"><a href="#cb47-342" aria-hidden="true" tabindex="-1"></a>7) And one more little step before plotting. Here we  **a**) specify a custom color palette and  **b**) specify the sample order. </span>
<span id="cb47-343"><a href="#cb47-343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-346"><a href="#cb47-346" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-347"><a href="#cb47-347" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb47-348"><a href="#cb47-348" aria-hidden="true" tabindex="-1"></a>top_level <span class="ot">&lt;-</span> <span class="st">"Phylum"</span></span>
<span id="cb47-349"><a href="#cb47-349" aria-hidden="true" tabindex="-1"></a>ssu18_colvec.tax <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#00463C"</span>,<span class="st">"#FFD5FD"</span>,<span class="st">"#00A51C"</span>,<span class="st">"#C80B2A"</span>,<span class="st">"#00C7F9"</span>,<span class="st">"#FFA035"</span>,</span>
<span id="cb47-350"><a href="#cb47-350" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"#ED0DFD"</span>,<span class="st">"#0063E5"</span>,<span class="st">"#5FFFDE"</span>,<span class="st">"#C00B6F"</span>,<span class="st">"#00A090"</span>,<span class="st">"#FF95BA"</span>)</span>
<span id="cb47-351"><a href="#cb47-351" aria-hidden="true" tabindex="-1"></a>ssu18_samp_order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"P02_D00_010_C0A"</span>, <span class="st">"P04_D00_010_C0B"</span>, <span class="st">"P06_D00_010_C0C"</span>, <span class="st">"P08_D00_010_C0D"</span>, <span class="st">"P10_D00_010_C0E"</span>, </span>
<span id="cb47-352"><a href="#cb47-352" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"P01_D00_010_W3A"</span>, <span class="st">"P03_D00_010_W3B"</span>, <span class="st">"P05_D00_010_W3C"</span>, <span class="st">"P07_D00_010_W3D"</span>, <span class="st">"P09_D00_010_W3E"</span>, </span>
<span id="cb47-353"><a href="#cb47-353" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"P01_D00_010_W8A"</span>, <span class="st">"P03_D00_010_W8B"</span>, <span class="st">"P05_D00_010_W8C"</span>, <span class="st">"P07_D00_010_W8D"</span>, <span class="st">"P09_D00_010_W8E"</span>)</span>
<span id="cb47-354"><a href="#cb47-354" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-355"><a href="#cb47-355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-356"><a href="#cb47-356" aria-hidden="true" tabindex="-1"></a>8) Now we can generate plots (in a loop) for each faceted data set.</span>
<span id="cb47-357"><a href="#cb47-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-360"><a href="#cb47-360" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-361"><a href="#cb47-361" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb47-362"><a href="#cb47-362" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb47-363"><a href="#cb47-363" aria-hidden="true" tabindex="-1"></a>  tmp_abund <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_me_abund"</span>)))</span>
<span id="cb47-364"><a href="#cb47-364" aria-hidden="true" tabindex="-1"></a>  tmp_tax_order <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_tax_ord"</span>)))</span>
<span id="cb47-365"><a href="#cb47-365" aria-hidden="true" tabindex="-1"></a>  tmp_abund<span class="sc">$</span>use_taxanames <span class="ot">&lt;-</span> tmp_tax_order</span>
<span id="cb47-366"><a href="#cb47-366" aria-hidden="true" tabindex="-1"></a>  tmp_facet_plot <span class="ot">&lt;-</span> tmp_abund<span class="sc">$</span><span class="fu">plot_bar</span>(<span class="at">use_colors =</span> ssu18_colvec.tax, </span>
<span id="cb47-367"><a href="#cb47-367" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">others_color =</span> <span class="st">"#323232"</span>, </span>
<span id="cb47-368"><a href="#cb47-368" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">facet =</span> <span class="st">"TEMP"</span>, </span>
<span id="cb47-369"><a href="#cb47-369" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_keep =</span> <span class="cn">TRUE</span>, </span>
<span id="cb47-370"><a href="#cb47-370" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_type_hor =</span> <span class="cn">FALSE</span>, </span>
<span id="cb47-371"><a href="#cb47-371" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">legend_text_italic =</span> <span class="cn">FALSE</span>, </span>
<span id="cb47-372"><a href="#cb47-372" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_size =</span> <span class="dv">6</span>, </span>
<span id="cb47-373"><a href="#cb47-373" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">facet_color =</span> <span class="st">"#cccccc"</span>, </span>
<span id="cb47-374"><a href="#cb47-374" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">order_x =</span> ssu18_samp_order)</span>
<span id="cb47-375"><a href="#cb47-375" aria-hidden="true" tabindex="-1"></a>  tmp_facet_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_facet_plot"</span>)) </span>
<span id="cb47-376"><a href="#cb47-376" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_facet_name, tmp_facet_plot)  </span>
<span id="cb47-377"><a href="#cb47-377" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-378"><a href="#cb47-378" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-379"><a href="#cb47-379" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-380"><a href="#cb47-380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-381"><a href="#cb47-381" aria-hidden="true" tabindex="-1"></a>Then add a little formatting to the faceted plots.</span>
<span id="cb47-382"><a href="#cb47-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-385"><a href="#cb47-385" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-386"><a href="#cb47-386" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb47-387"><a href="#cb47-387" aria-hidden="true" tabindex="-1"></a>set_to_plot <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ssu18_ps_work_facet_plot"</span>, <span class="st">"ssu18_ps_filt_facet_plot"</span>, <span class="st">"ssu18_ps_perfect_facet_plot"</span>, <span class="st">"ssu18_ps_pime_facet_plot"</span>)</span>
<span id="cb47-388"><a href="#cb47-388" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span>  set_to_plot) {</span>
<span id="cb47-389"><a href="#cb47-389" aria-hidden="true" tabindex="-1"></a>    tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb47-390"><a href="#cb47-390" aria-hidden="true" tabindex="-1"></a>    tmp_get <span class="ot">&lt;-</span> tmp_get <span class="sc">+</span></span>
<span id="cb47-391"><a href="#cb47-391" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb47-392"><a href="#cb47-392" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> top_level, <span class="at">reverse =</span> <span class="cn">FALSE</span>,</span>
<span id="cb47-393"><a href="#cb47-393" aria-hidden="true" tabindex="-1"></a>                               <span class="at">keywidth =</span> <span class="fl">0.7</span>, <span class="at">keyheight =</span> <span class="fl">0.7</span>)) <span class="sc">+</span></span>
<span id="cb47-394"><a href="#cb47-394" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Sample"</span>) <span class="sc">+</span></span>
<span id="cb47-395"><a href="#cb47-395" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-396"><a href="#cb47-396" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"transparent"</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb47-397"><a href="#cb47-397" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"transparent"</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb47-398"><a href="#cb47-398" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb47-399"><a href="#cb47-399" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>),</span>
<span id="cb47-400"><a href="#cb47-400" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb47-401"><a href="#cb47-401" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb47-402"><a href="#cb47-402" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb47-403"><a href="#cb47-403" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>, <span class="at">angle =</span> <span class="dv">90</span>),</span>
<span id="cb47-404"><a href="#cb47-404" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">0</span>),</span>
<span id="cb47-405"><a href="#cb47-405" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb47-406"><a href="#cb47-406" aria-hidden="true" tabindex="-1"></a>      <span class="fu">scale_y_continuous</span>()</span>
<span id="cb47-407"><a href="#cb47-407" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(i, tmp_get)</span>
<span id="cb47-408"><a href="#cb47-408" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>)) </span>
<span id="cb47-409"><a href="#cb47-409" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-410"><a href="#cb47-410" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-411"><a href="#cb47-411" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-412"><a href="#cb47-412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-413"><a href="#cb47-413" aria-hidden="true" tabindex="-1"></a>And now plots for the groupmeans sets. We can use the same taxa order since that should not have changed.</span>
<span id="cb47-414"><a href="#cb47-414" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-417"><a href="#cb47-417" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-418"><a href="#cb47-418" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb47-419"><a href="#cb47-419" aria-hidden="true" tabindex="-1"></a>set_to_plot <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ssu18_ps_work_group_plot"</span>, <span class="st">"ssu18_ps_filt_group_plot"</span>, <span class="st">"ssu18_ps_perfect_group_plot"</span>, <span class="st">"ssu18_ps_pime_group_plot"</span>)</span>
<span id="cb47-420"><a href="#cb47-420" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb47-421"><a href="#cb47-421" aria-hidden="true" tabindex="-1"></a>  tmp_abund <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_me_abund_group"</span>)))</span>
<span id="cb47-422"><a href="#cb47-422" aria-hidden="true" tabindex="-1"></a>  tmp_tax_order <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_tax_ord"</span>)))</span>
<span id="cb47-423"><a href="#cb47-423" aria-hidden="true" tabindex="-1"></a>  tmp_abund<span class="sc">$</span>use_taxanames <span class="ot">&lt;-</span> tmp_tax_order</span>
<span id="cb47-424"><a href="#cb47-424" aria-hidden="true" tabindex="-1"></a>  tmp_group_plot <span class="ot">&lt;-</span> tmp_abund<span class="sc">$</span><span class="fu">plot_bar</span>(<span class="at">use_colors =</span> ssu18_colvec.tax, </span>
<span id="cb47-425"><a href="#cb47-425" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">others_color =</span> <span class="st">"#323232"</span>, </span>
<span id="cb47-426"><a href="#cb47-426" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_keep =</span> <span class="cn">TRUE</span>, </span>
<span id="cb47-427"><a href="#cb47-427" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_type_hor =</span> <span class="cn">TRUE</span>, </span>
<span id="cb47-428"><a href="#cb47-428" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">legend_text_italic =</span> <span class="cn">FALSE</span>, </span>
<span id="cb47-429"><a href="#cb47-429" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_size =</span> <span class="dv">10</span>, </span>
<span id="cb47-430"><a href="#cb47-430" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">facet_color =</span> <span class="st">"#cccccc"</span>)</span>
<span id="cb47-431"><a href="#cb47-431" aria-hidden="true" tabindex="-1"></a>  tmp_group_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_group_plot"</span>)) </span>
<span id="cb47-432"><a href="#cb47-432" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_group_name, tmp_group_plot)  </span>
<span id="cb47-433"><a href="#cb47-433" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-434"><a href="#cb47-434" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-435"><a href="#cb47-435" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-436"><a href="#cb47-436" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-437"><a href="#cb47-437" aria-hidden="true" tabindex="-1"></a>Let's also add a little formatting to the groupmean plots.</span>
<span id="cb47-438"><a href="#cb47-438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-441"><a href="#cb47-441" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-442"><a href="#cb47-442" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb47-443"><a href="#cb47-443" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span>  set_to_plot) {</span>
<span id="cb47-444"><a href="#cb47-444" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb47-445"><a href="#cb47-445" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> tmp_get <span class="sc">+</span></span>
<span id="cb47-446"><a href="#cb47-446" aria-hidden="true" tabindex="-1"></a>       <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb47-447"><a href="#cb47-447" aria-hidden="true" tabindex="-1"></a>       <span class="fu">ylab</span>(<span class="st">"Relative Abundance (% total reads)"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Temperature"</span>) <span class="sc">+</span></span>
<span id="cb47-448"><a href="#cb47-448" aria-hidden="true" tabindex="-1"></a>       <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-449"><a href="#cb47-449" aria-hidden="true" tabindex="-1"></a>             <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"transparent"</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb47-450"><a href="#cb47-450" aria-hidden="true" tabindex="-1"></a>             <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"transparent"</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb47-451"><a href="#cb47-451" aria-hidden="true" tabindex="-1"></a>             <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb47-452"><a href="#cb47-452" aria-hidden="true" tabindex="-1"></a>             <span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb47-453"><a href="#cb47-453" aria-hidden="true" tabindex="-1"></a>             <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb47-454"><a href="#cb47-454" aria-hidden="true" tabindex="-1"></a>             <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)) <span class="sc">+</span> </span>
<span id="cb47-455"><a href="#cb47-455" aria-hidden="true" tabindex="-1"></a>       <span class="fu">scale_y_continuous</span>()</span>
<span id="cb47-456"><a href="#cb47-456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-457"><a href="#cb47-457" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(i, tmp_get)</span>
<span id="cb47-458"><a href="#cb47-458" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-459"><a href="#cb47-459" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-460"><a href="#cb47-460" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-461"><a href="#cb47-461" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-462"><a href="#cb47-462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-463"><a href="#cb47-463" aria-hidden="true" tabindex="-1"></a>10) Finally we use the <span class="in">`patchwork`</span> package to combine the two plots and customize the look.</span>
<span id="cb47-464"><a href="#cb47-464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-467"><a href="#cb47-467" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-468"><a href="#cb47-468" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb47-469"><a href="#cb47-469" aria-hidden="true" tabindex="-1"></a><span class="do">## single index that acts as an index for referencing elements (variables) in a list</span></span>
<span id="cb47-470"><a href="#cb47-470" aria-hidden="true" tabindex="-1"></a><span class="do">##  solution modified from this SO answerhttps://stackoverflow.com/a/54451460</span></span>
<span id="cb47-471"><a href="#cb47-471" aria-hidden="true" tabindex="-1"></a>var_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb47-472"><a href="#cb47-472" aria-hidden="true" tabindex="-1"></a>  <span class="at">var1 =</span> ssu18_data_sets,</span>
<span id="cb47-473"><a href="#cb47-473" aria-hidden="true" tabindex="-1"></a>  <span class="at">var2 =</span> <span class="fu">c</span>(<span class="st">"FULL"</span>, <span class="st">"Arbitrary filtered"</span>, <span class="st">"PERfect filtered"</span>, <span class="st">"PIME filtered"</span>)</span>
<span id="cb47-474"><a href="#cb47-474" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb47-475"><a href="#cb47-475" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span>  <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(var_list<span class="sc">$</span>var1)) {</span>
<span id="cb47-476"><a href="#cb47-476" aria-hidden="true" tabindex="-1"></a>     tmp_plot_final_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(var_list<span class="sc">$</span>var1[j], <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_"</span>, top_level, <span class="st">"_plot_final"</span>))</span>
<span id="cb47-477"><a href="#cb47-477" aria-hidden="true" tabindex="-1"></a>     tmp_set_type <span class="ot">&lt;-</span> var_list<span class="sc">$</span>var2[j]</span>
<span id="cb47-478"><a href="#cb47-478" aria-hidden="true" tabindex="-1"></a>     tmp_p_plot <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(var_list<span class="sc">$</span>var1[j], <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_group_plot"</span>)))</span>
<span id="cb47-479"><a href="#cb47-479" aria-hidden="true" tabindex="-1"></a>     tmp_m_plot <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(var_list<span class="sc">$</span>var1[j], <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_facet_plot"</span>)))</span>
<span id="cb47-480"><a href="#cb47-480" aria-hidden="true" tabindex="-1"></a>     tmp_plot_final <span class="ot">&lt;-</span> tmp_p_plot <span class="sc">+</span> tmp_m_plot</span>
<span id="cb47-481"><a href="#cb47-481" aria-hidden="true" tabindex="-1"></a>     tmp_plot_final <span class="ot">&lt;-</span> tmp_plot_final <span class="sc">+</span></span>
<span id="cb47-482"><a href="#cb47-482" aria-hidden="true" tabindex="-1"></a>         <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">"a"</span>, </span>
<span id="cb47-483"><a href="#cb47-483" aria-hidden="true" tabindex="-1"></a>                         <span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Top"</span>, top_level, <span class="st">"of"</span>, tmp_set_type, <span class="st">"data set"</span>), </span>
<span id="cb47-484"><a href="#cb47-484" aria-hidden="true" tabindex="-1"></a>                         <span class="co">#subtitle = 'Top taxa of non-filtered data',</span></span>
<span id="cb47-485"><a href="#cb47-485" aria-hidden="true" tabindex="-1"></a>                         <span class="at">caption =</span> <span class="st">"a) grouped by temperature.,</span></span>
<span id="cb47-486"><a href="#cb47-486" aria-hidden="true" tabindex="-1"></a><span class="st">                         b) All samples, faceted by temperature."</span>) <span class="sc">+</span> </span>
<span id="cb47-487"><a href="#cb47-487" aria-hidden="true" tabindex="-1"></a>       <span class="fu">plot_layout</span>(<span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)) <span class="sc">&amp;</span></span>
<span id="cb47-488"><a href="#cb47-488" aria-hidden="true" tabindex="-1"></a>       <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">13</span>),</span>
<span id="cb47-489"><a href="#cb47-489" aria-hidden="true" tabindex="-1"></a>             <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>), </span>
<span id="cb47-490"><a href="#cb47-490" aria-hidden="true" tabindex="-1"></a>             <span class="at">plot.tag =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>), </span>
<span id="cb47-491"><a href="#cb47-491" aria-hidden="true" tabindex="-1"></a>             <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>), </span>
<span id="cb47-492"><a href="#cb47-492" aria-hidden="true" tabindex="-1"></a>             <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>)) </span>
<span id="cb47-493"><a href="#cb47-493" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_plot_final_name, tmp_plot_final)</span>
<span id="cb47-494"><a href="#cb47-494" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-495"><a href="#cb47-495" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-496"><a href="#cb47-496" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-497"><a href="#cb47-497" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-500"><a href="#cb47-500" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-501"><a href="#cb47-501" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-502"><a href="#cb47-502" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb47-503"><a href="#cb47-503" aria-hidden="true" tabindex="-1"></a>   tmp_plot_final <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_"</span>, top_level,  <span class="st">"_plot_final"</span>)))</span>
<span id="cb47-504"><a href="#cb47-504" aria-hidden="true" tabindex="-1"></a>   ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_plot_final, <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"files/taxa/figures/"</span>, i, <span class="st">"_tax_div_bar_plots.png"</span>, <span class="at">sep =</span> <span class="st">""</span>), </span>
<span id="cb47-505"><a href="#cb47-505" aria-hidden="true" tabindex="-1"></a>                   <span class="at">height =</span> <span class="dv">14</span>, <span class="at">width =</span> <span class="dv">21</span>, <span class="at">units =</span> <span class="st">'cm'</span>, <span class="at">bg =</span> <span class="st">"white"</span>)</span>
<span id="cb47-506"><a href="#cb47-506" aria-hidden="true" tabindex="-1"></a>   ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_plot_final, <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"files/taxa/figures/"</span>, i, <span class="st">"_tax_div_bar_plots.pdf"</span>, <span class="at">sep =</span> <span class="st">""</span>), </span>
<span id="cb47-507"><a href="#cb47-507" aria-hidden="true" tabindex="-1"></a>                   <span class="at">height =</span> <span class="dv">14</span>, <span class="at">width =</span> <span class="dv">21</span>, <span class="at">units =</span> <span class="st">'cm'</span>, <span class="at">bg =</span> <span class="st">"white"</span>)</span>
<span id="cb47-508"><a href="#cb47-508" aria-hidden="true" tabindex="-1"></a>   <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-509"><a href="#cb47-509" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-510"><a href="#cb47-510" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-511"><a href="#cb47-511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-514"><a href="#cb47-514" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-515"><a href="#cb47-515" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-516"><a href="#cb47-516" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">"cp files/taxa/figures/ssu18_ps_work_tax_div_bar_plots.png include/taxa/ssu18_ps_work_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-517"><a href="#cb47-517" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">"cp files/taxa/figures/ssu18_ps_filt_tax_div_bar_plots.png include/taxa/ssu18_ps_filt_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-518"><a href="#cb47-518" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">"cp files/taxa/figures/ssu18_ps_perfect_tax_div_bar_plots.png include/taxa/ssu18_ps_perfect_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-519"><a href="#cb47-519" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">"cp files/taxa/figures/ssu18_ps_pime_tax_div_bar_plots.png include/taxa/ssu18_ps_pime_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-520"><a href="#cb47-520" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-521"><a href="#cb47-521" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-522"><a href="#cb47-522" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br/&gt;</span></span>
<span id="cb47-523"><a href="#cb47-523" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-524"><a href="#cb47-524" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; It is important to note that while most of the dominant taxa are the same across the FULL, Arbitrary, PIME, and PERfect data sets, there a few key difference:</span></span>
<span id="cb47-525"><a href="#cb47-525" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-526"><a href="#cb47-526" aria-hidden="true" tabindex="-1"></a>1) Planctomycetota was not among the dominant taxa in the PIME data set. Nitrospirota was substituted. </span>
<span id="cb47-527"><a href="#cb47-527" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-528"><a href="#cb47-528" aria-hidden="true" tabindex="-1"></a>2) Chloroflexi was not among the dominant taxa in either the Arbitrary, PERfect, or PIME  data sets. RCP2-54 was substituted in both cases.</span>
<span id="cb47-529"><a href="#cb47-529" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-530"><a href="#cb47-530" aria-hidden="true" tabindex="-1"></a>::: {.column-body-outset}</span>
<span id="cb47-531"><a href="#cb47-531" aria-hidden="true" tabindex="-1"></a>::: {.panel-tabset}</span>
<span id="cb47-532"><a href="#cb47-532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-533"><a href="#cb47-533" aria-hidden="true" tabindex="-1"></a><span class="fu">#### FULL (unfiltered)</span></span>
<span id="cb47-534"><a href="#cb47-534" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-537"><a href="#cb47-537" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-538"><a href="#cb47-538" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-539"><a href="#cb47-539" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-540"><a href="#cb47-540" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-541"><a href="#cb47-541" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/taxa/ssu18_ps_work_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-542"><a href="#cb47-542" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-543"><a href="#cb47-543" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("major_taxa_full_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb47-544"><a href="#cb47-544" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-545"><a href="#cb47-545" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Arbitrary filtered</span></span>
<span id="cb47-546"><a href="#cb47-546" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-549"><a href="#cb47-549" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-550"><a href="#cb47-550" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-551"><a href="#cb47-551" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-552"><a href="#cb47-552" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-553"><a href="#cb47-553" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/taxa/ssu18_ps_filt_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-554"><a href="#cb47-554" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-555"><a href="#cb47-555" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("major_taxa_filt_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb47-556"><a href="#cb47-556" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-557"><a href="#cb47-557" aria-hidden="true" tabindex="-1"></a><span class="fu">#### PERFECT filtered</span></span>
<span id="cb47-558"><a href="#cb47-558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-561"><a href="#cb47-561" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-562"><a href="#cb47-562" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-563"><a href="#cb47-563" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-564"><a href="#cb47-564" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-565"><a href="#cb47-565" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/taxa/ssu18_ps_perfect_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-566"><a href="#cb47-566" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-567"><a href="#cb47-567" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("major_taxa_perfect_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb47-568"><a href="#cb47-568" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-569"><a href="#cb47-569" aria-hidden="true" tabindex="-1"></a><span class="fu">#### PIME filtered</span></span>
<span id="cb47-570"><a href="#cb47-570" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-573"><a href="#cb47-573" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-574"><a href="#cb47-574" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-575"><a href="#cb47-575" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-576"><a href="#cb47-576" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-577"><a href="#cb47-577" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/taxa/ssu18_ps_pime_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-578"><a href="#cb47-578" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-579"><a href="#cb47-579" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("major_taxa_pime_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb47-580"><a href="#cb47-580" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb47-581"><a href="#cb47-581" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb47-582"><a href="#cb47-582" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-583"><a href="#cb47-583" aria-hidden="true" tabindex="-1"></a><span class="fu">## ITS</span></span>
<span id="cb47-584"><a href="#cb47-584" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-587"><a href="#cb47-587" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-588"><a href="#cb47-588" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-589"><a href="#cb47-589" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb47-590"><a href="#cb47-590" aria-hidden="true" tabindex="-1"></a><span class="do">## Initial Load for  ANALYSIS #1</span></span>
<span id="cb47-591"><a href="#cb47-591" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb47-592"><a href="#cb47-592" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">119</span>)</span>
<span id="cb47-593"><a href="#cb47-593" aria-hidden="true" tabindex="-1"></a>its18_ps_work <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"files/data-prep/rdata/its18_ps_work.rds"</span>)</span>
<span id="cb47-594"><a href="#cb47-594" aria-hidden="true" tabindex="-1"></a>its18_ps_filt <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"files/filtering/arbitrary/rdata/its18_ps_filt.rds"</span>)</span>
<span id="cb47-595"><a href="#cb47-595" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"files/filtering/perfect/rdata/its18_ps_perfect.rds"</span>)</span>
<span id="cb47-596"><a href="#cb47-596" aria-hidden="true" tabindex="-1"></a>its18_ps_pime <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"files/filtering/pime/rdata/its18_ps_pime.rds"</span>)</span>
<span id="cb47-597"><a href="#cb47-597" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-598"><a href="#cb47-598" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-601"><a href="#cb47-601" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-602"><a href="#cb47-602" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-603"><a href="#cb47-603" aria-hidden="true" tabindex="-1"></a>swel_col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#2271B2"</span>, <span class="st">"#71B222"</span>, <span class="st">"#B22271"</span>)</span>
<span id="cb47-604"><a href="#cb47-604" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-605"><a href="#cb47-605" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-606"><a href="#cb47-606" aria-hidden="true" tabindex="-1"></a>Here we compare the taxonomic breakdown of the FULL (unfiltered), PIME filtered, and PERfect filtered data sets, split by temperature treatment for the ITS data. The workflow is basically the same as the 16S rRNA data. Click the box below to see the entire workflow.</span>
<span id="cb47-607"><a href="#cb47-607" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-610"><a href="#cb47-610" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-611"><a href="#cb47-611" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-612"><a href="#cb47-612" aria-hidden="true" tabindex="-1"></a>its18_data_sets <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"its18_ps_work"</span>,  <span class="st">"its18_ps_filt"</span>,</span>
<span id="cb47-613"><a href="#cb47-613" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"its18_ps_perfect"</span>, <span class="st">"its18_ps_pime"</span>)</span>
<span id="cb47-614"><a href="#cb47-614" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-615"><a href="#cb47-615" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-616"><a href="#cb47-616" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb47-617"><a href="#cb47-617" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;&lt;strong&gt;</span>Detailed workflow<span class="kw">&lt;/strong&gt;</span> for ITS taxonomic analysis.<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb47-618"><a href="#cb47-618" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-619"><a href="#cb47-619" aria-hidden="true" tabindex="-1"></a>1) Choose the **number** of taxa to display and the taxonomic **level**. Aggregate the rest into "Other".</span>
<span id="cb47-620"><a href="#cb47-620" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-623"><a href="#cb47-623" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-624"><a href="#cb47-624" aria-hidden="true" tabindex="-1"></a>top_hits <span class="ot">&lt;-</span> <span class="dv">14</span></span>
<span id="cb47-625"><a href="#cb47-625" aria-hidden="true" tabindex="-1"></a>top_level <span class="ot">&lt;-</span> <span class="st">"Order"</span></span>
<span id="cb47-626"><a href="#cb47-626" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-627"><a href="#cb47-627" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-628"><a href="#cb47-628" aria-hidden="true" tabindex="-1"></a>As above, in order to use <span class="in">`microeco`</span>, we need to add the rank designation as a prefix to each taxa. For example, <span class="in">`Basidiomycota`</span> is changed to <span class="in">`p__Basidiomycota`</span>. </span>
<span id="cb47-629"><a href="#cb47-629" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-632"><a href="#cb47-632" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-633"><a href="#cb47-633" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its18_data_sets) {</span>
<span id="cb47-634"><a href="#cb47-634" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb47-635"><a href="#cb47-635" aria-hidden="true" tabindex="-1"></a>  tmp_sam_data <span class="ot">&lt;-</span> <span class="fu">sample_data</span>(tmp_get)</span>
<span id="cb47-636"><a href="#cb47-636" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(tmp_get))</span>
<span id="cb47-637"><a href="#cb47-637" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>ASV_ID <span class="ot">&lt;-</span> <span class="cn">NULL</span> <span class="co"># Some have, some do not</span></span>
<span id="cb47-638"><a href="#cb47-638" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>ASV_SEQ <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb47-639"><a href="#cb47-639" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb47-640"><a href="#cb47-640" aria-hidden="true" tabindex="-1"></a><span class="co">#  tmp_tax_data[] &lt;- data.frame(lapply(tmp_tax_data, gsub, </span></span>
<span id="cb47-641"><a href="#cb47-641" aria-hidden="true" tabindex="-1"></a><span class="co">#                                      pattern = "^[k | p | c | o | f]_.*", </span></span>
<span id="cb47-642"><a href="#cb47-642" aria-hidden="true" tabindex="-1"></a><span class="co">#                                      replacement = "", fixed = FALSE))</span></span>
<span id="cb47-643"><a href="#cb47-643" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Kingdom <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"k__"</span>, tmp_tax_data<span class="sc">$</span>Kingdom, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb47-644"><a href="#cb47-644" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Phylum <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"p__"</span>, tmp_tax_data<span class="sc">$</span>Phylum, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb47-645"><a href="#cb47-645" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Class <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"c__"</span>, tmp_tax_data<span class="sc">$</span>Class, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb47-646"><a href="#cb47-646" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Order <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"o__"</span>, tmp_tax_data<span class="sc">$</span>Order, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb47-647"><a href="#cb47-647" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Family <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"f__"</span>, tmp_tax_data<span class="sc">$</span>Family, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb47-648"><a href="#cb47-648" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data<span class="sc">$</span>Genus <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"g__"</span>, tmp_tax_data<span class="sc">$</span>Genus, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb47-649"><a href="#cb47-649" aria-hidden="true" tabindex="-1"></a>  tmp_tax_data <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_tax_data)</span>
<span id="cb47-650"><a href="#cb47-650" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb47-651"><a href="#cb47-651" aria-hidden="true" tabindex="-1"></a>  tmp_ps <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(<span class="fu">otu_table</span>(tmp_get),</span>
<span id="cb47-652"><a href="#cb47-652" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">tax_table</span>(tmp_tax_data),</span>
<span id="cb47-653"><a href="#cb47-653" aria-hidden="true" tabindex="-1"></a>                     tmp_sam_data)</span>
<span id="cb47-654"><a href="#cb47-654" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(i, tmp_ps)</span>
<span id="cb47-655"><a href="#cb47-655" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-656"><a href="#cb47-656" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-657"><a href="#cb47-657" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="fu">tax_table</span>(its18_ps_pime))</span>
<span id="cb47-658"><a href="#cb47-658" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-659"><a href="#cb47-659" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-662"><a href="#cb47-662" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-663"><a href="#cb47-663" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its18_data_sets) {</span>
<span id="cb47-664"><a href="#cb47-664" aria-hidden="true" tabindex="-1"></a>    tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb47-665"><a href="#cb47-665" aria-hidden="true" tabindex="-1"></a>    tmp_otu_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(tmp_get)))</span>
<span id="cb47-666"><a href="#cb47-666" aria-hidden="true" tabindex="-1"></a>    tmp_sample_info <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb47-667"><a href="#cb47-667" aria-hidden="true" tabindex="-1"></a>    tmp_taxonomy_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(tmp_get))</span>
<span id="cb47-668"><a href="#cb47-668" aria-hidden="true" tabindex="-1"></a>    <span class="co">#tmp_taxonomy_table %&lt;&gt;% tidy_taxonomy</span></span>
<span id="cb47-669"><a href="#cb47-669" aria-hidden="true" tabindex="-1"></a>    tmp_dataset <span class="ot">&lt;-</span> microtable<span class="sc">$</span><span class="fu">new</span>(<span class="at">sample_table =</span> tmp_sample_info, </span>
<span id="cb47-670"><a href="#cb47-670" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">otu_table =</span> tmp_otu_table, </span>
<span id="cb47-671"><a href="#cb47-671" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">tax_table =</span> tmp_taxonomy_table)</span>
<span id="cb47-672"><a href="#cb47-672" aria-hidden="true" tabindex="-1"></a>    tmp_dataset<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb47-673"><a href="#cb47-673" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(tmp_dataset)</span>
<span id="cb47-674"><a href="#cb47-674" aria-hidden="true" tabindex="-1"></a>    tmp_dataset<span class="sc">$</span>tax_table <span class="sc">%&lt;&gt;%</span> base<span class="sc">::</span><span class="fu">subset</span>(Kingdom <span class="sc">==</span> <span class="st">"k__Fungi"</span>)</span>
<span id="cb47-675"><a href="#cb47-675" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(tmp_dataset)</span>
<span id="cb47-676"><a href="#cb47-676" aria-hidden="true" tabindex="-1"></a>    tmp_dataset<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb47-677"><a href="#cb47-677" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(tmp_dataset)</span>
<span id="cb47-678"><a href="#cb47-678" aria-hidden="true" tabindex="-1"></a>    tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_me"</span>)) </span>
<span id="cb47-679"><a href="#cb47-679" aria-hidden="true" tabindex="-1"></a>    <span class="fu">assign</span>(tmp_name, tmp_dataset)</span>
<span id="cb47-680"><a href="#cb47-680" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-681"><a href="#cb47-681" aria-hidden="true" tabindex="-1"></a>}  </span>
<span id="cb47-682"><a href="#cb47-682" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span>
<span id="cb47-683"><a href="#cb47-683" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-684"><a href="#cb47-684" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-685"><a href="#cb47-685" aria-hidden="true" tabindex="-1"></a>Here is an example of what a new microeco object looks like when you call it.</span>
<span id="cb47-686"><a href="#cb47-686" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-689"><a href="#cb47-689" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-690"><a href="#cb47-690" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb47-691"><a href="#cb47-691" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-692"><a href="#cb47-692" aria-hidden="true" tabindex="-1"></a>its18_ps_work_me</span>
<span id="cb47-693"><a href="#cb47-693" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-694"><a href="#cb47-694" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-697"><a href="#cb47-697" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-698"><a href="#cb47-698" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-699"><a href="#cb47-699" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(its18_ps_work_me, <span class="st">"files/taxa/rdata/its18_ps_work_me.rds"</span>)</span>
<span id="cb47-700"><a href="#cb47-700" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(its18_ps_filt_me, <span class="st">"files/taxa/rdata/its18_ps_filt_me.rds"</span>)</span>
<span id="cb47-701"><a href="#cb47-701" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(its18_ps_perfect_me, <span class="st">"files/taxa/rdata/its18_ps_perfect_me.rds"</span>)</span>
<span id="cb47-702"><a href="#cb47-702" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(its18_ps_pime_me, <span class="st">"files/taxa/rdata/its18_ps_pime_me.rds"</span>)</span>
<span id="cb47-703"><a href="#cb47-703" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-704"><a href="#cb47-704" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-705"><a href="#cb47-705" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-706"><a href="#cb47-706" aria-hidden="true" tabindex="-1"></a>5) Now, we calculate the taxa abundance at each taxonomic rank using <span class="in">`cal_abund()`</span>. This function return a list called <span class="in">`taxa_abund`</span> containing several data frame of the abundance information at each taxonomic rank. The list is stored in the microtable object automatically. It’s worth noting that the <span class="in">`cal_abund()`</span> function can be used to solve some complex cases, such as supporting both the relative and absolute abundance calculation and selecting the partial taxonomic columns. More information can be found in the description of the <span class="co">[</span><span class="ot">file2meco package</span><span class="co">](https://github.com/ChiLiubio/file2meco)</span>. In the same loop we also create a <span class="in">`trans_abund`</span> class, which is used to transform taxonomic abundance data for plotting.</span>
<span id="cb47-707"><a href="#cb47-707" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-710"><a href="#cb47-710" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-711"><a href="#cb47-711" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its18_data_sets) {</span>
<span id="cb47-712"><a href="#cb47-712" aria-hidden="true" tabindex="-1"></a>  tmp_me <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_me"</span>)))</span>
<span id="cb47-713"><a href="#cb47-713" aria-hidden="true" tabindex="-1"></a>  tmp_me<span class="sc">$</span><span class="fu">cal_abund</span>()</span>
<span id="cb47-714"><a href="#cb47-714" aria-hidden="true" tabindex="-1"></a>  tmp_me_abund <span class="ot">&lt;-</span> trans_abund<span class="sc">$</span><span class="fu">new</span>(<span class="at">dataset =</span> tmp_me, <span class="at">taxrank =</span> top_level, <span class="at">ntaxa =</span> top_hits)</span>
<span id="cb47-715"><a href="#cb47-715" aria-hidden="true" tabindex="-1"></a>  tmp_me_abund<span class="sc">$</span>abund_data<span class="sc">$</span>Abundance <span class="ot">&lt;-</span> tmp_me_abund<span class="sc">$</span>abund_data<span class="sc">$</span>Abundance <span class="sc">/</span> <span class="dv">100</span></span>
<span id="cb47-716"><a href="#cb47-716" aria-hidden="true" tabindex="-1"></a>  tmp_me_abund_gr <span class="ot">&lt;-</span> trans_abund<span class="sc">$</span><span class="fu">new</span>(<span class="at">dataset =</span> tmp_me, <span class="at">taxrank =</span> top_level, <span class="at">ntaxa =</span> top_hits, </span>
<span id="cb47-717"><a href="#cb47-717" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">groupmean =</span> <span class="st">"TEMP"</span>)</span>
<span id="cb47-718"><a href="#cb47-718" aria-hidden="true" tabindex="-1"></a>  tmp_me_abund_gr<span class="sc">$</span>abund_data<span class="sc">$</span>Abundance <span class="ot">&lt;-</span> tmp_me_abund_gr<span class="sc">$</span>abund_data<span class="sc">$</span>Abundance <span class="sc">/</span> <span class="dv">100</span></span>
<span id="cb47-719"><a href="#cb47-719" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_me_abund"</span>)) </span>
<span id="cb47-720"><a href="#cb47-720" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_me_abund)  </span>
<span id="cb47-721"><a href="#cb47-721" aria-hidden="true" tabindex="-1"></a>  tmp_name_gr <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_me_abund_group"</span>)) </span>
<span id="cb47-722"><a href="#cb47-722" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name_gr, tmp_me_abund_gr)  </span>
<span id="cb47-723"><a href="#cb47-723" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-724"><a href="#cb47-724" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-725"><a href="#cb47-725" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-726"><a href="#cb47-726" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-727"><a href="#cb47-727" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-728"><a href="#cb47-728" aria-hidden="true" tabindex="-1"></a><span class="in">The result is stored in object$taxa_abund ...</span></span>
<span id="cb47-729"><a href="#cb47-729" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-730"><a href="#cb47-730" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-731"><a href="#cb47-731" aria-hidden="true" tabindex="-1"></a>6) I prefer to specify the order of taxa in these kinds of plots. We can look the top <span class="in">`ntaxa`</span> (defined above) by accessing the <span class="in">`use_taxanames`</span> character vector of each microtable object.</span>
<span id="cb47-732"><a href="#cb47-732" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-735"><a href="#cb47-735" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-736"><a href="#cb47-736" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-737"><a href="#cb47-737" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"FULL Data set (its18_ps_work_me)"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-738"><a href="#cb47-738" aria-hidden="true" tabindex="-1"></a>its18_ps_work_me_abund<span class="sc">$</span>use_taxanames</span>
<span id="cb47-739"><a href="#cb47-739" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"--------------------------------------------"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-740"><a href="#cb47-740" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Arbitrary filtered Data set (its18_ps_filt_me)"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-741"><a href="#cb47-741" aria-hidden="true" tabindex="-1"></a>its18_ps_filt_me_abund<span class="sc">$</span>use_taxanames</span>
<span id="cb47-742"><a href="#cb47-742" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"--------------------------------------------"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-743"><a href="#cb47-743" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"PERfect filtered Data set (its18_ps_perfect)"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-744"><a href="#cb47-744" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect_me_abund<span class="sc">$</span>use_taxanames</span>
<span id="cb47-745"><a href="#cb47-745" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"--------------------------------------------"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-746"><a href="#cb47-746" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"PIME filtered Data set (its18_ps_pime_me)"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-747"><a href="#cb47-747" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_me_abund<span class="sc">$</span>use_taxanames</span>
<span id="cb47-748"><a href="#cb47-748" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"--------------------------------------------"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-749"><a href="#cb47-749" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-750"><a href="#cb47-750" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-751"><a href="#cb47-751" aria-hidden="true" tabindex="-1"></a>Hum. Similar but not the same. This means we need to define the order for each separately. Once we do that, we will override the <span class="in">`use_taxanames`</span> character vectors with the reordered vectors.</span>
<span id="cb47-752"><a href="#cb47-752" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-755"><a href="#cb47-755" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-756"><a href="#cb47-756" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb47-757"><a href="#cb47-757" aria-hidden="true" tabindex="-1"></a>its18_ps_work_tax_ord <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(</span>
<span id="cb47-758"><a href="#cb47-758" aria-hidden="true" tabindex="-1"></a>  <span class="st">"k_Fungi"</span>,  <span class="st">"p_Ascomycota"</span>,  <span class="st">"c_Agaricomycetes"</span>, <span class="st">"Agaricales"</span>,  </span>
<span id="cb47-759"><a href="#cb47-759" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Archaeorhizomycetales"</span>,  <span class="st">"Capnodiales"</span>,  <span class="st">"Eurotiales"</span>,  <span class="st">"Geastrales"</span>,  </span>
<span id="cb47-760"><a href="#cb47-760" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Glomerales"</span>,  <span class="st">"Helotiales"</span>, <span class="st">"Hypocreales"</span>,  <span class="st">"Saccharomycetales"</span>,  </span>
<span id="cb47-761"><a href="#cb47-761" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Trichosporonales"</span>,  <span class="st">"Xylariales"</span>))   </span>
<span id="cb47-762"><a href="#cb47-762" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-763"><a href="#cb47-763" aria-hidden="true" tabindex="-1"></a>its18_ps_filt_tax_ord <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(</span>
<span id="cb47-764"><a href="#cb47-764" aria-hidden="true" tabindex="-1"></a>  <span class="st">"k_Fungi"</span>,  <span class="st">"p_Ascomycota"</span>,  <span class="st">"Polyporales"</span>, <span class="st">"Agaricales"</span>,  </span>
<span id="cb47-765"><a href="#cb47-765" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Archaeorhizomycetales"</span>,  <span class="st">"Capnodiales"</span>,  <span class="st">"Eurotiales"</span>,  <span class="st">"Geastrales"</span>,  </span>
<span id="cb47-766"><a href="#cb47-766" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Glomerales"</span>,  <span class="st">"Mortierellales"</span>, <span class="st">"Hypocreales"</span>,  <span class="st">"Saccharomycetales"</span>,  </span>
<span id="cb47-767"><a href="#cb47-767" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Trichosporonales"</span>,  <span class="st">"Xylariales"</span>))   </span>
<span id="cb47-768"><a href="#cb47-768" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-769"><a href="#cb47-769" aria-hidden="true" tabindex="-1"></a>its18_ps_perfect_tax_ord <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(</span>
<span id="cb47-770"><a href="#cb47-770" aria-hidden="true" tabindex="-1"></a>  <span class="st">"k_Fungi"</span>,  <span class="st">"p_Ascomycota"</span>,  <span class="st">"c_Agaricomycetes"</span>, <span class="st">"Agaricales"</span>,  </span>
<span id="cb47-771"><a href="#cb47-771" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Archaeorhizomycetales"</span>,  <span class="st">"Capnodiales"</span>,  <span class="st">"Eurotiales"</span>,  <span class="st">"Geastrales"</span>,  </span>
<span id="cb47-772"><a href="#cb47-772" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Glomerales"</span>,  <span class="st">"Helotiales"</span>,   <span class="st">"Hypocreales"</span>,  <span class="st">"Saccharomycetales"</span>, </span>
<span id="cb47-773"><a href="#cb47-773" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Trichosporonales"</span>,  <span class="st">"Pezizales"</span>))   </span>
<span id="cb47-774"><a href="#cb47-774" aria-hidden="true" tabindex="-1"></a>its18_ps_pime_tax_ord <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(</span>
<span id="cb47-775"><a href="#cb47-775" aria-hidden="true" tabindex="-1"></a>  <span class="st">"k_Fungi"</span>,  <span class="st">"p_Ascomycota"</span>,  <span class="st">"Polyporales"</span>,  <span class="st">"Agaricales"</span>,  </span>
<span id="cb47-776"><a href="#cb47-776" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Archaeorhizomycetales"</span>,  <span class="st">"Capnodiales"</span>,  <span class="st">"Eurotiales"</span>,  <span class="st">"Geastrales"</span>, </span>
<span id="cb47-777"><a href="#cb47-777" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Glomerales"</span>,  <span class="st">"Mortierellales"</span>,  <span class="st">"Hypocreales"</span>,  <span class="st">"Saccharomycetales"</span>,  </span>
<span id="cb47-778"><a href="#cb47-778" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Trichosporonales"</span>,  <span class="st">"Pezizales"</span>))   </span>
<span id="cb47-779"><a href="#cb47-779" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-780"><a href="#cb47-780" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-781"><a href="#cb47-781" aria-hidden="true" tabindex="-1"></a>7) And one more little step before plotting. Here we  **a**) specify a custom color palette and  **b**) specify the sample order. </span>
<span id="cb47-782"><a href="#cb47-782" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-785"><a href="#cb47-785" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-786"><a href="#cb47-786" aria-hidden="true" tabindex="-1"></a>its18_colvec.tax <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">"#323232"</span>, <span class="st">"#004949"</span>, <span class="st">"#924900"</span>, <span class="st">"#490092"</span>, </span>
<span id="cb47-787"><a href="#cb47-787" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"#6db6ff"</span>, <span class="st">"#920000"</span>, <span class="st">"#ffb6db"</span>, <span class="st">"#24ff24"</span>, </span>
<span id="cb47-788"><a href="#cb47-788" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"#006ddb"</span>, <span class="st">"#db6d00"</span>, <span class="st">"#b66dff"</span>, <span class="st">"#ffff6d"</span>, </span>
<span id="cb47-789"><a href="#cb47-789" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"#009292"</span>, <span class="st">"#b6dbff"</span>, <span class="st">"#ff6db6"</span>))</span>
<span id="cb47-790"><a href="#cb47-790" aria-hidden="true" tabindex="-1"></a>its18_samp_order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"P02_D00_010_C0A"</span>, <span class="st">"P04_D00_010_C0B"</span>, <span class="st">"P06_D00_010_C0C"</span>, <span class="st">"P08_D00_010_C0D"</span>, <span class="st">"P10_D00_010_C0E"</span>, </span>
<span id="cb47-791"><a href="#cb47-791" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"P01_D00_010_W3A"</span>, <span class="st">"P03_D00_010_W3B"</span>, <span class="st">"P07_D00_010_W3D"</span>, <span class="st">"P09_D00_010_W3E"</span>, </span>
<span id="cb47-792"><a href="#cb47-792" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"P01_D00_010_W8A"</span>, <span class="st">"P03_D00_010_W8B"</span>, <span class="st">"P05_D00_010_W8C"</span>, <span class="st">"P07_D00_010_W8D"</span>)</span>
<span id="cb47-793"><a href="#cb47-793" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-794"><a href="#cb47-794" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-795"><a href="#cb47-795" aria-hidden="true" tabindex="-1"></a>8) Now we can generate plots (in a loop) for each faceted data set.</span>
<span id="cb47-796"><a href="#cb47-796" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-799"><a href="#cb47-799" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-800"><a href="#cb47-800" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its18_data_sets) {</span>
<span id="cb47-801"><a href="#cb47-801" aria-hidden="true" tabindex="-1"></a>  tmp_abund <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_me_abund"</span>)))</span>
<span id="cb47-802"><a href="#cb47-802" aria-hidden="true" tabindex="-1"></a>  tmp_tax_order <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_tax_ord"</span>)))</span>
<span id="cb47-803"><a href="#cb47-803" aria-hidden="true" tabindex="-1"></a>  tmp_abund<span class="sc">$</span>use_taxanames <span class="ot">&lt;-</span> tmp_tax_order</span>
<span id="cb47-804"><a href="#cb47-804" aria-hidden="true" tabindex="-1"></a>  tmp_facet_plot <span class="ot">&lt;-</span> tmp_abund<span class="sc">$</span><span class="fu">plot_bar</span>(<span class="at">use_colors =</span> its18_colvec.tax, </span>
<span id="cb47-805"><a href="#cb47-805" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">others_color =</span> <span class="st">"#323232"</span>, </span>
<span id="cb47-806"><a href="#cb47-806" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">facet =</span> <span class="st">"TEMP"</span>, </span>
<span id="cb47-807"><a href="#cb47-807" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_keep =</span> <span class="cn">TRUE</span>, </span>
<span id="cb47-808"><a href="#cb47-808" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_type_hor =</span> <span class="cn">FALSE</span>, </span>
<span id="cb47-809"><a href="#cb47-809" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">legend_text_italic =</span> <span class="cn">FALSE</span>, </span>
<span id="cb47-810"><a href="#cb47-810" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_size =</span> <span class="dv">6</span>, </span>
<span id="cb47-811"><a href="#cb47-811" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">facet_color =</span> <span class="st">"#cccccc"</span>, </span>
<span id="cb47-812"><a href="#cb47-812" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">order_x =</span> its18_samp_order)</span>
<span id="cb47-813"><a href="#cb47-813" aria-hidden="true" tabindex="-1"></a>  tmp_facet_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_facet_plot"</span>)) </span>
<span id="cb47-814"><a href="#cb47-814" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_facet_name, tmp_facet_plot)  </span>
<span id="cb47-815"><a href="#cb47-815" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-816"><a href="#cb47-816" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-817"><a href="#cb47-817" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-818"><a href="#cb47-818" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-819"><a href="#cb47-819" aria-hidden="true" tabindex="-1"></a>Then add a little formatting to the faceted plots.</span>
<span id="cb47-820"><a href="#cb47-820" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-823"><a href="#cb47-823" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-824"><a href="#cb47-824" aria-hidden="true" tabindex="-1"></a>set_to_plot <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"its18_ps_work_facet_plot"</span>, <span class="st">"its18_ps_filt_facet_plot"</span>, <span class="st">"its18_ps_perfect_facet_plot"</span>, <span class="st">"its18_ps_pime_facet_plot"</span>)</span>
<span id="cb47-825"><a href="#cb47-825" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span>  set_to_plot) {</span>
<span id="cb47-826"><a href="#cb47-826" aria-hidden="true" tabindex="-1"></a>    tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb47-827"><a href="#cb47-827" aria-hidden="true" tabindex="-1"></a>    tmp_get <span class="ot">&lt;-</span> tmp_get <span class="sc">+</span></span>
<span id="cb47-828"><a href="#cb47-828" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb47-829"><a href="#cb47-829" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> top_level, <span class="at">reverse =</span> <span class="cn">FALSE</span>,</span>
<span id="cb47-830"><a href="#cb47-830" aria-hidden="true" tabindex="-1"></a>                               <span class="at">keywidth =</span> <span class="fl">0.7</span>, <span class="at">keyheight =</span> <span class="fl">0.7</span>)) <span class="sc">+</span></span>
<span id="cb47-831"><a href="#cb47-831" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Sample"</span>) <span class="sc">+</span></span>
<span id="cb47-832"><a href="#cb47-832" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-833"><a href="#cb47-833" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"transparent"</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb47-834"><a href="#cb47-834" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"transparent"</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb47-835"><a href="#cb47-835" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb47-836"><a href="#cb47-836" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>),</span>
<span id="cb47-837"><a href="#cb47-837" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb47-838"><a href="#cb47-838" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb47-839"><a href="#cb47-839" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb47-840"><a href="#cb47-840" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>, <span class="at">angle =</span> <span class="dv">90</span>),</span>
<span id="cb47-841"><a href="#cb47-841" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">0</span>),</span>
<span id="cb47-842"><a href="#cb47-842" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb47-843"><a href="#cb47-843" aria-hidden="true" tabindex="-1"></a>      <span class="fu">scale_y_continuous</span>()</span>
<span id="cb47-844"><a href="#cb47-844" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(i, tmp_get)</span>
<span id="cb47-845"><a href="#cb47-845" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>)) </span>
<span id="cb47-846"><a href="#cb47-846" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-847"><a href="#cb47-847" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-848"><a href="#cb47-848" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-849"><a href="#cb47-849" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-850"><a href="#cb47-850" aria-hidden="true" tabindex="-1"></a>And now plots for the groupmeans sets. We can use the same taxa order since that should not have changed.</span>
<span id="cb47-851"><a href="#cb47-851" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-854"><a href="#cb47-854" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-855"><a href="#cb47-855" aria-hidden="true" tabindex="-1"></a>set_to_plot <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"its18_ps_work_group_plot"</span>, <span class="st">"its18_ps_filt_group_plot"</span>, <span class="st">"its18_ps_perfect_group_plot"</span>, <span class="st">"its18_ps_pime_group_plot"</span>)</span>
<span id="cb47-856"><a href="#cb47-856" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its18_data_sets) {</span>
<span id="cb47-857"><a href="#cb47-857" aria-hidden="true" tabindex="-1"></a>  tmp_abund <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_me_abund_group"</span>)))</span>
<span id="cb47-858"><a href="#cb47-858" aria-hidden="true" tabindex="-1"></a>  tmp_tax_order <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_tax_ord"</span>)))</span>
<span id="cb47-859"><a href="#cb47-859" aria-hidden="true" tabindex="-1"></a>  tmp_abund<span class="sc">$</span>use_taxanames <span class="ot">&lt;-</span> tmp_tax_order</span>
<span id="cb47-860"><a href="#cb47-860" aria-hidden="true" tabindex="-1"></a>  tmp_group_plot <span class="ot">&lt;-</span> tmp_abund<span class="sc">$</span><span class="fu">plot_bar</span>(<span class="at">use_colors =</span> its18_colvec.tax, </span>
<span id="cb47-861"><a href="#cb47-861" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">others_color =</span> <span class="st">"#323232"</span>, </span>
<span id="cb47-862"><a href="#cb47-862" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_keep =</span> <span class="cn">TRUE</span>, </span>
<span id="cb47-863"><a href="#cb47-863" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_type_hor =</span> <span class="cn">TRUE</span>, </span>
<span id="cb47-864"><a href="#cb47-864" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">legend_text_italic =</span> <span class="cn">FALSE</span>, </span>
<span id="cb47-865"><a href="#cb47-865" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">xtext_size =</span> <span class="dv">10</span>, </span>
<span id="cb47-866"><a href="#cb47-866" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">facet_color =</span> <span class="st">"#cccccc"</span>)</span>
<span id="cb47-867"><a href="#cb47-867" aria-hidden="true" tabindex="-1"></a>  tmp_group_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_group_plot"</span>)) </span>
<span id="cb47-868"><a href="#cb47-868" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_group_name, tmp_group_plot)  </span>
<span id="cb47-869"><a href="#cb47-869" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-870"><a href="#cb47-870" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-871"><a href="#cb47-871" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-872"><a href="#cb47-872" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-873"><a href="#cb47-873" aria-hidden="true" tabindex="-1"></a>Let's also add a little formatting to the groupmean plots.</span>
<span id="cb47-874"><a href="#cb47-874" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-877"><a href="#cb47-877" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-878"><a href="#cb47-878" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span>  set_to_plot) {</span>
<span id="cb47-879"><a href="#cb47-879" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb47-880"><a href="#cb47-880" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> tmp_get <span class="sc">+</span></span>
<span id="cb47-881"><a href="#cb47-881" aria-hidden="true" tabindex="-1"></a>       <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb47-882"><a href="#cb47-882" aria-hidden="true" tabindex="-1"></a>       <span class="fu">ylab</span>(<span class="st">"Relative Abundance (% total reads)"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Temperature"</span>) <span class="sc">+</span></span>
<span id="cb47-883"><a href="#cb47-883" aria-hidden="true" tabindex="-1"></a>       <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-884"><a href="#cb47-884" aria-hidden="true" tabindex="-1"></a>             <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"transparent"</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb47-885"><a href="#cb47-885" aria-hidden="true" tabindex="-1"></a>             <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"transparent"</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb47-886"><a href="#cb47-886" aria-hidden="true" tabindex="-1"></a>             <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb47-887"><a href="#cb47-887" aria-hidden="true" tabindex="-1"></a>             <span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb47-888"><a href="#cb47-888" aria-hidden="true" tabindex="-1"></a>             <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb47-889"><a href="#cb47-889" aria-hidden="true" tabindex="-1"></a>             <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)) <span class="sc">+</span> </span>
<span id="cb47-890"><a href="#cb47-890" aria-hidden="true" tabindex="-1"></a>       <span class="fu">scale_y_continuous</span>()</span>
<span id="cb47-891"><a href="#cb47-891" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-892"><a href="#cb47-892" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(i, tmp_get)</span>
<span id="cb47-893"><a href="#cb47-893" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-894"><a href="#cb47-894" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-895"><a href="#cb47-895" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-896"><a href="#cb47-896" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-897"><a href="#cb47-897" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-898"><a href="#cb47-898" aria-hidden="true" tabindex="-1"></a>10) Finally we use the <span class="in">`patchwork`</span> package to combine the two plots and customize the look.</span>
<span id="cb47-899"><a href="#cb47-899" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-902"><a href="#cb47-902" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-903"><a href="#cb47-903" aria-hidden="true" tabindex="-1"></a><span class="do">## single index that acts as an index for referencing elements (variables) in a list</span></span>
<span id="cb47-904"><a href="#cb47-904" aria-hidden="true" tabindex="-1"></a><span class="do">##  solution modified from this SO answerhttps://stackoverflow.com/a/54451460</span></span>
<span id="cb47-905"><a href="#cb47-905" aria-hidden="true" tabindex="-1"></a>var_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb47-906"><a href="#cb47-906" aria-hidden="true" tabindex="-1"></a>  <span class="at">var1 =</span> its18_data_sets,</span>
<span id="cb47-907"><a href="#cb47-907" aria-hidden="true" tabindex="-1"></a>  <span class="at">var2 =</span> <span class="fu">c</span>(<span class="st">"FULL"</span>, <span class="st">"Arbitrary filtered"</span>, <span class="st">"PERfect filtered"</span>, <span class="st">"PIME filtered"</span>)</span>
<span id="cb47-908"><a href="#cb47-908" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb47-909"><a href="#cb47-909" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span>  <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(var_list<span class="sc">$</span>var1)) {</span>
<span id="cb47-910"><a href="#cb47-910" aria-hidden="true" tabindex="-1"></a>     tmp_plot_final_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(var_list<span class="sc">$</span>var1[j], <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_"</span>, top_level, <span class="st">"_plot_final"</span>))</span>
<span id="cb47-911"><a href="#cb47-911" aria-hidden="true" tabindex="-1"></a>     tmp_set_type <span class="ot">&lt;-</span> var_list<span class="sc">$</span>var2[j]</span>
<span id="cb47-912"><a href="#cb47-912" aria-hidden="true" tabindex="-1"></a>     tmp_p_plot <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(var_list<span class="sc">$</span>var1[j], <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_group_plot"</span>)))</span>
<span id="cb47-913"><a href="#cb47-913" aria-hidden="true" tabindex="-1"></a>     tmp_m_plot <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(var_list<span class="sc">$</span>var1[j], <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_facet_plot"</span>)))</span>
<span id="cb47-914"><a href="#cb47-914" aria-hidden="true" tabindex="-1"></a>     tmp_plot_final <span class="ot">&lt;-</span> tmp_p_plot <span class="sc">+</span> tmp_m_plot</span>
<span id="cb47-915"><a href="#cb47-915" aria-hidden="true" tabindex="-1"></a>     tmp_plot_final <span class="ot">&lt;-</span> tmp_plot_final <span class="sc">+</span></span>
<span id="cb47-916"><a href="#cb47-916" aria-hidden="true" tabindex="-1"></a>         <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">"a"</span>, </span>
<span id="cb47-917"><a href="#cb47-917" aria-hidden="true" tabindex="-1"></a>                         <span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Top"</span>, top_level, <span class="st">"of"</span>, tmp_set_type, <span class="st">"data set"</span>), </span>
<span id="cb47-918"><a href="#cb47-918" aria-hidden="true" tabindex="-1"></a>                         <span class="co">#subtitle = 'Top taxa of non-filtered data',</span></span>
<span id="cb47-919"><a href="#cb47-919" aria-hidden="true" tabindex="-1"></a>                         <span class="at">caption =</span> <span class="st">"a) grouped by temperature.,</span></span>
<span id="cb47-920"><a href="#cb47-920" aria-hidden="true" tabindex="-1"></a><span class="st">                         b) All samples, faceted by temperature."</span>) <span class="sc">+</span> </span>
<span id="cb47-921"><a href="#cb47-921" aria-hidden="true" tabindex="-1"></a>       <span class="fu">plot_layout</span>(<span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)) <span class="sc">&amp;</span></span>
<span id="cb47-922"><a href="#cb47-922" aria-hidden="true" tabindex="-1"></a>       <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">13</span>),</span>
<span id="cb47-923"><a href="#cb47-923" aria-hidden="true" tabindex="-1"></a>             <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>), </span>
<span id="cb47-924"><a href="#cb47-924" aria-hidden="true" tabindex="-1"></a>             <span class="at">plot.tag =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>), </span>
<span id="cb47-925"><a href="#cb47-925" aria-hidden="true" tabindex="-1"></a>             <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>), </span>
<span id="cb47-926"><a href="#cb47-926" aria-hidden="true" tabindex="-1"></a>             <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>)) </span>
<span id="cb47-927"><a href="#cb47-927" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_plot_final_name, tmp_plot_final)</span>
<span id="cb47-928"><a href="#cb47-928" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-929"><a href="#cb47-929" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-930"><a href="#cb47-930" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-931"><a href="#cb47-931" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-934"><a href="#cb47-934" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-935"><a href="#cb47-935" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-936"><a href="#cb47-936" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its18_data_sets) {</span>
<span id="cb47-937"><a href="#cb47-937" aria-hidden="true" tabindex="-1"></a>   tmp_plot_final <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_"</span>, top_level,  <span class="st">"_plot_final"</span>)))</span>
<span id="cb47-938"><a href="#cb47-938" aria-hidden="true" tabindex="-1"></a>   ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_plot_final, <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"files/taxa/figures/"</span>, i, <span class="st">"_tax_div_bar_plots.png"</span>, <span class="at">sep =</span> <span class="st">""</span>), </span>
<span id="cb47-939"><a href="#cb47-939" aria-hidden="true" tabindex="-1"></a>                   <span class="at">height =</span> <span class="dv">14</span>, <span class="at">width =</span> <span class="dv">21</span>, <span class="at">units =</span> <span class="st">'cm'</span>, <span class="at">bg =</span> <span class="st">"white"</span>)</span>
<span id="cb47-940"><a href="#cb47-940" aria-hidden="true" tabindex="-1"></a>   ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_plot_final, <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"files/taxa/figures/"</span>, i, <span class="st">"_tax_div_bar_plots.pdf"</span>, <span class="at">sep =</span> <span class="st">""</span>), </span>
<span id="cb47-941"><a href="#cb47-941" aria-hidden="true" tabindex="-1"></a>                   <span class="at">height =</span> <span class="dv">14</span>, <span class="at">width =</span> <span class="dv">21</span>, <span class="at">units =</span> <span class="st">'cm'</span>, <span class="at">bg =</span> <span class="st">"white"</span>)</span>
<span id="cb47-942"><a href="#cb47-942" aria-hidden="true" tabindex="-1"></a>   <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-943"><a href="#cb47-943" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-944"><a href="#cb47-944" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-945"><a href="#cb47-945" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-948"><a href="#cb47-948" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-949"><a href="#cb47-949" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-950"><a href="#cb47-950" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">"cp files/taxa/figures/its18_ps_work_tax_div_bar_plots.png include/taxa/its18_ps_work_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-951"><a href="#cb47-951" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">"cp files/taxa/figures/its18_ps_filt_tax_div_bar_plots.png include/taxa/its18_ps_filt_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-952"><a href="#cb47-952" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">"cp files/taxa/figures/its18_ps_perfect_tax_div_bar_plots.png include/taxa/its18_ps_perfect_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-953"><a href="#cb47-953" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">"cp files/taxa/figures/its18_ps_pime_tax_div_bar_plots.png include/taxa/its18_ps_pime_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-954"><a href="#cb47-954" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-955"><a href="#cb47-955" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb47-956"><a href="#cb47-956" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-957"><a href="#cb47-957" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br/&gt;</span></span>
<span id="cb47-958"><a href="#cb47-958" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-959"><a href="#cb47-959" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; It is important to note that while most of the dominant taxa are the same across the FULL, Arbitrary, PIME, and PERfect data sets, there a few key difference:</span></span>
<span id="cb47-960"><a href="#cb47-960" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-961"><a href="#cb47-961" aria-hidden="true" tabindex="-1"></a>1) c_Agaricomycetes &amp; Helotiales were not among the dominant taxa in the Arbitrary or PIME filtered data sets. Polyporales &amp; Mortierellales were substituted, respectively. </span>
<span id="cb47-962"><a href="#cb47-962" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-963"><a href="#cb47-963" aria-hidden="true" tabindex="-1"></a>2) Xylariales was not among the dominant taxa in either the PIME or PERfect data sets. Pezizales was substituted in both cases.</span>
<span id="cb47-964"><a href="#cb47-964" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-965"><a href="#cb47-965" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-966"><a href="#cb47-966" aria-hidden="true" tabindex="-1"></a>::: {.column-body-outset}</span>
<span id="cb47-967"><a href="#cb47-967" aria-hidden="true" tabindex="-1"></a>::: {.panel-tabset}</span>
<span id="cb47-968"><a href="#cb47-968" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-969"><a href="#cb47-969" aria-hidden="true" tabindex="-1"></a><span class="fu">#### FULL (unfiltered)</span></span>
<span id="cb47-970"><a href="#cb47-970" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-973"><a href="#cb47-973" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-974"><a href="#cb47-974" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-975"><a href="#cb47-975" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-976"><a href="#cb47-976" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-977"><a href="#cb47-977" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/taxa/its18_ps_work_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-978"><a href="#cb47-978" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-979"><a href="#cb47-979" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("major_taxa_full_its")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb47-980"><a href="#cb47-980" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-981"><a href="#cb47-981" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Arbitrary filtered</span></span>
<span id="cb47-982"><a href="#cb47-982" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-985"><a href="#cb47-985" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-986"><a href="#cb47-986" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-987"><a href="#cb47-987" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-988"><a href="#cb47-988" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-989"><a href="#cb47-989" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/taxa/its18_ps_filt_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-990"><a href="#cb47-990" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-991"><a href="#cb47-991" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("major_taxa_filt_its")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb47-992"><a href="#cb47-992" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-993"><a href="#cb47-993" aria-hidden="true" tabindex="-1"></a><span class="fu">#### PERFECT filtered</span></span>
<span id="cb47-994"><a href="#cb47-994" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-997"><a href="#cb47-997" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-998"><a href="#cb47-998" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-999"><a href="#cb47-999" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-1000"><a href="#cb47-1000" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-1001"><a href="#cb47-1001" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/taxa/its18_ps_perfect_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-1002"><a href="#cb47-1002" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1003"><a href="#cb47-1003" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("major_taxa_perfect_its")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb47-1004"><a href="#cb47-1004" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1005"><a href="#cb47-1005" aria-hidden="true" tabindex="-1"></a><span class="fu">#### PIME filtered</span></span>
<span id="cb47-1006"><a href="#cb47-1006" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1009"><a href="#cb47-1009" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1010"><a href="#cb47-1010" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-1011"><a href="#cb47-1011" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-1012"><a href="#cb47-1012" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-1013"><a href="#cb47-1013" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/taxa/its18_ps_pime_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-1014"><a href="#cb47-1014" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1015"><a href="#cb47-1015" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("major_taxa_pime_its")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb47-1016"><a href="#cb47-1016" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb47-1017"><a href="#cb47-1017" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb47-1018"><a href="#cb47-1018" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1019"><a href="#cb47-1019" aria-hidden="true" tabindex="-1"></a><span class="fu"># Major Bacterial taxa by Family</span></span>
<span id="cb47-1020"><a href="#cb47-1020" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1021"><a href="#cb47-1021" aria-hidden="true" tabindex="-1"></a>We also look at the relative abundance of groups within dominant Phyla. **Please note** these analyses are for the FULL data set only, before any type of filtering. For this analysis, we use the phyloseq object created in the Data Preparation wokflow section <span class="co">[</span><span class="ot">Rename NA taxonomic ranks</span><span class="co">](data-prep.html#rename-na-taxonomic-ranks)</span> where unclassified ranks were renamed by the next highest rank. See step #1 below for an example of the taxa table structure. </span>
<span id="cb47-1022"><a href="#cb47-1022" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1023"><a href="#cb47-1023" aria-hidden="true" tabindex="-1"></a>:::{.callout-note}</span>
<span id="cb47-1024"><a href="#cb47-1024" aria-hidden="true" tabindex="-1"></a>This code ***only*** works for multiple taxa from a *single* phyloseq object.</span>
<span id="cb47-1025"><a href="#cb47-1025" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb47-1026"><a href="#cb47-1026" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1027"><a href="#cb47-1027" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details</span> <span class="er">markdown</span><span class="ot">=</span><span class="st">"1"</span><span class="kw">&gt;</span></span>
<span id="cb47-1028"><a href="#cb47-1028" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;&lt;strong&gt;</span>Detailed workflow<span class="kw">&lt;/strong&gt;</span> for assessing Bacterial families.<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb47-1029"><a href="#cb47-1029" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1030"><a href="#cb47-1030" aria-hidden="true" tabindex="-1"></a>1) Subset Phyla of interest. </span>
<span id="cb47-1031"><a href="#cb47-1031" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1034"><a href="#cb47-1034" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1035"><a href="#cb47-1035" aria-hidden="true" tabindex="-1"></a>ssu18_ps_work <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"files/data-prep/rdata/ssu18_ps_work.rds"</span>)</span>
<span id="cb47-1036"><a href="#cb47-1036" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ssu18_ps_work)))[,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb47-1037"><a href="#cb47-1037" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1038"><a href="#cb47-1038" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1041"><a href="#cb47-1041" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1042"><a href="#cb47-1042" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-1043"><a href="#cb47-1043" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-1044"><a href="#cb47-1044" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ssu18_ps_work)))[,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb47-1045"><a href="#cb47-1045" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1046"><a href="#cb47-1046" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1047"><a href="#cb47-1047" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1050"><a href="#cb47-1050" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1051"><a href="#cb47-1051" aria-hidden="true" tabindex="-1"></a>ssu18_data_sets <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ssu18_ps_work"</span>)</span>
<span id="cb47-1052"><a href="#cb47-1052" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb47-1053"><a href="#cb47-1053" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_proteo"</span>))</span>
<span id="cb47-1054"><a href="#cb47-1054" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb47-1055"><a href="#cb47-1055" aria-hidden="true" tabindex="-1"></a>     tmp_df <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(tmp_get, Phylum <span class="sc">==</span> <span class="st">"Proteobacteria"</span>)</span>
<span id="cb47-1056"><a href="#cb47-1056" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_df)</span>
<span id="cb47-1057"><a href="#cb47-1057" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_name)</span>
<span id="cb47-1058"><a href="#cb47-1058" aria-hidden="true" tabindex="-1"></a>     tmp_get_taxa <span class="ot">&lt;-</span> <span class="fu">get_taxa_unique</span>(tmp_df,</span>
<span id="cb47-1059"><a href="#cb47-1059" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">taxonomic.rank =</span> <span class="fu">rank_names</span>(tmp_df)[<span class="dv">3</span>],</span>
<span id="cb47-1060"><a href="#cb47-1060" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">errorIfNULL=</span><span class="cn">TRUE</span>)</span>
<span id="cb47-1061"><a href="#cb47-1061" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_get_taxa)</span>
<span id="cb47-1062"><a href="#cb47-1062" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-1063"><a href="#cb47-1063" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"_proteo"</span>))</span>
<span id="cb47-1064"><a href="#cb47-1064" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-1065"><a href="#cb47-1065" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1066"><a href="#cb47-1066" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1067"><a href="#cb47-1067" aria-hidden="true" tabindex="-1"></a>2) Replace Phylum Proteobacteria with the Class name.</span>
<span id="cb47-1068"><a href="#cb47-1068" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1071"><a href="#cb47-1071" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1072"><a href="#cb47-1072" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> ssu18_data_sets) {</span>
<span id="cb47-1073"><a href="#cb47-1073" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(j, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_proteo_clean"</span>))</span>
<span id="cb47-1074"><a href="#cb47-1074" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(j)</span>
<span id="cb47-1075"><a href="#cb47-1075" aria-hidden="true" tabindex="-1"></a>  tmp_clean <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(tmp_get))</span>
<span id="cb47-1076"><a href="#cb47-1076" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1077"><a href="#cb47-1077" aria-hidden="true" tabindex="-1"></a>   <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(tmp_clean)){</span>
<span id="cb47-1078"><a href="#cb47-1078" aria-hidden="true" tabindex="-1"></a>       <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">"Proteobacteria"</span> <span class="sc">&amp;</span> tmp_clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">"Alphaproteobacteria"</span>){</span>
<span id="cb47-1079"><a href="#cb47-1079" aria-hidden="true" tabindex="-1"></a>           phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">"Alphaproteobacteria"</span>)</span>
<span id="cb47-1080"><a href="#cb47-1080" aria-hidden="true" tabindex="-1"></a>           tmp_clean[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb47-1081"><a href="#cb47-1081" aria-hidden="true" tabindex="-1"></a>   }   <span class="cf">else</span> <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">"Proteobacteria"</span> <span class="sc">&amp;</span> tmp_clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">"Gammaproteobacteria"</span>){</span>
<span id="cb47-1082"><a href="#cb47-1082" aria-hidden="true" tabindex="-1"></a>           phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">"Gammaproteobacteria"</span>)</span>
<span id="cb47-1083"><a href="#cb47-1083" aria-hidden="true" tabindex="-1"></a>           tmp_clean[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb47-1084"><a href="#cb47-1084" aria-hidden="true" tabindex="-1"></a>   }   <span class="cf">else</span> <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">"Proteobacteria"</span> <span class="sc">&amp;</span> tmp_clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">"Zetaproteobacteria"</span>){</span>
<span id="cb47-1085"><a href="#cb47-1085" aria-hidden="true" tabindex="-1"></a>              phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">"Zetaproteobacteria"</span>)</span>
<span id="cb47-1086"><a href="#cb47-1086" aria-hidden="true" tabindex="-1"></a>           tmp_clean[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb47-1087"><a href="#cb47-1087" aria-hidden="true" tabindex="-1"></a>   }   <span class="cf">else</span> <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="sc">==</span> <span class="st">"Proteobacteria"</span> <span class="sc">&amp;</span> tmp_clean[i,<span class="dv">3</span>] <span class="sc">==</span> <span class="st">"p_Proteobacteria"</span>){</span>
<span id="cb47-1088"><a href="#cb47-1088" aria-hidden="true" tabindex="-1"></a>           phylum <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">paste</span>(<span class="st">"p_Proteobacteria"</span>)</span>
<span id="cb47-1089"><a href="#cb47-1089" aria-hidden="true" tabindex="-1"></a>           tmp_clean[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> phylum</span>
<span id="cb47-1090"><a href="#cb47-1090" aria-hidden="true" tabindex="-1"></a>       }</span>
<span id="cb47-1091"><a href="#cb47-1091" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb47-1092"><a href="#cb47-1092" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tax_table</span>(tmp_get) <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_clean)</span>
<span id="cb47-1093"><a href="#cb47-1093" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rank_names</span>(tmp_get)</span>
<span id="cb47-1094"><a href="#cb47-1094" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_get)</span>
<span id="cb47-1095"><a href="#cb47-1095" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">c</span>(tmp_name, tmp_get))</span>
<span id="cb47-1096"><a href="#cb47-1096" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(<span class="fu">get_taxa_unique</span>(tmp_get,</span>
<span id="cb47-1097"><a href="#cb47-1097" aria-hidden="true" tabindex="-1"></a>                               <span class="at">taxonomic.rank =</span> <span class="fu">rank_names</span>(tmp_get)[<span class="dv">2</span>],</span>
<span id="cb47-1098"><a href="#cb47-1098" aria-hidden="true" tabindex="-1"></a>                               <span class="at">errorIfNULL=</span><span class="cn">TRUE</span>)))</span>
<span id="cb47-1099"><a href="#cb47-1099" aria-hidden="true" tabindex="-1"></a>  tmp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"files/taxa/rdata/"</span>)</span>
<span id="cb47-1100"><a href="#cb47-1100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(tmp_get, <span class="fu">paste</span>(tmp_path, j, <span class="st">"_clean.rds"</span>, <span class="at">sep =</span> <span class="st">""</span>))</span>
<span id="cb47-1101"><a href="#cb47-1101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-1102"><a href="#cb47-1102" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-1103"><a href="#cb47-1103" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(class, order, phylum)</span>
<span id="cb47-1104"><a href="#cb47-1104" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1105"><a href="#cb47-1105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1108"><a href="#cb47-1108" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1109"><a href="#cb47-1109" aria-hidden="true" tabindex="-1"></a>set_to_plot <span class="ot">&lt;-</span> <span class="st">"ssu18_ps_work"</span></span>
<span id="cb47-1110"><a href="#cb47-1110" aria-hidden="true" tabindex="-1"></a>tax_group <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Alphaproteobacteria"</span>, <span class="st">"Gammaproteobacteria"</span>, </span>
<span id="cb47-1111"><a href="#cb47-1111" aria-hidden="true" tabindex="-1"></a>               <span class="st">"Acidobacteriota"</span>, <span class="st">"Actinobacteriota"</span>, </span>
<span id="cb47-1112"><a href="#cb47-1112" aria-hidden="true" tabindex="-1"></a>               <span class="st">"Bacteroidota"</span>, <span class="st">"Firmicutes"</span>, </span>
<span id="cb47-1113"><a href="#cb47-1113" aria-hidden="true" tabindex="-1"></a>               <span class="st">"Myxococcota"</span>, <span class="st">"Verrucomicrobiota"</span>)</span>
<span id="cb47-1114"><a href="#cb47-1114" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> set_to_plot) {</span>
<span id="cb47-1115"><a href="#cb47-1115" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (j <span class="cf">in</span> tax_group) {</span>
<span id="cb47-1116"><a href="#cb47-1116" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_proteo_clean"</span>)))</span>
<span id="cb47-1117"><a href="#cb47-1117" aria-hidden="true" tabindex="-1"></a>  tmp_sub <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(tmp_get, Phylum <span class="sc">==</span> j)</span>
<span id="cb47-1118"><a href="#cb47-1118" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_"</span>, j))</span>
<span id="cb47-1119"><a href="#cb47-1119" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_sub)</span>
<span id="cb47-1120"><a href="#cb47-1120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-1121"><a href="#cb47-1121" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb47-1122"><a href="#cb47-1122" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-1123"><a href="#cb47-1123" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1124"><a href="#cb47-1124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1127"><a href="#cb47-1127" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1128"><a href="#cb47-1128" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-1129"><a href="#cb47-1129" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb47-1130"><a href="#cb47-1130" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(set_to_plot, <span class="st">"_"</span>, i)))</span>
<span id="cb47-1131"><a href="#cb47-1131" aria-hidden="true" tabindex="-1"></a>  tmp_list <span class="ot">&lt;-</span>  <span class="fu">get_taxa_unique</span>(tmp_get, <span class="at">taxonomic.rank =</span> <span class="fu">rank_names</span>(tmp_get)[<span class="dv">5</span>], <span class="at">errorIfNULL =</span> <span class="cn">TRUE</span>)</span>
<span id="cb47-1132"><a href="#cb47-1132" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb47-1133"><a href="#cb47-1133" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"####################################################"</span>, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb47-1134"><a href="#cb47-1134" aria-hidden="true" tabindex="-1"></a>  tmp_print <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Unique taxa:"</span>, i)</span>
<span id="cb47-1135"><a href="#cb47-1135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(tmp_print, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb47-1136"><a href="#cb47-1136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"####################################################"</span>)</span>
<span id="cb47-1137"><a href="#cb47-1137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb47-1138"><a href="#cb47-1138" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(tmp_list)</span>
<span id="cb47-1139"><a href="#cb47-1139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-1140"><a href="#cb47-1140" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-1141"><a href="#cb47-1141" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1142"><a href="#cb47-1142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1143"><a href="#cb47-1143" aria-hidden="true" tabindex="-1"></a>3) Choose the **number** of taxa to display and the taxonomic **level**. Aggregate the rest into "Other".</span>
<span id="cb47-1144"><a href="#cb47-1144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1147"><a href="#cb47-1147" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1148"><a href="#cb47-1148" aria-hidden="true" tabindex="-1"></a>top_hits <span class="ot">&lt;-</span> <span class="dv">12</span></span>
<span id="cb47-1149"><a href="#cb47-1149" aria-hidden="true" tabindex="-1"></a>top_level <span class="ot">&lt;-</span> <span class="st">"Family"</span></span>
<span id="cb47-1150"><a href="#cb47-1150" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb47-1151"><a href="#cb47-1151" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(set_to_plot, <span class="st">"_"</span>, i)))</span>
<span id="cb47-1152"><a href="#cb47-1152" aria-hidden="true" tabindex="-1"></a>  tmp_otu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(tmp_get)))</span>
<span id="cb47-1153"><a href="#cb47-1153" aria-hidden="true" tabindex="-1"></a>  tmp_otu[] <span class="ot">&lt;-</span> <span class="fu">lapply</span>(tmp_otu, as.numeric)</span>
<span id="cb47-1154"><a href="#cb47-1154" aria-hidden="true" tabindex="-1"></a>  tmp_otu <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_otu)</span>
<span id="cb47-1155"><a href="#cb47-1155" aria-hidden="true" tabindex="-1"></a>  tmp_tax <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">data.frame</span>(<span class="fu">tax_table</span>(tmp_get)))</span>
<span id="cb47-1156"><a href="#cb47-1156" aria-hidden="true" tabindex="-1"></a>  tmp_samples <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb47-1157"><a href="#cb47-1157" aria-hidden="true" tabindex="-1"></a>  tmp_clean_df <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(<span class="fu">otu_table</span>(tmp_otu, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>),</span>
<span id="cb47-1158"><a href="#cb47-1158" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">tax_table</span>(tmp_tax),</span>
<span id="cb47-1159"><a href="#cb47-1159" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">sample_data</span>(tmp_samples))</span>
<span id="cb47-1160"><a href="#cb47-1160" aria-hidden="true" tabindex="-1"></a>  tmp_agg_df <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">aggregate_top_taxa</span>(tmp_clean_df,</span>
<span id="cb47-1161"><a href="#cb47-1161" aria-hidden="true" tabindex="-1"></a>                                                    <span class="at">top =</span> top_hits,</span>
<span id="cb47-1162"><a href="#cb47-1162" aria-hidden="true" tabindex="-1"></a>                                                    <span class="at">level =</span> top_level)</span>
<span id="cb47-1163"><a href="#cb47-1163" aria-hidden="true" tabindex="-1"></a>  tmp_agg_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(set_to_plot, <span class="st">"_"</span>, i, <span class="st">"_agg"</span>))</span>
<span id="cb47-1164"><a href="#cb47-1164" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_agg_name, tmp_agg_df)</span>
<span id="cb47-1165"><a href="#cb47-1165" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"_sep_agg"</span>))</span>
<span id="cb47-1166"><a href="#cb47-1166" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-1167"><a href="#cb47-1167" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1168"><a href="#cb47-1168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1171"><a href="#cb47-1171" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1172"><a href="#cb47-1172" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group){</span>
<span id="cb47-1173"><a href="#cb47-1173" aria-hidden="true" tabindex="-1"></a>  tmp_data <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(set_to_plot, <span class="st">"_"</span>, i, <span class="st">"_agg"</span>))</span>
<span id="cb47-1174"><a href="#cb47-1174" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(tmp_data)</span>
<span id="cb47-1175"><a href="#cb47-1175" aria-hidden="true" tabindex="-1"></a>  tmp_list <span class="ot">&lt;-</span> <span class="fu">get_taxa_unique</span>(tmp_get, <span class="at">taxonomic.rank =</span> <span class="fu">rank_names</span>(tmp_get)[<span class="dv">2</span>],</span>
<span id="cb47-1176"><a href="#cb47-1176" aria-hidden="true" tabindex="-1"></a>                        <span class="at">errorIfNULL =</span> <span class="cn">TRUE</span>)</span>
<span id="cb47-1177"><a href="#cb47-1177" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(tmp_data, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">"_order"</span>))</span>
<span id="cb47-1178"><a href="#cb47-1178" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_list)</span>
<span id="cb47-1179"><a href="#cb47-1179" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-1180"><a href="#cb47-1180" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-1181"><a href="#cb47-1181" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1182"><a href="#cb47-1182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1185"><a href="#cb47-1185" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1186"><a href="#cb47-1186" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb47-1187"><a href="#cb47-1187" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(set_to_plot, <span class="st">"_"</span>, i, <span class="st">"_agg_order"</span>)))</span>
<span id="cb47-1188"><a href="#cb47-1188" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb47-1189"><a href="#cb47-1189" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"#########"</span>, i, <span class="st">"########"</span>, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb47-1190"><a href="#cb47-1190" aria-hidden="true" tabindex="-1"></a>  tmp_print <span class="ot">&lt;-</span> <span class="fu">c</span>(tmp_get)</span>
<span id="cb47-1191"><a href="#cb47-1191" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(tmp_print, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb47-1192"><a href="#cb47-1192" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"####################################################"</span>)</span>
<span id="cb47-1193"><a href="#cb47-1193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb47-1194"><a href="#cb47-1194" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-1195"><a href="#cb47-1195" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-1196"><a href="#cb47-1196" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(i, j)</span>
<span id="cb47-1197"><a href="#cb47-1197" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1198"><a href="#cb47-1198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1201"><a href="#cb47-1201" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1202"><a href="#cb47-1202" aria-hidden="true" tabindex="-1"></a>tmp_order <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">"Other"</span>, <span class="st">"c_Alphaproteobacteria"</span>, <span class="st">"o_Elsterales"</span>, <span class="st">"o_Azospirillales"</span>, <span class="st">"Rhizobiales_Incertae_Sedis"</span>, <span class="st">"Xanthobacteraceae"</span>, <span class="st">"Sphingomonadaceae"</span>, <span class="st">"Rhizobiaceae"</span>, <span class="st">"Micropepsaceae"</span>, <span class="st">"Methyloligellaceae"</span>, <span class="st">"KF-JG30-B3"</span>, <span class="st">"Hyphomicrobiaceae"</span>, <span class="st">"Dongiaceae"</span>))</span>
<span id="cb47-1203"><a href="#cb47-1203" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(<span class="fu">paste</span>(set_to_plot, <span class="st">"_"</span>, <span class="st">"Alphaproteobacteria"</span>, <span class="st">"_agg_order"</span>, <span class="at">sep =</span> <span class="st">""</span>), tmp_order) </span>
<span id="cb47-1204"><a href="#cb47-1204" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span>
<span id="cb47-1205"><a href="#cb47-1205" aria-hidden="true" tabindex="-1"></a>tmp_order <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">"Other"</span>, <span class="st">"c_Gammaproteobacteria"</span>, <span class="st">"o_PLTA13"</span>, <span class="st">"o_CCD24"</span>, <span class="st">"Unknown_Family"</span>, <span class="st">"Xanthomonadaceae"</span>, <span class="st">"TRA3-20"</span>, <span class="st">"Steroidobacteraceae"</span>, <span class="st">"SC-I-84"</span>, <span class="st">"Nitrosomonadaceae"</span>, <span class="st">"Comamonadaceae"</span>, <span class="st">"Burkholderiaceae"</span>, <span class="st">"B1-7BS"</span>))</span>
<span id="cb47-1206"><a href="#cb47-1206" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(<span class="fu">paste</span>(set_to_plot, <span class="st">"_"</span>, <span class="st">"Gammaproteobacteria"</span>, <span class="st">"_agg_order"</span>, <span class="at">sep =</span> <span class="st">""</span>), tmp_order) </span>
<span id="cb47-1207"><a href="#cb47-1207" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span>
<span id="cb47-1208"><a href="#cb47-1208" aria-hidden="true" tabindex="-1"></a>tmp_order <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">"Other"</span>, <span class="st">"c_Subgroup_22"</span>, <span class="st">"c_Subgroup_25"</span>, <span class="st">"c_Subgroup_5"</span>, <span class="st">"o_Vicinamibacterales"</span>, <span class="st">"o_11-24"</span>, <span class="st">"o_Subgroup_17"</span>, <span class="st">"o_Subgroup_2"</span>, <span class="st">"o_Subgroup_7"</span>, <span class="st">"o_Acidobacteriales"</span>, <span class="st">"Vicinamibacteraceae"</span>, <span class="st">"Solibacteraceae"</span>, <span class="st">"Pyrinomonadaceae"</span>))</span>
<span id="cb47-1209"><a href="#cb47-1209" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(<span class="fu">paste</span>(set_to_plot, <span class="st">"_"</span>, <span class="st">"Acidobacteriota"</span>, <span class="st">"_agg_order"</span>, <span class="at">sep =</span> <span class="st">""</span>), tmp_order) </span>
<span id="cb47-1210"><a href="#cb47-1210" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span>
<span id="cb47-1211"><a href="#cb47-1211" aria-hidden="true" tabindex="-1"></a>tmp_order <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">"Other"</span>, <span class="st">"c_MB-A2-108"</span>, <span class="st">"o_IMCC26256"</span>, <span class="st">"o_Gaiellales"</span>, <span class="st">"o_Frankiales"</span>, <span class="st">"Streptomycetaceae"</span>, <span class="st">"Solirubrobacteraceae"</span>, <span class="st">"Nocardioidaceae"</span>, <span class="st">"Mycobacteriaceae"</span>, <span class="st">"Micromonosporaceae"</span>, <span class="st">"Gaiellaceae"</span>, <span class="st">"Acidothermaceae"</span>, <span class="st">"67-14"</span>))</span>
<span id="cb47-1212"><a href="#cb47-1212" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(<span class="fu">paste</span>(set_to_plot, <span class="st">"_"</span>, <span class="st">"Actinobacteriota"</span>, <span class="st">"_agg_order"</span>, <span class="at">sep =</span> <span class="st">""</span>), tmp_order) </span>
<span id="cb47-1213"><a href="#cb47-1213" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span>
<span id="cb47-1214"><a href="#cb47-1214" aria-hidden="true" tabindex="-1"></a>tmp_order <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">"Other"</span>, <span class="st">"c_SJA-28"</span>, <span class="st">"o_Chitinophagales"</span>, <span class="st">"Sphingobacteriaceae"</span>, <span class="st">"Saprospiraceae"</span>, <span class="st">"Microscillaceae"</span>, <span class="st">"Hymenobacteraceae"</span>, <span class="st">"Flavobacteriaceae"</span>, <span class="st">"Cytophagaceae"</span>, <span class="st">"Chitinophagaceae"</span>, <span class="st">"BSV26"</span>, <span class="st">"env.OPS_17"</span>, <span class="st">"AKYH767"</span>))</span>
<span id="cb47-1215"><a href="#cb47-1215" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(<span class="fu">paste</span>(set_to_plot, <span class="st">"_"</span>, <span class="st">"Bacteroidota"</span>, <span class="st">"_agg_order"</span>, <span class="at">sep =</span> <span class="st">""</span>), tmp_order) </span>
<span id="cb47-1216"><a href="#cb47-1216" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span>
<span id="cb47-1217"><a href="#cb47-1217" aria-hidden="true" tabindex="-1"></a>tmp_order <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">"Other"</span>, <span class="st">"p_Firmicutes"</span>, <span class="st">"c_Bacilli"</span>, <span class="st">"o_Bacillales"</span>, <span class="st">"Thermoactinomycetaceae"</span>, <span class="st">"type_III"</span>, <span class="st">"Paenibacillaceae"</span>, <span class="st">"Lachnospiraceae"</span>, <span class="st">"Hungateiclostridiaceae"</span>, <span class="st">"Clostridiaceae"</span>, <span class="st">"Bacillaceae"</span>, <span class="st">"Alicyclobacillaceae"</span>, <span class="st">"Planococcaceae"</span>))</span>
<span id="cb47-1218"><a href="#cb47-1218" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(<span class="fu">paste</span>(set_to_plot, <span class="st">"_"</span>, <span class="st">"Firmicutes"</span>, <span class="st">"_agg_order"</span>, <span class="at">sep =</span> <span class="st">""</span>), tmp_order) </span>
<span id="cb47-1219"><a href="#cb47-1219" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span>
<span id="cb47-1220"><a href="#cb47-1220" aria-hidden="true" tabindex="-1"></a>tmp_order <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">"Other"</span>, <span class="st">"p_Myxococcota"</span>, <span class="st">"c_bacteriap25"</span>, <span class="st">"c_Polyangia"</span>, <span class="st">"o_mle1-27"</span>, <span class="st">"o_MSB-4B10"</span>, <span class="st">"Sandaracinaceae"</span>, <span class="st">"Polyangiaceae"</span>, <span class="st">"Phaselicystidaceae"</span>, <span class="st">"Myxococcaceae"</span>, <span class="st">"Haliangiaceae"</span>, <span class="st">"BIrii41"</span>, <span class="st">"Anaeromyxobacteraceae"</span>))</span>
<span id="cb47-1221"><a href="#cb47-1221" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(<span class="fu">paste</span>(set_to_plot, <span class="st">"_"</span>, <span class="st">"Myxococcota"</span>, <span class="st">"_agg_order"</span>, <span class="at">sep =</span> <span class="st">""</span>), tmp_order) </span>
<span id="cb47-1222"><a href="#cb47-1222" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span>
<span id="cb47-1223"><a href="#cb47-1223" aria-hidden="true" tabindex="-1"></a>tmp_order <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">"Other"</span>, <span class="st">"c_Verrucomicrobiae"</span>, <span class="st">"o_Chlamydiales"</span>, <span class="st">"o_LD1-PA32"</span>, <span class="st">"o_S-BQ2-57_soil_group"</span>, <span class="st">"Xiphinematobacteraceae"</span>, <span class="st">"Simkaniaceae"</span>, <span class="st">"Pedosphaeraceae"</span>, <span class="st">"Parachlamydiaceae"</span>, <span class="st">"Opitutaceae"</span>, <span class="st">"Omnitrophaceae"</span>, <span class="st">"cvE6"</span>, <span class="st">"Chthoniobacteraceae"</span>))</span>
<span id="cb47-1224"><a href="#cb47-1224" aria-hidden="true" tabindex="-1"></a><span class="fu">assign</span>(<span class="fu">paste</span>(set_to_plot, <span class="st">"_"</span>, <span class="st">"Verrucomicrobiota"</span>, <span class="st">"_agg_order"</span>, <span class="at">sep =</span> <span class="st">""</span>), tmp_order) </span>
<span id="cb47-1225"><a href="#cb47-1225" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span>
<span id="cb47-1226"><a href="#cb47-1226" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1227"><a href="#cb47-1227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1228"><a href="#cb47-1228" aria-hidden="true" tabindex="-1"></a>4) Now, transform the data to relative abundance.</span>
<span id="cb47-1229"><a href="#cb47-1229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1232"><a href="#cb47-1232" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1233"><a href="#cb47-1233" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb47-1234"><a href="#cb47-1234" aria-hidden="true" tabindex="-1"></a>  tmp_agg <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(set_to_plot, <span class="st">"_"</span>, i, <span class="st">"_agg"</span>))</span>
<span id="cb47-1235"><a href="#cb47-1235" aria-hidden="true" tabindex="-1"></a>  tmp_order <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(tmp_agg, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_order"</span>))</span>
<span id="cb47-1236"><a href="#cb47-1236" aria-hidden="true" tabindex="-1"></a>  tmp_get_agg <span class="ot">&lt;-</span> <span class="fu">get</span>(tmp_agg)</span>
<span id="cb47-1237"><a href="#cb47-1237" aria-hidden="true" tabindex="-1"></a>  tmp_get_order <span class="ot">&lt;-</span> <span class="fu">get</span>(tmp_order)</span>
<span id="cb47-1238"><a href="#cb47-1238" aria-hidden="true" tabindex="-1"></a>  tmp_df <span class="ot">&lt;-</span> tmp_get_agg <span class="sc">%&gt;%</span></span>
<span id="cb47-1239"><a href="#cb47-1239" aria-hidden="true" tabindex="-1"></a>            <span class="fu">transform_sample_counts</span>(<span class="cf">function</span>(x) {x<span class="sc">/</span><span class="fu">sum</span>(x)} ) <span class="sc">%&gt;%</span></span>
<span id="cb47-1240"><a href="#cb47-1240" aria-hidden="true" tabindex="-1"></a>            <span class="fu">psmelt</span>()</span>
<span id="cb47-1241"><a href="#cb47-1241" aria-hidden="true" tabindex="-1"></a>  tmp_df[[top_level]] <span class="ot">&lt;-</span> gdata<span class="sc">::</span><span class="fu">reorder.factor</span>(tmp_df[[top_level]],</span>
<span id="cb47-1242"><a href="#cb47-1242" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">new.order =</span> <span class="fu">rev</span>(tmp_get_order))</span>
<span id="cb47-1243"><a href="#cb47-1243" aria-hidden="true" tabindex="-1"></a>  tmp_df <span class="ot">&lt;-</span> tmp_df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">get</span>(top_level))</span>
<span id="cb47-1244"><a href="#cb47-1244" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(tmp_agg, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">"_tax"</span>))</span>
<span id="cb47-1245"><a href="#cb47-1245" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_df)</span>
<span id="cb47-1246"><a href="#cb47-1246" aria-hidden="true" tabindex="-1"></a>  <span class="co">#print(c(i, tmp_name, tmp_agg))</span></span>
<span id="cb47-1247"><a href="#cb47-1247" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-1248"><a href="#cb47-1248" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-1249"><a href="#cb47-1249" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1250"><a href="#cb47-1250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1251"><a href="#cb47-1251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1254"><a href="#cb47-1254" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1255"><a href="#cb47-1255" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb47-1256"><a href="#cb47-1256" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(set_to_plot, <span class="st">"_"</span>, i, <span class="st">"_agg_tax"</span>)))</span>
<span id="cb47-1257"><a href="#cb47-1257" aria-hidden="true" tabindex="-1"></a>  tmp_levels <span class="ot">&lt;-</span> <span class="fu">levels</span>(tmp_get[[top_level]])</span>
<span id="cb47-1258"><a href="#cb47-1258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">c</span>(i, tmp_levels))</span>
<span id="cb47-1259"><a href="#cb47-1259" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-1260"><a href="#cb47-1260" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1261"><a href="#cb47-1261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1262"><a href="#cb47-1262" aria-hidden="true" tabindex="-1"></a>5) Plot the data for a single phyloseq object. Here you use an aggregated tax file.</span>
<span id="cb47-1263"><a href="#cb47-1263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1266"><a href="#cb47-1266" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1267"><a href="#cb47-1267" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-1268"><a href="#cb47-1268" aria-hidden="true" tabindex="-1"></a>ssu18_colvec.tax <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb47-1269"><a href="#cb47-1269" aria-hidden="true" tabindex="-1"></a><span class="st">"#3D3C04"</span>,</span>
<span id="cb47-1270"><a href="#cb47-1270" aria-hidden="true" tabindex="-1"></a><span class="st">"#FF95BA"</span>,</span>
<span id="cb47-1271"><a href="#cb47-1271" aria-hidden="true" tabindex="-1"></a><span class="st">"#00A090"</span>,</span>
<span id="cb47-1272"><a href="#cb47-1272" aria-hidden="true" tabindex="-1"></a><span class="st">"#C00B6F"</span>,</span>
<span id="cb47-1273"><a href="#cb47-1273" aria-hidden="true" tabindex="-1"></a><span class="co">#"#9BFF2D",</span></span>
<span id="cb47-1274"><a href="#cb47-1274" aria-hidden="true" tabindex="-1"></a><span class="co">#"#590A87",</span></span>
<span id="cb47-1275"><a href="#cb47-1275" aria-hidden="true" tabindex="-1"></a><span class="st">"#5FFFDE"</span>,</span>
<span id="cb47-1276"><a href="#cb47-1276" aria-hidden="true" tabindex="-1"></a><span class="st">"#0063E5"</span>,</span>
<span id="cb47-1277"><a href="#cb47-1277" aria-hidden="true" tabindex="-1"></a><span class="st">"#ED0DFD"</span>,</span>
<span id="cb47-1278"><a href="#cb47-1278" aria-hidden="true" tabindex="-1"></a><span class="st">"#FFA035"</span>,</span>
<span id="cb47-1279"><a href="#cb47-1279" aria-hidden="true" tabindex="-1"></a><span class="st">"#00C7F9"</span>,</span>
<span id="cb47-1280"><a href="#cb47-1280" aria-hidden="true" tabindex="-1"></a><span class="st">"#C80B2A"</span>,</span>
<span id="cb47-1281"><a href="#cb47-1281" aria-hidden="true" tabindex="-1"></a><span class="st">"#00A51C"</span>,</span>
<span id="cb47-1282"><a href="#cb47-1282" aria-hidden="true" tabindex="-1"></a><span class="st">"#FFD5FD"</span>,</span>
<span id="cb47-1283"><a href="#cb47-1283" aria-hidden="true" tabindex="-1"></a><span class="st">"#00463C"</span></span>
<span id="cb47-1284"><a href="#cb47-1284" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb47-1285"><a href="#cb47-1285" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1286"><a href="#cb47-1286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1289"><a href="#cb47-1289" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1290"><a href="#cb47-1290" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb47-1291"><a href="#cb47-1291" aria-hidden="true" tabindex="-1"></a>      tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(set_to_plot, <span class="st">"_"</span>, i, <span class="st">"_agg_tax"</span>)))</span>
<span id="cb47-1292"><a href="#cb47-1292" aria-hidden="true" tabindex="-1"></a>      tmp_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(tmp_get, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(TEMP), </span>
<span id="cb47-1293"><a href="#cb47-1293" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">y =</span> Abundance, <span class="at">fill =</span> <span class="fu">get</span>(top_level))) <span class="sc">+</span></span>
<span id="cb47-1294"><a href="#cb47-1294" aria-hidden="true" tabindex="-1"></a>          <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"fill"</span>) <span class="sc">+</span></span>
<span id="cb47-1295"><a href="#cb47-1295" aria-hidden="true" tabindex="-1"></a>          <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> ssu18_colvec.tax) <span class="sc">+</span></span>
<span id="cb47-1296"><a href="#cb47-1296" aria-hidden="true" tabindex="-1"></a>          <span class="co">#scale_x_discrete("Temperature", expand = waiver(), position = "bottom", drop = FALSE) +</span></span>
<span id="cb47-1297"><a href="#cb47-1297" aria-hidden="true" tabindex="-1"></a>          <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb47-1298"><a href="#cb47-1298" aria-hidden="true" tabindex="-1"></a>          <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> top_level)) <span class="sc">+</span></span>
<span id="cb47-1299"><a href="#cb47-1299" aria-hidden="true" tabindex="-1"></a>          <span class="co">#guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +</span></span>
<span id="cb47-1300"><a href="#cb47-1300" aria-hidden="true" tabindex="-1"></a>          <span class="fu">ylab</span>(<span class="st">"Relative Abundance (% total reads)"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Temperature"</span>) <span class="sc">+</span></span>
<span id="cb47-1301"><a href="#cb47-1301" aria-hidden="true" tabindex="-1"></a>          <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-1302"><a href="#cb47-1302" aria-hidden="true" tabindex="-1"></a>                <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"transparent"</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb47-1303"><a href="#cb47-1303" aria-hidden="true" tabindex="-1"></a>                <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"transparent"</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb47-1304"><a href="#cb47-1304" aria-hidden="true" tabindex="-1"></a>                <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb47-1305"><a href="#cb47-1305" aria-hidden="true" tabindex="-1"></a>                <span class="co">#axis.text = element_text(size = 6),</span></span>
<span id="cb47-1306"><a href="#cb47-1306" aria-hidden="true" tabindex="-1"></a>                <span class="co">#axis.title = element_text(size = 8),</span></span>
<span id="cb47-1307"><a href="#cb47-1307" aria-hidden="true" tabindex="-1"></a>                <span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb47-1308"><a href="#cb47-1308" aria-hidden="true" tabindex="-1"></a>      tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(set_to_plot, <span class="st">"_"</span>, ., <span class="st">"_plot"</span>))</span>
<span id="cb47-1309"><a href="#cb47-1309" aria-hidden="true" tabindex="-1"></a>      <span class="fu">assign</span>(tmp_name, tmp_plot)</span>
<span id="cb47-1310"><a href="#cb47-1310" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-1311"><a href="#cb47-1311" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-1312"><a href="#cb47-1312" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1313"><a href="#cb47-1313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1314"><a href="#cb47-1314" aria-hidden="true" tabindex="-1"></a>6) Plot the data for multiple taxa. Here again you use an aggregated tax file. This code can be used to generate plots for multiple data frames by adding the base  phyloseq names to the <span class="in">`ssu18_split_df`</span> variable. This code will also facet the plots by a metadata variable. If you do not want to facet remove the line beginning with <span class="in">`facet_grid`</span>.</span>
<span id="cb47-1315"><a href="#cb47-1315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1318"><a href="#cb47-1318" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1319"><a href="#cb47-1319" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb47-1320"><a href="#cb47-1320" aria-hidden="true" tabindex="-1"></a>  tmp_level_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(set_to_plot, <span class="st">"_"</span>, .)))</span>
<span id="cb47-1321"><a href="#cb47-1321" aria-hidden="true" tabindex="-1"></a>  tmp_level <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_level_get))</span>
<span id="cb47-1322"><a href="#cb47-1322" aria-hidden="true" tabindex="-1"></a>  tmp_level <span class="ot">&lt;-</span> tmp_level[<span class="fu">order</span>(tmp_level<span class="sc">$</span>TEMP), ]</span>
<span id="cb47-1323"><a href="#cb47-1323" aria-hidden="true" tabindex="-1"></a>  tmp_level <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(tmp_level<span class="sc">$</span>SamName)</span>
<span id="cb47-1324"><a href="#cb47-1324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1325"><a href="#cb47-1325" aria-hidden="true" tabindex="-1"></a>  tmp_agg_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(set_to_plot, <span class="st">"_"</span>, .,  <span class="st">"_agg_tax"</span>))</span>
<span id="cb47-1326"><a href="#cb47-1326" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(tmp_agg_name)</span>
<span id="cb47-1327"><a href="#cb47-1327" aria-hidden="true" tabindex="-1"></a>  tmp_df <span class="ot">&lt;-</span> reshape<span class="sc">::</span><span class="fu">melt</span>(tmp_get, <span class="at">id.vars =</span> <span class="fu">c</span>(<span class="st">"Sample"</span>, <span class="st">"TEMP"</span>, <span class="st">"Abundance"</span>, <span class="st">"Family"</span>))</span>
<span id="cb47-1328"><a href="#cb47-1328" aria-hidden="true" tabindex="-1"></a>  tmp_plot_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(set_to_plot, <span class="st">"_"</span>, ., <span class="st">"_plot_melt"</span>))</span>
<span id="cb47-1329"><a href="#cb47-1329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1330"><a href="#cb47-1330" aria-hidden="true" tabindex="-1"></a>  tmp_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(tmp_df,</span>
<span id="cb47-1331"><a href="#cb47-1331" aria-hidden="true" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">x =</span> Sample,</span>
<span id="cb47-1332"><a href="#cb47-1332" aria-hidden="true" tabindex="-1"></a>                    <span class="at">y =</span> Abundance, <span class="at">fill =</span> <span class="fu">get</span>(top_level))) <span class="sc">+</span></span>
<span id="cb47-1333"><a href="#cb47-1333" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(. <span class="sc">~</span>TEMP, <span class="at">scale =</span> <span class="st">"free_x"</span>, <span class="at">space =</span> <span class="st">"free_x"</span>)<span class="sc">+</span></span>
<span id="cb47-1334"><a href="#cb47-1334" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"fill"</span>)  <span class="sc">+</span></span>
<span id="cb47-1335"><a href="#cb47-1335" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> ssu18_colvec.tax) <span class="sc">+</span></span>
<span id="cb47-1336"><a href="#cb47-1336" aria-hidden="true" tabindex="-1"></a>    <span class="co">#scale_x_discrete("Treatment", expand = waiver(),</span></span>
<span id="cb47-1337"><a href="#cb47-1337" aria-hidden="true" tabindex="-1"></a>    <span class="co">#                 position = "bottom", drop = FALSE, limits = tmp_level) +</span></span>
<span id="cb47-1338"><a href="#cb47-1338" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb47-1339"><a href="#cb47-1339" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> top_level, <span class="at">reverse =</span> <span class="cn">FALSE</span>,</span>
<span id="cb47-1340"><a href="#cb47-1340" aria-hidden="true" tabindex="-1"></a>                               <span class="at">keywidth =</span> <span class="fl">0.7</span>, <span class="at">keyheight =</span> <span class="fl">0.7</span>)) <span class="sc">+</span></span>
<span id="cb47-1341"><a href="#cb47-1341" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb47-1342"><a href="#cb47-1342" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-1343"><a href="#cb47-1343" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"transparent"</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb47-1344"><a href="#cb47-1344" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"transparent"</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb47-1345"><a href="#cb47-1345" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb47-1346"><a href="#cb47-1346" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb47-1347"><a href="#cb47-1347" aria-hidden="true" tabindex="-1"></a>        <span class="co">#legend.text = element_text(size = 6),</span></span>
<span id="cb47-1348"><a href="#cb47-1348" aria-hidden="true" tabindex="-1"></a>        <span class="co">#legend.title = element_text(size = 8),</span></span>
<span id="cb47-1349"><a href="#cb47-1349" aria-hidden="true" tabindex="-1"></a>        <span class="co">#legend.key.size = unit(1.5, "cm"),</span></span>
<span id="cb47-1350"><a href="#cb47-1350" aria-hidden="true" tabindex="-1"></a>        <span class="co">#axis.text.y = element_text(size = 6),</span></span>
<span id="cb47-1351"><a href="#cb47-1351" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>)</span>
<span id="cb47-1352"><a href="#cb47-1352" aria-hidden="true" tabindex="-1"></a>        <span class="co">#strip.text = element_text(size = 8, angle = 0),</span></span>
<span id="cb47-1353"><a href="#cb47-1353" aria-hidden="true" tabindex="-1"></a>        <span class="co">#axis.title = element_text(size = 8)</span></span>
<span id="cb47-1354"><a href="#cb47-1354" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">+</span> </span>
<span id="cb47-1355"><a href="#cb47-1355" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="cn">NULL</span>)</span>
<span id="cb47-1356"><a href="#cb47-1356" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb47-1357"><a href="#cb47-1357" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_plot_name, tmp_plot)</span>
<span id="cb47-1358"><a href="#cb47-1358" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-1359"><a href="#cb47-1359" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-1360"><a href="#cb47-1360" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1361"><a href="#cb47-1361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1362"><a href="#cb47-1362" aria-hidden="true" tabindex="-1"></a>7) Finally we use the <span class="in">`patchwork`</span> package to combine the two plots and customize the look.</span>
<span id="cb47-1363"><a href="#cb47-1363" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1366"><a href="#cb47-1366" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1367"><a href="#cb47-1367" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb47-1368"><a href="#cb47-1368" aria-hidden="true" tabindex="-1"></a>  tmp_plot_main <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(set_to_plot, <span class="st">"_"</span>, ., <span class="st">"_plot"</span>)))</span>
<span id="cb47-1369"><a href="#cb47-1369" aria-hidden="true" tabindex="-1"></a>  tmp_plot_melt <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(set_to_plot, <span class="st">"_"</span>, ., <span class="st">"_plot_melt"</span>)))</span>
<span id="cb47-1370"><a href="#cb47-1370" aria-hidden="true" tabindex="-1"></a>  tmp_final <span class="ot">&lt;-</span> tmp_plot_main <span class="sc">+</span> tmp_plot_melt</span>
<span id="cb47-1371"><a href="#cb47-1371" aria-hidden="true" tabindex="-1"></a>  tmp_final <span class="ot">&lt;-</span> tmp_final <span class="sc">+</span> </span>
<span id="cb47-1372"><a href="#cb47-1372" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">'a'</span>, </span>
<span id="cb47-1373"><a href="#cb47-1373" aria-hidden="true" tabindex="-1"></a>                    <span class="at">title =</span> i) <span class="sc">+</span> </span>
<span id="cb47-1374"><a href="#cb47-1374" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#subtitle = 'Top taxa of non-filtered data',</span></span>
<span id="cb47-1375"><a href="#cb47-1375" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#caption = 'A) grouped by temperature.,</span></span>
<span id="cb47-1376"><a href="#cb47-1376" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#B) All samples, faceted by temperature.') + </span></span>
<span id="cb47-1377"><a href="#cb47-1377" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_layout</span>(<span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)) <span class="sc">&amp;</span></span>
<span id="cb47-1378"><a href="#cb47-1378" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>),</span>
<span id="cb47-1379"><a href="#cb47-1379" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">1</span>),</span>
<span id="cb47-1380"><a href="#cb47-1380" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.tag =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>),</span>
<span id="cb47-1381"><a href="#cb47-1381" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>),</span>
<span id="cb47-1382"><a href="#cb47-1382" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>),</span>
<span id="cb47-1383"><a href="#cb47-1383" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">0</span>),</span>
<span id="cb47-1384"><a href="#cb47-1384" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>),</span>
<span id="cb47-1385"><a href="#cb47-1385" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>),</span>
<span id="cb47-1386"><a href="#cb47-1386" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb47-1387"><a href="#cb47-1387" aria-hidden="true" tabindex="-1"></a>        <span class="co">#legend.position = "right")</span></span>
<span id="cb47-1388"><a href="#cb47-1388" aria-hidden="true" tabindex="-1"></a>        <span class="co">#legend.position = "right",</span></span>
<span id="cb47-1389"><a href="#cb47-1389" aria-hidden="true" tabindex="-1"></a>        <span class="co">#legend.title = element_text(size = rel(1)),</span></span>
<span id="cb47-1390"><a href="#cb47-1390" aria-hidden="true" tabindex="-1"></a>        <span class="co">#legend.text = element_text(size = rel(1)))</span></span>
<span id="cb47-1391"><a href="#cb47-1391" aria-hidden="true" tabindex="-1"></a>  tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(set_to_plot, <span class="st">"_"</span>, ., <span class="st">"_final_plot"</span>))</span>
<span id="cb47-1392"><a href="#cb47-1392" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name, tmp_final)</span>
<span id="cb47-1393"><a href="#cb47-1393" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb47-1394"><a href="#cb47-1394" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-1395"><a href="#cb47-1395" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1396"><a href="#cb47-1396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1399"><a href="#cb47-1399" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1400"><a href="#cb47-1400" aria-hidden="true" tabindex="-1"></a>taxa_to_plot <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Acidobacteriota"</span>, <span class="st">"Actinobacteriota"</span>, <span class="st">"Alphaproteobacteria"</span>, </span>
<span id="cb47-1401"><a href="#cb47-1401" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Bacteroidota"</span>, <span class="st">"Firmicutes"</span>, <span class="st">"Gammaproteobacteria"</span>, <span class="st">"Myxococcota"</span>, </span>
<span id="cb47-1402"><a href="#cb47-1402" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Verrucomicrobiota"</span>)</span>
<span id="cb47-1403"><a href="#cb47-1403" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> taxa_to_plot) {</span>
<span id="cb47-1404"><a href="#cb47-1404" aria-hidden="true" tabindex="-1"></a>  tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"ssu18_ps_work_"</span>, i, <span class="st">"_final_plot"</span>, <span class="at">sep =</span> <span class="st">""</span>)))</span>
<span id="cb47-1405"><a href="#cb47-1405" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_get, </span>
<span id="cb47-1406"><a href="#cb47-1406" aria-hidden="true" tabindex="-1"></a>                <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"files/taxa/figures/"</span>, i, <span class="st">"_tax_div_bar_plots.png"</span>, <span class="at">sep =</span> <span class="st">""</span>), </span>
<span id="cb47-1407"><a href="#cb47-1407" aria-hidden="true" tabindex="-1"></a>                <span class="at">height =</span> <span class="dv">14</span>, <span class="at">width =</span> <span class="dv">21</span>, <span class="at">units =</span> <span class="st">'cm'</span>, <span class="at">bg =</span> <span class="st">"white"</span>)</span>
<span id="cb47-1408"><a href="#cb47-1408" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_get, </span>
<span id="cb47-1409"><a href="#cb47-1409" aria-hidden="true" tabindex="-1"></a>                <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"files/taxa/figures/"</span>, i, <span class="st">"_tax_div_bar_plots.pdf"</span>, <span class="at">sep =</span> <span class="st">""</span>), </span>
<span id="cb47-1410"><a href="#cb47-1410" aria-hidden="true" tabindex="-1"></a>                <span class="at">height =</span> <span class="dv">14</span>, <span class="at">width =</span> <span class="dv">21</span>, <span class="at">units =</span> <span class="st">'cm'</span>, <span class="at">bg =</span> <span class="st">"white"</span>)</span>
<span id="cb47-1411"><a href="#cb47-1411" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-1412"><a href="#cb47-1412" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1413"><a href="#cb47-1413" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb47-1414"><a href="#cb47-1414" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1415"><a href="#cb47-1415" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br/&gt;</span></span>
<span id="cb47-1416"><a href="#cb47-1416" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1417"><a href="#cb47-1417" aria-hidden="true" tabindex="-1"></a>::: {.column-body-outset}</span>
<span id="cb47-1418"><a href="#cb47-1418" aria-hidden="true" tabindex="-1"></a>::: {.panel-tabset}</span>
<span id="cb47-1419"><a href="#cb47-1419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1420"><a href="#cb47-1420" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Alphaproteobacteria</span></span>
<span id="cb47-1421"><a href="#cb47-1421" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1424"><a href="#cb47-1424" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1425"><a href="#cb47-1425" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-1426"><a href="#cb47-1426" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-1427"><a href="#cb47-1427" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-1428"><a href="#cb47-1428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1429"><a href="#cb47-1429" aria-hidden="true" tabindex="-1"></a><span class="co">#top_level &lt;- "Family"</span></span>
<span id="cb47-1430"><a href="#cb47-1430" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu18_ps_work_Alphaproteobacteria_final_plot</span></span>
<span id="cb47-1431"><a href="#cb47-1431" aria-hidden="true" tabindex="-1"></a><span class="co">#unique(ssu18_ps_work_Alphaproteobacteria_final_plot$data$Family)</span></span>
<span id="cb47-1432"><a href="#cb47-1432" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/taxa/Alphaproteobacteria_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-1433"><a href="#cb47-1433" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1434"><a href="#cb47-1434" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("alpha_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb47-1435"><a href="#cb47-1435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1436"><a href="#cb47-1436" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Gammaproteobacteria</span></span>
<span id="cb47-1437"><a href="#cb47-1437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1440"><a href="#cb47-1440" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1441"><a href="#cb47-1441" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-1442"><a href="#cb47-1442" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-1443"><a href="#cb47-1443" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-1444"><a href="#cb47-1444" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu18_ps_work_Gammaproteobacteria_final_plot </span></span>
<span id="cb47-1445"><a href="#cb47-1445" aria-hidden="true" tabindex="-1"></a><span class="co">#unique(ssu18_ps_work_Gammaproteobacteria_final_plot$data$Family)</span></span>
<span id="cb47-1446"><a href="#cb47-1446" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/taxa/Gammaproteobacteria_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-1447"><a href="#cb47-1447" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1448"><a href="#cb47-1448" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("gamma_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb47-1449"><a href="#cb47-1449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1450"><a href="#cb47-1450" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Acidobacteriota</span></span>
<span id="cb47-1451"><a href="#cb47-1451" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1454"><a href="#cb47-1454" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1455"><a href="#cb47-1455" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-1456"><a href="#cb47-1456" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-1457"><a href="#cb47-1457" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-1458"><a href="#cb47-1458" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu18_ps_work_Acidobacteriota_final_plot </span></span>
<span id="cb47-1459"><a href="#cb47-1459" aria-hidden="true" tabindex="-1"></a><span class="co">#unique(ssu18_ps_work_Acidobacteriota_final_plot$data$Family)</span></span>
<span id="cb47-1460"><a href="#cb47-1460" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/taxa/Acidobacteriota_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-1461"><a href="#cb47-1461" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1462"><a href="#cb47-1462" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("acido_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb47-1463"><a href="#cb47-1463" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1464"><a href="#cb47-1464" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Actinobacteriota</span></span>
<span id="cb47-1465"><a href="#cb47-1465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1468"><a href="#cb47-1468" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1469"><a href="#cb47-1469" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-1470"><a href="#cb47-1470" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-1471"><a href="#cb47-1471" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-1472"><a href="#cb47-1472" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu18_ps_work_Actinobacteriota_final_plot </span></span>
<span id="cb47-1473"><a href="#cb47-1473" aria-hidden="true" tabindex="-1"></a><span class="co">#unique(ssu18_ps_work_Actinobacteriota_final_plot$data$Family)</span></span>
<span id="cb47-1474"><a href="#cb47-1474" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/taxa/Actinobacteriota_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-1475"><a href="#cb47-1475" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1476"><a href="#cb47-1476" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("actino_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb47-1477"><a href="#cb47-1477" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1478"><a href="#cb47-1478" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Bacteroidota</span></span>
<span id="cb47-1479"><a href="#cb47-1479" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1482"><a href="#cb47-1482" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1483"><a href="#cb47-1483" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-1484"><a href="#cb47-1484" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-1485"><a href="#cb47-1485" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-1486"><a href="#cb47-1486" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu18_ps_work_Bacteroidota_final_plot </span></span>
<span id="cb47-1487"><a href="#cb47-1487" aria-hidden="true" tabindex="-1"></a><span class="co">#unique(ssu18_ps_work_Bacteroidota_final_plot$data$Family)</span></span>
<span id="cb47-1488"><a href="#cb47-1488" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/taxa/Bacteroidota_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-1489"><a href="#cb47-1489" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1490"><a href="#cb47-1490" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("bacteroid_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb47-1491"><a href="#cb47-1491" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1492"><a href="#cb47-1492" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Firmicutes</span></span>
<span id="cb47-1493"><a href="#cb47-1493" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1496"><a href="#cb47-1496" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1497"><a href="#cb47-1497" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-1498"><a href="#cb47-1498" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-1499"><a href="#cb47-1499" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-1500"><a href="#cb47-1500" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu18_ps_work_Firmicutes_final_plot </span></span>
<span id="cb47-1501"><a href="#cb47-1501" aria-hidden="true" tabindex="-1"></a><span class="co">#unique(ssu18_ps_work_Firmicutes_final_plot$data$Family)</span></span>
<span id="cb47-1502"><a href="#cb47-1502" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/taxa/Firmicutes_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-1503"><a href="#cb47-1503" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1504"><a href="#cb47-1504" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("firm_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb47-1505"><a href="#cb47-1505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1506"><a href="#cb47-1506" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Myxococcota</span></span>
<span id="cb47-1507"><a href="#cb47-1507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1510"><a href="#cb47-1510" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1511"><a href="#cb47-1511" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-1512"><a href="#cb47-1512" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-1513"><a href="#cb47-1513" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-1514"><a href="#cb47-1514" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu18_ps_work_Myxococcota_final_plot </span></span>
<span id="cb47-1515"><a href="#cb47-1515" aria-hidden="true" tabindex="-1"></a><span class="co">#unique(ssu18_ps_work_Myxococcota_final_plot$data$Family)</span></span>
<span id="cb47-1516"><a href="#cb47-1516" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/taxa/Myxococcota_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-1517"><a href="#cb47-1517" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1518"><a href="#cb47-1518" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("myxo_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb47-1519"><a href="#cb47-1519" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1520"><a href="#cb47-1520" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Verrucomicrobiota</span></span>
<span id="cb47-1521"><a href="#cb47-1521" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1524"><a href="#cb47-1524" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1525"><a href="#cb47-1525" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-1526"><a href="#cb47-1526" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-1527"><a href="#cb47-1527" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-1528"><a href="#cb47-1528" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu18_ps_work_Verrucomicrobiota_final_plot </span></span>
<span id="cb47-1529"><a href="#cb47-1529" aria-hidden="true" tabindex="-1"></a><span class="co">#unique(ssu18_ps_work_Verrucomicrobiota_final_plot$data$Family)</span></span>
<span id="cb47-1530"><a href="#cb47-1530" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/taxa/Verrucomicrobiota_tax_div_bar_plots.png"</span>)</span>
<span id="cb47-1531"><a href="#cb47-1531" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1532"><a href="#cb47-1532" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;small&gt;</span><span class="in">`r caption_fig("verruco_ssu")`</span><span class="kw">&lt;/small&gt;</span></span>
<span id="cb47-1533"><a href="#cb47-1533" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb47-1534"><a href="#cb47-1534" aria-hidden="true" tabindex="-1"></a>::: </span>
<span id="cb47-1535"><a href="#cb47-1535" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1536"><a href="#cb47-1536" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1539"><a href="#cb47-1539" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1540"><a href="#cb47-1540" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-1541"><a href="#cb47-1541" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="st">"page_build/taxa_wf.rdata"</span>)</span>
<span id="cb47-1542"><a href="#cb47-1542" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1543"><a href="#cb47-1543" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1546"><a href="#cb47-1546" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1547"><a href="#cb47-1547" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb47-1548"><a href="#cb47-1548" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-1549"><a href="#cb47-1549" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb47-1550"><a href="#cb47-1550" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1551"><a href="#cb47-1551" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1552"><a href="#cb47-1552" aria-hidden="true" tabindex="-1"></a><span class="fu"># Workflow Output </span></span>
<span id="cb47-1553"><a href="#cb47-1553" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1554"><a href="#cb47-1554" aria-hidden="true" tabindex="-1"></a>Data products generated in this workflow can be downloaded from figshare.</span>
<span id="cb47-1555"><a href="#cb47-1555" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1556"><a href="#cb47-1556" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;iframe</span> <span class="er">src</span><span class="ot">=</span><span class="st">"https://widgets.figshare.com/articles/16826737/embed?show_title=1"</span> <span class="er">width</span><span class="ot">=</span><span class="st">"100%"</span> <span class="er">height</span><span class="ot">=</span><span class="st">"221"</span> <span class="er">allowfullscreen</span> <span class="er">frameborder</span><span class="ot">=</span><span class="st">"0"</span><span class="kw">&gt;&lt;/iframe&gt;</span></span>
<span id="cb47-1557"><a href="#cb47-1557" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1558"><a href="#cb47-1558" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb47-1559"><a href="#cb47-1559" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1560"><a href="#cb47-1560" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"alpha.html"</span> <span class="er">class</span><span class="ot">=</span><span class="st">"btnnav button_round"</span> <span class="er">style</span><span class="ot">=</span><span class="st">"float: right;"</span><span class="kw">&gt;</span>Next workflow:</span>
<span id="cb47-1561"><a href="#cb47-1561" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br&gt;</span> 5. Alpha Diversity Estimates<span class="kw">&lt;/a&gt;</span></span>
<span id="cb47-1562"><a href="#cb47-1562" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"filtering.html"</span> <span class="er">class</span><span class="ot">=</span><span class="st">"btnnav button_round"</span> <span class="er">style</span><span class="ot">=</span><span class="st">"float: left;"</span><span class="kw">&gt;</span>Previous workflow:</span>
<span id="cb47-1563"><a href="#cb47-1563" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br&gt;</span> 3. Filtering<span class="kw">&lt;/a&gt;&lt;br&gt;&lt;br&gt;</span></span>
<span id="cb47-1564"><a href="#cb47-1564" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1567"><a href="#cb47-1567" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1568"><a href="#cb47-1568" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false </span></span>
<span id="cb47-1569"><a href="#cb47-1569" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb47-1570"><a href="#cb47-1570" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-1571"><a href="#cb47-1571" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-1572"><a href="#cb47-1572" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb47-1573"><a href="#cb47-1573" aria-hidden="true" tabindex="-1"></a><span class="do">### COmmon formatting scripts</span></span>
<span id="cb47-1574"><a href="#cb47-1574" aria-hidden="true" tabindex="-1"></a><span class="do">### </span><span class="al">NOTE</span><span class="do">: captioner.R must be read BEFORE captions_XYZ.R</span></span>
<span id="cb47-1575"><a href="#cb47-1575" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">"assets"</span>, <span class="st">"functions.R"</span>))</span>
<span id="cb47-1576"><a href="#cb47-1576" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1577"><a href="#cb47-1577" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1578"><a href="#cb47-1578" aria-hidden="true" tabindex="-1"></a><span class="fu">####  Source Code {.appendix}</span></span>
<span id="cb47-1579"><a href="#cb47-1579" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1580"><a href="#cb47-1580" aria-hidden="true" tabindex="-1"></a>The source code for this page can be accessed on GitHub <span class="in">`r fa(name = "github")`</span> by <span class="co">[</span><span class="ot">clicking this link</span><span class="co">]</span>(<span class="in">`r source_code()`</span>). </span>
<span id="cb47-1581"><a href="#cb47-1581" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1582"><a href="#cb47-1582" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Data Availability {.appendix}</span></span>
<span id="cb47-1583"><a href="#cb47-1583" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1584"><a href="#cb47-1584" aria-hidden="true" tabindex="-1"></a>Data generated in this workflow and the Rdata need to run the workflow can be accessed on figshare at <span class="co">[</span><span class="ot">10.25573/data.16826737</span><span class="co">](https://doi.org/10.25573/data.16826737)</span>.</span>
<span id="cb47-1585"><a href="#cb47-1585" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1586"><a href="#cb47-1586" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Last updated on {.appendix}</span></span>
<span id="cb47-1587"><a href="#cb47-1587" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1590"><a href="#cb47-1590" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1591"><a href="#cb47-1591" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb47-1592"><a href="#cb47-1592" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb47-1593"><a href="#cb47-1593" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb47-1594"><a href="#cb47-1594" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
      <div class="nav-footer-center">© Copyright 2022 <a href="https://metacrobe.github.io/">metacrobe</a> &amp; The SWELTR Project. Site constructed using <a href="https://quarto.org/">Quarto</a>.</div>
  </div>
</footer>


</body></html>